// @elrondnetwork/dapp-core/__commonjs/wrappers/DappProvider/CustomComponents.js
"use strict";var gi=Object.create;var Ve=Object.defineProperty,fi=Object.defineProperties,hi=Object.getOwnPropertyDescriptor,Ti=Object.getOwnPropertyDescriptors,xi=Object.getOwnPropertyNames,oo=Object.getOwnPropertySymbols,yi=Object.getPrototypeOf,Go=Object.prototype.hasOwnProperty,Zn=Object.prototype.propertyIsEnumerable;var Yn=(e,o,n)=>o in e?Ve(e,o,{enumerable:!0,configurable:!0,writable:!0,value:n}):e[o]=n,f=(e,o)=>{for(var n in o||(o={}))Go.call(o,n)&&Yn(e,n,o[n]);if(oo)for(var n of oo(o))Zn.call(o,n)&&Yn(e,n,o[n]);return e},v=(e,o)=>fi(e,Ti(o));var $o=(e,o)=>{var n={};for(var r in e)Go.call(e,r)&&o.indexOf(r)<0&&(n[r]=e[r]);if(e!=null&&oo)for(var r of oo(e))o.indexOf(r)<0&&Zn.call(e,r)&&(n[r]=e[r]);return n};var T=(e,o)=>()=>(e&&(o=e(e=0)),o);var no=(e,o)=>()=>(o||e((o={exports:{}}).exports,o),o.exports),le=(e,o)=>{for(var n in o)Ve(e,n,{get:o[n],enumerable:!0})},Jn=(e,o,n,r)=>{if(o&&typeof o=="object"||typeof o=="function")for(let a of xi(o))!Go.call(e,a)&&a!==n&&Ve(e,a,{get:()=>o[a],enumerable:!(r=hi(o,a))||r.enumerable});return e};var S=(e,o,n)=>(n=e!=null?gi(yi(e)):{},Jn(o||!e||!e.__esModule?Ve(n,"default",{value:e,enumerable:!0}):n,e)),He=e=>Jn(Ve({},"__esModule",{value:!0}),e);var x=(e,o,n)=>new Promise((r,a)=>{var m=d=>{try{l(n.next(d))}catch(u){a(u)}},s=d=>{try{l(n.throw(d))}catch(u){a(u)}},l=d=>d.done?r(d.value):Promise.resolve(d.value).then(m,s);l((n=n.apply(e,o)).next())});var ot=no(to=>{"use strict";t();to.byteLength=Si;to.toByteArray=vi;to.fromByteArray=Ai;var z=[],U=[],bi=typeof Uint8Array!="undefined"?Uint8Array:Array,Vo="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";for(ue=0,Qn=Vo.length;ue<Qn;++ue)z[ue]=Vo[ue],U[Vo.charCodeAt(ue)]=ue;var ue,Qn;U["-".charCodeAt(0)]=62;U["_".charCodeAt(0)]=63;function et(e){var o=e.length;if(o%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var n=e.indexOf("=");n===-1&&(n=o);var r=n===o?0:4-n%4;return[n,r]}function Si(e){var o=et(e),n=o[0],r=o[1];return(n+r)*3/4-r}function wi(e,o,n){return(o+n)*3/4-n}function vi(e){var o,n=et(e),r=n[0],a=n[1],m=new bi(wi(e,r,a)),s=0,l=a>0?r-4:r,d;for(d=0;d<l;d+=4)o=U[e.charCodeAt(d)]<<18|U[e.charCodeAt(d+1)]<<12|U[e.charCodeAt(d+2)]<<6|U[e.charCodeAt(d+3)],m[s++]=o>>16&255,m[s++]=o>>8&255,m[s++]=o&255;return a===2&&(o=U[e.charCodeAt(d)]<<2|U[e.charCodeAt(d+1)]>>4,m[s++]=o&255),a===1&&(o=U[e.charCodeAt(d)]<<10|U[e.charCodeAt(d+1)]<<4|U[e.charCodeAt(d+2)]>>2,m[s++]=o>>8&255,m[s++]=o&255),m}function ki(e){return z[e>>18&63]+z[e>>12&63]+z[e>>6&63]+z[e&63]}function Ii(e,o,n){for(var r,a=[],m=o;m<n;m+=3)r=(e[m]<<16&16711680)+(e[m+1]<<8&65280)+(e[m+2]&255),a.push(ki(r));return a.join("")}function Ai(e){for(var o,n=e.length,r=n%3,a=[],m=16383,s=0,l=n-r;s<l;s+=m)a.push(Ii(e,s,s+m>l?l:s+m));return r===1?(o=e[n-1],a.push(z[o>>2]+z[o<<4&63]+"==")):r===2&&(o=(e[n-2]<<8)+e[n-1],a.push(z[o>>10]+z[o>>4&63]+z[o<<2&63]+"=")),a.join("")}});var nt=no(Ho=>{t();Ho.read=function(e,o,n,r,a){var m,s,l=a*8-r-1,d=(1<<l)-1,u=d>>1,y=-7,h=n?a-1:0,L=n?-1:1,C=e[o+h];for(h+=L,m=C&(1<<-y)-1,C>>=-y,y+=l;y>0;m=m*256+e[o+h],h+=L,y-=8);for(s=m&(1<<-y)-1,m>>=-y,y+=r;y>0;s=s*256+e[o+h],h+=L,y-=8);if(m===0)m=1-u;else{if(m===d)return s?NaN:(C?-1:1)*(1/0);s=s+Math.pow(2,r),m=m-u}return(C?-1:1)*s*Math.pow(2,m-r)};Ho.write=function(e,o,n,r,a,m){var s,l,d,u=m*8-a-1,y=(1<<u)-1,h=y>>1,L=a===23?Math.pow(2,-24)-Math.pow(2,-77):0,C=r?0:m-1,O=r?1:-1,j=o<0||o===0&&1/o<0?1:0;for(o=Math.abs(o),isNaN(o)||o===1/0?(l=isNaN(o)?1:0,s=y):(s=Math.floor(Math.log(o)/Math.LN2),o*(d=Math.pow(2,-s))<1&&(s--,d*=2),s+h>=1?o+=L/d:o+=L*Math.pow(2,1-h),o*d>=2&&(s++,d/=2),s+h>=y?(l=0,s=y):s+h>=1?(l=(o*d-1)*Math.pow(2,a),s=s+h):(l=o*Math.pow(2,h-1)*Math.pow(2,a),s=0));a>=8;e[n+C]=l&255,C+=O,l/=256,a-=8);for(s=s<<a|l,u+=a;u>0;e[n+C]=s&255,C+=O,s/=256,u-=8);e[n+C-O]|=j*128}});var gt=no(Ae=>{"use strict";t();var jo=ot(),Ie=nt(),tt=typeof Symbol=="function"&&typeof Symbol.for=="function"?Symbol.for("nodejs.util.inspect.custom"):null;Ae.Buffer=_;Ae.SlowBuffer=Ni;Ae.INSPECT_MAX_BYTES=50;var ro=2147483647;Ae.kMaxLength=ro;_.TYPED_ARRAY_SUPPORT=Ei();!_.TYPED_ARRAY_SUPPORT&&typeof console!="undefined"&&typeof console.error=="function"&&console.error("This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.");function Ei(){try{var e=new Uint8Array(1),o={foo:function(){return 42}};return Object.setPrototypeOf(o,Uint8Array.prototype),Object.setPrototypeOf(e,o),e.foo()===42}catch(n){return!1}}Object.defineProperty(_.prototype,"parent",{enumerable:!0,get:function(){if(!!_.isBuffer(this))return this.buffer}});Object.defineProperty(_.prototype,"offset",{enumerable:!0,get:function(){if(!!_.isBuffer(this))return this.byteOffset}});function Y(e){if(e>ro)throw new RangeError('The value "'+e+'" is invalid for option "size"');var o=new Uint8Array(e);return Object.setPrototypeOf(o,_.prototype),o}function _(e,o,n){if(typeof e=="number"){if(typeof o=="string")throw new TypeError('The "string" argument must be of type string. Received type number');return Xo(e)}return it(e,o,n)}_.poolSize=8192;function it(e,o,n){if(typeof e=="string")return Di(e,o);if(ArrayBuffer.isView(e))return Ci(e);if(e==null)throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e);if(q(e,ArrayBuffer)||e&&q(e.buffer,ArrayBuffer)||typeof SharedArrayBuffer!="undefined"&&(q(e,SharedArrayBuffer)||e&&q(e.buffer,SharedArrayBuffer)))return qo(e,o,n);if(typeof e=="number")throw new TypeError('The "value" argument must not be of type number. Received type number');var r=e.valueOf&&e.valueOf();if(r!=null&&r!==e)return _.from(r,o,n);var a=Pi(e);if(a)return a;if(typeof Symbol!="undefined"&&Symbol.toPrimitive!=null&&typeof e[Symbol.toPrimitive]=="function")return _.from(e[Symbol.toPrimitive]("string"),o,n);throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e)}_.from=function(e,o,n){return it(e,o,n)};Object.setPrototypeOf(_.prototype,Uint8Array.prototype);Object.setPrototypeOf(_,Uint8Array);function ct(e){if(typeof e!="number")throw new TypeError('"size" argument must be of type number');if(e<0)throw new RangeError('The value "'+e+'" is invalid for option "size"')}function Li(e,o,n){return ct(e),e<=0?Y(e):o!==void 0?typeof n=="string"?Y(e).fill(o,n):Y(e).fill(o):Y(e)}_.alloc=function(e,o,n){return Li(e,o,n)};function Xo(e){return ct(e),Y(e<0?0:Yo(e)|0)}_.allocUnsafe=function(e){return Xo(e)};_.allocUnsafeSlow=function(e){return Xo(e)};function Di(e,o){if((typeof o!="string"||o==="")&&(o="utf8"),!_.isEncoding(o))throw new TypeError("Unknown encoding: "+o);var n=pt(e,o)|0,r=Y(n),a=r.write(e,o);return a!==n&&(r=r.slice(0,a)),r}function zo(e){for(var o=e.length<0?0:Yo(e.length)|0,n=Y(o),r=0;r<o;r+=1)n[r]=e[r]&255;return n}function Ci(e){if(q(e,Uint8Array)){var o=new Uint8Array(e);return qo(o.buffer,o.byteOffset,o.byteLength)}return zo(e)}function qo(e,o,n){if(o<0||e.byteLength<o)throw new RangeError('"offset" is outside of buffer bounds');if(e.byteLength<o+(n||0))throw new RangeError('"length" is outside of buffer bounds');var r;return o===void 0&&n===void 0?r=new Uint8Array(e):n===void 0?r=new Uint8Array(e,o):r=new Uint8Array(e,o,n),Object.setPrototypeOf(r,_.prototype),r}function Pi(e){if(_.isBuffer(e)){var o=Yo(e.length)|0,n=Y(o);return n.length===0||e.copy(n,0,0,o),n}if(e.length!==void 0)return typeof e.length!="number"||Zo(e.length)?Y(0):zo(e);if(e.type==="Buffer"&&Array.isArray(e.data))return zo(e.data)}function Yo(e){if(e>=ro)throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x"+ro.toString(16)+" bytes");return e|0}function Ni(e){return+e!=e&&(e=0),_.alloc(+e)}_.isBuffer=function(o){return o!=null&&o._isBuffer===!0&&o!==_.prototype};_.compare=function(o,n){if(q(o,Uint8Array)&&(o=_.from(o,o.offset,o.byteLength)),q(n,Uint8Array)&&(n=_.from(n,n.offset,n.byteLength)),!_.isBuffer(o)||!_.isBuffer(n))throw new TypeError('The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array');if(o===n)return 0;for(var r=o.length,a=n.length,m=0,s=Math.min(r,a);m<s;++m)if(o[m]!==n[m]){r=o[m],a=n[m];break}return r<a?-1:a<r?1:0};_.isEncoding=function(o){switch(String(o).toLowerCase()){case"hex":case"utf8":case"utf-8":case"ascii":case"latin1":case"binary":case"base64":case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return!0;default:return!1}};_.concat=function(o,n){if(!Array.isArray(o))throw new TypeError('"list" argument must be an Array of Buffers');if(o.length===0)return _.alloc(0);var r;if(n===void 0)for(n=0,r=0;r<o.length;++r)n+=o[r].length;var a=_.allocUnsafe(n),m=0;for(r=0;r<o.length;++r){var s=o[r];if(q(s,Uint8Array))m+s.length>a.length?_.from(s).copy(a,m):Uint8Array.prototype.set.call(a,s,m);else if(_.isBuffer(s))s.copy(a,m);else throw new TypeError('"list" argument must be an Array of Buffers');m+=s.length}return a};function pt(e,o){if(_.isBuffer(e))return e.length;if(ArrayBuffer.isView(e)||q(e,ArrayBuffer))return e.byteLength;if(typeof e!="string")throw new TypeError('The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type '+typeof e);var n=e.length,r=arguments.length>2&&arguments[2]===!0;if(!r&&n===0)return 0;for(var a=!1;;)switch(o){case"ascii":case"latin1":case"binary":return n;case"utf8":case"utf-8":return Ko(e).length;case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return n*2;case"hex":return n>>>1;case"base64":return ut(e).length;default:if(a)return r?-1:Ko(e).length;o=(""+o).toLowerCase(),a=!0}}_.byteLength=pt;function Ri(e,o,n){var r=!1;if((o===void 0||o<0)&&(o=0),o>this.length||((n===void 0||n>this.length)&&(n=this.length),n<=0)||(n>>>=0,o>>>=0,n<=o))return"";for(e||(e="utf8");;)switch(e){case"hex":return Hi(this,o,n);case"utf8":case"utf-8":return mt(this,o,n);case"ascii":return $i(this,o,n);case"latin1":case"binary":return Vi(this,o,n);case"base64":return Wi(this,o,n);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return ji(this,o,n);default:if(r)throw new TypeError("Unknown encoding: "+e);e=(e+"").toLowerCase(),r=!0}}_.prototype._isBuffer=!0;function ge(e,o,n){var r=e[o];e[o]=e[n],e[n]=r}_.prototype.swap16=function(){var o=this.length;if(o%2!==0)throw new RangeError("Buffer size must be a multiple of 16-bits");for(var n=0;n<o;n+=2)ge(this,n,n+1);return this};_.prototype.swap32=function(){var o=this.length;if(o%4!==0)throw new RangeError("Buffer size must be a multiple of 32-bits");for(var n=0;n<o;n+=4)ge(this,n,n+3),ge(this,n+1,n+2);return this};_.prototype.swap64=function(){var o=this.length;if(o%8!==0)throw new RangeError("Buffer size must be a multiple of 64-bits");for(var n=0;n<o;n+=8)ge(this,n,n+7),ge(this,n+1,n+6),ge(this,n+2,n+5),ge(this,n+3,n+4);return this};_.prototype.toString=function(){var o=this.length;return o===0?"":arguments.length===0?mt(this,0,o):Ri.apply(this,arguments)};_.prototype.toLocaleString=_.prototype.toString;_.prototype.equals=function(o){if(!_.isBuffer(o))throw new TypeError("Argument must be a Buffer");return this===o?!0:_.compare(this,o)===0};_.prototype.inspect=function(){var o="",n=Ae.INSPECT_MAX_BYTES;return o=this.toString("hex",0,n).replace(/(.{2})/g,"$1 ").trim(),this.length>n&&(o+=" ... "),"<Buffer "+o+">"};tt&&(_.prototype[tt]=_.prototype.inspect);_.prototype.compare=function(o,n,r,a,m){if(q(o,Uint8Array)&&(o=_.from(o,o.offset,o.byteLength)),!_.isBuffer(o))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+typeof o);if(n===void 0&&(n=0),r===void 0&&(r=o?o.length:0),a===void 0&&(a=0),m===void 0&&(m=this.length),n<0||r>o.length||a<0||m>this.length)throw new RangeError("out of range index");if(a>=m&&n>=r)return 0;if(a>=m)return-1;if(n>=r)return 1;if(n>>>=0,r>>>=0,a>>>=0,m>>>=0,this===o)return 0;for(var s=m-a,l=r-n,d=Math.min(s,l),u=this.slice(a,m),y=o.slice(n,r),h=0;h<d;++h)if(u[h]!==y[h]){s=u[h],l=y[h];break}return s<l?-1:l<s?1:0};function st(e,o,n,r,a){if(e.length===0)return-1;if(typeof n=="string"?(r=n,n=0):n>2147483647?n=2147483647:n<-2147483648&&(n=-2147483648),n=+n,Zo(n)&&(n=a?0:e.length-1),n<0&&(n=e.length+n),n>=e.length){if(a)return-1;n=e.length-1}else if(n<0)if(a)n=0;else return-1;if(typeof o=="string"&&(o=_.from(o,r)),_.isBuffer(o))return o.length===0?-1:rt(e,o,n,r,a);if(typeof o=="number")return o=o&255,typeof Uint8Array.prototype.indexOf=="function"?a?Uint8Array.prototype.indexOf.call(e,o,n):Uint8Array.prototype.lastIndexOf.call(e,o,n):rt(e,[o],n,r,a);throw new TypeError("val must be string, number or Buffer")}function rt(e,o,n,r,a){var m=1,s=e.length,l=o.length;if(r!==void 0&&(r=String(r).toLowerCase(),r==="ucs2"||r==="ucs-2"||r==="utf16le"||r==="utf-16le")){if(e.length<2||o.length<2)return-1;m=2,s/=2,l/=2,n/=2}function d(C,O){return m===1?C[O]:C.readUInt16BE(O*m)}var u;if(a){var y=-1;for(u=n;u<s;u++)if(d(e,u)===d(o,y===-1?0:u-y)){if(y===-1&&(y=u),u-y+1===l)return y*m}else y!==-1&&(u-=u-y),y=-1}else for(n+l>s&&(n=s-l),u=n;u>=0;u--){for(var h=!0,L=0;L<l;L++)if(d(e,u+L)!==d(o,L)){h=!1;break}if(h)return u}return-1}_.prototype.includes=function(o,n,r){return this.indexOf(o,n,r)!==-1};_.prototype.indexOf=function(o,n,r){return st(this,o,n,r,!0)};_.prototype.lastIndexOf=function(o,n,r){return st(this,o,n,r,!1)};function Mi(e,o,n,r){n=Number(n)||0;var a=e.length-n;r?(r=Number(r),r>a&&(r=a)):r=a;var m=o.length;r>m/2&&(r=m/2);for(var s=0;s<r;++s){var l=parseInt(o.substr(s*2,2),16);if(Zo(l))return s;e[n+s]=l}return s}function Fi(e,o,n,r){return ao(Ko(o,e.length-n),e,n,r)}function Oi(e,o,n,r){return ao(Ki(o),e,n,r)}function Bi(e,o,n,r){return ao(ut(o),e,n,r)}function Ui(e,o,n,r){return ao(Xi(o,e.length-n),e,n,r)}_.prototype.write=function(o,n,r,a){if(n===void 0)a="utf8",r=this.length,n=0;else if(r===void 0&&typeof n=="string")a=n,r=this.length,n=0;else if(isFinite(n))n=n>>>0,isFinite(r)?(r=r>>>0,a===void 0&&(a="utf8")):(a=r,r=void 0);else throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");var m=this.length-n;if((r===void 0||r>m)&&(r=m),o.length>0&&(r<0||n<0)||n>this.length)throw new RangeError("Attempt to write outside buffer bounds");a||(a="utf8");for(var s=!1;;)switch(a){case"hex":return Mi(this,o,n,r);case"utf8":case"utf-8":return Fi(this,o,n,r);case"ascii":case"latin1":case"binary":return Oi(this,o,n,r);case"base64":return Bi(this,o,n,r);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return Ui(this,o,n,r);default:if(s)throw new TypeError("Unknown encoding: "+a);a=(""+a).toLowerCase(),s=!0}};_.prototype.toJSON=function(){return{type:"Buffer",data:Array.prototype.slice.call(this._arr||this,0)}};function Wi(e,o,n){return o===0&&n===e.length?jo.fromByteArray(e):jo.fromByteArray(e.slice(o,n))}function mt(e,o,n){n=Math.min(e.length,n);for(var r=[],a=o;a<n;){var m=e[a],s=null,l=m>239?4:m>223?3:m>191?2:1;if(a+l<=n){var d,u,y,h;switch(l){case 1:m<128&&(s=m);break;case 2:d=e[a+1],(d&192)===128&&(h=(m&31)<<6|d&63,h>127&&(s=h));break;case 3:d=e[a+1],u=e[a+2],(d&192)===128&&(u&192)===128&&(h=(m&15)<<12|(d&63)<<6|u&63,h>2047&&(h<55296||h>57343)&&(s=h));break;case 4:d=e[a+1],u=e[a+2],y=e[a+3],(d&192)===128&&(u&192)===128&&(y&192)===128&&(h=(m&15)<<18|(d&63)<<12|(u&63)<<6|y&63,h>65535&&h<1114112&&(s=h))}}s===null?(s=65533,l=1):s>65535&&(s-=65536,r.push(s>>>10&1023|55296),s=56320|s&1023),r.push(s),a+=l}return Gi(r)}var at=4096;function Gi(e){var o=e.length;if(o<=at)return String.fromCharCode.apply(String,e);for(var n="",r=0;r<o;)n+=String.fromCharCode.apply(String,e.slice(r,r+=at));return n}function $i(e,o,n){var r="";n=Math.min(e.length,n);for(var a=o;a<n;++a)r+=String.fromCharCode(e[a]&127);return r}function Vi(e,o,n){var r="";n=Math.min(e.length,n);for(var a=o;a<n;++a)r+=String.fromCharCode(e[a]);return r}function Hi(e,o,n){var r=e.length;(!o||o<0)&&(o=0),(!n||n<0||n>r)&&(n=r);for(var a="",m=o;m<n;++m)a+=Yi[e[m]];return a}function ji(e,o,n){for(var r=e.slice(o,n),a="",m=0;m<r.length-1;m+=2)a+=String.fromCharCode(r[m]+r[m+1]*256);return a}_.prototype.slice=function(o,n){var r=this.length;o=~~o,n=n===void 0?r:~~n,o<0?(o+=r,o<0&&(o=0)):o>r&&(o=r),n<0?(n+=r,n<0&&(n=0)):n>r&&(n=r),n<o&&(n=o);var a=this.subarray(o,n);return Object.setPrototypeOf(a,_.prototype),a};function D(e,o,n){if(e%1!==0||e<0)throw new RangeError("offset is not uint");if(e+o>n)throw new RangeError("Trying to access beyond buffer length")}_.prototype.readUintLE=_.prototype.readUIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||D(o,n,this.length);for(var a=this[o],m=1,s=0;++s<n&&(m*=256);)a+=this[o+s]*m;return a};_.prototype.readUintBE=_.prototype.readUIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||D(o,n,this.length);for(var a=this[o+--n],m=1;n>0&&(m*=256);)a+=this[o+--n]*m;return a};_.prototype.readUint8=_.prototype.readUInt8=function(o,n){return o=o>>>0,n||D(o,1,this.length),this[o]};_.prototype.readUint16LE=_.prototype.readUInt16LE=function(o,n){return o=o>>>0,n||D(o,2,this.length),this[o]|this[o+1]<<8};_.prototype.readUint16BE=_.prototype.readUInt16BE=function(o,n){return o=o>>>0,n||D(o,2,this.length),this[o]<<8|this[o+1]};_.prototype.readUint32LE=_.prototype.readUInt32LE=function(o,n){return o=o>>>0,n||D(o,4,this.length),(this[o]|this[o+1]<<8|this[o+2]<<16)+this[o+3]*16777216};_.prototype.readUint32BE=_.prototype.readUInt32BE=function(o,n){return o=o>>>0,n||D(o,4,this.length),this[o]*16777216+(this[o+1]<<16|this[o+2]<<8|this[o+3])};_.prototype.readIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||D(o,n,this.length);for(var a=this[o],m=1,s=0;++s<n&&(m*=256);)a+=this[o+s]*m;return m*=128,a>=m&&(a-=Math.pow(2,8*n)),a};_.prototype.readIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||D(o,n,this.length);for(var a=n,m=1,s=this[o+--a];a>0&&(m*=256);)s+=this[o+--a]*m;return m*=128,s>=m&&(s-=Math.pow(2,8*n)),s};_.prototype.readInt8=function(o,n){return o=o>>>0,n||D(o,1,this.length),this[o]&128?(255-this[o]+1)*-1:this[o]};_.prototype.readInt16LE=function(o,n){o=o>>>0,n||D(o,2,this.length);var r=this[o]|this[o+1]<<8;return r&32768?r|4294901760:r};_.prototype.readInt16BE=function(o,n){o=o>>>0,n||D(o,2,this.length);var r=this[o+1]|this[o]<<8;return r&32768?r|4294901760:r};_.prototype.readInt32LE=function(o,n){return o=o>>>0,n||D(o,4,this.length),this[o]|this[o+1]<<8|this[o+2]<<16|this[o+3]<<24};_.prototype.readInt32BE=function(o,n){return o=o>>>0,n||D(o,4,this.length),this[o]<<24|this[o+1]<<16|this[o+2]<<8|this[o+3]};_.prototype.readFloatLE=function(o,n){return o=o>>>0,n||D(o,4,this.length),Ie.read(this,o,!0,23,4)};_.prototype.readFloatBE=function(o,n){return o=o>>>0,n||D(o,4,this.length),Ie.read(this,o,!1,23,4)};_.prototype.readDoubleLE=function(o,n){return o=o>>>0,n||D(o,8,this.length),Ie.read(this,o,!0,52,8)};_.prototype.readDoubleBE=function(o,n){return o=o>>>0,n||D(o,8,this.length),Ie.read(this,o,!1,52,8)};function R(e,o,n,r,a,m){if(!_.isBuffer(e))throw new TypeError('"buffer" argument must be a Buffer instance');if(o>a||o<m)throw new RangeError('"value" argument is out of bounds');if(n+r>e.length)throw new RangeError("Index out of range")}_.prototype.writeUintLE=_.prototype.writeUIntLE=function(o,n,r,a){if(o=+o,n=n>>>0,r=r>>>0,!a){var m=Math.pow(2,8*r)-1;R(this,o,n,r,m,0)}var s=1,l=0;for(this[n]=o&255;++l<r&&(s*=256);)this[n+l]=o/s&255;return n+r};_.prototype.writeUintBE=_.prototype.writeUIntBE=function(o,n,r,a){if(o=+o,n=n>>>0,r=r>>>0,!a){var m=Math.pow(2,8*r)-1;R(this,o,n,r,m,0)}var s=r-1,l=1;for(this[n+s]=o&255;--s>=0&&(l*=256);)this[n+s]=o/l&255;return n+r};_.prototype.writeUint8=_.prototype.writeUInt8=function(o,n,r){return o=+o,n=n>>>0,r||R(this,o,n,1,255,0),this[n]=o&255,n+1};_.prototype.writeUint16LE=_.prototype.writeUInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||R(this,o,n,2,65535,0),this[n]=o&255,this[n+1]=o>>>8,n+2};_.prototype.writeUint16BE=_.prototype.writeUInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||R(this,o,n,2,65535,0),this[n]=o>>>8,this[n+1]=o&255,n+2};_.prototype.writeUint32LE=_.prototype.writeUInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||R(this,o,n,4,4294967295,0),this[n+3]=o>>>24,this[n+2]=o>>>16,this[n+1]=o>>>8,this[n]=o&255,n+4};_.prototype.writeUint32BE=_.prototype.writeUInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||R(this,o,n,4,4294967295,0),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};_.prototype.writeIntLE=function(o,n,r,a){if(o=+o,n=n>>>0,!a){var m=Math.pow(2,8*r-1);R(this,o,n,r,m-1,-m)}var s=0,l=1,d=0;for(this[n]=o&255;++s<r&&(l*=256);)o<0&&d===0&&this[n+s-1]!==0&&(d=1),this[n+s]=(o/l>>0)-d&255;return n+r};_.prototype.writeIntBE=function(o,n,r,a){if(o=+o,n=n>>>0,!a){var m=Math.pow(2,8*r-1);R(this,o,n,r,m-1,-m)}var s=r-1,l=1,d=0;for(this[n+s]=o&255;--s>=0&&(l*=256);)o<0&&d===0&&this[n+s+1]!==0&&(d=1),this[n+s]=(o/l>>0)-d&255;return n+r};_.prototype.writeInt8=function(o,n,r){return o=+o,n=n>>>0,r||R(this,o,n,1,127,-128),o<0&&(o=255+o+1),this[n]=o&255,n+1};_.prototype.writeInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||R(this,o,n,2,32767,-32768),this[n]=o&255,this[n+1]=o>>>8,n+2};_.prototype.writeInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||R(this,o,n,2,32767,-32768),this[n]=o>>>8,this[n+1]=o&255,n+2};_.prototype.writeInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||R(this,o,n,4,2147483647,-2147483648),this[n]=o&255,this[n+1]=o>>>8,this[n+2]=o>>>16,this[n+3]=o>>>24,n+4};_.prototype.writeInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||R(this,o,n,4,2147483647,-2147483648),o<0&&(o=4294967295+o+1),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};function _t(e,o,n,r,a,m){if(n+r>e.length)throw new RangeError("Index out of range");if(n<0)throw new RangeError("Index out of range")}function dt(e,o,n,r,a){return o=+o,n=n>>>0,a||_t(e,o,n,4,34028234663852886e22,-34028234663852886e22),Ie.write(e,o,n,r,23,4),n+4}_.prototype.writeFloatLE=function(o,n,r){return dt(this,o,n,!0,r)};_.prototype.writeFloatBE=function(o,n,r){return dt(this,o,n,!1,r)};function lt(e,o,n,r,a){return o=+o,n=n>>>0,a||_t(e,o,n,8,17976931348623157e292,-17976931348623157e292),Ie.write(e,o,n,r,52,8),n+8}_.prototype.writeDoubleLE=function(o,n,r){return lt(this,o,n,!0,r)};_.prototype.writeDoubleBE=function(o,n,r){return lt(this,o,n,!1,r)};_.prototype.copy=function(o,n,r,a){if(!_.isBuffer(o))throw new TypeError("argument should be a Buffer");if(r||(r=0),!a&&a!==0&&(a=this.length),n>=o.length&&(n=o.length),n||(n=0),a>0&&a<r&&(a=r),a===r||o.length===0||this.length===0)return 0;if(n<0)throw new RangeError("targetStart out of bounds");if(r<0||r>=this.length)throw new RangeError("Index out of range");if(a<0)throw new RangeError("sourceEnd out of bounds");a>this.length&&(a=this.length),o.length-n<a-r&&(a=o.length-n+r);var m=a-r;return this===o&&typeof Uint8Array.prototype.copyWithin=="function"?this.copyWithin(n,r,a):Uint8Array.prototype.set.call(o,this.subarray(r,a),n),m};_.prototype.fill=function(o,n,r,a){if(typeof o=="string"){if(typeof n=="string"?(a=n,n=0,r=this.length):typeof r=="string"&&(a=r,r=this.length),a!==void 0&&typeof a!="string")throw new TypeError("encoding must be a string");if(typeof a=="string"&&!_.isEncoding(a))throw new TypeError("Unknown encoding: "+a);if(o.length===1){var m=o.charCodeAt(0);(a==="utf8"&&m<128||a==="latin1")&&(o=m)}}else typeof o=="number"?o=o&255:typeof o=="boolean"&&(o=Number(o));if(n<0||this.length<n||this.length<r)throw new RangeError("Out of range index");if(r<=n)return this;n=n>>>0,r=r===void 0?this.length:r>>>0,o||(o=0);var s;if(typeof o=="number")for(s=n;s<r;++s)this[s]=o;else{var l=_.isBuffer(o)?o:_.from(o,a),d=l.length;if(d===0)throw new TypeError('The value "'+o+'" is invalid for argument "value"');for(s=0;s<r-n;++s)this[s+n]=l[s%d]}return this};var zi=/[^+/0-9A-Za-z-_]/g;function qi(e){if(e=e.split("=")[0],e=e.trim().replace(zi,""),e.length<2)return"";for(;e.length%4!==0;)e=e+"=";return e}function Ko(e,o){o=o||1/0;for(var n,r=e.length,a=null,m=[],s=0;s<r;++s){if(n=e.charCodeAt(s),n>55295&&n<57344){if(!a){if(n>56319){(o-=3)>-1&&m.push(239,191,189);continue}else if(s+1===r){(o-=3)>-1&&m.push(239,191,189);continue}a=n;continue}if(n<56320){(o-=3)>-1&&m.push(239,191,189),a=n;continue}n=(a-55296<<10|n-56320)+65536}else a&&(o-=3)>-1&&m.push(239,191,189);if(a=null,n<128){if((o-=1)<0)break;m.push(n)}else if(n<2048){if((o-=2)<0)break;m.push(n>>6|192,n&63|128)}else if(n<65536){if((o-=3)<0)break;m.push(n>>12|224,n>>6&63|128,n&63|128)}else if(n<1114112){if((o-=4)<0)break;m.push(n>>18|240,n>>12&63|128,n>>6&63|128,n&63|128)}else throw new Error("Invalid code point")}return m}function Ki(e){for(var o=[],n=0;n<e.length;++n)o.push(e.charCodeAt(n)&255);return o}function Xi(e,o){for(var n,r,a,m=[],s=0;s<e.length&&!((o-=2)<0);++s)n=e.charCodeAt(s),r=n>>8,a=n%256,m.push(a),m.push(r);return m}function ut(e){return jo.toByteArray(qi(e))}function ao(e,o,n,r){for(var a=0;a<r&&!(a+n>=o.length||a>=e.length);++a)o[a+n]=e[a];return a}function q(e,o){return e instanceof o||e!=null&&e.constructor!=null&&e.constructor.name!=null&&e.constructor.name===o.name}function Zo(e){return e!==e}var Yi=function(){for(var e="0123456789abcdef",o=new Array(256),n=0;n<16;++n)for(var r=n*16,a=0;a<16;++a)o[r+a]=e[n]+e[a];return o}()});var yt=no((Bs,xt)=>{t();var E=xt.exports={},K,X;function Jo(){throw new Error("setTimeout has not been defined")}function Qo(){throw new Error("clearTimeout has not been defined")}(function(){try{typeof setTimeout=="function"?K=setTimeout:K=Jo}catch(e){K=Jo}try{typeof clearTimeout=="function"?X=clearTimeout:X=Qo}catch(e){X=Qo}})();function ft(e){if(K===setTimeout)return setTimeout(e,0);if((K===Jo||!K)&&setTimeout)return K=setTimeout,setTimeout(e,0);try{return K(e,0)}catch(o){try{return K.call(null,e,0)}catch(n){return K.call(this,e,0)}}}function Zi(e){if(X===clearTimeout)return clearTimeout(e);if((X===Qo||!X)&&clearTimeout)return X=clearTimeout,clearTimeout(e);try{return X(e)}catch(o){try{return X.call(null,e)}catch(n){return X.call(this,e)}}}var Z=[],Ee=!1,fe,io=-1;function Ji(){!Ee||!fe||(Ee=!1,fe.length?Z=fe.concat(Z):io=-1,Z.length&&ht())}function ht(){if(!Ee){var e=ft(Ji);Ee=!0;for(var o=Z.length;o;){for(fe=Z,Z=[];++io<o;)fe&&fe[io].run();io=-1,o=Z.length}fe=null,Ee=!1,Zi(e)}}E.nextTick=function(e){var o=new Array(arguments.length-1);if(arguments.length>1)for(var n=1;n<arguments.length;n++)o[n-1]=arguments[n];Z.push(new Tt(e,o)),Z.length===1&&!Ee&&ft(ht)};function Tt(e,o){this.fun=e,this.array=o}Tt.prototype.run=function(){this.fun.apply(null,this.array)};E.title="browser";E.browser=!0;E.env={};E.argv=[];E.version="";E.versions={};function J(){}E.on=J;E.addListener=J;E.once=J;E.off=J;E.removeListener=J;E.removeAllListeners=J;E.emit=J;E.prependListener=J;E.prependOnceListener=J;E.listeners=function(e){return[]};E.binding=function(e){throw new Error("process.binding is not supported")};E.cwd=function(){return"/"};E.chdir=function(e){throw new Error("process.chdir is not supported")};E.umask=function(){return 0}});var i,c,Us,t=T(()=>{i=S(gt()),c=S(yt()),Us=function(e){function o(){var r=this||self;return delete e.prototype.__magic__,r}if(typeof globalThis=="object")return globalThis;if(this)return o();e.defineProperty(e.prototype,"__magic__",{configurable:!0,get:o});var n=__magic__;return n}(Object)});var co,en=T(()=>{"use strict";t();co=()=>Date.now()/1e3});var bt=T(()=>{"use strict";t()});var St=T(()=>{"use strict";t()});var on=T(()=>{"use strict";t();en();bt();St()});var tn={};le(tn,{getItem:()=>ec,localStorageKeys:()=>W,removeItem:()=>oc,setItem:()=>Qi});var W,nn,Qi,ec,oc,he=T(()=>{"use strict";t();on();W={loginExpiresAt:"dapp-core-login-expires-at",logoutEvent:"dapp-core-logout-event"},nn=typeof localStorage!="undefined",Qi=({key:e,data:o,expires:n})=>{!nn||localStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},ec=e=>{if(!nn)return;let o=localStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:co()>=n.expires?(localStorage.removeItem(String(e)),null):n.data},oc=e=>{!nn||localStorage.removeItem(String(e))}});var rn={};le(rn,{clear:()=>It,getItem:()=>vt,removeItem:()=>kt,setItem:()=>wt,storage:()=>nc});var wt,vt,kt,It,nc,At=T(()=>{"use strict";t();wt=({key:e,data:o,expires:n})=>{sessionStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},vt=e=>{let o=sessionStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:Date.now()>=n.expires?(sessionStorage.removeItem(String(e)),null):n.data},kt=e=>sessionStorage.removeItem(String(e)),It=()=>sessionStorage.clear(),nc={setItem:wt,getItem:vt,removeItem:kt,clear:It}});var Q,Le=T(()=>{"use strict";t();he();At();Q={session:rn,local:tn}});function je(){return new Date().setHours(new Date().getHours()+24)}function ze(e){Q.local.setItem({key:W.loginExpiresAt,data:e,expires:e})}var an=T(()=>{"use strict";t();Le();he()});var A=T(()=>{"use strict";t()});var De=T(()=>{"use strict";t()});var Et=T(()=>{"use strict";t()});var N=T(()=>{"use strict";t()});var Lt=T(()=>{"use strict";t()});var Dt=T(()=>{"use strict";t()});var Ct=T(()=>{"use strict";t()});var Pt=T(()=>{"use strict";t()});var G=T(()=>{"use strict";t();Et();N();A();Lt();Dt();Ct();Pt()});var cn,Nt,lm,Rt,um,Mt,gm,fm,hm,Ft=T(()=>{"use strict";t();G();cn={devnet:{id:"devnet",chainId:"D",name:"Devnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://devnet-wallet.elrond.com",apiAddress:"https://devnet-api.elrond.com",explorerAddress:"http://devnet-explorer.elrond.com",apiTimeout:"4000"},testnet:{id:"testnet",chainId:"T",name:"Testnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://testnet-wallet.elrond.com",apiAddress:"https://testnet-api.elrond.com",explorerAddress:"http://testnet-explorer.elrond.com",apiTimeout:"4000"},mainnet:{id:"mainnet",chainId:"1",name:"Mainnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://wallet.elrond.com",apiAddress:"https://api.elrond.com",explorerAddress:"https://explorer.elrond.com",apiTimeout:"4000"}},{chainId:Nt,egldLabel:lm}=cn["devnet"],{chainId:Rt,egldLabel:um}=cn["testnet"],{chainId:Mt,egldLabel:gm}=cn["mainnet"],fm={["devnet"]:Nt,["testnet"]:Rt,["mainnet"]:Mt},hm={[Nt]:"devnet",[Rt]:"testnet",[Mt]:"mainnet"}});var pn=T(()=>{"use strict";t()});var Ot=T(()=>{"use strict";t()});var sn,mn=T(()=>{"use strict";t();sn="transaction-status-toast-id"});var qe,po,Bt,so,Ut,Ce,I=T(()=>{"use strict";t();De();Ft();pn();Ot();mn();qe=5e4,po=1e9,Bt=1,so="logout",Ut="login",Ce="0"});var _n,P,ee,M=T(()=>{"use strict";t();_n=require("@reduxjs/toolkit");I();P=(0,_n.createAction)(so),ee=(0,_n.createAction)(Ut,e=>({payload:e}))});var Gt,Wt,$t,Mm,Vt,tc,Ke,Ht,rc,jt,zt,dn=T(()=>{"use strict";t();Gt=require("@reduxjs/toolkit");an();A();M();Wt={loginMethod:"",walletConnectLogin:null,ledgerLogin:null,tokenLogin:null,walletLogin:null,extensionLogin:null,isLoginSessionInvalid:!1},$t=(0,Gt.createSlice)({name:"loginInfoSlice",initialState:Wt,reducers:{setLoginMethod:(e,o)=>{e.loginMethod=o.payload},setTokenLogin:(e,o)=>{e.tokenLogin=o.payload},setTokenLoginSignature:(e,o)=>{(e==null?void 0:e.tokenLogin)!=null&&(e.tokenLogin.signature=o.payload)},setWalletLogin:(e,o)=>{e.walletLogin=o.payload},setWalletConnectLogin:(e,o)=>{e.walletConnectLogin=o.payload},setLedgerLogin:(e,o)=>{e.ledgerLogin=o.payload},invalidateLoginSession:e=>{e.isLoginSessionInvalid=!0}},extraReducers:e=>{e.addCase(P,()=>Wt),e.addCase(ee,(o,n)=>{o.loginMethod=n.payload.loginMethod,ze(je())})}}),{setLoginMethod:Mm,setWalletConnectLogin:Vt,setLedgerLogin:tc,setTokenLogin:Ke,setTokenLoginSignature:Ht,setWalletLogin:rc,invalidateLoginSession:jt}=$t.actions,zt=$t.reducer});var qt,Kt,Pe,Ne=T(()=>{"use strict";t();qt=!1,Kt=(e=!0)=>{qt=e},Pe=(e,o=0)=>{qt||setTimeout(()=>{window.location.href=e},o)}});var Xt=T(()=>{"use strict";t();Ne()});function ln(e){return e[Math.floor(Math.random()*e.length)]}var Yt=T(()=>{"use strict";t()});var un=T(()=>{"use strict";t();pn()});var Re=T(()=>{"use strict";t();Xt();Yt();un()});var Zt,Jt,gn,ac,Qt,Ym,Zm,er,fn=T(()=>{"use strict";t();Zt=require("@reduxjs/toolkit"),Jt=S(require("lodash.omit"));Re();gn={id:"not-configured",chainId:"",name:"NOT CONFIGURED",egldLabel:"",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"",walletConnectBridgeAddress:"",walletConnectV2RelayAddress:"",walletConnectV2ProjectId:"",walletAddress:"",apiAddress:"",explorerAddress:"",apiTimeout:"4000"},ac={network:gn,chainID:"-1"},Qt=(0,Zt.createSlice)({name:"appConfig",initialState:ac,reducers:{initializeNetworkConfig:(e,o)=>{let n=ln(o.payload.walletConnectBridgeAddresses),r=ln(o.payload.walletConnectV2RelayAddresses),a=(0,Jt.default)(o.payload,"walletConnectBridgeAddresses","walletConnectV2RelayAddresses");e.network=v(f(f({},e.network),a),{walletConnectBridgeAddress:n,walletConnectV2RelayAddress:r})},setChainID:(e,o)=>{e.chainID=o.payload}}}),{initializeNetworkConfig:Ym,setChainID:Zm}=Qt.actions,er=Qt.reducer});var Tn,or,nr,mo,hn,tr,_o,a_,lo,i_,ic,cc,c_,p_,s_,rr,ar,xn=T(()=>{"use strict";t();Tn=require("@elrondnetwork/erdjs"),or=require("@reduxjs/toolkit"),nr=require("redux-persist");I();Le();he();M();mo={balance:"...",address:"",nonce:0,txCount:0,scrCount:0,claimableRewards:Ce},hn={address:"",websocketEvent:null,accounts:{"":mo},ledgerAccount:null,publicKey:"",walletConnectAccount:null,isAccountLoading:!0,accountLoadingError:null},tr=(0,or.createSlice)({name:"accountInfoSlice",initialState:hn,reducers:{setAddress:(e,o)=>{let n=o.payload;e.address=n,e.publicKey=new Tn.Address(n).hex()},setAccount:(e,o)=>{let n=e.address===o.payload.address;e.accounts={[e.address]:n?o.payload:mo},e.isAccountLoading=!1,e.accountLoadingError=null},setAccountNonce:(e,o)=>{let{address:n}=e;e.accounts[n].nonce=o.payload},setAccountShard:(e,o)=>{e.shard=o.payload},setLedgerAccount:(e,o)=>{e.ledgerAccount=o.payload},updateLedgerAccount:(e,o)=>{e.ledgerAccount!=null&&(e.ledgerAccount.index=o.payload.index,e.ledgerAccount.address=o.payload.address)},setWalletConnectAccount:(e,o)=>{e.walletConnectAccount=o.payload},setIsAccountLoading:(e,o)=>{e.isAccountLoading=o.payload,e.accountLoadingError=null},setAccountLoadingError:(e,o)=>{e.accountLoadingError=o.payload,e.isAccountLoading=!1},setWebsocketEvent:(e,o)=>{e.websocketEvent={timestamp:Date.now(),message:o.payload}}},extraReducers:e=>{e.addCase(P,()=>(Q.local.removeItem(W.loginExpiresAt),hn)),e.addCase(ee,(o,n)=>{let{address:r}=n.payload;o.address=r,o.publicKey=new Tn.Address(r).hex()}),e.addCase(nr.REHYDRATE,(o,n)=>{var u;if(!((u=n.payload)!=null&&u.account))return;let{account:r}=n.payload,{address:a,shard:m,accounts:s,publicKey:l}=r;o.address=a,o.shard=m;let d=s&&a in s;o.accounts=d?s:hn.accounts,o.publicKey=l})}}),{setAccount:_o,setAddress:a_,setAccountNonce:lo,setAccountShard:i_,setLedgerAccount:ic,updateLedgerAccount:cc,setWalletConnectAccount:c_,setIsAccountLoading:p_,setAccountLoadingError:s_,setWebsocketEvent:rr}=tr.actions,ar=tr.reducer});function oe(e){return e!=null&&(gc(e)||xc(e))}function Te(e){return e!=null&&(fc(e)||yc(e))}function xe(e){return e!=null&&(hc(e)||bc(e))}function yn(e){return e!=null&&Tc(e)}function gc(e){return e!=null&&pc.includes(e)}function fc(e){return e!=null&&sc.includes(e)}function hc(e){return e!=null&&mc.includes(e)}function Tc(e){return e!=null&&_c.includes(e)}function xc(e){return e!=null&&dc.includes(e)}function yc(e){return e!=null&&lc.includes(e)}function bc(e){return e!=null&&uc.includes(e)}var pc,sc,mc,_c,dc,lc,uc,Fe=T(()=>{"use strict";t();A();pc=["sent"],sc=["success"],mc=["fail","cancelled","timedOut"],_c=["timedOut"],dc=["pending"],lc=["success"],uc=["fail","invalid"]});var ir,cr,Xe,Sc,pr,Ye,ye,wc,Ze,vc,kc,T_,Ic,sr,bn,uo,go=T(()=>{"use strict";t();ir=require("@reduxjs/toolkit"),cr=require("redux-persist");A();Fe();M();Xe={signedTransactions:{},transactionsToSign:null,signTransactionsError:null,signTransactionsCancelMessage:null,customTransactionInformationForSessionId:{}},Sc={signWithoutSending:!1,sessionInformation:null,redirectAfterSign:!1},pr=(0,ir.createSlice)({name:"transactionsSlice",initialState:Xe,reducers:{moveTransactionsToSignedState:(e,o)=>{var d,u;let{sessionId:n,transactions:r,errorMessage:a,status:m,redirectRoute:s}=o.payload,l=((d=e.customTransactionInformationForSessionId)==null?void 0:d[n])||Sc;e.signedTransactions[n]={transactions:r,status:m,errorMessage:a,redirectRoute:s,customTransactionInformation:l},((u=e==null?void 0:e.transactionsToSign)==null?void 0:u.sessionId)===n&&(e.transactionsToSign=Xe.transactionsToSign)},clearSignedTransaction:(e,o)=>{e.signedTransactions[o.payload]&&delete e.signedTransactions[o.payload]},clearTransactionToSign:e=>{e!=null&&e.transactionsToSign&&(e.transactionsToSign=null)},updateSignedTransaction:(e,o)=>{e.signedTransactions=f(f({},e.signedTransactions),o.payload)},updateSignedTransactions:(e,o)=>{let{sessionId:n,status:r,errorMessage:a,transactions:m}=o.payload;e.signedTransactions[n]!=null&&(e.signedTransactions[n].status=r,a!=null&&(e.signedTransactions[n].errorMessage=a),m!=null&&(e.signedTransactions[n].transactions=m))},updateSignedTransactionStatus:(e,o)=>{var l,d,u,y,h,L;let{sessionId:n,status:r,errorMessage:a,transactionHash:m}=o.payload,s=(d=(l=e.signedTransactions)==null?void 0:l[n])==null?void 0:d.transactions;if(s!=null){e.signedTransactions[n].transactions=s.map(j=>j.hash===m?v(f({},j),{status:r,errorMessage:a}):j);let C=(y=(u=e.signedTransactions[n])==null?void 0:u.transactions)==null?void 0:y.every(j=>Te(j.status)),O=(L=(h=e.signedTransactions[n])==null?void 0:h.transactions)==null?void 0:L.every(j=>xe(j.status));C&&(e.signedTransactions[n].status="success"),O&&(e.signedTransactions[n].status="fail")}},setTransactionsToSign:(e,o)=>{e.transactionsToSign=o.payload;let{sessionId:n,customTransactionInformation:r}=o.payload;e.customTransactionInformationForSessionId[n]=r,e.signTransactionsError=null},clearAllTransactionsToSign:e=>{e.transactionsToSign=Xe.transactionsToSign,e.signTransactionsError=null},clearAllSignedTransactions:e=>{e.signedTransactions=Xe.signedTransactions},setSignTransactionsError:(e,o)=>{e.signTransactionsError=o.payload},setSignTransactionsCancelMessage:(e,o)=>{e.signTransactionsCancelMessage=o.payload}},extraReducers:e=>{e.addCase(P,()=>Xe),e.addCase(cr.REHYDRATE,(o,n)=>{var s;if(!((s=n.payload)!=null&&s.transactions))return;let{signedTransactions:r,customTransactionInformationForSessionId:a}=n.payload.transactions,m=Object.entries(r).reduce((l,[d,u])=>{let y=new Date(d),h=new Date;return h.setHours(h.getHours()+5),h-y>0||(l[d]=u),l},{});a!=null&&(o.customTransactionInformationForSessionId=a),r!=null&&(o.signedTransactions=m)})}}),{updateSignedTransactionStatus:Ye,updateSignedTransactions:ye,setTransactionsToSign:wc,clearAllTransactionsToSign:Ze,clearAllSignedTransactions:vc,clearSignedTransaction:kc,clearTransactionToSign:T_,setSignTransactionsError:Ic,setSignTransactionsCancelMessage:sr,moveTransactionsToSignedState:bn}=pr.actions,uo=pr.reducer});var mr,wn,vn,kn,Ac,Sn,_r,S_,Ec,dr,fo,ho=T(()=>{"use strict";t();mr=require("@reduxjs/toolkit");M();wn="Transaction failed",vn="Transaction successful",kn="Processing transaction",Ac="Transaction submitted",Sn={},_r=(0,mr.createSlice)({name:"transactionsInfo",initialState:Sn,reducers:{setTransactionsDisplayInfo(e,o){let{sessionId:n,transactionsDisplayInfo:r}=o.payload;n!=null&&(e[n]={errorMessage:(r==null?void 0:r.errorMessage)||wn,successMessage:(r==null?void 0:r.successMessage)||vn,processingMessage:(r==null?void 0:r.processingMessage)||kn,submittedMessage:(r==null?void 0:r.submittedMessage)||Ac,transactionDuration:r==null?void 0:r.transactionDuration})},clearTransactionsInfoForSessionId(e,o){o.payload!=null&&delete e[o.payload]},clearTransactionsInfo:()=>Sn},extraReducers:e=>{e.addCase(P,()=>Sn)}}),{clearTransactionsInfo:S_,setTransactionsDisplayInfo:Ec,clearTransactionsInfoForSessionId:dr}=_r.actions,fo=_r.reducer});var ur,lr,gr,fr,Lc,I_,Dc,hr,In=T(()=>{"use strict";t();ur=require("@reduxjs/toolkit");M();lr={},gr=(0,ur.createSlice)({name:"modalsSlice",initialState:lr,reducers:{setTxSubmittedModal:(e,o)=>{e.txSubmittedModal=o.payload},setNotificationModal:(e,o)=>{e.notificationModal=o.payload},clearTxSubmittedModal:e=>{e.txSubmittedModal=void 0},clearNotificationModal:e=>{e.notificationModal=void 0}},extraReducers:e=>{e.addCase(P,()=>lr)}}),{setTxSubmittedModal:fr,setNotificationModal:Lc,clearTxSubmittedModal:I_,clearNotificationModal:Dc}=gr.actions,hr=gr.reducer});var xr,Tr,yr,Cc,br,R_,M_,F_,O_,To,xo=T(()=>{"use strict";t();xr=require("@reduxjs/toolkit");mn();G();en();M();Tr={customToasts:[],transactionToasts:[],failTransactionToast:null},yr=(0,xr.createSlice)({name:"toastsSlice",initialState:Tr,reducers:{addCustomToast:(e,o)=>{e.customToasts.push(v(f({},o.payload),{type:"custom",toastId:o.payload.toastId||`custom-toast-${e.customToasts.length+1}`}))},removeCustomToast:(e,o)=>{e.customToasts=e.customToasts.filter(n=>n.toastId!==o.payload)},addTransactionToast:(e,o)=>{e.transactionToasts.push({type:"transaction",startTimestamp:co(),toastId:o.payload||`custom-toast-${e.transactionToasts.length+1}`})},removeTransactionToast:(e,o)=>{e.transactionToasts=e.transactionToasts.filter(n=>n.toastId!==o.payload)},addFailTransactionToast:(e,o)=>{e.failTransactionToast=v(f({},o.payload),{toastId:sn})},removeFailTransactionToast:e=>{e.failTransactionToast=null}},extraReducers:e=>{e.addCase(P,()=>Tr)}}),{addCustomToast:Cc,removeCustomToast:br,addTransactionToast:R_,removeTransactionToast:M_,addFailTransactionToast:F_,removeFailTransactionToast:O_}=yr.actions,To=yr.reducer});var k=T(()=>{"use strict";t();dn();fn();xn();go();ho();In();xo()});var An=T(()=>{"use strict";t()});var kr,vr,Ir,td,rd,ad,id,cd,pd,Ar,Er=T(()=>{"use strict";t();kr=require("@reduxjs/toolkit");M();vr={visible:!1,modal:{showHeader:!0,showFooter:!1,headerText:"",footerText:""}},Ir=(0,kr.createSlice)({name:"dappModalsSlice",initialState:vr,reducers:{setDappModalVisibility:(e,o)=>{e.visible=o.payload},setDappModalHeaderVisibility:(e,o)=>{e.modal.showHeader=o.payload},setDappModalFooterVisibility:(e,o)=>{e.modal.showFooter=o.payload},setDappModalHeaderText:(e,o)=>{e.modal.headerText=o.payload},setDappModalFooterText:(e,o)=>{e.modal.footerText=o.payload},setDappModalConfig:(e,o)=>{e.modal=f({},o.payload)}},extraReducers:e=>{e.addCase(P,()=>vr)}}),{setDappModalVisibility:td,setDappModalHeaderVisibility:rd,setDappModalFooterVisibility:ad,setDappModalHeaderText:id,setDappModalFooterText:cd,setDappModalConfig:pd}=Ir.actions,Ar=Ir.reducer});var Lr,Mc,Fc,Oe,bo=T(()=>{"use strict";t();Lr=require("@reduxjs/toolkit");An();xn();Er();dn();In();fn();xo();ho();go();Mc={["account"]:ar,["networkConfig"]:er,["loginInfo"]:zt,["modals"]:hr,["dappModal"]:Ar,["toasts"]:To,["transactions"]:uo,["transactionsInfo"]:fo},Fc=(e={})=>(0,Lr.combineReducers)(f(f({},Mc),e)),Oe=Fc});var Pr={};le(Pr,{default:()=>Vc});function En(e,o=[]){return{key:e,version:1,storage:Cr.default,blacklist:o}}var be,Dr,Cr,Oc,Bc,Uc,Wc,Gc,$c,Vc,Nr=T(()=>{"use strict";t();be=require("redux-persist"),Dr=S(require("redux-persist/lib/storage")),Cr=S(require("redux-persist/lib/storage/session"));bo();k();xo();ho();go();An();Oc={2:e=>v(f({},e),{networkConfig:gn})};Bc=En("dapp-core-transactionsInfo"),Uc=En("dapp-core-transactions",["transactionsToSign"]),Wc=En("dapp-core-toasts"),Gc={key:"dapp-core-store",version:2,storage:Dr.default,whitelist:["account","loginInfo","modals","networkConfig"],migrate:(0,be.createMigrate)(Oc,{debug:!1})},$c={["toasts"]:(0,be.persistReducer)(Wc,To),["transactions"]:(0,be.persistReducer)(Uc,uo),["transactionsInfo"]:(0,be.persistReducer)(Bc,fo)},Vc=(0,be.persistReducer)(Gc,Oe($c))});var Rr={};le(Rr,{default:()=>Hc});var Hc,Mr=T(()=>{"use strict";t();bo();Hc=Oe()});var Fr={};le(Fr,{default:()=>zc});var V,jc,zc,Or=T(()=>{"use strict";t();V=require("redux-persist"),jc=[V.FLUSH,V.REHYDRATE,V.PAUSE,V.PERSIST,V.PURGE,V.REGISTER],zc=jc});var Wr={};le(Wr,{default:()=>Ur});function Ur(e){return(0,Br.persistStore)(e)}var Br,Gr=T(()=>{"use strict";t();Br=require("redux-persist")});var Ps={};le(Ps,{CustomComponents:()=>Cs});module.exports=He(Ps);t();var $e=S(require("react"));t();var oi=require("react");t();t();t();var qr=require("react"),Be=require("react-redux");t();var Ln=require("@reduxjs/toolkit"),jr=require("react-redux/es/utils/Subscription");k();t();var Sr=S(require("lodash.throttle"));I();k();an();Le();he();var Pc=[so],Nc=(0,Sr.default)(()=>{ze(je())},5e3),wr=e=>o=>n=>{var d;if(Pc.includes(n.type))return o(n);let r=e.getState(),a=Q.local.getItem(W.loginExpiresAt);if(!((d=r==null?void 0:r.loginInfo)==null?void 0:d.isLoggedIn))return o(n);if(a==null)return ze(je());let s=Date.now();return a-s<0?setTimeout(()=>x(void 0,null,function*(){console.log("session expired");try{e.dispatch(jt())}catch(u){console.error("error logging out",u)}}),1e3):(Nc(),o(n))};t();t();function yo(){return typeof sessionStorage!="undefined"}var $r=yo()?(Nr(),He(Pr)).default:(Mr(),He(Rr)).default,Vr=yo()?(Or(),He(Fr)).default:[],Hr=yo()?(Gr(),He(Wr)).default:e=>e;bo();var b=(0,Ln.configureStore)({reducer:$r,middleware:e=>e({serializableCheck:{ignoredActions:[...Vr,lo.type,_o.type],ignoredPaths:["payload.nonce","account.account.nonce","providers.accountProvider"]}}).concat(wr)}),zr=(0,jr.createSubscription)(b),Nd=Hr(b),Rd=(0,Ln.configureStore)({reducer:Oe});var qc={store:b,subscription:zr},Dn=(0,qr.createContext)(qc),Bd=(0,Be.createStoreHook)(Dn),F=(0,Be.createDispatchHook)(Dn),w=(0,Be.createSelectorHook)(Dn);t();t();k();t();var Kr=S(require("lodash.isequal")),So=require("reselect"),g=(0,So.createSelectorCreator)(So.defaultMemoize,Kr.default);var ne=e=>e.account,Se=g(ne,e=>e.address),pe=g(ne,Se,(e,o)=>o in e.accounts?e.accounts[o]:mo),Xr=g(ne,pe,(e,o)=>{let a=e,{accounts:n}=a,r=$o(a,["accounts"]);return v(f({},r),{account:o})}),Kc=g(pe,e=>e.balance),Yr=g(pe,e=>{var o;return((o=e==null?void 0:e.nonce)==null?void 0:o.valueOf())||0}),Vd=g(ne,e=>e.shard),Xc=g(ne,e=>e.ledgerAccount),Hd=g(ne,e=>e.walletConnectAccount),jd=g(ne,e=>e.isAccountLoading),zd=g(ne,e=>e.accountLoadingError),Cn=g(ne,e=>e.websocketEvent);t();var te=e=>e.networkConfig,wo=g(te,e=>e.chainID),Yc=g(te,e=>e.network.walletConnectBridgeAddress),Zc=g(te,e=>e.network.walletConnectV2RelayAddress),Jc=g(te,e=>e.network.walletConnectV2ProjectId),Zr=g(te,e=>e.network.walletConnectDeepLink),se=g(te,e=>e.network),vo=g(se,e=>e.apiAddress),Jr=g(se,e=>e.explorerAddress),Qr=g(se,e=>e.egldLabel);t();A();var we=e=>e.loginInfo,Qd=g(we,e=>e.loginMethod),Ue=g(we,Se,(e,o)=>e.loginMethod!=""&&Boolean(o)),el=g(we,e=>e.walletConnectLogin),ol=g(we,e=>e.ledgerLogin),ea=g(we,e=>e.walletLogin),nl=g(we,e=>e.isLoginSessionInvalid);t();t();var B=require("@elrondnetwork/erdjs");I();t();function Pn(e){try{return i.Buffer.from(e,"base64").toString()===atob(e)}catch(o){return!1}}function We(e){var a,m,s;let{data:o}=e,n=Pn(o!=null?o:"")?B.TransactionPayload.fromEncoded(o):new B.TransactionPayload(o),r=new B.Transaction(f({value:e.value.valueOf(),data:n,nonce:e.nonce.valueOf(),receiver:new B.Address(e.receiver),sender:new B.Address(e.sender),gasLimit:(a=e.gasLimit.valueOf())!=null?a:qe,gasPrice:(m=e.gasPrice.valueOf())!=null?m:po,chainID:e.chainID.valueOf(),version:new B.TransactionVersion((s=e.version)!=null?s:Bt)},e.options?{options:new B.TransactionOptions(e.options)}:{}));return r.applySignature({hex:()=>e.signature||""},new B.Address(e.sender)),r}t();Fe();t();var ep=S(require("bignumber.js"));G();t();t();t();t();var np=require("@elrondnetwork/erdjs"),tp=S(require("bignumber.js"));A();t();var ta=require("@elrondnetwork/erdjs");function op(e){try{let o=new ta.Address(e);return Boolean(o.bech32())}catch(o){return!1}}function Je(e){return(e==null?void 0:e.startsWith("erd"))&&e.length===62&&/^\w+$/.test(e)&&op(e)}t();t();t();var ap=require("@elrondnetwork/erdjs/out");t();I();t();var ra=require("@elrondnetwork/erdjs");G();t();t();t();t();A();t();t();t();t();t();t();t();var ko="accounts";t();t();t();t();N();t();t();N();t();t();t();var up=S(require("bignumber.js"));I();t();var lp=require("@elrondnetwork/erdjs"),Nn=S(require("bignumber.js"));I();t();var _p=S(require("bignumber.js"));t();Nn.default.config({ROUNDING_MODE:Nn.default.ROUND_FLOOR});t();t();t();I();t();t();t();t();I();t();I();t();var gp=require("@elrondnetwork/erdjs");I();t();var Eo=require("@elrondnetwork/erdjs"),Tp=require("@elrondnetwork/erdjs"),xp=require("@elrondnetwork/erdjs-network-providers");I();t();t();var fp=S(require("bignumber.js"));I();t();I();t();t();t();t();t();I();t();N();var bp=["reDelegateRewards","claimRewards","unBond"],Sp=["wrapEgld","unwrapEgld"],wp=["unStake"],vp=["unDelegate"];t();t();t();I();t();var Ep=S(require("bignumber.js"));t();t();t();N();t();var Lp=S(require("bignumber.js"));t();t();t();N();t();N();t();var Pp=S(require("bignumber.js"));I();N();t();N();t();N();t();var Np=S(require("bignumber.js"));t();A();t();A();t();t();var Rp=require("react");N();var Gh=["create","localMint","ESDTLocalMint","addQuantity","burn","localBurn","ESDTLocalBurn","wipe","writeLog","signalError"];t();var Mp=S(require("anchorme"));t();var sa=require("react");G();t();t();N();t();var Fp=require("@elrondnetwork/erdjs/out"),Op=S(require("bignumber.js"));N();t();N();t();t();A();var TT=[{label:"Raw",value:"raw"},{label:"Text",value:"text"},{label:"Decimal",value:"decimal"},{label:"Smart",value:"smart"}];t();var ma=require("react");G();t();t();t();t();t();I();var Do=e=>e.transactions,_e=g(Do,e=>e.signedTransactions),Up=g(Do,e=>e.signTransactionsError),Wp=g(Do,e=>e.signTransactionsCancelMessage),Co=e=>o=>Object.entries(o).reduce((n,[r,a])=>(e(a.status)&&(n[r]=a),n),{}),Mn=g(_e,Co(oe)),_a=g(_e,Co(Te)),da=g(_e,Co(xe)),Gp=g(_e,Co(yn)),Fn=g(Do,e=>{var o;return(e==null?void 0:e.transactionsToSign)==null?null:v(f({},e.transactionsToSign),{transactions:((o=e==null?void 0:e.transactionsToSign)==null?void 0:o.transactions.map(n=>We(n)))||[]})}),$p=g(_e,(e,o)=>o,(e,o)=>o!=null?(e==null?void 0:e[o])||{}:{});t();k();var la={errorMessage:wn,successMessage:vn,processingMessage:kn},Vp=e=>e.transactionsInfo,Hp=g(Vp,(e,o)=>o,(e,o)=>o!=null&&(e==null?void 0:e[Number(o)])||la);t();var ua=e=>e.modals,gx=g(ua,e=>e.txSubmittedModal),jp=g(ua,e=>e.notificationModal);t();var On=e=>e.toasts,Tx=g(On,e=>e.customToasts),xx=g(On,e=>e.transactionToasts),yx=g(On,e=>e.failTransactionToast);var Bn=()=>w(te);t();var ga=require("react");t();k();t();t();t();var qp=require("react"),Kp=require("@elrondnetwork/erdjs-web-wallet-provider"),Xp=S(require("qs"));I();k();A();t();var $n=require("react"),os=require("@elrondnetwork/erdjs-extension-provider");I();t();t();t();t();var ha=require("@elrondnetwork/erdjs-extension-provider"),Ta=require("@elrondnetwork/erdjs-hw-provider"),No=require("@elrondnetwork/erdjs-wallet-connect-provider"),xa=require("@elrondnetwork/erdjs-web-wallet-provider");I();A();var Ge=e=>{switch(e==null?void 0:e.constructor){case xa.WalletProvider:return"wallet";case No.WalletConnectProvider:return"walletconnect";case No.WalletConnectV2Provider:return"walletconnectv2";case Ta.HWProvider:return"ledger";case ha.ExtensionProvider:return"extension";case Po:return"";default:return"extra"}};var re=e=>`Unable to perform ${e}, Provider not initialized`,Po=class{init(){return Promise.resolve(!1)}login(o){throw new Error(re(`login with options: ${o}`))}logout(o){throw new Error(re(`logout with options: ${o}`))}getAddress(){throw new Error(re("getAddress"))}isInitialized(){return!1}isConnected(){return Promise.resolve(!1)}sendTransaction(o,n){throw new Error(re(`sendTransaction with transactions: ${o} options: ${n}`))}signTransaction(o,n){throw new Error(re(`signTransaction with transactions: ${o} options: ${n}`))}signTransactions(o,n){throw new Error(re(`signTransactions with transactions: ${o} options: ${n}`))}signMessage(o,n){throw new Error(re(`signTransactions with ${o} and options ${n}`))}sendCustomMessage({method:o,params:n}){throw new Error(re(`sendCustomMessage with method: ${o} params: ${n}`))}sendCustomRequest(o){throw new Error(re(`sendSessionEvent with options: ${o}`))}ping(){return Promise.resolve(!1)}},Un=new Po;var Yp=Un;function H(){return Yp||Un}k();A();t();t();M();G();t();function ve(){return Ue(b.getState())}t();t();A();t();t();function ya(){let e=H();return Ge(e)}function ae(e){return ya()===e}function Qe(){let{search:e}=window.location,o=b.getState(),n=H(),r=Se(o),a=Ue(o),m=ea(o);if(!n)throw"provider not initialized";return ae("ledger")&&a?new Promise(s=>{s(r)}):!ae("")&&!ae("wallet")&&!ae("extra")?n.getAddress():new Promise(s=>{if(m!=null){let l=new URLSearchParams(e),d=Object.fromEntries(l);Je(d.address)&&s(d.address)}a&&s(r),s("")})}t();t();var Sa=S(require("axios"));t();t();var ba=e=>{let o=e!=null?e:vo(b.getState());return o.endsWith("/")?o.slice(0,-1):o};function wa(e){return x(this,null,function*(){if(!e)return null;let n=`${ba()}/${ko}/${e}`;try{let{data:r}=yield Sa.default.get(n);return r}catch(r){console.error("error fetching configuration for ",n)}return null})}function eo(e){return x(this,null,function*(){return yield wa(e)})}t();k();function va(e){b.dispatch(lo(e))}t();function Ro(e){let o=b.getState(),n=Yr(o);return e?n&&!isNaN(n)?Math.max(n,e.nonce):e.nonce:n}t();t();k();var ka=()=>x(void 0,null,function*(){try{let e=yield Qe();try{let o=yield eo(e);if(o!=null){let n=v(f({},o),{nonce:Ro(o)});return b.dispatch(_o(n)),n}}catch(o){console.error("Failed getting account ",o)}}catch(e){console.error("Failed getting address ",e)}return null});function Mo(){return x(this,null,function*(){let e=H();if(e==null)throw"Provider not initialized";if(!e.isInitialized||e.isInitialized())return ka();try{if(!e.init)throw"Current provider does not have init() function";return(yield e.init())?ka():void 0}catch(o){console.error("Failed initializing provider ",o)}})}t();var Ia=require("@elrondnetwork/erdjs");t();M();G();Ne();Le();he();var Zp=e=>{Q.local.setItem({key:W.logoutEvent,data:e,expires:0}),Q.local.removeItem(W.logoutEvent)};function Fo(e,o){return x(this,null,function*(){let n=H(),r=Ge(n),a=ve(),m=r==="wallet";if(!a||!n){Wn(e,o,!1);return}try{let s=yield Qe();Zp(s)}catch(s){Wn(e,o,!1),console.error("error fetching logout address",s)}m&&Kt(),b.dispatch(P());try{let l=m&&!e?window.location.origin:e;yield n.logout({callbackUrl:l}),Wn(e,o,m)}catch(s){console.error("error logging out",s)}})}function Wn(e,o,n){e&&!n&&(typeof o=="function"?o(e):Pe(e))}t();t();t();t();k();t();t();t();t();t();Le();Ne();on();t();t();t();function Aa(e){return new Promise(o=>{setTimeout(()=>o(),e)})}var Ea=(e,o,n,r=0)=>x(void 0,null,function*(){try{return yield e(...n)}catch(a){return r<o.retries?((o==null?void 0:o.delay)!=null&&(yield Aa(o.delay)),Ea(e,o,n,r+1)):null}}),La=(e,o={retries:5,delay:500})=>(...n)=>x(void 0,null,function*(){return Ea(e,o,n)});t();t();var Da=S(require("axios"));function Gn(e){return x(this,null,function*(){try{let{data:o}=yield Da.default.get(`${e}/websocket/config`);return`wss://${o.url}`}catch(o){throw console.error(o),new Error("Can not get websocket url")}})}t();t();var Jp=S(require("platform"));t();t();t();var ns=S(require("axios")),ts=S(require("swr"));t();t();var rs=S(require("axios"));t();var Vn=()=>w(Xr);t();var Na=require("react");t();var Pa=require("react");un();k();G();Ne();t();De();k();t();t();t();t();t();function Ra(){let e=w(Mn),o=Object.entries(e),n=(o==null?void 0:o.length)>0;return{pendingTransactions:e,pendingTransactionsArray:o,hasPendingTransactions:n}}Fe();t();t();t();var Ma=S(require("axios"));function Fa(e){return x(this,null,function*(){let{apiAddress:o,apiTimeout:n}=se(b.getState()),r=e.map(m=>Ma.default.post(`${o}/transactions`,m.toPlainObject(),{timeout:parseInt(n)}));return(yield Promise.all(r)).map(({data:m})=>m.txHash)})}t();var Oa=S(require("axios"));function Oo(e){return x(this,null,function*(){let o=vo(b.getState()),n=e.map(a=>a.hash),{data:r}=yield Oa.default.get(`${o}/transactions`,{params:{hashes:n.join(","),withScResults:!0}});return e.map(({hash:a,previousStatus:m})=>{let s=r.find(l=>(l==null?void 0:l.txHash)===a);return{hash:a,data:s==null?void 0:s.data,invalidTransaction:s==null,status:s==null?void 0:s.status,results:s==null?void 0:s.results,sender:s==null?void 0:s.sender,receiver:s==null?void 0:s.receiver,previousStatus:m,hasStatusChanged:s&&s.status!==m}})})}t();var cs=S(require("axios"));t();t();var ps=S(require("axios"));t();var ss=S(require("axios"));t();k();A();t();Fe();function Ua(e,o){return e.reduce((r,{status:a,hash:m})=>(m!=null&&!o.includes(m)&&oe(a)&&r.push({hash:m,previousStatus:a}),r),[])}t();k();A();function Wa({results:e,hash:o,sessionId:n}){let r=e==null?void 0:e.find(a=>(a==null?void 0:a.returnMessage)!=="");b.dispatch(Ye({transactionHash:o,sessionId:n,status:"fail",errorMessage:r==null?void 0:r.returnMessage})),b.dispatch(ye({sessionId:n,status:"fail",errorMessage:r==null?void 0:r.returnMessage}))}t();k();G();function Hn(e){b.dispatch(ye({sessionId:e,status:"timedOut"}))}var Bo={},jn=[];function ms({serverTransaction:e,sessionId:o,customTransactionInformation:n,shouldRefreshBalance:r}){let{hash:a,status:m,results:s,invalidTransaction:l,hasStatusChanged:d}=e;try{if(jn.includes(a))return;if(Bo[a]>30){Hn(o);return}if(l||oe(m)){Bo[a]=Bo[a]?Bo[a]+1:1;return}d&&(Te(m)&&(n==null?void 0:n.completedTransactionsDelay)!=null?(jn.push(a),setTimeout(()=>b.dispatch(Ye({sessionId:o,status:"success",transactionHash:a})),n==null?void 0:n.completedTransactionsDelay)):b.dispatch(Ye({sessionId:o,status:m,transactionHash:a}))),r||Mo(),xe(m)&&Wa({sessionId:o,hash:a,results:s})}catch(u){console.error(u),Hn(o)}}function Ga(m){return x(this,arguments,function*({sessionId:e,transactionBatch:{transactions:o,customTransactionInformation:n},getTransactionsByHash:r=Oo,shouldRefreshBalance:a}){try{if(o==null)return;let s=Ua(o,jn),l=yield r(s);for(let d of l)ms({serverTransaction:d,sessionId:e,customTransactionInformation:n,shouldRefreshBalance:a})}catch(s){console.error(s)}})}function $a(){let{pendingTransactionsArray:e}=Ra();function o(n){return x(this,null,function*(){let r=e.filter(([a,m])=>a!=null&&oe(m.status));if(r.length>0)for(let[a,m]of r)yield Ga(f({sessionId:a,transactionBatch:m},n));n.shouldRefreshBalance&&Mo()})}return o}t();var hs=require("react");t();t();t();k();t();t();var us=S(require("bignumber.js"));t();var _s=`.dapp-core-component__main__link-style {
  color: #1392ff;
}

.dapp-core-component__main__link-second-style {
  color: #9dabbd;
}
.dapp-core-component__main__link-second-style:hover {
  color: #1392ff;
}
.dapp-core-component__main__link-second-style:hover svg {
  color: #1392ff;
}

.dapp-core-component__main__wrapper {
  min-height: 100vh;
}

.dapp-core-component__main__opacity-6 {
  opacity: 0.6;
}

.dapp-core-component__main__qr-code-svg-container {
  width: 15rem;
  height: 15rem;
}

.dapp-core-component__main__pair-list {
  max-height: 20rem;
  overflow-y: auto;
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__pair-image {
  width: 4rem;
  overflow: hidden;
  height: auto;
  min-width: 2rem;
}

.dapp-core-component__main__pair-remove {
  padding: 0 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

.dapp-core-component__main__dapp-icon {
  padding: 5px;
  border-radius: 50%;
  background-color: #fafafa;
  width: 4.6428571429rem;
  height: 4.6428571429rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}
.dapp-core-component__main__dapp-icon.dapp-core-component__main__icon-medium {
  width: 80px;
  height: 80px;
}

.dapp-core-component__main__centering {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dapp-core-component__main__border-n {
  border: none !important;
}

.dapp-core-component__main__t-shadow {
  text-shadow: 0px 4px 7px rgba(0, 0, 0, 0.3);
}

.dapp-core-component__main__btn {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.04) !important;
}

.dapp-core-component__main__media {
  display: flex;
  align-items: flex-start;
}

.dapp-core-component__main__media-body {
  flex: 1;
}

.dapp-core-component__main__connect-btns .dapp-core-component__main__btn {
  display: flex;
  height: 48px;
  justify-content: center;
  align-items: center;
  color: #fff !important;
  background-color: #1392ff;
  padding: 15px 20px 15px 16px;
  border-radius: 6px;
  gap: 8px;
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__btn:hover, .dapp-core-component__main__connect-btns .dapp-core-component__main__btn:active {
  background-color: #1392ff;
  background-image: linear-gradient(rgba(0, 0, 0, 0.1) 0 0);
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__info {
  color: #1392ff;
  margin-left: 48px;
}

.dapp-core-component__main__btn {
  transition: none !important;
}

.dapp-core-component__main__trust-badge {
  width: 20px;
  height: 20px;
}

/* Navbar
  -------------------------------------------------- */
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name {
  display: flex;
  align-items: center;
  font-size: 1.7142857143rem;
  font-weight: 400;
  color: #9ba5b4;
  margin-left: 1.6428571429rem;
}
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name:before {
  content: "";
  display: block;
  width: 1px;
  height: 3rem;
  margin-right: 1.7142857143rem;
}

/* Dashboard
  -------------------------------------------------- */
.dapp-core-component__main__action-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  margin: 0 7px;
}
.dapp-core-component__main__action-btn button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #fff;
}
.dapp-core-component__main__action-btn button:hover {
  opacity: 0.9;
}

.dapp-core-component__main__transactions .dapp-core-component__main__transaction-icon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table td {
  white-space: nowrap;
  vertical-align: middle;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table tr:last-of-type td {
  border-bottom: 1px solid #dee2e6;
}

/* Loader, Transaction Success/Fail
  -------------------------------------------------- */
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div {
  position: absolute;
  top: 27px;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.1);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.dapp-core-component__main__page-state .dapp-core-component__main__bg-blue .dapp-core-component__main__lds-ellipsis div {
  background: rgba(255, 255, 255, 0.4);
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(1) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis1 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(2) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(3) {
  left: 26px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(4) {
  left: 45px;
  animation: dapp-core-component__main__lds-ellipsis3 0.6s infinite;
}
@keyframes dapp-core-component__main__lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(19px, 0);
  }
}

.dapp-core-component__main__btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
}
.dapp-core-component__main__btn .dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn .dapp-core-component__main__disabled {
  opacity: 0.65;
  cursor: default;
}
.dapp-core-component__main__btn:hover {
  color: #212529;
  text-decoration: none;
}
.dapp-core-component__main__btn:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn:disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn-primary {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary:hover {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-primary:focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary:hover {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-secondary:focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-warning {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning:hover {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning:focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-dark {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark:hover {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
}
.dapp-core-component__main__btn-dark:focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-link {
  font-weight: 400;
  color: #007bff;
  text-decoration: none;
}
.dapp-core-component__main__btn-link:hover {
  color: #0056b3;
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:focus, .dapp-core-component__main__btn-link.dapp-core-component__main__focus {
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:disabled, .dapp-core-component__main__btn-link.dapp-core-component__main__disabled {
  color: #6c757d;
  pointer-events: none;
}

.dapp-core-component__main__btn-light {
  color: #212529;
  background-color: #fafafa;
  border-color: #fafafa;
}
.dapp-core-component__main__btn-light .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}
.dapp-core-component__main__btn-light .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #f8f9fa;
  border-color: #f8f9fa;
}
.dapp-core-component__main__btn-light:hover {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
}
.dapp-core-component__main__btn-light:focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .dapp-core-component__main__badge {
    transition: none;
  }
}
a.dapp-core-component__main__badge:hover, a.dapp-core-component__main__badge:focus {
  text-decoration: none;
}

.dapp-core-component__main__badge:empty {
  display: none;
}

.dapp-core-component__main__btn .dapp-core-component__main__badge {
  position: relative;
  top: -1px;
}

.dapp-core-component__main__badge-pill {
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.dapp-core-component__main__badge-primary {
  color: #fff;
  background-color: #007bff;
}

a.dapp-core-component__main__badge-primary:hover, a.dapp-core-component__main__badge-primary:focus {
  color: #fff;
  background-color: #0062cc;
}

a.dapp-core-component__main__badge-primary:focus, a.dapp-core-component__main__badge-primary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.5);
}

.dapp-core-component__main__badge-secondary {
  color: #fff;
  background-color: #6c757d;
}

a.dapp-core-component__main__badge-secondary:hover, a.dapp-core-component__main__badge-secondary:focus {
  color: #fff;
  background-color: #545b62;
}

a.dapp-core-component__main__badge-secondary:focus, a.dapp-core-component__main__badge-secondary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108, 117, 125, 0.5);
}

.dapp-core-component__main__badge-success {
  color: #fff;
  background-color: #28a745;
}

a.dapp-core-component__main__badge-success:hover, a.dapp-core-component__main__badge-success:focus {
  color: #fff;
  background-color: #1e7e34;
}

a.dapp-core-component__main__badge-success:focus, a.dapp-core-component__main__badge-success.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
}

.dapp-core-component__main__badge-info {
  color: #fff;
  background-color: #17a2b8;
}

a.dapp-core-component__main__badge-info:hover, a.dapp-core-component__main__badge-info:focus {
  color: #fff;
  background-color: #117a8b;
}

a.dapp-core-component__main__badge-info:focus, a.dapp-core-component__main__badge-info.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(23, 162, 184, 0.5);
}

.dapp-core-component__main__badge-warning {
  color: #212529;
  background-color: #ffc107;
}

a.dapp-core-component__main__badge-warning:hover, a.dapp-core-component__main__badge-warning:focus {
  color: #212529;
  background-color: #d39e00;
}

a.dapp-core-component__main__badge-warning:focus, a.dapp-core-component__main__badge-warning.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 193, 7, 0.5);
}

.dapp-core-component__main__badge-danger {
  color: #fff;
  background-color: #dc3545;
}

a.dapp-core-component__main__badge-danger:hover, a.dapp-core-component__main__badge-danger:focus {
  color: #fff;
  background-color: #bd2130;
}

a.dapp-core-component__main__badge-danger:focus, a.dapp-core-component__main__badge-danger.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(220, 53, 69, 0.5);
}

.dapp-core-component__main__badge-light {
  color: #212529;
  background-color: #f8f9fa;
}

a.dapp-core-component__main__badge-light:hover, a.dapp-core-component__main__badge-light:focus {
  color: #212529;
  background-color: #dae0e5;
}

a.dapp-core-component__main__badge-light:focus, a.dapp-core-component__main__badge-light.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(248, 249, 250, 0.5);
}

.dapp-core-component__main__badge-dark {
  color: #fff;
  background-color: #343a40;
}

a.dapp-core-component__main__badge-dark:hover, a.dapp-core-component__main__badge-dark:focus {
  color: #fff;
  background-color: #1d2124;
}

a.dapp-core-component__main__badge-dark:focus, a.dapp-core-component__main__badge-dark.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}

.dapp-core-component__main__card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 5px;
}
.dapp-core-component__main__card > hr {
  margin-right: 0;
  margin-left: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group {
  border-top: inherit;
  border-bottom: inherit;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:first-child {
  border-top-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:last-child {
  border-bottom-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__card-header + .dapp-core-component__main__list-group,
.dapp-core-component__main__card > .dapp-core-component__main__list-group + .dapp-core-component__main__card-footer {
  border-top: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header {
  border: 0;
  padding: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header:first-child {
  border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item {
  border-bottom: 1px solid #dee2e6;
  padding: 1.1rem 1rem;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item h6,
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item .dapp-core-component__main__h6 {
  line-height: 2.5rem;
  letter-spacing: unset;
  margin: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item.dapp-core-component__main__compact {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fafafa;
}

.dapp-core-component__main__card-body {
  flex: 1 1 auto;
  min-height: 1px;
  padding: 1.25rem;
}

.dapp-core-component__main__card-title {
  margin-bottom: 0.75rem;
}

.dapp-core-component__main__card-subtitle {
  margin-top: -0.375rem;
  margin-bottom: 0;
}

.dapp-core-component__main__card-text:last-child {
  margin-bottom: 0;
}

.dapp-core-component__main__card-link:hover {
  text-decoration: none;
}
.dapp-core-component__main__card-link + .dapp-core-component__main__card-link {
  margin-left: 1.25rem;
}

.dapp-core-component__main__card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: #fff;
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__card-footer {
  padding: 0.75rem 1.25rem;
  background-color: #fff;
  border-top: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__text {
  color: #9dabbd;
}

.dapp-core-component__main__text-white {
  color: #fff;
}

.dapp-core-component__main__text-monospace {
  font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-size-xl .dapp-core-component__main__trim {
  max-width: 13rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim {
  max-width: 10rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-only-sm .dapp-core-component__main__trim {
  max-width: none;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-left {
  text-align: left;
}

.dapp-core-component__main__text-right {
  text-align: right;
}

.dapp-core-component__main__text-center {
  text-align: center;
}

.dapp-core-component__main__text-primary {
  color: #1b46c2;
}

.dapp-core-component__main__text-secondary {
  color: #6c757d;
}

.dapp-core-component__main__text-primary-highlight {
  color: #1b46c2;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-success {
  color: #28a745;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-body {
  color: #212529;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-black-50 {
  color: rgba(0, 0, 0, 0.5);
}

.dapp-core-component__main__text-white-50 {
  color: rgba(255, 255, 255, 0.5);
}

@media (min-width: 992px) {
  .dapp-core-component__main__text-lg-right {
    text-align: right;
  }
}

.dapp-core-component__main__text-hide {
  font: 0/0 a;
  color: rgba(0, 0, 0, 0);
  text-shadow: none;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
}

.dapp-core-component__main__text-decoration-none {
  text-decoration: none;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__text-reset {
  color: inherit;
}

.dapp-core-component__main__text-lowercase {
  text-transform: lowercase;
}

.dapp-core-component__main__text-uppercase {
  text-transform: uppercase;
}

.dapp-core-component__main__text-capitalize {
  text-transform: capitalize;
}

.dapp-core-component__main__text-danger {
  color: #d9534f;
}

.dapp-core-component__main__text-warning {
  color: #f0ad4e;
}

.dapp-core-component__main__trim-fs-sm {
  color: inherit;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__m-0 {
  margin: 0;
}

.dapp-core-component__main__mt-0 {
  margin-top: 0;
}

.dapp-core-component__main__my-0 {
  margin-top: 0;
  margin-bottom: 0;
}

.dapp-core-component__main__mr-0 {
  margin-right: 0;
}

.dapp-core-component__main__mx-0 {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__mb-0 {
  margin-bottom: 0;
}

.dapp-core-component__main__my-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.dapp-core-component__main__ml-0 {
  margin-left: 0;
}

.dapp-core-component__main__mx-0 {
  margin-left: 0;
  margin-right: 0;
}

.dapp-core-component__main__m-1 {
  margin: 0.25rem;
}

.dapp-core-component__main__mt-1 {
  margin-top: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__mr-1 {
  margin-right: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.dapp-core-component__main__mb-1 {
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-bottom: 0.25rem;
  margin-top: 0.25rem;
}

.dapp-core-component__main__ml-1 {
  margin-left: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-right: 0.25rem;
  margin-left: 0.25rem;
}

.dapp-core-component__main__m-2 {
  margin: 0.5rem;
}

.dapp-core-component__main__mt-2 {
  margin-top: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-bottom: 0.5rem;
  margin-top: 0.5rem;
}

.dapp-core-component__main__mr-2 {
  margin-right: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__mb-2 {
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__ml-2 {
  margin-left: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__m-3 {
  margin: 1rem;
}

.dapp-core-component__main__mt-3,
.dapp-core-component__main__my-3 {
  margin-top: 1rem;
}

.dapp-core-component__main__mr-3 {
  margin-right: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__mb-3 {
  margin-bottom: 1rem;
}

.dapp-core-component__main__my-3 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.dapp-core-component__main__ml-3 {
  margin-left: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__m-4 {
  margin: 1.5rem;
}

.dapp-core-component__main__mt-4 {
  margin-top: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__mr-4 {
  margin-right: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-right: 1.5rem;
  margin-left: 1.5rem;
}

.dapp-core-component__main__mb-4 {
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.dapp-core-component__main__ml-4 {
  margin-left: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}

.dapp-core-component__main__m-5 {
  margin: 3rem;
}

.dapp-core-component__main__mt-5 {
  margin-top: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__mr-5 {
  margin-right: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-right: 3rem;
  margin-left: 3rem;
}

.dapp-core-component__main__mb-5 {
  margin-bottom: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__ml-5 {
  margin-left: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-left: 3rem;
  margin-right: 3rem;
}

.dapp-core-component__main__p-0 {
  padding: 0;
}

.dapp-core-component__main__pt-0 {
  padding-top: 0;
}

.dapp-core-component__main__py-0 {
  padding-top: 0;
  padding-bottom: 0;
}

.dapp-core-component__main__pr-0 {
  padding-right: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__pb-0 {
  padding-bottom: 0;
}

.dapp-core-component__main__py-0 {
  padding-bottom: 0;
  padding-top: 0;
}

.dapp-core-component__main__pl-0 {
  padding-left: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__p-1 {
  padding: 0.25rem;
}

.dapp-core-component__main__pt-1 {
  padding-top: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pr-1 {
  padding-right: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-right: 0.25rem;
  padding-left: 0.25rem;
}

.dapp-core-component__main__pb-1 {
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pl-1 {
  padding-left: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

.dapp-core-component__main__p-2 {
  padding: 0.5rem;
}

.dapp-core-component__main__pt-2 {
  padding-top: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.dapp-core-component__main__pr-2 {
  padding-right: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__pb-2 {
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__pl-2 {
  padding-left: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__p-3 {
  padding: 1rem;
}

.dapp-core-component__main__pt-3 {
  padding-top: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.dapp-core-component__main__pr-3 {
  padding-right: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__pb-3 {
  padding-bottom: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.dapp-core-component__main__pl-3 {
  padding-left: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__p-4 {
  padding: 1.5rem;
}

.dapp-core-component__main__pt-4 {
  padding-top: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pr-4 {
  padding-right: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.dapp-core-component__main__pb-4 {
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pl-4 {
  padding-left: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
}

.dapp-core-component__main__p-5 {
  padding: 3rem;
}

.dapp-core-component__main__pt-5 {
  padding-top: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-bottom: 3rem;
  padding-top: 3rem;
}

.dapp-core-component__main__pr-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__pb-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.dapp-core-component__main__pl-5 {
  padding-left: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__m-n1 {
  margin: -0.25rem;
}

.dapp-core-component__main__mt-n1 {
  margin-top: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__mr-n1 {
  margin-right: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__mb-n1 {
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-bottom: -0.25rem;
  margin-top: -0.25rem;
}

.dapp-core-component__main__ml-n1 {
  margin-left: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__m-n2 {
  margin: -0.5rem;
}

.dapp-core-component__main__mt-n2 {
  margin-top: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-top: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mr-n2 {
  margin-right: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mb-n2 {
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-bottom: -0.5rem;
  margin-top: -0.5rem;
}

.dapp-core-component__main__ml-n2 {
  margin-left: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.dapp-core-component__main__m-n3 {
  margin: -1rem;
}

.dapp-core-component__main__mt-n3 {
  margin-top: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-top: -1rem;
  margin-bottom: -1rem;
}

.dapp-core-component__main__mr-n3 {
  margin-right: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-right: -1rem;
  margin-left: -1rem;
}

.dapp-core-component__main__mb-n3 {
  margin-bottom: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-bottom: -1rem;
  margin-top: -1rem;
}

.dapp-core-component__main__ml-n3 {
  margin-left: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-left: -1rem;
  margin-right: -1rem;
}

.dapp-core-component__main__m-n4 {
  margin: -1.5rem;
}

.dapp-core-component__main__mt-n4 {
  margin-top: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-top: -1.5rem;
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__mr-n4 {
  margin-right: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__mb-n4 {
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-bottom: -1.5rem;
  margin-top: -1.5rem;
}

.dapp-core-component__main__ml-n4 {
  margin-left: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__m-n5 {
  margin: -3rem;
}

.dapp-core-component__main__mt-n5 {
  margin-top: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-bottom: -3rem;
  margin-top: -3rem;
}

.dapp-core-component__main__mr-n5 {
  margin-right: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-left: -3rem;
  margin-right: -3rem;
}

.dapp-core-component__main__mb-n5 {
  margin-bottom: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-top: -3rem;
  margin-bottom: -3rem;
}

.dapp-core-component__main__ml-n5 {
  margin-left: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-right: -3rem;
  margin-left: -3rem;
}

.dapp-core-component__main__m-auto {
  margin: auto;
}

.dapp-core-component__main__mt-auto {
  margin-top: auto;
}

.dapp-core-component__main__my-auto {
  margin-bottom: auto;
  margin-top: auto;
}

.dapp-core-component__main__mr-auto {
  margin-right: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__mb-auto {
  margin-bottom: auto;
}

.dapp-core-component__main__my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.dapp-core-component__main__ml-auto {
  margin-left: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-1 {
    margin-right: 0.25rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-4 {
    margin-right: 1.5rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mx-lg-4 {
    margin-left: 1.5rem;
    margin-right: 1.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__mb-xl-2 {
    margin-bottom: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__pr-sm-2 {
    padding-right: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__px-sm-2 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__pr-xl-0,
.dapp-core-component__main__px-xl-0 {
    padding-right: 0;
  }
}

.dapp-core-component__main__col {
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

.dapp-core-component__main__col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__col-xl,
.dapp-core-component__main__col-xl-auto,
.dapp-core-component__main__col-xl-12,
.dapp-core-component__main__col-xl-11,
.dapp-core-component__main__col-xl-10,
.dapp-core-component__main__col-xl-9,
.dapp-core-component__main__col-xl-8,
.dapp-core-component__main__col-xl-7,
.dapp-core-component__main__col-xl-6,
.dapp-core-component__main__col-xl-5,
.dapp-core-component__main__col-xl-4,
.dapp-core-component__main__col-xl-3,
.dapp-core-component__main__col-xl-2,
.dapp-core-component__main__col-xl-1,
.dapp-core-component__main__col-lg,
.dapp-core-component__main__col-lg-auto,
.dapp-core-component__main__col-lg-12,
.dapp-core-component__main__col-lg-11,
.dapp-core-component__main__col-lg-10,
.dapp-core-component__main__col-lg-9,
.dapp-core-component__main__col-lg-8,
.dapp-core-component__main__col-lg-7,
.dapp-core-component__main__col-lg-6,
.dapp-core-component__main__col-lg-5,
.dapp-core-component__main__col-lg-4,
.dapp-core-component__main__col-lg-3,
.dapp-core-component__main__col-lg-2,
.dapp-core-component__main__col-lg-1,
.dapp-core-component__main__col-md,
.dapp-core-component__main__col-md-auto,
.dapp-core-component__main__col-md-12,
.dapp-core-component__main__col-md-11,
.dapp-core-component__main__col-md-10,
.dapp-core-component__main__col-md-9,
.dapp-core-component__main__col-md-8,
.dapp-core-component__main__col-md-7,
.dapp-core-component__main__col-md-6,
.dapp-core-component__main__col-md-5,
.dapp-core-component__main__col-md-4,
.dapp-core-component__main__col-md-3,
.dapp-core-component__main__col-md-2,
.dapp-core-component__main__col-md-1,
.dapp-core-component__main__col-sm,
.dapp-core-component__main__col-sm-auto,
.dapp-core-component__main__col-sm-12,
.dapp-core-component__main__col-sm-11,
.dapp-core-component__main__col-sm-10,
.dapp-core-component__main__col-sm-9,
.dapp-core-component__main__col-sm-8,
.dapp-core-component__main__col-sm-7,
.dapp-core-component__main__col-sm-6,
.dapp-core-component__main__col-sm-5,
.dapp-core-component__main__col-sm-4,
.dapp-core-component__main__col-sm-3,
.dapp-core-component__main__col-sm-2,
.dapp-core-component__main__col-sm-1,
.dapp-core-component__main__col,
.dapp-core-component__main__col-auto,
.dapp-core-component__main__col-12,
.dapp-core-component__main__col-11,
.dapp-core-component__main__col-10,
.dapp-core-component__main__col-9,
.dapp-core-component__main__col-8,
.dapp-core-component__main__col-7,
.dapp-core-component__main__col-6,
.dapp-core-component__main__col-5,
.dapp-core-component__main__col-4,
.dapp-core-component__main__col-3,
.dapp-core-component__main__col-2,
.dapp-core-component__main__col-1 {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
}

.dapp-core-component__main__col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

@media (min-width: 576px) {
  .dapp-core-component__main__col-sm-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-sm-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-sm-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-sm-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-sm-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-sm-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-sm-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-sm-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-sm-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-sm-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-sm-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-sm-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 768px) {
  .dapp-core-component__main__col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__col-xl-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-xl-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-xl-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-xl-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-xl-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-xl-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-xl-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-xl-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-xl-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-xl-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-xl-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-xl-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
.dapp-core-component__main__flex-row {
  flex-direction: row;
}

.dapp-core-component__main__flex-column {
  flex-direction: column;
}

.dapp-core-component__main__flex-row-reverse {
  flex-direction: row-reverse;
}

.dapp-core-component__main__flex-column-reverse {
  flex-direction: column-reverse;
}

.dapp-core-component__main__flex-wrap {
  flex-wrap: wrap;
}

.dapp-core-component__main__flex-nowrap {
  flex-wrap: nowrap;
}

.dapp-core-component__main__flex-wrap-reverse {
  flex-wrap: wrap-reverse;
}

.dapp-core-component__main__flex-fill {
  flex: 1 1 auto;
}

.dapp-core-component__main__flex-grow-0 {
  flex-grow: 0;
}

.dapp-core-component__main__flex-grow-1 {
  flex-grow: 1;
}

.dapp-core-component__main__flex-shrink-0 {
  flex-shrink: 0;
}

.dapp-core-component__main__flex-shrink-1 {
  flex-shrink: 1;
}

.dapp-core-component__main__justify-content-start {
  justify-content: flex-start;
}

.dapp-core-component__main__justify-content-end {
  justify-content: flex-end;
}

.dapp-core-component__main__justify-content-center {
  justify-content: center;
}

.dapp-core-component__main__justify-content-between {
  justify-content: space-between;
}

.dapp-core-component__main__justify-content-around {
  justify-content: space-around;
}

.dapp-core-component__main__align-items-start {
  align-items: flex-start;
}

.dapp-core-component__main__align-items-end {
  align-items: flex-end;
}

.dapp-core-component__main__align-items-center {
  align-items: center;
}

.dapp-core-component__main__align-items-baseline {
  align-items: baseline;
}

.dapp-core-component__main__align-items-stretch {
  align-items: stretch;
}

.dapp-core-component__main__align-content-start {
  align-content: flex-start;
}

.dapp-core-component__main__align-content-end {
  align-content: flex-end;
}

.dapp-core-component__main__align-content-center {
  align-content: center;
}

.dapp-core-component__main__align-content-between {
  align-content: space-between;
}

.dapp-core-component__main__align-content-around {
  align-content: space-around;
}

.dapp-core-component__main__align-content-stretch {
  align-content: stretch;
}

.dapp-core-component__main__align-self-auto {
  align-self: auto;
}

.dapp-core-component__main__align-self-start {
  align-self: flex-start;
}

.dapp-core-component__main__align-self-end {
  align-self: flex-end;
}

.dapp-core-component__main__align-self-center {
  align-self: center;
}

.dapp-core-component__main__align-self-baseline {
  align-self: baseline;
}

.dapp-core-component__main__align-self-stretch {
  align-self: stretch;
}

@media (min-width: 576px) {
  .dapp-core-component__main__flex-sm-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-wrap {
    flex-wrap: wrap;
  }
}

.dapp-core-component__main__form-check {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: auto;
  padding-left: 0;
}

.dapp-core-component__main__form-check-label {
  margin-bottom: 0;
}

.dapp-core-component__main__form-check-input {
  position: absolute;
  margin-top: 0.3rem;
  margin-left: -1.25rem;
}
.dapp-core-component__main__form-check-input[disabled] ~ .dapp-core-component__main__form-check-label, .dapp-core-component__main__form-check-input:disabled ~ .dapp-core-component__main__form-check-label {
  color: #6c757d;
}

.dapp-core-component__main__form-group {
  display: flex;
  flex: 0 0 auto;
  flex-flow: row wrap;
  align-items: center;
  margin-bottom: 0;
}

.dapp-core-component__main__form-control {
  display: block;
  width: 100%;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.dapp-core-component__main__form-label {
  padding-top: calc(0.375rem + 1px);
  padding-bottom: calc(0.375rem + 1px);
  margin-bottom: 0;
  font-size: inherit;
  line-height: 1.5;
}

input,
button,
select,
optgroup,
textarea {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.dapp-core-component__main__form-group {
  margin-bottom: 1rem;
}

.dapp-core-component__main__form-text {
  display: block;
  margin-top: 0.25rem;
  color: inherit;
}

.dapp-core-component__main__border {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__border-top {
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__border-right {
  border-right: 1px solid #dee2e6;
}

.dapp-core-component__main__border-bottom {
  border-bottom: 1px solid #dee2e6;
}

.dapp-core-component__main__border-left {
  border-left: 1px solid #dee2e6;
}

.dapp-core-component__main__border-0 {
  border: 0;
}

.dapp-core-component__main__border-top-0 {
  border-top: 0;
}

.dapp-core-component__main__border-right-0 {
  border-right: 0;
}

.dapp-core-component__main__border-bottom-0 {
  border-bottom: 0;
}

.dapp-core-component__main__border-left-0 {
  border-left: 0;
}

.dapp-core-component__main__rounded-sm {
  border-radius: 0.2rem;
}

.dapp-core-component__main__rounded {
  border-radius: 0.25rem;
}

.dapp-core-component__main__rounded-top {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-right {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-bottom {
  border-bottom-right-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-left {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-lg {
  border-radius: 0.3rem;
}

.dapp-core-component__main__rounded-circle {
  border-radius: 50%;
}

.dapp-core-component__main__rounded-pill {
  border-radius: 50rem;
}

.dapp-core-component__main__rounded-0 {
  border-radius: 0;
}

.dapp-core-component__main__h1,
.dapp-core-component__main__h2,
.dapp-core-component__main__h3,
.dapp-core-component__main__h4,
.dapp-core-component__main__h5,
.dapp-core-component__main__h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2;
}

.dapp-core-component__main__h1 {
  font-size: 2.5rem;
}

.dapp-core-component__main__h2 {
  font-size: 2rem;
}

.dapp-core-component__main__h3 {
  font-size: 1.75rem;
}

.dapp-core-component__main__h4 {
  font-size: 1.5rem;
}

.dapp-core-component__main__h5 {
  font-size: 1.25rem;
}

.dapp-core-component__main__h6 {
  font-size: 1rem;
}

.dapp-core-component__main__lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

.dapp-core-component__main__bg-warning,
.dapp-core-component__main__warning {
  background-color: #f0ad4e;
}

.dapp-core-component__main__bg-danger,
.dapp-core-component__main__danger {
  background-color: #d9534f;
}

.dapp-core-component__main__bg-success,
.dapp-core-component__main__success {
  background-color: #5cb85c;
}

.dapp-core-component__main__position-relative {
  position: relative;
}

.dapp-core-component__main__position-absolute {
  position: absolute;
}

.dapp-core-component__main__w-25 {
  width: 25%;
}

.dapp-core-component__main__w-50 {
  width: 50%;
}

.dapp-core-component__main__w-75 {
  width: 75%;
}

.dapp-core-component__main__w-100 {
  width: 100%;
}

.dapp-core-component__main__w-auto {
  width: auto;
}

.dapp-core-component__main__h-25 {
  height: 25%;
}

.dapp-core-component__main__h-50 {
  height: 50%;
}

.dapp-core-component__main__h-75 {
  height: 75%;
}

.dapp-core-component__main__h-100 {
  height: 100%;
}

.dapp-core-component__main__h-auto {
  height: auto;
}

.dapp-core-component__main__mw-100 {
  max-width: 100%;
}

.dapp-core-component__main__mh-100 {
  max-height: 100%;
}

.dapp-core-component__main__min-vw-100 {
  min-width: 100vw;
}

.dapp-core-component__main__min-vh-100 {
  min-height: 100vh;
}

.dapp-core-component__main__vw-100 {
  width: 100vw;
}

.dapp-core-component__main__vh-100 {
  height: 100vh;
}

.dapp-core-component__main__locked-icon {
  color: inherit;
}

.dapp-core-component__main__shadow-sm {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
}

.dapp-core-component__main__shadow {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.dapp-core-component__main__shadow-lg {
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.175);
}

.dapp-core-component__main__shadow-none {
  box-shadow: none;
}

.dapp-core-component__main__d-none {
  display: none;
}

.dapp-core-component__main__d-inline {
  display: inline;
}

.dapp-core-component__main__d-inline-block {
  display: inline-block;
}

.dapp-core-component__main__d-block {
  display: block;
}

.dapp-core-component__main__d-table {
  display: table;
}

.dapp-core-component__main__d-table-row {
  display: table-row;
}

.dapp-core-component__main__d-table-cell {
  display: table-cell;
}

.dapp-core-component__main__d-flex {
  display: -ms-flexbox;
  display: flex;
}

.dapp-core-component__main__d-inline-flex {
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media (min-width: 576px) {
  .dapp-core-component__main__d-sm-flex {
    display: flex;
  }
}

.dapp-core-component__main__table {
  width: 100%;
  margin-bottom: 1rem;
  color: #212529;
}
.dapp-core-component__main__table th,
.dapp-core-component__main__table td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #dee2e6;
}
.dapp-core-component__main__table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #dee2e6;
}
.dapp-core-component__main__table tbody + tbody {
  border-top: 2px solid #dee2e6;
}

.dapp-core-component__main__table-sm th,
.dapp-core-component__main__table-sm td {
  padding: 0.3rem;
}

.dapp-core-component__main__table-bordered {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered th,
.dapp-core-component__main__table-bordered td {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered thead th,
.dapp-core-component__main__table-bordered thead td {
  border-bottom-width: 2px;
}

.dapp-core-component__main__table-borderless th,
.dapp-core-component__main__table-borderless td,
.dapp-core-component__main__table-borderless thead th,
.dapp-core-component__main__table-borderless tbody + tbody {
  border: 0;
}

.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(0, 0, 0, 0.05);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-dark th {
  color: #fff;
  background-color: #343a40;
  border-color: #454d55;
}
.dapp-core-component__main__table .dapp-core-component__main__thead-light th {
  color: #495057;
  background-color: #e9ecef;
  border-color: #dee2e6;
}

.dapp-core-component__main__table-dark {
  color: #fff;
  background-color: #343a40;
}
.dapp-core-component__main__table-dark th,
.dapp-core-component__main__table-dark td,
.dapp-core-component__main__table-dark thead th {
  border-color: #454d55;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-bordered {
  border: 0;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(255, 255, 255, 0.05);
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-hover tbody tr {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.075);
}

.dapp-core-component__main__table-responsive {
  display: block;
  width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}
.dapp-core-component__main__table-responsive > .dapp-core-component__main__table-bordered {
  border: 0;
}

@media (max-width: 767.98px) {
  .dapp-core-component__main__transactions.dapp-core-component__main__table .dapp-core-component__main__transaction-hash .dapp-core-component__main__trim {
    max-width: 8rem;
  }
}
.dapp-core-component__main__font-weight-light {
  font-weight: 300 !important;
}

.dapp-core-component__main__font-weight-lighter {
  font-weight: lighter !important;
}

.dapp-core-component__main__font-weight-normal {
  font-weight: 400 !important;
}

.dapp-core-component__main__font-weight-medium {
  font-weight: 500 !important;
}

.dapp-core-component__main__font-weight-bold {
  font-weight: 700 !important;
}

.dapp-core-component__main__font-weight-bolder {
  font-weight: bolder !important;
}

.dapp-core-component__main__font-italic {
  font-style: italic !important;
}

.dapp-core-component__main__icon-state {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
}
.dapp-core-component__main__icon-state.dapp-core-component__main__half {
  width: 3rem;
  height: 3rem;
}

.dapp-core-component__main__side-icon {
  width: 1.5rem;
  height: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  color: transparent;
  border-radius: 50%;
  overflow: hidden;
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__side-icon svg {
  color: #cacbcc;
}

div.dapp-core-component__main__side-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
div.dapp-core-component__main__side-icon img {
  transform: scale(0.75);
}

.dapp-core-component__main__toast-messages {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast {
  display: block;
  flex-basis: 4rem;
  width: 100%;
  max-width: 28.875rem;
  transition: transform 0.3s ease-in-out;
  transform: translateX(120%);
  background-color: #fff;
  border-radius: 0.25rem;
  border-color: #dee2e6;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__clickable {
  cursor: pointer;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible {
  margin-bottom: 0.3rem;
  transform: translateX(0);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress {
  background-color: rgba(0, 0, 0, 0);
  height: inherit;
  font-size: inherit;
  line-height: inherit;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar {
  background-color: rgba(108, 117, 125, 0.2);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar .dapp-core-component__main__content-height {
  z-index: -1;
  visibility: hidden;
  width: 0 !important;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast .dapp-core-component__main__tx-description {
  margin-bottom: 0.2rem;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close {
  opacity: 1;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close svg {
  color: #6c757d;
}

.dapp-core-component__main__data-field .dapp-core-component__main__textarea {
  -moz-appearance: textfield-multiline;
  -webkit-appearance: textarea;
  border: 1px solid #000;
  background-color: #fff;
  border-radius: 5px;
  font: medium -moz-fixed;
  font-size: 0.8125rem;
  min-height: 5rem;
  overflow: auto;
  padding: 0.375rem 0.75rem;
  resize: both;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea:hover {
  border: 1px solid #000;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea .dapp-core-component__main__highlighted {
  font-size: 1rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea.dapp-core-component__main__sc-call {
  min-height: 3rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__form-control:disabled,
.dapp-core-component__main__data-field .dapp-core-component__main__form-control[readonly] {
  background-color: #fafafa;
}
.dapp-core-component__main__data-field .dapp-core-component__main__small-font {
  font-size: 0.8125rem;
}

.dapp-core-component__main__progress-steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps hr {
  background-color: #6c757d;
  border-color: #dee2e6;
  z-index: 2;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot {
  height: 0.625rem;
  width: 0.625rem;
  border-radius: 50%;
  margin: 0 1rem;
  z-index: 5;
  background-color: #6c757d;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:first-of-type {
  margin-left: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:last-of-type {
  margin-right: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

a.dapp-core-component__main__side-link::after {
  right: 0;
  left: unset;
  width: calc(100% - 1.85rem);
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(_s));I();k();G();t();var ls=S(require("bignumber.js"));I();t();var gs=require("@elrondnetwork/erdjs"),fs=S(require("bignumber.js"));I();A();Fe();t();t();t();t();t();var Va=()=>w(pe);t();t();t();t();var Ha=require("react");t();t();var Ts=require("react"),xs=require("@elrondnetwork/erdjs-extension-provider");De();M();k();A();Re();t();var ys=require("react");De();k();t();var za=require("react"),bs=require("@elrondnetwork/erdjs-hw-provider");De();M();k();A();Re();t();var Kn=require("react"),ws=require("@elrondnetwork/erdjs-wallet-connect-provider");M();k();A();Re();t();var qa=require("react"),vs=require("@elrondnetwork/erdjs-wallet-connect-provider");M();k();A();Re();t();t();var Ka=require("react");t();var Za=require("react");t();var ke=require("react"),Xa=require("socket.io-client");k();t();var ie={current:null,status:"not_initialized"};var ks=3e3,Is=3,As=500,Es=1e3;function Ya(){let e=(0,ke.useRef)(null),{address:o}=Va(),n=F(),{network:r}=Bn(),a=s=>{e.current&&clearTimeout(e.current),e.current=setTimeout(()=>{n(rr(s))},Es)},m=(0,ke.useCallback)(La(()=>x(this,null,function*(){ie.status="pending";let s=yield Gn(r.apiAddress);if(s==null){console.warn("Can not get websocket url");return}ie.current=(0,Xa.io)(s,{forceNew:!0,reconnectionAttempts:Is,timeout:ks,query:{address:o}}),ie.status="completed",ie.current.onAny(l=>{a(l)})}),{retries:2,delay:As}),[o]);(0,ke.useEffect)(()=>{o&&ie.status==="not_initialized"&&!ie.current&&m()},[o,ie.current]),(0,ke.useEffect)(()=>()=>{var s;(s=ie.current)==null||s.close(),e.current&&clearTimeout(e.current)},[])}function Ja(e){Ya();let o=w(Cn);(0,Za.useEffect)(()=>{let n=o==null?void 0:o.message;n&&e(n)},[e,o])}t();t();t();var Qa=e=>e.dappModal,Ls=g(Qa,e=>e.visible),Uo=g(Qa,e=>e.modal),vI=g(Uo,e=>e.showHeader),kI=g(Uo,e=>e.showFooter),II=g(Uo,e=>e.headerText),AI=g(Uo,e=>e.footerText);he();var{logoutEvent:Ds}=W,ei="storage",ni=()=>{let{address:e}=Vn();return(0,oi.useEffect)(()=>{let o=n=>{if(!(n.key!==Ds||!n.newValue))try{let{data:r}=JSON.parse(n.newValue);r===e&&Fo()}catch(r){return}};return window.addEventListener(ei,o),()=>{window.removeEventListener(ei,o)}},[e]),null};t();t();var Wo=require("react"),ri=require("@elrondnetwork/erdjs"),ai=require("@elrondnetwork/erdjs/out/signature");k();A();Ne();var ti=e=>{let o=e.redirectRoute;o&&Pe(o)},ii=({sendSignedTransactionsAsync:e=Fa})=>{let o=w(pe),n=w(_e),r=(0,Wo.useRef)(!1),a=F(),m=()=>{a(Ze()),r.current=!1};function s(){return x(this,null,function*(){var d;let l=Object.keys(n);for(let u of l){let y=n==null?void 0:n[u],h=(d=y==null?void 0:y.customTransactionInformation)==null?void 0:d.signWithoutSending;if(!u||h){ti(y);continue}try{if(!(n[u].status==="signed"&&!r.current))continue;let{transactions:O}=n[u];if(!O)continue;r.current=!0;let j=O.map(de=>{let li=new ri.Address(de.sender),Xn=We(de),ui=new ai.Signature(de.signature);return Xn.applySignature(ui,li),Xn}),mi=yield e(j),_i="pending",di=O.map(de=>mi.includes(de.hash)?v(f({},de),{status:_i}):de);a(fr({sessionId:u,submittedMessage:"submitted"})),a(ye({sessionId:u,status:"sent",transactions:di})),m(),va(o.nonce+O.length),ti(y),history.pushState({},document.title,"?")}catch(L){console.error("Unable to send transactions",L),a(ye({sessionId:u,status:"fail",errorMessage:L.message})),m()}finally{r.current=!1}}})}return(0,Wo.useEffect)(()=>{s()},[n,o]),null};t();t();t();var ci=require("react");function pi(e){var a;let o=$a(),n=(a=e==null?void 0:e.getTransactionsByHash)!=null?a:Oo,r=()=>{o({shouldRefreshBalance:!0,getTransactionsByHash:n})};Ja(r),(0,ci.useEffect)(()=>{let m=setInterval(r,3e4);return()=>{clearInterval(m)}},[r])}function si({getTransactionsByHash:e}){return pi({getTransactionsByHash:e}),null}function Cs({customComponents:e}){var m,s;let o=e==null?void 0:e.transactionSender,n=e==null?void 0:e.transactionTracker,r=(m=o==null?void 0:o.component)!=null?m:ii,a=(s=n==null?void 0:n.component)!=null?s:si;return $e.default.createElement($e.default.Fragment,null,$e.default.createElement(r,f({},o==null?void 0:o.props)),$e.default.createElement(a,f({},n==null?void 0:n.props)),$e.default.createElement(ni,null))}0&&(module.exports={CustomComponents});
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
