// @elrondnetwork/dapp-core/__commonjs/wrappers/DappProvider/index.js
"use strict";var Ic=Object.create;var Do=Object.defineProperty,Ec=Object.defineProperties,Lc=Object.getOwnPropertyDescriptor,Cc=Object.getOwnPropertyDescriptors,Dc=Object.getOwnPropertyNames,Vo=Object.getOwnPropertySymbols,Pc=Object.getPrototypeOf,Cn=Object.prototype.hasOwnProperty,zt=Object.prototype.propertyIsEnumerable;var Ht=(e,o,n)=>o in e?Do(e,o,{enumerable:!0,configurable:!0,writable:!0,value:n}):e[o]=n,h=(e,o)=>{for(var n in o||(o={}))Cn.call(o,n)&&Ht(e,n,o[n]);if(Vo)for(var n of Vo(o))zt.call(o,n)&&Ht(e,n,o[n]);return e},A=(e,o)=>Ec(e,Cc(o));var jt=e=>typeof e=="symbol"?e:e+"",Ve=(e,o)=>{var n={};for(var r in e)Cn.call(e,r)&&o.indexOf(r)<0&&(n[r]=e[r]);if(e!=null&&Vo)for(var r of Vo(e))o.indexOf(r)<0&&zt.call(e,r)&&(n[r]=e[r]);return n};var w=(e,o)=>()=>(e&&(o=e(e=0)),o);var Ho=(e,o)=>()=>(o||e((o={exports:{}}).exports,o),o.exports),He=(e,o)=>{for(var n in o)Do(e,n,{get:o[n],enumerable:!0})},qt=(e,o,n,r)=>{if(o&&typeof o=="object"||typeof o=="function")for(let a of Dc(o))!Cn.call(e,a)&&a!==n&&Do(e,a,{get:()=>o[a],enumerable:!(r=Lc(o,a))||r.enumerable});return e};var S=(e,o,n)=>(n=e!=null?Ic(Pc(e)):{},qt(o||!e||!e.__esModule?Do(n,"default",{value:e,enumerable:!0}):n,e)),Po=e=>qt(Do({},"__esModule",{value:!0}),e);var u=(e,o,n)=>new Promise((r,a)=>{var m=l=>{try{_(n.next(l))}catch(g){a(g)}},s=l=>{try{_(n.throw(l))}catch(g){a(g)}},_=l=>l.done?r(l.value):Promise.resolve(l.value).then(m,s);_((n=n.apply(e,o)).next())});var Yt=Ho(zo=>{"use strict";t();zo.byteLength=Rc;zo.toByteArray=Fc;zo.fromByteArray=Uc;var le=[],re=[],Nc=typeof Uint8Array!="undefined"?Uint8Array:Array,Dn="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";for(ze=0,Kt=Dn.length;ze<Kt;++ze)le[ze]=Dn[ze],re[Dn.charCodeAt(ze)]=ze;var ze,Kt;re["-".charCodeAt(0)]=62;re["_".charCodeAt(0)]=63;function Xt(e){var o=e.length;if(o%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var n=e.indexOf("=");n===-1&&(n=o);var r=n===o?0:4-n%4;return[n,r]}function Rc(e){var o=Xt(e),n=o[0],r=o[1];return(n+r)*3/4-r}function Mc(e,o,n){return(o+n)*3/4-n}function Fc(e){var o,n=Xt(e),r=n[0],a=n[1],m=new Nc(Mc(e,r,a)),s=0,_=a>0?r-4:r,l;for(l=0;l<_;l+=4)o=re[e.charCodeAt(l)]<<18|re[e.charCodeAt(l+1)]<<12|re[e.charCodeAt(l+2)]<<6|re[e.charCodeAt(l+3)],m[s++]=o>>16&255,m[s++]=o>>8&255,m[s++]=o&255;return a===2&&(o=re[e.charCodeAt(l)]<<2|re[e.charCodeAt(l+1)]>>4,m[s++]=o&255),a===1&&(o=re[e.charCodeAt(l)]<<10|re[e.charCodeAt(l+1)]<<4|re[e.charCodeAt(l+2)]>>2,m[s++]=o>>8&255,m[s++]=o&255),m}function Oc(e){return le[e>>18&63]+le[e>>12&63]+le[e>>6&63]+le[e&63]}function Bc(e,o,n){for(var r,a=[],m=o;m<n;m+=3)r=(e[m]<<16&16711680)+(e[m+1]<<8&65280)+(e[m+2]&255),a.push(Oc(r));return a.join("")}function Uc(e){for(var o,n=e.length,r=n%3,a=[],m=16383,s=0,_=n-r;s<_;s+=m)a.push(Bc(e,s,s+m>_?_:s+m));return r===1?(o=e[n-1],a.push(le[o>>2]+le[o<<4&63]+"==")):r===2&&(o=(e[n-2]<<8)+e[n-1],a.push(le[o>>10]+le[o>>4&63]+le[o<<2&63]+"=")),a.join("")}});var Zt=Ho(Pn=>{t();Pn.read=function(e,o,n,r,a){var m,s,_=a*8-r-1,l=(1<<_)-1,g=l>>1,f=-7,T=n?a-1:0,I=n?-1:1,k=e[o+T];for(T+=I,m=k&(1<<-f)-1,k>>=-f,f+=_;f>0;m=m*256+e[o+T],T+=I,f-=8);for(s=m&(1<<-f)-1,m>>=-f,f+=r;f>0;s=s*256+e[o+T],T+=I,f-=8);if(m===0)m=1-g;else{if(m===l)return s?NaN:(k?-1:1)*(1/0);s=s+Math.pow(2,r),m=m-g}return(k?-1:1)*s*Math.pow(2,m-r)};Pn.write=function(e,o,n,r,a,m){var s,_,l,g=m*8-a-1,f=(1<<g)-1,T=f>>1,I=a===23?Math.pow(2,-24)-Math.pow(2,-77):0,k=r?0:m-1,L=r?1:-1,F=o<0||o===0&&1/o<0?1:0;for(o=Math.abs(o),isNaN(o)||o===1/0?(_=isNaN(o)?1:0,s=f):(s=Math.floor(Math.log(o)/Math.LN2),o*(l=Math.pow(2,-s))<1&&(s--,l*=2),s+T>=1?o+=I/l:o+=I*Math.pow(2,1-T),o*l>=2&&(s++,l/=2),s+T>=f?(_=0,s=f):s+T>=1?(_=(o*l-1)*Math.pow(2,a),s=s+T):(_=o*Math.pow(2,T-1)*Math.pow(2,a),s=0));a>=8;e[n+k]=_&255,k+=L,_/=256,a-=8);for(s=s<<a|_,g+=a;g>0;e[n+k]=s&255,k+=L,s/=256,g-=8);e[n+k-L]|=F*128}});var mr=Ho(so=>{"use strict";t();var Nn=Yt(),po=Zt(),Jt=typeof Symbol=="function"&&typeof Symbol.for=="function"?Symbol.for("nodejs.util.inspect.custom"):null;so.Buffer=d;so.SlowBuffer=zc;so.INSPECT_MAX_BYTES=50;var jo=2147483647;so.kMaxLength=jo;d.TYPED_ARRAY_SUPPORT=Wc();!d.TYPED_ARRAY_SUPPORT&&typeof console!="undefined"&&typeof console.error=="function"&&console.error("This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.");function Wc(){try{var e=new Uint8Array(1),o={foo:function(){return 42}};return Object.setPrototypeOf(o,Uint8Array.prototype),Object.setPrototypeOf(e,o),e.foo()===42}catch(n){return!1}}Object.defineProperty(d.prototype,"parent",{enumerable:!0,get:function(){if(!!d.isBuffer(this))return this.buffer}});Object.defineProperty(d.prototype,"offset",{enumerable:!0,get:function(){if(!!d.isBuffer(this))return this.byteOffset}});function be(e){if(e>jo)throw new RangeError('The value "'+e+'" is invalid for option "size"');var o=new Uint8Array(e);return Object.setPrototypeOf(o,d.prototype),o}function d(e,o,n){if(typeof e=="number"){if(typeof o=="string")throw new TypeError('The "string" argument must be of type string. Received type number');return On(e)}return or(e,o,n)}d.poolSize=8192;function or(e,o,n){if(typeof e=="string")return $c(e,o);if(ArrayBuffer.isView(e))return Vc(e);if(e==null)throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e);if(ue(e,ArrayBuffer)||e&&ue(e.buffer,ArrayBuffer)||typeof SharedArrayBuffer!="undefined"&&(ue(e,SharedArrayBuffer)||e&&ue(e.buffer,SharedArrayBuffer)))return Mn(e,o,n);if(typeof e=="number")throw new TypeError('The "value" argument must not be of type number. Received type number');var r=e.valueOf&&e.valueOf();if(r!=null&&r!==e)return d.from(r,o,n);var a=Hc(e);if(a)return a;if(typeof Symbol!="undefined"&&Symbol.toPrimitive!=null&&typeof e[Symbol.toPrimitive]=="function")return d.from(e[Symbol.toPrimitive]("string"),o,n);throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e)}d.from=function(e,o,n){return or(e,o,n)};Object.setPrototypeOf(d.prototype,Uint8Array.prototype);Object.setPrototypeOf(d,Uint8Array);function nr(e){if(typeof e!="number")throw new TypeError('"size" argument must be of type number');if(e<0)throw new RangeError('The value "'+e+'" is invalid for option "size"')}function Gc(e,o,n){return nr(e),e<=0?be(e):o!==void 0?typeof n=="string"?be(e).fill(o,n):be(e).fill(o):be(e)}d.alloc=function(e,o,n){return Gc(e,o,n)};function On(e){return nr(e),be(e<0?0:Bn(e)|0)}d.allocUnsafe=function(e){return On(e)};d.allocUnsafeSlow=function(e){return On(e)};function $c(e,o){if((typeof o!="string"||o==="")&&(o="utf8"),!d.isEncoding(o))throw new TypeError("Unknown encoding: "+o);var n=tr(e,o)|0,r=be(n),a=r.write(e,o);return a!==n&&(r=r.slice(0,a)),r}function Rn(e){for(var o=e.length<0?0:Bn(e.length)|0,n=be(o),r=0;r<o;r+=1)n[r]=e[r]&255;return n}function Vc(e){if(ue(e,Uint8Array)){var o=new Uint8Array(e);return Mn(o.buffer,o.byteOffset,o.byteLength)}return Rn(e)}function Mn(e,o,n){if(o<0||e.byteLength<o)throw new RangeError('"offset" is outside of buffer bounds');if(e.byteLength<o+(n||0))throw new RangeError('"length" is outside of buffer bounds');var r;return o===void 0&&n===void 0?r=new Uint8Array(e):n===void 0?r=new Uint8Array(e,o):r=new Uint8Array(e,o,n),Object.setPrototypeOf(r,d.prototype),r}function Hc(e){if(d.isBuffer(e)){var o=Bn(e.length)|0,n=be(o);return n.length===0||e.copy(n,0,0,o),n}if(e.length!==void 0)return typeof e.length!="number"||Un(e.length)?be(0):Rn(e);if(e.type==="Buffer"&&Array.isArray(e.data))return Rn(e.data)}function Bn(e){if(e>=jo)throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x"+jo.toString(16)+" bytes");return e|0}function zc(e){return+e!=e&&(e=0),d.alloc(+e)}d.isBuffer=function(o){return o!=null&&o._isBuffer===!0&&o!==d.prototype};d.compare=function(o,n){if(ue(o,Uint8Array)&&(o=d.from(o,o.offset,o.byteLength)),ue(n,Uint8Array)&&(n=d.from(n,n.offset,n.byteLength)),!d.isBuffer(o)||!d.isBuffer(n))throw new TypeError('The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array');if(o===n)return 0;for(var r=o.length,a=n.length,m=0,s=Math.min(r,a);m<s;++m)if(o[m]!==n[m]){r=o[m],a=n[m];break}return r<a?-1:a<r?1:0};d.isEncoding=function(o){switch(String(o).toLowerCase()){case"hex":case"utf8":case"utf-8":case"ascii":case"latin1":case"binary":case"base64":case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return!0;default:return!1}};d.concat=function(o,n){if(!Array.isArray(o))throw new TypeError('"list" argument must be an Array of Buffers');if(o.length===0)return d.alloc(0);var r;if(n===void 0)for(n=0,r=0;r<o.length;++r)n+=o[r].length;var a=d.allocUnsafe(n),m=0;for(r=0;r<o.length;++r){var s=o[r];if(ue(s,Uint8Array))m+s.length>a.length?d.from(s).copy(a,m):Uint8Array.prototype.set.call(a,s,m);else if(d.isBuffer(s))s.copy(a,m);else throw new TypeError('"list" argument must be an Array of Buffers');m+=s.length}return a};function tr(e,o){if(d.isBuffer(e))return e.length;if(ArrayBuffer.isView(e)||ue(e,ArrayBuffer))return e.byteLength;if(typeof e!="string")throw new TypeError('The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type '+typeof e);var n=e.length,r=arguments.length>2&&arguments[2]===!0;if(!r&&n===0)return 0;for(var a=!1;;)switch(o){case"ascii":case"latin1":case"binary":return n;case"utf8":case"utf-8":return Fn(e).length;case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return n*2;case"hex":return n>>>1;case"base64":return sr(e).length;default:if(a)return r?-1:Fn(e).length;o=(""+o).toLowerCase(),a=!0}}d.byteLength=tr;function jc(e,o,n){var r=!1;if((o===void 0||o<0)&&(o=0),o>this.length||((n===void 0||n>this.length)&&(n=this.length),n<=0)||(n>>>=0,o>>>=0,n<=o))return"";for(e||(e="utf8");;)switch(e){case"hex":return np(this,o,n);case"utf8":case"utf-8":return ar(this,o,n);case"ascii":return ep(this,o,n);case"latin1":case"binary":return op(this,o,n);case"base64":return Jc(this,o,n);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return tp(this,o,n);default:if(r)throw new TypeError("Unknown encoding: "+e);e=(e+"").toLowerCase(),r=!0}}d.prototype._isBuffer=!0;function je(e,o,n){var r=e[o];e[o]=e[n],e[n]=r}d.prototype.swap16=function(){var o=this.length;if(o%2!==0)throw new RangeError("Buffer size must be a multiple of 16-bits");for(var n=0;n<o;n+=2)je(this,n,n+1);return this};d.prototype.swap32=function(){var o=this.length;if(o%4!==0)throw new RangeError("Buffer size must be a multiple of 32-bits");for(var n=0;n<o;n+=4)je(this,n,n+3),je(this,n+1,n+2);return this};d.prototype.swap64=function(){var o=this.length;if(o%8!==0)throw new RangeError("Buffer size must be a multiple of 64-bits");for(var n=0;n<o;n+=8)je(this,n,n+7),je(this,n+1,n+6),je(this,n+2,n+5),je(this,n+3,n+4);return this};d.prototype.toString=function(){var o=this.length;return o===0?"":arguments.length===0?ar(this,0,o):jc.apply(this,arguments)};d.prototype.toLocaleString=d.prototype.toString;d.prototype.equals=function(o){if(!d.isBuffer(o))throw new TypeError("Argument must be a Buffer");return this===o?!0:d.compare(this,o)===0};d.prototype.inspect=function(){var o="",n=so.INSPECT_MAX_BYTES;return o=this.toString("hex",0,n).replace(/(.{2})/g,"$1 ").trim(),this.length>n&&(o+=" ... "),"<Buffer "+o+">"};Jt&&(d.prototype[Jt]=d.prototype.inspect);d.prototype.compare=function(o,n,r,a,m){if(ue(o,Uint8Array)&&(o=d.from(o,o.offset,o.byteLength)),!d.isBuffer(o))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+typeof o);if(n===void 0&&(n=0),r===void 0&&(r=o?o.length:0),a===void 0&&(a=0),m===void 0&&(m=this.length),n<0||r>o.length||a<0||m>this.length)throw new RangeError("out of range index");if(a>=m&&n>=r)return 0;if(a>=m)return-1;if(n>=r)return 1;if(n>>>=0,r>>>=0,a>>>=0,m>>>=0,this===o)return 0;for(var s=m-a,_=r-n,l=Math.min(s,_),g=this.slice(a,m),f=o.slice(n,r),T=0;T<l;++T)if(g[T]!==f[T]){s=g[T],_=f[T];break}return s<_?-1:_<s?1:0};function rr(e,o,n,r,a){if(e.length===0)return-1;if(typeof n=="string"?(r=n,n=0):n>2147483647?n=2147483647:n<-2147483648&&(n=-2147483648),n=+n,Un(n)&&(n=a?0:e.length-1),n<0&&(n=e.length+n),n>=e.length){if(a)return-1;n=e.length-1}else if(n<0)if(a)n=0;else return-1;if(typeof o=="string"&&(o=d.from(o,r)),d.isBuffer(o))return o.length===0?-1:Qt(e,o,n,r,a);if(typeof o=="number")return o=o&255,typeof Uint8Array.prototype.indexOf=="function"?a?Uint8Array.prototype.indexOf.call(e,o,n):Uint8Array.prototype.lastIndexOf.call(e,o,n):Qt(e,[o],n,r,a);throw new TypeError("val must be string, number or Buffer")}function Qt(e,o,n,r,a){var m=1,s=e.length,_=o.length;if(r!==void 0&&(r=String(r).toLowerCase(),r==="ucs2"||r==="ucs-2"||r==="utf16le"||r==="utf-16le")){if(e.length<2||o.length<2)return-1;m=2,s/=2,_/=2,n/=2}function l(k,L){return m===1?k[L]:k.readUInt16BE(L*m)}var g;if(a){var f=-1;for(g=n;g<s;g++)if(l(e,g)===l(o,f===-1?0:g-f)){if(f===-1&&(f=g),g-f+1===_)return f*m}else f!==-1&&(g-=g-f),f=-1}else for(n+_>s&&(n=s-_),g=n;g>=0;g--){for(var T=!0,I=0;I<_;I++)if(l(e,g+I)!==l(o,I)){T=!1;break}if(T)return g}return-1}d.prototype.includes=function(o,n,r){return this.indexOf(o,n,r)!==-1};d.prototype.indexOf=function(o,n,r){return rr(this,o,n,r,!0)};d.prototype.lastIndexOf=function(o,n,r){return rr(this,o,n,r,!1)};function qc(e,o,n,r){n=Number(n)||0;var a=e.length-n;r?(r=Number(r),r>a&&(r=a)):r=a;var m=o.length;r>m/2&&(r=m/2);for(var s=0;s<r;++s){var _=parseInt(o.substr(s*2,2),16);if(Un(_))return s;e[n+s]=_}return s}function Kc(e,o,n,r){return qo(Fn(o,e.length-n),e,n,r)}function Xc(e,o,n,r){return qo(ip(o),e,n,r)}function Yc(e,o,n,r){return qo(sr(o),e,n,r)}function Zc(e,o,n,r){return qo(cp(o,e.length-n),e,n,r)}d.prototype.write=function(o,n,r,a){if(n===void 0)a="utf8",r=this.length,n=0;else if(r===void 0&&typeof n=="string")a=n,r=this.length,n=0;else if(isFinite(n))n=n>>>0,isFinite(r)?(r=r>>>0,a===void 0&&(a="utf8")):(a=r,r=void 0);else throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");var m=this.length-n;if((r===void 0||r>m)&&(r=m),o.length>0&&(r<0||n<0)||n>this.length)throw new RangeError("Attempt to write outside buffer bounds");a||(a="utf8");for(var s=!1;;)switch(a){case"hex":return qc(this,o,n,r);case"utf8":case"utf-8":return Kc(this,o,n,r);case"ascii":case"latin1":case"binary":return Xc(this,o,n,r);case"base64":return Yc(this,o,n,r);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return Zc(this,o,n,r);default:if(s)throw new TypeError("Unknown encoding: "+a);a=(""+a).toLowerCase(),s=!0}};d.prototype.toJSON=function(){return{type:"Buffer",data:Array.prototype.slice.call(this._arr||this,0)}};function Jc(e,o,n){return o===0&&n===e.length?Nn.fromByteArray(e):Nn.fromByteArray(e.slice(o,n))}function ar(e,o,n){n=Math.min(e.length,n);for(var r=[],a=o;a<n;){var m=e[a],s=null,_=m>239?4:m>223?3:m>191?2:1;if(a+_<=n){var l,g,f,T;switch(_){case 1:m<128&&(s=m);break;case 2:l=e[a+1],(l&192)===128&&(T=(m&31)<<6|l&63,T>127&&(s=T));break;case 3:l=e[a+1],g=e[a+2],(l&192)===128&&(g&192)===128&&(T=(m&15)<<12|(l&63)<<6|g&63,T>2047&&(T<55296||T>57343)&&(s=T));break;case 4:l=e[a+1],g=e[a+2],f=e[a+3],(l&192)===128&&(g&192)===128&&(f&192)===128&&(T=(m&15)<<18|(l&63)<<12|(g&63)<<6|f&63,T>65535&&T<1114112&&(s=T))}}s===null?(s=65533,_=1):s>65535&&(s-=65536,r.push(s>>>10&1023|55296),s=56320|s&1023),r.push(s),a+=_}return Qc(r)}var er=4096;function Qc(e){var o=e.length;if(o<=er)return String.fromCharCode.apply(String,e);for(var n="",r=0;r<o;)n+=String.fromCharCode.apply(String,e.slice(r,r+=er));return n}function ep(e,o,n){var r="";n=Math.min(e.length,n);for(var a=o;a<n;++a)r+=String.fromCharCode(e[a]&127);return r}function op(e,o,n){var r="";n=Math.min(e.length,n);for(var a=o;a<n;++a)r+=String.fromCharCode(e[a]);return r}function np(e,o,n){var r=e.length;(!o||o<0)&&(o=0),(!n||n<0||n>r)&&(n=r);for(var a="",m=o;m<n;++m)a+=pp[e[m]];return a}function tp(e,o,n){for(var r=e.slice(o,n),a="",m=0;m<r.length-1;m+=2)a+=String.fromCharCode(r[m]+r[m+1]*256);return a}d.prototype.slice=function(o,n){var r=this.length;o=~~o,n=n===void 0?r:~~n,o<0?(o+=r,o<0&&(o=0)):o>r&&(o=r),n<0?(n+=r,n<0&&(n=0)):n>r&&(n=r),n<o&&(n=o);var a=this.subarray(o,n);return Object.setPrototypeOf(a,d.prototype),a};function G(e,o,n){if(e%1!==0||e<0)throw new RangeError("offset is not uint");if(e+o>n)throw new RangeError("Trying to access beyond buffer length")}d.prototype.readUintLE=d.prototype.readUIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||G(o,n,this.length);for(var a=this[o],m=1,s=0;++s<n&&(m*=256);)a+=this[o+s]*m;return a};d.prototype.readUintBE=d.prototype.readUIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||G(o,n,this.length);for(var a=this[o+--n],m=1;n>0&&(m*=256);)a+=this[o+--n]*m;return a};d.prototype.readUint8=d.prototype.readUInt8=function(o,n){return o=o>>>0,n||G(o,1,this.length),this[o]};d.prototype.readUint16LE=d.prototype.readUInt16LE=function(o,n){return o=o>>>0,n||G(o,2,this.length),this[o]|this[o+1]<<8};d.prototype.readUint16BE=d.prototype.readUInt16BE=function(o,n){return o=o>>>0,n||G(o,2,this.length),this[o]<<8|this[o+1]};d.prototype.readUint32LE=d.prototype.readUInt32LE=function(o,n){return o=o>>>0,n||G(o,4,this.length),(this[o]|this[o+1]<<8|this[o+2]<<16)+this[o+3]*16777216};d.prototype.readUint32BE=d.prototype.readUInt32BE=function(o,n){return o=o>>>0,n||G(o,4,this.length),this[o]*16777216+(this[o+1]<<16|this[o+2]<<8|this[o+3])};d.prototype.readIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||G(o,n,this.length);for(var a=this[o],m=1,s=0;++s<n&&(m*=256);)a+=this[o+s]*m;return m*=128,a>=m&&(a-=Math.pow(2,8*n)),a};d.prototype.readIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||G(o,n,this.length);for(var a=n,m=1,s=this[o+--a];a>0&&(m*=256);)s+=this[o+--a]*m;return m*=128,s>=m&&(s-=Math.pow(2,8*n)),s};d.prototype.readInt8=function(o,n){return o=o>>>0,n||G(o,1,this.length),this[o]&128?(255-this[o]+1)*-1:this[o]};d.prototype.readInt16LE=function(o,n){o=o>>>0,n||G(o,2,this.length);var r=this[o]|this[o+1]<<8;return r&32768?r|4294901760:r};d.prototype.readInt16BE=function(o,n){o=o>>>0,n||G(o,2,this.length);var r=this[o+1]|this[o]<<8;return r&32768?r|4294901760:r};d.prototype.readInt32LE=function(o,n){return o=o>>>0,n||G(o,4,this.length),this[o]|this[o+1]<<8|this[o+2]<<16|this[o+3]<<24};d.prototype.readInt32BE=function(o,n){return o=o>>>0,n||G(o,4,this.length),this[o]<<24|this[o+1]<<16|this[o+2]<<8|this[o+3]};d.prototype.readFloatLE=function(o,n){return o=o>>>0,n||G(o,4,this.length),po.read(this,o,!0,23,4)};d.prototype.readFloatBE=function(o,n){return o=o>>>0,n||G(o,4,this.length),po.read(this,o,!1,23,4)};d.prototype.readDoubleLE=function(o,n){return o=o>>>0,n||G(o,8,this.length),po.read(this,o,!0,52,8)};d.prototype.readDoubleBE=function(o,n){return o=o>>>0,n||G(o,8,this.length),po.read(this,o,!1,52,8)};function Y(e,o,n,r,a,m){if(!d.isBuffer(e))throw new TypeError('"buffer" argument must be a Buffer instance');if(o>a||o<m)throw new RangeError('"value" argument is out of bounds');if(n+r>e.length)throw new RangeError("Index out of range")}d.prototype.writeUintLE=d.prototype.writeUIntLE=function(o,n,r,a){if(o=+o,n=n>>>0,r=r>>>0,!a){var m=Math.pow(2,8*r)-1;Y(this,o,n,r,m,0)}var s=1,_=0;for(this[n]=o&255;++_<r&&(s*=256);)this[n+_]=o/s&255;return n+r};d.prototype.writeUintBE=d.prototype.writeUIntBE=function(o,n,r,a){if(o=+o,n=n>>>0,r=r>>>0,!a){var m=Math.pow(2,8*r)-1;Y(this,o,n,r,m,0)}var s=r-1,_=1;for(this[n+s]=o&255;--s>=0&&(_*=256);)this[n+s]=o/_&255;return n+r};d.prototype.writeUint8=d.prototype.writeUInt8=function(o,n,r){return o=+o,n=n>>>0,r||Y(this,o,n,1,255,0),this[n]=o&255,n+1};d.prototype.writeUint16LE=d.prototype.writeUInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||Y(this,o,n,2,65535,0),this[n]=o&255,this[n+1]=o>>>8,n+2};d.prototype.writeUint16BE=d.prototype.writeUInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||Y(this,o,n,2,65535,0),this[n]=o>>>8,this[n+1]=o&255,n+2};d.prototype.writeUint32LE=d.prototype.writeUInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||Y(this,o,n,4,4294967295,0),this[n+3]=o>>>24,this[n+2]=o>>>16,this[n+1]=o>>>8,this[n]=o&255,n+4};d.prototype.writeUint32BE=d.prototype.writeUInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||Y(this,o,n,4,4294967295,0),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};d.prototype.writeIntLE=function(o,n,r,a){if(o=+o,n=n>>>0,!a){var m=Math.pow(2,8*r-1);Y(this,o,n,r,m-1,-m)}var s=0,_=1,l=0;for(this[n]=o&255;++s<r&&(_*=256);)o<0&&l===0&&this[n+s-1]!==0&&(l=1),this[n+s]=(o/_>>0)-l&255;return n+r};d.prototype.writeIntBE=function(o,n,r,a){if(o=+o,n=n>>>0,!a){var m=Math.pow(2,8*r-1);Y(this,o,n,r,m-1,-m)}var s=r-1,_=1,l=0;for(this[n+s]=o&255;--s>=0&&(_*=256);)o<0&&l===0&&this[n+s+1]!==0&&(l=1),this[n+s]=(o/_>>0)-l&255;return n+r};d.prototype.writeInt8=function(o,n,r){return o=+o,n=n>>>0,r||Y(this,o,n,1,127,-128),o<0&&(o=255+o+1),this[n]=o&255,n+1};d.prototype.writeInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||Y(this,o,n,2,32767,-32768),this[n]=o&255,this[n+1]=o>>>8,n+2};d.prototype.writeInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||Y(this,o,n,2,32767,-32768),this[n]=o>>>8,this[n+1]=o&255,n+2};d.prototype.writeInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||Y(this,o,n,4,2147483647,-2147483648),this[n]=o&255,this[n+1]=o>>>8,this[n+2]=o>>>16,this[n+3]=o>>>24,n+4};d.prototype.writeInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||Y(this,o,n,4,2147483647,-2147483648),o<0&&(o=4294967295+o+1),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};function ir(e,o,n,r,a,m){if(n+r>e.length)throw new RangeError("Index out of range");if(n<0)throw new RangeError("Index out of range")}function cr(e,o,n,r,a){return o=+o,n=n>>>0,a||ir(e,o,n,4,34028234663852886e22,-34028234663852886e22),po.write(e,o,n,r,23,4),n+4}d.prototype.writeFloatLE=function(o,n,r){return cr(this,o,n,!0,r)};d.prototype.writeFloatBE=function(o,n,r){return cr(this,o,n,!1,r)};function pr(e,o,n,r,a){return o=+o,n=n>>>0,a||ir(e,o,n,8,17976931348623157e292,-17976931348623157e292),po.write(e,o,n,r,52,8),n+8}d.prototype.writeDoubleLE=function(o,n,r){return pr(this,o,n,!0,r)};d.prototype.writeDoubleBE=function(o,n,r){return pr(this,o,n,!1,r)};d.prototype.copy=function(o,n,r,a){if(!d.isBuffer(o))throw new TypeError("argument should be a Buffer");if(r||(r=0),!a&&a!==0&&(a=this.length),n>=o.length&&(n=o.length),n||(n=0),a>0&&a<r&&(a=r),a===r||o.length===0||this.length===0)return 0;if(n<0)throw new RangeError("targetStart out of bounds");if(r<0||r>=this.length)throw new RangeError("Index out of range");if(a<0)throw new RangeError("sourceEnd out of bounds");a>this.length&&(a=this.length),o.length-n<a-r&&(a=o.length-n+r);var m=a-r;return this===o&&typeof Uint8Array.prototype.copyWithin=="function"?this.copyWithin(n,r,a):Uint8Array.prototype.set.call(o,this.subarray(r,a),n),m};d.prototype.fill=function(o,n,r,a){if(typeof o=="string"){if(typeof n=="string"?(a=n,n=0,r=this.length):typeof r=="string"&&(a=r,r=this.length),a!==void 0&&typeof a!="string")throw new TypeError("encoding must be a string");if(typeof a=="string"&&!d.isEncoding(a))throw new TypeError("Unknown encoding: "+a);if(o.length===1){var m=o.charCodeAt(0);(a==="utf8"&&m<128||a==="latin1")&&(o=m)}}else typeof o=="number"?o=o&255:typeof o=="boolean"&&(o=Number(o));if(n<0||this.length<n||this.length<r)throw new RangeError("Out of range index");if(r<=n)return this;n=n>>>0,r=r===void 0?this.length:r>>>0,o||(o=0);var s;if(typeof o=="number")for(s=n;s<r;++s)this[s]=o;else{var _=d.isBuffer(o)?o:d.from(o,a),l=_.length;if(l===0)throw new TypeError('The value "'+o+'" is invalid for argument "value"');for(s=0;s<r-n;++s)this[s+n]=_[s%l]}return this};var rp=/[^+/0-9A-Za-z-_]/g;function ap(e){if(e=e.split("=")[0],e=e.trim().replace(rp,""),e.length<2)return"";for(;e.length%4!==0;)e=e+"=";return e}function Fn(e,o){o=o||1/0;for(var n,r=e.length,a=null,m=[],s=0;s<r;++s){if(n=e.charCodeAt(s),n>55295&&n<57344){if(!a){if(n>56319){(o-=3)>-1&&m.push(239,191,189);continue}else if(s+1===r){(o-=3)>-1&&m.push(239,191,189);continue}a=n;continue}if(n<56320){(o-=3)>-1&&m.push(239,191,189),a=n;continue}n=(a-55296<<10|n-56320)+65536}else a&&(o-=3)>-1&&m.push(239,191,189);if(a=null,n<128){if((o-=1)<0)break;m.push(n)}else if(n<2048){if((o-=2)<0)break;m.push(n>>6|192,n&63|128)}else if(n<65536){if((o-=3)<0)break;m.push(n>>12|224,n>>6&63|128,n&63|128)}else if(n<1114112){if((o-=4)<0)break;m.push(n>>18|240,n>>12&63|128,n>>6&63|128,n&63|128)}else throw new Error("Invalid code point")}return m}function ip(e){for(var o=[],n=0;n<e.length;++n)o.push(e.charCodeAt(n)&255);return o}function cp(e,o){for(var n,r,a,m=[],s=0;s<e.length&&!((o-=2)<0);++s)n=e.charCodeAt(s),r=n>>8,a=n%256,m.push(a),m.push(r);return m}function sr(e){return Nn.toByteArray(ap(e))}function qo(e,o,n,r){for(var a=0;a<r&&!(a+n>=o.length||a>=e.length);++a)o[a+n]=e[a];return a}function ue(e,o){return e instanceof o||e!=null&&e.constructor!=null&&e.constructor.name!=null&&e.constructor.name===o.name}function Un(e){return e!==e}var pp=function(){for(var e="0123456789abcdef",o=new Array(256),n=0;n<16;++n)for(var r=n*16,a=0;a<16;++a)o[r+a]=e[n]+e[a];return o}()});var gr=Ho(($m,ur)=>{t();var O=ur.exports={},ge,fe;function Wn(){throw new Error("setTimeout has not been defined")}function Gn(){throw new Error("clearTimeout has not been defined")}(function(){try{typeof setTimeout=="function"?ge=setTimeout:ge=Wn}catch(e){ge=Wn}try{typeof clearTimeout=="function"?fe=clearTimeout:fe=Gn}catch(e){fe=Gn}})();function dr(e){if(ge===setTimeout)return setTimeout(e,0);if((ge===Wn||!ge)&&setTimeout)return ge=setTimeout,setTimeout(e,0);try{return ge(e,0)}catch(o){try{return ge.call(null,e,0)}catch(n){return ge.call(this,e,0)}}}function sp(e){if(fe===clearTimeout)return clearTimeout(e);if((fe===Gn||!fe)&&clearTimeout)return fe=clearTimeout,clearTimeout(e);try{return fe(e)}catch(o){try{return fe.call(null,e)}catch(n){return fe.call(this,e)}}}var we=[],mo=!1,qe,Ko=-1;function mp(){!mo||!qe||(mo=!1,qe.length?we=qe.concat(we):Ko=-1,we.length&&_r())}function _r(){if(!mo){var e=dr(mp);mo=!0;for(var o=we.length;o;){for(qe=we,we=[];++Ko<o;)qe&&qe[Ko].run();Ko=-1,o=we.length}qe=null,mo=!1,sp(e)}}O.nextTick=function(e){var o=new Array(arguments.length-1);if(arguments.length>1)for(var n=1;n<arguments.length;n++)o[n-1]=arguments[n];we.push(new lr(e,o)),we.length===1&&!mo&&dr(_r)};function lr(e,o){this.fun=e,this.array=o}lr.prototype.run=function(){this.fun.apply(null,this.array)};O.title="browser";O.browser=!0;O.env={};O.argv=[];O.version="";O.versions={};function Se(){}O.on=Se;O.addListener=Se;O.once=Se;O.off=Se;O.removeListener=Se;O.removeAllListeners=Se;O.emit=Se;O.prependListener=Se;O.prependOnceListener=Se;O.listeners=function(e){return[]};O.binding=function(e){throw new Error("process.binding is not supported")};O.cwd=function(){return"/"};O.chdir=function(e){throw new Error("process.chdir is not supported")};O.umask=function(){return 0}});var i,c,Vm,t=w(()=>{i=S(mr()),c=S(gr()),Vm=function(e){function o(){var r=this||self;return delete e.prototype.__magic__,r}if(typeof globalThis=="object")return globalThis;if(this)return o();e.defineProperty(e.prototype,"__magic__",{configurable:!0,get:o});var n=__magic__;return n}(Object)});var Xo,$n=w(()=>{"use strict";t();Xo=()=>Date.now()/1e3});var fr=w(()=>{"use strict";t()});var hr=w(()=>{"use strict";t()});var Vn=w(()=>{"use strict";t();$n();fr();hr()});var zn={};He(zn,{getItem:()=>_p,localStorageKeys:()=>ae,removeItem:()=>lp,setItem:()=>dp});var ae,Hn,dp,_p,lp,Ke=w(()=>{"use strict";t();Vn();ae={loginExpiresAt:"dapp-core-login-expires-at",logoutEvent:"dapp-core-logout-event"},Hn=typeof localStorage!="undefined",dp=({key:e,data:o,expires:n})=>{!Hn||localStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},_p=e=>{if(!Hn)return;let o=localStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:Xo()>=n.expires?(localStorage.removeItem(String(e)),null):n.data},lp=e=>{!Hn||localStorage.removeItem(String(e))}});var jn={};He(jn,{clear:()=>br,getItem:()=>xr,removeItem:()=>yr,setItem:()=>Tr,storage:()=>up});var Tr,xr,yr,br,up,wr=w(()=>{"use strict";t();Tr=({key:e,data:o,expires:n})=>{sessionStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},xr=e=>{let o=sessionStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:Date.now()>=n.expires?(sessionStorage.removeItem(String(e)),null):n.data},yr=e=>sessionStorage.removeItem(String(e)),br=()=>sessionStorage.clear(),up={setItem:Tr,getItem:xr,removeItem:yr,clear:br}});var ve,_o=w(()=>{"use strict";t();Ke();wr();ve={session:jn,local:zn}});function No(){return new Date().setHours(new Date().getHours()+24)}function Ro(e){ve.local.setItem({key:ae.loginExpiresAt,data:e,expires:e})}var qn=w(()=>{"use strict";t();_o();Ke()});var N=w(()=>{"use strict";t()});var lo=w(()=>{"use strict";t()});var Sr=w(()=>{"use strict";t()});var j=w(()=>{"use strict";t()});var vr=w(()=>{"use strict";t()});var kr=w(()=>{"use strict";t()});var Ar=w(()=>{"use strict";t()});var Ir=w(()=>{"use strict";t()});var ie=w(()=>{"use strict";t();Sr();j();N();vr();kr();Ar();Ir()});var Xe,Er,hd,Lr,Td,Cr,xd,yd,gp,uo=w(()=>{"use strict";t();ie();Xe={devnet:{id:"devnet",chainId:"D",name:"Devnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://devnet-wallet.elrond.com",apiAddress:"https://devnet-api.elrond.com",explorerAddress:"http://devnet-explorer.elrond.com",apiTimeout:"4000"},testnet:{id:"testnet",chainId:"T",name:"Testnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://testnet-wallet.elrond.com",apiAddress:"https://testnet-api.elrond.com",explorerAddress:"http://testnet-explorer.elrond.com",apiTimeout:"4000"},mainnet:{id:"mainnet",chainId:"1",name:"Mainnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://wallet.elrond.com",apiAddress:"https://api.elrond.com",explorerAddress:"https://explorer.elrond.com",apiTimeout:"4000"}},{chainId:Er,egldLabel:hd}=Xe["devnet"],{chainId:Lr,egldLabel:Td}=Xe["testnet"],{chainId:Cr,egldLabel:xd}=Xe["mainnet"],yd={["devnet"]:Er,["testnet"]:Lr,["mainnet"]:Cr},gp={[Er]:"devnet",[Lr]:"testnet",[Cr]:"mainnet"}});var Kn=w(()=>{"use strict";t()});var Dr=w(()=>{"use strict";t()});var Xn,Yn=w(()=>{"use strict";t();Xn="transaction-status-toast-id"});var Mo,Yo,Pr,Nr,Zo,Rr,go,D=w(()=>{"use strict";t();lo();uo();Kn();Dr();Yn();Mo=5e4,Yo=1e9,Pr=1,Nr=1,Zo="logout",Rr="login",go="0"});var Zn,H,Q,q=w(()=>{"use strict";t();Zn=require("@reduxjs/toolkit");D();H=(0,Zn.createAction)(Zo),Q=(0,Zn.createAction)(Rr,e=>({payload:e}))});var Fr,Mr,Or,Bd,fo,fp,Ne,ho,Jn,Br,Ur,Qn=w(()=>{"use strict";t();Fr=require("@reduxjs/toolkit");qn();N();q();Mr={loginMethod:"",walletConnectLogin:null,ledgerLogin:null,tokenLogin:null,walletLogin:null,extensionLogin:null,isLoginSessionInvalid:!1},Or=(0,Fr.createSlice)({name:"loginInfoSlice",initialState:Mr,reducers:{setLoginMethod:(e,o)=>{e.loginMethod=o.payload},setTokenLogin:(e,o)=>{e.tokenLogin=o.payload},setTokenLoginSignature:(e,o)=>{(e==null?void 0:e.tokenLogin)!=null&&(e.tokenLogin.signature=o.payload)},setWalletLogin:(e,o)=>{e.walletLogin=o.payload},setWalletConnectLogin:(e,o)=>{e.walletConnectLogin=o.payload},setLedgerLogin:(e,o)=>{e.ledgerLogin=o.payload},invalidateLoginSession:e=>{e.isLoginSessionInvalid=!0}},extraReducers:e=>{e.addCase(H,()=>Mr),e.addCase(Q,(o,n)=>{o.loginMethod=n.payload.loginMethod,Ro(No())})}}),{setLoginMethod:Bd,setWalletConnectLogin:fo,setLedgerLogin:fp,setTokenLogin:Ne,setTokenLoginSignature:ho,setWalletLogin:Jn,invalidateLoginSession:Br}=Or.actions,Ur=Or.reducer});var Wr,Gr,Ye,To=w(()=>{"use strict";t();Wr=!1,Gr=(e=!0)=>{Wr=e},Ye=(e,o=0)=>{Wr||setTimeout(()=>{window.location.href=e},o)}});function xo(e,o){Boolean(e)&&e!=null&&setTimeout(()=>{window.location.pathname.includes(e)||(typeof o=="function"?o(e):Ye(e))},200)}var $r=w(()=>{"use strict";t();To()});function et(e){return e[Math.floor(Math.random()*e.length)]}var Vr=w(()=>{"use strict";t()});var ot=w(()=>{"use strict";t();Kn()});var yo=w(()=>{"use strict";t();$r();Vr();ot()});var Hr,zr,nt,hp,jr,tt,qr,Kr,Jo=w(()=>{"use strict";t();Hr=require("@reduxjs/toolkit"),zr=S(require("lodash.omit"));yo();nt={id:"not-configured",chainId:"",name:"NOT CONFIGURED",egldLabel:"",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"",walletConnectBridgeAddress:"",walletConnectV2RelayAddress:"",walletConnectV2ProjectId:"",walletAddress:"",apiAddress:"",explorerAddress:"",apiTimeout:"4000"},hp={network:nt,chainID:"-1"},jr=(0,Hr.createSlice)({name:"appConfig",initialState:hp,reducers:{initializeNetworkConfig:(e,o)=>{let n=et(o.payload.walletConnectBridgeAddresses),r=et(o.payload.walletConnectV2RelayAddresses),a=(0,zr.default)(o.payload,"walletConnectBridgeAddresses","walletConnectV2RelayAddresses");e.network=A(h(h({},e.network),a),{walletConnectBridgeAddress:n,walletConnectV2RelayAddress:r})},setChainID:(e,o)=>{e.chainID=o.payload}}}),{initializeNetworkConfig:tt,setChainID:qr}=jr.actions,Kr=jr.reducer});var at,Xr,Yr,Qo,rt,Zr,Ze,i_,en,Jr,it,Tp,c_,ct,Qr,ea,oa,pt=w(()=>{"use strict";t();at=require("@elrondnetwork/erdjs"),Xr=require("@reduxjs/toolkit"),Yr=require("redux-persist");D();_o();Ke();q();Qo={balance:"...",address:"",nonce:0,txCount:0,scrCount:0,claimableRewards:go},rt={address:"",websocketEvent:null,accounts:{"":Qo},ledgerAccount:null,publicKey:"",walletConnectAccount:null,isAccountLoading:!0,accountLoadingError:null},Zr=(0,Xr.createSlice)({name:"accountInfoSlice",initialState:rt,reducers:{setAddress:(e,o)=>{let n=o.payload;e.address=n,e.publicKey=new at.Address(n).hex()},setAccount:(e,o)=>{let n=e.address===o.payload.address;e.accounts={[e.address]:n?o.payload:Qo},e.isAccountLoading=!1,e.accountLoadingError=null},setAccountNonce:(e,o)=>{let{address:n}=e;e.accounts[n].nonce=o.payload},setAccountShard:(e,o)=>{e.shard=o.payload},setLedgerAccount:(e,o)=>{e.ledgerAccount=o.payload},updateLedgerAccount:(e,o)=>{e.ledgerAccount!=null&&(e.ledgerAccount.index=o.payload.index,e.ledgerAccount.address=o.payload.address)},setWalletConnectAccount:(e,o)=>{e.walletConnectAccount=o.payload},setIsAccountLoading:(e,o)=>{e.isAccountLoading=o.payload,e.accountLoadingError=null},setAccountLoadingError:(e,o)=>{e.accountLoadingError=o.payload,e.isAccountLoading=!1},setWebsocketEvent:(e,o)=>{e.websocketEvent={timestamp:Date.now(),message:o.payload}}},extraReducers:e=>{e.addCase(H,()=>(ve.local.removeItem(ae.loginExpiresAt),rt)),e.addCase(Q,(o,n)=>{let{address:r}=n.payload;o.address=r,o.publicKey=new at.Address(r).hex()}),e.addCase(Yr.REHYDRATE,(o,n)=>{var g;if(!((g=n.payload)!=null&&g.account))return;let{account:r}=n.payload,{address:a,shard:m,accounts:s,publicKey:_}=r;o.address=a,o.shard=m;let l=s&&a in s;o.accounts=l?s:rt.accounts,o.publicKey=_})}}),{setAccount:Ze,setAddress:i_,setAccountNonce:en,setAccountShard:Jr,setLedgerAccount:it,updateLedgerAccount:Tp,setWalletConnectAccount:c_,setIsAccountLoading:ct,setAccountLoadingError:Qr,setWebsocketEvent:ea}=Zr.actions,oa=Zr.reducer});function ke(e){return e!=null&&(Ap(e)||Cp(e))}function Je(e){return e!=null&&(Ip(e)||Dp(e))}function Qe(e){return e!=null&&(Ep(e)||Pp(e))}function st(e){return e!=null&&Lp(e)}function Ap(e){return e!=null&&xp.includes(e)}function Ip(e){return e!=null&&yp.includes(e)}function Ep(e){return e!=null&&bp.includes(e)}function Lp(e){return e!=null&&wp.includes(e)}function Cp(e){return e!=null&&Sp.includes(e)}function Dp(e){return e!=null&&vp.includes(e)}function Pp(e){return e!=null&&kp.includes(e)}var xp,yp,bp,wp,Sp,vp,kp,wo=w(()=>{"use strict";t();N();xp=["sent"],yp=["success"],bp=["fail","cancelled","timedOut"],wp=["timedOut"],Sp=["pending"],vp=["success"],kp=["fail","invalid"]});var na,ta,Fo,Np,ra,Oo,eo,Rp,Bo,Mp,Fp,f_,Op,aa,mt,on,nn=w(()=>{"use strict";t();na=require("@reduxjs/toolkit"),ta=require("redux-persist");N();wo();q();Fo={signedTransactions:{},transactionsToSign:null,signTransactionsError:null,signTransactionsCancelMessage:null,customTransactionInformationForSessionId:{}},Np={signWithoutSending:!1,sessionInformation:null,redirectAfterSign:!1},ra=(0,na.createSlice)({name:"transactionsSlice",initialState:Fo,reducers:{moveTransactionsToSignedState:(e,o)=>{var l,g;let{sessionId:n,transactions:r,errorMessage:a,status:m,redirectRoute:s}=o.payload,_=((l=e.customTransactionInformationForSessionId)==null?void 0:l[n])||Np;e.signedTransactions[n]={transactions:r,status:m,errorMessage:a,redirectRoute:s,customTransactionInformation:_},((g=e==null?void 0:e.transactionsToSign)==null?void 0:g.sessionId)===n&&(e.transactionsToSign=Fo.transactionsToSign)},clearSignedTransaction:(e,o)=>{e.signedTransactions[o.payload]&&delete e.signedTransactions[o.payload]},clearTransactionToSign:e=>{e!=null&&e.transactionsToSign&&(e.transactionsToSign=null)},updateSignedTransaction:(e,o)=>{e.signedTransactions=h(h({},e.signedTransactions),o.payload)},updateSignedTransactions:(e,o)=>{let{sessionId:n,status:r,errorMessage:a,transactions:m}=o.payload;e.signedTransactions[n]!=null&&(e.signedTransactions[n].status=r,a!=null&&(e.signedTransactions[n].errorMessage=a),m!=null&&(e.signedTransactions[n].transactions=m))},updateSignedTransactionStatus:(e,o)=>{var _,l,g,f,T,I;let{sessionId:n,status:r,errorMessage:a,transactionHash:m}=o.payload,s=(l=(_=e.signedTransactions)==null?void 0:_[n])==null?void 0:l.transactions;if(s!=null){e.signedTransactions[n].transactions=s.map(F=>F.hash===m?A(h({},F),{status:r,errorMessage:a}):F);let k=(f=(g=e.signedTransactions[n])==null?void 0:g.transactions)==null?void 0:f.every(F=>Je(F.status)),L=(I=(T=e.signedTransactions[n])==null?void 0:T.transactions)==null?void 0:I.every(F=>Qe(F.status));k&&(e.signedTransactions[n].status="success"),L&&(e.signedTransactions[n].status="fail")}},setTransactionsToSign:(e,o)=>{e.transactionsToSign=o.payload;let{sessionId:n,customTransactionInformation:r}=o.payload;e.customTransactionInformationForSessionId[n]=r,e.signTransactionsError=null},clearAllTransactionsToSign:e=>{e.transactionsToSign=Fo.transactionsToSign,e.signTransactionsError=null},clearAllSignedTransactions:e=>{e.signedTransactions=Fo.signedTransactions},setSignTransactionsError:(e,o)=>{e.signTransactionsError=o.payload},setSignTransactionsCancelMessage:(e,o)=>{e.signTransactionsCancelMessage=o.payload}},extraReducers:e=>{e.addCase(H,()=>Fo),e.addCase(ta.REHYDRATE,(o,n)=>{var s;if(!((s=n.payload)!=null&&s.transactions))return;let{signedTransactions:r,customTransactionInformationForSessionId:a}=n.payload.transactions,m=Object.entries(r).reduce((_,[l,g])=>{let f=new Date(l),T=new Date;return T.setHours(T.getHours()+5),T-f>0||(_[l]=g),_},{});a!=null&&(o.customTransactionInformationForSessionId=a),r!=null&&(o.signedTransactions=m)})}}),{updateSignedTransactionStatus:Oo,updateSignedTransactions:eo,setTransactionsToSign:Rp,clearAllTransactionsToSign:Bo,clearAllSignedTransactions:Mp,clearSignedTransaction:Fp,clearTransactionToSign:f_,setSignTransactionsError:Op,setSignTransactionsCancelMessage:aa,moveTransactionsToSignedState:mt}=ra.actions,on=ra.reducer});var ia,_t,lt,ut,Bp,dt,ca,y_,Up,pa,tn,rn=w(()=>{"use strict";t();ia=require("@reduxjs/toolkit");q();_t="Transaction failed",lt="Transaction successful",ut="Processing transaction",Bp="Transaction submitted",dt={},ca=(0,ia.createSlice)({name:"transactionsInfo",initialState:dt,reducers:{setTransactionsDisplayInfo(e,o){let{sessionId:n,transactionsDisplayInfo:r}=o.payload;n!=null&&(e[n]={errorMessage:(r==null?void 0:r.errorMessage)||_t,successMessage:(r==null?void 0:r.successMessage)||lt,processingMessage:(r==null?void 0:r.processingMessage)||ut,submittedMessage:(r==null?void 0:r.submittedMessage)||Bp,transactionDuration:r==null?void 0:r.transactionDuration})},clearTransactionsInfoForSessionId(e,o){o.payload!=null&&delete e[o.payload]},clearTransactionsInfo:()=>dt},extraReducers:e=>{e.addCase(H,()=>dt)}}),{clearTransactionsInfo:y_,setTransactionsDisplayInfo:Up,clearTransactionsInfoForSessionId:pa}=ca.actions,tn=ca.reducer});var ma,sa,da,_a,Wp,v_,Gp,la,gt=w(()=>{"use strict";t();ma=require("@reduxjs/toolkit");q();sa={},da=(0,ma.createSlice)({name:"modalsSlice",initialState:sa,reducers:{setTxSubmittedModal:(e,o)=>{e.txSubmittedModal=o.payload},setNotificationModal:(e,o)=>{e.notificationModal=o.payload},clearTxSubmittedModal:e=>{e.txSubmittedModal=void 0},clearNotificationModal:e=>{e.notificationModal=void 0}},extraReducers:e=>{e.addCase(H,()=>sa)}}),{setTxSubmittedModal:_a,setNotificationModal:Wp,clearTxSubmittedModal:v_,clearNotificationModal:Gp}=da.actions,la=da.reducer});var ga,ua,fa,$p,ha,P_,N_,R_,M_,an,cn=w(()=>{"use strict";t();ga=require("@reduxjs/toolkit");Yn();ie();$n();q();ua={customToasts:[],transactionToasts:[],failTransactionToast:null},fa=(0,ga.createSlice)({name:"toastsSlice",initialState:ua,reducers:{addCustomToast:(e,o)=>{e.customToasts.push(A(h({},o.payload),{type:"custom",toastId:o.payload.toastId||`custom-toast-${e.customToasts.length+1}`}))},removeCustomToast:(e,o)=>{e.customToasts=e.customToasts.filter(n=>n.toastId!==o.payload)},addTransactionToast:(e,o)=>{e.transactionToasts.push({type:"transaction",startTimestamp:Xo(),toastId:o.payload||`custom-toast-${e.transactionToasts.length+1}`})},removeTransactionToast:(e,o)=>{e.transactionToasts=e.transactionToasts.filter(n=>n.toastId!==o.payload)},addFailTransactionToast:(e,o)=>{e.failTransactionToast=A(h({},o.payload),{toastId:Xn})},removeFailTransactionToast:e=>{e.failTransactionToast=null}},extraReducers:e=>{e.addCase(H,()=>ua)}}),{addCustomToast:$p,removeCustomToast:ha,addTransactionToast:P_,removeTransactionToast:N_,addFailTransactionToast:R_,removeFailTransactionToast:M_}=fa.actions,an=fa.reducer});var E=w(()=>{"use strict";t();Qn();Jo();pt();nn();rn();gt();cn()});var xt=w(()=>{"use strict";t()});var Ba,Oa,Ua,tu,ru,au,iu,cu,pu,Wa,Ga=w(()=>{"use strict";t();Ba=require("@reduxjs/toolkit");q();Oa={visible:!1,modal:{showHeader:!0,showFooter:!1,headerText:"",footerText:""}},Ua=(0,Ba.createSlice)({name:"dappModalsSlice",initialState:Oa,reducers:{setDappModalVisibility:(e,o)=>{e.visible=o.payload},setDappModalHeaderVisibility:(e,o)=>{e.modal.showHeader=o.payload},setDappModalFooterVisibility:(e,o)=>{e.modal.showFooter=o.payload},setDappModalHeaderText:(e,o)=>{e.modal.headerText=o.payload},setDappModalFooterText:(e,o)=>{e.modal.footerText=o.payload},setDappModalConfig:(e,o)=>{e.modal=h({},o.payload)}},extraReducers:e=>{e.addCase(H,()=>Oa)}}),{setDappModalVisibility:tu,setDappModalHeaderVisibility:ru,setDappModalFooterVisibility:au,setDappModalHeaderText:iu,setDappModalFooterText:cu,setDappModalConfig:pu}=Ua.actions,Wa=Ua.reducer});var $a,es,os,ko,_n=w(()=>{"use strict";t();$a=require("@reduxjs/toolkit");xt();pt();Ga();Qn();gt();Jo();cn();rn();nn();es={["account"]:oa,["networkConfig"]:Kr,["loginInfo"]:Ur,["modals"]:la,["dappModal"]:Wa,["toasts"]:an,["transactions"]:on,["transactionsInfo"]:tn},os=(e={})=>(0,$a.combineReducers)(h(h({},es),e)),ko=os});var za={};He(za,{default:()=>ps});function yt(e,o=[]){return{key:e,version:1,storage:Ha.default,blacklist:o}}var no,Va,Ha,ns,ts,rs,as,is,cs,ps,ja=w(()=>{"use strict";t();no=require("redux-persist"),Va=S(require("redux-persist/lib/storage")),Ha=S(require("redux-persist/lib/storage/session"));_n();E();cn();rn();nn();xt();ns={2:e=>A(h({},e),{networkConfig:nt})};ts=yt("dapp-core-transactionsInfo"),rs=yt("dapp-core-transactions",["transactionsToSign"]),as=yt("dapp-core-toasts"),is={key:"dapp-core-store",version:2,storage:Va.default,whitelist:["account","loginInfo","modals","networkConfig"],migrate:(0,no.createMigrate)(ns,{debug:!1})},cs={["toasts"]:(0,no.persistReducer)(as,an),["transactions"]:(0,no.persistReducer)(rs,on),["transactionsInfo"]:(0,no.persistReducer)(ts,tn)},ps=(0,no.persistReducer)(is,ko(cs))});var qa={};He(qa,{default:()=>ss});var ss,Ka=w(()=>{"use strict";t();_n();ss=ko()});var Xa={};He(Xa,{default:()=>ds});var ce,ms,ds,Ya=w(()=>{"use strict";t();ce=require("redux-persist"),ms=[ce.FLUSH,ce.REHYDRATE,ce.PAUSE,ce.PERSIST,ce.PURGE,ce.REGISTER],ds=ms});var Qa={};He(Qa,{default:()=>Ja});function Ja(e){return(0,Za.persistStore)(e)}var Za,ei=w(()=>{"use strict";t();Za=require("redux-persist")});var Fm={};He(Fm,{DappProvider:()=>Mm});module.exports=Po(Fm);t();var Co=S(require("react")),vc=require("react-redux"),kc=require("redux-persist/integration/react");t();var ao=require("react"),Ui=require("@elrondnetwork/erdjs-extension-provider"),Wi=require("@elrondnetwork/erdjs-hw-provider");t();t();t();var xi=S(require("axios"));t();t();E();t();var Ta=S(require("lodash.isequal")),pn=require("reselect"),x=(0,pn.createSelectorCreator)(pn.defaultMemoize,Ta.default);var Ae=e=>e.account,me=x(Ae,e=>e.address),Me=x(Ae,me,(e,o)=>o in e.accounts?e.accounts[o]:Qo),xa=x(Ae,Me,(e,o)=>{let a=e,{accounts:n}=a,r=Ve(a,["accounts"]);return A(h({},r),{account:o})}),Vp=x(Me,e=>e.balance),ya=x(Me,e=>{var o;return((o=e==null?void 0:e.nonce)==null?void 0:o.valueOf())||0}),ba=x(Ae,e=>e.shard),ft=x(Ae,e=>e.ledgerAccount),K_=x(Ae,e=>e.walletConnectAccount),X_=x(Ae,e=>e.isAccountLoading),Y_=x(Ae,e=>e.accountLoadingError),ht=x(Ae,e=>e.websocketEvent);t();var Ie=e=>e.networkConfig,Uo=x(Ie,e=>e.chainID),wa=x(Ie,e=>e.network.walletConnectBridgeAddress),Sa=x(Ie,e=>e.network.walletConnectV2RelayAddress),va=x(Ie,e=>e.network.walletConnectV2ProjectId),sn=x(Ie,e=>e.network.walletConnectDeepLink),he=x(Ie,e=>e.network),So=x(he,e=>e.apiAddress),ka=x(he,e=>e.explorerAddress),Aa=x(he,e=>e.egldLabel);t();N();var oo=e=>e.loginInfo,Ia=x(oo,e=>e.loginMethod),Fe=x(oo,me,(e,o)=>e.loginMethod!=""&&Boolean(o)),Ea=x(oo,e=>e.walletConnectLogin),La=x(oo,e=>e.ledgerLogin),mn=x(oo,e=>e.walletLogin),Ca=x(oo,e=>e.isLoginSessionInvalid);t();t();var oe=require("@elrondnetwork/erdjs");D();t();function Tt(e){try{return i.Buffer.from(e,"base64").toString()===atob(e)}catch(o){return!1}}function vo(e){var a,m,s;let{data:o}=e,n=Tt(o!=null?o:"")?oe.TransactionPayload.fromEncoded(o):new oe.TransactionPayload(o),r=new oe.Transaction(h({value:e.value.valueOf(),data:n,nonce:e.nonce.valueOf(),receiver:new oe.Address(e.receiver),sender:new oe.Address(e.sender),gasLimit:(a=e.gasLimit.valueOf())!=null?a:Mo,gasPrice:(m=e.gasPrice.valueOf())!=null?m:Yo,chainID:e.chainID.valueOf(),version:new oe.TransactionVersion((s=e.version)!=null?s:Pr)},e.options?{options:new oe.TransactionOptions(e.options)}:{}));return r.applySignature({hex:()=>e.signature||""},new oe.Address(e.sender)),r}t();wo();t();var zp=S(require("bignumber.js"));ie();t();t();t();t();var qp=require("@elrondnetwork/erdjs"),Kp=S(require("bignumber.js"));N();t();var Na=require("@elrondnetwork/erdjs");function jp(e){try{let o=new Na.Address(e);return Boolean(o.bech32())}catch(o){return!1}}function Wo(e){return(e==null?void 0:e.startsWith("erd"))&&e.length===62&&/^\w+$/.test(e)&&jp(e)}t();t();t();var Yp=require("@elrondnetwork/erdjs/out");t();D();t();var Ra=require("@elrondnetwork/erdjs");ie();t();t();t();t();N();t();t();var bt=require("@reduxjs/toolkit"),ri=require("react-redux/es/utils/Subscription");E();t();var Ma=S(require("lodash.throttle"));D();E();qn();_o();Ke();var Zp=[Zo],Jp=(0,Ma.default)(()=>{Ro(No())},5e3),Fa=e=>o=>n=>{var l;if(Zp.includes(n.type))return o(n);let r=e.getState(),a=ve.local.getItem(ae.loginExpiresAt);if(!((l=r==null?void 0:r.loginInfo)==null?void 0:l.isLoggedIn))return o(n);if(a==null)return Ro(No());let s=Date.now();return a-s<0?setTimeout(()=>u(void 0,null,function*(){console.log("session expired");try{e.dispatch(Br())}catch(g){console.error("error logging out",g)}}),1e3):(Jp(),o(n))};t();t();function dn(){return typeof sessionStorage!="undefined"}var oi=dn()?(ja(),Po(za)).default:(Ka(),Po(qa)).default,ni=dn()?(Ya(),Po(Xa)).default:[],ti=dn()?(ei(),Po(Qa)).default:e=>e;_n();var b=(0,bt.configureStore)({reducer:oi,middleware:e=>e({serializableCheck:{ignoredActions:[...ni,en.type,Ze.type],ignoredPaths:["payload.nonce","account.account.nonce","providers.accountProvider"]}}).concat(Fa)}),ai=(0,ri.createSubscription)(b),ii=ti(b),Nu=(0,bt.configureStore)({reducer:ko});t();t();t();t();t();t();var ci="dapp/config",pi="network/config";var ln="accounts";t();t();t();t();j();t();t();j();t();t();t();var ys=S(require("bignumber.js"));D();t();var xs=require("@elrondnetwork/erdjs"),wt=S(require("bignumber.js"));D();t();var hs=S(require("bignumber.js"));t();wt.default.config({ROUNDING_MODE:wt.default.ROUND_FLOOR});t();t();t();D();t();t();t();t();D();t();D();t();var bs=require("@elrondnetwork/erdjs");D();t();var fn=require("@elrondnetwork/erdjs"),vs=require("@elrondnetwork/erdjs"),ks=require("@elrondnetwork/erdjs-network-providers");D();t();t();var ws=S(require("bignumber.js"));D();t();D();t();t();t();t();t();D();t();j();var Is=["reDelegateRewards","claimRewards","unBond"],Es=["wrapEgld","unwrapEgld"],Ls=["unStake"],Cs=["unDelegate"];t();t();t();D();t();var Rs=S(require("bignumber.js"));t();t();t();j();t();var Ms=S(require("bignumber.js"));t();t();t();j();t();j();t();var Bs=S(require("bignumber.js"));D();j();t();j();t();j();t();var Us=S(require("bignumber.js"));t();N();t();N();t();t();var Ws=require("react");j();var LT=["create","localMint","ESDTLocalMint","addQuantity","burn","localBurn","ESDTLocalBurn","wipe","writeLog","signalError"];t();var Gs=S(require("anchorme"));t();var li=require("react");ie();t();t();j();t();var $s=require("@elrondnetwork/erdjs/out"),Vs=S(require("bignumber.js"));j();t();j();t();t();N();var ix=[{label:"Raw",value:"raw"},{label:"Text",value:"text"},{label:"Decimal",value:"decimal"},{label:"Smart",value:"smart"}];t();var ui=require("react");ie();t();t();t();t();t();D();var Tn=e=>e.transactions,Be=x(Tn,e=>e.signedTransactions),zs=x(Tn,e=>e.signTransactionsError),js=x(Tn,e=>e.signTransactionsCancelMessage),xn=e=>o=>Object.entries(o).reduce((n,[r,a])=>(e(a.status)&&(n[r]=a),n),{}),vt=x(Be,xn(ke)),gi=x(Be,xn(Je)),fi=x(Be,xn(Qe)),qs=x(Be,xn(st)),kt=x(Tn,e=>{var o;return(e==null?void 0:e.transactionsToSign)==null?null:A(h({},e.transactionsToSign),{transactions:((o=e==null?void 0:e.transactionsToSign)==null?void 0:o.transactions.map(n=>vo(n)))||[]})}),Ks=x(Be,(e,o)=>o,(e,o)=>o!=null?(e==null?void 0:e[o])||{}:{});t();E();var hi={errorMessage:_t,successMessage:lt,processingMessage:ut},Xs=e=>e.transactionsInfo,Ys=x(Xs,(e,o)=>o,(e,o)=>o!=null&&(e==null?void 0:e[Number(o)])||hi);t();var Ti=e=>e.modals,ty=x(Ti,e=>e.txSubmittedModal),Zs=x(Ti,e=>e.notificationModal);t();var At=e=>e.toasts,iy=x(At,e=>e.customToasts),cy=x(At,e=>e.transactionToasts),py=x(At,e=>e.failTransactionToast);function yi(e){return u(this,null,function*(){let{apiAddress:o,apiTimeout:n}=he(b.getState()),r=e.map(m=>xi.default.post(`${o}/transactions`,m.toPlainObject(),{timeout:parseInt(n)}));return(yield Promise.all(r)).map(({data:m})=>m.txHash)})}t();var bi=S(require("axios"));function yn(e){return u(this,null,function*(){let o=So(b.getState()),n=e.map(a=>a.hash),{data:r}=yield bi.default.get(`${o}/transactions`,{params:{hashes:n.join(","),withScResults:!0}});return e.map(({hash:a,previousStatus:m})=>{let s=r.find(_=>(_==null?void 0:_.txHash)===a);return{hash:a,data:s==null?void 0:s.data,invalidTransaction:s==null,status:s==null?void 0:s.status,results:s==null?void 0:s.results,sender:s==null?void 0:s.sender,receiver:s==null?void 0:s.receiver,previousStatus:m,hasStatusChanged:s&&s.status!==m}})})}t();var Qs=S(require("axios"));t();t();var em=S(require("axios"));t();var om=S(require("axios"));t();t();t();var Si=S(require("axios"));t();t();var bn=e=>{let o=e!=null?e:So(b.getState());return o.endsWith("/")?o.slice(0,-1):o};function vi(e){return u(this,null,function*(){if(!e)return null;let n=`${bn()}/${ln}/${e}`;try{let{data:r}=yield Si.default.get(n);return r}catch(r){console.error("error fetching configuration for ",n)}return null})}t();t();uo();t();var ki=S(require("axios"));function It(e){return u(this,null,function*(){let o=`${e}/${ci}`;try{let{data:r}=yield ki.default.get(o);if(r!=null){let a="egldDenomination";if(a in r){let n=r,{[a]:m,decimals:s}=n,_=Ve(n,[jt(a),"decimals"]);return A(h({},_),{decimals:m,digits:s})}return r}}catch(r){console.error("error fetching configuration for ",o)}return null})}t();var Ai=S(require("axios"));function Ii(){return u(this,null,function*(){var n;let o=`${bn()}/${pi}`;try{let{data:r}=yield Ai.default.get(o);if(r!=null)return(n=r==null?void 0:r.data)==null?void 0:n.config}catch(r){console.error("error fetching configuration for ",o)}return null})}t();uo();t();uo();t();var nm=S(require("axios"));t();var Ge=require("react"),Oi=require("@elrondnetwork/erdjs-wallet-connect-provider");t();t();t();t();var Ei=require("@elrondnetwork/erdjs-extension-provider"),Li=require("@elrondnetwork/erdjs-hw-provider"),Sn=require("@elrondnetwork/erdjs-wallet-connect-provider"),Et=require("@elrondnetwork/erdjs-web-wallet-provider");D();N();var tm="/dapp/init",to=e=>{switch(e==null?void 0:e.constructor){case Et.WalletProvider:return"wallet";case Sn.WalletConnectProvider:return"walletconnect";case Sn.WalletConnectV2Provider:return"walletconnectv2";case Li.HWProvider:return"ledger";case Ei.ExtensionProvider:return"extension";case wn:return"";default:return"extra"}},Lt=e=>new Et.WalletProvider(`${e}${tm}`),Ct=e=>u(void 0,null,function*(){if(!e.isInitialized())throw new Error("Unable to get version. Provider not initialized");let o=e.hwApp,{contractData:n,version:r}=yield o.getAppConfiguration();return{version:r,dataEnabled:n===Nr}}),Ee=e=>`Unable to perform ${e}, Provider not initialized`,wn=class{init(){return Promise.resolve(!1)}login(o){throw new Error(Ee(`login with options: ${o}`))}logout(o){throw new Error(Ee(`logout with options: ${o}`))}getAddress(){throw new Error(Ee("getAddress"))}isInitialized(){return!1}isConnected(){return Promise.resolve(!1)}sendTransaction(o,n){throw new Error(Ee(`sendTransaction with transactions: ${o} options: ${n}`))}signTransaction(o,n){throw new Error(Ee(`signTransaction with transactions: ${o} options: ${n}`))}signTransactions(o,n){throw new Error(Ee(`signTransactions with transactions: ${o} options: ${n}`))}signMessage(o,n){throw new Error(Ee(`signTransactions with ${o} and options ${n}`))}sendCustomMessage({method:o,params:n}){throw new Error(Ee(`sendCustomMessage with method: ${o} params: ${n}`))}sendCustomRequest(o){throw new Error(Ee(`sendSessionEvent with options: ${o}`))}ping(){return Promise.resolve(!1)}},Dt=new wn;var Nt=Dt,Pt=null;function Te(e){Nt=e}function Ci(e){Pt=e}function Di(){Pt!=null&&(Nt=Pt)}function Z(){return Nt||Dt}var vn=()=>{let e=Z(),o=to(e);return{provider:e,providerType:o}};t();var kn=require("react");function Ao(e,o=[]){let n=(0,kn.useRef)(!0);(0,kn.useEffect)(()=>{n.current?n.current=!1:e()},o)}q();t();var Pi=require("react"),Io=require("react-redux");var rm={store:b,subscription:ai},Go=(0,Pi.createContext)(rm),Vb=(0,Io.createStoreHook)(Go),B=(0,Io.createDispatchHook)(Go),y=(0,Io.createSelectorHook)(Go);E();N();t();q();ie();t();function de(){return Fe(b.getState())}t();t();N();t();t();function Ni(){let e=Z();return to(e)}function pe(e){return Ni()===e}function Ue(){let{search:e}=window.location,o=b.getState(),n=Z(),r=me(o),a=Fe(o),m=mn(o);if(!n)throw"provider not initialized";return pe("ledger")&&a?new Promise(s=>{s(r)}):!pe("")&&!pe("wallet")&&!pe("extra")?n.getAddress():new Promise(s=>{if(m!=null){let _=new URLSearchParams(e),l=Object.fromEntries(_);Wo(l.address)&&s(l.address)}a&&s(r),s("")})}t();function We(e){return u(this,null,function*(){return yield vi(e)})}t();E();function Ri(e){b.dispatch(en(e))}t();function Eo(e){let o=b.getState(),n=ya(o);return e?n&&!isNaN(n)?Math.max(n,e.nonce):e.nonce:n}t();t();E();var Mi=()=>u(void 0,null,function*(){try{let e=yield Ue();try{let o=yield We(e);if(o!=null){let n=A(h({},o),{nonce:Eo(o)});return b.dispatch(Ze(n)),n}}catch(o){console.error("Failed getting account ",o)}}catch(e){console.error("Failed getting address ",e)}return null});function An(){return u(this,null,function*(){let e=Z();if(e==null)throw"Provider not initialized";if(!e.isInitialized||e.isInitialized())return Mi();try{if(!e.init)throw"Current provider does not have init() function";return(yield e.init())?Mi():void 0}catch(o){console.error("Failed initializing provider ",o)}})}t();var Fi=require("@elrondnetwork/erdjs");t();q();ie();To();_o();Ke();var am=e=>{ve.local.setItem({key:ae.logoutEvent,data:e,expires:0}),ve.local.removeItem(ae.logoutEvent)};function se(e,o){return u(this,null,function*(){let n=Z(),r=to(n),a=de(),m=r==="wallet";if(!a||!n){Rt(e,o,!1);return}try{let s=yield Ue();am(s)}catch(s){Rt(e,o,!1),console.error("error fetching logout address",s)}m&&Gr(),b.dispatch(H());try{let _=m&&!e?window.location.origin:e;yield n.logout({callbackUrl:_}),Rt(e,o,m)}catch(s){console.error("error logging out",s)}})}function Rt(e,o,n){e&&!n&&(typeof o=="function"?o(e):Ye(e))}yo();var Mt=({logoutRoute:e,callbackRoute:o,token:n,onLoginRedirect:r})=>{let a=B(),m=15e3,[s,_]=(0,Ge.useState)(""),[l,g]=(0,Ge.useState)(""),{provider:f}=vn(),T=y(wa),I=y(sn),k=(0,Ge.useRef)(f),L=(0,Ge.useRef)(!0),F,R=Boolean(l),ne=!R,_e=R?`${I}?wallet-connect=${encodeURIComponent(l)}`:"";(0,Ge.useEffect)(()=>{Ce();let M=setInterval(()=>{Ce()},m);return()=>clearInterval(M)},[f]),Ao(()=>{co()},[n]),Ao(()=>{k.current=f},[f]);let ye=()=>{se(e)},J=()=>{L.current=!1};function Ce(){return u(this,null,function*(){var K,U;if(f==null||!pe("walletconnect")||!Boolean((U=(K=k.current)==null?void 0:K.walletConnector)==null?void 0:U.connected))return;let $={method:"heartbeat",params:{}};try{yield k.current.sendCustomMessage($)}catch(X){console.error("Connection lost",X),ye()}})}function De(){return u(this,null,function*(){var M;try{let $=k.current;if(de()||$==null||!pe("walletconnect"))return;if(!L.current){try{yield(M=k.current)==null?void 0:M.logout()}catch(z){console.warn("Unable to logout")}return}let U=yield $.getAddress();if(!U){console.warn("Login cancelled.");return}let X=yield $.getSignature(),P=Boolean(X),W={address:U,loginMethod:"walletconnect"},V={logoutRoute:e,loginType:"walletConnect",callbackRoute:o!=null?o:window.location.href};P?(a(fo(V)),a(ho(X))):a(fo(V)),a(Q(W)),$.walletConnector.on("heartbeat",()=>{clearInterval(F),F=setInterval(()=>{console.log("Maiar Wallet Connection Lost"),ye(),clearInterval(F)},15e4)}),xo(o,r)}catch($){_("Invalid address"),console.error($)}})}function $e(M=!0){return u(this,null,function*(){var X,P;let $=M&&!l;if(!T||((P=(X=k==null?void 0:k.current)==null?void 0:X.isInitialized)==null?void 0:P.call(X))&&!$)return;let K={onClientLogin:De,onClientLogout:ye},U=new Oi.WalletConnectProvider(T,K);yield U.init(),L.current=!0,Te(U),k.current=U,M&&co()})}function co(){return u(this,null,function*(){var U;if(!T)return;let M=yield(U=k.current)==null?void 0:U.login();if(!Boolean(M))return;if(!n){g(M);return}let K=`${M}&token=${n}`;g(K),a(Ne({loginToken:n}))})}let v=Boolean(s),C=de();return[$e,{error:s,loginFailed:v,isLoading:ne&&!v,isLoggedIn:C&&!v},{uriDeepLink:_e,walletConnectUri:l,cancelLogin:J}]};t();var ro=require("react"),Bi=require("@elrondnetwork/erdjs-wallet-connect-provider");q();E();N();yo();var Ft=({callbackRoute:e,logoutRoute:o,token:n,onLoginRedirect:r})=>{let a=B(),[m,s]=(0,ro.useState)(""),[_,l]=(0,ro.useState)(""),[g,f]=(0,ro.useState)([]),T=Z(),I=y(Sa),k=y(va),L=y(Uo),F=y(sn),R=(0,ro.useRef)(T),ne=(0,ro.useRef)(!0),_e=Boolean(_),ye=["erd_cancelAction"],J=!_e,Ce=_e?`${F}?wallet-connect=${encodeURIComponent(_)}`:"";Ao(()=>{K()},[n]),Ao(()=>{R.current=T},[T]);let De=()=>{se(o)},$e=P=>{console.log("wc2 session event: ",P)},co=()=>{ne.current=!1};function v(){return u(this,null,function*(){var P;try{let W=R.current;if(de()||W==null||!pe("walletconnectv2"))return;if(!ne.current){try{yield(P=R.current)==null?void 0:P.logout()}catch(Om){console.warn("Unable to logout")}return}let z=yield W.getAddress();if(!z){console.warn("Login cancelled.");return}let te=yield W.getSignature(),Pe=Boolean(te),Ac={address:z,loginMethod:"walletconnectv2"},Vt={logoutRoute:o,loginType:"walletconnectv2",callbackRoute:e!=null?e:window.location.href};Pe?(a(fo(Vt)),a(ho(te))):a(fo(Vt)),a(Q(Ac)),xo(e,r)}catch(W){s("Invalid address"),console.error(W)}})}function C(P=!0){return u(this,null,function*(){var te,Pe;let W=P&&!_;if(!k||!I||((Pe=(te=R==null?void 0:R.current)==null?void 0:te.isInitialized)==null?void 0:Pe.call(te))&&!W){s("Invalid WalletConnect setup");return}let V={onClientLogin:v,onClientLogout:De,onClientEvent:$e},z=new Bi.WalletConnectV2Provider(V,L,I,k);yield z.init(),ne.current=!0,Te(z),f(z.pairings),R.current=z,P&&K()})}function M(P){return u(this,null,function*(){var W,V,z;if(!I||!k){s("Invalid WalletConnect setup");return}if(!P||!P.topic){s("Expired connection");return}try{let{approval:te}=yield(W=R.current)==null?void 0:W.connect({topic:P.topic,methods:ye});n&&a(Ne({loginToken:n}));try{yield(V=R.current)==null?void 0:V.login({approval:te,token:n})}catch(Pe){console.warn("User rejected existing connection proposal",Pe),yield C()}}catch(te){console.error("Unable to connect to existing session",te),s("Unable to connect to existing session")}finally{f((z=R.current)==null?void 0:z.pairings)}})}function $(P){return u(this,null,function*(){var W,V;try{P&&(yield(W=R.current)==null?void 0:W.logout({topic:P}))}catch(z){console.error("Unable to remove existing pairing",z),s("Unable to remove existing pairing")}finally{f((V=R.current)==null?void 0:V.pairings)}})}function K(){return u(this,null,function*(){var P,W;if(!I||!k){s("Invalid WalletConnect setup");return}try{let{uri:V,approval:z}=yield(P=R.current)==null?void 0:P.connect({methods:ye});if(!Boolean(V))return;l(V),n&&a(Ne({loginToken:n}));try{yield(W=R.current)==null?void 0:W.login({approval:z,token:n})}catch(Pe){console.warn("User rejected connection proposal",Pe),yield C()}}catch(V){console.error("Unable to connect",V)}})}let U=Boolean(m),X=de();return[C,{error:m,loginFailed:U,isLoading:J&&!U,isLoggedIn:X&&!U},{uriDeepLink:Ce,walletConnectUri:_,cancelLogin:co,connectExisting:M,removeExistingPairing:$,wcPairings:g}]};q();E();N();function Gi(){let e=y(he),o=y(Ea),n=y(Ia),r=y(mn),a=y(me),m=y(ft),s=y(La),_=y(Fe),[l,g]=(0,ao.useState)(),f=B(),{callbackRoute:T,logoutRoute:I}=o||{callbackRoute:"",logoutRoute:""},[k]=Mt({callbackRoute:T,logoutRoute:I}),[L]=Ft({callbackRoute:T,logoutRoute:I});(0,ao.useEffect)(()=>{F()},[e]),(0,ao.useEffect)(()=>{co()},[n]),(0,ao.useEffect)(()=>{ne()},[a,_]),(0,ao.useEffect)(()=>{R()},[m,_,l]);function F(){return u(this,null,function*(){try{let v=yield Ii();v&&f(qr(v.erd_chain_id))}catch(v){console.error("failed refreshing chainId ",v)}})}function R(){m==null&&s!=null&&l&&f(it({index:s.index,address:a,hasContractDataEnabled:l.dataEnabled,version:l.version}))}function ne(){return u(this,null,function*(){if(f(ct(!0)),a&&_)try{let v=yield We(a);v&&f(Ze(A(h({},v),{address:a,nonce:v.nonce.valueOf()})))}catch(v){f(Qr("Failed getting account")),console.error("Failed getting account ",v)}f(ct(!1))})}function _e(){return u(this,null,function*(){let v=Lt(e.walletAddress);if(Te(v),r!=null){try{let C=yield Ue();if(C){f(Q({address:C,loginMethod:"wallet"}));let M=yield We(C);M&&f(Ze(A(h({},M),{nonce:Eo(M)})))}ye()}catch(C){console.error("Failed authenticating wallet user ",C)}f(Jn(null))}})}function ye(){var U;let v={};if((U=window==null?void 0:window.location)!=null&&U.search){let P=new URLSearchParams(window.location.search);v=Object.fromEntries(P)}let X=v,{signature:C,loginToken:M,address:$}=X,K=Ve(X,["signature","loginToken","address"]);C&&f(ho(C)),J(K)}function J(v){let C=new URLSearchParams(v).toString(),{pathname:M}=window.location,$=C?`?${C}`:"",K=M?`${M}${$}`:"./";window.history.replaceState({},document.title,K)}function Ce(){return u(this,null,function*(){let v=new Wi.HWProvider,C=v.isInitialized();if(C||(C=yield v.init()),!C&&_){console.warn("Could not initialise ledger app"),se();return}return(s==null?void 0:s.index)!=null&&(v.addressIndex=s.index),v})}function De(){return u(this,null,function*(){try{let v=yield Ce();if(!v)return;let C=yield Ct(v);Te(v),g(C)}catch(v){console.error("Could not initialise ledger app",v),se()}})}function $e(){return u(this,null,function*(){try{let v=yield Ue(),C=Ui.ExtensionProvider.getInstance().setAddress(v);(yield C.init())?Te(C):console.error("Could not initialise extension, make sure Elrond wallet extension is installed.")}catch(v){console.error("Unable to login to ExtensionProvider",v)}})}function co(){if(n!=null)switch(n){case"ledger":{De();break}case"walletconnect":{k(!1);break}case"walletconnectv2":{L(!1);break}case"extension":{$e();break}case"extra":{Di();break}case"wallet":case"":{_e();break}default:return}}return null}t();var xe=S(require("react")),Vi=require("@elrondnetwork/erdjs/out");uo();t();var $o=()=>y(xa);Jo();t();var $i=S(require("axios"));E();function Ot(){return u(this,null,function*(){let e=b.getState(),o=So(e),n=me(e),r=ba(e);try{if(r==null&&n){let{data:a}=yield $i.default.get(`${o}/accounts/${n}`);return b.dispatch(Jr(a.shard)),a.shard}else return r}catch(a){return console.error(a),null}})}function Hi({customNetworkConfig:e={},children:o,environment:n}){let[r,a]=(0,xe.useState)(!1),m=$o(),s=y(Ca),{address:_,publicKey:l}=m,g=B();function f(){return u(this,null,function*(){let I=!(e!=null&&e.skipFetchFromServer),k=e==null?void 0:e.apiAddress,L=Xe[n]||{},F=h(h({},L),e);if(I){let R=L==null?void 0:L.apiAddress,ne=yield It(k||R);if(ne!=null){let _e=h(h(h({},L),ne),e);g(tt(_e));return}}g(tt(F))})}function T(){return u(this,null,function*(){yield f(),a(!0),Ot()})}return(0,xe.useEffect)(()=>{!_||Ot()},[_]),(0,xe.useEffect)(()=>{_&&new Vi.Address(_).hex()!==l&&se()},[_,l]),(0,xe.useEffect)(()=>{T()},[e,n]),(0,xe.useEffect)(()=>{s&&se()},[s]),r?xe.default.createElement(xe.default.Fragment,null,o):null}t();var Lo=S(require("react"));t();var uc=require("react");t();t();var Bt=()=>y(Ie);t();E();t();t();t();var cm=require("react"),pm=require("@elrondnetwork/erdjs-web-wallet-provider"),sm=S(require("qs"));D();E();N();t();var Wt=require("react"),lm=require("@elrondnetwork/erdjs-extension-provider");D();E();N();t();t();t();t();t();E();t();t();t();t();t();_o();To();Vn();t();t();t();function ji(e){return new Promise(o=>{setTimeout(()=>o(),e)})}var qi=(e,o,n,r=0)=>u(void 0,null,function*(){try{return yield e(...n)}catch(a){return r<o.retries?((o==null?void 0:o.delay)!=null&&(yield ji(o.delay)),qi(e,o,n,r+1)):null}}),Ki=(e,o={retries:5,delay:500})=>(...n)=>u(void 0,null,function*(){return qi(e,o,n)});t();t();var Xi=S(require("axios"));function Ut(e){return u(this,null,function*(){try{let{data:o}=yield Xi.default.get(`${e}/websocket/config`);return`wss://${o.url}`}catch(o){throw console.error(o),new Error("Can not get websocket url")}})}t();t();var mm=S(require("platform"));t();t();t();var um=S(require("axios")),gm=S(require("swr"));t();t();var Zi=require("react");t();var Yi=require("react");ot();E();ie();To();t();lo();E();t();t();t();t();t();function Ji(){let e=y(vt),o=Object.entries(e),n=(o==null?void 0:o.length)>0;return{pendingTransactions:e,pendingTransactionsArray:o,hasPendingTransactions:n}}wo();t();E();N();t();wo();function Qi(e,o){return e.reduce((r,{status:a,hash:m})=>(m!=null&&!o.includes(m)&&ke(a)&&r.push({hash:m,previousStatus:a}),r),[])}t();E();N();function ec({results:e,hash:o,sessionId:n}){let r=e==null?void 0:e.find(a=>(a==null?void 0:a.returnMessage)!=="");b.dispatch(Oo({transactionHash:o,sessionId:n,status:"fail",errorMessage:r==null?void 0:r.returnMessage})),b.dispatch(eo({sessionId:n,status:"fail",errorMessage:r==null?void 0:r.returnMessage}))}t();E();ie();function Gt(e){b.dispatch(eo({sessionId:e,status:"timedOut"}))}var In={},$t=[];function hm({serverTransaction:e,sessionId:o,customTransactionInformation:n,shouldRefreshBalance:r}){let{hash:a,status:m,results:s,invalidTransaction:_,hasStatusChanged:l}=e;try{if($t.includes(a))return;if(In[a]>30){Gt(o);return}if(_||ke(m)){In[a]=In[a]?In[a]+1:1;return}l&&(Je(m)&&(n==null?void 0:n.completedTransactionsDelay)!=null?($t.push(a),setTimeout(()=>b.dispatch(Oo({sessionId:o,status:"success",transactionHash:a})),n==null?void 0:n.completedTransactionsDelay)):b.dispatch(Oo({sessionId:o,status:m,transactionHash:a}))),r||An(),Qe(m)&&ec({sessionId:o,hash:a,results:s})}catch(g){console.error(g),Gt(o)}}function oc(m){return u(this,arguments,function*({sessionId:e,transactionBatch:{transactions:o,customTransactionInformation:n},getTransactionsByHash:r=yn,shouldRefreshBalance:a}){try{if(o==null)return;let s=Qi(o,$t),_=yield r(s);for(let l of _)hm({serverTransaction:l,sessionId:e,customTransactionInformation:n,shouldRefreshBalance:a})}catch(s){console.error(s)}})}function nc(){let{pendingTransactionsArray:e}=Ji();function o(n){return u(this,null,function*(){let r=e.filter(([a,m])=>a!=null&&ke(m.status));if(r.length>0)for(let[a,m]of r)yield oc(h({sessionId:a,transactionBatch:m},n));n.shouldRefreshBalance&&An()})}return o}t();var vm=require("react");t();t();t();E();t();t();var bm=S(require("bignumber.js"));t();var Tm=`.dapp-core-component__main__link-style {
  color: #1392ff;
}

.dapp-core-component__main__link-second-style {
  color: #9dabbd;
}
.dapp-core-component__main__link-second-style:hover {
  color: #1392ff;
}
.dapp-core-component__main__link-second-style:hover svg {
  color: #1392ff;
}

.dapp-core-component__main__wrapper {
  min-height: 100vh;
}

.dapp-core-component__main__opacity-6 {
  opacity: 0.6;
}

.dapp-core-component__main__qr-code-svg-container {
  width: 15rem;
  height: 15rem;
}

.dapp-core-component__main__pair-list {
  max-height: 20rem;
  overflow-y: auto;
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__pair-image {
  width: 4rem;
  overflow: hidden;
  height: auto;
  min-width: 2rem;
}

.dapp-core-component__main__pair-remove {
  padding: 0 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

.dapp-core-component__main__dapp-icon {
  padding: 5px;
  border-radius: 50%;
  background-color: #fafafa;
  width: 4.6428571429rem;
  height: 4.6428571429rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}
.dapp-core-component__main__dapp-icon.dapp-core-component__main__icon-medium {
  width: 80px;
  height: 80px;
}

.dapp-core-component__main__centering {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dapp-core-component__main__border-n {
  border: none !important;
}

.dapp-core-component__main__t-shadow {
  text-shadow: 0px 4px 7px rgba(0, 0, 0, 0.3);
}

.dapp-core-component__main__btn {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.04) !important;
}

.dapp-core-component__main__media {
  display: flex;
  align-items: flex-start;
}

.dapp-core-component__main__media-body {
  flex: 1;
}

.dapp-core-component__main__connect-btns .dapp-core-component__main__btn {
  display: flex;
  height: 48px;
  justify-content: center;
  align-items: center;
  color: #fff !important;
  background-color: #1392ff;
  padding: 15px 20px 15px 16px;
  border-radius: 6px;
  gap: 8px;
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__btn:hover, .dapp-core-component__main__connect-btns .dapp-core-component__main__btn:active {
  background-color: #1392ff;
  background-image: linear-gradient(rgba(0, 0, 0, 0.1) 0 0);
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__info {
  color: #1392ff;
  margin-left: 48px;
}

.dapp-core-component__main__btn {
  transition: none !important;
}

.dapp-core-component__main__trust-badge {
  width: 20px;
  height: 20px;
}

/* Navbar
  -------------------------------------------------- */
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name {
  display: flex;
  align-items: center;
  font-size: 1.7142857143rem;
  font-weight: 400;
  color: #9ba5b4;
  margin-left: 1.6428571429rem;
}
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name:before {
  content: "";
  display: block;
  width: 1px;
  height: 3rem;
  margin-right: 1.7142857143rem;
}

/* Dashboard
  -------------------------------------------------- */
.dapp-core-component__main__action-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  margin: 0 7px;
}
.dapp-core-component__main__action-btn button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #fff;
}
.dapp-core-component__main__action-btn button:hover {
  opacity: 0.9;
}

.dapp-core-component__main__transactions .dapp-core-component__main__transaction-icon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table td {
  white-space: nowrap;
  vertical-align: middle;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table tr:last-of-type td {
  border-bottom: 1px solid #dee2e6;
}

/* Loader, Transaction Success/Fail
  -------------------------------------------------- */
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div {
  position: absolute;
  top: 27px;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.1);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.dapp-core-component__main__page-state .dapp-core-component__main__bg-blue .dapp-core-component__main__lds-ellipsis div {
  background: rgba(255, 255, 255, 0.4);
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(1) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis1 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(2) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(3) {
  left: 26px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(4) {
  left: 45px;
  animation: dapp-core-component__main__lds-ellipsis3 0.6s infinite;
}
@keyframes dapp-core-component__main__lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(19px, 0);
  }
}

.dapp-core-component__main__btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
}
.dapp-core-component__main__btn .dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn .dapp-core-component__main__disabled {
  opacity: 0.65;
  cursor: default;
}
.dapp-core-component__main__btn:hover {
  color: #212529;
  text-decoration: none;
}
.dapp-core-component__main__btn:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn:disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn-primary {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary:hover {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-primary:focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary:hover {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-secondary:focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-warning {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning:hover {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning:focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-dark {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark:hover {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
}
.dapp-core-component__main__btn-dark:focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-link {
  font-weight: 400;
  color: #007bff;
  text-decoration: none;
}
.dapp-core-component__main__btn-link:hover {
  color: #0056b3;
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:focus, .dapp-core-component__main__btn-link.dapp-core-component__main__focus {
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:disabled, .dapp-core-component__main__btn-link.dapp-core-component__main__disabled {
  color: #6c757d;
  pointer-events: none;
}

.dapp-core-component__main__btn-light {
  color: #212529;
  background-color: #fafafa;
  border-color: #fafafa;
}
.dapp-core-component__main__btn-light .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}
.dapp-core-component__main__btn-light .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #f8f9fa;
  border-color: #f8f9fa;
}
.dapp-core-component__main__btn-light:hover {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
}
.dapp-core-component__main__btn-light:focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .dapp-core-component__main__badge {
    transition: none;
  }
}
a.dapp-core-component__main__badge:hover, a.dapp-core-component__main__badge:focus {
  text-decoration: none;
}

.dapp-core-component__main__badge:empty {
  display: none;
}

.dapp-core-component__main__btn .dapp-core-component__main__badge {
  position: relative;
  top: -1px;
}

.dapp-core-component__main__badge-pill {
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.dapp-core-component__main__badge-primary {
  color: #fff;
  background-color: #007bff;
}

a.dapp-core-component__main__badge-primary:hover, a.dapp-core-component__main__badge-primary:focus {
  color: #fff;
  background-color: #0062cc;
}

a.dapp-core-component__main__badge-primary:focus, a.dapp-core-component__main__badge-primary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.5);
}

.dapp-core-component__main__badge-secondary {
  color: #fff;
  background-color: #6c757d;
}

a.dapp-core-component__main__badge-secondary:hover, a.dapp-core-component__main__badge-secondary:focus {
  color: #fff;
  background-color: #545b62;
}

a.dapp-core-component__main__badge-secondary:focus, a.dapp-core-component__main__badge-secondary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108, 117, 125, 0.5);
}

.dapp-core-component__main__badge-success {
  color: #fff;
  background-color: #28a745;
}

a.dapp-core-component__main__badge-success:hover, a.dapp-core-component__main__badge-success:focus {
  color: #fff;
  background-color: #1e7e34;
}

a.dapp-core-component__main__badge-success:focus, a.dapp-core-component__main__badge-success.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
}

.dapp-core-component__main__badge-info {
  color: #fff;
  background-color: #17a2b8;
}

a.dapp-core-component__main__badge-info:hover, a.dapp-core-component__main__badge-info:focus {
  color: #fff;
  background-color: #117a8b;
}

a.dapp-core-component__main__badge-info:focus, a.dapp-core-component__main__badge-info.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(23, 162, 184, 0.5);
}

.dapp-core-component__main__badge-warning {
  color: #212529;
  background-color: #ffc107;
}

a.dapp-core-component__main__badge-warning:hover, a.dapp-core-component__main__badge-warning:focus {
  color: #212529;
  background-color: #d39e00;
}

a.dapp-core-component__main__badge-warning:focus, a.dapp-core-component__main__badge-warning.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 193, 7, 0.5);
}

.dapp-core-component__main__badge-danger {
  color: #fff;
  background-color: #dc3545;
}

a.dapp-core-component__main__badge-danger:hover, a.dapp-core-component__main__badge-danger:focus {
  color: #fff;
  background-color: #bd2130;
}

a.dapp-core-component__main__badge-danger:focus, a.dapp-core-component__main__badge-danger.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(220, 53, 69, 0.5);
}

.dapp-core-component__main__badge-light {
  color: #212529;
  background-color: #f8f9fa;
}

a.dapp-core-component__main__badge-light:hover, a.dapp-core-component__main__badge-light:focus {
  color: #212529;
  background-color: #dae0e5;
}

a.dapp-core-component__main__badge-light:focus, a.dapp-core-component__main__badge-light.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(248, 249, 250, 0.5);
}

.dapp-core-component__main__badge-dark {
  color: #fff;
  background-color: #343a40;
}

a.dapp-core-component__main__badge-dark:hover, a.dapp-core-component__main__badge-dark:focus {
  color: #fff;
  background-color: #1d2124;
}

a.dapp-core-component__main__badge-dark:focus, a.dapp-core-component__main__badge-dark.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}

.dapp-core-component__main__card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 5px;
}
.dapp-core-component__main__card > hr {
  margin-right: 0;
  margin-left: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group {
  border-top: inherit;
  border-bottom: inherit;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:first-child {
  border-top-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:last-child {
  border-bottom-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__card-header + .dapp-core-component__main__list-group,
.dapp-core-component__main__card > .dapp-core-component__main__list-group + .dapp-core-component__main__card-footer {
  border-top: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header {
  border: 0;
  padding: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header:first-child {
  border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item {
  border-bottom: 1px solid #dee2e6;
  padding: 1.1rem 1rem;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item h6,
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item .dapp-core-component__main__h6 {
  line-height: 2.5rem;
  letter-spacing: unset;
  margin: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item.dapp-core-component__main__compact {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fafafa;
}

.dapp-core-component__main__card-body {
  flex: 1 1 auto;
  min-height: 1px;
  padding: 1.25rem;
}

.dapp-core-component__main__card-title {
  margin-bottom: 0.75rem;
}

.dapp-core-component__main__card-subtitle {
  margin-top: -0.375rem;
  margin-bottom: 0;
}

.dapp-core-component__main__card-text:last-child {
  margin-bottom: 0;
}

.dapp-core-component__main__card-link:hover {
  text-decoration: none;
}
.dapp-core-component__main__card-link + .dapp-core-component__main__card-link {
  margin-left: 1.25rem;
}

.dapp-core-component__main__card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: #fff;
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__card-footer {
  padding: 0.75rem 1.25rem;
  background-color: #fff;
  border-top: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__text {
  color: #9dabbd;
}

.dapp-core-component__main__text-white {
  color: #fff;
}

.dapp-core-component__main__text-monospace {
  font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-size-xl .dapp-core-component__main__trim {
  max-width: 13rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim {
  max-width: 10rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-only-sm .dapp-core-component__main__trim {
  max-width: none;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-left {
  text-align: left;
}

.dapp-core-component__main__text-right {
  text-align: right;
}

.dapp-core-component__main__text-center {
  text-align: center;
}

.dapp-core-component__main__text-primary {
  color: #1b46c2;
}

.dapp-core-component__main__text-secondary {
  color: #6c757d;
}

.dapp-core-component__main__text-primary-highlight {
  color: #1b46c2;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-success {
  color: #28a745;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-body {
  color: #212529;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-black-50 {
  color: rgba(0, 0, 0, 0.5);
}

.dapp-core-component__main__text-white-50 {
  color: rgba(255, 255, 255, 0.5);
}

@media (min-width: 992px) {
  .dapp-core-component__main__text-lg-right {
    text-align: right;
  }
}

.dapp-core-component__main__text-hide {
  font: 0/0 a;
  color: rgba(0, 0, 0, 0);
  text-shadow: none;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
}

.dapp-core-component__main__text-decoration-none {
  text-decoration: none;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__text-reset {
  color: inherit;
}

.dapp-core-component__main__text-lowercase {
  text-transform: lowercase;
}

.dapp-core-component__main__text-uppercase {
  text-transform: uppercase;
}

.dapp-core-component__main__text-capitalize {
  text-transform: capitalize;
}

.dapp-core-component__main__text-danger {
  color: #d9534f;
}

.dapp-core-component__main__text-warning {
  color: #f0ad4e;
}

.dapp-core-component__main__trim-fs-sm {
  color: inherit;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__m-0 {
  margin: 0;
}

.dapp-core-component__main__mt-0 {
  margin-top: 0;
}

.dapp-core-component__main__my-0 {
  margin-top: 0;
  margin-bottom: 0;
}

.dapp-core-component__main__mr-0 {
  margin-right: 0;
}

.dapp-core-component__main__mx-0 {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__mb-0 {
  margin-bottom: 0;
}

.dapp-core-component__main__my-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.dapp-core-component__main__ml-0 {
  margin-left: 0;
}

.dapp-core-component__main__mx-0 {
  margin-left: 0;
  margin-right: 0;
}

.dapp-core-component__main__m-1 {
  margin: 0.25rem;
}

.dapp-core-component__main__mt-1 {
  margin-top: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__mr-1 {
  margin-right: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.dapp-core-component__main__mb-1 {
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-bottom: 0.25rem;
  margin-top: 0.25rem;
}

.dapp-core-component__main__ml-1 {
  margin-left: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-right: 0.25rem;
  margin-left: 0.25rem;
}

.dapp-core-component__main__m-2 {
  margin: 0.5rem;
}

.dapp-core-component__main__mt-2 {
  margin-top: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-bottom: 0.5rem;
  margin-top: 0.5rem;
}

.dapp-core-component__main__mr-2 {
  margin-right: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__mb-2 {
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__ml-2 {
  margin-left: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__m-3 {
  margin: 1rem;
}

.dapp-core-component__main__mt-3,
.dapp-core-component__main__my-3 {
  margin-top: 1rem;
}

.dapp-core-component__main__mr-3 {
  margin-right: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__mb-3 {
  margin-bottom: 1rem;
}

.dapp-core-component__main__my-3 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.dapp-core-component__main__ml-3 {
  margin-left: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__m-4 {
  margin: 1.5rem;
}

.dapp-core-component__main__mt-4 {
  margin-top: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__mr-4 {
  margin-right: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-right: 1.5rem;
  margin-left: 1.5rem;
}

.dapp-core-component__main__mb-4 {
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.dapp-core-component__main__ml-4 {
  margin-left: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}

.dapp-core-component__main__m-5 {
  margin: 3rem;
}

.dapp-core-component__main__mt-5 {
  margin-top: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__mr-5 {
  margin-right: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-right: 3rem;
  margin-left: 3rem;
}

.dapp-core-component__main__mb-5 {
  margin-bottom: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__ml-5 {
  margin-left: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-left: 3rem;
  margin-right: 3rem;
}

.dapp-core-component__main__p-0 {
  padding: 0;
}

.dapp-core-component__main__pt-0 {
  padding-top: 0;
}

.dapp-core-component__main__py-0 {
  padding-top: 0;
  padding-bottom: 0;
}

.dapp-core-component__main__pr-0 {
  padding-right: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__pb-0 {
  padding-bottom: 0;
}

.dapp-core-component__main__py-0 {
  padding-bottom: 0;
  padding-top: 0;
}

.dapp-core-component__main__pl-0 {
  padding-left: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__p-1 {
  padding: 0.25rem;
}

.dapp-core-component__main__pt-1 {
  padding-top: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pr-1 {
  padding-right: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-right: 0.25rem;
  padding-left: 0.25rem;
}

.dapp-core-component__main__pb-1 {
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pl-1 {
  padding-left: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

.dapp-core-component__main__p-2 {
  padding: 0.5rem;
}

.dapp-core-component__main__pt-2 {
  padding-top: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.dapp-core-component__main__pr-2 {
  padding-right: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__pb-2 {
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__pl-2 {
  padding-left: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__p-3 {
  padding: 1rem;
}

.dapp-core-component__main__pt-3 {
  padding-top: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.dapp-core-component__main__pr-3 {
  padding-right: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__pb-3 {
  padding-bottom: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.dapp-core-component__main__pl-3 {
  padding-left: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__p-4 {
  padding: 1.5rem;
}

.dapp-core-component__main__pt-4 {
  padding-top: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pr-4 {
  padding-right: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.dapp-core-component__main__pb-4 {
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pl-4 {
  padding-left: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
}

.dapp-core-component__main__p-5 {
  padding: 3rem;
}

.dapp-core-component__main__pt-5 {
  padding-top: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-bottom: 3rem;
  padding-top: 3rem;
}

.dapp-core-component__main__pr-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__pb-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.dapp-core-component__main__pl-5 {
  padding-left: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__m-n1 {
  margin: -0.25rem;
}

.dapp-core-component__main__mt-n1 {
  margin-top: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__mr-n1 {
  margin-right: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__mb-n1 {
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-bottom: -0.25rem;
  margin-top: -0.25rem;
}

.dapp-core-component__main__ml-n1 {
  margin-left: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__m-n2 {
  margin: -0.5rem;
}

.dapp-core-component__main__mt-n2 {
  margin-top: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-top: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mr-n2 {
  margin-right: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mb-n2 {
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-bottom: -0.5rem;
  margin-top: -0.5rem;
}

.dapp-core-component__main__ml-n2 {
  margin-left: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.dapp-core-component__main__m-n3 {
  margin: -1rem;
}

.dapp-core-component__main__mt-n3 {
  margin-top: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-top: -1rem;
  margin-bottom: -1rem;
}

.dapp-core-component__main__mr-n3 {
  margin-right: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-right: -1rem;
  margin-left: -1rem;
}

.dapp-core-component__main__mb-n3 {
  margin-bottom: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-bottom: -1rem;
  margin-top: -1rem;
}

.dapp-core-component__main__ml-n3 {
  margin-left: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-left: -1rem;
  margin-right: -1rem;
}

.dapp-core-component__main__m-n4 {
  margin: -1.5rem;
}

.dapp-core-component__main__mt-n4 {
  margin-top: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-top: -1.5rem;
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__mr-n4 {
  margin-right: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__mb-n4 {
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-bottom: -1.5rem;
  margin-top: -1.5rem;
}

.dapp-core-component__main__ml-n4 {
  margin-left: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__m-n5 {
  margin: -3rem;
}

.dapp-core-component__main__mt-n5 {
  margin-top: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-bottom: -3rem;
  margin-top: -3rem;
}

.dapp-core-component__main__mr-n5 {
  margin-right: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-left: -3rem;
  margin-right: -3rem;
}

.dapp-core-component__main__mb-n5 {
  margin-bottom: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-top: -3rem;
  margin-bottom: -3rem;
}

.dapp-core-component__main__ml-n5 {
  margin-left: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-right: -3rem;
  margin-left: -3rem;
}

.dapp-core-component__main__m-auto {
  margin: auto;
}

.dapp-core-component__main__mt-auto {
  margin-top: auto;
}

.dapp-core-component__main__my-auto {
  margin-bottom: auto;
  margin-top: auto;
}

.dapp-core-component__main__mr-auto {
  margin-right: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__mb-auto {
  margin-bottom: auto;
}

.dapp-core-component__main__my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.dapp-core-component__main__ml-auto {
  margin-left: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-1 {
    margin-right: 0.25rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-4 {
    margin-right: 1.5rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mx-lg-4 {
    margin-left: 1.5rem;
    margin-right: 1.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__mb-xl-2 {
    margin-bottom: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__pr-sm-2 {
    padding-right: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__px-sm-2 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__pr-xl-0,
.dapp-core-component__main__px-xl-0 {
    padding-right: 0;
  }
}

.dapp-core-component__main__col {
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

.dapp-core-component__main__col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__col-xl,
.dapp-core-component__main__col-xl-auto,
.dapp-core-component__main__col-xl-12,
.dapp-core-component__main__col-xl-11,
.dapp-core-component__main__col-xl-10,
.dapp-core-component__main__col-xl-9,
.dapp-core-component__main__col-xl-8,
.dapp-core-component__main__col-xl-7,
.dapp-core-component__main__col-xl-6,
.dapp-core-component__main__col-xl-5,
.dapp-core-component__main__col-xl-4,
.dapp-core-component__main__col-xl-3,
.dapp-core-component__main__col-xl-2,
.dapp-core-component__main__col-xl-1,
.dapp-core-component__main__col-lg,
.dapp-core-component__main__col-lg-auto,
.dapp-core-component__main__col-lg-12,
.dapp-core-component__main__col-lg-11,
.dapp-core-component__main__col-lg-10,
.dapp-core-component__main__col-lg-9,
.dapp-core-component__main__col-lg-8,
.dapp-core-component__main__col-lg-7,
.dapp-core-component__main__col-lg-6,
.dapp-core-component__main__col-lg-5,
.dapp-core-component__main__col-lg-4,
.dapp-core-component__main__col-lg-3,
.dapp-core-component__main__col-lg-2,
.dapp-core-component__main__col-lg-1,
.dapp-core-component__main__col-md,
.dapp-core-component__main__col-md-auto,
.dapp-core-component__main__col-md-12,
.dapp-core-component__main__col-md-11,
.dapp-core-component__main__col-md-10,
.dapp-core-component__main__col-md-9,
.dapp-core-component__main__col-md-8,
.dapp-core-component__main__col-md-7,
.dapp-core-component__main__col-md-6,
.dapp-core-component__main__col-md-5,
.dapp-core-component__main__col-md-4,
.dapp-core-component__main__col-md-3,
.dapp-core-component__main__col-md-2,
.dapp-core-component__main__col-md-1,
.dapp-core-component__main__col-sm,
.dapp-core-component__main__col-sm-auto,
.dapp-core-component__main__col-sm-12,
.dapp-core-component__main__col-sm-11,
.dapp-core-component__main__col-sm-10,
.dapp-core-component__main__col-sm-9,
.dapp-core-component__main__col-sm-8,
.dapp-core-component__main__col-sm-7,
.dapp-core-component__main__col-sm-6,
.dapp-core-component__main__col-sm-5,
.dapp-core-component__main__col-sm-4,
.dapp-core-component__main__col-sm-3,
.dapp-core-component__main__col-sm-2,
.dapp-core-component__main__col-sm-1,
.dapp-core-component__main__col,
.dapp-core-component__main__col-auto,
.dapp-core-component__main__col-12,
.dapp-core-component__main__col-11,
.dapp-core-component__main__col-10,
.dapp-core-component__main__col-9,
.dapp-core-component__main__col-8,
.dapp-core-component__main__col-7,
.dapp-core-component__main__col-6,
.dapp-core-component__main__col-5,
.dapp-core-component__main__col-4,
.dapp-core-component__main__col-3,
.dapp-core-component__main__col-2,
.dapp-core-component__main__col-1 {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
}

.dapp-core-component__main__col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

@media (min-width: 576px) {
  .dapp-core-component__main__col-sm-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-sm-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-sm-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-sm-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-sm-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-sm-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-sm-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-sm-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-sm-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-sm-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-sm-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-sm-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 768px) {
  .dapp-core-component__main__col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__col-xl-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-xl-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-xl-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-xl-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-xl-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-xl-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-xl-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-xl-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-xl-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-xl-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-xl-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-xl-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
.dapp-core-component__main__flex-row {
  flex-direction: row;
}

.dapp-core-component__main__flex-column {
  flex-direction: column;
}

.dapp-core-component__main__flex-row-reverse {
  flex-direction: row-reverse;
}

.dapp-core-component__main__flex-column-reverse {
  flex-direction: column-reverse;
}

.dapp-core-component__main__flex-wrap {
  flex-wrap: wrap;
}

.dapp-core-component__main__flex-nowrap {
  flex-wrap: nowrap;
}

.dapp-core-component__main__flex-wrap-reverse {
  flex-wrap: wrap-reverse;
}

.dapp-core-component__main__flex-fill {
  flex: 1 1 auto;
}

.dapp-core-component__main__flex-grow-0 {
  flex-grow: 0;
}

.dapp-core-component__main__flex-grow-1 {
  flex-grow: 1;
}

.dapp-core-component__main__flex-shrink-0 {
  flex-shrink: 0;
}

.dapp-core-component__main__flex-shrink-1 {
  flex-shrink: 1;
}

.dapp-core-component__main__justify-content-start {
  justify-content: flex-start;
}

.dapp-core-component__main__justify-content-end {
  justify-content: flex-end;
}

.dapp-core-component__main__justify-content-center {
  justify-content: center;
}

.dapp-core-component__main__justify-content-between {
  justify-content: space-between;
}

.dapp-core-component__main__justify-content-around {
  justify-content: space-around;
}

.dapp-core-component__main__align-items-start {
  align-items: flex-start;
}

.dapp-core-component__main__align-items-end {
  align-items: flex-end;
}

.dapp-core-component__main__align-items-center {
  align-items: center;
}

.dapp-core-component__main__align-items-baseline {
  align-items: baseline;
}

.dapp-core-component__main__align-items-stretch {
  align-items: stretch;
}

.dapp-core-component__main__align-content-start {
  align-content: flex-start;
}

.dapp-core-component__main__align-content-end {
  align-content: flex-end;
}

.dapp-core-component__main__align-content-center {
  align-content: center;
}

.dapp-core-component__main__align-content-between {
  align-content: space-between;
}

.dapp-core-component__main__align-content-around {
  align-content: space-around;
}

.dapp-core-component__main__align-content-stretch {
  align-content: stretch;
}

.dapp-core-component__main__align-self-auto {
  align-self: auto;
}

.dapp-core-component__main__align-self-start {
  align-self: flex-start;
}

.dapp-core-component__main__align-self-end {
  align-self: flex-end;
}

.dapp-core-component__main__align-self-center {
  align-self: center;
}

.dapp-core-component__main__align-self-baseline {
  align-self: baseline;
}

.dapp-core-component__main__align-self-stretch {
  align-self: stretch;
}

@media (min-width: 576px) {
  .dapp-core-component__main__flex-sm-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-wrap {
    flex-wrap: wrap;
  }
}

.dapp-core-component__main__form-check {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: auto;
  padding-left: 0;
}

.dapp-core-component__main__form-check-label {
  margin-bottom: 0;
}

.dapp-core-component__main__form-check-input {
  position: absolute;
  margin-top: 0.3rem;
  margin-left: -1.25rem;
}
.dapp-core-component__main__form-check-input[disabled] ~ .dapp-core-component__main__form-check-label, .dapp-core-component__main__form-check-input:disabled ~ .dapp-core-component__main__form-check-label {
  color: #6c757d;
}

.dapp-core-component__main__form-group {
  display: flex;
  flex: 0 0 auto;
  flex-flow: row wrap;
  align-items: center;
  margin-bottom: 0;
}

.dapp-core-component__main__form-control {
  display: block;
  width: 100%;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.dapp-core-component__main__form-label {
  padding-top: calc(0.375rem + 1px);
  padding-bottom: calc(0.375rem + 1px);
  margin-bottom: 0;
  font-size: inherit;
  line-height: 1.5;
}

input,
button,
select,
optgroup,
textarea {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.dapp-core-component__main__form-group {
  margin-bottom: 1rem;
}

.dapp-core-component__main__form-text {
  display: block;
  margin-top: 0.25rem;
  color: inherit;
}

.dapp-core-component__main__border {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__border-top {
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__border-right {
  border-right: 1px solid #dee2e6;
}

.dapp-core-component__main__border-bottom {
  border-bottom: 1px solid #dee2e6;
}

.dapp-core-component__main__border-left {
  border-left: 1px solid #dee2e6;
}

.dapp-core-component__main__border-0 {
  border: 0;
}

.dapp-core-component__main__border-top-0 {
  border-top: 0;
}

.dapp-core-component__main__border-right-0 {
  border-right: 0;
}

.dapp-core-component__main__border-bottom-0 {
  border-bottom: 0;
}

.dapp-core-component__main__border-left-0 {
  border-left: 0;
}

.dapp-core-component__main__rounded-sm {
  border-radius: 0.2rem;
}

.dapp-core-component__main__rounded {
  border-radius: 0.25rem;
}

.dapp-core-component__main__rounded-top {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-right {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-bottom {
  border-bottom-right-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-left {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-lg {
  border-radius: 0.3rem;
}

.dapp-core-component__main__rounded-circle {
  border-radius: 50%;
}

.dapp-core-component__main__rounded-pill {
  border-radius: 50rem;
}

.dapp-core-component__main__rounded-0 {
  border-radius: 0;
}

.dapp-core-component__main__h1,
.dapp-core-component__main__h2,
.dapp-core-component__main__h3,
.dapp-core-component__main__h4,
.dapp-core-component__main__h5,
.dapp-core-component__main__h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2;
}

.dapp-core-component__main__h1 {
  font-size: 2.5rem;
}

.dapp-core-component__main__h2 {
  font-size: 2rem;
}

.dapp-core-component__main__h3 {
  font-size: 1.75rem;
}

.dapp-core-component__main__h4 {
  font-size: 1.5rem;
}

.dapp-core-component__main__h5 {
  font-size: 1.25rem;
}

.dapp-core-component__main__h6 {
  font-size: 1rem;
}

.dapp-core-component__main__lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

.dapp-core-component__main__bg-warning,
.dapp-core-component__main__warning {
  background-color: #f0ad4e;
}

.dapp-core-component__main__bg-danger,
.dapp-core-component__main__danger {
  background-color: #d9534f;
}

.dapp-core-component__main__bg-success,
.dapp-core-component__main__success {
  background-color: #5cb85c;
}

.dapp-core-component__main__position-relative {
  position: relative;
}

.dapp-core-component__main__position-absolute {
  position: absolute;
}

.dapp-core-component__main__w-25 {
  width: 25%;
}

.dapp-core-component__main__w-50 {
  width: 50%;
}

.dapp-core-component__main__w-75 {
  width: 75%;
}

.dapp-core-component__main__w-100 {
  width: 100%;
}

.dapp-core-component__main__w-auto {
  width: auto;
}

.dapp-core-component__main__h-25 {
  height: 25%;
}

.dapp-core-component__main__h-50 {
  height: 50%;
}

.dapp-core-component__main__h-75 {
  height: 75%;
}

.dapp-core-component__main__h-100 {
  height: 100%;
}

.dapp-core-component__main__h-auto {
  height: auto;
}

.dapp-core-component__main__mw-100 {
  max-width: 100%;
}

.dapp-core-component__main__mh-100 {
  max-height: 100%;
}

.dapp-core-component__main__min-vw-100 {
  min-width: 100vw;
}

.dapp-core-component__main__min-vh-100 {
  min-height: 100vh;
}

.dapp-core-component__main__vw-100 {
  width: 100vw;
}

.dapp-core-component__main__vh-100 {
  height: 100vh;
}

.dapp-core-component__main__locked-icon {
  color: inherit;
}

.dapp-core-component__main__shadow-sm {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
}

.dapp-core-component__main__shadow {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.dapp-core-component__main__shadow-lg {
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.175);
}

.dapp-core-component__main__shadow-none {
  box-shadow: none;
}

.dapp-core-component__main__d-none {
  display: none;
}

.dapp-core-component__main__d-inline {
  display: inline;
}

.dapp-core-component__main__d-inline-block {
  display: inline-block;
}

.dapp-core-component__main__d-block {
  display: block;
}

.dapp-core-component__main__d-table {
  display: table;
}

.dapp-core-component__main__d-table-row {
  display: table-row;
}

.dapp-core-component__main__d-table-cell {
  display: table-cell;
}

.dapp-core-component__main__d-flex {
  display: -ms-flexbox;
  display: flex;
}

.dapp-core-component__main__d-inline-flex {
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media (min-width: 576px) {
  .dapp-core-component__main__d-sm-flex {
    display: flex;
  }
}

.dapp-core-component__main__table {
  width: 100%;
  margin-bottom: 1rem;
  color: #212529;
}
.dapp-core-component__main__table th,
.dapp-core-component__main__table td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #dee2e6;
}
.dapp-core-component__main__table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #dee2e6;
}
.dapp-core-component__main__table tbody + tbody {
  border-top: 2px solid #dee2e6;
}

.dapp-core-component__main__table-sm th,
.dapp-core-component__main__table-sm td {
  padding: 0.3rem;
}

.dapp-core-component__main__table-bordered {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered th,
.dapp-core-component__main__table-bordered td {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered thead th,
.dapp-core-component__main__table-bordered thead td {
  border-bottom-width: 2px;
}

.dapp-core-component__main__table-borderless th,
.dapp-core-component__main__table-borderless td,
.dapp-core-component__main__table-borderless thead th,
.dapp-core-component__main__table-borderless tbody + tbody {
  border: 0;
}

.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(0, 0, 0, 0.05);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-dark th {
  color: #fff;
  background-color: #343a40;
  border-color: #454d55;
}
.dapp-core-component__main__table .dapp-core-component__main__thead-light th {
  color: #495057;
  background-color: #e9ecef;
  border-color: #dee2e6;
}

.dapp-core-component__main__table-dark {
  color: #fff;
  background-color: #343a40;
}
.dapp-core-component__main__table-dark th,
.dapp-core-component__main__table-dark td,
.dapp-core-component__main__table-dark thead th {
  border-color: #454d55;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-bordered {
  border: 0;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(255, 255, 255, 0.05);
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-hover tbody tr {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.075);
}

.dapp-core-component__main__table-responsive {
  display: block;
  width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}
.dapp-core-component__main__table-responsive > .dapp-core-component__main__table-bordered {
  border: 0;
}

@media (max-width: 767.98px) {
  .dapp-core-component__main__transactions.dapp-core-component__main__table .dapp-core-component__main__transaction-hash .dapp-core-component__main__trim {
    max-width: 8rem;
  }
}
.dapp-core-component__main__font-weight-light {
  font-weight: 300 !important;
}

.dapp-core-component__main__font-weight-lighter {
  font-weight: lighter !important;
}

.dapp-core-component__main__font-weight-normal {
  font-weight: 400 !important;
}

.dapp-core-component__main__font-weight-medium {
  font-weight: 500 !important;
}

.dapp-core-component__main__font-weight-bold {
  font-weight: 700 !important;
}

.dapp-core-component__main__font-weight-bolder {
  font-weight: bolder !important;
}

.dapp-core-component__main__font-italic {
  font-style: italic !important;
}

.dapp-core-component__main__icon-state {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
}
.dapp-core-component__main__icon-state.dapp-core-component__main__half {
  width: 3rem;
  height: 3rem;
}

.dapp-core-component__main__side-icon {
  width: 1.5rem;
  height: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  color: transparent;
  border-radius: 50%;
  overflow: hidden;
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__side-icon svg {
  color: #cacbcc;
}

div.dapp-core-component__main__side-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
div.dapp-core-component__main__side-icon img {
  transform: scale(0.75);
}

.dapp-core-component__main__toast-messages {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast {
  display: block;
  flex-basis: 4rem;
  width: 100%;
  max-width: 28.875rem;
  transition: transform 0.3s ease-in-out;
  transform: translateX(120%);
  background-color: #fff;
  border-radius: 0.25rem;
  border-color: #dee2e6;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__clickable {
  cursor: pointer;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible {
  margin-bottom: 0.3rem;
  transform: translateX(0);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress {
  background-color: rgba(0, 0, 0, 0);
  height: inherit;
  font-size: inherit;
  line-height: inherit;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar {
  background-color: rgba(108, 117, 125, 0.2);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar .dapp-core-component__main__content-height {
  z-index: -1;
  visibility: hidden;
  width: 0 !important;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast .dapp-core-component__main__tx-description {
  margin-bottom: 0.2rem;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close {
  opacity: 1;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close svg {
  color: #6c757d;
}

.dapp-core-component__main__data-field .dapp-core-component__main__textarea {
  -moz-appearance: textfield-multiline;
  -webkit-appearance: textarea;
  border: 1px solid #000;
  background-color: #fff;
  border-radius: 5px;
  font: medium -moz-fixed;
  font-size: 0.8125rem;
  min-height: 5rem;
  overflow: auto;
  padding: 0.375rem 0.75rem;
  resize: both;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea:hover {
  border: 1px solid #000;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea .dapp-core-component__main__highlighted {
  font-size: 1rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea.dapp-core-component__main__sc-call {
  min-height: 3rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__form-control:disabled,
.dapp-core-component__main__data-field .dapp-core-component__main__form-control[readonly] {
  background-color: #fafafa;
}
.dapp-core-component__main__data-field .dapp-core-component__main__small-font {
  font-size: 0.8125rem;
}

.dapp-core-component__main__progress-steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps hr {
  background-color: #6c757d;
  border-color: #dee2e6;
  z-index: 2;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot {
  height: 0.625rem;
  width: 0.625rem;
  border-radius: 50%;
  margin: 0 1rem;
  z-index: 5;
  background-color: #6c757d;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:first-of-type {
  margin-left: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:last-of-type {
  margin-right: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

a.dapp-core-component__main__side-link::after {
  right: 0;
  left: unset;
  width: calc(100% - 1.85rem);
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Tm));D();E();ie();t();var ym=S(require("bignumber.js"));D();t();var wm=require("@elrondnetwork/erdjs"),Sm=S(require("bignumber.js"));D();N();wo();t();t();t();t();t();var tc=()=>y(Me);t();t();t();t();var rc=require("react");t();t();var km=require("react"),Am=require("@elrondnetwork/erdjs-extension-provider");lo();q();E();N();yo();t();var Im=require("react");lo();E();t();var ic=require("react"),Em=require("@elrondnetwork/erdjs-hw-provider");lo();q();E();N();yo();t();t();var cc=require("react");t();var mc=require("react");t();var io=require("react"),pc=require("socket.io-client");E();t();var Le={current:null,status:"not_initialized"};var Lm=3e3,Cm=3,Dm=500,Pm=1e3;function sc(){let e=(0,io.useRef)(null),{address:o}=tc(),n=B(),{network:r}=Bt(),a=s=>{e.current&&clearTimeout(e.current),e.current=setTimeout(()=>{n(ea(s))},Pm)},m=(0,io.useCallback)(Ki(()=>u(this,null,function*(){Le.status="pending";let s=yield Ut(r.apiAddress);if(s==null){console.warn("Can not get websocket url");return}Le.current=(0,pc.io)(s,{forceNew:!0,reconnectionAttempts:Cm,timeout:Lm,query:{address:o}}),Le.status="completed",Le.current.onAny(_=>{a(_)})}),{retries:2,delay:Dm}),[o]);(0,io.useEffect)(()=>{o&&Le.status==="not_initialized"&&!Le.current&&m()},[o,Le.current]),(0,io.useEffect)(()=>()=>{var s;(s=Le.current)==null||s.close(),e.current&&clearTimeout(e.current)},[])}function dc(e){sc();let o=y(ht);(0,mc.useEffect)(()=>{let n=o==null?void 0:o.message;n&&e(n)},[e,o])}t();t();t();var _c=e=>e.dappModal,Nm=x(_c,e=>e.visible),En=x(_c,e=>e.modal),_E=x(En,e=>e.showHeader),lE=x(En,e=>e.showFooter),uE=x(En,e=>e.headerText),gE=x(En,e=>e.footerText);Ke();var{logoutEvent:Rm}=ae,lc="storage",gc=()=>{let{address:e}=$o();return(0,uc.useEffect)(()=>{let o=n=>{if(!(n.key!==Rm||!n.newValue))try{let{data:r}=JSON.parse(n.newValue);r===e&&se()}catch(r){return}};return window.addEventListener(lc,o),()=>{window.removeEventListener(lc,o)}},[e]),null};t();t();var Ln=require("react"),hc=require("@elrondnetwork/erdjs"),Tc=require("@elrondnetwork/erdjs/out/signature");E();N();To();var fc=e=>{let o=e.redirectRoute;o&&Ye(o)},xc=({sendSignedTransactionsAsync:e=yi})=>{let o=y(Me),n=y(Be),r=(0,Ln.useRef)(!1),a=B(),m=()=>{a(Bo()),r.current=!1};function s(){return u(this,null,function*(){var l;let _=Object.keys(n);for(let g of _){let f=n==null?void 0:n[g],T=(l=f==null?void 0:f.customTransactionInformation)==null?void 0:l.signWithoutSending;if(!g||T){fc(f);continue}try{if(!(n[g].status==="signed"&&!r.current))continue;let{transactions:L}=n[g];if(!L)continue;r.current=!0;let F=L.map(J=>{let Ce=new hc.Address(J.sender),De=vo(J),$e=new Tc.Signature(J.signature);return De.applySignature($e,Ce),De}),R=yield e(F),ne="pending",_e=L.map(J=>R.includes(J.hash)?A(h({},J),{status:ne}):J);a(_a({sessionId:g,submittedMessage:"submitted"})),a(eo({sessionId:g,status:"sent",transactions:_e})),m(),Ri(o.nonce+L.length),fc(f),history.pushState({},document.title,"?")}catch(I){console.error("Unable to send transactions",I),a(eo({sessionId:g,status:"fail",errorMessage:I.message})),m()}finally{r.current=!1}}})}return(0,Ln.useEffect)(()=>{s()},[n,o]),null};t();t();t();var yc=require("react");function bc(e){var a;let o=nc(),n=(a=e==null?void 0:e.getTransactionsByHash)!=null?a:yn,r=()=>{o({shouldRefreshBalance:!0,getTransactionsByHash:n})};dc(r),(0,yc.useEffect)(()=>{let m=setInterval(r,3e4);return()=>{clearInterval(m)}},[r])}function wc({getTransactionsByHash:e}){return bc({getTransactionsByHash:e}),null}function Sc({customComponents:e}){var m,s;let o=e==null?void 0:e.transactionSender,n=e==null?void 0:e.transactionTracker,r=(m=o==null?void 0:o.component)!=null?m:xc,a=(s=n==null?void 0:n.component)!=null?s:wc;return Lo.default.createElement(Lo.default.Fragment,null,Lo.default.createElement(r,h({},o==null?void 0:o.props)),Lo.default.createElement(a,h({},n==null?void 0:n.props)),Lo.default.createElement(gc,null))}var Mm=({children:e,customNetworkConfig:o={},externalProvider:n,environment:r,customComponents:a})=>{if(!r)throw new Error("missing environment flag");return n!=null&&Ci(n),Co.default.createElement(vc.Provider,{context:Go,store:b},Co.default.createElement(kc.PersistGate,{persistor:ii,loading:null},Co.default.createElement(Hi,{environment:r,customNetworkConfig:o},Co.default.createElement(Gi,null),Co.default.createElement(Sc,{customComponents:a}),e)))};0&&(module.exports={DappProvider});
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
