// @elrondnetwork/dapp-core/__commonjs/wrappers/index.js
"use strict";var Nc=Object.create;var Do=Object.defineProperty,Rc=Object.defineProperties,Mc=Object.getOwnPropertyDescriptor,Fc=Object.getOwnPropertyDescriptors,Oc=Object.getOwnPropertyNames,Ho=Object.getOwnPropertySymbols,Bc=Object.getPrototypeOf,Pn=Object.prototype.hasOwnProperty,qt=Object.prototype.propertyIsEnumerable;var jt=(e,o,n)=>o in e?Do(e,o,{enumerable:!0,configurable:!0,writable:!0,value:n}):e[o]=n,T=(e,o)=>{for(var n in o||(o={}))Pn.call(o,n)&&jt(e,n,o[n]);if(Ho)for(var n of Ho(o))qt.call(o,n)&&jt(e,n,o[n]);return e},I=(e,o)=>Rc(e,Fc(o));var Kt=e=>typeof e=="symbol"?e:e+"",He=(e,o)=>{var n={};for(var r in e)Pn.call(e,r)&&o.indexOf(r)<0&&(n[r]=e[r]);if(e!=null&&Ho)for(var r of Ho(e))o.indexOf(r)<0&&qt.call(e,r)&&(n[r]=e[r]);return n};var S=(e,o)=>()=>(e&&(o=e(e=0)),o);var zo=(e,o)=>()=>(o||e((o={exports:{}}).exports,o),o.exports),ze=(e,o)=>{for(var n in o)Do(e,n,{get:o[n],enumerable:!0})},Xt=(e,o,n,r)=>{if(o&&typeof o=="object"||typeof o=="function")for(let a of Oc(o))!Pn.call(e,a)&&a!==n&&Do(e,a,{get:()=>o[a],enumerable:!(r=Mc(o,a))||r.enumerable});return e};var w=(e,o,n)=>(n=e!=null?Nc(Bc(e)):{},Xt(o||!e||!e.__esModule?Do(n,"default",{value:e,enumerable:!0}):n,e)),No=e=>Xt(Do({},"__esModule",{value:!0}),e);var g=(e,o,n)=>new Promise((r,a)=>{var m=l=>{try{_(n.next(l))}catch(u){a(u)}},p=l=>{try{_(n.throw(l))}catch(u){a(u)}},_=l=>l.done?r(l.value):Promise.resolve(l.value).then(m,p);_((n=n.apply(e,o)).next())});var Jt=zo(jo=>{"use strict";t();jo.byteLength=Wc;jo.toByteArray=Gc;jo.fromByteArray=zc;var le=[],re=[],Uc=typeof Uint8Array!="undefined"?Uint8Array:Array,Cn="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";for(je=0,Yt=Cn.length;je<Yt;++je)le[je]=Cn[je],re[Cn.charCodeAt(je)]=je;var je,Yt;re["-".charCodeAt(0)]=62;re["_".charCodeAt(0)]=63;function Zt(e){var o=e.length;if(o%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var n=e.indexOf("=");n===-1&&(n=o);var r=n===o?0:4-n%4;return[n,r]}function Wc(e){var o=Zt(e),n=o[0],r=o[1];return(n+r)*3/4-r}function $c(e,o,n){return(o+n)*3/4-n}function Gc(e){var o,n=Zt(e),r=n[0],a=n[1],m=new Uc($c(e,r,a)),p=0,_=a>0?r-4:r,l;for(l=0;l<_;l+=4)o=re[e.charCodeAt(l)]<<18|re[e.charCodeAt(l+1)]<<12|re[e.charCodeAt(l+2)]<<6|re[e.charCodeAt(l+3)],m[p++]=o>>16&255,m[p++]=o>>8&255,m[p++]=o&255;return a===2&&(o=re[e.charCodeAt(l)]<<2|re[e.charCodeAt(l+1)]>>4,m[p++]=o&255),a===1&&(o=re[e.charCodeAt(l)]<<10|re[e.charCodeAt(l+1)]<<4|re[e.charCodeAt(l+2)]>>2,m[p++]=o>>8&255,m[p++]=o&255),m}function Vc(e){return le[e>>18&63]+le[e>>12&63]+le[e>>6&63]+le[e&63]}function Hc(e,o,n){for(var r,a=[],m=o;m<n;m+=3)r=(e[m]<<16&16711680)+(e[m+1]<<8&65280)+(e[m+2]&255),a.push(Vc(r));return a.join("")}function zc(e){for(var o,n=e.length,r=n%3,a=[],m=16383,p=0,_=n-r;p<_;p+=m)a.push(Hc(e,p,p+m>_?_:p+m));return r===1?(o=e[n-1],a.push(le[o>>2]+le[o<<4&63]+"==")):r===2&&(o=(e[n-2]<<8)+e[n-1],a.push(le[o>>10]+le[o>>4&63]+le[o<<2&63]+"=")),a.join("")}});var Qt=zo(Dn=>{t();Dn.read=function(e,o,n,r,a){var m,p,_=a*8-r-1,l=(1<<_)-1,u=l>>1,f=-7,h=n?a-1:0,A=n?-1:1,k=e[o+h];for(h+=A,m=k&(1<<-f)-1,k>>=-f,f+=_;f>0;m=m*256+e[o+h],h+=A,f-=8);for(p=m&(1<<-f)-1,m>>=-f,f+=r;f>0;p=p*256+e[o+h],h+=A,f-=8);if(m===0)m=1-u;else{if(m===l)return p?NaN:(k?-1:1)*(1/0);p=p+Math.pow(2,r),m=m-u}return(k?-1:1)*p*Math.pow(2,m-r)};Dn.write=function(e,o,n,r,a,m){var p,_,l,u=m*8-a-1,f=(1<<u)-1,h=f>>1,A=a===23?Math.pow(2,-24)-Math.pow(2,-77):0,k=r?0:m-1,L=r?1:-1,F=o<0||o===0&&1/o<0?1:0;for(o=Math.abs(o),isNaN(o)||o===1/0?(_=isNaN(o)?1:0,p=f):(p=Math.floor(Math.log(o)/Math.LN2),o*(l=Math.pow(2,-p))<1&&(p--,l*=2),p+h>=1?o+=A/l:o+=A*Math.pow(2,1-h),o*l>=2&&(p++,l/=2),p+h>=f?(_=0,p=f):p+h>=1?(_=(o*l-1)*Math.pow(2,a),p=p+h):(_=o*Math.pow(2,h-1)*Math.pow(2,a),p=0));a>=8;e[n+k]=_&255,k+=L,_/=256,a-=8);for(p=p<<a|_,u+=a;u>0;e[n+k]=p&255,k+=L,p/=256,u-=8);e[n+k-L]|=F*128}});var _r=zo(mo=>{"use strict";t();var Nn=Jt(),po=Qt(),er=typeof Symbol=="function"&&typeof Symbol.for=="function"?Symbol.for("nodejs.util.inspect.custom"):null;mo.Buffer=d;mo.SlowBuffer=Zc;mo.INSPECT_MAX_BYTES=50;var qo=2147483647;mo.kMaxLength=qo;d.TYPED_ARRAY_SUPPORT=jc();!d.TYPED_ARRAY_SUPPORT&&typeof console!="undefined"&&typeof console.error=="function"&&console.error("This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.");function jc(){try{var e=new Uint8Array(1),o={foo:function(){return 42}};return Object.setPrototypeOf(o,Uint8Array.prototype),Object.setPrototypeOf(e,o),e.foo()===42}catch(n){return!1}}Object.defineProperty(d.prototype,"parent",{enumerable:!0,get:function(){if(!!d.isBuffer(this))return this.buffer}});Object.defineProperty(d.prototype,"offset",{enumerable:!0,get:function(){if(!!d.isBuffer(this))return this.byteOffset}});function Se(e){if(e>qo)throw new RangeError('The value "'+e+'" is invalid for option "size"');var o=new Uint8Array(e);return Object.setPrototypeOf(o,d.prototype),o}function d(e,o,n){if(typeof e=="number"){if(typeof o=="string")throw new TypeError('The "string" argument must be of type string. Received type number');return On(e)}return tr(e,o,n)}d.poolSize=8192;function tr(e,o,n){if(typeof e=="string")return Kc(e,o);if(ArrayBuffer.isView(e))return Xc(e);if(e==null)throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e);if(ue(e,ArrayBuffer)||e&&ue(e.buffer,ArrayBuffer)||typeof SharedArrayBuffer!="undefined"&&(ue(e,SharedArrayBuffer)||e&&ue(e.buffer,SharedArrayBuffer)))return Mn(e,o,n);if(typeof e=="number")throw new TypeError('The "value" argument must not be of type number. Received type number');var r=e.valueOf&&e.valueOf();if(r!=null&&r!==e)return d.from(r,o,n);var a=Yc(e);if(a)return a;if(typeof Symbol!="undefined"&&Symbol.toPrimitive!=null&&typeof e[Symbol.toPrimitive]=="function")return d.from(e[Symbol.toPrimitive]("string"),o,n);throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e)}d.from=function(e,o,n){return tr(e,o,n)};Object.setPrototypeOf(d.prototype,Uint8Array.prototype);Object.setPrototypeOf(d,Uint8Array);function rr(e){if(typeof e!="number")throw new TypeError('"size" argument must be of type number');if(e<0)throw new RangeError('The value "'+e+'" is invalid for option "size"')}function qc(e,o,n){return rr(e),e<=0?Se(e):o!==void 0?typeof n=="string"?Se(e).fill(o,n):Se(e).fill(o):Se(e)}d.alloc=function(e,o,n){return qc(e,o,n)};function On(e){return rr(e),Se(e<0?0:Bn(e)|0)}d.allocUnsafe=function(e){return On(e)};d.allocUnsafeSlow=function(e){return On(e)};function Kc(e,o){if((typeof o!="string"||o==="")&&(o="utf8"),!d.isEncoding(o))throw new TypeError("Unknown encoding: "+o);var n=ar(e,o)|0,r=Se(n),a=r.write(e,o);return a!==n&&(r=r.slice(0,a)),r}function Rn(e){for(var o=e.length<0?0:Bn(e.length)|0,n=Se(o),r=0;r<o;r+=1)n[r]=e[r]&255;return n}function Xc(e){if(ue(e,Uint8Array)){var o=new Uint8Array(e);return Mn(o.buffer,o.byteOffset,o.byteLength)}return Rn(e)}function Mn(e,o,n){if(o<0||e.byteLength<o)throw new RangeError('"offset" is outside of buffer bounds');if(e.byteLength<o+(n||0))throw new RangeError('"length" is outside of buffer bounds');var r;return o===void 0&&n===void 0?r=new Uint8Array(e):n===void 0?r=new Uint8Array(e,o):r=new Uint8Array(e,o,n),Object.setPrototypeOf(r,d.prototype),r}function Yc(e){if(d.isBuffer(e)){var o=Bn(e.length)|0,n=Se(o);return n.length===0||e.copy(n,0,0,o),n}if(e.length!==void 0)return typeof e.length!="number"||Un(e.length)?Se(0):Rn(e);if(e.type==="Buffer"&&Array.isArray(e.data))return Rn(e.data)}function Bn(e){if(e>=qo)throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x"+qo.toString(16)+" bytes");return e|0}function Zc(e){return+e!=e&&(e=0),d.alloc(+e)}d.isBuffer=function(o){return o!=null&&o._isBuffer===!0&&o!==d.prototype};d.compare=function(o,n){if(ue(o,Uint8Array)&&(o=d.from(o,o.offset,o.byteLength)),ue(n,Uint8Array)&&(n=d.from(n,n.offset,n.byteLength)),!d.isBuffer(o)||!d.isBuffer(n))throw new TypeError('The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array');if(o===n)return 0;for(var r=o.length,a=n.length,m=0,p=Math.min(r,a);m<p;++m)if(o[m]!==n[m]){r=o[m],a=n[m];break}return r<a?-1:a<r?1:0};d.isEncoding=function(o){switch(String(o).toLowerCase()){case"hex":case"utf8":case"utf-8":case"ascii":case"latin1":case"binary":case"base64":case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return!0;default:return!1}};d.concat=function(o,n){if(!Array.isArray(o))throw new TypeError('"list" argument must be an Array of Buffers');if(o.length===0)return d.alloc(0);var r;if(n===void 0)for(n=0,r=0;r<o.length;++r)n+=o[r].length;var a=d.allocUnsafe(n),m=0;for(r=0;r<o.length;++r){var p=o[r];if(ue(p,Uint8Array))m+p.length>a.length?d.from(p).copy(a,m):Uint8Array.prototype.set.call(a,p,m);else if(d.isBuffer(p))p.copy(a,m);else throw new TypeError('"list" argument must be an Array of Buffers');m+=p.length}return a};function ar(e,o){if(d.isBuffer(e))return e.length;if(ArrayBuffer.isView(e)||ue(e,ArrayBuffer))return e.byteLength;if(typeof e!="string")throw new TypeError('The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type '+typeof e);var n=e.length,r=arguments.length>2&&arguments[2]===!0;if(!r&&n===0)return 0;for(var a=!1;;)switch(o){case"ascii":case"latin1":case"binary":return n;case"utf8":case"utf-8":return Fn(e).length;case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return n*2;case"hex":return n>>>1;case"base64":return dr(e).length;default:if(a)return r?-1:Fn(e).length;o=(""+o).toLowerCase(),a=!0}}d.byteLength=ar;function Jc(e,o,n){var r=!1;if((o===void 0||o<0)&&(o=0),o>this.length||((n===void 0||n>this.length)&&(n=this.length),n<=0)||(n>>>=0,o>>>=0,n<=o))return"";for(e||(e="utf8");;)switch(e){case"hex":return ss(this,o,n);case"utf8":case"utf-8":return cr(this,o,n);case"ascii":return is(this,o,n);case"latin1":case"binary":return cs(this,o,n);case"base64":return rs(this,o,n);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return ps(this,o,n);default:if(r)throw new TypeError("Unknown encoding: "+e);e=(e+"").toLowerCase(),r=!0}}d.prototype._isBuffer=!0;function qe(e,o,n){var r=e[o];e[o]=e[n],e[n]=r}d.prototype.swap16=function(){var o=this.length;if(o%2!==0)throw new RangeError("Buffer size must be a multiple of 16-bits");for(var n=0;n<o;n+=2)qe(this,n,n+1);return this};d.prototype.swap32=function(){var o=this.length;if(o%4!==0)throw new RangeError("Buffer size must be a multiple of 32-bits");for(var n=0;n<o;n+=4)qe(this,n,n+3),qe(this,n+1,n+2);return this};d.prototype.swap64=function(){var o=this.length;if(o%8!==0)throw new RangeError("Buffer size must be a multiple of 64-bits");for(var n=0;n<o;n+=8)qe(this,n,n+7),qe(this,n+1,n+6),qe(this,n+2,n+5),qe(this,n+3,n+4);return this};d.prototype.toString=function(){var o=this.length;return o===0?"":arguments.length===0?cr(this,0,o):Jc.apply(this,arguments)};d.prototype.toLocaleString=d.prototype.toString;d.prototype.equals=function(o){if(!d.isBuffer(o))throw new TypeError("Argument must be a Buffer");return this===o?!0:d.compare(this,o)===0};d.prototype.inspect=function(){var o="",n=mo.INSPECT_MAX_BYTES;return o=this.toString("hex",0,n).replace(/(.{2})/g,"$1 ").trim(),this.length>n&&(o+=" ... "),"<Buffer "+o+">"};er&&(d.prototype[er]=d.prototype.inspect);d.prototype.compare=function(o,n,r,a,m){if(ue(o,Uint8Array)&&(o=d.from(o,o.offset,o.byteLength)),!d.isBuffer(o))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+typeof o);if(n===void 0&&(n=0),r===void 0&&(r=o?o.length:0),a===void 0&&(a=0),m===void 0&&(m=this.length),n<0||r>o.length||a<0||m>this.length)throw new RangeError("out of range index");if(a>=m&&n>=r)return 0;if(a>=m)return-1;if(n>=r)return 1;if(n>>>=0,r>>>=0,a>>>=0,m>>>=0,this===o)return 0;for(var p=m-a,_=r-n,l=Math.min(p,_),u=this.slice(a,m),f=o.slice(n,r),h=0;h<l;++h)if(u[h]!==f[h]){p=u[h],_=f[h];break}return p<_?-1:_<p?1:0};function ir(e,o,n,r,a){if(e.length===0)return-1;if(typeof n=="string"?(r=n,n=0):n>2147483647?n=2147483647:n<-2147483648&&(n=-2147483648),n=+n,Un(n)&&(n=a?0:e.length-1),n<0&&(n=e.length+n),n>=e.length){if(a)return-1;n=e.length-1}else if(n<0)if(a)n=0;else return-1;if(typeof o=="string"&&(o=d.from(o,r)),d.isBuffer(o))return o.length===0?-1:or(e,o,n,r,a);if(typeof o=="number")return o=o&255,typeof Uint8Array.prototype.indexOf=="function"?a?Uint8Array.prototype.indexOf.call(e,o,n):Uint8Array.prototype.lastIndexOf.call(e,o,n):or(e,[o],n,r,a);throw new TypeError("val must be string, number or Buffer")}function or(e,o,n,r,a){var m=1,p=e.length,_=o.length;if(r!==void 0&&(r=String(r).toLowerCase(),r==="ucs2"||r==="ucs-2"||r==="utf16le"||r==="utf-16le")){if(e.length<2||o.length<2)return-1;m=2,p/=2,_/=2,n/=2}function l(k,L){return m===1?k[L]:k.readUInt16BE(L*m)}var u;if(a){var f=-1;for(u=n;u<p;u++)if(l(e,u)===l(o,f===-1?0:u-f)){if(f===-1&&(f=u),u-f+1===_)return f*m}else f!==-1&&(u-=u-f),f=-1}else for(n+_>p&&(n=p-_),u=n;u>=0;u--){for(var h=!0,A=0;A<_;A++)if(l(e,u+A)!==l(o,A)){h=!1;break}if(h)return u}return-1}d.prototype.includes=function(o,n,r){return this.indexOf(o,n,r)!==-1};d.prototype.indexOf=function(o,n,r){return ir(this,o,n,r,!0)};d.prototype.lastIndexOf=function(o,n,r){return ir(this,o,n,r,!1)};function Qc(e,o,n,r){n=Number(n)||0;var a=e.length-n;r?(r=Number(r),r>a&&(r=a)):r=a;var m=o.length;r>m/2&&(r=m/2);for(var p=0;p<r;++p){var _=parseInt(o.substr(p*2,2),16);if(Un(_))return p;e[n+p]=_}return p}function es(e,o,n,r){return Ko(Fn(o,e.length-n),e,n,r)}function os(e,o,n,r){return Ko(_s(o),e,n,r)}function ns(e,o,n,r){return Ko(dr(o),e,n,r)}function ts(e,o,n,r){return Ko(ls(o,e.length-n),e,n,r)}d.prototype.write=function(o,n,r,a){if(n===void 0)a="utf8",r=this.length,n=0;else if(r===void 0&&typeof n=="string")a=n,r=this.length,n=0;else if(isFinite(n))n=n>>>0,isFinite(r)?(r=r>>>0,a===void 0&&(a="utf8")):(a=r,r=void 0);else throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");var m=this.length-n;if((r===void 0||r>m)&&(r=m),o.length>0&&(r<0||n<0)||n>this.length)throw new RangeError("Attempt to write outside buffer bounds");a||(a="utf8");for(var p=!1;;)switch(a){case"hex":return Qc(this,o,n,r);case"utf8":case"utf-8":return es(this,o,n,r);case"ascii":case"latin1":case"binary":return os(this,o,n,r);case"base64":return ns(this,o,n,r);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return ts(this,o,n,r);default:if(p)throw new TypeError("Unknown encoding: "+a);a=(""+a).toLowerCase(),p=!0}};d.prototype.toJSON=function(){return{type:"Buffer",data:Array.prototype.slice.call(this._arr||this,0)}};function rs(e,o,n){return o===0&&n===e.length?Nn.fromByteArray(e):Nn.fromByteArray(e.slice(o,n))}function cr(e,o,n){n=Math.min(e.length,n);for(var r=[],a=o;a<n;){var m=e[a],p=null,_=m>239?4:m>223?3:m>191?2:1;if(a+_<=n){var l,u,f,h;switch(_){case 1:m<128&&(p=m);break;case 2:l=e[a+1],(l&192)===128&&(h=(m&31)<<6|l&63,h>127&&(p=h));break;case 3:l=e[a+1],u=e[a+2],(l&192)===128&&(u&192)===128&&(h=(m&15)<<12|(l&63)<<6|u&63,h>2047&&(h<55296||h>57343)&&(p=h));break;case 4:l=e[a+1],u=e[a+2],f=e[a+3],(l&192)===128&&(u&192)===128&&(f&192)===128&&(h=(m&15)<<18|(l&63)<<12|(u&63)<<6|f&63,h>65535&&h<1114112&&(p=h))}}p===null?(p=65533,_=1):p>65535&&(p-=65536,r.push(p>>>10&1023|55296),p=56320|p&1023),r.push(p),a+=_}return as(r)}var nr=4096;function as(e){var o=e.length;if(o<=nr)return String.fromCharCode.apply(String,e);for(var n="",r=0;r<o;)n+=String.fromCharCode.apply(String,e.slice(r,r+=nr));return n}function is(e,o,n){var r="";n=Math.min(e.length,n);for(var a=o;a<n;++a)r+=String.fromCharCode(e[a]&127);return r}function cs(e,o,n){var r="";n=Math.min(e.length,n);for(var a=o;a<n;++a)r+=String.fromCharCode(e[a]);return r}function ss(e,o,n){var r=e.length;(!o||o<0)&&(o=0),(!n||n<0||n>r)&&(n=r);for(var a="",m=o;m<n;++m)a+=us[e[m]];return a}function ps(e,o,n){for(var r=e.slice(o,n),a="",m=0;m<r.length-1;m+=2)a+=String.fromCharCode(r[m]+r[m+1]*256);return a}d.prototype.slice=function(o,n){var r=this.length;o=~~o,n=n===void 0?r:~~n,o<0?(o+=r,o<0&&(o=0)):o>r&&(o=r),n<0?(n+=r,n<0&&(n=0)):n>r&&(n=r),n<o&&(n=o);var a=this.subarray(o,n);return Object.setPrototypeOf(a,d.prototype),a};function $(e,o,n){if(e%1!==0||e<0)throw new RangeError("offset is not uint");if(e+o>n)throw new RangeError("Trying to access beyond buffer length")}d.prototype.readUintLE=d.prototype.readUIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||$(o,n,this.length);for(var a=this[o],m=1,p=0;++p<n&&(m*=256);)a+=this[o+p]*m;return a};d.prototype.readUintBE=d.prototype.readUIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||$(o,n,this.length);for(var a=this[o+--n],m=1;n>0&&(m*=256);)a+=this[o+--n]*m;return a};d.prototype.readUint8=d.prototype.readUInt8=function(o,n){return o=o>>>0,n||$(o,1,this.length),this[o]};d.prototype.readUint16LE=d.prototype.readUInt16LE=function(o,n){return o=o>>>0,n||$(o,2,this.length),this[o]|this[o+1]<<8};d.prototype.readUint16BE=d.prototype.readUInt16BE=function(o,n){return o=o>>>0,n||$(o,2,this.length),this[o]<<8|this[o+1]};d.prototype.readUint32LE=d.prototype.readUInt32LE=function(o,n){return o=o>>>0,n||$(o,4,this.length),(this[o]|this[o+1]<<8|this[o+2]<<16)+this[o+3]*16777216};d.prototype.readUint32BE=d.prototype.readUInt32BE=function(o,n){return o=o>>>0,n||$(o,4,this.length),this[o]*16777216+(this[o+1]<<16|this[o+2]<<8|this[o+3])};d.prototype.readIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||$(o,n,this.length);for(var a=this[o],m=1,p=0;++p<n&&(m*=256);)a+=this[o+p]*m;return m*=128,a>=m&&(a-=Math.pow(2,8*n)),a};d.prototype.readIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||$(o,n,this.length);for(var a=n,m=1,p=this[o+--a];a>0&&(m*=256);)p+=this[o+--a]*m;return m*=128,p>=m&&(p-=Math.pow(2,8*n)),p};d.prototype.readInt8=function(o,n){return o=o>>>0,n||$(o,1,this.length),this[o]&128?(255-this[o]+1)*-1:this[o]};d.prototype.readInt16LE=function(o,n){o=o>>>0,n||$(o,2,this.length);var r=this[o]|this[o+1]<<8;return r&32768?r|4294901760:r};d.prototype.readInt16BE=function(o,n){o=o>>>0,n||$(o,2,this.length);var r=this[o+1]|this[o]<<8;return r&32768?r|4294901760:r};d.prototype.readInt32LE=function(o,n){return o=o>>>0,n||$(o,4,this.length),this[o]|this[o+1]<<8|this[o+2]<<16|this[o+3]<<24};d.prototype.readInt32BE=function(o,n){return o=o>>>0,n||$(o,4,this.length),this[o]<<24|this[o+1]<<16|this[o+2]<<8|this[o+3]};d.prototype.readFloatLE=function(o,n){return o=o>>>0,n||$(o,4,this.length),po.read(this,o,!0,23,4)};d.prototype.readFloatBE=function(o,n){return o=o>>>0,n||$(o,4,this.length),po.read(this,o,!1,23,4)};d.prototype.readDoubleLE=function(o,n){return o=o>>>0,n||$(o,8,this.length),po.read(this,o,!0,52,8)};d.prototype.readDoubleBE=function(o,n){return o=o>>>0,n||$(o,8,this.length),po.read(this,o,!1,52,8)};function Y(e,o,n,r,a,m){if(!d.isBuffer(e))throw new TypeError('"buffer" argument must be a Buffer instance');if(o>a||o<m)throw new RangeError('"value" argument is out of bounds');if(n+r>e.length)throw new RangeError("Index out of range")}d.prototype.writeUintLE=d.prototype.writeUIntLE=function(o,n,r,a){if(o=+o,n=n>>>0,r=r>>>0,!a){var m=Math.pow(2,8*r)-1;Y(this,o,n,r,m,0)}var p=1,_=0;for(this[n]=o&255;++_<r&&(p*=256);)this[n+_]=o/p&255;return n+r};d.prototype.writeUintBE=d.prototype.writeUIntBE=function(o,n,r,a){if(o=+o,n=n>>>0,r=r>>>0,!a){var m=Math.pow(2,8*r)-1;Y(this,o,n,r,m,0)}var p=r-1,_=1;for(this[n+p]=o&255;--p>=0&&(_*=256);)this[n+p]=o/_&255;return n+r};d.prototype.writeUint8=d.prototype.writeUInt8=function(o,n,r){return o=+o,n=n>>>0,r||Y(this,o,n,1,255,0),this[n]=o&255,n+1};d.prototype.writeUint16LE=d.prototype.writeUInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||Y(this,o,n,2,65535,0),this[n]=o&255,this[n+1]=o>>>8,n+2};d.prototype.writeUint16BE=d.prototype.writeUInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||Y(this,o,n,2,65535,0),this[n]=o>>>8,this[n+1]=o&255,n+2};d.prototype.writeUint32LE=d.prototype.writeUInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||Y(this,o,n,4,4294967295,0),this[n+3]=o>>>24,this[n+2]=o>>>16,this[n+1]=o>>>8,this[n]=o&255,n+4};d.prototype.writeUint32BE=d.prototype.writeUInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||Y(this,o,n,4,4294967295,0),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};d.prototype.writeIntLE=function(o,n,r,a){if(o=+o,n=n>>>0,!a){var m=Math.pow(2,8*r-1);Y(this,o,n,r,m-1,-m)}var p=0,_=1,l=0;for(this[n]=o&255;++p<r&&(_*=256);)o<0&&l===0&&this[n+p-1]!==0&&(l=1),this[n+p]=(o/_>>0)-l&255;return n+r};d.prototype.writeIntBE=function(o,n,r,a){if(o=+o,n=n>>>0,!a){var m=Math.pow(2,8*r-1);Y(this,o,n,r,m-1,-m)}var p=r-1,_=1,l=0;for(this[n+p]=o&255;--p>=0&&(_*=256);)o<0&&l===0&&this[n+p+1]!==0&&(l=1),this[n+p]=(o/_>>0)-l&255;return n+r};d.prototype.writeInt8=function(o,n,r){return o=+o,n=n>>>0,r||Y(this,o,n,1,127,-128),o<0&&(o=255+o+1),this[n]=o&255,n+1};d.prototype.writeInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||Y(this,o,n,2,32767,-32768),this[n]=o&255,this[n+1]=o>>>8,n+2};d.prototype.writeInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||Y(this,o,n,2,32767,-32768),this[n]=o>>>8,this[n+1]=o&255,n+2};d.prototype.writeInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||Y(this,o,n,4,2147483647,-2147483648),this[n]=o&255,this[n+1]=o>>>8,this[n+2]=o>>>16,this[n+3]=o>>>24,n+4};d.prototype.writeInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||Y(this,o,n,4,2147483647,-2147483648),o<0&&(o=4294967295+o+1),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};function sr(e,o,n,r,a,m){if(n+r>e.length)throw new RangeError("Index out of range");if(n<0)throw new RangeError("Index out of range")}function pr(e,o,n,r,a){return o=+o,n=n>>>0,a||sr(e,o,n,4,34028234663852886e22,-34028234663852886e22),po.write(e,o,n,r,23,4),n+4}d.prototype.writeFloatLE=function(o,n,r){return pr(this,o,n,!0,r)};d.prototype.writeFloatBE=function(o,n,r){return pr(this,o,n,!1,r)};function mr(e,o,n,r,a){return o=+o,n=n>>>0,a||sr(e,o,n,8,17976931348623157e292,-17976931348623157e292),po.write(e,o,n,r,52,8),n+8}d.prototype.writeDoubleLE=function(o,n,r){return mr(this,o,n,!0,r)};d.prototype.writeDoubleBE=function(o,n,r){return mr(this,o,n,!1,r)};d.prototype.copy=function(o,n,r,a){if(!d.isBuffer(o))throw new TypeError("argument should be a Buffer");if(r||(r=0),!a&&a!==0&&(a=this.length),n>=o.length&&(n=o.length),n||(n=0),a>0&&a<r&&(a=r),a===r||o.length===0||this.length===0)return 0;if(n<0)throw new RangeError("targetStart out of bounds");if(r<0||r>=this.length)throw new RangeError("Index out of range");if(a<0)throw new RangeError("sourceEnd out of bounds");a>this.length&&(a=this.length),o.length-n<a-r&&(a=o.length-n+r);var m=a-r;return this===o&&typeof Uint8Array.prototype.copyWithin=="function"?this.copyWithin(n,r,a):Uint8Array.prototype.set.call(o,this.subarray(r,a),n),m};d.prototype.fill=function(o,n,r,a){if(typeof o=="string"){if(typeof n=="string"?(a=n,n=0,r=this.length):typeof r=="string"&&(a=r,r=this.length),a!==void 0&&typeof a!="string")throw new TypeError("encoding must be a string");if(typeof a=="string"&&!d.isEncoding(a))throw new TypeError("Unknown encoding: "+a);if(o.length===1){var m=o.charCodeAt(0);(a==="utf8"&&m<128||a==="latin1")&&(o=m)}}else typeof o=="number"?o=o&255:typeof o=="boolean"&&(o=Number(o));if(n<0||this.length<n||this.length<r)throw new RangeError("Out of range index");if(r<=n)return this;n=n>>>0,r=r===void 0?this.length:r>>>0,o||(o=0);var p;if(typeof o=="number")for(p=n;p<r;++p)this[p]=o;else{var _=d.isBuffer(o)?o:d.from(o,a),l=_.length;if(l===0)throw new TypeError('The value "'+o+'" is invalid for argument "value"');for(p=0;p<r-n;++p)this[p+n]=_[p%l]}return this};var ms=/[^+/0-9A-Za-z-_]/g;function ds(e){if(e=e.split("=")[0],e=e.trim().replace(ms,""),e.length<2)return"";for(;e.length%4!==0;)e=e+"=";return e}function Fn(e,o){o=o||1/0;for(var n,r=e.length,a=null,m=[],p=0;p<r;++p){if(n=e.charCodeAt(p),n>55295&&n<57344){if(!a){if(n>56319){(o-=3)>-1&&m.push(239,191,189);continue}else if(p+1===r){(o-=3)>-1&&m.push(239,191,189);continue}a=n;continue}if(n<56320){(o-=3)>-1&&m.push(239,191,189),a=n;continue}n=(a-55296<<10|n-56320)+65536}else a&&(o-=3)>-1&&m.push(239,191,189);if(a=null,n<128){if((o-=1)<0)break;m.push(n)}else if(n<2048){if((o-=2)<0)break;m.push(n>>6|192,n&63|128)}else if(n<65536){if((o-=3)<0)break;m.push(n>>12|224,n>>6&63|128,n&63|128)}else if(n<1114112){if((o-=4)<0)break;m.push(n>>18|240,n>>12&63|128,n>>6&63|128,n&63|128)}else throw new Error("Invalid code point")}return m}function _s(e){for(var o=[],n=0;n<e.length;++n)o.push(e.charCodeAt(n)&255);return o}function ls(e,o){for(var n,r,a,m=[],p=0;p<e.length&&!((o-=2)<0);++p)n=e.charCodeAt(p),r=n>>8,a=n%256,m.push(a),m.push(r);return m}function dr(e){return Nn.toByteArray(ds(e))}function Ko(e,o,n,r){for(var a=0;a<r&&!(a+n>=o.length||a>=e.length);++a)o[a+n]=e[a];return a}function ue(e,o){return e instanceof o||e!=null&&e.constructor!=null&&e.constructor.name!=null&&e.constructor.name===o.name}function Un(e){return e!==e}var us=function(){for(var e="0123456789abcdef",o=new Array(256),n=0;n<16;++n)for(var r=n*16,a=0;a<16;++a)o[r+a]=e[n]+e[a];return o}()});var hr=zo((Ym,fr)=>{t();var O=fr.exports={},ge,fe;function Wn(){throw new Error("setTimeout has not been defined")}function $n(){throw new Error("clearTimeout has not been defined")}(function(){try{typeof setTimeout=="function"?ge=setTimeout:ge=Wn}catch(e){ge=Wn}try{typeof clearTimeout=="function"?fe=clearTimeout:fe=$n}catch(e){fe=$n}})();function lr(e){if(ge===setTimeout)return setTimeout(e,0);if((ge===Wn||!ge)&&setTimeout)return ge=setTimeout,setTimeout(e,0);try{return ge(e,0)}catch(o){try{return ge.call(null,e,0)}catch(n){return ge.call(this,e,0)}}}function gs(e){if(fe===clearTimeout)return clearTimeout(e);if((fe===$n||!fe)&&clearTimeout)return fe=clearTimeout,clearTimeout(e);try{return fe(e)}catch(o){try{return fe.call(null,e)}catch(n){return fe.call(this,e)}}}var we=[],_o=!1,Ke,Xo=-1;function fs(){!_o||!Ke||(_o=!1,Ke.length?we=Ke.concat(we):Xo=-1,we.length&&ur())}function ur(){if(!_o){var e=lr(fs);_o=!0;for(var o=we.length;o;){for(Ke=we,we=[];++Xo<o;)Ke&&Ke[Xo].run();Xo=-1,o=we.length}Ke=null,_o=!1,gs(e)}}O.nextTick=function(e){var o=new Array(arguments.length-1);if(arguments.length>1)for(var n=1;n<arguments.length;n++)o[n-1]=arguments[n];we.push(new gr(e,o)),we.length===1&&!_o&&lr(ur)};function gr(e,o){this.fun=e,this.array=o}gr.prototype.run=function(){this.fun.apply(null,this.array)};O.title="browser";O.browser=!0;O.env={};O.argv=[];O.version="";O.versions={};function ve(){}O.on=ve;O.addListener=ve;O.once=ve;O.off=ve;O.removeListener=ve;O.removeAllListeners=ve;O.emit=ve;O.prependListener=ve;O.prependOnceListener=ve;O.listeners=function(e){return[]};O.binding=function(e){throw new Error("process.binding is not supported")};O.cwd=function(){return"/"};O.chdir=function(e){throw new Error("process.chdir is not supported")};O.umask=function(){return 0}});var i,c,Zm,t=S(()=>{i=w(_r()),c=w(hr()),Zm=function(e){function o(){var r=this||self;return delete e.prototype.__magic__,r}if(typeof globalThis=="object")return globalThis;if(this)return o();e.defineProperty(e.prototype,"__magic__",{configurable:!0,get:o});var n=__magic__;return n}(Object)});var Yo,Gn=S(()=>{"use strict";t();Yo=()=>Date.now()/1e3});var Tr=S(()=>{"use strict";t()});var xr=S(()=>{"use strict";t()});var Vn=S(()=>{"use strict";t();Gn();Tr();xr()});var zn={};ze(zn,{getItem:()=>Ts,localStorageKeys:()=>ae,removeItem:()=>xs,setItem:()=>hs});var ae,Hn,hs,Ts,xs,Xe=S(()=>{"use strict";t();Vn();ae={loginExpiresAt:"dapp-core-login-expires-at",logoutEvent:"dapp-core-logout-event"},Hn=typeof localStorage!="undefined",hs=({key:e,data:o,expires:n})=>{!Hn||localStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},Ts=e=>{if(!Hn)return;let o=localStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:Yo()>=n.expires?(localStorage.removeItem(String(e)),null):n.data},xs=e=>{!Hn||localStorage.removeItem(String(e))}});var jn={};ze(jn,{clear:()=>wr,getItem:()=>br,removeItem:()=>Sr,setItem:()=>yr,storage:()=>ys});var yr,br,Sr,wr,ys,vr=S(()=>{"use strict";t();yr=({key:e,data:o,expires:n})=>{sessionStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},br=e=>{let o=sessionStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:Date.now()>=n.expires?(sessionStorage.removeItem(String(e)),null):n.data},Sr=e=>sessionStorage.removeItem(String(e)),wr=()=>sessionStorage.clear(),ys={setItem:yr,getItem:br,removeItem:Sr,clear:wr}});var ke,lo=S(()=>{"use strict";t();Xe();vr();ke={session:jn,local:zn}});function Ro(){return new Date().setHours(new Date().getHours()+24)}function Mo(e){ke.local.setItem({key:ae.loginExpiresAt,data:e,expires:e})}var qn=S(()=>{"use strict";t();lo();Xe()});var N=S(()=>{"use strict";t()});var uo=S(()=>{"use strict";t()});var kr=S(()=>{"use strict";t()});var j=S(()=>{"use strict";t()});var Ar=S(()=>{"use strict";t()});var Ir=S(()=>{"use strict";t()});var Er=S(()=>{"use strict";t()});var Lr=S(()=>{"use strict";t()});var ie=S(()=>{"use strict";t();kr();j();N();Ar();Ir();Er();Lr()});var Ye,Pr,kd,Cr,Ad,Dr,Id,Ed,bs,go=S(()=>{"use strict";t();ie();Ye={devnet:{id:"devnet",chainId:"D",name:"Devnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://devnet-wallet.elrond.com",apiAddress:"https://devnet-api.elrond.com",explorerAddress:"http://devnet-explorer.elrond.com",apiTimeout:"4000"},testnet:{id:"testnet",chainId:"T",name:"Testnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://testnet-wallet.elrond.com",apiAddress:"https://testnet-api.elrond.com",explorerAddress:"http://testnet-explorer.elrond.com",apiTimeout:"4000"},mainnet:{id:"mainnet",chainId:"1",name:"Mainnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://wallet.elrond.com",apiAddress:"https://api.elrond.com",explorerAddress:"https://explorer.elrond.com",apiTimeout:"4000"}},{chainId:Pr,egldLabel:kd}=Ye["devnet"],{chainId:Cr,egldLabel:Ad}=Ye["testnet"],{chainId:Dr,egldLabel:Id}=Ye["mainnet"],Ed={["devnet"]:Pr,["testnet"]:Cr,["mainnet"]:Dr},bs={[Pr]:"devnet",[Cr]:"testnet",[Dr]:"mainnet"}});var Kn=S(()=>{"use strict";t()});var Nr=S(()=>{"use strict";t()});var Xn,Yn=S(()=>{"use strict";t();Xn="transaction-status-toast-id"});var Fo,Zo,Rr,Mr,Jo,Fr,fo,C=S(()=>{"use strict";t();uo();go();Kn();Nr();Yn();Fo=5e4,Zo=1e9,Rr=1,Mr=1,Jo="logout",Fr="login",fo="0"});var Zn,H,Q,q=S(()=>{"use strict";t();Zn=require("@reduxjs/toolkit");C();H=(0,Zn.createAction)(Jo),Q=(0,Zn.createAction)(Fr,e=>({payload:e}))});var Br,Or,Ur,jd,ho,Ss,Me,To,Jn,Wr,$r,Qn=S(()=>{"use strict";t();Br=require("@reduxjs/toolkit");qn();N();q();Or={loginMethod:"",walletConnectLogin:null,ledgerLogin:null,tokenLogin:null,walletLogin:null,extensionLogin:null,isLoginSessionInvalid:!1},Ur=(0,Br.createSlice)({name:"loginInfoSlice",initialState:Or,reducers:{setLoginMethod:(e,o)=>{e.loginMethod=o.payload},setTokenLogin:(e,o)=>{e.tokenLogin=o.payload},setTokenLoginSignature:(e,o)=>{(e==null?void 0:e.tokenLogin)!=null&&(e.tokenLogin.signature=o.payload)},setWalletLogin:(e,o)=>{e.walletLogin=o.payload},setWalletConnectLogin:(e,o)=>{e.walletConnectLogin=o.payload},setLedgerLogin:(e,o)=>{e.ledgerLogin=o.payload},invalidateLoginSession:e=>{e.isLoginSessionInvalid=!0}},extraReducers:e=>{e.addCase(H,()=>Or),e.addCase(Q,(o,n)=>{o.loginMethod=n.payload.loginMethod,Mo(Ro())})}}),{setLoginMethod:jd,setWalletConnectLogin:ho,setLedgerLogin:Ss,setTokenLogin:Me,setTokenLoginSignature:To,setWalletLogin:Jn,invalidateLoginSession:Wr}=Ur.actions,$r=Ur.reducer});var Gr,Vr,Ae,Ze=S(()=>{"use strict";t();Gr=!1,Vr=(e=!0)=>{Gr=e},Ae=(e,o=0)=>{Gr||setTimeout(()=>{window.location.href=e},o)}});function xo(e,o){Boolean(e)&&e!=null&&setTimeout(()=>{window.location.pathname.includes(e)||(typeof o=="function"?o(e):Ae(e))},200)}var Hr=S(()=>{"use strict";t();Ze()});function et(e){return e[Math.floor(Math.random()*e.length)]}var zr=S(()=>{"use strict";t()});var ot=S(()=>{"use strict";t();Kn()});var yo=S(()=>{"use strict";t();Hr();zr();ot()});var jr,qr,nt,ws,Kr,tt,Xr,Yr,Qo=S(()=>{"use strict";t();jr=require("@reduxjs/toolkit"),qr=w(require("lodash.omit"));yo();nt={id:"not-configured",chainId:"",name:"NOT CONFIGURED",egldLabel:"",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"",walletConnectBridgeAddress:"",walletConnectV2RelayAddress:"",walletConnectV2ProjectId:"",walletAddress:"",apiAddress:"",explorerAddress:"",apiTimeout:"4000"},ws={network:nt,chainID:"-1"},Kr=(0,jr.createSlice)({name:"appConfig",initialState:ws,reducers:{initializeNetworkConfig:(e,o)=>{let n=et(o.payload.walletConnectBridgeAddresses),r=et(o.payload.walletConnectV2RelayAddresses),a=(0,qr.default)(o.payload,"walletConnectBridgeAddresses","walletConnectV2RelayAddresses");e.network=I(T(T({},e.network),a),{walletConnectBridgeAddress:n,walletConnectV2RelayAddress:r})},setChainID:(e,o)=>{e.chainID=o.payload}}}),{initializeNetworkConfig:tt,setChainID:Xr}=Kr.actions,Yr=Kr.reducer});var at,Zr,Jr,en,rt,Qr,Je,u_,on,ea,it,vs,g_,ct,oa,na,ta,st=S(()=>{"use strict";t();at=require("@elrondnetwork/erdjs"),Zr=require("@reduxjs/toolkit"),Jr=require("redux-persist");C();lo();Xe();q();en={balance:"...",address:"",nonce:0,txCount:0,scrCount:0,claimableRewards:fo},rt={address:"",websocketEvent:null,accounts:{"":en},ledgerAccount:null,publicKey:"",walletConnectAccount:null,isAccountLoading:!0,accountLoadingError:null},Qr=(0,Zr.createSlice)({name:"accountInfoSlice",initialState:rt,reducers:{setAddress:(e,o)=>{let n=o.payload;e.address=n,e.publicKey=new at.Address(n).hex()},setAccount:(e,o)=>{let n=e.address===o.payload.address;e.accounts={[e.address]:n?o.payload:en},e.isAccountLoading=!1,e.accountLoadingError=null},setAccountNonce:(e,o)=>{let{address:n}=e;e.accounts[n].nonce=o.payload},setAccountShard:(e,o)=>{e.shard=o.payload},setLedgerAccount:(e,o)=>{e.ledgerAccount=o.payload},updateLedgerAccount:(e,o)=>{e.ledgerAccount!=null&&(e.ledgerAccount.index=o.payload.index,e.ledgerAccount.address=o.payload.address)},setWalletConnectAccount:(e,o)=>{e.walletConnectAccount=o.payload},setIsAccountLoading:(e,o)=>{e.isAccountLoading=o.payload,e.accountLoadingError=null},setAccountLoadingError:(e,o)=>{e.accountLoadingError=o.payload,e.isAccountLoading=!1},setWebsocketEvent:(e,o)=>{e.websocketEvent={timestamp:Date.now(),message:o.payload}}},extraReducers:e=>{e.addCase(H,()=>(ke.local.removeItem(ae.loginExpiresAt),rt)),e.addCase(Q,(o,n)=>{let{address:r}=n.payload;o.address=r,o.publicKey=new at.Address(r).hex()}),e.addCase(Jr.REHYDRATE,(o,n)=>{var u;if(!((u=n.payload)!=null&&u.account))return;let{account:r}=n.payload,{address:a,shard:m,accounts:p,publicKey:_}=r;o.address=a,o.shard=m;let l=p&&a in p;o.accounts=l?p:rt.accounts,o.publicKey=_})}}),{setAccount:Je,setAddress:u_,setAccountNonce:on,setAccountShard:ea,setLedgerAccount:it,updateLedgerAccount:vs,setWalletConnectAccount:g_,setIsAccountLoading:ct,setAccountLoadingError:oa,setWebsocketEvent:na}=Qr.actions,ta=Qr.reducer});function Ie(e){return e!=null&&(Ds(e)||Fs(e))}function Qe(e){return e!=null&&(Ns(e)||Os(e))}function eo(e){return e!=null&&(Rs(e)||Bs(e))}function pt(e){return e!=null&&Ms(e)}function Ds(e){return e!=null&&ks.includes(e)}function Ns(e){return e!=null&&As.includes(e)}function Rs(e){return e!=null&&Is.includes(e)}function Ms(e){return e!=null&&Es.includes(e)}function Fs(e){return e!=null&&Ls.includes(e)}function Os(e){return e!=null&&Ps.includes(e)}function Bs(e){return e!=null&&Cs.includes(e)}var ks,As,Is,Es,Ls,Ps,Cs,So=S(()=>{"use strict";t();N();ks=["sent"],As=["success"],Is=["fail","cancelled","timedOut"],Es=["timedOut"],Ls=["pending"],Ps=["success"],Cs=["fail","invalid"]});var ra,aa,Oo,Us,ia,Bo,oo,Ws,Uo,$s,Gs,v_,Vs,ca,mt,nn,tn=S(()=>{"use strict";t();ra=require("@reduxjs/toolkit"),aa=require("redux-persist");N();So();q();Oo={signedTransactions:{},transactionsToSign:null,signTransactionsError:null,signTransactionsCancelMessage:null,customTransactionInformationForSessionId:{}},Us={signWithoutSending:!1,sessionInformation:null,redirectAfterSign:!1},ia=(0,ra.createSlice)({name:"transactionsSlice",initialState:Oo,reducers:{moveTransactionsToSignedState:(e,o)=>{var l,u;let{sessionId:n,transactions:r,errorMessage:a,status:m,redirectRoute:p}=o.payload,_=((l=e.customTransactionInformationForSessionId)==null?void 0:l[n])||Us;e.signedTransactions[n]={transactions:r,status:m,errorMessage:a,redirectRoute:p,customTransactionInformation:_},((u=e==null?void 0:e.transactionsToSign)==null?void 0:u.sessionId)===n&&(e.transactionsToSign=Oo.transactionsToSign)},clearSignedTransaction:(e,o)=>{e.signedTransactions[o.payload]&&delete e.signedTransactions[o.payload]},clearTransactionToSign:e=>{e!=null&&e.transactionsToSign&&(e.transactionsToSign=null)},updateSignedTransaction:(e,o)=>{e.signedTransactions=T(T({},e.signedTransactions),o.payload)},updateSignedTransactions:(e,o)=>{let{sessionId:n,status:r,errorMessage:a,transactions:m}=o.payload;e.signedTransactions[n]!=null&&(e.signedTransactions[n].status=r,a!=null&&(e.signedTransactions[n].errorMessage=a),m!=null&&(e.signedTransactions[n].transactions=m))},updateSignedTransactionStatus:(e,o)=>{var _,l,u,f,h,A;let{sessionId:n,status:r,errorMessage:a,transactionHash:m}=o.payload,p=(l=(_=e.signedTransactions)==null?void 0:_[n])==null?void 0:l.transactions;if(p!=null){e.signedTransactions[n].transactions=p.map(F=>F.hash===m?I(T({},F),{status:r,errorMessage:a}):F);let k=(f=(u=e.signedTransactions[n])==null?void 0:u.transactions)==null?void 0:f.every(F=>Qe(F.status)),L=(A=(h=e.signedTransactions[n])==null?void 0:h.transactions)==null?void 0:A.every(F=>eo(F.status));k&&(e.signedTransactions[n].status="success"),L&&(e.signedTransactions[n].status="fail")}},setTransactionsToSign:(e,o)=>{e.transactionsToSign=o.payload;let{sessionId:n,customTransactionInformation:r}=o.payload;e.customTransactionInformationForSessionId[n]=r,e.signTransactionsError=null},clearAllTransactionsToSign:e=>{e.transactionsToSign=Oo.transactionsToSign,e.signTransactionsError=null},clearAllSignedTransactions:e=>{e.signedTransactions=Oo.signedTransactions},setSignTransactionsError:(e,o)=>{e.signTransactionsError=o.payload},setSignTransactionsCancelMessage:(e,o)=>{e.signTransactionsCancelMessage=o.payload}},extraReducers:e=>{e.addCase(H,()=>Oo),e.addCase(aa.REHYDRATE,(o,n)=>{var p;if(!((p=n.payload)!=null&&p.transactions))return;let{signedTransactions:r,customTransactionInformationForSessionId:a}=n.payload.transactions,m=Object.entries(r).reduce((_,[l,u])=>{let f=new Date(l),h=new Date;return h.setHours(h.getHours()+5),h-f>0||(_[l]=u),_},{});a!=null&&(o.customTransactionInformationForSessionId=a),r!=null&&(o.signedTransactions=m)})}}),{updateSignedTransactionStatus:Bo,updateSignedTransactions:oo,setTransactionsToSign:Ws,clearAllTransactionsToSign:Uo,clearAllSignedTransactions:$s,clearSignedTransaction:Gs,clearTransactionToSign:v_,setSignTransactionsError:Vs,setSignTransactionsCancelMessage:ca,moveTransactionsToSignedState:mt}=ia.actions,nn=ia.reducer});var sa,_t,lt,ut,Hs,dt,pa,E_,zs,ma,rn,an=S(()=>{"use strict";t();sa=require("@reduxjs/toolkit");q();_t="Transaction failed",lt="Transaction successful",ut="Processing transaction",Hs="Transaction submitted",dt={},pa=(0,sa.createSlice)({name:"transactionsInfo",initialState:dt,reducers:{setTransactionsDisplayInfo(e,o){let{sessionId:n,transactionsDisplayInfo:r}=o.payload;n!=null&&(e[n]={errorMessage:(r==null?void 0:r.errorMessage)||_t,successMessage:(r==null?void 0:r.successMessage)||lt,processingMessage:(r==null?void 0:r.processingMessage)||ut,submittedMessage:(r==null?void 0:r.submittedMessage)||Hs,transactionDuration:r==null?void 0:r.transactionDuration})},clearTransactionsInfoForSessionId(e,o){o.payload!=null&&delete e[o.payload]},clearTransactionsInfo:()=>dt},extraReducers:e=>{e.addCase(H,()=>dt)}}),{clearTransactionsInfo:E_,setTransactionsDisplayInfo:zs,clearTransactionsInfoForSessionId:ma}=pa.actions,rn=pa.reducer});var _a,da,la,ua,js,D_,qs,ga,gt=S(()=>{"use strict";t();_a=require("@reduxjs/toolkit");q();da={},la=(0,_a.createSlice)({name:"modalsSlice",initialState:da,reducers:{setTxSubmittedModal:(e,o)=>{e.txSubmittedModal=o.payload},setNotificationModal:(e,o)=>{e.notificationModal=o.payload},clearTxSubmittedModal:e=>{e.txSubmittedModal=void 0},clearNotificationModal:e=>{e.notificationModal=void 0}},extraReducers:e=>{e.addCase(H,()=>da)}}),{setTxSubmittedModal:ua,setNotificationModal:js,clearTxSubmittedModal:D_,clearNotificationModal:qs}=la.actions,ga=la.reducer});var ha,fa,Ta,Ks,xa,W_,$_,G_,V_,cn,sn=S(()=>{"use strict";t();ha=require("@reduxjs/toolkit");Yn();ie();Gn();q();fa={customToasts:[],transactionToasts:[],failTransactionToast:null},Ta=(0,ha.createSlice)({name:"toastsSlice",initialState:fa,reducers:{addCustomToast:(e,o)=>{e.customToasts.push(I(T({},o.payload),{type:"custom",toastId:o.payload.toastId||`custom-toast-${e.customToasts.length+1}`}))},removeCustomToast:(e,o)=>{e.customToasts=e.customToasts.filter(n=>n.toastId!==o.payload)},addTransactionToast:(e,o)=>{e.transactionToasts.push({type:"transaction",startTimestamp:Yo(),toastId:o.payload||`custom-toast-${e.transactionToasts.length+1}`})},removeTransactionToast:(e,o)=>{e.transactionToasts=e.transactionToasts.filter(n=>n.toastId!==o.payload)},addFailTransactionToast:(e,o)=>{e.failTransactionToast=I(T({},o.payload),{toastId:Xn})},removeFailTransactionToast:e=>{e.failTransactionToast=null}},extraReducers:e=>{e.addCase(H,()=>fa)}}),{addCustomToast:Ks,removeCustomToast:xa,addTransactionToast:W_,removeTransactionToast:$_,addFailTransactionToast:G_,removeFailTransactionToast:V_}=Ta.actions,cn=Ta.reducer});var E=S(()=>{"use strict";t();Qn();Qo();st();tn();an();gt();sn()});var ft=S(()=>{"use strict";t()});var wa,Sa,va,pl,ml,dl,_l,ll,ul,ka,Aa=S(()=>{"use strict";t();wa=require("@reduxjs/toolkit");q();Sa={visible:!1,modal:{showHeader:!0,showFooter:!1,headerText:"",footerText:""}},va=(0,wa.createSlice)({name:"dappModalsSlice",initialState:Sa,reducers:{setDappModalVisibility:(e,o)=>{e.visible=o.payload},setDappModalHeaderVisibility:(e,o)=>{e.modal.showHeader=o.payload},setDappModalFooterVisibility:(e,o)=>{e.modal.showFooter=o.payload},setDappModalHeaderText:(e,o)=>{e.modal.headerText=o.payload},setDappModalFooterText:(e,o)=>{e.modal.footerText=o.payload},setDappModalConfig:(e,o)=>{e.modal=T({},o.payload)}},extraReducers:e=>{e.addCase(H,()=>Sa)}}),{setDappModalVisibility:pl,setDappModalHeaderVisibility:ml,setDappModalFooterVisibility:dl,setDappModalHeaderText:_l,setDappModalFooterText:ll,setDappModalConfig:ul}=va.actions,ka=va.reducer});var Ia,Js,Qs,wo,mn=S(()=>{"use strict";t();Ia=require("@reduxjs/toolkit");ft();st();Aa();Qn();gt();Qo();sn();an();tn();Js={["account"]:ta,["networkConfig"]:Yr,["loginInfo"]:$r,["modals"]:ga,["dappModal"]:ka,["toasts"]:cn,["transactions"]:nn,["transactionsInfo"]:rn},Qs=(e={})=>(0,Ia.combineReducers)(T(T({},Js),e)),wo=Qs});var Pa={};ze(Pa,{default:()=>ip});function ht(e,o=[]){return{key:e,version:1,storage:La.default,blacklist:o}}var no,Ea,La,ep,op,np,tp,rp,ap,ip,Ca=S(()=>{"use strict";t();no=require("redux-persist"),Ea=w(require("redux-persist/lib/storage")),La=w(require("redux-persist/lib/storage/session"));mn();E();sn();an();tn();ft();ep={2:e=>I(T({},e),{networkConfig:nt})};op=ht("dapp-core-transactionsInfo"),np=ht("dapp-core-transactions",["transactionsToSign"]),tp=ht("dapp-core-toasts"),rp={key:"dapp-core-store",version:2,storage:Ea.default,whitelist:["account","loginInfo","modals","networkConfig"],migrate:(0,no.createMigrate)(ep,{debug:!1})},ap={["toasts"]:(0,no.persistReducer)(tp,cn),["transactions"]:(0,no.persistReducer)(np,nn),["transactionsInfo"]:(0,no.persistReducer)(op,rn)},ip=(0,no.persistReducer)(rp,wo(ap))});var Da={};ze(Da,{default:()=>cp});var cp,Na=S(()=>{"use strict";t();mn();cp=wo()});var Ra={};ze(Ra,{default:()=>pp});var ce,sp,pp,Ma=S(()=>{"use strict";t();ce=require("redux-persist"),sp=[ce.FLUSH,ce.REHYDRATE,ce.PAUSE,ce.PERSIST,ce.PURGE,ce.REGISTER],pp=sp});var Ba={};ze(Ba,{default:()=>Oa});function Oa(e){return(0,Fa.persistStore)(e)}var Fa,Ua=S(()=>{"use strict";t();Fa=require("redux-persist")});var Hm={};ze(Hm,{AppInitializer:()=>Ot,AuthenticatedRoutesWrapper:()=>Ai,DappProvider:()=>Vm});module.exports=No(Hm);t();t();var It=w(require("react"));t();var ja=require("react"),vo=require("react-redux");t();var Tt=require("@reduxjs/toolkit"),Va=require("react-redux/es/utils/Subscription");E();t();var ya=w(require("lodash.throttle"));C();E();qn();lo();Xe();var Xs=[Jo],Ys=(0,ya.default)(()=>{Mo(Ro())},5e3),ba=e=>o=>n=>{var l;if(Xs.includes(n.type))return o(n);let r=e.getState(),a=ke.local.getItem(ae.loginExpiresAt);if(!((l=r==null?void 0:r.loginInfo)==null?void 0:l.isLoggedIn))return o(n);if(a==null)return Mo(Ro());let p=Date.now();return a-p<0?setTimeout(()=>g(void 0,null,function*(){console.log("session expired");try{e.dispatch(Wr())}catch(u){console.error("error logging out",u)}}),1e3):(Ys(),o(n))};t();t();function pn(){return typeof sessionStorage!="undefined"}var Wa=pn()?(Ca(),No(Pa)).default:(Na(),No(Da)).default,$a=pn()?(Ma(),No(Ra)).default:[],Ga=pn()?(Ua(),No(Ba)).default:e=>e;mn();var b=(0,Tt.configureStore)({reducer:Wa,middleware:e=>e({serializableCheck:{ignoredActions:[...$a,on.type,Je.type],ignoredPaths:["payload.nonce","account.account.nonce","providers.accountProvider"]}}).concat(ba)}),Ha=(0,Va.createSubscription)(b),za=Ga(b),Ul=(0,Tt.configureStore)({reducer:wo});var mp={store:b,subscription:Ha},Wo=(0,ja.createContext)(mp),Vl=(0,vo.createStoreHook)(Wo),B=(0,vo.createDispatchHook)(Wo),y=(0,vo.createSelectorHook)(Wo);t();t();E();t();var qa=w(require("lodash.isequal")),dn=require("reselect"),x=(0,dn.createSelectorCreator)(dn.defaultMemoize,qa.default);var Ee=e=>e.account,me=x(Ee,e=>e.address),Oe=x(Ee,me,(e,o)=>o in e.accounts?e.accounts[o]:en),Ka=x(Ee,Oe,(e,o)=>{let a=e,{accounts:n}=a,r=He(a,["accounts"]);return I(T({},r),{account:o})}),dp=x(Oe,e=>e.balance),Xa=x(Oe,e=>{var o;return((o=e==null?void 0:e.nonce)==null?void 0:o.valueOf())||0}),Ya=x(Ee,e=>e.shard),xt=x(Ee,e=>e.ledgerAccount),Kl=x(Ee,e=>e.walletConnectAccount),Za=x(Ee,e=>e.isAccountLoading),Xl=x(Ee,e=>e.accountLoadingError),yt=x(Ee,e=>e.websocketEvent);t();var Le=e=>e.networkConfig,$o=x(Le,e=>e.chainID),Ja=x(Le,e=>e.network.walletConnectBridgeAddress),Qa=x(Le,e=>e.network.walletConnectV2RelayAddress),ei=x(Le,e=>e.network.walletConnectV2ProjectId),_n=x(Le,e=>e.network.walletConnectDeepLink),he=x(Le,e=>e.network),ko=x(he,e=>e.apiAddress),oi=x(he,e=>e.explorerAddress),ni=x(he,e=>e.egldLabel);t();N();var to=e=>e.loginInfo,ti=x(to,e=>e.loginMethod),Te=x(to,me,(e,o)=>e.loginMethod!=""&&Boolean(o)),ri=x(to,e=>e.walletConnectLogin),ai=x(to,e=>e.ledgerLogin),Ao=x(to,e=>e.walletLogin),ii=x(to,e=>e.isLoginSessionInvalid);t();t();var oe=require("@elrondnetwork/erdjs");C();t();function bt(e){try{return i.Buffer.from(e,"base64").toString()===atob(e)}catch(o){return!1}}function Io(e){var a,m,p;let{data:o}=e,n=bt(o!=null?o:"")?oe.TransactionPayload.fromEncoded(o):new oe.TransactionPayload(o),r=new oe.Transaction(T({value:e.value.valueOf(),data:n,nonce:e.nonce.valueOf(),receiver:new oe.Address(e.receiver),sender:new oe.Address(e.sender),gasLimit:(a=e.gasLimit.valueOf())!=null?a:Fo,gasPrice:(m=e.gasPrice.valueOf())!=null?m:Zo,chainID:e.chainID.valueOf(),version:new oe.TransactionVersion((p=e.version)!=null?p:Rr)},e.options?{options:new oe.TransactionOptions(e.options)}:{}));return r.applySignature({hex:()=>e.signature||""},new oe.Address(e.sender)),r}t();So();t();var lp=w(require("bignumber.js"));ie();t();t();t();t();var gp=require("@elrondnetwork/erdjs"),fp=w(require("bignumber.js"));N();t();var pi=require("@elrondnetwork/erdjs");function up(e){try{let o=new pi.Address(e);return Boolean(o.bech32())}catch(o){return!1}}function Go(e){return(e==null?void 0:e.startsWith("erd"))&&e.length===62&&/^\w+$/.test(e)&&up(e)}t();t();t();var Tp=require("@elrondnetwork/erdjs/out");t();C();t();var mi=require("@elrondnetwork/erdjs");ie();t();t();t();t();N();t();t();t();t();t();t();t();var di="dapp/config",_i="network/config";var ln="accounts";t();t();t();t();j();t();t();j();t();t();t();var Ip=w(require("bignumber.js"));C();t();var Ap=require("@elrondnetwork/erdjs"),St=w(require("bignumber.js"));C();t();var vp=w(require("bignumber.js"));t();St.default.config({ROUNDING_MODE:St.default.ROUND_FLOOR});t();t();t();C();t();t();t();t();C();t();C();t();var Ep=require("@elrondnetwork/erdjs");C();t();var fn=require("@elrondnetwork/erdjs"),Cp=require("@elrondnetwork/erdjs"),Dp=require("@elrondnetwork/erdjs-network-providers");C();t();t();var Lp=w(require("bignumber.js"));C();t();C();t();t();t();t();t();C();t();j();var Rp=["reDelegateRewards","claimRewards","unBond"],Mp=["wrapEgld","unwrapEgld"],Fp=["unStake"],Op=["unDelegate"];t();t();t();C();t();var $p=w(require("bignumber.js"));t();t();t();j();t();var Gp=w(require("bignumber.js"));t();t();t();j();t();j();t();var zp=w(require("bignumber.js"));C();j();t();j();t();j();t();var jp=w(require("bignumber.js"));t();N();t();N();t();t();var qp=require("react");j();var WT=["create","localMint","ESDTLocalMint","addQuantity","burn","localBurn","ESDTLocalBurn","wipe","writeLog","signalError"];t();var Kp=w(require("anchorme"));t();var hi=require("react");ie();t();t();j();t();var Xp=require("@elrondnetwork/erdjs/out"),Yp=w(require("bignumber.js"));j();t();j();t();t();N();var hx=[{label:"Raw",value:"raw"},{label:"Text",value:"text"},{label:"Decimal",value:"decimal"},{label:"Smart",value:"smart"}];t();var Ti=require("react");ie();t();t();t();t();t();C();var Tn=e=>e.transactions,Ue=x(Tn,e=>e.signedTransactions),Jp=x(Tn,e=>e.signTransactionsError),Qp=x(Tn,e=>e.signTransactionsCancelMessage),xn=e=>o=>Object.entries(o).reduce((n,[r,a])=>(e(a.status)&&(n[r]=a),n),{}),vt=x(Ue,xn(Ie)),xi=x(Ue,xn(Qe)),yi=x(Ue,xn(eo)),em=x(Ue,xn(pt)),kt=x(Tn,e=>{var o;return(e==null?void 0:e.transactionsToSign)==null?null:I(T({},e.transactionsToSign),{transactions:((o=e==null?void 0:e.transactionsToSign)==null?void 0:o.transactions.map(n=>Io(n)))||[]})}),om=x(Ue,(e,o)=>o,(e,o)=>o!=null?(e==null?void 0:e[o])||{}:{});t();E();var bi={errorMessage:_t,successMessage:lt,processingMessage:ut},nm=e=>e.transactionsInfo,tm=x(nm,(e,o)=>o,(e,o)=>o!=null&&(e==null?void 0:e[Number(o)])||bi);t();var Si=e=>e.modals,uy=x(Si,e=>e.txSubmittedModal),rm=x(Si,e=>e.notificationModal);t();var At=e=>e.toasts,hy=x(At,e=>e.customToasts),Ty=x(At,e=>e.transactionToasts),xy=x(At,e=>e.failTransactionToast);Ze();t();t();function wi(e,o){if(!e){typeof console!="undefined"&&console.warn(o);try{throw new Error(o)}catch(n){}}}function vi(e,o){typeof e=="string"&&(e={path:e,caseSensitive:!1,end:!0});let[n,r]=am(e.path,e.caseSensitive,e.end),a=o.match(n);if(!a)return null;let m=a[0],p=m.replace(/(.)\/+$/,"$1"),_=a.slice(1);return{params:r.reduce((u,f,h)=>{if(f==="*"){let A=_[h]||"";p=m.slice(0,m.length-A.length).replace(/(.)\/+$/,"$1")}return u[f]=im(_[h]||"",f),u},{}),pathname:m,pathnameBase:p,pattern:e}}function am(e,o=!1,n=!0){wi(e==="*"||!e.endsWith("*")||e.endsWith("/*"),`Route path "${e}" will be treated as if it were "${e.replace(/\*$/,"/*")}" because the \`*\` character must always follow a \`/\` in the pattern. To get rid of this warning, please change the route path to "${e.replace(/\*$/,"/*")}".`);let r=[],a="^"+e.replace(/\/*\*?$/,"").replace(/^\/*/,"/").replace(/[\\.*+^$?{}|()[\]]/g,"\\$&").replace(/:(\w+)/g,(p,_)=>(r.push(_),"([^\\/]+)"));return e.endsWith("*")?(r.push("*"),a+=e==="*"||e==="/*"?"(.*)$":"(?:\\/(.+)|\\/*)$"):a+=n?"\\/*$":"(?:(?=[.~-]|%[0-9A-F]{2})|\\b|\\/|$)",[new RegExp(a,o?void 0:"i"),r]}function im(e,o){try{return decodeURIComponent(e)}catch(n){return wi(!1,`The value for the URL param "${o}" will not be decoded because the string "${e}" is a malformed URL segment. This is probably due to a bad percent encoding (${n}).`),e}}var ki=(e,o)=>e.filter(({authenticatedRoute:a})=>Boolean(a)).some(({path:a})=>vi(a,o)!==null);var Ai=({children:e,routes:o,unlockRoute:n,onRedirect:r})=>{let a=y(Te),m=y(Za),p=y(Ao),l=ki(o,window.location.pathname)&&!a&&p==null;return m||p?null:l?(r?r(n):Ae(n),null):It.default.createElement(It.default.Fragment,null,e)};t();var ye=w(require("react")),zi=require("@elrondnetwork/erdjs/out");t();t();t();var Ii=w(require("axios"));function Ei(e){return g(this,null,function*(){let{apiAddress:o,apiTimeout:n}=he(b.getState()),r=e.map(m=>Ii.default.post(`${o}/transactions`,m.toPlainObject(),{timeout:parseInt(n)}));return(yield Promise.all(r)).map(({data:m})=>m.txHash)})}t();var Li=w(require("axios"));function yn(e){return g(this,null,function*(){let o=ko(b.getState()),n=e.map(a=>a.hash),{data:r}=yield Li.default.get(`${o}/transactions`,{params:{hashes:n.join(","),withScResults:!0}});return e.map(({hash:a,previousStatus:m})=>{let p=r.find(_=>(_==null?void 0:_.txHash)===a);return{hash:a,data:p==null?void 0:p.data,invalidTransaction:p==null,status:p==null?void 0:p.status,results:p==null?void 0:p.results,sender:p==null?void 0:p.sender,receiver:p==null?void 0:p.receiver,previousStatus:m,hasStatusChanged:p&&p.status!==m}})})}t();var sm=w(require("axios"));t();t();var pm=w(require("axios"));t();var mm=w(require("axios"));t();t();t();var Ci=w(require("axios"));t();t();var bn=e=>{let o=e!=null?e:ko(b.getState());return o.endsWith("/")?o.slice(0,-1):o};function Di(e){return g(this,null,function*(){if(!e)return null;let n=`${bn()}/${ln}/${e}`;try{let{data:r}=yield Ci.default.get(n);return r}catch(r){console.error("error fetching configuration for ",n)}return null})}t();t();go();t();var Ni=w(require("axios"));function Et(e){return g(this,null,function*(){let o=`${e}/${di}`;try{let{data:r}=yield Ni.default.get(o);if(r!=null){let a="egldDenomination";if(a in r){let n=r,{[a]:m,decimals:p}=n,_=He(n,[Kt(a),"decimals"]);return I(T({},_),{decimals:m,digits:p})}return r}}catch(r){console.error("error fetching configuration for ",o)}return null})}t();var Ri=w(require("axios"));function Mi(){return g(this,null,function*(){var n;let o=`${bn()}/${_i}`;try{let{data:r}=yield Ri.default.get(o);if(r!=null)return(n=r==null?void 0:r.data)==null?void 0:n.config}catch(r){console.error("error fetching configuration for ",o)}return null})}t();go();t();go();t();var dm=w(require("axios"));go();t();var Vo=()=>y(Ka);Qo();t();var Fi=w(require("axios"));E();function Lt(){return g(this,null,function*(){let e=b.getState(),o=ko(e),n=me(e),r=Ya(e);try{if(r==null&&n){let{data:a}=yield Fi.default.get(`${o}/accounts/${n}`);return b.dispatch(ea(a.shard)),a.shard}else return r}catch(a){return console.error(a),null}})}t();t();t();t();var Oi=require("@elrondnetwork/erdjs-extension-provider"),Bi=require("@elrondnetwork/erdjs-hw-provider"),wn=require("@elrondnetwork/erdjs-wallet-connect-provider"),Pt=require("@elrondnetwork/erdjs-web-wallet-provider");C();N();var _m="/dapp/init",ro=e=>{switch(e==null?void 0:e.constructor){case Pt.WalletProvider:return"wallet";case wn.WalletConnectProvider:return"walletconnect";case wn.WalletConnectV2Provider:return"walletconnectv2";case Bi.HWProvider:return"ledger";case Oi.ExtensionProvider:return"extension";case Sn:return"";default:return"extra"}},Ct=e=>new Pt.WalletProvider(`${e}${_m}`),Dt=e=>g(void 0,null,function*(){if(!e.isInitialized())throw new Error("Unable to get version. Provider not initialized");let o=e.hwApp,{contractData:n,version:r}=yield o.getAppConfiguration();return{version:r,dataEnabled:n===Mr}}),Pe=e=>`Unable to perform ${e}, Provider not initialized`,Sn=class{init(){return Promise.resolve(!1)}login(o){throw new Error(Pe(`login with options: ${o}`))}logout(o){throw new Error(Pe(`logout with options: ${o}`))}getAddress(){throw new Error(Pe("getAddress"))}isInitialized(){return!1}isConnected(){return Promise.resolve(!1)}sendTransaction(o,n){throw new Error(Pe(`sendTransaction with transactions: ${o} options: ${n}`))}signTransaction(o,n){throw new Error(Pe(`signTransaction with transactions: ${o} options: ${n}`))}signTransactions(o,n){throw new Error(Pe(`signTransactions with transactions: ${o} options: ${n}`))}signMessage(o,n){throw new Error(Pe(`signTransactions with ${o} and options ${n}`))}sendCustomMessage({method:o,params:n}){throw new Error(Pe(`sendCustomMessage with method: ${o} params: ${n}`))}sendCustomRequest(o){throw new Error(Pe(`sendSessionEvent with options: ${o}`))}ping(){return Promise.resolve(!1)}},Nt=new Sn;var Mt=Nt,Rt=null;function xe(e){Mt=e}function Ui(e){Rt=e}function Wi(){Rt!=null&&(Mt=Rt)}function Z(){return Mt||Nt}q();ie();t();function de(){return Te(b.getState())}t();t();N();t();t();function $i(){let e=Z();return ro(e)}function se(e){return $i()===e}function We(){let{search:e}=window.location,o=b.getState(),n=Z(),r=me(o),a=Te(o),m=Ao(o);if(!n)throw"provider not initialized";return se("ledger")&&a?new Promise(p=>{p(r)}):!se("")&&!se("wallet")&&!se("extra")?n.getAddress():new Promise(p=>{if(m!=null){let _=new URLSearchParams(e),l=Object.fromEntries(_);Go(l.address)&&p(l.address)}a&&p(r),p("")})}t();function $e(e){return g(this,null,function*(){return yield Di(e)})}t();E();function Gi(e){b.dispatch(on(e))}t();function Eo(e){let o=b.getState(),n=Xa(o);return e?n&&!isNaN(n)?Math.max(n,e.nonce):e.nonce:n}t();t();E();var Vi=()=>g(void 0,null,function*(){try{let e=yield We();try{let o=yield $e(e);if(o!=null){let n=I(T({},o),{nonce:Eo(o)});return b.dispatch(Je(n)),n}}catch(o){console.error("Failed getting account ",o)}}catch(e){console.error("Failed getting address ",e)}return null});function vn(){return g(this,null,function*(){let e=Z();if(e==null)throw"Provider not initialized";if(!e.isInitialized||e.isInitialized())return Vi();try{if(!e.init)throw"Current provider does not have init() function";return(yield e.init())?Vi():void 0}catch(o){console.error("Failed initializing provider ",o)}})}t();var Hi=require("@elrondnetwork/erdjs");t();q();ie();Ze();lo();Xe();var lm=e=>{ke.local.setItem({key:ae.logoutEvent,data:e,expires:0}),ke.local.removeItem(ae.logoutEvent)};function pe(e,o){return g(this,null,function*(){let n=Z(),r=ro(n),a=de(),m=r==="wallet";if(!a||!n){Ft(e,o,!1);return}try{let p=yield We();lm(p)}catch(p){Ft(e,o,!1),console.error("error fetching logout address",p)}m&&Vr(),b.dispatch(H());try{let _=m&&!e?window.location.origin:e;yield n.logout({callbackUrl:_}),Ft(e,o,m)}catch(p){console.error("error logging out",p)}})}function Ft(e,o,n){e&&!n&&(typeof o=="function"?o(e):Ae(e))}function Ot({customNetworkConfig:e={},children:o,environment:n}){let[r,a]=(0,ye.useState)(!1),m=Vo(),p=y(ii),{address:_,publicKey:l}=m,u=B();function f(){return g(this,null,function*(){let A=!(e!=null&&e.skipFetchFromServer),k=e==null?void 0:e.apiAddress,L=Ye[n]||{},F=T(T({},L),e);if(A){let R=L==null?void 0:L.apiAddress,ne=yield Et(k||R);if(ne!=null){let _e=T(T(T({},L),ne),e);u(tt(_e));return}}u(tt(F))})}function h(){return g(this,null,function*(){yield f(),a(!0),Lt()})}return(0,ye.useEffect)(()=>{!_||Lt()},[_]),(0,ye.useEffect)(()=>{_&&new zi.Address(_).hex()!==l&&pe()},[_,l]),(0,ye.useEffect)(()=>{h()},[e,n]),(0,ye.useEffect)(()=>{p&&pe()},[p]),r?ye.default.createElement(ye.default.Fragment,null,o):null}t();var Co=w(require("react")),Pc=require("react-redux"),Cc=require("redux-persist/integration/react");t();var io=require("react"),Ki=require("@elrondnetwork/erdjs-extension-provider"),Xi=require("@elrondnetwork/erdjs-hw-provider");t();var Ge=require("react"),ji=require("@elrondnetwork/erdjs-wallet-connect-provider");t();var kn=()=>{let e=Z(),o=ro(e);return{provider:e,providerType:o}};t();var An=require("react");function Lo(e,o=[]){let n=(0,An.useRef)(!0);(0,An.useEffect)(()=>{n.current?n.current=!1:e()},o)}q();E();N();yo();var Bt=({logoutRoute:e,callbackRoute:o,token:n,onLoginRedirect:r})=>{let a=B(),m=15e3,[p,_]=(0,Ge.useState)(""),[l,u]=(0,Ge.useState)(""),{provider:f}=kn(),h=y(Ja),A=y(_n),k=(0,Ge.useRef)(f),L=(0,Ge.useRef)(!0),F,R=Boolean(l),ne=!R,_e=R?`${A}?wallet-connect=${encodeURIComponent(l)}`:"";(0,Ge.useEffect)(()=>{De();let M=setInterval(()=>{De()},m);return()=>clearInterval(M)},[f]),Lo(()=>{so()},[n]),Lo(()=>{k.current=f},[f]);let be=()=>{pe(e)},J=()=>{L.current=!1};function De(){return g(this,null,function*(){var K,U;if(f==null||!se("walletconnect")||!Boolean((U=(K=k.current)==null?void 0:K.walletConnector)==null?void 0:U.connected))return;let G={method:"heartbeat",params:{}};try{yield k.current.sendCustomMessage(G)}catch(X){console.error("Connection lost",X),be()}})}function Ne(){return g(this,null,function*(){var M;try{let G=k.current;if(de()||G==null||!se("walletconnect"))return;if(!L.current){try{yield(M=k.current)==null?void 0:M.logout()}catch(z){console.warn("Unable to logout")}return}let U=yield G.getAddress();if(!U){console.warn("Login cancelled.");return}let X=yield G.getSignature(),D=Boolean(X),W={address:U,loginMethod:"walletconnect"},V={logoutRoute:e,loginType:"walletConnect",callbackRoute:o!=null?o:window.location.href};D?(a(ho(V)),a(To(X))):a(ho(V)),a(Q(W)),G.walletConnector.on("heartbeat",()=>{clearInterval(F),F=setInterval(()=>{console.log("Maiar Wallet Connection Lost"),be(),clearInterval(F)},15e4)}),xo(o,r)}catch(G){_("Invalid address"),console.error(G)}})}function Ve(M=!0){return g(this,null,function*(){var X,D;let G=M&&!l;if(!h||((D=(X=k==null?void 0:k.current)==null?void 0:X.isInitialized)==null?void 0:D.call(X))&&!G)return;let K={onClientLogin:Ne,onClientLogout:be},U=new ji.WalletConnectProvider(h,K);yield U.init(),L.current=!0,xe(U),k.current=U,M&&so()})}function so(){return g(this,null,function*(){var U;if(!h)return;let M=yield(U=k.current)==null?void 0:U.login();if(!Boolean(M))return;if(!n){u(M);return}let K=`${M}&token=${n}`;u(K),a(Me({loginToken:n}))})}let v=Boolean(p),P=de();return[Ve,{error:p,loginFailed:v,isLoading:ne&&!v,isLoggedIn:P&&!v},{uriDeepLink:_e,walletConnectUri:l,cancelLogin:J}]};t();var ao=require("react"),qi=require("@elrondnetwork/erdjs-wallet-connect-provider");q();E();N();yo();var Ut=({callbackRoute:e,logoutRoute:o,token:n,onLoginRedirect:r})=>{let a=B(),[m,p]=(0,ao.useState)(""),[_,l]=(0,ao.useState)(""),[u,f]=(0,ao.useState)([]),h=Z(),A=y(Qa),k=y(ei),L=y($o),F=y(_n),R=(0,ao.useRef)(h),ne=(0,ao.useRef)(!0),_e=Boolean(_),be=["erd_cancelAction"],J=!_e,De=_e?`${F}?wallet-connect=${encodeURIComponent(_)}`:"";Lo(()=>{K()},[n]),Lo(()=>{R.current=h},[h]);let Ne=()=>{pe(o)},Ve=D=>{console.log("wc2 session event: ",D)},so=()=>{ne.current=!1};function v(){return g(this,null,function*(){var D;try{let W=R.current;if(de()||W==null||!se("walletconnectv2"))return;if(!ne.current){try{yield(D=R.current)==null?void 0:D.logout()}catch(zm){console.warn("Unable to logout")}return}let z=yield W.getAddress();if(!z){console.warn("Login cancelled.");return}let te=yield W.getSignature(),Re=Boolean(te),Dc={address:z,loginMethod:"walletconnectv2"},zt={logoutRoute:o,loginType:"walletconnectv2",callbackRoute:e!=null?e:window.location.href};Re?(a(ho(zt)),a(To(te))):a(ho(zt)),a(Q(Dc)),xo(e,r)}catch(W){p("Invalid address"),console.error(W)}})}function P(D=!0){return g(this,null,function*(){var te,Re;let W=D&&!_;if(!k||!A||((Re=(te=R==null?void 0:R.current)==null?void 0:te.isInitialized)==null?void 0:Re.call(te))&&!W){p("Invalid WalletConnect setup");return}let V={onClientLogin:v,onClientLogout:Ne,onClientEvent:Ve},z=new qi.WalletConnectV2Provider(V,L,A,k);yield z.init(),ne.current=!0,xe(z),f(z.pairings),R.current=z,D&&K()})}function M(D){return g(this,null,function*(){var W,V,z;if(!A||!k){p("Invalid WalletConnect setup");return}if(!D||!D.topic){p("Expired connection");return}try{let{approval:te}=yield(W=R.current)==null?void 0:W.connect({topic:D.topic,methods:be});n&&a(Me({loginToken:n}));try{yield(V=R.current)==null?void 0:V.login({approval:te,token:n})}catch(Re){console.warn("User rejected existing connection proposal",Re),yield P()}}catch(te){console.error("Unable to connect to existing session",te),p("Unable to connect to existing session")}finally{f((z=R.current)==null?void 0:z.pairings)}})}function G(D){return g(this,null,function*(){var W,V;try{D&&(yield(W=R.current)==null?void 0:W.logout({topic:D}))}catch(z){console.error("Unable to remove existing pairing",z),p("Unable to remove existing pairing")}finally{f((V=R.current)==null?void 0:V.pairings)}})}function K(){return g(this,null,function*(){var D,W;if(!A||!k){p("Invalid WalletConnect setup");return}try{let{uri:V,approval:z}=yield(D=R.current)==null?void 0:D.connect({methods:be});if(!Boolean(V))return;l(V),n&&a(Me({loginToken:n}));try{yield(W=R.current)==null?void 0:W.login({approval:z,token:n})}catch(Re){console.warn("User rejected connection proposal",Re),yield P()}}catch(V){console.error("Unable to connect",V)}})}let U=Boolean(m),X=de();return[P,{error:m,loginFailed:U,isLoading:J&&!U,isLoggedIn:X&&!U},{uriDeepLink:De,walletConnectUri:_,cancelLogin:so,connectExisting:M,removeExistingPairing:G,wcPairings:u}]};q();E();N();function Yi(){let e=y(he),o=y(ri),n=y(ti),r=y(Ao),a=y(me),m=y(xt),p=y(ai),_=y(Te),[l,u]=(0,io.useState)(),f=B(),{callbackRoute:h,logoutRoute:A}=o||{callbackRoute:"",logoutRoute:""},[k]=Bt({callbackRoute:h,logoutRoute:A}),[L]=Ut({callbackRoute:h,logoutRoute:A});(0,io.useEffect)(()=>{F()},[e]),(0,io.useEffect)(()=>{so()},[n]),(0,io.useEffect)(()=>{ne()},[a,_]),(0,io.useEffect)(()=>{R()},[m,_,l]);function F(){return g(this,null,function*(){try{let v=yield Mi();v&&f(Xr(v.erd_chain_id))}catch(v){console.error("failed refreshing chainId ",v)}})}function R(){m==null&&p!=null&&l&&f(it({index:p.index,address:a,hasContractDataEnabled:l.dataEnabled,version:l.version}))}function ne(){return g(this,null,function*(){if(f(ct(!0)),a&&_)try{let v=yield $e(a);v&&f(Je(I(T({},v),{address:a,nonce:v.nonce.valueOf()})))}catch(v){f(oa("Failed getting account")),console.error("Failed getting account ",v)}f(ct(!1))})}function _e(){return g(this,null,function*(){let v=Ct(e.walletAddress);if(xe(v),r!=null){try{let P=yield We();if(P){f(Q({address:P,loginMethod:"wallet"}));let M=yield $e(P);M&&f(Je(I(T({},M),{nonce:Eo(M)})))}be()}catch(P){console.error("Failed authenticating wallet user ",P)}f(Jn(null))}})}function be(){var U;let v={};if((U=window==null?void 0:window.location)!=null&&U.search){let D=new URLSearchParams(window.location.search);v=Object.fromEntries(D)}let X=v,{signature:P,loginToken:M,address:G}=X,K=He(X,["signature","loginToken","address"]);P&&f(To(P)),J(K)}function J(v){let P=new URLSearchParams(v).toString(),{pathname:M}=window.location,G=P?`?${P}`:"",K=M?`${M}${G}`:"./";window.history.replaceState({},document.title,K)}function De(){return g(this,null,function*(){let v=new Xi.HWProvider,P=v.isInitialized();if(P||(P=yield v.init()),!P&&_){console.warn("Could not initialise ledger app"),pe();return}return(p==null?void 0:p.index)!=null&&(v.addressIndex=p.index),v})}function Ne(){return g(this,null,function*(){try{let v=yield De();if(!v)return;let P=yield Dt(v);xe(v),u(P)}catch(v){console.error("Could not initialise ledger app",v),pe()}})}function Ve(){return g(this,null,function*(){try{let v=yield We(),P=Ki.ExtensionProvider.getInstance().setAddress(v);(yield P.init())?xe(P):console.error("Could not initialise extension, make sure Elrond wallet extension is installed.")}catch(v){console.error("Unable to login to ExtensionProvider",v)}})}function so(){if(n!=null)switch(n){case"ledger":{Ne();break}case"walletconnect":{k(!1);break}case"walletconnectv2":{L(!1);break}case"extension":{Ve();break}case"extra":{Wi();break}case"wallet":case"":{_e();break}default:return}}return null}t();var Po=w(require("react"));t();var yc=require("react");t();t();var Wt=()=>y(Le);t();E();t();t();t();var gm=require("react"),fm=require("@elrondnetwork/erdjs-web-wallet-provider"),hm=w(require("qs"));C();E();N();t();var Gt=require("react"),bm=require("@elrondnetwork/erdjs-extension-provider");C();E();N();t();t();t();t();t();E();t();t();t();t();t();lo();Ze();Vn();t();t();t();function Ji(e){return new Promise(o=>{setTimeout(()=>o(),e)})}var Qi=(e,o,n,r=0)=>g(void 0,null,function*(){try{return yield e(...n)}catch(a){return r<o.retries?((o==null?void 0:o.delay)!=null&&(yield Ji(o.delay)),Qi(e,o,n,r+1)):null}}),ec=(e,o={retries:5,delay:500})=>(...n)=>g(void 0,null,function*(){return Qi(e,o,n)});t();t();var oc=w(require("axios"));function $t(e){return g(this,null,function*(){try{let{data:o}=yield oc.default.get(`${e}/websocket/config`);return`wss://${o.url}`}catch(o){throw console.error(o),new Error("Can not get websocket url")}})}t();t();var Tm=w(require("platform"));t();t();t();var Sm=w(require("axios")),wm=w(require("swr"));t();t();var tc=require("react");t();var nc=require("react");ot();E();ie();Ze();t();uo();E();t();t();t();t();t();function rc(){let e=y(vt),o=Object.entries(e),n=(o==null?void 0:o.length)>0;return{pendingTransactions:e,pendingTransactionsArray:o,hasPendingTransactions:n}}So();t();E();N();t();So();function ac(e,o){return e.reduce((r,{status:a,hash:m})=>(m!=null&&!o.includes(m)&&Ie(a)&&r.push({hash:m,previousStatus:a}),r),[])}t();E();N();function ic({results:e,hash:o,sessionId:n}){let r=e==null?void 0:e.find(a=>(a==null?void 0:a.returnMessage)!=="");b.dispatch(Bo({transactionHash:o,sessionId:n,status:"fail",errorMessage:r==null?void 0:r.returnMessage})),b.dispatch(oo({sessionId:n,status:"fail",errorMessage:r==null?void 0:r.returnMessage}))}t();E();ie();function Vt(e){b.dispatch(oo({sessionId:e,status:"timedOut"}))}var In={},Ht=[];function km({serverTransaction:e,sessionId:o,customTransactionInformation:n,shouldRefreshBalance:r}){let{hash:a,status:m,results:p,invalidTransaction:_,hasStatusChanged:l}=e;try{if(Ht.includes(a))return;if(In[a]>30){Vt(o);return}if(_||Ie(m)){In[a]=In[a]?In[a]+1:1;return}l&&(Qe(m)&&(n==null?void 0:n.completedTransactionsDelay)!=null?(Ht.push(a),setTimeout(()=>b.dispatch(Bo({sessionId:o,status:"success",transactionHash:a})),n==null?void 0:n.completedTransactionsDelay)):b.dispatch(Bo({sessionId:o,status:m,transactionHash:a}))),r||vn(),eo(m)&&ic({sessionId:o,hash:a,results:p})}catch(u){console.error(u),Vt(o)}}function cc(m){return g(this,arguments,function*({sessionId:e,transactionBatch:{transactions:o,customTransactionInformation:n},getTransactionsByHash:r=yn,shouldRefreshBalance:a}){try{if(o==null)return;let p=ac(o,Ht),_=yield r(p);for(let l of _)km({serverTransaction:l,sessionId:e,customTransactionInformation:n,shouldRefreshBalance:a})}catch(p){console.error(p)}})}function sc(){let{pendingTransactionsArray:e}=rc();function o(n){return g(this,null,function*(){let r=e.filter(([a,m])=>a!=null&&Ie(m.status));if(r.length>0)for(let[a,m]of r)yield cc(T({sessionId:a,transactionBatch:m},n));n.shouldRefreshBalance&&vn()})}return o}t();var Dm=require("react");t();t();t();E();t();t();var Lm=w(require("bignumber.js"));t();var Am=`.dapp-core-component__main__link-style {
  color: #1392ff;
}

.dapp-core-component__main__link-second-style {
  color: #9dabbd;
}
.dapp-core-component__main__link-second-style:hover {
  color: #1392ff;
}
.dapp-core-component__main__link-second-style:hover svg {
  color: #1392ff;
}

.dapp-core-component__main__wrapper {
  min-height: 100vh;
}

.dapp-core-component__main__opacity-6 {
  opacity: 0.6;
}

.dapp-core-component__main__qr-code-svg-container {
  width: 15rem;
  height: 15rem;
}

.dapp-core-component__main__pair-list {
  max-height: 20rem;
  overflow-y: auto;
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__pair-image {
  width: 4rem;
  overflow: hidden;
  height: auto;
  min-width: 2rem;
}

.dapp-core-component__main__pair-remove {
  padding: 0 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

.dapp-core-component__main__dapp-icon {
  padding: 5px;
  border-radius: 50%;
  background-color: #fafafa;
  width: 4.6428571429rem;
  height: 4.6428571429rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}
.dapp-core-component__main__dapp-icon.dapp-core-component__main__icon-medium {
  width: 80px;
  height: 80px;
}

.dapp-core-component__main__centering {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dapp-core-component__main__border-n {
  border: none !important;
}

.dapp-core-component__main__t-shadow {
  text-shadow: 0px 4px 7px rgba(0, 0, 0, 0.3);
}

.dapp-core-component__main__btn {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.04) !important;
}

.dapp-core-component__main__media {
  display: flex;
  align-items: flex-start;
}

.dapp-core-component__main__media-body {
  flex: 1;
}

.dapp-core-component__main__connect-btns .dapp-core-component__main__btn {
  display: flex;
  height: 48px;
  justify-content: center;
  align-items: center;
  color: #fff !important;
  background-color: #1392ff;
  padding: 15px 20px 15px 16px;
  border-radius: 6px;
  gap: 8px;
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__btn:hover, .dapp-core-component__main__connect-btns .dapp-core-component__main__btn:active {
  background-color: #1392ff;
  background-image: linear-gradient(rgba(0, 0, 0, 0.1) 0 0);
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__info {
  color: #1392ff;
  margin-left: 48px;
}

.dapp-core-component__main__btn {
  transition: none !important;
}

.dapp-core-component__main__trust-badge {
  width: 20px;
  height: 20px;
}

/* Navbar
  -------------------------------------------------- */
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name {
  display: flex;
  align-items: center;
  font-size: 1.7142857143rem;
  font-weight: 400;
  color: #9ba5b4;
  margin-left: 1.6428571429rem;
}
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name:before {
  content: "";
  display: block;
  width: 1px;
  height: 3rem;
  margin-right: 1.7142857143rem;
}

/* Dashboard
  -------------------------------------------------- */
.dapp-core-component__main__action-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  margin: 0 7px;
}
.dapp-core-component__main__action-btn button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #fff;
}
.dapp-core-component__main__action-btn button:hover {
  opacity: 0.9;
}

.dapp-core-component__main__transactions .dapp-core-component__main__transaction-icon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table td {
  white-space: nowrap;
  vertical-align: middle;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table tr:last-of-type td {
  border-bottom: 1px solid #dee2e6;
}

/* Loader, Transaction Success/Fail
  -------------------------------------------------- */
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div {
  position: absolute;
  top: 27px;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.1);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.dapp-core-component__main__page-state .dapp-core-component__main__bg-blue .dapp-core-component__main__lds-ellipsis div {
  background: rgba(255, 255, 255, 0.4);
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(1) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis1 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(2) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(3) {
  left: 26px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(4) {
  left: 45px;
  animation: dapp-core-component__main__lds-ellipsis3 0.6s infinite;
}
@keyframes dapp-core-component__main__lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(19px, 0);
  }
}

.dapp-core-component__main__btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
}
.dapp-core-component__main__btn .dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn .dapp-core-component__main__disabled {
  opacity: 0.65;
  cursor: default;
}
.dapp-core-component__main__btn:hover {
  color: #212529;
  text-decoration: none;
}
.dapp-core-component__main__btn:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn:disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn-primary {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary:hover {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-primary:focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary:hover {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-secondary:focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-warning {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning:hover {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning:focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-dark {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark:hover {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
}
.dapp-core-component__main__btn-dark:focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-link {
  font-weight: 400;
  color: #007bff;
  text-decoration: none;
}
.dapp-core-component__main__btn-link:hover {
  color: #0056b3;
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:focus, .dapp-core-component__main__btn-link.dapp-core-component__main__focus {
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:disabled, .dapp-core-component__main__btn-link.dapp-core-component__main__disabled {
  color: #6c757d;
  pointer-events: none;
}

.dapp-core-component__main__btn-light {
  color: #212529;
  background-color: #fafafa;
  border-color: #fafafa;
}
.dapp-core-component__main__btn-light .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}
.dapp-core-component__main__btn-light .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #f8f9fa;
  border-color: #f8f9fa;
}
.dapp-core-component__main__btn-light:hover {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
}
.dapp-core-component__main__btn-light:focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .dapp-core-component__main__badge {
    transition: none;
  }
}
a.dapp-core-component__main__badge:hover, a.dapp-core-component__main__badge:focus {
  text-decoration: none;
}

.dapp-core-component__main__badge:empty {
  display: none;
}

.dapp-core-component__main__btn .dapp-core-component__main__badge {
  position: relative;
  top: -1px;
}

.dapp-core-component__main__badge-pill {
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.dapp-core-component__main__badge-primary {
  color: #fff;
  background-color: #007bff;
}

a.dapp-core-component__main__badge-primary:hover, a.dapp-core-component__main__badge-primary:focus {
  color: #fff;
  background-color: #0062cc;
}

a.dapp-core-component__main__badge-primary:focus, a.dapp-core-component__main__badge-primary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.5);
}

.dapp-core-component__main__badge-secondary {
  color: #fff;
  background-color: #6c757d;
}

a.dapp-core-component__main__badge-secondary:hover, a.dapp-core-component__main__badge-secondary:focus {
  color: #fff;
  background-color: #545b62;
}

a.dapp-core-component__main__badge-secondary:focus, a.dapp-core-component__main__badge-secondary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108, 117, 125, 0.5);
}

.dapp-core-component__main__badge-success {
  color: #fff;
  background-color: #28a745;
}

a.dapp-core-component__main__badge-success:hover, a.dapp-core-component__main__badge-success:focus {
  color: #fff;
  background-color: #1e7e34;
}

a.dapp-core-component__main__badge-success:focus, a.dapp-core-component__main__badge-success.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
}

.dapp-core-component__main__badge-info {
  color: #fff;
  background-color: #17a2b8;
}

a.dapp-core-component__main__badge-info:hover, a.dapp-core-component__main__badge-info:focus {
  color: #fff;
  background-color: #117a8b;
}

a.dapp-core-component__main__badge-info:focus, a.dapp-core-component__main__badge-info.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(23, 162, 184, 0.5);
}

.dapp-core-component__main__badge-warning {
  color: #212529;
  background-color: #ffc107;
}

a.dapp-core-component__main__badge-warning:hover, a.dapp-core-component__main__badge-warning:focus {
  color: #212529;
  background-color: #d39e00;
}

a.dapp-core-component__main__badge-warning:focus, a.dapp-core-component__main__badge-warning.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 193, 7, 0.5);
}

.dapp-core-component__main__badge-danger {
  color: #fff;
  background-color: #dc3545;
}

a.dapp-core-component__main__badge-danger:hover, a.dapp-core-component__main__badge-danger:focus {
  color: #fff;
  background-color: #bd2130;
}

a.dapp-core-component__main__badge-danger:focus, a.dapp-core-component__main__badge-danger.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(220, 53, 69, 0.5);
}

.dapp-core-component__main__badge-light {
  color: #212529;
  background-color: #f8f9fa;
}

a.dapp-core-component__main__badge-light:hover, a.dapp-core-component__main__badge-light:focus {
  color: #212529;
  background-color: #dae0e5;
}

a.dapp-core-component__main__badge-light:focus, a.dapp-core-component__main__badge-light.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(248, 249, 250, 0.5);
}

.dapp-core-component__main__badge-dark {
  color: #fff;
  background-color: #343a40;
}

a.dapp-core-component__main__badge-dark:hover, a.dapp-core-component__main__badge-dark:focus {
  color: #fff;
  background-color: #1d2124;
}

a.dapp-core-component__main__badge-dark:focus, a.dapp-core-component__main__badge-dark.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}

.dapp-core-component__main__card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 5px;
}
.dapp-core-component__main__card > hr {
  margin-right: 0;
  margin-left: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group {
  border-top: inherit;
  border-bottom: inherit;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:first-child {
  border-top-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:last-child {
  border-bottom-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__card-header + .dapp-core-component__main__list-group,
.dapp-core-component__main__card > .dapp-core-component__main__list-group + .dapp-core-component__main__card-footer {
  border-top: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header {
  border: 0;
  padding: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header:first-child {
  border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item {
  border-bottom: 1px solid #dee2e6;
  padding: 1.1rem 1rem;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item h6,
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item .dapp-core-component__main__h6 {
  line-height: 2.5rem;
  letter-spacing: unset;
  margin: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item.dapp-core-component__main__compact {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fafafa;
}

.dapp-core-component__main__card-body {
  flex: 1 1 auto;
  min-height: 1px;
  padding: 1.25rem;
}

.dapp-core-component__main__card-title {
  margin-bottom: 0.75rem;
}

.dapp-core-component__main__card-subtitle {
  margin-top: -0.375rem;
  margin-bottom: 0;
}

.dapp-core-component__main__card-text:last-child {
  margin-bottom: 0;
}

.dapp-core-component__main__card-link:hover {
  text-decoration: none;
}
.dapp-core-component__main__card-link + .dapp-core-component__main__card-link {
  margin-left: 1.25rem;
}

.dapp-core-component__main__card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: #fff;
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__card-footer {
  padding: 0.75rem 1.25rem;
  background-color: #fff;
  border-top: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__text {
  color: #9dabbd;
}

.dapp-core-component__main__text-white {
  color: #fff;
}

.dapp-core-component__main__text-monospace {
  font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-size-xl .dapp-core-component__main__trim {
  max-width: 13rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim {
  max-width: 10rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-only-sm .dapp-core-component__main__trim {
  max-width: none;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-left {
  text-align: left;
}

.dapp-core-component__main__text-right {
  text-align: right;
}

.dapp-core-component__main__text-center {
  text-align: center;
}

.dapp-core-component__main__text-primary {
  color: #1b46c2;
}

.dapp-core-component__main__text-secondary {
  color: #6c757d;
}

.dapp-core-component__main__text-primary-highlight {
  color: #1b46c2;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-success {
  color: #28a745;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-body {
  color: #212529;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-black-50 {
  color: rgba(0, 0, 0, 0.5);
}

.dapp-core-component__main__text-white-50 {
  color: rgba(255, 255, 255, 0.5);
}

@media (min-width: 992px) {
  .dapp-core-component__main__text-lg-right {
    text-align: right;
  }
}

.dapp-core-component__main__text-hide {
  font: 0/0 a;
  color: rgba(0, 0, 0, 0);
  text-shadow: none;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
}

.dapp-core-component__main__text-decoration-none {
  text-decoration: none;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__text-reset {
  color: inherit;
}

.dapp-core-component__main__text-lowercase {
  text-transform: lowercase;
}

.dapp-core-component__main__text-uppercase {
  text-transform: uppercase;
}

.dapp-core-component__main__text-capitalize {
  text-transform: capitalize;
}

.dapp-core-component__main__text-danger {
  color: #d9534f;
}

.dapp-core-component__main__text-warning {
  color: #f0ad4e;
}

.dapp-core-component__main__trim-fs-sm {
  color: inherit;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__m-0 {
  margin: 0;
}

.dapp-core-component__main__mt-0 {
  margin-top: 0;
}

.dapp-core-component__main__my-0 {
  margin-top: 0;
  margin-bottom: 0;
}

.dapp-core-component__main__mr-0 {
  margin-right: 0;
}

.dapp-core-component__main__mx-0 {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__mb-0 {
  margin-bottom: 0;
}

.dapp-core-component__main__my-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.dapp-core-component__main__ml-0 {
  margin-left: 0;
}

.dapp-core-component__main__mx-0 {
  margin-left: 0;
  margin-right: 0;
}

.dapp-core-component__main__m-1 {
  margin: 0.25rem;
}

.dapp-core-component__main__mt-1 {
  margin-top: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__mr-1 {
  margin-right: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.dapp-core-component__main__mb-1 {
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-bottom: 0.25rem;
  margin-top: 0.25rem;
}

.dapp-core-component__main__ml-1 {
  margin-left: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-right: 0.25rem;
  margin-left: 0.25rem;
}

.dapp-core-component__main__m-2 {
  margin: 0.5rem;
}

.dapp-core-component__main__mt-2 {
  margin-top: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-bottom: 0.5rem;
  margin-top: 0.5rem;
}

.dapp-core-component__main__mr-2 {
  margin-right: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__mb-2 {
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__ml-2 {
  margin-left: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__m-3 {
  margin: 1rem;
}

.dapp-core-component__main__mt-3,
.dapp-core-component__main__my-3 {
  margin-top: 1rem;
}

.dapp-core-component__main__mr-3 {
  margin-right: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__mb-3 {
  margin-bottom: 1rem;
}

.dapp-core-component__main__my-3 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.dapp-core-component__main__ml-3 {
  margin-left: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__m-4 {
  margin: 1.5rem;
}

.dapp-core-component__main__mt-4 {
  margin-top: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__mr-4 {
  margin-right: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-right: 1.5rem;
  margin-left: 1.5rem;
}

.dapp-core-component__main__mb-4 {
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.dapp-core-component__main__ml-4 {
  margin-left: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}

.dapp-core-component__main__m-5 {
  margin: 3rem;
}

.dapp-core-component__main__mt-5 {
  margin-top: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__mr-5 {
  margin-right: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-right: 3rem;
  margin-left: 3rem;
}

.dapp-core-component__main__mb-5 {
  margin-bottom: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__ml-5 {
  margin-left: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-left: 3rem;
  margin-right: 3rem;
}

.dapp-core-component__main__p-0 {
  padding: 0;
}

.dapp-core-component__main__pt-0 {
  padding-top: 0;
}

.dapp-core-component__main__py-0 {
  padding-top: 0;
  padding-bottom: 0;
}

.dapp-core-component__main__pr-0 {
  padding-right: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__pb-0 {
  padding-bottom: 0;
}

.dapp-core-component__main__py-0 {
  padding-bottom: 0;
  padding-top: 0;
}

.dapp-core-component__main__pl-0 {
  padding-left: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__p-1 {
  padding: 0.25rem;
}

.dapp-core-component__main__pt-1 {
  padding-top: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pr-1 {
  padding-right: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-right: 0.25rem;
  padding-left: 0.25rem;
}

.dapp-core-component__main__pb-1 {
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pl-1 {
  padding-left: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

.dapp-core-component__main__p-2 {
  padding: 0.5rem;
}

.dapp-core-component__main__pt-2 {
  padding-top: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.dapp-core-component__main__pr-2 {
  padding-right: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__pb-2 {
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__pl-2 {
  padding-left: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__p-3 {
  padding: 1rem;
}

.dapp-core-component__main__pt-3 {
  padding-top: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.dapp-core-component__main__pr-3 {
  padding-right: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__pb-3 {
  padding-bottom: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.dapp-core-component__main__pl-3 {
  padding-left: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__p-4 {
  padding: 1.5rem;
}

.dapp-core-component__main__pt-4 {
  padding-top: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pr-4 {
  padding-right: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.dapp-core-component__main__pb-4 {
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pl-4 {
  padding-left: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
}

.dapp-core-component__main__p-5 {
  padding: 3rem;
}

.dapp-core-component__main__pt-5 {
  padding-top: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-bottom: 3rem;
  padding-top: 3rem;
}

.dapp-core-component__main__pr-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__pb-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.dapp-core-component__main__pl-5 {
  padding-left: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__m-n1 {
  margin: -0.25rem;
}

.dapp-core-component__main__mt-n1 {
  margin-top: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__mr-n1 {
  margin-right: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__mb-n1 {
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-bottom: -0.25rem;
  margin-top: -0.25rem;
}

.dapp-core-component__main__ml-n1 {
  margin-left: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__m-n2 {
  margin: -0.5rem;
}

.dapp-core-component__main__mt-n2 {
  margin-top: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-top: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mr-n2 {
  margin-right: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mb-n2 {
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-bottom: -0.5rem;
  margin-top: -0.5rem;
}

.dapp-core-component__main__ml-n2 {
  margin-left: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.dapp-core-component__main__m-n3 {
  margin: -1rem;
}

.dapp-core-component__main__mt-n3 {
  margin-top: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-top: -1rem;
  margin-bottom: -1rem;
}

.dapp-core-component__main__mr-n3 {
  margin-right: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-right: -1rem;
  margin-left: -1rem;
}

.dapp-core-component__main__mb-n3 {
  margin-bottom: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-bottom: -1rem;
  margin-top: -1rem;
}

.dapp-core-component__main__ml-n3 {
  margin-left: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-left: -1rem;
  margin-right: -1rem;
}

.dapp-core-component__main__m-n4 {
  margin: -1.5rem;
}

.dapp-core-component__main__mt-n4 {
  margin-top: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-top: -1.5rem;
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__mr-n4 {
  margin-right: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__mb-n4 {
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-bottom: -1.5rem;
  margin-top: -1.5rem;
}

.dapp-core-component__main__ml-n4 {
  margin-left: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__m-n5 {
  margin: -3rem;
}

.dapp-core-component__main__mt-n5 {
  margin-top: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-bottom: -3rem;
  margin-top: -3rem;
}

.dapp-core-component__main__mr-n5 {
  margin-right: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-left: -3rem;
  margin-right: -3rem;
}

.dapp-core-component__main__mb-n5 {
  margin-bottom: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-top: -3rem;
  margin-bottom: -3rem;
}

.dapp-core-component__main__ml-n5 {
  margin-left: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-right: -3rem;
  margin-left: -3rem;
}

.dapp-core-component__main__m-auto {
  margin: auto;
}

.dapp-core-component__main__mt-auto {
  margin-top: auto;
}

.dapp-core-component__main__my-auto {
  margin-bottom: auto;
  margin-top: auto;
}

.dapp-core-component__main__mr-auto {
  margin-right: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__mb-auto {
  margin-bottom: auto;
}

.dapp-core-component__main__my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.dapp-core-component__main__ml-auto {
  margin-left: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-1 {
    margin-right: 0.25rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-4 {
    margin-right: 1.5rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mx-lg-4 {
    margin-left: 1.5rem;
    margin-right: 1.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__mb-xl-2 {
    margin-bottom: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__pr-sm-2 {
    padding-right: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__px-sm-2 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__pr-xl-0,
.dapp-core-component__main__px-xl-0 {
    padding-right: 0;
  }
}

.dapp-core-component__main__col {
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

.dapp-core-component__main__col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__col-xl,
.dapp-core-component__main__col-xl-auto,
.dapp-core-component__main__col-xl-12,
.dapp-core-component__main__col-xl-11,
.dapp-core-component__main__col-xl-10,
.dapp-core-component__main__col-xl-9,
.dapp-core-component__main__col-xl-8,
.dapp-core-component__main__col-xl-7,
.dapp-core-component__main__col-xl-6,
.dapp-core-component__main__col-xl-5,
.dapp-core-component__main__col-xl-4,
.dapp-core-component__main__col-xl-3,
.dapp-core-component__main__col-xl-2,
.dapp-core-component__main__col-xl-1,
.dapp-core-component__main__col-lg,
.dapp-core-component__main__col-lg-auto,
.dapp-core-component__main__col-lg-12,
.dapp-core-component__main__col-lg-11,
.dapp-core-component__main__col-lg-10,
.dapp-core-component__main__col-lg-9,
.dapp-core-component__main__col-lg-8,
.dapp-core-component__main__col-lg-7,
.dapp-core-component__main__col-lg-6,
.dapp-core-component__main__col-lg-5,
.dapp-core-component__main__col-lg-4,
.dapp-core-component__main__col-lg-3,
.dapp-core-component__main__col-lg-2,
.dapp-core-component__main__col-lg-1,
.dapp-core-component__main__col-md,
.dapp-core-component__main__col-md-auto,
.dapp-core-component__main__col-md-12,
.dapp-core-component__main__col-md-11,
.dapp-core-component__main__col-md-10,
.dapp-core-component__main__col-md-9,
.dapp-core-component__main__col-md-8,
.dapp-core-component__main__col-md-7,
.dapp-core-component__main__col-md-6,
.dapp-core-component__main__col-md-5,
.dapp-core-component__main__col-md-4,
.dapp-core-component__main__col-md-3,
.dapp-core-component__main__col-md-2,
.dapp-core-component__main__col-md-1,
.dapp-core-component__main__col-sm,
.dapp-core-component__main__col-sm-auto,
.dapp-core-component__main__col-sm-12,
.dapp-core-component__main__col-sm-11,
.dapp-core-component__main__col-sm-10,
.dapp-core-component__main__col-sm-9,
.dapp-core-component__main__col-sm-8,
.dapp-core-component__main__col-sm-7,
.dapp-core-component__main__col-sm-6,
.dapp-core-component__main__col-sm-5,
.dapp-core-component__main__col-sm-4,
.dapp-core-component__main__col-sm-3,
.dapp-core-component__main__col-sm-2,
.dapp-core-component__main__col-sm-1,
.dapp-core-component__main__col,
.dapp-core-component__main__col-auto,
.dapp-core-component__main__col-12,
.dapp-core-component__main__col-11,
.dapp-core-component__main__col-10,
.dapp-core-component__main__col-9,
.dapp-core-component__main__col-8,
.dapp-core-component__main__col-7,
.dapp-core-component__main__col-6,
.dapp-core-component__main__col-5,
.dapp-core-component__main__col-4,
.dapp-core-component__main__col-3,
.dapp-core-component__main__col-2,
.dapp-core-component__main__col-1 {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
}

.dapp-core-component__main__col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

@media (min-width: 576px) {
  .dapp-core-component__main__col-sm-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-sm-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-sm-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-sm-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-sm-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-sm-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-sm-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-sm-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-sm-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-sm-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-sm-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-sm-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 768px) {
  .dapp-core-component__main__col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__col-xl-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-xl-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-xl-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-xl-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-xl-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-xl-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-xl-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-xl-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-xl-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-xl-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-xl-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-xl-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
.dapp-core-component__main__flex-row {
  flex-direction: row;
}

.dapp-core-component__main__flex-column {
  flex-direction: column;
}

.dapp-core-component__main__flex-row-reverse {
  flex-direction: row-reverse;
}

.dapp-core-component__main__flex-column-reverse {
  flex-direction: column-reverse;
}

.dapp-core-component__main__flex-wrap {
  flex-wrap: wrap;
}

.dapp-core-component__main__flex-nowrap {
  flex-wrap: nowrap;
}

.dapp-core-component__main__flex-wrap-reverse {
  flex-wrap: wrap-reverse;
}

.dapp-core-component__main__flex-fill {
  flex: 1 1 auto;
}

.dapp-core-component__main__flex-grow-0 {
  flex-grow: 0;
}

.dapp-core-component__main__flex-grow-1 {
  flex-grow: 1;
}

.dapp-core-component__main__flex-shrink-0 {
  flex-shrink: 0;
}

.dapp-core-component__main__flex-shrink-1 {
  flex-shrink: 1;
}

.dapp-core-component__main__justify-content-start {
  justify-content: flex-start;
}

.dapp-core-component__main__justify-content-end {
  justify-content: flex-end;
}

.dapp-core-component__main__justify-content-center {
  justify-content: center;
}

.dapp-core-component__main__justify-content-between {
  justify-content: space-between;
}

.dapp-core-component__main__justify-content-around {
  justify-content: space-around;
}

.dapp-core-component__main__align-items-start {
  align-items: flex-start;
}

.dapp-core-component__main__align-items-end {
  align-items: flex-end;
}

.dapp-core-component__main__align-items-center {
  align-items: center;
}

.dapp-core-component__main__align-items-baseline {
  align-items: baseline;
}

.dapp-core-component__main__align-items-stretch {
  align-items: stretch;
}

.dapp-core-component__main__align-content-start {
  align-content: flex-start;
}

.dapp-core-component__main__align-content-end {
  align-content: flex-end;
}

.dapp-core-component__main__align-content-center {
  align-content: center;
}

.dapp-core-component__main__align-content-between {
  align-content: space-between;
}

.dapp-core-component__main__align-content-around {
  align-content: space-around;
}

.dapp-core-component__main__align-content-stretch {
  align-content: stretch;
}

.dapp-core-component__main__align-self-auto {
  align-self: auto;
}

.dapp-core-component__main__align-self-start {
  align-self: flex-start;
}

.dapp-core-component__main__align-self-end {
  align-self: flex-end;
}

.dapp-core-component__main__align-self-center {
  align-self: center;
}

.dapp-core-component__main__align-self-baseline {
  align-self: baseline;
}

.dapp-core-component__main__align-self-stretch {
  align-self: stretch;
}

@media (min-width: 576px) {
  .dapp-core-component__main__flex-sm-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-wrap {
    flex-wrap: wrap;
  }
}

.dapp-core-component__main__form-check {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: auto;
  padding-left: 0;
}

.dapp-core-component__main__form-check-label {
  margin-bottom: 0;
}

.dapp-core-component__main__form-check-input {
  position: absolute;
  margin-top: 0.3rem;
  margin-left: -1.25rem;
}
.dapp-core-component__main__form-check-input[disabled] ~ .dapp-core-component__main__form-check-label, .dapp-core-component__main__form-check-input:disabled ~ .dapp-core-component__main__form-check-label {
  color: #6c757d;
}

.dapp-core-component__main__form-group {
  display: flex;
  flex: 0 0 auto;
  flex-flow: row wrap;
  align-items: center;
  margin-bottom: 0;
}

.dapp-core-component__main__form-control {
  display: block;
  width: 100%;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.dapp-core-component__main__form-label {
  padding-top: calc(0.375rem + 1px);
  padding-bottom: calc(0.375rem + 1px);
  margin-bottom: 0;
  font-size: inherit;
  line-height: 1.5;
}

input,
button,
select,
optgroup,
textarea {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.dapp-core-component__main__form-group {
  margin-bottom: 1rem;
}

.dapp-core-component__main__form-text {
  display: block;
  margin-top: 0.25rem;
  color: inherit;
}

.dapp-core-component__main__border {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__border-top {
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__border-right {
  border-right: 1px solid #dee2e6;
}

.dapp-core-component__main__border-bottom {
  border-bottom: 1px solid #dee2e6;
}

.dapp-core-component__main__border-left {
  border-left: 1px solid #dee2e6;
}

.dapp-core-component__main__border-0 {
  border: 0;
}

.dapp-core-component__main__border-top-0 {
  border-top: 0;
}

.dapp-core-component__main__border-right-0 {
  border-right: 0;
}

.dapp-core-component__main__border-bottom-0 {
  border-bottom: 0;
}

.dapp-core-component__main__border-left-0 {
  border-left: 0;
}

.dapp-core-component__main__rounded-sm {
  border-radius: 0.2rem;
}

.dapp-core-component__main__rounded {
  border-radius: 0.25rem;
}

.dapp-core-component__main__rounded-top {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-right {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-bottom {
  border-bottom-right-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-left {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-lg {
  border-radius: 0.3rem;
}

.dapp-core-component__main__rounded-circle {
  border-radius: 50%;
}

.dapp-core-component__main__rounded-pill {
  border-radius: 50rem;
}

.dapp-core-component__main__rounded-0 {
  border-radius: 0;
}

.dapp-core-component__main__h1,
.dapp-core-component__main__h2,
.dapp-core-component__main__h3,
.dapp-core-component__main__h4,
.dapp-core-component__main__h5,
.dapp-core-component__main__h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2;
}

.dapp-core-component__main__h1 {
  font-size: 2.5rem;
}

.dapp-core-component__main__h2 {
  font-size: 2rem;
}

.dapp-core-component__main__h3 {
  font-size: 1.75rem;
}

.dapp-core-component__main__h4 {
  font-size: 1.5rem;
}

.dapp-core-component__main__h5 {
  font-size: 1.25rem;
}

.dapp-core-component__main__h6 {
  font-size: 1rem;
}

.dapp-core-component__main__lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

.dapp-core-component__main__bg-warning,
.dapp-core-component__main__warning {
  background-color: #f0ad4e;
}

.dapp-core-component__main__bg-danger,
.dapp-core-component__main__danger {
  background-color: #d9534f;
}

.dapp-core-component__main__bg-success,
.dapp-core-component__main__success {
  background-color: #5cb85c;
}

.dapp-core-component__main__position-relative {
  position: relative;
}

.dapp-core-component__main__position-absolute {
  position: absolute;
}

.dapp-core-component__main__w-25 {
  width: 25%;
}

.dapp-core-component__main__w-50 {
  width: 50%;
}

.dapp-core-component__main__w-75 {
  width: 75%;
}

.dapp-core-component__main__w-100 {
  width: 100%;
}

.dapp-core-component__main__w-auto {
  width: auto;
}

.dapp-core-component__main__h-25 {
  height: 25%;
}

.dapp-core-component__main__h-50 {
  height: 50%;
}

.dapp-core-component__main__h-75 {
  height: 75%;
}

.dapp-core-component__main__h-100 {
  height: 100%;
}

.dapp-core-component__main__h-auto {
  height: auto;
}

.dapp-core-component__main__mw-100 {
  max-width: 100%;
}

.dapp-core-component__main__mh-100 {
  max-height: 100%;
}

.dapp-core-component__main__min-vw-100 {
  min-width: 100vw;
}

.dapp-core-component__main__min-vh-100 {
  min-height: 100vh;
}

.dapp-core-component__main__vw-100 {
  width: 100vw;
}

.dapp-core-component__main__vh-100 {
  height: 100vh;
}

.dapp-core-component__main__locked-icon {
  color: inherit;
}

.dapp-core-component__main__shadow-sm {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
}

.dapp-core-component__main__shadow {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.dapp-core-component__main__shadow-lg {
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.175);
}

.dapp-core-component__main__shadow-none {
  box-shadow: none;
}

.dapp-core-component__main__d-none {
  display: none;
}

.dapp-core-component__main__d-inline {
  display: inline;
}

.dapp-core-component__main__d-inline-block {
  display: inline-block;
}

.dapp-core-component__main__d-block {
  display: block;
}

.dapp-core-component__main__d-table {
  display: table;
}

.dapp-core-component__main__d-table-row {
  display: table-row;
}

.dapp-core-component__main__d-table-cell {
  display: table-cell;
}

.dapp-core-component__main__d-flex {
  display: -ms-flexbox;
  display: flex;
}

.dapp-core-component__main__d-inline-flex {
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media (min-width: 576px) {
  .dapp-core-component__main__d-sm-flex {
    display: flex;
  }
}

.dapp-core-component__main__table {
  width: 100%;
  margin-bottom: 1rem;
  color: #212529;
}
.dapp-core-component__main__table th,
.dapp-core-component__main__table td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #dee2e6;
}
.dapp-core-component__main__table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #dee2e6;
}
.dapp-core-component__main__table tbody + tbody {
  border-top: 2px solid #dee2e6;
}

.dapp-core-component__main__table-sm th,
.dapp-core-component__main__table-sm td {
  padding: 0.3rem;
}

.dapp-core-component__main__table-bordered {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered th,
.dapp-core-component__main__table-bordered td {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered thead th,
.dapp-core-component__main__table-bordered thead td {
  border-bottom-width: 2px;
}

.dapp-core-component__main__table-borderless th,
.dapp-core-component__main__table-borderless td,
.dapp-core-component__main__table-borderless thead th,
.dapp-core-component__main__table-borderless tbody + tbody {
  border: 0;
}

.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(0, 0, 0, 0.05);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-dark th {
  color: #fff;
  background-color: #343a40;
  border-color: #454d55;
}
.dapp-core-component__main__table .dapp-core-component__main__thead-light th {
  color: #495057;
  background-color: #e9ecef;
  border-color: #dee2e6;
}

.dapp-core-component__main__table-dark {
  color: #fff;
  background-color: #343a40;
}
.dapp-core-component__main__table-dark th,
.dapp-core-component__main__table-dark td,
.dapp-core-component__main__table-dark thead th {
  border-color: #454d55;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-bordered {
  border: 0;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(255, 255, 255, 0.05);
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-hover tbody tr {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.075);
}

.dapp-core-component__main__table-responsive {
  display: block;
  width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}
.dapp-core-component__main__table-responsive > .dapp-core-component__main__table-bordered {
  border: 0;
}

@media (max-width: 767.98px) {
  .dapp-core-component__main__transactions.dapp-core-component__main__table .dapp-core-component__main__transaction-hash .dapp-core-component__main__trim {
    max-width: 8rem;
  }
}
.dapp-core-component__main__font-weight-light {
  font-weight: 300 !important;
}

.dapp-core-component__main__font-weight-lighter {
  font-weight: lighter !important;
}

.dapp-core-component__main__font-weight-normal {
  font-weight: 400 !important;
}

.dapp-core-component__main__font-weight-medium {
  font-weight: 500 !important;
}

.dapp-core-component__main__font-weight-bold {
  font-weight: 700 !important;
}

.dapp-core-component__main__font-weight-bolder {
  font-weight: bolder !important;
}

.dapp-core-component__main__font-italic {
  font-style: italic !important;
}

.dapp-core-component__main__icon-state {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
}
.dapp-core-component__main__icon-state.dapp-core-component__main__half {
  width: 3rem;
  height: 3rem;
}

.dapp-core-component__main__side-icon {
  width: 1.5rem;
  height: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  color: transparent;
  border-radius: 50%;
  overflow: hidden;
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__side-icon svg {
  color: #cacbcc;
}

div.dapp-core-component__main__side-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
div.dapp-core-component__main__side-icon img {
  transform: scale(0.75);
}

.dapp-core-component__main__toast-messages {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast {
  display: block;
  flex-basis: 4rem;
  width: 100%;
  max-width: 28.875rem;
  transition: transform 0.3s ease-in-out;
  transform: translateX(120%);
  background-color: #fff;
  border-radius: 0.25rem;
  border-color: #dee2e6;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__clickable {
  cursor: pointer;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible {
  margin-bottom: 0.3rem;
  transform: translateX(0);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress {
  background-color: rgba(0, 0, 0, 0);
  height: inherit;
  font-size: inherit;
  line-height: inherit;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar {
  background-color: rgba(108, 117, 125, 0.2);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar .dapp-core-component__main__content-height {
  z-index: -1;
  visibility: hidden;
  width: 0 !important;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast .dapp-core-component__main__tx-description {
  margin-bottom: 0.2rem;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close {
  opacity: 1;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close svg {
  color: #6c757d;
}

.dapp-core-component__main__data-field .dapp-core-component__main__textarea {
  -moz-appearance: textfield-multiline;
  -webkit-appearance: textarea;
  border: 1px solid #000;
  background-color: #fff;
  border-radius: 5px;
  font: medium -moz-fixed;
  font-size: 0.8125rem;
  min-height: 5rem;
  overflow: auto;
  padding: 0.375rem 0.75rem;
  resize: both;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea:hover {
  border: 1px solid #000;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea .dapp-core-component__main__highlighted {
  font-size: 1rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea.dapp-core-component__main__sc-call {
  min-height: 3rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__form-control:disabled,
.dapp-core-component__main__data-field .dapp-core-component__main__form-control[readonly] {
  background-color: #fafafa;
}
.dapp-core-component__main__data-field .dapp-core-component__main__small-font {
  font-size: 0.8125rem;
}

.dapp-core-component__main__progress-steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps hr {
  background-color: #6c757d;
  border-color: #dee2e6;
  z-index: 2;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot {
  height: 0.625rem;
  width: 0.625rem;
  border-radius: 50%;
  margin: 0 1rem;
  z-index: 5;
  background-color: #6c757d;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:first-of-type {
  margin-left: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:last-of-type {
  margin-right: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

a.dapp-core-component__main__side-link::after {
  right: 0;
  left: unset;
  width: calc(100% - 1.85rem);
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Am));C();E();ie();t();var Em=w(require("bignumber.js"));C();t();var Pm=require("@elrondnetwork/erdjs"),Cm=w(require("bignumber.js"));C();N();So();t();t();t();t();t();var pc=()=>y(Oe);t();t();t();t();var mc=require("react");t();t();var Nm=require("react"),Rm=require("@elrondnetwork/erdjs-extension-provider");uo();q();E();N();yo();t();var Mm=require("react");uo();E();t();var _c=require("react"),Fm=require("@elrondnetwork/erdjs-hw-provider");uo();q();E();N();yo();t();t();var lc=require("react");t();var fc=require("react");t();var co=require("react"),uc=require("socket.io-client");E();t();var Ce={current:null,status:"not_initialized"};var Om=3e3,Bm=3,Um=500,Wm=1e3;function gc(){let e=(0,co.useRef)(null),{address:o}=pc(),n=B(),{network:r}=Wt(),a=p=>{e.current&&clearTimeout(e.current),e.current=setTimeout(()=>{n(na(p))},Wm)},m=(0,co.useCallback)(ec(()=>g(this,null,function*(){Ce.status="pending";let p=yield $t(r.apiAddress);if(p==null){console.warn("Can not get websocket url");return}Ce.current=(0,uc.io)(p,{forceNew:!0,reconnectionAttempts:Bm,timeout:Om,query:{address:o}}),Ce.status="completed",Ce.current.onAny(_=>{a(_)})}),{retries:2,delay:Um}),[o]);(0,co.useEffect)(()=>{o&&Ce.status==="not_initialized"&&!Ce.current&&m()},[o,Ce.current]),(0,co.useEffect)(()=>()=>{var p;(p=Ce.current)==null||p.close(),e.current&&clearTimeout(e.current)},[])}function hc(e){gc();let o=y(yt);(0,fc.useEffect)(()=>{let n=o==null?void 0:o.message;n&&e(n)},[e,o])}t();t();t();var Tc=e=>e.dappModal,$m=x(Tc,e=>e.visible),En=x(Tc,e=>e.modal),EE=x(En,e=>e.showHeader),LE=x(En,e=>e.showFooter),PE=x(En,e=>e.headerText),CE=x(En,e=>e.footerText);Xe();var{logoutEvent:Gm}=ae,xc="storage",bc=()=>{let{address:e}=Vo();return(0,yc.useEffect)(()=>{let o=n=>{if(!(n.key!==Gm||!n.newValue))try{let{data:r}=JSON.parse(n.newValue);r===e&&pe()}catch(r){return}};return window.addEventListener(xc,o),()=>{window.removeEventListener(xc,o)}},[e]),null};t();t();var Ln=require("react"),wc=require("@elrondnetwork/erdjs"),vc=require("@elrondnetwork/erdjs/out/signature");E();N();Ze();var Sc=e=>{let o=e.redirectRoute;o&&Ae(o)},kc=({sendSignedTransactionsAsync:e=Ei})=>{let o=y(Oe),n=y(Ue),r=(0,Ln.useRef)(!1),a=B(),m=()=>{a(Uo()),r.current=!1};function p(){return g(this,null,function*(){var l;let _=Object.keys(n);for(let u of _){let f=n==null?void 0:n[u],h=(l=f==null?void 0:f.customTransactionInformation)==null?void 0:l.signWithoutSending;if(!u||h){Sc(f);continue}try{if(!(n[u].status==="signed"&&!r.current))continue;let{transactions:L}=n[u];if(!L)continue;r.current=!0;let F=L.map(J=>{let De=new wc.Address(J.sender),Ne=Io(J),Ve=new vc.Signature(J.signature);return Ne.applySignature(Ve,De),Ne}),R=yield e(F),ne="pending",_e=L.map(J=>R.includes(J.hash)?I(T({},J),{status:ne}):J);a(ua({sessionId:u,submittedMessage:"submitted"})),a(oo({sessionId:u,status:"sent",transactions:_e})),m(),Gi(o.nonce+L.length),Sc(f),history.pushState({},document.title,"?")}catch(A){console.error("Unable to send transactions",A),a(oo({sessionId:u,status:"fail",errorMessage:A.message})),m()}finally{r.current=!1}}})}return(0,Ln.useEffect)(()=>{p()},[n,o]),null};t();t();t();var Ac=require("react");function Ic(e){var a;let o=sc(),n=(a=e==null?void 0:e.getTransactionsByHash)!=null?a:yn,r=()=>{o({shouldRefreshBalance:!0,getTransactionsByHash:n})};hc(r),(0,Ac.useEffect)(()=>{let m=setInterval(r,3e4);return()=>{clearInterval(m)}},[r])}function Ec({getTransactionsByHash:e}){return Ic({getTransactionsByHash:e}),null}function Lc({customComponents:e}){var m,p;let o=e==null?void 0:e.transactionSender,n=e==null?void 0:e.transactionTracker,r=(m=o==null?void 0:o.component)!=null?m:kc,a=(p=n==null?void 0:n.component)!=null?p:Ec;return Po.default.createElement(Po.default.Fragment,null,Po.default.createElement(r,T({},o==null?void 0:o.props)),Po.default.createElement(a,T({},n==null?void 0:n.props)),Po.default.createElement(bc,null))}var Vm=({children:e,customNetworkConfig:o={},externalProvider:n,environment:r,customComponents:a})=>{if(!r)throw new Error("missing environment flag");return n!=null&&Ui(n),Co.default.createElement(Pc.Provider,{context:Wo,store:b},Co.default.createElement(Cc.PersistGate,{persistor:za,loading:null},Co.default.createElement(Ot,{environment:r,customNetworkConfig:o},Co.default.createElement(Yi,null),Co.default.createElement(Lc,{customComponents:a}),e)))};0&&(module.exports={AppInitializer,AuthenticatedRoutesWrapper,DappProvider});
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
