// @elrondnetwork/erdjs-web-wallet-provider/out/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WALLET_PROVIDER_CALLBACK_PARAM_TX_SIGNED = exports.WALLET_PROVIDER_CALLBACK_PARAM = exports.WALLET_PROVIDER_SIGN_TRANSACTION_URL = exports.WALLET_PROVIDER_SEND_TRANSACTION_URL = exports.WALLET_PROVIDER_DISCONNECT_URL = exports.WALLET_PROVIDER_CONNECT_URL = exports.WALLET_PROVIDER_TESTNET = exports.WALLET_PROVIDER_DEVNET = exports.WALLET_PROVIDER_MAINNET = void 0;
exports.WALLET_PROVIDER_MAINNET = "https://wallet.elrond.com/dapp/init";
exports.WALLET_PROVIDER_DEVNET = "https://devnet-wallet.elrond.com/dapp/init";
exports.WALLET_PROVIDER_TESTNET = "https://testnet-wallet.elrond.com/dapp/init";
exports.WALLET_PROVIDER_CONNECT_URL = "hook/login";
exports.WALLET_PROVIDER_DISCONNECT_URL = "hook/logout";
exports.WALLET_PROVIDER_SEND_TRANSACTION_URL = "hook/transaction";
exports.WALLET_PROVIDER_SIGN_TRANSACTION_URL = "hook/sign";
exports.WALLET_PROVIDER_CALLBACK_PARAM = "walletProviderStatus";
exports.WALLET_PROVIDER_CALLBACK_PARAM_TX_SIGNED = "transactionsSigned";
