// @elrondnetwork/dapp-core/__commonjs/UI/TransactionInfo/components/TransactionAction/TransactionAction.js
"use strict";var Va=Object.create;var Ue=Object.defineProperty,$a=Object.defineProperties,za=Object.getOwnPropertyDescriptor,Wa=Object.getOwnPropertyDescriptors,Ha=Object.getOwnPropertyNames,Qe=Object.getOwnPropertySymbols,ja=Object.getPrototypeOf,Co=Object.prototype.hasOwnProperty,On=Object.prototype.propertyIsEnumerable;var Bn=(e,o,n)=>o in e?Ue(e,o,{enumerable:!0,configurable:!0,writable:!0,value:n}):e[o]=n,b=(e,o)=>{for(var n in o||(o={}))Co.call(o,n)&&Bn(e,n,o[n]);if(Qe)for(var n of Qe(o))On.call(o,n)&&Bn(e,n,o[n]);return e},S=(e,o)=>$a(e,Wa(o));var Te=(e,o)=>{var n={};for(var t in e)Co.call(e,t)&&o.indexOf(t)<0&&(n[t]=e[t]);if(e!=null&&Qe)for(var t of Qe(e))o.indexOf(t)<0&&On.call(e,t)&&(n[t]=e[t]);return n};var y=(e,o)=>()=>(e&&(o=e(e=0)),o);var eo=(e,o)=>()=>(o||e((o={exports:{}}).exports,o),o.exports),ce=(e,o)=>{for(var n in o)Ue(e,n,{get:o[n],enumerable:!0})},Un=(e,o,n,t)=>{if(o&&typeof o=="object"||typeof o=="function")for(let a of Ha(o))!Co.call(e,a)&&a!==n&&Ue(e,a,{get:()=>o[a],enumerable:!(t=za(o,a))||t.enumerable});return e};var T=(e,o,n)=>(n=e!=null?Va(ja(e)):{},Un(o||!e||!e.__esModule?Ue(n,"default",{value:e,enumerable:!0}):n,e)),Ve=e=>Un(Ue({},"__esModule",{value:!0}),e);var Vn=(e,o,n)=>new Promise((t,a)=>{var i=d=>{try{l(n.next(d))}catch(u){a(u)}},_=d=>{try{l(n.throw(d))}catch(u){a(u)}},l=d=>d.done?t(d.value):Promise.resolve(d.value).then(i,_);l((n=n.apply(e,o)).next())});var Wn=eo(oo=>{"use strict";r();oo.byteLength=qa;oo.toByteArray=Xa;oo.fromByteArray=Ja;var W=[],O=[],Ga=typeof Uint8Array!="undefined"?Uint8Array:Array,Fo="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";for(me=0,$n=Fo.length;me<$n;++me)W[me]=Fo[me],O[Fo.charCodeAt(me)]=me;var me,$n;O["-".charCodeAt(0)]=62;O["_".charCodeAt(0)]=63;function zn(e){var o=e.length;if(o%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var n=e.indexOf("=");n===-1&&(n=o);var t=n===o?0:4-n%4;return[n,t]}function qa(e){var o=zn(e),n=o[0],t=o[1];return(n+t)*3/4-t}function Ka(e,o,n){return(o+n)*3/4-n}function Xa(e){var o,n=zn(e),t=n[0],a=n[1],i=new Ga(Ka(e,t,a)),_=0,l=a>0?t-4:t,d;for(d=0;d<l;d+=4)o=O[e.charCodeAt(d)]<<18|O[e.charCodeAt(d+1)]<<12|O[e.charCodeAt(d+2)]<<6|O[e.charCodeAt(d+3)],i[_++]=o>>16&255,i[_++]=o>>8&255,i[_++]=o&255;return a===2&&(o=O[e.charCodeAt(d)]<<2|O[e.charCodeAt(d+1)]>>4,i[_++]=o&255),a===1&&(o=O[e.charCodeAt(d)]<<10|O[e.charCodeAt(d+1)]<<4|O[e.charCodeAt(d+2)]>>2,i[_++]=o>>8&255,i[_++]=o&255),i}function Za(e){return W[e>>18&63]+W[e>>12&63]+W[e>>6&63]+W[e&63]}function Ya(e,o,n){for(var t,a=[],i=o;i<n;i+=3)t=(e[i]<<16&16711680)+(e[i+1]<<8&65280)+(e[i+2]&255),a.push(Za(t));return a.join("")}function Ja(e){for(var o,n=e.length,t=n%3,a=[],i=16383,_=0,l=n-t;_<l;_+=i)a.push(Ya(e,_,_+i>l?l:_+i));return t===1?(o=e[n-1],a.push(W[o>>2]+W[o<<4&63]+"==")):t===2&&(o=(e[n-2]<<8)+e[n-1],a.push(W[o>>10]+W[o>>4&63]+W[o<<2&63]+"=")),a.join("")}});var Hn=eo(Po=>{r();Po.read=function(e,o,n,t,a){var i,_,l=a*8-t-1,d=(1<<l)-1,u=d>>1,h=-7,f=n?a-1:0,w=n?-1:1,k=e[o+f];for(f+=w,i=k&(1<<-h)-1,k>>=-h,h+=l;h>0;i=i*256+e[o+f],f+=w,h-=8);for(_=i&(1<<-h)-1,i>>=-h,h+=t;h>0;_=_*256+e[o+f],f+=w,h-=8);if(i===0)i=1-u;else{if(i===d)return _?NaN:(k?-1:1)*(1/0);_=_+Math.pow(2,t),i=i-u}return(k?-1:1)*_*Math.pow(2,i-t)};Po.write=function(e,o,n,t,a,i){var _,l,d,u=i*8-a-1,h=(1<<u)-1,f=h>>1,w=a===23?Math.pow(2,-24)-Math.pow(2,-77):0,k=t?0:i-1,M=t?1:-1,C=o<0||o===0&&1/o<0?1:0;for(o=Math.abs(o),isNaN(o)||o===1/0?(l=isNaN(o)?1:0,_=h):(_=Math.floor(Math.log(o)/Math.LN2),o*(d=Math.pow(2,-_))<1&&(_--,d*=2),_+f>=1?o+=w/d:o+=w*Math.pow(2,1-f),o*d>=2&&(_++,d/=2),_+f>=h?(l=0,_=h):_+f>=1?(l=(o*d-1)*Math.pow(2,a),_=_+f):(l=o*Math.pow(2,f-1)*Math.pow(2,a),_=0));a>=8;e[n+k]=l&255,k+=M,l/=256,a-=8);for(_=_<<a|l,u+=a;u>0;e[n+k]=_&255,k+=M,_/=256,u-=8);e[n+k-M]|=C*128}});var tt=eo(Se=>{"use strict";r();var Mo=Wn(),we=Hn(),jn=typeof Symbol=="function"&&typeof Symbol.for=="function"?Symbol.for("nodejs.util.inspect.custom"):null;Se.Buffer=s;Se.SlowBuffer=ri;Se.INSPECT_MAX_BYTES=50;var no=2147483647;Se.kMaxLength=no;s.TYPED_ARRAY_SUPPORT=Qa();!s.TYPED_ARRAY_SUPPORT&&typeof console!="undefined"&&typeof console.error=="function"&&console.error("This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.");function Qa(){try{var e=new Uint8Array(1),o={foo:function(){return 42}};return Object.setPrototypeOf(o,Uint8Array.prototype),Object.setPrototypeOf(e,o),e.foo()===42}catch(n){return!1}}Object.defineProperty(s.prototype,"parent",{enumerable:!0,get:function(){if(!!s.isBuffer(this))return this.buffer}});Object.defineProperty(s.prototype,"offset",{enumerable:!0,get:function(){if(!!s.isBuffer(this))return this.byteOffset}});function K(e){if(e>no)throw new RangeError('The value "'+e+'" is invalid for option "size"');var o=new Uint8Array(e);return Object.setPrototypeOf(o,s.prototype),o}function s(e,o,n){if(typeof e=="number"){if(typeof o=="string")throw new TypeError('The "string" argument must be of type string. Received type number');return Uo(e)}return Kn(e,o,n)}s.poolSize=8192;function Kn(e,o,n){if(typeof e=="string")return oi(e,o);if(ArrayBuffer.isView(e))return ni(e);if(e==null)throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e);if(H(e,ArrayBuffer)||e&&H(e.buffer,ArrayBuffer)||typeof SharedArrayBuffer!="undefined"&&(H(e,SharedArrayBuffer)||e&&H(e.buffer,SharedArrayBuffer)))return Bo(e,o,n);if(typeof e=="number")throw new TypeError('The "value" argument must not be of type number. Received type number');var t=e.valueOf&&e.valueOf();if(t!=null&&t!==e)return s.from(t,o,n);var a=ti(e);if(a)return a;if(typeof Symbol!="undefined"&&Symbol.toPrimitive!=null&&typeof e[Symbol.toPrimitive]=="function")return s.from(e[Symbol.toPrimitive]("string"),o,n);throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e)}s.from=function(e,o,n){return Kn(e,o,n)};Object.setPrototypeOf(s.prototype,Uint8Array.prototype);Object.setPrototypeOf(s,Uint8Array);function Xn(e){if(typeof e!="number")throw new TypeError('"size" argument must be of type number');if(e<0)throw new RangeError('The value "'+e+'" is invalid for option "size"')}function ei(e,o,n){return Xn(e),e<=0?K(e):o!==void 0?typeof n=="string"?K(e).fill(o,n):K(e).fill(o):K(e)}s.alloc=function(e,o,n){return ei(e,o,n)};function Uo(e){return Xn(e),K(e<0?0:Vo(e)|0)}s.allocUnsafe=function(e){return Uo(e)};s.allocUnsafeSlow=function(e){return Uo(e)};function oi(e,o){if((typeof o!="string"||o==="")&&(o="utf8"),!s.isEncoding(o))throw new TypeError("Unknown encoding: "+o);var n=Zn(e,o)|0,t=K(n),a=t.write(e,o);return a!==n&&(t=t.slice(0,a)),t}function Ro(e){for(var o=e.length<0?0:Vo(e.length)|0,n=K(o),t=0;t<o;t+=1)n[t]=e[t]&255;return n}function ni(e){if(H(e,Uint8Array)){var o=new Uint8Array(e);return Bo(o.buffer,o.byteOffset,o.byteLength)}return Ro(e)}function Bo(e,o,n){if(o<0||e.byteLength<o)throw new RangeError('"offset" is outside of buffer bounds');if(e.byteLength<o+(n||0))throw new RangeError('"length" is outside of buffer bounds');var t;return o===void 0&&n===void 0?t=new Uint8Array(e):n===void 0?t=new Uint8Array(e,o):t=new Uint8Array(e,o,n),Object.setPrototypeOf(t,s.prototype),t}function ti(e){if(s.isBuffer(e)){var o=Vo(e.length)|0,n=K(o);return n.length===0||e.copy(n,0,0,o),n}if(e.length!==void 0)return typeof e.length!="number"||$o(e.length)?K(0):Ro(e);if(e.type==="Buffer"&&Array.isArray(e.data))return Ro(e.data)}function Vo(e){if(e>=no)throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x"+no.toString(16)+" bytes");return e|0}function ri(e){return+e!=e&&(e=0),s.alloc(+e)}s.isBuffer=function(o){return o!=null&&o._isBuffer===!0&&o!==s.prototype};s.compare=function(o,n){if(H(o,Uint8Array)&&(o=s.from(o,o.offset,o.byteLength)),H(n,Uint8Array)&&(n=s.from(n,n.offset,n.byteLength)),!s.isBuffer(o)||!s.isBuffer(n))throw new TypeError('The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array');if(o===n)return 0;for(var t=o.length,a=n.length,i=0,_=Math.min(t,a);i<_;++i)if(o[i]!==n[i]){t=o[i],a=n[i];break}return t<a?-1:a<t?1:0};s.isEncoding=function(o){switch(String(o).toLowerCase()){case"hex":case"utf8":case"utf-8":case"ascii":case"latin1":case"binary":case"base64":case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return!0;default:return!1}};s.concat=function(o,n){if(!Array.isArray(o))throw new TypeError('"list" argument must be an Array of Buffers');if(o.length===0)return s.alloc(0);var t;if(n===void 0)for(n=0,t=0;t<o.length;++t)n+=o[t].length;var a=s.allocUnsafe(n),i=0;for(t=0;t<o.length;++t){var _=o[t];if(H(_,Uint8Array))i+_.length>a.length?s.from(_).copy(a,i):Uint8Array.prototype.set.call(a,_,i);else if(s.isBuffer(_))_.copy(a,i);else throw new TypeError('"list" argument must be an Array of Buffers');i+=_.length}return a};function Zn(e,o){if(s.isBuffer(e))return e.length;if(ArrayBuffer.isView(e)||H(e,ArrayBuffer))return e.byteLength;if(typeof e!="string")throw new TypeError('The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type '+typeof e);var n=e.length,t=arguments.length>2&&arguments[2]===!0;if(!t&&n===0)return 0;for(var a=!1;;)switch(o){case"ascii":case"latin1":case"binary":return n;case"utf8":case"utf-8":return Oo(e).length;case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return n*2;case"hex":return n>>>1;case"base64":return nt(e).length;default:if(a)return t?-1:Oo(e).length;o=(""+o).toLowerCase(),a=!0}}s.byteLength=Zn;function ai(e,o,n){var t=!1;if((o===void 0||o<0)&&(o=0),o>this.length||((n===void 0||n>this.length)&&(n=this.length),n<=0)||(n>>>=0,o>>>=0,n<=o))return"";for(e||(e="utf8");;)switch(e){case"hex":return gi(this,o,n);case"utf8":case"utf-8":return Jn(this,o,n);case"ascii":return li(this,o,n);case"latin1":case"binary":return ui(this,o,n);case"base64":return si(this,o,n);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return fi(this,o,n);default:if(t)throw new TypeError("Unknown encoding: "+e);e=(e+"").toLowerCase(),t=!0}}s.prototype._isBuffer=!0;function _e(e,o,n){var t=e[o];e[o]=e[n],e[n]=t}s.prototype.swap16=function(){var o=this.length;if(o%2!==0)throw new RangeError("Buffer size must be a multiple of 16-bits");for(var n=0;n<o;n+=2)_e(this,n,n+1);return this};s.prototype.swap32=function(){var o=this.length;if(o%4!==0)throw new RangeError("Buffer size must be a multiple of 32-bits");for(var n=0;n<o;n+=4)_e(this,n,n+3),_e(this,n+1,n+2);return this};s.prototype.swap64=function(){var o=this.length;if(o%8!==0)throw new RangeError("Buffer size must be a multiple of 64-bits");for(var n=0;n<o;n+=8)_e(this,n,n+7),_e(this,n+1,n+6),_e(this,n+2,n+5),_e(this,n+3,n+4);return this};s.prototype.toString=function(){var o=this.length;return o===0?"":arguments.length===0?Jn(this,0,o):ai.apply(this,arguments)};s.prototype.toLocaleString=s.prototype.toString;s.prototype.equals=function(o){if(!s.isBuffer(o))throw new TypeError("Argument must be a Buffer");return this===o?!0:s.compare(this,o)===0};s.prototype.inspect=function(){var o="",n=Se.INSPECT_MAX_BYTES;return o=this.toString("hex",0,n).replace(/(.{2})/g,"$1 ").trim(),this.length>n&&(o+=" ... "),"<Buffer "+o+">"};jn&&(s.prototype[jn]=s.prototype.inspect);s.prototype.compare=function(o,n,t,a,i){if(H(o,Uint8Array)&&(o=s.from(o,o.offset,o.byteLength)),!s.isBuffer(o))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+typeof o);if(n===void 0&&(n=0),t===void 0&&(t=o?o.length:0),a===void 0&&(a=0),i===void 0&&(i=this.length),n<0||t>o.length||a<0||i>this.length)throw new RangeError("out of range index");if(a>=i&&n>=t)return 0;if(a>=i)return-1;if(n>=t)return 1;if(n>>>=0,t>>>=0,a>>>=0,i>>>=0,this===o)return 0;for(var _=i-a,l=t-n,d=Math.min(_,l),u=this.slice(a,i),h=o.slice(n,t),f=0;f<d;++f)if(u[f]!==h[f]){_=u[f],l=h[f];break}return _<l?-1:l<_?1:0};function Yn(e,o,n,t,a){if(e.length===0)return-1;if(typeof n=="string"?(t=n,n=0):n>2147483647?n=2147483647:n<-2147483648&&(n=-2147483648),n=+n,$o(n)&&(n=a?0:e.length-1),n<0&&(n=e.length+n),n>=e.length){if(a)return-1;n=e.length-1}else if(n<0)if(a)n=0;else return-1;if(typeof o=="string"&&(o=s.from(o,t)),s.isBuffer(o))return o.length===0?-1:Gn(e,o,n,t,a);if(typeof o=="number")return o=o&255,typeof Uint8Array.prototype.indexOf=="function"?a?Uint8Array.prototype.indexOf.call(e,o,n):Uint8Array.prototype.lastIndexOf.call(e,o,n):Gn(e,[o],n,t,a);throw new TypeError("val must be string, number or Buffer")}function Gn(e,o,n,t,a){var i=1,_=e.length,l=o.length;if(t!==void 0&&(t=String(t).toLowerCase(),t==="ucs2"||t==="ucs-2"||t==="utf16le"||t==="utf-16le")){if(e.length<2||o.length<2)return-1;i=2,_/=2,l/=2,n/=2}function d(k,M){return i===1?k[M]:k.readUInt16BE(M*i)}var u;if(a){var h=-1;for(u=n;u<_;u++)if(d(e,u)===d(o,h===-1?0:u-h)){if(h===-1&&(h=u),u-h+1===l)return h*i}else h!==-1&&(u-=u-h),h=-1}else for(n+l>_&&(n=_-l),u=n;u>=0;u--){for(var f=!0,w=0;w<l;w++)if(d(e,u+w)!==d(o,w)){f=!1;break}if(f)return u}return-1}s.prototype.includes=function(o,n,t){return this.indexOf(o,n,t)!==-1};s.prototype.indexOf=function(o,n,t){return Yn(this,o,n,t,!0)};s.prototype.lastIndexOf=function(o,n,t){return Yn(this,o,n,t,!1)};function ii(e,o,n,t){n=Number(n)||0;var a=e.length-n;t?(t=Number(t),t>a&&(t=a)):t=a;var i=o.length;t>i/2&&(t=i/2);for(var _=0;_<t;++_){var l=parseInt(o.substr(_*2,2),16);if($o(l))return _;e[n+_]=l}return _}function pi(e,o,n,t){return to(Oo(o,e.length-n),e,n,t)}function ci(e,o,n,t){return to(yi(o),e,n,t)}function mi(e,o,n,t){return to(nt(o),e,n,t)}function _i(e,o,n,t){return to(bi(o,e.length-n),e,n,t)}s.prototype.write=function(o,n,t,a){if(n===void 0)a="utf8",t=this.length,n=0;else if(t===void 0&&typeof n=="string")a=n,t=this.length,n=0;else if(isFinite(n))n=n>>>0,isFinite(t)?(t=t>>>0,a===void 0&&(a="utf8")):(a=t,t=void 0);else throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");var i=this.length-n;if((t===void 0||t>i)&&(t=i),o.length>0&&(t<0||n<0)||n>this.length)throw new RangeError("Attempt to write outside buffer bounds");a||(a="utf8");for(var _=!1;;)switch(a){case"hex":return ii(this,o,n,t);case"utf8":case"utf-8":return pi(this,o,n,t);case"ascii":case"latin1":case"binary":return ci(this,o,n,t);case"base64":return mi(this,o,n,t);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return _i(this,o,n,t);default:if(_)throw new TypeError("Unknown encoding: "+a);a=(""+a).toLowerCase(),_=!0}};s.prototype.toJSON=function(){return{type:"Buffer",data:Array.prototype.slice.call(this._arr||this,0)}};function si(e,o,n){return o===0&&n===e.length?Mo.fromByteArray(e):Mo.fromByteArray(e.slice(o,n))}function Jn(e,o,n){n=Math.min(e.length,n);for(var t=[],a=o;a<n;){var i=e[a],_=null,l=i>239?4:i>223?3:i>191?2:1;if(a+l<=n){var d,u,h,f;switch(l){case 1:i<128&&(_=i);break;case 2:d=e[a+1],(d&192)===128&&(f=(i&31)<<6|d&63,f>127&&(_=f));break;case 3:d=e[a+1],u=e[a+2],(d&192)===128&&(u&192)===128&&(f=(i&15)<<12|(d&63)<<6|u&63,f>2047&&(f<55296||f>57343)&&(_=f));break;case 4:d=e[a+1],u=e[a+2],h=e[a+3],(d&192)===128&&(u&192)===128&&(h&192)===128&&(f=(i&15)<<18|(d&63)<<12|(u&63)<<6|h&63,f>65535&&f<1114112&&(_=f))}}_===null?(_=65533,l=1):_>65535&&(_-=65536,t.push(_>>>10&1023|55296),_=56320|_&1023),t.push(_),a+=l}return di(t)}var qn=4096;function di(e){var o=e.length;if(o<=qn)return String.fromCharCode.apply(String,e);for(var n="",t=0;t<o;)n+=String.fromCharCode.apply(String,e.slice(t,t+=qn));return n}function li(e,o,n){var t="";n=Math.min(e.length,n);for(var a=o;a<n;++a)t+=String.fromCharCode(e[a]&127);return t}function ui(e,o,n){var t="";n=Math.min(e.length,n);for(var a=o;a<n;++a)t+=String.fromCharCode(e[a]);return t}function gi(e,o,n){var t=e.length;(!o||o<0)&&(o=0),(!n||n<0||n>t)&&(n=t);for(var a="",i=o;i<n;++i)a+=Ti[e[i]];return a}function fi(e,o,n){for(var t=e.slice(o,n),a="",i=0;i<t.length-1;i+=2)a+=String.fromCharCode(t[i]+t[i+1]*256);return a}s.prototype.slice=function(o,n){var t=this.length;o=~~o,n=n===void 0?t:~~n,o<0?(o+=t,o<0&&(o=0)):o>t&&(o=t),n<0?(n+=t,n<0&&(n=0)):n>t&&(n=t),n<o&&(n=o);var a=this.subarray(o,n);return Object.setPrototypeOf(a,s.prototype),a};function N(e,o,n){if(e%1!==0||e<0)throw new RangeError("offset is not uint");if(e+o>n)throw new RangeError("Trying to access beyond buffer length")}s.prototype.readUintLE=s.prototype.readUIntLE=function(o,n,t){o=o>>>0,n=n>>>0,t||N(o,n,this.length);for(var a=this[o],i=1,_=0;++_<n&&(i*=256);)a+=this[o+_]*i;return a};s.prototype.readUintBE=s.prototype.readUIntBE=function(o,n,t){o=o>>>0,n=n>>>0,t||N(o,n,this.length);for(var a=this[o+--n],i=1;n>0&&(i*=256);)a+=this[o+--n]*i;return a};s.prototype.readUint8=s.prototype.readUInt8=function(o,n){return o=o>>>0,n||N(o,1,this.length),this[o]};s.prototype.readUint16LE=s.prototype.readUInt16LE=function(o,n){return o=o>>>0,n||N(o,2,this.length),this[o]|this[o+1]<<8};s.prototype.readUint16BE=s.prototype.readUInt16BE=function(o,n){return o=o>>>0,n||N(o,2,this.length),this[o]<<8|this[o+1]};s.prototype.readUint32LE=s.prototype.readUInt32LE=function(o,n){return o=o>>>0,n||N(o,4,this.length),(this[o]|this[o+1]<<8|this[o+2]<<16)+this[o+3]*16777216};s.prototype.readUint32BE=s.prototype.readUInt32BE=function(o,n){return o=o>>>0,n||N(o,4,this.length),this[o]*16777216+(this[o+1]<<16|this[o+2]<<8|this[o+3])};s.prototype.readIntLE=function(o,n,t){o=o>>>0,n=n>>>0,t||N(o,n,this.length);for(var a=this[o],i=1,_=0;++_<n&&(i*=256);)a+=this[o+_]*i;return i*=128,a>=i&&(a-=Math.pow(2,8*n)),a};s.prototype.readIntBE=function(o,n,t){o=o>>>0,n=n>>>0,t||N(o,n,this.length);for(var a=n,i=1,_=this[o+--a];a>0&&(i*=256);)_+=this[o+--a]*i;return i*=128,_>=i&&(_-=Math.pow(2,8*n)),_};s.prototype.readInt8=function(o,n){return o=o>>>0,n||N(o,1,this.length),this[o]&128?(255-this[o]+1)*-1:this[o]};s.prototype.readInt16LE=function(o,n){o=o>>>0,n||N(o,2,this.length);var t=this[o]|this[o+1]<<8;return t&32768?t|4294901760:t};s.prototype.readInt16BE=function(o,n){o=o>>>0,n||N(o,2,this.length);var t=this[o+1]|this[o]<<8;return t&32768?t|4294901760:t};s.prototype.readInt32LE=function(o,n){return o=o>>>0,n||N(o,4,this.length),this[o]|this[o+1]<<8|this[o+2]<<16|this[o+3]<<24};s.prototype.readInt32BE=function(o,n){return o=o>>>0,n||N(o,4,this.length),this[o]<<24|this[o+1]<<16|this[o+2]<<8|this[o+3]};s.prototype.readFloatLE=function(o,n){return o=o>>>0,n||N(o,4,this.length),we.read(this,o,!0,23,4)};s.prototype.readFloatBE=function(o,n){return o=o>>>0,n||N(o,4,this.length),we.read(this,o,!1,23,4)};s.prototype.readDoubleLE=function(o,n){return o=o>>>0,n||N(o,8,this.length),we.read(this,o,!0,52,8)};s.prototype.readDoubleBE=function(o,n){return o=o>>>0,n||N(o,8,this.length),we.read(this,o,!1,52,8)};function F(e,o,n,t,a,i){if(!s.isBuffer(e))throw new TypeError('"buffer" argument must be a Buffer instance');if(o>a||o<i)throw new RangeError('"value" argument is out of bounds');if(n+t>e.length)throw new RangeError("Index out of range")}s.prototype.writeUintLE=s.prototype.writeUIntLE=function(o,n,t,a){if(o=+o,n=n>>>0,t=t>>>0,!a){var i=Math.pow(2,8*t)-1;F(this,o,n,t,i,0)}var _=1,l=0;for(this[n]=o&255;++l<t&&(_*=256);)this[n+l]=o/_&255;return n+t};s.prototype.writeUintBE=s.prototype.writeUIntBE=function(o,n,t,a){if(o=+o,n=n>>>0,t=t>>>0,!a){var i=Math.pow(2,8*t)-1;F(this,o,n,t,i,0)}var _=t-1,l=1;for(this[n+_]=o&255;--_>=0&&(l*=256);)this[n+_]=o/l&255;return n+t};s.prototype.writeUint8=s.prototype.writeUInt8=function(o,n,t){return o=+o,n=n>>>0,t||F(this,o,n,1,255,0),this[n]=o&255,n+1};s.prototype.writeUint16LE=s.prototype.writeUInt16LE=function(o,n,t){return o=+o,n=n>>>0,t||F(this,o,n,2,65535,0),this[n]=o&255,this[n+1]=o>>>8,n+2};s.prototype.writeUint16BE=s.prototype.writeUInt16BE=function(o,n,t){return o=+o,n=n>>>0,t||F(this,o,n,2,65535,0),this[n]=o>>>8,this[n+1]=o&255,n+2};s.prototype.writeUint32LE=s.prototype.writeUInt32LE=function(o,n,t){return o=+o,n=n>>>0,t||F(this,o,n,4,4294967295,0),this[n+3]=o>>>24,this[n+2]=o>>>16,this[n+1]=o>>>8,this[n]=o&255,n+4};s.prototype.writeUint32BE=s.prototype.writeUInt32BE=function(o,n,t){return o=+o,n=n>>>0,t||F(this,o,n,4,4294967295,0),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};s.prototype.writeIntLE=function(o,n,t,a){if(o=+o,n=n>>>0,!a){var i=Math.pow(2,8*t-1);F(this,o,n,t,i-1,-i)}var _=0,l=1,d=0;for(this[n]=o&255;++_<t&&(l*=256);)o<0&&d===0&&this[n+_-1]!==0&&(d=1),this[n+_]=(o/l>>0)-d&255;return n+t};s.prototype.writeIntBE=function(o,n,t,a){if(o=+o,n=n>>>0,!a){var i=Math.pow(2,8*t-1);F(this,o,n,t,i-1,-i)}var _=t-1,l=1,d=0;for(this[n+_]=o&255;--_>=0&&(l*=256);)o<0&&d===0&&this[n+_+1]!==0&&(d=1),this[n+_]=(o/l>>0)-d&255;return n+t};s.prototype.writeInt8=function(o,n,t){return o=+o,n=n>>>0,t||F(this,o,n,1,127,-128),o<0&&(o=255+o+1),this[n]=o&255,n+1};s.prototype.writeInt16LE=function(o,n,t){return o=+o,n=n>>>0,t||F(this,o,n,2,32767,-32768),this[n]=o&255,this[n+1]=o>>>8,n+2};s.prototype.writeInt16BE=function(o,n,t){return o=+o,n=n>>>0,t||F(this,o,n,2,32767,-32768),this[n]=o>>>8,this[n+1]=o&255,n+2};s.prototype.writeInt32LE=function(o,n,t){return o=+o,n=n>>>0,t||F(this,o,n,4,2147483647,-2147483648),this[n]=o&255,this[n+1]=o>>>8,this[n+2]=o>>>16,this[n+3]=o>>>24,n+4};s.prototype.writeInt32BE=function(o,n,t){return o=+o,n=n>>>0,t||F(this,o,n,4,2147483647,-2147483648),o<0&&(o=4294967295+o+1),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};function Qn(e,o,n,t,a,i){if(n+t>e.length)throw new RangeError("Index out of range");if(n<0)throw new RangeError("Index out of range")}function et(e,o,n,t,a){return o=+o,n=n>>>0,a||Qn(e,o,n,4,34028234663852886e22,-34028234663852886e22),we.write(e,o,n,t,23,4),n+4}s.prototype.writeFloatLE=function(o,n,t){return et(this,o,n,!0,t)};s.prototype.writeFloatBE=function(o,n,t){return et(this,o,n,!1,t)};function ot(e,o,n,t,a){return o=+o,n=n>>>0,a||Qn(e,o,n,8,17976931348623157e292,-17976931348623157e292),we.write(e,o,n,t,52,8),n+8}s.prototype.writeDoubleLE=function(o,n,t){return ot(this,o,n,!0,t)};s.prototype.writeDoubleBE=function(o,n,t){return ot(this,o,n,!1,t)};s.prototype.copy=function(o,n,t,a){if(!s.isBuffer(o))throw new TypeError("argument should be a Buffer");if(t||(t=0),!a&&a!==0&&(a=this.length),n>=o.length&&(n=o.length),n||(n=0),a>0&&a<t&&(a=t),a===t||o.length===0||this.length===0)return 0;if(n<0)throw new RangeError("targetStart out of bounds");if(t<0||t>=this.length)throw new RangeError("Index out of range");if(a<0)throw new RangeError("sourceEnd out of bounds");a>this.length&&(a=this.length),o.length-n<a-t&&(a=o.length-n+t);var i=a-t;return this===o&&typeof Uint8Array.prototype.copyWithin=="function"?this.copyWithin(n,t,a):Uint8Array.prototype.set.call(o,this.subarray(t,a),n),i};s.prototype.fill=function(o,n,t,a){if(typeof o=="string"){if(typeof n=="string"?(a=n,n=0,t=this.length):typeof t=="string"&&(a=t,t=this.length),a!==void 0&&typeof a!="string")throw new TypeError("encoding must be a string");if(typeof a=="string"&&!s.isEncoding(a))throw new TypeError("Unknown encoding: "+a);if(o.length===1){var i=o.charCodeAt(0);(a==="utf8"&&i<128||a==="latin1")&&(o=i)}}else typeof o=="number"?o=o&255:typeof o=="boolean"&&(o=Number(o));if(n<0||this.length<n||this.length<t)throw new RangeError("Out of range index");if(t<=n)return this;n=n>>>0,t=t===void 0?this.length:t>>>0,o||(o=0);var _;if(typeof o=="number")for(_=n;_<t;++_)this[_]=o;else{var l=s.isBuffer(o)?o:s.from(o,a),d=l.length;if(d===0)throw new TypeError('The value "'+o+'" is invalid for argument "value"');for(_=0;_<t-n;++_)this[_+n]=l[_%d]}return this};var hi=/[^+/0-9A-Za-z-_]/g;function xi(e){if(e=e.split("=")[0],e=e.trim().replace(hi,""),e.length<2)return"";for(;e.length%4!==0;)e=e+"=";return e}function Oo(e,o){o=o||1/0;for(var n,t=e.length,a=null,i=[],_=0;_<t;++_){if(n=e.charCodeAt(_),n>55295&&n<57344){if(!a){if(n>56319){(o-=3)>-1&&i.push(239,191,189);continue}else if(_+1===t){(o-=3)>-1&&i.push(239,191,189);continue}a=n;continue}if(n<56320){(o-=3)>-1&&i.push(239,191,189),a=n;continue}n=(a-55296<<10|n-56320)+65536}else a&&(o-=3)>-1&&i.push(239,191,189);if(a=null,n<128){if((o-=1)<0)break;i.push(n)}else if(n<2048){if((o-=2)<0)break;i.push(n>>6|192,n&63|128)}else if(n<65536){if((o-=3)<0)break;i.push(n>>12|224,n>>6&63|128,n&63|128)}else if(n<1114112){if((o-=4)<0)break;i.push(n>>18|240,n>>12&63|128,n>>6&63|128,n&63|128)}else throw new Error("Invalid code point")}return i}function yi(e){for(var o=[],n=0;n<e.length;++n)o.push(e.charCodeAt(n)&255);return o}function bi(e,o){for(var n,t,a,i=[],_=0;_<e.length&&!((o-=2)<0);++_)n=e.charCodeAt(_),t=n>>8,a=n%256,i.push(a),i.push(t);return i}function nt(e){return Mo.toByteArray(xi(e))}function to(e,o,n,t){for(var a=0;a<t&&!(a+n>=o.length||a>=e.length);++a)o[a+n]=e[a];return a}function H(e,o){return e instanceof o||e!=null&&e.constructor!=null&&e.constructor.name!=null&&e.constructor.name===o.name}function $o(e){return e!==e}var Ti=function(){for(var e="0123456789abcdef",o=new Array(256),n=0;n<16;++n)for(var t=n*16,a=0;a<16;++a)o[t+a]=e[n]+e[a];return o}()});var ct=eo((mc,pt)=>{r();var A=pt.exports={},j,G;function zo(){throw new Error("setTimeout has not been defined")}function Wo(){throw new Error("clearTimeout has not been defined")}(function(){try{typeof setTimeout=="function"?j=setTimeout:j=zo}catch(e){j=zo}try{typeof clearTimeout=="function"?G=clearTimeout:G=Wo}catch(e){G=Wo}})();function rt(e){if(j===setTimeout)return setTimeout(e,0);if((j===zo||!j)&&setTimeout)return j=setTimeout,setTimeout(e,0);try{return j(e,0)}catch(o){try{return j.call(null,e,0)}catch(n){return j.call(this,e,0)}}}function wi(e){if(G===clearTimeout)return clearTimeout(e);if((G===Wo||!G)&&clearTimeout)return G=clearTimeout,clearTimeout(e);try{return G(e)}catch(o){try{return G.call(null,e)}catch(n){return G.call(this,e)}}}var X=[],ke=!1,se,ro=-1;function Si(){!ke||!se||(ke=!1,se.length?X=se.concat(X):ro=-1,X.length&&at())}function at(){if(!ke){var e=rt(Si);ke=!0;for(var o=X.length;o;){for(se=X,X=[];++ro<o;)se&&se[ro].run();ro=-1,o=X.length}se=null,ke=!1,wi(e)}}A.nextTick=function(e){var o=new Array(arguments.length-1);if(arguments.length>1)for(var n=1;n<arguments.length;n++)o[n-1]=arguments[n];X.push(new it(e,o)),X.length===1&&!ke&&rt(at)};function it(e,o){this.fun=e,this.array=o}it.prototype.run=function(){this.fun.apply(null,this.array)};A.title="browser";A.browser=!0;A.env={};A.argv=[];A.version="";A.versions={};function Z(){}A.on=Z;A.addListener=Z;A.once=Z;A.off=Z;A.removeListener=Z;A.removeAllListeners=Z;A.emit=Z;A.prependListener=Z;A.prependOnceListener=Z;A.listeners=function(e){return[]};A.binding=function(e){throw new Error("process.binding is not supported")};A.cwd=function(){return"/"};A.chdir=function(e){throw new Error("process.chdir is not supported")};A.umask=function(){return 0}});var p,c,_c,r=y(()=>{p=T(tt()),c=T(ct()),_c=function(e){function o(){var t=this||self;return delete e.prototype.__magic__,t}if(typeof globalThis=="object")return globalThis;if(this)return o();e.defineProperty(e.prototype,"__magic__",{configurable:!0,get:o});var n=__magic__;return n}(Object)});var L=y(()=>{"use strict";r()});var _t=y(()=>{"use strict";r()});var st=y(()=>{"use strict";r()});var $=y(()=>{"use strict";r()});var dt=y(()=>{"use strict";r()});var lt=y(()=>{"use strict";r()});var ut=y(()=>{"use strict";r()});var gt=y(()=>{"use strict";r()});var de=y(()=>{"use strict";r();st();L();$();dt();lt();ut();gt()});var Ho,ft,Lc,ht,Dc,xt,yt,Cc,Fc,bt=y(()=>{"use strict";r();de();Ho={devnet:{id:"devnet",chainId:"D",name:"Devnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://devnet-wallet.elrond.com",apiAddress:"https://devnet-api.elrond.com",explorerAddress:"http://devnet-explorer.elrond.com",apiTimeout:"4000"},testnet:{id:"testnet",chainId:"T",name:"Testnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://testnet-wallet.elrond.com",apiAddress:"https://testnet-api.elrond.com",explorerAddress:"http://testnet-explorer.elrond.com",apiTimeout:"4000"},mainnet:{id:"mainnet",chainId:"1",name:"Mainnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://wallet.elrond.com",apiAddress:"https://api.elrond.com",explorerAddress:"https://explorer.elrond.com",apiTimeout:"4000"}},{chainId:ft,egldLabel:Lc}=Ho["devnet"],{chainId:ht,egldLabel:Dc}=Ho["testnet"],{chainId:xt,egldLabel:yt}=Ho["mainnet"],Cc={["devnet"]:ft,["testnet"]:ht,["mainnet"]:xt},Fc={[ft]:"devnet",[ht]:"testnet",[xt]:"mainnet"}});var jo=y(()=>{"use strict";r()});var Tt=y(()=>{"use strict";r()});var wt,Go=y(()=>{"use strict";r();wt="transaction-status-toast-id"});var qo,Ko,D,ve,St,io,kt,z,I=y(()=>{"use strict";r();_t();bt();jo();Tt();Go();qo=5e4,Ko=1e9,D=18,ve=4,St=1,io="logout",kt="login",z="0"});var po,Zo=y(()=>{"use strict";r();po=()=>Date.now()/1e3});var Lt=y(()=>{"use strict";r()});var Dt=y(()=>{"use strict";r()});var Ct=y(()=>{"use strict";r();Zo();Lt();Dt()});var Jo={};ce(Jo,{getItem:()=>Ii,localStorageKeys:()=>le,removeItem:()=>Ei,setItem:()=>Ai});var le,Yo,Ai,Ii,Ei,$e=y(()=>{"use strict";r();Ct();le={loginExpiresAt:"dapp-core-login-expires-at",logoutEvent:"dapp-core-logout-event"},Yo=typeof localStorage!="undefined",Ai=({key:e,data:o,expires:n})=>{!Yo||localStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},Ii=e=>{if(!Yo)return;let o=localStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:po()>=n.expires?(localStorage.removeItem(String(e)),null):n.data},Ei=e=>{!Yo||localStorage.removeItem(String(e))}});var Qo={};ce(Qo,{clear:()=>Rt,getItem:()=>Pt,removeItem:()=>Mt,setItem:()=>Ft,storage:()=>Ni});var Ft,Pt,Mt,Rt,Ni,Bt=y(()=>{"use strict";r();Ft=({key:e,data:o,expires:n})=>{sessionStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},Pt=e=>{let o=sessionStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:Date.now()>=n.expires?(sessionStorage.removeItem(String(e)),null):n.data},Mt=e=>sessionStorage.removeItem(String(e)),Rt=()=>sessionStorage.clear(),Ni={setItem:Ft,getItem:Pt,removeItem:Mt,clear:Rt}});var Ae,co=y(()=>{"use strict";r();$e();Bt();Ae={session:Qo,local:Jo}});function ze(){return new Date().setHours(new Date().getHours()+24)}function We(e){Ae.local.setItem({key:le.loginExpiresAt,data:e,expires:e})}var en=y(()=>{"use strict";r();co();$e()});var on,P,mo,te=y(()=>{"use strict";r();on=require("@reduxjs/toolkit");I();P=(0,on.createAction)(io),mo=(0,on.createAction)(kt,e=>({payload:e}))});var Ut,Ot,Vt,Am,Im,Em,Nm,Lm,Dm,$t,zt,nn=y(()=>{"use strict";r();Ut=require("@reduxjs/toolkit");en();$();te();Ot={loginMethod:"",walletConnectLogin:null,ledgerLogin:null,tokenLogin:null,walletLogin:null,extensionLogin:null,isLoginSessionInvalid:!1},Vt=(0,Ut.createSlice)({name:"loginInfoSlice",initialState:Ot,reducers:{setLoginMethod:(e,o)=>{e.loginMethod=o.payload},setTokenLogin:(e,o)=>{e.tokenLogin=o.payload},setTokenLoginSignature:(e,o)=>{(e==null?void 0:e.tokenLogin)!=null&&(e.tokenLogin.signature=o.payload)},setWalletLogin:(e,o)=>{e.walletLogin=o.payload},setWalletConnectLogin:(e,o)=>{e.walletConnectLogin=o.payload},setLedgerLogin:(e,o)=>{e.ledgerLogin=o.payload},invalidateLoginSession:e=>{e.isLoginSessionInvalid=!0}},extraReducers:e=>{e.addCase(P,()=>Ot),e.addCase(mo,(o,n)=>{o.loginMethod=n.payload.loginMethod,We(ze())})}}),{setLoginMethod:Am,setWalletConnectLogin:Im,setLedgerLogin:Em,setTokenLogin:Nm,setTokenLoginSignature:Lm,setWalletLogin:Dm,invalidateLoginSession:$t}=Vt.actions,zt=Vt.reducer});var Wt=y(()=>{"use strict";r()});var Ht=y(()=>{"use strict";r();Wt()});function tn(e){return e[Math.floor(Math.random()*e.length)]}var jt=y(()=>{"use strict";r()});var Gt=y(()=>{"use strict";r();jo()});var qt=y(()=>{"use strict";r();Ht();jt();Gt()});var Kt,Xt,rn,Di,Zt,qm,Km,Yt,an=y(()=>{"use strict";r();Kt=require("@reduxjs/toolkit"),Xt=T(require("lodash.omit"));qt();rn={id:"not-configured",chainId:"",name:"NOT CONFIGURED",egldLabel:"",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"",walletConnectBridgeAddress:"",walletConnectV2RelayAddress:"",walletConnectV2ProjectId:"",walletAddress:"",apiAddress:"",explorerAddress:"",apiTimeout:"4000"},Di={network:rn,chainID:"-1"},Zt=(0,Kt.createSlice)({name:"appConfig",initialState:Di,reducers:{initializeNetworkConfig:(e,o)=>{let n=tn(o.payload.walletConnectBridgeAddresses),t=tn(o.payload.walletConnectV2RelayAddresses),a=(0,Xt.default)(o.payload,"walletConnectBridgeAddresses","walletConnectV2RelayAddresses");e.network=S(b(b({},e.network),a),{walletConnectBridgeAddress:n,walletConnectV2RelayAddress:t})},setChainID:(e,o)=>{e.chainID=o.payload}}}),{initializeNetworkConfig:qm,setChainID:Km}=Zt.actions,Yt=Zt.reducer});var cn,Jt,Qt,_o,pn,er,or,n_,nr,t_,r_,a_,i_,p_,c_,m_,tr,mn=y(()=>{"use strict";r();cn=require("@elrondnetwork/erdjs"),Jt=require("@reduxjs/toolkit"),Qt=require("redux-persist");I();co();$e();te();_o={balance:"...",address:"",nonce:0,txCount:0,scrCount:0,claimableRewards:z},pn={address:"",websocketEvent:null,accounts:{"":_o},ledgerAccount:null,publicKey:"",walletConnectAccount:null,isAccountLoading:!0,accountLoadingError:null},er=(0,Jt.createSlice)({name:"accountInfoSlice",initialState:pn,reducers:{setAddress:(e,o)=>{let n=o.payload;e.address=n,e.publicKey=new cn.Address(n).hex()},setAccount:(e,o)=>{let n=e.address===o.payload.address;e.accounts={[e.address]:n?o.payload:_o},e.isAccountLoading=!1,e.accountLoadingError=null},setAccountNonce:(e,o)=>{let{address:n}=e;e.accounts[n].nonce=o.payload},setAccountShard:(e,o)=>{e.shard=o.payload},setLedgerAccount:(e,o)=>{e.ledgerAccount=o.payload},updateLedgerAccount:(e,o)=>{e.ledgerAccount!=null&&(e.ledgerAccount.index=o.payload.index,e.ledgerAccount.address=o.payload.address)},setWalletConnectAccount:(e,o)=>{e.walletConnectAccount=o.payload},setIsAccountLoading:(e,o)=>{e.isAccountLoading=o.payload,e.accountLoadingError=null},setAccountLoadingError:(e,o)=>{e.accountLoadingError=o.payload,e.isAccountLoading=!1},setWebsocketEvent:(e,o)=>{e.websocketEvent={timestamp:Date.now(),message:o.payload}}},extraReducers:e=>{e.addCase(P,()=>(Ae.local.removeItem(le.loginExpiresAt),pn)),e.addCase(mo,(o,n)=>{let{address:t}=n.payload;o.address=t,o.publicKey=new cn.Address(t).hex()}),e.addCase(Qt.REHYDRATE,(o,n)=>{var u;if(!((u=n.payload)!=null&&u.account))return;let{account:t}=n.payload,{address:a,shard:i,accounts:_,publicKey:l}=t;o.address=a,o.shard=i;let d=_&&a in _;o.accounts=d?_:pn.accounts,o.publicKey=l})}}),{setAccount:or,setAddress:n_,setAccountNonce:nr,setAccountShard:t_,setLedgerAccount:r_,updateLedgerAccount:a_,setWalletConnectAccount:i_,setIsAccountLoading:p_,setAccountLoadingError:c_,setWebsocketEvent:m_}=er.actions,tr=er.reducer});function rr(e){return e!=null&&(Vi(e)||Hi(e))}function so(e){return e!=null&&($i(e)||ji(e))}function lo(e){return e!=null&&(zi(e)||Gi(e))}function ar(e){return e!=null&&Wi(e)}function Vi(e){return e!=null&&Fi.includes(e)}function $i(e){return e!=null&&Pi.includes(e)}function zi(e){return e!=null&&Mi.includes(e)}function Wi(e){return e!=null&&Ri.includes(e)}function Hi(e){return e!=null&&Bi.includes(e)}function ji(e){return e!=null&&Oi.includes(e)}function Gi(e){return e!=null&&Ui.includes(e)}var Fi,Pi,Mi,Ri,Bi,Oi,Ui,sn=y(()=>{"use strict";r();$();Fi=["sent"],Pi=["success"],Mi=["fail","cancelled","timedOut"],Ri=["timedOut"],Bi=["pending"],Oi=["success"],Ui=["fail","invalid"]});var ir,pr,He,qi,cr,x_,y_,b_,T_,w_,S_,k_,v_,A_,I_,uo,go=y(()=>{"use strict";r();ir=require("@reduxjs/toolkit"),pr=require("redux-persist");$();sn();te();He={signedTransactions:{},transactionsToSign:null,signTransactionsError:null,signTransactionsCancelMessage:null,customTransactionInformationForSessionId:{}},qi={signWithoutSending:!1,sessionInformation:null,redirectAfterSign:!1},cr=(0,ir.createSlice)({name:"transactionsSlice",initialState:He,reducers:{moveTransactionsToSignedState:(e,o)=>{var d,u;let{sessionId:n,transactions:t,errorMessage:a,status:i,redirectRoute:_}=o.payload,l=((d=e.customTransactionInformationForSessionId)==null?void 0:d[n])||qi;e.signedTransactions[n]={transactions:t,status:i,errorMessage:a,redirectRoute:_,customTransactionInformation:l},((u=e==null?void 0:e.transactionsToSign)==null?void 0:u.sessionId)===n&&(e.transactionsToSign=He.transactionsToSign)},clearSignedTransaction:(e,o)=>{e.signedTransactions[o.payload]&&delete e.signedTransactions[o.payload]},clearTransactionToSign:e=>{e!=null&&e.transactionsToSign&&(e.transactionsToSign=null)},updateSignedTransaction:(e,o)=>{e.signedTransactions=b(b({},e.signedTransactions),o.payload)},updateSignedTransactions:(e,o)=>{let{sessionId:n,status:t,errorMessage:a,transactions:i}=o.payload;e.signedTransactions[n]!=null&&(e.signedTransactions[n].status=t,a!=null&&(e.signedTransactions[n].errorMessage=a),i!=null&&(e.signedTransactions[n].transactions=i))},updateSignedTransactionStatus:(e,o)=>{var l,d,u,h,f,w;let{sessionId:n,status:t,errorMessage:a,transactionHash:i}=o.payload,_=(d=(l=e.signedTransactions)==null?void 0:l[n])==null?void 0:d.transactions;if(_!=null){e.signedTransactions[n].transactions=_.map(C=>C.hash===i?S(b({},C),{status:t,errorMessage:a}):C);let k=(h=(u=e.signedTransactions[n])==null?void 0:u.transactions)==null?void 0:h.every(C=>so(C.status)),M=(w=(f=e.signedTransactions[n])==null?void 0:f.transactions)==null?void 0:w.every(C=>lo(C.status));k&&(e.signedTransactions[n].status="success"),M&&(e.signedTransactions[n].status="fail")}},setTransactionsToSign:(e,o)=>{e.transactionsToSign=o.payload;let{sessionId:n,customTransactionInformation:t}=o.payload;e.customTransactionInformationForSessionId[n]=t,e.signTransactionsError=null},clearAllTransactionsToSign:e=>{e.transactionsToSign=He.transactionsToSign,e.signTransactionsError=null},clearAllSignedTransactions:e=>{e.signedTransactions=He.signedTransactions},setSignTransactionsError:(e,o)=>{e.signTransactionsError=o.payload},setSignTransactionsCancelMessage:(e,o)=>{e.signTransactionsCancelMessage=o.payload}},extraReducers:e=>{e.addCase(P,()=>He),e.addCase(pr.REHYDRATE,(o,n)=>{var _;if(!((_=n.payload)!=null&&_.transactions))return;let{signedTransactions:t,customTransactionInformationForSessionId:a}=n.payload.transactions,i=Object.entries(t).reduce((l,[d,u])=>{let h=new Date(d),f=new Date;return f.setHours(f.getHours()+5),f-h>0||(l[d]=u),l},{});a!=null&&(o.customTransactionInformationForSessionId=a),t!=null&&(o.signedTransactions=i)})}}),{updateSignedTransactionStatus:x_,updateSignedTransactions:y_,setTransactionsToSign:b_,clearAllTransactionsToSign:T_,clearAllSignedTransactions:w_,clearSignedTransaction:S_,clearTransactionToSign:k_,setSignTransactionsError:v_,setSignTransactionsCancelMessage:A_,moveTransactionsToSignedState:I_}=cr.actions,uo=cr.reducer});var mr,ln,un,gn,Ki,dn,_r,D_,C_,F_,fo,ho=y(()=>{"use strict";r();mr=require("@reduxjs/toolkit");te();ln="Transaction failed",un="Transaction successful",gn="Processing transaction",Ki="Transaction submitted",dn={},_r=(0,mr.createSlice)({name:"transactionsInfo",initialState:dn,reducers:{setTransactionsDisplayInfo(e,o){let{sessionId:n,transactionsDisplayInfo:t}=o.payload;n!=null&&(e[n]={errorMessage:(t==null?void 0:t.errorMessage)||ln,successMessage:(t==null?void 0:t.successMessage)||un,processingMessage:(t==null?void 0:t.processingMessage)||gn,submittedMessage:(t==null?void 0:t.submittedMessage)||Ki,transactionDuration:t==null?void 0:t.transactionDuration})},clearTransactionsInfoForSessionId(e,o){o.payload!=null&&delete e[o.payload]},clearTransactionsInfo:()=>dn},extraReducers:e=>{e.addCase(P,()=>dn)}}),{clearTransactionsInfo:D_,setTransactionsDisplayInfo:C_,clearTransactionsInfoForSessionId:F_}=_r.actions,fo=_r.reducer});var dr,sr,lr,B_,O_,U_,V_,ur,fn=y(()=>{"use strict";r();dr=require("@reduxjs/toolkit");te();sr={},lr=(0,dr.createSlice)({name:"modalsSlice",initialState:sr,reducers:{setTxSubmittedModal:(e,o)=>{e.txSubmittedModal=o.payload},setNotificationModal:(e,o)=>{e.notificationModal=o.payload},clearTxSubmittedModal:e=>{e.txSubmittedModal=void 0},clearNotificationModal:e=>{e.notificationModal=void 0}},extraReducers:e=>{e.addCase(P,()=>sr)}}),{setTxSubmittedModal:B_,setNotificationModal:O_,clearTxSubmittedModal:U_,clearNotificationModal:V_}=lr.actions,ur=lr.reducer});var fr,gr,hr,K_,X_,Z_,Y_,J_,Q_,xo,yo=y(()=>{"use strict";r();fr=require("@reduxjs/toolkit");Go();de();Zo();te();gr={customToasts:[],transactionToasts:[],failTransactionToast:null},hr=(0,fr.createSlice)({name:"toastsSlice",initialState:gr,reducers:{addCustomToast:(e,o)=>{e.customToasts.push(S(b({},o.payload),{type:"custom",toastId:o.payload.toastId||`custom-toast-${e.customToasts.length+1}`}))},removeCustomToast:(e,o)=>{e.customToasts=e.customToasts.filter(n=>n.toastId!==o.payload)},addTransactionToast:(e,o)=>{e.transactionToasts.push({type:"transaction",startTimestamp:po(),toastId:o.payload||`custom-toast-${e.transactionToasts.length+1}`})},removeTransactionToast:(e,o)=>{e.transactionToasts=e.transactionToasts.filter(n=>n.toastId!==o.payload)},addFailTransactionToast:(e,o)=>{e.failTransactionToast=S(b({},o.payload),{toastId:wt})},removeFailTransactionToast:e=>{e.failTransactionToast=null}},extraReducers:e=>{e.addCase(P,()=>gr)}}),{addCustomToast:K_,removeCustomToast:X_,addTransactionToast:Z_,removeTransactionToast:Y_,addFailTransactionToast:J_,removeFailTransactionToast:Q_}=hr.actions,xo=hr.reducer});var Ie=y(()=>{"use strict";r();nn();an();mn();go();ho();fn();yo()});var hn=y(()=>{"use strict";r()});var Tr,br,wr,ys,bs,Ts,ws,Ss,ks,Sr,kr=y(()=>{"use strict";r();Tr=require("@reduxjs/toolkit");te();br={visible:!1,modal:{showHeader:!0,showFooter:!1,headerText:"",footerText:""}},wr=(0,Tr.createSlice)({name:"dappModalsSlice",initialState:br,reducers:{setDappModalVisibility:(e,o)=>{e.visible=o.payload},setDappModalHeaderVisibility:(e,o)=>{e.modal.showHeader=o.payload},setDappModalFooterVisibility:(e,o)=>{e.modal.showFooter=o.payload},setDappModalHeaderText:(e,o)=>{e.modal.headerText=o.payload},setDappModalFooterText:(e,o)=>{e.modal.footerText=o.payload},setDappModalConfig:(e,o)=>{e.modal=b({},o.payload)}},extraReducers:e=>{e.addCase(P,()=>br)}}),{setDappModalVisibility:ys,setDappModalHeaderVisibility:bs,setDappModalFooterVisibility:Ts,setDappModalHeaderText:ws,setDappModalFooterText:Ss,setDappModalConfig:ks}=wr.actions,Sr=wr.reducer});var vr,Ji,Qi,Ee,To=y(()=>{"use strict";r();vr=require("@reduxjs/toolkit");hn();mn();kr();nn();fn();an();yo();ho();go();Ji={["account"]:tr,["networkConfig"]:Yt,["loginInfo"]:zt,["modals"]:ur,["dappModal"]:Sr,["toasts"]:xo,["transactions"]:uo,["transactionsInfo"]:fo},Qi=(e={})=>(0,vr.combineReducers)(b(b({},Ji),e)),Ee=Qi});var Er={};ce(Er,{default:()=>ip});function xn(e,o=[]){return{key:e,version:1,storage:Ir.default,blacklist:o}}var ue,Ar,Ir,ep,op,np,tp,rp,ap,ip,Nr=y(()=>{"use strict";r();ue=require("redux-persist"),Ar=T(require("redux-persist/lib/storage")),Ir=T(require("redux-persist/lib/storage/session"));To();Ie();yo();ho();go();hn();ep={2:e=>S(b({},e),{networkConfig:rn})};op=xn("dapp-core-transactionsInfo"),np=xn("dapp-core-transactions",["transactionsToSign"]),tp=xn("dapp-core-toasts"),rp={key:"dapp-core-store",version:2,storage:Ar.default,whitelist:["account","loginInfo","modals","networkConfig"],migrate:(0,ue.createMigrate)(ep,{debug:!1})},ap={["toasts"]:(0,ue.persistReducer)(tp,xo),["transactions"]:(0,ue.persistReducer)(np,uo),["transactionsInfo"]:(0,ue.persistReducer)(op,fo)},ip=(0,ue.persistReducer)(rp,Ee(ap))});var Lr={};ce(Lr,{default:()=>pp});var pp,Dr=y(()=>{"use strict";r();To();pp=Ee()});var Cr={};ce(Cr,{default:()=>mp});var U,cp,mp,Fr=y(()=>{"use strict";r();U=require("redux-persist"),cp=[U.FLUSH,U.REHYDRATE,U.PAUSE,U.PERSIST,U.PURGE,U.REGISTER],mp=cp});var Rr={};ce(Rr,{default:()=>Mr});function Mr(e){return(0,Pr.persistStore)(e)}var Pr,Br=y(()=>{"use strict";r();Pr=require("redux-persist")});var tc={};ce(tc,{TransactionAction:()=>nc});module.exports=Ve(tc);r();var Be=T(require("react"));r();L();r();L();var mt=e=>{var o,n;switch(e.name){case"transfer":return["Transfer",{token:(o=e.arguments)==null?void 0:o.transfers},"to",{address:(n=e.arguments)==null?void 0:n.receiver}];default:return[]}};r();var vt=T(require("bignumber.js"));I();L();var At=e=>{var o,n,t,a,i,_,l,d,u,h,f,w;switch(e.name){case"claimLockedAssets":return["Claim locked assets"];case"enterFarm":case"enterFarmProxy":return["Enter farm with",{token:(o=e.arguments)==null?void 0:o.transfers}];case"enterFarmAndLockRewards":case"enterFarmAndLockRewardsProxy":return["Enter farm and lock rewards with",{token:(n=e.arguments)==null?void 0:n.transfers}];case"exitFarm":case"exitFarmProxy":return["Exit farm with",{token:(t=e.arguments)==null?void 0:t.transfers}];case"claimRewards":case"claimRewardsProxy":return["Claim rewards",{token:(a=e.arguments)==null?void 0:a.transfers}];case"compoundRewards":case"compoundRewardsProxy":return["Reinvest rewards",{token:(i=e.arguments)==null?void 0:i.transfers}];case"swapTokensFixedInput":case"swap":return e.description?[e.description]:[];case"swapTokensFixedOutput":return e.description?[e.description]:[];case"addLiquidity":case"addLiquidityProxy":return["Added liquidity for",{token:[(_=e.arguments)==null?void 0:_.transfers[0]]},"and",{token:[(l=e.arguments)==null?void 0:l.transfers[1]]}];case"removeLiquidity":case"removeLiquidityProxy":return["Removed liquidity with ",{token:(d=e.arguments)==null?void 0:d.transfers}];case"mergeLockedAssetTokens":let k=z;if((u=e.arguments)!=null&&u.transfers){let M=e.arguments.transfers.map(({value:C})=>C);k=vt.default.sum.apply(null,M).toString(10)}return[`Merge ${(h=e.arguments)==null?void 0:h.transfers.length}`,{tokenNoLink:[(f=e.arguments)==null?void 0:f.transfers[0]]},"positions into a single",{tokenNoLink:[(w=e.arguments)==null?void 0:w.transfers[0]]},"position of value",{value:k}];case"wrapEgld":case"unwrapEgld":default:return e.description?[e.description]:[]}};r();L();var It=e=>{var o,n,t,a,i,_,l,d,u,h,f,w;switch(e.name){case"delegate":case"stake":return["Delegate",{egldValue:(o=e.arguments)==null?void 0:o.value},"to staking provider",{providerName:(n=e.arguments)==null?void 0:n.providerName,providerAvatar:(t=e.arguments)==null?void 0:t.providerAvatar}];case"unDelegate":return["Undelegate",{egldValue:(a=e.arguments)==null?void 0:a.value},"from staking provider",{providerName:(i=e.arguments)==null?void 0:i.providerName,providerAvatar:(_=e.arguments)==null?void 0:_.providerAvatar}];case"claimRewards":return["Claim rewards from staking provider",{providerName:(l=e.arguments)==null?void 0:l.providerName,providerAvatar:(d=e.arguments)==null?void 0:d.providerAvatar}];case"reDelegateRewards":return["Redelegate rewards from staking provider",{providerName:(u=e.arguments)==null?void 0:u.providerName,providerAvatar:(h=e.arguments)==null?void 0:h.providerAvatar}];case"withdraw":return["Withdraw from staking provider",{providerName:(f=e.arguments)==null?void 0:f.providerName,providerAvatar:(w=e.arguments)==null?void 0:w.providerAvatar}];default:return[]}};var Et=e=>{if(!e.arguments)return e.description?[e.description]:[e.name];switch(e.category){case"esdtNft":return mt(e);case"mex":return At(e);case"stake":return It(e);default:return e.description?[e.description]:[]}};r();var v=T(require("react")),be=T(require("classnames"));r();var Xo=T(require("react")),vi=e=>Xo.createElement("svg",b({"aria-hidden":"true",focusable:"false","data-prefix":"far","data-icon":"globe",role:"img",xmlns:"http://www.w3.org/2000/svg",viewBox:"0 0 496 512",className:"svg-inline--fa fa-globe fa-w-16 fa-5x"},e),Xo.createElement("path",{fill:"#6c757d",d:"M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm179.3 160h-67.2c-6.7-36.5-17.5-68.8-31.2-94.7 42.9 19 77.7 52.7 98.4 94.7zM248 56c18.6 0 48.6 41.2 63.2 112H184.8C199.4 97.2 229.4 56 248 56zM48 256c0-13.7 1.4-27.1 4-40h77.7c-1 13.1-1.7 26.3-1.7 40s.7 26.9 1.7 40H52c-2.6-12.9-4-26.3-4-40zm20.7 88h67.2c6.7 36.5 17.5 68.8 31.2 94.7-42.9-19-77.7-52.7-98.4-94.7zm67.2-176H68.7c20.7-42 55.5-75.7 98.4-94.7-13.7 25.9-24.5 58.2-31.2 94.7zM248 456c-18.6 0-48.6-41.2-63.2-112h126.5c-14.7 70.8-44.7 112-63.3 112zm70.1-160H177.9c-1.1-12.8-1.9-26-1.9-40s.8-27.2 1.9-40h140.3c1.1 12.8 1.9 26 1.9 40s-.9 27.2-2 40zm10.8 142.7c13.7-25.9 24.4-58.2 31.2-94.7h67.2c-20.7 42-55.5 75.7-98.4 94.7zM366.3 296c1-13.1 1.7-26.3 1.7-40s-.7-26.9-1.7-40H444c2.6 12.9 4 26.3 4 40s-1.4 27.1-4 40h-77.7z",className:""})),Nt=vi;r();r();var Cn=T(require("react")),fa=require("@fortawesome/free-solid-svg-icons"),ha=require("@fortawesome/react-fontawesome"),xa=T(require("classnames"));r();r();var Wr=require("react"),Le=require("react-redux");r();var yn=require("@reduxjs/toolkit"),$r=require("react-redux/es/utils/Subscription");Ie();r();var xr=T(require("lodash.throttle"));I();Ie();en();co();$e();var Xi=[io],Zi=(0,xr.default)(()=>{We(ze())},5e3),yr=e=>o=>n=>{var d;if(Xi.includes(n.type))return o(n);let t=e.getState(),a=Ae.local.getItem(le.loginExpiresAt);if(!((d=t==null?void 0:t.loginInfo)==null?void 0:d.isLoggedIn))return o(n);if(a==null)return We(ze());let _=Date.now();return a-_<0?setTimeout(()=>Vn(void 0,null,function*(){console.log("session expired");try{e.dispatch($t())}catch(u){console.error("error logging out",u)}}),1e3):(Zi(),o(n))};r();r();function bo(){return typeof sessionStorage!="undefined"}var Or=bo()?(Nr(),Ve(Er)).default:(Dr(),Ve(Lr)).default,Ur=bo()?(Fr(),Ve(Cr)).default:[],Vr=bo()?(Br(),Ve(Rr)).default:e=>e;To();var Ne=(0,yn.configureStore)({reducer:Or,middleware:e=>e({serializableCheck:{ignoredActions:[...Ur,nr.type,or.type],ignoredPaths:["payload.nonce","account.account.nonce","providers.accountProvider"]}}).concat(yr)}),zr=(0,$r.createSubscription)(Ne),Xs=Vr(Ne),Zs=(0,yn.configureStore)({reducer:Ee});var _p={store:Ne,subscription:zr},bn=(0,Wr.createContext)(_p),ed=(0,Le.createStoreHook)(bn),od=(0,Le.createDispatchHook)(bn),Hr=(0,Le.createSelectorHook)(bn);r();r();Ie();r();var jr=T(require("lodash.isequal")),wo=require("reselect"),x=(0,wo.createSelectorCreator)(wo.defaultMemoize,jr.default);var Y=e=>e.account,Tn=x(Y,e=>e.address),wn=x(Y,Tn,(e,o)=>o in e.accounts?e.accounts[o]:_o),id=x(Y,wn,(e,o)=>{let a=e,{accounts:n}=a,t=Te(a,["accounts"]);return S(b({},t),{account:o})}),pd=x(wn,e=>e.balance),cd=x(wn,e=>{var o;return((o=e==null?void 0:e.nonce)==null?void 0:o.valueOf())||0}),md=x(Y,e=>e.shard),_d=x(Y,e=>e.ledgerAccount),sd=x(Y,e=>e.walletConnectAccount),dd=x(Y,e=>e.isAccountLoading),ld=x(Y,e=>e.accountLoadingError),ud=x(Y,e=>e.websocketEvent);r();var re=e=>e.networkConfig,hd=x(re,e=>e.chainID),xd=x(re,e=>e.network.walletConnectBridgeAddress),yd=x(re,e=>e.network.walletConnectV2RelayAddress),bd=x(re,e=>e.network.walletConnectV2ProjectId),Td=x(re,e=>e.network.walletConnectDeepLink),Sn=x(re,e=>e.network),wd=x(Sn,e=>e.apiAddress),Gr=x(Sn,e=>e.explorerAddress),Sd=x(Sn,e=>e.egldLabel);r();$();var De=e=>e.loginInfo,Ed=x(De,e=>e.loginMethod),Nd=x(De,Tn,(e,o)=>e.loginMethod!=""&&Boolean(o)),Ld=x(De,e=>e.walletConnectLogin),Dd=x(De,e=>e.ledgerLogin),Cd=x(De,e=>e.walletLogin),Fd=x(De,e=>e.isLoginSessionInvalid);r();r();var R=require("@elrondnetwork/erdjs");I();r();function So(e){try{return p.Buffer.from(e,"base64").toString()===atob(e)}catch(o){return!1}}function kn(e){var a,i,_;let{data:o}=e,n=So(o!=null?o:"")?R.TransactionPayload.fromEncoded(o):new R.TransactionPayload(o),t=new R.Transaction(b({value:e.value.valueOf(),data:n,nonce:e.nonce.valueOf(),receiver:new R.Address(e.receiver),sender:new R.Address(e.sender),gasLimit:(a=e.gasLimit.valueOf())!=null?a:qo,gasPrice:(i=e.gasPrice.valueOf())!=null?i:Ko,chainID:e.chainID.valueOf(),version:new R.TransactionVersion((_=e.version)!=null?_:St)},e.options?{options:new R.TransactionOptions(e.options)}:{}));return t.applySignature({hex:()=>e.signature||""},new R.Address(e.sender)),t}r();sn();r();var dp=T(require("bignumber.js"));de();r();r();r();r();var up=require("@elrondnetwork/erdjs"),gp=T(require("bignumber.js"));$();r();var Xr=require("@elrondnetwork/erdjs");function lp(e){try{let o=new Xr.Address(e);return Boolean(o.bech32())}catch(o){return!1}}function Ce(e){return(e==null?void 0:e.startsWith("erd"))&&e.length===62&&/^\w+$/.test(e)&&lp(e)}r();r();r();var hp=require("@elrondnetwork/erdjs/out");r();I();r();var je=require("@elrondnetwork/erdjs");de();var xp=["ESDTNFTTransfer","ESDTNFTBurn","ESDTNFTAddQuantity","ESDTNFTCreate","MultiESDTNFTTransfer","ESDTTransfer","ESDTBurn","ESDTLocalMint","ESDTLocalBurn","ESDTWipe","ESDTFreeze"];function ko(e,o,n=""){if(!Ce(e))return!1;if(new je.Address(e).isContractAddress())return!0;let i=Zr({receiver:e,data:n});return i?new je.Address(i).isContractAddress()||Tp(e,o,n):!1}var yp=e=>e.toLowerCase().match(/[0-9a-f]/g),bp=e=>e.length%2===0;function Tp(e,o,n){let t=n==null?void 0:n.split("@");if(t==null)return!1;let[a,...i]=t,_=o!=null&&e!=null&&e===o,l=xp.includes(a),d=i.every(u=>yp(u)&&bp(u));return _&&l&&d}function Zr({receiver:e,data:o}){try{if(!o)return e;let n=So(o)?je.TransactionPayload.fromEncoded(o).toString():o,t=wp(n),a=n.split("@");return t>-1?a[t]:e}catch(n){console.log(n);return}}function wp(e){return e.includes("MultiESDTNFTTransfer")?1:e.includes("ESDTNFTTransfer")?4:-1}r();r();r();r();$();r();r();r();r();r();r();r();var ae="accounts",ge="transactions";var Fe="tokens",Yr="blocks";var Jr="nodes",Qr="identities",ea="collections",vn="nfts",An="providers",In="sc-results",oa="contracts",na="logs",ta="code",ra="locked-accounts",aa="roles";var q={shard:e=>`/${Yr}?shard=${e}`,receiverShard:e=>`/${ge}?receivershard=${e}`,senderShard:e=>`/${ge}?sendershard=${e}`,transactionDetails:e=>`/${ge}/${e}`,transactionDetailsScResults:e=>`/${ge}/${e}/${In}`,transactionDetailsLogs:e=>`/${ge}/${e}/${na}`,nodeDetails:e=>`/${Jr}/${e}`,accountDetails:e=>`/${ae}/${e}`,accountDetailsContractCode:e=>`/${ae}/${e}/${ta}`,accountDetailsTokens:e=>`/${ae}/${e}/${Fe}`,accountDetailsNfts:e=>`/${ae}/${e}/${vn}`,accountDetailsScResults:e=>`/${ae}/${e}/${In}`,accountDetailsContracts:e=>`/${ae}/${e}/${oa}`,identityDetails:e=>`/${Qr}/${e}`,tokenDetails:e=>`/${Fe}/${e}`,tokenDetailsAccounts:e=>`/${Fe}/${e}/${ae}`,tokenDetailsLockedAccounts:e=>`/${Fe}/${e}/${ra}`,tokenDetailsRoles:e=>`/${Fe}/${e}/${aa}`,collectionDetails:e=>`/${ea}/${e}`,nftDetails:e=>`/${vn}/${e}`,providerDetails:e=>`/${An}/${e}`,providerDetailsTransactions:e=>`/${An}/${e}/${ge}`};r();r();r();r();L();r();var ia=!1;function kp(e){ia||(console.error(e),ia=!0)}function En({explorerAddress:e,to:o}){try{return new URL(o).href}catch(n){return o.startsWith("/")||(kp(`Link not prepended by / : ${o}`),o=`/${o}`),e?`${e}${o}`:o}}r();L();r();r();r();var Ap=T(require("bignumber.js"));I();r();var ca=require("@elrondnetwork/erdjs"),ie=T(require("bignumber.js"));I();r();var pa=T(require("bignumber.js")),Ge=(e,o=!0)=>{let n=String(e);if(!n.match(/^[-]?\d+$/))return!1;let t=new pa.default(n),a=o?0:-1;return t.toString(10)===n&&t.comparedTo(0)>=a};r();function J(e){return{if:function(o){return o?{then:n=>n instanceof Function?J(n(e)):J(n)}:{then:()=>J(e)}},then:o=>o instanceof Function?J(o(e)):J(o),valueOf:function(){return e}}}ie.default.config({ROUNDING_MODE:ie.default.ROUND_FLOOR});function Q({input:e,decimals:o=D,digits:n=ve,showLastNonZeroDecimal:t=!0,showIsLessThanDecimalsLabel:a=!1,addCommas:i=!1}){if(!Ge(e,!1))throw new Error("Invalid input");let _=new ie.default(e).isNegative(),l=e;return _&&(l=e.substring(1)),J(l).then(()=>ca.TokenPayment.fungibleFromBigInteger("",l,o).toRationalNumber()).then(d=>{let u=new ie.default(d);if(u.isZero())return z;let h=u.toString(10),[f,w]=h.split("."),k=new ie.default(w||0),M=J(0).if(Boolean(w&&t)).then(()=>Math.max(w.length,n)).if(k.isZero()&&!t).then(0).if(Boolean(w&&!t)).then(()=>Math.min(w.length,n)).valueOf(),C=w&&n>=1&&n<=w.length&&k.isGreaterThan(0)&&new ie.default(w.substring(0,n)).isZero(),Ba=u.toFormat(M);return J(h).if(i).then(Ba).if(Boolean(C)).then(Ze=>{let Oe=new ie.default(f).isZero(),[V,Ye]=Ze.split("."),Je=new Array(n-1).fill(0),Oa=[...Je,0].join(""),Ua=[...Je,1].join("");return Oe?a?`<${V}.${Ua}`:t?`${V}.${Ye}`:V:`${V}.${Oa}`}).if(Boolean(!C&&w)).then(Ze=>{let[Oe]=Ze.split("."),V=w.substring(0,M);if(t){let Ye=n-V.length;if(Ye>0){let Je=Array(Ye).fill(0).join("");return V=`${V}${Je}`,`${Oe}.${V}`}return Ze}return V?`${Oe}.${V}`:Oe}).valueOf()}).if(_).then(d=>`-${d}`).valueOf()}r();r();r();I();r();r();r();r();I();r();I();r();var Ip=require("@elrondnetwork/erdjs");I();r();var vo=require("@elrondnetwork/erdjs"),Np=require("@elrondnetwork/erdjs"),Lp=require("@elrondnetwork/erdjs-network-providers");I();r();r();var Ep=T(require("bignumber.js"));I();r();I();r();r();r();r();function Pe({token:e,noValue:o,showLastNonZeroDecimal:n}){var u;let t=!o&&e.value&&e.type!=="NonFungibleESDT",a=e.decimals!=null,i="";e.type==="NonFungibleESDT"&&(i="NFT"),e.type==="SemiFungibleESDT"&&(i="SFT"),e.type==="MetaESDT"&&(i="Meta-ESDT");let _="";t&&a&&(_=a?Q({input:e.value,decimals:e.decimals,digits:2,showLastNonZeroDecimal:n}):Number(e.value).toLocaleString("en"));let l=q.nftDetails(String(e.identifier)),d=e.ticker===e.collection&&(u=e.identifier)!=null?u:e.ticker;return{badgeText:i,tokenFormattedAmount:t?_:null,tokenExplorerLink:l,tokenLinkText:d,token:e,noValue:o,showLastNonZeroDecimal:n}}r();I();function Ao({token:e,noValue:o,showLastNonZeroDecimal:n}){var l;let t=Boolean(!o&&e.value),a=t?Q({input:e.value,decimals:(l=e.decimals)!=null?l:D,digits:2,showLastNonZeroDecimal:n,addCommas:!0}):null,i=q.tokenDetails(String(e.token)),_=e.ticker;return{tokenExplorerLink:i,tokenFormattedAmount:a,showFormattedAmount:t,tokenLinkText:_,token:e,showLastNonZeroDecimal:n}}r();L();var Dp=["reDelegateRewards","claimRewards","unBond"],Cp=["wrapEgld","unwrapEgld"],Fp=["unStake"],Pp=["unDelegate"];r();r();r();I();r();var Mp=T(require("bignumber.js"));r();r();r();L();r();var Rp=T(require("bignumber.js"));r();r();r();L();r();var Op=T(require("bignumber.js"));r();$();r();$();r();r();var Up=require("react");L();var Bf=["create","localMint","ESDTLocalMint","addQuantity","burn","localBurn","ESDTLocalBurn","wipe","writeLog","signalError"];r();var Vp=T(require("anchorme"));r();var sa=require("react");de();r();r();L();r();var $p=require("@elrondnetwork/erdjs/out"),zp=T(require("bignumber.js"));L();r();L();r();r();$();var uh=[{label:"Raw",value:"raw"},{label:"Text",value:"text"},{label:"Decimal",value:"decimal"},{label:"Smart",value:"smart"}];r();var da=require("react");de();r();r();r();r();r();I();var Eo=e=>e.transactions,qe=x(Eo,e=>e.signedTransactions),ix=x(Eo,e=>e.signTransactionsError),px=x(Eo,e=>e.signTransactionsCancelMessage),No=e=>o=>Object.entries(o).reduce((n,[t,a])=>(e(a.status)&&(n[t]=a),n),{}),cx=x(qe,No(rr)),mx=x(qe,No(so)),_x=x(qe,No(lo)),sx=x(qe,No(ar)),dx=x(Eo,e=>{var o;return(e==null?void 0:e.transactionsToSign)==null?null:S(b({},e.transactionsToSign),{transactions:((o=e==null?void 0:e.transactionsToSign)==null?void 0:o.transactions.map(n=>kn(n)))||[]})}),lx=x(qe,(e,o)=>o,(e,o)=>o!=null?(e==null?void 0:e[o])||{}:{});r();Ie();var la={errorMessage:ln,successMessage:un,processingMessage:gn},Hp=e=>e.transactionsInfo,hx=x(Hp,(e,o)=>o,(e,o)=>o!=null&&(e==null?void 0:e[Number(o)])||la);r();var ua=e=>e.modals,bx=x(ua,e=>e.txSubmittedModal),Tx=x(ua,e=>e.notificationModal);r();var Ln=e=>e.toasts,kx=x(Ln,e=>e.customToasts),vx=x(Ln,e=>e.transactionToasts),Ax=x(Ln,e=>e.failTransactionToast);var ga=()=>Hr(re);r();var jp=`.dapp-core-component__main__link-style {
  color: #1392ff;
}

.dapp-core-component__main__link-second-style {
  color: #9dabbd;
}
.dapp-core-component__main__link-second-style:hover {
  color: #1392ff;
}
.dapp-core-component__main__link-second-style:hover svg {
  color: #1392ff;
}

.dapp-core-component__main__wrapper {
  min-height: 100vh;
}

.dapp-core-component__main__opacity-6 {
  opacity: 0.6;
}

.dapp-core-component__main__qr-code-svg-container {
  width: 15rem;
  height: 15rem;
}

.dapp-core-component__main__pair-list {
  max-height: 20rem;
  overflow-y: auto;
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__pair-image {
  width: 4rem;
  overflow: hidden;
  height: auto;
  min-width: 2rem;
}

.dapp-core-component__main__pair-remove {
  padding: 0 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

.dapp-core-component__main__dapp-icon {
  padding: 5px;
  border-radius: 50%;
  background-color: #fafafa;
  width: 4.6428571429rem;
  height: 4.6428571429rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}
.dapp-core-component__main__dapp-icon.dapp-core-component__main__icon-medium {
  width: 80px;
  height: 80px;
}

.dapp-core-component__main__centering {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dapp-core-component__main__border-n {
  border: none !important;
}

.dapp-core-component__main__t-shadow {
  text-shadow: 0px 4px 7px rgba(0, 0, 0, 0.3);
}

.dapp-core-component__main__btn {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.04) !important;
}

.dapp-core-component__main__media {
  display: flex;
  align-items: flex-start;
}

.dapp-core-component__main__media-body {
  flex: 1;
}

.dapp-core-component__main__connect-btns .dapp-core-component__main__btn {
  display: flex;
  height: 48px;
  justify-content: center;
  align-items: center;
  color: #fff !important;
  background-color: #1392ff;
  padding: 15px 20px 15px 16px;
  border-radius: 6px;
  gap: 8px;
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__btn:hover, .dapp-core-component__main__connect-btns .dapp-core-component__main__btn:active {
  background-color: #1392ff;
  background-image: linear-gradient(rgba(0, 0, 0, 0.1) 0 0);
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__info {
  color: #1392ff;
  margin-left: 48px;
}

.dapp-core-component__main__btn {
  transition: none !important;
}

.dapp-core-component__main__trust-badge {
  width: 20px;
  height: 20px;
}

/* Navbar
  -------------------------------------------------- */
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name {
  display: flex;
  align-items: center;
  font-size: 1.7142857143rem;
  font-weight: 400;
  color: #9ba5b4;
  margin-left: 1.6428571429rem;
}
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name:before {
  content: "";
  display: block;
  width: 1px;
  height: 3rem;
  margin-right: 1.7142857143rem;
}

/* Dashboard
  -------------------------------------------------- */
.dapp-core-component__main__action-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  margin: 0 7px;
}
.dapp-core-component__main__action-btn button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #fff;
}
.dapp-core-component__main__action-btn button:hover {
  opacity: 0.9;
}

.dapp-core-component__main__transactions .dapp-core-component__main__transaction-icon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table td {
  white-space: nowrap;
  vertical-align: middle;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table tr:last-of-type td {
  border-bottom: 1px solid #dee2e6;
}

/* Loader, Transaction Success/Fail
  -------------------------------------------------- */
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div {
  position: absolute;
  top: 27px;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.1);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.dapp-core-component__main__page-state .dapp-core-component__main__bg-blue .dapp-core-component__main__lds-ellipsis div {
  background: rgba(255, 255, 255, 0.4);
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(1) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis1 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(2) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(3) {
  left: 26px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(4) {
  left: 45px;
  animation: dapp-core-component__main__lds-ellipsis3 0.6s infinite;
}
@keyframes dapp-core-component__main__lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(19px, 0);
  }
}

.dapp-core-component__main__btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
}
.dapp-core-component__main__btn .dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn .dapp-core-component__main__disabled {
  opacity: 0.65;
  cursor: default;
}
.dapp-core-component__main__btn:hover {
  color: #212529;
  text-decoration: none;
}
.dapp-core-component__main__btn:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn:disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn-primary {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary:hover {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-primary:focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary:hover {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-secondary:focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-warning {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning:hover {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning:focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-dark {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark:hover {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
}
.dapp-core-component__main__btn-dark:focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-link {
  font-weight: 400;
  color: #007bff;
  text-decoration: none;
}
.dapp-core-component__main__btn-link:hover {
  color: #0056b3;
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:focus, .dapp-core-component__main__btn-link.dapp-core-component__main__focus {
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:disabled, .dapp-core-component__main__btn-link.dapp-core-component__main__disabled {
  color: #6c757d;
  pointer-events: none;
}

.dapp-core-component__main__btn-light {
  color: #212529;
  background-color: #fafafa;
  border-color: #fafafa;
}
.dapp-core-component__main__btn-light .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}
.dapp-core-component__main__btn-light .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #f8f9fa;
  border-color: #f8f9fa;
}
.dapp-core-component__main__btn-light:hover {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
}
.dapp-core-component__main__btn-light:focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .dapp-core-component__main__badge {
    transition: none;
  }
}
a.dapp-core-component__main__badge:hover, a.dapp-core-component__main__badge:focus {
  text-decoration: none;
}

.dapp-core-component__main__badge:empty {
  display: none;
}

.dapp-core-component__main__btn .dapp-core-component__main__badge {
  position: relative;
  top: -1px;
}

.dapp-core-component__main__badge-pill {
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.dapp-core-component__main__badge-primary {
  color: #fff;
  background-color: #007bff;
}

a.dapp-core-component__main__badge-primary:hover, a.dapp-core-component__main__badge-primary:focus {
  color: #fff;
  background-color: #0062cc;
}

a.dapp-core-component__main__badge-primary:focus, a.dapp-core-component__main__badge-primary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.5);
}

.dapp-core-component__main__badge-secondary {
  color: #fff;
  background-color: #6c757d;
}

a.dapp-core-component__main__badge-secondary:hover, a.dapp-core-component__main__badge-secondary:focus {
  color: #fff;
  background-color: #545b62;
}

a.dapp-core-component__main__badge-secondary:focus, a.dapp-core-component__main__badge-secondary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108, 117, 125, 0.5);
}

.dapp-core-component__main__badge-success {
  color: #fff;
  background-color: #28a745;
}

a.dapp-core-component__main__badge-success:hover, a.dapp-core-component__main__badge-success:focus {
  color: #fff;
  background-color: #1e7e34;
}

a.dapp-core-component__main__badge-success:focus, a.dapp-core-component__main__badge-success.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
}

.dapp-core-component__main__badge-info {
  color: #fff;
  background-color: #17a2b8;
}

a.dapp-core-component__main__badge-info:hover, a.dapp-core-component__main__badge-info:focus {
  color: #fff;
  background-color: #117a8b;
}

a.dapp-core-component__main__badge-info:focus, a.dapp-core-component__main__badge-info.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(23, 162, 184, 0.5);
}

.dapp-core-component__main__badge-warning {
  color: #212529;
  background-color: #ffc107;
}

a.dapp-core-component__main__badge-warning:hover, a.dapp-core-component__main__badge-warning:focus {
  color: #212529;
  background-color: #d39e00;
}

a.dapp-core-component__main__badge-warning:focus, a.dapp-core-component__main__badge-warning.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 193, 7, 0.5);
}

.dapp-core-component__main__badge-danger {
  color: #fff;
  background-color: #dc3545;
}

a.dapp-core-component__main__badge-danger:hover, a.dapp-core-component__main__badge-danger:focus {
  color: #fff;
  background-color: #bd2130;
}

a.dapp-core-component__main__badge-danger:focus, a.dapp-core-component__main__badge-danger.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(220, 53, 69, 0.5);
}

.dapp-core-component__main__badge-light {
  color: #212529;
  background-color: #f8f9fa;
}

a.dapp-core-component__main__badge-light:hover, a.dapp-core-component__main__badge-light:focus {
  color: #212529;
  background-color: #dae0e5;
}

a.dapp-core-component__main__badge-light:focus, a.dapp-core-component__main__badge-light.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(248, 249, 250, 0.5);
}

.dapp-core-component__main__badge-dark {
  color: #fff;
  background-color: #343a40;
}

a.dapp-core-component__main__badge-dark:hover, a.dapp-core-component__main__badge-dark:focus {
  color: #fff;
  background-color: #1d2124;
}

a.dapp-core-component__main__badge-dark:focus, a.dapp-core-component__main__badge-dark.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}

.dapp-core-component__main__card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 5px;
}
.dapp-core-component__main__card > hr {
  margin-right: 0;
  margin-left: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group {
  border-top: inherit;
  border-bottom: inherit;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:first-child {
  border-top-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:last-child {
  border-bottom-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__card-header + .dapp-core-component__main__list-group,
.dapp-core-component__main__card > .dapp-core-component__main__list-group + .dapp-core-component__main__card-footer {
  border-top: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header {
  border: 0;
  padding: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header:first-child {
  border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item {
  border-bottom: 1px solid #dee2e6;
  padding: 1.1rem 1rem;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item h6,
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item .dapp-core-component__main__h6 {
  line-height: 2.5rem;
  letter-spacing: unset;
  margin: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item.dapp-core-component__main__compact {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fafafa;
}

.dapp-core-component__main__card-body {
  flex: 1 1 auto;
  min-height: 1px;
  padding: 1.25rem;
}

.dapp-core-component__main__card-title {
  margin-bottom: 0.75rem;
}

.dapp-core-component__main__card-subtitle {
  margin-top: -0.375rem;
  margin-bottom: 0;
}

.dapp-core-component__main__card-text:last-child {
  margin-bottom: 0;
}

.dapp-core-component__main__card-link:hover {
  text-decoration: none;
}
.dapp-core-component__main__card-link + .dapp-core-component__main__card-link {
  margin-left: 1.25rem;
}

.dapp-core-component__main__card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: #fff;
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__card-footer {
  padding: 0.75rem 1.25rem;
  background-color: #fff;
  border-top: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__text {
  color: #9dabbd;
}

.dapp-core-component__main__text-white {
  color: #fff;
}

.dapp-core-component__main__text-monospace {
  font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-size-xl .dapp-core-component__main__trim {
  max-width: 13rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim {
  max-width: 10rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-only-sm .dapp-core-component__main__trim {
  max-width: none;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-left {
  text-align: left;
}

.dapp-core-component__main__text-right {
  text-align: right;
}

.dapp-core-component__main__text-center {
  text-align: center;
}

.dapp-core-component__main__text-primary {
  color: #1b46c2;
}

.dapp-core-component__main__text-secondary {
  color: #6c757d;
}

.dapp-core-component__main__text-primary-highlight {
  color: #1b46c2;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-success {
  color: #28a745;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-body {
  color: #212529;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-black-50 {
  color: rgba(0, 0, 0, 0.5);
}

.dapp-core-component__main__text-white-50 {
  color: rgba(255, 255, 255, 0.5);
}

@media (min-width: 992px) {
  .dapp-core-component__main__text-lg-right {
    text-align: right;
  }
}

.dapp-core-component__main__text-hide {
  font: 0/0 a;
  color: rgba(0, 0, 0, 0);
  text-shadow: none;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
}

.dapp-core-component__main__text-decoration-none {
  text-decoration: none;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__text-reset {
  color: inherit;
}

.dapp-core-component__main__text-lowercase {
  text-transform: lowercase;
}

.dapp-core-component__main__text-uppercase {
  text-transform: uppercase;
}

.dapp-core-component__main__text-capitalize {
  text-transform: capitalize;
}

.dapp-core-component__main__text-danger {
  color: #d9534f;
}

.dapp-core-component__main__text-warning {
  color: #f0ad4e;
}

.dapp-core-component__main__trim-fs-sm {
  color: inherit;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__m-0 {
  margin: 0;
}

.dapp-core-component__main__mt-0 {
  margin-top: 0;
}

.dapp-core-component__main__my-0 {
  margin-top: 0;
  margin-bottom: 0;
}

.dapp-core-component__main__mr-0 {
  margin-right: 0;
}

.dapp-core-component__main__mx-0 {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__mb-0 {
  margin-bottom: 0;
}

.dapp-core-component__main__my-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.dapp-core-component__main__ml-0 {
  margin-left: 0;
}

.dapp-core-component__main__mx-0 {
  margin-left: 0;
  margin-right: 0;
}

.dapp-core-component__main__m-1 {
  margin: 0.25rem;
}

.dapp-core-component__main__mt-1 {
  margin-top: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__mr-1 {
  margin-right: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.dapp-core-component__main__mb-1 {
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-bottom: 0.25rem;
  margin-top: 0.25rem;
}

.dapp-core-component__main__ml-1 {
  margin-left: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-right: 0.25rem;
  margin-left: 0.25rem;
}

.dapp-core-component__main__m-2 {
  margin: 0.5rem;
}

.dapp-core-component__main__mt-2 {
  margin-top: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-bottom: 0.5rem;
  margin-top: 0.5rem;
}

.dapp-core-component__main__mr-2 {
  margin-right: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__mb-2 {
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__ml-2 {
  margin-left: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__m-3 {
  margin: 1rem;
}

.dapp-core-component__main__mt-3,
.dapp-core-component__main__my-3 {
  margin-top: 1rem;
}

.dapp-core-component__main__mr-3 {
  margin-right: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__mb-3 {
  margin-bottom: 1rem;
}

.dapp-core-component__main__my-3 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.dapp-core-component__main__ml-3 {
  margin-left: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__m-4 {
  margin: 1.5rem;
}

.dapp-core-component__main__mt-4 {
  margin-top: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__mr-4 {
  margin-right: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-right: 1.5rem;
  margin-left: 1.5rem;
}

.dapp-core-component__main__mb-4 {
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.dapp-core-component__main__ml-4 {
  margin-left: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}

.dapp-core-component__main__m-5 {
  margin: 3rem;
}

.dapp-core-component__main__mt-5 {
  margin-top: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__mr-5 {
  margin-right: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-right: 3rem;
  margin-left: 3rem;
}

.dapp-core-component__main__mb-5 {
  margin-bottom: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__ml-5 {
  margin-left: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-left: 3rem;
  margin-right: 3rem;
}

.dapp-core-component__main__p-0 {
  padding: 0;
}

.dapp-core-component__main__pt-0 {
  padding-top: 0;
}

.dapp-core-component__main__py-0 {
  padding-top: 0;
  padding-bottom: 0;
}

.dapp-core-component__main__pr-0 {
  padding-right: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__pb-0 {
  padding-bottom: 0;
}

.dapp-core-component__main__py-0 {
  padding-bottom: 0;
  padding-top: 0;
}

.dapp-core-component__main__pl-0 {
  padding-left: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__p-1 {
  padding: 0.25rem;
}

.dapp-core-component__main__pt-1 {
  padding-top: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pr-1 {
  padding-right: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-right: 0.25rem;
  padding-left: 0.25rem;
}

.dapp-core-component__main__pb-1 {
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pl-1 {
  padding-left: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

.dapp-core-component__main__p-2 {
  padding: 0.5rem;
}

.dapp-core-component__main__pt-2 {
  padding-top: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.dapp-core-component__main__pr-2 {
  padding-right: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__pb-2 {
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__pl-2 {
  padding-left: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__p-3 {
  padding: 1rem;
}

.dapp-core-component__main__pt-3 {
  padding-top: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.dapp-core-component__main__pr-3 {
  padding-right: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__pb-3 {
  padding-bottom: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.dapp-core-component__main__pl-3 {
  padding-left: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__p-4 {
  padding: 1.5rem;
}

.dapp-core-component__main__pt-4 {
  padding-top: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pr-4 {
  padding-right: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.dapp-core-component__main__pb-4 {
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pl-4 {
  padding-left: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
}

.dapp-core-component__main__p-5 {
  padding: 3rem;
}

.dapp-core-component__main__pt-5 {
  padding-top: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-bottom: 3rem;
  padding-top: 3rem;
}

.dapp-core-component__main__pr-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__pb-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.dapp-core-component__main__pl-5 {
  padding-left: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__m-n1 {
  margin: -0.25rem;
}

.dapp-core-component__main__mt-n1 {
  margin-top: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__mr-n1 {
  margin-right: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__mb-n1 {
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-bottom: -0.25rem;
  margin-top: -0.25rem;
}

.dapp-core-component__main__ml-n1 {
  margin-left: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__m-n2 {
  margin: -0.5rem;
}

.dapp-core-component__main__mt-n2 {
  margin-top: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-top: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mr-n2 {
  margin-right: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mb-n2 {
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-bottom: -0.5rem;
  margin-top: -0.5rem;
}

.dapp-core-component__main__ml-n2 {
  margin-left: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.dapp-core-component__main__m-n3 {
  margin: -1rem;
}

.dapp-core-component__main__mt-n3 {
  margin-top: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-top: -1rem;
  margin-bottom: -1rem;
}

.dapp-core-component__main__mr-n3 {
  margin-right: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-right: -1rem;
  margin-left: -1rem;
}

.dapp-core-component__main__mb-n3 {
  margin-bottom: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-bottom: -1rem;
  margin-top: -1rem;
}

.dapp-core-component__main__ml-n3 {
  margin-left: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-left: -1rem;
  margin-right: -1rem;
}

.dapp-core-component__main__m-n4 {
  margin: -1.5rem;
}

.dapp-core-component__main__mt-n4 {
  margin-top: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-top: -1.5rem;
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__mr-n4 {
  margin-right: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__mb-n4 {
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-bottom: -1.5rem;
  margin-top: -1.5rem;
}

.dapp-core-component__main__ml-n4 {
  margin-left: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__m-n5 {
  margin: -3rem;
}

.dapp-core-component__main__mt-n5 {
  margin-top: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-bottom: -3rem;
  margin-top: -3rem;
}

.dapp-core-component__main__mr-n5 {
  margin-right: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-left: -3rem;
  margin-right: -3rem;
}

.dapp-core-component__main__mb-n5 {
  margin-bottom: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-top: -3rem;
  margin-bottom: -3rem;
}

.dapp-core-component__main__ml-n5 {
  margin-left: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-right: -3rem;
  margin-left: -3rem;
}

.dapp-core-component__main__m-auto {
  margin: auto;
}

.dapp-core-component__main__mt-auto {
  margin-top: auto;
}

.dapp-core-component__main__my-auto {
  margin-bottom: auto;
  margin-top: auto;
}

.dapp-core-component__main__mr-auto {
  margin-right: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__mb-auto {
  margin-bottom: auto;
}

.dapp-core-component__main__my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.dapp-core-component__main__ml-auto {
  margin-left: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-1 {
    margin-right: 0.25rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-4 {
    margin-right: 1.5rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mx-lg-4 {
    margin-left: 1.5rem;
    margin-right: 1.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__mb-xl-2 {
    margin-bottom: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__pr-sm-2 {
    padding-right: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__px-sm-2 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__pr-xl-0,
.dapp-core-component__main__px-xl-0 {
    padding-right: 0;
  }
}

.dapp-core-component__main__col {
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

.dapp-core-component__main__col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__col-xl,
.dapp-core-component__main__col-xl-auto,
.dapp-core-component__main__col-xl-12,
.dapp-core-component__main__col-xl-11,
.dapp-core-component__main__col-xl-10,
.dapp-core-component__main__col-xl-9,
.dapp-core-component__main__col-xl-8,
.dapp-core-component__main__col-xl-7,
.dapp-core-component__main__col-xl-6,
.dapp-core-component__main__col-xl-5,
.dapp-core-component__main__col-xl-4,
.dapp-core-component__main__col-xl-3,
.dapp-core-component__main__col-xl-2,
.dapp-core-component__main__col-xl-1,
.dapp-core-component__main__col-lg,
.dapp-core-component__main__col-lg-auto,
.dapp-core-component__main__col-lg-12,
.dapp-core-component__main__col-lg-11,
.dapp-core-component__main__col-lg-10,
.dapp-core-component__main__col-lg-9,
.dapp-core-component__main__col-lg-8,
.dapp-core-component__main__col-lg-7,
.dapp-core-component__main__col-lg-6,
.dapp-core-component__main__col-lg-5,
.dapp-core-component__main__col-lg-4,
.dapp-core-component__main__col-lg-3,
.dapp-core-component__main__col-lg-2,
.dapp-core-component__main__col-lg-1,
.dapp-core-component__main__col-md,
.dapp-core-component__main__col-md-auto,
.dapp-core-component__main__col-md-12,
.dapp-core-component__main__col-md-11,
.dapp-core-component__main__col-md-10,
.dapp-core-component__main__col-md-9,
.dapp-core-component__main__col-md-8,
.dapp-core-component__main__col-md-7,
.dapp-core-component__main__col-md-6,
.dapp-core-component__main__col-md-5,
.dapp-core-component__main__col-md-4,
.dapp-core-component__main__col-md-3,
.dapp-core-component__main__col-md-2,
.dapp-core-component__main__col-md-1,
.dapp-core-component__main__col-sm,
.dapp-core-component__main__col-sm-auto,
.dapp-core-component__main__col-sm-12,
.dapp-core-component__main__col-sm-11,
.dapp-core-component__main__col-sm-10,
.dapp-core-component__main__col-sm-9,
.dapp-core-component__main__col-sm-8,
.dapp-core-component__main__col-sm-7,
.dapp-core-component__main__col-sm-6,
.dapp-core-component__main__col-sm-5,
.dapp-core-component__main__col-sm-4,
.dapp-core-component__main__col-sm-3,
.dapp-core-component__main__col-sm-2,
.dapp-core-component__main__col-sm-1,
.dapp-core-component__main__col,
.dapp-core-component__main__col-auto,
.dapp-core-component__main__col-12,
.dapp-core-component__main__col-11,
.dapp-core-component__main__col-10,
.dapp-core-component__main__col-9,
.dapp-core-component__main__col-8,
.dapp-core-component__main__col-7,
.dapp-core-component__main__col-6,
.dapp-core-component__main__col-5,
.dapp-core-component__main__col-4,
.dapp-core-component__main__col-3,
.dapp-core-component__main__col-2,
.dapp-core-component__main__col-1 {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
}

.dapp-core-component__main__col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

@media (min-width: 576px) {
  .dapp-core-component__main__col-sm-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-sm-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-sm-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-sm-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-sm-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-sm-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-sm-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-sm-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-sm-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-sm-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-sm-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-sm-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 768px) {
  .dapp-core-component__main__col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__col-xl-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-xl-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-xl-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-xl-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-xl-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-xl-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-xl-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-xl-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-xl-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-xl-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-xl-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-xl-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
.dapp-core-component__main__flex-row {
  flex-direction: row;
}

.dapp-core-component__main__flex-column {
  flex-direction: column;
}

.dapp-core-component__main__flex-row-reverse {
  flex-direction: row-reverse;
}

.dapp-core-component__main__flex-column-reverse {
  flex-direction: column-reverse;
}

.dapp-core-component__main__flex-wrap {
  flex-wrap: wrap;
}

.dapp-core-component__main__flex-nowrap {
  flex-wrap: nowrap;
}

.dapp-core-component__main__flex-wrap-reverse {
  flex-wrap: wrap-reverse;
}

.dapp-core-component__main__flex-fill {
  flex: 1 1 auto;
}

.dapp-core-component__main__flex-grow-0 {
  flex-grow: 0;
}

.dapp-core-component__main__flex-grow-1 {
  flex-grow: 1;
}

.dapp-core-component__main__flex-shrink-0 {
  flex-shrink: 0;
}

.dapp-core-component__main__flex-shrink-1 {
  flex-shrink: 1;
}

.dapp-core-component__main__justify-content-start {
  justify-content: flex-start;
}

.dapp-core-component__main__justify-content-end {
  justify-content: flex-end;
}

.dapp-core-component__main__justify-content-center {
  justify-content: center;
}

.dapp-core-component__main__justify-content-between {
  justify-content: space-between;
}

.dapp-core-component__main__justify-content-around {
  justify-content: space-around;
}

.dapp-core-component__main__align-items-start {
  align-items: flex-start;
}

.dapp-core-component__main__align-items-end {
  align-items: flex-end;
}

.dapp-core-component__main__align-items-center {
  align-items: center;
}

.dapp-core-component__main__align-items-baseline {
  align-items: baseline;
}

.dapp-core-component__main__align-items-stretch {
  align-items: stretch;
}

.dapp-core-component__main__align-content-start {
  align-content: flex-start;
}

.dapp-core-component__main__align-content-end {
  align-content: flex-end;
}

.dapp-core-component__main__align-content-center {
  align-content: center;
}

.dapp-core-component__main__align-content-between {
  align-content: space-between;
}

.dapp-core-component__main__align-content-around {
  align-content: space-around;
}

.dapp-core-component__main__align-content-stretch {
  align-content: stretch;
}

.dapp-core-component__main__align-self-auto {
  align-self: auto;
}

.dapp-core-component__main__align-self-start {
  align-self: flex-start;
}

.dapp-core-component__main__align-self-end {
  align-self: flex-end;
}

.dapp-core-component__main__align-self-center {
  align-self: center;
}

.dapp-core-component__main__align-self-baseline {
  align-self: baseline;
}

.dapp-core-component__main__align-self-stretch {
  align-self: stretch;
}

@media (min-width: 576px) {
  .dapp-core-component__main__flex-sm-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-wrap {
    flex-wrap: wrap;
  }
}

.dapp-core-component__main__form-check {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: auto;
  padding-left: 0;
}

.dapp-core-component__main__form-check-label {
  margin-bottom: 0;
}

.dapp-core-component__main__form-check-input {
  position: absolute;
  margin-top: 0.3rem;
  margin-left: -1.25rem;
}
.dapp-core-component__main__form-check-input[disabled] ~ .dapp-core-component__main__form-check-label, .dapp-core-component__main__form-check-input:disabled ~ .dapp-core-component__main__form-check-label {
  color: #6c757d;
}

.dapp-core-component__main__form-group {
  display: flex;
  flex: 0 0 auto;
  flex-flow: row wrap;
  align-items: center;
  margin-bottom: 0;
}

.dapp-core-component__main__form-control {
  display: block;
  width: 100%;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.dapp-core-component__main__form-label {
  padding-top: calc(0.375rem + 1px);
  padding-bottom: calc(0.375rem + 1px);
  margin-bottom: 0;
  font-size: inherit;
  line-height: 1.5;
}

input,
button,
select,
optgroup,
textarea {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.dapp-core-component__main__form-group {
  margin-bottom: 1rem;
}

.dapp-core-component__main__form-text {
  display: block;
  margin-top: 0.25rem;
  color: inherit;
}

.dapp-core-component__main__border {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__border-top {
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__border-right {
  border-right: 1px solid #dee2e6;
}

.dapp-core-component__main__border-bottom {
  border-bottom: 1px solid #dee2e6;
}

.dapp-core-component__main__border-left {
  border-left: 1px solid #dee2e6;
}

.dapp-core-component__main__border-0 {
  border: 0;
}

.dapp-core-component__main__border-top-0 {
  border-top: 0;
}

.dapp-core-component__main__border-right-0 {
  border-right: 0;
}

.dapp-core-component__main__border-bottom-0 {
  border-bottom: 0;
}

.dapp-core-component__main__border-left-0 {
  border-left: 0;
}

.dapp-core-component__main__rounded-sm {
  border-radius: 0.2rem;
}

.dapp-core-component__main__rounded {
  border-radius: 0.25rem;
}

.dapp-core-component__main__rounded-top {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-right {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-bottom {
  border-bottom-right-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-left {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-lg {
  border-radius: 0.3rem;
}

.dapp-core-component__main__rounded-circle {
  border-radius: 50%;
}

.dapp-core-component__main__rounded-pill {
  border-radius: 50rem;
}

.dapp-core-component__main__rounded-0 {
  border-radius: 0;
}

.dapp-core-component__main__h1,
.dapp-core-component__main__h2,
.dapp-core-component__main__h3,
.dapp-core-component__main__h4,
.dapp-core-component__main__h5,
.dapp-core-component__main__h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2;
}

.dapp-core-component__main__h1 {
  font-size: 2.5rem;
}

.dapp-core-component__main__h2 {
  font-size: 2rem;
}

.dapp-core-component__main__h3 {
  font-size: 1.75rem;
}

.dapp-core-component__main__h4 {
  font-size: 1.5rem;
}

.dapp-core-component__main__h5 {
  font-size: 1.25rem;
}

.dapp-core-component__main__h6 {
  font-size: 1rem;
}

.dapp-core-component__main__lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

.dapp-core-component__main__bg-warning,
.dapp-core-component__main__warning {
  background-color: #f0ad4e;
}

.dapp-core-component__main__bg-danger,
.dapp-core-component__main__danger {
  background-color: #d9534f;
}

.dapp-core-component__main__bg-success,
.dapp-core-component__main__success {
  background-color: #5cb85c;
}

.dapp-core-component__main__position-relative {
  position: relative;
}

.dapp-core-component__main__position-absolute {
  position: absolute;
}

.dapp-core-component__main__w-25 {
  width: 25%;
}

.dapp-core-component__main__w-50 {
  width: 50%;
}

.dapp-core-component__main__w-75 {
  width: 75%;
}

.dapp-core-component__main__w-100 {
  width: 100%;
}

.dapp-core-component__main__w-auto {
  width: auto;
}

.dapp-core-component__main__h-25 {
  height: 25%;
}

.dapp-core-component__main__h-50 {
  height: 50%;
}

.dapp-core-component__main__h-75 {
  height: 75%;
}

.dapp-core-component__main__h-100 {
  height: 100%;
}

.dapp-core-component__main__h-auto {
  height: auto;
}

.dapp-core-component__main__mw-100 {
  max-width: 100%;
}

.dapp-core-component__main__mh-100 {
  max-height: 100%;
}

.dapp-core-component__main__min-vw-100 {
  min-width: 100vw;
}

.dapp-core-component__main__min-vh-100 {
  min-height: 100vh;
}

.dapp-core-component__main__vw-100 {
  width: 100vw;
}

.dapp-core-component__main__vh-100 {
  height: 100vh;
}

.dapp-core-component__main__locked-icon {
  color: inherit;
}

.dapp-core-component__main__shadow-sm {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
}

.dapp-core-component__main__shadow {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.dapp-core-component__main__shadow-lg {
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.175);
}

.dapp-core-component__main__shadow-none {
  box-shadow: none;
}

.dapp-core-component__main__d-none {
  display: none;
}

.dapp-core-component__main__d-inline {
  display: inline;
}

.dapp-core-component__main__d-inline-block {
  display: inline-block;
}

.dapp-core-component__main__d-block {
  display: block;
}

.dapp-core-component__main__d-table {
  display: table;
}

.dapp-core-component__main__d-table-row {
  display: table-row;
}

.dapp-core-component__main__d-table-cell {
  display: table-cell;
}

.dapp-core-component__main__d-flex {
  display: -ms-flexbox;
  display: flex;
}

.dapp-core-component__main__d-inline-flex {
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media (min-width: 576px) {
  .dapp-core-component__main__d-sm-flex {
    display: flex;
  }
}

.dapp-core-component__main__table {
  width: 100%;
  margin-bottom: 1rem;
  color: #212529;
}
.dapp-core-component__main__table th,
.dapp-core-component__main__table td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #dee2e6;
}
.dapp-core-component__main__table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #dee2e6;
}
.dapp-core-component__main__table tbody + tbody {
  border-top: 2px solid #dee2e6;
}

.dapp-core-component__main__table-sm th,
.dapp-core-component__main__table-sm td {
  padding: 0.3rem;
}

.dapp-core-component__main__table-bordered {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered th,
.dapp-core-component__main__table-bordered td {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered thead th,
.dapp-core-component__main__table-bordered thead td {
  border-bottom-width: 2px;
}

.dapp-core-component__main__table-borderless th,
.dapp-core-component__main__table-borderless td,
.dapp-core-component__main__table-borderless thead th,
.dapp-core-component__main__table-borderless tbody + tbody {
  border: 0;
}

.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(0, 0, 0, 0.05);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-dark th {
  color: #fff;
  background-color: #343a40;
  border-color: #454d55;
}
.dapp-core-component__main__table .dapp-core-component__main__thead-light th {
  color: #495057;
  background-color: #e9ecef;
  border-color: #dee2e6;
}

.dapp-core-component__main__table-dark {
  color: #fff;
  background-color: #343a40;
}
.dapp-core-component__main__table-dark th,
.dapp-core-component__main__table-dark td,
.dapp-core-component__main__table-dark thead th {
  border-color: #454d55;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-bordered {
  border: 0;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(255, 255, 255, 0.05);
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-hover tbody tr {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.075);
}

.dapp-core-component__main__table-responsive {
  display: block;
  width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}
.dapp-core-component__main__table-responsive > .dapp-core-component__main__table-bordered {
  border: 0;
}

@media (max-width: 767.98px) {
  .dapp-core-component__main__transactions.dapp-core-component__main__table .dapp-core-component__main__transaction-hash .dapp-core-component__main__trim {
    max-width: 8rem;
  }
}
.dapp-core-component__main__font-weight-light {
  font-weight: 300 !important;
}

.dapp-core-component__main__font-weight-lighter {
  font-weight: lighter !important;
}

.dapp-core-component__main__font-weight-normal {
  font-weight: 400 !important;
}

.dapp-core-component__main__font-weight-medium {
  font-weight: 500 !important;
}

.dapp-core-component__main__font-weight-bold {
  font-weight: 700 !important;
}

.dapp-core-component__main__font-weight-bolder {
  font-weight: bolder !important;
}

.dapp-core-component__main__font-italic {
  font-style: italic !important;
}

.dapp-core-component__main__icon-state {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
}
.dapp-core-component__main__icon-state.dapp-core-component__main__half {
  width: 3rem;
  height: 3rem;
}

.dapp-core-component__main__side-icon {
  width: 1.5rem;
  height: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  color: transparent;
  border-radius: 50%;
  overflow: hidden;
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__side-icon svg {
  color: #cacbcc;
}

div.dapp-core-component__main__side-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
div.dapp-core-component__main__side-icon img {
  transform: scale(0.75);
}

.dapp-core-component__main__toast-messages {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast {
  display: block;
  flex-basis: 4rem;
  width: 100%;
  max-width: 28.875rem;
  transition: transform 0.3s ease-in-out;
  transform: translateX(120%);
  background-color: #fff;
  border-radius: 0.25rem;
  border-color: #dee2e6;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__clickable {
  cursor: pointer;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible {
  margin-bottom: 0.3rem;
  transform: translateX(0);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress {
  background-color: rgba(0, 0, 0, 0);
  height: inherit;
  font-size: inherit;
  line-height: inherit;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar {
  background-color: rgba(108, 117, 125, 0.2);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar .dapp-core-component__main__content-height {
  z-index: -1;
  visibility: hidden;
  width: 0 !important;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast .dapp-core-component__main__tx-description {
  margin-bottom: 0.2rem;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close {
  opacity: 1;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close svg {
  color: #6c757d;
}

.dapp-core-component__main__data-field .dapp-core-component__main__textarea {
  -moz-appearance: textfield-multiline;
  -webkit-appearance: textarea;
  border: 1px solid #000;
  background-color: #fff;
  border-radius: 5px;
  font: medium -moz-fixed;
  font-size: 0.8125rem;
  min-height: 5rem;
  overflow: auto;
  padding: 0.375rem 0.75rem;
  resize: both;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea:hover {
  border: 1px solid #000;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea .dapp-core-component__main__highlighted {
  font-size: 1rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea.dapp-core-component__main__sc-call {
  min-height: 3rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__form-control:disabled,
.dapp-core-component__main__data-field .dapp-core-component__main__form-control[readonly] {
  background-color: #fafafa;
}
.dapp-core-component__main__data-field .dapp-core-component__main__small-font {
  font-size: 0.8125rem;
}

.dapp-core-component__main__progress-steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps hr {
  background-color: #6c757d;
  border-color: #dee2e6;
  z-index: 2;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot {
  height: 0.625rem;
  width: 0.625rem;
  border-radius: 50%;
  margin: 0 1rem;
  z-index: 5;
  background-color: #6c757d;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:first-of-type {
  margin-left: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:last-of-type {
  margin-right: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

a.dapp-core-component__main__side-link::after {
  right: 0;
  left: unset;
  width: calc(100% - 1.85rem);
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(jp));var g={"link-style":"dapp-core-component__main__link-style",linkStyle:"dapp-core-component__main__link-style","link-second-style":"dapp-core-component__main__link-second-style",linkSecondStyle:"dapp-core-component__main__link-second-style",wrapper:"dapp-core-component__main__wrapper","opacity-6":"dapp-core-component__main__opacity-6",opacity6:"dapp-core-component__main__opacity-6","qr-code-svg-container":"dapp-core-component__main__qr-code-svg-container",qrCodeSvgContainer:"dapp-core-component__main__qr-code-svg-container","pair-list":"dapp-core-component__main__pair-list",pairList:"dapp-core-component__main__pair-list","pair-image":"dapp-core-component__main__pair-image",pairImage:"dapp-core-component__main__pair-image","pair-remove":"dapp-core-component__main__pair-remove",pairRemove:"dapp-core-component__main__pair-remove","dapp-icon":"dapp-core-component__main__dapp-icon",dappIcon:"dapp-core-component__main__dapp-icon","icon-medium":"dapp-core-component__main__icon-medium",iconMedium:"dapp-core-component__main__icon-medium",centering:"dapp-core-component__main__centering","border-n":"dapp-core-component__main__border-n",borderN:"dapp-core-component__main__border-n","t-shadow":"dapp-core-component__main__t-shadow",tShadow:"dapp-core-component__main__t-shadow",btn:"dapp-core-component__main__btn",media:"dapp-core-component__main__media","media-body":"dapp-core-component__main__media-body",mediaBody:"dapp-core-component__main__media-body","connect-btns":"dapp-core-component__main__connect-btns",connectBtns:"dapp-core-component__main__connect-btns",info:"dapp-core-component__main__info","trust-badge":"dapp-core-component__main__trust-badge",trustBadge:"dapp-core-component__main__trust-badge",navbar:"dapp-core-component__main__navbar","dapp-name":"dapp-core-component__main__dapp-name",dappName:"dapp-core-component__main__dapp-name","action-btn":"dapp-core-component__main__action-btn",actionBtn:"dapp-core-component__main__action-btn",transactions:"dapp-core-component__main__transactions","transaction-icon":"dapp-core-component__main__transaction-icon",transactionIcon:"dapp-core-component__main__transaction-icon",table:"dapp-core-component__main__table","page-state":"dapp-core-component__main__page-state",pageState:"dapp-core-component__main__page-state","lds-ellipsis":"dapp-core-component__main__lds-ellipsis",ldsEllipsis:"dapp-core-component__main__lds-ellipsis","bg-blue":"dapp-core-component__main__bg-blue",bgBlue:"dapp-core-component__main__bg-blue","lds-ellipsis1":"dapp-core-component__main__lds-ellipsis1",ldsEllipsis1:"dapp-core-component__main__lds-ellipsis1","lds-ellipsis2":"dapp-core-component__main__lds-ellipsis2",ldsEllipsis2:"dapp-core-component__main__lds-ellipsis2","lds-ellipsis3":"dapp-core-component__main__lds-ellipsis3",ldsEllipsis3:"dapp-core-component__main__lds-ellipsis3",focus:"dapp-core-component__main__focus",disabled:"dapp-core-component__main__disabled","btn-primary":"dapp-core-component__main__btn-primary",btnPrimary:"dapp-core-component__main__btn-primary","btn-secondary":"dapp-core-component__main__btn-secondary",btnSecondary:"dapp-core-component__main__btn-secondary","btn-warning":"dapp-core-component__main__btn-warning",btnWarning:"dapp-core-component__main__btn-warning","btn-dark":"dapp-core-component__main__btn-dark",btnDark:"dapp-core-component__main__btn-dark","btn-link":"dapp-core-component__main__btn-link",btnLink:"dapp-core-component__main__btn-link","btn-light":"dapp-core-component__main__btn-light",btnLight:"dapp-core-component__main__btn-light",badge:"dapp-core-component__main__badge","badge-pill":"dapp-core-component__main__badge-pill",badgePill:"dapp-core-component__main__badge-pill","badge-primary":"dapp-core-component__main__badge-primary",badgePrimary:"dapp-core-component__main__badge-primary","badge-secondary":"dapp-core-component__main__badge-secondary",badgeSecondary:"dapp-core-component__main__badge-secondary","badge-success":"dapp-core-component__main__badge-success",badgeSuccess:"dapp-core-component__main__badge-success","badge-info":"dapp-core-component__main__badge-info",badgeInfo:"dapp-core-component__main__badge-info","badge-warning":"dapp-core-component__main__badge-warning",badgeWarning:"dapp-core-component__main__badge-warning","badge-danger":"dapp-core-component__main__badge-danger",badgeDanger:"dapp-core-component__main__badge-danger","badge-light":"dapp-core-component__main__badge-light",badgeLight:"dapp-core-component__main__badge-light","badge-dark":"dapp-core-component__main__badge-dark",badgeDark:"dapp-core-component__main__badge-dark",card:"dapp-core-component__main__card","list-group":"dapp-core-component__main__list-group",listGroup:"dapp-core-component__main__list-group","card-header":"dapp-core-component__main__card-header",cardHeader:"dapp-core-component__main__card-header","card-footer":"dapp-core-component__main__card-footer",cardFooter:"dapp-core-component__main__card-footer","card-header-item":"dapp-core-component__main__card-header-item",cardHeaderItem:"dapp-core-component__main__card-header-item",h6:"dapp-core-component__main__h6",compact:"dapp-core-component__main__compact","card-body":"dapp-core-component__main__card-body",cardBody:"dapp-core-component__main__card-body","card-title":"dapp-core-component__main__card-title",cardTitle:"dapp-core-component__main__card-title","card-subtitle":"dapp-core-component__main__card-subtitle",cardSubtitle:"dapp-core-component__main__card-subtitle","card-text":"dapp-core-component__main__card-text",cardText:"dapp-core-component__main__card-text","card-link":"dapp-core-component__main__card-link",cardLink:"dapp-core-component__main__card-link",text:"dapp-core-component__main__text","text-white":"dapp-core-component__main__text-white",textWhite:"dapp-core-component__main__text-white","text-monospace":"dapp-core-component__main__text-monospace",textMonospace:"dapp-core-component__main__text-monospace","trim-size-xl":"dapp-core-component__main__trim-size-xl",trimSizeXl:"dapp-core-component__main__trim-size-xl",trim:"dapp-core-component__main__trim","trim-only-sm":"dapp-core-component__main__trim-only-sm",trimOnlySm:"dapp-core-component__main__trim-only-sm","text-justify":"dapp-core-component__main__text-justify",textJustify:"dapp-core-component__main__text-justify","text-wrap":"dapp-core-component__main__text-wrap",textWrap:"dapp-core-component__main__text-wrap","text-nowrap":"dapp-core-component__main__text-nowrap",textNowrap:"dapp-core-component__main__text-nowrap","text-truncate":"dapp-core-component__main__text-truncate",textTruncate:"dapp-core-component__main__text-truncate","text-left":"dapp-core-component__main__text-left",textLeft:"dapp-core-component__main__text-left","text-right":"dapp-core-component__main__text-right",textRight:"dapp-core-component__main__text-right","text-center":"dapp-core-component__main__text-center",textCenter:"dapp-core-component__main__text-center","text-primary":"dapp-core-component__main__text-primary",textPrimary:"dapp-core-component__main__text-primary","text-secondary":"dapp-core-component__main__text-secondary",textSecondary:"dapp-core-component__main__text-secondary","text-primary-highlight":"dapp-core-component__main__text-primary-highlight",textPrimaryHighlight:"dapp-core-component__main__text-primary-highlight","text-muted":"dapp-core-component__main__text-muted",textMuted:"dapp-core-component__main__text-muted","text-success":"dapp-core-component__main__text-success",textSuccess:"dapp-core-component__main__text-success","text-body":"dapp-core-component__main__text-body",textBody:"dapp-core-component__main__text-body","text-black-50":"dapp-core-component__main__text-black-50",textBlack50:"dapp-core-component__main__text-black-50","text-white-50":"dapp-core-component__main__text-white-50",textWhite50:"dapp-core-component__main__text-white-50","text-lg-right":"dapp-core-component__main__text-lg-right",textLgRight:"dapp-core-component__main__text-lg-right","text-hide":"dapp-core-component__main__text-hide",textHide:"dapp-core-component__main__text-hide","text-decoration-none":"dapp-core-component__main__text-decoration-none",textDecorationNone:"dapp-core-component__main__text-decoration-none","text-break":"dapp-core-component__main__text-break",textBreak:"dapp-core-component__main__text-break","text-reset":"dapp-core-component__main__text-reset",textReset:"dapp-core-component__main__text-reset","text-lowercase":"dapp-core-component__main__text-lowercase",textLowercase:"dapp-core-component__main__text-lowercase","text-uppercase":"dapp-core-component__main__text-uppercase",textUppercase:"dapp-core-component__main__text-uppercase","text-capitalize":"dapp-core-component__main__text-capitalize",textCapitalize:"dapp-core-component__main__text-capitalize","text-danger":"dapp-core-component__main__text-danger",textDanger:"dapp-core-component__main__text-danger","text-warning":"dapp-core-component__main__text-warning",textWarning:"dapp-core-component__main__text-warning","trim-fs-sm":"dapp-core-component__main__trim-fs-sm",trimFsSm:"dapp-core-component__main__trim-fs-sm","mx-auto":"dapp-core-component__main__mx-auto",mxAuto:"dapp-core-component__main__mx-auto","m-0":"dapp-core-component__main__m-0",m0:"dapp-core-component__main__m-0","mt-0":"dapp-core-component__main__mt-0",mt0:"dapp-core-component__main__mt-0","my-0":"dapp-core-component__main__my-0",my0:"dapp-core-component__main__my-0","mr-0":"dapp-core-component__main__mr-0",mr0:"dapp-core-component__main__mr-0","mx-0":"dapp-core-component__main__mx-0",mx0:"dapp-core-component__main__mx-0","mb-0":"dapp-core-component__main__mb-0",mb0:"dapp-core-component__main__mb-0","ml-0":"dapp-core-component__main__ml-0",ml0:"dapp-core-component__main__ml-0","m-1":"dapp-core-component__main__m-1",m1:"dapp-core-component__main__m-1","mt-1":"dapp-core-component__main__mt-1",mt1:"dapp-core-component__main__mt-1","my-1":"dapp-core-component__main__my-1",my1:"dapp-core-component__main__my-1","mr-1":"dapp-core-component__main__mr-1",mr1:"dapp-core-component__main__mr-1","mx-1":"dapp-core-component__main__mx-1",mx1:"dapp-core-component__main__mx-1","mb-1":"dapp-core-component__main__mb-1",mb1:"dapp-core-component__main__mb-1","ml-1":"dapp-core-component__main__ml-1",ml1:"dapp-core-component__main__ml-1","m-2":"dapp-core-component__main__m-2",m2:"dapp-core-component__main__m-2","mt-2":"dapp-core-component__main__mt-2",mt2:"dapp-core-component__main__mt-2","my-2":"dapp-core-component__main__my-2",my2:"dapp-core-component__main__my-2","mr-2":"dapp-core-component__main__mr-2",mr2:"dapp-core-component__main__mr-2","mx-2":"dapp-core-component__main__mx-2",mx2:"dapp-core-component__main__mx-2","mb-2":"dapp-core-component__main__mb-2",mb2:"dapp-core-component__main__mb-2","ml-2":"dapp-core-component__main__ml-2",ml2:"dapp-core-component__main__ml-2","m-3":"dapp-core-component__main__m-3",m3:"dapp-core-component__main__m-3","mt-3":"dapp-core-component__main__mt-3",mt3:"dapp-core-component__main__mt-3","my-3":"dapp-core-component__main__my-3",my3:"dapp-core-component__main__my-3","mr-3":"dapp-core-component__main__mr-3",mr3:"dapp-core-component__main__mr-3","mx-3":"dapp-core-component__main__mx-3",mx3:"dapp-core-component__main__mx-3","mb-3":"dapp-core-component__main__mb-3",mb3:"dapp-core-component__main__mb-3","ml-3":"dapp-core-component__main__ml-3",ml3:"dapp-core-component__main__ml-3","m-4":"dapp-core-component__main__m-4",m4:"dapp-core-component__main__m-4","mt-4":"dapp-core-component__main__mt-4",mt4:"dapp-core-component__main__mt-4","my-4":"dapp-core-component__main__my-4",my4:"dapp-core-component__main__my-4","mr-4":"dapp-core-component__main__mr-4",mr4:"dapp-core-component__main__mr-4","mx-4":"dapp-core-component__main__mx-4",mx4:"dapp-core-component__main__mx-4","mb-4":"dapp-core-component__main__mb-4",mb4:"dapp-core-component__main__mb-4","ml-4":"dapp-core-component__main__ml-4",ml4:"dapp-core-component__main__ml-4","m-5":"dapp-core-component__main__m-5",m5:"dapp-core-component__main__m-5","mt-5":"dapp-core-component__main__mt-5",mt5:"dapp-core-component__main__mt-5","my-5":"dapp-core-component__main__my-5",my5:"dapp-core-component__main__my-5","mr-5":"dapp-core-component__main__mr-5",mr5:"dapp-core-component__main__mr-5","mx-5":"dapp-core-component__main__mx-5",mx5:"dapp-core-component__main__mx-5","mb-5":"dapp-core-component__main__mb-5",mb5:"dapp-core-component__main__mb-5","ml-5":"dapp-core-component__main__ml-5",ml5:"dapp-core-component__main__ml-5","p-0":"dapp-core-component__main__p-0",p0:"dapp-core-component__main__p-0","pt-0":"dapp-core-component__main__pt-0",pt0:"dapp-core-component__main__pt-0","py-0":"dapp-core-component__main__py-0",py0:"dapp-core-component__main__py-0","pr-0":"dapp-core-component__main__pr-0",pr0:"dapp-core-component__main__pr-0","px-0":"dapp-core-component__main__px-0",px0:"dapp-core-component__main__px-0","pb-0":"dapp-core-component__main__pb-0",pb0:"dapp-core-component__main__pb-0","pl-0":"dapp-core-component__main__pl-0",pl0:"dapp-core-component__main__pl-0","p-1":"dapp-core-component__main__p-1",p1:"dapp-core-component__main__p-1","pt-1":"dapp-core-component__main__pt-1",pt1:"dapp-core-component__main__pt-1","py-1":"dapp-core-component__main__py-1",py1:"dapp-core-component__main__py-1","pr-1":"dapp-core-component__main__pr-1",pr1:"dapp-core-component__main__pr-1","px-1":"dapp-core-component__main__px-1",px1:"dapp-core-component__main__px-1","pb-1":"dapp-core-component__main__pb-1",pb1:"dapp-core-component__main__pb-1","pl-1":"dapp-core-component__main__pl-1",pl1:"dapp-core-component__main__pl-1","p-2":"dapp-core-component__main__p-2",p2:"dapp-core-component__main__p-2","pt-2":"dapp-core-component__main__pt-2",pt2:"dapp-core-component__main__pt-2","py-2":"dapp-core-component__main__py-2",py2:"dapp-core-component__main__py-2","pr-2":"dapp-core-component__main__pr-2",pr2:"dapp-core-component__main__pr-2","px-2":"dapp-core-component__main__px-2",px2:"dapp-core-component__main__px-2","pb-2":"dapp-core-component__main__pb-2",pb2:"dapp-core-component__main__pb-2","pl-2":"dapp-core-component__main__pl-2",pl2:"dapp-core-component__main__pl-2","p-3":"dapp-core-component__main__p-3",p3:"dapp-core-component__main__p-3","pt-3":"dapp-core-component__main__pt-3",pt3:"dapp-core-component__main__pt-3","py-3":"dapp-core-component__main__py-3",py3:"dapp-core-component__main__py-3","pr-3":"dapp-core-component__main__pr-3",pr3:"dapp-core-component__main__pr-3","px-3":"dapp-core-component__main__px-3",px3:"dapp-core-component__main__px-3","pb-3":"dapp-core-component__main__pb-3",pb3:"dapp-core-component__main__pb-3","pl-3":"dapp-core-component__main__pl-3",pl3:"dapp-core-component__main__pl-3","p-4":"dapp-core-component__main__p-4",p4:"dapp-core-component__main__p-4","pt-4":"dapp-core-component__main__pt-4",pt4:"dapp-core-component__main__pt-4","py-4":"dapp-core-component__main__py-4",py4:"dapp-core-component__main__py-4","pr-4":"dapp-core-component__main__pr-4",pr4:"dapp-core-component__main__pr-4","px-4":"dapp-core-component__main__px-4",px4:"dapp-core-component__main__px-4","pb-4":"dapp-core-component__main__pb-4",pb4:"dapp-core-component__main__pb-4","pl-4":"dapp-core-component__main__pl-4",pl4:"dapp-core-component__main__pl-4","p-5":"dapp-core-component__main__p-5",p5:"dapp-core-component__main__p-5","pt-5":"dapp-core-component__main__pt-5",pt5:"dapp-core-component__main__pt-5","py-5":"dapp-core-component__main__py-5",py5:"dapp-core-component__main__py-5","pr-5":"dapp-core-component__main__pr-5",pr5:"dapp-core-component__main__pr-5","px-5":"dapp-core-component__main__px-5",px5:"dapp-core-component__main__px-5","pb-5":"dapp-core-component__main__pb-5",pb5:"dapp-core-component__main__pb-5","pl-5":"dapp-core-component__main__pl-5",pl5:"dapp-core-component__main__pl-5","m-n1":"dapp-core-component__main__m-n1",mN1:"dapp-core-component__main__m-n1","mt-n1":"dapp-core-component__main__mt-n1",mtN1:"dapp-core-component__main__mt-n1","my-n1":"dapp-core-component__main__my-n1",myN1:"dapp-core-component__main__my-n1","mr-n1":"dapp-core-component__main__mr-n1",mrN1:"dapp-core-component__main__mr-n1","mx-n1":"dapp-core-component__main__mx-n1",mxN1:"dapp-core-component__main__mx-n1","mb-n1":"dapp-core-component__main__mb-n1",mbN1:"dapp-core-component__main__mb-n1","ml-n1":"dapp-core-component__main__ml-n1",mlN1:"dapp-core-component__main__ml-n1","m-n2":"dapp-core-component__main__m-n2",mN2:"dapp-core-component__main__m-n2","mt-n2":"dapp-core-component__main__mt-n2",mtN2:"dapp-core-component__main__mt-n2","my-n2":"dapp-core-component__main__my-n2",myN2:"dapp-core-component__main__my-n2","mr-n2":"dapp-core-component__main__mr-n2",mrN2:"dapp-core-component__main__mr-n2","mx-n2":"dapp-core-component__main__mx-n2",mxN2:"dapp-core-component__main__mx-n2","mb-n2":"dapp-core-component__main__mb-n2",mbN2:"dapp-core-component__main__mb-n2","ml-n2":"dapp-core-component__main__ml-n2",mlN2:"dapp-core-component__main__ml-n2","m-n3":"dapp-core-component__main__m-n3",mN3:"dapp-core-component__main__m-n3","mt-n3":"dapp-core-component__main__mt-n3",mtN3:"dapp-core-component__main__mt-n3","my-n3":"dapp-core-component__main__my-n3",myN3:"dapp-core-component__main__my-n3","mr-n3":"dapp-core-component__main__mr-n3",mrN3:"dapp-core-component__main__mr-n3","mx-n3":"dapp-core-component__main__mx-n3",mxN3:"dapp-core-component__main__mx-n3","mb-n3":"dapp-core-component__main__mb-n3",mbN3:"dapp-core-component__main__mb-n3","ml-n3":"dapp-core-component__main__ml-n3",mlN3:"dapp-core-component__main__ml-n3","m-n4":"dapp-core-component__main__m-n4",mN4:"dapp-core-component__main__m-n4","mt-n4":"dapp-core-component__main__mt-n4",mtN4:"dapp-core-component__main__mt-n4","my-n4":"dapp-core-component__main__my-n4",myN4:"dapp-core-component__main__my-n4","mr-n4":"dapp-core-component__main__mr-n4",mrN4:"dapp-core-component__main__mr-n4","mx-n4":"dapp-core-component__main__mx-n4",mxN4:"dapp-core-component__main__mx-n4","mb-n4":"dapp-core-component__main__mb-n4",mbN4:"dapp-core-component__main__mb-n4","ml-n4":"dapp-core-component__main__ml-n4",mlN4:"dapp-core-component__main__ml-n4","m-n5":"dapp-core-component__main__m-n5",mN5:"dapp-core-component__main__m-n5","mt-n5":"dapp-core-component__main__mt-n5",mtN5:"dapp-core-component__main__mt-n5","my-n5":"dapp-core-component__main__my-n5",myN5:"dapp-core-component__main__my-n5","mr-n5":"dapp-core-component__main__mr-n5",mrN5:"dapp-core-component__main__mr-n5","mx-n5":"dapp-core-component__main__mx-n5",mxN5:"dapp-core-component__main__mx-n5","mb-n5":"dapp-core-component__main__mb-n5",mbN5:"dapp-core-component__main__mb-n5","ml-n5":"dapp-core-component__main__ml-n5",mlN5:"dapp-core-component__main__ml-n5","m-auto":"dapp-core-component__main__m-auto",mAuto:"dapp-core-component__main__m-auto","mt-auto":"dapp-core-component__main__mt-auto",mtAuto:"dapp-core-component__main__mt-auto","my-auto":"dapp-core-component__main__my-auto",myAuto:"dapp-core-component__main__my-auto","mr-auto":"dapp-core-component__main__mr-auto",mrAuto:"dapp-core-component__main__mr-auto","mb-auto":"dapp-core-component__main__mb-auto",mbAuto:"dapp-core-component__main__mb-auto","ml-auto":"dapp-core-component__main__ml-auto",mlAuto:"dapp-core-component__main__ml-auto","mr-lg-1":"dapp-core-component__main__mr-lg-1",mrLg1:"dapp-core-component__main__mr-lg-1","mr-lg-4":"dapp-core-component__main__mr-lg-4",mrLg4:"dapp-core-component__main__mr-lg-4","mx-lg-4":"dapp-core-component__main__mx-lg-4",mxLg4:"dapp-core-component__main__mx-lg-4","mb-xl-2":"dapp-core-component__main__mb-xl-2",mbXl2:"dapp-core-component__main__mb-xl-2","pr-sm-2":"dapp-core-component__main__pr-sm-2",prSm2:"dapp-core-component__main__pr-sm-2","px-sm-2":"dapp-core-component__main__px-sm-2",pxSm2:"dapp-core-component__main__px-sm-2","pr-xl-0":"dapp-core-component__main__pr-xl-0",prXl0:"dapp-core-component__main__pr-xl-0","px-xl-0":"dapp-core-component__main__px-xl-0",pxXl0:"dapp-core-component__main__px-xl-0",col:"dapp-core-component__main__col",row:"dapp-core-component__main__row","col-xl":"dapp-core-component__main__col-xl",colXl:"dapp-core-component__main__col-xl","col-xl-auto":"dapp-core-component__main__col-xl-auto",colXlAuto:"dapp-core-component__main__col-xl-auto","col-xl-12":"dapp-core-component__main__col-xl-12",colXl12:"dapp-core-component__main__col-xl-12","col-xl-11":"dapp-core-component__main__col-xl-11",colXl11:"dapp-core-component__main__col-xl-11","col-xl-10":"dapp-core-component__main__col-xl-10",colXl10:"dapp-core-component__main__col-xl-10","col-xl-9":"dapp-core-component__main__col-xl-9",colXl9:"dapp-core-component__main__col-xl-9","col-xl-8":"dapp-core-component__main__col-xl-8",colXl8:"dapp-core-component__main__col-xl-8","col-xl-7":"dapp-core-component__main__col-xl-7",colXl7:"dapp-core-component__main__col-xl-7","col-xl-6":"dapp-core-component__main__col-xl-6",colXl6:"dapp-core-component__main__col-xl-6","col-xl-5":"dapp-core-component__main__col-xl-5",colXl5:"dapp-core-component__main__col-xl-5","col-xl-4":"dapp-core-component__main__col-xl-4",colXl4:"dapp-core-component__main__col-xl-4","col-xl-3":"dapp-core-component__main__col-xl-3",colXl3:"dapp-core-component__main__col-xl-3","col-xl-2":"dapp-core-component__main__col-xl-2",colXl2:"dapp-core-component__main__col-xl-2","col-xl-1":"dapp-core-component__main__col-xl-1",colXl1:"dapp-core-component__main__col-xl-1","col-lg":"dapp-core-component__main__col-lg",colLg:"dapp-core-component__main__col-lg","col-lg-auto":"dapp-core-component__main__col-lg-auto",colLgAuto:"dapp-core-component__main__col-lg-auto","col-lg-12":"dapp-core-component__main__col-lg-12",colLg12:"dapp-core-component__main__col-lg-12","col-lg-11":"dapp-core-component__main__col-lg-11",colLg11:"dapp-core-component__main__col-lg-11","col-lg-10":"dapp-core-component__main__col-lg-10",colLg10:"dapp-core-component__main__col-lg-10","col-lg-9":"dapp-core-component__main__col-lg-9",colLg9:"dapp-core-component__main__col-lg-9","col-lg-8":"dapp-core-component__main__col-lg-8",colLg8:"dapp-core-component__main__col-lg-8","col-lg-7":"dapp-core-component__main__col-lg-7",colLg7:"dapp-core-component__main__col-lg-7","col-lg-6":"dapp-core-component__main__col-lg-6",colLg6:"dapp-core-component__main__col-lg-6","col-lg-5":"dapp-core-component__main__col-lg-5",colLg5:"dapp-core-component__main__col-lg-5","col-lg-4":"dapp-core-component__main__col-lg-4",colLg4:"dapp-core-component__main__col-lg-4","col-lg-3":"dapp-core-component__main__col-lg-3",colLg3:"dapp-core-component__main__col-lg-3","col-lg-2":"dapp-core-component__main__col-lg-2",colLg2:"dapp-core-component__main__col-lg-2","col-lg-1":"dapp-core-component__main__col-lg-1",colLg1:"dapp-core-component__main__col-lg-1","col-md":"dapp-core-component__main__col-md",colMd:"dapp-core-component__main__col-md","col-md-auto":"dapp-core-component__main__col-md-auto",colMdAuto:"dapp-core-component__main__col-md-auto","col-md-12":"dapp-core-component__main__col-md-12",colMd12:"dapp-core-component__main__col-md-12","col-md-11":"dapp-core-component__main__col-md-11",colMd11:"dapp-core-component__main__col-md-11","col-md-10":"dapp-core-component__main__col-md-10",colMd10:"dapp-core-component__main__col-md-10","col-md-9":"dapp-core-component__main__col-md-9",colMd9:"dapp-core-component__main__col-md-9","col-md-8":"dapp-core-component__main__col-md-8",colMd8:"dapp-core-component__main__col-md-8","col-md-7":"dapp-core-component__main__col-md-7",colMd7:"dapp-core-component__main__col-md-7","col-md-6":"dapp-core-component__main__col-md-6",colMd6:"dapp-core-component__main__col-md-6","col-md-5":"dapp-core-component__main__col-md-5",colMd5:"dapp-core-component__main__col-md-5","col-md-4":"dapp-core-component__main__col-md-4",colMd4:"dapp-core-component__main__col-md-4","col-md-3":"dapp-core-component__main__col-md-3",colMd3:"dapp-core-component__main__col-md-3","col-md-2":"dapp-core-component__main__col-md-2",colMd2:"dapp-core-component__main__col-md-2","col-md-1":"dapp-core-component__main__col-md-1",colMd1:"dapp-core-component__main__col-md-1","col-sm":"dapp-core-component__main__col-sm",colSm:"dapp-core-component__main__col-sm","col-sm-auto":"dapp-core-component__main__col-sm-auto",colSmAuto:"dapp-core-component__main__col-sm-auto","col-sm-12":"dapp-core-component__main__col-sm-12",colSm12:"dapp-core-component__main__col-sm-12","col-sm-11":"dapp-core-component__main__col-sm-11",colSm11:"dapp-core-component__main__col-sm-11","col-sm-10":"dapp-core-component__main__col-sm-10",colSm10:"dapp-core-component__main__col-sm-10","col-sm-9":"dapp-core-component__main__col-sm-9",colSm9:"dapp-core-component__main__col-sm-9","col-sm-8":"dapp-core-component__main__col-sm-8",colSm8:"dapp-core-component__main__col-sm-8","col-sm-7":"dapp-core-component__main__col-sm-7",colSm7:"dapp-core-component__main__col-sm-7","col-sm-6":"dapp-core-component__main__col-sm-6",colSm6:"dapp-core-component__main__col-sm-6","col-sm-5":"dapp-core-component__main__col-sm-5",colSm5:"dapp-core-component__main__col-sm-5","col-sm-4":"dapp-core-component__main__col-sm-4",colSm4:"dapp-core-component__main__col-sm-4","col-sm-3":"dapp-core-component__main__col-sm-3",colSm3:"dapp-core-component__main__col-sm-3","col-sm-2":"dapp-core-component__main__col-sm-2",colSm2:"dapp-core-component__main__col-sm-2","col-sm-1":"dapp-core-component__main__col-sm-1",colSm1:"dapp-core-component__main__col-sm-1","col-auto":"dapp-core-component__main__col-auto",colAuto:"dapp-core-component__main__col-auto","col-12":"dapp-core-component__main__col-12",col12:"dapp-core-component__main__col-12","col-11":"dapp-core-component__main__col-11",col11:"dapp-core-component__main__col-11","col-10":"dapp-core-component__main__col-10",col10:"dapp-core-component__main__col-10","col-9":"dapp-core-component__main__col-9",col9:"dapp-core-component__main__col-9","col-8":"dapp-core-component__main__col-8",col8:"dapp-core-component__main__col-8","col-7":"dapp-core-component__main__col-7",col7:"dapp-core-component__main__col-7","col-6":"dapp-core-component__main__col-6",col6:"dapp-core-component__main__col-6","col-5":"dapp-core-component__main__col-5",col5:"dapp-core-component__main__col-5","col-4":"dapp-core-component__main__col-4",col4:"dapp-core-component__main__col-4","col-3":"dapp-core-component__main__col-3",col3:"dapp-core-component__main__col-3","col-2":"dapp-core-component__main__col-2",col2:"dapp-core-component__main__col-2","col-1":"dapp-core-component__main__col-1",col1:"dapp-core-component__main__col-1","flex-row":"dapp-core-component__main__flex-row",flexRow:"dapp-core-component__main__flex-row","flex-column":"dapp-core-component__main__flex-column",flexColumn:"dapp-core-component__main__flex-column","flex-row-reverse":"dapp-core-component__main__flex-row-reverse",flexRowReverse:"dapp-core-component__main__flex-row-reverse","flex-column-reverse":"dapp-core-component__main__flex-column-reverse",flexColumnReverse:"dapp-core-component__main__flex-column-reverse","flex-wrap":"dapp-core-component__main__flex-wrap",flexWrap:"dapp-core-component__main__flex-wrap","flex-nowrap":"dapp-core-component__main__flex-nowrap",flexNowrap:"dapp-core-component__main__flex-nowrap","flex-wrap-reverse":"dapp-core-component__main__flex-wrap-reverse",flexWrapReverse:"dapp-core-component__main__flex-wrap-reverse","flex-fill":"dapp-core-component__main__flex-fill",flexFill:"dapp-core-component__main__flex-fill","flex-grow-0":"dapp-core-component__main__flex-grow-0",flexGrow0:"dapp-core-component__main__flex-grow-0","flex-grow-1":"dapp-core-component__main__flex-grow-1",flexGrow1:"dapp-core-component__main__flex-grow-1","flex-shrink-0":"dapp-core-component__main__flex-shrink-0",flexShrink0:"dapp-core-component__main__flex-shrink-0","flex-shrink-1":"dapp-core-component__main__flex-shrink-1",flexShrink1:"dapp-core-component__main__flex-shrink-1","justify-content-start":"dapp-core-component__main__justify-content-start",justifyContentStart:"dapp-core-component__main__justify-content-start","justify-content-end":"dapp-core-component__main__justify-content-end",justifyContentEnd:"dapp-core-component__main__justify-content-end","justify-content-center":"dapp-core-component__main__justify-content-center",justifyContentCenter:"dapp-core-component__main__justify-content-center","justify-content-between":"dapp-core-component__main__justify-content-between",justifyContentBetween:"dapp-core-component__main__justify-content-between","justify-content-around":"dapp-core-component__main__justify-content-around",justifyContentAround:"dapp-core-component__main__justify-content-around","align-items-start":"dapp-core-component__main__align-items-start",alignItemsStart:"dapp-core-component__main__align-items-start","align-items-end":"dapp-core-component__main__align-items-end",alignItemsEnd:"dapp-core-component__main__align-items-end","align-items-center":"dapp-core-component__main__align-items-center",alignItemsCenter:"dapp-core-component__main__align-items-center","align-items-baseline":"dapp-core-component__main__align-items-baseline",alignItemsBaseline:"dapp-core-component__main__align-items-baseline","align-items-stretch":"dapp-core-component__main__align-items-stretch",alignItemsStretch:"dapp-core-component__main__align-items-stretch","align-content-start":"dapp-core-component__main__align-content-start",alignContentStart:"dapp-core-component__main__align-content-start","align-content-end":"dapp-core-component__main__align-content-end",alignContentEnd:"dapp-core-component__main__align-content-end","align-content-center":"dapp-core-component__main__align-content-center",alignContentCenter:"dapp-core-component__main__align-content-center","align-content-between":"dapp-core-component__main__align-content-between",alignContentBetween:"dapp-core-component__main__align-content-between","align-content-around":"dapp-core-component__main__align-content-around",alignContentAround:"dapp-core-component__main__align-content-around","align-content-stretch":"dapp-core-component__main__align-content-stretch",alignContentStretch:"dapp-core-component__main__align-content-stretch","align-self-auto":"dapp-core-component__main__align-self-auto",alignSelfAuto:"dapp-core-component__main__align-self-auto","align-self-start":"dapp-core-component__main__align-self-start",alignSelfStart:"dapp-core-component__main__align-self-start","align-self-end":"dapp-core-component__main__align-self-end",alignSelfEnd:"dapp-core-component__main__align-self-end","align-self-center":"dapp-core-component__main__align-self-center",alignSelfCenter:"dapp-core-component__main__align-self-center","align-self-baseline":"dapp-core-component__main__align-self-baseline",alignSelfBaseline:"dapp-core-component__main__align-self-baseline","align-self-stretch":"dapp-core-component__main__align-self-stretch",alignSelfStretch:"dapp-core-component__main__align-self-stretch","flex-sm-row":"dapp-core-component__main__flex-sm-row",flexSmRow:"dapp-core-component__main__flex-sm-row","flex-lg-row":"dapp-core-component__main__flex-lg-row",flexLgRow:"dapp-core-component__main__flex-lg-row","flex-lg-wrap":"dapp-core-component__main__flex-lg-wrap",flexLgWrap:"dapp-core-component__main__flex-lg-wrap","form-check":"dapp-core-component__main__form-check",formCheck:"dapp-core-component__main__form-check","form-check-label":"dapp-core-component__main__form-check-label",formCheckLabel:"dapp-core-component__main__form-check-label","form-check-input":"dapp-core-component__main__form-check-input",formCheckInput:"dapp-core-component__main__form-check-input","form-group":"dapp-core-component__main__form-group",formGroup:"dapp-core-component__main__form-group","form-control":"dapp-core-component__main__form-control",formControl:"dapp-core-component__main__form-control","form-label":"dapp-core-component__main__form-label",formLabel:"dapp-core-component__main__form-label","form-text":"dapp-core-component__main__form-text",formText:"dapp-core-component__main__form-text",border:"dapp-core-component__main__border","border-top":"dapp-core-component__main__border-top",borderTop:"dapp-core-component__main__border-top","border-right":"dapp-core-component__main__border-right",borderRight:"dapp-core-component__main__border-right","border-bottom":"dapp-core-component__main__border-bottom",borderBottom:"dapp-core-component__main__border-bottom","border-left":"dapp-core-component__main__border-left",borderLeft:"dapp-core-component__main__border-left","border-0":"dapp-core-component__main__border-0",border0:"dapp-core-component__main__border-0","border-top-0":"dapp-core-component__main__border-top-0",borderTop0:"dapp-core-component__main__border-top-0","border-right-0":"dapp-core-component__main__border-right-0",borderRight0:"dapp-core-component__main__border-right-0","border-bottom-0":"dapp-core-component__main__border-bottom-0",borderBottom0:"dapp-core-component__main__border-bottom-0","border-left-0":"dapp-core-component__main__border-left-0",borderLeft0:"dapp-core-component__main__border-left-0","rounded-sm":"dapp-core-component__main__rounded-sm",roundedSm:"dapp-core-component__main__rounded-sm",rounded:"dapp-core-component__main__rounded","rounded-top":"dapp-core-component__main__rounded-top",roundedTop:"dapp-core-component__main__rounded-top","rounded-right":"dapp-core-component__main__rounded-right",roundedRight:"dapp-core-component__main__rounded-right","rounded-bottom":"dapp-core-component__main__rounded-bottom",roundedBottom:"dapp-core-component__main__rounded-bottom","rounded-left":"dapp-core-component__main__rounded-left",roundedLeft:"dapp-core-component__main__rounded-left","rounded-lg":"dapp-core-component__main__rounded-lg",roundedLg:"dapp-core-component__main__rounded-lg","rounded-circle":"dapp-core-component__main__rounded-circle",roundedCircle:"dapp-core-component__main__rounded-circle","rounded-pill":"dapp-core-component__main__rounded-pill",roundedPill:"dapp-core-component__main__rounded-pill","rounded-0":"dapp-core-component__main__rounded-0",rounded0:"dapp-core-component__main__rounded-0",h1:"dapp-core-component__main__h1",h2:"dapp-core-component__main__h2",h3:"dapp-core-component__main__h3",h4:"dapp-core-component__main__h4",h5:"dapp-core-component__main__h5",lead:"dapp-core-component__main__lead","bg-primary":"dapp-core-component__main__bg-primary",bgPrimary:"dapp-core-component__main__bg-primary","bg-warning":"dapp-core-component__main__bg-warning",bgWarning:"dapp-core-component__main__bg-warning",warning:"dapp-core-component__main__warning","bg-danger":"dapp-core-component__main__bg-danger",bgDanger:"dapp-core-component__main__bg-danger",danger:"dapp-core-component__main__danger","bg-success":"dapp-core-component__main__bg-success",bgSuccess:"dapp-core-component__main__bg-success",success:"dapp-core-component__main__success","position-relative":"dapp-core-component__main__position-relative",positionRelative:"dapp-core-component__main__position-relative","position-absolute":"dapp-core-component__main__position-absolute",positionAbsolute:"dapp-core-component__main__position-absolute","w-25":"dapp-core-component__main__w-25",w25:"dapp-core-component__main__w-25","w-50":"dapp-core-component__main__w-50",w50:"dapp-core-component__main__w-50","w-75":"dapp-core-component__main__w-75",w75:"dapp-core-component__main__w-75","w-100":"dapp-core-component__main__w-100",w100:"dapp-core-component__main__w-100","w-auto":"dapp-core-component__main__w-auto",wAuto:"dapp-core-component__main__w-auto","h-25":"dapp-core-component__main__h-25",h25:"dapp-core-component__main__h-25","h-50":"dapp-core-component__main__h-50",h50:"dapp-core-component__main__h-50","h-75":"dapp-core-component__main__h-75",h75:"dapp-core-component__main__h-75","h-100":"dapp-core-component__main__h-100",h100:"dapp-core-component__main__h-100","h-auto":"dapp-core-component__main__h-auto",hAuto:"dapp-core-component__main__h-auto","mw-100":"dapp-core-component__main__mw-100",mw100:"dapp-core-component__main__mw-100","mh-100":"dapp-core-component__main__mh-100",mh100:"dapp-core-component__main__mh-100","min-vw-100":"dapp-core-component__main__min-vw-100",minVw100:"dapp-core-component__main__min-vw-100","min-vh-100":"dapp-core-component__main__min-vh-100",minVh100:"dapp-core-component__main__min-vh-100","vw-100":"dapp-core-component__main__vw-100",vw100:"dapp-core-component__main__vw-100","vh-100":"dapp-core-component__main__vh-100",vh100:"dapp-core-component__main__vh-100","locked-icon":"dapp-core-component__main__locked-icon",lockedIcon:"dapp-core-component__main__locked-icon","shadow-sm":"dapp-core-component__main__shadow-sm",shadowSm:"dapp-core-component__main__shadow-sm",shadow:"dapp-core-component__main__shadow","shadow-lg":"dapp-core-component__main__shadow-lg",shadowLg:"dapp-core-component__main__shadow-lg","shadow-none":"dapp-core-component__main__shadow-none",shadowNone:"dapp-core-component__main__shadow-none","d-none":"dapp-core-component__main__d-none",dNone:"dapp-core-component__main__d-none","d-inline":"dapp-core-component__main__d-inline",dInline:"dapp-core-component__main__d-inline","d-inline-block":"dapp-core-component__main__d-inline-block",dInlineBlock:"dapp-core-component__main__d-inline-block","d-block":"dapp-core-component__main__d-block",dBlock:"dapp-core-component__main__d-block","d-table":"dapp-core-component__main__d-table",dTable:"dapp-core-component__main__d-table","d-table-row":"dapp-core-component__main__d-table-row",dTableRow:"dapp-core-component__main__d-table-row","d-table-cell":"dapp-core-component__main__d-table-cell",dTableCell:"dapp-core-component__main__d-table-cell","d-flex":"dapp-core-component__main__d-flex",dFlex:"dapp-core-component__main__d-flex","d-inline-flex":"dapp-core-component__main__d-inline-flex",dInlineFlex:"dapp-core-component__main__d-inline-flex","d-sm-flex":"dapp-core-component__main__d-sm-flex",dSmFlex:"dapp-core-component__main__d-sm-flex","table-sm":"dapp-core-component__main__table-sm",tableSm:"dapp-core-component__main__table-sm","table-bordered":"dapp-core-component__main__table-bordered",tableBordered:"dapp-core-component__main__table-bordered","table-borderless":"dapp-core-component__main__table-borderless",tableBorderless:"dapp-core-component__main__table-borderless","table-striped":"dapp-core-component__main__table-striped",tableStriped:"dapp-core-component__main__table-striped","thead-dark":"dapp-core-component__main__thead-dark",theadDark:"dapp-core-component__main__thead-dark","thead-light":"dapp-core-component__main__thead-light",theadLight:"dapp-core-component__main__thead-light","table-dark":"dapp-core-component__main__table-dark",tableDark:"dapp-core-component__main__table-dark","table-hover":"dapp-core-component__main__table-hover",tableHover:"dapp-core-component__main__table-hover","table-responsive":"dapp-core-component__main__table-responsive",tableResponsive:"dapp-core-component__main__table-responsive","transaction-hash":"dapp-core-component__main__transaction-hash",transactionHash:"dapp-core-component__main__transaction-hash","font-weight-light":"dapp-core-component__main__font-weight-light",fontWeightLight:"dapp-core-component__main__font-weight-light","font-weight-lighter":"dapp-core-component__main__font-weight-lighter",fontWeightLighter:"dapp-core-component__main__font-weight-lighter","font-weight-normal":"dapp-core-component__main__font-weight-normal",fontWeightNormal:"dapp-core-component__main__font-weight-normal","font-weight-medium":"dapp-core-component__main__font-weight-medium",fontWeightMedium:"dapp-core-component__main__font-weight-medium","font-weight-bold":"dapp-core-component__main__font-weight-bold",fontWeightBold:"dapp-core-component__main__font-weight-bold","font-weight-bolder":"dapp-core-component__main__font-weight-bolder",fontWeightBolder:"dapp-core-component__main__font-weight-bolder","font-italic":"dapp-core-component__main__font-italic",fontItalic:"dapp-core-component__main__font-italic","icon-state":"dapp-core-component__main__icon-state",iconState:"dapp-core-component__main__icon-state",half:"dapp-core-component__main__half","side-icon":"dapp-core-component__main__side-icon",sideIcon:"dapp-core-component__main__side-icon","toast-messages":"dapp-core-component__main__toast-messages",toastMessages:"dapp-core-component__main__toast-messages",toast:"dapp-core-component__main__toast",clickable:"dapp-core-component__main__clickable","toast-visible":"dapp-core-component__main__toast-visible",toastVisible:"dapp-core-component__main__toast-visible",progress:"dapp-core-component__main__progress","progress-bar":"dapp-core-component__main__progress-bar",progressBar:"dapp-core-component__main__progress-bar","content-height":"dapp-core-component__main__content-height",contentHeight:"dapp-core-component__main__content-height","tx-description":"dapp-core-component__main__tx-description",txDescription:"dapp-core-component__main__tx-description",close:"dapp-core-component__main__close","data-field":"dapp-core-component__main__data-field",dataField:"dapp-core-component__main__data-field",textarea:"dapp-core-component__main__textarea",highlighted:"dapp-core-component__main__highlighted","sc-call":"dapp-core-component__main__sc-call",scCall:"dapp-core-component__main__sc-call","small-font":"dapp-core-component__main__small-font",smallFont:"dapp-core-component__main__small-font","progress-steps":"dapp-core-component__main__progress-steps",progressSteps:"dapp-core-component__main__progress-steps",steps:"dapp-core-component__main__steps",dot:"dapp-core-component__main__dot","side-link":"dapp-core-component__main__side-link",sideLink:"dapp-core-component__main__side-link"};r();var Gp=`.dapp-core-component__explorerLinkStyles__link .dapp-core-component__explorerLinkStyles__search {
  color: #6c757d;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Gp));var Dn={link:"dapp-core-component__explorerLinkStyles__link",search:"dapp-core-component__explorerLinkStyles__search"};var ee=i=>{var _=i,{page:e,text:o,className:n="dapp-explorer-link",children:t}=_,a=Te(_,["page","text","className","children"]);let{network:{explorerAddress:l}}=ga(),d=o!=null?o:Cn.default.createElement(ha.FontAwesomeIcon,{icon:fa.faSearch,className:Dn.search}),u=En({explorerAddress:String(l),to:e});return Cn.default.createElement("a",b({href:u,target:"_blank",className:(0,xa.default)(Dn.link,g.ml2,n),rel:"noreferrer"},a),t!=null?t:d)};r();r();var pe=T(require("react")),ya=T(require("classnames"));I();r();var qp=`.dapp-core-component__formatAmountStyles__int-amount {
  color: inherit;
}

.dapp-core-component__formatAmountStyles__decimals {
  color: inherit;
}

.dapp-core-component__formatAmountStyles__symbol {
  color: inherit;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(qp));var Ke={"int-amount":"dapp-core-component__formatAmountStyles__int-amount",intAmount:"dapp-core-component__formatAmountStyles__int-amount",decimals:"dapp-core-component__formatAmountStyles__decimals",symbol:"dapp-core-component__formatAmountStyles__symbol"};var Kp=e=>pe.default.createElement("span",{"data-testid":e["data-testid"]?e["data-testid"]:"formatAmountComponent",className:e.className},pe.default.createElement("span",{className:Ke["int-amount"],"data-testid":"formatAmountInt"},"...")),Xp=(e,o)=>{var f;let{value:n,showLastNonZeroDecimal:t=!1,showLabel:a=!0}=e,i=e.digits!=null?e.digits:ve,_=e.decimals!=null?e.decimals:D,l=Q({input:n,decimals:_,digits:i,showLastNonZeroDecimal:t,addCommas:!0}),d=l.split("."),u=d.length===1,h=l!==z;if(i>0&&u&&h){let w="";for(let k=1;k<=i;k++)w=w+z;d.push(w)}return pe.default.createElement("span",{"data-testid":e["data-testid"]?e["data-testid"]:"formatAmountComponent",className:e.className},pe.default.createElement("span",{className:Ke["int-amount"],"data-testid":"formatAmountInt"},d[0]),d.length>1&&pe.default.createElement("span",{className:Ke.decimals,"data-testid":"formatAmountDecimals"},".",d[1]),a&&pe.default.createElement("span",{className:(0,ya.default)(Ke.symbol,e.token&&g.textMuted),"data-testid":"formatAmountSymbol"},` ${(f=e.token)!=null?f:o}`))},Zp=e=>{let{value:o}=e;return Ge(o)?Xp(e,e.egldLabel||""):Kp(e)},fe=e=>{let o=e.egldLabel||yt,n=S(b({},e),{egldLabel:o});return pe.default.createElement(Zp,b({},n))};r();r();var Fn=T(require("react")),Sa=T(require("classnames"));r();var E=T(require("react")),ba=T(require("classnames")),Ta=T(require("lodash.debounce"));r();var Yp=`.dapp-core-component__trim-styles__trim {
  display: flex;
  max-width: 100%;
  overflow: hidden;
  position: relative;
  white-space: nowrap;
  /* IE fix */
  /* SAFARI fix */
}
.dapp-core-component__trim-styles__trim.dapp-core-component__trim-styles__overflow .dapp-core-component__trim-styles__ellipsis {
  display: block;
}
.dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__left {
  flex-shrink: 1;
  font-size: 1px;
  overflow: hidden;
  text-overflow: ellipsis;
  text-align: left;
}
.dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__right {
  flex-shrink: 1;
  font-size: 1px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  direction: rtl;
  text-align: right;
}
.dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__left span,
.dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__right span {
  font-size: 0.875rem;
  pointer-events: none;
  user-select: none;
}
.dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__ellipsis {
  flex-shrink: 0;
  display: none;
  pointer-events: none;
  user-select: none;
}
@media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
  .dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__right {
    text-overflow: clip;
  }
}
@supports (-webkit-hyphens: none) {
  .dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__right,
.dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__left {
    letter-spacing: -0.001em;
  }
  .dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__right {
    text-overflow: clip;
  }
}
.dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__hidden-text-ref {
  position: absolute;
  display: block;
  color: transparent;
}
@media (max-width: 1199.98px) {
  .dapp-core-component__trim-styles__trim {
    max-width: 26rem;
  }
}
@media (max-width: 991.98px) {
  .dapp-core-component__trim-styles__trim {
    max-width: 12rem;
  }
}
@media (max-width: 768px) {
  .dapp-core-component__trim-styles__trim {
    max-width: 8rem;
  }
}

.dapp-core-component__trim-styles__trim-wrapper {
  display: flex;
  max-width: 100%;
  overflow: hidden;
}

a:hover > .dapp-core-component__trim-styles__trim span {
  color: #0056b3 !important;
}
a:hover > .dapp-core-component__trim-styles__trim span.dapp-core-component__trim-styles__hidden-text-ref {
  color: transparent !important;
}

a > .dapp-core-component__trim-styles__trim span,
.dapp-core-component__trim-styles__text-primary > .dapp-core-component__trim-styles__trim span {
  color: #1b46c2 !important;
}
a > .dapp-core-component__trim-styles__trim span.dapp-core-component__trim-styles__hidden-text-ref,
.dapp-core-component__trim-styles__text-primary > .dapp-core-component__trim-styles__trim span.dapp-core-component__trim-styles__hidden-text-ref {
  color: transparent !important;
}

.dapp-core-component__trim-styles__table .dapp-core-component__trim-styles__trim {
  max-width: 10rem;
}

.dapp-core-component__trim-styles__table .dapp-core-component__trim-styles__trim-only-sm .dapp-core-component__trim-styles__trim {
  max-width: none;
}
@media (max-width: 768px) {
  .dapp-core-component__trim-styles__table .dapp-core-component__trim-styles__trim-only-sm .dapp-core-component__trim-styles__trim {
    max-width: 13rem;
  }
}

.dapp-core-component__trim-styles__trim-fs-sm .dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__left span,
.dapp-core-component__trim-styles__trim-fs-sm .dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__right span,
.dapp-core-component__trim-styles__trim-fs-sm .dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__ellipsis {
  font-size: 0.875rem;
}

@media (max-width: 768px) {
  .dapp-core-component__trim-styles__table .dapp-core-component__trim-styles__trim-size-xl .dapp-core-component__trim-styles__trim {
    max-width: 13rem;
  }
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Yp));var he={trim:"dapp-core-component__trim-styles__trim",overflow:"dapp-core-component__trim-styles__overflow",ellipsis:"dapp-core-component__trim-styles__ellipsis",left:"dapp-core-component__trim-styles__left",right:"dapp-core-component__trim-styles__right","hidden-text-ref":"dapp-core-component__trim-styles__hidden-text-ref",hiddenTextRef:"dapp-core-component__trim-styles__hidden-text-ref","trim-wrapper":"dapp-core-component__trim-styles__trim-wrapper",trimWrapper:"dapp-core-component__trim-styles__trim-wrapper","text-primary":"dapp-core-component__trim-styles__text-primary",textPrimary:"dapp-core-component__trim-styles__text-primary",table:"dapp-core-component__trim-styles__table","trim-only-sm":"dapp-core-component__trim-styles__trim-only-sm",trimOnlySm:"dapp-core-component__trim-styles__trim-only-sm","trim-fs-sm":"dapp-core-component__trim-styles__trim-fs-sm",trimFsSm:"dapp-core-component__trim-styles__trim-fs-sm","trim-size-xl":"dapp-core-component__trim-styles__trim-size-xl",trimSizeXl:"dapp-core-component__trim-styles__trim-size-xl"};var wa=({text:e,className:o="dapp-trim",dataTestId:n="trim-text-component",color:t})=>{let[a,i]=(0,E.useState)(!1),_=(0,E.useRef)(document.createElement("span")),l=(0,E.useRef)(document.createElement("span")),d=(0,E.useCallback)((0,Ta.default)(()=>{if(_.current&&l.current){let h=l.current.offsetWidth-_.current.offsetWidth;i(h>1)}},300),[]);return(0,E.useEffect)(()=>(window.addEventListener("resize",d),()=>{window.removeEventListener("resize",d)})),(0,E.useEffect)(()=>{d()},[]),E.default.createElement("span",{ref:_,className:(0,ba.default)(he.trim,t!=null?t:"",{[he.overflow]:a},o),"data-testid":n},E.default.createElement("span",{ref:l,className:he.hiddenTextRef},e),a?E.default.createElement(E.default.Fragment,null,E.default.createElement("span",{className:he.left},E.default.createElement("span",null,String(e).substring(0,Math.floor(e.length/2)))),E.default.createElement("span",{className:he.ellipsis},"..."),E.default.createElement("span",{className:he.right},E.default.createElement("span",null,String(e).substring(Math.ceil(e.length/2))))):E.default.createElement("span",null,e))};var ka=a=>{var i=a,{address:e,assets:o,color:n}=i,t=Te(i,["address","assets","color"]);if(o&&o.name){let _=o.name.replace(new RegExp("\\p{Emoji}","gu"),""),l=`${_} (${e})`;return Fn.default.createElement("span",S(b({className:(0,Sa.default)(g.textTruncate,{[`text-${n}`]:n})},t),{title:l}),_)}return Fn.default.createElement(wa,b({text:e,color:n},t))};r();var va=T(require("react")),Aa=T(require("classnames")),Ia=require("@fortawesome/free-solid-svg-icons/faFileAlt"),Ea=require("@fortawesome/react-fontawesome");var Na=({initiator:e,secondInitiator:o})=>ko(e)||ko(o!=null?o:"")?va.default.createElement(Ea.FontAwesomeIcon,{title:"Smart Contract",icon:Ia.faFileAlt,className:(0,Aa.default)(g.mr1,g.textSecondary),"data-testid":"scIcon"}):null;r();var B=T(require("react"));r();r();var Xe=T(require("react")),Lo=T(require("classnames"));r();var Jp=`.dapp-core-component__styles__transaction-action-collection {
  color: #1f43f4 !important;
  display: inline-flex;
  margin-left: 0 !important;
}
.dapp-core-component__styles__transaction-action-collection:hover {
  text-decoration: none;
}

.dapp-core-component__styles__transaction-action-nft {
  display: contents;
}
.dapp-core-component__styles__transaction-action-nft .dapp-core-component__styles__explorer {
  margin-left: 0;
  color: #1f43f4 !important;
}
.dapp-core-component__styles__transaction-action-nft .dapp-core-component__styles__explorer:hover {
  text-decoration: none;
}
.dapp-core-component__styles__transaction-action-nft .dapp-core-component__styles__explorer .dapp-core-component__styles__data {
  margin: 0;
  margin-right: 0.25rem;
}
.dapp-core-component__styles__transaction-action-nft .dapp-core-component__styles__truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Jp));var xe={"transaction-action-collection":"dapp-core-component__styles__transaction-action-collection",transactionActionCollection:"dapp-core-component__styles__transaction-action-collection","transaction-action-nft":"dapp-core-component__styles__transaction-action-nft",transactionActionNft:"dapp-core-component__styles__transaction-action-nft",explorer:"dapp-core-component__styles__explorer",data:"dapp-core-component__styles__data",truncate:"dapp-core-component__styles__truncate"};var La=({token:e})=>e.collection?Xe.default.createElement(ee,{page:q.collectionDetails(e.collection),className:(0,Lo.default)(xe.transactionActionCollection)},Xe.default.createElement("div",{className:(0,Lo.default)(g.dFlex,g.alignItemsCenter)},e.svgUrl&&Xe.default.createElement("img",{src:e.svgUrl,alt:e.name,className:(0,Lo.default)(g.sideIcon,g.mr1)}),Xe.default.createElement("span",null,e.ticker))):null;r();var oe=T(require("react")),Me=T(require("classnames"));r();var Da=T(require("react")),Ca=T(require("classnames"));var Fa=({text:e,className:o,"data-testid":n="nftBadge"})=>Da.default.createElement("div",{"data-testid":n,className:(0,Ca.default)(g.badge,g.badgeSecondary,g.badgePill,g.fontWeightLight,o)},e);var Pa=({badgeText:e,tokenFormattedAmount:o,tokenExplorerLink:n,tokenLinkText:t,token:a,showLastNonZeroDecimal:i})=>a.identifier?oe.default.createElement("div",{className:xe.transactionActionNft},e!=null&&oe.default.createElement(Fa,{text:e,className:(0,Me.default)(g.mr1,g.myAuto)}),o!=null&&oe.default.createElement("div",{className:(0,Me.default)(g.mr1,{[g.textTruncate]:a.svgUrl})},oe.default.createElement(fe,{value:a.value,digits:2,showLabel:!1,showLastNonZeroDecimal:i,decimals:a.decimals,"data-testid":"nftFormattedAmount"})),oe.default.createElement(ee,{page:n,"data-testid":"nftExplorerLink",className:(0,Me.default)(xe.explorer,{[g.sideLink]:a.svgUrl,[g.dFlex]:a.svgUrl,[g.textTruncate]:!a.svgUrl})},oe.default.createElement("div",{className:xe.data},a.svgUrl&&oe.default.createElement("img",{src:a.svgUrl,alt:a.name,className:(0,Me.default)(g.sideIcon,g.mr1)}),oe.default.createElement("span",{className:(0,Me.default)({[xe.truncate]:a.ticker===a.collection})},t)))):null;r();var ne=T(require("react")),Do=T(require("classnames"));I();var Ma=({tokenExplorerLink:e,showFormattedAmount:o,tokenLinkText:n,token:t,showLastNonZeroDecimal:a})=>{var i;return t.token?ne.default.createElement(ne.default.Fragment,null,o&&ne.default.createElement("div",{className:g.textTruncate},ne.default.createElement(fe,{value:t.value,digits:2,showLabel:!1,decimals:(i=t.decimals)!=null?i:D,showLastNonZeroDecimal:a,"data-testid":"tokenFormattedAmount"})),ne.default.createElement(ee,{page:e,"data-testid":"tokenExplorerLink",className:(0,Do.default)(g.dFlex,{[g.sideLink]:t.svgUrl})},ne.default.createElement("div",{className:(0,Do.default)(g.dFlex,g.alignItemsCenter)},t.svgUrl&&ne.default.createElement("img",{src:t.svgUrl,alt:t.name,className:(0,Do.default)(g.sideIcon,g.mr1)}),ne.default.createElement("span",null,n)))):null};var ye={Collection:La,Nft:Pa,Token:Ma};r();var Qp=`.dapp-core-component__styles__group {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Qp));var Pn={group:"dapp-core-component__styles__group"};var Mn=({token:e,noValue:o,showLastNonZeroDecimal:n})=>{if(["MetaESDT".toString(),"SemiFungibleESDT","NonFungibleESDT"].includes(e.type))switch(e.type){case"SemiFungibleESDT":{let a=Pe({token:e,showLastNonZeroDecimal:n});return B.default.createElement("div",{className:Pn.group},B.default.createElement("span",{className:g.mr1},"SFT quantity"),B.default.createElement(ye.Nft,S(b({},a),{badgeText:null})),B.default.createElement("span",{className:g.mr1},"of collection"),B.default.createElement(ye.Collection,{token:e}))}case"NonFungibleESDT":{let a=Pe({token:e,noValue:o,showLastNonZeroDecimal:n});return B.default.createElement("div",{className:Pn.group},B.default.createElement("span",{className:g.mr1},"NFT"),B.default.createElement(ye.Nft,S(b({},a),{badgeText:null})),B.default.createElement("span",{className:g.mr1},"of collection"),B.default.createElement(ye.Collection,{token:e}))}case"MetaESDT":{let a=Pe({token:e,noValue:o,showLastNonZeroDecimal:n});return B.default.createElement(ye.Nft,S(b({},a),{badgeText:null}))}default:return null}else{let a=Ao({token:e,noValue:o,showLastNonZeroDecimal:n});return B.default.createElement(ye.Token,S(b({},a),{showLastNonZeroDecimal:!0}))}};r();var ec=`.dapp-core-component__styles__address {
  display: flex;
  align-items: center;
}
.dapp-core-component__styles__address .dapp-core-component__styles__explorer {
  margin-left: 0 !important;
  position: relative;
  display: flex;
  max-width: 100%;
  overflow: hidden;
  color: #1f43f4 !important;
}
.dapp-core-component__styles__address .dapp-core-component__styles__explorer:hover {
  text-decoration: none;
}
.dapp-core-component__styles__address .dapp-core-component__styles__explorer span {
  max-width: none;
}
.dapp-core-component__styles__address .dapp-core-component__styles__explorer span [class*=left] span,
.dapp-core-component__styles__address .dapp-core-component__styles__explorer span [class*=right] span {
  font-size: 1rem;
  color: #1f43f4 !important;
}
.dapp-core-component__styles__address .dapp-core-component__styles__explorer span:last-child {
  color: #1f43f4 !important;
}

.dapp-core-component__styles__token {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
}
@media (min-width: 992px) {
  .dapp-core-component__styles__token.dapp-core-component__styles__spaced {
    margin-right: 0.25rem;
  }
}
.dapp-core-component__styles__token .dapp-core-component__styles__comma {
  display: none;
}
@media (min-width: 992px) {
  .dapp-core-component__styles__token .dapp-core-component__styles__comma {
    display: block;
  }
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(ec));var Re={address:"dapp-core-component__styles__address",explorer:"dapp-core-component__styles__explorer",token:"dapp-core-component__styles__token",spaced:"dapp-core-component__styles__spaced",comma:"dapp-core-component__styles__comma"};var Ra=({entry:e,transaction:o})=>{if(typeof e=="string")return v.default.createElement("span",{className:g.mr1},e.replace("eGLD","EGLD"));if(Boolean(e.address)){let n;return e.address===o.sender&&(n=o.senderAssets),e.address===o.receiver&&(n=o.receiverAssets),Ce(e.address)?v.default.createElement("div",{className:Re.address},v.default.createElement(Na,{initiator:e.address}),v.default.createElement(ee,{page:q.accountDetails(e.address),"data-testid":"receiverLink",className:Re.explorer},v.default.createElement(ka,{address:e.address,assets:n}))):""}if(Boolean(e.token&&e.token.length>0))return e.token.map((n,t)=>v.default.createElement("div",{key:`tx-${n.identifier}-${t}`,className:(0,be.default)(Re.token,{[Re.spaced]:e.token.length>1})},v.default.createElement(Mn,{token:n,showLastNonZeroDecimal:!0}),t<e.token.length-1&&v.default.createElement("span",{className:Re.comma},",")));if(Boolean(e.tokenNoValue&&e.tokenNoValue.length>0))return e.tokenNoValue.map((n,t)=>v.default.createElement("div",{key:`tx-${n.token}-${t}`},v.default.createElement(Mn,{token:n,noValue:!0,showLastNonZeroDecimal:!0}),t<e.tokenNoValue.length-1&&v.default.createElement("span",{className:(0,be.default)(g.mlN1,g.mr1,g.dNone,g.dSmFlex)},",")));if(Boolean(e.tokenNoLink&&e.tokenNoLink.length>0))return e.tokenNoLink.map((n,t)=>v.default.createElement("div",{key:`tx-${n.token}-${t}`},v.default.createElement("span",{className:g.mr1},n.ticker),t<e.tokenNoLink.length-1&&v.default.createElement("span",{className:(0,be.default)(g.mlN1,g.mr1,g.dNone,g.dSmFlex)},",")));if(Boolean(e.value))return v.default.createElement("span",null,v.default.createElement(fe,{value:e.value,showLabel:!1,showLastNonZeroDecimal:!0}));if(Boolean(e.egldValue))return v.default.createElement("span",null,v.default.createElement(fe,{value:e.egldValue,showLastNonZeroDecimal:!0}));if(Boolean(e.providerName)){let n=(0,be.default)(g.sideIcon,g.mr1,g.mrLg1,g.roundedCircle);return v.default.createElement("span",{className:(0,be.default)(g.dFlex,g.mr1)},v.default.createElement(ee,{page:q.providerDetails(o.receiver),className:(0,be.default)(g.dFlex,g.alignSelfCenter)},e.providerAvatar?v.default.createElement("img",{src:e.providerAvatar,className:n,alt:" "}):v.default.createElement(Nt,{className:n}),e.providerName))}return null};r();var oc=`.dapp-core-component__styles__action {
  display: flex;
  flex-direction: column;
}
@media (min-width: 992px) {
  .dapp-core-component__styles__action {
    flex-wrap: wrap;
    flex-direction: row;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__styles__action .dapp-core-component__styles__result {
    flex-direction: row;
    align-items: center;
    display: contents;
    margin-right: 0.25rem;
  }
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(oc));var Rn={action:"dapp-core-component__styles__action",result:"dapp-core-component__styles__result"};var nc=({transaction:e})=>{let o=(0,Be.useMemo)(()=>e.action?Et(e.action):[],[e.action]);return Be.default.createElement("div",{"data-testid":"transactionAction",className:Rn.action},o.map((n,t)=>Be.default.createElement("div",{key:JSON.stringify(o)+t,className:Rn.result},Be.default.createElement(Ra,{transaction:e,entry:n}))))};0&&(module.exports={TransactionAction});
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
