// @elrondnetwork/dapp-core/__commonjs/hooks/transactions/index.js
"use strict";var Oi=Object.create;var Je=Object.defineProperty,Bi=Object.defineProperties,Ui=Object.getOwnPropertyDescriptor,$i=Object.getOwnPropertyDescriptors,Gi=Object.getOwnPropertyNames,bo=Object.getOwnPropertySymbols,Vi=Object.getPrototypeOf,un=Object.prototype.hasOwnProperty,Ot=Object.prototype.propertyIsEnumerable;var Ft=(e,o,n)=>o in e?Je(e,o,{enumerable:!0,configurable:!0,writable:!0,value:n}):e[o]=n,T=(e,o)=>{for(var n in o||(o={}))un.call(o,n)&&Ft(e,n,o[n]);if(bo)for(var n of bo(o))Ot.call(o,n)&&Ft(e,n,o[n]);return e},I=(e,o)=>Bi(e,$i(o));var gn=(e,o)=>{var n={};for(var r in e)un.call(e,r)&&o.indexOf(r)<0&&(n[r]=e[r]);if(e!=null&&bo)for(var r of bo(e))o.indexOf(r)<0&&Ot.call(e,r)&&(n[r]=e[r]);return n};var b=(e,o)=>()=>(e&&(o=e(e=0)),o);var So=(e,o)=>()=>(o||e((o={exports:{}}).exports,o),o.exports),be=(e,o)=>{for(var n in o)Je(e,n,{get:o[n],enumerable:!0})},Bt=(e,o,n,r)=>{if(o&&typeof o=="object"||typeof o=="function")for(let a of Gi(o))!un.call(e,a)&&a!==n&&Je(e,a,{get:()=>o[a],enumerable:!(r=Ui(o,a))||r.enumerable});return e};var v=(e,o,n)=>(n=e!=null?Oi(Vi(e)):{},Bt(o||!e||!e.__esModule?Je(n,"default",{value:e,enumerable:!0}):n,e)),Qe=e=>Bt(Je({},"__esModule",{value:!0}),e);var w=(e,o,n)=>new Promise((r,a)=>{var m=l=>{try{d(n.next(l))}catch(u){a(u)}},s=l=>{try{d(n.throw(l))}catch(u){a(u)}},d=l=>l.done?r(l.value):Promise.resolve(l.value).then(m,s);d((n=n.apply(e,o)).next())});var Gt=So(wo=>{"use strict";t();wo.byteLength=Hi;wo.toByteArray=zi;wo.fromByteArray=Xi;var J=[],z=[],Wi=typeof Uint8Array!="undefined"?Uint8Array:Array,fn="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";for(Se=0,Ut=fn.length;Se<Ut;++Se)J[Se]=fn[Se],z[fn.charCodeAt(Se)]=Se;var Se,Ut;z["-".charCodeAt(0)]=62;z["_".charCodeAt(0)]=63;function $t(e){var o=e.length;if(o%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var n=e.indexOf("=");n===-1&&(n=o);var r=n===o?0:4-n%4;return[n,r]}function Hi(e){var o=$t(e),n=o[0],r=o[1];return(n+r)*3/4-r}function ji(e,o,n){return(o+n)*3/4-n}function zi(e){var o,n=$t(e),r=n[0],a=n[1],m=new Wi(ji(e,r,a)),s=0,d=a>0?r-4:r,l;for(l=0;l<d;l+=4)o=z[e.charCodeAt(l)]<<18|z[e.charCodeAt(l+1)]<<12|z[e.charCodeAt(l+2)]<<6|z[e.charCodeAt(l+3)],m[s++]=o>>16&255,m[s++]=o>>8&255,m[s++]=o&255;return a===2&&(o=z[e.charCodeAt(l)]<<2|z[e.charCodeAt(l+1)]>>4,m[s++]=o&255),a===1&&(o=z[e.charCodeAt(l)]<<10|z[e.charCodeAt(l+1)]<<4|z[e.charCodeAt(l+2)]>>2,m[s++]=o>>8&255,m[s++]=o&255),m}function qi(e){return J[e>>18&63]+J[e>>12&63]+J[e>>6&63]+J[e&63]}function Ki(e,o,n){for(var r,a=[],m=o;m<n;m+=3)r=(e[m]<<16&16711680)+(e[m+1]<<8&65280)+(e[m+2]&255),a.push(qi(r));return a.join("")}function Xi(e){for(var o,n=e.length,r=n%3,a=[],m=16383,s=0,d=n-r;s<d;s+=m)a.push(Ki(e,s,s+m>d?d:s+m));return r===1?(o=e[n-1],a.push(J[o>>2]+J[o<<4&63]+"==")):r===2&&(o=(e[n-2]<<8)+e[n-1],a.push(J[o>>10]+J[o>>4&63]+J[o<<2&63]+"=")),a.join("")}});var Vt=So(hn=>{t();hn.read=function(e,o,n,r,a){var m,s,d=a*8-r-1,l=(1<<d)-1,u=l>>1,f=-7,g=n?a-1:0,y=n?-1:1,h=e[o+g];for(g+=y,m=h&(1<<-f)-1,h>>=-f,f+=d;f>0;m=m*256+e[o+g],g+=y,f-=8);for(s=m&(1<<-f)-1,m>>=-f,f+=r;f>0;s=s*256+e[o+g],g+=y,f-=8);if(m===0)m=1-u;else{if(m===l)return s?NaN:(h?-1:1)*(1/0);s=s+Math.pow(2,r),m=m-u}return(h?-1:1)*s*Math.pow(2,m-r)};hn.write=function(e,o,n,r,a,m){var s,d,l,u=m*8-a-1,f=(1<<u)-1,g=f>>1,y=a===23?Math.pow(2,-24)-Math.pow(2,-77):0,h=r?0:m-1,E=r?1:-1,L=o<0||o===0&&1/o<0?1:0;for(o=Math.abs(o),isNaN(o)||o===1/0?(d=isNaN(o)?1:0,s=f):(s=Math.floor(Math.log(o)/Math.LN2),o*(l=Math.pow(2,-s))<1&&(s--,l*=2),s+g>=1?o+=y/l:o+=y*Math.pow(2,1-g),o*l>=2&&(s++,l/=2),s+g>=f?(d=0,s=f):s+g>=1?(d=(o*l-1)*Math.pow(2,a),s=s+g):(d=o*Math.pow(2,g-1)*Math.pow(2,a),s=0));a>=8;e[n+h]=d&255,h+=E,d/=256,a-=8);for(s=s<<a|d,u+=a;u>0;e[n+h]=s&255,h+=E,s/=256,u-=8);e[n+h-E]|=L*128}});var or=So(Ce=>{"use strict";t();var Tn=Gt(),Pe=Vt(),Wt=typeof Symbol=="function"&&typeof Symbol.for=="function"?Symbol.for("nodejs.util.inspect.custom"):null;Ce.Buffer=_;Ce.SlowBuffer=oc;Ce.INSPECT_MAX_BYTES=50;var vo=2147483647;Ce.kMaxLength=vo;_.TYPED_ARRAY_SUPPORT=Yi();!_.TYPED_ARRAY_SUPPORT&&typeof console!="undefined"&&typeof console.error=="function"&&console.error("This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.");function Yi(){try{var e=new Uint8Array(1),o={foo:function(){return 42}};return Object.setPrototypeOf(o,Uint8Array.prototype),Object.setPrototypeOf(e,o),e.foo()===42}catch(n){return!1}}Object.defineProperty(_.prototype,"parent",{enumerable:!0,get:function(){if(!!_.isBuffer(this))return this.buffer}});Object.defineProperty(_.prototype,"offset",{enumerable:!0,get:function(){if(!!_.isBuffer(this))return this.byteOffset}});function ce(e){if(e>vo)throw new RangeError('The value "'+e+'" is invalid for option "size"');var o=new Uint8Array(e);return Object.setPrototypeOf(o,_.prototype),o}function _(e,o,n){if(typeof e=="number"){if(typeof o=="string")throw new TypeError('The "string" argument must be of type string. Received type number');return Sn(e)}return zt(e,o,n)}_.poolSize=8192;function zt(e,o,n){if(typeof e=="string")return Ji(e,o);if(ArrayBuffer.isView(e))return Qi(e);if(e==null)throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e);if(Q(e,ArrayBuffer)||e&&Q(e.buffer,ArrayBuffer)||typeof SharedArrayBuffer!="undefined"&&(Q(e,SharedArrayBuffer)||e&&Q(e.buffer,SharedArrayBuffer)))return yn(e,o,n);if(typeof e=="number")throw new TypeError('The "value" argument must not be of type number. Received type number');var r=e.valueOf&&e.valueOf();if(r!=null&&r!==e)return _.from(r,o,n);var a=ec(e);if(a)return a;if(typeof Symbol!="undefined"&&Symbol.toPrimitive!=null&&typeof e[Symbol.toPrimitive]=="function")return _.from(e[Symbol.toPrimitive]("string"),o,n);throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e)}_.from=function(e,o,n){return zt(e,o,n)};Object.setPrototypeOf(_.prototype,Uint8Array.prototype);Object.setPrototypeOf(_,Uint8Array);function qt(e){if(typeof e!="number")throw new TypeError('"size" argument must be of type number');if(e<0)throw new RangeError('The value "'+e+'" is invalid for option "size"')}function Zi(e,o,n){return qt(e),e<=0?ce(e):o!==void 0?typeof n=="string"?ce(e).fill(o,n):ce(e).fill(o):ce(e)}_.alloc=function(e,o,n){return Zi(e,o,n)};function Sn(e){return qt(e),ce(e<0?0:wn(e)|0)}_.allocUnsafe=function(e){return Sn(e)};_.allocUnsafeSlow=function(e){return Sn(e)};function Ji(e,o){if((typeof o!="string"||o==="")&&(o="utf8"),!_.isEncoding(o))throw new TypeError("Unknown encoding: "+o);var n=Kt(e,o)|0,r=ce(n),a=r.write(e,o);return a!==n&&(r=r.slice(0,a)),r}function xn(e){for(var o=e.length<0?0:wn(e.length)|0,n=ce(o),r=0;r<o;r+=1)n[r]=e[r]&255;return n}function Qi(e){if(Q(e,Uint8Array)){var o=new Uint8Array(e);return yn(o.buffer,o.byteOffset,o.byteLength)}return xn(e)}function yn(e,o,n){if(o<0||e.byteLength<o)throw new RangeError('"offset" is outside of buffer bounds');if(e.byteLength<o+(n||0))throw new RangeError('"length" is outside of buffer bounds');var r;return o===void 0&&n===void 0?r=new Uint8Array(e):n===void 0?r=new Uint8Array(e,o):r=new Uint8Array(e,o,n),Object.setPrototypeOf(r,_.prototype),r}function ec(e){if(_.isBuffer(e)){var o=wn(e.length)|0,n=ce(o);return n.length===0||e.copy(n,0,0,o),n}if(e.length!==void 0)return typeof e.length!="number"||vn(e.length)?ce(0):xn(e);if(e.type==="Buffer"&&Array.isArray(e.data))return xn(e.data)}function wn(e){if(e>=vo)throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x"+vo.toString(16)+" bytes");return e|0}function oc(e){return+e!=e&&(e=0),_.alloc(+e)}_.isBuffer=function(o){return o!=null&&o._isBuffer===!0&&o!==_.prototype};_.compare=function(o,n){if(Q(o,Uint8Array)&&(o=_.from(o,o.offset,o.byteLength)),Q(n,Uint8Array)&&(n=_.from(n,n.offset,n.byteLength)),!_.isBuffer(o)||!_.isBuffer(n))throw new TypeError('The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array');if(o===n)return 0;for(var r=o.length,a=n.length,m=0,s=Math.min(r,a);m<s;++m)if(o[m]!==n[m]){r=o[m],a=n[m];break}return r<a?-1:a<r?1:0};_.isEncoding=function(o){switch(String(o).toLowerCase()){case"hex":case"utf8":case"utf-8":case"ascii":case"latin1":case"binary":case"base64":case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return!0;default:return!1}};_.concat=function(o,n){if(!Array.isArray(o))throw new TypeError('"list" argument must be an Array of Buffers');if(o.length===0)return _.alloc(0);var r;if(n===void 0)for(n=0,r=0;r<o.length;++r)n+=o[r].length;var a=_.allocUnsafe(n),m=0;for(r=0;r<o.length;++r){var s=o[r];if(Q(s,Uint8Array))m+s.length>a.length?_.from(s).copy(a,m):Uint8Array.prototype.set.call(a,s,m);else if(_.isBuffer(s))s.copy(a,m);else throw new TypeError('"list" argument must be an Array of Buffers');m+=s.length}return a};function Kt(e,o){if(_.isBuffer(e))return e.length;if(ArrayBuffer.isView(e)||Q(e,ArrayBuffer))return e.byteLength;if(typeof e!="string")throw new TypeError('The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type '+typeof e);var n=e.length,r=arguments.length>2&&arguments[2]===!0;if(!r&&n===0)return 0;for(var a=!1;;)switch(o){case"ascii":case"latin1":case"binary":return n;case"utf8":case"utf-8":return bn(e).length;case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return n*2;case"hex":return n>>>1;case"base64":return er(e).length;default:if(a)return r?-1:bn(e).length;o=(""+o).toLowerCase(),a=!0}}_.byteLength=Kt;function nc(e,o,n){var r=!1;if((o===void 0||o<0)&&(o=0),o>this.length||((n===void 0||n>this.length)&&(n=this.length),n<=0)||(n>>>=0,o>>>=0,n<=o))return"";for(e||(e="utf8");;)switch(e){case"hex":return dc(this,o,n);case"utf8":case"utf-8":return Yt(this,o,n);case"ascii":return mc(this,o,n);case"latin1":case"binary":return _c(this,o,n);case"base64":return pc(this,o,n);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return lc(this,o,n);default:if(r)throw new TypeError("Unknown encoding: "+e);e=(e+"").toLowerCase(),r=!0}}_.prototype._isBuffer=!0;function we(e,o,n){var r=e[o];e[o]=e[n],e[n]=r}_.prototype.swap16=function(){var o=this.length;if(o%2!==0)throw new RangeError("Buffer size must be a multiple of 16-bits");for(var n=0;n<o;n+=2)we(this,n,n+1);return this};_.prototype.swap32=function(){var o=this.length;if(o%4!==0)throw new RangeError("Buffer size must be a multiple of 32-bits");for(var n=0;n<o;n+=4)we(this,n,n+3),we(this,n+1,n+2);return this};_.prototype.swap64=function(){var o=this.length;if(o%8!==0)throw new RangeError("Buffer size must be a multiple of 64-bits");for(var n=0;n<o;n+=8)we(this,n,n+7),we(this,n+1,n+6),we(this,n+2,n+5),we(this,n+3,n+4);return this};_.prototype.toString=function(){var o=this.length;return o===0?"":arguments.length===0?Yt(this,0,o):nc.apply(this,arguments)};_.prototype.toLocaleString=_.prototype.toString;_.prototype.equals=function(o){if(!_.isBuffer(o))throw new TypeError("Argument must be a Buffer");return this===o?!0:_.compare(this,o)===0};_.prototype.inspect=function(){var o="",n=Ce.INSPECT_MAX_BYTES;return o=this.toString("hex",0,n).replace(/(.{2})/g,"$1 ").trim(),this.length>n&&(o+=" ... "),"<Buffer "+o+">"};Wt&&(_.prototype[Wt]=_.prototype.inspect);_.prototype.compare=function(o,n,r,a,m){if(Q(o,Uint8Array)&&(o=_.from(o,o.offset,o.byteLength)),!_.isBuffer(o))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+typeof o);if(n===void 0&&(n=0),r===void 0&&(r=o?o.length:0),a===void 0&&(a=0),m===void 0&&(m=this.length),n<0||r>o.length||a<0||m>this.length)throw new RangeError("out of range index");if(a>=m&&n>=r)return 0;if(a>=m)return-1;if(n>=r)return 1;if(n>>>=0,r>>>=0,a>>>=0,m>>>=0,this===o)return 0;for(var s=m-a,d=r-n,l=Math.min(s,d),u=this.slice(a,m),f=o.slice(n,r),g=0;g<l;++g)if(u[g]!==f[g]){s=u[g],d=f[g];break}return s<d?-1:d<s?1:0};function Xt(e,o,n,r,a){if(e.length===0)return-1;if(typeof n=="string"?(r=n,n=0):n>2147483647?n=2147483647:n<-2147483648&&(n=-2147483648),n=+n,vn(n)&&(n=a?0:e.length-1),n<0&&(n=e.length+n),n>=e.length){if(a)return-1;n=e.length-1}else if(n<0)if(a)n=0;else return-1;if(typeof o=="string"&&(o=_.from(o,r)),_.isBuffer(o))return o.length===0?-1:Ht(e,o,n,r,a);if(typeof o=="number")return o=o&255,typeof Uint8Array.prototype.indexOf=="function"?a?Uint8Array.prototype.indexOf.call(e,o,n):Uint8Array.prototype.lastIndexOf.call(e,o,n):Ht(e,[o],n,r,a);throw new TypeError("val must be string, number or Buffer")}function Ht(e,o,n,r,a){var m=1,s=e.length,d=o.length;if(r!==void 0&&(r=String(r).toLowerCase(),r==="ucs2"||r==="ucs-2"||r==="utf16le"||r==="utf-16le")){if(e.length<2||o.length<2)return-1;m=2,s/=2,d/=2,n/=2}function l(h,E){return m===1?h[E]:h.readUInt16BE(E*m)}var u;if(a){var f=-1;for(u=n;u<s;u++)if(l(e,u)===l(o,f===-1?0:u-f)){if(f===-1&&(f=u),u-f+1===d)return f*m}else f!==-1&&(u-=u-f),f=-1}else for(n+d>s&&(n=s-d),u=n;u>=0;u--){for(var g=!0,y=0;y<d;y++)if(l(e,u+y)!==l(o,y)){g=!1;break}if(g)return u}return-1}_.prototype.includes=function(o,n,r){return this.indexOf(o,n,r)!==-1};_.prototype.indexOf=function(o,n,r){return Xt(this,o,n,r,!0)};_.prototype.lastIndexOf=function(o,n,r){return Xt(this,o,n,r,!1)};function tc(e,o,n,r){n=Number(n)||0;var a=e.length-n;r?(r=Number(r),r>a&&(r=a)):r=a;var m=o.length;r>m/2&&(r=m/2);for(var s=0;s<r;++s){var d=parseInt(o.substr(s*2,2),16);if(vn(d))return s;e[n+s]=d}return s}function rc(e,o,n,r){return ko(bn(o,e.length-n),e,n,r)}function ac(e,o,n,r){return ko(fc(o),e,n,r)}function ic(e,o,n,r){return ko(er(o),e,n,r)}function cc(e,o,n,r){return ko(hc(o,e.length-n),e,n,r)}_.prototype.write=function(o,n,r,a){if(n===void 0)a="utf8",r=this.length,n=0;else if(r===void 0&&typeof n=="string")a=n,r=this.length,n=0;else if(isFinite(n))n=n>>>0,isFinite(r)?(r=r>>>0,a===void 0&&(a="utf8")):(a=r,r=void 0);else throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");var m=this.length-n;if((r===void 0||r>m)&&(r=m),o.length>0&&(r<0||n<0)||n>this.length)throw new RangeError("Attempt to write outside buffer bounds");a||(a="utf8");for(var s=!1;;)switch(a){case"hex":return tc(this,o,n,r);case"utf8":case"utf-8":return rc(this,o,n,r);case"ascii":case"latin1":case"binary":return ac(this,o,n,r);case"base64":return ic(this,o,n,r);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return cc(this,o,n,r);default:if(s)throw new TypeError("Unknown encoding: "+a);a=(""+a).toLowerCase(),s=!0}};_.prototype.toJSON=function(){return{type:"Buffer",data:Array.prototype.slice.call(this._arr||this,0)}};function pc(e,o,n){return o===0&&n===e.length?Tn.fromByteArray(e):Tn.fromByteArray(e.slice(o,n))}function Yt(e,o,n){n=Math.min(e.length,n);for(var r=[],a=o;a<n;){var m=e[a],s=null,d=m>239?4:m>223?3:m>191?2:1;if(a+d<=n){var l,u,f,g;switch(d){case 1:m<128&&(s=m);break;case 2:l=e[a+1],(l&192)===128&&(g=(m&31)<<6|l&63,g>127&&(s=g));break;case 3:l=e[a+1],u=e[a+2],(l&192)===128&&(u&192)===128&&(g=(m&15)<<12|(l&63)<<6|u&63,g>2047&&(g<55296||g>57343)&&(s=g));break;case 4:l=e[a+1],u=e[a+2],f=e[a+3],(l&192)===128&&(u&192)===128&&(f&192)===128&&(g=(m&15)<<18|(l&63)<<12|(u&63)<<6|f&63,g>65535&&g<1114112&&(s=g))}}s===null?(s=65533,d=1):s>65535&&(s-=65536,r.push(s>>>10&1023|55296),s=56320|s&1023),r.push(s),a+=d}return sc(r)}var jt=4096;function sc(e){var o=e.length;if(o<=jt)return String.fromCharCode.apply(String,e);for(var n="",r=0;r<o;)n+=String.fromCharCode.apply(String,e.slice(r,r+=jt));return n}function mc(e,o,n){var r="";n=Math.min(e.length,n);for(var a=o;a<n;++a)r+=String.fromCharCode(e[a]&127);return r}function _c(e,o,n){var r="";n=Math.min(e.length,n);for(var a=o;a<n;++a)r+=String.fromCharCode(e[a]);return r}function dc(e,o,n){var r=e.length;(!o||o<0)&&(o=0),(!n||n<0||n>r)&&(n=r);for(var a="",m=o;m<n;++m)a+=Tc[e[m]];return a}function lc(e,o,n){for(var r=e.slice(o,n),a="",m=0;m<r.length-1;m+=2)a+=String.fromCharCode(r[m]+r[m+1]*256);return a}_.prototype.slice=function(o,n){var r=this.length;o=~~o,n=n===void 0?r:~~n,o<0?(o+=r,o<0&&(o=0)):o>r&&(o=r),n<0?(n+=r,n<0&&(n=0)):n>r&&(n=r),n<o&&(n=o);var a=this.subarray(o,n);return Object.setPrototypeOf(a,_.prototype),a};function F(e,o,n){if(e%1!==0||e<0)throw new RangeError("offset is not uint");if(e+o>n)throw new RangeError("Trying to access beyond buffer length")}_.prototype.readUintLE=_.prototype.readUIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||F(o,n,this.length);for(var a=this[o],m=1,s=0;++s<n&&(m*=256);)a+=this[o+s]*m;return a};_.prototype.readUintBE=_.prototype.readUIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||F(o,n,this.length);for(var a=this[o+--n],m=1;n>0&&(m*=256);)a+=this[o+--n]*m;return a};_.prototype.readUint8=_.prototype.readUInt8=function(o,n){return o=o>>>0,n||F(o,1,this.length),this[o]};_.prototype.readUint16LE=_.prototype.readUInt16LE=function(o,n){return o=o>>>0,n||F(o,2,this.length),this[o]|this[o+1]<<8};_.prototype.readUint16BE=_.prototype.readUInt16BE=function(o,n){return o=o>>>0,n||F(o,2,this.length),this[o]<<8|this[o+1]};_.prototype.readUint32LE=_.prototype.readUInt32LE=function(o,n){return o=o>>>0,n||F(o,4,this.length),(this[o]|this[o+1]<<8|this[o+2]<<16)+this[o+3]*16777216};_.prototype.readUint32BE=_.prototype.readUInt32BE=function(o,n){return o=o>>>0,n||F(o,4,this.length),this[o]*16777216+(this[o+1]<<16|this[o+2]<<8|this[o+3])};_.prototype.readIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||F(o,n,this.length);for(var a=this[o],m=1,s=0;++s<n&&(m*=256);)a+=this[o+s]*m;return m*=128,a>=m&&(a-=Math.pow(2,8*n)),a};_.prototype.readIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||F(o,n,this.length);for(var a=n,m=1,s=this[o+--a];a>0&&(m*=256);)s+=this[o+--a]*m;return m*=128,s>=m&&(s-=Math.pow(2,8*n)),s};_.prototype.readInt8=function(o,n){return o=o>>>0,n||F(o,1,this.length),this[o]&128?(255-this[o]+1)*-1:this[o]};_.prototype.readInt16LE=function(o,n){o=o>>>0,n||F(o,2,this.length);var r=this[o]|this[o+1]<<8;return r&32768?r|4294901760:r};_.prototype.readInt16BE=function(o,n){o=o>>>0,n||F(o,2,this.length);var r=this[o+1]|this[o]<<8;return r&32768?r|4294901760:r};_.prototype.readInt32LE=function(o,n){return o=o>>>0,n||F(o,4,this.length),this[o]|this[o+1]<<8|this[o+2]<<16|this[o+3]<<24};_.prototype.readInt32BE=function(o,n){return o=o>>>0,n||F(o,4,this.length),this[o]<<24|this[o+1]<<16|this[o+2]<<8|this[o+3]};_.prototype.readFloatLE=function(o,n){return o=o>>>0,n||F(o,4,this.length),Pe.read(this,o,!0,23,4)};_.prototype.readFloatBE=function(o,n){return o=o>>>0,n||F(o,4,this.length),Pe.read(this,o,!1,23,4)};_.prototype.readDoubleLE=function(o,n){return o=o>>>0,n||F(o,8,this.length),Pe.read(this,o,!0,52,8)};_.prototype.readDoubleBE=function(o,n){return o=o>>>0,n||F(o,8,this.length),Pe.read(this,o,!1,52,8)};function W(e,o,n,r,a,m){if(!_.isBuffer(e))throw new TypeError('"buffer" argument must be a Buffer instance');if(o>a||o<m)throw new RangeError('"value" argument is out of bounds');if(n+r>e.length)throw new RangeError("Index out of range")}_.prototype.writeUintLE=_.prototype.writeUIntLE=function(o,n,r,a){if(o=+o,n=n>>>0,r=r>>>0,!a){var m=Math.pow(2,8*r)-1;W(this,o,n,r,m,0)}var s=1,d=0;for(this[n]=o&255;++d<r&&(s*=256);)this[n+d]=o/s&255;return n+r};_.prototype.writeUintBE=_.prototype.writeUIntBE=function(o,n,r,a){if(o=+o,n=n>>>0,r=r>>>0,!a){var m=Math.pow(2,8*r)-1;W(this,o,n,r,m,0)}var s=r-1,d=1;for(this[n+s]=o&255;--s>=0&&(d*=256);)this[n+s]=o/d&255;return n+r};_.prototype.writeUint8=_.prototype.writeUInt8=function(o,n,r){return o=+o,n=n>>>0,r||W(this,o,n,1,255,0),this[n]=o&255,n+1};_.prototype.writeUint16LE=_.prototype.writeUInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||W(this,o,n,2,65535,0),this[n]=o&255,this[n+1]=o>>>8,n+2};_.prototype.writeUint16BE=_.prototype.writeUInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||W(this,o,n,2,65535,0),this[n]=o>>>8,this[n+1]=o&255,n+2};_.prototype.writeUint32LE=_.prototype.writeUInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||W(this,o,n,4,4294967295,0),this[n+3]=o>>>24,this[n+2]=o>>>16,this[n+1]=o>>>8,this[n]=o&255,n+4};_.prototype.writeUint32BE=_.prototype.writeUInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||W(this,o,n,4,4294967295,0),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};_.prototype.writeIntLE=function(o,n,r,a){if(o=+o,n=n>>>0,!a){var m=Math.pow(2,8*r-1);W(this,o,n,r,m-1,-m)}var s=0,d=1,l=0;for(this[n]=o&255;++s<r&&(d*=256);)o<0&&l===0&&this[n+s-1]!==0&&(l=1),this[n+s]=(o/d>>0)-l&255;return n+r};_.prototype.writeIntBE=function(o,n,r,a){if(o=+o,n=n>>>0,!a){var m=Math.pow(2,8*r-1);W(this,o,n,r,m-1,-m)}var s=r-1,d=1,l=0;for(this[n+s]=o&255;--s>=0&&(d*=256);)o<0&&l===0&&this[n+s+1]!==0&&(l=1),this[n+s]=(o/d>>0)-l&255;return n+r};_.prototype.writeInt8=function(o,n,r){return o=+o,n=n>>>0,r||W(this,o,n,1,127,-128),o<0&&(o=255+o+1),this[n]=o&255,n+1};_.prototype.writeInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||W(this,o,n,2,32767,-32768),this[n]=o&255,this[n+1]=o>>>8,n+2};_.prototype.writeInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||W(this,o,n,2,32767,-32768),this[n]=o>>>8,this[n+1]=o&255,n+2};_.prototype.writeInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||W(this,o,n,4,2147483647,-2147483648),this[n]=o&255,this[n+1]=o>>>8,this[n+2]=o>>>16,this[n+3]=o>>>24,n+4};_.prototype.writeInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||W(this,o,n,4,2147483647,-2147483648),o<0&&(o=4294967295+o+1),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};function Zt(e,o,n,r,a,m){if(n+r>e.length)throw new RangeError("Index out of range");if(n<0)throw new RangeError("Index out of range")}function Jt(e,o,n,r,a){return o=+o,n=n>>>0,a||Zt(e,o,n,4,34028234663852886e22,-34028234663852886e22),Pe.write(e,o,n,r,23,4),n+4}_.prototype.writeFloatLE=function(o,n,r){return Jt(this,o,n,!0,r)};_.prototype.writeFloatBE=function(o,n,r){return Jt(this,o,n,!1,r)};function Qt(e,o,n,r,a){return o=+o,n=n>>>0,a||Zt(e,o,n,8,17976931348623157e292,-17976931348623157e292),Pe.write(e,o,n,r,52,8),n+8}_.prototype.writeDoubleLE=function(o,n,r){return Qt(this,o,n,!0,r)};_.prototype.writeDoubleBE=function(o,n,r){return Qt(this,o,n,!1,r)};_.prototype.copy=function(o,n,r,a){if(!_.isBuffer(o))throw new TypeError("argument should be a Buffer");if(r||(r=0),!a&&a!==0&&(a=this.length),n>=o.length&&(n=o.length),n||(n=0),a>0&&a<r&&(a=r),a===r||o.length===0||this.length===0)return 0;if(n<0)throw new RangeError("targetStart out of bounds");if(r<0||r>=this.length)throw new RangeError("Index out of range");if(a<0)throw new RangeError("sourceEnd out of bounds");a>this.length&&(a=this.length),o.length-n<a-r&&(a=o.length-n+r);var m=a-r;return this===o&&typeof Uint8Array.prototype.copyWithin=="function"?this.copyWithin(n,r,a):Uint8Array.prototype.set.call(o,this.subarray(r,a),n),m};_.prototype.fill=function(o,n,r,a){if(typeof o=="string"){if(typeof n=="string"?(a=n,n=0,r=this.length):typeof r=="string"&&(a=r,r=this.length),a!==void 0&&typeof a!="string")throw new TypeError("encoding must be a string");if(typeof a=="string"&&!_.isEncoding(a))throw new TypeError("Unknown encoding: "+a);if(o.length===1){var m=o.charCodeAt(0);(a==="utf8"&&m<128||a==="latin1")&&(o=m)}}else typeof o=="number"?o=o&255:typeof o=="boolean"&&(o=Number(o));if(n<0||this.length<n||this.length<r)throw new RangeError("Out of range index");if(r<=n)return this;n=n>>>0,r=r===void 0?this.length:r>>>0,o||(o=0);var s;if(typeof o=="number")for(s=n;s<r;++s)this[s]=o;else{var d=_.isBuffer(o)?o:_.from(o,a),l=d.length;if(l===0)throw new TypeError('The value "'+o+'" is invalid for argument "value"');for(s=0;s<r-n;++s)this[s+n]=d[s%l]}return this};var uc=/[^+/0-9A-Za-z-_]/g;function gc(e){if(e=e.split("=")[0],e=e.trim().replace(uc,""),e.length<2)return"";for(;e.length%4!==0;)e=e+"=";return e}function bn(e,o){o=o||1/0;for(var n,r=e.length,a=null,m=[],s=0;s<r;++s){if(n=e.charCodeAt(s),n>55295&&n<57344){if(!a){if(n>56319){(o-=3)>-1&&m.push(239,191,189);continue}else if(s+1===r){(o-=3)>-1&&m.push(239,191,189);continue}a=n;continue}if(n<56320){(o-=3)>-1&&m.push(239,191,189),a=n;continue}n=(a-55296<<10|n-56320)+65536}else a&&(o-=3)>-1&&m.push(239,191,189);if(a=null,n<128){if((o-=1)<0)break;m.push(n)}else if(n<2048){if((o-=2)<0)break;m.push(n>>6|192,n&63|128)}else if(n<65536){if((o-=3)<0)break;m.push(n>>12|224,n>>6&63|128,n&63|128)}else if(n<1114112){if((o-=4)<0)break;m.push(n>>18|240,n>>12&63|128,n>>6&63|128,n&63|128)}else throw new Error("Invalid code point")}return m}function fc(e){for(var o=[],n=0;n<e.length;++n)o.push(e.charCodeAt(n)&255);return o}function hc(e,o){for(var n,r,a,m=[],s=0;s<e.length&&!((o-=2)<0);++s)n=e.charCodeAt(s),r=n>>8,a=n%256,m.push(a),m.push(r);return m}function er(e){return Tn.toByteArray(gc(e))}function ko(e,o,n,r){for(var a=0;a<r&&!(a+n>=o.length||a>=e.length);++a)o[a+n]=e[a];return a}function Q(e,o){return e instanceof o||e!=null&&e.constructor!=null&&e.constructor.name!=null&&e.constructor.name===o.name}function vn(e){return e!==e}var Tc=function(){for(var e="0123456789abcdef",o=new Array(256),n=0;n<16;++n)for(var r=n*16,a=0;a<16;++a)o[r+a]=e[n]+e[a];return o}()});var ir=So((Is,ar)=>{t();var R=ar.exports={},ee,oe;function kn(){throw new Error("setTimeout has not been defined")}function An(){throw new Error("clearTimeout has not been defined")}(function(){try{typeof setTimeout=="function"?ee=setTimeout:ee=kn}catch(e){ee=kn}try{typeof clearTimeout=="function"?oe=clearTimeout:oe=An}catch(e){oe=An}})();function nr(e){if(ee===setTimeout)return setTimeout(e,0);if((ee===kn||!ee)&&setTimeout)return ee=setTimeout,setTimeout(e,0);try{return ee(e,0)}catch(o){try{return ee.call(null,e,0)}catch(n){return ee.call(this,e,0)}}}function xc(e){if(oe===clearTimeout)return clearTimeout(e);if((oe===An||!oe)&&clearTimeout)return oe=clearTimeout,clearTimeout(e);try{return oe(e)}catch(o){try{return oe.call(null,e)}catch(n){return oe.call(this,e)}}}var pe=[],Re=!1,ve,Ao=-1;function yc(){!Re||!ve||(Re=!1,ve.length?pe=ve.concat(pe):Ao=-1,pe.length&&tr())}function tr(){if(!Re){var e=nr(yc);Re=!0;for(var o=pe.length;o;){for(ve=pe,pe=[];++Ao<o;)ve&&ve[Ao].run();Ao=-1,o=pe.length}ve=null,Re=!1,xc(e)}}R.nextTick=function(e){var o=new Array(arguments.length-1);if(arguments.length>1)for(var n=1;n<arguments.length;n++)o[n-1]=arguments[n];pe.push(new rr(e,o)),pe.length===1&&!Re&&nr(tr)};function rr(e,o){this.fun=e,this.array=o}rr.prototype.run=function(){this.fun.apply(null,this.array)};R.title="browser";R.browser=!0;R.env={};R.argv=[];R.version="";R.versions={};function se(){}R.on=se;R.addListener=se;R.once=se;R.off=se;R.removeListener=se;R.removeAllListeners=se;R.emit=se;R.prependListener=se;R.prependOnceListener=se;R.listeners=function(e){return[]};R.binding=function(e){throw new Error("process.binding is not supported")};R.cwd=function(){return"/"};R.chdir=function(e){throw new Error("process.chdir is not supported")};R.umask=function(){return 0}});var i,c,Es,t=b(()=>{i=v(or()),c=v(ir()),Es=function(e){function o(){var r=this||self;return delete e.prototype.__magic__,r}if(typeof globalThis=="object")return globalThis;if(this)return o();e.defineProperty(e.prototype,"__magic__",{configurable:!0,get:o});var n=__magic__;return n}(Object)});var Io,In=b(()=>{"use strict";t();Io=()=>Date.now()/1e3});var cr=b(()=>{"use strict";t()});var pr=b(()=>{"use strict";t()});var En=b(()=>{"use strict";t();In();cr();pr()});var Nn={};be(Nn,{getItem:()=>Sc,localStorageKeys:()=>ue,removeItem:()=>wc,setItem:()=>bc});var ue,Dn,bc,Sc,wc,Me=b(()=>{"use strict";t();En();ue={loginExpiresAt:"dapp-core-login-expires-at",logoutEvent:"dapp-core-logout-event"},Dn=typeof localStorage!="undefined",bc=({key:e,data:o,expires:n})=>{!Dn||localStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},Sc=e=>{if(!Dn)return;let o=localStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:Io()>=n.expires?(localStorage.removeItem(String(e)),null):n.data},wc=e=>{!Dn||localStorage.removeItem(String(e))}});var Ln={};be(Ln,{clear:()=>dr,getItem:()=>mr,removeItem:()=>_r,setItem:()=>sr,storage:()=>vc});var sr,mr,_r,dr,vc,lr=b(()=>{"use strict";t();sr=({key:e,data:o,expires:n})=>{sessionStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},mr=e=>{let o=sessionStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:Date.now()>=n.expires?(sessionStorage.removeItem(String(e)),null):n.data},_r=e=>sessionStorage.removeItem(String(e)),dr=()=>sessionStorage.clear(),vc={setItem:sr,getItem:mr,removeItem:_r,clear:dr}});var ke,Fe=b(()=>{"use strict";t();Me();lr();ke={session:Ln,local:Nn}});function eo(){return new Date().setHours(new Date().getHours()+24)}function oo(e){ke.local.setItem({key:ue.loginExpiresAt,data:e,expires:e})}var Pn=b(()=>{"use strict";t();Fe();Me()});var O=b(()=>{"use strict";t()});var ur,Oe,gr,fr,hr,Cn=b(()=>{"use strict";t();ur="error when signing",Oe="Transaction canceled",gr="error signing transaction",fr="provider not initialized",hr="You need a signer/valid signer to send a transaction,use either WalletProvider, LedgerProvider or WalletConnect"});var Tr=b(()=>{"use strict";t()});var G=b(()=>{"use strict";t()});var xr=b(()=>{"use strict";t()});var yr=b(()=>{"use strict";t()});var br=b(()=>{"use strict";t()});var Sr=b(()=>{"use strict";t()});var q=b(()=>{"use strict";t();Tr();G();O();xr();yr();br();Sr()});var Rn,wr,nm,vr,tm,kr,rm,am,im,Ar=b(()=>{"use strict";t();q();Rn={devnet:{id:"devnet",chainId:"D",name:"Devnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://devnet-wallet.elrond.com",apiAddress:"https://devnet-api.elrond.com",explorerAddress:"http://devnet-explorer.elrond.com",apiTimeout:"4000"},testnet:{id:"testnet",chainId:"T",name:"Testnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://testnet-wallet.elrond.com",apiAddress:"https://testnet-api.elrond.com",explorerAddress:"http://testnet-explorer.elrond.com",apiTimeout:"4000"},mainnet:{id:"mainnet",chainId:"1",name:"Mainnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://wallet.elrond.com",apiAddress:"https://api.elrond.com",explorerAddress:"https://explorer.elrond.com",apiTimeout:"4000"}},{chainId:wr,egldLabel:nm}=Rn["devnet"],{chainId:vr,egldLabel:tm}=Rn["testnet"],{chainId:kr,egldLabel:rm}=Rn["mainnet"],am={["devnet"]:wr,["testnet"]:vr,["mainnet"]:kr},im={[wr]:"devnet",[vr]:"testnet",[kr]:"mainnet"}});var Mn,Fn=b(()=>{"use strict";t();Mn={36864:{code:"codeSuccess",message:"Success"},27013:{code:"ERR_USER_DENIED",message:"Rejected by user"},27904:{code:"ERR_UNKNOWN_INSTRUCTION",message:"Unknown instruction"},28160:{code:"ERR_WRONG_CLA",message:"Wrong CLA"},28161:{code:"ERR_INVALID_ARGUMENTS",message:"Invalid arguments"},28162:{code:"ERR_INVALID_MESSAGE",message:"Invalid message"},28163:{code:"ERR_INVALID_P1",message:"Invalid P1"},28164:{code:"ERR_MESSAGE_TOO_LONG",message:"Message too long"},28165:{code:"ERR_RECEIVER_TOO_LONG",message:"Receiver too long"},28166:{code:"ERR_AMOUNT_TOO_LONG",message:"Amount too long"},28167:{code:"ERR_CONTRACT_DATA_DISABLED",message:"Contract data disabled in app options"},28168:{code:"ERR_MESSAGE_INCOMPLETE",message:"Message incomplete"},28176:{code:"ERR_SIGNATURE_FAILED",message:"Signature failed"},28169:{code:"ERR_WRONG_TX_VERSION",message:"Wrong TX version"},28170:{code:"ERR_NONCE_TOO_LONG",message:"Nonce too long"},28171:{code:"ERR_INVALID_AMOUNT",message:"Invalid amount"},28172:{code:"ERR_INVALID_FEE",message:"Invalid fee"},28173:{code:"ERR_PRETTY_FAILED",message:"Pretty failed"},28174:{code:"ERR_DATA_TOO_LONG",message:"Data too long"},28175:{code:"ERR_WRONG_TX_OPTIONS",message:"Invalid transaction options"},28177:{code:"ERR_SIGN_TX_DEPRECATED",message:"Regular transaction signing is deprecated in this version. Use hash signing."}}});var Ir=b(()=>{"use strict";t()});var Eo,On=b(()=>{"use strict";t();Eo="transaction-status-toast-id"});var no,Do,Er,Dr,to,No,Nr,Be,D=b(()=>{"use strict";t();Cn();Ar();Fn();Ir();On();no=5e4,Do=1e9,Er=1,Dr="/dapp/init",to="signSession",No="logout",Nr="login",Be="0"});var Bn,V,ro,ne=b(()=>{"use strict";t();Bn=require("@reduxjs/toolkit");D();V=(0,Bn.createAction)(No),ro=(0,Bn.createAction)(Nr,e=>({payload:e}))});var Pr,Lr,Cr,vm,km,Am,Im,Em,Dm,Rr,Mr,Un=b(()=>{"use strict";t();Pr=require("@reduxjs/toolkit");Pn();O();ne();Lr={loginMethod:"",walletConnectLogin:null,ledgerLogin:null,tokenLogin:null,walletLogin:null,extensionLogin:null,isLoginSessionInvalid:!1},Cr=(0,Pr.createSlice)({name:"loginInfoSlice",initialState:Lr,reducers:{setLoginMethod:(e,o)=>{e.loginMethod=o.payload},setTokenLogin:(e,o)=>{e.tokenLogin=o.payload},setTokenLoginSignature:(e,o)=>{(e==null?void 0:e.tokenLogin)!=null&&(e.tokenLogin.signature=o.payload)},setWalletLogin:(e,o)=>{e.walletLogin=o.payload},setWalletConnectLogin:(e,o)=>{e.walletConnectLogin=o.payload},setLedgerLogin:(e,o)=>{e.ledgerLogin=o.payload},invalidateLoginSession:e=>{e.isLoginSessionInvalid=!0}},extraReducers:e=>{e.addCase(V,()=>Lr),e.addCase(ro,(o,n)=>{o.loginMethod=n.payload.loginMethod,oo(eo())})}}),{setLoginMethod:vm,setWalletConnectLogin:km,setLedgerLogin:Am,setTokenLogin:Im,setTokenLoginSignature:Em,setWalletLogin:Dm,invalidateLoginSession:Rr}=Cr.actions,Mr=Cr.reducer});var kc,Lo,ao=b(()=>{"use strict";t();kc=!1,Lo=(e,o=0)=>{kc||setTimeout(()=>{window.location.href=e},o)}});var Fr=b(()=>{"use strict";t();ao()});function $n(e){return e[Math.floor(Math.random()*e.length)]}var Or=b(()=>{"use strict";t()});function Ur(e){let o=null;if((e==null?void 0:e.statusCode)in Mn){let n=e==null?void 0:e.statusCode,{message:r}=Mn[n];o=Ec.includes(n)?Br:r}return{errorMessage:o,defaultErrorMessage:Br}}var Br,Ac,Ic,Ec,Gn=b(()=>{"use strict";t();Fn();Br="Check if Elrond app is open on Ledger",Ac=28161,Ic=28160,Ec=[Ac,Ic]});var $r=b(()=>{"use strict";t();Fr();Or();Gn()});var Gr,Vr,Vn,Dc,Wr,Wm,Hm,Hr,Wn=b(()=>{"use strict";t();Gr=require("@reduxjs/toolkit"),Vr=v(require("lodash.omit"));$r();Vn={id:"not-configured",chainId:"",name:"NOT CONFIGURED",egldLabel:"",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"",walletConnectBridgeAddress:"",walletConnectV2RelayAddress:"",walletConnectV2ProjectId:"",walletAddress:"",apiAddress:"",explorerAddress:"",apiTimeout:"4000"},Dc={network:Vn,chainID:"-1"},Wr=(0,Gr.createSlice)({name:"appConfig",initialState:Dc,reducers:{initializeNetworkConfig:(e,o)=>{let n=$n(o.payload.walletConnectBridgeAddresses),r=$n(o.payload.walletConnectV2RelayAddresses),a=(0,Vr.default)(o.payload,"walletConnectBridgeAddresses","walletConnectV2RelayAddresses");e.network=I(T(T({},e.network),a),{walletConnectBridgeAddress:n,walletConnectV2RelayAddress:r})},setChainID:(e,o)=>{e.chainID=o.payload}}}),{initializeNetworkConfig:Wm,setChainID:Hm}=Wr.actions,Hr=Wr.reducer});var jn,jr,zr,Po,Hn,qr,Co,Jm,zn,Qm,e_,o_,n_,t_,r_,a_,Kr,qn=b(()=>{"use strict";t();jn=require("@elrondnetwork/erdjs"),jr=require("@reduxjs/toolkit"),zr=require("redux-persist");D();Fe();Me();ne();Po={balance:"...",address:"",nonce:0,txCount:0,scrCount:0,claimableRewards:Be},Hn={address:"",websocketEvent:null,accounts:{"":Po},ledgerAccount:null,publicKey:"",walletConnectAccount:null,isAccountLoading:!0,accountLoadingError:null},qr=(0,jr.createSlice)({name:"accountInfoSlice",initialState:Hn,reducers:{setAddress:(e,o)=>{let n=o.payload;e.address=n,e.publicKey=new jn.Address(n).hex()},setAccount:(e,o)=>{let n=e.address===o.payload.address;e.accounts={[e.address]:n?o.payload:Po},e.isAccountLoading=!1,e.accountLoadingError=null},setAccountNonce:(e,o)=>{let{address:n}=e;e.accounts[n].nonce=o.payload},setAccountShard:(e,o)=>{e.shard=o.payload},setLedgerAccount:(e,o)=>{e.ledgerAccount=o.payload},updateLedgerAccount:(e,o)=>{e.ledgerAccount!=null&&(e.ledgerAccount.index=o.payload.index,e.ledgerAccount.address=o.payload.address)},setWalletConnectAccount:(e,o)=>{e.walletConnectAccount=o.payload},setIsAccountLoading:(e,o)=>{e.isAccountLoading=o.payload,e.accountLoadingError=null},setAccountLoadingError:(e,o)=>{e.accountLoadingError=o.payload,e.isAccountLoading=!1},setWebsocketEvent:(e,o)=>{e.websocketEvent={timestamp:Date.now(),message:o.payload}}},extraReducers:e=>{e.addCase(V,()=>(ke.local.removeItem(ue.loginExpiresAt),Hn)),e.addCase(ro,(o,n)=>{let{address:r}=n.payload;o.address=r,o.publicKey=new jn.Address(r).hex()}),e.addCase(zr.REHYDRATE,(o,n)=>{var u;if(!((u=n.payload)!=null&&u.account))return;let{account:r}=n.payload,{address:a,shard:m,accounts:s,publicKey:d}=r;o.address=a,o.shard=m;let l=s&&a in s;o.accounts=l?s:Hn.accounts,o.publicKey=d})}}),{setAccount:Co,setAddress:Jm,setAccountNonce:zn,setAccountShard:Qm,setLedgerAccount:e_,updateLedgerAccount:o_,setWalletConnectAccount:n_,setIsAccountLoading:t_,setAccountLoadingError:r_,setWebsocketEvent:a_}=qr.actions,Kr=qr.reducer});function te(e){return e!=null&&(Oc(e)||Gc(e))}function fe(e){return e!=null&&(Bc(e)||Vc(e))}function he(e){return e!=null&&(Uc(e)||Wc(e))}function Ro(e){return e!=null&&$c(e)}function Oc(e){return e!=null&&Nc.includes(e)}function Bc(e){return e!=null&&Lc.includes(e)}function Uc(e){return e!=null&&Pc.includes(e)}function $c(e){return e!=null&&Cc.includes(e)}function Gc(e){return e!=null&&Rc.includes(e)}function Vc(e){return e!=null&&Mc.includes(e)}function Wc(e){return e!=null&&Fc.includes(e)}var Nc,Lc,Pc,Cc,Rc,Mc,Fc,Ue=b(()=>{"use strict";t();O();Nc=["sent"],Lc=["success"],Pc=["fail","cancelled","timedOut"],Cc=["timedOut"],Rc=["pending"],Mc=["success"],Fc=["fail","invalid"]});var Xr,Yr,co,Hc,Zr,po,Mo,jc,$e,Jr,zc,u_,Qr,so,me,Fo,Oo=b(()=>{"use strict";t();Xr=require("@reduxjs/toolkit"),Yr=require("redux-persist");O();Ue();ne();co={signedTransactions:{},transactionsToSign:null,signTransactionsError:null,signTransactionsCancelMessage:null,customTransactionInformationForSessionId:{}},Hc={signWithoutSending:!1,sessionInformation:null,redirectAfterSign:!1},Zr=(0,Xr.createSlice)({name:"transactionsSlice",initialState:co,reducers:{moveTransactionsToSignedState:(e,o)=>{var l,u;let{sessionId:n,transactions:r,errorMessage:a,status:m,redirectRoute:s}=o.payload,d=((l=e.customTransactionInformationForSessionId)==null?void 0:l[n])||Hc;e.signedTransactions[n]={transactions:r,status:m,errorMessage:a,redirectRoute:s,customTransactionInformation:d},((u=e==null?void 0:e.transactionsToSign)==null?void 0:u.sessionId)===n&&(e.transactionsToSign=co.transactionsToSign)},clearSignedTransaction:(e,o)=>{e.signedTransactions[o.payload]&&delete e.signedTransactions[o.payload]},clearTransactionToSign:e=>{e!=null&&e.transactionsToSign&&(e.transactionsToSign=null)},updateSignedTransaction:(e,o)=>{e.signedTransactions=T(T({},e.signedTransactions),o.payload)},updateSignedTransactions:(e,o)=>{let{sessionId:n,status:r,errorMessage:a,transactions:m}=o.payload;e.signedTransactions[n]!=null&&(e.signedTransactions[n].status=r,a!=null&&(e.signedTransactions[n].errorMessage=a),m!=null&&(e.signedTransactions[n].transactions=m))},updateSignedTransactionStatus:(e,o)=>{var d,l,u,f,g,y;let{sessionId:n,status:r,errorMessage:a,transactionHash:m}=o.payload,s=(l=(d=e.signedTransactions)==null?void 0:d[n])==null?void 0:l.transactions;if(s!=null){e.signedTransactions[n].transactions=s.map(L=>L.hash===m?I(T({},L),{status:r,errorMessage:a}):L);let h=(f=(u=e.signedTransactions[n])==null?void 0:u.transactions)==null?void 0:f.every(L=>fe(L.status)),E=(y=(g=e.signedTransactions[n])==null?void 0:g.transactions)==null?void 0:y.every(L=>he(L.status));h&&(e.signedTransactions[n].status="success"),E&&(e.signedTransactions[n].status="fail")}},setTransactionsToSign:(e,o)=>{e.transactionsToSign=o.payload;let{sessionId:n,customTransactionInformation:r}=o.payload;e.customTransactionInformationForSessionId[n]=r,e.signTransactionsError=null},clearAllTransactionsToSign:e=>{e.transactionsToSign=co.transactionsToSign,e.signTransactionsError=null},clearAllSignedTransactions:e=>{e.signedTransactions=co.signedTransactions},setSignTransactionsError:(e,o)=>{e.signTransactionsError=o.payload},setSignTransactionsCancelMessage:(e,o)=>{e.signTransactionsCancelMessage=o.payload}},extraReducers:e=>{e.addCase(V,()=>co),e.addCase(Yr.REHYDRATE,(o,n)=>{var s;if(!((s=n.payload)!=null&&s.transactions))return;let{signedTransactions:r,customTransactionInformationForSessionId:a}=n.payload.transactions,m=Object.entries(r).reduce((d,[l,u])=>{let f=new Date(l),g=new Date;return g.setHours(g.getHours()+5),g-f>0||(d[l]=u),d},{});a!=null&&(o.customTransactionInformationForSessionId=a),r!=null&&(o.signedTransactions=m)})}}),{updateSignedTransactionStatus:po,updateSignedTransactions:Mo,setTransactionsToSign:jc,clearAllTransactionsToSign:$e,clearAllSignedTransactions:Jr,clearSignedTransaction:zc,clearTransactionToSign:u_,setSignTransactionsError:Qr,setSignTransactionsCancelMessage:so,moveTransactionsToSignedState:me}=Zr.actions,Fo=Zr.reducer});var ea,Xn,Yn,Zn,qc,Kn,oa,T_,Kc,Bo,Uo,$o=b(()=>{"use strict";t();ea=require("@reduxjs/toolkit");ne();Xn="Transaction failed",Yn="Transaction successful",Zn="Processing transaction",qc="Transaction submitted",Kn={},oa=(0,ea.createSlice)({name:"transactionsInfo",initialState:Kn,reducers:{setTransactionsDisplayInfo(e,o){let{sessionId:n,transactionsDisplayInfo:r}=o.payload;n!=null&&(e[n]={errorMessage:(r==null?void 0:r.errorMessage)||Xn,successMessage:(r==null?void 0:r.successMessage)||Yn,processingMessage:(r==null?void 0:r.processingMessage)||Zn,submittedMessage:(r==null?void 0:r.submittedMessage)||qc,transactionDuration:r==null?void 0:r.transactionDuration})},clearTransactionsInfoForSessionId(e,o){o.payload!=null&&delete e[o.payload]},clearTransactionsInfo:()=>Kn},extraReducers:e=>{e.addCase(V,()=>Kn)}}),{clearTransactionsInfo:T_,setTransactionsDisplayInfo:Kc,clearTransactionsInfoForSessionId:Bo}=oa.actions,Uo=oa.reducer});var ta,na,ra,S_,Xc,w_,v_,aa,Jn=b(()=>{"use strict";t();ta=require("@reduxjs/toolkit");ne();na={},ra=(0,ta.createSlice)({name:"modalsSlice",initialState:na,reducers:{setTxSubmittedModal:(e,o)=>{e.txSubmittedModal=o.payload},setNotificationModal:(e,o)=>{e.notificationModal=o.payload},clearTxSubmittedModal:e=>{e.txSubmittedModal=void 0},clearNotificationModal:e=>{e.notificationModal=void 0}},extraReducers:e=>{e.addCase(V,()=>na)}}),{setTxSubmittedModal:S_,setNotificationModal:Xc,clearTxSubmittedModal:w_,clearNotificationModal:v_}=ra.actions,aa=ra.reducer});var ca,ia,pa,Yc,Qn,P_,C_,R_,M_,Go,Vo=b(()=>{"use strict";t();ca=require("@reduxjs/toolkit");On();q();In();ne();ia={customToasts:[],transactionToasts:[],failTransactionToast:null},pa=(0,ca.createSlice)({name:"toastsSlice",initialState:ia,reducers:{addCustomToast:(e,o)=>{e.customToasts.push(I(T({},o.payload),{type:"custom",toastId:o.payload.toastId||`custom-toast-${e.customToasts.length+1}`}))},removeCustomToast:(e,o)=>{e.customToasts=e.customToasts.filter(n=>n.toastId!==o.payload)},addTransactionToast:(e,o)=>{e.transactionToasts.push({type:"transaction",startTimestamp:Io(),toastId:o.payload||`custom-toast-${e.transactionToasts.length+1}`})},removeTransactionToast:(e,o)=>{e.transactionToasts=e.transactionToasts.filter(n=>n.toastId!==o.payload)},addFailTransactionToast:(e,o)=>{e.failTransactionToast=I(T({},o.payload),{toastId:Eo})},removeFailTransactionToast:e=>{e.failTransactionToast=null}},extraReducers:e=>{e.addCase(V,()=>ia)}}),{addCustomToast:Yc,removeCustomToast:Qn,addTransactionToast:P_,removeTransactionToast:C_,addFailTransactionToast:R_,removeFailTransactionToast:M_}=pa.actions,Go=pa.reducer});var B=b(()=>{"use strict";t();Un();Wn();qn();Oo();$o();Jn();Vo()});var et=b(()=>{"use strict";t()});var da,_a,la,od,nd,td,rd,ad,id,ua,ga=b(()=>{"use strict";t();da=require("@reduxjs/toolkit");ne();_a={visible:!1,modal:{showHeader:!0,showFooter:!1,headerText:"",footerText:""}},la=(0,da.createSlice)({name:"dappModalsSlice",initialState:_a,reducers:{setDappModalVisibility:(e,o)=>{e.visible=o.payload},setDappModalHeaderVisibility:(e,o)=>{e.modal.showHeader=o.payload},setDappModalFooterVisibility:(e,o)=>{e.modal.showFooter=o.payload},setDappModalHeaderText:(e,o)=>{e.modal.headerText=o.payload},setDappModalFooterText:(e,o)=>{e.modal.footerText=o.payload},setDappModalConfig:(e,o)=>{e.modal=T({},o.payload)}},extraReducers:e=>{e.addCase(V,()=>_a)}}),{setDappModalVisibility:od,setDappModalHeaderVisibility:nd,setDappModalFooterVisibility:td,setDappModalHeaderText:rd,setDappModalFooterText:ad,setDappModalConfig:id}=la.actions,ua=la.reducer});var fa,ep,op,Ge,Ho=b(()=>{"use strict";t();fa=require("@reduxjs/toolkit");et();qn();ga();Un();Jn();Wn();Vo();$o();Oo();ep={["account"]:Kr,["networkConfig"]:Hr,["loginInfo"]:Mr,["modals"]:aa,["dappModal"]:ua,["toasts"]:Go,["transactions"]:Fo,["transactionsInfo"]:Uo},op=(e={})=>(0,fa.combineReducers)(T(T({},ep),e)),Ge=op});var xa={};be(xa,{default:()=>pp});function ot(e,o=[]){return{key:e,version:1,storage:Ta.default,blacklist:o}}var Ie,ha,Ta,np,tp,rp,ap,ip,cp,pp,ya=b(()=>{"use strict";t();Ie=require("redux-persist"),ha=v(require("redux-persist/lib/storage")),Ta=v(require("redux-persist/lib/storage/session"));Ho();B();Vo();$o();Oo();et();np={2:e=>I(T({},e),{networkConfig:Vn})};tp=ot("dapp-core-transactionsInfo"),rp=ot("dapp-core-transactions",["transactionsToSign"]),ap=ot("dapp-core-toasts"),ip={key:"dapp-core-store",version:2,storage:ha.default,whitelist:["account","loginInfo","modals","networkConfig"],migrate:(0,Ie.createMigrate)(np,{debug:!1})},cp={["toasts"]:(0,Ie.persistReducer)(ap,Go),["transactions"]:(0,Ie.persistReducer)(rp,Fo),["transactionsInfo"]:(0,Ie.persistReducer)(tp,Uo)},pp=(0,Ie.persistReducer)(ip,Ge(cp))});var ba={};be(ba,{default:()=>sp});var sp,Sa=b(()=>{"use strict";t();Ho();sp=Ge()});var wa={};be(wa,{default:()=>_p});var K,mp,_p,va=b(()=>{"use strict";t();K=require("redux-persist"),mp=[K.FLUSH,K.REHYDRATE,K.PAUSE,K.PERSIST,K.PURGE,K.REGISTER],_p=mp});var Ia={};be(Ia,{default:()=>Aa});function Aa(e){return(0,ka.persistStore)(e)}var ka,Ea=b(()=>{"use strict";t();ka=require("redux-persist")});var Ss={};be(Ss,{useCheckTransactionStatus:()=>_s,useGetActiveTransactionsStatus:()=>bs,useGetFailedTransactions:()=>xs,useGetPendingTransactions:()=>Et,useGetSignTransactionsError:()=>rs,useGetSignedTransactions:()=>vt,useGetSuccessfulTransactions:()=>ys,useGetTokenDetails:()=>ns,useGetTransactionDisplayInfo:()=>Xp,useParseMultiEsdtTransferData:()=>kt,useParseSignedTransactions:()=>bt,useSignMultipleTransactions:()=>At,useSignTransactions:()=>es,useSignTransactionsWithDevice:()=>It,useSignTransactionsWithLedger:()=>ts,useTrackTransactionStatus:()=>Ts});module.exports=Qe(Ss);t();t();t();var Ra=require("react"),Ve=require("react-redux");t();var nt=require("@reduxjs/toolkit"),Pa=require("react-redux/es/utils/Subscription");B();t();var sa=v(require("lodash.throttle"));D();B();Pn();Fe();Me();var Zc=[No],Jc=(0,sa.default)(()=>{oo(eo())},5e3),ma=e=>o=>n=>{var l;if(Zc.includes(n.type))return o(n);let r=e.getState(),a=ke.local.getItem(ue.loginExpiresAt);if(!((l=r==null?void 0:r.loginInfo)==null?void 0:l.isLoggedIn))return o(n);if(a==null)return oo(eo());let s=Date.now();return a-s<0?setTimeout(()=>w(void 0,null,function*(){console.log("session expired");try{e.dispatch(Rr())}catch(u){console.error("error logging out",u)}}),1e3):(Jc(),o(n))};t();t();function Wo(){return typeof sessionStorage!="undefined"}var Da=Wo()?(ya(),Qe(xa)).default:(Sa(),Qe(ba)).default,Na=Wo()?(va(),Qe(wa)).default:[],La=Wo()?(Ea(),Qe(Ia)).default:e=>e;Ho();var S=(0,nt.configureStore)({reducer:Da,middleware:e=>e({serializableCheck:{ignoredActions:[...Na,zn.type,Co.type],ignoredPaths:["payload.nonce","account.account.nonce","providers.accountProvider"]}}).concat(ma)}),Ca=(0,Pa.createSubscription)(S),Ld=La(S),Pd=(0,nt.configureStore)({reducer:Ge});var dp={store:S,subscription:Ca},tt=(0,Ra.createContext)(dp),Fd=(0,Ve.createStoreHook)(tt),Te=(0,Ve.createDispatchHook)(tt),k=(0,Ve.createSelectorHook)(tt);t();t();B();t();var Ma=v(require("lodash.isequal")),jo=require("reselect"),x=(0,jo.createSelectorCreator)(jo.defaultMemoize,Ma.default);var _e=e=>e.account,xe=x(_e,e=>e.address),zo=x(_e,xe,(e,o)=>o in e.accounts?e.accounts[o]:Po),Fa=x(_e,zo,(e,o)=>{let a=e,{accounts:n}=a,r=gn(a,["accounts"]);return I(T({},r),{account:o})}),lp=x(zo,e=>e.balance),Oa=x(zo,e=>{var o;return((o=e==null?void 0:e.nonce)==null?void 0:o.valueOf())||0}),Gd=x(_e,e=>e.shard),Vd=x(_e,e=>e.ledgerAccount),Wd=x(_e,e=>e.walletConnectAccount),Hd=x(_e,e=>e.isAccountLoading),jd=x(_e,e=>e.accountLoadingError),zd=x(_e,e=>e.websocketEvent);t();var re=e=>e.networkConfig,rt=x(re,e=>e.chainID),Xd=x(re,e=>e.network.walletConnectBridgeAddress),Yd=x(re,e=>e.network.walletConnectV2RelayAddress),Zd=x(re,e=>e.network.walletConnectV2ProjectId),Jd=x(re,e=>e.network.walletConnectDeepLink),Ee=x(re,e=>e.network),qo=x(Ee,e=>e.apiAddress),Ba=x(Ee,e=>e.explorerAddress),at=x(Ee,e=>e.egldLabel);t();O();var We=e=>e.loginInfo,tl=x(We,e=>e.loginMethod),it=x(We,xe,(e,o)=>e.loginMethod!=""&&Boolean(o)),rl=x(We,e=>e.walletConnectLogin),al=x(We,e=>e.ledgerLogin),Ua=x(We,e=>e.walletLogin),il=x(We,e=>e.isLoginSessionInvalid);t();t();var j=require("@elrondnetwork/erdjs");D();t();function ct(e){try{return i.Buffer.from(e,"base64").toString()===atob(e)}catch(o){return!1}}function mo(e){var a,m,s;let{data:o}=e,n=ct(o!=null?o:"")?j.TransactionPayload.fromEncoded(o):new j.TransactionPayload(o),r=new j.Transaction(T({value:e.value.valueOf(),data:n,nonce:e.nonce.valueOf(),receiver:new j.Address(e.receiver),sender:new j.Address(e.sender),gasLimit:(a=e.gasLimit.valueOf())!=null?a:no,gasPrice:(m=e.gasPrice.valueOf())!=null?m:Do,chainID:e.chainID.valueOf(),version:new j.TransactionVersion((s=e.version)!=null?s:Er)},e.options?{options:new j.TransactionOptions(e.options)}:{}));return r.applySignature({hex:()=>e.signature||""},new j.Address(e.sender)),r}t();Ue();t();var st=v(require("bignumber.js"));q();t();t();function pt(e){for(let o=0;o<e.length;o++)if(e.charCodeAt(o)>127)return!1;return!0}function _o(e){let o=e;try{let n=i.Buffer.from(e,"hex").toString();pt(n)&&n.length>1&&(o=n)}catch(n){}return o}t();var $a=(e,o)=>{let n=[],r=e.indexOf(o,0);for(;r>=0;)n.push(r),r=e.indexOf(o,r+1);return n};function Va(e){let o=[],n=0;try{if((e==null?void 0:e.startsWith("MultiESDTNFTTransfer"))&&(e==null?void 0:e.includes("@"))){let[,r,a,...m]=e==null?void 0:e.split("@");if(r){let s=new st.default(a,16).toNumber(),d=0;for(let g=0;g<s;g++){let y={type:"nftTransaction",data:"",receiver:r};for(let h=0;h<3;h++){switch(h){case 0:y.token=_o(m[d]),y.data=m[d];break;case 1:{let E=m[d]&&m[d].length?m[d]:"";E?y.nonce=E:y.type="esdtTransaction",y.data=`${y.data}@${m[d]}`;break}case 2:y.amount=new st.default(m[d],16).toString(10),y.data=`${y.data}@${m[d]}`;break;default:break}n=d+1,d++}o[g]=y}let l=o.length!==s,u=o.some(g=>$a(g.data,"@").length!==2),f=o.some(g=>g.data.startsWith("@"));if(l||u||f)return[];if(m[n]){let g=m[n];for(let y=n+1;y<m.length;y++)g+="@"+m[y];o[s]={type:"scCall",data:g,receiver:r}}}}}catch(r){return console.error("failed parsing tx",r),o}return o}t();var _t=require("@elrondnetwork/erdjs"),dt=v(require("bignumber.js"));O();t();var Wa=require("@elrondnetwork/erdjs");function up(e){try{let o=new Wa.Address(e);return Boolean(o.bech32())}catch(o){return!1}}function He(e){return(e==null?void 0:e.startsWith("erd"))&&e.length===62&&/^\w+$/.test(e)&&up(e)}var mt={tokenId:"",amount:""},gp=e=>e.split("@").map((m,s)=>[2,3].includes(s)?m:_o(m));function lt(e){if(!e)return mt;let o=e.startsWith("ESDTTransfer"),n=e.startsWith("ESDTNFTTransfer")&&e.includes("@");if(o){let[,r,a]=e.split("@");try{let m=i.Buffer.from(r,"hex").toString("ascii");if(!m)return mt;let s=new dt.default("0x"+a.replace("0x","")).toString(10);return{tokenId:m,amount:s}}catch(m){console.error("Error getting token from transaction data",m)}}if(n)try{let[,r,a,m,s]=gp(e);if([r,a,m,s].every(d=>Boolean(d))&&He(new _t.Address(s).bech32()))return{tokenId:`${r}-${a}`,amount:new dt.default(m,16).toString(10),collection:r,nonce:a,receiver:new _t.Address(s).bech32()}}catch(r){}return mt}t();function Ha({tokenId:e,erdLabel:o}){return Boolean(e&&e!==o)}t();t();var hp=require("@elrondnetwork/erdjs/out");t();D();t();var ja=require("@elrondnetwork/erdjs");q();t();function Tp(e,o){let n=new URLSearchParams(e),r=Object.fromEntries(n);return{nextUrlParams:new URLSearchParams(T(T({},r),o)).toString(),params:r}}function za({callbackUrl:e,urlParams:o={}}){let n=e;if(Object.entries(o).length>0){let{search:r,origin:a,pathname:m}=new URL(e),{nextUrlParams:s}=Tp(r,o);n=`${a}${m}?${s}`}return n}t();t();t();O();function je(e){let n=Object.getPrototypeOf(e).toPlainObject!=null?e:mo(e);return I(T({},n.toPlainObject()),{hash:n.getHash().hex(),status:"pending"})}t();t();t();t();t();t();t();var lo="accounts";var ut="tokens";var gt="collections";t();t();t();t();G();t();t();G();t();t();t();var kp=v(require("bignumber.js"));D();t();var vp=require("@elrondnetwork/erdjs"),ft=v(require("bignumber.js"));D();t();var Sp=v(require("bignumber.js"));t();ft.default.config({ROUNDING_MODE:ft.default.ROUND_FLOOR});t();t();t();D();t();t();t();t();D();t();D();t();var Ap=require("@elrondnetwork/erdjs");D();t();var Yo=require("@elrondnetwork/erdjs"),Dp=require("@elrondnetwork/erdjs"),Np=require("@elrondnetwork/erdjs-network-providers");D();t();t();var Ip=v(require("bignumber.js"));D();t();D();t();var ht={isEsdt:!1,isNft:!1,isEgld:!1};function Tt(e){let o=e==null?void 0:e.split("-").length;return o===2?I(T({},ht),{isEsdt:!0}):o===3?I(T({},ht),{isNft:!0}):I(T({},ht),{isEgld:!0})}t();t();t();t();D();t();G();var Pp=["reDelegateRewards","claimRewards","unBond"],Cp=["wrapEgld","unwrapEgld"],Rp=["unStake"],Mp=["unDelegate"];t();t();t();D();t();var Bp=v(require("bignumber.js"));t();t();t();G();t();var Up=v(require("bignumber.js"));t();t();t();G();t();G();t();var Gp=v(require("bignumber.js"));D();G();t();G();t();G();t();var Vp=v(require("bignumber.js"));t();O();t();O();t();t();var Wp=require("react");G();var Hh=["create","localMint","ESDTLocalMint","addQuantity","burn","localBurn","ESDTLocalBurn","wipe","writeLog","signalError"];t();var Hp=v(require("anchorme"));t();var Za=require("react");q();t();t();G();t();var jp=require("@elrondnetwork/erdjs/out"),zp=v(require("bignumber.js"));G();t();G();t();t();O();var bT=[{label:"Raw",value:"raw"},{label:"Text",value:"text"},{label:"Decimal",value:"decimal"},{label:"Smart",value:"smart"}];t();var Ja=require("react");q();t();t();t();t();t();D();var Jo=e=>e.transactions,De=x(Jo,e=>e.signedTransactions),Qa=x(Jo,e=>e.signTransactionsError),ei=x(Jo,e=>e.signTransactionsCancelMessage),Qo=e=>o=>Object.entries(o).reduce((n,[r,a])=>(e(a.status)&&(n[r]=a),n),{}),en=x(De,Qo(te)),on=x(De,Qo(fe)),nn=x(De,Qo(he)),oi=x(De,Qo(Ro)),ze=x(Jo,e=>{var o;return(e==null?void 0:e.transactionsToSign)==null?null:I(T({},e.transactionsToSign),{transactions:((o=e==null?void 0:e.transactionsToSign)==null?void 0:o.transactions.map(n=>mo(n)))||[]})}),ni=x(De,(e,o)=>o,(e,o)=>o!=null?(e==null?void 0:e[o])||{}:{});t();B();var ti={errorMessage:Xn,successMessage:Yn,processingMessage:Zn},Kp=e=>e.transactionsInfo,ri=x(Kp,(e,o)=>o,(e,o)=>o!=null&&(e==null?void 0:e[Number(o)])||ti);t();var ai=e=>e.modals,Tx=x(ai,e=>e.txSubmittedModal),xx=x(ai,e=>e.notificationModal);t();var yt=e=>e.toasts,Sx=x(yt,e=>e.customToasts),wx=x(yt,e=>e.transactionToasts),vx=x(yt,e=>e.failTransactionToast);function Xp(e){return k(o=>ri(o,e))}t();var ii=require("react"),ci=require("@elrondnetwork/erdjs-web-wallet-provider"),pi=v(require("qs"));D();B();O();function bt(e){let{search:o}=window.location,n=k(Ee),r=Te();(0,ii.useEffect)(()=>{if(o!=null){let a=pi.default.parse(o.replace("?",""));if(a&&to in a){let m=String(a[to]),s=new ci.WalletProvider(`${n.walletAddress}${Dr}`).getTransactionsFromWalletUrl();if(a.status==="cancelled"){r(me({sessionId:m,status:"cancelled"})),e(),history.pushState({},document.title,"?");return}s.length>0&&(r(me({sessionId:m,status:"signed",transactions:s.map(d=>je(d))})),history.pushState({},document.title,"?"))}}},[o])}t();var Ke=require("react"),wt=require("@elrondnetwork/erdjs-extension-provider");D();t();t();t();t();var si=require("@elrondnetwork/erdjs-extension-provider"),mi=require("@elrondnetwork/erdjs-hw-provider"),rn=require("@elrondnetwork/erdjs-wallet-connect-provider"),_i=require("@elrondnetwork/erdjs-web-wallet-provider");D();O();var Ne=e=>{switch(e==null?void 0:e.constructor){case _i.WalletProvider:return"wallet";case rn.WalletConnectProvider:return"walletconnect";case rn.WalletConnectV2Provider:return"walletconnectv2";case mi.HWProvider:return"ledger";case si.ExtensionProvider:return"extension";case tn:return"";default:return"extra"}};var de=e=>`Unable to perform ${e}, Provider not initialized`,tn=class{init(){return Promise.resolve(!1)}login(o){throw new Error(de(`login with options: ${o}`))}logout(o){throw new Error(de(`logout with options: ${o}`))}getAddress(){throw new Error(de("getAddress"))}isInitialized(){return!1}isConnected(){return Promise.resolve(!1)}sendTransaction(o,n){throw new Error(de(`sendTransaction with transactions: ${o} options: ${n}`))}signTransaction(o,n){throw new Error(de(`signTransaction with transactions: ${o} options: ${n}`))}signTransactions(o,n){throw new Error(de(`signTransactions with transactions: ${o} options: ${n}`))}signMessage(o,n){throw new Error(de(`signTransactions with ${o} and options ${n}`))}sendCustomMessage({method:o,params:n}){throw new Error(de(`sendCustomMessage with method: ${o} params: ${n}`))}sendCustomRequest(o){throw new Error(de(`sendSessionEvent with options: ${o}`))}ping(){return Promise.resolve(!1)}},St=new tn;var Yp=St;function Y(){return Yp||St}var an=()=>{let e=Y(),o=Ne(e);return{provider:e,providerType:o}};B();O();t();t();ne();q();t();t();t();O();t();t();function di(){let e=Y();return Ne(e)}function le(e){return di()===e}function cn(){let{search:e}=window.location,o=S.getState(),n=Y(),r=xe(o),a=it(o),m=Ua(o);if(!n)throw"provider not initialized";return le("ledger")&&a?new Promise(s=>{s(r)}):!le("")&&!le("wallet")&&!le("extra")?n.getAddress():new Promise(s=>{if(m!=null){let d=new URLSearchParams(e),l=Object.fromEntries(d);He(l.address)&&s(l.address)}a&&s(r),s("")})}t();t();var ui=v(require("axios"));t();t();var li=e=>{let o=e!=null?e:qo(S.getState());return o.endsWith("/")?o.slice(0,-1):o};function gi(e){return w(this,null,function*(){if(!e)return null;let n=`${li()}/${lo}/${e}`;try{let{data:r}=yield ui.default.get(n);return r}catch(r){console.error("error fetching configuration for ",n)}return null})}function qe(e){return w(this,null,function*(){return yield gi(e)})}t();B();t();function uo(e){let o=S.getState(),n=Oa(o);return e?n&&!isNaN(n)?Math.max(n,e.nonce):e.nonce:n}t();t();B();var fi=()=>w(void 0,null,function*(){try{let e=yield cn();try{let o=yield qe(e);if(o!=null){let n=I(T({},o),{nonce:uo(o)});return S.dispatch(Co(n)),n}}catch(o){console.error("Failed getting account ",o)}}catch(e){console.error("Failed getting address ",e)}return null});function pn(){return w(this,null,function*(){let e=Y();if(e==null)throw"Provider not initialized";if(!e.isInitialized||e.isInitialized())return fi();try{if(!e.init)throw"Current provider does not have init() function";return(yield e.init())?fi():void 0}catch(o){console.error("Failed initializing provider ",o)}})}t();var hi=require("@elrondnetwork/erdjs");t();ne();q();ao();Fe();Me();t();t();t();t();B();t();t();t();t();t();Fe();ao();En();t();t();t();t();t();var Zp=v(require("axios"));t();t();var Jp=v(require("platform"));t();function sn(e){let o=ze(S.getState()),n=Object.keys(e).length===(o==null?void 0:o.transactions.length);return e&&(e&&n)}var Qp=(e,o)=>o.map((n,r)=>(n.setNonce(e+r),n)),es=()=>{let e=Te(),o=(0,Ke.useRef)("/"),n=k(xe),{provider:r}=an(),a=Ne(r),[m,s]=(0,Ke.useState)(null),d=k(ze),l=k(ei),u=Boolean(d==null?void 0:d.transactions),f=()=>{s(null),e(so(null))},g=N=>{f(),y(N)};bt(g);function y(N){var P,C;let A=r instanceof wt.ExtensionProvider;e($e()),e(Bo(N)),e(Qn(Eo)),A&&(f(),(C=(P=wt.ExtensionProvider.getInstance())==null?void 0:P.cancelAction)==null||C.call(P))}let h=(N,A)=>{let P=N.includes(Oe);if(y(A),P){e(so(Oe));return}s(N)},E=(N,A,P="")=>{let C={[to]:A},X=`${window.location.origin}${P}`,H=za({callbackUrl:X,urlParams:C});r.signTransactions(N,{callbackUrl:encodeURIComponent(H)})},L=()=>w(void 0,null,function*(){var ho;let{sessionId:N,transactions:A,callbackRoute:P,customTransactionInformation:C}=d,{redirectAfterSign:X}=C,H=P||window.location.pathname,Xe=window.location.pathname.includes(H),Ye=X&&!Xe;try{if(!(yield(ho=r==null?void 0:r.init)==null?void 0:ho.call(r)))return}catch($){let ae=($==null?void 0:$.message)||$||fr;console.error(ae),h(ae);return}try{let $=yield r.signTransactions(A);if(!sn($))return;let dn=Object.values($).map(xo=>je(xo)),To={sessionId:N,transactions:dn,status:"signed"};Ye&&(To.redirectRoute=H),e(me(To))}catch($){let ae=($==null?void 0:$.message)||$||gr;console.error(ae),e(me({sessionId:N,status:"cancelled"})),h(ae.includes("cancel")?Oe:ae,N)}}),Z=()=>w(void 0,null,function*(){if(!d)return;f();let{sessionId:N,transactions:A,callbackRoute:P}=d;if(!r){console.error(hr);return}o.current=P||window.location.pathname;try{let C=yield qe(n);if(C==null)return;let X=a==="wallet",H=!["wallet","ledger","extra"].includes(a),Xe=uo(C),Ye=Qp(Xe,A);X&&E(Ye,N,P),H&&L()}catch(C){let H=(C==null?void 0:C.message)||ur;console.error(H),h(H,N),e(me({sessionId:N,status:"cancelled"})),console.error(H,C)}});return(0,Ke.useEffect)(()=>{Z()},[d]),{error:m,canceledTransactionsMessage:l,onAbort:g,hasTransactions:u,callbackRoute:o.current,sessionId:d==null?void 0:d.sessionId,transactions:d==null?void 0:d.transactions}};t();function vt(){let e=k(De),o=Object.entries(e),n=(o==null?void 0:o.length)>0;return{signedTransactions:e,signedTransactionsArray:o,hasSignedTransactions:n}}t();var xi=v(require("axios")),yi=v(require("swr"));t();var Ti=()=>k(re);var os=e=>xi.default.get(e).then(o=>o.data);function ns({tokenId:e}){var y;let{network:o}=Ti(),{isEsdt:n,isNft:r}=Tt(e),a=n?ut:gt,m=e;if(r){let[h,E]=e.split("-");m=`${h}-${E}`}let{data:s,error:d}=(0,yi.default)(Boolean(m)?`${o.apiAddress}/${a}/${m}`:null,os);if(!m)return{tokenDecimals:Number(o.decimals),tokenLabel:"",tokenAvatar:""};let l=s?s==null?void 0:s.decimals:Number(o.decimals),u=s?s==null?void 0:s.name:"",f=s?`${(y=s==null?void 0:s.assets)==null?void 0:y.svgUrl}`:"",g=s==null?void 0:s.assets;return{tokenDecimals:l,tokenLabel:u,tokenAvatar:f,assets:g,error:d}}t();t();var bi=v(require("axios"));function Si(e){return w(this,null,function*(){let{network:{apiAddress:o,apiTimeout:n}}=re(S.getState()),{data:r}=yield bi.default.get(`/${lo}/${e}`,{baseURL:o,timeout:Number(n)});return r})}t();var wi=()=>k(Fa);t();var Le=require("react");t();var go=require("react");var vi={tokenId:"",amount:"",type:"",multiTxData:"",receiver:""};function kt({transactions:e}){let[o,n]=(0,go.useState)({}),[r,a]=(0,go.useState)([]);function m(l,u){n(f=>I(T({},f),{[l]:u}))}function s(l,u){return o==null?vi:l in o?o[l]:u!=null&&String(u)in o?o[u]:vi}function d(){if(e&&e.length>0){let l=[];e.forEach((u,f)=>{let g=u.getData().toString(),y=Va(g);if(y.length>0)y.forEach((h,E)=>{let L={transaction:u,multiTxData:h.data,transactionIndex:E};m(h.data,{tokenId:h.token?h.token:"",amount:h.amount?h.amount:"",type:h.type,nonce:h.nonce?h.nonce:"",multiTxData:h.data,receiver:h.receiver}),l.push(L)});else{let{tokenId:h,amount:E}=lt(u.getData().toString());h&&m(u.getData().toString(),{tokenId:h,amount:E,receiver:u.getReceiver().bech32()}),l.push({transaction:u,transactionIndex:f})}}),a(l)}}return(0,go.useEffect)(()=>{d()},[e==null?void 0:e.length]),{parsedTransactionsByDataField:o,getTxInfoByDataField:s,allTransactions:r}}Gn();var mn={};function At({isLedger:e=!1,transactionsToSign:o,egldLabel:n,address:r,onCancel:a,onSignTransaction:m,onTransactionsSignError:s,onTransactionsSignSuccess:d,onGetScamAddressData:l}){var xo,Lt,Pt;let[u,f]=(0,Le.useState)(0),[g,y]=(0,Le.useState)(),[h,E]=(0,Le.useState)(null),[L,Z]=(0,Le.useState)(!1),{getTxInfoByDataField:N,allTransactions:A}=kt({transactions:o}),P=u===A.length-1;(0,Le.useEffect)(()=>{C()},[u,A]);function C(){return w(this,null,function*(){var Mt;let M=A[u];if(M==null)return;let{transaction:U,multiTxData:ie}=M,ln=U.getData().toString(),Ct=N(U.getData().toString(),ie),{tokenId:Rt}=Ct,Ze=U.getReceiver().toString(),Ri=r!==Ze,Mi=Ze in mn;if(Ri&&!Mi&&l!=null){let yo=yield l(Ze);mn=T(T({},mn),yo!=null&&yo.scamInfo?{[Ze]:yo.scamInfo}:{})}let Fi=Boolean(Rt&&Ha({tokenId:Rt,erdLabel:n}));E({transaction:U,receiverScamInfo:((Mt=mn[Ze])==null?void 0:Mt.info)||null,transactionTokenInfo:Ct,isTokenTransaction:Fi,dataField:ln})})}function X(){f(0),y(void 0),Z(!1)}function H(){return w(this,null,function*(){try{if(h==null)return;Z(e);let M=yield m(h.transaction),U={[u]:M},ie=g?T(T({},g),U):U;y(ie),P?ie&&(d(Object.values(ie)),X()):(f(ln=>ln+1),Z(!1))}catch(M){console.error(M,"sign error");let{message:U}=M,ie=e?Ur(M).errorMessage:null;X(),s(ie!=null?ie:U)}})}function Xe(){try{if(h==null)return;h.transaction.getSignature().hex()?P||f(U=>U+1):H()}catch(M){H()}}let Ye=u===0;function ho(){Ye?a==null||a():f(M=>M-1)}let $=Boolean(((xo=h==null?void 0:h.transactionTokenInfo)==null?void 0:xo.type)&&((Lt=h==null?void 0:h.transactionTokenInfo)==null?void 0:Lt.multiTxData)&&!(h!=null&&h.dataField.endsWith((Pt=h==null?void 0:h.transactionTokenInfo)==null?void 0:Pt.multiTxData)));function ae(){$?f(M=>M+1):Xe()}function dn(){f(M=>{let U=M+1;return U>(A==null?void 0:A.length)?M:U})}function To(){f(M=>{let U=M-1;return U<0?M:U})}return{allTransactions:A,onSignTransaction:ae,onNext:dn,onPrev:To,waitingForDevice:L,onAbort:ho,isLastTransaction:P,isFirstTransaction:u===0,hasMultipleTransactions:A.length>1,shouldContinueWithoutSigning:$,currentStep:u,signedTransactions:g,currentTransaction:h}}B();q();ao();t();Cn();B();function ki(){let e=Te();return o=>{e($e()),e(Bo(o)),e(so(Oe))}}function It({onCancel:e,verifyReceiverScam:o=!0}){let n=k(ze),r=k(at),{account:{address:a}}=wi(),{provider:m}=an(),s=Te(),d=ki(),{transactions:l,sessionId:u,callbackRoute:f,customTransactionInformation:g}=n||{};function y(A){s(Qr(A))}let h=f!=null&&window.location.pathname.includes(f);function E(A){!sn(A)||u&&(s(me({sessionId:u,status:"signed",transactions:A.map(C=>je(C))})),f!=null&&(g==null?void 0:g.redirectAfterSign)&&!h&&Lo(f))}function L(){e(),d(u)}function Z(A){return w(this,null,function*(){return yield m.signTransaction(A)})}let N=At({address:a,egldLabel:r,transactionsToSign:l,onGetScamAddressData:o?Si:null,isLedger:le("ledger"),onCancel:L,onSignTransaction:Z,onTransactionsSignError:y,onTransactionsSignSuccess:E});return I(T({},N),{callbackRoute:f})}t();function ts(e){return It(e)}t();var rs=()=>k(Qa);t();t();t();function Et(){let e=k(en),o=Object.entries(e),n=(o==null?void 0:o.length)>0;return{pendingTransactions:e,pendingTransactionsArray:o,hasPendingTransactions:n}}Ue();t();t();t();var as=v(require("axios"));t();var Ai=v(require("axios"));function Ii(e){return w(this,null,function*(){let o=qo(S.getState()),n=e.map(a=>a.hash),{data:r}=yield Ai.default.get(`${o}/transactions`,{params:{hashes:n.join(","),withScResults:!0}});return e.map(({hash:a,previousStatus:m})=>{let s=r.find(d=>(d==null?void 0:d.txHash)===a);return{hash:a,data:s==null?void 0:s.data,invalidTransaction:s==null,status:s==null?void 0:s.status,results:s==null?void 0:s.results,sender:s==null?void 0:s.sender,receiver:s==null?void 0:s.receiver,previousStatus:m,hasStatusChanged:s&&s.status!==m}})})}t();var cs=v(require("axios"));t();t();var ps=v(require("axios"));t();var ss=v(require("axios"));t();B();O();t();Ue();function Di(e,o){return e.reduce((r,{status:a,hash:m})=>(m!=null&&!o.includes(m)&&te(a)&&r.push({hash:m,previousStatus:a}),r),[])}t();B();O();function Ni({results:e,hash:o,sessionId:n}){let r=e==null?void 0:e.find(a=>(a==null?void 0:a.returnMessage)!=="");S.dispatch(po({transactionHash:o,sessionId:n,status:"fail",errorMessage:r==null?void 0:r.returnMessage})),S.dispatch(Mo({sessionId:n,status:"fail",errorMessage:r==null?void 0:r.returnMessage}))}t();B();q();function Dt(e){S.dispatch(Mo({sessionId:e,status:"timedOut"}))}var _n={},Nt=[];function ms({serverTransaction:e,sessionId:o,customTransactionInformation:n,shouldRefreshBalance:r}){let{hash:a,status:m,results:s,invalidTransaction:d,hasStatusChanged:l}=e;try{if(Nt.includes(a))return;if(_n[a]>30){Dt(o);return}if(d||te(m)){_n[a]=_n[a]?_n[a]+1:1;return}l&&(fe(m)&&(n==null?void 0:n.completedTransactionsDelay)!=null?(Nt.push(a),setTimeout(()=>S.dispatch(po({sessionId:o,status:"success",transactionHash:a})),n==null?void 0:n.completedTransactionsDelay)):S.dispatch(po({sessionId:o,status:m,transactionHash:a}))),r||pn(),he(m)&&Ni({sessionId:o,hash:a,results:s})}catch(u){console.error(u),Dt(o)}}function Li(m){return w(this,arguments,function*({sessionId:e,transactionBatch:{transactions:o,customTransactionInformation:n},getTransactionsByHash:r=Ii,shouldRefreshBalance:a}){try{if(o==null)return;let s=Di(o,Nt),d=yield r(s);for(let l of d)ms({serverTransaction:l,sessionId:e,customTransactionInformation:n,shouldRefreshBalance:a})}catch(s){console.error(s)}})}function _s(){let{pendingTransactionsArray:e}=Et();function o(n){return w(this,null,function*(){let r=e.filter(([a,m])=>a!=null&&te(m.status));if(r.length>0)for(let[a,m]of r)yield Li(T({sessionId:a,transactionBatch:m},n));n.shouldRefreshBalance&&pn()})}return o}t();var fo=require("react");t();t();t();B();function Pi(){S.dispatch(Jr())}function Ci(){S.dispatch($e())}t();t();var gs=v(require("bignumber.js"));t();var ds=`.dapp-core-component__main__link-style {
  color: #1392ff;
}

.dapp-core-component__main__link-second-style {
  color: #9dabbd;
}
.dapp-core-component__main__link-second-style:hover {
  color: #1392ff;
}
.dapp-core-component__main__link-second-style:hover svg {
  color: #1392ff;
}

.dapp-core-component__main__wrapper {
  min-height: 100vh;
}

.dapp-core-component__main__opacity-6 {
  opacity: 0.6;
}

.dapp-core-component__main__qr-code-svg-container {
  width: 15rem;
  height: 15rem;
}

.dapp-core-component__main__pair-list {
  max-height: 20rem;
  overflow-y: auto;
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__pair-image {
  width: 4rem;
  overflow: hidden;
  height: auto;
  min-width: 2rem;
}

.dapp-core-component__main__pair-remove {
  padding: 0 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

.dapp-core-component__main__dapp-icon {
  padding: 5px;
  border-radius: 50%;
  background-color: #fafafa;
  width: 4.6428571429rem;
  height: 4.6428571429rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}
.dapp-core-component__main__dapp-icon.dapp-core-component__main__icon-medium {
  width: 80px;
  height: 80px;
}

.dapp-core-component__main__centering {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dapp-core-component__main__border-n {
  border: none !important;
}

.dapp-core-component__main__t-shadow {
  text-shadow: 0px 4px 7px rgba(0, 0, 0, 0.3);
}

.dapp-core-component__main__btn {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.04) !important;
}

.dapp-core-component__main__media {
  display: flex;
  align-items: flex-start;
}

.dapp-core-component__main__media-body {
  flex: 1;
}

.dapp-core-component__main__connect-btns .dapp-core-component__main__btn {
  display: flex;
  height: 48px;
  justify-content: center;
  align-items: center;
  color: #fff !important;
  background-color: #1392ff;
  padding: 15px 20px 15px 16px;
  border-radius: 6px;
  gap: 8px;
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__btn:hover, .dapp-core-component__main__connect-btns .dapp-core-component__main__btn:active {
  background-color: #1392ff;
  background-image: linear-gradient(rgba(0, 0, 0, 0.1) 0 0);
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__info {
  color: #1392ff;
  margin-left: 48px;
}

.dapp-core-component__main__btn {
  transition: none !important;
}

.dapp-core-component__main__trust-badge {
  width: 20px;
  height: 20px;
}

/* Navbar
  -------------------------------------------------- */
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name {
  display: flex;
  align-items: center;
  font-size: 1.7142857143rem;
  font-weight: 400;
  color: #9ba5b4;
  margin-left: 1.6428571429rem;
}
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name:before {
  content: "";
  display: block;
  width: 1px;
  height: 3rem;
  margin-right: 1.7142857143rem;
}

/* Dashboard
  -------------------------------------------------- */
.dapp-core-component__main__action-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  margin: 0 7px;
}
.dapp-core-component__main__action-btn button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #fff;
}
.dapp-core-component__main__action-btn button:hover {
  opacity: 0.9;
}

.dapp-core-component__main__transactions .dapp-core-component__main__transaction-icon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table td {
  white-space: nowrap;
  vertical-align: middle;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table tr:last-of-type td {
  border-bottom: 1px solid #dee2e6;
}

/* Loader, Transaction Success/Fail
  -------------------------------------------------- */
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div {
  position: absolute;
  top: 27px;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.1);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.dapp-core-component__main__page-state .dapp-core-component__main__bg-blue .dapp-core-component__main__lds-ellipsis div {
  background: rgba(255, 255, 255, 0.4);
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(1) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis1 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(2) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(3) {
  left: 26px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(4) {
  left: 45px;
  animation: dapp-core-component__main__lds-ellipsis3 0.6s infinite;
}
@keyframes dapp-core-component__main__lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(19px, 0);
  }
}

.dapp-core-component__main__btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
}
.dapp-core-component__main__btn .dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn .dapp-core-component__main__disabled {
  opacity: 0.65;
  cursor: default;
}
.dapp-core-component__main__btn:hover {
  color: #212529;
  text-decoration: none;
}
.dapp-core-component__main__btn:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn:disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn-primary {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary:hover {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-primary:focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary:hover {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-secondary:focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-warning {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning:hover {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning:focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-dark {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark:hover {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
}
.dapp-core-component__main__btn-dark:focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-link {
  font-weight: 400;
  color: #007bff;
  text-decoration: none;
}
.dapp-core-component__main__btn-link:hover {
  color: #0056b3;
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:focus, .dapp-core-component__main__btn-link.dapp-core-component__main__focus {
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:disabled, .dapp-core-component__main__btn-link.dapp-core-component__main__disabled {
  color: #6c757d;
  pointer-events: none;
}

.dapp-core-component__main__btn-light {
  color: #212529;
  background-color: #fafafa;
  border-color: #fafafa;
}
.dapp-core-component__main__btn-light .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}
.dapp-core-component__main__btn-light .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #f8f9fa;
  border-color: #f8f9fa;
}
.dapp-core-component__main__btn-light:hover {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
}
.dapp-core-component__main__btn-light:focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .dapp-core-component__main__badge {
    transition: none;
  }
}
a.dapp-core-component__main__badge:hover, a.dapp-core-component__main__badge:focus {
  text-decoration: none;
}

.dapp-core-component__main__badge:empty {
  display: none;
}

.dapp-core-component__main__btn .dapp-core-component__main__badge {
  position: relative;
  top: -1px;
}

.dapp-core-component__main__badge-pill {
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.dapp-core-component__main__badge-primary {
  color: #fff;
  background-color: #007bff;
}

a.dapp-core-component__main__badge-primary:hover, a.dapp-core-component__main__badge-primary:focus {
  color: #fff;
  background-color: #0062cc;
}

a.dapp-core-component__main__badge-primary:focus, a.dapp-core-component__main__badge-primary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.5);
}

.dapp-core-component__main__badge-secondary {
  color: #fff;
  background-color: #6c757d;
}

a.dapp-core-component__main__badge-secondary:hover, a.dapp-core-component__main__badge-secondary:focus {
  color: #fff;
  background-color: #545b62;
}

a.dapp-core-component__main__badge-secondary:focus, a.dapp-core-component__main__badge-secondary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108, 117, 125, 0.5);
}

.dapp-core-component__main__badge-success {
  color: #fff;
  background-color: #28a745;
}

a.dapp-core-component__main__badge-success:hover, a.dapp-core-component__main__badge-success:focus {
  color: #fff;
  background-color: #1e7e34;
}

a.dapp-core-component__main__badge-success:focus, a.dapp-core-component__main__badge-success.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
}

.dapp-core-component__main__badge-info {
  color: #fff;
  background-color: #17a2b8;
}

a.dapp-core-component__main__badge-info:hover, a.dapp-core-component__main__badge-info:focus {
  color: #fff;
  background-color: #117a8b;
}

a.dapp-core-component__main__badge-info:focus, a.dapp-core-component__main__badge-info.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(23, 162, 184, 0.5);
}

.dapp-core-component__main__badge-warning {
  color: #212529;
  background-color: #ffc107;
}

a.dapp-core-component__main__badge-warning:hover, a.dapp-core-component__main__badge-warning:focus {
  color: #212529;
  background-color: #d39e00;
}

a.dapp-core-component__main__badge-warning:focus, a.dapp-core-component__main__badge-warning.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 193, 7, 0.5);
}

.dapp-core-component__main__badge-danger {
  color: #fff;
  background-color: #dc3545;
}

a.dapp-core-component__main__badge-danger:hover, a.dapp-core-component__main__badge-danger:focus {
  color: #fff;
  background-color: #bd2130;
}

a.dapp-core-component__main__badge-danger:focus, a.dapp-core-component__main__badge-danger.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(220, 53, 69, 0.5);
}

.dapp-core-component__main__badge-light {
  color: #212529;
  background-color: #f8f9fa;
}

a.dapp-core-component__main__badge-light:hover, a.dapp-core-component__main__badge-light:focus {
  color: #212529;
  background-color: #dae0e5;
}

a.dapp-core-component__main__badge-light:focus, a.dapp-core-component__main__badge-light.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(248, 249, 250, 0.5);
}

.dapp-core-component__main__badge-dark {
  color: #fff;
  background-color: #343a40;
}

a.dapp-core-component__main__badge-dark:hover, a.dapp-core-component__main__badge-dark:focus {
  color: #fff;
  background-color: #1d2124;
}

a.dapp-core-component__main__badge-dark:focus, a.dapp-core-component__main__badge-dark.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}

.dapp-core-component__main__card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 5px;
}
.dapp-core-component__main__card > hr {
  margin-right: 0;
  margin-left: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group {
  border-top: inherit;
  border-bottom: inherit;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:first-child {
  border-top-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:last-child {
  border-bottom-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__card-header + .dapp-core-component__main__list-group,
.dapp-core-component__main__card > .dapp-core-component__main__list-group + .dapp-core-component__main__card-footer {
  border-top: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header {
  border: 0;
  padding: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header:first-child {
  border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item {
  border-bottom: 1px solid #dee2e6;
  padding: 1.1rem 1rem;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item h6,
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item .dapp-core-component__main__h6 {
  line-height: 2.5rem;
  letter-spacing: unset;
  margin: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item.dapp-core-component__main__compact {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fafafa;
}

.dapp-core-component__main__card-body {
  flex: 1 1 auto;
  min-height: 1px;
  padding: 1.25rem;
}

.dapp-core-component__main__card-title {
  margin-bottom: 0.75rem;
}

.dapp-core-component__main__card-subtitle {
  margin-top: -0.375rem;
  margin-bottom: 0;
}

.dapp-core-component__main__card-text:last-child {
  margin-bottom: 0;
}

.dapp-core-component__main__card-link:hover {
  text-decoration: none;
}
.dapp-core-component__main__card-link + .dapp-core-component__main__card-link {
  margin-left: 1.25rem;
}

.dapp-core-component__main__card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: #fff;
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__card-footer {
  padding: 0.75rem 1.25rem;
  background-color: #fff;
  border-top: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__text {
  color: #9dabbd;
}

.dapp-core-component__main__text-white {
  color: #fff;
}

.dapp-core-component__main__text-monospace {
  font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-size-xl .dapp-core-component__main__trim {
  max-width: 13rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim {
  max-width: 10rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-only-sm .dapp-core-component__main__trim {
  max-width: none;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-left {
  text-align: left;
}

.dapp-core-component__main__text-right {
  text-align: right;
}

.dapp-core-component__main__text-center {
  text-align: center;
}

.dapp-core-component__main__text-primary {
  color: #1b46c2;
}

.dapp-core-component__main__text-secondary {
  color: #6c757d;
}

.dapp-core-component__main__text-primary-highlight {
  color: #1b46c2;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-success {
  color: #28a745;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-body {
  color: #212529;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-black-50 {
  color: rgba(0, 0, 0, 0.5);
}

.dapp-core-component__main__text-white-50 {
  color: rgba(255, 255, 255, 0.5);
}

@media (min-width: 992px) {
  .dapp-core-component__main__text-lg-right {
    text-align: right;
  }
}

.dapp-core-component__main__text-hide {
  font: 0/0 a;
  color: rgba(0, 0, 0, 0);
  text-shadow: none;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
}

.dapp-core-component__main__text-decoration-none {
  text-decoration: none;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__text-reset {
  color: inherit;
}

.dapp-core-component__main__text-lowercase {
  text-transform: lowercase;
}

.dapp-core-component__main__text-uppercase {
  text-transform: uppercase;
}

.dapp-core-component__main__text-capitalize {
  text-transform: capitalize;
}

.dapp-core-component__main__text-danger {
  color: #d9534f;
}

.dapp-core-component__main__text-warning {
  color: #f0ad4e;
}

.dapp-core-component__main__trim-fs-sm {
  color: inherit;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__m-0 {
  margin: 0;
}

.dapp-core-component__main__mt-0 {
  margin-top: 0;
}

.dapp-core-component__main__my-0 {
  margin-top: 0;
  margin-bottom: 0;
}

.dapp-core-component__main__mr-0 {
  margin-right: 0;
}

.dapp-core-component__main__mx-0 {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__mb-0 {
  margin-bottom: 0;
}

.dapp-core-component__main__my-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.dapp-core-component__main__ml-0 {
  margin-left: 0;
}

.dapp-core-component__main__mx-0 {
  margin-left: 0;
  margin-right: 0;
}

.dapp-core-component__main__m-1 {
  margin: 0.25rem;
}

.dapp-core-component__main__mt-1 {
  margin-top: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__mr-1 {
  margin-right: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.dapp-core-component__main__mb-1 {
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-bottom: 0.25rem;
  margin-top: 0.25rem;
}

.dapp-core-component__main__ml-1 {
  margin-left: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-right: 0.25rem;
  margin-left: 0.25rem;
}

.dapp-core-component__main__m-2 {
  margin: 0.5rem;
}

.dapp-core-component__main__mt-2 {
  margin-top: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-bottom: 0.5rem;
  margin-top: 0.5rem;
}

.dapp-core-component__main__mr-2 {
  margin-right: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__mb-2 {
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__ml-2 {
  margin-left: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__m-3 {
  margin: 1rem;
}

.dapp-core-component__main__mt-3,
.dapp-core-component__main__my-3 {
  margin-top: 1rem;
}

.dapp-core-component__main__mr-3 {
  margin-right: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__mb-3 {
  margin-bottom: 1rem;
}

.dapp-core-component__main__my-3 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.dapp-core-component__main__ml-3 {
  margin-left: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__m-4 {
  margin: 1.5rem;
}

.dapp-core-component__main__mt-4 {
  margin-top: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__mr-4 {
  margin-right: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-right: 1.5rem;
  margin-left: 1.5rem;
}

.dapp-core-component__main__mb-4 {
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.dapp-core-component__main__ml-4 {
  margin-left: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}

.dapp-core-component__main__m-5 {
  margin: 3rem;
}

.dapp-core-component__main__mt-5 {
  margin-top: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__mr-5 {
  margin-right: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-right: 3rem;
  margin-left: 3rem;
}

.dapp-core-component__main__mb-5 {
  margin-bottom: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__ml-5 {
  margin-left: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-left: 3rem;
  margin-right: 3rem;
}

.dapp-core-component__main__p-0 {
  padding: 0;
}

.dapp-core-component__main__pt-0 {
  padding-top: 0;
}

.dapp-core-component__main__py-0 {
  padding-top: 0;
  padding-bottom: 0;
}

.dapp-core-component__main__pr-0 {
  padding-right: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__pb-0 {
  padding-bottom: 0;
}

.dapp-core-component__main__py-0 {
  padding-bottom: 0;
  padding-top: 0;
}

.dapp-core-component__main__pl-0 {
  padding-left: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__p-1 {
  padding: 0.25rem;
}

.dapp-core-component__main__pt-1 {
  padding-top: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pr-1 {
  padding-right: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-right: 0.25rem;
  padding-left: 0.25rem;
}

.dapp-core-component__main__pb-1 {
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pl-1 {
  padding-left: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

.dapp-core-component__main__p-2 {
  padding: 0.5rem;
}

.dapp-core-component__main__pt-2 {
  padding-top: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.dapp-core-component__main__pr-2 {
  padding-right: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__pb-2 {
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__pl-2 {
  padding-left: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__p-3 {
  padding: 1rem;
}

.dapp-core-component__main__pt-3 {
  padding-top: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.dapp-core-component__main__pr-3 {
  padding-right: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__pb-3 {
  padding-bottom: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.dapp-core-component__main__pl-3 {
  padding-left: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__p-4 {
  padding: 1.5rem;
}

.dapp-core-component__main__pt-4 {
  padding-top: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pr-4 {
  padding-right: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.dapp-core-component__main__pb-4 {
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pl-4 {
  padding-left: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
}

.dapp-core-component__main__p-5 {
  padding: 3rem;
}

.dapp-core-component__main__pt-5 {
  padding-top: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-bottom: 3rem;
  padding-top: 3rem;
}

.dapp-core-component__main__pr-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__pb-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.dapp-core-component__main__pl-5 {
  padding-left: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__m-n1 {
  margin: -0.25rem;
}

.dapp-core-component__main__mt-n1 {
  margin-top: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__mr-n1 {
  margin-right: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__mb-n1 {
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-bottom: -0.25rem;
  margin-top: -0.25rem;
}

.dapp-core-component__main__ml-n1 {
  margin-left: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__m-n2 {
  margin: -0.5rem;
}

.dapp-core-component__main__mt-n2 {
  margin-top: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-top: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mr-n2 {
  margin-right: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mb-n2 {
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-bottom: -0.5rem;
  margin-top: -0.5rem;
}

.dapp-core-component__main__ml-n2 {
  margin-left: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.dapp-core-component__main__m-n3 {
  margin: -1rem;
}

.dapp-core-component__main__mt-n3 {
  margin-top: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-top: -1rem;
  margin-bottom: -1rem;
}

.dapp-core-component__main__mr-n3 {
  margin-right: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-right: -1rem;
  margin-left: -1rem;
}

.dapp-core-component__main__mb-n3 {
  margin-bottom: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-bottom: -1rem;
  margin-top: -1rem;
}

.dapp-core-component__main__ml-n3 {
  margin-left: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-left: -1rem;
  margin-right: -1rem;
}

.dapp-core-component__main__m-n4 {
  margin: -1.5rem;
}

.dapp-core-component__main__mt-n4 {
  margin-top: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-top: -1.5rem;
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__mr-n4 {
  margin-right: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__mb-n4 {
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-bottom: -1.5rem;
  margin-top: -1.5rem;
}

.dapp-core-component__main__ml-n4 {
  margin-left: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__m-n5 {
  margin: -3rem;
}

.dapp-core-component__main__mt-n5 {
  margin-top: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-bottom: -3rem;
  margin-top: -3rem;
}

.dapp-core-component__main__mr-n5 {
  margin-right: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-left: -3rem;
  margin-right: -3rem;
}

.dapp-core-component__main__mb-n5 {
  margin-bottom: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-top: -3rem;
  margin-bottom: -3rem;
}

.dapp-core-component__main__ml-n5 {
  margin-left: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-right: -3rem;
  margin-left: -3rem;
}

.dapp-core-component__main__m-auto {
  margin: auto;
}

.dapp-core-component__main__mt-auto {
  margin-top: auto;
}

.dapp-core-component__main__my-auto {
  margin-bottom: auto;
  margin-top: auto;
}

.dapp-core-component__main__mr-auto {
  margin-right: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__mb-auto {
  margin-bottom: auto;
}

.dapp-core-component__main__my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.dapp-core-component__main__ml-auto {
  margin-left: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-1 {
    margin-right: 0.25rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-4 {
    margin-right: 1.5rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mx-lg-4 {
    margin-left: 1.5rem;
    margin-right: 1.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__mb-xl-2 {
    margin-bottom: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__pr-sm-2 {
    padding-right: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__px-sm-2 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__pr-xl-0,
.dapp-core-component__main__px-xl-0 {
    padding-right: 0;
  }
}

.dapp-core-component__main__col {
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

.dapp-core-component__main__col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__col-xl,
.dapp-core-component__main__col-xl-auto,
.dapp-core-component__main__col-xl-12,
.dapp-core-component__main__col-xl-11,
.dapp-core-component__main__col-xl-10,
.dapp-core-component__main__col-xl-9,
.dapp-core-component__main__col-xl-8,
.dapp-core-component__main__col-xl-7,
.dapp-core-component__main__col-xl-6,
.dapp-core-component__main__col-xl-5,
.dapp-core-component__main__col-xl-4,
.dapp-core-component__main__col-xl-3,
.dapp-core-component__main__col-xl-2,
.dapp-core-component__main__col-xl-1,
.dapp-core-component__main__col-lg,
.dapp-core-component__main__col-lg-auto,
.dapp-core-component__main__col-lg-12,
.dapp-core-component__main__col-lg-11,
.dapp-core-component__main__col-lg-10,
.dapp-core-component__main__col-lg-9,
.dapp-core-component__main__col-lg-8,
.dapp-core-component__main__col-lg-7,
.dapp-core-component__main__col-lg-6,
.dapp-core-component__main__col-lg-5,
.dapp-core-component__main__col-lg-4,
.dapp-core-component__main__col-lg-3,
.dapp-core-component__main__col-lg-2,
.dapp-core-component__main__col-lg-1,
.dapp-core-component__main__col-md,
.dapp-core-component__main__col-md-auto,
.dapp-core-component__main__col-md-12,
.dapp-core-component__main__col-md-11,
.dapp-core-component__main__col-md-10,
.dapp-core-component__main__col-md-9,
.dapp-core-component__main__col-md-8,
.dapp-core-component__main__col-md-7,
.dapp-core-component__main__col-md-6,
.dapp-core-component__main__col-md-5,
.dapp-core-component__main__col-md-4,
.dapp-core-component__main__col-md-3,
.dapp-core-component__main__col-md-2,
.dapp-core-component__main__col-md-1,
.dapp-core-component__main__col-sm,
.dapp-core-component__main__col-sm-auto,
.dapp-core-component__main__col-sm-12,
.dapp-core-component__main__col-sm-11,
.dapp-core-component__main__col-sm-10,
.dapp-core-component__main__col-sm-9,
.dapp-core-component__main__col-sm-8,
.dapp-core-component__main__col-sm-7,
.dapp-core-component__main__col-sm-6,
.dapp-core-component__main__col-sm-5,
.dapp-core-component__main__col-sm-4,
.dapp-core-component__main__col-sm-3,
.dapp-core-component__main__col-sm-2,
.dapp-core-component__main__col-sm-1,
.dapp-core-component__main__col,
.dapp-core-component__main__col-auto,
.dapp-core-component__main__col-12,
.dapp-core-component__main__col-11,
.dapp-core-component__main__col-10,
.dapp-core-component__main__col-9,
.dapp-core-component__main__col-8,
.dapp-core-component__main__col-7,
.dapp-core-component__main__col-6,
.dapp-core-component__main__col-5,
.dapp-core-component__main__col-4,
.dapp-core-component__main__col-3,
.dapp-core-component__main__col-2,
.dapp-core-component__main__col-1 {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
}

.dapp-core-component__main__col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

@media (min-width: 576px) {
  .dapp-core-component__main__col-sm-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-sm-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-sm-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-sm-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-sm-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-sm-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-sm-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-sm-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-sm-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-sm-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-sm-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-sm-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 768px) {
  .dapp-core-component__main__col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__col-xl-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-xl-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-xl-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-xl-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-xl-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-xl-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-xl-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-xl-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-xl-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-xl-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-xl-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-xl-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
.dapp-core-component__main__flex-row {
  flex-direction: row;
}

.dapp-core-component__main__flex-column {
  flex-direction: column;
}

.dapp-core-component__main__flex-row-reverse {
  flex-direction: row-reverse;
}

.dapp-core-component__main__flex-column-reverse {
  flex-direction: column-reverse;
}

.dapp-core-component__main__flex-wrap {
  flex-wrap: wrap;
}

.dapp-core-component__main__flex-nowrap {
  flex-wrap: nowrap;
}

.dapp-core-component__main__flex-wrap-reverse {
  flex-wrap: wrap-reverse;
}

.dapp-core-component__main__flex-fill {
  flex: 1 1 auto;
}

.dapp-core-component__main__flex-grow-0 {
  flex-grow: 0;
}

.dapp-core-component__main__flex-grow-1 {
  flex-grow: 1;
}

.dapp-core-component__main__flex-shrink-0 {
  flex-shrink: 0;
}

.dapp-core-component__main__flex-shrink-1 {
  flex-shrink: 1;
}

.dapp-core-component__main__justify-content-start {
  justify-content: flex-start;
}

.dapp-core-component__main__justify-content-end {
  justify-content: flex-end;
}

.dapp-core-component__main__justify-content-center {
  justify-content: center;
}

.dapp-core-component__main__justify-content-between {
  justify-content: space-between;
}

.dapp-core-component__main__justify-content-around {
  justify-content: space-around;
}

.dapp-core-component__main__align-items-start {
  align-items: flex-start;
}

.dapp-core-component__main__align-items-end {
  align-items: flex-end;
}

.dapp-core-component__main__align-items-center {
  align-items: center;
}

.dapp-core-component__main__align-items-baseline {
  align-items: baseline;
}

.dapp-core-component__main__align-items-stretch {
  align-items: stretch;
}

.dapp-core-component__main__align-content-start {
  align-content: flex-start;
}

.dapp-core-component__main__align-content-end {
  align-content: flex-end;
}

.dapp-core-component__main__align-content-center {
  align-content: center;
}

.dapp-core-component__main__align-content-between {
  align-content: space-between;
}

.dapp-core-component__main__align-content-around {
  align-content: space-around;
}

.dapp-core-component__main__align-content-stretch {
  align-content: stretch;
}

.dapp-core-component__main__align-self-auto {
  align-self: auto;
}

.dapp-core-component__main__align-self-start {
  align-self: flex-start;
}

.dapp-core-component__main__align-self-end {
  align-self: flex-end;
}

.dapp-core-component__main__align-self-center {
  align-self: center;
}

.dapp-core-component__main__align-self-baseline {
  align-self: baseline;
}

.dapp-core-component__main__align-self-stretch {
  align-self: stretch;
}

@media (min-width: 576px) {
  .dapp-core-component__main__flex-sm-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-wrap {
    flex-wrap: wrap;
  }
}

.dapp-core-component__main__form-check {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: auto;
  padding-left: 0;
}

.dapp-core-component__main__form-check-label {
  margin-bottom: 0;
}

.dapp-core-component__main__form-check-input {
  position: absolute;
  margin-top: 0.3rem;
  margin-left: -1.25rem;
}
.dapp-core-component__main__form-check-input[disabled] ~ .dapp-core-component__main__form-check-label, .dapp-core-component__main__form-check-input:disabled ~ .dapp-core-component__main__form-check-label {
  color: #6c757d;
}

.dapp-core-component__main__form-group {
  display: flex;
  flex: 0 0 auto;
  flex-flow: row wrap;
  align-items: center;
  margin-bottom: 0;
}

.dapp-core-component__main__form-control {
  display: block;
  width: 100%;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.dapp-core-component__main__form-label {
  padding-top: calc(0.375rem + 1px);
  padding-bottom: calc(0.375rem + 1px);
  margin-bottom: 0;
  font-size: inherit;
  line-height: 1.5;
}

input,
button,
select,
optgroup,
textarea {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.dapp-core-component__main__form-group {
  margin-bottom: 1rem;
}

.dapp-core-component__main__form-text {
  display: block;
  margin-top: 0.25rem;
  color: inherit;
}

.dapp-core-component__main__border {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__border-top {
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__border-right {
  border-right: 1px solid #dee2e6;
}

.dapp-core-component__main__border-bottom {
  border-bottom: 1px solid #dee2e6;
}

.dapp-core-component__main__border-left {
  border-left: 1px solid #dee2e6;
}

.dapp-core-component__main__border-0 {
  border: 0;
}

.dapp-core-component__main__border-top-0 {
  border-top: 0;
}

.dapp-core-component__main__border-right-0 {
  border-right: 0;
}

.dapp-core-component__main__border-bottom-0 {
  border-bottom: 0;
}

.dapp-core-component__main__border-left-0 {
  border-left: 0;
}

.dapp-core-component__main__rounded-sm {
  border-radius: 0.2rem;
}

.dapp-core-component__main__rounded {
  border-radius: 0.25rem;
}

.dapp-core-component__main__rounded-top {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-right {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-bottom {
  border-bottom-right-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-left {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-lg {
  border-radius: 0.3rem;
}

.dapp-core-component__main__rounded-circle {
  border-radius: 50%;
}

.dapp-core-component__main__rounded-pill {
  border-radius: 50rem;
}

.dapp-core-component__main__rounded-0 {
  border-radius: 0;
}

.dapp-core-component__main__h1,
.dapp-core-component__main__h2,
.dapp-core-component__main__h3,
.dapp-core-component__main__h4,
.dapp-core-component__main__h5,
.dapp-core-component__main__h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2;
}

.dapp-core-component__main__h1 {
  font-size: 2.5rem;
}

.dapp-core-component__main__h2 {
  font-size: 2rem;
}

.dapp-core-component__main__h3 {
  font-size: 1.75rem;
}

.dapp-core-component__main__h4 {
  font-size: 1.5rem;
}

.dapp-core-component__main__h5 {
  font-size: 1.25rem;
}

.dapp-core-component__main__h6 {
  font-size: 1rem;
}

.dapp-core-component__main__lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

.dapp-core-component__main__bg-warning,
.dapp-core-component__main__warning {
  background-color: #f0ad4e;
}

.dapp-core-component__main__bg-danger,
.dapp-core-component__main__danger {
  background-color: #d9534f;
}

.dapp-core-component__main__bg-success,
.dapp-core-component__main__success {
  background-color: #5cb85c;
}

.dapp-core-component__main__position-relative {
  position: relative;
}

.dapp-core-component__main__position-absolute {
  position: absolute;
}

.dapp-core-component__main__w-25 {
  width: 25%;
}

.dapp-core-component__main__w-50 {
  width: 50%;
}

.dapp-core-component__main__w-75 {
  width: 75%;
}

.dapp-core-component__main__w-100 {
  width: 100%;
}

.dapp-core-component__main__w-auto {
  width: auto;
}

.dapp-core-component__main__h-25 {
  height: 25%;
}

.dapp-core-component__main__h-50 {
  height: 50%;
}

.dapp-core-component__main__h-75 {
  height: 75%;
}

.dapp-core-component__main__h-100 {
  height: 100%;
}

.dapp-core-component__main__h-auto {
  height: auto;
}

.dapp-core-component__main__mw-100 {
  max-width: 100%;
}

.dapp-core-component__main__mh-100 {
  max-height: 100%;
}

.dapp-core-component__main__min-vw-100 {
  min-width: 100vw;
}

.dapp-core-component__main__min-vh-100 {
  min-height: 100vh;
}

.dapp-core-component__main__vw-100 {
  width: 100vw;
}

.dapp-core-component__main__vh-100 {
  height: 100vh;
}

.dapp-core-component__main__locked-icon {
  color: inherit;
}

.dapp-core-component__main__shadow-sm {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
}

.dapp-core-component__main__shadow {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.dapp-core-component__main__shadow-lg {
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.175);
}

.dapp-core-component__main__shadow-none {
  box-shadow: none;
}

.dapp-core-component__main__d-none {
  display: none;
}

.dapp-core-component__main__d-inline {
  display: inline;
}

.dapp-core-component__main__d-inline-block {
  display: inline-block;
}

.dapp-core-component__main__d-block {
  display: block;
}

.dapp-core-component__main__d-table {
  display: table;
}

.dapp-core-component__main__d-table-row {
  display: table-row;
}

.dapp-core-component__main__d-table-cell {
  display: table-cell;
}

.dapp-core-component__main__d-flex {
  display: -ms-flexbox;
  display: flex;
}

.dapp-core-component__main__d-inline-flex {
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media (min-width: 576px) {
  .dapp-core-component__main__d-sm-flex {
    display: flex;
  }
}

.dapp-core-component__main__table {
  width: 100%;
  margin-bottom: 1rem;
  color: #212529;
}
.dapp-core-component__main__table th,
.dapp-core-component__main__table td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #dee2e6;
}
.dapp-core-component__main__table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #dee2e6;
}
.dapp-core-component__main__table tbody + tbody {
  border-top: 2px solid #dee2e6;
}

.dapp-core-component__main__table-sm th,
.dapp-core-component__main__table-sm td {
  padding: 0.3rem;
}

.dapp-core-component__main__table-bordered {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered th,
.dapp-core-component__main__table-bordered td {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered thead th,
.dapp-core-component__main__table-bordered thead td {
  border-bottom-width: 2px;
}

.dapp-core-component__main__table-borderless th,
.dapp-core-component__main__table-borderless td,
.dapp-core-component__main__table-borderless thead th,
.dapp-core-component__main__table-borderless tbody + tbody {
  border: 0;
}

.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(0, 0, 0, 0.05);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-dark th {
  color: #fff;
  background-color: #343a40;
  border-color: #454d55;
}
.dapp-core-component__main__table .dapp-core-component__main__thead-light th {
  color: #495057;
  background-color: #e9ecef;
  border-color: #dee2e6;
}

.dapp-core-component__main__table-dark {
  color: #fff;
  background-color: #343a40;
}
.dapp-core-component__main__table-dark th,
.dapp-core-component__main__table-dark td,
.dapp-core-component__main__table-dark thead th {
  border-color: #454d55;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-bordered {
  border: 0;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(255, 255, 255, 0.05);
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-hover tbody tr {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.075);
}

.dapp-core-component__main__table-responsive {
  display: block;
  width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}
.dapp-core-component__main__table-responsive > .dapp-core-component__main__table-bordered {
  border: 0;
}

@media (max-width: 767.98px) {
  .dapp-core-component__main__transactions.dapp-core-component__main__table .dapp-core-component__main__transaction-hash .dapp-core-component__main__trim {
    max-width: 8rem;
  }
}
.dapp-core-component__main__font-weight-light {
  font-weight: 300 !important;
}

.dapp-core-component__main__font-weight-lighter {
  font-weight: lighter !important;
}

.dapp-core-component__main__font-weight-normal {
  font-weight: 400 !important;
}

.dapp-core-component__main__font-weight-medium {
  font-weight: 500 !important;
}

.dapp-core-component__main__font-weight-bold {
  font-weight: 700 !important;
}

.dapp-core-component__main__font-weight-bolder {
  font-weight: bolder !important;
}

.dapp-core-component__main__font-italic {
  font-style: italic !important;
}

.dapp-core-component__main__icon-state {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
}
.dapp-core-component__main__icon-state.dapp-core-component__main__half {
  width: 3rem;
  height: 3rem;
}

.dapp-core-component__main__side-icon {
  width: 1.5rem;
  height: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  color: transparent;
  border-radius: 50%;
  overflow: hidden;
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__side-icon svg {
  color: #cacbcc;
}

div.dapp-core-component__main__side-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
div.dapp-core-component__main__side-icon img {
  transform: scale(0.75);
}

.dapp-core-component__main__toast-messages {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast {
  display: block;
  flex-basis: 4rem;
  width: 100%;
  max-width: 28.875rem;
  transition: transform 0.3s ease-in-out;
  transform: translateX(120%);
  background-color: #fff;
  border-radius: 0.25rem;
  border-color: #dee2e6;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__clickable {
  cursor: pointer;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible {
  margin-bottom: 0.3rem;
  transform: translateX(0);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress {
  background-color: rgba(0, 0, 0, 0);
  height: inherit;
  font-size: inherit;
  line-height: inherit;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar {
  background-color: rgba(108, 117, 125, 0.2);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar .dapp-core-component__main__content-height {
  z-index: -1;
  visibility: hidden;
  width: 0 !important;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast .dapp-core-component__main__tx-description {
  margin-bottom: 0.2rem;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close {
  opacity: 1;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close svg {
  color: #6c757d;
}

.dapp-core-component__main__data-field .dapp-core-component__main__textarea {
  -moz-appearance: textfield-multiline;
  -webkit-appearance: textarea;
  border: 1px solid #000;
  background-color: #fff;
  border-radius: 5px;
  font: medium -moz-fixed;
  font-size: 0.8125rem;
  min-height: 5rem;
  overflow: auto;
  padding: 0.375rem 0.75rem;
  resize: both;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea:hover {
  border: 1px solid #000;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea .dapp-core-component__main__highlighted {
  font-size: 1rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea.dapp-core-component__main__sc-call {
  min-height: 3rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__form-control:disabled,
.dapp-core-component__main__data-field .dapp-core-component__main__form-control[readonly] {
  background-color: #fafafa;
}
.dapp-core-component__main__data-field .dapp-core-component__main__small-font {
  font-size: 0.8125rem;
}

.dapp-core-component__main__progress-steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps hr {
  background-color: #6c757d;
  border-color: #dee2e6;
  z-index: 2;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot {
  height: 0.625rem;
  width: 0.625rem;
  border-radius: 50%;
  margin: 0 1rem;
  z-index: 5;
  background-color: #6c757d;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:first-of-type {
  margin-left: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:last-of-type {
  margin-right: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

a.dapp-core-component__main__side-link::after {
  right: 0;
  left: unset;
  width: calc(100% - 1.85rem);
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(ds));D();B();q();t();var us=v(require("bignumber.js"));D();t();var fs=require("@elrondnetwork/erdjs"),hs=v(require("bignumber.js"));D();O();Ue();function Ts({transactionId:e,onSuccess:o,onFail:n,onCancelled:r,onTimedOut:a}){var C;let{signedTransactionsArray:m}=vt(),s=le("wallet"),[d]=m.length>0?m[m.length-1]:[],l=(C=e!=null?e:d)!=null?C:null,u=s?l:e,f=k(X=>ni(X,u)),{status:g,transactions:y,errorMessage:h}=f,E=te(g),L=he(g),Z=Ro(g),N=fe(g),A=g==="cancelled";function P(){s&&(Pi(),Ci())}return(0,fo.useEffect)(()=>{N&&o&&o(u)},[N]),(0,fo.useEffect)(()=>{L&&n&&(n(u,h),P())},[L]),(0,fo.useEffect)(()=>{A&&r&&(r(u),P())},[A]),(0,fo.useEffect)(()=>{Z&&(a?a(u):n==null||n(u,"timeout"))},[Z]),u==null?{}:f==null?{errorMessage:"No transaction to track"}:{isPending:E,isSuccessful:N,isFailed:L,isCancelled:A,errorMessage:h,status:g,transactions:y}}t();function xs(){let e=k(nn),o=Object.entries(e),n=(o==null?void 0:o.length)>0;return{failedTransactions:e,failedTransactionsArray:o,hasFailedTransactions:n}}t();function ys(){let e=k(on),o=Object.entries(e),n=(o==null?void 0:o.length)>0;return{successfulTransactions:e,successfulTransactionsArray:o,hasSuccessfulTransactions:n}}t();function bs(){var l,u,f;let e=k(oi),o=k(nn),n=k(on),r=k(en),a=((l=Object.keys(r))==null?void 0:l.length)>0,m=!a&&((u=Object.keys(e))==null?void 0:u.length)>0,s=!a&&!m&&((f=Object.keys(o))==null?void 0:f.length)>0,d=!a&&!m&&!s&&Object.keys(n).length>0;return{pending:a,timedOut:m,fail:s,success:d}}0&&(module.exports={useCheckTransactionStatus,useGetActiveTransactionsStatus,useGetFailedTransactions,useGetPendingTransactions,useGetSignTransactionsError,useGetSignedTransactions,useGetSuccessfulTransactions,useGetTokenDetails,useGetTransactionDisplayInfo,useParseMultiEsdtTransferData,useParseSignedTransactions,useSignMultipleTransactions,useSignTransactions,useSignTransactionsWithDevice,useSignTransactionsWithLedger,useTrackTransactionStatus});
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
