// @elrondnetwork/dapp-core/__commonjs/UI/TransactionInfo/components/transactionInfoFields/tests/TransactionInfoFee.test.js
"use strict";var ia=Object.create;var Te=Object.defineProperty,ca=Object.defineProperties,pa=Object.getOwnPropertyDescriptor,ma=Object.getOwnPropertyDescriptors,sa=Object.getOwnPropertyNames,Re=Object.getOwnPropertySymbols,da=Object.getPrototypeOf,hn=Object.prototype.hasOwnProperty,yo=Object.prototype.propertyIsEnumerable;var xo=(e,n,o)=>n in e?Te(e,n,{enumerable:!0,configurable:!0,writable:!0,value:o}):e[n]=o,x=(e,n)=>{for(var o in n||(n={}))hn.call(n,o)&&xo(e,o,n[o]);if(Re)for(var o of Re(n))yo.call(n,o)&&xo(e,o,n[o]);return e},v=(e,n)=>ca(e,ma(n));var To=(e,n)=>{var o={};for(var t in e)hn.call(e,t)&&n.indexOf(t)<0&&(o[t]=e[t]);if(e!=null&&Re)for(var t of Re(e))n.indexOf(t)<0&&yo.call(e,t)&&(o[t]=e[t]);return o};var h=(e,n)=>()=>(e&&(n=e(e=0)),n);var Ce=(e,n)=>()=>(n||e((n={exports:{}}).exports,n),n.exports),pe=(e,n)=>{for(var o in n)Te(e,o,{get:n[o],enumerable:!0})},wo=(e,n,o,t)=>{if(n&&typeof n=="object"||typeof n=="function")for(let a of sa(n))!hn.call(e,a)&&a!==o&&Te(e,a,{get:()=>n[a],enumerable:!(t=pa(n,a))||t.enumerable});return e};var T=(e,n,o)=>(o=e!=null?ia(da(e)):{},wo(n||!e||!e.__esModule?Te(o,"default",{value:e,enumerable:!0}):o,e)),Fe=e=>wo(Te({},"__esModule",{value:!0}),e);var qo=(e,n,o)=>new Promise((t,a)=>{var i=_=>{try{l(o.next(_))}catch(g){a(g)}},c=_=>{try{l(o.throw(_))}catch(g){a(g)}},l=_=>_.done?t(_.value):Promise.resolve(_.value).then(i,c);l((o=o.apply(e,n)).next())});var So=Ce(Pe=>{"use strict";r();Pe.byteLength=la;Pe.toByteArray=ua;Pe.fromByteArray=ba;var j=[],L=[],_a=typeof Uint8Array!="undefined"?Uint8Array:Array,bn="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";for(K=0,ko=bn.length;K<ko;++K)j[K]=bn[K],L[bn.charCodeAt(K)]=K;var K,ko;L["-".charCodeAt(0)]=62;L["_".charCodeAt(0)]=63;function vo(e){var n=e.length;if(n%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var o=e.indexOf("=");o===-1&&(o=n);var t=o===n?0:4-o%4;return[o,t]}function la(e){var n=vo(e),o=n[0],t=n[1];return(o+t)*3/4-t}function ga(e,n,o){return(n+o)*3/4-o}function ua(e){var n,o=vo(e),t=o[0],a=o[1],i=new _a(ga(e,t,a)),c=0,l=a>0?t-4:t,_;for(_=0;_<l;_+=4)n=L[e.charCodeAt(_)]<<18|L[e.charCodeAt(_+1)]<<12|L[e.charCodeAt(_+2)]<<6|L[e.charCodeAt(_+3)],i[c++]=n>>16&255,i[c++]=n>>8&255,i[c++]=n&255;return a===2&&(n=L[e.charCodeAt(_)]<<2|L[e.charCodeAt(_+1)]>>4,i[c++]=n&255),a===1&&(n=L[e.charCodeAt(_)]<<10|L[e.charCodeAt(_+1)]<<4|L[e.charCodeAt(_+2)]>>2,i[c++]=n>>8&255,i[c++]=n&255),i}function fa(e){return j[e>>18&63]+j[e>>12&63]+j[e>>6&63]+j[e&63]}function ha(e,n,o){for(var t,a=[],i=n;i<o;i+=3)t=(e[i]<<16&16711680)+(e[i+1]<<8&65280)+(e[i+2]&255),a.push(fa(t));return a.join("")}function ba(e){for(var n,o=e.length,t=o%3,a=[],i=16383,c=0,l=o-t;c<l;c+=i)a.push(ha(e,c,c+i>l?l:c+i));return t===1?(n=e[o-1],a.push(j[n>>2]+j[n<<4&63]+"==")):t===2&&(n=(e[o-2]<<8)+e[o-1],a.push(j[n>>10]+j[n>>4&63]+j[n<<2&63]+"=")),a.join("")}});var Ao=Ce(xn=>{r();xn.read=function(e,n,o,t,a){var i,c,l=a*8-t-1,_=(1<<l)-1,g=_>>1,b=-7,f=o?a-1:0,y=o?-1:1,k=e[n+f];for(f+=y,i=k&(1<<-b)-1,k>>=-b,b+=l;b>0;i=i*256+e[n+f],f+=y,b-=8);for(c=i&(1<<-b)-1,i>>=-b,b+=t;b>0;c=c*256+e[n+f],f+=y,b-=8);if(i===0)i=1-g;else{if(i===_)return c?NaN:(k?-1:1)*(1/0);c=c+Math.pow(2,t),i=i-g}return(k?-1:1)*c*Math.pow(2,i-t)};xn.write=function(e,n,o,t,a,i){var c,l,_,g=i*8-a-1,b=(1<<g)-1,f=b>>1,y=a===23?Math.pow(2,-24)-Math.pow(2,-77):0,k=t?0:i-1,M=t?1:-1,A=n<0||n===0&&1/n<0?1:0;for(n=Math.abs(n),isNaN(n)||n===1/0?(l=isNaN(n)?1:0,c=b):(c=Math.floor(Math.log(n)/Math.LN2),n*(_=Math.pow(2,-c))<1&&(c--,_*=2),c+f>=1?n+=y/_:n+=y*Math.pow(2,1-f),n*_>=2&&(c++,_/=2),c+f>=b?(l=0,c=b):c+f>=1?(l=(n*_-1)*Math.pow(2,a),c=c+f):(l=n*Math.pow(2,f-1)*Math.pow(2,a),c=0));a>=8;e[o+k]=l&255,k+=M,l/=256,a-=8);for(c=c<<a|l,g+=a;g>0;e[o+k]=c&255,k+=M,c/=256,g-=8);e[o+k-M]|=A*128}});var Uo=Ce(se=>{"use strict";r();var yn=So(),me=Ao(),No=typeof Symbol=="function"&&typeof Symbol.for=="function"?Symbol.for("nodejs.util.inspect.custom"):null;se.Buffer=d;se.SlowBuffer=ka;se.INSPECT_MAX_BYTES=50;var je=2147483647;se.kMaxLength=je;d.TYPED_ARRAY_SUPPORT=xa();!d.TYPED_ARRAY_SUPPORT&&typeof console!="undefined"&&typeof console.error=="function"&&console.error("This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.");function xa(){try{var e=new Uint8Array(1),n={foo:function(){return 42}};return Object.setPrototypeOf(n,Uint8Array.prototype),Object.setPrototypeOf(e,n),e.foo()===42}catch(o){return!1}}Object.defineProperty(d.prototype,"parent",{enumerable:!0,get:function(){if(!!d.isBuffer(this))return this.buffer}});Object.defineProperty(d.prototype,"offset",{enumerable:!0,get:function(){if(!!d.isBuffer(this))return this.byteOffset}});function V(e){if(e>je)throw new RangeError('The value "'+e+'" is invalid for option "size"');var n=new Uint8Array(e);return Object.setPrototypeOf(n,d.prototype),n}function d(e,n,o){if(typeof e=="number"){if(typeof n=="string")throw new TypeError('The "string" argument must be of type string. Received type number');return kn(e)}return Mo(e,n,o)}d.poolSize=8192;function Mo(e,n,o){if(typeof e=="string")return Ta(e,n);if(ArrayBuffer.isView(e))return wa(e);if(e==null)throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e);if(z(e,ArrayBuffer)||e&&z(e.buffer,ArrayBuffer)||typeof SharedArrayBuffer!="undefined"&&(z(e,SharedArrayBuffer)||e&&z(e.buffer,SharedArrayBuffer)))return wn(e,n,o);if(typeof e=="number")throw new TypeError('The "value" argument must not be of type number. Received type number');var t=e.valueOf&&e.valueOf();if(t!=null&&t!==e)return d.from(t,n,o);var a=qa(e);if(a)return a;if(typeof Symbol!="undefined"&&Symbol.toPrimitive!=null&&typeof e[Symbol.toPrimitive]=="function")return d.from(e[Symbol.toPrimitive]("string"),n,o);throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e)}d.from=function(e,n,o){return Mo(e,n,o)};Object.setPrototypeOf(d.prototype,Uint8Array.prototype);Object.setPrototypeOf(d,Uint8Array);function Io(e){if(typeof e!="number")throw new TypeError('"size" argument must be of type number');if(e<0)throw new RangeError('The value "'+e+'" is invalid for option "size"')}function ya(e,n,o){return Io(e),e<=0?V(e):n!==void 0?typeof o=="string"?V(e).fill(n,o):V(e).fill(n):V(e)}d.alloc=function(e,n,o){return ya(e,n,o)};function kn(e){return Io(e),V(e<0?0:vn(e)|0)}d.allocUnsafe=function(e){return kn(e)};d.allocUnsafeSlow=function(e){return kn(e)};function Ta(e,n){if((typeof n!="string"||n==="")&&(n="utf8"),!d.isEncoding(n))throw new TypeError("Unknown encoding: "+n);var o=Lo(e,n)|0,t=V(o),a=t.write(e,n);return a!==o&&(t=t.slice(0,a)),t}function Tn(e){for(var n=e.length<0?0:vn(e.length)|0,o=V(n),t=0;t<n;t+=1)o[t]=e[t]&255;return o}function wa(e){if(z(e,Uint8Array)){var n=new Uint8Array(e);return wn(n.buffer,n.byteOffset,n.byteLength)}return Tn(e)}function wn(e,n,o){if(n<0||e.byteLength<n)throw new RangeError('"offset" is outside of buffer bounds');if(e.byteLength<n+(o||0))throw new RangeError('"length" is outside of buffer bounds');var t;return n===void 0&&o===void 0?t=new Uint8Array(e):o===void 0?t=new Uint8Array(e,n):t=new Uint8Array(e,n,o),Object.setPrototypeOf(t,d.prototype),t}function qa(e){if(d.isBuffer(e)){var n=vn(e.length)|0,o=V(n);return o.length===0||e.copy(o,0,0,n),o}if(e.length!==void 0)return typeof e.length!="number"||Sn(e.length)?V(0):Tn(e);if(e.type==="Buffer"&&Array.isArray(e.data))return Tn(e.data)}function vn(e){if(e>=je)throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x"+je.toString(16)+" bytes");return e|0}function ka(e){return+e!=e&&(e=0),d.alloc(+e)}d.isBuffer=function(n){return n!=null&&n._isBuffer===!0&&n!==d.prototype};d.compare=function(n,o){if(z(n,Uint8Array)&&(n=d.from(n,n.offset,n.byteLength)),z(o,Uint8Array)&&(o=d.from(o,o.offset,o.byteLength)),!d.isBuffer(n)||!d.isBuffer(o))throw new TypeError('The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array');if(n===o)return 0;for(var t=n.length,a=o.length,i=0,c=Math.min(t,a);i<c;++i)if(n[i]!==o[i]){t=n[i],a=o[i];break}return t<a?-1:a<t?1:0};d.isEncoding=function(n){switch(String(n).toLowerCase()){case"hex":case"utf8":case"utf-8":case"ascii":case"latin1":case"binary":case"base64":case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return!0;default:return!1}};d.concat=function(n,o){if(!Array.isArray(n))throw new TypeError('"list" argument must be an Array of Buffers');if(n.length===0)return d.alloc(0);var t;if(o===void 0)for(o=0,t=0;t<n.length;++t)o+=n[t].length;var a=d.allocUnsafe(o),i=0;for(t=0;t<n.length;++t){var c=n[t];if(z(c,Uint8Array))i+c.length>a.length?d.from(c).copy(a,i):Uint8Array.prototype.set.call(a,c,i);else if(d.isBuffer(c))c.copy(a,i);else throw new TypeError('"list" argument must be an Array of Buffers');i+=c.length}return a};function Lo(e,n){if(d.isBuffer(e))return e.length;if(ArrayBuffer.isView(e)||z(e,ArrayBuffer))return e.byteLength;if(typeof e!="string")throw new TypeError('The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type '+typeof e);var o=e.length,t=arguments.length>2&&arguments[2]===!0;if(!t&&o===0)return 0;for(var a=!1;;)switch(n){case"ascii":case"latin1":case"binary":return o;case"utf8":case"utf-8":return qn(e).length;case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return o*2;case"hex":return o>>>1;case"base64":return zo(e).length;default:if(a)return t?-1:qn(e).length;n=(""+n).toLowerCase(),a=!0}}d.byteLength=Lo;function va(e,n,o){var t=!1;if((n===void 0||n<0)&&(n=0),n>this.length||((o===void 0||o>this.length)&&(o=this.length),o<=0)||(o>>>=0,n>>>=0,o<=n))return"";for(e||(e="utf8");;)switch(e){case"hex":return Ca(this,n,o);case"utf8":case"utf-8":return Co(this,n,o);case"ascii":return La(this,n,o);case"latin1":case"binary":return Ra(this,n,o);case"base64":return Ma(this,n,o);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return Fa(this,n,o);default:if(t)throw new TypeError("Unknown encoding: "+e);e=(e+"").toLowerCase(),t=!0}}d.prototype._isBuffer=!0;function J(e,n,o){var t=e[n];e[n]=e[o],e[o]=t}d.prototype.swap16=function(){var n=this.length;if(n%2!==0)throw new RangeError("Buffer size must be a multiple of 16-bits");for(var o=0;o<n;o+=2)J(this,o,o+1);return this};d.prototype.swap32=function(){var n=this.length;if(n%4!==0)throw new RangeError("Buffer size must be a multiple of 32-bits");for(var o=0;o<n;o+=4)J(this,o,o+3),J(this,o+1,o+2);return this};d.prototype.swap64=function(){var n=this.length;if(n%8!==0)throw new RangeError("Buffer size must be a multiple of 64-bits");for(var o=0;o<n;o+=8)J(this,o,o+7),J(this,o+1,o+6),J(this,o+2,o+5),J(this,o+3,o+4);return this};d.prototype.toString=function(){var n=this.length;return n===0?"":arguments.length===0?Co(this,0,n):va.apply(this,arguments)};d.prototype.toLocaleString=d.prototype.toString;d.prototype.equals=function(n){if(!d.isBuffer(n))throw new TypeError("Argument must be a Buffer");return this===n?!0:d.compare(this,n)===0};d.prototype.inspect=function(){var n="",o=se.INSPECT_MAX_BYTES;return n=this.toString("hex",0,o).replace(/(.{2})/g,"$1 ").trim(),this.length>o&&(n+=" ... "),"<Buffer "+n+">"};No&&(d.prototype[No]=d.prototype.inspect);d.prototype.compare=function(n,o,t,a,i){if(z(n,Uint8Array)&&(n=d.from(n,n.offset,n.byteLength)),!d.isBuffer(n))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+typeof n);if(o===void 0&&(o=0),t===void 0&&(t=n?n.length:0),a===void 0&&(a=0),i===void 0&&(i=this.length),o<0||t>n.length||a<0||i>this.length)throw new RangeError("out of range index");if(a>=i&&o>=t)return 0;if(a>=i)return-1;if(o>=t)return 1;if(o>>>=0,t>>>=0,a>>>=0,i>>>=0,this===n)return 0;for(var c=i-a,l=t-o,_=Math.min(c,l),g=this.slice(a,i),b=n.slice(o,t),f=0;f<_;++f)if(g[f]!==b[f]){c=g[f],l=b[f];break}return c<l?-1:l<c?1:0};function Ro(e,n,o,t,a){if(e.length===0)return-1;if(typeof o=="string"?(t=o,o=0):o>2147483647?o=2147483647:o<-2147483648&&(o=-2147483648),o=+o,Sn(o)&&(o=a?0:e.length-1),o<0&&(o=e.length+o),o>=e.length){if(a)return-1;o=e.length-1}else if(o<0)if(a)o=0;else return-1;if(typeof n=="string"&&(n=d.from(n,t)),d.isBuffer(n))return n.length===0?-1:Eo(e,n,o,t,a);if(typeof n=="number")return n=n&255,typeof Uint8Array.prototype.indexOf=="function"?a?Uint8Array.prototype.indexOf.call(e,n,o):Uint8Array.prototype.lastIndexOf.call(e,n,o):Eo(e,[n],o,t,a);throw new TypeError("val must be string, number or Buffer")}function Eo(e,n,o,t,a){var i=1,c=e.length,l=n.length;if(t!==void 0&&(t=String(t).toLowerCase(),t==="ucs2"||t==="ucs-2"||t==="utf16le"||t==="utf-16le")){if(e.length<2||n.length<2)return-1;i=2,c/=2,l/=2,o/=2}function _(k,M){return i===1?k[M]:k.readUInt16BE(M*i)}var g;if(a){var b=-1;for(g=o;g<c;g++)if(_(e,g)===_(n,b===-1?0:g-b)){if(b===-1&&(b=g),g-b+1===l)return b*i}else b!==-1&&(g-=g-b),b=-1}else for(o+l>c&&(o=c-l),g=o;g>=0;g--){for(var f=!0,y=0;y<l;y++)if(_(e,g+y)!==_(n,y)){f=!1;break}if(f)return g}return-1}d.prototype.includes=function(n,o,t){return this.indexOf(n,o,t)!==-1};d.prototype.indexOf=function(n,o,t){return Ro(this,n,o,t,!0)};d.prototype.lastIndexOf=function(n,o,t){return Ro(this,n,o,t,!1)};function Sa(e,n,o,t){o=Number(o)||0;var a=e.length-o;t?(t=Number(t),t>a&&(t=a)):t=a;var i=n.length;t>i/2&&(t=i/2);for(var c=0;c<t;++c){var l=parseInt(n.substr(c*2,2),16);if(Sn(l))return c;e[o+c]=l}return c}function Aa(e,n,o,t){return ze(qn(n,e.length-o),e,o,t)}function Na(e,n,o,t){return ze(za(n),e,o,t)}function Ea(e,n,o,t){return ze(zo(n),e,o,t)}function Da(e,n,o,t){return ze(Ua(n,e.length-o),e,o,t)}d.prototype.write=function(n,o,t,a){if(o===void 0)a="utf8",t=this.length,o=0;else if(t===void 0&&typeof o=="string")a=o,t=this.length,o=0;else if(isFinite(o))o=o>>>0,isFinite(t)?(t=t>>>0,a===void 0&&(a="utf8")):(a=t,t=void 0);else throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");var i=this.length-o;if((t===void 0||t>i)&&(t=i),n.length>0&&(t<0||o<0)||o>this.length)throw new RangeError("Attempt to write outside buffer bounds");a||(a="utf8");for(var c=!1;;)switch(a){case"hex":return Sa(this,n,o,t);case"utf8":case"utf-8":return Aa(this,n,o,t);case"ascii":case"latin1":case"binary":return Na(this,n,o,t);case"base64":return Ea(this,n,o,t);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return Da(this,n,o,t);default:if(c)throw new TypeError("Unknown encoding: "+a);a=(""+a).toLowerCase(),c=!0}};d.prototype.toJSON=function(){return{type:"Buffer",data:Array.prototype.slice.call(this._arr||this,0)}};function Ma(e,n,o){return n===0&&o===e.length?yn.fromByteArray(e):yn.fromByteArray(e.slice(n,o))}function Co(e,n,o){o=Math.min(e.length,o);for(var t=[],a=n;a<o;){var i=e[a],c=null,l=i>239?4:i>223?3:i>191?2:1;if(a+l<=o){var _,g,b,f;switch(l){case 1:i<128&&(c=i);break;case 2:_=e[a+1],(_&192)===128&&(f=(i&31)<<6|_&63,f>127&&(c=f));break;case 3:_=e[a+1],g=e[a+2],(_&192)===128&&(g&192)===128&&(f=(i&15)<<12|(_&63)<<6|g&63,f>2047&&(f<55296||f>57343)&&(c=f));break;case 4:_=e[a+1],g=e[a+2],b=e[a+3],(_&192)===128&&(g&192)===128&&(b&192)===128&&(f=(i&15)<<18|(_&63)<<12|(g&63)<<6|b&63,f>65535&&f<1114112&&(c=f))}}c===null?(c=65533,l=1):c>65535&&(c-=65536,t.push(c>>>10&1023|55296),c=56320|c&1023),t.push(c),a+=l}return Ia(t)}var Do=4096;function Ia(e){var n=e.length;if(n<=Do)return String.fromCharCode.apply(String,e);for(var o="",t=0;t<n;)o+=String.fromCharCode.apply(String,e.slice(t,t+=Do));return o}function La(e,n,o){var t="";o=Math.min(e.length,o);for(var a=n;a<o;++a)t+=String.fromCharCode(e[a]&127);return t}function Ra(e,n,o){var t="";o=Math.min(e.length,o);for(var a=n;a<o;++a)t+=String.fromCharCode(e[a]);return t}function Ca(e,n,o){var t=e.length;(!n||n<0)&&(n=0),(!o||o<0||o>t)&&(o=t);for(var a="",i=n;i<o;++i)a+=Ba[e[i]];return a}function Fa(e,n,o){for(var t=e.slice(n,o),a="",i=0;i<t.length-1;i+=2)a+=String.fromCharCode(t[i]+t[i+1]*256);return a}d.prototype.slice=function(n,o){var t=this.length;n=~~n,o=o===void 0?t:~~o,n<0?(n+=t,n<0&&(n=0)):n>t&&(n=t),o<0?(o+=t,o<0&&(o=0)):o>t&&(o=t),o<n&&(o=n);var a=this.subarray(n,o);return Object.setPrototypeOf(a,d.prototype),a};function S(e,n,o){if(e%1!==0||e<0)throw new RangeError("offset is not uint");if(e+n>o)throw new RangeError("Trying to access beyond buffer length")}d.prototype.readUintLE=d.prototype.readUIntLE=function(n,o,t){n=n>>>0,o=o>>>0,t||S(n,o,this.length);for(var a=this[n],i=1,c=0;++c<o&&(i*=256);)a+=this[n+c]*i;return a};d.prototype.readUintBE=d.prototype.readUIntBE=function(n,o,t){n=n>>>0,o=o>>>0,t||S(n,o,this.length);for(var a=this[n+--o],i=1;o>0&&(i*=256);)a+=this[n+--o]*i;return a};d.prototype.readUint8=d.prototype.readUInt8=function(n,o){return n=n>>>0,o||S(n,1,this.length),this[n]};d.prototype.readUint16LE=d.prototype.readUInt16LE=function(n,o){return n=n>>>0,o||S(n,2,this.length),this[n]|this[n+1]<<8};d.prototype.readUint16BE=d.prototype.readUInt16BE=function(n,o){return n=n>>>0,o||S(n,2,this.length),this[n]<<8|this[n+1]};d.prototype.readUint32LE=d.prototype.readUInt32LE=function(n,o){return n=n>>>0,o||S(n,4,this.length),(this[n]|this[n+1]<<8|this[n+2]<<16)+this[n+3]*16777216};d.prototype.readUint32BE=d.prototype.readUInt32BE=function(n,o){return n=n>>>0,o||S(n,4,this.length),this[n]*16777216+(this[n+1]<<16|this[n+2]<<8|this[n+3])};d.prototype.readIntLE=function(n,o,t){n=n>>>0,o=o>>>0,t||S(n,o,this.length);for(var a=this[n],i=1,c=0;++c<o&&(i*=256);)a+=this[n+c]*i;return i*=128,a>=i&&(a-=Math.pow(2,8*o)),a};d.prototype.readIntBE=function(n,o,t){n=n>>>0,o=o>>>0,t||S(n,o,this.length);for(var a=o,i=1,c=this[n+--a];a>0&&(i*=256);)c+=this[n+--a]*i;return i*=128,c>=i&&(c-=Math.pow(2,8*o)),c};d.prototype.readInt8=function(n,o){return n=n>>>0,o||S(n,1,this.length),this[n]&128?(255-this[n]+1)*-1:this[n]};d.prototype.readInt16LE=function(n,o){n=n>>>0,o||S(n,2,this.length);var t=this[n]|this[n+1]<<8;return t&32768?t|4294901760:t};d.prototype.readInt16BE=function(n,o){n=n>>>0,o||S(n,2,this.length);var t=this[n+1]|this[n]<<8;return t&32768?t|4294901760:t};d.prototype.readInt32LE=function(n,o){return n=n>>>0,o||S(n,4,this.length),this[n]|this[n+1]<<8|this[n+2]<<16|this[n+3]<<24};d.prototype.readInt32BE=function(n,o){return n=n>>>0,o||S(n,4,this.length),this[n]<<24|this[n+1]<<16|this[n+2]<<8|this[n+3]};d.prototype.readFloatLE=function(n,o){return n=n>>>0,o||S(n,4,this.length),me.read(this,n,!0,23,4)};d.prototype.readFloatBE=function(n,o){return n=n>>>0,o||S(n,4,this.length),me.read(this,n,!1,23,4)};d.prototype.readDoubleLE=function(n,o){return n=n>>>0,o||S(n,8,this.length),me.read(this,n,!0,52,8)};d.prototype.readDoubleBE=function(n,o){return n=n>>>0,o||S(n,8,this.length),me.read(this,n,!1,52,8)};function E(e,n,o,t,a,i){if(!d.isBuffer(e))throw new TypeError('"buffer" argument must be a Buffer instance');if(n>a||n<i)throw new RangeError('"value" argument is out of bounds');if(o+t>e.length)throw new RangeError("Index out of range")}d.prototype.writeUintLE=d.prototype.writeUIntLE=function(n,o,t,a){if(n=+n,o=o>>>0,t=t>>>0,!a){var i=Math.pow(2,8*t)-1;E(this,n,o,t,i,0)}var c=1,l=0;for(this[o]=n&255;++l<t&&(c*=256);)this[o+l]=n/c&255;return o+t};d.prototype.writeUintBE=d.prototype.writeUIntBE=function(n,o,t,a){if(n=+n,o=o>>>0,t=t>>>0,!a){var i=Math.pow(2,8*t)-1;E(this,n,o,t,i,0)}var c=t-1,l=1;for(this[o+c]=n&255;--c>=0&&(l*=256);)this[o+c]=n/l&255;return o+t};d.prototype.writeUint8=d.prototype.writeUInt8=function(n,o,t){return n=+n,o=o>>>0,t||E(this,n,o,1,255,0),this[o]=n&255,o+1};d.prototype.writeUint16LE=d.prototype.writeUInt16LE=function(n,o,t){return n=+n,o=o>>>0,t||E(this,n,o,2,65535,0),this[o]=n&255,this[o+1]=n>>>8,o+2};d.prototype.writeUint16BE=d.prototype.writeUInt16BE=function(n,o,t){return n=+n,o=o>>>0,t||E(this,n,o,2,65535,0),this[o]=n>>>8,this[o+1]=n&255,o+2};d.prototype.writeUint32LE=d.prototype.writeUInt32LE=function(n,o,t){return n=+n,o=o>>>0,t||E(this,n,o,4,4294967295,0),this[o+3]=n>>>24,this[o+2]=n>>>16,this[o+1]=n>>>8,this[o]=n&255,o+4};d.prototype.writeUint32BE=d.prototype.writeUInt32BE=function(n,o,t){return n=+n,o=o>>>0,t||E(this,n,o,4,4294967295,0),this[o]=n>>>24,this[o+1]=n>>>16,this[o+2]=n>>>8,this[o+3]=n&255,o+4};d.prototype.writeIntLE=function(n,o,t,a){if(n=+n,o=o>>>0,!a){var i=Math.pow(2,8*t-1);E(this,n,o,t,i-1,-i)}var c=0,l=1,_=0;for(this[o]=n&255;++c<t&&(l*=256);)n<0&&_===0&&this[o+c-1]!==0&&(_=1),this[o+c]=(n/l>>0)-_&255;return o+t};d.prototype.writeIntBE=function(n,o,t,a){if(n=+n,o=o>>>0,!a){var i=Math.pow(2,8*t-1);E(this,n,o,t,i-1,-i)}var c=t-1,l=1,_=0;for(this[o+c]=n&255;--c>=0&&(l*=256);)n<0&&_===0&&this[o+c+1]!==0&&(_=1),this[o+c]=(n/l>>0)-_&255;return o+t};d.prototype.writeInt8=function(n,o,t){return n=+n,o=o>>>0,t||E(this,n,o,1,127,-128),n<0&&(n=255+n+1),this[o]=n&255,o+1};d.prototype.writeInt16LE=function(n,o,t){return n=+n,o=o>>>0,t||E(this,n,o,2,32767,-32768),this[o]=n&255,this[o+1]=n>>>8,o+2};d.prototype.writeInt16BE=function(n,o,t){return n=+n,o=o>>>0,t||E(this,n,o,2,32767,-32768),this[o]=n>>>8,this[o+1]=n&255,o+2};d.prototype.writeInt32LE=function(n,o,t){return n=+n,o=o>>>0,t||E(this,n,o,4,2147483647,-2147483648),this[o]=n&255,this[o+1]=n>>>8,this[o+2]=n>>>16,this[o+3]=n>>>24,o+4};d.prototype.writeInt32BE=function(n,o,t){return n=+n,o=o>>>0,t||E(this,n,o,4,2147483647,-2147483648),n<0&&(n=4294967295+n+1),this[o]=n>>>24,this[o+1]=n>>>16,this[o+2]=n>>>8,this[o+3]=n&255,o+4};function Fo(e,n,o,t,a,i){if(o+t>e.length)throw new RangeError("Index out of range");if(o<0)throw new RangeError("Index out of range")}function Po(e,n,o,t,a){return n=+n,o=o>>>0,a||Fo(e,n,o,4,34028234663852886e22,-34028234663852886e22),me.write(e,n,o,t,23,4),o+4}d.prototype.writeFloatLE=function(n,o,t){return Po(this,n,o,!0,t)};d.prototype.writeFloatBE=function(n,o,t){return Po(this,n,o,!1,t)};function jo(e,n,o,t,a){return n=+n,o=o>>>0,a||Fo(e,n,o,8,17976931348623157e292,-17976931348623157e292),me.write(e,n,o,t,52,8),o+8}d.prototype.writeDoubleLE=function(n,o,t){return jo(this,n,o,!0,t)};d.prototype.writeDoubleBE=function(n,o,t){return jo(this,n,o,!1,t)};d.prototype.copy=function(n,o,t,a){if(!d.isBuffer(n))throw new TypeError("argument should be a Buffer");if(t||(t=0),!a&&a!==0&&(a=this.length),o>=n.length&&(o=n.length),o||(o=0),a>0&&a<t&&(a=t),a===t||n.length===0||this.length===0)return 0;if(o<0)throw new RangeError("targetStart out of bounds");if(t<0||t>=this.length)throw new RangeError("Index out of range");if(a<0)throw new RangeError("sourceEnd out of bounds");a>this.length&&(a=this.length),n.length-o<a-t&&(a=n.length-o+t);var i=a-t;return this===n&&typeof Uint8Array.prototype.copyWithin=="function"?this.copyWithin(o,t,a):Uint8Array.prototype.set.call(n,this.subarray(t,a),o),i};d.prototype.fill=function(n,o,t,a){if(typeof n=="string"){if(typeof o=="string"?(a=o,o=0,t=this.length):typeof t=="string"&&(a=t,t=this.length),a!==void 0&&typeof a!="string")throw new TypeError("encoding must be a string");if(typeof a=="string"&&!d.isEncoding(a))throw new TypeError("Unknown encoding: "+a);if(n.length===1){var i=n.charCodeAt(0);(a==="utf8"&&i<128||a==="latin1")&&(n=i)}}else typeof n=="number"?n=n&255:typeof n=="boolean"&&(n=Number(n));if(o<0||this.length<o||this.length<t)throw new RangeError("Out of range index");if(t<=o)return this;o=o>>>0,t=t===void 0?this.length:t>>>0,n||(n=0);var c;if(typeof n=="number")for(c=o;c<t;++c)this[c]=n;else{var l=d.isBuffer(n)?n:d.from(n,a),_=l.length;if(_===0)throw new TypeError('The value "'+n+'" is invalid for argument "value"');for(c=0;c<t-o;++c)this[c+o]=l[c%_]}return this};var Pa=/[^+/0-9A-Za-z-_]/g;function ja(e){if(e=e.split("=")[0],e=e.trim().replace(Pa,""),e.length<2)return"";for(;e.length%4!==0;)e=e+"=";return e}function qn(e,n){n=n||1/0;for(var o,t=e.length,a=null,i=[],c=0;c<t;++c){if(o=e.charCodeAt(c),o>55295&&o<57344){if(!a){if(o>56319){(n-=3)>-1&&i.push(239,191,189);continue}else if(c+1===t){(n-=3)>-1&&i.push(239,191,189);continue}a=o;continue}if(o<56320){(n-=3)>-1&&i.push(239,191,189),a=o;continue}o=(a-55296<<10|o-56320)+65536}else a&&(n-=3)>-1&&i.push(239,191,189);if(a=null,o<128){if((n-=1)<0)break;i.push(o)}else if(o<2048){if((n-=2)<0)break;i.push(o>>6|192,o&63|128)}else if(o<65536){if((n-=3)<0)break;i.push(o>>12|224,o>>6&63|128,o&63|128)}else if(o<1114112){if((n-=4)<0)break;i.push(o>>18|240,o>>12&63|128,o>>6&63|128,o&63|128)}else throw new Error("Invalid code point")}return i}function za(e){for(var n=[],o=0;o<e.length;++o)n.push(e.charCodeAt(o)&255);return n}function Ua(e,n){for(var o,t,a,i=[],c=0;c<e.length&&!((n-=2)<0);++c)o=e.charCodeAt(c),t=o>>8,a=o%256,i.push(a),i.push(t);return i}function zo(e){return yn.toByteArray(ja(e))}function ze(e,n,o,t){for(var a=0;a<t&&!(a+o>=n.length||a>=e.length);++a)n[a+o]=e[a];return a}function z(e,n){return e instanceof n||e!=null&&e.constructor!=null&&e.constructor.name!=null&&e.constructor.name===n.name}function Sn(e){return e!==e}var Ba=function(){for(var e="0123456789abcdef",n=new Array(256),o=0;o<16;++o)for(var t=o*16,a=0;a<16;++a)n[t+a]=e[o]+e[a];return n}()});var Wo=Ce((hc,Go)=>{r();var w=Go.exports={},U,B;function An(){throw new Error("setTimeout has not been defined")}function Nn(){throw new Error("clearTimeout has not been defined")}(function(){try{typeof setTimeout=="function"?U=setTimeout:U=An}catch(e){U=An}try{typeof clearTimeout=="function"?B=clearTimeout:B=Nn}catch(e){B=Nn}})();function Bo(e){if(U===setTimeout)return setTimeout(e,0);if((U===An||!U)&&setTimeout)return U=setTimeout,setTimeout(e,0);try{return U(e,0)}catch(n){try{return U.call(null,e,0)}catch(o){return U.call(this,e,0)}}}function Oa(e){if(B===clearTimeout)return clearTimeout(e);if((B===Nn||!B)&&clearTimeout)return B=clearTimeout,clearTimeout(e);try{return B(e)}catch(n){try{return B.call(null,e)}catch(o){return B.call(this,e)}}}var G=[],de=!1,ee,Ue=-1;function Va(){!de||!ee||(de=!1,ee.length?G=ee.concat(G):Ue=-1,G.length&&Oo())}function Oo(){if(!de){var e=Bo(Va);de=!0;for(var n=G.length;n;){for(ee=G,G=[];++Ue<n;)ee&&ee[Ue].run();Ue=-1,n=G.length}ee=null,de=!1,Oa(e)}}w.nextTick=function(e){var n=new Array(arguments.length-1);if(arguments.length>1)for(var o=1;o<arguments.length;o++)n[o-1]=arguments[o];G.push(new Vo(e,n)),G.length===1&&!de&&Bo(Oo)};function Vo(e,n){this.fun=e,this.array=n}Vo.prototype.run=function(){this.fun.apply(null,this.array)};w.title="browser";w.browser=!0;w.env={};w.argv=[];w.version="";w.versions={};function W(){}w.on=W;w.addListener=W;w.once=W;w.off=W;w.removeListener=W;w.removeAllListeners=W;w.emit=W;w.prependListener=W;w.prependOnceListener=W;w.listeners=function(e){return[]};w.binding=function(e){throw new Error("process.binding is not supported")};w.cwd=function(){return"/"};w.chdir=function(e){throw new Error("process.chdir is not supported")};w.umask=function(){return 0}});var p,m,bc,r=h(()=>{p=T(Uo()),m=T(Wo()),bc=function(e){function n(){var t=this||self;return delete e.prototype.__magic__,t}if(typeof globalThis=="object")return globalThis;if(this)return n();e.defineProperty(e.prototype,"__magic__",{configurable:!0,get:n});var o=__magic__;return o}(Object)});var Ho=h(()=>{"use strict";r()});var Zo=h(()=>{"use strict";r()});var N=h(()=>{"use strict";r()});var P=h(()=>{"use strict";r()});var Yo=h(()=>{"use strict";r()});var $o=h(()=>{"use strict";r()});var Qo=h(()=>{"use strict";r()});var Xo=h(()=>{"use strict";r()});var ne=h(()=>{"use strict";r();Zo();N();P();Yo();$o();Qo();Xo()});var we,Ko,Rc,Jo,Cc,et,Fc,Pc,jc,nt=h(()=>{"use strict";r();ne();we={devnet:{id:"devnet",chainId:"D",name:"Devnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://devnet-wallet.elrond.com",apiAddress:"https://devnet-api.elrond.com",explorerAddress:"http://devnet-explorer.elrond.com",apiTimeout:"4000"},testnet:{id:"testnet",chainId:"T",name:"Testnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://testnet-wallet.elrond.com",apiAddress:"https://testnet-api.elrond.com",explorerAddress:"http://testnet-explorer.elrond.com",apiTimeout:"4000"},mainnet:{id:"mainnet",chainId:"1",name:"Mainnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://wallet.elrond.com",apiAddress:"https://api.elrond.com",explorerAddress:"https://explorer.elrond.com",apiTimeout:"4000"}},{chainId:Ko,egldLabel:Rc}=we["devnet"],{chainId:Jo,egldLabel:Cc}=we["testnet"],{chainId:et,egldLabel:Fc}=we["mainnet"],Pc={["devnet"]:Ko,["testnet"]:Jo,["mainnet"]:et},jc={[Ko]:"devnet",[Jo]:"testnet",[et]:"mainnet"}});var En=h(()=>{"use strict";r()});var ot=h(()=>{"use strict";r()});var tt,Dn=h(()=>{"use strict";r();tt="transaction-status-toast-id"});var Mn,In,O,Be,rt,Oe,at,Ve,oe,q=h(()=>{"use strict";r();Ho();nt();En();ot();Dn();Mn=5e4,In=1e9,O=18,Be=4,rt=1,Oe="logout",at="login",Ve="N/A",oe="0"});var $e,Vn=h(()=>{"use strict";r();$e=()=>Date.now()/1e3});var Dt=h(()=>{"use strict";r()});var Mt=h(()=>{"use strict";r()});var It=h(()=>{"use strict";r();Vn();Dt();Mt()});var Wn={};pe(Wn,{getItem:()=>ui,localStorageKeys:()=>ae,removeItem:()=>fi,setItem:()=>gi});var ae,Gn,gi,ui,fi,Se=h(()=>{"use strict";r();It();ae={loginExpiresAt:"dapp-core-login-expires-at",logoutEvent:"dapp-core-logout-event"},Gn=typeof localStorage!="undefined",gi=({key:e,data:n,expires:o})=>{!Gn||localStorage.setItem(String(e),JSON.stringify({expires:o,data:n}))},ui=e=>{if(!Gn)return;let n=localStorage.getItem(String(e));if(!n)return null;let o=JSON.parse(n);return!o||!o.hasOwnProperty("expires")||!o.hasOwnProperty("data")?null:$e()>=o.expires?(localStorage.removeItem(String(e)),null):o.data},fi=e=>{!Gn||localStorage.removeItem(String(e))}});var Hn={};pe(Hn,{clear:()=>Ft,getItem:()=>Rt,removeItem:()=>Ct,setItem:()=>Lt,storage:()=>hi});var Lt,Rt,Ct,Ft,hi,Pt=h(()=>{"use strict";r();Lt=({key:e,data:n,expires:o})=>{sessionStorage.setItem(String(e),JSON.stringify({expires:o,data:n}))},Rt=e=>{let n=sessionStorage.getItem(String(e));if(!n)return null;let o=JSON.parse(n);return!o||!o.hasOwnProperty("expires")||!o.hasOwnProperty("data")?null:Date.now()>=o.expires?(sessionStorage.removeItem(String(e)),null):o.data},Ct=e=>sessionStorage.removeItem(String(e)),Ft=()=>sessionStorage.clear(),hi={setItem:Lt,getItem:Rt,removeItem:Ct,clear:Ft}});var ge,Qe=h(()=>{"use strict";r();Se();Pt();ge={session:Hn,local:Wn}});function Ae(){return new Date().setHours(new Date().getHours()+24)}function Ne(e){ge.local.setItem({key:ae.loginExpiresAt,data:e,expires:e})}var Zn=h(()=>{"use strict";r();Qe();Se()});var Yn,D,Xe,X=h(()=>{"use strict";r();Yn=require("@reduxjs/toolkit");q();D=(0,Yn.createAction)(Oe),Xe=(0,Yn.createAction)(at,e=>({payload:e}))});var zt,jt,Ut,gd,ud,fd,hd,bd,xd,Bt,Ot,$n=h(()=>{"use strict";r();zt=require("@reduxjs/toolkit");Zn();P();X();jt={loginMethod:"",walletConnectLogin:null,ledgerLogin:null,tokenLogin:null,walletLogin:null,extensionLogin:null,isLoginSessionInvalid:!1},Ut=(0,zt.createSlice)({name:"loginInfoSlice",initialState:jt,reducers:{setLoginMethod:(e,n)=>{e.loginMethod=n.payload},setTokenLogin:(e,n)=>{e.tokenLogin=n.payload},setTokenLoginSignature:(e,n)=>{(e==null?void 0:e.tokenLogin)!=null&&(e.tokenLogin.signature=n.payload)},setWalletLogin:(e,n)=>{e.walletLogin=n.payload},setWalletConnectLogin:(e,n)=>{e.walletConnectLogin=n.payload},setLedgerLogin:(e,n)=>{e.ledgerLogin=n.payload},invalidateLoginSession:e=>{e.isLoginSessionInvalid=!0}},extraReducers:e=>{e.addCase(D,()=>jt),e.addCase(Xe,(n,o)=>{n.loginMethod=o.payload.loginMethod,Ne(Ae())})}}),{setLoginMethod:gd,setWalletConnectLogin:ud,setLedgerLogin:fd,setTokenLogin:hd,setTokenLoginSignature:bd,setWalletLogin:xd,invalidateLoginSession:Bt}=Ut.actions,Ot=Ut.reducer});var Vt=h(()=>{"use strict";r()});var Gt=h(()=>{"use strict";r();Vt()});function Qn(e){return e[Math.floor(Math.random()*e.length)]}var Wt=h(()=>{"use strict";r()});var Ht=h(()=>{"use strict";r();En()});var Zt=h(()=>{"use strict";r();Gt();Wt();Ht()});var Yt,$t,Xn,xi,Qt,Cd,Fd,Xt,Kn=h(()=>{"use strict";r();Yt=require("@reduxjs/toolkit"),$t=T(require("lodash.omit"));Zt();Xn={id:"not-configured",chainId:"",name:"NOT CONFIGURED",egldLabel:"",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"",walletConnectBridgeAddress:"",walletConnectV2RelayAddress:"",walletConnectV2ProjectId:"",walletAddress:"",apiAddress:"",explorerAddress:"",apiTimeout:"4000"},xi={network:Xn,chainID:"-1"},Qt=(0,Yt.createSlice)({name:"appConfig",initialState:xi,reducers:{initializeNetworkConfig:(e,n)=>{let o=Qn(n.payload.walletConnectBridgeAddresses),t=Qn(n.payload.walletConnectV2RelayAddresses),a=(0,$t.default)(n.payload,"walletConnectBridgeAddresses","walletConnectV2RelayAddresses");e.network=v(x(x({},e.network),a),{walletConnectBridgeAddress:o,walletConnectV2RelayAddress:t})},setChainID:(e,n)=>{e.chainID=n.payload}}}),{initializeNetworkConfig:Cd,setChainID:Fd}=Qt.actions,Xt=Qt.reducer});var eo,Kt,Jt,Ke,Jn,er,nr,Gd,or,Wd,Hd,Zd,Yd,$d,Qd,Xd,tr,no=h(()=>{"use strict";r();eo=require("@elrondnetwork/erdjs"),Kt=require("@reduxjs/toolkit"),Jt=require("redux-persist");q();Qe();Se();X();Ke={balance:"...",address:"",nonce:0,txCount:0,scrCount:0,claimableRewards:oe},Jn={address:"",websocketEvent:null,accounts:{"":Ke},ledgerAccount:null,publicKey:"",walletConnectAccount:null,isAccountLoading:!0,accountLoadingError:null},er=(0,Kt.createSlice)({name:"accountInfoSlice",initialState:Jn,reducers:{setAddress:(e,n)=>{let o=n.payload;e.address=o,e.publicKey=new eo.Address(o).hex()},setAccount:(e,n)=>{let o=e.address===n.payload.address;e.accounts={[e.address]:o?n.payload:Ke},e.isAccountLoading=!1,e.accountLoadingError=null},setAccountNonce:(e,n)=>{let{address:o}=e;e.accounts[o].nonce=n.payload},setAccountShard:(e,n)=>{e.shard=n.payload},setLedgerAccount:(e,n)=>{e.ledgerAccount=n.payload},updateLedgerAccount:(e,n)=>{e.ledgerAccount!=null&&(e.ledgerAccount.index=n.payload.index,e.ledgerAccount.address=n.payload.address)},setWalletConnectAccount:(e,n)=>{e.walletConnectAccount=n.payload},setIsAccountLoading:(e,n)=>{e.isAccountLoading=n.payload,e.accountLoadingError=null},setAccountLoadingError:(e,n)=>{e.accountLoadingError=n.payload,e.isAccountLoading=!1},setWebsocketEvent:(e,n)=>{e.websocketEvent={timestamp:Date.now(),message:n.payload}}},extraReducers:e=>{e.addCase(D,()=>(ge.local.removeItem(ae.loginExpiresAt),Jn)),e.addCase(Xe,(n,o)=>{let{address:t}=o.payload;n.address=t,n.publicKey=new eo.Address(t).hex()}),e.addCase(Jt.REHYDRATE,(n,o)=>{var g;if(!((g=o.payload)!=null&&g.account))return;let{account:t}=o.payload,{address:a,shard:i,accounts:c,publicKey:l}=t;n.address=a,n.shard=i;let _=c&&a in c;n.accounts=_?c:Jn.accounts,n.publicKey=l})}}),{setAccount:nr,setAddress:Gd,setAccountNonce:or,setAccountShard:Wd,setLedgerAccount:Hd,updateLedgerAccount:Zd,setWalletConnectAccount:Yd,setIsAccountLoading:$d,setAccountLoadingError:Qd,setWebsocketEvent:Xd}=er.actions,tr=er.reducer});function rr(e){return e!=null&&(Ni(e)||Ii(e))}function Je(e){return e!=null&&(Ei(e)||Li(e))}function en(e){return e!=null&&(Di(e)||Ri(e))}function ar(e){return e!=null&&Mi(e)}function Ni(e){return e!=null&&Ti.includes(e)}function Ei(e){return e!=null&&wi.includes(e)}function Di(e){return e!=null&&qi.includes(e)}function Mi(e){return e!=null&&ki.includes(e)}function Ii(e){return e!=null&&vi.includes(e)}function Li(e){return e!=null&&Si.includes(e)}function Ri(e){return e!=null&&Ai.includes(e)}var Ti,wi,qi,ki,vi,Si,Ai,to=h(()=>{"use strict";r();P();Ti=["sent"],wi=["success"],qi=["fail","cancelled","timedOut"],ki=["timedOut"],vi=["pending"],Si=["success"],Ai=["fail","invalid"]});var ir,cr,Ee,Ci,pr,i_,c_,p_,m_,s_,d_,__,l_,g_,u_,nn,on=h(()=>{"use strict";r();ir=require("@reduxjs/toolkit"),cr=require("redux-persist");P();to();X();Ee={signedTransactions:{},transactionsToSign:null,signTransactionsError:null,signTransactionsCancelMessage:null,customTransactionInformationForSessionId:{}},Ci={signWithoutSending:!1,sessionInformation:null,redirectAfterSign:!1},pr=(0,ir.createSlice)({name:"transactionsSlice",initialState:Ee,reducers:{moveTransactionsToSignedState:(e,n)=>{var _,g;let{sessionId:o,transactions:t,errorMessage:a,status:i,redirectRoute:c}=n.payload,l=((_=e.customTransactionInformationForSessionId)==null?void 0:_[o])||Ci;e.signedTransactions[o]={transactions:t,status:i,errorMessage:a,redirectRoute:c,customTransactionInformation:l},((g=e==null?void 0:e.transactionsToSign)==null?void 0:g.sessionId)===o&&(e.transactionsToSign=Ee.transactionsToSign)},clearSignedTransaction:(e,n)=>{e.signedTransactions[n.payload]&&delete e.signedTransactions[n.payload]},clearTransactionToSign:e=>{e!=null&&e.transactionsToSign&&(e.transactionsToSign=null)},updateSignedTransaction:(e,n)=>{e.signedTransactions=x(x({},e.signedTransactions),n.payload)},updateSignedTransactions:(e,n)=>{let{sessionId:o,status:t,errorMessage:a,transactions:i}=n.payload;e.signedTransactions[o]!=null&&(e.signedTransactions[o].status=t,a!=null&&(e.signedTransactions[o].errorMessage=a),i!=null&&(e.signedTransactions[o].transactions=i))},updateSignedTransactionStatus:(e,n)=>{var l,_,g,b,f,y;let{sessionId:o,status:t,errorMessage:a,transactionHash:i}=n.payload,c=(_=(l=e.signedTransactions)==null?void 0:l[o])==null?void 0:_.transactions;if(c!=null){e.signedTransactions[o].transactions=c.map(A=>A.hash===i?v(x({},A),{status:t,errorMessage:a}):A);let k=(b=(g=e.signedTransactions[o])==null?void 0:g.transactions)==null?void 0:b.every(A=>Je(A.status)),M=(y=(f=e.signedTransactions[o])==null?void 0:f.transactions)==null?void 0:y.every(A=>en(A.status));k&&(e.signedTransactions[o].status="success"),M&&(e.signedTransactions[o].status="fail")}},setTransactionsToSign:(e,n)=>{e.transactionsToSign=n.payload;let{sessionId:o,customTransactionInformation:t}=n.payload;e.customTransactionInformationForSessionId[o]=t,e.signTransactionsError=null},clearAllTransactionsToSign:e=>{e.transactionsToSign=Ee.transactionsToSign,e.signTransactionsError=null},clearAllSignedTransactions:e=>{e.signedTransactions=Ee.signedTransactions},setSignTransactionsError:(e,n)=>{e.signTransactionsError=n.payload},setSignTransactionsCancelMessage:(e,n)=>{e.signTransactionsCancelMessage=n.payload}},extraReducers:e=>{e.addCase(D,()=>Ee),e.addCase(cr.REHYDRATE,(n,o)=>{var c;if(!((c=o.payload)!=null&&c.transactions))return;let{signedTransactions:t,customTransactionInformationForSessionId:a}=o.payload.transactions,i=Object.entries(t).reduce((l,[_,g])=>{let b=new Date(_),f=new Date;return f.setHours(f.getHours()+5),f-b>0||(l[_]=g),l},{});a!=null&&(n.customTransactionInformationForSessionId=a),t!=null&&(n.signedTransactions=i)})}}),{updateSignedTransactionStatus:i_,updateSignedTransactions:c_,setTransactionsToSign:p_,clearAllTransactionsToSign:m_,clearAllSignedTransactions:s_,clearSignedTransaction:d_,clearTransactionToSign:__,setSignTransactionsError:l_,setSignTransactionsCancelMessage:g_,moveTransactionsToSignedState:u_}=pr.actions,nn=pr.reducer});var mr,ao,io,co,Fi,ro,sr,x_,y_,T_,tn,rn=h(()=>{"use strict";r();mr=require("@reduxjs/toolkit");X();ao="Transaction failed",io="Transaction successful",co="Processing transaction",Fi="Transaction submitted",ro={},sr=(0,mr.createSlice)({name:"transactionsInfo",initialState:ro,reducers:{setTransactionsDisplayInfo(e,n){let{sessionId:o,transactionsDisplayInfo:t}=n.payload;o!=null&&(e[o]={errorMessage:(t==null?void 0:t.errorMessage)||ao,successMessage:(t==null?void 0:t.successMessage)||io,processingMessage:(t==null?void 0:t.processingMessage)||co,submittedMessage:(t==null?void 0:t.submittedMessage)||Fi,transactionDuration:t==null?void 0:t.transactionDuration})},clearTransactionsInfoForSessionId(e,n){n.payload!=null&&delete e[n.payload]},clearTransactionsInfo:()=>ro},extraReducers:e=>{e.addCase(D,()=>ro)}}),{clearTransactionsInfo:x_,setTransactionsDisplayInfo:y_,clearTransactionsInfoForSessionId:T_}=sr.actions,tn=sr.reducer});var _r,dr,lr,v_,S_,A_,N_,gr,po=h(()=>{"use strict";r();_r=require("@reduxjs/toolkit");X();dr={},lr=(0,_r.createSlice)({name:"modalsSlice",initialState:dr,reducers:{setTxSubmittedModal:(e,n)=>{e.txSubmittedModal=n.payload},setNotificationModal:(e,n)=>{e.notificationModal=n.payload},clearTxSubmittedModal:e=>{e.txSubmittedModal=void 0},clearNotificationModal:e=>{e.notificationModal=void 0}},extraReducers:e=>{e.addCase(D,()=>dr)}}),{setTxSubmittedModal:v_,setNotificationModal:S_,clearTxSubmittedModal:A_,clearNotificationModal:N_}=lr.actions,gr=lr.reducer});var fr,ur,hr,F_,P_,j_,z_,U_,B_,an,cn=h(()=>{"use strict";r();fr=require("@reduxjs/toolkit");Dn();ne();Vn();X();ur={customToasts:[],transactionToasts:[],failTransactionToast:null},hr=(0,fr.createSlice)({name:"toastsSlice",initialState:ur,reducers:{addCustomToast:(e,n)=>{e.customToasts.push(v(x({},n.payload),{type:"custom",toastId:n.payload.toastId||`custom-toast-${e.customToasts.length+1}`}))},removeCustomToast:(e,n)=>{e.customToasts=e.customToasts.filter(o=>o.toastId!==n.payload)},addTransactionToast:(e,n)=>{e.transactionToasts.push({type:"transaction",startTimestamp:$e(),toastId:n.payload||`custom-toast-${e.transactionToasts.length+1}`})},removeTransactionToast:(e,n)=>{e.transactionToasts=e.transactionToasts.filter(o=>o.toastId!==n.payload)},addFailTransactionToast:(e,n)=>{e.failTransactionToast=v(x({},n.payload),{toastId:tt})},removeFailTransactionToast:e=>{e.failTransactionToast=null}},extraReducers:e=>{e.addCase(D,()=>ur)}}),{addCustomToast:F_,removeCustomToast:P_,addTransactionToast:j_,removeTransactionToast:z_,addFailTransactionToast:U_,removeFailTransactionToast:B_}=hr.actions,an=hr.reducer});var ue=h(()=>{"use strict";r();$n();Kn();no();on();rn();po();cn()});var go=h(()=>{"use strict";r()});var kr,qr,vr,_g,lg,gg,ug,fg,hg,Sr,Ar=h(()=>{"use strict";r();kr=require("@reduxjs/toolkit");X();qr={visible:!1,modal:{showHeader:!0,showFooter:!1,headerText:"",footerText:""}},vr=(0,kr.createSlice)({name:"dappModalsSlice",initialState:qr,reducers:{setDappModalVisibility:(e,n)=>{e.visible=n.payload},setDappModalHeaderVisibility:(e,n)=>{e.modal.showHeader=n.payload},setDappModalFooterVisibility:(e,n)=>{e.modal.showFooter=n.payload},setDappModalHeaderText:(e,n)=>{e.modal.headerText=n.payload},setDappModalFooterText:(e,n)=>{e.modal.footerText=n.payload},setDappModalConfig:(e,n)=>{e.modal=x({},n.payload)}},extraReducers:e=>{e.addCase(D,()=>qr)}}),{setDappModalVisibility:_g,setDappModalHeaderVisibility:lg,setDappModalFooterVisibility:gg,setDappModalHeaderText:ug,setDappModalFooterText:fg,setDappModalConfig:hg}=vr.actions,Sr=vr.reducer});var Nr,Vi,Gi,be,sn=h(()=>{"use strict";r();Nr=require("@reduxjs/toolkit");go();no();Ar();$n();po();Kn();cn();rn();on();Vi={["account"]:tr,["networkConfig"]:Xt,["loginInfo"]:Ot,["modals"]:gr,["dappModal"]:Sr,["toasts"]:an,["transactions"]:nn,["transactionsInfo"]:tn},Gi=(e={})=>(0,Nr.combineReducers)(x(x({},Vi),e)),be=Gi});var Mr={};pe(Mr,{default:()=>Xi});function uo(e,n=[]){return{key:e,version:1,storage:Dr.default,blacklist:n}}var ie,Er,Dr,Wi,Hi,Zi,Yi,$i,Qi,Xi,Ir=h(()=>{"use strict";r();ie=require("redux-persist"),Er=T(require("redux-persist/lib/storage")),Dr=T(require("redux-persist/lib/storage/session"));sn();ue();cn();rn();on();go();Wi={2:e=>v(x({},e),{networkConfig:Xn})};Hi=uo("dapp-core-transactionsInfo"),Zi=uo("dapp-core-transactions",["transactionsToSign"]),Yi=uo("dapp-core-toasts"),$i={key:"dapp-core-store",version:2,storage:Er.default,whitelist:["account","loginInfo","modals","networkConfig"],migrate:(0,ie.createMigrate)(Wi,{debug:!1})},Qi={["toasts"]:(0,ie.persistReducer)(Yi,an),["transactions"]:(0,ie.persistReducer)(Zi,nn),["transactionsInfo"]:(0,ie.persistReducer)(Hi,tn)},Xi=(0,ie.persistReducer)($i,be(Qi))});var Lr={};pe(Lr,{default:()=>Ki});var Ki,Rr=h(()=>{"use strict";r();sn();Ki=be()});var Cr={};pe(Cr,{default:()=>ec});var R,Ji,ec,Fr=h(()=>{"use strict";r();R=require("redux-persist"),Ji=[R.FLUSH,R.REHYDRATE,R.PAUSE,R.PERSIST,R.PURGE,R.REGISTER],ec=Ji});var zr={};pe(zr,{default:()=>jr});function jr(e){return(0,Pr.persistStore)(e)}var Pr,Ur=h(()=>{"use strict";r();Pr=require("redux-persist")});r();var na=T(require("react")),oa=require("@testing-library/react");r();q();r();r();var ke=require("@elrondnetwork/erdjs");ne();r();var ct=require("@elrondnetwork/erdjs");function Ga(e){try{let n=new ct.Address(e);return Boolean(n.bech32())}catch(n){return!1}}function qe(e){return(e==null?void 0:e.startsWith("erd"))&&e.length===62&&/^\w+$/.test(e)&&Ga(e)}r();function Ge(e){try{return p.Buffer.from(e,"base64").toString()===atob(e)}catch(n){return!1}}var Wa=["ESDTNFTTransfer","ESDTNFTBurn","ESDTNFTAddQuantity","ESDTNFTCreate","MultiESDTNFTTransfer","ESDTTransfer","ESDTBurn","ESDTLocalMint","ESDTLocalBurn","ESDTWipe","ESDTFreeze"];function pt(e,n,o=""){if(!qe(e))return!1;if(new ke.Address(e).isContractAddress())return!0;let i=mt({receiver:e,data:o});return i?new ke.Address(i).isContractAddress()||Ya(e,n,o):!1}var Ha=e=>e.toLowerCase().match(/[0-9a-f]/g),Za=e=>e.length%2===0;function Ya(e,n,o){let t=o==null?void 0:o.split("@");if(t==null)return!1;let[a,...i]=t,c=n!=null&&e!=null&&e===n,l=Wa.includes(a),_=i.every(g=>Ha(g)&&Za(g));return c&&l&&_}function mt({receiver:e,data:n}){try{if(!n)return e;let o=Ge(n)?ke.TransactionPayload.fromEncoded(n).toString():n,t=$a(o),a=o.split("@");return t>-1?a[t]:e}catch(o){console.log(o);return}}function $a(e){return e.includes("MultiESDTNFTTransfer")?1:e.includes("ESDTNFTTransfer")?4:-1}r();var Cn=require("@elrondnetwork/erdjs"),Fn=T(require("bignumber.js"));P();r();r();function Ln(e){for(let n=0;n<e.length;n++)if(e.charCodeAt(n)>127)return!1;return!0}function We(e){let n=e;try{let o=p.Buffer.from(e,"hex").toString();Ln(o)&&o.length>1&&(n=o)}catch(o){}return n}var Rn={tokenId:"",amount:""},Qa=e=>e.split("@").map((i,c)=>[2,3].includes(c)?i:We(i));function dt(e){if(!e)return Rn;let n=e.startsWith("ESDTTransfer"),o=e.startsWith("ESDTNFTTransfer")&&e.includes("@");if(n){let[,t,a]=e.split("@");try{let i=p.Buffer.from(t,"hex").toString("ascii");if(!i)return Rn;let c=new Fn.default("0x"+a.replace("0x","")).toString(10);return{tokenId:i,amount:c}}catch(i){console.error("Error getting token from transaction data",i)}}if(o)try{let[,t,a,i,c]=Qa(e);if([t,a,i,c].every(l=>Boolean(l))&&qe(new Cn.Address(c).bech32()))return{tokenId:`${t}-${a}`,amount:new Fn.default(i,16).toString(10),collection:t,nonce:a,receiver:new Cn.Address(c).bech32()}}catch(t){}return Rn}r();r();r();var Y="accounts",te="transactions";var _e="tokens",_t="blocks";var lt="nodes",gt="identities",ut="collections",Pn="nfts",jn="providers",zn="sc-results",ft="contracts",ht="logs",bt="code",xt="locked-accounts",yt="roles";var $={shard:e=>`/${_t}?shard=${e}`,receiverShard:e=>`/${te}?receivershard=${e}`,senderShard:e=>`/${te}?sendershard=${e}`,transactionDetails:e=>`/${te}/${e}`,transactionDetailsScResults:e=>`/${te}/${e}/${zn}`,transactionDetailsLogs:e=>`/${te}/${e}/${ht}`,nodeDetails:e=>`/${lt}/${e}`,accountDetails:e=>`/${Y}/${e}`,accountDetailsContractCode:e=>`/${Y}/${e}/${bt}`,accountDetailsTokens:e=>`/${Y}/${e}/${_e}`,accountDetailsNfts:e=>`/${Y}/${e}/${Pn}`,accountDetailsScResults:e=>`/${Y}/${e}/${zn}`,accountDetailsContracts:e=>`/${Y}/${e}/${ft}`,identityDetails:e=>`/${gt}/${e}`,tokenDetails:e=>`/${_e}/${e}`,tokenDetailsAccounts:e=>`/${_e}/${e}/${Y}`,tokenDetailsLockedAccounts:e=>`/${_e}/${e}/${xt}`,tokenDetailsRoles:e=>`/${_e}/${e}/${yt}`,collectionDetails:e=>`/${ut}/${e}`,nftDetails:e=>`/${Pn}/${e}`,providerDetails:e=>`/${jn}/${e}`,providerDetailsTransactions:e=>`/${jn}/${e}/${te}`};r();var Un=e=>{var n,o,t,a;if(e.action){let i=[(n=e.action.arguments)==null?void 0:n.token,(o=e.action.arguments)==null?void 0:o.token1,(t=e.action.arguments)==null?void 0:t.token2,(a=e.action.arguments)==null?void 0:a.transfers].filter(l=>l!=null);return[].concat(...i)}return[]};r();r();function He(e){var o,t;let n=e.receiver;return(t=(o=e.action)==null?void 0:o.arguments)!=null&&t.receiver&&(n=e.action.arguments.receiver),n}function Tt(e){let n=He(e);return e.receiver===n?e.receiverAssets:void 0}r();N();var wt=e=>{var t,a,i,c,l,_;let n="Transaction";return((t=e.action)==null?void 0:t.name)&&((a=e.action)==null?void 0:a.category)&&(((i=e.action)==null?void 0:i.category)==="esdtNft"&&((c=e.action)==null?void 0:c.name)==="transfer"?n="Transaction":e.action&&(n=e.action.name),(_=(l=e.action)==null?void 0:l.arguments)!=null&&_.functionName&&(n=e.action.arguments.functionName)),n};r();var qt=!1;function Ka(e){qt||(console.error(e),qt=!0)}function le({explorerAddress:e,to:n}){try{return new URL(n).href}catch(o){return n.startsWith("/")||(Ka(`Link not prepended by / : ${n}`),n=`/${n}`),e?`${e}${n}`:n}}r();N();function kt(e,n,o){let t=e===n.sender,a=e===o,i=t&&a,c=(n==null?void 0:n.type)==="SmartContractResult";switch(!0){case c:return"Internal";case i:return"Self";case a:return"In";case t:default:return"Out"}}r();r();r();var ei=T(require("bignumber.js"));q();r();var St=require("@elrondnetwork/erdjs"),Q=T(require("bignumber.js"));q();r();var vt=T(require("bignumber.js")),ve=(e,n=!0)=>{let o=String(e);if(!o.match(/^[-]?\d+$/))return!1;let t=new vt.default(o),a=n?0:-1;return t.toString(10)===o&&t.comparedTo(0)>=a};r();function H(e){return{if:function(n){return n?{then:o=>o instanceof Function?H(o(e)):H(o)}:{then:()=>H(e)}},then:n=>n instanceof Function?H(n(e)):H(n),valueOf:function(){return e}}}Q.default.config({ROUNDING_MODE:Q.default.ROUND_FLOOR});function re({input:e,decimals:n=O,digits:o=Be,showLastNonZeroDecimal:t=!0,showIsLessThanDecimalsLabel:a=!1,addCommas:i=!1}){if(!ve(e,!1))throw new Error("Invalid input");let c=new Q.default(e).isNegative(),l=e;return c&&(l=e.substring(1)),H(l).then(()=>St.TokenPayment.fungibleFromBigInteger("",l,n).toRationalNumber()).then(_=>{let g=new Q.default(_);if(g.isZero())return oe;let b=g.toString(10),[f,y]=b.split("."),k=new Q.default(y||0),M=H(0).if(Boolean(y&&t)).then(()=>Math.max(y.length,o)).if(k.isZero()&&!t).then(0).if(Boolean(y&&!t)).then(()=>Math.min(y.length,o)).valueOf(),A=y&&o>=1&&o<=y.length&&k.isGreaterThan(0)&&new Q.default(y.substring(0,o)).isZero(),ta=g.toFormat(M);return H(b).if(i).then(ta).if(Boolean(A)).then(Me=>{let ye=new Q.default(f).isZero(),[F,Ie]=Me.split("."),Le=new Array(o-1).fill(0),ra=[...Le,0].join(""),aa=[...Le,1].join("");return ye?a?`<${F}.${aa}`:t?`${F}.${Ie}`:F:`${F}.${ra}`}).if(Boolean(!A&&y)).then(Me=>{let[ye]=Me.split("."),F=y.substring(0,M);if(t){let Ie=o-F.length;if(Ie>0){let Le=Array(Ie).fill(0).join("");return F=`${F}${Le}`,`${ye}.${F}`}return Me}return F?`${ye}.${F}`:ye}).valueOf()}).if(c).then(_=>`-${_}`).valueOf()}r();r();r();q();r();r();r();r();q();r();q();r();var ni=require("@elrondnetwork/erdjs");q();r();var Ye=require("@elrondnetwork/erdjs"),ti=require("@elrondnetwork/erdjs"),ri=require("@elrondnetwork/erdjs-network-providers");q();r();r();var oi=T(require("bignumber.js"));q();r();q();r();r();var At=({amount:e,usd:n,decimals:o=2,addEqualSign:t})=>{let a=(parseFloat(e)*n).toFixed(o),i=parseFloat(a).toLocaleString("en",{maximumFractionDigits:o,minimumFractionDigits:o}),c=parseFloat(e)>0?"\u2248":"=";return`${t?`${c} `:""}$${i}`};r();r();r();q();r();N();var ii=["reDelegateRewards","claimRewards","unBond"],ci=["wrapEgld","unwrapEgld"],pi=["unStake"],mi=["unDelegate"];r();r();r();q();r();var _i=T(require("bignumber.js"));r();r();r();N();r();var li=T(require("bignumber.js"));function Nt({transaction:e,address:n,explorerAddress:o}){var A;let t=(A=e.tokenIdentifier)!=null?A:dt(e.data).tokenId,a=He(e),i=Tt(e),c=kt(n,e,a),l=wt(e),_=Un(e),g=le({explorerAddress:o,to:$.accountDetails(e.sender)}),b=le({explorerAddress:o,to:$.accountDetails(a)}),f=le({explorerAddress:o,to:$.senderShard(e.senderShard)}),y=le({explorerAddress:o,to:$.receiverShard(e.receiverShard)}),k=e.originalTxHash?`${e.originalTxHash}#${e.txHash}`:e.txHash,M=le({explorerAddress:o,to:$.transactionDetails(k)});return v(x({},e),{tokenIdentifier:t,receiver:a,receiverAssets:i,transactionDetails:{direction:c,method:l,transactionTokens:_,isContract:pt(e.sender)},links:{senderLink:g,receiverLink:b,senderShardLink:f,receiverShardLink:y,transactionLink:M}})}r();var On=[{txHash:"5996572de1fc5df2c456602467e8cc1a446b8145ae79df8f6c670e0732ab3b04",gasLimit:12e6,gasPrice:1e9,gasUsed:6062e3,miniBlockHash:"54d07f7c1e4d3f66e9f2c30d44d1e35d768a2da98dfe50cc437d485b9c2afb92",nonce:241,receiver:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqz8llllsh6u4jp",receiverAssets:{name:"Staking: ARC Stake",description:"ARC Staking provides staking services on the Elrond Blockchain, the most advanced blockchain platform in the world.",tags:["staking","provider"],iconPng:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"},receiverShard:4294967295,round:7430784,sender:"erd1zwq3qaa3vk5suenlkj4cf0ullwefa6h3n2394k25pxv4sz0pwhhsj9u9vk",senderShard:1,signature:"b9d589aa9aabf521864ccd52647850d224bb0ce76e89aaa8ad62bf67b4b9434d586e1b5be0db10ffc251be2fa495a3284585a226f57ee147ae2158efe651900b",status:"success",value:"14167802221131358682",fee:"122000000000000",timestamp:1640702304,data:"ZGVsZWdhdGU=",function:"delegate",action:{category:"stake",name:"delegate",description:"Delegate 14.167802221131358682 eGLD to staking provider ARC Stake",arguments:{value:"14167802221131358682",providerName:"ARC Stake",providerAvatar:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"}},results:[{hash:"8970fe5f5548caa994e495798ee1874b07f4ac2156d6a3a9636f450211350784",timestamp:1640702304,nonce:0,gasLimit:0,gasPrice:1e9,value:"14167802221131358682",sender:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqz8llllsh6u4jp",receiver:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqplllst77y4l",senderAssets:{name:"Staking: ARC Stake",description:"ARC Staking provides staking services on the Elrond Blockchain, the most advanced blockchain platform in the world.",tags:["staking","provider"],iconPng:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"},receiverAssets:{name:"System: Staking module",description:"Smart contract containing all staked eGLD on the network",tags:["system","staking"],iconPng:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.png",iconSvg:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.svg"},prevTxHash:"5996572de1fc5df2c456602467e8cc1a446b8145ae79df8f6c670e0732ab3b04",originalTxHash:"5996572de1fc5df2c456602467e8cc1a446b8145ae79df8f6c670e0732ab3b04",callType:"0",miniBlockHash:"432d6b8022301b67f1d76b5a96d525758e934b026128251d7ad462edb003c4b9"},{hash:"d4d84be4aa09cf5d45f79ffed87c971e5d883e7581d31e00d585f65b2da0564d",timestamp:1640702316,nonce:242,gasLimit:0,gasPrice:1e9,value:"59380000000000",sender:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqz8llllsh6u4jp",receiver:"erd1zwq3qaa3vk5suenlkj4cf0ullwefa6h3n2394k25pxv4sz0pwhhsj9u9vk",senderAssets:{name:"Staking: ARC Stake",description:"ARC Staking provides staking services on the Elrond Blockchain, the most advanced blockchain platform in the world.",tags:["staking","provider"],iconPng:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"},data:"QDZmNmI=",prevTxHash:"5996572de1fc5df2c456602467e8cc1a446b8145ae79df8f6c670e0732ab3b04",originalTxHash:"5996572de1fc5df2c456602467e8cc1a446b8145ae79df8f6c670e0732ab3b04",callType:"0",miniBlockHash:"578fc3b632fa3899ae00a6c7c399bb80d9b8f3e7c9e9396f6730efa73900cf3e",logs:{id:"d4d84be4aa09cf5d45f79ffed87c971e5d883e7581d31e00d585f65b2da0564d",address:"erd1zwq3qaa3vk5suenlkj4cf0ullwefa6h3n2394k25pxv4sz0pwhhsj9u9vk",events:[{address:"erd1zwq3qaa3vk5suenlkj4cf0ullwefa6h3n2394k25pxv4sz0pwhhsj9u9vk",identifier:"completedTxEvent",topics:["WZZXLeH8XfLEVmAkZ+jMGkRrgUWued+PbGcOBzKrOwQ="],order:0}]}}],price:240.75,logs:{id:"5996572de1fc5df2c456602467e8cc1a446b8145ae79df8f6c670e0732ab3b04",address:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqz8llllsh6u4jp",addressAssets:{name:"Staking: ARC Stake",description:"ARC Staking provides staking services on the Elrond Blockchain, the most advanced blockchain platform in the world.",tags:["staking","provider"],iconPng:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"},events:[{address:"erd1zwq3qaa3vk5suenlkj4cf0ullwefa6h3n2394k25pxv4sz0pwhhsj9u9vk",identifier:"delegate",topics:["xJ4lDk6GYdo=","BKAbisZRO5yq","E8I=","Ha6dNo2YLEc53A=="],order:0}]},operations:[{id:"8970fe5f5548caa994e495798ee1874b07f4ac2156d6a3a9636f450211350784",action:"transfer",type:"egld",sender:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqz8llllsh6u4jp",receiver:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqplllst77y4l",senderAssets:{name:"Staking: ARC Stake",description:"ARC Staking provides staking services on the Elrond Blockchain, the most advanced blockchain platform in the world.",tags:["staking","provider"],iconPng:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"},receiverAssets:{name:"System: Staking module",description:"Smart contract containing all staked eGLD on the network",tags:["system","staking"],iconPng:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.png",iconSvg:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.svg"},value:"14167802221131358682"}]},{txHash:"5e32a3f1808e7909a25381784731d6fc247325cc0a2737595129e783d50a3065",gasLimit:12e6,gasPrice:1e9,gasUsed:12e6,miniBlockHash:"e233dd9bc6f3b886ef3c4c5dfd722b291ea189bd4cd7065720730bee84553774",nonce:348,receiver:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqz8llllsh6u4jp",receiverAssets:{name:"Staking: ARC Stake",description:"ARC Staking provides staking services on the Elrond Blockchain, the most advanced blockchain platform in the world.",tags:["staking","provider"],iconPng:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"},receiverShard:4294967295,round:7430390,sender:"erd122h3x0nk8jddu3tfqujs67jffwuwu300c23da5lyxl9z5hyqtq8shceft6",senderShard:1,signature:"c449a086173d86c6c136278cd129c402cc64b88c757c85273bafcaad40be2b5128bf638d549d123448b09e1b4467954b8d1db43ff5aaf0a2cec66f63e779ec0f",status:"success",value:"0",fee:"209595000000000",timestamp:1640699940,data:"dW5EZWxlZ2F0ZUA0NTYzOTE4MjQ0ZjQwMDAw",function:"unDelegate",action:{category:"stake",name:"unDelegate",description:"Undelegate 5 eGLD from staking provider ARC Stake",arguments:{value:"5000000000000000000",providerName:"ARC Stake",providerAvatar:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"}}},{txHash:"813b7314a27b5aa4fd7e7c9565a826eca12984b620f674c756b01595b74a7ced",gasLimit:6e6,gasPrice:1e9,gasUsed:1068e3,miniBlockHash:"4bd1cc674dffdb322aa9d88d023fa8ec8a4f1f93bee73b00ee261bef7e5c4ce0",nonce:55,receiver:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqz8llllsh6u4jp",receiverAssets:{name:"Staking: ARC Stake",description:"ARC Staking provides staking services on the Elrond Blockchain, the most advanced blockchain platform in the world.",tags:["staking","provider"],iconPng:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"},receiverShard:4294967295,round:7430236,sender:"erd194ym3hhkw5dl6qxnxvpnnafjj54dyr8lycnev6sg32mqx270pnpq98ryvf",senderShard:2,signature:"b03ebdd464c37c1906bf2c2f1d957736c551fa311f6f5ce2c34c5b6d7b6516e46cac110d2ce8cc846530f2f3ac136d1d3d496b15ec421a73610b73da32480207",status:"success",value:"0",fee:"78000000000000",timestamp:1640699016,data:"Y2xhaW1SZXdhcmRz",function:"claimRewards",action:{category:"stake",name:"claimRewards",description:"Claim rewards from staking provider ARC Stake",arguments:{providerName:"ARC Stake",providerAvatar:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"}}},{txHash:"96f70d60de9fd2b29ea42841a507fe8b8aa0de07e9ed877c9f0fa8ec1feccc51",gasLimit:12e6,gasPrice:1e9,gasUsed:12e6,miniBlockHash:"f7ce09752494703b0f7e70f65a6e8ce7cf91684db94056e95300bc94e0f2a4da",nonce:174,receiver:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqz8llllsh6u4jp",receiverAssets:{name:"Staking: ARC Stake",description:"ARC Staking provides staking services on the Elrond Blockchain, the most advanced blockchain platform in the world.",tags:["staking","provider"],iconPng:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"},receiverShard:4294967295,round:7429921,sender:"erd1vcsum7fapl6crfrpqg7petydufgxm9qtqnuvhdqygdc6xc6uutksuzfatm",senderShard:1,signature:"ae7c8ae1cfa88474088bf886f1aff12e0ff85c9e1b61facfc46c79aa5314746e7c0ccb4df79197879c4c2171278399f833f8173ec5e7c0f25d64cfb6558d8401",status:"success",value:"0",fee:"181380000000000",timestamp:1640697126,data:"d2l0aGRyYXc=",function:"withdraw",action:{category:"stake",name:"withdraw",description:"Withdraw from staking provider ARC Stake",arguments:{providerName:"ARC Stake",providerAvatar:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"}}},{txHash:"6f570b2e75883df0a525786bbe69da141e0e4c4fc39accac17c9819162e77fbc",gasLimit:1175500,gasPrice:1e9,gasUsed:463200,miniBlockHash:"599c1a26a61dcd9efa5de88deeb856a0278a7e4abeaee017dc72c22d6a118425",nonce:408,receiver:"erd1y8jm2342j4jcyy4kz7hv823lr36wd9ur2mmlq4ea5wn6jndhpnjqh5shcg",receiverShard:0,round:7228515,sender:"erd1y8jm2342j4jcyy4kz7hv823lr36wd9ur2mmlq4ea5wn6jndhpnjqh5shcg",senderShard:0,signature:"b1ee08ee093949d5974461ea8d673d1761139a9bf208cac75a271b93f5f6e3a1b1ebd1bac62bf04c240a7f39e097c0e5f2730df8fc33f113069917a4d1c9c509",status:"success",value:"0",fee:"227877000000000",timestamp:1639488690,data:"RVNEVE5GVFRyYW5zZmVyQDQxNTI1NDQzNTI0MTQ2NTQyZDMzMzIzMjYzMzY2NUAxMUAwMUA3MTc3YjMwYjU0OGM3N2VjMjVlNzQ2NmYwMzg2ZjZlNGQ0NmFlNGZhMzk3Mzg0ZGEzYmVkYWQxNmEwNTA0NzA3",action:{category:"esdtNft",name:"transfer",description:"Transfer",arguments:{transfers:[{type:"NonFungibleESDT",name:"Genesis",ticker:"ARTCRAFT-322c6e",collection:"ARTCRAFT-322c6e",identifier:"ARTCRAFT-322c6e-11",value:"1"}],receiver:"erd1w9mmxz6533m7cf08gehs8phkun2x4e8689ecfk3makk3dgzsgurszhsxk4"}}},{txHash:"d51a4d6c74f1ab359e01dd18148f01ef64ab49162570f0b325fa35103d9784fb",gasLimit:12e6,gasPrice:1e9,gasUsed:6075500,miniBlockHash:"f5d84f705198e940905b36bc6231a90bdb3c1cf6f7af1776bc33bae34e103180",nonce:256,receiver:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqq2lllllspsw9sj",receiverAssets:{name:"Staking: Unity Stake",description:"We are a dedicated Elrond Network staking provider aiming to make things as simple and safe as possible by providing personal and caring support.",tags:["staking","provider"],iconPng:"https://s3.amazonaws.com/keybase_processed_uploads/4dd9f0dc7321d396f61f271951bba405_360_360.jpg"},receiverShard:4294967295,round:7071577,sender:"erd1vt2c72styh9eyrlsss855tprxgvtjdhmjpw8kyuapnjudlggatks0q0gm4",senderShard:1,signature:"5d198321defff0c4f451bc16434913f827efb25bb9b86fbf60aeca5ccefc896e27faa7de821f3c998b47dba9f18bae60aa5cba882cef9ee9d37389b5ed262200",status:"success",value:"0",fee:"135500000000000",timestamp:1638547062,data:"cmVEZWxlZ2F0ZVJld2FyZHM=",function:"reDelegateRewards",action:{category:"stake",name:"reDelegateRewards",description:"Redelegate rewards from staking provider Unity Stake",arguments:{providerName:"Unity Stake",providerAvatar:"https://s3.amazonaws.com/keybase_processed_uploads/4dd9f0dc7321d396f61f271951bba405_360_360.jpg"}}},{txHash:"20ab299a06956e3ea122445b3d8a181a226c4dacb0730dd9c13ede7ab5ee9328",gasLimit:14e6,gasPrice:1e9,gasUsed:11520759,miniBlockHash:"4e53661c4ac0803d4e53aa3fd6fa1eae0b4da337bd4630d107229ad0d3af0a85",nonce:112,receiver:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",receiverShard:0,round:7057306,sender:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",senderShard:0,signature:"07ca4425308793064273435b2fdbfea61acdd7d34bbe6371c3a0e459256003c8f72d52c31cf4538e173965da8a91729686814f5d321426a56b85746f9ddcb002",status:"success",value:"0",fee:"501802590000000",timestamp:1638461436,data:"TXVsdGlFU0RUTkZUVHJhbnNmZXJAMDAwMDAwMDAwMDAwMDAwMDA1MDA5MDU2ZjAwMTEwMWI2ZTE1MTgwNDQwNjRiNDhjMmU0ZmI0OTJjZDg4NTQ4M0AwMkA0YzRiNGQ0NTU4MmQ2MTYxNjIzOTMxMzBAMDNAMDEzZmJlODVlZGM5MDAwMEA0YzRiNGQ0NTU4MmQ2MTYxNjIzOTMxMzBAMGQzZEAxMTJkNmFiZGQ5NWUxZDVlNDBANmQ2NTcyNjc2NTRjNmY2MzZiNjU2NDQxNzM3MzY1NzQ1NDZmNmI2NTZlNzM=",function:"mergeLockedAssetTokens",action:{category:"mex",name:"mergeLockedAssetTokens",description:"Merge 2 LKMEX positions into a single LKMEX position of value 316.957286073585",arguments:{transfers:[{type:"MetaESDT",name:"LockedMEX",ticker:"LKMEX",svgUrl:"https://media.elrond.com/tokens/asset/LKMEX-aab910/logo.svg",collection:"LKMEX-aab910",decimals:18,identifier:"LKMEX-aab910-03",value:"90000000000000000"},{type:"MetaESDT",name:"LockedMEX",ticker:"LKMEX",svgUrl:"https://media.elrond.com/tokens/asset/LKMEX-aab910/logo.svg",collection:"LKMEX-aab910",decimals:18,identifier:"LKMEX-aab910-0d3d",value:"316867286073585000000"}],receiver:"erd1qqqqqqqqqqqqqpgqjpt0qqgsrdhp2xqygpjtfrpwf76f9nvg2jpsg4q7th",functionName:"mergeLockedAssetTokens",receiverAssets:{name:"Maiar Exchange: Locked asset Contract",tags:["mex","lockedasset"]}}}},{txHash:"91d1be9345b8b93983f105bca08ec62c99e13cacb40098773c549a7450aacf3b",gasLimit:1e6,gasPrice:1e9,gasUsed:459600,miniBlockHash:"2c5add74b4224c785440fc957ea5389f288953fdd4c9ddab91d11a1e7f436939",nonce:170,receiver:"erd180yut8q4ve7mzuqtens26ct0tdg0hwmhjfr0s8t2z0ujukmet8rqtt2px0",receiverShard:2,round:7044527,sender:"erd180yut8q4ve7mzuqtens26ct0tdg0hwmhjfr0s8t2z0ujukmet8rqtt2px0",senderShard:2,signature:"cda937a0e6c62ff1f745c5500d1aa9cf0a2009051dc3fdfb2abeaa3c6701946bd3946c8811a8cea6addbdf598f57d27d4ab059d765c88af2085758c4ca2c5807",status:"success",value:"0",fee:"248631000000000",timestamp:1638384762,data:"RVNEVE5GVFRyYW5zZmVyQDRjNGI0NjQxNTI0ZDJkMzk2NDMxNjU2MTM4QDEyNmMxN0AxNTU1ODA4OWJkYmJmNmI5QGNjZmUxOTY4N2JkN2QxMjgwZTY1YTQ3ZTI5NDI0OGE3ODhiNjFkZmUxYTIyNDRmODBhNDEwMTEyMDFlN2MwMTU=",action:{category:"esdtNft",name:"transfer",description:"Transfer",arguments:{transfers:[{type:"MetaESDT",name:"LockedLPStaked",ticker:"LKFARM",svgUrl:"https://media.elrond.com/tokens/asset/LKFARM-9d1ea8/logo.svg",collection:"LKFARM-9d1ea8",decimals:18,identifier:"LKFARM-9d1ea8-126c17",value:"1537276176898979513"}],receiver:"erd1enlpj6rm6lgjsrn953lzjsjg57ytv807rg3yf7q2gyq3yq08cq2sfphazl"}}},{txHash:"5bbe1a53027e6850362fcb4633108627bbe241adbb48001e25e3cd8129861189",gasLimit:105e5,gasPrice:1e9,gasUsed:9755137,miniBlockHash:"29e8111d14131cfc1b80b354ef47899aecff43873b333636e80997bb58d39f6d",nonce:864,receiver:"erd1qqqqqqqqqqqqqpgqa0fsfshnff4n76jhcye6k7uvd7qacsq42jpsp6shh2",receiverAssets:{name:"Maiar Exchange: MEX/WEGLD Liquidity Pool",tags:["mex","liquiditypool"]},receiverShard:1,round:6885918,sender:"erd1dwpev7tgza3hv58u8syzlk8nxsnkm94slwvvsfs44qwu3dqega7sqq3uxx",senderShard:1,signature:"8f028a4635be9cd8e02c2ae180da1a65f95a2b65dea4f65c3b25863f9b2be0c1d6c1059d1c9aafe88dd05e80a709685a120042099a4f182116a071ed7e11500b",status:"success",value:"0",fee:"350496370000000",timestamp:1637433108,data:"RVNEVFRyYW5zZmVyQDU3NDU0NzRjNDQyZDYyNjQzNDY0MzczOUAwZGUwYjZiM2E3NjQwMDAwQDczNzc2MTcwNTQ2ZjZiNjU2ZTczNDY2OTc4NjU2NDQ5NmU3MDc1NzRANGQ0NTU4MmQzNDM1MzU2MzM1MzdAM2I4NGE4YzZmYmY2YjJhZDA5Y2Y=",function:"swapTokensFixedInput",action:{category:"mex",name:"swap",description:"Swap 1 WEGLD for a minimum of 281066.754391919467235791 MEX",arguments:{transfers:[{type:"FungibleESDT",name:"WrappedEGLD",ticker:"WEGLD",svgUrl:"https://media.elrond.com/tokens/asset/WEGLD-bd4d79/logo.svg",token:"WEGLD-bd4d79",decimals:18,value:"1000000000000000000"},{type:"FungibleESDT",name:"MEX",ticker:"MEX",svgUrl:"https://media.elrond.com/tokens/asset/MEX-455c57/logo.svg",token:"MEX-455c57",decimals:18,value:"281066754391919467235791"}],receiver:"erd1qqqqqqqqqqqqqpgqa0fsfshnff4n76jhcye6k7uvd7qacsq42jpsp6shh2",functionName:"swapTokensFixedInput",functionArgs:["4d45582d343535633537","3b84a8c6fbf6b2ad09cf"],receiverAssets:{name:"Maiar Exchange: MEX/WEGLD Liquidity Pool",tags:["mex","liquiditypool"]}}}},{txHash:"6b65404efa716ad6dc6a2f37b6b6a7677cea70815fb699451d6ff1875f2111e2",gasLimit:4e6,gasPrice:1e9,gasUsed:2111255,miniBlockHash:"29e8111d14131cfc1b80b354ef47899aecff43873b333636e80997bb58d39f6d",nonce:863,receiver:"erd1qqqqqqqqqqqqqpgqhe8t5jewej70zupmh44jurgn29psua5l2jps3ntjj3",receiverAssets:{name:"ESDT: WrappedEGLD Contract Shard 1",tags:["mex","wegld"]},receiverShard:1,round:6885918,sender:"erd1dwpev7tgza3hv58u8syzlk8nxsnkm94slwvvsfs44qwu3dqega7sqq3uxx",senderShard:1,signature:"521ce31af51e02f2bf31b9198807784cf7a37c97798acc7859e7c5c5c3e24651b15fe392af24febf90d5a2864614f12f1852eb6ad541e61c456448b6d1c86807",status:"success",value:"1000000000000000000",fee:"82492550000000",timestamp:1637433108,data:"d3JhcEVnbGQ=",function:"wrapEgld",action:{category:"mex",name:"wrapEgld",description:"Wrap 1 eGLD",arguments:{token:{type:"FungibleESDT",name:"WrappedEGLD",token:"WEGLD-bd4d79",ticker:"WEGLD",decimals:18,value:"1000000000000000000"},receiver:"erd1qqqqqqqqqqqqqpgqhe8t5jewej70zupmh44jurgn29psua5l2jps3ntjj3",receiverAssets:{name:"ESDT: WrappedEGLD Contract Shard 1",tags:["mex","wegld"]}}}},{txHash:"8db34cc6b7a8f8bcdfb4dbf0ca55cb885f625da1dfe316fc9e9d722ebc687524",gasLimit:5e5,gasPrice:1e9,gasUsed:316e3,miniBlockHash:"966bcf89dd731f1ed563e37dcab3ec8d923382a769eec6d4ab03eb490d573dee",nonce:813,receiver:"erd1k50edvacs6qvsfyuex38d8plzswus35knd4wxh39apjlpz0j308qpeywwr",receiverShard:2,round:6885918,sender:"erd1fdcfhqsa2qp0l6grzpehndgeum8kzavd5hxjn9h7f0llafwrvyfqvt6k2y",senderShard:2,signature:"8f7caff6c24b5a685008cb93b3bf4c8c53d50215d4e3c8de3f69aeeec4220f183840031a0b220dd38630f4456af11c80d550c486b174273a91fe47f27ae9c302",status:"success",value:"0",fee:"118000000000000",timestamp:1637433108,data:"RVNEVFRyYW5zZmVyQDU1NTM0NDQzMmQ2MzM3MzY2NjMxNjZAYWZkNTZkODA=",action:{category:"esdtNft",name:"transfer",description:"Transfer",arguments:{transfers:[{type:"FungibleESDT",name:"WrappedUSDC",ticker:"USDC",svgUrl:"https://media.elrond.com/tokens/asset/USDC-c76f1f/logo.svg",token:"USDC-c76f1f",decimals:6,value:"2950000000"}],receiver:"erd1k50edvacs6qvsfyuex38d8plzswus35knd4wxh39apjlpz0j308qpeywwr"}}},{txHash:"4abc9f3a18b2685bc8e34f26e0815512899787e10ef66b8ace105a8f48e3f694",gasLimit:4e6,gasPrice:1e9,gasUsed:2213320,miniBlockHash:"29e8111d14131cfc1b80b354ef47899aecff43873b333636e80997bb58d39f6d",nonce:761,receiver:"erd1qqqqqqqqqqqqqpgqhe8t5jewej70zupmh44jurgn29psua5l2jps3ntjj3",receiverAssets:{name:"ESDT: WrappedEGLD Contract Shard 1",tags:["mex","wegld"]},receiverShard:1,round:6885918,sender:"erd1e7ycjzurhan9tcryu5t65sv3pcj0rg5xh490cg9mmpy7k582zsgsyufld5",senderShard:1,signature:"27b2329b6fbd6b717dc664d5c081f9763f4731b8d6d8905001f07dce2781faa98c8fc99deccf2a7b7bf1e3e6666b732ab99c420ebdc1af9f8c5a8cc47808820e",status:"success",value:"0",fee:"183008200000000",timestamp:1637433108,data:"RVNEVFRyYW5zZmVyQDU3NDU0NzRjNDQyZDYyNjQzNDY0MzczOUA0NGVjYzgzNzEwYzU3OWQzQDc1NmU3NzcyNjE3MDQ1Njc2YzY0",function:"unwrapEgld",action:{category:"mex",name:"unwrapEgld",description:"Unwrap 4.966564627912948179 WEGLD",arguments:{transfers:[{type:"FungibleESDT",name:"WrappedEGLD",ticker:"WEGLD",svgUrl:"https://media.elrond.com/tokens/asset/WEGLD-bd4d79/logo.svg",token:"WEGLD-bd4d79",decimals:18,value:"4966564627912948179"}],receiver:"erd1qqqqqqqqqqqqqpgqhe8t5jewej70zupmh44jurgn29psua5l2jps3ntjj3",functionName:"unwrapEgld",receiverAssets:{name:"ESDT: WrappedEGLD Contract Shard 1",tags:["mex","wegld"]}}}},{txHash:"e060a7820af21bae38686ec036c2e28df6295d2492dde473c23c5ebbf83a40cf",gasLimit:6e7,gasPrice:1e9,gasUsed:51054664,miniBlockHash:"29e8111d14131cfc1b80b354ef47899aecff43873b333636e80997bb58d39f6d",nonce:377,receiver:"erd1096gglmcpgrfudgfr6ya8vjxev7gslzfnngyn5qgspt3df9nvlfsttsfwl",receiverShard:1,round:6885918,sender:"erd1096gglmcpgrfudgfr6ya8vjxev7gslzfnngyn5qgspt3df9nvlfsttsfwl",senderShard:1,signature:"bea080cfc7329bc4b181332935100f7772f0ba4133d5e8be807ca78dc9b09218ff4b9a02f730a2cfd28a3256f40324373d78df90c4b559ad6947207d40383500",status:"success",value:"0",fee:"903081640000000",timestamp:1637433108,data:"RVNEVE5GVFRyYW5zZmVyQDRjNGI0NjQxNTI0ZDJkMzk2NDMxNjU2MTM4QDAxNjcxMUA3MDlhNWYxNjJiOGM3YmU2QDAwMDAwMDAwMDAwMDAwMDAwNTAwMWUyYTE0MjhkZDFlM2E1MTQ2YjM5NjBkOWUwZjRhNTAzNjk5MDRlZTU0ODNANjM2YzYxNjk2ZDUyNjU3NzYxNzI2NDczNTA3MjZmNzg3OUAwMDAwMDAwMDAwMDAwMDAwMDUwMDI2NzUxODkzZDY3ODliZTllNWE5OTg2M2JhOWVlYWE4MDg4ZGQyNWY1NDgz",function:"claimRewardsProxy",action:{category:"mex",name:"claimRewards",description:"Claim rewards for 8.113902227485916134 LKFARM",arguments:{transfers:[{type:"MetaESDT",name:"LockedLPStaked",ticker:"LKFARM",svgUrl:"https://media.elrond.com/tokens/asset/LKFARM-9d1ea8/logo.svg",collection:"LKFARM-9d1ea8",decimals:18,identifier:"LKFARM-9d1ea8-016711",value:"8113902227485916134"}],receiver:"erd1qqqqqqqqqqqqqpgqrc4pg2xarca9z34njcxeur622qmfjp8w2jps89fxnl",functionName:"claimRewardsProxy",functionArgs:["0000000000000000050026751893d6789be9e5a99863ba9eeaa8088dd25f5483"],receiverAssets:{name:"Maiar Exchange: Proxy",tags:["maiar","exchange","proxy"],iconPng:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.png",iconSvg:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.svg"}}}},{txHash:"7b4ac0ef53f9b43c80cc2af6ae0a1e4e9065a3054a1358775edb6161eaa02811",gasLimit:12e6,gasPrice:1e9,gasUsed:10717267,miniBlockHash:"29e8111d14131cfc1b80b354ef47899aecff43873b333636e80997bb58d39f6d",nonce:274,receiver:"erd1x0syqh7tcauqx70qdakmaa48mkmnczh8kl5zw3y0tnp08kfuz9zsqtmmtk",receiverShard:1,round:6885918,sender:"erd1x0syqh7tcauqx70qdakmaa48mkmnczh8kl5zw3y0tnp08kfuz9zsqtmmtk",senderShard:1,signature:"c8ae158d499004313c8d7e0cbede71c9302eb080ea131bd6a53fc7d7fc9d71364f0ff80408948e2a09652003903352a0a13b8f0625497bda3dc1e03297150f07",status:"success",value:"0",fee:"508617670000000",timestamp:1637433108,data:"TXVsdGlFU0RUTkZUVHJhbnNmZXJAMDAwMDAwMDAwMDAwMDAwMDA1MDBlYmQzMDRjMmYzNGE2YjNmNmE1N2MxMzNhYjdiOGM2ZjgxZGM0MDE1NTQ4M0AwMkA1NzQ1NDc0YzQ0MmQ2MjY0MzQ2NDM3MzlAQDA0NGNmNjhjMjkwODUwMDlANGQ0NTU4MmQzNDM1MzU2MzM1MzdAQDEyYWZlMGQzODQ2MTMzMjBjNTQ0QDYxNjQ2NDRjNjk3MTc1Njk2NDY5NzQ3OUAwNDQxZjQxNGZmYTk4MjZmQDEyODAwYTE5MTRmOWQzZTU5MDE3",function:"addLiquidity",action:{category:"mex",name:"addLiquidity",description:"Added liquidity for 0.309893556225986569 WEGLD, 88246.97734239037166522 MEX",arguments:{transfers:[{type:"FungibleESDT",name:"WrappedEGLD",ticker:"WEGLD",svgUrl:"https://media.elrond.com/tokens/asset/WEGLD-bd4d79/logo.svg",token:"WEGLD-bd4d79",decimals:18,value:"309893556225986569"},{type:"FungibleESDT",name:"MEX",ticker:"MEX",svgUrl:"https://media.elrond.com/tokens/asset/MEX-455c57/logo.svg",token:"MEX-455c57",decimals:18,value:"88246977342390371665220"}],receiver:"erd1qqqqqqqqqqqqqpgqa0fsfshnff4n76jhcye6k7uvd7qacsq42jpsp6shh2",functionName:"addLiquidity",functionArgs:["0441f414ffa9826f","12800a1914f9d3e59017"],receiverAssets:{name:"Maiar Exchange: MEX/WEGLD Liquidity Pool",tags:["mex","liquiditypool"]}}}},{txHash:"a13bf18c6d91edb7dc403d915fd89298806f67d59fdd72f12370f2c6fe31966c",gasLimit:38e6,gasPrice:1e9,gasUsed:34290478,miniBlockHash:"29e8111d14131cfc1b80b354ef47899aecff43873b333636e80997bb58d39f6d",nonce:222,receiver:"erd1znusjpmfpukrtf0wscvf4su8yjqg393c092j8sxxvvrfk9m0rdss0646y0",receiverShard:1,round:6885918,sender:"erd1znusjpmfpukrtf0wscvf4su8yjqg393c092j8sxxvvrfk9m0rdss0646y0",senderShard:1,signature:"e06d88a54af0d9fc220050b69d5c2c4368765ce2a382871cc341531ad309cc695b983a11a24efe99697407ac028e7f4cbdd8ff1a79ed17765760fa33897e340b",status:"success",value:"0",fee:"753259780000000",timestamp:1637433108,data:"RVNEVE5GVFRyYW5zZmVyQDRjNGI0NjQxNTI0ZDJkMzk2NDMxNjU2MTM4QDAxNmEzMUAwYTAxMTBiYjNkMGZmNDJmN2ZlZDQ4QDAwMDAwMDAwMDAwMDAwMDAwNTAwMWUyYTE0MjhkZDFlM2E1MTQ2YjM5NjBkOWUwZjRhNTAzNjk5MDRlZTU0ODNANjM2ZjZkNzA2Zjc1NmU2NDUyNjU3NzYxNzI2NDczNTA3MjZmNzg3OUAwMDAwMDAwMDAwMDAwMDAwMDUwMDY1NmQwYWNjNTM1NjFjNWQ2ZjZmZDdkN2U4MmJmMTMyNDcwMTRmNjE1NDgz",function:"compoundRewardsProxy",action:{category:"mex",name:"compoundRewards",description:"Reinvest rewards for 12094289.202491940860325192 LKFARM",arguments:{transfers:[{type:"MetaESDT",name:"LockedLPStaked",ticker:"LKFARM",svgUrl:"https://media.elrond.com/tokens/asset/LKFARM-9d1ea8/logo.svg",collection:"LKFARM-9d1ea8",decimals:18,identifier:"LKFARM-9d1ea8-016a31",value:"12094289202491940860325192"}],receiver:"erd1qqqqqqqqqqqqqpgqrc4pg2xarca9z34njcxeur622qmfjp8w2jps89fxnl",functionName:"compoundRewardsProxy",functionArgs:["00000000000000000500656d0acc53561c5d6f6fd7d7e82bf13247014f615483"],receiverAssets:{name:"Maiar Exchange: Proxy",tags:["maiar","exchange","proxy"],iconPng:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.png",iconSvg:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.svg"}}}},{txHash:"8f3b73756ecf8500737d0f68d300585c19e9d9f38f3b112171ca71e6890c2ab2",gasLimit:165e5,gasPrice:1e9,gasUsed:11554861,miniBlockHash:"7b6cf2eb70d414973ac1294fb02ceec49d5b2f48a8fcf56e96eb13e71299912d",nonce:150,receiver:"erd1z3q7g93vj2vqz6sevse7cugauh6e5uszelwsfjledpzz9qc2xg2q97sqvr",receiverShard:0,round:6885918,sender:"erd1z3q7g93vj2vqz6sevse7cugauh6e5uszelwsfjledpzz9qc2xg2q97sqvr",senderShard:0,signature:"9ef7c45815f4147b75aaff150f55347f73efbbc7158731273c9c363c163aa34cabdd324d24449d8657a9faba896cc33fc64f86224123dab6ca71d7aa556b5004",status:"success",value:"0",fee:"438288610000000",timestamp:1637433108,data:"TXVsdGlFU0RUTkZUVHJhbnNmZXJAMDAwMDAwMDAwMDAwMDAwMDA1MDAyNjc1MTg5M2Q2Nzg5YmU5ZTVhOTk4NjNiYTllZWFhODA4OGRkMjVmNTQ4M0AwMUA0NTQ3NGM0NDRkNDU1ODJkMzA2MjY1Mzk2NTM1QEAwMzUxYmFiNTY3YWFkODE2NTlANjU2ZTc0NjU3MjQ2NjE3MjZkNDE2ZTY0NGM2ZjYzNmI1MjY1Nzc2MTcyNjQ3Mw==",function:"enterFarmAndLockRewards",action:{category:"mex",name:"enterFarm",description:"Enter farm and lock rewards with 61.229451140721546841 EGLDMEX",arguments:{transfers:[{type:"FungibleESDT",name:"EGLDMEXLP",ticker:"EGLDMEX",svgUrl:"https://media.elrond.com/tokens/asset/EGLDMEX-0be9e5/logo.svg",token:"EGLDMEX-0be9e5",decimals:18,value:"61229451140721546841"}],receiver:"erd1qqqqqqqqqqqqqpgqye633y7k0zd7nedfnp3m48h24qygm5jl2jpslxallh",functionName:"enterFarmAndLockRewards",receiverAssets:{name:"Maiar Exchange: EGLDMEXLPStaked Farm",tags:["mex","farm"]}}}},{txHash:"6e29bb7454bad9c145d76308f7c36efb5cc8a83619d2ca26a5ed165a302e50e7",gasLimit:36e6,gasPrice:1e9,gasUsed:34299559,miniBlockHash:"7b6cf2eb70d414973ac1294fb02ceec49d5b2f48a8fcf56e96eb13e71299912d",nonce:70,receiver:"erd1dasthqh3ydgtjnxj4xmzrm02rru7dk4ytq6vqsjdrq4p9lul7hjqms7q4m",receiverShard:0,round:6885918,sender:"erd1dasthqh3ydgtjnxj4xmzrm02rru7dk4ytq6vqsjdrq4p9lul7hjqms7q4m",senderShard:0,signature:"edafd4d9f8255f1bf8e416fd3944b74784faaf86baa242978e11a52f67e07b6273d4bc209f46e11d192b60b1181075b36af243d938da64bdf8940b4426a0900d",status:"success",value:"0",fee:"738500590000000",timestamp:1637433108,data:"TXVsdGlFU0RUTkZUVHJhbnNmZXJAMDAwMDAwMDAwMDAwMDAwMDA1MDAxZTJhMTQyOGRkMWUzYTUxNDZiMzk2MGQ5ZTBmNGE1MDM2OTkwNGVlNTQ4M0AwMUA0YzRiNGQ0NTU4MmQ2MTYxNjIzOTMxMzBAMDRAMDUxZDVkZDAzNjA3Y2ZhMmJlMDAwMEA2NTZlNzQ2NTcyNDY2MTcyNmQ1MDcyNmY3ODc5QDAwMDAwMDAwMDAwMDAwMDAwNTAwNjU2ZDBhY2M1MzU2MWM1ZDZmNmZkN2Q3ZTgyYmYxMzI0NzAxNGY2MTU0ODM=",function:"enterFarmProxy",action:{category:"mex",name:"enterFarm",description:"Enter farm with 6183308.2764630175744 LKMEX",arguments:{transfers:[{type:"MetaESDT",name:"LockedMEX",ticker:"LKMEX",svgUrl:"https://media.elrond.com/tokens/asset/LKMEX-aab910/logo.svg",collection:"LKMEX-aab910",decimals:18,identifier:"LKMEX-aab910-04",value:"6183308276463017574400000"}],receiver:"erd1qqqqqqqqqqqqqpgqrc4pg2xarca9z34njcxeur622qmfjp8w2jps89fxnl",functionName:"enterFarmProxy",functionArgs:["00000000000000000500656d0acc53561c5d6f6fd7d7e82bf13247014f615483"],receiverAssets:{name:"Maiar Exchange: Proxy",tags:["maiar","exchange","proxy"],iconPng:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.png",iconSvg:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.svg"}}}},{txHash:"e47917728c85e355baf0bec96f3f51e0fe7b732594905c4b7ce8e03f49866d0f",gasLimit:115e5,gasPrice:1e9,gasUsed:9757888,miniBlockHash:"452b91830aebdfe7c73028af68221b84977a26455eb1d074bb71ce8f35591eaf",nonce:806,receiver:"erd1qqqqqqqqqqqqqpgqa0fsfshnff4n76jhcye6k7uvd7qacsq42jpsp6shh2",receiverAssets:{name:"Maiar Exchange: MEX/WEGLD Liquidity Pool",tags:["mex","liquiditypool"]},receiverShard:1,round:6885907,sender:"erd1esdzudw5wfzq4egwjwaxc9yueqes3cxw88cumaqs9dak9tsdseps95k0xt",senderShard:1,signature:"d3a30b6660ef25612e7ce551e2a7daab257dfb362bf363a86795e4aa7c3153be78d3f6e1223cc19b867ff37fa2ba769a2157bbd2e1a75f589d1f705b1fd3d505",status:"success",value:"0",fee:"335673880000000",timestamp:1637433042,data:"RVNEVFRyYW5zZmVyQDQ1NDc0YzQ0NGQ0NTU4MmQzMDYyNjUzOTY1MzVAMDQxZjI3ZWQyYTRhODAwMEA3MjY1NmQ2Zjc2NjU0YzY5NzE3NTY5NjQ2OTc0NzlAMDlhN2I0OTUyMDA1MmVkYkAyOWY1MTU3ODBmNDI1ZGYzNjMzMQ==",function:"removeLiquidity",action:{category:"mex",name:"removeLiquidity",description:"Removed liquidity with 0.297 EGLDMEX",arguments:{transfers:[{type:"FungibleESDT",name:"EGLDMEXLP",ticker:"EGLDMEX",svgUrl:"https://media.elrond.com/tokens/asset/EGLDMEX-0be9e5/logo.svg",token:"EGLDMEX-0be9e5",decimals:18,value:"297000000000000000"}],receiver:"erd1qqqqqqqqqqqqqpgqa0fsfshnff4n76jhcye6k7uvd7qacsq42jpsp6shh2",functionName:"removeLiquidity",functionArgs:["09a7b49520052edb","29f515780f425df36331"],receiverAssets:{name:"Maiar Exchange: MEX/WEGLD Liquidity Pool",tags:["mex","liquiditypool"]}}}},{txHash:"4f6559320c4687c8bfa065e0ed7d3de8c45ef4c1606af29616306cd5dabf33f1",gasLimit:615e5,gasPrice:1e9,gasUsed:48392131,miniBlockHash:"452b91830aebdfe7c73028af68221b84977a26455eb1d074bb71ce8f35591eaf",nonce:33,receiver:"erd1ecr8cdkpafrh9nw688hn2fq80lll53lqd4w9wl72kvde8qurntrssqgez2",receiverShard:1,round:6885907,sender:"erd1ecr8cdkpafrh9nw688hn2fq80lll53lqd4w9wl72kvde8qurntrssqgez2",senderShard:1,signature:"95e286ce67ec20bf7097e9a8f38d68a5fb318df31b8a73fffb7c1b729af0c65bf36f1d205d1d7cefbfd76fbb52c306f0f1f7c703aa0a974a8f04aaf3d8fbfc05",status:"success",value:"0",fee:"870516310000000",timestamp:1637433042,data:"RVNEVE5GVFRyYW5zZmVyQDRjNGI0NjQxNTI0ZDJkMzk2NDMxNjU2MTM4QDdmZjFAMDExZmE2OWY4ZTdiMDgzNDBkMDAwMEAwMDAwMDAwMDAwMDAwMDAwMDUwMDFlMmExNDI4ZGQxZTNhNTE0NmIzOTYwZDllMGY0YTUwMzY5OTA0ZWU1NDgzQDY1Nzg2OTc0NDY2MTcyNmQ1MDcyNmY3ODc5QDAwMDAwMDAwMDAwMDAwMDAwNTAwNjU2ZDBhY2M1MzU2MWM1ZDZmNmZkN2Q3ZTgyYmYxMzI0NzAxNGY2MTU0ODM=",function:"exitFarmProxy",action:{category:"mex",name:"exitFarm",description:"Exit farm with 1358392.837361997848576 LKFARM",arguments:{transfers:[{type:"MetaESDT",name:"LockedLPStaked",ticker:"LKFARM",svgUrl:"https://media.elrond.com/tokens/asset/LKFARM-9d1ea8/logo.svg",collection:"LKFARM-9d1ea8",decimals:18,identifier:"LKFARM-9d1ea8-7ff1",value:"1358392837361997848576000"}],receiver:"erd1qqqqqqqqqqqqqpgqrc4pg2xarca9z34njcxeur622qmfjp8w2jps89fxnl",functionName:"exitFarmProxy",functionArgs:["00000000000000000500656d0acc53561c5d6f6fd7d7e82bf13247014f615483"],receiverAssets:{name:"Maiar Exchange: Proxy",tags:["maiar","exchange","proxy"],iconPng:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.png",iconSvg:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.svg"}}}},{txHash:"02c3c2660f6c2bcfa767f9539201cd9062ab0f398c3db8bcdca1ae2cd718c639",gasLimit:75e6,gasPrice:1e9,gasUsed:7953928,miniBlockHash:"8c56f215588dd57815a0ca0147d643b958eda75ddbafb95fa24a84a9c98101e2",nonce:448,receiver:"erd1qqqqqqqqqqqqqpgqp699jngundfqw07d8jzkepucvpzush6k3wvqyc44rx",receiverShard:0,round:1812254,sender:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",senderShard:0,signature:"425d8db21db62692983095d9a2148bd0c12a337c743b50a480961ae0f135c1e48a7bcd3bb4e9eb5ce11bd8f96537aefd5660152748d7a09e294e90cb543d7904",status:"success",value:"0",fee:"167649280000000",timestamp:1664195124,data:"dW5TdGFrZUAwMTE1OGU0NjA5MTNkMDAwMDA=",function:"unStake",action:{category:"scCall",name:"unStake"}},{txHash:"ce1c1d6c4b36a3f239f822307624491cae6f0c4872970a3d0d02c3cc48b2844f",gasLimit:6e6,gasPrice:1e9,gasUsed:1068e3,miniBlockHash:"c72fca9867293054d9182aad167674142cf57c2fc3958282792ecddc55a74cca",nonce:446,receiver:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqphllllsndz99p",receiverShard:4294967295,round:1812226,sender:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",senderShard:0,signature:"6ed3d00d103e1ade45bd9a6e5ca8194661e414e5a2c8a30cf3360ea9bf6ccc71f14cbc56a68eefbc020502f9a2835278c9cb027c6407face5c6dc225e85c430e",status:"success",value:"0",fee:"78000000000000",timestamp:1664194956,data:"Y2xhaW1SZXdhcmRz",function:"claimRewards",action:{category:"stake",name:"claimRewards",description:"Claim rewards from staking provider https://keybase.io/drfonix",arguments:{providerName:"https://keybase.io/drfonix"}},results:[{hash:"03ec460f4037114403828f357dc91495149ab35c3556c6ad0155acd8c65c4b04",timestamp:1664194968,nonce:0,gasLimit:0,gasPrice:1e9,value:"67861704531764158",sender:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqphllllsndz99p",receiver:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",prevTxHash:"ce1c1d6c4b36a3f239f822307624491cae6f0c4872970a3d0d02c3cc48b2844f",originalTxHash:"ce1c1d6c4b36a3f239f822307624491cae6f0c4872970a3d0d02c3cc48b2844f",callType:"0",miniBlockHash:"8da210d07a8e099ced579f7ff634b3159f9589cbad106c55c51b4e1c5bf98c7d"},{hash:"ff5d9b11e3fddf3ee4bdf3dec0e82915f1c0b36233e1d05366b9950d78f96fea",timestamp:1664194968,nonce:447,gasLimit:0,gasPrice:1e9,value:"49320000000000",sender:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqphllllsndz99p",receiver:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",data:"QDZmNmI=",prevTxHash:"ce1c1d6c4b36a3f239f822307624491cae6f0c4872970a3d0d02c3cc48b2844f",originalTxHash:"ce1c1d6c4b36a3f239f822307624491cae6f0c4872970a3d0d02c3cc48b2844f",callType:"0",miniBlockHash:"8da210d07a8e099ced579f7ff634b3159f9589cbad106c55c51b4e1c5bf98c7d"}],operations:[{id:"03ec460f4037114403828f357dc91495149ab35c3556c6ad0155acd8c65c4b04",action:"transfer",type:"egld",sender:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqphllllsndz99p",receiver:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",value:"67861704531764158"}]},{txHash:"5e73704d40fdef680bd8839cd277827209f5decd3679b235efddd921f1c9fb3b",gasLimit:12e6,gasPrice:1e9,gasUsed:6075500,miniBlockHash:"3a7470a369d5e5bf6b5e984df1f6fdbed2fb260bb81574b1392e8f42d3c17f96",nonce:445,receiver:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqq9lllllsf3mp40",receiverShard:4294967295,round:1809456,sender:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",senderShard:0,signature:"3a25067722f2c8a98f693ccb2861f6ea1f91565d99e6b6a1aee57788d612a78d22cf1c3bfb55e94d9f5e881efad8632a4ad5be83e0830ad8f6181c8ac1552205",status:"success",value:"0",fee:"135500000000000",timestamp:1664178336,data:"cmVEZWxlZ2F0ZVJld2FyZHM=",function:"reDelegateRewards",action:{category:"stake",name:"reDelegateRewards",description:"Redelegate rewards from staking provider figment.io_testnet",arguments:{providerName:"figment.io_testnet",providerAvatar:"https://s3.amazonaws.com/keybase_processed_uploads/8cba5dc15c79565402849fd869654705_360_360.jpg"}},results:[{hash:"1bf45b782580cd035efdf600aafbfdd6d3fc437e93b4fb9219fe76f7526fad65",timestamp:1664178336,nonce:0,gasLimit:0,gasPrice:1e9,value:"392502810808819111",sender:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqq9lllllsf3mp40",receiver:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqplllst77y4l",prevTxHash:"5e73704d40fdef680bd8839cd277827209f5decd3679b235efddd921f1c9fb3b",originalTxHash:"5e73704d40fdef680bd8839cd277827209f5decd3679b235efddd921f1c9fb3b",callType:"0",miniBlockHash:"e74217b57b02069cf1b3294db1acbce08c5174811464c283cd1c739dbbf27b74"},{hash:"5fef839b3abc2f07be0923acfcb086a19ad2cf9522e9cf2eee4f398c84b301ef",timestamp:1664178348,nonce:446,gasLimit:0,gasPrice:1e9,value:"59245000000000",sender:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqq9lllllsf3mp40",receiver:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",data:"QDZmNmI=",prevTxHash:"5e73704d40fdef680bd8839cd277827209f5decd3679b235efddd921f1c9fb3b",originalTxHash:"5e73704d40fdef680bd8839cd277827209f5decd3679b235efddd921f1c9fb3b",callType:"0",miniBlockHash:"8f15889818ae5f2b63bebfae42deb1265ef500832aa5d12510246535140e1465"}],operations:[{id:"1bf45b782580cd035efdf600aafbfdd6d3fc437e93b4fb9219fe76f7526fad65",action:"transfer",type:"egld",sender:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqq9lllllsf3mp40",receiver:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqplllst77y4l",value:"392502810808819111"}]},{txHash:"5bdfd566dcc2c19a8bd62caa138acd2662a9d5f3c76213698e11c0f74c28cda6",gasLimit:75e6,gasPrice:1e9,gasUsed:75e6,miniBlockHash:"bde6aad6833c6eac6a58f26fbed6ae3960615b900ba41cf634138eb3c4902a1e",nonce:449,receiver:"erd1qqqqqqqqqqqqqpgqp699jngundfqw07d8jzkepucvpzush6k3wvqyc44rx",receiverShard:0,round:1813482,sender:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",senderShard:0,signature:"414b703c1e998daacf0d7ddf809b88ccd9ded2e673dc4ff978ec1c01f59a2457ddef21196ae7c566e2268be4fd0cb09be876cca938352f7acd6793b377c99e0f",status:"success",value:"0",fee:"808410000000000",timestamp:1664202492,data:"dW5Cb25k",function:"unBond",action:{category:"scCall",name:"unBond"},results:[{hash:"62ad6e43a172cfc39ae1620e61a0e200cb5cd51e9cf1675a19cc4963971fdc65",timestamp:1664202492,nonce:0,gasLimit:0,gasPrice:1e9,value:"20000000000000000000",sender:"erd1qqqqqqqqqqqqqpgqp699jngundfqw07d8jzkepucvpzush6k3wvqyc44rx",receiver:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",data:"ZGVsZWdhdGlvbiBzdGFrZSB1bmJvbmQ=",prevTxHash:"5bdfd566dcc2c19a8bd62caa138acd2662a9d5f3c76213698e11c0f74c28cda6",originalTxHash:"5bdfd566dcc2c19a8bd62caa138acd2662a9d5f3c76213698e11c0f74c28cda6",callType:"0"}],operations:[{id:"5bdfd566dcc2c19a8bd62caa138acd2662a9d5f3c76213698e11c0f74c28cda6",action:"transfer",type:"egld",sender:"erd1qqqqqqqqqqqqqpgqp699jngundfqw07d8jzkepucvpzush6k3wvqyc44rx",receiver:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",value:"20000000000000000000"},{id:"5bdfd566dcc2c19a8bd62caa138acd2662a9d5f3c76213698e11c0f74c28cda6",action:"writeLog",type:"log",sender:"erd1qqqqqqqqqqqqqpgqp699jngundfqw07d8jzkepucvpzush6k3wvqyc44rx",receiver:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",data:"@6f6b@01158e460913d00000",message:"@too much gas provided for processing: gas provided = 74941000, gas used = 6578498"}]}];var Et=(e=0)=>Nt({transaction:On[e],address:On[e].sender,explorerAddress:we.devnet.explorerAddress});r();r();var ce=T(require("react"));q();r();r();r();ue();r();var br=T(require("lodash.isequal")),pn=require("reselect"),u=(0,pn.createSelectorCreator)(pn.defaultMemoize,br.default);var Z=e=>e.account,mo=u(Z,e=>e.address),so=u(Z,mo,(e,n)=>n in e.accounts?e.accounts[n]:Ke),el=u(Z,so,(e,n)=>{let a=e,{accounts:o}=a,t=To(a,["accounts"]);return v(x({},t),{account:n})}),nl=u(so,e=>e.balance),ol=u(so,e=>{var n;return((n=e==null?void 0:e.nonce)==null?void 0:n.valueOf())||0}),tl=u(Z,e=>e.shard),rl=u(Z,e=>e.ledgerAccount),al=u(Z,e=>e.walletConnectAccount),il=u(Z,e=>e.isAccountLoading),cl=u(Z,e=>e.accountLoadingError),pl=u(Z,e=>e.websocketEvent);r();var fe=e=>e.networkConfig,dl=u(fe,e=>e.chainID),_l=u(fe,e=>e.network.walletConnectBridgeAddress),ll=u(fe,e=>e.network.walletConnectV2RelayAddress),gl=u(fe,e=>e.network.walletConnectV2ProjectId),ul=u(fe,e=>e.network.walletConnectDeepLink),_o=u(fe,e=>e.network),fl=u(_o,e=>e.apiAddress),xr=u(_o,e=>e.explorerAddress),yr=u(_o,e=>e.egldLabel);r();P();var he=e=>e.loginInfo,Tl=u(he,e=>e.loginMethod),wl=u(he,mo,(e,n)=>e.loginMethod!=""&&Boolean(n)),ql=u(he,e=>e.walletConnectLogin),kl=u(he,e=>e.ledgerLogin),vl=u(he,e=>e.walletLogin),Sl=u(he,e=>e.isLoginSessionInvalid);r();r();var I=require("@elrondnetwork/erdjs");q();function lo(e){var a,i,c;let{data:n}=e,o=Ge(n!=null?n:"")?I.TransactionPayload.fromEncoded(n):new I.TransactionPayload(n),t=new I.Transaction(x({value:e.value.valueOf(),data:o,nonce:e.nonce.valueOf(),receiver:new I.Address(e.receiver),sender:new I.Address(e.sender),gasLimit:(a=e.gasLimit.valueOf())!=null?a:Mn,gasPrice:(i=e.gasPrice.valueOf())!=null?i:In,chainID:e.chainID.valueOf(),version:new I.TransactionVersion((c=e.version)!=null?c:rt)},e.options?{options:new I.TransactionOptions(e.options)}:{}));return t.applySignature({hex:()=>e.signature||""},new I.Address(e.sender)),t}r();to();r();var Pi=T(require("bignumber.js"));ne();r();r();r();r();var zi=require("@elrondnetwork/erdjs/out");r();q();r();r();r();r();P();r();r();var fo=require("@reduxjs/toolkit"),Gr=require("react-redux/es/utils/Subscription");ue();r();var Tr=T(require("lodash.throttle"));q();ue();Zn();Qe();Se();var Ui=[Oe],Bi=(0,Tr.default)(()=>{Ne(Ae())},5e3),wr=e=>n=>o=>{var _;if(Ui.includes(o.type))return n(o);let t=e.getState(),a=ge.local.getItem(ae.loginExpiresAt);if(!((_=t==null?void 0:t.loginInfo)==null?void 0:_.isLoggedIn))return n(o);if(a==null)return Ne(Ae());let c=Date.now();return a-c<0?setTimeout(()=>qo(void 0,null,function*(){console.log("session expired");try{e.dispatch(Bt())}catch(g){console.error("error logging out",g)}}),1e3):(Bi(),n(o))};r();r();function mn(){return typeof sessionStorage!="undefined"}var Br=mn()?(Ir(),Fe(Mr)).default:(Rr(),Fe(Lr)).default,Or=mn()?(Fr(),Fe(Cr)).default:[],Vr=mn()?(Ur(),Fe(zr)).default:e=>e;sn();var xe=(0,fo.configureStore)({reducer:Br,middleware:e=>e({serializableCheck:{ignoredActions:[...Or,or.type,nr.type],ignoredPaths:["payload.nonce","account.account.nonce","providers.accountProvider"]}}).concat(wr)}),Og=(0,Gr.createSubscription)(xe),Vg=Vr(xe),Gg=(0,fo.configureStore)({reducer:be});r();r();r();r();r();N();r();N();r();var nc=T(require("bignumber.js"));q();N();r();N();r();N();r();var ho=T(require("bignumber.js")),oc=e=>{let n=new ho.default(e.gasPrice),o=new ho.default(e.gasUsed);return n.times(o).toString()};function Wr(e){return e.fee?e.fee:e.gasUsed===void 0?null:oc(e)}r();P();r();P();r();r();var tc=require("react");N();var Vu=["create","localMint","ESDTLocalMint","addQuantity","burn","localBurn","ESDTLocalBurn","wipe","writeLog","signalError"];r();var rc=T(require("anchorme"));r();var Zr=require("react");ne();r();r();N();r();var ac=require("@elrondnetwork/erdjs/out"),ic=T(require("bignumber.js"));N();r();N();r();r();P();var Tf=[{label:"Raw",value:"raw"},{label:"Text",value:"text"},{label:"Decimal",value:"decimal"},{label:"Smart",value:"smart"}];r();var Yr=require("react");ne();r();r();r();r();r();q();var _n=e=>e.transactions,De=u(_n,e=>e.signedTransactions),l0=u(_n,e=>e.signTransactionsError),g0=u(_n,e=>e.signTransactionsCancelMessage),ln=e=>n=>Object.entries(n).reduce((o,[t,a])=>(e(a.status)&&(o[t]=a),o),{}),u0=u(De,ln(rr)),f0=u(De,ln(Je)),h0=u(De,ln(en)),b0=u(De,ln(ar)),x0=u(_n,e=>{var n;return(e==null?void 0:e.transactionsToSign)==null?null:v(x({},e.transactionsToSign),{transactions:((n=e==null?void 0:e.transactionsToSign)==null?void 0:n.transactions.map(o=>lo(o)))||[]})}),y0=u(De,(e,n)=>n,(e,n)=>n!=null?(e==null?void 0:e[n])||{}:{});r();ue();var $r={errorMessage:ao,successMessage:io,processingMessage:co},pc=e=>e.transactionsInfo,k0=u(pc,(e,n)=>n,(e,n)=>n!=null&&(e==null?void 0:e[Number(n)])||$r);r();var Qr=e=>e.modals,A0=u(Qr,e=>e.txSubmittedModal),N0=u(Qr,e=>e.notificationModal);r();var bo=e=>e.toasts,M0=u(bo,e=>e.customToasts),I0=u(bo,e=>e.transactionToasts),L0=u(bo,e=>e.failTransactionToast);function Xr(){return yr(xe.getState())}r();var gn=T(require("react")),un=T(require("classnames"));r();var mc=`.dapp-core-component__detailItem-module__detail-item:last-child {
  border-bottom: 0 !important;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(mc));var Kr={"detail-item":"dapp-core-component__detailItem-module__detail-item",detailItem:"dapp-core-component__detailItem-module__detail-item"};r();var sc=`.dapp-core-component__main__link-style {
  color: #1392ff;
}

.dapp-core-component__main__link-second-style {
  color: #9dabbd;
}
.dapp-core-component__main__link-second-style:hover {
  color: #1392ff;
}
.dapp-core-component__main__link-second-style:hover svg {
  color: #1392ff;
}

.dapp-core-component__main__wrapper {
  min-height: 100vh;
}

.dapp-core-component__main__opacity-6 {
  opacity: 0.6;
}

.dapp-core-component__main__qr-code-svg-container {
  width: 15rem;
  height: 15rem;
}

.dapp-core-component__main__pair-list {
  max-height: 20rem;
  overflow-y: auto;
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__pair-image {
  width: 4rem;
  overflow: hidden;
  height: auto;
  min-width: 2rem;
}

.dapp-core-component__main__pair-remove {
  padding: 0 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

.dapp-core-component__main__dapp-icon {
  padding: 5px;
  border-radius: 50%;
  background-color: #fafafa;
  width: 4.6428571429rem;
  height: 4.6428571429rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}
.dapp-core-component__main__dapp-icon.dapp-core-component__main__icon-medium {
  width: 80px;
  height: 80px;
}

.dapp-core-component__main__centering {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dapp-core-component__main__border-n {
  border: none !important;
}

.dapp-core-component__main__t-shadow {
  text-shadow: 0px 4px 7px rgba(0, 0, 0, 0.3);
}

.dapp-core-component__main__btn {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.04) !important;
}

.dapp-core-component__main__media {
  display: flex;
  align-items: flex-start;
}

.dapp-core-component__main__media-body {
  flex: 1;
}

.dapp-core-component__main__connect-btns .dapp-core-component__main__btn {
  display: flex;
  height: 48px;
  justify-content: center;
  align-items: center;
  color: #fff !important;
  background-color: #1392ff;
  padding: 15px 20px 15px 16px;
  border-radius: 6px;
  gap: 8px;
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__btn:hover, .dapp-core-component__main__connect-btns .dapp-core-component__main__btn:active {
  background-color: #1392ff;
  background-image: linear-gradient(rgba(0, 0, 0, 0.1) 0 0);
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__info {
  color: #1392ff;
  margin-left: 48px;
}

.dapp-core-component__main__btn {
  transition: none !important;
}

.dapp-core-component__main__trust-badge {
  width: 20px;
  height: 20px;
}

/* Navbar
  -------------------------------------------------- */
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name {
  display: flex;
  align-items: center;
  font-size: 1.7142857143rem;
  font-weight: 400;
  color: #9ba5b4;
  margin-left: 1.6428571429rem;
}
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name:before {
  content: "";
  display: block;
  width: 1px;
  height: 3rem;
  margin-right: 1.7142857143rem;
}

/* Dashboard
  -------------------------------------------------- */
.dapp-core-component__main__action-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  margin: 0 7px;
}
.dapp-core-component__main__action-btn button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #fff;
}
.dapp-core-component__main__action-btn button:hover {
  opacity: 0.9;
}

.dapp-core-component__main__transactions .dapp-core-component__main__transaction-icon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table td {
  white-space: nowrap;
  vertical-align: middle;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table tr:last-of-type td {
  border-bottom: 1px solid #dee2e6;
}

/* Loader, Transaction Success/Fail
  -------------------------------------------------- */
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div {
  position: absolute;
  top: 27px;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.1);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.dapp-core-component__main__page-state .dapp-core-component__main__bg-blue .dapp-core-component__main__lds-ellipsis div {
  background: rgba(255, 255, 255, 0.4);
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(1) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis1 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(2) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(3) {
  left: 26px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(4) {
  left: 45px;
  animation: dapp-core-component__main__lds-ellipsis3 0.6s infinite;
}
@keyframes dapp-core-component__main__lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(19px, 0);
  }
}

.dapp-core-component__main__btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
}
.dapp-core-component__main__btn .dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn .dapp-core-component__main__disabled {
  opacity: 0.65;
  cursor: default;
}
.dapp-core-component__main__btn:hover {
  color: #212529;
  text-decoration: none;
}
.dapp-core-component__main__btn:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn:disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn-primary {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary:hover {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-primary:focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary:hover {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-secondary:focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-warning {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning:hover {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning:focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-dark {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark:hover {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
}
.dapp-core-component__main__btn-dark:focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-link {
  font-weight: 400;
  color: #007bff;
  text-decoration: none;
}
.dapp-core-component__main__btn-link:hover {
  color: #0056b3;
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:focus, .dapp-core-component__main__btn-link.dapp-core-component__main__focus {
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:disabled, .dapp-core-component__main__btn-link.dapp-core-component__main__disabled {
  color: #6c757d;
  pointer-events: none;
}

.dapp-core-component__main__btn-light {
  color: #212529;
  background-color: #fafafa;
  border-color: #fafafa;
}
.dapp-core-component__main__btn-light .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}
.dapp-core-component__main__btn-light .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #f8f9fa;
  border-color: #f8f9fa;
}
.dapp-core-component__main__btn-light:hover {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
}
.dapp-core-component__main__btn-light:focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .dapp-core-component__main__badge {
    transition: none;
  }
}
a.dapp-core-component__main__badge:hover, a.dapp-core-component__main__badge:focus {
  text-decoration: none;
}

.dapp-core-component__main__badge:empty {
  display: none;
}

.dapp-core-component__main__btn .dapp-core-component__main__badge {
  position: relative;
  top: -1px;
}

.dapp-core-component__main__badge-pill {
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.dapp-core-component__main__badge-primary {
  color: #fff;
  background-color: #007bff;
}

a.dapp-core-component__main__badge-primary:hover, a.dapp-core-component__main__badge-primary:focus {
  color: #fff;
  background-color: #0062cc;
}

a.dapp-core-component__main__badge-primary:focus, a.dapp-core-component__main__badge-primary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.5);
}

.dapp-core-component__main__badge-secondary {
  color: #fff;
  background-color: #6c757d;
}

a.dapp-core-component__main__badge-secondary:hover, a.dapp-core-component__main__badge-secondary:focus {
  color: #fff;
  background-color: #545b62;
}

a.dapp-core-component__main__badge-secondary:focus, a.dapp-core-component__main__badge-secondary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108, 117, 125, 0.5);
}

.dapp-core-component__main__badge-success {
  color: #fff;
  background-color: #28a745;
}

a.dapp-core-component__main__badge-success:hover, a.dapp-core-component__main__badge-success:focus {
  color: #fff;
  background-color: #1e7e34;
}

a.dapp-core-component__main__badge-success:focus, a.dapp-core-component__main__badge-success.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
}

.dapp-core-component__main__badge-info {
  color: #fff;
  background-color: #17a2b8;
}

a.dapp-core-component__main__badge-info:hover, a.dapp-core-component__main__badge-info:focus {
  color: #fff;
  background-color: #117a8b;
}

a.dapp-core-component__main__badge-info:focus, a.dapp-core-component__main__badge-info.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(23, 162, 184, 0.5);
}

.dapp-core-component__main__badge-warning {
  color: #212529;
  background-color: #ffc107;
}

a.dapp-core-component__main__badge-warning:hover, a.dapp-core-component__main__badge-warning:focus {
  color: #212529;
  background-color: #d39e00;
}

a.dapp-core-component__main__badge-warning:focus, a.dapp-core-component__main__badge-warning.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 193, 7, 0.5);
}

.dapp-core-component__main__badge-danger {
  color: #fff;
  background-color: #dc3545;
}

a.dapp-core-component__main__badge-danger:hover, a.dapp-core-component__main__badge-danger:focus {
  color: #fff;
  background-color: #bd2130;
}

a.dapp-core-component__main__badge-danger:focus, a.dapp-core-component__main__badge-danger.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(220, 53, 69, 0.5);
}

.dapp-core-component__main__badge-light {
  color: #212529;
  background-color: #f8f9fa;
}

a.dapp-core-component__main__badge-light:hover, a.dapp-core-component__main__badge-light:focus {
  color: #212529;
  background-color: #dae0e5;
}

a.dapp-core-component__main__badge-light:focus, a.dapp-core-component__main__badge-light.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(248, 249, 250, 0.5);
}

.dapp-core-component__main__badge-dark {
  color: #fff;
  background-color: #343a40;
}

a.dapp-core-component__main__badge-dark:hover, a.dapp-core-component__main__badge-dark:focus {
  color: #fff;
  background-color: #1d2124;
}

a.dapp-core-component__main__badge-dark:focus, a.dapp-core-component__main__badge-dark.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}

.dapp-core-component__main__card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 5px;
}
.dapp-core-component__main__card > hr {
  margin-right: 0;
  margin-left: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group {
  border-top: inherit;
  border-bottom: inherit;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:first-child {
  border-top-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:last-child {
  border-bottom-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__card-header + .dapp-core-component__main__list-group,
.dapp-core-component__main__card > .dapp-core-component__main__list-group + .dapp-core-component__main__card-footer {
  border-top: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header {
  border: 0;
  padding: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header:first-child {
  border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item {
  border-bottom: 1px solid #dee2e6;
  padding: 1.1rem 1rem;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item h6,
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item .dapp-core-component__main__h6 {
  line-height: 2.5rem;
  letter-spacing: unset;
  margin: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item.dapp-core-component__main__compact {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fafafa;
}

.dapp-core-component__main__card-body {
  flex: 1 1 auto;
  min-height: 1px;
  padding: 1.25rem;
}

.dapp-core-component__main__card-title {
  margin-bottom: 0.75rem;
}

.dapp-core-component__main__card-subtitle {
  margin-top: -0.375rem;
  margin-bottom: 0;
}

.dapp-core-component__main__card-text:last-child {
  margin-bottom: 0;
}

.dapp-core-component__main__card-link:hover {
  text-decoration: none;
}
.dapp-core-component__main__card-link + .dapp-core-component__main__card-link {
  margin-left: 1.25rem;
}

.dapp-core-component__main__card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: #fff;
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__card-footer {
  padding: 0.75rem 1.25rem;
  background-color: #fff;
  border-top: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__text {
  color: #9dabbd;
}

.dapp-core-component__main__text-white {
  color: #fff;
}

.dapp-core-component__main__text-monospace {
  font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-size-xl .dapp-core-component__main__trim {
  max-width: 13rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim {
  max-width: 10rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-only-sm .dapp-core-component__main__trim {
  max-width: none;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-left {
  text-align: left;
}

.dapp-core-component__main__text-right {
  text-align: right;
}

.dapp-core-component__main__text-center {
  text-align: center;
}

.dapp-core-component__main__text-primary {
  color: #1b46c2;
}

.dapp-core-component__main__text-secondary {
  color: #6c757d;
}

.dapp-core-component__main__text-primary-highlight {
  color: #1b46c2;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-success {
  color: #28a745;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-body {
  color: #212529;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-black-50 {
  color: rgba(0, 0, 0, 0.5);
}

.dapp-core-component__main__text-white-50 {
  color: rgba(255, 255, 255, 0.5);
}

@media (min-width: 992px) {
  .dapp-core-component__main__text-lg-right {
    text-align: right;
  }
}

.dapp-core-component__main__text-hide {
  font: 0/0 a;
  color: rgba(0, 0, 0, 0);
  text-shadow: none;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
}

.dapp-core-component__main__text-decoration-none {
  text-decoration: none;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__text-reset {
  color: inherit;
}

.dapp-core-component__main__text-lowercase {
  text-transform: lowercase;
}

.dapp-core-component__main__text-uppercase {
  text-transform: uppercase;
}

.dapp-core-component__main__text-capitalize {
  text-transform: capitalize;
}

.dapp-core-component__main__text-danger {
  color: #d9534f;
}

.dapp-core-component__main__text-warning {
  color: #f0ad4e;
}

.dapp-core-component__main__trim-fs-sm {
  color: inherit;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__m-0 {
  margin: 0;
}

.dapp-core-component__main__mt-0 {
  margin-top: 0;
}

.dapp-core-component__main__my-0 {
  margin-top: 0;
  margin-bottom: 0;
}

.dapp-core-component__main__mr-0 {
  margin-right: 0;
}

.dapp-core-component__main__mx-0 {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__mb-0 {
  margin-bottom: 0;
}

.dapp-core-component__main__my-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.dapp-core-component__main__ml-0 {
  margin-left: 0;
}

.dapp-core-component__main__mx-0 {
  margin-left: 0;
  margin-right: 0;
}

.dapp-core-component__main__m-1 {
  margin: 0.25rem;
}

.dapp-core-component__main__mt-1 {
  margin-top: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__mr-1 {
  margin-right: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.dapp-core-component__main__mb-1 {
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-bottom: 0.25rem;
  margin-top: 0.25rem;
}

.dapp-core-component__main__ml-1 {
  margin-left: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-right: 0.25rem;
  margin-left: 0.25rem;
}

.dapp-core-component__main__m-2 {
  margin: 0.5rem;
}

.dapp-core-component__main__mt-2 {
  margin-top: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-bottom: 0.5rem;
  margin-top: 0.5rem;
}

.dapp-core-component__main__mr-2 {
  margin-right: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__mb-2 {
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__ml-2 {
  margin-left: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__m-3 {
  margin: 1rem;
}

.dapp-core-component__main__mt-3,
.dapp-core-component__main__my-3 {
  margin-top: 1rem;
}

.dapp-core-component__main__mr-3 {
  margin-right: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__mb-3 {
  margin-bottom: 1rem;
}

.dapp-core-component__main__my-3 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.dapp-core-component__main__ml-3 {
  margin-left: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__m-4 {
  margin: 1.5rem;
}

.dapp-core-component__main__mt-4 {
  margin-top: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__mr-4 {
  margin-right: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-right: 1.5rem;
  margin-left: 1.5rem;
}

.dapp-core-component__main__mb-4 {
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.dapp-core-component__main__ml-4 {
  margin-left: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}

.dapp-core-component__main__m-5 {
  margin: 3rem;
}

.dapp-core-component__main__mt-5 {
  margin-top: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__mr-5 {
  margin-right: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-right: 3rem;
  margin-left: 3rem;
}

.dapp-core-component__main__mb-5 {
  margin-bottom: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__ml-5 {
  margin-left: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-left: 3rem;
  margin-right: 3rem;
}

.dapp-core-component__main__p-0 {
  padding: 0;
}

.dapp-core-component__main__pt-0 {
  padding-top: 0;
}

.dapp-core-component__main__py-0 {
  padding-top: 0;
  padding-bottom: 0;
}

.dapp-core-component__main__pr-0 {
  padding-right: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__pb-0 {
  padding-bottom: 0;
}

.dapp-core-component__main__py-0 {
  padding-bottom: 0;
  padding-top: 0;
}

.dapp-core-component__main__pl-0 {
  padding-left: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__p-1 {
  padding: 0.25rem;
}

.dapp-core-component__main__pt-1 {
  padding-top: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pr-1 {
  padding-right: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-right: 0.25rem;
  padding-left: 0.25rem;
}

.dapp-core-component__main__pb-1 {
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pl-1 {
  padding-left: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

.dapp-core-component__main__p-2 {
  padding: 0.5rem;
}

.dapp-core-component__main__pt-2 {
  padding-top: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.dapp-core-component__main__pr-2 {
  padding-right: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__pb-2 {
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__pl-2 {
  padding-left: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__p-3 {
  padding: 1rem;
}

.dapp-core-component__main__pt-3 {
  padding-top: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.dapp-core-component__main__pr-3 {
  padding-right: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__pb-3 {
  padding-bottom: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.dapp-core-component__main__pl-3 {
  padding-left: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__p-4 {
  padding: 1.5rem;
}

.dapp-core-component__main__pt-4 {
  padding-top: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pr-4 {
  padding-right: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.dapp-core-component__main__pb-4 {
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pl-4 {
  padding-left: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
}

.dapp-core-component__main__p-5 {
  padding: 3rem;
}

.dapp-core-component__main__pt-5 {
  padding-top: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-bottom: 3rem;
  padding-top: 3rem;
}

.dapp-core-component__main__pr-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__pb-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.dapp-core-component__main__pl-5 {
  padding-left: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__m-n1 {
  margin: -0.25rem;
}

.dapp-core-component__main__mt-n1 {
  margin-top: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__mr-n1 {
  margin-right: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__mb-n1 {
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-bottom: -0.25rem;
  margin-top: -0.25rem;
}

.dapp-core-component__main__ml-n1 {
  margin-left: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__m-n2 {
  margin: -0.5rem;
}

.dapp-core-component__main__mt-n2 {
  margin-top: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-top: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mr-n2 {
  margin-right: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mb-n2 {
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-bottom: -0.5rem;
  margin-top: -0.5rem;
}

.dapp-core-component__main__ml-n2 {
  margin-left: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.dapp-core-component__main__m-n3 {
  margin: -1rem;
}

.dapp-core-component__main__mt-n3 {
  margin-top: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-top: -1rem;
  margin-bottom: -1rem;
}

.dapp-core-component__main__mr-n3 {
  margin-right: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-right: -1rem;
  margin-left: -1rem;
}

.dapp-core-component__main__mb-n3 {
  margin-bottom: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-bottom: -1rem;
  margin-top: -1rem;
}

.dapp-core-component__main__ml-n3 {
  margin-left: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-left: -1rem;
  margin-right: -1rem;
}

.dapp-core-component__main__m-n4 {
  margin: -1.5rem;
}

.dapp-core-component__main__mt-n4 {
  margin-top: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-top: -1.5rem;
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__mr-n4 {
  margin-right: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__mb-n4 {
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-bottom: -1.5rem;
  margin-top: -1.5rem;
}

.dapp-core-component__main__ml-n4 {
  margin-left: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__m-n5 {
  margin: -3rem;
}

.dapp-core-component__main__mt-n5 {
  margin-top: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-bottom: -3rem;
  margin-top: -3rem;
}

.dapp-core-component__main__mr-n5 {
  margin-right: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-left: -3rem;
  margin-right: -3rem;
}

.dapp-core-component__main__mb-n5 {
  margin-bottom: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-top: -3rem;
  margin-bottom: -3rem;
}

.dapp-core-component__main__ml-n5 {
  margin-left: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-right: -3rem;
  margin-left: -3rem;
}

.dapp-core-component__main__m-auto {
  margin: auto;
}

.dapp-core-component__main__mt-auto {
  margin-top: auto;
}

.dapp-core-component__main__my-auto {
  margin-bottom: auto;
  margin-top: auto;
}

.dapp-core-component__main__mr-auto {
  margin-right: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__mb-auto {
  margin-bottom: auto;
}

.dapp-core-component__main__my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.dapp-core-component__main__ml-auto {
  margin-left: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-1 {
    margin-right: 0.25rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-4 {
    margin-right: 1.5rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mx-lg-4 {
    margin-left: 1.5rem;
    margin-right: 1.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__mb-xl-2 {
    margin-bottom: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__pr-sm-2 {
    padding-right: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__px-sm-2 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__pr-xl-0,
.dapp-core-component__main__px-xl-0 {
    padding-right: 0;
  }
}

.dapp-core-component__main__col {
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

.dapp-core-component__main__col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__col-xl,
.dapp-core-component__main__col-xl-auto,
.dapp-core-component__main__col-xl-12,
.dapp-core-component__main__col-xl-11,
.dapp-core-component__main__col-xl-10,
.dapp-core-component__main__col-xl-9,
.dapp-core-component__main__col-xl-8,
.dapp-core-component__main__col-xl-7,
.dapp-core-component__main__col-xl-6,
.dapp-core-component__main__col-xl-5,
.dapp-core-component__main__col-xl-4,
.dapp-core-component__main__col-xl-3,
.dapp-core-component__main__col-xl-2,
.dapp-core-component__main__col-xl-1,
.dapp-core-component__main__col-lg,
.dapp-core-component__main__col-lg-auto,
.dapp-core-component__main__col-lg-12,
.dapp-core-component__main__col-lg-11,
.dapp-core-component__main__col-lg-10,
.dapp-core-component__main__col-lg-9,
.dapp-core-component__main__col-lg-8,
.dapp-core-component__main__col-lg-7,
.dapp-core-component__main__col-lg-6,
.dapp-core-component__main__col-lg-5,
.dapp-core-component__main__col-lg-4,
.dapp-core-component__main__col-lg-3,
.dapp-core-component__main__col-lg-2,
.dapp-core-component__main__col-lg-1,
.dapp-core-component__main__col-md,
.dapp-core-component__main__col-md-auto,
.dapp-core-component__main__col-md-12,
.dapp-core-component__main__col-md-11,
.dapp-core-component__main__col-md-10,
.dapp-core-component__main__col-md-9,
.dapp-core-component__main__col-md-8,
.dapp-core-component__main__col-md-7,
.dapp-core-component__main__col-md-6,
.dapp-core-component__main__col-md-5,
.dapp-core-component__main__col-md-4,
.dapp-core-component__main__col-md-3,
.dapp-core-component__main__col-md-2,
.dapp-core-component__main__col-md-1,
.dapp-core-component__main__col-sm,
.dapp-core-component__main__col-sm-auto,
.dapp-core-component__main__col-sm-12,
.dapp-core-component__main__col-sm-11,
.dapp-core-component__main__col-sm-10,
.dapp-core-component__main__col-sm-9,
.dapp-core-component__main__col-sm-8,
.dapp-core-component__main__col-sm-7,
.dapp-core-component__main__col-sm-6,
.dapp-core-component__main__col-sm-5,
.dapp-core-component__main__col-sm-4,
.dapp-core-component__main__col-sm-3,
.dapp-core-component__main__col-sm-2,
.dapp-core-component__main__col-sm-1,
.dapp-core-component__main__col,
.dapp-core-component__main__col-auto,
.dapp-core-component__main__col-12,
.dapp-core-component__main__col-11,
.dapp-core-component__main__col-10,
.dapp-core-component__main__col-9,
.dapp-core-component__main__col-8,
.dapp-core-component__main__col-7,
.dapp-core-component__main__col-6,
.dapp-core-component__main__col-5,
.dapp-core-component__main__col-4,
.dapp-core-component__main__col-3,
.dapp-core-component__main__col-2,
.dapp-core-component__main__col-1 {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
}

.dapp-core-component__main__col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

@media (min-width: 576px) {
  .dapp-core-component__main__col-sm-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-sm-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-sm-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-sm-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-sm-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-sm-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-sm-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-sm-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-sm-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-sm-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-sm-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-sm-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 768px) {
  .dapp-core-component__main__col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__col-xl-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-xl-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-xl-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-xl-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-xl-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-xl-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-xl-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-xl-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-xl-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-xl-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-xl-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-xl-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
.dapp-core-component__main__flex-row {
  flex-direction: row;
}

.dapp-core-component__main__flex-column {
  flex-direction: column;
}

.dapp-core-component__main__flex-row-reverse {
  flex-direction: row-reverse;
}

.dapp-core-component__main__flex-column-reverse {
  flex-direction: column-reverse;
}

.dapp-core-component__main__flex-wrap {
  flex-wrap: wrap;
}

.dapp-core-component__main__flex-nowrap {
  flex-wrap: nowrap;
}

.dapp-core-component__main__flex-wrap-reverse {
  flex-wrap: wrap-reverse;
}

.dapp-core-component__main__flex-fill {
  flex: 1 1 auto;
}

.dapp-core-component__main__flex-grow-0 {
  flex-grow: 0;
}

.dapp-core-component__main__flex-grow-1 {
  flex-grow: 1;
}

.dapp-core-component__main__flex-shrink-0 {
  flex-shrink: 0;
}

.dapp-core-component__main__flex-shrink-1 {
  flex-shrink: 1;
}

.dapp-core-component__main__justify-content-start {
  justify-content: flex-start;
}

.dapp-core-component__main__justify-content-end {
  justify-content: flex-end;
}

.dapp-core-component__main__justify-content-center {
  justify-content: center;
}

.dapp-core-component__main__justify-content-between {
  justify-content: space-between;
}

.dapp-core-component__main__justify-content-around {
  justify-content: space-around;
}

.dapp-core-component__main__align-items-start {
  align-items: flex-start;
}

.dapp-core-component__main__align-items-end {
  align-items: flex-end;
}

.dapp-core-component__main__align-items-center {
  align-items: center;
}

.dapp-core-component__main__align-items-baseline {
  align-items: baseline;
}

.dapp-core-component__main__align-items-stretch {
  align-items: stretch;
}

.dapp-core-component__main__align-content-start {
  align-content: flex-start;
}

.dapp-core-component__main__align-content-end {
  align-content: flex-end;
}

.dapp-core-component__main__align-content-center {
  align-content: center;
}

.dapp-core-component__main__align-content-between {
  align-content: space-between;
}

.dapp-core-component__main__align-content-around {
  align-content: space-around;
}

.dapp-core-component__main__align-content-stretch {
  align-content: stretch;
}

.dapp-core-component__main__align-self-auto {
  align-self: auto;
}

.dapp-core-component__main__align-self-start {
  align-self: flex-start;
}

.dapp-core-component__main__align-self-end {
  align-self: flex-end;
}

.dapp-core-component__main__align-self-center {
  align-self: center;
}

.dapp-core-component__main__align-self-baseline {
  align-self: baseline;
}

.dapp-core-component__main__align-self-stretch {
  align-self: stretch;
}

@media (min-width: 576px) {
  .dapp-core-component__main__flex-sm-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-wrap {
    flex-wrap: wrap;
  }
}

.dapp-core-component__main__form-check {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: auto;
  padding-left: 0;
}

.dapp-core-component__main__form-check-label {
  margin-bottom: 0;
}

.dapp-core-component__main__form-check-input {
  position: absolute;
  margin-top: 0.3rem;
  margin-left: -1.25rem;
}
.dapp-core-component__main__form-check-input[disabled] ~ .dapp-core-component__main__form-check-label, .dapp-core-component__main__form-check-input:disabled ~ .dapp-core-component__main__form-check-label {
  color: #6c757d;
}

.dapp-core-component__main__form-group {
  display: flex;
  flex: 0 0 auto;
  flex-flow: row wrap;
  align-items: center;
  margin-bottom: 0;
}

.dapp-core-component__main__form-control {
  display: block;
  width: 100%;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.dapp-core-component__main__form-label {
  padding-top: calc(0.375rem + 1px);
  padding-bottom: calc(0.375rem + 1px);
  margin-bottom: 0;
  font-size: inherit;
  line-height: 1.5;
}

input,
button,
select,
optgroup,
textarea {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.dapp-core-component__main__form-group {
  margin-bottom: 1rem;
}

.dapp-core-component__main__form-text {
  display: block;
  margin-top: 0.25rem;
  color: inherit;
}

.dapp-core-component__main__border {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__border-top {
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__border-right {
  border-right: 1px solid #dee2e6;
}

.dapp-core-component__main__border-bottom {
  border-bottom: 1px solid #dee2e6;
}

.dapp-core-component__main__border-left {
  border-left: 1px solid #dee2e6;
}

.dapp-core-component__main__border-0 {
  border: 0;
}

.dapp-core-component__main__border-top-0 {
  border-top: 0;
}

.dapp-core-component__main__border-right-0 {
  border-right: 0;
}

.dapp-core-component__main__border-bottom-0 {
  border-bottom: 0;
}

.dapp-core-component__main__border-left-0 {
  border-left: 0;
}

.dapp-core-component__main__rounded-sm {
  border-radius: 0.2rem;
}

.dapp-core-component__main__rounded {
  border-radius: 0.25rem;
}

.dapp-core-component__main__rounded-top {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-right {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-bottom {
  border-bottom-right-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-left {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-lg {
  border-radius: 0.3rem;
}

.dapp-core-component__main__rounded-circle {
  border-radius: 50%;
}

.dapp-core-component__main__rounded-pill {
  border-radius: 50rem;
}

.dapp-core-component__main__rounded-0 {
  border-radius: 0;
}

.dapp-core-component__main__h1,
.dapp-core-component__main__h2,
.dapp-core-component__main__h3,
.dapp-core-component__main__h4,
.dapp-core-component__main__h5,
.dapp-core-component__main__h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2;
}

.dapp-core-component__main__h1 {
  font-size: 2.5rem;
}

.dapp-core-component__main__h2 {
  font-size: 2rem;
}

.dapp-core-component__main__h3 {
  font-size: 1.75rem;
}

.dapp-core-component__main__h4 {
  font-size: 1.5rem;
}

.dapp-core-component__main__h5 {
  font-size: 1.25rem;
}

.dapp-core-component__main__h6 {
  font-size: 1rem;
}

.dapp-core-component__main__lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

.dapp-core-component__main__bg-warning,
.dapp-core-component__main__warning {
  background-color: #f0ad4e;
}

.dapp-core-component__main__bg-danger,
.dapp-core-component__main__danger {
  background-color: #d9534f;
}

.dapp-core-component__main__bg-success,
.dapp-core-component__main__success {
  background-color: #5cb85c;
}

.dapp-core-component__main__position-relative {
  position: relative;
}

.dapp-core-component__main__position-absolute {
  position: absolute;
}

.dapp-core-component__main__w-25 {
  width: 25%;
}

.dapp-core-component__main__w-50 {
  width: 50%;
}

.dapp-core-component__main__w-75 {
  width: 75%;
}

.dapp-core-component__main__w-100 {
  width: 100%;
}

.dapp-core-component__main__w-auto {
  width: auto;
}

.dapp-core-component__main__h-25 {
  height: 25%;
}

.dapp-core-component__main__h-50 {
  height: 50%;
}

.dapp-core-component__main__h-75 {
  height: 75%;
}

.dapp-core-component__main__h-100 {
  height: 100%;
}

.dapp-core-component__main__h-auto {
  height: auto;
}

.dapp-core-component__main__mw-100 {
  max-width: 100%;
}

.dapp-core-component__main__mh-100 {
  max-height: 100%;
}

.dapp-core-component__main__min-vw-100 {
  min-width: 100vw;
}

.dapp-core-component__main__min-vh-100 {
  min-height: 100vh;
}

.dapp-core-component__main__vw-100 {
  width: 100vw;
}

.dapp-core-component__main__vh-100 {
  height: 100vh;
}

.dapp-core-component__main__locked-icon {
  color: inherit;
}

.dapp-core-component__main__shadow-sm {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
}

.dapp-core-component__main__shadow {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.dapp-core-component__main__shadow-lg {
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.175);
}

.dapp-core-component__main__shadow-none {
  box-shadow: none;
}

.dapp-core-component__main__d-none {
  display: none;
}

.dapp-core-component__main__d-inline {
  display: inline;
}

.dapp-core-component__main__d-inline-block {
  display: inline-block;
}

.dapp-core-component__main__d-block {
  display: block;
}

.dapp-core-component__main__d-table {
  display: table;
}

.dapp-core-component__main__d-table-row {
  display: table-row;
}

.dapp-core-component__main__d-table-cell {
  display: table-cell;
}

.dapp-core-component__main__d-flex {
  display: -ms-flexbox;
  display: flex;
}

.dapp-core-component__main__d-inline-flex {
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media (min-width: 576px) {
  .dapp-core-component__main__d-sm-flex {
    display: flex;
  }
}

.dapp-core-component__main__table {
  width: 100%;
  margin-bottom: 1rem;
  color: #212529;
}
.dapp-core-component__main__table th,
.dapp-core-component__main__table td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #dee2e6;
}
.dapp-core-component__main__table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #dee2e6;
}
.dapp-core-component__main__table tbody + tbody {
  border-top: 2px solid #dee2e6;
}

.dapp-core-component__main__table-sm th,
.dapp-core-component__main__table-sm td {
  padding: 0.3rem;
}

.dapp-core-component__main__table-bordered {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered th,
.dapp-core-component__main__table-bordered td {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered thead th,
.dapp-core-component__main__table-bordered thead td {
  border-bottom-width: 2px;
}

.dapp-core-component__main__table-borderless th,
.dapp-core-component__main__table-borderless td,
.dapp-core-component__main__table-borderless thead th,
.dapp-core-component__main__table-borderless tbody + tbody {
  border: 0;
}

.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(0, 0, 0, 0.05);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-dark th {
  color: #fff;
  background-color: #343a40;
  border-color: #454d55;
}
.dapp-core-component__main__table .dapp-core-component__main__thead-light th {
  color: #495057;
  background-color: #e9ecef;
  border-color: #dee2e6;
}

.dapp-core-component__main__table-dark {
  color: #fff;
  background-color: #343a40;
}
.dapp-core-component__main__table-dark th,
.dapp-core-component__main__table-dark td,
.dapp-core-component__main__table-dark thead th {
  border-color: #454d55;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-bordered {
  border: 0;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(255, 255, 255, 0.05);
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-hover tbody tr {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.075);
}

.dapp-core-component__main__table-responsive {
  display: block;
  width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}
.dapp-core-component__main__table-responsive > .dapp-core-component__main__table-bordered {
  border: 0;
}

@media (max-width: 767.98px) {
  .dapp-core-component__main__transactions.dapp-core-component__main__table .dapp-core-component__main__transaction-hash .dapp-core-component__main__trim {
    max-width: 8rem;
  }
}
.dapp-core-component__main__font-weight-light {
  font-weight: 300 !important;
}

.dapp-core-component__main__font-weight-lighter {
  font-weight: lighter !important;
}

.dapp-core-component__main__font-weight-normal {
  font-weight: 400 !important;
}

.dapp-core-component__main__font-weight-medium {
  font-weight: 500 !important;
}

.dapp-core-component__main__font-weight-bold {
  font-weight: 700 !important;
}

.dapp-core-component__main__font-weight-bolder {
  font-weight: bolder !important;
}

.dapp-core-component__main__font-italic {
  font-style: italic !important;
}

.dapp-core-component__main__icon-state {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
}
.dapp-core-component__main__icon-state.dapp-core-component__main__half {
  width: 3rem;
  height: 3rem;
}

.dapp-core-component__main__side-icon {
  width: 1.5rem;
  height: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  color: transparent;
  border-radius: 50%;
  overflow: hidden;
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__side-icon svg {
  color: #cacbcc;
}

div.dapp-core-component__main__side-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
div.dapp-core-component__main__side-icon img {
  transform: scale(0.75);
}

.dapp-core-component__main__toast-messages {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast {
  display: block;
  flex-basis: 4rem;
  width: 100%;
  max-width: 28.875rem;
  transition: transform 0.3s ease-in-out;
  transform: translateX(120%);
  background-color: #fff;
  border-radius: 0.25rem;
  border-color: #dee2e6;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__clickable {
  cursor: pointer;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible {
  margin-bottom: 0.3rem;
  transform: translateX(0);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress {
  background-color: rgba(0, 0, 0, 0);
  height: inherit;
  font-size: inherit;
  line-height: inherit;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar {
  background-color: rgba(108, 117, 125, 0.2);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar .dapp-core-component__main__content-height {
  z-index: -1;
  visibility: hidden;
  width: 0 !important;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast .dapp-core-component__main__tx-description {
  margin-bottom: 0.2rem;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close {
  opacity: 1;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close svg {
  color: #6c757d;
}

.dapp-core-component__main__data-field .dapp-core-component__main__textarea {
  -moz-appearance: textfield-multiline;
  -webkit-appearance: textarea;
  border: 1px solid #000;
  background-color: #fff;
  border-radius: 5px;
  font: medium -moz-fixed;
  font-size: 0.8125rem;
  min-height: 5rem;
  overflow: auto;
  padding: 0.375rem 0.75rem;
  resize: both;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea:hover {
  border: 1px solid #000;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea .dapp-core-component__main__highlighted {
  font-size: 1rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea.dapp-core-component__main__sc-call {
  min-height: 3rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__form-control:disabled,
.dapp-core-component__main__data-field .dapp-core-component__main__form-control[readonly] {
  background-color: #fafafa;
}
.dapp-core-component__main__data-field .dapp-core-component__main__small-font {
  font-size: 0.8125rem;
}

.dapp-core-component__main__progress-steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps hr {
  background-color: #6c757d;
  border-color: #dee2e6;
  z-index: 2;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot {
  height: 0.625rem;
  width: 0.625rem;
  border-radius: 50%;
  margin: 0 1rem;
  z-index: 5;
  background-color: #6c757d;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:first-of-type {
  margin-left: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:last-of-type {
  margin-right: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

a.dapp-core-component__main__side-link::after {
  right: 0;
  left: unset;
  width: calc(100% - 1.85rem);
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(sc));var C={"link-style":"dapp-core-component__main__link-style",linkStyle:"dapp-core-component__main__link-style","link-second-style":"dapp-core-component__main__link-second-style",linkSecondStyle:"dapp-core-component__main__link-second-style",wrapper:"dapp-core-component__main__wrapper","opacity-6":"dapp-core-component__main__opacity-6",opacity6:"dapp-core-component__main__opacity-6","qr-code-svg-container":"dapp-core-component__main__qr-code-svg-container",qrCodeSvgContainer:"dapp-core-component__main__qr-code-svg-container","pair-list":"dapp-core-component__main__pair-list",pairList:"dapp-core-component__main__pair-list","pair-image":"dapp-core-component__main__pair-image",pairImage:"dapp-core-component__main__pair-image","pair-remove":"dapp-core-component__main__pair-remove",pairRemove:"dapp-core-component__main__pair-remove","dapp-icon":"dapp-core-component__main__dapp-icon",dappIcon:"dapp-core-component__main__dapp-icon","icon-medium":"dapp-core-component__main__icon-medium",iconMedium:"dapp-core-component__main__icon-medium",centering:"dapp-core-component__main__centering","border-n":"dapp-core-component__main__border-n",borderN:"dapp-core-component__main__border-n","t-shadow":"dapp-core-component__main__t-shadow",tShadow:"dapp-core-component__main__t-shadow",btn:"dapp-core-component__main__btn",media:"dapp-core-component__main__media","media-body":"dapp-core-component__main__media-body",mediaBody:"dapp-core-component__main__media-body","connect-btns":"dapp-core-component__main__connect-btns",connectBtns:"dapp-core-component__main__connect-btns",info:"dapp-core-component__main__info","trust-badge":"dapp-core-component__main__trust-badge",trustBadge:"dapp-core-component__main__trust-badge",navbar:"dapp-core-component__main__navbar","dapp-name":"dapp-core-component__main__dapp-name",dappName:"dapp-core-component__main__dapp-name","action-btn":"dapp-core-component__main__action-btn",actionBtn:"dapp-core-component__main__action-btn",transactions:"dapp-core-component__main__transactions","transaction-icon":"dapp-core-component__main__transaction-icon",transactionIcon:"dapp-core-component__main__transaction-icon",table:"dapp-core-component__main__table","page-state":"dapp-core-component__main__page-state",pageState:"dapp-core-component__main__page-state","lds-ellipsis":"dapp-core-component__main__lds-ellipsis",ldsEllipsis:"dapp-core-component__main__lds-ellipsis","bg-blue":"dapp-core-component__main__bg-blue",bgBlue:"dapp-core-component__main__bg-blue","lds-ellipsis1":"dapp-core-component__main__lds-ellipsis1",ldsEllipsis1:"dapp-core-component__main__lds-ellipsis1","lds-ellipsis2":"dapp-core-component__main__lds-ellipsis2",ldsEllipsis2:"dapp-core-component__main__lds-ellipsis2","lds-ellipsis3":"dapp-core-component__main__lds-ellipsis3",ldsEllipsis3:"dapp-core-component__main__lds-ellipsis3",focus:"dapp-core-component__main__focus",disabled:"dapp-core-component__main__disabled","btn-primary":"dapp-core-component__main__btn-primary",btnPrimary:"dapp-core-component__main__btn-primary","btn-secondary":"dapp-core-component__main__btn-secondary",btnSecondary:"dapp-core-component__main__btn-secondary","btn-warning":"dapp-core-component__main__btn-warning",btnWarning:"dapp-core-component__main__btn-warning","btn-dark":"dapp-core-component__main__btn-dark",btnDark:"dapp-core-component__main__btn-dark","btn-link":"dapp-core-component__main__btn-link",btnLink:"dapp-core-component__main__btn-link","btn-light":"dapp-core-component__main__btn-light",btnLight:"dapp-core-component__main__btn-light",badge:"dapp-core-component__main__badge","badge-pill":"dapp-core-component__main__badge-pill",badgePill:"dapp-core-component__main__badge-pill","badge-primary":"dapp-core-component__main__badge-primary",badgePrimary:"dapp-core-component__main__badge-primary","badge-secondary":"dapp-core-component__main__badge-secondary",badgeSecondary:"dapp-core-component__main__badge-secondary","badge-success":"dapp-core-component__main__badge-success",badgeSuccess:"dapp-core-component__main__badge-success","badge-info":"dapp-core-component__main__badge-info",badgeInfo:"dapp-core-component__main__badge-info","badge-warning":"dapp-core-component__main__badge-warning",badgeWarning:"dapp-core-component__main__badge-warning","badge-danger":"dapp-core-component__main__badge-danger",badgeDanger:"dapp-core-component__main__badge-danger","badge-light":"dapp-core-component__main__badge-light",badgeLight:"dapp-core-component__main__badge-light","badge-dark":"dapp-core-component__main__badge-dark",badgeDark:"dapp-core-component__main__badge-dark",card:"dapp-core-component__main__card","list-group":"dapp-core-component__main__list-group",listGroup:"dapp-core-component__main__list-group","card-header":"dapp-core-component__main__card-header",cardHeader:"dapp-core-component__main__card-header","card-footer":"dapp-core-component__main__card-footer",cardFooter:"dapp-core-component__main__card-footer","card-header-item":"dapp-core-component__main__card-header-item",cardHeaderItem:"dapp-core-component__main__card-header-item",h6:"dapp-core-component__main__h6",compact:"dapp-core-component__main__compact","card-body":"dapp-core-component__main__card-body",cardBody:"dapp-core-component__main__card-body","card-title":"dapp-core-component__main__card-title",cardTitle:"dapp-core-component__main__card-title","card-subtitle":"dapp-core-component__main__card-subtitle",cardSubtitle:"dapp-core-component__main__card-subtitle","card-text":"dapp-core-component__main__card-text",cardText:"dapp-core-component__main__card-text","card-link":"dapp-core-component__main__card-link",cardLink:"dapp-core-component__main__card-link",text:"dapp-core-component__main__text","text-white":"dapp-core-component__main__text-white",textWhite:"dapp-core-component__main__text-white","text-monospace":"dapp-core-component__main__text-monospace",textMonospace:"dapp-core-component__main__text-monospace","trim-size-xl":"dapp-core-component__main__trim-size-xl",trimSizeXl:"dapp-core-component__main__trim-size-xl",trim:"dapp-core-component__main__trim","trim-only-sm":"dapp-core-component__main__trim-only-sm",trimOnlySm:"dapp-core-component__main__trim-only-sm","text-justify":"dapp-core-component__main__text-justify",textJustify:"dapp-core-component__main__text-justify","text-wrap":"dapp-core-component__main__text-wrap",textWrap:"dapp-core-component__main__text-wrap","text-nowrap":"dapp-core-component__main__text-nowrap",textNowrap:"dapp-core-component__main__text-nowrap","text-truncate":"dapp-core-component__main__text-truncate",textTruncate:"dapp-core-component__main__text-truncate","text-left":"dapp-core-component__main__text-left",textLeft:"dapp-core-component__main__text-left","text-right":"dapp-core-component__main__text-right",textRight:"dapp-core-component__main__text-right","text-center":"dapp-core-component__main__text-center",textCenter:"dapp-core-component__main__text-center","text-primary":"dapp-core-component__main__text-primary",textPrimary:"dapp-core-component__main__text-primary","text-secondary":"dapp-core-component__main__text-secondary",textSecondary:"dapp-core-component__main__text-secondary","text-primary-highlight":"dapp-core-component__main__text-primary-highlight",textPrimaryHighlight:"dapp-core-component__main__text-primary-highlight","text-muted":"dapp-core-component__main__text-muted",textMuted:"dapp-core-component__main__text-muted","text-success":"dapp-core-component__main__text-success",textSuccess:"dapp-core-component__main__text-success","text-body":"dapp-core-component__main__text-body",textBody:"dapp-core-component__main__text-body","text-black-50":"dapp-core-component__main__text-black-50",textBlack50:"dapp-core-component__main__text-black-50","text-white-50":"dapp-core-component__main__text-white-50",textWhite50:"dapp-core-component__main__text-white-50","text-lg-right":"dapp-core-component__main__text-lg-right",textLgRight:"dapp-core-component__main__text-lg-right","text-hide":"dapp-core-component__main__text-hide",textHide:"dapp-core-component__main__text-hide","text-decoration-none":"dapp-core-component__main__text-decoration-none",textDecorationNone:"dapp-core-component__main__text-decoration-none","text-break":"dapp-core-component__main__text-break",textBreak:"dapp-core-component__main__text-break","text-reset":"dapp-core-component__main__text-reset",textReset:"dapp-core-component__main__text-reset","text-lowercase":"dapp-core-component__main__text-lowercase",textLowercase:"dapp-core-component__main__text-lowercase","text-uppercase":"dapp-core-component__main__text-uppercase",textUppercase:"dapp-core-component__main__text-uppercase","text-capitalize":"dapp-core-component__main__text-capitalize",textCapitalize:"dapp-core-component__main__text-capitalize","text-danger":"dapp-core-component__main__text-danger",textDanger:"dapp-core-component__main__text-danger","text-warning":"dapp-core-component__main__text-warning",textWarning:"dapp-core-component__main__text-warning","trim-fs-sm":"dapp-core-component__main__trim-fs-sm",trimFsSm:"dapp-core-component__main__trim-fs-sm","mx-auto":"dapp-core-component__main__mx-auto",mxAuto:"dapp-core-component__main__mx-auto","m-0":"dapp-core-component__main__m-0",m0:"dapp-core-component__main__m-0","mt-0":"dapp-core-component__main__mt-0",mt0:"dapp-core-component__main__mt-0","my-0":"dapp-core-component__main__my-0",my0:"dapp-core-component__main__my-0","mr-0":"dapp-core-component__main__mr-0",mr0:"dapp-core-component__main__mr-0","mx-0":"dapp-core-component__main__mx-0",mx0:"dapp-core-component__main__mx-0","mb-0":"dapp-core-component__main__mb-0",mb0:"dapp-core-component__main__mb-0","ml-0":"dapp-core-component__main__ml-0",ml0:"dapp-core-component__main__ml-0","m-1":"dapp-core-component__main__m-1",m1:"dapp-core-component__main__m-1","mt-1":"dapp-core-component__main__mt-1",mt1:"dapp-core-component__main__mt-1","my-1":"dapp-core-component__main__my-1",my1:"dapp-core-component__main__my-1","mr-1":"dapp-core-component__main__mr-1",mr1:"dapp-core-component__main__mr-1","mx-1":"dapp-core-component__main__mx-1",mx1:"dapp-core-component__main__mx-1","mb-1":"dapp-core-component__main__mb-1",mb1:"dapp-core-component__main__mb-1","ml-1":"dapp-core-component__main__ml-1",ml1:"dapp-core-component__main__ml-1","m-2":"dapp-core-component__main__m-2",m2:"dapp-core-component__main__m-2","mt-2":"dapp-core-component__main__mt-2",mt2:"dapp-core-component__main__mt-2","my-2":"dapp-core-component__main__my-2",my2:"dapp-core-component__main__my-2","mr-2":"dapp-core-component__main__mr-2",mr2:"dapp-core-component__main__mr-2","mx-2":"dapp-core-component__main__mx-2",mx2:"dapp-core-component__main__mx-2","mb-2":"dapp-core-component__main__mb-2",mb2:"dapp-core-component__main__mb-2","ml-2":"dapp-core-component__main__ml-2",ml2:"dapp-core-component__main__ml-2","m-3":"dapp-core-component__main__m-3",m3:"dapp-core-component__main__m-3","mt-3":"dapp-core-component__main__mt-3",mt3:"dapp-core-component__main__mt-3","my-3":"dapp-core-component__main__my-3",my3:"dapp-core-component__main__my-3","mr-3":"dapp-core-component__main__mr-3",mr3:"dapp-core-component__main__mr-3","mx-3":"dapp-core-component__main__mx-3",mx3:"dapp-core-component__main__mx-3","mb-3":"dapp-core-component__main__mb-3",mb3:"dapp-core-component__main__mb-3","ml-3":"dapp-core-component__main__ml-3",ml3:"dapp-core-component__main__ml-3","m-4":"dapp-core-component__main__m-4",m4:"dapp-core-component__main__m-4","mt-4":"dapp-core-component__main__mt-4",mt4:"dapp-core-component__main__mt-4","my-4":"dapp-core-component__main__my-4",my4:"dapp-core-component__main__my-4","mr-4":"dapp-core-component__main__mr-4",mr4:"dapp-core-component__main__mr-4","mx-4":"dapp-core-component__main__mx-4",mx4:"dapp-core-component__main__mx-4","mb-4":"dapp-core-component__main__mb-4",mb4:"dapp-core-component__main__mb-4","ml-4":"dapp-core-component__main__ml-4",ml4:"dapp-core-component__main__ml-4","m-5":"dapp-core-component__main__m-5",m5:"dapp-core-component__main__m-5","mt-5":"dapp-core-component__main__mt-5",mt5:"dapp-core-component__main__mt-5","my-5":"dapp-core-component__main__my-5",my5:"dapp-core-component__main__my-5","mr-5":"dapp-core-component__main__mr-5",mr5:"dapp-core-component__main__mr-5","mx-5":"dapp-core-component__main__mx-5",mx5:"dapp-core-component__main__mx-5","mb-5":"dapp-core-component__main__mb-5",mb5:"dapp-core-component__main__mb-5","ml-5":"dapp-core-component__main__ml-5",ml5:"dapp-core-component__main__ml-5","p-0":"dapp-core-component__main__p-0",p0:"dapp-core-component__main__p-0","pt-0":"dapp-core-component__main__pt-0",pt0:"dapp-core-component__main__pt-0","py-0":"dapp-core-component__main__py-0",py0:"dapp-core-component__main__py-0","pr-0":"dapp-core-component__main__pr-0",pr0:"dapp-core-component__main__pr-0","px-0":"dapp-core-component__main__px-0",px0:"dapp-core-component__main__px-0","pb-0":"dapp-core-component__main__pb-0",pb0:"dapp-core-component__main__pb-0","pl-0":"dapp-core-component__main__pl-0",pl0:"dapp-core-component__main__pl-0","p-1":"dapp-core-component__main__p-1",p1:"dapp-core-component__main__p-1","pt-1":"dapp-core-component__main__pt-1",pt1:"dapp-core-component__main__pt-1","py-1":"dapp-core-component__main__py-1",py1:"dapp-core-component__main__py-1","pr-1":"dapp-core-component__main__pr-1",pr1:"dapp-core-component__main__pr-1","px-1":"dapp-core-component__main__px-1",px1:"dapp-core-component__main__px-1","pb-1":"dapp-core-component__main__pb-1",pb1:"dapp-core-component__main__pb-1","pl-1":"dapp-core-component__main__pl-1",pl1:"dapp-core-component__main__pl-1","p-2":"dapp-core-component__main__p-2",p2:"dapp-core-component__main__p-2","pt-2":"dapp-core-component__main__pt-2",pt2:"dapp-core-component__main__pt-2","py-2":"dapp-core-component__main__py-2",py2:"dapp-core-component__main__py-2","pr-2":"dapp-core-component__main__pr-2",pr2:"dapp-core-component__main__pr-2","px-2":"dapp-core-component__main__px-2",px2:"dapp-core-component__main__px-2","pb-2":"dapp-core-component__main__pb-2",pb2:"dapp-core-component__main__pb-2","pl-2":"dapp-core-component__main__pl-2",pl2:"dapp-core-component__main__pl-2","p-3":"dapp-core-component__main__p-3",p3:"dapp-core-component__main__p-3","pt-3":"dapp-core-component__main__pt-3",pt3:"dapp-core-component__main__pt-3","py-3":"dapp-core-component__main__py-3",py3:"dapp-core-component__main__py-3","pr-3":"dapp-core-component__main__pr-3",pr3:"dapp-core-component__main__pr-3","px-3":"dapp-core-component__main__px-3",px3:"dapp-core-component__main__px-3","pb-3":"dapp-core-component__main__pb-3",pb3:"dapp-core-component__main__pb-3","pl-3":"dapp-core-component__main__pl-3",pl3:"dapp-core-component__main__pl-3","p-4":"dapp-core-component__main__p-4",p4:"dapp-core-component__main__p-4","pt-4":"dapp-core-component__main__pt-4",pt4:"dapp-core-component__main__pt-4","py-4":"dapp-core-component__main__py-4",py4:"dapp-core-component__main__py-4","pr-4":"dapp-core-component__main__pr-4",pr4:"dapp-core-component__main__pr-4","px-4":"dapp-core-component__main__px-4",px4:"dapp-core-component__main__px-4","pb-4":"dapp-core-component__main__pb-4",pb4:"dapp-core-component__main__pb-4","pl-4":"dapp-core-component__main__pl-4",pl4:"dapp-core-component__main__pl-4","p-5":"dapp-core-component__main__p-5",p5:"dapp-core-component__main__p-5","pt-5":"dapp-core-component__main__pt-5",pt5:"dapp-core-component__main__pt-5","py-5":"dapp-core-component__main__py-5",py5:"dapp-core-component__main__py-5","pr-5":"dapp-core-component__main__pr-5",pr5:"dapp-core-component__main__pr-5","px-5":"dapp-core-component__main__px-5",px5:"dapp-core-component__main__px-5","pb-5":"dapp-core-component__main__pb-5",pb5:"dapp-core-component__main__pb-5","pl-5":"dapp-core-component__main__pl-5",pl5:"dapp-core-component__main__pl-5","m-n1":"dapp-core-component__main__m-n1",mN1:"dapp-core-component__main__m-n1","mt-n1":"dapp-core-component__main__mt-n1",mtN1:"dapp-core-component__main__mt-n1","my-n1":"dapp-core-component__main__my-n1",myN1:"dapp-core-component__main__my-n1","mr-n1":"dapp-core-component__main__mr-n1",mrN1:"dapp-core-component__main__mr-n1","mx-n1":"dapp-core-component__main__mx-n1",mxN1:"dapp-core-component__main__mx-n1","mb-n1":"dapp-core-component__main__mb-n1",mbN1:"dapp-core-component__main__mb-n1","ml-n1":"dapp-core-component__main__ml-n1",mlN1:"dapp-core-component__main__ml-n1","m-n2":"dapp-core-component__main__m-n2",mN2:"dapp-core-component__main__m-n2","mt-n2":"dapp-core-component__main__mt-n2",mtN2:"dapp-core-component__main__mt-n2","my-n2":"dapp-core-component__main__my-n2",myN2:"dapp-core-component__main__my-n2","mr-n2":"dapp-core-component__main__mr-n2",mrN2:"dapp-core-component__main__mr-n2","mx-n2":"dapp-core-component__main__mx-n2",mxN2:"dapp-core-component__main__mx-n2","mb-n2":"dapp-core-component__main__mb-n2",mbN2:"dapp-core-component__main__mb-n2","ml-n2":"dapp-core-component__main__ml-n2",mlN2:"dapp-core-component__main__ml-n2","m-n3":"dapp-core-component__main__m-n3",mN3:"dapp-core-component__main__m-n3","mt-n3":"dapp-core-component__main__mt-n3",mtN3:"dapp-core-component__main__mt-n3","my-n3":"dapp-core-component__main__my-n3",myN3:"dapp-core-component__main__my-n3","mr-n3":"dapp-core-component__main__mr-n3",mrN3:"dapp-core-component__main__mr-n3","mx-n3":"dapp-core-component__main__mx-n3",mxN3:"dapp-core-component__main__mx-n3","mb-n3":"dapp-core-component__main__mb-n3",mbN3:"dapp-core-component__main__mb-n3","ml-n3":"dapp-core-component__main__ml-n3",mlN3:"dapp-core-component__main__ml-n3","m-n4":"dapp-core-component__main__m-n4",mN4:"dapp-core-component__main__m-n4","mt-n4":"dapp-core-component__main__mt-n4",mtN4:"dapp-core-component__main__mt-n4","my-n4":"dapp-core-component__main__my-n4",myN4:"dapp-core-component__main__my-n4","mr-n4":"dapp-core-component__main__mr-n4",mrN4:"dapp-core-component__main__mr-n4","mx-n4":"dapp-core-component__main__mx-n4",mxN4:"dapp-core-component__main__mx-n4","mb-n4":"dapp-core-component__main__mb-n4",mbN4:"dapp-core-component__main__mb-n4","ml-n4":"dapp-core-component__main__ml-n4",mlN4:"dapp-core-component__main__ml-n4","m-n5":"dapp-core-component__main__m-n5",mN5:"dapp-core-component__main__m-n5","mt-n5":"dapp-core-component__main__mt-n5",mtN5:"dapp-core-component__main__mt-n5","my-n5":"dapp-core-component__main__my-n5",myN5:"dapp-core-component__main__my-n5","mr-n5":"dapp-core-component__main__mr-n5",mrN5:"dapp-core-component__main__mr-n5","mx-n5":"dapp-core-component__main__mx-n5",mxN5:"dapp-core-component__main__mx-n5","mb-n5":"dapp-core-component__main__mb-n5",mbN5:"dapp-core-component__main__mb-n5","ml-n5":"dapp-core-component__main__ml-n5",mlN5:"dapp-core-component__main__ml-n5","m-auto":"dapp-core-component__main__m-auto",mAuto:"dapp-core-component__main__m-auto","mt-auto":"dapp-core-component__main__mt-auto",mtAuto:"dapp-core-component__main__mt-auto","my-auto":"dapp-core-component__main__my-auto",myAuto:"dapp-core-component__main__my-auto","mr-auto":"dapp-core-component__main__mr-auto",mrAuto:"dapp-core-component__main__mr-auto","mb-auto":"dapp-core-component__main__mb-auto",mbAuto:"dapp-core-component__main__mb-auto","ml-auto":"dapp-core-component__main__ml-auto",mlAuto:"dapp-core-component__main__ml-auto","mr-lg-1":"dapp-core-component__main__mr-lg-1",mrLg1:"dapp-core-component__main__mr-lg-1","mr-lg-4":"dapp-core-component__main__mr-lg-4",mrLg4:"dapp-core-component__main__mr-lg-4","mx-lg-4":"dapp-core-component__main__mx-lg-4",mxLg4:"dapp-core-component__main__mx-lg-4","mb-xl-2":"dapp-core-component__main__mb-xl-2",mbXl2:"dapp-core-component__main__mb-xl-2","pr-sm-2":"dapp-core-component__main__pr-sm-2",prSm2:"dapp-core-component__main__pr-sm-2","px-sm-2":"dapp-core-component__main__px-sm-2",pxSm2:"dapp-core-component__main__px-sm-2","pr-xl-0":"dapp-core-component__main__pr-xl-0",prXl0:"dapp-core-component__main__pr-xl-0","px-xl-0":"dapp-core-component__main__px-xl-0",pxXl0:"dapp-core-component__main__px-xl-0",col:"dapp-core-component__main__col",row:"dapp-core-component__main__row","col-xl":"dapp-core-component__main__col-xl",colXl:"dapp-core-component__main__col-xl","col-xl-auto":"dapp-core-component__main__col-xl-auto",colXlAuto:"dapp-core-component__main__col-xl-auto","col-xl-12":"dapp-core-component__main__col-xl-12",colXl12:"dapp-core-component__main__col-xl-12","col-xl-11":"dapp-core-component__main__col-xl-11",colXl11:"dapp-core-component__main__col-xl-11","col-xl-10":"dapp-core-component__main__col-xl-10",colXl10:"dapp-core-component__main__col-xl-10","col-xl-9":"dapp-core-component__main__col-xl-9",colXl9:"dapp-core-component__main__col-xl-9","col-xl-8":"dapp-core-component__main__col-xl-8",colXl8:"dapp-core-component__main__col-xl-8","col-xl-7":"dapp-core-component__main__col-xl-7",colXl7:"dapp-core-component__main__col-xl-7","col-xl-6":"dapp-core-component__main__col-xl-6",colXl6:"dapp-core-component__main__col-xl-6","col-xl-5":"dapp-core-component__main__col-xl-5",colXl5:"dapp-core-component__main__col-xl-5","col-xl-4":"dapp-core-component__main__col-xl-4",colXl4:"dapp-core-component__main__col-xl-4","col-xl-3":"dapp-core-component__main__col-xl-3",colXl3:"dapp-core-component__main__col-xl-3","col-xl-2":"dapp-core-component__main__col-xl-2",colXl2:"dapp-core-component__main__col-xl-2","col-xl-1":"dapp-core-component__main__col-xl-1",colXl1:"dapp-core-component__main__col-xl-1","col-lg":"dapp-core-component__main__col-lg",colLg:"dapp-core-component__main__col-lg","col-lg-auto":"dapp-core-component__main__col-lg-auto",colLgAuto:"dapp-core-component__main__col-lg-auto","col-lg-12":"dapp-core-component__main__col-lg-12",colLg12:"dapp-core-component__main__col-lg-12","col-lg-11":"dapp-core-component__main__col-lg-11",colLg11:"dapp-core-component__main__col-lg-11","col-lg-10":"dapp-core-component__main__col-lg-10",colLg10:"dapp-core-component__main__col-lg-10","col-lg-9":"dapp-core-component__main__col-lg-9",colLg9:"dapp-core-component__main__col-lg-9","col-lg-8":"dapp-core-component__main__col-lg-8",colLg8:"dapp-core-component__main__col-lg-8","col-lg-7":"dapp-core-component__main__col-lg-7",colLg7:"dapp-core-component__main__col-lg-7","col-lg-6":"dapp-core-component__main__col-lg-6",colLg6:"dapp-core-component__main__col-lg-6","col-lg-5":"dapp-core-component__main__col-lg-5",colLg5:"dapp-core-component__main__col-lg-5","col-lg-4":"dapp-core-component__main__col-lg-4",colLg4:"dapp-core-component__main__col-lg-4","col-lg-3":"dapp-core-component__main__col-lg-3",colLg3:"dapp-core-component__main__col-lg-3","col-lg-2":"dapp-core-component__main__col-lg-2",colLg2:"dapp-core-component__main__col-lg-2","col-lg-1":"dapp-core-component__main__col-lg-1",colLg1:"dapp-core-component__main__col-lg-1","col-md":"dapp-core-component__main__col-md",colMd:"dapp-core-component__main__col-md","col-md-auto":"dapp-core-component__main__col-md-auto",colMdAuto:"dapp-core-component__main__col-md-auto","col-md-12":"dapp-core-component__main__col-md-12",colMd12:"dapp-core-component__main__col-md-12","col-md-11":"dapp-core-component__main__col-md-11",colMd11:"dapp-core-component__main__col-md-11","col-md-10":"dapp-core-component__main__col-md-10",colMd10:"dapp-core-component__main__col-md-10","col-md-9":"dapp-core-component__main__col-md-9",colMd9:"dapp-core-component__main__col-md-9","col-md-8":"dapp-core-component__main__col-md-8",colMd8:"dapp-core-component__main__col-md-8","col-md-7":"dapp-core-component__main__col-md-7",colMd7:"dapp-core-component__main__col-md-7","col-md-6":"dapp-core-component__main__col-md-6",colMd6:"dapp-core-component__main__col-md-6","col-md-5":"dapp-core-component__main__col-md-5",colMd5:"dapp-core-component__main__col-md-5","col-md-4":"dapp-core-component__main__col-md-4",colMd4:"dapp-core-component__main__col-md-4","col-md-3":"dapp-core-component__main__col-md-3",colMd3:"dapp-core-component__main__col-md-3","col-md-2":"dapp-core-component__main__col-md-2",colMd2:"dapp-core-component__main__col-md-2","col-md-1":"dapp-core-component__main__col-md-1",colMd1:"dapp-core-component__main__col-md-1","col-sm":"dapp-core-component__main__col-sm",colSm:"dapp-core-component__main__col-sm","col-sm-auto":"dapp-core-component__main__col-sm-auto",colSmAuto:"dapp-core-component__main__col-sm-auto","col-sm-12":"dapp-core-component__main__col-sm-12",colSm12:"dapp-core-component__main__col-sm-12","col-sm-11":"dapp-core-component__main__col-sm-11",colSm11:"dapp-core-component__main__col-sm-11","col-sm-10":"dapp-core-component__main__col-sm-10",colSm10:"dapp-core-component__main__col-sm-10","col-sm-9":"dapp-core-component__main__col-sm-9",colSm9:"dapp-core-component__main__col-sm-9","col-sm-8":"dapp-core-component__main__col-sm-8",colSm8:"dapp-core-component__main__col-sm-8","col-sm-7":"dapp-core-component__main__col-sm-7",colSm7:"dapp-core-component__main__col-sm-7","col-sm-6":"dapp-core-component__main__col-sm-6",colSm6:"dapp-core-component__main__col-sm-6","col-sm-5":"dapp-core-component__main__col-sm-5",colSm5:"dapp-core-component__main__col-sm-5","col-sm-4":"dapp-core-component__main__col-sm-4",colSm4:"dapp-core-component__main__col-sm-4","col-sm-3":"dapp-core-component__main__col-sm-3",colSm3:"dapp-core-component__main__col-sm-3","col-sm-2":"dapp-core-component__main__col-sm-2",colSm2:"dapp-core-component__main__col-sm-2","col-sm-1":"dapp-core-component__main__col-sm-1",colSm1:"dapp-core-component__main__col-sm-1","col-auto":"dapp-core-component__main__col-auto",colAuto:"dapp-core-component__main__col-auto","col-12":"dapp-core-component__main__col-12",col12:"dapp-core-component__main__col-12","col-11":"dapp-core-component__main__col-11",col11:"dapp-core-component__main__col-11","col-10":"dapp-core-component__main__col-10",col10:"dapp-core-component__main__col-10","col-9":"dapp-core-component__main__col-9",col9:"dapp-core-component__main__col-9","col-8":"dapp-core-component__main__col-8",col8:"dapp-core-component__main__col-8","col-7":"dapp-core-component__main__col-7",col7:"dapp-core-component__main__col-7","col-6":"dapp-core-component__main__col-6",col6:"dapp-core-component__main__col-6","col-5":"dapp-core-component__main__col-5",col5:"dapp-core-component__main__col-5","col-4":"dapp-core-component__main__col-4",col4:"dapp-core-component__main__col-4","col-3":"dapp-core-component__main__col-3",col3:"dapp-core-component__main__col-3","col-2":"dapp-core-component__main__col-2",col2:"dapp-core-component__main__col-2","col-1":"dapp-core-component__main__col-1",col1:"dapp-core-component__main__col-1","flex-row":"dapp-core-component__main__flex-row",flexRow:"dapp-core-component__main__flex-row","flex-column":"dapp-core-component__main__flex-column",flexColumn:"dapp-core-component__main__flex-column","flex-row-reverse":"dapp-core-component__main__flex-row-reverse",flexRowReverse:"dapp-core-component__main__flex-row-reverse","flex-column-reverse":"dapp-core-component__main__flex-column-reverse",flexColumnReverse:"dapp-core-component__main__flex-column-reverse","flex-wrap":"dapp-core-component__main__flex-wrap",flexWrap:"dapp-core-component__main__flex-wrap","flex-nowrap":"dapp-core-component__main__flex-nowrap",flexNowrap:"dapp-core-component__main__flex-nowrap","flex-wrap-reverse":"dapp-core-component__main__flex-wrap-reverse",flexWrapReverse:"dapp-core-component__main__flex-wrap-reverse","flex-fill":"dapp-core-component__main__flex-fill",flexFill:"dapp-core-component__main__flex-fill","flex-grow-0":"dapp-core-component__main__flex-grow-0",flexGrow0:"dapp-core-component__main__flex-grow-0","flex-grow-1":"dapp-core-component__main__flex-grow-1",flexGrow1:"dapp-core-component__main__flex-grow-1","flex-shrink-0":"dapp-core-component__main__flex-shrink-0",flexShrink0:"dapp-core-component__main__flex-shrink-0","flex-shrink-1":"dapp-core-component__main__flex-shrink-1",flexShrink1:"dapp-core-component__main__flex-shrink-1","justify-content-start":"dapp-core-component__main__justify-content-start",justifyContentStart:"dapp-core-component__main__justify-content-start","justify-content-end":"dapp-core-component__main__justify-content-end",justifyContentEnd:"dapp-core-component__main__justify-content-end","justify-content-center":"dapp-core-component__main__justify-content-center",justifyContentCenter:"dapp-core-component__main__justify-content-center","justify-content-between":"dapp-core-component__main__justify-content-between",justifyContentBetween:"dapp-core-component__main__justify-content-between","justify-content-around":"dapp-core-component__main__justify-content-around",justifyContentAround:"dapp-core-component__main__justify-content-around","align-items-start":"dapp-core-component__main__align-items-start",alignItemsStart:"dapp-core-component__main__align-items-start","align-items-end":"dapp-core-component__main__align-items-end",alignItemsEnd:"dapp-core-component__main__align-items-end","align-items-center":"dapp-core-component__main__align-items-center",alignItemsCenter:"dapp-core-component__main__align-items-center","align-items-baseline":"dapp-core-component__main__align-items-baseline",alignItemsBaseline:"dapp-core-component__main__align-items-baseline","align-items-stretch":"dapp-core-component__main__align-items-stretch",alignItemsStretch:"dapp-core-component__main__align-items-stretch","align-content-start":"dapp-core-component__main__align-content-start",alignContentStart:"dapp-core-component__main__align-content-start","align-content-end":"dapp-core-component__main__align-content-end",alignContentEnd:"dapp-core-component__main__align-content-end","align-content-center":"dapp-core-component__main__align-content-center",alignContentCenter:"dapp-core-component__main__align-content-center","align-content-between":"dapp-core-component__main__align-content-between",alignContentBetween:"dapp-core-component__main__align-content-between","align-content-around":"dapp-core-component__main__align-content-around",alignContentAround:"dapp-core-component__main__align-content-around","align-content-stretch":"dapp-core-component__main__align-content-stretch",alignContentStretch:"dapp-core-component__main__align-content-stretch","align-self-auto":"dapp-core-component__main__align-self-auto",alignSelfAuto:"dapp-core-component__main__align-self-auto","align-self-start":"dapp-core-component__main__align-self-start",alignSelfStart:"dapp-core-component__main__align-self-start","align-self-end":"dapp-core-component__main__align-self-end",alignSelfEnd:"dapp-core-component__main__align-self-end","align-self-center":"dapp-core-component__main__align-self-center",alignSelfCenter:"dapp-core-component__main__align-self-center","align-self-baseline":"dapp-core-component__main__align-self-baseline",alignSelfBaseline:"dapp-core-component__main__align-self-baseline","align-self-stretch":"dapp-core-component__main__align-self-stretch",alignSelfStretch:"dapp-core-component__main__align-self-stretch","flex-sm-row":"dapp-core-component__main__flex-sm-row",flexSmRow:"dapp-core-component__main__flex-sm-row","flex-lg-row":"dapp-core-component__main__flex-lg-row",flexLgRow:"dapp-core-component__main__flex-lg-row","flex-lg-wrap":"dapp-core-component__main__flex-lg-wrap",flexLgWrap:"dapp-core-component__main__flex-lg-wrap","form-check":"dapp-core-component__main__form-check",formCheck:"dapp-core-component__main__form-check","form-check-label":"dapp-core-component__main__form-check-label",formCheckLabel:"dapp-core-component__main__form-check-label","form-check-input":"dapp-core-component__main__form-check-input",formCheckInput:"dapp-core-component__main__form-check-input","form-group":"dapp-core-component__main__form-group",formGroup:"dapp-core-component__main__form-group","form-control":"dapp-core-component__main__form-control",formControl:"dapp-core-component__main__form-control","form-label":"dapp-core-component__main__form-label",formLabel:"dapp-core-component__main__form-label","form-text":"dapp-core-component__main__form-text",formText:"dapp-core-component__main__form-text",border:"dapp-core-component__main__border","border-top":"dapp-core-component__main__border-top",borderTop:"dapp-core-component__main__border-top","border-right":"dapp-core-component__main__border-right",borderRight:"dapp-core-component__main__border-right","border-bottom":"dapp-core-component__main__border-bottom",borderBottom:"dapp-core-component__main__border-bottom","border-left":"dapp-core-component__main__border-left",borderLeft:"dapp-core-component__main__border-left","border-0":"dapp-core-component__main__border-0",border0:"dapp-core-component__main__border-0","border-top-0":"dapp-core-component__main__border-top-0",borderTop0:"dapp-core-component__main__border-top-0","border-right-0":"dapp-core-component__main__border-right-0",borderRight0:"dapp-core-component__main__border-right-0","border-bottom-0":"dapp-core-component__main__border-bottom-0",borderBottom0:"dapp-core-component__main__border-bottom-0","border-left-0":"dapp-core-component__main__border-left-0",borderLeft0:"dapp-core-component__main__border-left-0","rounded-sm":"dapp-core-component__main__rounded-sm",roundedSm:"dapp-core-component__main__rounded-sm",rounded:"dapp-core-component__main__rounded","rounded-top":"dapp-core-component__main__rounded-top",roundedTop:"dapp-core-component__main__rounded-top","rounded-right":"dapp-core-component__main__rounded-right",roundedRight:"dapp-core-component__main__rounded-right","rounded-bottom":"dapp-core-component__main__rounded-bottom",roundedBottom:"dapp-core-component__main__rounded-bottom","rounded-left":"dapp-core-component__main__rounded-left",roundedLeft:"dapp-core-component__main__rounded-left","rounded-lg":"dapp-core-component__main__rounded-lg",roundedLg:"dapp-core-component__main__rounded-lg","rounded-circle":"dapp-core-component__main__rounded-circle",roundedCircle:"dapp-core-component__main__rounded-circle","rounded-pill":"dapp-core-component__main__rounded-pill",roundedPill:"dapp-core-component__main__rounded-pill","rounded-0":"dapp-core-component__main__rounded-0",rounded0:"dapp-core-component__main__rounded-0",h1:"dapp-core-component__main__h1",h2:"dapp-core-component__main__h2",h3:"dapp-core-component__main__h3",h4:"dapp-core-component__main__h4",h5:"dapp-core-component__main__h5",lead:"dapp-core-component__main__lead","bg-primary":"dapp-core-component__main__bg-primary",bgPrimary:"dapp-core-component__main__bg-primary","bg-warning":"dapp-core-component__main__bg-warning",bgWarning:"dapp-core-component__main__bg-warning",warning:"dapp-core-component__main__warning","bg-danger":"dapp-core-component__main__bg-danger",bgDanger:"dapp-core-component__main__bg-danger",danger:"dapp-core-component__main__danger","bg-success":"dapp-core-component__main__bg-success",bgSuccess:"dapp-core-component__main__bg-success",success:"dapp-core-component__main__success","position-relative":"dapp-core-component__main__position-relative",positionRelative:"dapp-core-component__main__position-relative","position-absolute":"dapp-core-component__main__position-absolute",positionAbsolute:"dapp-core-component__main__position-absolute","w-25":"dapp-core-component__main__w-25",w25:"dapp-core-component__main__w-25","w-50":"dapp-core-component__main__w-50",w50:"dapp-core-component__main__w-50","w-75":"dapp-core-component__main__w-75",w75:"dapp-core-component__main__w-75","w-100":"dapp-core-component__main__w-100",w100:"dapp-core-component__main__w-100","w-auto":"dapp-core-component__main__w-auto",wAuto:"dapp-core-component__main__w-auto","h-25":"dapp-core-component__main__h-25",h25:"dapp-core-component__main__h-25","h-50":"dapp-core-component__main__h-50",h50:"dapp-core-component__main__h-50","h-75":"dapp-core-component__main__h-75",h75:"dapp-core-component__main__h-75","h-100":"dapp-core-component__main__h-100",h100:"dapp-core-component__main__h-100","h-auto":"dapp-core-component__main__h-auto",hAuto:"dapp-core-component__main__h-auto","mw-100":"dapp-core-component__main__mw-100",mw100:"dapp-core-component__main__mw-100","mh-100":"dapp-core-component__main__mh-100",mh100:"dapp-core-component__main__mh-100","min-vw-100":"dapp-core-component__main__min-vw-100",minVw100:"dapp-core-component__main__min-vw-100","min-vh-100":"dapp-core-component__main__min-vh-100",minVh100:"dapp-core-component__main__min-vh-100","vw-100":"dapp-core-component__main__vw-100",vw100:"dapp-core-component__main__vw-100","vh-100":"dapp-core-component__main__vh-100",vh100:"dapp-core-component__main__vh-100","locked-icon":"dapp-core-component__main__locked-icon",lockedIcon:"dapp-core-component__main__locked-icon","shadow-sm":"dapp-core-component__main__shadow-sm",shadowSm:"dapp-core-component__main__shadow-sm",shadow:"dapp-core-component__main__shadow","shadow-lg":"dapp-core-component__main__shadow-lg",shadowLg:"dapp-core-component__main__shadow-lg","shadow-none":"dapp-core-component__main__shadow-none",shadowNone:"dapp-core-component__main__shadow-none","d-none":"dapp-core-component__main__d-none",dNone:"dapp-core-component__main__d-none","d-inline":"dapp-core-component__main__d-inline",dInline:"dapp-core-component__main__d-inline","d-inline-block":"dapp-core-component__main__d-inline-block",dInlineBlock:"dapp-core-component__main__d-inline-block","d-block":"dapp-core-component__main__d-block",dBlock:"dapp-core-component__main__d-block","d-table":"dapp-core-component__main__d-table",dTable:"dapp-core-component__main__d-table","d-table-row":"dapp-core-component__main__d-table-row",dTableRow:"dapp-core-component__main__d-table-row","d-table-cell":"dapp-core-component__main__d-table-cell",dTableCell:"dapp-core-component__main__d-table-cell","d-flex":"dapp-core-component__main__d-flex",dFlex:"dapp-core-component__main__d-flex","d-inline-flex":"dapp-core-component__main__d-inline-flex",dInlineFlex:"dapp-core-component__main__d-inline-flex","d-sm-flex":"dapp-core-component__main__d-sm-flex",dSmFlex:"dapp-core-component__main__d-sm-flex","table-sm":"dapp-core-component__main__table-sm",tableSm:"dapp-core-component__main__table-sm","table-bordered":"dapp-core-component__main__table-bordered",tableBordered:"dapp-core-component__main__table-bordered","table-borderless":"dapp-core-component__main__table-borderless",tableBorderless:"dapp-core-component__main__table-borderless","table-striped":"dapp-core-component__main__table-striped",tableStriped:"dapp-core-component__main__table-striped","thead-dark":"dapp-core-component__main__thead-dark",theadDark:"dapp-core-component__main__thead-dark","thead-light":"dapp-core-component__main__thead-light",theadLight:"dapp-core-component__main__thead-light","table-dark":"dapp-core-component__main__table-dark",tableDark:"dapp-core-component__main__table-dark","table-hover":"dapp-core-component__main__table-hover",tableHover:"dapp-core-component__main__table-hover","table-responsive":"dapp-core-component__main__table-responsive",tableResponsive:"dapp-core-component__main__table-responsive","transaction-hash":"dapp-core-component__main__transaction-hash",transactionHash:"dapp-core-component__main__transaction-hash","font-weight-light":"dapp-core-component__main__font-weight-light",fontWeightLight:"dapp-core-component__main__font-weight-light","font-weight-lighter":"dapp-core-component__main__font-weight-lighter",fontWeightLighter:"dapp-core-component__main__font-weight-lighter","font-weight-normal":"dapp-core-component__main__font-weight-normal",fontWeightNormal:"dapp-core-component__main__font-weight-normal","font-weight-medium":"dapp-core-component__main__font-weight-medium",fontWeightMedium:"dapp-core-component__main__font-weight-medium","font-weight-bold":"dapp-core-component__main__font-weight-bold",fontWeightBold:"dapp-core-component__main__font-weight-bold","font-weight-bolder":"dapp-core-component__main__font-weight-bolder",fontWeightBolder:"dapp-core-component__main__font-weight-bolder","font-italic":"dapp-core-component__main__font-italic",fontItalic:"dapp-core-component__main__font-italic","icon-state":"dapp-core-component__main__icon-state",iconState:"dapp-core-component__main__icon-state",half:"dapp-core-component__main__half","side-icon":"dapp-core-component__main__side-icon",sideIcon:"dapp-core-component__main__side-icon","toast-messages":"dapp-core-component__main__toast-messages",toastMessages:"dapp-core-component__main__toast-messages",toast:"dapp-core-component__main__toast",clickable:"dapp-core-component__main__clickable","toast-visible":"dapp-core-component__main__toast-visible",toastVisible:"dapp-core-component__main__toast-visible",progress:"dapp-core-component__main__progress","progress-bar":"dapp-core-component__main__progress-bar",progressBar:"dapp-core-component__main__progress-bar","content-height":"dapp-core-component__main__content-height",contentHeight:"dapp-core-component__main__content-height","tx-description":"dapp-core-component__main__tx-description",txDescription:"dapp-core-component__main__tx-description",close:"dapp-core-component__main__close","data-field":"dapp-core-component__main__data-field",dataField:"dapp-core-component__main__data-field",textarea:"dapp-core-component__main__textarea",highlighted:"dapp-core-component__main__highlighted","sc-call":"dapp-core-component__main__sc-call",scCall:"dapp-core-component__main__sc-call","small-font":"dapp-core-component__main__small-font",smallFont:"dapp-core-component__main__small-font","progress-steps":"dapp-core-component__main__progress-steps",progressSteps:"dapp-core-component__main__progress-steps",steps:"dapp-core-component__main__steps",dot:"dapp-core-component__main__dot","side-link":"dapp-core-component__main__side-link",sideLink:"dapp-core-component__main__side-link"};var Jr=({children:e,title:n,className:o="",colWidth:t="2",noBorder:a=!1})=>gn.default.createElement("div",{className:(0,un.default)(C.row,Kr.detailItem,o,{[C.pt3]:a,[C.pb1]:a,[C.borderBottom]:!a,[C.py3]:!a})},gn.default.createElement("div",{className:(0,un.default)(C.textSecondary,C.textLgRight,C.fontWeightMedium,C[`colLg${t}`])},n),gn.default.createElement("div",{className:(0,un.default)(C.fontWeightMedium,C[`colLg${12-Number(t)}`])},e));r();var dc=`.dapp-core-component__styles__fee .dapp-core-component__styles__price {
  color: #6c757d;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(dc));var fn={fee:"dapp-core-component__styles__fee",price:"dapp-core-component__styles__price"};var ea=({transaction:e})=>{let n=Xr(),o=Wr(e),t=o&&ve(o)?re({input:o,showLastNonZeroDecimal:!0}):Ve,a=e.price!=null?At({amount:t,usd:e.price,decimals:4,addEqualSign:!0}):Ve,i=e.gasUsed!=null?ce.default.createElement(ce.default.Fragment,null,t," ",n," ",ce.default.createElement("span",{className:fn.price},"(",a,")")):ce.default.createElement("span",{className:fn.price},Ve);return ce.default.createElement(Jr,{title:"Transaction Fee",className:fn.fee},ce.default.createElement("span",{"data-testid":"transactionInfoFee"},i))};describe("TransactionInfoFee tests",()=>{it("should display fee",()=>{let n=(0,oa.render)(na.default.createElement(ea,{transaction:Et(0)})).getByTestId("transactionInfoFee");expect(n.textContent).toContain("0.000122 EGLD (\u2248 $0.0294)")})});
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
