// @elrondnetwork/dapp-core/__commonjs/UI/TransactionsToastList/components/TransactionToast/hooks/useTransactionToast.js
"use strict";var va=Object.create;var Ce=Object.defineProperty,ka=Object.defineProperties,Aa=Object.getOwnPropertyDescriptor,Ia=Object.getOwnPropertyDescriptors,Ea=Object.getOwnPropertyNames,He=Object.getOwnPropertySymbols,La=Object.getPrototypeOf,So=Object.prototype.hasOwnProperty,Rn=Object.prototype.propertyIsEnumerable;var Nn=(e,o,n)=>o in e?Ce(e,o,{enumerable:!0,configurable:!0,writable:!0,value:n}):e[o]=n,T=(e,o)=>{for(var n in o||(o={}))So.call(o,n)&&Nn(e,n,o[n]);if(He)for(var n of He(o))Rn.call(o,n)&&Nn(e,n,o[n]);return e},v=(e,o)=>ka(e,Ia(o));var vo=(e,o)=>{var n={};for(var r in e)So.call(e,r)&&o.indexOf(r)<0&&(n[r]=e[r]);if(e!=null&&He)for(var r of He(e))o.indexOf(r)<0&&Rn.call(e,r)&&(n[r]=e[r]);return n};var h=(e,o)=>()=>(e&&(o=e(e=0)),o);var je=(e,o)=>()=>(o||e((o={exports:{}}).exports,o),o.exports),ae=(e,o)=>{for(var n in o)Ce(e,n,{get:o[n],enumerable:!0})},Mn=(e,o,n,r)=>{if(o&&typeof o=="object"||typeof o=="function")for(let p of Ea(o))!So.call(e,p)&&p!==n&&Ce(e,p,{get:()=>o[p],enumerable:!(r=Aa(o,p))||r.enumerable});return e};var y=(e,o,n)=>(n=e!=null?va(La(e)):{},Mn(o||!e||!e.__esModule?Ce(n,"default",{value:e,enumerable:!0}):n,e)),Pe=e=>Mn(Ce({},"__esModule",{value:!0}),e);var b=(e,o,n)=>new Promise((r,p)=>{var s=d=>{try{l(n.next(d))}catch(g){p(g)}},m=d=>{try{l(n.throw(d))}catch(g){p(g)}},l=d=>d.done?r(d.value):Promise.resolve(d.value).then(s,m);l((n=n.apply(e,o)).next())});var Bn=je(ze=>{"use strict";t();ze.byteLength=Ca;ze.toByteArray=Na;ze.fromByteArray=Fa;var H=[],B=[],Da=typeof Uint8Array!="undefined"?Uint8Array:Array,ko="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";for(ie=0,Fn=ko.length;ie<Fn;++ie)H[ie]=ko[ie],B[ko.charCodeAt(ie)]=ie;var ie,Fn;B["-".charCodeAt(0)]=62;B["_".charCodeAt(0)]=63;function On(e){var o=e.length;if(o%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var n=e.indexOf("=");n===-1&&(n=o);var r=n===o?0:4-n%4;return[n,r]}function Ca(e){var o=On(e),n=o[0],r=o[1];return(n+r)*3/4-r}function Pa(e,o,n){return(o+n)*3/4-n}function Na(e){var o,n=On(e),r=n[0],p=n[1],s=new Da(Pa(e,r,p)),m=0,l=p>0?r-4:r,d;for(d=0;d<l;d+=4)o=B[e.charCodeAt(d)]<<18|B[e.charCodeAt(d+1)]<<12|B[e.charCodeAt(d+2)]<<6|B[e.charCodeAt(d+3)],s[m++]=o>>16&255,s[m++]=o>>8&255,s[m++]=o&255;return p===2&&(o=B[e.charCodeAt(d)]<<2|B[e.charCodeAt(d+1)]>>4,s[m++]=o&255),p===1&&(o=B[e.charCodeAt(d)]<<10|B[e.charCodeAt(d+1)]<<4|B[e.charCodeAt(d+2)]>>2,s[m++]=o>>8&255,s[m++]=o&255),s}function Ra(e){return H[e>>18&63]+H[e>>12&63]+H[e>>6&63]+H[e&63]}function Ma(e,o,n){for(var r,p=[],s=o;s<n;s+=3)r=(e[s]<<16&16711680)+(e[s+1]<<8&65280)+(e[s+2]&255),p.push(Ra(r));return p.join("")}function Fa(e){for(var o,n=e.length,r=n%3,p=[],s=16383,m=0,l=n-r;m<l;m+=s)p.push(Ma(e,m,m+s>l?l:m+s));return r===1?(o=e[n-1],p.push(H[o>>2]+H[o<<4&63]+"==")):r===2&&(o=(e[n-2]<<8)+e[n-1],p.push(H[o>>10]+H[o>>4&63]+H[o<<2&63]+"=")),p.join("")}});var Un=je(Ao=>{t();Ao.read=function(e,o,n,r,p){var s,m,l=p*8-r-1,d=(1<<l)-1,g=d>>1,x=-7,f=n?p-1:0,D=n?-1:1,C=e[o+f];for(f+=D,s=C&(1<<-x)-1,C>>=-x,x+=l;x>0;s=s*256+e[o+f],f+=D,x-=8);for(m=s&(1<<-x)-1,s>>=-x,x+=r;x>0;m=m*256+e[o+f],f+=D,x-=8);if(s===0)s=1-g;else{if(s===d)return m?NaN:(C?-1:1)*(1/0);m=m+Math.pow(2,r),s=s-g}return(C?-1:1)*m*Math.pow(2,s-r)};Ao.write=function(e,o,n,r,p,s){var m,l,d,g=s*8-p-1,x=(1<<g)-1,f=x>>1,D=p===23?Math.pow(2,-24)-Math.pow(2,-77):0,C=r?0:s-1,V=r?1:-1,G=o<0||o===0&&1/o<0?1:0;for(o=Math.abs(o),isNaN(o)||o===1/0?(l=isNaN(o)?1:0,m=x):(m=Math.floor(Math.log(o)/Math.LN2),o*(d=Math.pow(2,-m))<1&&(m--,d*=2),m+f>=1?o+=D/d:o+=D*Math.pow(2,1-f),o*d>=2&&(m++,d/=2),m+f>=x?(l=0,m=x):m+f>=1?(l=(o*d-1)*Math.pow(2,p),m=m+f):(l=o*Math.pow(2,f-1)*Math.pow(2,p),m=0));p>=8;e[n+C]=l&255,C+=V,l/=256,p-=8);for(m=m<<p|l,g+=p;g>0;e[n+C]=m&255,C+=V,m/=256,g-=8);e[n+C-V]|=G*128}});var Jn=je(he=>{"use strict";t();var Io=Bn(),fe=Un(),Wn=typeof Symbol=="function"&&typeof Symbol.for=="function"?Symbol.for("nodejs.util.inspect.custom"):null;he.Buffer=_;he.SlowBuffer=$a;he.INSPECT_MAX_BYTES=50;var qe=2147483647;he.kMaxLength=qe;_.TYPED_ARRAY_SUPPORT=Oa();!_.TYPED_ARRAY_SUPPORT&&typeof console!="undefined"&&typeof console.error=="function"&&console.error("This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.");function Oa(){try{var e=new Uint8Array(1),o={foo:function(){return 42}};return Object.setPrototypeOf(o,Uint8Array.prototype),Object.setPrototypeOf(e,o),e.foo()===42}catch(n){return!1}}Object.defineProperty(_.prototype,"parent",{enumerable:!0,get:function(){if(!!_.isBuffer(this))return this.buffer}});Object.defineProperty(_.prototype,"offset",{enumerable:!0,get:function(){if(!!_.isBuffer(this))return this.byteOffset}});function K(e){if(e>qe)throw new RangeError('The value "'+e+'" is invalid for option "size"');var o=new Uint8Array(e);return Object.setPrototypeOf(o,_.prototype),o}function _(e,o,n){if(typeof e=="number"){if(typeof o=="string")throw new TypeError('The "string" argument must be of type string. Received type number');return Co(e)}return Vn(e,o,n)}_.poolSize=8192;function Vn(e,o,n){if(typeof e=="string")return Ua(e,o);if(ArrayBuffer.isView(e))return Wa(e);if(e==null)throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e);if(j(e,ArrayBuffer)||e&&j(e.buffer,ArrayBuffer)||typeof SharedArrayBuffer!="undefined"&&(j(e,SharedArrayBuffer)||e&&j(e.buffer,SharedArrayBuffer)))return Lo(e,o,n);if(typeof e=="number")throw new TypeError('The "value" argument must not be of type number. Received type number');var r=e.valueOf&&e.valueOf();if(r!=null&&r!==e)return _.from(r,o,n);var p=Ga(e);if(p)return p;if(typeof Symbol!="undefined"&&Symbol.toPrimitive!=null&&typeof e[Symbol.toPrimitive]=="function")return _.from(e[Symbol.toPrimitive]("string"),o,n);throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e)}_.from=function(e,o,n){return Vn(e,o,n)};Object.setPrototypeOf(_.prototype,Uint8Array.prototype);Object.setPrototypeOf(_,Uint8Array);function Hn(e){if(typeof e!="number")throw new TypeError('"size" argument must be of type number');if(e<0)throw new RangeError('The value "'+e+'" is invalid for option "size"')}function Ba(e,o,n){return Hn(e),e<=0?K(e):o!==void 0?typeof n=="string"?K(e).fill(o,n):K(e).fill(o):K(e)}_.alloc=function(e,o,n){return Ba(e,o,n)};function Co(e){return Hn(e),K(e<0?0:Po(e)|0)}_.allocUnsafe=function(e){return Co(e)};_.allocUnsafeSlow=function(e){return Co(e)};function Ua(e,o){if((typeof o!="string"||o==="")&&(o="utf8"),!_.isEncoding(o))throw new TypeError("Unknown encoding: "+o);var n=jn(e,o)|0,r=K(n),p=r.write(e,o);return p!==n&&(r=r.slice(0,p)),r}function Eo(e){for(var o=e.length<0?0:Po(e.length)|0,n=K(o),r=0;r<o;r+=1)n[r]=e[r]&255;return n}function Wa(e){if(j(e,Uint8Array)){var o=new Uint8Array(e);return Lo(o.buffer,o.byteOffset,o.byteLength)}return Eo(e)}function Lo(e,o,n){if(o<0||e.byteLength<o)throw new RangeError('"offset" is outside of buffer bounds');if(e.byteLength<o+(n||0))throw new RangeError('"length" is outside of buffer bounds');var r;return o===void 0&&n===void 0?r=new Uint8Array(e):n===void 0?r=new Uint8Array(e,o):r=new Uint8Array(e,o,n),Object.setPrototypeOf(r,_.prototype),r}function Ga(e){if(_.isBuffer(e)){var o=Po(e.length)|0,n=K(o);return n.length===0||e.copy(n,0,0,o),n}if(e.length!==void 0)return typeof e.length!="number"||No(e.length)?K(0):Eo(e);if(e.type==="Buffer"&&Array.isArray(e.data))return Eo(e.data)}function Po(e){if(e>=qe)throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x"+qe.toString(16)+" bytes");return e|0}function $a(e){return+e!=e&&(e=0),_.alloc(+e)}_.isBuffer=function(o){return o!=null&&o._isBuffer===!0&&o!==_.prototype};_.compare=function(o,n){if(j(o,Uint8Array)&&(o=_.from(o,o.offset,o.byteLength)),j(n,Uint8Array)&&(n=_.from(n,n.offset,n.byteLength)),!_.isBuffer(o)||!_.isBuffer(n))throw new TypeError('The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array');if(o===n)return 0;for(var r=o.length,p=n.length,s=0,m=Math.min(r,p);s<m;++s)if(o[s]!==n[s]){r=o[s],p=n[s];break}return r<p?-1:p<r?1:0};_.isEncoding=function(o){switch(String(o).toLowerCase()){case"hex":case"utf8":case"utf-8":case"ascii":case"latin1":case"binary":case"base64":case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return!0;default:return!1}};_.concat=function(o,n){if(!Array.isArray(o))throw new TypeError('"list" argument must be an Array of Buffers');if(o.length===0)return _.alloc(0);var r;if(n===void 0)for(n=0,r=0;r<o.length;++r)n+=o[r].length;var p=_.allocUnsafe(n),s=0;for(r=0;r<o.length;++r){var m=o[r];if(j(m,Uint8Array))s+m.length>p.length?_.from(m).copy(p,s):Uint8Array.prototype.set.call(p,m,s);else if(_.isBuffer(m))m.copy(p,s);else throw new TypeError('"list" argument must be an Array of Buffers');s+=m.length}return p};function jn(e,o){if(_.isBuffer(e))return e.length;if(ArrayBuffer.isView(e)||j(e,ArrayBuffer))return e.byteLength;if(typeof e!="string")throw new TypeError('The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type '+typeof e);var n=e.length,r=arguments.length>2&&arguments[2]===!0;if(!r&&n===0)return 0;for(var p=!1;;)switch(o){case"ascii":case"latin1":case"binary":return n;case"utf8":case"utf-8":return Do(e).length;case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return n*2;case"hex":return n>>>1;case"base64":return Zn(e).length;default:if(p)return r?-1:Do(e).length;o=(""+o).toLowerCase(),p=!0}}_.byteLength=jn;function Va(e,o,n){var r=!1;if((o===void 0||o<0)&&(o=0),o>this.length||((n===void 0||n>this.length)&&(n=this.length),n<=0)||(n>>>=0,o>>>=0,n<=o))return"";for(e||(e="utf8");;)switch(e){case"hex":return Qa(this,o,n);case"utf8":case"utf-8":return qn(this,o,n);case"ascii":return Za(this,o,n);case"latin1":case"binary":return Ja(this,o,n);case"base64":return Xa(this,o,n);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return ei(this,o,n);default:if(r)throw new TypeError("Unknown encoding: "+e);e=(e+"").toLowerCase(),r=!0}}_.prototype._isBuffer=!0;function ce(e,o,n){var r=e[o];e[o]=e[n],e[n]=r}_.prototype.swap16=function(){var o=this.length;if(o%2!==0)throw new RangeError("Buffer size must be a multiple of 16-bits");for(var n=0;n<o;n+=2)ce(this,n,n+1);return this};_.prototype.swap32=function(){var o=this.length;if(o%4!==0)throw new RangeError("Buffer size must be a multiple of 32-bits");for(var n=0;n<o;n+=4)ce(this,n,n+3),ce(this,n+1,n+2);return this};_.prototype.swap64=function(){var o=this.length;if(o%8!==0)throw new RangeError("Buffer size must be a multiple of 64-bits");for(var n=0;n<o;n+=8)ce(this,n,n+7),ce(this,n+1,n+6),ce(this,n+2,n+5),ce(this,n+3,n+4);return this};_.prototype.toString=function(){var o=this.length;return o===0?"":arguments.length===0?qn(this,0,o):Va.apply(this,arguments)};_.prototype.toLocaleString=_.prototype.toString;_.prototype.equals=function(o){if(!_.isBuffer(o))throw new TypeError("Argument must be a Buffer");return this===o?!0:_.compare(this,o)===0};_.prototype.inspect=function(){var o="",n=he.INSPECT_MAX_BYTES;return o=this.toString("hex",0,n).replace(/(.{2})/g,"$1 ").trim(),this.length>n&&(o+=" ... "),"<Buffer "+o+">"};Wn&&(_.prototype[Wn]=_.prototype.inspect);_.prototype.compare=function(o,n,r,p,s){if(j(o,Uint8Array)&&(o=_.from(o,o.offset,o.byteLength)),!_.isBuffer(o))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+typeof o);if(n===void 0&&(n=0),r===void 0&&(r=o?o.length:0),p===void 0&&(p=0),s===void 0&&(s=this.length),n<0||r>o.length||p<0||s>this.length)throw new RangeError("out of range index");if(p>=s&&n>=r)return 0;if(p>=s)return-1;if(n>=r)return 1;if(n>>>=0,r>>>=0,p>>>=0,s>>>=0,this===o)return 0;for(var m=s-p,l=r-n,d=Math.min(m,l),g=this.slice(p,s),x=o.slice(n,r),f=0;f<d;++f)if(g[f]!==x[f]){m=g[f],l=x[f];break}return m<l?-1:l<m?1:0};function zn(e,o,n,r,p){if(e.length===0)return-1;if(typeof n=="string"?(r=n,n=0):n>2147483647?n=2147483647:n<-2147483648&&(n=-2147483648),n=+n,No(n)&&(n=p?0:e.length-1),n<0&&(n=e.length+n),n>=e.length){if(p)return-1;n=e.length-1}else if(n<0)if(p)n=0;else return-1;if(typeof o=="string"&&(o=_.from(o,r)),_.isBuffer(o))return o.length===0?-1:Gn(e,o,n,r,p);if(typeof o=="number")return o=o&255,typeof Uint8Array.prototype.indexOf=="function"?p?Uint8Array.prototype.indexOf.call(e,o,n):Uint8Array.prototype.lastIndexOf.call(e,o,n):Gn(e,[o],n,r,p);throw new TypeError("val must be string, number or Buffer")}function Gn(e,o,n,r,p){var s=1,m=e.length,l=o.length;if(r!==void 0&&(r=String(r).toLowerCase(),r==="ucs2"||r==="ucs-2"||r==="utf16le"||r==="utf-16le")){if(e.length<2||o.length<2)return-1;s=2,m/=2,l/=2,n/=2}function d(C,V){return s===1?C[V]:C.readUInt16BE(V*s)}var g;if(p){var x=-1;for(g=n;g<m;g++)if(d(e,g)===d(o,x===-1?0:g-x)){if(x===-1&&(x=g),g-x+1===l)return x*s}else x!==-1&&(g-=g-x),x=-1}else for(n+l>m&&(n=m-l),g=n;g>=0;g--){for(var f=!0,D=0;D<l;D++)if(d(e,g+D)!==d(o,D)){f=!1;break}if(f)return g}return-1}_.prototype.includes=function(o,n,r){return this.indexOf(o,n,r)!==-1};_.prototype.indexOf=function(o,n,r){return zn(this,o,n,r,!0)};_.prototype.lastIndexOf=function(o,n,r){return zn(this,o,n,r,!1)};function Ha(e,o,n,r){n=Number(n)||0;var p=e.length-n;r?(r=Number(r),r>p&&(r=p)):r=p;var s=o.length;r>s/2&&(r=s/2);for(var m=0;m<r;++m){var l=parseInt(o.substr(m*2,2),16);if(No(l))return m;e[n+m]=l}return m}function ja(e,o,n,r){return Ke(Do(o,e.length-n),e,n,r)}function za(e,o,n,r){return Ke(ti(o),e,n,r)}function qa(e,o,n,r){return Ke(Zn(o),e,n,r)}function Ka(e,o,n,r){return Ke(ri(o,e.length-n),e,n,r)}_.prototype.write=function(o,n,r,p){if(n===void 0)p="utf8",r=this.length,n=0;else if(r===void 0&&typeof n=="string")p=n,r=this.length,n=0;else if(isFinite(n))n=n>>>0,isFinite(r)?(r=r>>>0,p===void 0&&(p="utf8")):(p=r,r=void 0);else throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");var s=this.length-n;if((r===void 0||r>s)&&(r=s),o.length>0&&(r<0||n<0)||n>this.length)throw new RangeError("Attempt to write outside buffer bounds");p||(p="utf8");for(var m=!1;;)switch(p){case"hex":return Ha(this,o,n,r);case"utf8":case"utf-8":return ja(this,o,n,r);case"ascii":case"latin1":case"binary":return za(this,o,n,r);case"base64":return qa(this,o,n,r);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return Ka(this,o,n,r);default:if(m)throw new TypeError("Unknown encoding: "+p);p=(""+p).toLowerCase(),m=!0}};_.prototype.toJSON=function(){return{type:"Buffer",data:Array.prototype.slice.call(this._arr||this,0)}};function Xa(e,o,n){return o===0&&n===e.length?Io.fromByteArray(e):Io.fromByteArray(e.slice(o,n))}function qn(e,o,n){n=Math.min(e.length,n);for(var r=[],p=o;p<n;){var s=e[p],m=null,l=s>239?4:s>223?3:s>191?2:1;if(p+l<=n){var d,g,x,f;switch(l){case 1:s<128&&(m=s);break;case 2:d=e[p+1],(d&192)===128&&(f=(s&31)<<6|d&63,f>127&&(m=f));break;case 3:d=e[p+1],g=e[p+2],(d&192)===128&&(g&192)===128&&(f=(s&15)<<12|(d&63)<<6|g&63,f>2047&&(f<55296||f>57343)&&(m=f));break;case 4:d=e[p+1],g=e[p+2],x=e[p+3],(d&192)===128&&(g&192)===128&&(x&192)===128&&(f=(s&15)<<18|(d&63)<<12|(g&63)<<6|x&63,f>65535&&f<1114112&&(m=f))}}m===null?(m=65533,l=1):m>65535&&(m-=65536,r.push(m>>>10&1023|55296),m=56320|m&1023),r.push(m),p+=l}return Ya(r)}var $n=4096;function Ya(e){var o=e.length;if(o<=$n)return String.fromCharCode.apply(String,e);for(var n="",r=0;r<o;)n+=String.fromCharCode.apply(String,e.slice(r,r+=$n));return n}function Za(e,o,n){var r="";n=Math.min(e.length,n);for(var p=o;p<n;++p)r+=String.fromCharCode(e[p]&127);return r}function Ja(e,o,n){var r="";n=Math.min(e.length,n);for(var p=o;p<n;++p)r+=String.fromCharCode(e[p]);return r}function Qa(e,o,n){var r=e.length;(!o||o<0)&&(o=0),(!n||n<0||n>r)&&(n=r);for(var p="",s=o;s<n;++s)p+=ai[e[s]];return p}function ei(e,o,n){for(var r=e.slice(o,n),p="",s=0;s<r.length-1;s+=2)p+=String.fromCharCode(r[s]+r[s+1]*256);return p}_.prototype.slice=function(o,n){var r=this.length;o=~~o,n=n===void 0?r:~~n,o<0?(o+=r,o<0&&(o=0)):o>r&&(o=r),n<0?(n+=r,n<0&&(n=0)):n>r&&(n=r),n<o&&(n=o);var p=this.subarray(o,n);return Object.setPrototypeOf(p,_.prototype),p};function L(e,o,n){if(e%1!==0||e<0)throw new RangeError("offset is not uint");if(e+o>n)throw new RangeError("Trying to access beyond buffer length")}_.prototype.readUintLE=_.prototype.readUIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||L(o,n,this.length);for(var p=this[o],s=1,m=0;++m<n&&(s*=256);)p+=this[o+m]*s;return p};_.prototype.readUintBE=_.prototype.readUIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||L(o,n,this.length);for(var p=this[o+--n],s=1;n>0&&(s*=256);)p+=this[o+--n]*s;return p};_.prototype.readUint8=_.prototype.readUInt8=function(o,n){return o=o>>>0,n||L(o,1,this.length),this[o]};_.prototype.readUint16LE=_.prototype.readUInt16LE=function(o,n){return o=o>>>0,n||L(o,2,this.length),this[o]|this[o+1]<<8};_.prototype.readUint16BE=_.prototype.readUInt16BE=function(o,n){return o=o>>>0,n||L(o,2,this.length),this[o]<<8|this[o+1]};_.prototype.readUint32LE=_.prototype.readUInt32LE=function(o,n){return o=o>>>0,n||L(o,4,this.length),(this[o]|this[o+1]<<8|this[o+2]<<16)+this[o+3]*16777216};_.prototype.readUint32BE=_.prototype.readUInt32BE=function(o,n){return o=o>>>0,n||L(o,4,this.length),this[o]*16777216+(this[o+1]<<16|this[o+2]<<8|this[o+3])};_.prototype.readIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||L(o,n,this.length);for(var p=this[o],s=1,m=0;++m<n&&(s*=256);)p+=this[o+m]*s;return s*=128,p>=s&&(p-=Math.pow(2,8*n)),p};_.prototype.readIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||L(o,n,this.length);for(var p=n,s=1,m=this[o+--p];p>0&&(s*=256);)m+=this[o+--p]*s;return s*=128,m>=s&&(m-=Math.pow(2,8*n)),m};_.prototype.readInt8=function(o,n){return o=o>>>0,n||L(o,1,this.length),this[o]&128?(255-this[o]+1)*-1:this[o]};_.prototype.readInt16LE=function(o,n){o=o>>>0,n||L(o,2,this.length);var r=this[o]|this[o+1]<<8;return r&32768?r|4294901760:r};_.prototype.readInt16BE=function(o,n){o=o>>>0,n||L(o,2,this.length);var r=this[o+1]|this[o]<<8;return r&32768?r|4294901760:r};_.prototype.readInt32LE=function(o,n){return o=o>>>0,n||L(o,4,this.length),this[o]|this[o+1]<<8|this[o+2]<<16|this[o+3]<<24};_.prototype.readInt32BE=function(o,n){return o=o>>>0,n||L(o,4,this.length),this[o]<<24|this[o+1]<<16|this[o+2]<<8|this[o+3]};_.prototype.readFloatLE=function(o,n){return o=o>>>0,n||L(o,4,this.length),fe.read(this,o,!0,23,4)};_.prototype.readFloatBE=function(o,n){return o=o>>>0,n||L(o,4,this.length),fe.read(this,o,!1,23,4)};_.prototype.readDoubleLE=function(o,n){return o=o>>>0,n||L(o,8,this.length),fe.read(this,o,!0,52,8)};_.prototype.readDoubleBE=function(o,n){return o=o>>>0,n||L(o,8,this.length),fe.read(this,o,!1,52,8)};function R(e,o,n,r,p,s){if(!_.isBuffer(e))throw new TypeError('"buffer" argument must be a Buffer instance');if(o>p||o<s)throw new RangeError('"value" argument is out of bounds');if(n+r>e.length)throw new RangeError("Index out of range")}_.prototype.writeUintLE=_.prototype.writeUIntLE=function(o,n,r,p){if(o=+o,n=n>>>0,r=r>>>0,!p){var s=Math.pow(2,8*r)-1;R(this,o,n,r,s,0)}var m=1,l=0;for(this[n]=o&255;++l<r&&(m*=256);)this[n+l]=o/m&255;return n+r};_.prototype.writeUintBE=_.prototype.writeUIntBE=function(o,n,r,p){if(o=+o,n=n>>>0,r=r>>>0,!p){var s=Math.pow(2,8*r)-1;R(this,o,n,r,s,0)}var m=r-1,l=1;for(this[n+m]=o&255;--m>=0&&(l*=256);)this[n+m]=o/l&255;return n+r};_.prototype.writeUint8=_.prototype.writeUInt8=function(o,n,r){return o=+o,n=n>>>0,r||R(this,o,n,1,255,0),this[n]=o&255,n+1};_.prototype.writeUint16LE=_.prototype.writeUInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||R(this,o,n,2,65535,0),this[n]=o&255,this[n+1]=o>>>8,n+2};_.prototype.writeUint16BE=_.prototype.writeUInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||R(this,o,n,2,65535,0),this[n]=o>>>8,this[n+1]=o&255,n+2};_.prototype.writeUint32LE=_.prototype.writeUInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||R(this,o,n,4,4294967295,0),this[n+3]=o>>>24,this[n+2]=o>>>16,this[n+1]=o>>>8,this[n]=o&255,n+4};_.prototype.writeUint32BE=_.prototype.writeUInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||R(this,o,n,4,4294967295,0),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};_.prototype.writeIntLE=function(o,n,r,p){if(o=+o,n=n>>>0,!p){var s=Math.pow(2,8*r-1);R(this,o,n,r,s-1,-s)}var m=0,l=1,d=0;for(this[n]=o&255;++m<r&&(l*=256);)o<0&&d===0&&this[n+m-1]!==0&&(d=1),this[n+m]=(o/l>>0)-d&255;return n+r};_.prototype.writeIntBE=function(o,n,r,p){if(o=+o,n=n>>>0,!p){var s=Math.pow(2,8*r-1);R(this,o,n,r,s-1,-s)}var m=r-1,l=1,d=0;for(this[n+m]=o&255;--m>=0&&(l*=256);)o<0&&d===0&&this[n+m+1]!==0&&(d=1),this[n+m]=(o/l>>0)-d&255;return n+r};_.prototype.writeInt8=function(o,n,r){return o=+o,n=n>>>0,r||R(this,o,n,1,127,-128),o<0&&(o=255+o+1),this[n]=o&255,n+1};_.prototype.writeInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||R(this,o,n,2,32767,-32768),this[n]=o&255,this[n+1]=o>>>8,n+2};_.prototype.writeInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||R(this,o,n,2,32767,-32768),this[n]=o>>>8,this[n+1]=o&255,n+2};_.prototype.writeInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||R(this,o,n,4,2147483647,-2147483648),this[n]=o&255,this[n+1]=o>>>8,this[n+2]=o>>>16,this[n+3]=o>>>24,n+4};_.prototype.writeInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||R(this,o,n,4,2147483647,-2147483648),o<0&&(o=4294967295+o+1),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};function Kn(e,o,n,r,p,s){if(n+r>e.length)throw new RangeError("Index out of range");if(n<0)throw new RangeError("Index out of range")}function Xn(e,o,n,r,p){return o=+o,n=n>>>0,p||Kn(e,o,n,4,34028234663852886e22,-34028234663852886e22),fe.write(e,o,n,r,23,4),n+4}_.prototype.writeFloatLE=function(o,n,r){return Xn(this,o,n,!0,r)};_.prototype.writeFloatBE=function(o,n,r){return Xn(this,o,n,!1,r)};function Yn(e,o,n,r,p){return o=+o,n=n>>>0,p||Kn(e,o,n,8,17976931348623157e292,-17976931348623157e292),fe.write(e,o,n,r,52,8),n+8}_.prototype.writeDoubleLE=function(o,n,r){return Yn(this,o,n,!0,r)};_.prototype.writeDoubleBE=function(o,n,r){return Yn(this,o,n,!1,r)};_.prototype.copy=function(o,n,r,p){if(!_.isBuffer(o))throw new TypeError("argument should be a Buffer");if(r||(r=0),!p&&p!==0&&(p=this.length),n>=o.length&&(n=o.length),n||(n=0),p>0&&p<r&&(p=r),p===r||o.length===0||this.length===0)return 0;if(n<0)throw new RangeError("targetStart out of bounds");if(r<0||r>=this.length)throw new RangeError("Index out of range");if(p<0)throw new RangeError("sourceEnd out of bounds");p>this.length&&(p=this.length),o.length-n<p-r&&(p=o.length-n+r);var s=p-r;return this===o&&typeof Uint8Array.prototype.copyWithin=="function"?this.copyWithin(n,r,p):Uint8Array.prototype.set.call(o,this.subarray(r,p),n),s};_.prototype.fill=function(o,n,r,p){if(typeof o=="string"){if(typeof n=="string"?(p=n,n=0,r=this.length):typeof r=="string"&&(p=r,r=this.length),p!==void 0&&typeof p!="string")throw new TypeError("encoding must be a string");if(typeof p=="string"&&!_.isEncoding(p))throw new TypeError("Unknown encoding: "+p);if(o.length===1){var s=o.charCodeAt(0);(p==="utf8"&&s<128||p==="latin1")&&(o=s)}}else typeof o=="number"?o=o&255:typeof o=="boolean"&&(o=Number(o));if(n<0||this.length<n||this.length<r)throw new RangeError("Out of range index");if(r<=n)return this;n=n>>>0,r=r===void 0?this.length:r>>>0,o||(o=0);var m;if(typeof o=="number")for(m=n;m<r;++m)this[m]=o;else{var l=_.isBuffer(o)?o:_.from(o,p),d=l.length;if(d===0)throw new TypeError('The value "'+o+'" is invalid for argument "value"');for(m=0;m<r-n;++m)this[m+n]=l[m%d]}return this};var oi=/[^+/0-9A-Za-z-_]/g;function ni(e){if(e=e.split("=")[0],e=e.trim().replace(oi,""),e.length<2)return"";for(;e.length%4!==0;)e=e+"=";return e}function Do(e,o){o=o||1/0;for(var n,r=e.length,p=null,s=[],m=0;m<r;++m){if(n=e.charCodeAt(m),n>55295&&n<57344){if(!p){if(n>56319){(o-=3)>-1&&s.push(239,191,189);continue}else if(m+1===r){(o-=3)>-1&&s.push(239,191,189);continue}p=n;continue}if(n<56320){(o-=3)>-1&&s.push(239,191,189),p=n;continue}n=(p-55296<<10|n-56320)+65536}else p&&(o-=3)>-1&&s.push(239,191,189);if(p=null,n<128){if((o-=1)<0)break;s.push(n)}else if(n<2048){if((o-=2)<0)break;s.push(n>>6|192,n&63|128)}else if(n<65536){if((o-=3)<0)break;s.push(n>>12|224,n>>6&63|128,n&63|128)}else if(n<1114112){if((o-=4)<0)break;s.push(n>>18|240,n>>12&63|128,n>>6&63|128,n&63|128)}else throw new Error("Invalid code point")}return s}function ti(e){for(var o=[],n=0;n<e.length;++n)o.push(e.charCodeAt(n)&255);return o}function ri(e,o){for(var n,r,p,s=[],m=0;m<e.length&&!((o-=2)<0);++m)n=e.charCodeAt(m),r=n>>8,p=n%256,s.push(p),s.push(r);return s}function Zn(e){return Io.toByteArray(ni(e))}function Ke(e,o,n,r){for(var p=0;p<r&&!(p+n>=o.length||p>=e.length);++p)o[p+n]=e[p];return p}function j(e,o){return e instanceof o||e!=null&&e.constructor!=null&&e.constructor.name!=null&&e.constructor.name===o.name}function No(e){return e!==e}var ai=function(){for(var e="0123456789abcdef",o=new Array(256),n=0;n<16;++n)for(var r=n*16,p=0;p<16;++p)o[r+p]=e[n]+e[p];return o}()});var tt=je((ts,nt)=>{t();var E=nt.exports={},z,q;function Ro(){throw new Error("setTimeout has not been defined")}function Mo(){throw new Error("clearTimeout has not been defined")}(function(){try{typeof setTimeout=="function"?z=setTimeout:z=Ro}catch(e){z=Ro}try{typeof clearTimeout=="function"?q=clearTimeout:q=Mo}catch(e){q=Mo}})();function Qn(e){if(z===setTimeout)return setTimeout(e,0);if((z===Ro||!z)&&setTimeout)return z=setTimeout,setTimeout(e,0);try{return z(e,0)}catch(o){try{return z.call(null,e,0)}catch(n){return z.call(this,e,0)}}}function ii(e){if(q===clearTimeout)return clearTimeout(e);if((q===Mo||!q)&&clearTimeout)return q=clearTimeout,clearTimeout(e);try{return q(e)}catch(o){try{return q.call(null,e)}catch(n){return q.call(this,e)}}}var X=[],Te=!1,pe,Xe=-1;function ci(){!Te||!pe||(Te=!1,pe.length?X=pe.concat(X):Xe=-1,X.length&&et())}function et(){if(!Te){var e=Qn(ci);Te=!0;for(var o=X.length;o;){for(pe=X,X=[];++Xe<o;)pe&&pe[Xe].run();Xe=-1,o=X.length}pe=null,Te=!1,ii(e)}}E.nextTick=function(e){var o=new Array(arguments.length-1);if(arguments.length>1)for(var n=1;n<arguments.length;n++)o[n-1]=arguments[n];X.push(new ot(e,o)),X.length===1&&!Te&&Qn(et)};function ot(e,o){this.fun=e,this.array=o}ot.prototype.run=function(){this.fun.apply(null,this.array)};E.title="browser";E.browser=!0;E.env={};E.argv=[];E.version="";E.versions={};function Y(){}E.on=Y;E.addListener=Y;E.once=Y;E.off=Y;E.removeListener=Y;E.removeAllListeners=Y;E.emit=Y;E.prependListener=Y;E.prependOnceListener=Y;E.listeners=function(e){return[]};E.binding=function(e){throw new Error("process.binding is not supported")};E.cwd=function(){return"/"};E.chdir=function(e){throw new Error("process.chdir is not supported")};E.umask=function(){return 0}});var a,i,rs,t=h(()=>{a=y(Jn()),i=y(tt()),rs=function(e){function o(){var r=this||self;return delete e.prototype.__magic__,r}if(typeof globalThis=="object")return globalThis;if(this)return o();e.defineProperty(e.prototype,"__magic__",{configurable:!0,get:o});var n=__magic__;return n}(Object)});var xe,Ye=h(()=>{"use strict";t();xe=()=>Date.now()/1e3});var rt,Fo=h(()=>{"use strict";t();rt=e=>new Date().setMilliseconds(new Date().getMilliseconds()+e)/1e3});var at=h(()=>{"use strict";t()});var Oo=h(()=>{"use strict";t();Ye();Fo();at()});var Uo={};ae(Uo,{getItem:()=>si,localStorageKeys:()=>oe,removeItem:()=>mi,setItem:()=>pi});var oe,Bo,pi,si,mi,ye=h(()=>{"use strict";t();Oo();oe={loginExpiresAt:"dapp-core-login-expires-at",logoutEvent:"dapp-core-logout-event"},Bo=typeof localStorage!="undefined",pi=({key:e,data:o,expires:n})=>{!Bo||localStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},si=e=>{if(!Bo)return;let o=localStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:xe()>=n.expires?(localStorage.removeItem(String(e)),null):n.data},mi=e=>{!Bo||localStorage.removeItem(String(e))}});var Wo={};ae(Wo,{clear:()=>st,getItem:()=>ct,removeItem:()=>pt,setItem:()=>it,storage:()=>_i});var it,ct,pt,st,_i,mt=h(()=>{"use strict";t();it=({key:e,data:o,expires:n})=>{sessionStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},ct=e=>{let o=sessionStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:Date.now()>=n.expires?(sessionStorage.removeItem(String(e)),null):n.data},pt=e=>sessionStorage.removeItem(String(e)),st=()=>sessionStorage.clear(),_i={setItem:it,getItem:ct,removeItem:pt,clear:st}});var se,be=h(()=>{"use strict";t();ye();mt();se={session:Wo,local:Uo}});function Ne(){return new Date().setHours(new Date().getHours()+24)}function Re(e){se.local.setItem({key:oe.loginExpiresAt,data:e,expires:e})}var Go=h(()=>{"use strict";t();be();ye()});var I=h(()=>{"use strict";t()});var we=h(()=>{"use strict";t()});var _t=h(()=>{"use strict";t()});var P=h(()=>{"use strict";t()});var dt=h(()=>{"use strict";t()});var lt=h(()=>{"use strict";t()});var ut=h(()=>{"use strict";t()});var gt=h(()=>{"use strict";t()});var M=h(()=>{"use strict";t();_t();P();I();dt();lt();ut();gt()});var $o,ft,Ps,ht,Ns,Tt,Rs,Ms,Fs,xt=h(()=>{"use strict";t();M();$o={devnet:{id:"devnet",chainId:"D",name:"Devnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://devnet-wallet.elrond.com",apiAddress:"https://devnet-api.elrond.com",explorerAddress:"http://devnet-explorer.elrond.com",apiTimeout:"4000"},testnet:{id:"testnet",chainId:"T",name:"Testnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://testnet-wallet.elrond.com",apiAddress:"https://testnet-api.elrond.com",explorerAddress:"http://testnet-explorer.elrond.com",apiTimeout:"4000"},mainnet:{id:"mainnet",chainId:"1",name:"Mainnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://wallet.elrond.com",apiAddress:"https://api.elrond.com",explorerAddress:"https://explorer.elrond.com",apiTimeout:"4000"}},{chainId:ft,egldLabel:Ps}=$o["devnet"],{chainId:ht,egldLabel:Ns}=$o["testnet"],{chainId:Tt,egldLabel:Rs}=$o["mainnet"],Ms={["devnet"]:ft,["testnet"]:ht,["mainnet"]:Tt},Fs={[ft]:"devnet",[ht]:"testnet",[Tt]:"mainnet"}});var Vo=h(()=>{"use strict";t()});var yt=h(()=>{"use strict";t()});var Ho,jo=h(()=>{"use strict";t();Ho="transaction-status-toast-id"});var Me,Ze,bt,zo,Je,wt,Se,k=h(()=>{"use strict";t();we();xt();Vo();yt();jo();Me=5e4,Ze=1e9,bt=1,zo=4294967295,Je="logout",wt="login",Se="0"});var qo,N,Z,F=h(()=>{"use strict";t();qo=require("@reduxjs/toolkit");k();N=(0,qo.createAction)(Je),Z=(0,qo.createAction)(wt,e=>({payload:e}))});var vt,St,kt,em,At,di,Fe,It,li,Et,Lt,Ko=h(()=>{"use strict";t();vt=require("@reduxjs/toolkit");Go();I();F();St={loginMethod:"",walletConnectLogin:null,ledgerLogin:null,tokenLogin:null,walletLogin:null,extensionLogin:null,isLoginSessionInvalid:!1},kt=(0,vt.createSlice)({name:"loginInfoSlice",initialState:St,reducers:{setLoginMethod:(e,o)=>{e.loginMethod=o.payload},setTokenLogin:(e,o)=>{e.tokenLogin=o.payload},setTokenLoginSignature:(e,o)=>{(e==null?void 0:e.tokenLogin)!=null&&(e.tokenLogin.signature=o.payload)},setWalletLogin:(e,o)=>{e.walletLogin=o.payload},setWalletConnectLogin:(e,o)=>{e.walletConnectLogin=o.payload},setLedgerLogin:(e,o)=>{e.ledgerLogin=o.payload},invalidateLoginSession:e=>{e.isLoginSessionInvalid=!0}},extraReducers:e=>{e.addCase(N,()=>St),e.addCase(Z,(o,n)=>{o.loginMethod=n.payload.loginMethod,Re(Ne())})}}),{setLoginMethod:em,setWalletConnectLogin:At,setLedgerLogin:di,setTokenLogin:Fe,setTokenLoginSignature:It,setWalletLogin:li,invalidateLoginSession:Et}=kt.actions,Lt=kt.reducer});var Oe=h(()=>{"use strict";t()});var Ct=h(()=>{"use strict";t();Oe()});function Xo(e){return e[Math.floor(Math.random()*e.length)]}var Pt=h(()=>{"use strict";t()});var Yo=h(()=>{"use strict";t();Vo()});var ve=h(()=>{"use strict";t();Ct();Pt();Yo()});var Nt,Rt,Zo,ui,Mt,gm,fm,Ft,Jo=h(()=>{"use strict";t();Nt=require("@reduxjs/toolkit"),Rt=y(require("lodash.omit"));ve();Zo={id:"not-configured",chainId:"",name:"NOT CONFIGURED",egldLabel:"",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"",walletConnectBridgeAddress:"",walletConnectV2RelayAddress:"",walletConnectV2ProjectId:"",walletAddress:"",apiAddress:"",explorerAddress:"",apiTimeout:"4000"},ui={network:Zo,chainID:"-1"},Mt=(0,Nt.createSlice)({name:"appConfig",initialState:ui,reducers:{initializeNetworkConfig:(e,o)=>{let n=Xo(o.payload.walletConnectBridgeAddresses),r=Xo(o.payload.walletConnectV2RelayAddresses),p=(0,Rt.default)(o.payload,"walletConnectBridgeAddresses","walletConnectV2RelayAddresses");e.network=v(T(T({},e.network),p),{walletConnectBridgeAddress:n,walletConnectV2RelayAddress:r})},setChainID:(e,o)=>{e.chainID=o.payload}}}),{initializeNetworkConfig:gm,setChainID:fm}=Mt.actions,Ft=Mt.reducer});var en,Ot,Bt,Qe,Qo,Ut,on,vm,nn,km,gi,fi,Am,Im,Em,hi,Wt,tn=h(()=>{"use strict";t();en=require("@elrondnetwork/erdjs"),Ot=require("@reduxjs/toolkit"),Bt=require("redux-persist");k();be();ye();F();Qe={balance:"...",address:"",nonce:0,txCount:0,scrCount:0,claimableRewards:Se},Qo={address:"",websocketEvent:null,accounts:{"":Qe},ledgerAccount:null,publicKey:"",walletConnectAccount:null,isAccountLoading:!0,accountLoadingError:null},Ut=(0,Ot.createSlice)({name:"accountInfoSlice",initialState:Qo,reducers:{setAddress:(e,o)=>{let n=o.payload;e.address=n,e.publicKey=new en.Address(n).hex()},setAccount:(e,o)=>{let n=e.address===o.payload.address;e.accounts={[e.address]:n?o.payload:Qe},e.isAccountLoading=!1,e.accountLoadingError=null},setAccountNonce:(e,o)=>{let{address:n}=e;e.accounts[n].nonce=o.payload},setAccountShard:(e,o)=>{e.shard=o.payload},setLedgerAccount:(e,o)=>{e.ledgerAccount=o.payload},updateLedgerAccount:(e,o)=>{e.ledgerAccount!=null&&(e.ledgerAccount.index=o.payload.index,e.ledgerAccount.address=o.payload.address)},setWalletConnectAccount:(e,o)=>{e.walletConnectAccount=o.payload},setIsAccountLoading:(e,o)=>{e.isAccountLoading=o.payload,e.accountLoadingError=null},setAccountLoadingError:(e,o)=>{e.accountLoadingError=o.payload,e.isAccountLoading=!1},setWebsocketEvent:(e,o)=>{e.websocketEvent={timestamp:Date.now(),message:o.payload}}},extraReducers:e=>{e.addCase(N,()=>(se.local.removeItem(oe.loginExpiresAt),Qo)),e.addCase(Z,(o,n)=>{let{address:r}=n.payload;o.address=r,o.publicKey=new en.Address(r).hex()}),e.addCase(Bt.REHYDRATE,(o,n)=>{var g;if(!((g=n.payload)!=null&&g.account))return;let{account:r}=n.payload,{address:p,shard:s,accounts:m,publicKey:l}=r;o.address=p,o.shard=s;let d=m&&p in m;o.accounts=d?m:Qo.accounts,o.publicKey=l})}}),{setAccount:on,setAddress:vm,setAccountNonce:nn,setAccountShard:km,setLedgerAccount:gi,updateLedgerAccount:fi,setWalletConnectAccount:Am,setIsAccountLoading:Im,setAccountLoadingError:Em,setWebsocketEvent:hi}=Ut.actions,Wt=Ut.reducer});function ne(e){return e!=null&&(ki(e)||Li(e))}function ke(e){return e!=null&&(Ai(e)||Di(e))}function Ae(e){return e!=null&&(Ii(e)||Ci(e))}function Ue(e){return e!=null&&Ei(e)}function ki(e){return e!=null&&Ti.includes(e)}function Ai(e){return e!=null&&xi.includes(e)}function Ii(e){return e!=null&&yi.includes(e)}function Ei(e){return e!=null&&bi.includes(e)}function Li(e){return e!=null&&wi.includes(e)}function Di(e){return e!=null&&Si.includes(e)}function Ci(e){return e!=null&&vi.includes(e)}var Ti,xi,yi,bi,wi,Si,vi,me=h(()=>{"use strict";t();I();Ti=["sent"],xi=["success"],yi=["fail","cancelled","timedOut"],bi=["timedOut"],wi=["pending"],Si=["success"],vi=["fail","invalid"]});var Gt,$t,We,Pi,Vt,Ht,jt,Ni,rn,Ri,Mi,Om,Fi,zt,an,eo,oo=h(()=>{"use strict";t();Gt=require("@reduxjs/toolkit"),$t=require("redux-persist");I();me();F();We={signedTransactions:{},transactionsToSign:null,signTransactionsError:null,signTransactionsCancelMessage:null,customTransactionInformationForSessionId:{}},Pi={signWithoutSending:!1,sessionInformation:null,redirectAfterSign:!1},Vt=(0,Gt.createSlice)({name:"transactionsSlice",initialState:We,reducers:{moveTransactionsToSignedState:(e,o)=>{var d,g;let{sessionId:n,transactions:r,errorMessage:p,status:s,redirectRoute:m}=o.payload,l=((d=e.customTransactionInformationForSessionId)==null?void 0:d[n])||Pi;e.signedTransactions[n]={transactions:r,status:s,errorMessage:p,redirectRoute:m,customTransactionInformation:l},((g=e==null?void 0:e.transactionsToSign)==null?void 0:g.sessionId)===n&&(e.transactionsToSign=We.transactionsToSign)},clearSignedTransaction:(e,o)=>{e.signedTransactions[o.payload]&&delete e.signedTransactions[o.payload]},clearTransactionToSign:e=>{e!=null&&e.transactionsToSign&&(e.transactionsToSign=null)},updateSignedTransaction:(e,o)=>{e.signedTransactions=T(T({},e.signedTransactions),o.payload)},updateSignedTransactions:(e,o)=>{let{sessionId:n,status:r,errorMessage:p,transactions:s}=o.payload;e.signedTransactions[n]!=null&&(e.signedTransactions[n].status=r,p!=null&&(e.signedTransactions[n].errorMessage=p),s!=null&&(e.signedTransactions[n].transactions=s))},updateSignedTransactionStatus:(e,o)=>{var l,d,g,x,f,D;let{sessionId:n,status:r,errorMessage:p,transactionHash:s}=o.payload,m=(d=(l=e.signedTransactions)==null?void 0:l[n])==null?void 0:d.transactions;if(m!=null){e.signedTransactions[n].transactions=m.map(G=>G.hash===s?v(T({},G),{status:r,errorMessage:p}):G);let C=(x=(g=e.signedTransactions[n])==null?void 0:g.transactions)==null?void 0:x.every(G=>ke(G.status)),V=(D=(f=e.signedTransactions[n])==null?void 0:f.transactions)==null?void 0:D.every(G=>Ae(G.status));C&&(e.signedTransactions[n].status="success"),V&&(e.signedTransactions[n].status="fail")}},setTransactionsToSign:(e,o)=>{e.transactionsToSign=o.payload;let{sessionId:n,customTransactionInformation:r}=o.payload;e.customTransactionInformationForSessionId[n]=r,e.signTransactionsError=null},clearAllTransactionsToSign:e=>{e.transactionsToSign=We.transactionsToSign,e.signTransactionsError=null},clearAllSignedTransactions:e=>{e.signedTransactions=We.signedTransactions},setSignTransactionsError:(e,o)=>{e.signTransactionsError=o.payload},setSignTransactionsCancelMessage:(e,o)=>{e.signTransactionsCancelMessage=o.payload}},extraReducers:e=>{e.addCase(N,()=>We),e.addCase($t.REHYDRATE,(o,n)=>{var m;if(!((m=n.payload)!=null&&m.transactions))return;let{signedTransactions:r,customTransactionInformationForSessionId:p}=n.payload.transactions,s=Object.entries(r).reduce((l,[d,g])=>{let x=new Date(d),f=new Date;return f.setHours(f.getHours()+5),f-x>0||(l[d]=g),l},{});p!=null&&(o.customTransactionInformationForSessionId=p),r!=null&&(o.signedTransactions=s)})}}),{updateSignedTransactionStatus:Ht,updateSignedTransactions:jt,setTransactionsToSign:Ni,clearAllTransactionsToSign:rn,clearAllSignedTransactions:Ri,clearSignedTransaction:Mi,clearTransactionToSign:Om,setSignTransactionsError:Fi,setSignTransactionsCancelMessage:zt,moveTransactionsToSignedState:an}=Vt.actions,eo=Vt.reducer});var qt,pn,sn,mn,Oi,cn,Kt,Gm,Bi,Xt,no,to=h(()=>{"use strict";t();qt=require("@reduxjs/toolkit");F();pn="Transaction failed",sn="Transaction successful",mn="Processing transaction",Oi="Transaction submitted",cn={},Kt=(0,qt.createSlice)({name:"transactionsInfo",initialState:cn,reducers:{setTransactionsDisplayInfo(e,o){let{sessionId:n,transactionsDisplayInfo:r}=o.payload;n!=null&&(e[n]={errorMessage:(r==null?void 0:r.errorMessage)||pn,successMessage:(r==null?void 0:r.successMessage)||sn,processingMessage:(r==null?void 0:r.processingMessage)||mn,submittedMessage:(r==null?void 0:r.submittedMessage)||Oi,transactionDuration:r==null?void 0:r.transactionDuration})},clearTransactionsInfoForSessionId(e,o){o.payload!=null&&delete e[o.payload]},clearTransactionsInfo:()=>cn},extraReducers:e=>{e.addCase(N,()=>cn)}}),{clearTransactionsInfo:Gm,setTransactionsDisplayInfo:Bi,clearTransactionsInfoForSessionId:Xt}=Kt.actions,no=Kt.reducer});var Zt,Yt,Jt,jm,Ui,zm,Wi,Qt,_n=h(()=>{"use strict";t();Zt=require("@reduxjs/toolkit");F();Yt={},Jt=(0,Zt.createSlice)({name:"modalsSlice",initialState:Yt,reducers:{setTxSubmittedModal:(e,o)=>{e.txSubmittedModal=o.payload},setNotificationModal:(e,o)=>{e.notificationModal=o.payload},clearTxSubmittedModal:e=>{e.txSubmittedModal=void 0},clearNotificationModal:e=>{e.notificationModal=void 0}},extraReducers:e=>{e.addCase(N,()=>Yt)}}),{setTxSubmittedModal:jm,setNotificationModal:Ui,clearTxSubmittedModal:zm,clearNotificationModal:Wi}=Jt.actions,Qt=Jt.reducer});var or,er,nr,Gi,tr,e_,o_,n_,t_,ro,ao=h(()=>{"use strict";t();or=require("@reduxjs/toolkit");jo();M();Ye();F();er={customToasts:[],transactionToasts:[],failTransactionToast:null},nr=(0,or.createSlice)({name:"toastsSlice",initialState:er,reducers:{addCustomToast:(e,o)=>{e.customToasts.push(v(T({},o.payload),{type:"custom",toastId:o.payload.toastId||`custom-toast-${e.customToasts.length+1}`}))},removeCustomToast:(e,o)=>{e.customToasts=e.customToasts.filter(n=>n.toastId!==o.payload)},addTransactionToast:(e,o)=>{e.transactionToasts.push({type:"transaction",startTimestamp:xe(),toastId:o.payload||`custom-toast-${e.transactionToasts.length+1}`})},removeTransactionToast:(e,o)=>{e.transactionToasts=e.transactionToasts.filter(n=>n.toastId!==o.payload)},addFailTransactionToast:(e,o)=>{e.failTransactionToast=v(T({},o.payload),{toastId:Ho})},removeFailTransactionToast:e=>{e.failTransactionToast=null}},extraReducers:e=>{e.addCase(N,()=>er)}}),{addCustomToast:Gi,removeCustomToast:tr,addTransactionToast:e_,removeTransactionToast:o_,addFailTransactionToast:n_,removeFailTransactionToast:t_}=nr.actions,ro=nr.reducer});var A=h(()=>{"use strict";t();Ko();Jo();tn();oo();to();_n();ao()});var dn=h(()=>{"use strict";t()});var cr,ir,pr,S_,v_,k_,A_,I_,E_,sr,mr=h(()=>{"use strict";t();cr=require("@reduxjs/toolkit");F();ir={visible:!1,modal:{showHeader:!0,showFooter:!1,headerText:"",footerText:""}},pr=(0,cr.createSlice)({name:"dappModalsSlice",initialState:ir,reducers:{setDappModalVisibility:(e,o)=>{e.visible=o.payload},setDappModalHeaderVisibility:(e,o)=>{e.modal.showHeader=o.payload},setDappModalFooterVisibility:(e,o)=>{e.modal.showFooter=o.payload},setDappModalHeaderText:(e,o)=>{e.modal.headerText=o.payload},setDappModalFooterText:(e,o)=>{e.modal.footerText=o.payload},setDappModalConfig:(e,o)=>{e.modal=T({},o.payload)}},extraReducers:e=>{e.addCase(N,()=>ir)}}),{setDappModalVisibility:S_,setDappModalHeaderVisibility:v_,setDappModalFooterVisibility:k_,setDappModalHeaderText:A_,setDappModalFooterText:I_,setDappModalConfig:E_}=pr.actions,sr=pr.reducer});var _r,ji,zi,Ie,co=h(()=>{"use strict";t();_r=require("@reduxjs/toolkit");dn();tn();mr();Ko();_n();Jo();ao();to();oo();ji={["account"]:Wt,["networkConfig"]:Ft,["loginInfo"]:Lt,["modals"]:Qt,["dappModal"]:sr,["toasts"]:ro,["transactions"]:eo,["transactionsInfo"]:no},zi=(e={})=>(0,_r.combineReducers)(T(T({},ji),e)),Ie=zi});var ur={};ae(ur,{default:()=>Qi});function ln(e,o=[]){return{key:e,version:1,storage:lr.default,blacklist:o}}var _e,dr,lr,qi,Ki,Xi,Yi,Zi,Ji,Qi,gr=h(()=>{"use strict";t();_e=require("redux-persist"),dr=y(require("redux-persist/lib/storage")),lr=y(require("redux-persist/lib/storage/session"));co();A();ao();to();oo();dn();qi={2:e=>v(T({},e),{networkConfig:Zo})};Ki=ln("dapp-core-transactionsInfo"),Xi=ln("dapp-core-transactions",["transactionsToSign"]),Yi=ln("dapp-core-toasts"),Zi={key:"dapp-core-store",version:2,storage:dr.default,whitelist:["account","loginInfo","modals","networkConfig"],migrate:(0,_e.createMigrate)(qi,{debug:!1})},Ji={["toasts"]:(0,_e.persistReducer)(Yi,ro),["transactions"]:(0,_e.persistReducer)(Xi,eo),["transactionsInfo"]:(0,_e.persistReducer)(Ki,no)},Qi=(0,_e.persistReducer)(Zi,Ie(Ji))});var fr={};ae(fr,{default:()=>ec});var ec,hr=h(()=>{"use strict";t();co();ec=Ie()});var Tr={};ae(Tr,{default:()=>nc});var W,oc,nc,xr=h(()=>{"use strict";t();W=require("redux-persist"),oc=[W.FLUSH,W.REHYDRATE,W.PAUSE,W.PERSIST,W.PURGE,W.REGISTER],nc=oc});var wr={};ae(wr,{default:()=>br});function br(e){return(0,yr.persistStore)(e)}var yr,Sr=h(()=>{"use strict";t();yr=require("redux-persist")});var Jp={};ae(Jp,{useTransactionToast:()=>Zp});module.exports=Pe(Jp);t();var Ve=require("react");t();t();t();var Lr=require("react"),Ee=require("react-redux");t();var un=require("@reduxjs/toolkit"),Ir=require("react-redux/es/utils/Subscription");A();t();var rr=y(require("lodash.throttle"));k();A();Go();be();ye();var $i=[Je],Vi=(0,rr.default)(()=>{Re(Ne())},5e3),ar=e=>o=>n=>{var d;if($i.includes(n.type))return o(n);let r=e.getState(),p=se.local.getItem(oe.loginExpiresAt);if(!((d=r==null?void 0:r.loginInfo)==null?void 0:d.isLoggedIn))return o(n);if(p==null)return Re(Ne());let m=Date.now();return p-m<0?setTimeout(()=>b(void 0,null,function*(){console.log("session expired");try{e.dispatch(Et())}catch(g){console.error("error logging out",g)}}),1e3):(Vi(),o(n))};t();t();function io(){return typeof sessionStorage!="undefined"}var vr=io()?(gr(),Pe(ur)).default:(hr(),Pe(fr)).default,kr=io()?(xr(),Pe(Tr)).default:[],Ar=io()?(Sr(),Pe(wr)).default:e=>e;co();var w=(0,un.configureStore)({reducer:vr,middleware:e=>e({serializableCheck:{ignoredActions:[...kr,nn.type,on.type],ignoredPaths:["payload.nonce","account.account.nonce","providers.accountProvider"]}}).concat(ar)}),Er=(0,Ir.createSubscription)(w),Q_=Ar(w),ed=(0,un.configureStore)({reducer:Ie});var tc={store:w,subscription:Er},gn=(0,Lr.createContext)(tc),rd=(0,Ee.createStoreHook)(gn),$=(0,Ee.createDispatchHook)(gn),S=(0,Ee.createSelectorHook)(gn);t();t();A();t();var Dr=y(require("lodash.isequal")),po=require("reselect"),u=(0,po.createSelectorCreator)(po.defaultMemoize,Dr.default);var Q=e=>e.account,Le=u(Q,e=>e.address),Ge=u(Q,Le,(e,o)=>o in e.accounts?e.accounts[o]:Qe),rc=u(Q,Ge,(e,o)=>{let p=e,{accounts:n}=p,r=vo(p,["accounts"]);return v(T({},r),{account:o})}),ac=u(Ge,e=>e.balance),ic=u(Ge,e=>{var o;return((o=e==null?void 0:e.nonce)==null?void 0:o.valueOf())||0}),Cr=u(Q,e=>e.shard),cc=u(Q,e=>e.ledgerAccount),sd=u(Q,e=>e.walletConnectAccount),md=u(Q,e=>e.isAccountLoading),_d=u(Q,e=>e.accountLoadingError),Pr=u(Q,e=>e.websocketEvent);t();var te=e=>e.networkConfig,so=u(te,e=>e.chainID),pc=u(te,e=>e.network.walletConnectBridgeAddress),sc=u(te,e=>e.network.walletConnectV2RelayAddress),mc=u(te,e=>e.network.walletConnectV2ProjectId),Nr=u(te,e=>e.network.walletConnectDeepLink),de=u(te,e=>e.network),Rr=u(de,e=>e.apiAddress),Mr=u(de,e=>e.explorerAddress),Fr=u(de,e=>e.egldLabel);t();I();var le=e=>e.loginInfo,Td=u(le,e=>e.loginMethod),mo=u(le,Le,(e,o)=>e.loginMethod!=""&&Boolean(o)),xd=u(le,e=>e.walletConnectLogin),yd=u(le,e=>e.ledgerLogin),_c=u(le,e=>e.walletLogin),bd=u(le,e=>e.isLoginSessionInvalid);t();t();var O=require("@elrondnetwork/erdjs");k();t();function _o(e){try{return a.Buffer.from(e,"base64").toString()===atob(e)}catch(o){return!1}}function lo(e){var p,s,m;let{data:o}=e,n=_o(o!=null?o:"")?O.TransactionPayload.fromEncoded(o):new O.TransactionPayload(o),r=new O.Transaction(T({value:e.value.valueOf(),data:n,nonce:e.nonce.valueOf(),receiver:new O.Address(e.receiver),sender:new O.Address(e.sender),gasLimit:(p=e.gasLimit.valueOf())!=null?p:Me,gasPrice:(s=e.gasPrice.valueOf())!=null?s:Ze,chainID:e.chainID.valueOf(),version:new O.TransactionVersion((m=e.version)!=null?m:bt)},e.options?{options:new O.TransactionOptions(e.options)}:{}));return r.applySignature({hex:()=>e.signature||""},new O.Address(e.sender)),r}t();me();t();var lc=y(require("bignumber.js"));M();t();t();t();t();var gc=require("@elrondnetwork/erdjs"),fc=y(require("bignumber.js"));I();t();var uc=require("@elrondnetwork/erdjs");t();t();t();var Tn=require("@elrondnetwork/erdjs/out");t();k();var hc=e=>{let o=a.Buffer.from([0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]);if(e.slice(0,o.length).equals(o))return!0;let r=a.Buffer.alloc(32).fill(0);return e.equals(r)},hn=e=>{try{let n=parseInt("11",2),r=parseInt("01",2),p=a.Buffer.from(e,"hex"),s=p[31];if(hc(p))return zo;let m=s&n;return m>3-1&&(m=s&r),m}catch(o){return-1}};function Ur({receiverAddress:e,senderShard:o,senderAddress:n}){try{let r=new Tn.Address(e),p=hn(r.pubkey());if(o==null&&n!=null){let s=new Tn.Address(n);return hn(s)===p}return p===o}catch(r){return!1}}t();var xn=require("@elrondnetwork/erdjs");M();function Wr({receiver:e,data:o}){try{if(!o)return e;let n=_o(o)?xn.TransactionPayload.fromEncoded(o).toString():o,r=Tc(n),p=n.split("@");return r>-1?p[r]:e}catch(n){console.log(n);return}}function Tc(e){return e.includes("MultiESDTNFTTransfer")?1:e.includes("ESDTNFTTransfer")?4:-1}var Gr=(e,o=1)=>e!=null?e.reduce((n,{receiver:r,data:p})=>{let s=Wr({receiver:r,data:p!=null?p:""});return s==null?n:n&&Ur({receiverAddress:s,senderShard:o})},!0):!0;t();t();t();t();I();t();t();t();t();t();t();t();t();t();t();t();P();t();t();P();t();t();t();var Ic=y(require("bignumber.js"));k();t();var Ac=require("@elrondnetwork/erdjs"),yn=y(require("bignumber.js"));k();t();var vc=y(require("bignumber.js"));t();yn.default.config({ROUNDING_MODE:yn.default.ROUND_FLOOR});t();t();t();k();t();t();t();t();k();t();k();t();var Ec=require("@elrondnetwork/erdjs");k();t();var fo=require("@elrondnetwork/erdjs"),Cc=require("@elrondnetwork/erdjs"),Pc=require("@elrondnetwork/erdjs-network-providers");k();t();t();var Lc=y(require("bignumber.js"));k();t();k();t();t();t();t();t();k();t();P();var Rc=["reDelegateRewards","claimRewards","unBond"],Mc=["wrapEgld","unwrapEgld"],Fc=["unStake"],Oc=["unDelegate"];t();t();t();k();t();var Gc=y(require("bignumber.js"));t();t();t();P();t();var $c=y(require("bignumber.js"));t();t();t();P();t();P();t();var jc=y(require("bignumber.js"));k();P();t();P();t();P();t();var zc=y(require("bignumber.js"));t();I();t();I();t();t();var qc=require("react");P();var th=["create","localMint","ESDTLocalMint","addQuantity","burn","localBurn","ESDTLocalBurn","wipe","writeLog","signalError"];t();var Kc=y(require("anchorme"));t();var qr=require("react");M();t();t();P();t();var Xc=require("@elrondnetwork/erdjs/out"),Yc=y(require("bignumber.js"));P();t();P();t();t();I();var Rh=[{label:"Raw",value:"raw"},{label:"Text",value:"text"},{label:"Decimal",value:"decimal"},{label:"Smart",value:"smart"}];t();var Kr=require("react");M();t();t();t();t();t();k();var To=e=>e.transactions,De=u(To,e=>e.signedTransactions),Jc=u(To,e=>e.signTransactionsError),Qc=u(To,e=>e.signTransactionsCancelMessage),xo=e=>o=>Object.entries(o).reduce((n,[r,p])=>(e(p.status)&&(n[r]=p),n),{}),Xr=u(De,xo(ne)),Yr=u(De,xo(ke)),Zr=u(De,xo(Ae)),ep=u(De,xo(Ue)),wn=u(To,e=>{var o;return(e==null?void 0:e.transactionsToSign)==null?null:v(T({},e.transactionsToSign),{transactions:((o=e==null?void 0:e.transactionsToSign)==null?void 0:o.transactions.map(n=>lo(n)))||[]})}),op=u(De,(e,o)=>o,(e,o)=>o!=null?(e==null?void 0:e[o])||{}:{});t();A();var Jr={errorMessage:pn,successMessage:sn,processingMessage:mn},np=e=>e.transactionsInfo,Qr=u(np,(e,o)=>o,(e,o)=>o!=null&&(e==null?void 0:e[Number(o)])||Jr);t();var ea=e=>e.modals,CT=u(ea,e=>e.txSubmittedModal),tp=u(ea,e=>e.notificationModal);t();var Sn=e=>e.toasts,RT=u(Sn,e=>e.customToasts),MT=u(Sn,e=>e.transactionToasts),FT=u(Sn,e=>e.failTransactionToast);t();var oa=require("react");t();A();t();t();function na(e){return S(o=>Qr(o,e))}t();var ap=require("react"),ip=require("@elrondnetwork/erdjs-web-wallet-provider"),cp=y(require("qs"));k();A();I();t();var In=require("react"),hp=require("@elrondnetwork/erdjs-extension-provider");k();t();t();t();t();var pp=require("@elrondnetwork/erdjs-extension-provider"),sp=require("@elrondnetwork/erdjs-hw-provider"),ra=require("@elrondnetwork/erdjs-wallet-connect-provider"),mp=require("@elrondnetwork/erdjs-web-wallet-provider");k();I();var ee=e=>`Unable to perform ${e}, Provider not initialized`,vn=class{init(){return Promise.resolve(!1)}login(o){throw new Error(ee(`login with options: ${o}`))}logout(o){throw new Error(ee(`logout with options: ${o}`))}getAddress(){throw new Error(ee("getAddress"))}isInitialized(){return!1}isConnected(){return Promise.resolve(!1)}sendTransaction(o,n){throw new Error(ee(`sendTransaction with transactions: ${o} options: ${n}`))}signTransaction(o,n){throw new Error(ee(`signTransaction with transactions: ${o} options: ${n}`))}signTransactions(o,n){throw new Error(ee(`signTransactions with transactions: ${o} options: ${n}`))}signMessage(o,n){throw new Error(ee(`signTransactions with ${o} and options ${n}`))}sendCustomMessage({method:o,params:n}){throw new Error(ee(`sendCustomMessage with method: ${o} params: ${n}`))}sendCustomRequest(o){throw new Error(ee(`sendSessionEvent with options: ${o}`))}ping(){return Promise.resolve(!1)}},_p=new vn;A();I();t();t();F();M();t();t();t();I();t();t();t();t();var dp=y(require("axios"));t();t();t();A();t();t();t();A();t();var ca=require("@elrondnetwork/erdjs");t();F();M();Oe();be();ye();t();t();t();t();A();t();t();t();t();t();be();Oe();Oo();t();t();t();t();t();var lp=y(require("axios"));t();t();var up=y(require("platform"));t();t();t();var xp=y(require("axios")),yp=y(require("swr"));t();t();var bp=y(require("axios"));t();t();var ma=require("react");t();var sa=require("react");Yo();A();M();Oe();t();we();A();t();t();t();t();t();me();t();t();t();var Sp=y(require("axios"));t();var vp=y(require("axios"));t();var Ap=y(require("axios"));t();t();var Ip=y(require("axios"));t();var Ep=y(require("axios"));t();A();I();t();me();t();A();I();t();A();M();t();var Fp=require("react");t();t();t();A();t();t();var Np=y(require("bignumber.js"));t();var Dp=`.dapp-core-component__main__link-style {
  color: #1392ff;
}

.dapp-core-component__main__link-second-style {
  color: #9dabbd;
}
.dapp-core-component__main__link-second-style:hover {
  color: #1392ff;
}
.dapp-core-component__main__link-second-style:hover svg {
  color: #1392ff;
}

.dapp-core-component__main__wrapper {
  min-height: 100vh;
}

.dapp-core-component__main__opacity-6 {
  opacity: 0.6;
}

.dapp-core-component__main__qr-code-svg-container {
  width: 15rem;
  height: 15rem;
}

.dapp-core-component__main__pair-list {
  max-height: 20rem;
  overflow-y: auto;
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__pair-image {
  width: 4rem;
  overflow: hidden;
  height: auto;
  min-width: 2rem;
}

.dapp-core-component__main__pair-remove {
  padding: 0 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

.dapp-core-component__main__dapp-icon {
  padding: 5px;
  border-radius: 50%;
  background-color: #fafafa;
  width: 4.6428571429rem;
  height: 4.6428571429rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}
.dapp-core-component__main__dapp-icon.dapp-core-component__main__icon-medium {
  width: 80px;
  height: 80px;
}

.dapp-core-component__main__centering {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dapp-core-component__main__border-n {
  border: none !important;
}

.dapp-core-component__main__t-shadow {
  text-shadow: 0px 4px 7px rgba(0, 0, 0, 0.3);
}

.dapp-core-component__main__btn {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.04) !important;
}

.dapp-core-component__main__media {
  display: flex;
  align-items: flex-start;
}

.dapp-core-component__main__media-body {
  flex: 1;
}

.dapp-core-component__main__connect-btns .dapp-core-component__main__btn {
  display: flex;
  height: 48px;
  justify-content: center;
  align-items: center;
  color: #fff !important;
  background-color: #1392ff;
  padding: 15px 20px 15px 16px;
  border-radius: 6px;
  gap: 8px;
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__btn:hover, .dapp-core-component__main__connect-btns .dapp-core-component__main__btn:active {
  background-color: #1392ff;
  background-image: linear-gradient(rgba(0, 0, 0, 0.1) 0 0);
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__info {
  color: #1392ff;
  margin-left: 48px;
}

.dapp-core-component__main__btn {
  transition: none !important;
}

.dapp-core-component__main__trust-badge {
  width: 20px;
  height: 20px;
}

/* Navbar
  -------------------------------------------------- */
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name {
  display: flex;
  align-items: center;
  font-size: 1.7142857143rem;
  font-weight: 400;
  color: #9ba5b4;
  margin-left: 1.6428571429rem;
}
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name:before {
  content: "";
  display: block;
  width: 1px;
  height: 3rem;
  margin-right: 1.7142857143rem;
}

/* Dashboard
  -------------------------------------------------- */
.dapp-core-component__main__action-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  margin: 0 7px;
}
.dapp-core-component__main__action-btn button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #fff;
}
.dapp-core-component__main__action-btn button:hover {
  opacity: 0.9;
}

.dapp-core-component__main__transactions .dapp-core-component__main__transaction-icon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table td {
  white-space: nowrap;
  vertical-align: middle;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table tr:last-of-type td {
  border-bottom: 1px solid #dee2e6;
}

/* Loader, Transaction Success/Fail
  -------------------------------------------------- */
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div {
  position: absolute;
  top: 27px;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.1);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.dapp-core-component__main__page-state .dapp-core-component__main__bg-blue .dapp-core-component__main__lds-ellipsis div {
  background: rgba(255, 255, 255, 0.4);
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(1) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis1 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(2) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(3) {
  left: 26px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(4) {
  left: 45px;
  animation: dapp-core-component__main__lds-ellipsis3 0.6s infinite;
}
@keyframes dapp-core-component__main__lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(19px, 0);
  }
}

.dapp-core-component__main__btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
}
.dapp-core-component__main__btn .dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn .dapp-core-component__main__disabled {
  opacity: 0.65;
  cursor: default;
}
.dapp-core-component__main__btn:hover {
  color: #212529;
  text-decoration: none;
}
.dapp-core-component__main__btn:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn:disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn-primary {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary:hover {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-primary:focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary:hover {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-secondary:focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-warning {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning:hover {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning:focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-dark {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark:hover {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
}
.dapp-core-component__main__btn-dark:focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-link {
  font-weight: 400;
  color: #007bff;
  text-decoration: none;
}
.dapp-core-component__main__btn-link:hover {
  color: #0056b3;
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:focus, .dapp-core-component__main__btn-link.dapp-core-component__main__focus {
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:disabled, .dapp-core-component__main__btn-link.dapp-core-component__main__disabled {
  color: #6c757d;
  pointer-events: none;
}

.dapp-core-component__main__btn-light {
  color: #212529;
  background-color: #fafafa;
  border-color: #fafafa;
}
.dapp-core-component__main__btn-light .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}
.dapp-core-component__main__btn-light .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #f8f9fa;
  border-color: #f8f9fa;
}
.dapp-core-component__main__btn-light:hover {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
}
.dapp-core-component__main__btn-light:focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .dapp-core-component__main__badge {
    transition: none;
  }
}
a.dapp-core-component__main__badge:hover, a.dapp-core-component__main__badge:focus {
  text-decoration: none;
}

.dapp-core-component__main__badge:empty {
  display: none;
}

.dapp-core-component__main__btn .dapp-core-component__main__badge {
  position: relative;
  top: -1px;
}

.dapp-core-component__main__badge-pill {
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.dapp-core-component__main__badge-primary {
  color: #fff;
  background-color: #007bff;
}

a.dapp-core-component__main__badge-primary:hover, a.dapp-core-component__main__badge-primary:focus {
  color: #fff;
  background-color: #0062cc;
}

a.dapp-core-component__main__badge-primary:focus, a.dapp-core-component__main__badge-primary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.5);
}

.dapp-core-component__main__badge-secondary {
  color: #fff;
  background-color: #6c757d;
}

a.dapp-core-component__main__badge-secondary:hover, a.dapp-core-component__main__badge-secondary:focus {
  color: #fff;
  background-color: #545b62;
}

a.dapp-core-component__main__badge-secondary:focus, a.dapp-core-component__main__badge-secondary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108, 117, 125, 0.5);
}

.dapp-core-component__main__badge-success {
  color: #fff;
  background-color: #28a745;
}

a.dapp-core-component__main__badge-success:hover, a.dapp-core-component__main__badge-success:focus {
  color: #fff;
  background-color: #1e7e34;
}

a.dapp-core-component__main__badge-success:focus, a.dapp-core-component__main__badge-success.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
}

.dapp-core-component__main__badge-info {
  color: #fff;
  background-color: #17a2b8;
}

a.dapp-core-component__main__badge-info:hover, a.dapp-core-component__main__badge-info:focus {
  color: #fff;
  background-color: #117a8b;
}

a.dapp-core-component__main__badge-info:focus, a.dapp-core-component__main__badge-info.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(23, 162, 184, 0.5);
}

.dapp-core-component__main__badge-warning {
  color: #212529;
  background-color: #ffc107;
}

a.dapp-core-component__main__badge-warning:hover, a.dapp-core-component__main__badge-warning:focus {
  color: #212529;
  background-color: #d39e00;
}

a.dapp-core-component__main__badge-warning:focus, a.dapp-core-component__main__badge-warning.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 193, 7, 0.5);
}

.dapp-core-component__main__badge-danger {
  color: #fff;
  background-color: #dc3545;
}

a.dapp-core-component__main__badge-danger:hover, a.dapp-core-component__main__badge-danger:focus {
  color: #fff;
  background-color: #bd2130;
}

a.dapp-core-component__main__badge-danger:focus, a.dapp-core-component__main__badge-danger.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(220, 53, 69, 0.5);
}

.dapp-core-component__main__badge-light {
  color: #212529;
  background-color: #f8f9fa;
}

a.dapp-core-component__main__badge-light:hover, a.dapp-core-component__main__badge-light:focus {
  color: #212529;
  background-color: #dae0e5;
}

a.dapp-core-component__main__badge-light:focus, a.dapp-core-component__main__badge-light.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(248, 249, 250, 0.5);
}

.dapp-core-component__main__badge-dark {
  color: #fff;
  background-color: #343a40;
}

a.dapp-core-component__main__badge-dark:hover, a.dapp-core-component__main__badge-dark:focus {
  color: #fff;
  background-color: #1d2124;
}

a.dapp-core-component__main__badge-dark:focus, a.dapp-core-component__main__badge-dark.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}

.dapp-core-component__main__card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 5px;
}
.dapp-core-component__main__card > hr {
  margin-right: 0;
  margin-left: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group {
  border-top: inherit;
  border-bottom: inherit;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:first-child {
  border-top-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:last-child {
  border-bottom-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__card-header + .dapp-core-component__main__list-group,
.dapp-core-component__main__card > .dapp-core-component__main__list-group + .dapp-core-component__main__card-footer {
  border-top: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header {
  border: 0;
  padding: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header:first-child {
  border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item {
  border-bottom: 1px solid #dee2e6;
  padding: 1.1rem 1rem;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item h6,
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item .dapp-core-component__main__h6 {
  line-height: 2.5rem;
  letter-spacing: unset;
  margin: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item.dapp-core-component__main__compact {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fafafa;
}

.dapp-core-component__main__card-body {
  flex: 1 1 auto;
  min-height: 1px;
  padding: 1.25rem;
}

.dapp-core-component__main__card-title {
  margin-bottom: 0.75rem;
}

.dapp-core-component__main__card-subtitle {
  margin-top: -0.375rem;
  margin-bottom: 0;
}

.dapp-core-component__main__card-text:last-child {
  margin-bottom: 0;
}

.dapp-core-component__main__card-link:hover {
  text-decoration: none;
}
.dapp-core-component__main__card-link + .dapp-core-component__main__card-link {
  margin-left: 1.25rem;
}

.dapp-core-component__main__card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: #fff;
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__card-footer {
  padding: 0.75rem 1.25rem;
  background-color: #fff;
  border-top: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__text {
  color: #9dabbd;
}

.dapp-core-component__main__text-white {
  color: #fff;
}

.dapp-core-component__main__text-monospace {
  font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-size-xl .dapp-core-component__main__trim {
  max-width: 13rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim {
  max-width: 10rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-only-sm .dapp-core-component__main__trim {
  max-width: none;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-left {
  text-align: left;
}

.dapp-core-component__main__text-right {
  text-align: right;
}

.dapp-core-component__main__text-center {
  text-align: center;
}

.dapp-core-component__main__text-primary {
  color: #1b46c2;
}

.dapp-core-component__main__text-secondary {
  color: #6c757d;
}

.dapp-core-component__main__text-primary-highlight {
  color: #1b46c2;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-success {
  color: #28a745;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-body {
  color: #212529;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-black-50 {
  color: rgba(0, 0, 0, 0.5);
}

.dapp-core-component__main__text-white-50 {
  color: rgba(255, 255, 255, 0.5);
}

@media (min-width: 992px) {
  .dapp-core-component__main__text-lg-right {
    text-align: right;
  }
}

.dapp-core-component__main__text-hide {
  font: 0/0 a;
  color: rgba(0, 0, 0, 0);
  text-shadow: none;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
}

.dapp-core-component__main__text-decoration-none {
  text-decoration: none;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__text-reset {
  color: inherit;
}

.dapp-core-component__main__text-lowercase {
  text-transform: lowercase;
}

.dapp-core-component__main__text-uppercase {
  text-transform: uppercase;
}

.dapp-core-component__main__text-capitalize {
  text-transform: capitalize;
}

.dapp-core-component__main__text-danger {
  color: #d9534f;
}

.dapp-core-component__main__text-warning {
  color: #f0ad4e;
}

.dapp-core-component__main__trim-fs-sm {
  color: inherit;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__m-0 {
  margin: 0;
}

.dapp-core-component__main__mt-0 {
  margin-top: 0;
}

.dapp-core-component__main__my-0 {
  margin-top: 0;
  margin-bottom: 0;
}

.dapp-core-component__main__mr-0 {
  margin-right: 0;
}

.dapp-core-component__main__mx-0 {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__mb-0 {
  margin-bottom: 0;
}

.dapp-core-component__main__my-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.dapp-core-component__main__ml-0 {
  margin-left: 0;
}

.dapp-core-component__main__mx-0 {
  margin-left: 0;
  margin-right: 0;
}

.dapp-core-component__main__m-1 {
  margin: 0.25rem;
}

.dapp-core-component__main__mt-1 {
  margin-top: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__mr-1 {
  margin-right: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.dapp-core-component__main__mb-1 {
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-bottom: 0.25rem;
  margin-top: 0.25rem;
}

.dapp-core-component__main__ml-1 {
  margin-left: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-right: 0.25rem;
  margin-left: 0.25rem;
}

.dapp-core-component__main__m-2 {
  margin: 0.5rem;
}

.dapp-core-component__main__mt-2 {
  margin-top: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-bottom: 0.5rem;
  margin-top: 0.5rem;
}

.dapp-core-component__main__mr-2 {
  margin-right: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__mb-2 {
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__ml-2 {
  margin-left: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__m-3 {
  margin: 1rem;
}

.dapp-core-component__main__mt-3,
.dapp-core-component__main__my-3 {
  margin-top: 1rem;
}

.dapp-core-component__main__mr-3 {
  margin-right: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__mb-3 {
  margin-bottom: 1rem;
}

.dapp-core-component__main__my-3 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.dapp-core-component__main__ml-3 {
  margin-left: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__m-4 {
  margin: 1.5rem;
}

.dapp-core-component__main__mt-4 {
  margin-top: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__mr-4 {
  margin-right: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-right: 1.5rem;
  margin-left: 1.5rem;
}

.dapp-core-component__main__mb-4 {
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.dapp-core-component__main__ml-4 {
  margin-left: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}

.dapp-core-component__main__m-5 {
  margin: 3rem;
}

.dapp-core-component__main__mt-5 {
  margin-top: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__mr-5 {
  margin-right: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-right: 3rem;
  margin-left: 3rem;
}

.dapp-core-component__main__mb-5 {
  margin-bottom: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__ml-5 {
  margin-left: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-left: 3rem;
  margin-right: 3rem;
}

.dapp-core-component__main__p-0 {
  padding: 0;
}

.dapp-core-component__main__pt-0 {
  padding-top: 0;
}

.dapp-core-component__main__py-0 {
  padding-top: 0;
  padding-bottom: 0;
}

.dapp-core-component__main__pr-0 {
  padding-right: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__pb-0 {
  padding-bottom: 0;
}

.dapp-core-component__main__py-0 {
  padding-bottom: 0;
  padding-top: 0;
}

.dapp-core-component__main__pl-0 {
  padding-left: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__p-1 {
  padding: 0.25rem;
}

.dapp-core-component__main__pt-1 {
  padding-top: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pr-1 {
  padding-right: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-right: 0.25rem;
  padding-left: 0.25rem;
}

.dapp-core-component__main__pb-1 {
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pl-1 {
  padding-left: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

.dapp-core-component__main__p-2 {
  padding: 0.5rem;
}

.dapp-core-component__main__pt-2 {
  padding-top: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.dapp-core-component__main__pr-2 {
  padding-right: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__pb-2 {
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__pl-2 {
  padding-left: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__p-3 {
  padding: 1rem;
}

.dapp-core-component__main__pt-3 {
  padding-top: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.dapp-core-component__main__pr-3 {
  padding-right: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__pb-3 {
  padding-bottom: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.dapp-core-component__main__pl-3 {
  padding-left: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__p-4 {
  padding: 1.5rem;
}

.dapp-core-component__main__pt-4 {
  padding-top: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pr-4 {
  padding-right: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.dapp-core-component__main__pb-4 {
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pl-4 {
  padding-left: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
}

.dapp-core-component__main__p-5 {
  padding: 3rem;
}

.dapp-core-component__main__pt-5 {
  padding-top: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-bottom: 3rem;
  padding-top: 3rem;
}

.dapp-core-component__main__pr-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__pb-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.dapp-core-component__main__pl-5 {
  padding-left: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__m-n1 {
  margin: -0.25rem;
}

.dapp-core-component__main__mt-n1 {
  margin-top: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__mr-n1 {
  margin-right: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__mb-n1 {
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-bottom: -0.25rem;
  margin-top: -0.25rem;
}

.dapp-core-component__main__ml-n1 {
  margin-left: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__m-n2 {
  margin: -0.5rem;
}

.dapp-core-component__main__mt-n2 {
  margin-top: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-top: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mr-n2 {
  margin-right: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mb-n2 {
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-bottom: -0.5rem;
  margin-top: -0.5rem;
}

.dapp-core-component__main__ml-n2 {
  margin-left: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.dapp-core-component__main__m-n3 {
  margin: -1rem;
}

.dapp-core-component__main__mt-n3 {
  margin-top: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-top: -1rem;
  margin-bottom: -1rem;
}

.dapp-core-component__main__mr-n3 {
  margin-right: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-right: -1rem;
  margin-left: -1rem;
}

.dapp-core-component__main__mb-n3 {
  margin-bottom: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-bottom: -1rem;
  margin-top: -1rem;
}

.dapp-core-component__main__ml-n3 {
  margin-left: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-left: -1rem;
  margin-right: -1rem;
}

.dapp-core-component__main__m-n4 {
  margin: -1.5rem;
}

.dapp-core-component__main__mt-n4 {
  margin-top: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-top: -1.5rem;
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__mr-n4 {
  margin-right: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__mb-n4 {
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-bottom: -1.5rem;
  margin-top: -1.5rem;
}

.dapp-core-component__main__ml-n4 {
  margin-left: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__m-n5 {
  margin: -3rem;
}

.dapp-core-component__main__mt-n5 {
  margin-top: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-bottom: -3rem;
  margin-top: -3rem;
}

.dapp-core-component__main__mr-n5 {
  margin-right: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-left: -3rem;
  margin-right: -3rem;
}

.dapp-core-component__main__mb-n5 {
  margin-bottom: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-top: -3rem;
  margin-bottom: -3rem;
}

.dapp-core-component__main__ml-n5 {
  margin-left: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-right: -3rem;
  margin-left: -3rem;
}

.dapp-core-component__main__m-auto {
  margin: auto;
}

.dapp-core-component__main__mt-auto {
  margin-top: auto;
}

.dapp-core-component__main__my-auto {
  margin-bottom: auto;
  margin-top: auto;
}

.dapp-core-component__main__mr-auto {
  margin-right: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__mb-auto {
  margin-bottom: auto;
}

.dapp-core-component__main__my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.dapp-core-component__main__ml-auto {
  margin-left: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-1 {
    margin-right: 0.25rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-4 {
    margin-right: 1.5rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mx-lg-4 {
    margin-left: 1.5rem;
    margin-right: 1.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__mb-xl-2 {
    margin-bottom: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__pr-sm-2 {
    padding-right: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__px-sm-2 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__pr-xl-0,
.dapp-core-component__main__px-xl-0 {
    padding-right: 0;
  }
}

.dapp-core-component__main__col {
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

.dapp-core-component__main__col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__col-xl,
.dapp-core-component__main__col-xl-auto,
.dapp-core-component__main__col-xl-12,
.dapp-core-component__main__col-xl-11,
.dapp-core-component__main__col-xl-10,
.dapp-core-component__main__col-xl-9,
.dapp-core-component__main__col-xl-8,
.dapp-core-component__main__col-xl-7,
.dapp-core-component__main__col-xl-6,
.dapp-core-component__main__col-xl-5,
.dapp-core-component__main__col-xl-4,
.dapp-core-component__main__col-xl-3,
.dapp-core-component__main__col-xl-2,
.dapp-core-component__main__col-xl-1,
.dapp-core-component__main__col-lg,
.dapp-core-component__main__col-lg-auto,
.dapp-core-component__main__col-lg-12,
.dapp-core-component__main__col-lg-11,
.dapp-core-component__main__col-lg-10,
.dapp-core-component__main__col-lg-9,
.dapp-core-component__main__col-lg-8,
.dapp-core-component__main__col-lg-7,
.dapp-core-component__main__col-lg-6,
.dapp-core-component__main__col-lg-5,
.dapp-core-component__main__col-lg-4,
.dapp-core-component__main__col-lg-3,
.dapp-core-component__main__col-lg-2,
.dapp-core-component__main__col-lg-1,
.dapp-core-component__main__col-md,
.dapp-core-component__main__col-md-auto,
.dapp-core-component__main__col-md-12,
.dapp-core-component__main__col-md-11,
.dapp-core-component__main__col-md-10,
.dapp-core-component__main__col-md-9,
.dapp-core-component__main__col-md-8,
.dapp-core-component__main__col-md-7,
.dapp-core-component__main__col-md-6,
.dapp-core-component__main__col-md-5,
.dapp-core-component__main__col-md-4,
.dapp-core-component__main__col-md-3,
.dapp-core-component__main__col-md-2,
.dapp-core-component__main__col-md-1,
.dapp-core-component__main__col-sm,
.dapp-core-component__main__col-sm-auto,
.dapp-core-component__main__col-sm-12,
.dapp-core-component__main__col-sm-11,
.dapp-core-component__main__col-sm-10,
.dapp-core-component__main__col-sm-9,
.dapp-core-component__main__col-sm-8,
.dapp-core-component__main__col-sm-7,
.dapp-core-component__main__col-sm-6,
.dapp-core-component__main__col-sm-5,
.dapp-core-component__main__col-sm-4,
.dapp-core-component__main__col-sm-3,
.dapp-core-component__main__col-sm-2,
.dapp-core-component__main__col-sm-1,
.dapp-core-component__main__col,
.dapp-core-component__main__col-auto,
.dapp-core-component__main__col-12,
.dapp-core-component__main__col-11,
.dapp-core-component__main__col-10,
.dapp-core-component__main__col-9,
.dapp-core-component__main__col-8,
.dapp-core-component__main__col-7,
.dapp-core-component__main__col-6,
.dapp-core-component__main__col-5,
.dapp-core-component__main__col-4,
.dapp-core-component__main__col-3,
.dapp-core-component__main__col-2,
.dapp-core-component__main__col-1 {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
}

.dapp-core-component__main__col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

@media (min-width: 576px) {
  .dapp-core-component__main__col-sm-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-sm-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-sm-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-sm-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-sm-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-sm-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-sm-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-sm-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-sm-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-sm-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-sm-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-sm-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 768px) {
  .dapp-core-component__main__col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__col-xl-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-xl-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-xl-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-xl-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-xl-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-xl-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-xl-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-xl-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-xl-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-xl-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-xl-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-xl-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
.dapp-core-component__main__flex-row {
  flex-direction: row;
}

.dapp-core-component__main__flex-column {
  flex-direction: column;
}

.dapp-core-component__main__flex-row-reverse {
  flex-direction: row-reverse;
}

.dapp-core-component__main__flex-column-reverse {
  flex-direction: column-reverse;
}

.dapp-core-component__main__flex-wrap {
  flex-wrap: wrap;
}

.dapp-core-component__main__flex-nowrap {
  flex-wrap: nowrap;
}

.dapp-core-component__main__flex-wrap-reverse {
  flex-wrap: wrap-reverse;
}

.dapp-core-component__main__flex-fill {
  flex: 1 1 auto;
}

.dapp-core-component__main__flex-grow-0 {
  flex-grow: 0;
}

.dapp-core-component__main__flex-grow-1 {
  flex-grow: 1;
}

.dapp-core-component__main__flex-shrink-0 {
  flex-shrink: 0;
}

.dapp-core-component__main__flex-shrink-1 {
  flex-shrink: 1;
}

.dapp-core-component__main__justify-content-start {
  justify-content: flex-start;
}

.dapp-core-component__main__justify-content-end {
  justify-content: flex-end;
}

.dapp-core-component__main__justify-content-center {
  justify-content: center;
}

.dapp-core-component__main__justify-content-between {
  justify-content: space-between;
}

.dapp-core-component__main__justify-content-around {
  justify-content: space-around;
}

.dapp-core-component__main__align-items-start {
  align-items: flex-start;
}

.dapp-core-component__main__align-items-end {
  align-items: flex-end;
}

.dapp-core-component__main__align-items-center {
  align-items: center;
}

.dapp-core-component__main__align-items-baseline {
  align-items: baseline;
}

.dapp-core-component__main__align-items-stretch {
  align-items: stretch;
}

.dapp-core-component__main__align-content-start {
  align-content: flex-start;
}

.dapp-core-component__main__align-content-end {
  align-content: flex-end;
}

.dapp-core-component__main__align-content-center {
  align-content: center;
}

.dapp-core-component__main__align-content-between {
  align-content: space-between;
}

.dapp-core-component__main__align-content-around {
  align-content: space-around;
}

.dapp-core-component__main__align-content-stretch {
  align-content: stretch;
}

.dapp-core-component__main__align-self-auto {
  align-self: auto;
}

.dapp-core-component__main__align-self-start {
  align-self: flex-start;
}

.dapp-core-component__main__align-self-end {
  align-self: flex-end;
}

.dapp-core-component__main__align-self-center {
  align-self: center;
}

.dapp-core-component__main__align-self-baseline {
  align-self: baseline;
}

.dapp-core-component__main__align-self-stretch {
  align-self: stretch;
}

@media (min-width: 576px) {
  .dapp-core-component__main__flex-sm-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-wrap {
    flex-wrap: wrap;
  }
}

.dapp-core-component__main__form-check {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: auto;
  padding-left: 0;
}

.dapp-core-component__main__form-check-label {
  margin-bottom: 0;
}

.dapp-core-component__main__form-check-input {
  position: absolute;
  margin-top: 0.3rem;
  margin-left: -1.25rem;
}
.dapp-core-component__main__form-check-input[disabled] ~ .dapp-core-component__main__form-check-label, .dapp-core-component__main__form-check-input:disabled ~ .dapp-core-component__main__form-check-label {
  color: #6c757d;
}

.dapp-core-component__main__form-group {
  display: flex;
  flex: 0 0 auto;
  flex-flow: row wrap;
  align-items: center;
  margin-bottom: 0;
}

.dapp-core-component__main__form-control {
  display: block;
  width: 100%;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.dapp-core-component__main__form-label {
  padding-top: calc(0.375rem + 1px);
  padding-bottom: calc(0.375rem + 1px);
  margin-bottom: 0;
  font-size: inherit;
  line-height: 1.5;
}

input,
button,
select,
optgroup,
textarea {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.dapp-core-component__main__form-group {
  margin-bottom: 1rem;
}

.dapp-core-component__main__form-text {
  display: block;
  margin-top: 0.25rem;
  color: inherit;
}

.dapp-core-component__main__border {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__border-top {
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__border-right {
  border-right: 1px solid #dee2e6;
}

.dapp-core-component__main__border-bottom {
  border-bottom: 1px solid #dee2e6;
}

.dapp-core-component__main__border-left {
  border-left: 1px solid #dee2e6;
}

.dapp-core-component__main__border-0 {
  border: 0;
}

.dapp-core-component__main__border-top-0 {
  border-top: 0;
}

.dapp-core-component__main__border-right-0 {
  border-right: 0;
}

.dapp-core-component__main__border-bottom-0 {
  border-bottom: 0;
}

.dapp-core-component__main__border-left-0 {
  border-left: 0;
}

.dapp-core-component__main__rounded-sm {
  border-radius: 0.2rem;
}

.dapp-core-component__main__rounded {
  border-radius: 0.25rem;
}

.dapp-core-component__main__rounded-top {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-right {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-bottom {
  border-bottom-right-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-left {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-lg {
  border-radius: 0.3rem;
}

.dapp-core-component__main__rounded-circle {
  border-radius: 50%;
}

.dapp-core-component__main__rounded-pill {
  border-radius: 50rem;
}

.dapp-core-component__main__rounded-0 {
  border-radius: 0;
}

.dapp-core-component__main__h1,
.dapp-core-component__main__h2,
.dapp-core-component__main__h3,
.dapp-core-component__main__h4,
.dapp-core-component__main__h5,
.dapp-core-component__main__h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2;
}

.dapp-core-component__main__h1 {
  font-size: 2.5rem;
}

.dapp-core-component__main__h2 {
  font-size: 2rem;
}

.dapp-core-component__main__h3 {
  font-size: 1.75rem;
}

.dapp-core-component__main__h4 {
  font-size: 1.5rem;
}

.dapp-core-component__main__h5 {
  font-size: 1.25rem;
}

.dapp-core-component__main__h6 {
  font-size: 1rem;
}

.dapp-core-component__main__lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

.dapp-core-component__main__bg-warning,
.dapp-core-component__main__warning {
  background-color: #f0ad4e;
}

.dapp-core-component__main__bg-danger,
.dapp-core-component__main__danger {
  background-color: #d9534f;
}

.dapp-core-component__main__bg-success,
.dapp-core-component__main__success {
  background-color: #5cb85c;
}

.dapp-core-component__main__position-relative {
  position: relative;
}

.dapp-core-component__main__position-absolute {
  position: absolute;
}

.dapp-core-component__main__w-25 {
  width: 25%;
}

.dapp-core-component__main__w-50 {
  width: 50%;
}

.dapp-core-component__main__w-75 {
  width: 75%;
}

.dapp-core-component__main__w-100 {
  width: 100%;
}

.dapp-core-component__main__w-auto {
  width: auto;
}

.dapp-core-component__main__h-25 {
  height: 25%;
}

.dapp-core-component__main__h-50 {
  height: 50%;
}

.dapp-core-component__main__h-75 {
  height: 75%;
}

.dapp-core-component__main__h-100 {
  height: 100%;
}

.dapp-core-component__main__h-auto {
  height: auto;
}

.dapp-core-component__main__mw-100 {
  max-width: 100%;
}

.dapp-core-component__main__mh-100 {
  max-height: 100%;
}

.dapp-core-component__main__min-vw-100 {
  min-width: 100vw;
}

.dapp-core-component__main__min-vh-100 {
  min-height: 100vh;
}

.dapp-core-component__main__vw-100 {
  width: 100vw;
}

.dapp-core-component__main__vh-100 {
  height: 100vh;
}

.dapp-core-component__main__locked-icon {
  color: inherit;
}

.dapp-core-component__main__shadow-sm {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
}

.dapp-core-component__main__shadow {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.dapp-core-component__main__shadow-lg {
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.175);
}

.dapp-core-component__main__shadow-none {
  box-shadow: none;
}

.dapp-core-component__main__d-none {
  display: none;
}

.dapp-core-component__main__d-inline {
  display: inline;
}

.dapp-core-component__main__d-inline-block {
  display: inline-block;
}

.dapp-core-component__main__d-block {
  display: block;
}

.dapp-core-component__main__d-table {
  display: table;
}

.dapp-core-component__main__d-table-row {
  display: table-row;
}

.dapp-core-component__main__d-table-cell {
  display: table-cell;
}

.dapp-core-component__main__d-flex {
  display: -ms-flexbox;
  display: flex;
}

.dapp-core-component__main__d-inline-flex {
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media (min-width: 576px) {
  .dapp-core-component__main__d-sm-flex {
    display: flex;
  }
}

.dapp-core-component__main__table {
  width: 100%;
  margin-bottom: 1rem;
  color: #212529;
}
.dapp-core-component__main__table th,
.dapp-core-component__main__table td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #dee2e6;
}
.dapp-core-component__main__table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #dee2e6;
}
.dapp-core-component__main__table tbody + tbody {
  border-top: 2px solid #dee2e6;
}

.dapp-core-component__main__table-sm th,
.dapp-core-component__main__table-sm td {
  padding: 0.3rem;
}

.dapp-core-component__main__table-bordered {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered th,
.dapp-core-component__main__table-bordered td {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered thead th,
.dapp-core-component__main__table-bordered thead td {
  border-bottom-width: 2px;
}

.dapp-core-component__main__table-borderless th,
.dapp-core-component__main__table-borderless td,
.dapp-core-component__main__table-borderless thead th,
.dapp-core-component__main__table-borderless tbody + tbody {
  border: 0;
}

.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(0, 0, 0, 0.05);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-dark th {
  color: #fff;
  background-color: #343a40;
  border-color: #454d55;
}
.dapp-core-component__main__table .dapp-core-component__main__thead-light th {
  color: #495057;
  background-color: #e9ecef;
  border-color: #dee2e6;
}

.dapp-core-component__main__table-dark {
  color: #fff;
  background-color: #343a40;
}
.dapp-core-component__main__table-dark th,
.dapp-core-component__main__table-dark td,
.dapp-core-component__main__table-dark thead th {
  border-color: #454d55;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-bordered {
  border: 0;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(255, 255, 255, 0.05);
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-hover tbody tr {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.075);
}

.dapp-core-component__main__table-responsive {
  display: block;
  width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}
.dapp-core-component__main__table-responsive > .dapp-core-component__main__table-bordered {
  border: 0;
}

@media (max-width: 767.98px) {
  .dapp-core-component__main__transactions.dapp-core-component__main__table .dapp-core-component__main__transaction-hash .dapp-core-component__main__trim {
    max-width: 8rem;
  }
}
.dapp-core-component__main__font-weight-light {
  font-weight: 300 !important;
}

.dapp-core-component__main__font-weight-lighter {
  font-weight: lighter !important;
}

.dapp-core-component__main__font-weight-normal {
  font-weight: 400 !important;
}

.dapp-core-component__main__font-weight-medium {
  font-weight: 500 !important;
}

.dapp-core-component__main__font-weight-bold {
  font-weight: 700 !important;
}

.dapp-core-component__main__font-weight-bolder {
  font-weight: bolder !important;
}

.dapp-core-component__main__font-italic {
  font-style: italic !important;
}

.dapp-core-component__main__icon-state {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
}
.dapp-core-component__main__icon-state.dapp-core-component__main__half {
  width: 3rem;
  height: 3rem;
}

.dapp-core-component__main__side-icon {
  width: 1.5rem;
  height: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  color: transparent;
  border-radius: 50%;
  overflow: hidden;
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__side-icon svg {
  color: #cacbcc;
}

div.dapp-core-component__main__side-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
div.dapp-core-component__main__side-icon img {
  transform: scale(0.75);
}

.dapp-core-component__main__toast-messages {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast {
  display: block;
  flex-basis: 4rem;
  width: 100%;
  max-width: 28.875rem;
  transition: transform 0.3s ease-in-out;
  transform: translateX(120%);
  background-color: #fff;
  border-radius: 0.25rem;
  border-color: #dee2e6;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__clickable {
  cursor: pointer;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible {
  margin-bottom: 0.3rem;
  transform: translateX(0);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress {
  background-color: rgba(0, 0, 0, 0);
  height: inherit;
  font-size: inherit;
  line-height: inherit;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar {
  background-color: rgba(108, 117, 125, 0.2);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar .dapp-core-component__main__content-height {
  z-index: -1;
  visibility: hidden;
  width: 0 !important;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast .dapp-core-component__main__tx-description {
  margin-bottom: 0.2rem;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close {
  opacity: 1;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close svg {
  color: #6c757d;
}

.dapp-core-component__main__data-field .dapp-core-component__main__textarea {
  -moz-appearance: textfield-multiline;
  -webkit-appearance: textarea;
  border: 1px solid #000;
  background-color: #fff;
  border-radius: 5px;
  font: medium -moz-fixed;
  font-size: 0.8125rem;
  min-height: 5rem;
  overflow: auto;
  padding: 0.375rem 0.75rem;
  resize: both;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea:hover {
  border: 1px solid #000;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea .dapp-core-component__main__highlighted {
  font-size: 1rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea.dapp-core-component__main__sc-call {
  min-height: 3rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__form-control:disabled,
.dapp-core-component__main__data-field .dapp-core-component__main__form-control[readonly] {
  background-color: #fafafa;
}
.dapp-core-component__main__data-field .dapp-core-component__main__small-font {
  font-size: 0.8125rem;
}

.dapp-core-component__main__progress-steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps hr {
  background-color: #6c757d;
  border-color: #dee2e6;
  z-index: 2;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot {
  height: 0.625rem;
  width: 0.625rem;
  border-radius: 50%;
  margin: 0 1rem;
  z-index: 5;
  background-color: #6c757d;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:first-of-type {
  margin-left: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:last-of-type {
  margin-right: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

a.dapp-core-component__main__side-link::after {
  right: 0;
  left: unset;
  width: calc(100% - 1.85rem);
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Dp));k();A();M();t();var Pp=y(require("bignumber.js"));k();t();var Rp=require("@elrondnetwork/erdjs"),Mp=y(require("bignumber.js"));k();I();me();t();t();t();t();t();t();t();t();t();var da=require("react");t();t();var Op=require("react"),Bp=require("@elrondnetwork/erdjs-extension-provider");we();F();A();I();ve();t();var Up=require("react");we();A();t();var ua=require("react"),Wp=require("@elrondnetwork/erdjs-hw-provider");we();F();A();I();ve();t();var Dn=require("react"),Vp=require("@elrondnetwork/erdjs-wallet-connect-provider");F();A();I();ve();t();var ga=require("react"),Hp=require("@elrondnetwork/erdjs-wallet-connect-provider");F();A();I();ve();t();t();var fa=require("react");t();var qp=require("react");t();var Cn=require("react"),zp=require("socket.io-client");A();t();t();t();t();var ha=e=>e.dappModal,Kp=u(ha,e=>e.visible),bo=u(ha,e=>e.modal),nA=u(bo,e=>e.showHeader),tA=u(bo,e=>e.showFooter),rA=u(bo,e=>e.headerText),aA=u(bo,e=>e.footerText);M();me();Ye();Fo();t();var Xp=`.dapp-core-component__styles__content {
  display: flex;
  align-items: center;
}
.dapp-core-component__styles__content .dapp-core-component__styles__left {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  padding-left: 0.5rem;
  padding-right: 0.75rem;
}
.dapp-core-component__styles__content .dapp-core-component__styles__left .dapp-core-component__styles__warning {
  background-color: #f0ad4e;
}
.dapp-core-component__styles__content .dapp-core-component__styles__left .dapp-core-component__styles__danger {
  background-color: #d9534f;
}
.dapp-core-component__styles__content .dapp-core-component__styles__left .dapp-core-component__styles__success {
  background-color: #5cb85c;
}
.dapp-core-component__styles__content .dapp-core-component__styles__left .dapp-core-component__styles__icon {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 3rem;
  height: 3rem;
  border-radius: 50%;
}
.dapp-core-component__styles__content .dapp-core-component__styles__left .dapp-core-component__styles__icon .dapp-core-component__styles__svg {
  height: auto;
  width: 20px;
}
.dapp-core-component__styles__content .dapp-core-component__styles__left .dapp-core-component__styles__icon .dapp-core-component__styles__svg path {
  fill: #fff;
}
.dapp-core-component__styles__content .dapp-core-component__styles__right {
  display: block;
  overflow: hidden;
  width: 100%;
}
.dapp-core-component__styles__content .dapp-core-component__styles__right .dapp-core-component__styles__heading {
  display: flex;
  justify-content: space-between;
  margin-bottom: 0.25rem;
}
.dapp-core-component__styles__content .dapp-core-component__styles__right .dapp-core-component__styles__heading .dapp-core-component__styles__title {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  margin: 0;
  font-weight: 400 !important;
}
.dapp-core-component__styles__content .dapp-core-component__styles__right .dapp-core-component__styles__heading .dapp-core-component__styles__close {
  cursor: pointer;
  float: right;
  font-size: 1.5rem;
  font-weight: 700;
  line-height: 1;
  color: #000;
  text-shadow: 0 1px 0 #fff;
  opacity: 0.5;
  padding: 0;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
  display: flex;
  align-items: center;
  margin: 0 0.5rem;
}
.dapp-core-component__styles__content .dapp-core-component__styles__right .dapp-core-component__styles__heading .dapp-core-component__styles__footer {
  display: block;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Xp));var Ta={content:"dapp-core-component__styles__content",left:"dapp-core-component__styles__left",warning:"dapp-core-component__styles__warning",danger:"dapp-core-component__styles__danger",success:"dapp-core-component__styles__success",icon:"dapp-core-component__styles__icon",svg:"dapp-core-component__styles__svg",right:"dapp-core-component__styles__right",heading:"dapp-core-component__styles__heading",title:"dapp-core-component__styles__title",close:"dapp-core-component__styles__close",footer:"dapp-core-component__styles__footer"};t();t();var ge=require("@fortawesome/free-solid-svg-icons");M();var xa=({status:e,toastId:o,classes:n={success:"success",danger:"danger",warning:"warning"},transactionDisplayInfo:r})=>{var d,g,x,f;let p={id:o,icon:ge.faCheck,expires:3e4,hasCloseButton:!0,title:(d=r==null?void 0:r.successMessage)!=null?d:"Transaction successful",iconClassName:n.success},s={id:o,expires:!1,icon:ge.faHourglass,hasCloseButton:!1,title:(g=r==null?void 0:r.processingMessage)!=null?g:"Processing transaction",iconClassName:n.warning},m={id:o,icon:ge.faTimes,title:(x=r==null?void 0:r.errorMessage)!=null?x:"Transaction failed",hasCloseButton:!0,iconClassName:n.danger},l={id:o,icon:ge.faTimes,title:(f=r==null?void 0:r.timedOutMessage)!=null?f:"Transaction timed out",hasCloseButton:!0,iconClassName:n.warning};switch(e){case"signed":case"sent":return s;case"success":return p;case"cancelled":case"fail":return m;case"timedOut":return l;default:return s}};var ya=6e3,Yp=5,Zp=({toastId:e,transactions:o,status:n,lifetimeAfterSuccess:r,startTimestamp:p,endTimeProgress:s,onDelete:m})=>{let l=na(e),d=S(Cr),x=(0,Ve.useMemo)(()=>Gr(o,d),[o,d])?ya:Yp*ya,f=(l==null?void 0:l.transactionDuration)||x,[D,C]=(0,Ve.useMemo)(()=>{let wo=p||xe(),Sa=s||rt(f);return[wo,Sa]},[]),V={startTime:D,endTime:C},G=ne(n),ba=Ue(n),wa=xa({status:n,toastId:e,transactionDisplayInfo:l,classes:Ta}),Pn=()=>{m==null||m(e)};return(0,Ve.useEffect)(()=>{if(n!=="success"||!r)return;let wo=setTimeout(()=>{Pn()},r);return()=>{clearTimeout(wo)}},[r,n]),{progress:V,isPending:G,isTimedOut:ba,toastDataState:wa,handleDeleteToast:Pn}};0&&(module.exports={useTransactionToast});
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
