// @elrondnetwork/dapp-core/__commonjs/hooks/index.js
"use strict";var Hc=Object.create;var Eo=Object.defineProperty,jc=Object.defineProperties,zc=Object.getOwnPropertyDescriptor,qc=Object.getOwnPropertyDescriptors,Kc=Object.getOwnPropertyNames,Zo=Object.getOwnPropertySymbols,Xc=Object.getPrototypeOf,qn=Object.prototype.hasOwnProperty,hr=Object.prototype.propertyIsEnumerable;var fr=(e,o,n)=>o in e?Eo(e,o,{enumerable:!0,configurable:!0,writable:!0,value:n}):e[o]=n,b=(e,o)=>{for(var n in o||(o={}))qn.call(o,n)&&fr(e,n,o[n]);if(Zo)for(var n of Zo(o))hr.call(o,n)&&fr(e,n,o[n]);return e},P=(e,o)=>jc(e,qc(o));var Kn=(e,o)=>{var n={};for(var r in e)qn.call(e,r)&&o.indexOf(r)<0&&(n[r]=e[r]);if(e!=null&&Zo)for(var r of Zo(e))o.indexOf(r)<0&&hr.call(e,r)&&(n[r]=e[r]);return n};var k=(e,o)=>()=>(e&&(o=e(e=0)),o);var Jo=(e,o)=>()=>(o||e((o={exports:{}}).exports,o),o.exports),je=(e,o)=>{for(var n in o)Eo(e,n,{get:o[n],enumerable:!0})},Tr=(e,o,n,r)=>{if(o&&typeof o=="object"||typeof o=="function")for(let a of Kc(o))!qn.call(e,a)&&a!==n&&Eo(e,a,{get:()=>o[a],enumerable:!(r=zc(o,a))||r.enumerable});return e};var M=(e,o,n)=>(n=e!=null?Hc(Xc(e)):{},Tr(o||!e||!e.__esModule?Eo(n,"default",{value:e,enumerable:!0}):n,e)),Lo=e=>Tr(Eo({},"__esModule",{value:!0}),e);var y=(e,o,n)=>new Promise((r,a)=>{var m=d=>{try{_(n.next(d))}catch(u){a(u)}},i=d=>{try{_(n.throw(d))}catch(u){a(u)}},_=d=>d.done?r(d.value):Promise.resolve(d.value).then(m,i);_((n=n.apply(e,o)).next())});var br=Jo(Qo=>{"use strict";t();Qo.byteLength=Zc;Qo.toByteArray=Qc;Qo.fromByteArray=np;var Te=[],de=[],Yc=typeof Uint8Array!="undefined"?Uint8Array:Array,Xn="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";for(ze=0,xr=Xn.length;ze<xr;++ze)Te[ze]=Xn[ze],de[Xn.charCodeAt(ze)]=ze;var ze,xr;de["-".charCodeAt(0)]=62;de["_".charCodeAt(0)]=63;function yr(e){var o=e.length;if(o%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var n=e.indexOf("=");n===-1&&(n=o);var r=n===o?0:4-n%4;return[n,r]}function Zc(e){var o=yr(e),n=o[0],r=o[1];return(n+r)*3/4-r}function Jc(e,o,n){return(o+n)*3/4-n}function Qc(e){var o,n=yr(e),r=n[0],a=n[1],m=new Yc(Jc(e,r,a)),i=0,_=a>0?r-4:r,d;for(d=0;d<_;d+=4)o=de[e.charCodeAt(d)]<<18|de[e.charCodeAt(d+1)]<<12|de[e.charCodeAt(d+2)]<<6|de[e.charCodeAt(d+3)],m[i++]=o>>16&255,m[i++]=o>>8&255,m[i++]=o&255;return a===2&&(o=de[e.charCodeAt(d)]<<2|de[e.charCodeAt(d+1)]>>4,m[i++]=o&255),a===1&&(o=de[e.charCodeAt(d)]<<10|de[e.charCodeAt(d+1)]<<4|de[e.charCodeAt(d+2)]>>2,m[i++]=o>>8&255,m[i++]=o&255),m}function ep(e){return Te[e>>18&63]+Te[e>>12&63]+Te[e>>6&63]+Te[e&63]}function op(e,o,n){for(var r,a=[],m=o;m<n;m+=3)r=(e[m]<<16&16711680)+(e[m+1]<<8&65280)+(e[m+2]&255),a.push(ep(r));return a.join("")}function np(e){for(var o,n=e.length,r=n%3,a=[],m=16383,i=0,_=n-r;i<_;i+=m)a.push(op(e,i,i+m>_?_:i+m));return r===1?(o=e[n-1],a.push(Te[o>>2]+Te[o<<4&63]+"==")):r===2&&(o=(e[n-2]<<8)+e[n-1],a.push(Te[o>>10]+Te[o>>4&63]+Te[o<<2&63]+"=")),a.join("")}});var wr=Jo(Yn=>{t();Yn.read=function(e,o,n,r,a){var m,i,_=a*8-r-1,d=(1<<_)-1,u=d>>1,g=-7,f=n?a-1:0,T=n?-1:1,h=e[o+f];for(f+=T,m=h&(1<<-g)-1,h>>=-g,g+=_;g>0;m=m*256+e[o+f],f+=T,g-=8);for(i=m&(1<<-g)-1,m>>=-g,g+=r;g>0;i=i*256+e[o+f],f+=T,g-=8);if(m===0)m=1-u;else{if(m===d)return i?NaN:(h?-1:1)*(1/0);i=i+Math.pow(2,r),m=m-u}return(h?-1:1)*i*Math.pow(2,m-r)};Yn.write=function(e,o,n,r,a,m){var i,_,d,u=m*8-a-1,g=(1<<u)-1,f=g>>1,T=a===23?Math.pow(2,-24)-Math.pow(2,-77):0,h=r?0:m-1,E=r?1:-1,F=o<0||o===0&&1/o<0?1:0;for(o=Math.abs(o),isNaN(o)||o===1/0?(_=isNaN(o)?1:0,i=g):(i=Math.floor(Math.log(o)/Math.LN2),o*(d=Math.pow(2,-i))<1&&(i--,d*=2),i+f>=1?o+=T/d:o+=T*Math.pow(2,1-f),o*d>=2&&(i++,d/=2),i+f>=g?(_=0,i=g):i+f>=1?(_=(o*d-1)*Math.pow(2,a),i=i+f):(_=o*Math.pow(2,f-1)*Math.pow(2,a),i=0));a>=8;e[n+h]=_&255,h+=E,_/=256,a-=8);for(i=i<<a|_,u+=a;u>0;e[n+h]=i&255,h+=E,i/=256,u-=8);e[n+h-E]|=F*128}});var Mr=Jo(ao=>{"use strict";t();var Zn=br(),ro=wr(),Sr=typeof Symbol=="function"&&typeof Symbol.for=="function"?Symbol.for("nodejs.util.inspect.custom"):null;ao.Buffer=l;ao.SlowBuffer=pp;ao.INSPECT_MAX_BYTES=50;var en=2147483647;ao.kMaxLength=en;l.TYPED_ARRAY_SUPPORT=tp();!l.TYPED_ARRAY_SUPPORT&&typeof console!="undefined"&&typeof console.error=="function"&&console.error("This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.");function tp(){try{var e=new Uint8Array(1),o={foo:function(){return 42}};return Object.setPrototypeOf(o,Uint8Array.prototype),Object.setPrototypeOf(e,o),e.foo()===42}catch(n){return!1}}Object.defineProperty(l.prototype,"parent",{enumerable:!0,get:function(){if(!!l.isBuffer(this))return this.buffer}});Object.defineProperty(l.prototype,"offset",{enumerable:!0,get:function(){if(!!l.isBuffer(this))return this.byteOffset}});function ke(e){if(e>en)throw new RangeError('The value "'+e+'" is invalid for option "size"');var o=new Uint8Array(e);return Object.setPrototypeOf(o,l.prototype),o}function l(e,o,n){if(typeof e=="number"){if(typeof o=="string")throw new TypeError('The "string" argument must be of type string. Received type number');return ot(e)}return Ir(e,o,n)}l.poolSize=8192;function Ir(e,o,n){if(typeof e=="string")return ap(e,o);if(ArrayBuffer.isView(e))return ip(e);if(e==null)throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e);if(xe(e,ArrayBuffer)||e&&xe(e.buffer,ArrayBuffer)||typeof SharedArrayBuffer!="undefined"&&(xe(e,SharedArrayBuffer)||e&&xe(e.buffer,SharedArrayBuffer)))return Qn(e,o,n);if(typeof e=="number")throw new TypeError('The "value" argument must not be of type number. Received type number');var r=e.valueOf&&e.valueOf();if(r!=null&&r!==e)return l.from(r,o,n);var a=cp(e);if(a)return a;if(typeof Symbol!="undefined"&&Symbol.toPrimitive!=null&&typeof e[Symbol.toPrimitive]=="function")return l.from(e[Symbol.toPrimitive]("string"),o,n);throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e)}l.from=function(e,o,n){return Ir(e,o,n)};Object.setPrototypeOf(l.prototype,Uint8Array.prototype);Object.setPrototypeOf(l,Uint8Array);function Ar(e){if(typeof e!="number")throw new TypeError('"size" argument must be of type number');if(e<0)throw new RangeError('The value "'+e+'" is invalid for option "size"')}function rp(e,o,n){return Ar(e),e<=0?ke(e):o!==void 0?typeof n=="string"?ke(e).fill(o,n):ke(e).fill(o):ke(e)}l.alloc=function(e,o,n){return rp(e,o,n)};function ot(e){return Ar(e),ke(e<0?0:nt(e)|0)}l.allocUnsafe=function(e){return ot(e)};l.allocUnsafeSlow=function(e){return ot(e)};function ap(e,o){if((typeof o!="string"||o==="")&&(o="utf8"),!l.isEncoding(o))throw new TypeError("Unknown encoding: "+o);var n=Er(e,o)|0,r=ke(n),a=r.write(e,o);return a!==n&&(r=r.slice(0,a)),r}function Jn(e){for(var o=e.length<0?0:nt(e.length)|0,n=ke(o),r=0;r<o;r+=1)n[r]=e[r]&255;return n}function ip(e){if(xe(e,Uint8Array)){var o=new Uint8Array(e);return Qn(o.buffer,o.byteOffset,o.byteLength)}return Jn(e)}function Qn(e,o,n){if(o<0||e.byteLength<o)throw new RangeError('"offset" is outside of buffer bounds');if(e.byteLength<o+(n||0))throw new RangeError('"length" is outside of buffer bounds');var r;return o===void 0&&n===void 0?r=new Uint8Array(e):n===void 0?r=new Uint8Array(e,o):r=new Uint8Array(e,o,n),Object.setPrototypeOf(r,l.prototype),r}function cp(e){if(l.isBuffer(e)){var o=nt(e.length)|0,n=ke(o);return n.length===0||e.copy(n,0,0,o),n}if(e.length!==void 0)return typeof e.length!="number"||tt(e.length)?ke(0):Jn(e);if(e.type==="Buffer"&&Array.isArray(e.data))return Jn(e.data)}function nt(e){if(e>=en)throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x"+en.toString(16)+" bytes");return e|0}function pp(e){return+e!=e&&(e=0),l.alloc(+e)}l.isBuffer=function(o){return o!=null&&o._isBuffer===!0&&o!==l.prototype};l.compare=function(o,n){if(xe(o,Uint8Array)&&(o=l.from(o,o.offset,o.byteLength)),xe(n,Uint8Array)&&(n=l.from(n,n.offset,n.byteLength)),!l.isBuffer(o)||!l.isBuffer(n))throw new TypeError('The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array');if(o===n)return 0;for(var r=o.length,a=n.length,m=0,i=Math.min(r,a);m<i;++m)if(o[m]!==n[m]){r=o[m],a=n[m];break}return r<a?-1:a<r?1:0};l.isEncoding=function(o){switch(String(o).toLowerCase()){case"hex":case"utf8":case"utf-8":case"ascii":case"latin1":case"binary":case"base64":case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return!0;default:return!1}};l.concat=function(o,n){if(!Array.isArray(o))throw new TypeError('"list" argument must be an Array of Buffers');if(o.length===0)return l.alloc(0);var r;if(n===void 0)for(n=0,r=0;r<o.length;++r)n+=o[r].length;var a=l.allocUnsafe(n),m=0;for(r=0;r<o.length;++r){var i=o[r];if(xe(i,Uint8Array))m+i.length>a.length?l.from(i).copy(a,m):Uint8Array.prototype.set.call(a,i,m);else if(l.isBuffer(i))i.copy(a,m);else throw new TypeError('"list" argument must be an Array of Buffers');m+=i.length}return a};function Er(e,o){if(l.isBuffer(e))return e.length;if(ArrayBuffer.isView(e)||xe(e,ArrayBuffer))return e.byteLength;if(typeof e!="string")throw new TypeError('The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type '+typeof e);var n=e.length,r=arguments.length>2&&arguments[2]===!0;if(!r&&n===0)return 0;for(var a=!1;;)switch(o){case"ascii":case"latin1":case"binary":return n;case"utf8":case"utf-8":return et(e).length;case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return n*2;case"hex":return n>>>1;case"base64":return Rr(e).length;default:if(a)return r?-1:et(e).length;o=(""+o).toLowerCase(),a=!0}}l.byteLength=Er;function sp(e,o,n){var r=!1;if((o===void 0||o<0)&&(o=0),o>this.length||((n===void 0||n>this.length)&&(n=this.length),n<=0)||(n>>>=0,o>>>=0,n<=o))return"";for(e||(e="utf8");;)switch(e){case"hex":return xp(this,o,n);case"utf8":case"utf-8":return Dr(this,o,n);case"ascii":return hp(this,o,n);case"latin1":case"binary":return Tp(this,o,n);case"base64":return gp(this,o,n);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return yp(this,o,n);default:if(r)throw new TypeError("Unknown encoding: "+e);e=(e+"").toLowerCase(),r=!0}}l.prototype._isBuffer=!0;function qe(e,o,n){var r=e[o];e[o]=e[n],e[n]=r}l.prototype.swap16=function(){var o=this.length;if(o%2!==0)throw new RangeError("Buffer size must be a multiple of 16-bits");for(var n=0;n<o;n+=2)qe(this,n,n+1);return this};l.prototype.swap32=function(){var o=this.length;if(o%4!==0)throw new RangeError("Buffer size must be a multiple of 32-bits");for(var n=0;n<o;n+=4)qe(this,n,n+3),qe(this,n+1,n+2);return this};l.prototype.swap64=function(){var o=this.length;if(o%8!==0)throw new RangeError("Buffer size must be a multiple of 64-bits");for(var n=0;n<o;n+=8)qe(this,n,n+7),qe(this,n+1,n+6),qe(this,n+2,n+5),qe(this,n+3,n+4);return this};l.prototype.toString=function(){var o=this.length;return o===0?"":arguments.length===0?Dr(this,0,o):sp.apply(this,arguments)};l.prototype.toLocaleString=l.prototype.toString;l.prototype.equals=function(o){if(!l.isBuffer(o))throw new TypeError("Argument must be a Buffer");return this===o?!0:l.compare(this,o)===0};l.prototype.inspect=function(){var o="",n=ao.INSPECT_MAX_BYTES;return o=this.toString("hex",0,n).replace(/(.{2})/g,"$1 ").trim(),this.length>n&&(o+=" ... "),"<Buffer "+o+">"};Sr&&(l.prototype[Sr]=l.prototype.inspect);l.prototype.compare=function(o,n,r,a,m){if(xe(o,Uint8Array)&&(o=l.from(o,o.offset,o.byteLength)),!l.isBuffer(o))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+typeof o);if(n===void 0&&(n=0),r===void 0&&(r=o?o.length:0),a===void 0&&(a=0),m===void 0&&(m=this.length),n<0||r>o.length||a<0||m>this.length)throw new RangeError("out of range index");if(a>=m&&n>=r)return 0;if(a>=m)return-1;if(n>=r)return 1;if(n>>>=0,r>>>=0,a>>>=0,m>>>=0,this===o)return 0;for(var i=m-a,_=r-n,d=Math.min(i,_),u=this.slice(a,m),g=o.slice(n,r),f=0;f<d;++f)if(u[f]!==g[f]){i=u[f],_=g[f];break}return i<_?-1:_<i?1:0};function Lr(e,o,n,r,a){if(e.length===0)return-1;if(typeof n=="string"?(r=n,n=0):n>2147483647?n=2147483647:n<-2147483648&&(n=-2147483648),n=+n,tt(n)&&(n=a?0:e.length-1),n<0&&(n=e.length+n),n>=e.length){if(a)return-1;n=e.length-1}else if(n<0)if(a)n=0;else return-1;if(typeof o=="string"&&(o=l.from(o,r)),l.isBuffer(o))return o.length===0?-1:vr(e,o,n,r,a);if(typeof o=="number")return o=o&255,typeof Uint8Array.prototype.indexOf=="function"?a?Uint8Array.prototype.indexOf.call(e,o,n):Uint8Array.prototype.lastIndexOf.call(e,o,n):vr(e,[o],n,r,a);throw new TypeError("val must be string, number or Buffer")}function vr(e,o,n,r,a){var m=1,i=e.length,_=o.length;if(r!==void 0&&(r=String(r).toLowerCase(),r==="ucs2"||r==="ucs-2"||r==="utf16le"||r==="utf-16le")){if(e.length<2||o.length<2)return-1;m=2,i/=2,_/=2,n/=2}function d(h,E){return m===1?h[E]:h.readUInt16BE(E*m)}var u;if(a){var g=-1;for(u=n;u<i;u++)if(d(e,u)===d(o,g===-1?0:u-g)){if(g===-1&&(g=u),u-g+1===_)return g*m}else g!==-1&&(u-=u-g),g=-1}else for(n+_>i&&(n=i-_),u=n;u>=0;u--){for(var f=!0,T=0;T<_;T++)if(d(e,u+T)!==d(o,T)){f=!1;break}if(f)return u}return-1}l.prototype.includes=function(o,n,r){return this.indexOf(o,n,r)!==-1};l.prototype.indexOf=function(o,n,r){return Lr(this,o,n,r,!0)};l.prototype.lastIndexOf=function(o,n,r){return Lr(this,o,n,r,!1)};function mp(e,o,n,r){n=Number(n)||0;var a=e.length-n;r?(r=Number(r),r>a&&(r=a)):r=a;var m=o.length;r>m/2&&(r=m/2);for(var i=0;i<r;++i){var _=parseInt(o.substr(i*2,2),16);if(tt(_))return i;e[n+i]=_}return i}function _p(e,o,n,r){return on(et(o,e.length-n),e,n,r)}function dp(e,o,n,r){return on(Sp(o),e,n,r)}function lp(e,o,n,r){return on(Rr(o),e,n,r)}function up(e,o,n,r){return on(vp(o,e.length-n),e,n,r)}l.prototype.write=function(o,n,r,a){if(n===void 0)a="utf8",r=this.length,n=0;else if(r===void 0&&typeof n=="string")a=n,r=this.length,n=0;else if(isFinite(n))n=n>>>0,isFinite(r)?(r=r>>>0,a===void 0&&(a="utf8")):(a=r,r=void 0);else throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");var m=this.length-n;if((r===void 0||r>m)&&(r=m),o.length>0&&(r<0||n<0)||n>this.length)throw new RangeError("Attempt to write outside buffer bounds");a||(a="utf8");for(var i=!1;;)switch(a){case"hex":return mp(this,o,n,r);case"utf8":case"utf-8":return _p(this,o,n,r);case"ascii":case"latin1":case"binary":return dp(this,o,n,r);case"base64":return lp(this,o,n,r);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return up(this,o,n,r);default:if(i)throw new TypeError("Unknown encoding: "+a);a=(""+a).toLowerCase(),i=!0}};l.prototype.toJSON=function(){return{type:"Buffer",data:Array.prototype.slice.call(this._arr||this,0)}};function gp(e,o,n){return o===0&&n===e.length?Zn.fromByteArray(e):Zn.fromByteArray(e.slice(o,n))}function Dr(e,o,n){n=Math.min(e.length,n);for(var r=[],a=o;a<n;){var m=e[a],i=null,_=m>239?4:m>223?3:m>191?2:1;if(a+_<=n){var d,u,g,f;switch(_){case 1:m<128&&(i=m);break;case 2:d=e[a+1],(d&192)===128&&(f=(m&31)<<6|d&63,f>127&&(i=f));break;case 3:d=e[a+1],u=e[a+2],(d&192)===128&&(u&192)===128&&(f=(m&15)<<12|(d&63)<<6|u&63,f>2047&&(f<55296||f>57343)&&(i=f));break;case 4:d=e[a+1],u=e[a+2],g=e[a+3],(d&192)===128&&(u&192)===128&&(g&192)===128&&(f=(m&15)<<18|(d&63)<<12|(u&63)<<6|g&63,f>65535&&f<1114112&&(i=f))}}i===null?(i=65533,_=1):i>65535&&(i-=65536,r.push(i>>>10&1023|55296),i=56320|i&1023),r.push(i),a+=_}return fp(r)}var kr=4096;function fp(e){var o=e.length;if(o<=kr)return String.fromCharCode.apply(String,e);for(var n="",r=0;r<o;)n+=String.fromCharCode.apply(String,e.slice(r,r+=kr));return n}function hp(e,o,n){var r="";n=Math.min(e.length,n);for(var a=o;a<n;++a)r+=String.fromCharCode(e[a]&127);return r}function Tp(e,o,n){var r="";n=Math.min(e.length,n);for(var a=o;a<n;++a)r+=String.fromCharCode(e[a]);return r}function xp(e,o,n){var r=e.length;(!o||o<0)&&(o=0),(!n||n<0||n>r)&&(n=r);for(var a="",m=o;m<n;++m)a+=kp[e[m]];return a}function yp(e,o,n){for(var r=e.slice(o,n),a="",m=0;m<r.length-1;m+=2)a+=String.fromCharCode(r[m]+r[m+1]*256);return a}l.prototype.slice=function(o,n){var r=this.length;o=~~o,n=n===void 0?r:~~n,o<0?(o+=r,o<0&&(o=0)):o>r&&(o=r),n<0?(n+=r,n<0&&(n=0)):n>r&&(n=r),n<o&&(n=o);var a=this.subarray(o,n);return Object.setPrototypeOf(a,l.prototype),a};function Y(e,o,n){if(e%1!==0||e<0)throw new RangeError("offset is not uint");if(e+o>n)throw new RangeError("Trying to access beyond buffer length")}l.prototype.readUintLE=l.prototype.readUIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||Y(o,n,this.length);for(var a=this[o],m=1,i=0;++i<n&&(m*=256);)a+=this[o+i]*m;return a};l.prototype.readUintBE=l.prototype.readUIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||Y(o,n,this.length);for(var a=this[o+--n],m=1;n>0&&(m*=256);)a+=this[o+--n]*m;return a};l.prototype.readUint8=l.prototype.readUInt8=function(o,n){return o=o>>>0,n||Y(o,1,this.length),this[o]};l.prototype.readUint16LE=l.prototype.readUInt16LE=function(o,n){return o=o>>>0,n||Y(o,2,this.length),this[o]|this[o+1]<<8};l.prototype.readUint16BE=l.prototype.readUInt16BE=function(o,n){return o=o>>>0,n||Y(o,2,this.length),this[o]<<8|this[o+1]};l.prototype.readUint32LE=l.prototype.readUInt32LE=function(o,n){return o=o>>>0,n||Y(o,4,this.length),(this[o]|this[o+1]<<8|this[o+2]<<16)+this[o+3]*16777216};l.prototype.readUint32BE=l.prototype.readUInt32BE=function(o,n){return o=o>>>0,n||Y(o,4,this.length),this[o]*16777216+(this[o+1]<<16|this[o+2]<<8|this[o+3])};l.prototype.readIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||Y(o,n,this.length);for(var a=this[o],m=1,i=0;++i<n&&(m*=256);)a+=this[o+i]*m;return m*=128,a>=m&&(a-=Math.pow(2,8*n)),a};l.prototype.readIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||Y(o,n,this.length);for(var a=n,m=1,i=this[o+--a];a>0&&(m*=256);)i+=this[o+--a]*m;return m*=128,i>=m&&(i-=Math.pow(2,8*n)),i};l.prototype.readInt8=function(o,n){return o=o>>>0,n||Y(o,1,this.length),this[o]&128?(255-this[o]+1)*-1:this[o]};l.prototype.readInt16LE=function(o,n){o=o>>>0,n||Y(o,2,this.length);var r=this[o]|this[o+1]<<8;return r&32768?r|4294901760:r};l.prototype.readInt16BE=function(o,n){o=o>>>0,n||Y(o,2,this.length);var r=this[o+1]|this[o]<<8;return r&32768?r|4294901760:r};l.prototype.readInt32LE=function(o,n){return o=o>>>0,n||Y(o,4,this.length),this[o]|this[o+1]<<8|this[o+2]<<16|this[o+3]<<24};l.prototype.readInt32BE=function(o,n){return o=o>>>0,n||Y(o,4,this.length),this[o]<<24|this[o+1]<<16|this[o+2]<<8|this[o+3]};l.prototype.readFloatLE=function(o,n){return o=o>>>0,n||Y(o,4,this.length),ro.read(this,o,!0,23,4)};l.prototype.readFloatBE=function(o,n){return o=o>>>0,n||Y(o,4,this.length),ro.read(this,o,!1,23,4)};l.prototype.readDoubleLE=function(o,n){return o=o>>>0,n||Y(o,8,this.length),ro.read(this,o,!0,52,8)};l.prototype.readDoubleBE=function(o,n){return o=o>>>0,n||Y(o,8,this.length),ro.read(this,o,!1,52,8)};function ne(e,o,n,r,a,m){if(!l.isBuffer(e))throw new TypeError('"buffer" argument must be a Buffer instance');if(o>a||o<m)throw new RangeError('"value" argument is out of bounds');if(n+r>e.length)throw new RangeError("Index out of range")}l.prototype.writeUintLE=l.prototype.writeUIntLE=function(o,n,r,a){if(o=+o,n=n>>>0,r=r>>>0,!a){var m=Math.pow(2,8*r)-1;ne(this,o,n,r,m,0)}var i=1,_=0;for(this[n]=o&255;++_<r&&(i*=256);)this[n+_]=o/i&255;return n+r};l.prototype.writeUintBE=l.prototype.writeUIntBE=function(o,n,r,a){if(o=+o,n=n>>>0,r=r>>>0,!a){var m=Math.pow(2,8*r)-1;ne(this,o,n,r,m,0)}var i=r-1,_=1;for(this[n+i]=o&255;--i>=0&&(_*=256);)this[n+i]=o/_&255;return n+r};l.prototype.writeUint8=l.prototype.writeUInt8=function(o,n,r){return o=+o,n=n>>>0,r||ne(this,o,n,1,255,0),this[n]=o&255,n+1};l.prototype.writeUint16LE=l.prototype.writeUInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||ne(this,o,n,2,65535,0),this[n]=o&255,this[n+1]=o>>>8,n+2};l.prototype.writeUint16BE=l.prototype.writeUInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||ne(this,o,n,2,65535,0),this[n]=o>>>8,this[n+1]=o&255,n+2};l.prototype.writeUint32LE=l.prototype.writeUInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||ne(this,o,n,4,4294967295,0),this[n+3]=o>>>24,this[n+2]=o>>>16,this[n+1]=o>>>8,this[n]=o&255,n+4};l.prototype.writeUint32BE=l.prototype.writeUInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||ne(this,o,n,4,4294967295,0),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};l.prototype.writeIntLE=function(o,n,r,a){if(o=+o,n=n>>>0,!a){var m=Math.pow(2,8*r-1);ne(this,o,n,r,m-1,-m)}var i=0,_=1,d=0;for(this[n]=o&255;++i<r&&(_*=256);)o<0&&d===0&&this[n+i-1]!==0&&(d=1),this[n+i]=(o/_>>0)-d&255;return n+r};l.prototype.writeIntBE=function(o,n,r,a){if(o=+o,n=n>>>0,!a){var m=Math.pow(2,8*r-1);ne(this,o,n,r,m-1,-m)}var i=r-1,_=1,d=0;for(this[n+i]=o&255;--i>=0&&(_*=256);)o<0&&d===0&&this[n+i+1]!==0&&(d=1),this[n+i]=(o/_>>0)-d&255;return n+r};l.prototype.writeInt8=function(o,n,r){return o=+o,n=n>>>0,r||ne(this,o,n,1,127,-128),o<0&&(o=255+o+1),this[n]=o&255,n+1};l.prototype.writeInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||ne(this,o,n,2,32767,-32768),this[n]=o&255,this[n+1]=o>>>8,n+2};l.prototype.writeInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||ne(this,o,n,2,32767,-32768),this[n]=o>>>8,this[n+1]=o&255,n+2};l.prototype.writeInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||ne(this,o,n,4,2147483647,-2147483648),this[n]=o&255,this[n+1]=o>>>8,this[n+2]=o>>>16,this[n+3]=o>>>24,n+4};l.prototype.writeInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||ne(this,o,n,4,2147483647,-2147483648),o<0&&(o=4294967295+o+1),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};function Cr(e,o,n,r,a,m){if(n+r>e.length)throw new RangeError("Index out of range");if(n<0)throw new RangeError("Index out of range")}function Pr(e,o,n,r,a){return o=+o,n=n>>>0,a||Cr(e,o,n,4,34028234663852886e22,-34028234663852886e22),ro.write(e,o,n,r,23,4),n+4}l.prototype.writeFloatLE=function(o,n,r){return Pr(this,o,n,!0,r)};l.prototype.writeFloatBE=function(o,n,r){return Pr(this,o,n,!1,r)};function Nr(e,o,n,r,a){return o=+o,n=n>>>0,a||Cr(e,o,n,8,17976931348623157e292,-17976931348623157e292),ro.write(e,o,n,r,52,8),n+8}l.prototype.writeDoubleLE=function(o,n,r){return Nr(this,o,n,!0,r)};l.prototype.writeDoubleBE=function(o,n,r){return Nr(this,o,n,!1,r)};l.prototype.copy=function(o,n,r,a){if(!l.isBuffer(o))throw new TypeError("argument should be a Buffer");if(r||(r=0),!a&&a!==0&&(a=this.length),n>=o.length&&(n=o.length),n||(n=0),a>0&&a<r&&(a=r),a===r||o.length===0||this.length===0)return 0;if(n<0)throw new RangeError("targetStart out of bounds");if(r<0||r>=this.length)throw new RangeError("Index out of range");if(a<0)throw new RangeError("sourceEnd out of bounds");a>this.length&&(a=this.length),o.length-n<a-r&&(a=o.length-n+r);var m=a-r;return this===o&&typeof Uint8Array.prototype.copyWithin=="function"?this.copyWithin(n,r,a):Uint8Array.prototype.set.call(o,this.subarray(r,a),n),m};l.prototype.fill=function(o,n,r,a){if(typeof o=="string"){if(typeof n=="string"?(a=n,n=0,r=this.length):typeof r=="string"&&(a=r,r=this.length),a!==void 0&&typeof a!="string")throw new TypeError("encoding must be a string");if(typeof a=="string"&&!l.isEncoding(a))throw new TypeError("Unknown encoding: "+a);if(o.length===1){var m=o.charCodeAt(0);(a==="utf8"&&m<128||a==="latin1")&&(o=m)}}else typeof o=="number"?o=o&255:typeof o=="boolean"&&(o=Number(o));if(n<0||this.length<n||this.length<r)throw new RangeError("Out of range index");if(r<=n)return this;n=n>>>0,r=r===void 0?this.length:r>>>0,o||(o=0);var i;if(typeof o=="number")for(i=n;i<r;++i)this[i]=o;else{var _=l.isBuffer(o)?o:l.from(o,a),d=_.length;if(d===0)throw new TypeError('The value "'+o+'" is invalid for argument "value"');for(i=0;i<r-n;++i)this[i+n]=_[i%d]}return this};var bp=/[^+/0-9A-Za-z-_]/g;function wp(e){if(e=e.split("=")[0],e=e.trim().replace(bp,""),e.length<2)return"";for(;e.length%4!==0;)e=e+"=";return e}function et(e,o){o=o||1/0;for(var n,r=e.length,a=null,m=[],i=0;i<r;++i){if(n=e.charCodeAt(i),n>55295&&n<57344){if(!a){if(n>56319){(o-=3)>-1&&m.push(239,191,189);continue}else if(i+1===r){(o-=3)>-1&&m.push(239,191,189);continue}a=n;continue}if(n<56320){(o-=3)>-1&&m.push(239,191,189),a=n;continue}n=(a-55296<<10|n-56320)+65536}else a&&(o-=3)>-1&&m.push(239,191,189);if(a=null,n<128){if((o-=1)<0)break;m.push(n)}else if(n<2048){if((o-=2)<0)break;m.push(n>>6|192,n&63|128)}else if(n<65536){if((o-=3)<0)break;m.push(n>>12|224,n>>6&63|128,n&63|128)}else if(n<1114112){if((o-=4)<0)break;m.push(n>>18|240,n>>12&63|128,n>>6&63|128,n&63|128)}else throw new Error("Invalid code point")}return m}function Sp(e){for(var o=[],n=0;n<e.length;++n)o.push(e.charCodeAt(n)&255);return o}function vp(e,o){for(var n,r,a,m=[],i=0;i<e.length&&!((o-=2)<0);++i)n=e.charCodeAt(i),r=n>>8,a=n%256,m.push(a),m.push(r);return m}function Rr(e){return Zn.toByteArray(wp(e))}function on(e,o,n,r){for(var a=0;a<r&&!(a+n>=o.length||a>=e.length);++a)o[a+n]=e[a];return a}function xe(e,o){return e instanceof o||e!=null&&e.constructor!=null&&e.constructor.name!=null&&e.constructor.name===o.name}function tt(e){return e!==e}var kp=function(){for(var e="0123456789abcdef",o=new Array(256),n=0;n<16;++n)for(var r=n*16,a=0;a<16;++a)o[r+a]=e[n]+e[a];return o}()});var Wr=Jo((zm,Ur)=>{t();var z=Ur.exports={},ye,be;function rt(){throw new Error("setTimeout has not been defined")}function at(){throw new Error("clearTimeout has not been defined")}(function(){try{typeof setTimeout=="function"?ye=setTimeout:ye=rt}catch(e){ye=rt}try{typeof clearTimeout=="function"?be=clearTimeout:be=at}catch(e){be=at}})();function Fr(e){if(ye===setTimeout)return setTimeout(e,0);if((ye===rt||!ye)&&setTimeout)return ye=setTimeout,setTimeout(e,0);try{return ye(e,0)}catch(o){try{return ye.call(null,e,0)}catch(n){return ye.call(this,e,0)}}}function Ip(e){if(be===clearTimeout)return clearTimeout(e);if((be===at||!be)&&clearTimeout)return be=clearTimeout,clearTimeout(e);try{return be(e)}catch(o){try{return be.call(null,e)}catch(n){return be.call(this,e)}}}var Ie=[],io=!1,Ke,nn=-1;function Ap(){!io||!Ke||(io=!1,Ke.length?Ie=Ke.concat(Ie):nn=-1,Ie.length&&Or())}function Or(){if(!io){var e=Fr(Ap);io=!0;for(var o=Ie.length;o;){for(Ke=Ie,Ie=[];++nn<o;)Ke&&Ke[nn].run();nn=-1,o=Ie.length}Ke=null,io=!1,Ip(e)}}z.nextTick=function(e){var o=new Array(arguments.length-1);if(arguments.length>1)for(var n=1;n<arguments.length;n++)o[n-1]=arguments[n];Ie.push(new Br(e,o)),Ie.length===1&&!io&&Fr(Or)};function Br(e,o){this.fun=e,this.array=o}Br.prototype.run=function(){this.fun.apply(null,this.array)};z.title="browser";z.browser=!0;z.env={};z.argv=[];z.version="";z.versions={};function Ae(){}z.on=Ae;z.addListener=Ae;z.once=Ae;z.off=Ae;z.removeListener=Ae;z.removeAllListeners=Ae;z.emit=Ae;z.prependListener=Ae;z.prependOnceListener=Ae;z.listeners=function(e){return[]};z.binding=function(e){throw new Error("process.binding is not supported")};z.cwd=function(){return"/"};z.chdir=function(e){throw new Error("process.chdir is not supported")};z.umask=function(){return 0}});var c,p,qm,t=k(()=>{c=M(Mr()),p=M(Wr()),qm=function(e){function o(){var r=this||self;return delete e.prototype.__magic__,r}if(typeof globalThis=="object")return globalThis;if(this)return o();e.defineProperty(e.prototype,"__magic__",{configurable:!0,get:o});var n=__magic__;return n}(Object)});var tn,it=k(()=>{"use strict";t();tn=()=>Date.now()/1e3});var Gr=k(()=>{"use strict";t()});var $r=k(()=>{"use strict";t()});var ct=k(()=>{"use strict";t();it();Gr();$r()});var st={};je(st,{getItem:()=>Lp,localStorageKeys:()=>we,removeItem:()=>Dp,setItem:()=>Ep});var we,pt,Ep,Lp,Dp,co=k(()=>{"use strict";t();ct();we={loginExpiresAt:"dapp-core-login-expires-at",logoutEvent:"dapp-core-logout-event"},pt=typeof localStorage!="undefined",Ep=({key:e,data:o,expires:n})=>{!pt||localStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},Lp=e=>{if(!pt)return;let o=localStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:tn()>=n.expires?(localStorage.removeItem(String(e)),null):n.data},Dp=e=>{!pt||localStorage.removeItem(String(e))}});var mt={};je(mt,{clear:()=>zr,getItem:()=>Hr,removeItem:()=>jr,setItem:()=>Vr,storage:()=>Cp});var Vr,Hr,jr,zr,Cp,qr=k(()=>{"use strict";t();Vr=({key:e,data:o,expires:n})=>{sessionStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},Hr=e=>{let o=sessionStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:Date.now()>=n.expires?(sessionStorage.removeItem(String(e)),null):n.data},jr=e=>sessionStorage.removeItem(String(e)),zr=()=>sessionStorage.clear(),Cp={setItem:Vr,getItem:Hr,removeItem:jr,clear:zr}});var Ee,po=k(()=>{"use strict";t();co();qr();Ee={session:mt,local:st}});function Do(){return new Date().setHours(new Date().getHours()+24)}function Co(e){Ee.local.setItem({key:we.loginExpiresAt,data:e,expires:e})}var _t=k(()=>{"use strict";t();po();co()});var H=k(()=>{"use strict";t()});var Kr,so,Xr,Yr,Zr,mo,_o=k(()=>{"use strict";t();Kr="error when signing",so="Transaction canceled",Xr="error signing transaction",Yr="provider not initialized",Zr="You need a signer/valid signer to send a transaction,use either WalletProvider, LedgerProvider or WalletConnect",mo="Action not allowed. User is logged in. Call logout() first"});var Jr=k(()=>{"use strict";t()});var ee=k(()=>{"use strict";t()});var Qr=k(()=>{"use strict";t()});var ea=k(()=>{"use strict";t()});var oa=k(()=>{"use strict";t()});var na=k(()=>{"use strict";t()});var le=k(()=>{"use strict";t();Jr();ee();H();Qr();ea();oa();na()});var dt,ta,b_,ra,w_,aa,S_,v_,k_,ia=k(()=>{"use strict";t();le();dt={devnet:{id:"devnet",chainId:"D",name:"Devnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://devnet-wallet.elrond.com",apiAddress:"https://devnet-api.elrond.com",explorerAddress:"http://devnet-explorer.elrond.com",apiTimeout:"4000"},testnet:{id:"testnet",chainId:"T",name:"Testnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://testnet-wallet.elrond.com",apiAddress:"https://testnet-api.elrond.com",explorerAddress:"http://testnet-explorer.elrond.com",apiTimeout:"4000"},mainnet:{id:"mainnet",chainId:"1",name:"Mainnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://wallet.elrond.com",apiAddress:"https://api.elrond.com",explorerAddress:"https://explorer.elrond.com",apiTimeout:"4000"}},{chainId:ta,egldLabel:b_}=dt["devnet"],{chainId:ra,egldLabel:w_}=dt["testnet"],{chainId:aa,egldLabel:S_}=dt["mainnet"],v_={["devnet"]:ta,["testnet"]:ra,["mainnet"]:aa},k_={[ta]:"devnet",[ra]:"testnet",[aa]:"mainnet"}});var lt,ut=k(()=>{"use strict";t();lt={36864:{code:"codeSuccess",message:"Success"},27013:{code:"ERR_USER_DENIED",message:"Rejected by user"},27904:{code:"ERR_UNKNOWN_INSTRUCTION",message:"Unknown instruction"},28160:{code:"ERR_WRONG_CLA",message:"Wrong CLA"},28161:{code:"ERR_INVALID_ARGUMENTS",message:"Invalid arguments"},28162:{code:"ERR_INVALID_MESSAGE",message:"Invalid message"},28163:{code:"ERR_INVALID_P1",message:"Invalid P1"},28164:{code:"ERR_MESSAGE_TOO_LONG",message:"Message too long"},28165:{code:"ERR_RECEIVER_TOO_LONG",message:"Receiver too long"},28166:{code:"ERR_AMOUNT_TOO_LONG",message:"Amount too long"},28167:{code:"ERR_CONTRACT_DATA_DISABLED",message:"Contract data disabled in app options"},28168:{code:"ERR_MESSAGE_INCOMPLETE",message:"Message incomplete"},28176:{code:"ERR_SIGNATURE_FAILED",message:"Signature failed"},28169:{code:"ERR_WRONG_TX_VERSION",message:"Wrong TX version"},28170:{code:"ERR_NONCE_TOO_LONG",message:"Nonce too long"},28171:{code:"ERR_INVALID_AMOUNT",message:"Invalid amount"},28172:{code:"ERR_INVALID_FEE",message:"Invalid fee"},28173:{code:"ERR_PRETTY_FAILED",message:"Pretty failed"},28174:{code:"ERR_DATA_TOO_LONG",message:"Data too long"},28175:{code:"ERR_WRONG_TX_OPTIONS",message:"Invalid transaction options"},28177:{code:"ERR_SIGN_TX_DEPRECATED",message:"Regular transaction signing is deprecated in this version. Use hash signing."}}});var ca=k(()=>{"use strict";t()});var rn,gt=k(()=>{"use strict";t();rn="transaction-status-toast-id"});var Po,an,pa,sa,ma,No,cn,_a,lo,W=k(()=>{"use strict";t();_o();ia();ut();ca();gt();Po=5e4,an=1e9,pa=1,sa=1,ma="/dapp/init",No="signSession",cn="logout",_a="login",lo="0"});var ft,Q,te,re=k(()=>{"use strict";t();ft=require("@reduxjs/toolkit");W();Q=(0,ft.createAction)(cn),te=(0,ft.createAction)(_a,e=>({payload:e}))});var la,da,ua,V_,uo,ga,fe,pn,fa,ha,Ta,ht=k(()=>{"use strict";t();la=require("@reduxjs/toolkit");_t();H();re();da={loginMethod:"",walletConnectLogin:null,ledgerLogin:null,tokenLogin:null,walletLogin:null,extensionLogin:null,isLoginSessionInvalid:!1},ua=(0,la.createSlice)({name:"loginInfoSlice",initialState:da,reducers:{setLoginMethod:(e,o)=>{e.loginMethod=o.payload},setTokenLogin:(e,o)=>{e.tokenLogin=o.payload},setTokenLoginSignature:(e,o)=>{(e==null?void 0:e.tokenLogin)!=null&&(e.tokenLogin.signature=o.payload)},setWalletLogin:(e,o)=>{e.walletLogin=o.payload},setWalletConnectLogin:(e,o)=>{e.walletConnectLogin=o.payload},setLedgerLogin:(e,o)=>{e.ledgerLogin=o.payload},invalidateLoginSession:e=>{e.isLoginSessionInvalid=!0}},extraReducers:e=>{e.addCase(Q,()=>da),e.addCase(te,(o,n)=>{o.loginMethod=n.payload.loginMethod,Co(Do())})}}),{setLoginMethod:V_,setWalletConnectLogin:uo,setLedgerLogin:ga,setTokenLogin:fe,setTokenLoginSignature:pn,setWalletLogin:fa,invalidateLoginSession:ha}=ua.actions,Ta=ua.reducer});var xa,ya,go,Ro=k(()=>{"use strict";t();xa=!1,ya=(e=!0)=>{xa=e},go=(e,o=0)=>{xa||setTimeout(()=>{window.location.href=e},o)}});function Le(e,o){Boolean(e)&&e!=null&&setTimeout(()=>{window.location.pathname.includes(e)||(typeof o=="function"?o(e):go(e))},200)}var ba=k(()=>{"use strict";t();Ro()});function Tt(e){return e[Math.floor(Math.random()*e.length)]}var wa=k(()=>{"use strict";t()});function Xe(e){let o=null;if((e==null?void 0:e.statusCode)in lt){let n=e==null?void 0:e.statusCode,{message:r}=lt[n];o=Rp.includes(n)?Sa:r}return{errorMessage:o,defaultErrorMessage:Sa}}var Sa,Pp,Np,Rp,xt=k(()=>{"use strict";t();ut();Sa="Check if Elrond app is open on Ledger",Pp=28161,Np=28160,Rp=[Pp,Np]});var fo=k(()=>{"use strict";t();ba();wa();xt()});var va,ka,yt,Mp,Ia,td,rd,Aa,bt=k(()=>{"use strict";t();va=require("@reduxjs/toolkit"),ka=M(require("lodash.omit"));fo();yt={id:"not-configured",chainId:"",name:"NOT CONFIGURED",egldLabel:"",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"",walletConnectBridgeAddress:"",walletConnectV2RelayAddress:"",walletConnectV2ProjectId:"",walletAddress:"",apiAddress:"",explorerAddress:"",apiTimeout:"4000"},Mp={network:yt,chainID:"-1"},Ia=(0,va.createSlice)({name:"appConfig",initialState:Mp,reducers:{initializeNetworkConfig:(e,o)=>{let n=Tt(o.payload.walletConnectBridgeAddresses),r=Tt(o.payload.walletConnectV2RelayAddresses),a=(0,ka.default)(o.payload,"walletConnectBridgeAddresses","walletConnectV2RelayAddresses");e.network=P(b(b({},e.network),a),{walletConnectBridgeAddress:n,walletConnectV2RelayAddress:r})},setChainID:(e,o)=>{e.chainID=o.payload}}}),{initializeNetworkConfig:td,setChainID:rd}=Ia.actions,Aa=Ia.reducer});var St,Ea,La,sn,wt,Da,mn,dd,vt,ld,kt,Ca,ud,gd,fd,Pa,Na,It=k(()=>{"use strict";t();St=require("@elrondnetwork/erdjs"),Ea=require("@reduxjs/toolkit"),La=require("redux-persist");W();po();co();re();sn={balance:"...",address:"",nonce:0,txCount:0,scrCount:0,claimableRewards:lo},wt={address:"",websocketEvent:null,accounts:{"":sn},ledgerAccount:null,publicKey:"",walletConnectAccount:null,isAccountLoading:!0,accountLoadingError:null},Da=(0,Ea.createSlice)({name:"accountInfoSlice",initialState:wt,reducers:{setAddress:(e,o)=>{let n=o.payload;e.address=n,e.publicKey=new St.Address(n).hex()},setAccount:(e,o)=>{let n=e.address===o.payload.address;e.accounts={[e.address]:n?o.payload:sn},e.isAccountLoading=!1,e.accountLoadingError=null},setAccountNonce:(e,o)=>{let{address:n}=e;e.accounts[n].nonce=o.payload},setAccountShard:(e,o)=>{e.shard=o.payload},setLedgerAccount:(e,o)=>{e.ledgerAccount=o.payload},updateLedgerAccount:(e,o)=>{e.ledgerAccount!=null&&(e.ledgerAccount.index=o.payload.index,e.ledgerAccount.address=o.payload.address)},setWalletConnectAccount:(e,o)=>{e.walletConnectAccount=o.payload},setIsAccountLoading:(e,o)=>{e.isAccountLoading=o.payload,e.accountLoadingError=null},setAccountLoadingError:(e,o)=>{e.accountLoadingError=o.payload,e.isAccountLoading=!1},setWebsocketEvent:(e,o)=>{e.websocketEvent={timestamp:Date.now(),message:o.payload}}},extraReducers:e=>{e.addCase(Q,()=>(Ee.local.removeItem(we.loginExpiresAt),wt)),e.addCase(te,(o,n)=>{let{address:r}=n.payload;o.address=r,o.publicKey=new St.Address(r).hex()}),e.addCase(La.REHYDRATE,(o,n)=>{var u;if(!((u=n.payload)!=null&&u.account))return;let{account:r}=n.payload,{address:a,shard:m,accounts:i,publicKey:_}=r;o.address=a,o.shard=m;let d=i&&a in i;o.accounts=d?i:wt.accounts,o.publicKey=_})}}),{setAccount:mn,setAddress:dd,setAccountNonce:vt,setAccountShard:ld,setLedgerAccount:kt,updateLedgerAccount:Ca,setWalletConnectAccount:ud,setIsAccountLoading:gd,setAccountLoadingError:fd,setWebsocketEvent:Pa}=Da.actions,Na=Da.reducer});function Se(e){return e!=null&&(Vp(e)||qp(e))}function Oe(e){return e!=null&&(Hp(e)||Kp(e))}function Be(e){return e!=null&&(jp(e)||Xp(e))}function _n(e){return e!=null&&zp(e)}function Vp(e){return e!=null&&Fp.includes(e)}function Hp(e){return e!=null&&Op.includes(e)}function jp(e){return e!=null&&Bp.includes(e)}function zp(e){return e!=null&&Up.includes(e)}function qp(e){return e!=null&&Wp.includes(e)}function Kp(e){return e!=null&&Gp.includes(e)}function Xp(e){return e!=null&&$p.includes(e)}var Fp,Op,Bp,Up,Wp,Gp,$p,ho=k(()=>{"use strict";t();H();Fp=["sent"],Op=["success"],Bp=["fail","cancelled","timedOut"],Up=["timedOut"],Wp=["pending"],Gp=["success"],$p=["fail","invalid"]});var Ra,Ma,Fo,Yp,Fa,Oo,dn,Zp,To,Oa,Jp,kd,Ba,Bo,De,ln,un=k(()=>{"use strict";t();Ra=require("@reduxjs/toolkit"),Ma=require("redux-persist");H();ho();re();Fo={signedTransactions:{},transactionsToSign:null,signTransactionsError:null,signTransactionsCancelMessage:null,customTransactionInformationForSessionId:{}},Yp={signWithoutSending:!1,sessionInformation:null,redirectAfterSign:!1},Fa=(0,Ra.createSlice)({name:"transactionsSlice",initialState:Fo,reducers:{moveTransactionsToSignedState:(e,o)=>{var d,u;let{sessionId:n,transactions:r,errorMessage:a,status:m,redirectRoute:i}=o.payload,_=((d=e.customTransactionInformationForSessionId)==null?void 0:d[n])||Yp;e.signedTransactions[n]={transactions:r,status:m,errorMessage:a,redirectRoute:i,customTransactionInformation:_},((u=e==null?void 0:e.transactionsToSign)==null?void 0:u.sessionId)===n&&(e.transactionsToSign=Fo.transactionsToSign)},clearSignedTransaction:(e,o)=>{e.signedTransactions[o.payload]&&delete e.signedTransactions[o.payload]},clearTransactionToSign:e=>{e!=null&&e.transactionsToSign&&(e.transactionsToSign=null)},updateSignedTransaction:(e,o)=>{e.signedTransactions=b(b({},e.signedTransactions),o.payload)},updateSignedTransactions:(e,o)=>{let{sessionId:n,status:r,errorMessage:a,transactions:m}=o.payload;e.signedTransactions[n]!=null&&(e.signedTransactions[n].status=r,a!=null&&(e.signedTransactions[n].errorMessage=a),m!=null&&(e.signedTransactions[n].transactions=m))},updateSignedTransactionStatus:(e,o)=>{var _,d,u,g,f,T;let{sessionId:n,status:r,errorMessage:a,transactionHash:m}=o.payload,i=(d=(_=e.signedTransactions)==null?void 0:_[n])==null?void 0:d.transactions;if(i!=null){e.signedTransactions[n].transactions=i.map(F=>F.hash===m?P(b({},F),{status:r,errorMessage:a}):F);let h=(g=(u=e.signedTransactions[n])==null?void 0:u.transactions)==null?void 0:g.every(F=>Oe(F.status)),E=(T=(f=e.signedTransactions[n])==null?void 0:f.transactions)==null?void 0:T.every(F=>Be(F.status));h&&(e.signedTransactions[n].status="success"),E&&(e.signedTransactions[n].status="fail")}},setTransactionsToSign:(e,o)=>{e.transactionsToSign=o.payload;let{sessionId:n,customTransactionInformation:r}=o.payload;e.customTransactionInformationForSessionId[n]=r,e.signTransactionsError=null},clearAllTransactionsToSign:e=>{e.transactionsToSign=Fo.transactionsToSign,e.signTransactionsError=null},clearAllSignedTransactions:e=>{e.signedTransactions=Fo.signedTransactions},setSignTransactionsError:(e,o)=>{e.signTransactionsError=o.payload},setSignTransactionsCancelMessage:(e,o)=>{e.signTransactionsCancelMessage=o.payload}},extraReducers:e=>{e.addCase(Q,()=>Fo),e.addCase(Ma.REHYDRATE,(o,n)=>{var i;if(!((i=n.payload)!=null&&i.transactions))return;let{signedTransactions:r,customTransactionInformationForSessionId:a}=n.payload.transactions,m=Object.entries(r).reduce((_,[d,u])=>{let g=new Date(d),f=new Date;return f.setHours(f.getHours()+5),f-g>0||(_[d]=u),_},{});a!=null&&(o.customTransactionInformationForSessionId=a),r!=null&&(o.signedTransactions=m)})}}),{updateSignedTransactionStatus:Oo,updateSignedTransactions:dn,setTransactionsToSign:Zp,clearAllTransactionsToSign:To,clearAllSignedTransactions:Oa,clearSignedTransaction:Jp,clearTransactionToSign:kd,setSignTransactionsError:Ba,setSignTransactionsCancelMessage:Bo,moveTransactionsToSignedState:De}=Fa.actions,ln=Fa.reducer});var Ua,Et,Lt,Dt,Qp,At,Wa,Ld,es,gn,fn,hn=k(()=>{"use strict";t();Ua=require("@reduxjs/toolkit");re();Et="Transaction failed",Lt="Transaction successful",Dt="Processing transaction",Qp="Transaction submitted",At={},Wa=(0,Ua.createSlice)({name:"transactionsInfo",initialState:At,reducers:{setTransactionsDisplayInfo(e,o){let{sessionId:n,transactionsDisplayInfo:r}=o.payload;n!=null&&(e[n]={errorMessage:(r==null?void 0:r.errorMessage)||Et,successMessage:(r==null?void 0:r.successMessage)||Lt,processingMessage:(r==null?void 0:r.processingMessage)||Dt,submittedMessage:(r==null?void 0:r.submittedMessage)||Qp,transactionDuration:r==null?void 0:r.transactionDuration})},clearTransactionsInfoForSessionId(e,o){o.payload!=null&&delete e[o.payload]},clearTransactionsInfo:()=>At},extraReducers:e=>{e.addCase(Q,()=>At)}}),{clearTransactionsInfo:Ld,setTransactionsDisplayInfo:es,clearTransactionsInfoForSessionId:gn}=Wa.actions,fn=Wa.reducer});var $a,Ga,Va,Nd,os,Rd,Ha,ja,Ct=k(()=>{"use strict";t();$a=require("@reduxjs/toolkit");re();Ga={},Va=(0,$a.createSlice)({name:"modalsSlice",initialState:Ga,reducers:{setTxSubmittedModal:(e,o)=>{e.txSubmittedModal=o.payload},setNotificationModal:(e,o)=>{e.notificationModal=o.payload},clearTxSubmittedModal:e=>{e.txSubmittedModal=void 0},clearNotificationModal:e=>{e.notificationModal=void 0}},extraReducers:e=>{e.addCase(Q,()=>Ga)}}),{setTxSubmittedModal:Nd,setNotificationModal:os,clearTxSubmittedModal:Rd,clearNotificationModal:Ha}=Va.actions,ja=Va.reducer});var qa,za,Ka,ns,Pt,$d,Vd,Hd,jd,Tn,xn=k(()=>{"use strict";t();qa=require("@reduxjs/toolkit");gt();le();it();re();za={customToasts:[],transactionToasts:[],failTransactionToast:null},Ka=(0,qa.createSlice)({name:"toastsSlice",initialState:za,reducers:{addCustomToast:(e,o)=>{e.customToasts.push(P(b({},o.payload),{type:"custom",toastId:o.payload.toastId||`custom-toast-${e.customToasts.length+1}`}))},removeCustomToast:(e,o)=>{e.customToasts=e.customToasts.filter(n=>n.toastId!==o.payload)},addTransactionToast:(e,o)=>{e.transactionToasts.push({type:"transaction",startTimestamp:tn(),toastId:o.payload||`custom-toast-${e.transactionToasts.length+1}`})},removeTransactionToast:(e,o)=>{e.transactionToasts=e.transactionToasts.filter(n=>n.toastId!==o.payload)},addFailTransactionToast:(e,o)=>{e.failTransactionToast=P(b({},o.payload),{toastId:rn})},removeFailTransactionToast:e=>{e.failTransactionToast=null}},extraReducers:e=>{e.addCase(Q,()=>za)}}),{addCustomToast:ns,removeCustomToast:Pt,addTransactionToast:$d,removeTransactionToast:Vd,addFailTransactionToast:Hd,removeFailTransactionToast:jd}=Ka.actions,Tn=Ka.reducer});var G=k(()=>{"use strict";t();ht();bt();It();un();hn();Ct();xn()});var Nt=k(()=>{"use strict";t()});var Ja,Za,Qa,_l,dl,ll,ul,gl,fl,ei,oi=k(()=>{"use strict";t();Ja=require("@reduxjs/toolkit");re();Za={visible:!1,modal:{showHeader:!0,showFooter:!1,headerText:"",footerText:""}},Qa=(0,Ja.createSlice)({name:"dappModalsSlice",initialState:Za,reducers:{setDappModalVisibility:(e,o)=>{e.visible=o.payload},setDappModalHeaderVisibility:(e,o)=>{e.modal.showHeader=o.payload},setDappModalFooterVisibility:(e,o)=>{e.modal.showFooter=o.payload},setDappModalHeaderText:(e,o)=>{e.modal.headerText=o.payload},setDappModalFooterText:(e,o)=>{e.modal.footerText=o.payload},setDappModalConfig:(e,o)=>{e.modal=b({},o.payload)}},extraReducers:e=>{e.addCase(Q,()=>Za)}}),{setDappModalVisibility:_l,setDappModalHeaderVisibility:dl,setDappModalFooterVisibility:ll,setDappModalHeaderText:ul,setDappModalFooterText:gl,setDappModalConfig:fl}=Qa.actions,ei=Qa.reducer});var ni,is,cs,xo,bn=k(()=>{"use strict";t();ni=require("@reduxjs/toolkit");Nt();It();oi();ht();Ct();bt();xn();hn();un();is={["account"]:Na,["networkConfig"]:Aa,["loginInfo"]:Ta,["modals"]:ja,["dappModal"]:ei,["toasts"]:Tn,["transactions"]:ln,["transactionsInfo"]:fn},cs=(e={})=>(0,ni.combineReducers)(b(b({},is),e)),xo=cs});var ai={};je(ai,{default:()=>us});function Rt(e,o=[]){return{key:e,version:1,storage:ri.default,blacklist:o}}var Ze,ti,ri,ps,ss,ms,_s,ds,ls,us,ii=k(()=>{"use strict";t();Ze=require("redux-persist"),ti=M(require("redux-persist/lib/storage")),ri=M(require("redux-persist/lib/storage/session"));bn();G();xn();hn();un();Nt();ps={2:e=>P(b({},e),{networkConfig:yt})};ss=Rt("dapp-core-transactionsInfo"),ms=Rt("dapp-core-transactions",["transactionsToSign"]),_s=Rt("dapp-core-toasts"),ds={key:"dapp-core-store",version:2,storage:ti.default,whitelist:["account","loginInfo","modals","networkConfig"],migrate:(0,Ze.createMigrate)(ps,{debug:!1})},ls={["toasts"]:(0,Ze.persistReducer)(_s,Tn),["transactions"]:(0,Ze.persistReducer)(ms,ln),["transactionsInfo"]:(0,Ze.persistReducer)(ss,fn)},us=(0,Ze.persistReducer)(ds,xo(ls))});var ci={};je(ci,{default:()=>gs});var gs,pi=k(()=>{"use strict";t();bn();gs=xo()});var si={};je(si,{default:()=>hs});var ge,fs,hs,mi=k(()=>{"use strict";t();ge=require("redux-persist"),fs=[ge.FLUSH,ge.REHYDRATE,ge.PAUSE,ge.PERSIST,ge.PURGE,ge.REGISTER],hs=fs});var li={};je(li,{default:()=>di});function di(e){return(0,_i.persistStore)(e)}var _i,ui=k(()=>{"use strict";t();_i=require("redux-persist")});var Gm={};je(Gm,{useCheckTransactionStatus:()=>Tm,useDebounce:()=>Pm,useExtensionLogin:()=>Dc,useGetAccount:()=>_r,useGetAccountInfo:()=>rr,useGetAccountProvider:()=>eo,useGetActiveTransactionsStatus:()=>Em,useGetFailedTransactions:()=>Im,useGetIsLoggedIn:()=>Dm,useGetLoginInfo:()=>Lm,useGetNetworkConfig:()=>Vo,useGetNotification:()=>om,useGetPendingTransactions:()=>pr,useGetSignTransactionsError:()=>_m,useGetSignedTransactions:()=>tr,useGetSuccessfulTransactions:()=>Am,useGetTokenDetails:()=>sm,useGetTransactionDisplayInfo:()=>nm,useGetWebsocketEvent:()=>Cm,useGetWebsocketUrl:()=>Rm,useIsDappModalVisible:()=>Wm,useLedgerLogin:()=>Pc,useParseMultiEsdtTransferData:()=>ar,useParseSignedTransactions:()=>Jt,useRegisterWebsocketListener:()=>Um,useSignMultipleTransactions:()=>ir,useSignTransactions:()=>cm,useSignTransactionsWithDevice:()=>cr,useSignTransactionsWithLedger:()=>mm,useTrackTransactionStatus:()=>km,useUpdateEffect:()=>Qe,useWalletConnectLogin:()=>Rc,useWalletConnectV2Login:()=>Fc,useWebWalletLogin:()=>Cc});module.exports=Lo(Gm);t();t();t();var yi=require("react"),yo=require("react-redux");t();var Mt=require("@reduxjs/toolkit"),Ti=require("react-redux/es/utils/Subscription");G();t();var Xa=M(require("lodash.throttle"));W();G();_t();po();co();var ts=[cn],rs=(0,Xa.default)(()=>{Co(Do())},5e3),Ya=e=>o=>n=>{var d;if(ts.includes(n.type))return o(n);let r=e.getState(),a=Ee.local.getItem(we.loginExpiresAt);if(!((d=r==null?void 0:r.loginInfo)==null?void 0:d.isLoggedIn))return o(n);if(a==null)return Co(Do());let i=Date.now();return a-i<0?setTimeout(()=>y(void 0,null,function*(){console.log("session expired");try{e.dispatch(ha())}catch(u){console.error("error logging out",u)}}),1e3):(rs(),o(n))};t();t();function yn(){return typeof sessionStorage!="undefined"}var gi=yn()?(ii(),Lo(ai)).default:(pi(),Lo(ci)).default,fi=yn()?(mi(),Lo(si)).default:[],hi=yn()?(ui(),Lo(li)).default:e=>e;bn();var I=(0,Mt.configureStore)({reducer:gi,middleware:e=>e({serializableCheck:{ignoredActions:[...fi,vt.type,mn.type],ignoredPaths:["payload.nonce","account.account.nonce","providers.accountProvider"]}}).concat(Ya)}),xi=(0,Ti.createSubscription)(I),Gl=hi(I),$l=(0,Mt.configureStore)({reducer:xo});var Ts={store:I,subscription:xi},Ft=(0,yi.createContext)(Ts),zl=(0,yo.createStoreHook)(Ft),j=(0,yo.createDispatchHook)(Ft),w=(0,yo.createSelectorHook)(Ft);t();t();G();t();var bi=M(require("lodash.isequal")),wn=require("reselect"),S=(0,wn.createSelectorCreator)(wn.defaultMemoize,bi.default);var Ce=e=>e.account,Ue=S(Ce,e=>e.address),bo=S(Ce,Ue,(e,o)=>o in e.accounts?e.accounts[o]:sn),wi=S(Ce,bo,(e,o)=>{let a=e,{accounts:n}=a,r=Kn(a,["accounts"]);return P(b({},r),{account:o})}),xs=S(bo,e=>e.balance),Si=S(bo,e=>{var o;return((o=e==null?void 0:e.nonce)==null?void 0:o.valueOf())||0}),Zl=S(Ce,e=>e.shard),vi=S(Ce,e=>e.ledgerAccount),Jl=S(Ce,e=>e.walletConnectAccount),Ql=S(Ce,e=>e.isAccountLoading),eu=S(Ce,e=>e.accountLoadingError),Sn=S(Ce,e=>e.websocketEvent);t();var ve=e=>e.networkConfig,Uo=S(ve,e=>e.chainID),ki=S(ve,e=>e.network.walletConnectBridgeAddress),Ii=S(ve,e=>e.network.walletConnectV2RelayAddress),Ai=S(ve,e=>e.network.walletConnectV2ProjectId),vn=S(ve,e=>e.network.walletConnectDeepLink),Pe=S(ve,e=>e.network),kn=S(Pe,e=>e.apiAddress),Ei=S(Pe,e=>e.explorerAddress),Ot=S(Pe,e=>e.egldLabel);t();H();var We=e=>e.loginInfo,cu=S(We,e=>e.loginMethod),Ge=S(We,Ue,(e,o)=>e.loginMethod!=""&&Boolean(o)),pu=S(We,e=>e.walletConnectLogin),su=S(We,e=>e.ledgerLogin),Li=S(We,e=>e.walletLogin),mu=S(We,e=>e.isLoginSessionInvalid);t();t();var _e=require("@elrondnetwork/erdjs");W();t();function Bt(e){try{return c.Buffer.from(e,"base64").toString()===atob(e)}catch(o){return!1}}function Wo(e){var a,m,i;let{data:o}=e,n=Bt(o!=null?o:"")?_e.TransactionPayload.fromEncoded(o):new _e.TransactionPayload(o),r=new _e.Transaction(b({value:e.value.valueOf(),data:n,nonce:e.nonce.valueOf(),receiver:new _e.Address(e.receiver),sender:new _e.Address(e.sender),gasLimit:(a=e.gasLimit.valueOf())!=null?a:Po,gasPrice:(m=e.gasPrice.valueOf())!=null?m:an,chainID:e.chainID.valueOf(),version:new _e.TransactionVersion((i=e.version)!=null?i:pa)},e.options?{options:new _e.TransactionOptions(e.options)}:{}));return r.applySignature({hex:()=>e.signature||""},new _e.Address(e.sender)),r}t();ho();t();var Wt=M(require("bignumber.js"));le();t();t();function Ut(e){for(let o=0;o<e.length;o++)if(e.charCodeAt(o)>127)return!1;return!0}function Go(e){let o=e;try{let n=c.Buffer.from(e,"hex").toString();Ut(n)&&n.length>1&&(o=n)}catch(n){}return o}t();var Di=(e,o)=>{let n=[],r=e.indexOf(o,0);for(;r>=0;)n.push(r),r=e.indexOf(o,r+1);return n};function Pi(e){let o=[],n=0;try{if((e==null?void 0:e.startsWith("MultiESDTNFTTransfer"))&&(e==null?void 0:e.includes("@"))){let[,r,a,...m]=e==null?void 0:e.split("@");if(r){let i=new Wt.default(a,16).toNumber(),_=0;for(let f=0;f<i;f++){let T={type:"nftTransaction",data:"",receiver:r};for(let h=0;h<3;h++){switch(h){case 0:T.token=Go(m[_]),T.data=m[_];break;case 1:{let E=m[_]&&m[_].length?m[_]:"";E?T.nonce=E:T.type="esdtTransaction",T.data=`${T.data}@${m[_]}`;break}case 2:T.amount=new Wt.default(m[_],16).toString(10),T.data=`${T.data}@${m[_]}`;break;default:break}n=_+1,_++}o[f]=T}let d=o.length!==i,u=o.some(f=>Di(f.data,"@").length!==2),g=o.some(f=>f.data.startsWith("@"));if(d||u||g)return[];if(m[n]){let f=m[n];for(let T=n+1;T<m.length;T++)f+="@"+m[T];o[i]={type:"scCall",data:f,receiver:r}}}}}catch(r){return console.error("failed parsing tx",r),o}return o}t();var $t=require("@elrondnetwork/erdjs"),Vt=M(require("bignumber.js"));H();t();var Ni=require("@elrondnetwork/erdjs");function ys(e){try{let o=new Ni.Address(e);return Boolean(o.bech32())}catch(o){return!1}}function wo(e){return(e==null?void 0:e.startsWith("erd"))&&e.length===62&&/^\w+$/.test(e)&&ys(e)}var Gt={tokenId:"",amount:""},bs=e=>e.split("@").map((m,i)=>[2,3].includes(i)?m:Go(m));function Ht(e){if(!e)return Gt;let o=e.startsWith("ESDTTransfer"),n=e.startsWith("ESDTNFTTransfer")&&e.includes("@");if(o){let[,r,a]=e.split("@");try{let m=c.Buffer.from(r,"hex").toString("ascii");if(!m)return Gt;let i=new Vt.default("0x"+a.replace("0x","")).toString(10);return{tokenId:m,amount:i}}catch(m){console.error("Error getting token from transaction data",m)}}if(n)try{let[,r,a,m,i]=bs(e);if([r,a,m,i].every(_=>Boolean(_))&&wo(new $t.Address(i).bech32()))return{tokenId:`${r}-${a}`,amount:new Vt.default(m,16).toString(10),collection:r,nonce:a,receiver:new $t.Address(i).bech32()}}catch(r){}return Gt}t();function Ri({tokenId:e,erdLabel:o}){return Boolean(e&&e!==o)}t();t();var Ss=require("@elrondnetwork/erdjs/out");t();W();t();var Mi=require("@elrondnetwork/erdjs");le();t();function vs(e,o){let n=new URLSearchParams(e),r=Object.fromEntries(n);return{nextUrlParams:new URLSearchParams(b(b({},r),o)).toString(),params:r}}function Fi({callbackUrl:e,urlParams:o={}}){let n=e;if(Object.entries(o).length>0){let{search:r,origin:a,pathname:m}=new URL(e),{nextUrlParams:i}=vs(r,o);n=`${a}${m}?${i}`}return n}t();t();t();H();function So(e){let n=Object.getPrototypeOf(e).toPlainObject!=null?e:Wo(e);return P(b({},n.toPlainObject()),{hash:n.getHash().hex(),status:"pending"})}t();t();t();t();t();t();t();var $o="accounts";var jt="tokens";var zt="collections";t();t();t();t();ee();t();t();ee();t();t();t();var Cs=M(require("bignumber.js"));W();t();var Ds=require("@elrondnetwork/erdjs"),qt=M(require("bignumber.js"));W();t();var Es=M(require("bignumber.js"));t();qt.default.config({ROUNDING_MODE:qt.default.ROUND_FLOOR});t();t();t();W();t();t();t();t();W();t();W();t();var Ps=require("@elrondnetwork/erdjs");W();t();var En=require("@elrondnetwork/erdjs"),Ms=require("@elrondnetwork/erdjs"),Fs=require("@elrondnetwork/erdjs-network-providers");W();t();t();var Ns=M(require("bignumber.js"));W();t();W();t();var Kt={isEsdt:!1,isNft:!1,isEgld:!1};function Xt(e){let o=e==null?void 0:e.split("-").length;return o===2?P(b({},Kt),{isEsdt:!0}):o===3?P(b({},Kt),{isNft:!0}):P(b({},Kt),{isEgld:!0})}t();t();t();t();W();t();ee();var Bs=["reDelegateRewards","claimRewards","unBond"],Us=["wrapEgld","unwrapEgld"],Ws=["unStake"],Gs=["unDelegate"];t();t();t();W();t();var Hs=M(require("bignumber.js"));t();t();t();ee();t();var js=M(require("bignumber.js"));t();t();t();ee();t();ee();t();var qs=M(require("bignumber.js"));W();ee();t();ee();t();ee();t();var Ks=M(require("bignumber.js"));t();H();t();H();t();t();var Xs=require("react");ee();var KT=["create","localMint","ESDTLocalMint","addQuantity","burn","localBurn","ESDTLocalBurn","wipe","writeLog","signalError"];t();var Ys=M(require("anchorme"));t();var Gi=require("react");le();t();t();ee();t();var Zs=require("@elrondnetwork/erdjs/out"),Js=M(require("bignumber.js"));ee();t();ee();t();t();H();var kx=[{label:"Raw",value:"raw"},{label:"Text",value:"text"},{label:"Decimal",value:"decimal"},{label:"Smart",value:"smart"}];t();var $i=require("react");le();t();t();t();t();t();W();var Dn=e=>e.transactions,Je=S(Dn,e=>e.signedTransactions),Vi=S(Dn,e=>e.signTransactionsError),Hi=S(Dn,e=>e.signTransactionsCancelMessage),Cn=e=>o=>Object.entries(o).reduce((n,[r,a])=>(e(a.status)&&(n[r]=a),n),{}),Pn=S(Je,Cn(Se)),Nn=S(Je,Cn(Oe)),Rn=S(Je,Cn(Be)),ji=S(Je,Cn(_n)),vo=S(Dn,e=>{var o;return(e==null?void 0:e.transactionsToSign)==null?null:P(b({},e.transactionsToSign),{transactions:((o=e==null?void 0:e.transactionsToSign)==null?void 0:o.transactions.map(n=>Wo(n)))||[]})}),zi=S(Je,(e,o)=>o,(e,o)=>o!=null?(e==null?void 0:e[o])||{}:{});t();G();var qi={errorMessage:Et,successMessage:Lt,processingMessage:Dt},em=e=>e.transactionsInfo,Ki=S(em,(e,o)=>o,(e,o)=>o!=null&&(e==null?void 0:e[Number(o)])||qi);t();var Xi=e=>e.modals,wy=S(Xi,e=>e.txSubmittedModal),Yi=S(Xi,e=>e.notificationModal);t();var Zt=e=>e.toasts,ky=S(Zt,e=>e.customToasts),Iy=S(Zt,e=>e.transactionToasts),Ay=S(Zt,e=>e.failTransactionToast);var Vo=()=>w(ve);t();var Mn=require("react");function Qe(e,o=[]){let n=(0,Mn.useRef)(!0);(0,Mn.useEffect)(()=>{n.current?n.current=!1:e()},o)}t();G();function om(){let e=w(Yi),o=j();return{notification:e,clearNotification:()=>o(Ha())}}t();t();function nm(e){return w(o=>Ki(o,e))}t();var Zi=require("react"),Ji=require("@elrondnetwork/erdjs-web-wallet-provider"),Qi=M(require("qs"));W();G();H();function Jt(e){let{search:o}=window.location,n=w(Pe),r=j();(0,Zi.useEffect)(()=>{if(o!=null){let a=Qi.default.parse(o.replace("?",""));if(a&&No in a){let m=String(a[No]),i=new Ji.WalletProvider(`${n.walletAddress}${ma}`).getTransactionsFromWalletUrl();if(a.status==="cancelled"){r(De({sessionId:m,status:"cancelled"})),e(),history.pushState({},document.title,"?");return}i.length>0&&(r(De({sessionId:m,status:"signed",transactions:i.map(_=>So(_))})),history.pushState({},document.title,"?"))}}},[o])}t();var Io=require("react"),nr=require("@elrondnetwork/erdjs-extension-provider");W();t();t();t();t();var ec=require("@elrondnetwork/erdjs-extension-provider"),oc=require("@elrondnetwork/erdjs-hw-provider"),On=require("@elrondnetwork/erdjs-wallet-connect-provider"),Qt=require("@elrondnetwork/erdjs-web-wallet-provider");W();H();var tm="/dapp/init",Ve=e=>{switch(e==null?void 0:e.constructor){case Qt.WalletProvider:return"wallet";case On.WalletConnectProvider:return"walletconnect";case On.WalletConnectV2Provider:return"walletconnectv2";case oc.HWProvider:return"ledger";case ec.ExtensionProvider:return"extension";case Fn:return"";default:return"extra"}},nc=e=>new Qt.WalletProvider(`${e}${tm}`),tc=e=>y(void 0,null,function*(){if(!e.isInitialized())throw new Error("Unable to get version. Provider not initialized");let o=e.hwApp,{contractData:n,version:r}=yield o.getAppConfiguration();return{version:r,dataEnabled:n===sa}}),Ne=e=>`Unable to perform ${e}, Provider not initialized`,Fn=class{init(){return Promise.resolve(!1)}login(o){throw new Error(Ne(`login with options: ${o}`))}logout(o){throw new Error(Ne(`logout with options: ${o}`))}getAddress(){throw new Error(Ne("getAddress"))}isInitialized(){return!1}isConnected(){return Promise.resolve(!1)}sendTransaction(o,n){throw new Error(Ne(`sendTransaction with transactions: ${o} options: ${n}`))}signTransaction(o,n){throw new Error(Ne(`signTransaction with transactions: ${o} options: ${n}`))}signTransactions(o,n){throw new Error(Ne(`signTransactions with transactions: ${o} options: ${n}`))}signMessage(o,n){throw new Error(Ne(`signTransactions with ${o} and options ${n}`))}sendCustomMessage({method:o,params:n}){throw new Error(Ne(`sendCustomMessage with method: ${o} params: ${n}`))}sendCustomRequest(o){throw new Error(Ne(`sendSessionEvent with options: ${o}`))}ping(){return Promise.resolve(!1)}},er=new Fn;var rc=er;function Re(e){rc=e}function ae(){return rc||er}var eo=()=>{let e=ae(),o=Ve(e);return{provider:e,providerType:o}};G();H();t();t();re();le();t();function ie(){return Ge(I.getState())}t();t();H();t();t();function ac(){let e=ae();return Ve(e)}function ce(e){return ac()===e}function Ho(){let{search:e}=window.location,o=I.getState(),n=ae(),r=Ue(o),a=Ge(o),m=Li(o);if(!n)throw"provider not initialized";return ce("ledger")&&a?new Promise(i=>{i(r)}):!ce("")&&!ce("wallet")&&!ce("extra")?n.getAddress():new Promise(i=>{if(m!=null){let _=new URLSearchParams(e),d=Object.fromEntries(_);wo(d.address)&&i(d.address)}a&&i(r),i("")})}t();t();var cc=M(require("axios"));t();t();var ic=e=>{let o=e!=null?e:kn(I.getState());return o.endsWith("/")?o.slice(0,-1):o};function pc(e){return y(this,null,function*(){if(!e)return null;let n=`${ic()}/${$o}/${e}`;try{let{data:r}=yield cc.default.get(n);return r}catch(r){console.error("error fetching configuration for ",n)}return null})}function ko(e){return y(this,null,function*(){return yield pc(e)})}t();G();t();function jo(e){let o=I.getState(),n=Si(o);return e?n&&!isNaN(n)?Math.max(n,e.nonce):e.nonce:n}t();t();G();var sc=()=>y(void 0,null,function*(){try{let e=yield Ho();try{let o=yield ko(e);if(o!=null){let n=P(b({},o),{nonce:jo(o)});return I.dispatch(mn(n)),n}}catch(o){console.error("Failed getting account ",o)}}catch(e){console.error("Failed getting address ",e)}return null});function Bn(){return y(this,null,function*(){let e=ae();if(e==null)throw"Provider not initialized";if(!e.isInitialized||e.isInitialized())return sc();try{if(!e.init)throw"Current provider does not have init() function";return(yield e.init())?sc():void 0}catch(o){console.error("Failed initializing provider ",o)}})}t();var mc=require("@elrondnetwork/erdjs");t();re();le();Ro();po();co();var rm=e=>{Ee.local.setItem({key:we.logoutEvent,data:e,expires:0}),Ee.local.removeItem(we.logoutEvent)};function Un(e,o){return y(this,null,function*(){let n=ae(),r=Ve(n),a=ie(),m=r==="wallet";if(!a||!n){or(e,o,!1);return}try{let i=yield Ho();rm(i)}catch(i){or(e,o,!1),console.error("error fetching logout address",i)}m&&ya(),I.dispatch(Q());try{let _=m&&!e?window.location.origin:e;yield n.logout({callbackUrl:_}),or(e,o,m)}catch(i){console.error("error logging out",i)}})}function or(e,o,n){e&&!n&&(typeof o=="function"?o(e):go(e))}t();t();t();t();G();t();t();t();t();t();po();Ro();ct();t();t();t();function _c(e){return new Promise(o=>{setTimeout(()=>o(),e)})}var dc=(e,o,n,r=0)=>y(void 0,null,function*(){try{return yield e(...n)}catch(a){return r<o.retries?((o==null?void 0:o.delay)!=null&&(yield _c(o.delay)),dc(e,o,n,r+1)):null}}),lc=(e,o={retries:5,delay:500})=>(...n)=>y(void 0,null,function*(){return dc(e,o,n)});t();t();var uc=M(require("axios"));function Wn(e){return y(this,null,function*(){try{let{data:o}=yield uc.default.get(`${e}/websocket/config`);return`wss://${o.url}`}catch(o){throw console.error(o),new Error("Can not get websocket url")}})}t();t();var am=M(require("platform"));t();function Gn(e){let o=vo(I.getState()),n=Object.keys(e).length===(o==null?void 0:o.transactions.length);return e&&(e&&n)}var im=(e,o)=>o.map((n,r)=>(n.setNonce(e+r),n)),cm=()=>{let e=j(),o=(0,Io.useRef)("/"),n=w(Ue),{provider:r}=eo(),a=Ve(r),[m,i]=(0,Io.useState)(null),_=w(vo),d=w(Hi),u=Boolean(_==null?void 0:_.transactions),g=()=>{i(null),e(Bo(null))},f=N=>{g(),T(N)};Jt(f);function T(N){var O,R;let D=r instanceof nr.ExtensionProvider;e(To()),e(gn(N)),e(Pt(rn)),D&&(g(),(R=(O=nr.ExtensionProvider.getInstance())==null?void 0:O.cancelAction)==null||R.call(O))}let h=(N,D)=>{let O=N.includes(so);if(T(D),O){e(Bo(so));return}i(N)},E=(N,D,O="")=>{let R={[No]:D},q=`${window.location.origin}${O}`,X=Fi({callbackUrl:q,urlParams:R});r.signTransactions(N,{callbackUrl:encodeURIComponent(X)})},F=()=>y(void 0,null,function*(){var me;let{sessionId:N,transactions:D,callbackRoute:O,customTransactionInformation:R}=_,{redirectAfterSign:q}=R,X=O||window.location.pathname,pe=window.location.pathname.includes(X),se=q&&!pe;try{if(!(yield(me=r==null?void 0:r.init)==null?void 0:me.call(r)))return}catch($){let U=($==null?void 0:$.message)||$||Yr;console.error(U),h(U);return}try{let $=yield r.signTransactions(D);if(!Gn($))return;let K=Object.values($).map(V=>So(V)),Z={sessionId:N,transactions:K,status:"signed"};se&&(Z.redirectRoute=X),e(De(Z))}catch($){let U=($==null?void 0:$.message)||$||Xr;console.error(U),e(De({sessionId:N,status:"cancelled"})),h(U.includes("cancel")?so:U,N)}}),C=()=>y(void 0,null,function*(){if(!_)return;g();let{sessionId:N,transactions:D,callbackRoute:O}=_;if(!r){console.error(Zr);return}o.current=O||window.location.pathname;try{let R=yield ko(n);if(R==null)return;let q=a==="wallet",X=!["wallet","ledger","extra"].includes(a),pe=jo(R),se=im(pe,D);q&&E(se,N,O),X&&F()}catch(R){let X=(R==null?void 0:R.message)||Kr;console.error(X),h(X,N),e(De({sessionId:N,status:"cancelled"})),console.error(X,R)}});return(0,Io.useEffect)(()=>{C()},[_]),{error:m,canceledTransactionsMessage:d,onAbort:f,hasTransactions:u,callbackRoute:o.current,sessionId:_==null?void 0:_.sessionId,transactions:_==null?void 0:_.transactions}};t();function tr(){let e=w(Je),o=Object.entries(e),n=(o==null?void 0:o.length)>0;return{signedTransactions:e,signedTransactionsArray:o,hasSignedTransactions:n}}t();var gc=M(require("axios")),fc=M(require("swr"));var pm=e=>gc.default.get(e).then(o=>o.data);function sm({tokenId:e}){var T;let{network:o}=Vo(),{isEsdt:n,isNft:r}=Xt(e),a=n?jt:zt,m=e;if(r){let[h,E]=e.split("-");m=`${h}-${E}`}let{data:i,error:_}=(0,fc.default)(Boolean(m)?`${o.apiAddress}/${a}/${m}`:null,pm);if(!m)return{tokenDecimals:Number(o.decimals),tokenLabel:"",tokenAvatar:""};let d=i?i==null?void 0:i.decimals:Number(o.decimals),u=i?i==null?void 0:i.name:"",g=i?`${(T=i==null?void 0:i.assets)==null?void 0:T.svgUrl}`:"",f=i==null?void 0:i.assets;return{tokenDecimals:d,tokenLabel:u,tokenAvatar:g,assets:f,error:_}}t();t();var hc=M(require("axios"));function Tc(e){return y(this,null,function*(){let{network:{apiAddress:o,apiTimeout:n}}=ve(I.getState()),{data:r}=yield hc.default.get(`/${$o}/${e}`,{baseURL:o,timeout:Number(n)});return r})}t();var rr=()=>w(wi);t();var oo=require("react");t();var zo=require("react");var xc={tokenId:"",amount:"",type:"",multiTxData:"",receiver:""};function ar({transactions:e}){let[o,n]=(0,zo.useState)({}),[r,a]=(0,zo.useState)([]);function m(d,u){n(g=>P(b({},g),{[d]:u}))}function i(d,u){return o==null?xc:d in o?o[d]:u!=null&&String(u)in o?o[u]:xc}function _(){if(e&&e.length>0){let d=[];e.forEach((u,g)=>{let f=u.getData().toString(),T=Pi(f);if(T.length>0)T.forEach((h,E)=>{let F={transaction:u,multiTxData:h.data,transactionIndex:E};m(h.data,{tokenId:h.token?h.token:"",amount:h.amount?h.amount:"",type:h.type,nonce:h.nonce?h.nonce:"",multiTxData:h.data,receiver:h.receiver}),d.push(F)});else{let{tokenId:h,amount:E}=Ht(u.getData().toString());h&&m(u.getData().toString(),{tokenId:h,amount:E,receiver:u.getReceiver().bech32()}),d.push({transaction:u,transactionIndex:g})}}),a(d)}}return(0,zo.useEffect)(()=>{_()},[e==null?void 0:e.length]),{parsedTransactionsByDataField:o,getTxInfoByDataField:i,allTransactions:r}}xt();var $n={};function ir({isLedger:e=!1,transactionsToSign:o,egldLabel:n,address:r,onCancel:a,onSignTransaction:m,onTransactionsSignError:i,onTransactionsSignSuccess:_,onGetScamAddressData:d}){var V,J,B;let[u,g]=(0,oo.useState)(0),[f,T]=(0,oo.useState)(),[h,E]=(0,oo.useState)(null),[F,C]=(0,oo.useState)(!1),{getTxInfoByDataField:N,allTransactions:D}=ar({transactions:o}),O=u===D.length-1;(0,oo.useEffect)(()=>{R()},[u,D]);function R(){return y(this,null,function*(){var gr;let A=D[u];if(A==null)return;let{transaction:x,multiTxData:v}=A,L=x.getData().toString(),oe=N(x.getData().toString(),v),{tokenId:Xo}=oe,Fe=x.getReceiver().toString(),ur=r!==Fe,$c=Fe in $n;if(ur&&!$c&&d!=null){let Yo=yield d(Fe);$n=b(b({},$n),Yo!=null&&Yo.scamInfo?{[Fe]:Yo.scamInfo}:{})}let Vc=Boolean(Xo&&Ri({tokenId:Xo,erdLabel:n}));E({transaction:x,receiverScamInfo:((gr=$n[Fe])==null?void 0:gr.info)||null,transactionTokenInfo:oe,isTokenTransaction:Vc,dataField:L})})}function q(){g(0),T(void 0),C(!1)}function X(){return y(this,null,function*(){try{if(h==null)return;C(e);let A=yield m(h.transaction),x={[u]:A},v=f?b(b({},f),x):x;T(v),O?v&&(_(Object.values(v)),q()):(g(L=>L+1),C(!1))}catch(A){console.error(A,"sign error");let{message:x}=A,v=e?Xe(A).errorMessage:null;q(),i(v!=null?v:x)}})}function pe(){try{if(h==null)return;h.transaction.getSignature().hex()?O||g(x=>x+1):X()}catch(A){X()}}let se=u===0;function me(){se?a==null||a():g(A=>A-1)}let $=Boolean(((V=h==null?void 0:h.transactionTokenInfo)==null?void 0:V.type)&&((J=h==null?void 0:h.transactionTokenInfo)==null?void 0:J.multiTxData)&&!(h!=null&&h.dataField.endsWith((B=h==null?void 0:h.transactionTokenInfo)==null?void 0:B.multiTxData)));function U(){$?g(A=>A+1):pe()}function K(){g(A=>{let x=A+1;return x>(D==null?void 0:D.length)?A:x})}function Z(){g(A=>{let x=A-1;return x<0?A:x})}return{allTransactions:D,onSignTransaction:U,onNext:K,onPrev:Z,waitingForDevice:F,onAbort:me,isLastTransaction:O,isFirstTransaction:u===0,hasMultipleTransactions:D.length>1,shouldContinueWithoutSigning:$,currentStep:u,signedTransactions:f,currentTransaction:h}}G();le();Ro();t();_o();G();function yc(){let e=j();return o=>{e(To()),e(gn(o)),e(Bo(so))}}function cr({onCancel:e,verifyReceiverScam:o=!0}){let n=w(vo),r=w(Ot),{account:{address:a}}=rr(),{provider:m}=eo(),i=j(),_=yc(),{transactions:d,sessionId:u,callbackRoute:g,customTransactionInformation:f}=n||{};function T(D){i(Ba(D))}let h=g!=null&&window.location.pathname.includes(g);function E(D){!Gn(D)||u&&(i(De({sessionId:u,status:"signed",transactions:D.map(R=>So(R))})),g!=null&&(f==null?void 0:f.redirectAfterSign)&&!h&&go(g))}function F(){e(),_(u)}function C(D){return y(this,null,function*(){return yield m.signTransaction(D)})}let N=ir({address:a,egldLabel:r,transactionsToSign:d,onGetScamAddressData:o?Tc:null,isLedger:ce("ledger"),onCancel:F,onSignTransaction:C,onTransactionsSignError:T,onTransactionsSignSuccess:E});return P(b({},N),{callbackRoute:g})}t();function mm(e){return cr(e)}t();var _m=()=>w(Vi);t();t();t();function pr(){let e=w(Pn),o=Object.entries(e),n=(o==null?void 0:o.length)>0;return{pendingTransactions:e,pendingTransactionsArray:o,hasPendingTransactions:n}}ho();t();t();t();var dm=M(require("axios"));t();var bc=M(require("axios"));function wc(e){return y(this,null,function*(){let o=kn(I.getState()),n=e.map(a=>a.hash),{data:r}=yield bc.default.get(`${o}/transactions`,{params:{hashes:n.join(","),withScResults:!0}});return e.map(({hash:a,previousStatus:m})=>{let i=r.find(_=>(_==null?void 0:_.txHash)===a);return{hash:a,data:i==null?void 0:i.data,invalidTransaction:i==null,status:i==null?void 0:i.status,results:i==null?void 0:i.results,sender:i==null?void 0:i.sender,receiver:i==null?void 0:i.receiver,previousStatus:m,hasStatusChanged:i&&i.status!==m}})})}t();var um=M(require("axios"));t();t();var gm=M(require("axios"));t();var fm=M(require("axios"));t();G();H();t();ho();function vc(e,o){return e.reduce((r,{status:a,hash:m})=>(m!=null&&!o.includes(m)&&Se(a)&&r.push({hash:m,previousStatus:a}),r),[])}t();G();H();function kc({results:e,hash:o,sessionId:n}){let r=e==null?void 0:e.find(a=>(a==null?void 0:a.returnMessage)!=="");I.dispatch(Oo({transactionHash:o,sessionId:n,status:"fail",errorMessage:r==null?void 0:r.returnMessage})),I.dispatch(dn({sessionId:n,status:"fail",errorMessage:r==null?void 0:r.returnMessage}))}t();G();le();function sr(e){I.dispatch(dn({sessionId:e,status:"timedOut"}))}var Vn={},mr=[];function hm({serverTransaction:e,sessionId:o,customTransactionInformation:n,shouldRefreshBalance:r}){let{hash:a,status:m,results:i,invalidTransaction:_,hasStatusChanged:d}=e;try{if(mr.includes(a))return;if(Vn[a]>30){sr(o);return}if(_||Se(m)){Vn[a]=Vn[a]?Vn[a]+1:1;return}d&&(Oe(m)&&(n==null?void 0:n.completedTransactionsDelay)!=null?(mr.push(a),setTimeout(()=>I.dispatch(Oo({sessionId:o,status:"success",transactionHash:a})),n==null?void 0:n.completedTransactionsDelay)):I.dispatch(Oo({sessionId:o,status:m,transactionHash:a}))),r||Bn(),Be(m)&&kc({sessionId:o,hash:a,results:i})}catch(u){console.error(u),sr(o)}}function Ic(m){return y(this,arguments,function*({sessionId:e,transactionBatch:{transactions:o,customTransactionInformation:n},getTransactionsByHash:r=wc,shouldRefreshBalance:a}){try{if(o==null)return;let i=vc(o,mr),_=yield r(i);for(let d of _)hm({serverTransaction:d,sessionId:e,customTransactionInformation:n,shouldRefreshBalance:a})}catch(i){console.error(i)}})}function Tm(){let{pendingTransactionsArray:e}=pr();function o(n){return y(this,null,function*(){let r=e.filter(([a,m])=>a!=null&&Se(m.status));if(r.length>0)for(let[a,m]of r)yield Ic(b({sessionId:a,transactionBatch:m},n));n.shouldRefreshBalance&&Bn()})}return o}t();var qo=require("react");t();t();t();G();function Ac(){I.dispatch(Oa())}function Ec(){I.dispatch(To())}t();t();var wm=M(require("bignumber.js"));t();var xm=`.dapp-core-component__main__link-style {
  color: #1392ff;
}

.dapp-core-component__main__link-second-style {
  color: #9dabbd;
}
.dapp-core-component__main__link-second-style:hover {
  color: #1392ff;
}
.dapp-core-component__main__link-second-style:hover svg {
  color: #1392ff;
}

.dapp-core-component__main__wrapper {
  min-height: 100vh;
}

.dapp-core-component__main__opacity-6 {
  opacity: 0.6;
}

.dapp-core-component__main__qr-code-svg-container {
  width: 15rem;
  height: 15rem;
}

.dapp-core-component__main__pair-list {
  max-height: 20rem;
  overflow-y: auto;
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__pair-image {
  width: 4rem;
  overflow: hidden;
  height: auto;
  min-width: 2rem;
}

.dapp-core-component__main__pair-remove {
  padding: 0 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

.dapp-core-component__main__dapp-icon {
  padding: 5px;
  border-radius: 50%;
  background-color: #fafafa;
  width: 4.6428571429rem;
  height: 4.6428571429rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}
.dapp-core-component__main__dapp-icon.dapp-core-component__main__icon-medium {
  width: 80px;
  height: 80px;
}

.dapp-core-component__main__centering {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dapp-core-component__main__border-n {
  border: none !important;
}

.dapp-core-component__main__t-shadow {
  text-shadow: 0px 4px 7px rgba(0, 0, 0, 0.3);
}

.dapp-core-component__main__btn {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.04) !important;
}

.dapp-core-component__main__media {
  display: flex;
  align-items: flex-start;
}

.dapp-core-component__main__media-body {
  flex: 1;
}

.dapp-core-component__main__connect-btns .dapp-core-component__main__btn {
  display: flex;
  height: 48px;
  justify-content: center;
  align-items: center;
  color: #fff !important;
  background-color: #1392ff;
  padding: 15px 20px 15px 16px;
  border-radius: 6px;
  gap: 8px;
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__btn:hover, .dapp-core-component__main__connect-btns .dapp-core-component__main__btn:active {
  background-color: #1392ff;
  background-image: linear-gradient(rgba(0, 0, 0, 0.1) 0 0);
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__info {
  color: #1392ff;
  margin-left: 48px;
}

.dapp-core-component__main__btn {
  transition: none !important;
}

.dapp-core-component__main__trust-badge {
  width: 20px;
  height: 20px;
}

/* Navbar
  -------------------------------------------------- */
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name {
  display: flex;
  align-items: center;
  font-size: 1.7142857143rem;
  font-weight: 400;
  color: #9ba5b4;
  margin-left: 1.6428571429rem;
}
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name:before {
  content: "";
  display: block;
  width: 1px;
  height: 3rem;
  margin-right: 1.7142857143rem;
}

/* Dashboard
  -------------------------------------------------- */
.dapp-core-component__main__action-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  margin: 0 7px;
}
.dapp-core-component__main__action-btn button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #fff;
}
.dapp-core-component__main__action-btn button:hover {
  opacity: 0.9;
}

.dapp-core-component__main__transactions .dapp-core-component__main__transaction-icon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table td {
  white-space: nowrap;
  vertical-align: middle;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table tr:last-of-type td {
  border-bottom: 1px solid #dee2e6;
}

/* Loader, Transaction Success/Fail
  -------------------------------------------------- */
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div {
  position: absolute;
  top: 27px;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.1);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.dapp-core-component__main__page-state .dapp-core-component__main__bg-blue .dapp-core-component__main__lds-ellipsis div {
  background: rgba(255, 255, 255, 0.4);
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(1) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis1 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(2) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(3) {
  left: 26px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(4) {
  left: 45px;
  animation: dapp-core-component__main__lds-ellipsis3 0.6s infinite;
}
@keyframes dapp-core-component__main__lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(19px, 0);
  }
}

.dapp-core-component__main__btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
}
.dapp-core-component__main__btn .dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn .dapp-core-component__main__disabled {
  opacity: 0.65;
  cursor: default;
}
.dapp-core-component__main__btn:hover {
  color: #212529;
  text-decoration: none;
}
.dapp-core-component__main__btn:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn:disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn-primary {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary:hover {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-primary:focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary:hover {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-secondary:focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-warning {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning:hover {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning:focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-dark {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark:hover {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
}
.dapp-core-component__main__btn-dark:focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-link {
  font-weight: 400;
  color: #007bff;
  text-decoration: none;
}
.dapp-core-component__main__btn-link:hover {
  color: #0056b3;
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:focus, .dapp-core-component__main__btn-link.dapp-core-component__main__focus {
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:disabled, .dapp-core-component__main__btn-link.dapp-core-component__main__disabled {
  color: #6c757d;
  pointer-events: none;
}

.dapp-core-component__main__btn-light {
  color: #212529;
  background-color: #fafafa;
  border-color: #fafafa;
}
.dapp-core-component__main__btn-light .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}
.dapp-core-component__main__btn-light .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #f8f9fa;
  border-color: #f8f9fa;
}
.dapp-core-component__main__btn-light:hover {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
}
.dapp-core-component__main__btn-light:focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .dapp-core-component__main__badge {
    transition: none;
  }
}
a.dapp-core-component__main__badge:hover, a.dapp-core-component__main__badge:focus {
  text-decoration: none;
}

.dapp-core-component__main__badge:empty {
  display: none;
}

.dapp-core-component__main__btn .dapp-core-component__main__badge {
  position: relative;
  top: -1px;
}

.dapp-core-component__main__badge-pill {
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.dapp-core-component__main__badge-primary {
  color: #fff;
  background-color: #007bff;
}

a.dapp-core-component__main__badge-primary:hover, a.dapp-core-component__main__badge-primary:focus {
  color: #fff;
  background-color: #0062cc;
}

a.dapp-core-component__main__badge-primary:focus, a.dapp-core-component__main__badge-primary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.5);
}

.dapp-core-component__main__badge-secondary {
  color: #fff;
  background-color: #6c757d;
}

a.dapp-core-component__main__badge-secondary:hover, a.dapp-core-component__main__badge-secondary:focus {
  color: #fff;
  background-color: #545b62;
}

a.dapp-core-component__main__badge-secondary:focus, a.dapp-core-component__main__badge-secondary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108, 117, 125, 0.5);
}

.dapp-core-component__main__badge-success {
  color: #fff;
  background-color: #28a745;
}

a.dapp-core-component__main__badge-success:hover, a.dapp-core-component__main__badge-success:focus {
  color: #fff;
  background-color: #1e7e34;
}

a.dapp-core-component__main__badge-success:focus, a.dapp-core-component__main__badge-success.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
}

.dapp-core-component__main__badge-info {
  color: #fff;
  background-color: #17a2b8;
}

a.dapp-core-component__main__badge-info:hover, a.dapp-core-component__main__badge-info:focus {
  color: #fff;
  background-color: #117a8b;
}

a.dapp-core-component__main__badge-info:focus, a.dapp-core-component__main__badge-info.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(23, 162, 184, 0.5);
}

.dapp-core-component__main__badge-warning {
  color: #212529;
  background-color: #ffc107;
}

a.dapp-core-component__main__badge-warning:hover, a.dapp-core-component__main__badge-warning:focus {
  color: #212529;
  background-color: #d39e00;
}

a.dapp-core-component__main__badge-warning:focus, a.dapp-core-component__main__badge-warning.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 193, 7, 0.5);
}

.dapp-core-component__main__badge-danger {
  color: #fff;
  background-color: #dc3545;
}

a.dapp-core-component__main__badge-danger:hover, a.dapp-core-component__main__badge-danger:focus {
  color: #fff;
  background-color: #bd2130;
}

a.dapp-core-component__main__badge-danger:focus, a.dapp-core-component__main__badge-danger.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(220, 53, 69, 0.5);
}

.dapp-core-component__main__badge-light {
  color: #212529;
  background-color: #f8f9fa;
}

a.dapp-core-component__main__badge-light:hover, a.dapp-core-component__main__badge-light:focus {
  color: #212529;
  background-color: #dae0e5;
}

a.dapp-core-component__main__badge-light:focus, a.dapp-core-component__main__badge-light.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(248, 249, 250, 0.5);
}

.dapp-core-component__main__badge-dark {
  color: #fff;
  background-color: #343a40;
}

a.dapp-core-component__main__badge-dark:hover, a.dapp-core-component__main__badge-dark:focus {
  color: #fff;
  background-color: #1d2124;
}

a.dapp-core-component__main__badge-dark:focus, a.dapp-core-component__main__badge-dark.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}

.dapp-core-component__main__card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 5px;
}
.dapp-core-component__main__card > hr {
  margin-right: 0;
  margin-left: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group {
  border-top: inherit;
  border-bottom: inherit;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:first-child {
  border-top-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:last-child {
  border-bottom-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__card-header + .dapp-core-component__main__list-group,
.dapp-core-component__main__card > .dapp-core-component__main__list-group + .dapp-core-component__main__card-footer {
  border-top: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header {
  border: 0;
  padding: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header:first-child {
  border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item {
  border-bottom: 1px solid #dee2e6;
  padding: 1.1rem 1rem;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item h6,
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item .dapp-core-component__main__h6 {
  line-height: 2.5rem;
  letter-spacing: unset;
  margin: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item.dapp-core-component__main__compact {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fafafa;
}

.dapp-core-component__main__card-body {
  flex: 1 1 auto;
  min-height: 1px;
  padding: 1.25rem;
}

.dapp-core-component__main__card-title {
  margin-bottom: 0.75rem;
}

.dapp-core-component__main__card-subtitle {
  margin-top: -0.375rem;
  margin-bottom: 0;
}

.dapp-core-component__main__card-text:last-child {
  margin-bottom: 0;
}

.dapp-core-component__main__card-link:hover {
  text-decoration: none;
}
.dapp-core-component__main__card-link + .dapp-core-component__main__card-link {
  margin-left: 1.25rem;
}

.dapp-core-component__main__card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: #fff;
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__card-footer {
  padding: 0.75rem 1.25rem;
  background-color: #fff;
  border-top: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__text {
  color: #9dabbd;
}

.dapp-core-component__main__text-white {
  color: #fff;
}

.dapp-core-component__main__text-monospace {
  font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-size-xl .dapp-core-component__main__trim {
  max-width: 13rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim {
  max-width: 10rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-only-sm .dapp-core-component__main__trim {
  max-width: none;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-left {
  text-align: left;
}

.dapp-core-component__main__text-right {
  text-align: right;
}

.dapp-core-component__main__text-center {
  text-align: center;
}

.dapp-core-component__main__text-primary {
  color: #1b46c2;
}

.dapp-core-component__main__text-secondary {
  color: #6c757d;
}

.dapp-core-component__main__text-primary-highlight {
  color: #1b46c2;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-success {
  color: #28a745;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-body {
  color: #212529;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-black-50 {
  color: rgba(0, 0, 0, 0.5);
}

.dapp-core-component__main__text-white-50 {
  color: rgba(255, 255, 255, 0.5);
}

@media (min-width: 992px) {
  .dapp-core-component__main__text-lg-right {
    text-align: right;
  }
}

.dapp-core-component__main__text-hide {
  font: 0/0 a;
  color: rgba(0, 0, 0, 0);
  text-shadow: none;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
}

.dapp-core-component__main__text-decoration-none {
  text-decoration: none;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__text-reset {
  color: inherit;
}

.dapp-core-component__main__text-lowercase {
  text-transform: lowercase;
}

.dapp-core-component__main__text-uppercase {
  text-transform: uppercase;
}

.dapp-core-component__main__text-capitalize {
  text-transform: capitalize;
}

.dapp-core-component__main__text-danger {
  color: #d9534f;
}

.dapp-core-component__main__text-warning {
  color: #f0ad4e;
}

.dapp-core-component__main__trim-fs-sm {
  color: inherit;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__m-0 {
  margin: 0;
}

.dapp-core-component__main__mt-0 {
  margin-top: 0;
}

.dapp-core-component__main__my-0 {
  margin-top: 0;
  margin-bottom: 0;
}

.dapp-core-component__main__mr-0 {
  margin-right: 0;
}

.dapp-core-component__main__mx-0 {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__mb-0 {
  margin-bottom: 0;
}

.dapp-core-component__main__my-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.dapp-core-component__main__ml-0 {
  margin-left: 0;
}

.dapp-core-component__main__mx-0 {
  margin-left: 0;
  margin-right: 0;
}

.dapp-core-component__main__m-1 {
  margin: 0.25rem;
}

.dapp-core-component__main__mt-1 {
  margin-top: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__mr-1 {
  margin-right: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.dapp-core-component__main__mb-1 {
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-bottom: 0.25rem;
  margin-top: 0.25rem;
}

.dapp-core-component__main__ml-1 {
  margin-left: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-right: 0.25rem;
  margin-left: 0.25rem;
}

.dapp-core-component__main__m-2 {
  margin: 0.5rem;
}

.dapp-core-component__main__mt-2 {
  margin-top: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-bottom: 0.5rem;
  margin-top: 0.5rem;
}

.dapp-core-component__main__mr-2 {
  margin-right: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__mb-2 {
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__ml-2 {
  margin-left: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__m-3 {
  margin: 1rem;
}

.dapp-core-component__main__mt-3,
.dapp-core-component__main__my-3 {
  margin-top: 1rem;
}

.dapp-core-component__main__mr-3 {
  margin-right: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__mb-3 {
  margin-bottom: 1rem;
}

.dapp-core-component__main__my-3 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.dapp-core-component__main__ml-3 {
  margin-left: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__m-4 {
  margin: 1.5rem;
}

.dapp-core-component__main__mt-4 {
  margin-top: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__mr-4 {
  margin-right: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-right: 1.5rem;
  margin-left: 1.5rem;
}

.dapp-core-component__main__mb-4 {
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.dapp-core-component__main__ml-4 {
  margin-left: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}

.dapp-core-component__main__m-5 {
  margin: 3rem;
}

.dapp-core-component__main__mt-5 {
  margin-top: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__mr-5 {
  margin-right: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-right: 3rem;
  margin-left: 3rem;
}

.dapp-core-component__main__mb-5 {
  margin-bottom: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__ml-5 {
  margin-left: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-left: 3rem;
  margin-right: 3rem;
}

.dapp-core-component__main__p-0 {
  padding: 0;
}

.dapp-core-component__main__pt-0 {
  padding-top: 0;
}

.dapp-core-component__main__py-0 {
  padding-top: 0;
  padding-bottom: 0;
}

.dapp-core-component__main__pr-0 {
  padding-right: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__pb-0 {
  padding-bottom: 0;
}

.dapp-core-component__main__py-0 {
  padding-bottom: 0;
  padding-top: 0;
}

.dapp-core-component__main__pl-0 {
  padding-left: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__p-1 {
  padding: 0.25rem;
}

.dapp-core-component__main__pt-1 {
  padding-top: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pr-1 {
  padding-right: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-right: 0.25rem;
  padding-left: 0.25rem;
}

.dapp-core-component__main__pb-1 {
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pl-1 {
  padding-left: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

.dapp-core-component__main__p-2 {
  padding: 0.5rem;
}

.dapp-core-component__main__pt-2 {
  padding-top: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.dapp-core-component__main__pr-2 {
  padding-right: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__pb-2 {
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__pl-2 {
  padding-left: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__p-3 {
  padding: 1rem;
}

.dapp-core-component__main__pt-3 {
  padding-top: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.dapp-core-component__main__pr-3 {
  padding-right: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__pb-3 {
  padding-bottom: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.dapp-core-component__main__pl-3 {
  padding-left: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__p-4 {
  padding: 1.5rem;
}

.dapp-core-component__main__pt-4 {
  padding-top: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pr-4 {
  padding-right: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.dapp-core-component__main__pb-4 {
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pl-4 {
  padding-left: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
}

.dapp-core-component__main__p-5 {
  padding: 3rem;
}

.dapp-core-component__main__pt-5 {
  padding-top: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-bottom: 3rem;
  padding-top: 3rem;
}

.dapp-core-component__main__pr-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__pb-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.dapp-core-component__main__pl-5 {
  padding-left: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__m-n1 {
  margin: -0.25rem;
}

.dapp-core-component__main__mt-n1 {
  margin-top: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__mr-n1 {
  margin-right: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__mb-n1 {
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-bottom: -0.25rem;
  margin-top: -0.25rem;
}

.dapp-core-component__main__ml-n1 {
  margin-left: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__m-n2 {
  margin: -0.5rem;
}

.dapp-core-component__main__mt-n2 {
  margin-top: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-top: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mr-n2 {
  margin-right: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mb-n2 {
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-bottom: -0.5rem;
  margin-top: -0.5rem;
}

.dapp-core-component__main__ml-n2 {
  margin-left: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.dapp-core-component__main__m-n3 {
  margin: -1rem;
}

.dapp-core-component__main__mt-n3 {
  margin-top: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-top: -1rem;
  margin-bottom: -1rem;
}

.dapp-core-component__main__mr-n3 {
  margin-right: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-right: -1rem;
  margin-left: -1rem;
}

.dapp-core-component__main__mb-n3 {
  margin-bottom: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-bottom: -1rem;
  margin-top: -1rem;
}

.dapp-core-component__main__ml-n3 {
  margin-left: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-left: -1rem;
  margin-right: -1rem;
}

.dapp-core-component__main__m-n4 {
  margin: -1.5rem;
}

.dapp-core-component__main__mt-n4 {
  margin-top: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-top: -1.5rem;
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__mr-n4 {
  margin-right: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__mb-n4 {
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-bottom: -1.5rem;
  margin-top: -1.5rem;
}

.dapp-core-component__main__ml-n4 {
  margin-left: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__m-n5 {
  margin: -3rem;
}

.dapp-core-component__main__mt-n5 {
  margin-top: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-bottom: -3rem;
  margin-top: -3rem;
}

.dapp-core-component__main__mr-n5 {
  margin-right: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-left: -3rem;
  margin-right: -3rem;
}

.dapp-core-component__main__mb-n5 {
  margin-bottom: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-top: -3rem;
  margin-bottom: -3rem;
}

.dapp-core-component__main__ml-n5 {
  margin-left: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-right: -3rem;
  margin-left: -3rem;
}

.dapp-core-component__main__m-auto {
  margin: auto;
}

.dapp-core-component__main__mt-auto {
  margin-top: auto;
}

.dapp-core-component__main__my-auto {
  margin-bottom: auto;
  margin-top: auto;
}

.dapp-core-component__main__mr-auto {
  margin-right: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__mb-auto {
  margin-bottom: auto;
}

.dapp-core-component__main__my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.dapp-core-component__main__ml-auto {
  margin-left: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-1 {
    margin-right: 0.25rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-4 {
    margin-right: 1.5rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mx-lg-4 {
    margin-left: 1.5rem;
    margin-right: 1.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__mb-xl-2 {
    margin-bottom: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__pr-sm-2 {
    padding-right: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__px-sm-2 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__pr-xl-0,
.dapp-core-component__main__px-xl-0 {
    padding-right: 0;
  }
}

.dapp-core-component__main__col {
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

.dapp-core-component__main__col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__col-xl,
.dapp-core-component__main__col-xl-auto,
.dapp-core-component__main__col-xl-12,
.dapp-core-component__main__col-xl-11,
.dapp-core-component__main__col-xl-10,
.dapp-core-component__main__col-xl-9,
.dapp-core-component__main__col-xl-8,
.dapp-core-component__main__col-xl-7,
.dapp-core-component__main__col-xl-6,
.dapp-core-component__main__col-xl-5,
.dapp-core-component__main__col-xl-4,
.dapp-core-component__main__col-xl-3,
.dapp-core-component__main__col-xl-2,
.dapp-core-component__main__col-xl-1,
.dapp-core-component__main__col-lg,
.dapp-core-component__main__col-lg-auto,
.dapp-core-component__main__col-lg-12,
.dapp-core-component__main__col-lg-11,
.dapp-core-component__main__col-lg-10,
.dapp-core-component__main__col-lg-9,
.dapp-core-component__main__col-lg-8,
.dapp-core-component__main__col-lg-7,
.dapp-core-component__main__col-lg-6,
.dapp-core-component__main__col-lg-5,
.dapp-core-component__main__col-lg-4,
.dapp-core-component__main__col-lg-3,
.dapp-core-component__main__col-lg-2,
.dapp-core-component__main__col-lg-1,
.dapp-core-component__main__col-md,
.dapp-core-component__main__col-md-auto,
.dapp-core-component__main__col-md-12,
.dapp-core-component__main__col-md-11,
.dapp-core-component__main__col-md-10,
.dapp-core-component__main__col-md-9,
.dapp-core-component__main__col-md-8,
.dapp-core-component__main__col-md-7,
.dapp-core-component__main__col-md-6,
.dapp-core-component__main__col-md-5,
.dapp-core-component__main__col-md-4,
.dapp-core-component__main__col-md-3,
.dapp-core-component__main__col-md-2,
.dapp-core-component__main__col-md-1,
.dapp-core-component__main__col-sm,
.dapp-core-component__main__col-sm-auto,
.dapp-core-component__main__col-sm-12,
.dapp-core-component__main__col-sm-11,
.dapp-core-component__main__col-sm-10,
.dapp-core-component__main__col-sm-9,
.dapp-core-component__main__col-sm-8,
.dapp-core-component__main__col-sm-7,
.dapp-core-component__main__col-sm-6,
.dapp-core-component__main__col-sm-5,
.dapp-core-component__main__col-sm-4,
.dapp-core-component__main__col-sm-3,
.dapp-core-component__main__col-sm-2,
.dapp-core-component__main__col-sm-1,
.dapp-core-component__main__col,
.dapp-core-component__main__col-auto,
.dapp-core-component__main__col-12,
.dapp-core-component__main__col-11,
.dapp-core-component__main__col-10,
.dapp-core-component__main__col-9,
.dapp-core-component__main__col-8,
.dapp-core-component__main__col-7,
.dapp-core-component__main__col-6,
.dapp-core-component__main__col-5,
.dapp-core-component__main__col-4,
.dapp-core-component__main__col-3,
.dapp-core-component__main__col-2,
.dapp-core-component__main__col-1 {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
}

.dapp-core-component__main__col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

@media (min-width: 576px) {
  .dapp-core-component__main__col-sm-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-sm-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-sm-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-sm-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-sm-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-sm-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-sm-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-sm-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-sm-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-sm-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-sm-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-sm-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 768px) {
  .dapp-core-component__main__col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__col-xl-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-xl-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-xl-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-xl-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-xl-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-xl-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-xl-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-xl-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-xl-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-xl-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-xl-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-xl-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
.dapp-core-component__main__flex-row {
  flex-direction: row;
}

.dapp-core-component__main__flex-column {
  flex-direction: column;
}

.dapp-core-component__main__flex-row-reverse {
  flex-direction: row-reverse;
}

.dapp-core-component__main__flex-column-reverse {
  flex-direction: column-reverse;
}

.dapp-core-component__main__flex-wrap {
  flex-wrap: wrap;
}

.dapp-core-component__main__flex-nowrap {
  flex-wrap: nowrap;
}

.dapp-core-component__main__flex-wrap-reverse {
  flex-wrap: wrap-reverse;
}

.dapp-core-component__main__flex-fill {
  flex: 1 1 auto;
}

.dapp-core-component__main__flex-grow-0 {
  flex-grow: 0;
}

.dapp-core-component__main__flex-grow-1 {
  flex-grow: 1;
}

.dapp-core-component__main__flex-shrink-0 {
  flex-shrink: 0;
}

.dapp-core-component__main__flex-shrink-1 {
  flex-shrink: 1;
}

.dapp-core-component__main__justify-content-start {
  justify-content: flex-start;
}

.dapp-core-component__main__justify-content-end {
  justify-content: flex-end;
}

.dapp-core-component__main__justify-content-center {
  justify-content: center;
}

.dapp-core-component__main__justify-content-between {
  justify-content: space-between;
}

.dapp-core-component__main__justify-content-around {
  justify-content: space-around;
}

.dapp-core-component__main__align-items-start {
  align-items: flex-start;
}

.dapp-core-component__main__align-items-end {
  align-items: flex-end;
}

.dapp-core-component__main__align-items-center {
  align-items: center;
}

.dapp-core-component__main__align-items-baseline {
  align-items: baseline;
}

.dapp-core-component__main__align-items-stretch {
  align-items: stretch;
}

.dapp-core-component__main__align-content-start {
  align-content: flex-start;
}

.dapp-core-component__main__align-content-end {
  align-content: flex-end;
}

.dapp-core-component__main__align-content-center {
  align-content: center;
}

.dapp-core-component__main__align-content-between {
  align-content: space-between;
}

.dapp-core-component__main__align-content-around {
  align-content: space-around;
}

.dapp-core-component__main__align-content-stretch {
  align-content: stretch;
}

.dapp-core-component__main__align-self-auto {
  align-self: auto;
}

.dapp-core-component__main__align-self-start {
  align-self: flex-start;
}

.dapp-core-component__main__align-self-end {
  align-self: flex-end;
}

.dapp-core-component__main__align-self-center {
  align-self: center;
}

.dapp-core-component__main__align-self-baseline {
  align-self: baseline;
}

.dapp-core-component__main__align-self-stretch {
  align-self: stretch;
}

@media (min-width: 576px) {
  .dapp-core-component__main__flex-sm-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-wrap {
    flex-wrap: wrap;
  }
}

.dapp-core-component__main__form-check {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: auto;
  padding-left: 0;
}

.dapp-core-component__main__form-check-label {
  margin-bottom: 0;
}

.dapp-core-component__main__form-check-input {
  position: absolute;
  margin-top: 0.3rem;
  margin-left: -1.25rem;
}
.dapp-core-component__main__form-check-input[disabled] ~ .dapp-core-component__main__form-check-label, .dapp-core-component__main__form-check-input:disabled ~ .dapp-core-component__main__form-check-label {
  color: #6c757d;
}

.dapp-core-component__main__form-group {
  display: flex;
  flex: 0 0 auto;
  flex-flow: row wrap;
  align-items: center;
  margin-bottom: 0;
}

.dapp-core-component__main__form-control {
  display: block;
  width: 100%;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.dapp-core-component__main__form-label {
  padding-top: calc(0.375rem + 1px);
  padding-bottom: calc(0.375rem + 1px);
  margin-bottom: 0;
  font-size: inherit;
  line-height: 1.5;
}

input,
button,
select,
optgroup,
textarea {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.dapp-core-component__main__form-group {
  margin-bottom: 1rem;
}

.dapp-core-component__main__form-text {
  display: block;
  margin-top: 0.25rem;
  color: inherit;
}

.dapp-core-component__main__border {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__border-top {
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__border-right {
  border-right: 1px solid #dee2e6;
}

.dapp-core-component__main__border-bottom {
  border-bottom: 1px solid #dee2e6;
}

.dapp-core-component__main__border-left {
  border-left: 1px solid #dee2e6;
}

.dapp-core-component__main__border-0 {
  border: 0;
}

.dapp-core-component__main__border-top-0 {
  border-top: 0;
}

.dapp-core-component__main__border-right-0 {
  border-right: 0;
}

.dapp-core-component__main__border-bottom-0 {
  border-bottom: 0;
}

.dapp-core-component__main__border-left-0 {
  border-left: 0;
}

.dapp-core-component__main__rounded-sm {
  border-radius: 0.2rem;
}

.dapp-core-component__main__rounded {
  border-radius: 0.25rem;
}

.dapp-core-component__main__rounded-top {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-right {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-bottom {
  border-bottom-right-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-left {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-lg {
  border-radius: 0.3rem;
}

.dapp-core-component__main__rounded-circle {
  border-radius: 50%;
}

.dapp-core-component__main__rounded-pill {
  border-radius: 50rem;
}

.dapp-core-component__main__rounded-0 {
  border-radius: 0;
}

.dapp-core-component__main__h1,
.dapp-core-component__main__h2,
.dapp-core-component__main__h3,
.dapp-core-component__main__h4,
.dapp-core-component__main__h5,
.dapp-core-component__main__h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2;
}

.dapp-core-component__main__h1 {
  font-size: 2.5rem;
}

.dapp-core-component__main__h2 {
  font-size: 2rem;
}

.dapp-core-component__main__h3 {
  font-size: 1.75rem;
}

.dapp-core-component__main__h4 {
  font-size: 1.5rem;
}

.dapp-core-component__main__h5 {
  font-size: 1.25rem;
}

.dapp-core-component__main__h6 {
  font-size: 1rem;
}

.dapp-core-component__main__lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

.dapp-core-component__main__bg-warning,
.dapp-core-component__main__warning {
  background-color: #f0ad4e;
}

.dapp-core-component__main__bg-danger,
.dapp-core-component__main__danger {
  background-color: #d9534f;
}

.dapp-core-component__main__bg-success,
.dapp-core-component__main__success {
  background-color: #5cb85c;
}

.dapp-core-component__main__position-relative {
  position: relative;
}

.dapp-core-component__main__position-absolute {
  position: absolute;
}

.dapp-core-component__main__w-25 {
  width: 25%;
}

.dapp-core-component__main__w-50 {
  width: 50%;
}

.dapp-core-component__main__w-75 {
  width: 75%;
}

.dapp-core-component__main__w-100 {
  width: 100%;
}

.dapp-core-component__main__w-auto {
  width: auto;
}

.dapp-core-component__main__h-25 {
  height: 25%;
}

.dapp-core-component__main__h-50 {
  height: 50%;
}

.dapp-core-component__main__h-75 {
  height: 75%;
}

.dapp-core-component__main__h-100 {
  height: 100%;
}

.dapp-core-component__main__h-auto {
  height: auto;
}

.dapp-core-component__main__mw-100 {
  max-width: 100%;
}

.dapp-core-component__main__mh-100 {
  max-height: 100%;
}

.dapp-core-component__main__min-vw-100 {
  min-width: 100vw;
}

.dapp-core-component__main__min-vh-100 {
  min-height: 100vh;
}

.dapp-core-component__main__vw-100 {
  width: 100vw;
}

.dapp-core-component__main__vh-100 {
  height: 100vh;
}

.dapp-core-component__main__locked-icon {
  color: inherit;
}

.dapp-core-component__main__shadow-sm {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
}

.dapp-core-component__main__shadow {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.dapp-core-component__main__shadow-lg {
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.175);
}

.dapp-core-component__main__shadow-none {
  box-shadow: none;
}

.dapp-core-component__main__d-none {
  display: none;
}

.dapp-core-component__main__d-inline {
  display: inline;
}

.dapp-core-component__main__d-inline-block {
  display: inline-block;
}

.dapp-core-component__main__d-block {
  display: block;
}

.dapp-core-component__main__d-table {
  display: table;
}

.dapp-core-component__main__d-table-row {
  display: table-row;
}

.dapp-core-component__main__d-table-cell {
  display: table-cell;
}

.dapp-core-component__main__d-flex {
  display: -ms-flexbox;
  display: flex;
}

.dapp-core-component__main__d-inline-flex {
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media (min-width: 576px) {
  .dapp-core-component__main__d-sm-flex {
    display: flex;
  }
}

.dapp-core-component__main__table {
  width: 100%;
  margin-bottom: 1rem;
  color: #212529;
}
.dapp-core-component__main__table th,
.dapp-core-component__main__table td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #dee2e6;
}
.dapp-core-component__main__table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #dee2e6;
}
.dapp-core-component__main__table tbody + tbody {
  border-top: 2px solid #dee2e6;
}

.dapp-core-component__main__table-sm th,
.dapp-core-component__main__table-sm td {
  padding: 0.3rem;
}

.dapp-core-component__main__table-bordered {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered th,
.dapp-core-component__main__table-bordered td {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered thead th,
.dapp-core-component__main__table-bordered thead td {
  border-bottom-width: 2px;
}

.dapp-core-component__main__table-borderless th,
.dapp-core-component__main__table-borderless td,
.dapp-core-component__main__table-borderless thead th,
.dapp-core-component__main__table-borderless tbody + tbody {
  border: 0;
}

.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(0, 0, 0, 0.05);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-dark th {
  color: #fff;
  background-color: #343a40;
  border-color: #454d55;
}
.dapp-core-component__main__table .dapp-core-component__main__thead-light th {
  color: #495057;
  background-color: #e9ecef;
  border-color: #dee2e6;
}

.dapp-core-component__main__table-dark {
  color: #fff;
  background-color: #343a40;
}
.dapp-core-component__main__table-dark th,
.dapp-core-component__main__table-dark td,
.dapp-core-component__main__table-dark thead th {
  border-color: #454d55;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-bordered {
  border: 0;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(255, 255, 255, 0.05);
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-hover tbody tr {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.075);
}

.dapp-core-component__main__table-responsive {
  display: block;
  width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}
.dapp-core-component__main__table-responsive > .dapp-core-component__main__table-bordered {
  border: 0;
}

@media (max-width: 767.98px) {
  .dapp-core-component__main__transactions.dapp-core-component__main__table .dapp-core-component__main__transaction-hash .dapp-core-component__main__trim {
    max-width: 8rem;
  }
}
.dapp-core-component__main__font-weight-light {
  font-weight: 300 !important;
}

.dapp-core-component__main__font-weight-lighter {
  font-weight: lighter !important;
}

.dapp-core-component__main__font-weight-normal {
  font-weight: 400 !important;
}

.dapp-core-component__main__font-weight-medium {
  font-weight: 500 !important;
}

.dapp-core-component__main__font-weight-bold {
  font-weight: 700 !important;
}

.dapp-core-component__main__font-weight-bolder {
  font-weight: bolder !important;
}

.dapp-core-component__main__font-italic {
  font-style: italic !important;
}

.dapp-core-component__main__icon-state {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
}
.dapp-core-component__main__icon-state.dapp-core-component__main__half {
  width: 3rem;
  height: 3rem;
}

.dapp-core-component__main__side-icon {
  width: 1.5rem;
  height: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  color: transparent;
  border-radius: 50%;
  overflow: hidden;
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__side-icon svg {
  color: #cacbcc;
}

div.dapp-core-component__main__side-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
div.dapp-core-component__main__side-icon img {
  transform: scale(0.75);
}

.dapp-core-component__main__toast-messages {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast {
  display: block;
  flex-basis: 4rem;
  width: 100%;
  max-width: 28.875rem;
  transition: transform 0.3s ease-in-out;
  transform: translateX(120%);
  background-color: #fff;
  border-radius: 0.25rem;
  border-color: #dee2e6;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__clickable {
  cursor: pointer;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible {
  margin-bottom: 0.3rem;
  transform: translateX(0);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress {
  background-color: rgba(0, 0, 0, 0);
  height: inherit;
  font-size: inherit;
  line-height: inherit;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar {
  background-color: rgba(108, 117, 125, 0.2);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar .dapp-core-component__main__content-height {
  z-index: -1;
  visibility: hidden;
  width: 0 !important;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast .dapp-core-component__main__tx-description {
  margin-bottom: 0.2rem;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close {
  opacity: 1;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close svg {
  color: #6c757d;
}

.dapp-core-component__main__data-field .dapp-core-component__main__textarea {
  -moz-appearance: textfield-multiline;
  -webkit-appearance: textarea;
  border: 1px solid #000;
  background-color: #fff;
  border-radius: 5px;
  font: medium -moz-fixed;
  font-size: 0.8125rem;
  min-height: 5rem;
  overflow: auto;
  padding: 0.375rem 0.75rem;
  resize: both;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea:hover {
  border: 1px solid #000;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea .dapp-core-component__main__highlighted {
  font-size: 1rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea.dapp-core-component__main__sc-call {
  min-height: 3rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__form-control:disabled,
.dapp-core-component__main__data-field .dapp-core-component__main__form-control[readonly] {
  background-color: #fafafa;
}
.dapp-core-component__main__data-field .dapp-core-component__main__small-font {
  font-size: 0.8125rem;
}

.dapp-core-component__main__progress-steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps hr {
  background-color: #6c757d;
  border-color: #dee2e6;
  z-index: 2;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot {
  height: 0.625rem;
  width: 0.625rem;
  border-radius: 50%;
  margin: 0 1rem;
  z-index: 5;
  background-color: #6c757d;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:first-of-type {
  margin-left: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:last-of-type {
  margin-right: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

a.dapp-core-component__main__side-link::after {
  right: 0;
  left: unset;
  width: calc(100% - 1.85rem);
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(xm));W();G();le();t();var bm=M(require("bignumber.js"));W();t();var Sm=require("@elrondnetwork/erdjs"),vm=M(require("bignumber.js"));W();H();ho();function km({transactionId:e,onSuccess:o,onFail:n,onCancelled:r,onTimedOut:a}){var R;let{signedTransactionsArray:m}=tr(),i=ce("wallet"),[_]=m.length>0?m[m.length-1]:[],d=(R=e!=null?e:_)!=null?R:null,u=i?d:e,g=w(q=>zi(q,u)),{status:f,transactions:T,errorMessage:h}=g,E=Se(f),F=Be(f),C=_n(f),N=Oe(f),D=f==="cancelled";function O(){i&&(Ac(),Ec())}return(0,qo.useEffect)(()=>{N&&o&&o(u)},[N]),(0,qo.useEffect)(()=>{F&&n&&(n(u,h),O())},[F]),(0,qo.useEffect)(()=>{D&&r&&(r(u),O())},[D]),(0,qo.useEffect)(()=>{C&&(a?a(u):n==null||n(u,"timeout"))},[C]),u==null?{}:g==null?{errorMessage:"No transaction to track"}:{isPending:E,isSuccessful:N,isFailed:F,isCancelled:D,errorMessage:h,status:f,transactions:T}}t();function Im(){let e=w(Rn),o=Object.entries(e),n=(o==null?void 0:o.length)>0;return{failedTransactions:e,failedTransactionsArray:o,hasFailedTransactions:n}}t();function Am(){let e=w(Nn),o=Object.entries(e),n=(o==null?void 0:o.length)>0;return{successfulTransactions:e,successfulTransactionsArray:o,hasSuccessfulTransactions:n}}t();function Em(){var d,u,g;let e=w(ji),o=w(Rn),n=w(Nn),r=w(Pn),a=((d=Object.keys(r))==null?void 0:d.length)>0,m=!a&&((u=Object.keys(e))==null?void 0:u.length)>0,i=!a&&!m&&((g=Object.keys(o))==null?void 0:g.length)>0,_=!a&&!m&&!i&&Object.keys(n).length>0;return{pending:a,timedOut:m,fail:i,success:_}}t();t();var _r=()=>w(bo);t();var Lm=()=>{let e=w(We),o=w(Ge);return P(b({},e),{isLoggedIn:o})};t();var Dm=()=>w(Ge);t();var Cm=()=>w(Sn);t();var Hn=require("react"),Pm=(e,o)=>{let[n,r]=(0,Hn.useState)(e);return(0,Hn.useEffect)(()=>{let m=setTimeout(()=>r(e),o);return()=>clearTimeout(m)},[e]),n};t();t();var dr=require("react"),Lc=require("@elrondnetwork/erdjs-extension-provider");_o();re();G();H();fo();var Dc=({callbackRoute:e,token:o,onLoginRedirect:n})=>{let[r,a]=(0,dr.useState)(""),[m,i]=(0,dr.useState)(!1),_=j(),d=ie();function u(){return y(this,null,function*(){if(d)throw new Error(mo);i(!0);let f=Lc.ExtensionProvider.getInstance();try{if(!(yield f.init())){console.warn("Something went wrong trying to redirect to wallet login..");return}let h=encodeURIComponent(`${window.location.origin}${e!=null?e:window.location.pathname}`),E=b({callbackUrl:h},o&&{token:o});yield f.login(E),Re(f);let{signature:F,address:C}=f.account;if(!C){i(!1),console.warn("Login cancelled.");return}F&&_(fe({loginToken:String(o),signature:F})),_(te({address:C,loginMethod:"extension"})),Le(e,n)}catch(T){console.error("error loging in",T),a("error logging in"+T.message)}finally{i(!1)}})}let g=Boolean(r);return[u,{loginFailed:g,error:r,isLoading:m&&!g,isLoggedIn:d&&!g}]};t();var lr=require("react");_o();G();var Cc=({callbackRoute:e,token:o,redirectDelayMilliseconds:n=100})=>{let[r,a]=(0,lr.useState)(""),[m,i]=(0,lr.useState)(!1),_=w(Pe),d=j(),u=ie();function g(){return y(this,null,function*(){if(u)throw new Error(mo);try{i(!0);let T=nc(_.walletAddress),h=new Date,E=h.setMinutes(h.getMinutes()+3)/1e3;d(fa({data:{},expires:E})),o&&d(fe({loginToken:o}));let C=encodeURIComponent(`${window.location.origin}${e}`),N=P(b({callbackUrl:C},o&&{token:o}),{redirectDelayMilliseconds:n});yield T.login(N)}catch(T){console.error(T),a("error logging in"+T.message)}finally{i(!1)}})}let f=Boolean(r);return[g,{error:r,loginFailed:f,isLoading:m&&!f,isLoggedIn:u&&!f}]};t();var he=require("react"),jn=require("@elrondnetwork/erdjs-hw-provider");_o();re();G();H();fo();var Ao="Could not initialise ledger app, make sure Elrond app is open",Nm=10;function Pc({callbackRoute:e,token:o,addressesPerPage:n=Nm,onLoginRedirect:r}){let a=w(vi),m=j(),i=ie(),[_,d]=(0,he.useState)(""),[u,g]=(0,he.useState)(!1),f=new jn.HWProvider,[T,h]=(0,he.useState)(0),[E,F]=(0,he.useState)([]),[C,N]=(0,he.useState)(""),[D,O]=(0,he.useState)(!1),[R,q]=(0,he.useState)(null),[X,pe]=(0,he.useState)(!1);function se({provider:x,address:v,index:L,signature:oe}){Re(x),m(ga({index:L,loginType:"ledger"})),oe&&m(fe({loginToken:String(o),signature:oe})),m(te({address:v,loginMethod:"ledger"})),Le(e,r)}let me=(x,v="")=>{let{errorMessage:L}=Xe(x);L&&d(`${L}.${v}`),g(!1),console.warn(x),m(kt(null))};function $(x){return y(this,null,function*(){if(R==null)return!1;let{index:v}=R;if(o)try{let L=yield x.tokenLogin({token:c.Buffer.from(`${o}{}`),addressIndex:v});se({address:L.address,provider:x,index:v,signature:L.signature.hex()})}catch(L){me(L,". Update Elrond App to continue.")}else try{let L=yield x.login({addressIndex:v});se({address:L,provider:x,index:v})}catch(L){return me(L),!1}return!0})}function U(){return y(this,null,function*(){try{if(g(!0),R==null)return!1;m(a?Ca(R):kt(P(b({},R),{version:C,hasContractDataEnabled:D})));let x=new jn.HWProvider;if(!(yield x.init()))return d(Ao),console.warn(Ao),!1;g(!1),yield $(x)}catch(x){let{errorMessage:v}=Xe(x);v&&d(v),console.warn(Ao,x)}finally{g(!1)}return pe(!1),!0})}function K(){return y(this,null,function*(){try{if(g(!0),!(yield f.init())){d(Ao),console.warn(Ao),g(!1);return}let v=yield f.getAccounts(T,n),L=yield tc(f);N(L.version),O(L.dataEnabled),F(v),g(!1)}catch(x){let{errorMessage:v,defaultErrorMessage:L}=Xe(x);d(v!=null?v:L),console.error("error",x),g(!1)}})}function Z(){return y(this,null,function*(){if(i)throw new Error(mo);d("");try{if(g(!0),a!=null){let x=new jn.HWProvider;if(!(yield x.init())||!R){console.warn(Ao);return}let L=yield x.login({addressIndex:R.index.valueOf()});if(Re(x),!L){g(!1),console.warn("Login cancelled.");return}m(te({address:L,loginMethod:"ledger"})),Le(e,r)}else(E==null?void 0:E.length)>0||(yield K()),pe(!0)}catch(x){console.error("error ",x);let{defaultErrorMessage:v}=Xe();d(v)}finally{g(!1)}})}function V(x){q(x)}function J(){q(null),h(x=>x+1)}function B(){q(null),h(x=>x===0?0:x-1)}(0,he.useEffect)(()=>{K()},[T]);let A=Boolean(_);return[Z,{loginFailed:A,isLoggedIn:i&&!A,error:_,isLoading:u&&!A},{accounts:E,showAddressList:X,startIndex:T,selectedAddress:R,version:C,contractDataEnabled:D,onGoToPrevPage:B,onGoToNextPage:J,onSelectAddress:V,onConfirmSelectedAddress:U}]}t();var He=require("react"),Nc=require("@elrondnetwork/erdjs-wallet-connect-provider");re();G();H();fo();var Rc=({logoutRoute:e,callbackRoute:o,token:n,onLoginRedirect:r})=>{let a=j(),m=15e3,[i,_]=(0,He.useState)(""),[d,u]=(0,He.useState)(""),{provider:g}=eo(),f=w(ki),T=w(vn),h=(0,He.useRef)(g),E=(0,He.useRef)(!0),F,C=Boolean(d),N=!C,D=C?`${T}?wallet-connect=${encodeURIComponent(d)}`:"";(0,He.useEffect)(()=>{q();let U=setInterval(()=>{q()},m);return()=>clearInterval(U)},[g]),Qe(()=>{se()},[n]),Qe(()=>{h.current=g},[g]);let O=()=>{Un(e)},R=()=>{E.current=!1};function q(){return y(this,null,function*(){var Z,V;if(g==null||!ce("walletconnect")||!Boolean((V=(Z=h.current)==null?void 0:Z.walletConnector)==null?void 0:V.connected))return;let K={method:"heartbeat",params:{}};try{yield h.current.sendCustomMessage(K)}catch(J){console.error("Connection lost",J),O()}})}function X(){return y(this,null,function*(){var U;try{let K=h.current;if(ie()||K==null||!ce("walletconnect"))return;if(!E.current){try{yield(U=h.current)==null?void 0:U.logout()}catch(v){console.warn("Unable to logout")}return}let V=yield K.getAddress();if(!V){console.warn("Login cancelled.");return}let J=yield K.getSignature(),B=Boolean(J),A={address:V,loginMethod:"walletconnect"},x={logoutRoute:e,loginType:"walletConnect",callbackRoute:o!=null?o:window.location.href};B?(a(uo(x)),a(pn(J))):a(uo(x)),a(te(A)),K.walletConnector.on("heartbeat",()=>{clearInterval(F),F=setInterval(()=>{console.log("Maiar Wallet Connection Lost"),O(),clearInterval(F)},15e4)}),Le(o,r)}catch(K){_("Invalid address"),console.error(K)}})}function pe(U=!0){return y(this,null,function*(){var J,B;let K=U&&!d;if(!f||((B=(J=h==null?void 0:h.current)==null?void 0:J.isInitialized)==null?void 0:B.call(J))&&!K)return;let Z={onClientLogin:X,onClientLogout:O},V=new Nc.WalletConnectProvider(f,Z);yield V.init(),E.current=!0,Re(V),h.current=V,U&&se()})}function se(){return y(this,null,function*(){var V;if(!f)return;let U=yield(V=h.current)==null?void 0:V.login();if(!Boolean(U))return;if(!n){u(U);return}let Z=`${U}&token=${n}`;u(Z),a(fe({loginToken:n}))})}let me=Boolean(i),$=ie();return[pe,{error:i,loginFailed:me,isLoading:N&&!me,isLoggedIn:$&&!me},{uriDeepLink:D,walletConnectUri:d,cancelLogin:R}]};t();var no=require("react"),Mc=require("@elrondnetwork/erdjs-wallet-connect-provider");re();G();H();fo();var Fc=({callbackRoute:e,logoutRoute:o,token:n,onLoginRedirect:r})=>{let a=j(),[m,i]=(0,no.useState)(""),[_,d]=(0,no.useState)(""),[u,g]=(0,no.useState)([]),f=ae(),T=w(Ii),h=w(Ai),E=w(Uo),F=w(vn),C=(0,no.useRef)(f),N=(0,no.useRef)(!0),D=Boolean(_),O=["erd_cancelAction"],R=!D,q=D?`${F}?wallet-connect=${encodeURIComponent(_)}`:"";Qe(()=>{Z()},[n]),Qe(()=>{C.current=f},[f]);let X=()=>{Un(o)},pe=B=>{console.log("wc2 session event: ",B)},se=()=>{N.current=!1};function me(){return y(this,null,function*(){var B;try{let A=C.current;if(ie()||A==null||!ce("walletconnectv2"))return;if(!N.current){try{yield(B=C.current)==null?void 0:B.logout()}catch(ur){console.warn("Unable to logout")}return}let v=yield A.getAddress();if(!v){console.warn("Login cancelled.");return}let L=yield A.getSignature(),oe=Boolean(L),Xo={address:v,loginMethod:"walletconnectv2"},Fe={logoutRoute:o,loginType:"walletconnectv2",callbackRoute:e!=null?e:window.location.href};oe?(a(uo(Fe)),a(pn(L))):a(uo(Fe)),a(te(Xo)),Le(e,r)}catch(A){i("Invalid address"),console.error(A)}})}function $(B=!0){return y(this,null,function*(){var L,oe;let A=B&&!_;if(!h||!T||((oe=(L=C==null?void 0:C.current)==null?void 0:L.isInitialized)==null?void 0:oe.call(L))&&!A){i("Invalid WalletConnect setup");return}let x={onClientLogin:me,onClientLogout:X,onClientEvent:pe},v=new Mc.WalletConnectV2Provider(x,E,T,h);yield v.init(),N.current=!0,Re(v),g(v.pairings),C.current=v,B&&Z()})}function U(B){return y(this,null,function*(){var A,x,v;if(!T||!h){i("Invalid WalletConnect setup");return}if(!B||!B.topic){i("Expired connection");return}try{let{approval:L}=yield(A=C.current)==null?void 0:A.connect({topic:B.topic,methods:O});n&&a(fe({loginToken:n}));try{yield(x=C.current)==null?void 0:x.login({approval:L,token:n})}catch(oe){console.warn("User rejected existing connection proposal",oe),yield $()}}catch(L){console.error("Unable to connect to existing session",L),i("Unable to connect to existing session")}finally{g((v=C.current)==null?void 0:v.pairings)}})}function K(B){return y(this,null,function*(){var A,x;try{B&&(yield(A=C.current)==null?void 0:A.logout({topic:B}))}catch(v){console.error("Unable to remove existing pairing",v),i("Unable to remove existing pairing")}finally{g((x=C.current)==null?void 0:x.pairings)}})}function Z(){return y(this,null,function*(){var B,A;if(!T||!h){i("Invalid WalletConnect setup");return}try{let{uri:x,approval:v}=yield(B=C.current)==null?void 0:B.connect({methods:O});if(!Boolean(x))return;d(x),n&&a(fe({loginToken:n}));try{yield(A=C.current)==null?void 0:A.login({approval:v,token:n})}catch(oe){console.warn("User rejected connection proposal",oe),yield $()}}catch(x){console.error("Unable to connect",x)}})}let V=Boolean(m),J=ie();return[$,{error:m,loginFailed:V,isLoading:R&&!V,isLoggedIn:J&&!V},{uriDeepLink:q,walletConnectUri:_,cancelLogin:se,connectExisting:U,removeExistingPairing:K,wcPairings:u}]};t();t();var Ko=require("react");function Rm(e){let[o,n]=(0,Ko.useState)(),[r,a]=(0,Ko.useState)();function m(){return y(this,null,function*(){try{let i=yield Wn(e);n(i)}catch(i){a(i)}})}return(0,Ko.useEffect)(()=>{m()},[]),{data:o,error:r,getUrl:m}}t();var Uc=require("react");t();var to=require("react"),Oc=require("socket.io-client");G();t();var Me={current:null,status:"not_initialized"};var Mm=3e3,Fm=3,Om=500,Bm=1e3;function Bc(){let e=(0,to.useRef)(null),{address:o}=_r(),n=j(),{network:r}=Vo(),a=i=>{e.current&&clearTimeout(e.current),e.current=setTimeout(()=>{n(Pa(i))},Bm)},m=(0,to.useCallback)(lc(()=>y(this,null,function*(){Me.status="pending";let i=yield Wn(r.apiAddress);if(i==null){console.warn("Can not get websocket url");return}Me.current=(0,Oc.io)(i,{forceNew:!0,reconnectionAttempts:Fm,timeout:Mm,query:{address:o}}),Me.status="completed",Me.current.onAny(_=>{a(_)})}),{retries:2,delay:Om}),[o]);(0,to.useEffect)(()=>{o&&Me.status==="not_initialized"&&!Me.current&&m()},[o,Me.current]),(0,to.useEffect)(()=>()=>{var i;(i=Me.current)==null||i.close(),e.current&&clearTimeout(e.current)},[])}function Um(e){Bc();let o=w(Sn);(0,Uc.useEffect)(()=>{let n=o==null?void 0:o.message;n&&e(n)},[e,o])}t();t();t();var Wc=e=>e.dappModal,Gc=S(Wc,e=>e.visible),zn=S(Wc,e=>e.modal),iA=S(zn,e=>e.showHeader),cA=S(zn,e=>e.showFooter),pA=S(zn,e=>e.headerText),sA=S(zn,e=>e.footerText);var Wm=()=>w(Gc);0&&(module.exports={useCheckTransactionStatus,useDebounce,useExtensionLogin,useGetAccount,useGetAccountInfo,useGetAccountProvider,useGetActiveTransactionsStatus,useGetFailedTransactions,useGetIsLoggedIn,useGetLoginInfo,useGetNetworkConfig,useGetNotification,useGetPendingTransactions,useGetSignTransactionsError,useGetSignedTransactions,useGetSuccessfulTransactions,useGetTokenDetails,useGetTransactionDisplayInfo,useGetWebsocketEvent,useGetWebsocketUrl,useIsDappModalVisible,useLedgerLogin,useParseMultiEsdtTransferData,useParseSignedTransactions,useRegisterWebsocketListener,useSignMultipleTransactions,useSignTransactions,useSignTransactionsWithDevice,useSignTransactionsWithLedger,useTrackTransactionStatus,useUpdateEffect,useWalletConnectLogin,useWalletConnectV2Login,useWebWalletLogin});
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
