// @elrondnetwork/dapp-core/__commonjs/UI/SignTransactionsModals/SignWithDeviceModal/index.js
"use strict";var Mc=Object.create;var lo=Object.defineProperty,Rc=Object.defineProperties,Fc=Object.getOwnPropertyDescriptor,Bc=Object.getOwnPropertyDescriptors,Oc=Object.getOwnPropertyNames,Mo=Object.getOwnPropertySymbols,Uc=Object.getPrototypeOf,Cn=Object.prototype.hasOwnProperty,dr=Object.prototype.propertyIsEnumerable;var mr=(e,o,n)=>o in e?lo(e,o,{enumerable:!0,configurable:!0,writable:!0,value:n}):e[o]=n,b=(e,o)=>{for(var n in o||(o={}))Cn.call(o,n)&&mr(e,n,o[n]);if(Mo)for(var n of Mo(o))dr.call(o,n)&&mr(e,n,o[n]);return e},I=(e,o)=>Rc(e,Bc(o));var Ln=(e,o)=>{var n={};for(var r in e)Cn.call(e,r)&&o.indexOf(r)<0&&(n[r]=e[r]);if(e!=null&&Mo)for(var r of Mo(e))o.indexOf(r)<0&&dr.call(e,r)&&(n[r]=e[r]);return n};var w=(e,o)=>()=>(e&&(o=e(e=0)),o);var Ro=(e,o)=>()=>(o||e((o={exports:{}}).exports,o),o.exports),Ie=(e,o)=>{for(var n in o)lo(e,n,{get:o[n],enumerable:!0})},lr=(e,o,n,r)=>{if(o&&typeof o=="object"||typeof o=="function")for(let p of Oc(o))!Cn.call(e,p)&&p!==n&&lo(e,p,{get:()=>o[p],enumerable:!(r=Fc(o,p))||r.enumerable});return e};var x=(e,o,n)=>(n=e!=null?Mc(Uc(e)):{},lr(o||!e||!e.__esModule?lo(n,"default",{value:e,enumerable:!0}):n,e)),_o=e=>lr(lo({},"__esModule",{value:!0}),e);var A=(e,o,n)=>new Promise((r,p)=>{var s=_=>{try{g(n.next(_))}catch(u){p(u)}},m=_=>{try{g(n.throw(_))}catch(u){p(u)}},g=_=>_.done?r(_.value):Promise.resolve(_.value).then(s,m);g((n=n.apply(e,o)).next())});var ur=Ro(Fo=>{"use strict";t();Fo.byteLength=Gc;Fo.toByteArray=$c;Fo.fromByteArray=zc;var ie=[],J=[],Wc=typeof Uint8Array!="undefined"?Uint8Array:Array,Dn="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";for(Ae=0,_r=Dn.length;Ae<_r;++Ae)ie[Ae]=Dn[Ae],J[Dn.charCodeAt(Ae)]=Ae;var Ae,_r;J["-".charCodeAt(0)]=62;J["_".charCodeAt(0)]=63;function gr(e){var o=e.length;if(o%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var n=e.indexOf("=");n===-1&&(n=o);var r=n===o?0:4-n%4;return[n,r]}function Gc(e){var o=gr(e),n=o[0],r=o[1];return(n+r)*3/4-r}function Hc(e,o,n){return(o+n)*3/4-n}function $c(e){var o,n=gr(e),r=n[0],p=n[1],s=new Wc(Hc(e,r,p)),m=0,g=p>0?r-4:r,_;for(_=0;_<g;_+=4)o=J[e.charCodeAt(_)]<<18|J[e.charCodeAt(_+1)]<<12|J[e.charCodeAt(_+2)]<<6|J[e.charCodeAt(_+3)],s[m++]=o>>16&255,s[m++]=o>>8&255,s[m++]=o&255;return p===2&&(o=J[e.charCodeAt(_)]<<2|J[e.charCodeAt(_+1)]>>4,s[m++]=o&255),p===1&&(o=J[e.charCodeAt(_)]<<10|J[e.charCodeAt(_+1)]<<4|J[e.charCodeAt(_+2)]>>2,s[m++]=o>>8&255,s[m++]=o&255),s}function Vc(e){return ie[e>>18&63]+ie[e>>12&63]+ie[e>>6&63]+ie[e&63]}function jc(e,o,n){for(var r,p=[],s=o;s<n;s+=3)r=(e[s]<<16&16711680)+(e[s+1]<<8&65280)+(e[s+2]&255),p.push(Vc(r));return p.join("")}function zc(e){for(var o,n=e.length,r=n%3,p=[],s=16383,m=0,g=n-r;m<g;m+=s)p.push(jc(e,m,m+s>g?g:m+s));return r===1?(o=e[n-1],p.push(ie[o>>2]+ie[o<<4&63]+"==")):r===2&&(o=(e[n-2]<<8)+e[n-1],p.push(ie[o>>10]+ie[o>>4&63]+ie[o<<2&63]+"=")),p.join("")}});var fr=Ro(Nn=>{t();Nn.read=function(e,o,n,r,p){var s,m,g=p*8-r-1,_=(1<<g)-1,u=_>>1,h=-7,f=n?p-1:0,y=n?-1:1,T=e[o+f];for(f+=y,s=T&(1<<-h)-1,T>>=-h,h+=g;h>0;s=s*256+e[o+f],f+=y,h-=8);for(m=s&(1<<-h)-1,s>>=-h,h+=r;h>0;m=m*256+e[o+f],f+=y,h-=8);if(s===0)s=1-u;else{if(s===_)return m?NaN:(T?-1:1)*(1/0);m=m+Math.pow(2,r),s=s-u}return(T?-1:1)*m*Math.pow(2,s-r)};Nn.write=function(e,o,n,r,p,s){var m,g,_,u=s*8-p-1,h=(1<<u)-1,f=h>>1,y=p===23?Math.pow(2,-24)-Math.pow(2,-77):0,T=r?0:s-1,v=r?1:-1,M=o<0||o===0&&1/o<0?1:0;for(o=Math.abs(o),isNaN(o)||o===1/0?(g=isNaN(o)?1:0,m=h):(m=Math.floor(Math.log(o)/Math.LN2),o*(_=Math.pow(2,-m))<1&&(m--,_*=2),m+f>=1?o+=y/_:o+=y*Math.pow(2,1-f),o*_>=2&&(m++,_/=2),m+f>=h?(g=0,m=h):m+f>=1?(g=(o*_-1)*Math.pow(2,p),m=m+f):(g=o*Math.pow(2,f-1)*Math.pow(2,p),m=0));p>=8;e[n+T]=g&255,T+=v,g/=256,p-=8);for(m=m<<p|g,u+=p;u>0;e[n+T]=m&255,T+=v,m/=256,u-=8);e[n+T-v]|=M*128}});var Cr=Ro(Ge=>{"use strict";t();var Pn=ur(),We=fr(),hr=typeof Symbol=="function"&&typeof Symbol.for=="function"?Symbol.for("nodejs.util.inspect.custom"):null;Ge.Buffer=l;Ge.SlowBuffer=Jc;Ge.INSPECT_MAX_BYTES=50;var Bo=2147483647;Ge.kMaxLength=Bo;l.TYPED_ARRAY_SUPPORT=qc();!l.TYPED_ARRAY_SUPPORT&&typeof console!="undefined"&&typeof console.error=="function"&&console.error("This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.");function qc(){try{var e=new Uint8Array(1),o={foo:function(){return 42}};return Object.setPrototypeOf(o,Uint8Array.prototype),Object.setPrototypeOf(e,o),e.foo()===42}catch(n){return!1}}Object.defineProperty(l.prototype,"parent",{enumerable:!0,get:function(){if(!!l.isBuffer(this))return this.buffer}});Object.defineProperty(l.prototype,"offset",{enumerable:!0,get:function(){if(!!l.isBuffer(this))return this.byteOffset}});function fe(e){if(e>Bo)throw new RangeError('The value "'+e+'" is invalid for option "size"');var o=new Uint8Array(e);return Object.setPrototypeOf(o,l.prototype),o}function l(e,o,n){if(typeof e=="number"){if(typeof o=="string")throw new TypeError('The "string" argument must be of type string. Received type number');return Bn(e)}return yr(e,o,n)}l.poolSize=8192;function yr(e,o,n){if(typeof e=="string")return Xc(e,o);if(ArrayBuffer.isView(e))return Yc(e);if(e==null)throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e);if(ce(e,ArrayBuffer)||e&&ce(e.buffer,ArrayBuffer)||typeof SharedArrayBuffer!="undefined"&&(ce(e,SharedArrayBuffer)||e&&ce(e.buffer,SharedArrayBuffer)))return Rn(e,o,n);if(typeof e=="number")throw new TypeError('The "value" argument must not be of type number. Received type number');var r=e.valueOf&&e.valueOf();if(r!=null&&r!==e)return l.from(r,o,n);var p=Zc(e);if(p)return p;if(typeof Symbol!="undefined"&&Symbol.toPrimitive!=null&&typeof e[Symbol.toPrimitive]=="function")return l.from(e[Symbol.toPrimitive]("string"),o,n);throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e)}l.from=function(e,o,n){return yr(e,o,n)};Object.setPrototypeOf(l.prototype,Uint8Array.prototype);Object.setPrototypeOf(l,Uint8Array);function br(e){if(typeof e!="number")throw new TypeError('"size" argument must be of type number');if(e<0)throw new RangeError('The value "'+e+'" is invalid for option "size"')}function Kc(e,o,n){return br(e),e<=0?fe(e):o!==void 0?typeof n=="string"?fe(e).fill(o,n):fe(e).fill(o):fe(e)}l.alloc=function(e,o,n){return Kc(e,o,n)};function Bn(e){return br(e),fe(e<0?0:On(e)|0)}l.allocUnsafe=function(e){return Bn(e)};l.allocUnsafeSlow=function(e){return Bn(e)};function Xc(e,o){if((typeof o!="string"||o==="")&&(o="utf8"),!l.isEncoding(o))throw new TypeError("Unknown encoding: "+o);var n=Sr(e,o)|0,r=fe(n),p=r.write(e,o);return p!==n&&(r=r.slice(0,p)),r}function Mn(e){for(var o=e.length<0?0:On(e.length)|0,n=fe(o),r=0;r<o;r+=1)n[r]=e[r]&255;return n}function Yc(e){if(ce(e,Uint8Array)){var o=new Uint8Array(e);return Rn(o.buffer,o.byteOffset,o.byteLength)}return Mn(e)}function Rn(e,o,n){if(o<0||e.byteLength<o)throw new RangeError('"offset" is outside of buffer bounds');if(e.byteLength<o+(n||0))throw new RangeError('"length" is outside of buffer bounds');var r;return o===void 0&&n===void 0?r=new Uint8Array(e):n===void 0?r=new Uint8Array(e,o):r=new Uint8Array(e,o,n),Object.setPrototypeOf(r,l.prototype),r}function Zc(e){if(l.isBuffer(e)){var o=On(e.length)|0,n=fe(o);return n.length===0||e.copy(n,0,0,o),n}if(e.length!==void 0)return typeof e.length!="number"||Un(e.length)?fe(0):Mn(e);if(e.type==="Buffer"&&Array.isArray(e.data))return Mn(e.data)}function On(e){if(e>=Bo)throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x"+Bo.toString(16)+" bytes");return e|0}function Jc(e){return+e!=e&&(e=0),l.alloc(+e)}l.isBuffer=function(o){return o!=null&&o._isBuffer===!0&&o!==l.prototype};l.compare=function(o,n){if(ce(o,Uint8Array)&&(o=l.from(o,o.offset,o.byteLength)),ce(n,Uint8Array)&&(n=l.from(n,n.offset,n.byteLength)),!l.isBuffer(o)||!l.isBuffer(n))throw new TypeError('The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array');if(o===n)return 0;for(var r=o.length,p=n.length,s=0,m=Math.min(r,p);s<m;++s)if(o[s]!==n[s]){r=o[s],p=n[s];break}return r<p?-1:p<r?1:0};l.isEncoding=function(o){switch(String(o).toLowerCase()){case"hex":case"utf8":case"utf-8":case"ascii":case"latin1":case"binary":case"base64":case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return!0;default:return!1}};l.concat=function(o,n){if(!Array.isArray(o))throw new TypeError('"list" argument must be an Array of Buffers');if(o.length===0)return l.alloc(0);var r;if(n===void 0)for(n=0,r=0;r<o.length;++r)n+=o[r].length;var p=l.allocUnsafe(n),s=0;for(r=0;r<o.length;++r){var m=o[r];if(ce(m,Uint8Array))s+m.length>p.length?l.from(m).copy(p,s):Uint8Array.prototype.set.call(p,m,s);else if(l.isBuffer(m))m.copy(p,s);else throw new TypeError('"list" argument must be an Array of Buffers');s+=m.length}return p};function Sr(e,o){if(l.isBuffer(e))return e.length;if(ArrayBuffer.isView(e)||ce(e,ArrayBuffer))return e.byteLength;if(typeof e!="string")throw new TypeError('The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type '+typeof e);var n=e.length,r=arguments.length>2&&arguments[2]===!0;if(!r&&n===0)return 0;for(var p=!1;;)switch(o){case"ascii":case"latin1":case"binary":return n;case"utf8":case"utf-8":return Fn(e).length;case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return n*2;case"hex":return n>>>1;case"base64":return Er(e).length;default:if(p)return r?-1:Fn(e).length;o=(""+o).toLowerCase(),p=!0}}l.byteLength=Sr;function Qc(e,o,n){var r=!1;if((o===void 0||o<0)&&(o=0),o>this.length||((n===void 0||n>this.length)&&(n=this.length),n<=0)||(n>>>=0,o>>>=0,n<=o))return"";for(e||(e="utf8");;)switch(e){case"hex":return sp(this,o,n);case"utf8":case"utf-8":return vr(this,o,n);case"ascii":return cp(this,o,n);case"latin1":case"binary":return pp(this,o,n);case"base64":return ap(this,o,n);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return mp(this,o,n);default:if(r)throw new TypeError("Unknown encoding: "+e);e=(e+"").toLowerCase(),r=!0}}l.prototype._isBuffer=!0;function Ee(e,o,n){var r=e[o];e[o]=e[n],e[n]=r}l.prototype.swap16=function(){var o=this.length;if(o%2!==0)throw new RangeError("Buffer size must be a multiple of 16-bits");for(var n=0;n<o;n+=2)Ee(this,n,n+1);return this};l.prototype.swap32=function(){var o=this.length;if(o%4!==0)throw new RangeError("Buffer size must be a multiple of 32-bits");for(var n=0;n<o;n+=4)Ee(this,n,n+3),Ee(this,n+1,n+2);return this};l.prototype.swap64=function(){var o=this.length;if(o%8!==0)throw new RangeError("Buffer size must be a multiple of 64-bits");for(var n=0;n<o;n+=8)Ee(this,n,n+7),Ee(this,n+1,n+6),Ee(this,n+2,n+5),Ee(this,n+3,n+4);return this};l.prototype.toString=function(){var o=this.length;return o===0?"":arguments.length===0?vr(this,0,o):Qc.apply(this,arguments)};l.prototype.toLocaleString=l.prototype.toString;l.prototype.equals=function(o){if(!l.isBuffer(o))throw new TypeError("Argument must be a Buffer");return this===o?!0:l.compare(this,o)===0};l.prototype.inspect=function(){var o="",n=Ge.INSPECT_MAX_BYTES;return o=this.toString("hex",0,n).replace(/(.{2})/g,"$1 ").trim(),this.length>n&&(o+=" ... "),"<Buffer "+o+">"};hr&&(l.prototype[hr]=l.prototype.inspect);l.prototype.compare=function(o,n,r,p,s){if(ce(o,Uint8Array)&&(o=l.from(o,o.offset,o.byteLength)),!l.isBuffer(o))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+typeof o);if(n===void 0&&(n=0),r===void 0&&(r=o?o.length:0),p===void 0&&(p=0),s===void 0&&(s=this.length),n<0||r>o.length||p<0||s>this.length)throw new RangeError("out of range index");if(p>=s&&n>=r)return 0;if(p>=s)return-1;if(n>=r)return 1;if(n>>>=0,r>>>=0,p>>>=0,s>>>=0,this===o)return 0;for(var m=s-p,g=r-n,_=Math.min(m,g),u=this.slice(p,s),h=o.slice(n,r),f=0;f<_;++f)if(u[f]!==h[f]){m=u[f],g=h[f];break}return m<g?-1:g<m?1:0};function wr(e,o,n,r,p){if(e.length===0)return-1;if(typeof n=="string"?(r=n,n=0):n>2147483647?n=2147483647:n<-2147483648&&(n=-2147483648),n=+n,Un(n)&&(n=p?0:e.length-1),n<0&&(n=e.length+n),n>=e.length){if(p)return-1;n=e.length-1}else if(n<0)if(p)n=0;else return-1;if(typeof o=="string"&&(o=l.from(o,r)),l.isBuffer(o))return o.length===0?-1:Tr(e,o,n,r,p);if(typeof o=="number")return o=o&255,typeof Uint8Array.prototype.indexOf=="function"?p?Uint8Array.prototype.indexOf.call(e,o,n):Uint8Array.prototype.lastIndexOf.call(e,o,n):Tr(e,[o],n,r,p);throw new TypeError("val must be string, number or Buffer")}function Tr(e,o,n,r,p){var s=1,m=e.length,g=o.length;if(r!==void 0&&(r=String(r).toLowerCase(),r==="ucs2"||r==="ucs-2"||r==="utf16le"||r==="utf-16le")){if(e.length<2||o.length<2)return-1;s=2,m/=2,g/=2,n/=2}function _(T,v){return s===1?T[v]:T.readUInt16BE(v*s)}var u;if(p){var h=-1;for(u=n;u<m;u++)if(_(e,u)===_(o,h===-1?0:u-h)){if(h===-1&&(h=u),u-h+1===g)return h*s}else h!==-1&&(u-=u-h),h=-1}else for(n+g>m&&(n=m-g),u=n;u>=0;u--){for(var f=!0,y=0;y<g;y++)if(_(e,u+y)!==_(o,y)){f=!1;break}if(f)return u}return-1}l.prototype.includes=function(o,n,r){return this.indexOf(o,n,r)!==-1};l.prototype.indexOf=function(o,n,r){return wr(this,o,n,r,!0)};l.prototype.lastIndexOf=function(o,n,r){return wr(this,o,n,r,!1)};function ep(e,o,n,r){n=Number(n)||0;var p=e.length-n;r?(r=Number(r),r>p&&(r=p)):r=p;var s=o.length;r>s/2&&(r=s/2);for(var m=0;m<r;++m){var g=parseInt(o.substr(m*2,2),16);if(Un(g))return m;e[n+m]=g}return m}function op(e,o,n,r){return Oo(Fn(o,e.length-n),e,n,r)}function np(e,o,n,r){return Oo(_p(o),e,n,r)}function tp(e,o,n,r){return Oo(Er(o),e,n,r)}function rp(e,o,n,r){return Oo(gp(o,e.length-n),e,n,r)}l.prototype.write=function(o,n,r,p){if(n===void 0)p="utf8",r=this.length,n=0;else if(r===void 0&&typeof n=="string")p=n,r=this.length,n=0;else if(isFinite(n))n=n>>>0,isFinite(r)?(r=r>>>0,p===void 0&&(p="utf8")):(p=r,r=void 0);else throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");var s=this.length-n;if((r===void 0||r>s)&&(r=s),o.length>0&&(r<0||n<0)||n>this.length)throw new RangeError("Attempt to write outside buffer bounds");p||(p="utf8");for(var m=!1;;)switch(p){case"hex":return ep(this,o,n,r);case"utf8":case"utf-8":return op(this,o,n,r);case"ascii":case"latin1":case"binary":return np(this,o,n,r);case"base64":return tp(this,o,n,r);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return rp(this,o,n,r);default:if(m)throw new TypeError("Unknown encoding: "+p);p=(""+p).toLowerCase(),m=!0}};l.prototype.toJSON=function(){return{type:"Buffer",data:Array.prototype.slice.call(this._arr||this,0)}};function ap(e,o,n){return o===0&&n===e.length?Pn.fromByteArray(e):Pn.fromByteArray(e.slice(o,n))}function vr(e,o,n){n=Math.min(e.length,n);for(var r=[],p=o;p<n;){var s=e[p],m=null,g=s>239?4:s>223?3:s>191?2:1;if(p+g<=n){var _,u,h,f;switch(g){case 1:s<128&&(m=s);break;case 2:_=e[p+1],(_&192)===128&&(f=(s&31)<<6|_&63,f>127&&(m=f));break;case 3:_=e[p+1],u=e[p+2],(_&192)===128&&(u&192)===128&&(f=(s&15)<<12|(_&63)<<6|u&63,f>2047&&(f<55296||f>57343)&&(m=f));break;case 4:_=e[p+1],u=e[p+2],h=e[p+3],(_&192)===128&&(u&192)===128&&(h&192)===128&&(f=(s&15)<<18|(_&63)<<12|(u&63)<<6|h&63,f>65535&&f<1114112&&(m=f))}}m===null?(m=65533,g=1):m>65535&&(m-=65536,r.push(m>>>10&1023|55296),m=56320|m&1023),r.push(m),p+=g}return ip(r)}var xr=4096;function ip(e){var o=e.length;if(o<=xr)return String.fromCharCode.apply(String,e);for(var n="",r=0;r<o;)n+=String.fromCharCode.apply(String,e.slice(r,r+=xr));return n}function cp(e,o,n){var r="";n=Math.min(e.length,n);for(var p=o;p<n;++p)r+=String.fromCharCode(e[p]&127);return r}function pp(e,o,n){var r="";n=Math.min(e.length,n);for(var p=o;p<n;++p)r+=String.fromCharCode(e[p]);return r}function sp(e,o,n){var r=e.length;(!o||o<0)&&(o=0),(!n||n<0||n>r)&&(n=r);for(var p="",s=o;s<n;++s)p+=up[e[s]];return p}function mp(e,o,n){for(var r=e.slice(o,n),p="",s=0;s<r.length-1;s+=2)p+=String.fromCharCode(r[s]+r[s+1]*256);return p}l.prototype.slice=function(o,n){var r=this.length;o=~~o,n=n===void 0?r:~~n,o<0?(o+=r,o<0&&(o=0)):o>r&&(o=r),n<0?(n+=r,n<0&&(n=0)):n>r&&(n=r),n<o&&(n=o);var p=this.subarray(o,n);return Object.setPrototypeOf(p,l.prototype),p};function G(e,o,n){if(e%1!==0||e<0)throw new RangeError("offset is not uint");if(e+o>n)throw new RangeError("Trying to access beyond buffer length")}l.prototype.readUintLE=l.prototype.readUIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||G(o,n,this.length);for(var p=this[o],s=1,m=0;++m<n&&(s*=256);)p+=this[o+m]*s;return p};l.prototype.readUintBE=l.prototype.readUIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||G(o,n,this.length);for(var p=this[o+--n],s=1;n>0&&(s*=256);)p+=this[o+--n]*s;return p};l.prototype.readUint8=l.prototype.readUInt8=function(o,n){return o=o>>>0,n||G(o,1,this.length),this[o]};l.prototype.readUint16LE=l.prototype.readUInt16LE=function(o,n){return o=o>>>0,n||G(o,2,this.length),this[o]|this[o+1]<<8};l.prototype.readUint16BE=l.prototype.readUInt16BE=function(o,n){return o=o>>>0,n||G(o,2,this.length),this[o]<<8|this[o+1]};l.prototype.readUint32LE=l.prototype.readUInt32LE=function(o,n){return o=o>>>0,n||G(o,4,this.length),(this[o]|this[o+1]<<8|this[o+2]<<16)+this[o+3]*16777216};l.prototype.readUint32BE=l.prototype.readUInt32BE=function(o,n){return o=o>>>0,n||G(o,4,this.length),this[o]*16777216+(this[o+1]<<16|this[o+2]<<8|this[o+3])};l.prototype.readIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||G(o,n,this.length);for(var p=this[o],s=1,m=0;++m<n&&(s*=256);)p+=this[o+m]*s;return s*=128,p>=s&&(p-=Math.pow(2,8*n)),p};l.prototype.readIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||G(o,n,this.length);for(var p=n,s=1,m=this[o+--p];p>0&&(s*=256);)m+=this[o+--p]*s;return s*=128,m>=s&&(m-=Math.pow(2,8*n)),m};l.prototype.readInt8=function(o,n){return o=o>>>0,n||G(o,1,this.length),this[o]&128?(255-this[o]+1)*-1:this[o]};l.prototype.readInt16LE=function(o,n){o=o>>>0,n||G(o,2,this.length);var r=this[o]|this[o+1]<<8;return r&32768?r|4294901760:r};l.prototype.readInt16BE=function(o,n){o=o>>>0,n||G(o,2,this.length);var r=this[o+1]|this[o]<<8;return r&32768?r|4294901760:r};l.prototype.readInt32LE=function(o,n){return o=o>>>0,n||G(o,4,this.length),this[o]|this[o+1]<<8|this[o+2]<<16|this[o+3]<<24};l.prototype.readInt32BE=function(o,n){return o=o>>>0,n||G(o,4,this.length),this[o]<<24|this[o+1]<<16|this[o+2]<<8|this[o+3]};l.prototype.readFloatLE=function(o,n){return o=o>>>0,n||G(o,4,this.length),We.read(this,o,!0,23,4)};l.prototype.readFloatBE=function(o,n){return o=o>>>0,n||G(o,4,this.length),We.read(this,o,!1,23,4)};l.prototype.readDoubleLE=function(o,n){return o=o>>>0,n||G(o,8,this.length),We.read(this,o,!0,52,8)};l.prototype.readDoubleBE=function(o,n){return o=o>>>0,n||G(o,8,this.length),We.read(this,o,!1,52,8)};function K(e,o,n,r,p,s){if(!l.isBuffer(e))throw new TypeError('"buffer" argument must be a Buffer instance');if(o>p||o<s)throw new RangeError('"value" argument is out of bounds');if(n+r>e.length)throw new RangeError("Index out of range")}l.prototype.writeUintLE=l.prototype.writeUIntLE=function(o,n,r,p){if(o=+o,n=n>>>0,r=r>>>0,!p){var s=Math.pow(2,8*r)-1;K(this,o,n,r,s,0)}var m=1,g=0;for(this[n]=o&255;++g<r&&(m*=256);)this[n+g]=o/m&255;return n+r};l.prototype.writeUintBE=l.prototype.writeUIntBE=function(o,n,r,p){if(o=+o,n=n>>>0,r=r>>>0,!p){var s=Math.pow(2,8*r)-1;K(this,o,n,r,s,0)}var m=r-1,g=1;for(this[n+m]=o&255;--m>=0&&(g*=256);)this[n+m]=o/g&255;return n+r};l.prototype.writeUint8=l.prototype.writeUInt8=function(o,n,r){return o=+o,n=n>>>0,r||K(this,o,n,1,255,0),this[n]=o&255,n+1};l.prototype.writeUint16LE=l.prototype.writeUInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||K(this,o,n,2,65535,0),this[n]=o&255,this[n+1]=o>>>8,n+2};l.prototype.writeUint16BE=l.prototype.writeUInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||K(this,o,n,2,65535,0),this[n]=o>>>8,this[n+1]=o&255,n+2};l.prototype.writeUint32LE=l.prototype.writeUInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||K(this,o,n,4,4294967295,0),this[n+3]=o>>>24,this[n+2]=o>>>16,this[n+1]=o>>>8,this[n]=o&255,n+4};l.prototype.writeUint32BE=l.prototype.writeUInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||K(this,o,n,4,4294967295,0),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};l.prototype.writeIntLE=function(o,n,r,p){if(o=+o,n=n>>>0,!p){var s=Math.pow(2,8*r-1);K(this,o,n,r,s-1,-s)}var m=0,g=1,_=0;for(this[n]=o&255;++m<r&&(g*=256);)o<0&&_===0&&this[n+m-1]!==0&&(_=1),this[n+m]=(o/g>>0)-_&255;return n+r};l.prototype.writeIntBE=function(o,n,r,p){if(o=+o,n=n>>>0,!p){var s=Math.pow(2,8*r-1);K(this,o,n,r,s-1,-s)}var m=r-1,g=1,_=0;for(this[n+m]=o&255;--m>=0&&(g*=256);)o<0&&_===0&&this[n+m+1]!==0&&(_=1),this[n+m]=(o/g>>0)-_&255;return n+r};l.prototype.writeInt8=function(o,n,r){return o=+o,n=n>>>0,r||K(this,o,n,1,127,-128),o<0&&(o=255+o+1),this[n]=o&255,n+1};l.prototype.writeInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||K(this,o,n,2,32767,-32768),this[n]=o&255,this[n+1]=o>>>8,n+2};l.prototype.writeInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||K(this,o,n,2,32767,-32768),this[n]=o>>>8,this[n+1]=o&255,n+2};l.prototype.writeInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||K(this,o,n,4,2147483647,-2147483648),this[n]=o&255,this[n+1]=o>>>8,this[n+2]=o>>>16,this[n+3]=o>>>24,n+4};l.prototype.writeInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||K(this,o,n,4,2147483647,-2147483648),o<0&&(o=4294967295+o+1),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};function kr(e,o,n,r,p,s){if(n+r>e.length)throw new RangeError("Index out of range");if(n<0)throw new RangeError("Index out of range")}function Ir(e,o,n,r,p){return o=+o,n=n>>>0,p||kr(e,o,n,4,34028234663852886e22,-34028234663852886e22),We.write(e,o,n,r,23,4),n+4}l.prototype.writeFloatLE=function(o,n,r){return Ir(this,o,n,!0,r)};l.prototype.writeFloatBE=function(o,n,r){return Ir(this,o,n,!1,r)};function Ar(e,o,n,r,p){return o=+o,n=n>>>0,p||kr(e,o,n,8,17976931348623157e292,-17976931348623157e292),We.write(e,o,n,r,52,8),n+8}l.prototype.writeDoubleLE=function(o,n,r){return Ar(this,o,n,!0,r)};l.prototype.writeDoubleBE=function(o,n,r){return Ar(this,o,n,!1,r)};l.prototype.copy=function(o,n,r,p){if(!l.isBuffer(o))throw new TypeError("argument should be a Buffer");if(r||(r=0),!p&&p!==0&&(p=this.length),n>=o.length&&(n=o.length),n||(n=0),p>0&&p<r&&(p=r),p===r||o.length===0||this.length===0)return 0;if(n<0)throw new RangeError("targetStart out of bounds");if(r<0||r>=this.length)throw new RangeError("Index out of range");if(p<0)throw new RangeError("sourceEnd out of bounds");p>this.length&&(p=this.length),o.length-n<p-r&&(p=o.length-n+r);var s=p-r;return this===o&&typeof Uint8Array.prototype.copyWithin=="function"?this.copyWithin(n,r,p):Uint8Array.prototype.set.call(o,this.subarray(r,p),n),s};l.prototype.fill=function(o,n,r,p){if(typeof o=="string"){if(typeof n=="string"?(p=n,n=0,r=this.length):typeof r=="string"&&(p=r,r=this.length),p!==void 0&&typeof p!="string")throw new TypeError("encoding must be a string");if(typeof p=="string"&&!l.isEncoding(p))throw new TypeError("Unknown encoding: "+p);if(o.length===1){var s=o.charCodeAt(0);(p==="utf8"&&s<128||p==="latin1")&&(o=s)}}else typeof o=="number"?o=o&255:typeof o=="boolean"&&(o=Number(o));if(n<0||this.length<n||this.length<r)throw new RangeError("Out of range index");if(r<=n)return this;n=n>>>0,r=r===void 0?this.length:r>>>0,o||(o=0);var m;if(typeof o=="number")for(m=n;m<r;++m)this[m]=o;else{var g=l.isBuffer(o)?o:l.from(o,p),_=g.length;if(_===0)throw new TypeError('The value "'+o+'" is invalid for argument "value"');for(m=0;m<r-n;++m)this[m+n]=g[m%_]}return this};var dp=/[^+/0-9A-Za-z-_]/g;function lp(e){if(e=e.split("=")[0],e=e.trim().replace(dp,""),e.length<2)return"";for(;e.length%4!==0;)e=e+"=";return e}function Fn(e,o){o=o||1/0;for(var n,r=e.length,p=null,s=[],m=0;m<r;++m){if(n=e.charCodeAt(m),n>55295&&n<57344){if(!p){if(n>56319){(o-=3)>-1&&s.push(239,191,189);continue}else if(m+1===r){(o-=3)>-1&&s.push(239,191,189);continue}p=n;continue}if(n<56320){(o-=3)>-1&&s.push(239,191,189),p=n;continue}n=(p-55296<<10|n-56320)+65536}else p&&(o-=3)>-1&&s.push(239,191,189);if(p=null,n<128){if((o-=1)<0)break;s.push(n)}else if(n<2048){if((o-=2)<0)break;s.push(n>>6|192,n&63|128)}else if(n<65536){if((o-=3)<0)break;s.push(n>>12|224,n>>6&63|128,n&63|128)}else if(n<1114112){if((o-=4)<0)break;s.push(n>>18|240,n>>12&63|128,n>>6&63|128,n&63|128)}else throw new Error("Invalid code point")}return s}function _p(e){for(var o=[],n=0;n<e.length;++n)o.push(e.charCodeAt(n)&255);return o}function gp(e,o){for(var n,r,p,s=[],m=0;m<e.length&&!((o-=2)<0);++m)n=e.charCodeAt(m),r=n>>8,p=n%256,s.push(p),s.push(r);return s}function Er(e){return Pn.toByteArray(lp(e))}function Oo(e,o,n,r){for(var p=0;p<r&&!(p+n>=o.length||p>=e.length);++p)o[p+n]=e[p];return p}function ce(e,o){return e instanceof o||e!=null&&e.constructor!=null&&e.constructor.name!=null&&e.constructor.name===o.name}function Un(e){return e!==e}var up=function(){for(var e="0123456789abcdef",o=new Array(256),n=0;n<16;++n)for(var r=n*16,p=0;p<16;++p)o[r+p]=e[n]+e[p];return o}()});var Mr=Ro((td,Pr)=>{t();var B=Pr.exports={},pe,se;function Wn(){throw new Error("setTimeout has not been defined")}function Gn(){throw new Error("clearTimeout has not been defined")}(function(){try{typeof setTimeout=="function"?pe=setTimeout:pe=Wn}catch(e){pe=Wn}try{typeof clearTimeout=="function"?se=clearTimeout:se=Gn}catch(e){se=Gn}})();function Lr(e){if(pe===setTimeout)return setTimeout(e,0);if((pe===Wn||!pe)&&setTimeout)return pe=setTimeout,setTimeout(e,0);try{return pe(e,0)}catch(o){try{return pe.call(null,e,0)}catch(n){return pe.call(this,e,0)}}}function fp(e){if(se===clearTimeout)return clearTimeout(e);if((se===Gn||!se)&&clearTimeout)return se=clearTimeout,clearTimeout(e);try{return se(e)}catch(o){try{return se.call(null,e)}catch(n){return se.call(this,e)}}}var he=[],He=!1,Ce,Uo=-1;function hp(){!He||!Ce||(He=!1,Ce.length?he=Ce.concat(he):Uo=-1,he.length&&Dr())}function Dr(){if(!He){var e=Lr(hp);He=!0;for(var o=he.length;o;){for(Ce=he,he=[];++Uo<o;)Ce&&Ce[Uo].run();Uo=-1,o=he.length}Ce=null,He=!1,fp(e)}}B.nextTick=function(e){var o=new Array(arguments.length-1);if(arguments.length>1)for(var n=1;n<arguments.length;n++)o[n-1]=arguments[n];he.push(new Nr(e,o)),he.length===1&&!He&&Lr(Dr)};function Nr(e,o){this.fun=e,this.array=o}Nr.prototype.run=function(){this.fun.apply(null,this.array)};B.title="browser";B.browser=!0;B.env={};B.argv=[];B.version="";B.versions={};function Te(){}B.on=Te;B.addListener=Te;B.once=Te;B.off=Te;B.removeListener=Te;B.removeAllListeners=Te;B.emit=Te;B.prependListener=Te;B.prependOnceListener=Te;B.listeners=function(e){return[]};B.binding=function(e){throw new Error("process.binding is not supported")};B.cwd=function(){return"/"};B.chdir=function(e){throw new Error("process.chdir is not supported")};B.umask=function(){return 0}});var a,i,rd,t=w(()=>{a=x(Cr()),i=x(Mr()),rd=function(e){function o(){var r=this||self;return delete e.prototype.__magic__,r}if(typeof globalThis=="object")return globalThis;if(this)return o();e.defineProperty(e.prototype,"__magic__",{configurable:!0,get:o});var n=__magic__;return n}(Object)});var Wo,Hn=w(()=>{"use strict";t();Wo=()=>Date.now()/1e3});var Rr=w(()=>{"use strict";t()});var Fr=w(()=>{"use strict";t()});var $n=w(()=>{"use strict";t();Hn();Rr();Fr()});var jn={};Ie(jn,{getItem:()=>yp,localStorageKeys:()=>ve,removeItem:()=>bp,setItem:()=>xp});var ve,Vn,xp,yp,bp,$e=w(()=>{"use strict";t();$n();ve={loginExpiresAt:"dapp-core-login-expires-at",logoutEvent:"dapp-core-logout-event"},Vn=typeof localStorage!="undefined",xp=({key:e,data:o,expires:n})=>{!Vn||localStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},yp=e=>{if(!Vn)return;let o=localStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:Wo()>=n.expires?(localStorage.removeItem(String(e)),null):n.data},bp=e=>{!Vn||localStorage.removeItem(String(e))}});var zn={};Ie(zn,{clear:()=>Wr,getItem:()=>Or,removeItem:()=>Ur,setItem:()=>Br,storage:()=>Sp});var Br,Or,Ur,Wr,Sp,Gr=w(()=>{"use strict";t();Br=({key:e,data:o,expires:n})=>{sessionStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},Or=e=>{let o=sessionStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:Date.now()>=n.expires?(sessionStorage.removeItem(String(e)),null):n.data},Ur=e=>sessionStorage.removeItem(String(e)),Wr=()=>sessionStorage.clear(),Sp={setItem:Br,getItem:Or,removeItem:Ur,clear:Wr}});var Le,Ve=w(()=>{"use strict";t();$e();Gr();Le={session:zn,local:jn}});function go(){return new Date().setHours(new Date().getHours()+24)}function uo(e){Le.local.setItem({key:ve.loginExpiresAt,data:e,expires:e})}var qn=w(()=>{"use strict";t();Ve();$e()});var R=w(()=>{"use strict";t()});var Kn,je=w(()=>{"use strict";t();Kn="Transaction canceled"});var Hr=w(()=>{"use strict";t()});var V=w(()=>{"use strict";t()});var $r=w(()=>{"use strict";t()});var Vr=w(()=>{"use strict";t()});var jr=w(()=>{"use strict";t()});var zr=w(()=>{"use strict";t()});var Q=w(()=>{"use strict";t();Hr();V();R();$r();Vr();jr();zr()});var Xn,qr,Pd,Kr,Md,Xr,Rd,Fd,Bd,Yr=w(()=>{"use strict";t();Q();Xn={devnet:{id:"devnet",chainId:"D",name:"Devnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://devnet-wallet.elrond.com",apiAddress:"https://devnet-api.elrond.com",explorerAddress:"http://devnet-explorer.elrond.com",apiTimeout:"4000"},testnet:{id:"testnet",chainId:"T",name:"Testnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://testnet-wallet.elrond.com",apiAddress:"https://testnet-api.elrond.com",explorerAddress:"http://testnet-explorer.elrond.com",apiTimeout:"4000"},mainnet:{id:"mainnet",chainId:"1",name:"Mainnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://wallet.elrond.com",apiAddress:"https://api.elrond.com",explorerAddress:"https://explorer.elrond.com",apiTimeout:"4000"}},{chainId:qr,egldLabel:Pd}=Xn["devnet"],{chainId:Kr,egldLabel:Md}=Xn["testnet"],{chainId:Xr,egldLabel:Rd}=Xn["mainnet"],Fd={["devnet"]:qr,["testnet"]:Kr,["mainnet"]:Xr},Bd={[qr]:"devnet",[Kr]:"testnet",[Xr]:"mainnet"}});var Yn,Zn=w(()=>{"use strict";t();Yn={36864:{code:"codeSuccess",message:"Success"},27013:{code:"ERR_USER_DENIED",message:"Rejected by user"},27904:{code:"ERR_UNKNOWN_INSTRUCTION",message:"Unknown instruction"},28160:{code:"ERR_WRONG_CLA",message:"Wrong CLA"},28161:{code:"ERR_INVALID_ARGUMENTS",message:"Invalid arguments"},28162:{code:"ERR_INVALID_MESSAGE",message:"Invalid message"},28163:{code:"ERR_INVALID_P1",message:"Invalid P1"},28164:{code:"ERR_MESSAGE_TOO_LONG",message:"Message too long"},28165:{code:"ERR_RECEIVER_TOO_LONG",message:"Receiver too long"},28166:{code:"ERR_AMOUNT_TOO_LONG",message:"Amount too long"},28167:{code:"ERR_CONTRACT_DATA_DISABLED",message:"Contract data disabled in app options"},28168:{code:"ERR_MESSAGE_INCOMPLETE",message:"Message incomplete"},28176:{code:"ERR_SIGNATURE_FAILED",message:"Signature failed"},28169:{code:"ERR_WRONG_TX_VERSION",message:"Wrong TX version"},28170:{code:"ERR_NONCE_TOO_LONG",message:"Nonce too long"},28171:{code:"ERR_INVALID_AMOUNT",message:"Invalid amount"},28172:{code:"ERR_INVALID_FEE",message:"Invalid fee"},28173:{code:"ERR_PRETTY_FAILED",message:"Pretty failed"},28174:{code:"ERR_DATA_TOO_LONG",message:"Data too long"},28175:{code:"ERR_WRONG_TX_OPTIONS",message:"Invalid transaction options"},28177:{code:"ERR_SIGN_TX_DEPRECATED",message:"Regular transaction signing is deprecated in this version. Use hash signing."}}});var Zr=w(()=>{"use strict";t()});var Jn,Qn=w(()=>{"use strict";t();Jn="transaction-status-toast-id"});var fo,Go,me,Ho,Jr,$o,Qr,ea,De,N=w(()=>{"use strict";t();je();Yr();Zn();Zr();Qn();fo=5e4,Go=1e9,me=18,Ho=4,Jr=1,$o="logout",Qr="login",ea="N/A",De="0"});var et,j,xe,X=w(()=>{"use strict";t();et=require("@reduxjs/toolkit");N();j=(0,et.createAction)($o),xe=(0,et.createAction)(Qr,e=>({payload:e}))});var na,oa,ta,ol,ra,wp,ho,aa,vp,ia,ca,ot=w(()=>{"use strict";t();na=require("@reduxjs/toolkit");qn();R();X();oa={loginMethod:"",walletConnectLogin:null,ledgerLogin:null,tokenLogin:null,walletLogin:null,extensionLogin:null,isLoginSessionInvalid:!1},ta=(0,na.createSlice)({name:"loginInfoSlice",initialState:oa,reducers:{setLoginMethod:(e,o)=>{e.loginMethod=o.payload},setTokenLogin:(e,o)=>{e.tokenLogin=o.payload},setTokenLoginSignature:(e,o)=>{(e==null?void 0:e.tokenLogin)!=null&&(e.tokenLogin.signature=o.payload)},setWalletLogin:(e,o)=>{e.walletLogin=o.payload},setWalletConnectLogin:(e,o)=>{e.walletConnectLogin=o.payload},setLedgerLogin:(e,o)=>{e.ledgerLogin=o.payload},invalidateLoginSession:e=>{e.isLoginSessionInvalid=!0}},extraReducers:e=>{e.addCase(j,()=>oa),e.addCase(xe,(o,n)=>{o.loginMethod=n.payload.loginMethod,uo(go())})}}),{setLoginMethod:ol,setWalletConnectLogin:ra,setLedgerLogin:wp,setTokenLogin:ho,setTokenLoginSignature:aa,setWalletLogin:vp,invalidateLoginSession:ia}=ta.actions,ca=ta.reducer});var kp,Vo,To=w(()=>{"use strict";t();kp=!1,Vo=(e,o=0)=>{kp||setTimeout(()=>{window.location.href=e},o)}});var pa=w(()=>{"use strict";t();To()});function nt(e){return e[Math.floor(Math.random()*e.length)]}var sa=w(()=>{"use strict";t()});function tt(e){let o=null;if((e==null?void 0:e.statusCode)in Yn){let n=e==null?void 0:e.statusCode,{message:r}=Yn[n];o=Ep.includes(n)?ma:r}return{errorMessage:o,defaultErrorMessage:ma}}var ma,Ip,Ap,Ep,rt=w(()=>{"use strict";t();Zn();ma="Check if Elrond app is open on Ledger",Ip=28161,Ap=28160,Ep=[Ip,Ap]});var ze=w(()=>{"use strict";t();pa();sa();rt()});var da,la,at,Cp,_a,ul,fl,ga,it=w(()=>{"use strict";t();da=require("@reduxjs/toolkit"),la=x(require("lodash.omit"));ze();at={id:"not-configured",chainId:"",name:"NOT CONFIGURED",egldLabel:"",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"",walletConnectBridgeAddress:"",walletConnectV2RelayAddress:"",walletConnectV2ProjectId:"",walletAddress:"",apiAddress:"",explorerAddress:"",apiTimeout:"4000"},Cp={network:at,chainID:"-1"},_a=(0,da.createSlice)({name:"appConfig",initialState:Cp,reducers:{initializeNetworkConfig:(e,o)=>{let n=nt(o.payload.walletConnectBridgeAddresses),r=nt(o.payload.walletConnectV2RelayAddresses),p=(0,la.default)(o.payload,"walletConnectBridgeAddresses","walletConnectV2RelayAddresses");e.network=I(b(b({},e.network),p),{walletConnectBridgeAddress:n,walletConnectV2RelayAddress:r})},setChainID:(e,o)=>{e.chainID=o.payload}}}),{initializeNetworkConfig:ul,setChainID:fl}=_a.actions,ga=_a.reducer});var pt,ua,fa,jo,ct,ha,st,vl,mt,kl,Lp,Dp,Il,Al,El,Np,Ta,dt=w(()=>{"use strict";t();pt=require("@elrondnetwork/erdjs"),ua=require("@reduxjs/toolkit"),fa=require("redux-persist");N();Ve();$e();X();jo={balance:"...",address:"",nonce:0,txCount:0,scrCount:0,claimableRewards:De},ct={address:"",websocketEvent:null,accounts:{"":jo},ledgerAccount:null,publicKey:"",walletConnectAccount:null,isAccountLoading:!0,accountLoadingError:null},ha=(0,ua.createSlice)({name:"accountInfoSlice",initialState:ct,reducers:{setAddress:(e,o)=>{let n=o.payload;e.address=n,e.publicKey=new pt.Address(n).hex()},setAccount:(e,o)=>{let n=e.address===o.payload.address;e.accounts={[e.address]:n?o.payload:jo},e.isAccountLoading=!1,e.accountLoadingError=null},setAccountNonce:(e,o)=>{let{address:n}=e;e.accounts[n].nonce=o.payload},setAccountShard:(e,o)=>{e.shard=o.payload},setLedgerAccount:(e,o)=>{e.ledgerAccount=o.payload},updateLedgerAccount:(e,o)=>{e.ledgerAccount!=null&&(e.ledgerAccount.index=o.payload.index,e.ledgerAccount.address=o.payload.address)},setWalletConnectAccount:(e,o)=>{e.walletConnectAccount=o.payload},setIsAccountLoading:(e,o)=>{e.isAccountLoading=o.payload,e.accountLoadingError=null},setAccountLoadingError:(e,o)=>{e.accountLoadingError=o.payload,e.isAccountLoading=!1},setWebsocketEvent:(e,o)=>{e.websocketEvent={timestamp:Date.now(),message:o.payload}}},extraReducers:e=>{e.addCase(j,()=>(Le.local.removeItem(ve.loginExpiresAt),ct)),e.addCase(xe,(o,n)=>{let{address:r}=n.payload;o.address=r,o.publicKey=new pt.Address(r).hex()}),e.addCase(fa.REHYDRATE,(o,n)=>{var u;if(!((u=n.payload)!=null&&u.account))return;let{account:r}=n.payload,{address:p,shard:s,accounts:m,publicKey:g}=r;o.address=p,o.shard=s;let _=m&&p in m;o.accounts=_?m:ct.accounts,o.publicKey=g})}}),{setAccount:st,setAddress:vl,setAccountNonce:mt,setAccountShard:kl,setLedgerAccount:Lp,updateLedgerAccount:Dp,setWalletConnectAccount:Il,setIsAccountLoading:Al,setAccountLoadingError:El,setWebsocketEvent:Np}=ha.actions,Ta=ha.reducer});function qe(e){return e!=null&&(Wp(e)||Vp(e))}function Ke(e){return e!=null&&(Gp(e)||jp(e))}function Xe(e){return e!=null&&(Hp(e)||zp(e))}function lt(e){return e!=null&&$p(e)}function Wp(e){return e!=null&&Pp.includes(e)}function Gp(e){return e!=null&&Mp.includes(e)}function Hp(e){return e!=null&&Rp.includes(e)}function $p(e){return e!=null&&Fp.includes(e)}function Vp(e){return e!=null&&Bp.includes(e)}function jp(e){return e!=null&&Op.includes(e)}function zp(e){return e!=null&&Up.includes(e)}var Pp,Mp,Rp,Fp,Bp,Op,Up,Ye=w(()=>{"use strict";t();R();Pp=["sent"],Mp=["success"],Rp=["fail","cancelled","timedOut"],Fp=["timedOut"],Bp=["pending"],Op=["success"],Up=["fail","invalid"]});var xa,ya,yo,qp,ba,Sa,wa,Kp,zo,Xp,Yp,Bl,va,_t,qo,Ko,Xo=w(()=>{"use strict";t();xa=require("@reduxjs/toolkit"),ya=require("redux-persist");R();Ye();X();yo={signedTransactions:{},transactionsToSign:null,signTransactionsError:null,signTransactionsCancelMessage:null,customTransactionInformationForSessionId:{}},qp={signWithoutSending:!1,sessionInformation:null,redirectAfterSign:!1},ba=(0,xa.createSlice)({name:"transactionsSlice",initialState:yo,reducers:{moveTransactionsToSignedState:(e,o)=>{var _,u;let{sessionId:n,transactions:r,errorMessage:p,status:s,redirectRoute:m}=o.payload,g=((_=e.customTransactionInformationForSessionId)==null?void 0:_[n])||qp;e.signedTransactions[n]={transactions:r,status:s,errorMessage:p,redirectRoute:m,customTransactionInformation:g},((u=e==null?void 0:e.transactionsToSign)==null?void 0:u.sessionId)===n&&(e.transactionsToSign=yo.transactionsToSign)},clearSignedTransaction:(e,o)=>{e.signedTransactions[o.payload]&&delete e.signedTransactions[o.payload]},clearTransactionToSign:e=>{e!=null&&e.transactionsToSign&&(e.transactionsToSign=null)},updateSignedTransaction:(e,o)=>{e.signedTransactions=b(b({},e.signedTransactions),o.payload)},updateSignedTransactions:(e,o)=>{let{sessionId:n,status:r,errorMessage:p,transactions:s}=o.payload;e.signedTransactions[n]!=null&&(e.signedTransactions[n].status=r,p!=null&&(e.signedTransactions[n].errorMessage=p),s!=null&&(e.signedTransactions[n].transactions=s))},updateSignedTransactionStatus:(e,o)=>{var g,_,u,h,f,y;let{sessionId:n,status:r,errorMessage:p,transactionHash:s}=o.payload,m=(_=(g=e.signedTransactions)==null?void 0:g[n])==null?void 0:_.transactions;if(m!=null){e.signedTransactions[n].transactions=m.map(M=>M.hash===s?I(b({},M),{status:r,errorMessage:p}):M);let T=(h=(u=e.signedTransactions[n])==null?void 0:u.transactions)==null?void 0:h.every(M=>Ke(M.status)),v=(y=(f=e.signedTransactions[n])==null?void 0:f.transactions)==null?void 0:y.every(M=>Xe(M.status));T&&(e.signedTransactions[n].status="success"),v&&(e.signedTransactions[n].status="fail")}},setTransactionsToSign:(e,o)=>{e.transactionsToSign=o.payload;let{sessionId:n,customTransactionInformation:r}=o.payload;e.customTransactionInformationForSessionId[n]=r,e.signTransactionsError=null},clearAllTransactionsToSign:e=>{e.transactionsToSign=yo.transactionsToSign,e.signTransactionsError=null},clearAllSignedTransactions:e=>{e.signedTransactions=yo.signedTransactions},setSignTransactionsError:(e,o)=>{e.signTransactionsError=o.payload},setSignTransactionsCancelMessage:(e,o)=>{e.signTransactionsCancelMessage=o.payload}},extraReducers:e=>{e.addCase(j,()=>yo),e.addCase(ya.REHYDRATE,(o,n)=>{var m;if(!((m=n.payload)!=null&&m.transactions))return;let{signedTransactions:r,customTransactionInformationForSessionId:p}=n.payload.transactions,s=Object.entries(r).reduce((g,[_,u])=>{let h=new Date(_),f=new Date;return f.setHours(f.getHours()+5),f-h>0||(g[_]=u),g},{});p!=null&&(o.customTransactionInformationForSessionId=p),r!=null&&(o.signedTransactions=s)})}}),{updateSignedTransactionStatus:Sa,updateSignedTransactions:wa,setTransactionsToSign:Kp,clearAllTransactionsToSign:zo,clearAllSignedTransactions:Xp,clearSignedTransaction:Yp,clearTransactionToSign:Bl,setSignTransactionsError:va,setSignTransactionsCancelMessage:_t,moveTransactionsToSignedState:qo}=ba.actions,Ko=ba.reducer});var ka,ut,ft,ht,Zp,gt,Ia,Gl,Jp,Tt,Yo,Zo=w(()=>{"use strict";t();ka=require("@reduxjs/toolkit");X();ut="Transaction failed",ft="Transaction successful",ht="Processing transaction",Zp="Transaction submitted",gt={},Ia=(0,ka.createSlice)({name:"transactionsInfo",initialState:gt,reducers:{setTransactionsDisplayInfo(e,o){let{sessionId:n,transactionsDisplayInfo:r}=o.payload;n!=null&&(e[n]={errorMessage:(r==null?void 0:r.errorMessage)||ut,successMessage:(r==null?void 0:r.successMessage)||ft,processingMessage:(r==null?void 0:r.processingMessage)||ht,submittedMessage:(r==null?void 0:r.submittedMessage)||Zp,transactionDuration:r==null?void 0:r.transactionDuration})},clearTransactionsInfoForSessionId(e,o){o.payload!=null&&delete e[o.payload]},clearTransactionsInfo:()=>gt},extraReducers:e=>{e.addCase(j,()=>gt)}}),{clearTransactionsInfo:Gl,setTransactionsDisplayInfo:Jp,clearTransactionsInfoForSessionId:Tt}=Ia.actions,Yo=Ia.reducer});var Ea,Aa,Ca,jl,Qp,zl,es,La,xt=w(()=>{"use strict";t();Ea=require("@reduxjs/toolkit");X();Aa={},Ca=(0,Ea.createSlice)({name:"modalsSlice",initialState:Aa,reducers:{setTxSubmittedModal:(e,o)=>{e.txSubmittedModal=o.payload},setNotificationModal:(e,o)=>{e.notificationModal=o.payload},clearTxSubmittedModal:e=>{e.txSubmittedModal=void 0},clearNotificationModal:e=>{e.notificationModal=void 0}},extraReducers:e=>{e.addCase(j,()=>Aa)}}),{setTxSubmittedModal:jl,setNotificationModal:Qp,clearTxSubmittedModal:zl,clearNotificationModal:es}=Ca.actions,La=Ca.reducer});var Na,Da,Pa,os,Ma,e_,o_,n_,t_,Jo,Qo=w(()=>{"use strict";t();Na=require("@reduxjs/toolkit");Qn();Q();Hn();X();Da={customToasts:[],transactionToasts:[],failTransactionToast:null},Pa=(0,Na.createSlice)({name:"toastsSlice",initialState:Da,reducers:{addCustomToast:(e,o)=>{e.customToasts.push(I(b({},o.payload),{type:"custom",toastId:o.payload.toastId||`custom-toast-${e.customToasts.length+1}`}))},removeCustomToast:(e,o)=>{e.customToasts=e.customToasts.filter(n=>n.toastId!==o.payload)},addTransactionToast:(e,o)=>{e.transactionToasts.push({type:"transaction",startTimestamp:Wo(),toastId:o.payload||`custom-toast-${e.transactionToasts.length+1}`})},removeTransactionToast:(e,o)=>{e.transactionToasts=e.transactionToasts.filter(n=>n.toastId!==o.payload)},addFailTransactionToast:(e,o)=>{e.failTransactionToast=I(b({},o.payload),{toastId:Jn})},removeFailTransactionToast:e=>{e.failTransactionToast=null}},extraReducers:e=>{e.addCase(j,()=>Da)}}),{addCustomToast:os,removeCustomToast:Ma,addTransactionToast:e_,removeTransactionToast:o_,addFailTransactionToast:n_,removeFailTransactionToast:t_}=Pa.actions,Jo=Pa.reducer});var P=w(()=>{"use strict";t();ot();it();dt();Xo();Zo();xt();Qo()});var yt=w(()=>{"use strict";t()});var Oa,Ba,Ua,bt,w_,v_,k_,I_,Wa,Ga,St=w(()=>{"use strict";t();Oa=require("@reduxjs/toolkit");X();Ba={visible:!1,modal:{showHeader:!0,showFooter:!1,headerText:"",footerText:""}},Ua=(0,Oa.createSlice)({name:"dappModalsSlice",initialState:Ba,reducers:{setDappModalVisibility:(e,o)=>{e.visible=o.payload},setDappModalHeaderVisibility:(e,o)=>{e.modal.showHeader=o.payload},setDappModalFooterVisibility:(e,o)=>{e.modal.showFooter=o.payload},setDappModalHeaderText:(e,o)=>{e.modal.headerText=o.payload},setDappModalFooterText:(e,o)=>{e.modal.footerText=o.payload},setDappModalConfig:(e,o)=>{e.modal=b({},o.payload)}},extraReducers:e=>{e.addCase(j,()=>Ba)}}),{setDappModalVisibility:bt,setDappModalHeaderVisibility:w_,setDappModalFooterVisibility:v_,setDappModalHeaderText:k_,setDappModalFooterText:I_,setDappModalConfig:Wa}=Ua.actions,Ga=Ua.reducer});var Ha,as,is,Ze,on=w(()=>{"use strict";t();Ha=require("@reduxjs/toolkit");yt();dt();St();ot();xt();it();Qo();Zo();Xo();as={["account"]:Ta,["networkConfig"]:ga,["loginInfo"]:ca,["modals"]:La,["dappModal"]:Ga,["toasts"]:Jo,["transactions"]:Ko,["transactionsInfo"]:Yo},is=(e={})=>(0,Ha.combineReducers)(b(b({},as),e)),Ze=is});var ja={};Ie(ja,{default:()=>_s});function wt(e,o=[]){return{key:e,version:1,storage:Va.default,blacklist:o}}var Pe,$a,Va,cs,ps,ss,ms,ds,ls,_s,za=w(()=>{"use strict";t();Pe=require("redux-persist"),$a=x(require("redux-persist/lib/storage")),Va=x(require("redux-persist/lib/storage/session"));on();P();Qo();Zo();Xo();yt();cs={2:e=>I(b({},e),{networkConfig:at})};ps=wt("dapp-core-transactionsInfo"),ss=wt("dapp-core-transactions",["transactionsToSign"]),ms=wt("dapp-core-toasts"),ds={key:"dapp-core-store",version:2,storage:$a.default,whitelist:["account","loginInfo","modals","networkConfig"],migrate:(0,Pe.createMigrate)(cs,{debug:!1})},ls={["toasts"]:(0,Pe.persistReducer)(ms,Jo),["transactions"]:(0,Pe.persistReducer)(ss,Ko),["transactionsInfo"]:(0,Pe.persistReducer)(ps,Yo)},_s=(0,Pe.persistReducer)(ds,Ze(ls))});var qa={};Ie(qa,{default:()=>gs});var gs,Ka=w(()=>{"use strict";t();on();gs=Ze()});var Xa={};Ie(Xa,{default:()=>fs});var oe,us,fs,Ya=w(()=>{"use strict";t();oe=require("redux-persist"),us=[oe.FLUSH,oe.REHYDRATE,oe.PAUSE,oe.PERSIST,oe.PURGE,oe.REGISTER],fs=us});var Qa={};Ie(Qa,{default:()=>Ja});function Ja(e){return(0,Za.persistStore)(e)}var Za,ei=w(()=>{"use strict";t();Za=require("redux-persist")});var Jm={};Ie(Jm,{SignWithDeviceModal:()=>Zm});module.exports=_o(Jm);t();var vn=x(require("react")),ir=x(require("classnames"));t();var Tp=`.dapp-core-component__main__link-style {
  color: #1392ff;
}

.dapp-core-component__main__link-second-style {
  color: #9dabbd;
}
.dapp-core-component__main__link-second-style:hover {
  color: #1392ff;
}
.dapp-core-component__main__link-second-style:hover svg {
  color: #1392ff;
}

.dapp-core-component__main__wrapper {
  min-height: 100vh;
}

.dapp-core-component__main__opacity-6 {
  opacity: 0.6;
}

.dapp-core-component__main__qr-code-svg-container {
  width: 15rem;
  height: 15rem;
}

.dapp-core-component__main__pair-list {
  max-height: 20rem;
  overflow-y: auto;
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__pair-image {
  width: 4rem;
  overflow: hidden;
  height: auto;
  min-width: 2rem;
}

.dapp-core-component__main__pair-remove {
  padding: 0 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

.dapp-core-component__main__dapp-icon {
  padding: 5px;
  border-radius: 50%;
  background-color: #fafafa;
  width: 4.6428571429rem;
  height: 4.6428571429rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}
.dapp-core-component__main__dapp-icon.dapp-core-component__main__icon-medium {
  width: 80px;
  height: 80px;
}

.dapp-core-component__main__centering {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dapp-core-component__main__border-n {
  border: none !important;
}

.dapp-core-component__main__t-shadow {
  text-shadow: 0px 4px 7px rgba(0, 0, 0, 0.3);
}

.dapp-core-component__main__btn {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.04) !important;
}

.dapp-core-component__main__media {
  display: flex;
  align-items: flex-start;
}

.dapp-core-component__main__media-body {
  flex: 1;
}

.dapp-core-component__main__connect-btns .dapp-core-component__main__btn {
  display: flex;
  height: 48px;
  justify-content: center;
  align-items: center;
  color: #fff !important;
  background-color: #1392ff;
  padding: 15px 20px 15px 16px;
  border-radius: 6px;
  gap: 8px;
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__btn:hover, .dapp-core-component__main__connect-btns .dapp-core-component__main__btn:active {
  background-color: #1392ff;
  background-image: linear-gradient(rgba(0, 0, 0, 0.1) 0 0);
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__info {
  color: #1392ff;
  margin-left: 48px;
}

.dapp-core-component__main__btn {
  transition: none !important;
}

.dapp-core-component__main__trust-badge {
  width: 20px;
  height: 20px;
}

/* Navbar
  -------------------------------------------------- */
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name {
  display: flex;
  align-items: center;
  font-size: 1.7142857143rem;
  font-weight: 400;
  color: #9ba5b4;
  margin-left: 1.6428571429rem;
}
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name:before {
  content: "";
  display: block;
  width: 1px;
  height: 3rem;
  margin-right: 1.7142857143rem;
}

/* Dashboard
  -------------------------------------------------- */
.dapp-core-component__main__action-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  margin: 0 7px;
}
.dapp-core-component__main__action-btn button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #fff;
}
.dapp-core-component__main__action-btn button:hover {
  opacity: 0.9;
}

.dapp-core-component__main__transactions .dapp-core-component__main__transaction-icon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table td {
  white-space: nowrap;
  vertical-align: middle;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table tr:last-of-type td {
  border-bottom: 1px solid #dee2e6;
}

/* Loader, Transaction Success/Fail
  -------------------------------------------------- */
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div {
  position: absolute;
  top: 27px;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.1);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.dapp-core-component__main__page-state .dapp-core-component__main__bg-blue .dapp-core-component__main__lds-ellipsis div {
  background: rgba(255, 255, 255, 0.4);
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(1) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis1 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(2) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(3) {
  left: 26px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(4) {
  left: 45px;
  animation: dapp-core-component__main__lds-ellipsis3 0.6s infinite;
}
@keyframes dapp-core-component__main__lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(19px, 0);
  }
}

.dapp-core-component__main__btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
}
.dapp-core-component__main__btn .dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn .dapp-core-component__main__disabled {
  opacity: 0.65;
  cursor: default;
}
.dapp-core-component__main__btn:hover {
  color: #212529;
  text-decoration: none;
}
.dapp-core-component__main__btn:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn:disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn-primary {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary:hover {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-primary:focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary:hover {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-secondary:focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-warning {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning:hover {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning:focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-dark {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark:hover {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
}
.dapp-core-component__main__btn-dark:focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-link {
  font-weight: 400;
  color: #007bff;
  text-decoration: none;
}
.dapp-core-component__main__btn-link:hover {
  color: #0056b3;
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:focus, .dapp-core-component__main__btn-link.dapp-core-component__main__focus {
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:disabled, .dapp-core-component__main__btn-link.dapp-core-component__main__disabled {
  color: #6c757d;
  pointer-events: none;
}

.dapp-core-component__main__btn-light {
  color: #212529;
  background-color: #fafafa;
  border-color: #fafafa;
}
.dapp-core-component__main__btn-light .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}
.dapp-core-component__main__btn-light .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #f8f9fa;
  border-color: #f8f9fa;
}
.dapp-core-component__main__btn-light:hover {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
}
.dapp-core-component__main__btn-light:focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .dapp-core-component__main__badge {
    transition: none;
  }
}
a.dapp-core-component__main__badge:hover, a.dapp-core-component__main__badge:focus {
  text-decoration: none;
}

.dapp-core-component__main__badge:empty {
  display: none;
}

.dapp-core-component__main__btn .dapp-core-component__main__badge {
  position: relative;
  top: -1px;
}

.dapp-core-component__main__badge-pill {
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.dapp-core-component__main__badge-primary {
  color: #fff;
  background-color: #007bff;
}

a.dapp-core-component__main__badge-primary:hover, a.dapp-core-component__main__badge-primary:focus {
  color: #fff;
  background-color: #0062cc;
}

a.dapp-core-component__main__badge-primary:focus, a.dapp-core-component__main__badge-primary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.5);
}

.dapp-core-component__main__badge-secondary {
  color: #fff;
  background-color: #6c757d;
}

a.dapp-core-component__main__badge-secondary:hover, a.dapp-core-component__main__badge-secondary:focus {
  color: #fff;
  background-color: #545b62;
}

a.dapp-core-component__main__badge-secondary:focus, a.dapp-core-component__main__badge-secondary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108, 117, 125, 0.5);
}

.dapp-core-component__main__badge-success {
  color: #fff;
  background-color: #28a745;
}

a.dapp-core-component__main__badge-success:hover, a.dapp-core-component__main__badge-success:focus {
  color: #fff;
  background-color: #1e7e34;
}

a.dapp-core-component__main__badge-success:focus, a.dapp-core-component__main__badge-success.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
}

.dapp-core-component__main__badge-info {
  color: #fff;
  background-color: #17a2b8;
}

a.dapp-core-component__main__badge-info:hover, a.dapp-core-component__main__badge-info:focus {
  color: #fff;
  background-color: #117a8b;
}

a.dapp-core-component__main__badge-info:focus, a.dapp-core-component__main__badge-info.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(23, 162, 184, 0.5);
}

.dapp-core-component__main__badge-warning {
  color: #212529;
  background-color: #ffc107;
}

a.dapp-core-component__main__badge-warning:hover, a.dapp-core-component__main__badge-warning:focus {
  color: #212529;
  background-color: #d39e00;
}

a.dapp-core-component__main__badge-warning:focus, a.dapp-core-component__main__badge-warning.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 193, 7, 0.5);
}

.dapp-core-component__main__badge-danger {
  color: #fff;
  background-color: #dc3545;
}

a.dapp-core-component__main__badge-danger:hover, a.dapp-core-component__main__badge-danger:focus {
  color: #fff;
  background-color: #bd2130;
}

a.dapp-core-component__main__badge-danger:focus, a.dapp-core-component__main__badge-danger.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(220, 53, 69, 0.5);
}

.dapp-core-component__main__badge-light {
  color: #212529;
  background-color: #f8f9fa;
}

a.dapp-core-component__main__badge-light:hover, a.dapp-core-component__main__badge-light:focus {
  color: #212529;
  background-color: #dae0e5;
}

a.dapp-core-component__main__badge-light:focus, a.dapp-core-component__main__badge-light.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(248, 249, 250, 0.5);
}

.dapp-core-component__main__badge-dark {
  color: #fff;
  background-color: #343a40;
}

a.dapp-core-component__main__badge-dark:hover, a.dapp-core-component__main__badge-dark:focus {
  color: #fff;
  background-color: #1d2124;
}

a.dapp-core-component__main__badge-dark:focus, a.dapp-core-component__main__badge-dark.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}

.dapp-core-component__main__card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 5px;
}
.dapp-core-component__main__card > hr {
  margin-right: 0;
  margin-left: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group {
  border-top: inherit;
  border-bottom: inherit;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:first-child {
  border-top-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:last-child {
  border-bottom-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__card-header + .dapp-core-component__main__list-group,
.dapp-core-component__main__card > .dapp-core-component__main__list-group + .dapp-core-component__main__card-footer {
  border-top: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header {
  border: 0;
  padding: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header:first-child {
  border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item {
  border-bottom: 1px solid #dee2e6;
  padding: 1.1rem 1rem;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item h6,
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item .dapp-core-component__main__h6 {
  line-height: 2.5rem;
  letter-spacing: unset;
  margin: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item.dapp-core-component__main__compact {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fafafa;
}

.dapp-core-component__main__card-body {
  flex: 1 1 auto;
  min-height: 1px;
  padding: 1.25rem;
}

.dapp-core-component__main__card-title {
  margin-bottom: 0.75rem;
}

.dapp-core-component__main__card-subtitle {
  margin-top: -0.375rem;
  margin-bottom: 0;
}

.dapp-core-component__main__card-text:last-child {
  margin-bottom: 0;
}

.dapp-core-component__main__card-link:hover {
  text-decoration: none;
}
.dapp-core-component__main__card-link + .dapp-core-component__main__card-link {
  margin-left: 1.25rem;
}

.dapp-core-component__main__card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: #fff;
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__card-footer {
  padding: 0.75rem 1.25rem;
  background-color: #fff;
  border-top: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__text {
  color: #9dabbd;
}

.dapp-core-component__main__text-white {
  color: #fff;
}

.dapp-core-component__main__text-monospace {
  font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-size-xl .dapp-core-component__main__trim {
  max-width: 13rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim {
  max-width: 10rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-only-sm .dapp-core-component__main__trim {
  max-width: none;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-left {
  text-align: left;
}

.dapp-core-component__main__text-right {
  text-align: right;
}

.dapp-core-component__main__text-center {
  text-align: center;
}

.dapp-core-component__main__text-primary {
  color: #1b46c2;
}

.dapp-core-component__main__text-secondary {
  color: #6c757d;
}

.dapp-core-component__main__text-primary-highlight {
  color: #1b46c2;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-success {
  color: #28a745;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-body {
  color: #212529;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-black-50 {
  color: rgba(0, 0, 0, 0.5);
}

.dapp-core-component__main__text-white-50 {
  color: rgba(255, 255, 255, 0.5);
}

@media (min-width: 992px) {
  .dapp-core-component__main__text-lg-right {
    text-align: right;
  }
}

.dapp-core-component__main__text-hide {
  font: 0/0 a;
  color: rgba(0, 0, 0, 0);
  text-shadow: none;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
}

.dapp-core-component__main__text-decoration-none {
  text-decoration: none;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__text-reset {
  color: inherit;
}

.dapp-core-component__main__text-lowercase {
  text-transform: lowercase;
}

.dapp-core-component__main__text-uppercase {
  text-transform: uppercase;
}

.dapp-core-component__main__text-capitalize {
  text-transform: capitalize;
}

.dapp-core-component__main__text-danger {
  color: #d9534f;
}

.dapp-core-component__main__text-warning {
  color: #f0ad4e;
}

.dapp-core-component__main__trim-fs-sm {
  color: inherit;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__m-0 {
  margin: 0;
}

.dapp-core-component__main__mt-0 {
  margin-top: 0;
}

.dapp-core-component__main__my-0 {
  margin-top: 0;
  margin-bottom: 0;
}

.dapp-core-component__main__mr-0 {
  margin-right: 0;
}

.dapp-core-component__main__mx-0 {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__mb-0 {
  margin-bottom: 0;
}

.dapp-core-component__main__my-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.dapp-core-component__main__ml-0 {
  margin-left: 0;
}

.dapp-core-component__main__mx-0 {
  margin-left: 0;
  margin-right: 0;
}

.dapp-core-component__main__m-1 {
  margin: 0.25rem;
}

.dapp-core-component__main__mt-1 {
  margin-top: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__mr-1 {
  margin-right: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.dapp-core-component__main__mb-1 {
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-bottom: 0.25rem;
  margin-top: 0.25rem;
}

.dapp-core-component__main__ml-1 {
  margin-left: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-right: 0.25rem;
  margin-left: 0.25rem;
}

.dapp-core-component__main__m-2 {
  margin: 0.5rem;
}

.dapp-core-component__main__mt-2 {
  margin-top: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-bottom: 0.5rem;
  margin-top: 0.5rem;
}

.dapp-core-component__main__mr-2 {
  margin-right: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__mb-2 {
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__ml-2 {
  margin-left: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__m-3 {
  margin: 1rem;
}

.dapp-core-component__main__mt-3,
.dapp-core-component__main__my-3 {
  margin-top: 1rem;
}

.dapp-core-component__main__mr-3 {
  margin-right: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__mb-3 {
  margin-bottom: 1rem;
}

.dapp-core-component__main__my-3 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.dapp-core-component__main__ml-3 {
  margin-left: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__m-4 {
  margin: 1.5rem;
}

.dapp-core-component__main__mt-4 {
  margin-top: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__mr-4 {
  margin-right: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-right: 1.5rem;
  margin-left: 1.5rem;
}

.dapp-core-component__main__mb-4 {
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.dapp-core-component__main__ml-4 {
  margin-left: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}

.dapp-core-component__main__m-5 {
  margin: 3rem;
}

.dapp-core-component__main__mt-5 {
  margin-top: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__mr-5 {
  margin-right: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-right: 3rem;
  margin-left: 3rem;
}

.dapp-core-component__main__mb-5 {
  margin-bottom: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__ml-5 {
  margin-left: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-left: 3rem;
  margin-right: 3rem;
}

.dapp-core-component__main__p-0 {
  padding: 0;
}

.dapp-core-component__main__pt-0 {
  padding-top: 0;
}

.dapp-core-component__main__py-0 {
  padding-top: 0;
  padding-bottom: 0;
}

.dapp-core-component__main__pr-0 {
  padding-right: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__pb-0 {
  padding-bottom: 0;
}

.dapp-core-component__main__py-0 {
  padding-bottom: 0;
  padding-top: 0;
}

.dapp-core-component__main__pl-0 {
  padding-left: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__p-1 {
  padding: 0.25rem;
}

.dapp-core-component__main__pt-1 {
  padding-top: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pr-1 {
  padding-right: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-right: 0.25rem;
  padding-left: 0.25rem;
}

.dapp-core-component__main__pb-1 {
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pl-1 {
  padding-left: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

.dapp-core-component__main__p-2 {
  padding: 0.5rem;
}

.dapp-core-component__main__pt-2 {
  padding-top: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.dapp-core-component__main__pr-2 {
  padding-right: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__pb-2 {
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__pl-2 {
  padding-left: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__p-3 {
  padding: 1rem;
}

.dapp-core-component__main__pt-3 {
  padding-top: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.dapp-core-component__main__pr-3 {
  padding-right: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__pb-3 {
  padding-bottom: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.dapp-core-component__main__pl-3 {
  padding-left: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__p-4 {
  padding: 1.5rem;
}

.dapp-core-component__main__pt-4 {
  padding-top: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pr-4 {
  padding-right: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.dapp-core-component__main__pb-4 {
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pl-4 {
  padding-left: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
}

.dapp-core-component__main__p-5 {
  padding: 3rem;
}

.dapp-core-component__main__pt-5 {
  padding-top: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-bottom: 3rem;
  padding-top: 3rem;
}

.dapp-core-component__main__pr-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__pb-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.dapp-core-component__main__pl-5 {
  padding-left: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__m-n1 {
  margin: -0.25rem;
}

.dapp-core-component__main__mt-n1 {
  margin-top: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__mr-n1 {
  margin-right: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__mb-n1 {
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-bottom: -0.25rem;
  margin-top: -0.25rem;
}

.dapp-core-component__main__ml-n1 {
  margin-left: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__m-n2 {
  margin: -0.5rem;
}

.dapp-core-component__main__mt-n2 {
  margin-top: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-top: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mr-n2 {
  margin-right: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mb-n2 {
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-bottom: -0.5rem;
  margin-top: -0.5rem;
}

.dapp-core-component__main__ml-n2 {
  margin-left: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.dapp-core-component__main__m-n3 {
  margin: -1rem;
}

.dapp-core-component__main__mt-n3 {
  margin-top: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-top: -1rem;
  margin-bottom: -1rem;
}

.dapp-core-component__main__mr-n3 {
  margin-right: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-right: -1rem;
  margin-left: -1rem;
}

.dapp-core-component__main__mb-n3 {
  margin-bottom: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-bottom: -1rem;
  margin-top: -1rem;
}

.dapp-core-component__main__ml-n3 {
  margin-left: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-left: -1rem;
  margin-right: -1rem;
}

.dapp-core-component__main__m-n4 {
  margin: -1.5rem;
}

.dapp-core-component__main__mt-n4 {
  margin-top: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-top: -1.5rem;
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__mr-n4 {
  margin-right: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__mb-n4 {
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-bottom: -1.5rem;
  margin-top: -1.5rem;
}

.dapp-core-component__main__ml-n4 {
  margin-left: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__m-n5 {
  margin: -3rem;
}

.dapp-core-component__main__mt-n5 {
  margin-top: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-bottom: -3rem;
  margin-top: -3rem;
}

.dapp-core-component__main__mr-n5 {
  margin-right: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-left: -3rem;
  margin-right: -3rem;
}

.dapp-core-component__main__mb-n5 {
  margin-bottom: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-top: -3rem;
  margin-bottom: -3rem;
}

.dapp-core-component__main__ml-n5 {
  margin-left: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-right: -3rem;
  margin-left: -3rem;
}

.dapp-core-component__main__m-auto {
  margin: auto;
}

.dapp-core-component__main__mt-auto {
  margin-top: auto;
}

.dapp-core-component__main__my-auto {
  margin-bottom: auto;
  margin-top: auto;
}

.dapp-core-component__main__mr-auto {
  margin-right: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__mb-auto {
  margin-bottom: auto;
}

.dapp-core-component__main__my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.dapp-core-component__main__ml-auto {
  margin-left: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-1 {
    margin-right: 0.25rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-4 {
    margin-right: 1.5rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mx-lg-4 {
    margin-left: 1.5rem;
    margin-right: 1.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__mb-xl-2 {
    margin-bottom: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__pr-sm-2 {
    padding-right: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__px-sm-2 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__pr-xl-0,
.dapp-core-component__main__px-xl-0 {
    padding-right: 0;
  }
}

.dapp-core-component__main__col {
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

.dapp-core-component__main__col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__col-xl,
.dapp-core-component__main__col-xl-auto,
.dapp-core-component__main__col-xl-12,
.dapp-core-component__main__col-xl-11,
.dapp-core-component__main__col-xl-10,
.dapp-core-component__main__col-xl-9,
.dapp-core-component__main__col-xl-8,
.dapp-core-component__main__col-xl-7,
.dapp-core-component__main__col-xl-6,
.dapp-core-component__main__col-xl-5,
.dapp-core-component__main__col-xl-4,
.dapp-core-component__main__col-xl-3,
.dapp-core-component__main__col-xl-2,
.dapp-core-component__main__col-xl-1,
.dapp-core-component__main__col-lg,
.dapp-core-component__main__col-lg-auto,
.dapp-core-component__main__col-lg-12,
.dapp-core-component__main__col-lg-11,
.dapp-core-component__main__col-lg-10,
.dapp-core-component__main__col-lg-9,
.dapp-core-component__main__col-lg-8,
.dapp-core-component__main__col-lg-7,
.dapp-core-component__main__col-lg-6,
.dapp-core-component__main__col-lg-5,
.dapp-core-component__main__col-lg-4,
.dapp-core-component__main__col-lg-3,
.dapp-core-component__main__col-lg-2,
.dapp-core-component__main__col-lg-1,
.dapp-core-component__main__col-md,
.dapp-core-component__main__col-md-auto,
.dapp-core-component__main__col-md-12,
.dapp-core-component__main__col-md-11,
.dapp-core-component__main__col-md-10,
.dapp-core-component__main__col-md-9,
.dapp-core-component__main__col-md-8,
.dapp-core-component__main__col-md-7,
.dapp-core-component__main__col-md-6,
.dapp-core-component__main__col-md-5,
.dapp-core-component__main__col-md-4,
.dapp-core-component__main__col-md-3,
.dapp-core-component__main__col-md-2,
.dapp-core-component__main__col-md-1,
.dapp-core-component__main__col-sm,
.dapp-core-component__main__col-sm-auto,
.dapp-core-component__main__col-sm-12,
.dapp-core-component__main__col-sm-11,
.dapp-core-component__main__col-sm-10,
.dapp-core-component__main__col-sm-9,
.dapp-core-component__main__col-sm-8,
.dapp-core-component__main__col-sm-7,
.dapp-core-component__main__col-sm-6,
.dapp-core-component__main__col-sm-5,
.dapp-core-component__main__col-sm-4,
.dapp-core-component__main__col-sm-3,
.dapp-core-component__main__col-sm-2,
.dapp-core-component__main__col-sm-1,
.dapp-core-component__main__col,
.dapp-core-component__main__col-auto,
.dapp-core-component__main__col-12,
.dapp-core-component__main__col-11,
.dapp-core-component__main__col-10,
.dapp-core-component__main__col-9,
.dapp-core-component__main__col-8,
.dapp-core-component__main__col-7,
.dapp-core-component__main__col-6,
.dapp-core-component__main__col-5,
.dapp-core-component__main__col-4,
.dapp-core-component__main__col-3,
.dapp-core-component__main__col-2,
.dapp-core-component__main__col-1 {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
}

.dapp-core-component__main__col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

@media (min-width: 576px) {
  .dapp-core-component__main__col-sm-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-sm-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-sm-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-sm-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-sm-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-sm-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-sm-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-sm-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-sm-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-sm-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-sm-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-sm-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 768px) {
  .dapp-core-component__main__col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__col-xl-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-xl-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-xl-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-xl-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-xl-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-xl-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-xl-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-xl-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-xl-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-xl-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-xl-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-xl-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
.dapp-core-component__main__flex-row {
  flex-direction: row;
}

.dapp-core-component__main__flex-column {
  flex-direction: column;
}

.dapp-core-component__main__flex-row-reverse {
  flex-direction: row-reverse;
}

.dapp-core-component__main__flex-column-reverse {
  flex-direction: column-reverse;
}

.dapp-core-component__main__flex-wrap {
  flex-wrap: wrap;
}

.dapp-core-component__main__flex-nowrap {
  flex-wrap: nowrap;
}

.dapp-core-component__main__flex-wrap-reverse {
  flex-wrap: wrap-reverse;
}

.dapp-core-component__main__flex-fill {
  flex: 1 1 auto;
}

.dapp-core-component__main__flex-grow-0 {
  flex-grow: 0;
}

.dapp-core-component__main__flex-grow-1 {
  flex-grow: 1;
}

.dapp-core-component__main__flex-shrink-0 {
  flex-shrink: 0;
}

.dapp-core-component__main__flex-shrink-1 {
  flex-shrink: 1;
}

.dapp-core-component__main__justify-content-start {
  justify-content: flex-start;
}

.dapp-core-component__main__justify-content-end {
  justify-content: flex-end;
}

.dapp-core-component__main__justify-content-center {
  justify-content: center;
}

.dapp-core-component__main__justify-content-between {
  justify-content: space-between;
}

.dapp-core-component__main__justify-content-around {
  justify-content: space-around;
}

.dapp-core-component__main__align-items-start {
  align-items: flex-start;
}

.dapp-core-component__main__align-items-end {
  align-items: flex-end;
}

.dapp-core-component__main__align-items-center {
  align-items: center;
}

.dapp-core-component__main__align-items-baseline {
  align-items: baseline;
}

.dapp-core-component__main__align-items-stretch {
  align-items: stretch;
}

.dapp-core-component__main__align-content-start {
  align-content: flex-start;
}

.dapp-core-component__main__align-content-end {
  align-content: flex-end;
}

.dapp-core-component__main__align-content-center {
  align-content: center;
}

.dapp-core-component__main__align-content-between {
  align-content: space-between;
}

.dapp-core-component__main__align-content-around {
  align-content: space-around;
}

.dapp-core-component__main__align-content-stretch {
  align-content: stretch;
}

.dapp-core-component__main__align-self-auto {
  align-self: auto;
}

.dapp-core-component__main__align-self-start {
  align-self: flex-start;
}

.dapp-core-component__main__align-self-end {
  align-self: flex-end;
}

.dapp-core-component__main__align-self-center {
  align-self: center;
}

.dapp-core-component__main__align-self-baseline {
  align-self: baseline;
}

.dapp-core-component__main__align-self-stretch {
  align-self: stretch;
}

@media (min-width: 576px) {
  .dapp-core-component__main__flex-sm-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-wrap {
    flex-wrap: wrap;
  }
}

.dapp-core-component__main__form-check {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: auto;
  padding-left: 0;
}

.dapp-core-component__main__form-check-label {
  margin-bottom: 0;
}

.dapp-core-component__main__form-check-input {
  position: absolute;
  margin-top: 0.3rem;
  margin-left: -1.25rem;
}
.dapp-core-component__main__form-check-input[disabled] ~ .dapp-core-component__main__form-check-label, .dapp-core-component__main__form-check-input:disabled ~ .dapp-core-component__main__form-check-label {
  color: #6c757d;
}

.dapp-core-component__main__form-group {
  display: flex;
  flex: 0 0 auto;
  flex-flow: row wrap;
  align-items: center;
  margin-bottom: 0;
}

.dapp-core-component__main__form-control {
  display: block;
  width: 100%;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.dapp-core-component__main__form-label {
  padding-top: calc(0.375rem + 1px);
  padding-bottom: calc(0.375rem + 1px);
  margin-bottom: 0;
  font-size: inherit;
  line-height: 1.5;
}

input,
button,
select,
optgroup,
textarea {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.dapp-core-component__main__form-group {
  margin-bottom: 1rem;
}

.dapp-core-component__main__form-text {
  display: block;
  margin-top: 0.25rem;
  color: inherit;
}

.dapp-core-component__main__border {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__border-top {
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__border-right {
  border-right: 1px solid #dee2e6;
}

.dapp-core-component__main__border-bottom {
  border-bottom: 1px solid #dee2e6;
}

.dapp-core-component__main__border-left {
  border-left: 1px solid #dee2e6;
}

.dapp-core-component__main__border-0 {
  border: 0;
}

.dapp-core-component__main__border-top-0 {
  border-top: 0;
}

.dapp-core-component__main__border-right-0 {
  border-right: 0;
}

.dapp-core-component__main__border-bottom-0 {
  border-bottom: 0;
}

.dapp-core-component__main__border-left-0 {
  border-left: 0;
}

.dapp-core-component__main__rounded-sm {
  border-radius: 0.2rem;
}

.dapp-core-component__main__rounded {
  border-radius: 0.25rem;
}

.dapp-core-component__main__rounded-top {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-right {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-bottom {
  border-bottom-right-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-left {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-lg {
  border-radius: 0.3rem;
}

.dapp-core-component__main__rounded-circle {
  border-radius: 50%;
}

.dapp-core-component__main__rounded-pill {
  border-radius: 50rem;
}

.dapp-core-component__main__rounded-0 {
  border-radius: 0;
}

.dapp-core-component__main__h1,
.dapp-core-component__main__h2,
.dapp-core-component__main__h3,
.dapp-core-component__main__h4,
.dapp-core-component__main__h5,
.dapp-core-component__main__h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2;
}

.dapp-core-component__main__h1 {
  font-size: 2.5rem;
}

.dapp-core-component__main__h2 {
  font-size: 2rem;
}

.dapp-core-component__main__h3 {
  font-size: 1.75rem;
}

.dapp-core-component__main__h4 {
  font-size: 1.5rem;
}

.dapp-core-component__main__h5 {
  font-size: 1.25rem;
}

.dapp-core-component__main__h6 {
  font-size: 1rem;
}

.dapp-core-component__main__lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

.dapp-core-component__main__bg-warning,
.dapp-core-component__main__warning {
  background-color: #f0ad4e;
}

.dapp-core-component__main__bg-danger,
.dapp-core-component__main__danger {
  background-color: #d9534f;
}

.dapp-core-component__main__bg-success,
.dapp-core-component__main__success {
  background-color: #5cb85c;
}

.dapp-core-component__main__position-relative {
  position: relative;
}

.dapp-core-component__main__position-absolute {
  position: absolute;
}

.dapp-core-component__main__w-25 {
  width: 25%;
}

.dapp-core-component__main__w-50 {
  width: 50%;
}

.dapp-core-component__main__w-75 {
  width: 75%;
}

.dapp-core-component__main__w-100 {
  width: 100%;
}

.dapp-core-component__main__w-auto {
  width: auto;
}

.dapp-core-component__main__h-25 {
  height: 25%;
}

.dapp-core-component__main__h-50 {
  height: 50%;
}

.dapp-core-component__main__h-75 {
  height: 75%;
}

.dapp-core-component__main__h-100 {
  height: 100%;
}

.dapp-core-component__main__h-auto {
  height: auto;
}

.dapp-core-component__main__mw-100 {
  max-width: 100%;
}

.dapp-core-component__main__mh-100 {
  max-height: 100%;
}

.dapp-core-component__main__min-vw-100 {
  min-width: 100vw;
}

.dapp-core-component__main__min-vh-100 {
  min-height: 100vh;
}

.dapp-core-component__main__vw-100 {
  width: 100vw;
}

.dapp-core-component__main__vh-100 {
  height: 100vh;
}

.dapp-core-component__main__locked-icon {
  color: inherit;
}

.dapp-core-component__main__shadow-sm {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
}

.dapp-core-component__main__shadow {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.dapp-core-component__main__shadow-lg {
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.175);
}

.dapp-core-component__main__shadow-none {
  box-shadow: none;
}

.dapp-core-component__main__d-none {
  display: none;
}

.dapp-core-component__main__d-inline {
  display: inline;
}

.dapp-core-component__main__d-inline-block {
  display: inline-block;
}

.dapp-core-component__main__d-block {
  display: block;
}

.dapp-core-component__main__d-table {
  display: table;
}

.dapp-core-component__main__d-table-row {
  display: table-row;
}

.dapp-core-component__main__d-table-cell {
  display: table-cell;
}

.dapp-core-component__main__d-flex {
  display: -ms-flexbox;
  display: flex;
}

.dapp-core-component__main__d-inline-flex {
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media (min-width: 576px) {
  .dapp-core-component__main__d-sm-flex {
    display: flex;
  }
}

.dapp-core-component__main__table {
  width: 100%;
  margin-bottom: 1rem;
  color: #212529;
}
.dapp-core-component__main__table th,
.dapp-core-component__main__table td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #dee2e6;
}
.dapp-core-component__main__table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #dee2e6;
}
.dapp-core-component__main__table tbody + tbody {
  border-top: 2px solid #dee2e6;
}

.dapp-core-component__main__table-sm th,
.dapp-core-component__main__table-sm td {
  padding: 0.3rem;
}

.dapp-core-component__main__table-bordered {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered th,
.dapp-core-component__main__table-bordered td {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered thead th,
.dapp-core-component__main__table-bordered thead td {
  border-bottom-width: 2px;
}

.dapp-core-component__main__table-borderless th,
.dapp-core-component__main__table-borderless td,
.dapp-core-component__main__table-borderless thead th,
.dapp-core-component__main__table-borderless tbody + tbody {
  border: 0;
}

.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(0, 0, 0, 0.05);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-dark th {
  color: #fff;
  background-color: #343a40;
  border-color: #454d55;
}
.dapp-core-component__main__table .dapp-core-component__main__thead-light th {
  color: #495057;
  background-color: #e9ecef;
  border-color: #dee2e6;
}

.dapp-core-component__main__table-dark {
  color: #fff;
  background-color: #343a40;
}
.dapp-core-component__main__table-dark th,
.dapp-core-component__main__table-dark td,
.dapp-core-component__main__table-dark thead th {
  border-color: #454d55;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-bordered {
  border: 0;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(255, 255, 255, 0.05);
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-hover tbody tr {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.075);
}

.dapp-core-component__main__table-responsive {
  display: block;
  width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}
.dapp-core-component__main__table-responsive > .dapp-core-component__main__table-bordered {
  border: 0;
}

@media (max-width: 767.98px) {
  .dapp-core-component__main__transactions.dapp-core-component__main__table .dapp-core-component__main__transaction-hash .dapp-core-component__main__trim {
    max-width: 8rem;
  }
}
.dapp-core-component__main__font-weight-light {
  font-weight: 300 !important;
}

.dapp-core-component__main__font-weight-lighter {
  font-weight: lighter !important;
}

.dapp-core-component__main__font-weight-normal {
  font-weight: 400 !important;
}

.dapp-core-component__main__font-weight-medium {
  font-weight: 500 !important;
}

.dapp-core-component__main__font-weight-bold {
  font-weight: 700 !important;
}

.dapp-core-component__main__font-weight-bolder {
  font-weight: bolder !important;
}

.dapp-core-component__main__font-italic {
  font-style: italic !important;
}

.dapp-core-component__main__icon-state {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
}
.dapp-core-component__main__icon-state.dapp-core-component__main__half {
  width: 3rem;
  height: 3rem;
}

.dapp-core-component__main__side-icon {
  width: 1.5rem;
  height: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  color: transparent;
  border-radius: 50%;
  overflow: hidden;
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__side-icon svg {
  color: #cacbcc;
}

div.dapp-core-component__main__side-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
div.dapp-core-component__main__side-icon img {
  transform: scale(0.75);
}

.dapp-core-component__main__toast-messages {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast {
  display: block;
  flex-basis: 4rem;
  width: 100%;
  max-width: 28.875rem;
  transition: transform 0.3s ease-in-out;
  transform: translateX(120%);
  background-color: #fff;
  border-radius: 0.25rem;
  border-color: #dee2e6;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__clickable {
  cursor: pointer;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible {
  margin-bottom: 0.3rem;
  transform: translateX(0);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress {
  background-color: rgba(0, 0, 0, 0);
  height: inherit;
  font-size: inherit;
  line-height: inherit;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar {
  background-color: rgba(108, 117, 125, 0.2);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar .dapp-core-component__main__content-height {
  z-index: -1;
  visibility: hidden;
  width: 0 !important;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast .dapp-core-component__main__tx-description {
  margin-bottom: 0.2rem;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close {
  opacity: 1;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close svg {
  color: #6c757d;
}

.dapp-core-component__main__data-field .dapp-core-component__main__textarea {
  -moz-appearance: textfield-multiline;
  -webkit-appearance: textarea;
  border: 1px solid #000;
  background-color: #fff;
  border-radius: 5px;
  font: medium -moz-fixed;
  font-size: 0.8125rem;
  min-height: 5rem;
  overflow: auto;
  padding: 0.375rem 0.75rem;
  resize: both;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea:hover {
  border: 1px solid #000;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea .dapp-core-component__main__highlighted {
  font-size: 1rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea.dapp-core-component__main__sc-call {
  min-height: 3rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__form-control:disabled,
.dapp-core-component__main__data-field .dapp-core-component__main__form-control[readonly] {
  background-color: #fafafa;
}
.dapp-core-component__main__data-field .dapp-core-component__main__small-font {
  font-size: 0.8125rem;
}

.dapp-core-component__main__progress-steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps hr {
  background-color: #6c757d;
  border-color: #dee2e6;
  z-index: 2;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot {
  height: 0.625rem;
  width: 0.625rem;
  border-radius: 50%;
  margin: 0 1rem;
  z-index: 5;
  background-color: #6c757d;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:first-of-type {
  margin-left: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:last-of-type {
  margin-right: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

a.dapp-core-component__main__side-link::after {
  right: 0;
  left: unset;
  width: calc(100% - 1.85rem);
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Tp));var d={"link-style":"dapp-core-component__main__link-style",linkStyle:"dapp-core-component__main__link-style","link-second-style":"dapp-core-component__main__link-second-style",linkSecondStyle:"dapp-core-component__main__link-second-style",wrapper:"dapp-core-component__main__wrapper","opacity-6":"dapp-core-component__main__opacity-6",opacity6:"dapp-core-component__main__opacity-6","qr-code-svg-container":"dapp-core-component__main__qr-code-svg-container",qrCodeSvgContainer:"dapp-core-component__main__qr-code-svg-container","pair-list":"dapp-core-component__main__pair-list",pairList:"dapp-core-component__main__pair-list","pair-image":"dapp-core-component__main__pair-image",pairImage:"dapp-core-component__main__pair-image","pair-remove":"dapp-core-component__main__pair-remove",pairRemove:"dapp-core-component__main__pair-remove","dapp-icon":"dapp-core-component__main__dapp-icon",dappIcon:"dapp-core-component__main__dapp-icon","icon-medium":"dapp-core-component__main__icon-medium",iconMedium:"dapp-core-component__main__icon-medium",centering:"dapp-core-component__main__centering","border-n":"dapp-core-component__main__border-n",borderN:"dapp-core-component__main__border-n","t-shadow":"dapp-core-component__main__t-shadow",tShadow:"dapp-core-component__main__t-shadow",btn:"dapp-core-component__main__btn",media:"dapp-core-component__main__media","media-body":"dapp-core-component__main__media-body",mediaBody:"dapp-core-component__main__media-body","connect-btns":"dapp-core-component__main__connect-btns",connectBtns:"dapp-core-component__main__connect-btns",info:"dapp-core-component__main__info","trust-badge":"dapp-core-component__main__trust-badge",trustBadge:"dapp-core-component__main__trust-badge",navbar:"dapp-core-component__main__navbar","dapp-name":"dapp-core-component__main__dapp-name",dappName:"dapp-core-component__main__dapp-name","action-btn":"dapp-core-component__main__action-btn",actionBtn:"dapp-core-component__main__action-btn",transactions:"dapp-core-component__main__transactions","transaction-icon":"dapp-core-component__main__transaction-icon",transactionIcon:"dapp-core-component__main__transaction-icon",table:"dapp-core-component__main__table","page-state":"dapp-core-component__main__page-state",pageState:"dapp-core-component__main__page-state","lds-ellipsis":"dapp-core-component__main__lds-ellipsis",ldsEllipsis:"dapp-core-component__main__lds-ellipsis","bg-blue":"dapp-core-component__main__bg-blue",bgBlue:"dapp-core-component__main__bg-blue","lds-ellipsis1":"dapp-core-component__main__lds-ellipsis1",ldsEllipsis1:"dapp-core-component__main__lds-ellipsis1","lds-ellipsis2":"dapp-core-component__main__lds-ellipsis2",ldsEllipsis2:"dapp-core-component__main__lds-ellipsis2","lds-ellipsis3":"dapp-core-component__main__lds-ellipsis3",ldsEllipsis3:"dapp-core-component__main__lds-ellipsis3",focus:"dapp-core-component__main__focus",disabled:"dapp-core-component__main__disabled","btn-primary":"dapp-core-component__main__btn-primary",btnPrimary:"dapp-core-component__main__btn-primary","btn-secondary":"dapp-core-component__main__btn-secondary",btnSecondary:"dapp-core-component__main__btn-secondary","btn-warning":"dapp-core-component__main__btn-warning",btnWarning:"dapp-core-component__main__btn-warning","btn-dark":"dapp-core-component__main__btn-dark",btnDark:"dapp-core-component__main__btn-dark","btn-link":"dapp-core-component__main__btn-link",btnLink:"dapp-core-component__main__btn-link","btn-light":"dapp-core-component__main__btn-light",btnLight:"dapp-core-component__main__btn-light",badge:"dapp-core-component__main__badge","badge-pill":"dapp-core-component__main__badge-pill",badgePill:"dapp-core-component__main__badge-pill","badge-primary":"dapp-core-component__main__badge-primary",badgePrimary:"dapp-core-component__main__badge-primary","badge-secondary":"dapp-core-component__main__badge-secondary",badgeSecondary:"dapp-core-component__main__badge-secondary","badge-success":"dapp-core-component__main__badge-success",badgeSuccess:"dapp-core-component__main__badge-success","badge-info":"dapp-core-component__main__badge-info",badgeInfo:"dapp-core-component__main__badge-info","badge-warning":"dapp-core-component__main__badge-warning",badgeWarning:"dapp-core-component__main__badge-warning","badge-danger":"dapp-core-component__main__badge-danger",badgeDanger:"dapp-core-component__main__badge-danger","badge-light":"dapp-core-component__main__badge-light",badgeLight:"dapp-core-component__main__badge-light","badge-dark":"dapp-core-component__main__badge-dark",badgeDark:"dapp-core-component__main__badge-dark",card:"dapp-core-component__main__card","list-group":"dapp-core-component__main__list-group",listGroup:"dapp-core-component__main__list-group","card-header":"dapp-core-component__main__card-header",cardHeader:"dapp-core-component__main__card-header","card-footer":"dapp-core-component__main__card-footer",cardFooter:"dapp-core-component__main__card-footer","card-header-item":"dapp-core-component__main__card-header-item",cardHeaderItem:"dapp-core-component__main__card-header-item",h6:"dapp-core-component__main__h6",compact:"dapp-core-component__main__compact","card-body":"dapp-core-component__main__card-body",cardBody:"dapp-core-component__main__card-body","card-title":"dapp-core-component__main__card-title",cardTitle:"dapp-core-component__main__card-title","card-subtitle":"dapp-core-component__main__card-subtitle",cardSubtitle:"dapp-core-component__main__card-subtitle","card-text":"dapp-core-component__main__card-text",cardText:"dapp-core-component__main__card-text","card-link":"dapp-core-component__main__card-link",cardLink:"dapp-core-component__main__card-link",text:"dapp-core-component__main__text","text-white":"dapp-core-component__main__text-white",textWhite:"dapp-core-component__main__text-white","text-monospace":"dapp-core-component__main__text-monospace",textMonospace:"dapp-core-component__main__text-monospace","trim-size-xl":"dapp-core-component__main__trim-size-xl",trimSizeXl:"dapp-core-component__main__trim-size-xl",trim:"dapp-core-component__main__trim","trim-only-sm":"dapp-core-component__main__trim-only-sm",trimOnlySm:"dapp-core-component__main__trim-only-sm","text-justify":"dapp-core-component__main__text-justify",textJustify:"dapp-core-component__main__text-justify","text-wrap":"dapp-core-component__main__text-wrap",textWrap:"dapp-core-component__main__text-wrap","text-nowrap":"dapp-core-component__main__text-nowrap",textNowrap:"dapp-core-component__main__text-nowrap","text-truncate":"dapp-core-component__main__text-truncate",textTruncate:"dapp-core-component__main__text-truncate","text-left":"dapp-core-component__main__text-left",textLeft:"dapp-core-component__main__text-left","text-right":"dapp-core-component__main__text-right",textRight:"dapp-core-component__main__text-right","text-center":"dapp-core-component__main__text-center",textCenter:"dapp-core-component__main__text-center","text-primary":"dapp-core-component__main__text-primary",textPrimary:"dapp-core-component__main__text-primary","text-secondary":"dapp-core-component__main__text-secondary",textSecondary:"dapp-core-component__main__text-secondary","text-primary-highlight":"dapp-core-component__main__text-primary-highlight",textPrimaryHighlight:"dapp-core-component__main__text-primary-highlight","text-muted":"dapp-core-component__main__text-muted",textMuted:"dapp-core-component__main__text-muted","text-success":"dapp-core-component__main__text-success",textSuccess:"dapp-core-component__main__text-success","text-body":"dapp-core-component__main__text-body",textBody:"dapp-core-component__main__text-body","text-black-50":"dapp-core-component__main__text-black-50",textBlack50:"dapp-core-component__main__text-black-50","text-white-50":"dapp-core-component__main__text-white-50",textWhite50:"dapp-core-component__main__text-white-50","text-lg-right":"dapp-core-component__main__text-lg-right",textLgRight:"dapp-core-component__main__text-lg-right","text-hide":"dapp-core-component__main__text-hide",textHide:"dapp-core-component__main__text-hide","text-decoration-none":"dapp-core-component__main__text-decoration-none",textDecorationNone:"dapp-core-component__main__text-decoration-none","text-break":"dapp-core-component__main__text-break",textBreak:"dapp-core-component__main__text-break","text-reset":"dapp-core-component__main__text-reset",textReset:"dapp-core-component__main__text-reset","text-lowercase":"dapp-core-component__main__text-lowercase",textLowercase:"dapp-core-component__main__text-lowercase","text-uppercase":"dapp-core-component__main__text-uppercase",textUppercase:"dapp-core-component__main__text-uppercase","text-capitalize":"dapp-core-component__main__text-capitalize",textCapitalize:"dapp-core-component__main__text-capitalize","text-danger":"dapp-core-component__main__text-danger",textDanger:"dapp-core-component__main__text-danger","text-warning":"dapp-core-component__main__text-warning",textWarning:"dapp-core-component__main__text-warning","trim-fs-sm":"dapp-core-component__main__trim-fs-sm",trimFsSm:"dapp-core-component__main__trim-fs-sm","mx-auto":"dapp-core-component__main__mx-auto",mxAuto:"dapp-core-component__main__mx-auto","m-0":"dapp-core-component__main__m-0",m0:"dapp-core-component__main__m-0","mt-0":"dapp-core-component__main__mt-0",mt0:"dapp-core-component__main__mt-0","my-0":"dapp-core-component__main__my-0",my0:"dapp-core-component__main__my-0","mr-0":"dapp-core-component__main__mr-0",mr0:"dapp-core-component__main__mr-0","mx-0":"dapp-core-component__main__mx-0",mx0:"dapp-core-component__main__mx-0","mb-0":"dapp-core-component__main__mb-0",mb0:"dapp-core-component__main__mb-0","ml-0":"dapp-core-component__main__ml-0",ml0:"dapp-core-component__main__ml-0","m-1":"dapp-core-component__main__m-1",m1:"dapp-core-component__main__m-1","mt-1":"dapp-core-component__main__mt-1",mt1:"dapp-core-component__main__mt-1","my-1":"dapp-core-component__main__my-1",my1:"dapp-core-component__main__my-1","mr-1":"dapp-core-component__main__mr-1",mr1:"dapp-core-component__main__mr-1","mx-1":"dapp-core-component__main__mx-1",mx1:"dapp-core-component__main__mx-1","mb-1":"dapp-core-component__main__mb-1",mb1:"dapp-core-component__main__mb-1","ml-1":"dapp-core-component__main__ml-1",ml1:"dapp-core-component__main__ml-1","m-2":"dapp-core-component__main__m-2",m2:"dapp-core-component__main__m-2","mt-2":"dapp-core-component__main__mt-2",mt2:"dapp-core-component__main__mt-2","my-2":"dapp-core-component__main__my-2",my2:"dapp-core-component__main__my-2","mr-2":"dapp-core-component__main__mr-2",mr2:"dapp-core-component__main__mr-2","mx-2":"dapp-core-component__main__mx-2",mx2:"dapp-core-component__main__mx-2","mb-2":"dapp-core-component__main__mb-2",mb2:"dapp-core-component__main__mb-2","ml-2":"dapp-core-component__main__ml-2",ml2:"dapp-core-component__main__ml-2","m-3":"dapp-core-component__main__m-3",m3:"dapp-core-component__main__m-3","mt-3":"dapp-core-component__main__mt-3",mt3:"dapp-core-component__main__mt-3","my-3":"dapp-core-component__main__my-3",my3:"dapp-core-component__main__my-3","mr-3":"dapp-core-component__main__mr-3",mr3:"dapp-core-component__main__mr-3","mx-3":"dapp-core-component__main__mx-3",mx3:"dapp-core-component__main__mx-3","mb-3":"dapp-core-component__main__mb-3",mb3:"dapp-core-component__main__mb-3","ml-3":"dapp-core-component__main__ml-3",ml3:"dapp-core-component__main__ml-3","m-4":"dapp-core-component__main__m-4",m4:"dapp-core-component__main__m-4","mt-4":"dapp-core-component__main__mt-4",mt4:"dapp-core-component__main__mt-4","my-4":"dapp-core-component__main__my-4",my4:"dapp-core-component__main__my-4","mr-4":"dapp-core-component__main__mr-4",mr4:"dapp-core-component__main__mr-4","mx-4":"dapp-core-component__main__mx-4",mx4:"dapp-core-component__main__mx-4","mb-4":"dapp-core-component__main__mb-4",mb4:"dapp-core-component__main__mb-4","ml-4":"dapp-core-component__main__ml-4",ml4:"dapp-core-component__main__ml-4","m-5":"dapp-core-component__main__m-5",m5:"dapp-core-component__main__m-5","mt-5":"dapp-core-component__main__mt-5",mt5:"dapp-core-component__main__mt-5","my-5":"dapp-core-component__main__my-5",my5:"dapp-core-component__main__my-5","mr-5":"dapp-core-component__main__mr-5",mr5:"dapp-core-component__main__mr-5","mx-5":"dapp-core-component__main__mx-5",mx5:"dapp-core-component__main__mx-5","mb-5":"dapp-core-component__main__mb-5",mb5:"dapp-core-component__main__mb-5","ml-5":"dapp-core-component__main__ml-5",ml5:"dapp-core-component__main__ml-5","p-0":"dapp-core-component__main__p-0",p0:"dapp-core-component__main__p-0","pt-0":"dapp-core-component__main__pt-0",pt0:"dapp-core-component__main__pt-0","py-0":"dapp-core-component__main__py-0",py0:"dapp-core-component__main__py-0","pr-0":"dapp-core-component__main__pr-0",pr0:"dapp-core-component__main__pr-0","px-0":"dapp-core-component__main__px-0",px0:"dapp-core-component__main__px-0","pb-0":"dapp-core-component__main__pb-0",pb0:"dapp-core-component__main__pb-0","pl-0":"dapp-core-component__main__pl-0",pl0:"dapp-core-component__main__pl-0","p-1":"dapp-core-component__main__p-1",p1:"dapp-core-component__main__p-1","pt-1":"dapp-core-component__main__pt-1",pt1:"dapp-core-component__main__pt-1","py-1":"dapp-core-component__main__py-1",py1:"dapp-core-component__main__py-1","pr-1":"dapp-core-component__main__pr-1",pr1:"dapp-core-component__main__pr-1","px-1":"dapp-core-component__main__px-1",px1:"dapp-core-component__main__px-1","pb-1":"dapp-core-component__main__pb-1",pb1:"dapp-core-component__main__pb-1","pl-1":"dapp-core-component__main__pl-1",pl1:"dapp-core-component__main__pl-1","p-2":"dapp-core-component__main__p-2",p2:"dapp-core-component__main__p-2","pt-2":"dapp-core-component__main__pt-2",pt2:"dapp-core-component__main__pt-2","py-2":"dapp-core-component__main__py-2",py2:"dapp-core-component__main__py-2","pr-2":"dapp-core-component__main__pr-2",pr2:"dapp-core-component__main__pr-2","px-2":"dapp-core-component__main__px-2",px2:"dapp-core-component__main__px-2","pb-2":"dapp-core-component__main__pb-2",pb2:"dapp-core-component__main__pb-2","pl-2":"dapp-core-component__main__pl-2",pl2:"dapp-core-component__main__pl-2","p-3":"dapp-core-component__main__p-3",p3:"dapp-core-component__main__p-3","pt-3":"dapp-core-component__main__pt-3",pt3:"dapp-core-component__main__pt-3","py-3":"dapp-core-component__main__py-3",py3:"dapp-core-component__main__py-3","pr-3":"dapp-core-component__main__pr-3",pr3:"dapp-core-component__main__pr-3","px-3":"dapp-core-component__main__px-3",px3:"dapp-core-component__main__px-3","pb-3":"dapp-core-component__main__pb-3",pb3:"dapp-core-component__main__pb-3","pl-3":"dapp-core-component__main__pl-3",pl3:"dapp-core-component__main__pl-3","p-4":"dapp-core-component__main__p-4",p4:"dapp-core-component__main__p-4","pt-4":"dapp-core-component__main__pt-4",pt4:"dapp-core-component__main__pt-4","py-4":"dapp-core-component__main__py-4",py4:"dapp-core-component__main__py-4","pr-4":"dapp-core-component__main__pr-4",pr4:"dapp-core-component__main__pr-4","px-4":"dapp-core-component__main__px-4",px4:"dapp-core-component__main__px-4","pb-4":"dapp-core-component__main__pb-4",pb4:"dapp-core-component__main__pb-4","pl-4":"dapp-core-component__main__pl-4",pl4:"dapp-core-component__main__pl-4","p-5":"dapp-core-component__main__p-5",p5:"dapp-core-component__main__p-5","pt-5":"dapp-core-component__main__pt-5",pt5:"dapp-core-component__main__pt-5","py-5":"dapp-core-component__main__py-5",py5:"dapp-core-component__main__py-5","pr-5":"dapp-core-component__main__pr-5",pr5:"dapp-core-component__main__pr-5","px-5":"dapp-core-component__main__px-5",px5:"dapp-core-component__main__px-5","pb-5":"dapp-core-component__main__pb-5",pb5:"dapp-core-component__main__pb-5","pl-5":"dapp-core-component__main__pl-5",pl5:"dapp-core-component__main__pl-5","m-n1":"dapp-core-component__main__m-n1",mN1:"dapp-core-component__main__m-n1","mt-n1":"dapp-core-component__main__mt-n1",mtN1:"dapp-core-component__main__mt-n1","my-n1":"dapp-core-component__main__my-n1",myN1:"dapp-core-component__main__my-n1","mr-n1":"dapp-core-component__main__mr-n1",mrN1:"dapp-core-component__main__mr-n1","mx-n1":"dapp-core-component__main__mx-n1",mxN1:"dapp-core-component__main__mx-n1","mb-n1":"dapp-core-component__main__mb-n1",mbN1:"dapp-core-component__main__mb-n1","ml-n1":"dapp-core-component__main__ml-n1",mlN1:"dapp-core-component__main__ml-n1","m-n2":"dapp-core-component__main__m-n2",mN2:"dapp-core-component__main__m-n2","mt-n2":"dapp-core-component__main__mt-n2",mtN2:"dapp-core-component__main__mt-n2","my-n2":"dapp-core-component__main__my-n2",myN2:"dapp-core-component__main__my-n2","mr-n2":"dapp-core-component__main__mr-n2",mrN2:"dapp-core-component__main__mr-n2","mx-n2":"dapp-core-component__main__mx-n2",mxN2:"dapp-core-component__main__mx-n2","mb-n2":"dapp-core-component__main__mb-n2",mbN2:"dapp-core-component__main__mb-n2","ml-n2":"dapp-core-component__main__ml-n2",mlN2:"dapp-core-component__main__ml-n2","m-n3":"dapp-core-component__main__m-n3",mN3:"dapp-core-component__main__m-n3","mt-n3":"dapp-core-component__main__mt-n3",mtN3:"dapp-core-component__main__mt-n3","my-n3":"dapp-core-component__main__my-n3",myN3:"dapp-core-component__main__my-n3","mr-n3":"dapp-core-component__main__mr-n3",mrN3:"dapp-core-component__main__mr-n3","mx-n3":"dapp-core-component__main__mx-n3",mxN3:"dapp-core-component__main__mx-n3","mb-n3":"dapp-core-component__main__mb-n3",mbN3:"dapp-core-component__main__mb-n3","ml-n3":"dapp-core-component__main__ml-n3",mlN3:"dapp-core-component__main__ml-n3","m-n4":"dapp-core-component__main__m-n4",mN4:"dapp-core-component__main__m-n4","mt-n4":"dapp-core-component__main__mt-n4",mtN4:"dapp-core-component__main__mt-n4","my-n4":"dapp-core-component__main__my-n4",myN4:"dapp-core-component__main__my-n4","mr-n4":"dapp-core-component__main__mr-n4",mrN4:"dapp-core-component__main__mr-n4","mx-n4":"dapp-core-component__main__mx-n4",mxN4:"dapp-core-component__main__mx-n4","mb-n4":"dapp-core-component__main__mb-n4",mbN4:"dapp-core-component__main__mb-n4","ml-n4":"dapp-core-component__main__ml-n4",mlN4:"dapp-core-component__main__ml-n4","m-n5":"dapp-core-component__main__m-n5",mN5:"dapp-core-component__main__m-n5","mt-n5":"dapp-core-component__main__mt-n5",mtN5:"dapp-core-component__main__mt-n5","my-n5":"dapp-core-component__main__my-n5",myN5:"dapp-core-component__main__my-n5","mr-n5":"dapp-core-component__main__mr-n5",mrN5:"dapp-core-component__main__mr-n5","mx-n5":"dapp-core-component__main__mx-n5",mxN5:"dapp-core-component__main__mx-n5","mb-n5":"dapp-core-component__main__mb-n5",mbN5:"dapp-core-component__main__mb-n5","ml-n5":"dapp-core-component__main__ml-n5",mlN5:"dapp-core-component__main__ml-n5","m-auto":"dapp-core-component__main__m-auto",mAuto:"dapp-core-component__main__m-auto","mt-auto":"dapp-core-component__main__mt-auto",mtAuto:"dapp-core-component__main__mt-auto","my-auto":"dapp-core-component__main__my-auto",myAuto:"dapp-core-component__main__my-auto","mr-auto":"dapp-core-component__main__mr-auto",mrAuto:"dapp-core-component__main__mr-auto","mb-auto":"dapp-core-component__main__mb-auto",mbAuto:"dapp-core-component__main__mb-auto","ml-auto":"dapp-core-component__main__ml-auto",mlAuto:"dapp-core-component__main__ml-auto","mr-lg-1":"dapp-core-component__main__mr-lg-1",mrLg1:"dapp-core-component__main__mr-lg-1","mr-lg-4":"dapp-core-component__main__mr-lg-4",mrLg4:"dapp-core-component__main__mr-lg-4","mx-lg-4":"dapp-core-component__main__mx-lg-4",mxLg4:"dapp-core-component__main__mx-lg-4","mb-xl-2":"dapp-core-component__main__mb-xl-2",mbXl2:"dapp-core-component__main__mb-xl-2","pr-sm-2":"dapp-core-component__main__pr-sm-2",prSm2:"dapp-core-component__main__pr-sm-2","px-sm-2":"dapp-core-component__main__px-sm-2",pxSm2:"dapp-core-component__main__px-sm-2","pr-xl-0":"dapp-core-component__main__pr-xl-0",prXl0:"dapp-core-component__main__pr-xl-0","px-xl-0":"dapp-core-component__main__px-xl-0",pxXl0:"dapp-core-component__main__px-xl-0",col:"dapp-core-component__main__col",row:"dapp-core-component__main__row","col-xl":"dapp-core-component__main__col-xl",colXl:"dapp-core-component__main__col-xl","col-xl-auto":"dapp-core-component__main__col-xl-auto",colXlAuto:"dapp-core-component__main__col-xl-auto","col-xl-12":"dapp-core-component__main__col-xl-12",colXl12:"dapp-core-component__main__col-xl-12","col-xl-11":"dapp-core-component__main__col-xl-11",colXl11:"dapp-core-component__main__col-xl-11","col-xl-10":"dapp-core-component__main__col-xl-10",colXl10:"dapp-core-component__main__col-xl-10","col-xl-9":"dapp-core-component__main__col-xl-9",colXl9:"dapp-core-component__main__col-xl-9","col-xl-8":"dapp-core-component__main__col-xl-8",colXl8:"dapp-core-component__main__col-xl-8","col-xl-7":"dapp-core-component__main__col-xl-7",colXl7:"dapp-core-component__main__col-xl-7","col-xl-6":"dapp-core-component__main__col-xl-6",colXl6:"dapp-core-component__main__col-xl-6","col-xl-5":"dapp-core-component__main__col-xl-5",colXl5:"dapp-core-component__main__col-xl-5","col-xl-4":"dapp-core-component__main__col-xl-4",colXl4:"dapp-core-component__main__col-xl-4","col-xl-3":"dapp-core-component__main__col-xl-3",colXl3:"dapp-core-component__main__col-xl-3","col-xl-2":"dapp-core-component__main__col-xl-2",colXl2:"dapp-core-component__main__col-xl-2","col-xl-1":"dapp-core-component__main__col-xl-1",colXl1:"dapp-core-component__main__col-xl-1","col-lg":"dapp-core-component__main__col-lg",colLg:"dapp-core-component__main__col-lg","col-lg-auto":"dapp-core-component__main__col-lg-auto",colLgAuto:"dapp-core-component__main__col-lg-auto","col-lg-12":"dapp-core-component__main__col-lg-12",colLg12:"dapp-core-component__main__col-lg-12","col-lg-11":"dapp-core-component__main__col-lg-11",colLg11:"dapp-core-component__main__col-lg-11","col-lg-10":"dapp-core-component__main__col-lg-10",colLg10:"dapp-core-component__main__col-lg-10","col-lg-9":"dapp-core-component__main__col-lg-9",colLg9:"dapp-core-component__main__col-lg-9","col-lg-8":"dapp-core-component__main__col-lg-8",colLg8:"dapp-core-component__main__col-lg-8","col-lg-7":"dapp-core-component__main__col-lg-7",colLg7:"dapp-core-component__main__col-lg-7","col-lg-6":"dapp-core-component__main__col-lg-6",colLg6:"dapp-core-component__main__col-lg-6","col-lg-5":"dapp-core-component__main__col-lg-5",colLg5:"dapp-core-component__main__col-lg-5","col-lg-4":"dapp-core-component__main__col-lg-4",colLg4:"dapp-core-component__main__col-lg-4","col-lg-3":"dapp-core-component__main__col-lg-3",colLg3:"dapp-core-component__main__col-lg-3","col-lg-2":"dapp-core-component__main__col-lg-2",colLg2:"dapp-core-component__main__col-lg-2","col-lg-1":"dapp-core-component__main__col-lg-1",colLg1:"dapp-core-component__main__col-lg-1","col-md":"dapp-core-component__main__col-md",colMd:"dapp-core-component__main__col-md","col-md-auto":"dapp-core-component__main__col-md-auto",colMdAuto:"dapp-core-component__main__col-md-auto","col-md-12":"dapp-core-component__main__col-md-12",colMd12:"dapp-core-component__main__col-md-12","col-md-11":"dapp-core-component__main__col-md-11",colMd11:"dapp-core-component__main__col-md-11","col-md-10":"dapp-core-component__main__col-md-10",colMd10:"dapp-core-component__main__col-md-10","col-md-9":"dapp-core-component__main__col-md-9",colMd9:"dapp-core-component__main__col-md-9","col-md-8":"dapp-core-component__main__col-md-8",colMd8:"dapp-core-component__main__col-md-8","col-md-7":"dapp-core-component__main__col-md-7",colMd7:"dapp-core-component__main__col-md-7","col-md-6":"dapp-core-component__main__col-md-6",colMd6:"dapp-core-component__main__col-md-6","col-md-5":"dapp-core-component__main__col-md-5",colMd5:"dapp-core-component__main__col-md-5","col-md-4":"dapp-core-component__main__col-md-4",colMd4:"dapp-core-component__main__col-md-4","col-md-3":"dapp-core-component__main__col-md-3",colMd3:"dapp-core-component__main__col-md-3","col-md-2":"dapp-core-component__main__col-md-2",colMd2:"dapp-core-component__main__col-md-2","col-md-1":"dapp-core-component__main__col-md-1",colMd1:"dapp-core-component__main__col-md-1","col-sm":"dapp-core-component__main__col-sm",colSm:"dapp-core-component__main__col-sm","col-sm-auto":"dapp-core-component__main__col-sm-auto",colSmAuto:"dapp-core-component__main__col-sm-auto","col-sm-12":"dapp-core-component__main__col-sm-12",colSm12:"dapp-core-component__main__col-sm-12","col-sm-11":"dapp-core-component__main__col-sm-11",colSm11:"dapp-core-component__main__col-sm-11","col-sm-10":"dapp-core-component__main__col-sm-10",colSm10:"dapp-core-component__main__col-sm-10","col-sm-9":"dapp-core-component__main__col-sm-9",colSm9:"dapp-core-component__main__col-sm-9","col-sm-8":"dapp-core-component__main__col-sm-8",colSm8:"dapp-core-component__main__col-sm-8","col-sm-7":"dapp-core-component__main__col-sm-7",colSm7:"dapp-core-component__main__col-sm-7","col-sm-6":"dapp-core-component__main__col-sm-6",colSm6:"dapp-core-component__main__col-sm-6","col-sm-5":"dapp-core-component__main__col-sm-5",colSm5:"dapp-core-component__main__col-sm-5","col-sm-4":"dapp-core-component__main__col-sm-4",colSm4:"dapp-core-component__main__col-sm-4","col-sm-3":"dapp-core-component__main__col-sm-3",colSm3:"dapp-core-component__main__col-sm-3","col-sm-2":"dapp-core-component__main__col-sm-2",colSm2:"dapp-core-component__main__col-sm-2","col-sm-1":"dapp-core-component__main__col-sm-1",colSm1:"dapp-core-component__main__col-sm-1","col-auto":"dapp-core-component__main__col-auto",colAuto:"dapp-core-component__main__col-auto","col-12":"dapp-core-component__main__col-12",col12:"dapp-core-component__main__col-12","col-11":"dapp-core-component__main__col-11",col11:"dapp-core-component__main__col-11","col-10":"dapp-core-component__main__col-10",col10:"dapp-core-component__main__col-10","col-9":"dapp-core-component__main__col-9",col9:"dapp-core-component__main__col-9","col-8":"dapp-core-component__main__col-8",col8:"dapp-core-component__main__col-8","col-7":"dapp-core-component__main__col-7",col7:"dapp-core-component__main__col-7","col-6":"dapp-core-component__main__col-6",col6:"dapp-core-component__main__col-6","col-5":"dapp-core-component__main__col-5",col5:"dapp-core-component__main__col-5","col-4":"dapp-core-component__main__col-4",col4:"dapp-core-component__main__col-4","col-3":"dapp-core-component__main__col-3",col3:"dapp-core-component__main__col-3","col-2":"dapp-core-component__main__col-2",col2:"dapp-core-component__main__col-2","col-1":"dapp-core-component__main__col-1",col1:"dapp-core-component__main__col-1","flex-row":"dapp-core-component__main__flex-row",flexRow:"dapp-core-component__main__flex-row","flex-column":"dapp-core-component__main__flex-column",flexColumn:"dapp-core-component__main__flex-column","flex-row-reverse":"dapp-core-component__main__flex-row-reverse",flexRowReverse:"dapp-core-component__main__flex-row-reverse","flex-column-reverse":"dapp-core-component__main__flex-column-reverse",flexColumnReverse:"dapp-core-component__main__flex-column-reverse","flex-wrap":"dapp-core-component__main__flex-wrap",flexWrap:"dapp-core-component__main__flex-wrap","flex-nowrap":"dapp-core-component__main__flex-nowrap",flexNowrap:"dapp-core-component__main__flex-nowrap","flex-wrap-reverse":"dapp-core-component__main__flex-wrap-reverse",flexWrapReverse:"dapp-core-component__main__flex-wrap-reverse","flex-fill":"dapp-core-component__main__flex-fill",flexFill:"dapp-core-component__main__flex-fill","flex-grow-0":"dapp-core-component__main__flex-grow-0",flexGrow0:"dapp-core-component__main__flex-grow-0","flex-grow-1":"dapp-core-component__main__flex-grow-1",flexGrow1:"dapp-core-component__main__flex-grow-1","flex-shrink-0":"dapp-core-component__main__flex-shrink-0",flexShrink0:"dapp-core-component__main__flex-shrink-0","flex-shrink-1":"dapp-core-component__main__flex-shrink-1",flexShrink1:"dapp-core-component__main__flex-shrink-1","justify-content-start":"dapp-core-component__main__justify-content-start",justifyContentStart:"dapp-core-component__main__justify-content-start","justify-content-end":"dapp-core-component__main__justify-content-end",justifyContentEnd:"dapp-core-component__main__justify-content-end","justify-content-center":"dapp-core-component__main__justify-content-center",justifyContentCenter:"dapp-core-component__main__justify-content-center","justify-content-between":"dapp-core-component__main__justify-content-between",justifyContentBetween:"dapp-core-component__main__justify-content-between","justify-content-around":"dapp-core-component__main__justify-content-around",justifyContentAround:"dapp-core-component__main__justify-content-around","align-items-start":"dapp-core-component__main__align-items-start",alignItemsStart:"dapp-core-component__main__align-items-start","align-items-end":"dapp-core-component__main__align-items-end",alignItemsEnd:"dapp-core-component__main__align-items-end","align-items-center":"dapp-core-component__main__align-items-center",alignItemsCenter:"dapp-core-component__main__align-items-center","align-items-baseline":"dapp-core-component__main__align-items-baseline",alignItemsBaseline:"dapp-core-component__main__align-items-baseline","align-items-stretch":"dapp-core-component__main__align-items-stretch",alignItemsStretch:"dapp-core-component__main__align-items-stretch","align-content-start":"dapp-core-component__main__align-content-start",alignContentStart:"dapp-core-component__main__align-content-start","align-content-end":"dapp-core-component__main__align-content-end",alignContentEnd:"dapp-core-component__main__align-content-end","align-content-center":"dapp-core-component__main__align-content-center",alignContentCenter:"dapp-core-component__main__align-content-center","align-content-between":"dapp-core-component__main__align-content-between",alignContentBetween:"dapp-core-component__main__align-content-between","align-content-around":"dapp-core-component__main__align-content-around",alignContentAround:"dapp-core-component__main__align-content-around","align-content-stretch":"dapp-core-component__main__align-content-stretch",alignContentStretch:"dapp-core-component__main__align-content-stretch","align-self-auto":"dapp-core-component__main__align-self-auto",alignSelfAuto:"dapp-core-component__main__align-self-auto","align-self-start":"dapp-core-component__main__align-self-start",alignSelfStart:"dapp-core-component__main__align-self-start","align-self-end":"dapp-core-component__main__align-self-end",alignSelfEnd:"dapp-core-component__main__align-self-end","align-self-center":"dapp-core-component__main__align-self-center",alignSelfCenter:"dapp-core-component__main__align-self-center","align-self-baseline":"dapp-core-component__main__align-self-baseline",alignSelfBaseline:"dapp-core-component__main__align-self-baseline","align-self-stretch":"dapp-core-component__main__align-self-stretch",alignSelfStretch:"dapp-core-component__main__align-self-stretch","flex-sm-row":"dapp-core-component__main__flex-sm-row",flexSmRow:"dapp-core-component__main__flex-sm-row","flex-lg-row":"dapp-core-component__main__flex-lg-row",flexLgRow:"dapp-core-component__main__flex-lg-row","flex-lg-wrap":"dapp-core-component__main__flex-lg-wrap",flexLgWrap:"dapp-core-component__main__flex-lg-wrap","form-check":"dapp-core-component__main__form-check",formCheck:"dapp-core-component__main__form-check","form-check-label":"dapp-core-component__main__form-check-label",formCheckLabel:"dapp-core-component__main__form-check-label","form-check-input":"dapp-core-component__main__form-check-input",formCheckInput:"dapp-core-component__main__form-check-input","form-group":"dapp-core-component__main__form-group",formGroup:"dapp-core-component__main__form-group","form-control":"dapp-core-component__main__form-control",formControl:"dapp-core-component__main__form-control","form-label":"dapp-core-component__main__form-label",formLabel:"dapp-core-component__main__form-label","form-text":"dapp-core-component__main__form-text",formText:"dapp-core-component__main__form-text",border:"dapp-core-component__main__border","border-top":"dapp-core-component__main__border-top",borderTop:"dapp-core-component__main__border-top","border-right":"dapp-core-component__main__border-right",borderRight:"dapp-core-component__main__border-right","border-bottom":"dapp-core-component__main__border-bottom",borderBottom:"dapp-core-component__main__border-bottom","border-left":"dapp-core-component__main__border-left",borderLeft:"dapp-core-component__main__border-left","border-0":"dapp-core-component__main__border-0",border0:"dapp-core-component__main__border-0","border-top-0":"dapp-core-component__main__border-top-0",borderTop0:"dapp-core-component__main__border-top-0","border-right-0":"dapp-core-component__main__border-right-0",borderRight0:"dapp-core-component__main__border-right-0","border-bottom-0":"dapp-core-component__main__border-bottom-0",borderBottom0:"dapp-core-component__main__border-bottom-0","border-left-0":"dapp-core-component__main__border-left-0",borderLeft0:"dapp-core-component__main__border-left-0","rounded-sm":"dapp-core-component__main__rounded-sm",roundedSm:"dapp-core-component__main__rounded-sm",rounded:"dapp-core-component__main__rounded","rounded-top":"dapp-core-component__main__rounded-top",roundedTop:"dapp-core-component__main__rounded-top","rounded-right":"dapp-core-component__main__rounded-right",roundedRight:"dapp-core-component__main__rounded-right","rounded-bottom":"dapp-core-component__main__rounded-bottom",roundedBottom:"dapp-core-component__main__rounded-bottom","rounded-left":"dapp-core-component__main__rounded-left",roundedLeft:"dapp-core-component__main__rounded-left","rounded-lg":"dapp-core-component__main__rounded-lg",roundedLg:"dapp-core-component__main__rounded-lg","rounded-circle":"dapp-core-component__main__rounded-circle",roundedCircle:"dapp-core-component__main__rounded-circle","rounded-pill":"dapp-core-component__main__rounded-pill",roundedPill:"dapp-core-component__main__rounded-pill","rounded-0":"dapp-core-component__main__rounded-0",rounded0:"dapp-core-component__main__rounded-0",h1:"dapp-core-component__main__h1",h2:"dapp-core-component__main__h2",h3:"dapp-core-component__main__h3",h4:"dapp-core-component__main__h4",h5:"dapp-core-component__main__h5",lead:"dapp-core-component__main__lead","bg-primary":"dapp-core-component__main__bg-primary",bgPrimary:"dapp-core-component__main__bg-primary","bg-warning":"dapp-core-component__main__bg-warning",bgWarning:"dapp-core-component__main__bg-warning",warning:"dapp-core-component__main__warning","bg-danger":"dapp-core-component__main__bg-danger",bgDanger:"dapp-core-component__main__bg-danger",danger:"dapp-core-component__main__danger","bg-success":"dapp-core-component__main__bg-success",bgSuccess:"dapp-core-component__main__bg-success",success:"dapp-core-component__main__success","position-relative":"dapp-core-component__main__position-relative",positionRelative:"dapp-core-component__main__position-relative","position-absolute":"dapp-core-component__main__position-absolute",positionAbsolute:"dapp-core-component__main__position-absolute","w-25":"dapp-core-component__main__w-25",w25:"dapp-core-component__main__w-25","w-50":"dapp-core-component__main__w-50",w50:"dapp-core-component__main__w-50","w-75":"dapp-core-component__main__w-75",w75:"dapp-core-component__main__w-75","w-100":"dapp-core-component__main__w-100",w100:"dapp-core-component__main__w-100","w-auto":"dapp-core-component__main__w-auto",wAuto:"dapp-core-component__main__w-auto","h-25":"dapp-core-component__main__h-25",h25:"dapp-core-component__main__h-25","h-50":"dapp-core-component__main__h-50",h50:"dapp-core-component__main__h-50","h-75":"dapp-core-component__main__h-75",h75:"dapp-core-component__main__h-75","h-100":"dapp-core-component__main__h-100",h100:"dapp-core-component__main__h-100","h-auto":"dapp-core-component__main__h-auto",hAuto:"dapp-core-component__main__h-auto","mw-100":"dapp-core-component__main__mw-100",mw100:"dapp-core-component__main__mw-100","mh-100":"dapp-core-component__main__mh-100",mh100:"dapp-core-component__main__mh-100","min-vw-100":"dapp-core-component__main__min-vw-100",minVw100:"dapp-core-component__main__min-vw-100","min-vh-100":"dapp-core-component__main__min-vh-100",minVh100:"dapp-core-component__main__min-vh-100","vw-100":"dapp-core-component__main__vw-100",vw100:"dapp-core-component__main__vw-100","vh-100":"dapp-core-component__main__vh-100",vh100:"dapp-core-component__main__vh-100","locked-icon":"dapp-core-component__main__locked-icon",lockedIcon:"dapp-core-component__main__locked-icon","shadow-sm":"dapp-core-component__main__shadow-sm",shadowSm:"dapp-core-component__main__shadow-sm",shadow:"dapp-core-component__main__shadow","shadow-lg":"dapp-core-component__main__shadow-lg",shadowLg:"dapp-core-component__main__shadow-lg","shadow-none":"dapp-core-component__main__shadow-none",shadowNone:"dapp-core-component__main__shadow-none","d-none":"dapp-core-component__main__d-none",dNone:"dapp-core-component__main__d-none","d-inline":"dapp-core-component__main__d-inline",dInline:"dapp-core-component__main__d-inline","d-inline-block":"dapp-core-component__main__d-inline-block",dInlineBlock:"dapp-core-component__main__d-inline-block","d-block":"dapp-core-component__main__d-block",dBlock:"dapp-core-component__main__d-block","d-table":"dapp-core-component__main__d-table",dTable:"dapp-core-component__main__d-table","d-table-row":"dapp-core-component__main__d-table-row",dTableRow:"dapp-core-component__main__d-table-row","d-table-cell":"dapp-core-component__main__d-table-cell",dTableCell:"dapp-core-component__main__d-table-cell","d-flex":"dapp-core-component__main__d-flex",dFlex:"dapp-core-component__main__d-flex","d-inline-flex":"dapp-core-component__main__d-inline-flex",dInlineFlex:"dapp-core-component__main__d-inline-flex","d-sm-flex":"dapp-core-component__main__d-sm-flex",dSmFlex:"dapp-core-component__main__d-sm-flex","table-sm":"dapp-core-component__main__table-sm",tableSm:"dapp-core-component__main__table-sm","table-bordered":"dapp-core-component__main__table-bordered",tableBordered:"dapp-core-component__main__table-bordered","table-borderless":"dapp-core-component__main__table-borderless",tableBorderless:"dapp-core-component__main__table-borderless","table-striped":"dapp-core-component__main__table-striped",tableStriped:"dapp-core-component__main__table-striped","thead-dark":"dapp-core-component__main__thead-dark",theadDark:"dapp-core-component__main__thead-dark","thead-light":"dapp-core-component__main__thead-light",theadLight:"dapp-core-component__main__thead-light","table-dark":"dapp-core-component__main__table-dark",tableDark:"dapp-core-component__main__table-dark","table-hover":"dapp-core-component__main__table-hover",tableHover:"dapp-core-component__main__table-hover","table-responsive":"dapp-core-component__main__table-responsive",tableResponsive:"dapp-core-component__main__table-responsive","transaction-hash":"dapp-core-component__main__transaction-hash",transactionHash:"dapp-core-component__main__transaction-hash","font-weight-light":"dapp-core-component__main__font-weight-light",fontWeightLight:"dapp-core-component__main__font-weight-light","font-weight-lighter":"dapp-core-component__main__font-weight-lighter",fontWeightLighter:"dapp-core-component__main__font-weight-lighter","font-weight-normal":"dapp-core-component__main__font-weight-normal",fontWeightNormal:"dapp-core-component__main__font-weight-normal","font-weight-medium":"dapp-core-component__main__font-weight-medium",fontWeightMedium:"dapp-core-component__main__font-weight-medium","font-weight-bold":"dapp-core-component__main__font-weight-bold",fontWeightBold:"dapp-core-component__main__font-weight-bold","font-weight-bolder":"dapp-core-component__main__font-weight-bolder",fontWeightBolder:"dapp-core-component__main__font-weight-bolder","font-italic":"dapp-core-component__main__font-italic",fontItalic:"dapp-core-component__main__font-italic","icon-state":"dapp-core-component__main__icon-state",iconState:"dapp-core-component__main__icon-state",half:"dapp-core-component__main__half","side-icon":"dapp-core-component__main__side-icon",sideIcon:"dapp-core-component__main__side-icon","toast-messages":"dapp-core-component__main__toast-messages",toastMessages:"dapp-core-component__main__toast-messages",toast:"dapp-core-component__main__toast",clickable:"dapp-core-component__main__clickable","toast-visible":"dapp-core-component__main__toast-visible",toastVisible:"dapp-core-component__main__toast-visible",progress:"dapp-core-component__main__progress","progress-bar":"dapp-core-component__main__progress-bar",progressBar:"dapp-core-component__main__progress-bar","content-height":"dapp-core-component__main__content-height",contentHeight:"dapp-core-component__main__content-height","tx-description":"dapp-core-component__main__tx-description",txDescription:"dapp-core-component__main__tx-description",close:"dapp-core-component__main__close","data-field":"dapp-core-component__main__data-field",dataField:"dapp-core-component__main__data-field",textarea:"dapp-core-component__main__textarea",highlighted:"dapp-core-component__main__highlighted","sc-call":"dapp-core-component__main__sc-call",scCall:"dapp-core-component__main__sc-call","small-font":"dapp-core-component__main__small-font",smallFont:"dapp-core-component__main__small-font","progress-steps":"dapp-core-component__main__progress-steps",progressSteps:"dapp-core-component__main__progress-steps",steps:"dapp-core-component__main__steps",dot:"dapp-core-component__main__dot","side-link":"dapp-core-component__main__side-link",sideLink:"dapp-core-component__main__side-link"};t();t();t();var ii=require("react"),Je=require("react-redux");t();var vt=require("@reduxjs/toolkit"),ri=require("react-redux/es/utils/Subscription");P();t();var Ra=x(require("lodash.throttle"));N();P();qn();Ve();$e();var ns=[$o],ts=(0,Ra.default)(()=>{uo(go())},5e3),Fa=e=>o=>n=>{var _;if(ns.includes(n.type))return o(n);let r=e.getState(),p=Le.local.getItem(ve.loginExpiresAt);if(!((_=r==null?void 0:r.loginInfo)==null?void 0:_.isLoggedIn))return o(n);if(p==null)return uo(go());let m=Date.now();return p-m<0?setTimeout(()=>A(void 0,null,function*(){console.log("session expired");try{e.dispatch(ia())}catch(u){console.error("error logging out",u)}}),1e3):(ts(),o(n))};t();t();function en(){return typeof sessionStorage!="undefined"}var oi=en()?(za(),_o(ja)).default:(Ka(),_o(qa)).default,ni=en()?(Ya(),_o(Xa)).default:[],ti=en()?(ei(),_o(Qa)).default:e=>e;on();var E=(0,vt.configureStore)({reducer:oi,middleware:e=>e({serializableCheck:{ignoredActions:[...ni,mt.type,st.type],ignoredPaths:["payload.nonce","account.account.nonce","providers.accountProvider"]}}).concat(Fa)}),ai=(0,ri.createSubscription)(E),Z_=ti(E),J_=(0,vt.configureStore)({reducer:Ze});var hs={store:E,subscription:ai},kt=(0,ii.createContext)(hs),ng=(0,Je.createStoreHook)(kt),z=(0,Je.createDispatchHook)(kt),C=(0,Je.createSelectorHook)(kt);t();t();P();t();var ci=x(require("lodash.isequal")),nn=require("reselect"),S=(0,nn.createSelectorCreator)(nn.defaultMemoize,ci.default);var ye=e=>e.account,Qe=S(ye,e=>e.address),bo=S(ye,Qe,(e,o)=>o in e.accounts?e.accounts[o]:jo),pi=S(ye,bo,(e,o)=>{let p=e,{accounts:n}=p,r=Ln(p,["accounts"]);return I(b({},r),{account:o})}),Ts=S(bo,e=>e.balance),xs=S(bo,e=>{var o;return((o=e==null?void 0:e.nonce)==null?void 0:o.valueOf())||0}),cg=S(ye,e=>e.shard),ys=S(ye,e=>e.ledgerAccount),pg=S(ye,e=>e.walletConnectAccount),sg=S(ye,e=>e.isAccountLoading),mg=S(ye,e=>e.accountLoadingError),si=S(ye,e=>e.websocketEvent);t();var de=e=>e.networkConfig,tn=S(de,e=>e.chainID),bs=S(de,e=>e.network.walletConnectBridgeAddress),Ss=S(de,e=>e.network.walletConnectV2RelayAddress),ws=S(de,e=>e.network.walletConnectV2ProjectId),mi=S(de,e=>e.network.walletConnectDeepLink),Me=S(de,e=>e.network),di=S(Me,e=>e.apiAddress),li=S(Me,e=>e.explorerAddress),rn=S(Me,e=>e.egldLabel);t();R();var Re=e=>e.loginInfo,hg=S(Re,e=>e.loginMethod),an=S(Re,Qe,(e,o)=>e.loginMethod!=""&&Boolean(o)),Tg=S(Re,e=>e.walletConnectLogin),xg=S(Re,e=>e.ledgerLogin),vs=S(Re,e=>e.walletLogin),yg=S(Re,e=>e.isLoginSessionInvalid);t();t();var Y=require("@elrondnetwork/erdjs");N();t();function It(e){try{return a.Buffer.from(e,"base64").toString()===atob(e)}catch(o){return!1}}function So(e){var p,s,m;let{data:o}=e,n=It(o!=null?o:"")?Y.TransactionPayload.fromEncoded(o):new Y.TransactionPayload(o),r=new Y.Transaction(b({value:e.value.valueOf(),data:n,nonce:e.nonce.valueOf(),receiver:new Y.Address(e.receiver),sender:new Y.Address(e.sender),gasLimit:(p=e.gasLimit.valueOf())!=null?p:fo,gasPrice:(s=e.gasPrice.valueOf())!=null?s:Go,chainID:e.chainID.valueOf(),version:new Y.TransactionVersion((m=e.version)!=null?m:Jr)},e.options?{options:new Y.TransactionOptions(e.options)}:{}));return r.applySignature({hex:()=>e.signature||""},new Y.Address(e.sender)),r}t();Ye();t();var Et=x(require("bignumber.js"));Q();t();t();function At(e){for(let o=0;o<e.length;o++)if(e.charCodeAt(o)>127)return!1;return!0}function Fe(e){let o=e;try{let n=a.Buffer.from(e,"hex").toString();At(n)&&n.length>1&&(o=n)}catch(n){}return o}t();var _i=(e,o)=>{let n=[],r=e.indexOf(o,0);for(;r>=0;)n.push(r),r=e.indexOf(o,r+1);return n};function ui(e){let o=[],n=0;try{if((e==null?void 0:e.startsWith("MultiESDTNFTTransfer"))&&(e==null?void 0:e.includes("@"))){let[,r,p,...s]=e==null?void 0:e.split("@");if(r){let m=new Et.default(p,16).toNumber(),g=0;for(let f=0;f<m;f++){let y={type:"nftTransaction",data:"",receiver:r};for(let T=0;T<3;T++){switch(T){case 0:y.token=Fe(s[g]),y.data=s[g];break;case 1:{let v=s[g]&&s[g].length?s[g]:"";v?y.nonce=v:y.type="esdtTransaction",y.data=`${y.data}@${s[g]}`;break}case 2:y.amount=new Et.default(s[g],16).toString(10),y.data=`${y.data}@${s[g]}`;break;default:break}n=g+1,g++}o[f]=y}let _=o.length!==m,u=o.some(f=>_i(f.data,"@").length!==2),h=o.some(f=>f.data.startsWith("@"));if(_||u||h)return[];if(s[n]){let f=s[n];for(let y=n+1;y<s.length;y++)f+="@"+s[y];o[m]={type:"scCall",data:f,receiver:r}}}}}catch(r){return console.error("failed parsing tx",r),o}return o}t();var Lt=require("@elrondnetwork/erdjs"),Dt=x(require("bignumber.js"));R();t();var fi=require("@elrondnetwork/erdjs");function ks(e){try{let o=new fi.Address(e);return Boolean(o.bech32())}catch(o){return!1}}function wo(e){return(e==null?void 0:e.startsWith("erd"))&&e.length===62&&/^\w+$/.test(e)&&ks(e)}var Ct={tokenId:"",amount:""},Is=e=>e.split("@").map((s,m)=>[2,3].includes(m)?s:Fe(s));function Nt(e){if(!e)return Ct;let o=e.startsWith("ESDTTransfer"),n=e.startsWith("ESDTNFTTransfer")&&e.includes("@");if(o){let[,r,p]=e.split("@");try{let s=a.Buffer.from(r,"hex").toString("ascii");if(!s)return Ct;let m=new Dt.default("0x"+p.replace("0x","")).toString(10);return{tokenId:s,amount:m}}catch(s){console.error("Error getting token from transaction data",s)}}if(n)try{let[,r,p,s,m]=Is(e);if([r,p,s,m].every(g=>Boolean(g))&&wo(new Lt.Address(m).bech32()))return{tokenId:`${r}-${p}`,amount:new Dt.default(s,16).toString(10),collection:r,nonce:p,receiver:new Lt.Address(m).bech32()}}catch(r){}return Ct}t();function cn({tokenId:e,erdLabel:o}){return Boolean(e&&e!==o)}t();t();var Es=require("@elrondnetwork/erdjs/out");t();N();t();var hi=require("@elrondnetwork/erdjs");Q();t();t();t();t();R();function pn(e){let n=Object.getPrototypeOf(e).toPlainObject!=null?e:So(e);return I(b({},n.toPlainObject()),{hash:n.getHash().hex(),status:"pending"})}t();t();t();t();t();t();t();var sn="accounts";var Pt="tokens";var Mt="collections";t();t();t();t();V();t();t();V();t();t();t();var Ns=x(require("bignumber.js"));N();t();var yi=require("@elrondnetwork/erdjs"),ke=x(require("bignumber.js"));N();t();var xi=x(require("bignumber.js")),Rt=(e,o=!0)=>{let n=String(e);if(!n.match(/^[-]?\d+$/))return!1;let r=new xi.default(n),p=o?0:-1;return r.toString(10)===n&&r.comparedTo(0)>=p};t();function be(e){return{if:function(o){return o?{then:n=>n instanceof Function?be(n(e)):be(n)}:{then:()=>be(e)}},then:o=>o instanceof Function?be(o(e)):be(o),valueOf:function(){return e}}}ke.default.config({ROUNDING_MODE:ke.default.ROUND_FLOOR});function Be({input:e,decimals:o=me,digits:n=Ho,showLastNonZeroDecimal:r=!0,showIsLessThanDecimalsLabel:p=!1,addCommas:s=!1}){if(!Rt(e,!1))throw new Error("Invalid input");let m=new ke.default(e).isNegative(),g=e;return m&&(g=e.substring(1)),be(g).then(()=>yi.TokenPayment.fungibleFromBigInteger("",g,o).toRationalNumber()).then(_=>{let u=new ke.default(_);if(u.isZero())return De;let h=u.toString(10),[f,y]=h.split("."),T=new ke.default(y||0),v=be(0).if(Boolean(y&&r)).then(()=>Math.max(y.length,n)).if(T.isZero()&&!r).then(0).if(Boolean(y&&!r)).then(()=>Math.min(y.length,n)).valueOf(),M=y&&n>=1&&n<=y.length&&T.isGreaterThan(0)&&new ke.default(y.substring(0,n)).isZero(),te=u.toFormat(v);return be(h).if(s).then(te).if(Boolean(M)).then(D=>{let Z=new ke.default(f).isZero(),[W,ae]=D.split("."),ge=new Array(n-1).fill(0),po=[...ge,0].join(""),re=[...ge,1].join("");return Z?p?`<${W}.${re}`:r?`${W}.${ae}`:W:`${W}.${po}`}).if(Boolean(!M&&y)).then(D=>{let[Z]=D.split("."),W=y.substring(0,v);if(r){let ae=n-W.length;if(ae>0){let ge=Array(ae).fill(0).join("");return W=`${W}${ge}`,`${Z}.${W}`}return D}return W?`${Z}.${W}`:Z}).valueOf()}).if(m).then(_=>`-${_}`).valueOf()}t();t();t();N();t();t();t();t();N();t();N();t();var Ps=require("@elrondnetwork/erdjs");N();t();var dn=require("@elrondnetwork/erdjs"),Fs=require("@elrondnetwork/erdjs"),Bs=require("@elrondnetwork/erdjs-network-providers");N();t();t();var Ms=x(require("bignumber.js"));N();t();N();t();var Ft={isEsdt:!1,isNft:!1,isEgld:!1};function Bt(e){let o=e==null?void 0:e.split("-").length;return o===2?I(b({},Ft),{isEsdt:!0}):o===3?I(b({},Ft),{isNft:!0}):I(b({},Ft),{isEgld:!0})}t();t();t();t();N();t();V();var Us=["reDelegateRewards","claimRewards","unBond"],Ws=["wrapEgld","unwrapEgld"],Gs=["unStake"],Hs=["unDelegate"];t();t();t();N();t();var js=x(require("bignumber.js"));t();t();t();V();t();var zs=x(require("bignumber.js"));t();t();t();V();t();V();t();var Ks=x(require("bignumber.js"));N();V();t();V();t();V();t();var Xs=x(require("bignumber.js"));t();R();t();R();t();t();var Ys=require("react");V();var tT=["create","localMint","ESDTLocalMint","addQuantity","burn","localBurn","ESDTLocalBurn","wipe","writeLog","signalError"];t();var Zs=x(require("anchorme"));t();var wi=require("react");Q();t();t();V();t();var Js=require("@elrondnetwork/erdjs/out"),Qs=x(require("bignumber.js"));V();t();V();t();t();R();var MT=[{label:"Raw",value:"raw"},{label:"Text",value:"text"},{label:"Decimal",value:"decimal"},{label:"Smart",value:"smart"}];t();var vi=require("react");Q();t();t();t();t();t();N();var _n=e=>e.transactions,eo=S(_n,e=>e.signedTransactions),om=S(_n,e=>e.signTransactionsError),nm=S(_n,e=>e.signTransactionsCancelMessage),gn=e=>o=>Object.entries(o).reduce((n,[r,p])=>(e(p.status)&&(n[r]=p),n),{}),ki=S(eo,gn(qe)),Ii=S(eo,gn(Ke)),Ai=S(eo,gn(Xe)),tm=S(eo,gn(lt)),vo=S(_n,e=>{var o;return(e==null?void 0:e.transactionsToSign)==null?null:I(b({},e.transactionsToSign),{transactions:((o=e==null?void 0:e.transactionsToSign)==null?void 0:o.transactions.map(n=>So(n)))||[]})}),rm=S(eo,(e,o)=>o,(e,o)=>o!=null?(e==null?void 0:e[o])||{}:{});t();P();var Ei={errorMessage:ut,successMessage:ft,processingMessage:ht},am=e=>e.transactionsInfo,im=S(am,(e,o)=>o,(e,o)=>o!=null&&(e==null?void 0:e[Number(o)])||Ei);t();var Ci=e=>e.modals,Dx=S(Ci,e=>e.txSubmittedModal),cm=S(Ci,e=>e.notificationModal);t();var Ut=e=>e.toasts,Mx=S(Ut,e=>e.customToasts),Rx=S(Ut,e=>e.transactionToasts),Fx=S(Ut,e=>e.failTransactionToast);var ko=()=>C(de);t();var Li=require("react");t();P();t();t();t();var sm=require("react"),mm=require("@elrondnetwork/erdjs-web-wallet-provider"),dm=x(require("qs"));N();P();R();t();var $t=require("react"),fm=require("@elrondnetwork/erdjs-extension-provider");N();t();t();t();t();var Di=require("@elrondnetwork/erdjs-extension-provider"),Ni=require("@elrondnetwork/erdjs-hw-provider"),fn=require("@elrondnetwork/erdjs-wallet-connect-provider"),Pi=require("@elrondnetwork/erdjs-web-wallet-provider");N();R();var oo=e=>{switch(e==null?void 0:e.constructor){case Pi.WalletProvider:return"wallet";case fn.WalletConnectProvider:return"walletconnect";case fn.WalletConnectV2Provider:return"walletconnectv2";case Ni.HWProvider:return"ledger";case Di.ExtensionProvider:return"extension";case un:return"";default:return"extra"}};var Se=e=>`Unable to perform ${e}, Provider not initialized`,un=class{init(){return Promise.resolve(!1)}login(o){throw new Error(Se(`login with options: ${o}`))}logout(o){throw new Error(Se(`logout with options: ${o}`))}getAddress(){throw new Error(Se("getAddress"))}isInitialized(){return!1}isConnected(){return Promise.resolve(!1)}sendTransaction(o,n){throw new Error(Se(`sendTransaction with transactions: ${o} options: ${n}`))}signTransaction(o,n){throw new Error(Se(`signTransaction with transactions: ${o} options: ${n}`))}signTransactions(o,n){throw new Error(Se(`signTransactions with transactions: ${o} options: ${n}`))}signMessage(o,n){throw new Error(Se(`signTransactions with ${o} and options ${n}`))}sendCustomMessage({method:o,params:n}){throw new Error(Se(`sendCustomMessage with method: ${o} params: ${n}`))}sendCustomRequest(o){throw new Error(Se(`sendSessionEvent with options: ${o}`))}ping(){return Promise.resolve(!1)}},Wt=new un;var lm=Wt;function le(){return lm||Wt}var hn=()=>{let e=le(),o=oo(e);return{provider:e,providerType:o}};P();R();t();t();X();Q();t();t();t();R();t();t();function Mi(){let e=le();return oo(e)}function no(e){return Mi()===e}t();t();var _m=x(require("axios"));t();t();t();P();t();t();t();P();t();var Bi=require("@elrondnetwork/erdjs");t();X();Q();To();Ve();$e();t();t();t();t();P();t();t();t();t();t();function Tn(){return rn(E.getState())}Ve();To();$n();t();t();t();t();t();var gm=x(require("axios"));t();t();var um=x(require("platform"));t();function Ht(e){let o=vo(E.getState()),n=Object.keys(e).length===(o==null?void 0:o.transactions.length);return e&&(e&&n)}t();t();var Oi=x(require("axios")),Ui=x(require("swr"));var hm=e=>Oi.default.get(e).then(o=>o.data);function Wi({tokenId:e}){var y;let{network:o}=ko(),{isEsdt:n,isNft:r}=Bt(e),p=n?Pt:Mt,s=e;if(r){let[T,v]=e.split("-");s=`${T}-${v}`}let{data:m,error:g}=(0,Ui.default)(Boolean(s)?`${o.apiAddress}/${p}/${s}`:null,hm);if(!s)return{tokenDecimals:Number(o.decimals),tokenLabel:"",tokenAvatar:""};let _=m?m==null?void 0:m.decimals:Number(o.decimals),u=m?m==null?void 0:m.name:"",h=m?`${(y=m==null?void 0:m.assets)==null?void 0:y.svgUrl}`:"",f=m==null?void 0:m.assets;return{tokenDecimals:_,tokenLabel:u,tokenAvatar:h,assets:f,error:g}}t();t();var Gi=x(require("axios"));function Hi(e){return A(this,null,function*(){let{network:{apiAddress:o,apiTimeout:n}}=de(E.getState()),{data:r}=yield Gi.default.get(`/${sn}/${e}`,{baseURL:o,timeout:Number(n)});return r})}t();var $i=()=>C(pi);t();var Oe=require("react");t();var Ao=require("react");var Vi={tokenId:"",amount:"",type:"",multiTxData:"",receiver:""};function ji({transactions:e}){let[o,n]=(0,Ao.useState)({}),[r,p]=(0,Ao.useState)([]);function s(_,u){n(h=>I(b({},h),{[_]:u}))}function m(_,u){return o==null?Vi:_ in o?o[_]:u!=null&&String(u)in o?o[u]:Vi}function g(){if(e&&e.length>0){let _=[];e.forEach((u,h)=>{let f=u.getData().toString(),y=ui(f);if(y.length>0)y.forEach((T,v)=>{let M={transaction:u,multiTxData:T.data,transactionIndex:v};s(T.data,{tokenId:T.token?T.token:"",amount:T.amount?T.amount:"",type:T.type,nonce:T.nonce?T.nonce:"",multiTxData:T.data,receiver:T.receiver}),_.push(M)});else{let{tokenId:T,amount:v}=Nt(u.getData().toString());T&&s(u.getData().toString(),{tokenId:T,amount:v,receiver:u.getReceiver().bech32()}),_.push({transaction:u,transactionIndex:h})}}),p(_)}}return(0,Ao.useEffect)(()=>{g()},[e==null?void 0:e.length]),{parsedTransactionsByDataField:o,getTxInfoByDataField:m,allTransactions:r}}rt();var xn={};function zi({isLedger:e=!1,transactionsToSign:o,egldLabel:n,address:r,onCancel:p,onSignTransaction:s,onTransactionsSignError:m,onTransactionsSignSuccess:g,onGetScamAddressData:_}){var Do,U,No;let[u,h]=(0,Oe.useState)(0),[f,y]=(0,Oe.useState)(),[T,v]=(0,Oe.useState)(null),[M,te]=(0,Oe.useState)(!1),{getTxInfoByDataField:we,allTransactions:D}=ji({transactions:o}),Z=u===D.length-1;(0,Oe.useEffect)(()=>{W()},[u,D]);function W(){return A(this,null,function*(){var sr;let O=D[u];if(O==null)return;let{transaction:H,multiTxData:ue}=O,En=H.getData().toString(),cr=we(H.getData().toString(),ue),{tokenId:pr}=cr,mo=H.getReceiver().toString(),Dc=r!==mo,Nc=mo in xn;if(Dc&&!Nc&&_!=null){let Po=yield _(mo);xn=b(b({},xn),Po!=null&&Po.scamInfo?{[mo]:Po.scamInfo}:{})}let Pc=Boolean(pr&&cn({tokenId:pr,erdLabel:n}));v({transaction:H,receiverScamInfo:((sr=xn[mo])==null?void 0:sr.info)||null,transactionTokenInfo:cr,isTokenTransaction:Pc,dataField:En})})}function ae(){h(0),y(void 0),te(!1)}function ge(){return A(this,null,function*(){try{if(T==null)return;te(e);let O=yield s(T.transaction),H={[u]:O},ue=f?b(b({},f),H):H;y(ue),Z?ue&&(g(Object.values(ue)),ae()):(h(En=>En+1),te(!1))}catch(O){console.error(O,"sign error");let{message:H}=O,ue=e?tt(O).errorMessage:null;ae(),m(ue!=null?ue:H)}})}function po(){try{if(T==null)return;T.transaction.getSignature().hex()?Z||h(H=>H+1):ge()}catch(O){ge()}}let re=u===0;function kn(){re?p==null||p():h(O=>O-1)}let Lo=Boolean(((Do=T==null?void 0:T.transactionTokenInfo)==null?void 0:Do.type)&&((U=T==null?void 0:T.transactionTokenInfo)==null?void 0:U.multiTxData)&&!(T!=null&&T.dataField.endsWith((No=T==null?void 0:T.transactionTokenInfo)==null?void 0:No.multiTxData)));function In(){Lo?h(O=>O+1):po()}function An(){h(O=>{let H=O+1;return H>(D==null?void 0:D.length)?O:H})}function so(){h(O=>{let H=O-1;return H<0?O:H})}return{allTransactions:D,onSignTransaction:In,onNext:An,onPrev:so,waitingForDevice:M,onAbort:kn,isLastTransaction:Z,isFirstTransaction:u===0,hasMultipleTransactions:D.length>1,shouldContinueWithoutSigning:Lo,currentStep:u,signedTransactions:f,currentTransaction:T}}P();Q();To();t();je();P();function qi(){let e=z();return o=>{e(zo()),e(Tt(o)),e(_t(Kn))}}function Vt({onCancel:e,verifyReceiverScam:o=!0}){let n=C(vo),r=C(rn),{account:{address:p}}=$i(),{provider:s}=hn(),m=z(),g=qi(),{transactions:_,sessionId:u,callbackRoute:h,customTransactionInformation:f}=n||{};function y(D){m(va(D))}let T=h!=null&&window.location.pathname.includes(h);function v(D){!Ht(D)||u&&(m(qo({sessionId:u,status:"signed",transactions:D.map(W=>pn(W))})),h!=null&&(f==null?void 0:f.redirectAfterSign)&&!T&&Vo(h))}function M(){e(),g(u)}function te(D){return A(this,null,function*(){return yield s.signTransaction(D)})}let we=zi({address:p,egldLabel:r,transactionsToSign:_,onGetScamAddressData:o?Hi:null,isLedger:no("ledger"),onCancel:M,onSignTransaction:te,onTransactionsSignError:y,onTransactionsSignSuccess:v});return I(b({},we),{callbackRoute:h})}t();t();t();t();t();Ye();t();t();t();var Tm=x(require("axios"));t();var xm=x(require("axios"));t();var bm=x(require("axios"));t();t();var Sm=x(require("axios"));t();var wm=x(require("axios"));t();P();R();t();Ye();t();P();R();t();P();Q();t();var Lm=require("react");t();t();t();P();t();t();var Am=x(require("bignumber.js"));N();P();Q();t();var Im=x(require("bignumber.js"));N();t();var Em=require("@elrondnetwork/erdjs"),Cm=x(require("bignumber.js"));N();R();Ye();t();t();t();t();t();t();t();t();t();var Xi=require("react");t();t();var Dm=require("react"),Nm=require("@elrondnetwork/erdjs-extension-provider");je();X();P();R();ze();t();var Pm=require("react");je();P();t();var Zi=require("react"),Mm=require("@elrondnetwork/erdjs-hw-provider");je();X();P();R();ze();t();var qt=require("react"),Bm=require("@elrondnetwork/erdjs-wallet-connect-provider");X();P();R();ze();t();var Ji=require("react"),Om=require("@elrondnetwork/erdjs-wallet-connect-provider");X();P();R();ze();t();t();var Qi=require("react");t();var Gm=require("react");t();var Kt=require("react"),Wm=require("socket.io-client");P();t();t();t();t();var ec=e=>e.dappModal,oc=S(ec,e=>e.visible),to=S(ec,e=>e.modal),jv=S(to,e=>e.showHeader),zv=S(to,e=>e.showFooter),qv=S(to,e=>e.headerText),Kv=S(to,e=>e.footerText);var nc=()=>C(oc);t();var mc=x(require("react"));t();t();t();var ao=x(require("react")),Qt=x(require("classnames")),pc=x(require("react-dom"));t();var tc=x(require("react")),rc=x(require("classnames"));t();var Hm=`.dapp-core-component__dappModalStyles__dappModal {
  position: fixed;
  z-index: 999;
  padding-top: 100px;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 100%;
  height: 100%;
  overflow: auto;
  background-color: rgba(0, 0, 0, 0.4);
  display: flex;
  align-items: center;
}

.dapp-core-component__dappModalStyles__dappModalContent {
  background: #fff;
  border: 2px solid #aaa;
  border-radius: 5px;
  z-index: 999;
  max-width: 500px;
  margin: auto;
  padding: 0;
  position: relative;
  width: 100%;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.dapp-core-component__dappModalStyles__dappModalHeader {
  padding: 2px 16px;
  background-color: #fff;
  font-size: 1.25rem;
}

.dapp-core-component__dappModalStyles__dappModalHeaderText {
  padding: 0 1rem;
}

.dapp-core-component__dappModalStyles__dappModalBody {
  padding: 16px 0;
}

.dapp-core-component__dappModalStyles__dappModalFooter {
  background-color: #fff;
  padding: 1rem 0.5rem;
}

.dapp-core-component__dappModalStyles__dappModalCloseButton {
  position: absolute;
  top: 0;
  right: 0;
  float: right;
  font-weight: bold;
  border-radius: 5px;
  font-size: 1rem;
  padding: 0.5rem 1rem;
  border: 0;
}

.dapp-core-component__dappModalStyles__dappModalCloseButton:hover,
.dapp-core-component__dappModalStyles__dappModalCloseButton:focus {
  text-decoration: none;
  cursor: pointer;
  color: #212529;
  background-color: #dee2e6;
  border-color: #e9ecef;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Hm));var ne={dappModal:"dapp-core-component__dappModalStyles__dappModal",dappModalContent:"dapp-core-component__dappModalStyles__dappModalContent",dappModalHeader:"dapp-core-component__dappModalStyles__dappModalHeader",dappModalHeaderText:"dapp-core-component__dappModalStyles__dappModalHeaderText",dappModalBody:"dapp-core-component__dappModalStyles__dappModalBody",dappModalFooter:"dapp-core-component__dappModalStyles__dappModalFooter",dappModalCloseButton:"dapp-core-component__dappModalStyles__dappModalCloseButton"};var Xt=({className:e,children:o})=>tc.default.createElement("div",{className:(0,rc.default)(ne.dappModalBody,e)},o);t();var Yt=x(require("react")),ac=x(require("classnames"));var Zt=({visible:e,customFooter:o,className:n,footerText:r})=>e?Yt.default.createElement("div",{className:(0,ac.default)(ne.dappModalFooter,n)},o!=null?o:Yt.default.createElement("div",null,r)):null;t();var ro=x(require("react")),ic=require("@fortawesome/free-solid-svg-icons"),cc=require("@fortawesome/react-fontawesome"),yn=x(require("classnames"));var Jt=({visible:e,headerText:o,customHeader:n,className:r,closeButtonClassName:p,onHide:s})=>e?n?ro.default.createElement("div",{className:(0,yn.default)(ne.dappModalHeader,r)},n):ro.default.createElement("div",{className:(0,yn.default)(ne.dappModalHeader,r)},ro.default.createElement("div",{className:ne.dappModalHeaderText},o),ro.default.createElement("button",{onClick:s,className:(0,yn.default)(ne.dappModalCloseButton,d.btn,d.btnLight,p)},ro.default.createElement(cc.FontAwesomeIcon,{size:"lg",icon:ic.faTimes}))):null;var $m={showHeader:!0,showFooter:!1,headerText:"",footerText:""},er=({id:e="dapp-modal",visible:o,onHide:n,parentElement:r,config:p=$m,children:s,className:m="dapp-modal-dialog-wrapper"})=>{if(!o)return null;let{showHeader:g,showFooter:_,headerText:u,footerText:h,modalDialogClassName:f="dapp-modal-dialog",modalContentClassName:y="dapp-modal-dialog-content",modalHeaderClassName:T="dapp-modal-dialog-header",modalCloseButtonClassName:v="dapp-modal-dialog-close-button",modalBodyClassName:M="dapp-modal-dialog-content-body",modalFooterClassName:te="dapp-modal-dialog-footer",customModalHeader:we,customModalFooter:D}=p;return pc.default.createPortal(ao.default.createElement("div",{id:e,role:"dialog","aria-modal":"true",className:(0,Qt.default)(f,ne.dappModal,m)},ao.default.createElement("div",{className:(0,Qt.default)(ne.dappModalContent,y)},ao.default.createElement(Jt,{visible:g,headerText:u,customHeader:we,className:T,closeButtonClassName:v,onHide:n}),ao.default.createElement(Xt,{className:M},s),ao.default.createElement(Zt,{visible:_,customFooter:D,footerText:h,className:te}))),r!=null?r:document.body)};t();t();var io=require("react");St();var sc=e=>{var g;let o=z(),n=nc(),r=C(to),p=()=>{o(bt(!0))},s=()=>{o(bt(!1))},m=(0,io.useCallback)(_=>{o(Wa(_))},[]);return(0,io.useEffect)(()=>{e!=null&&e.modalConfig&&m(e==null?void 0:e.modalConfig)},[e==null?void 0:e.modalConfig]),(0,io.useEffect)(()=>{var _;return(_=e==null?void 0:e.config)!=null&&_.openOnMount&&p(),()=>{s()}},[]),(0,io.useEffect)(()=>{var _;typeof((_=e==null?void 0:e.config)==null?void 0:_.visible)=="boolean"&&(e.config.visible?p():s())},[(g=e==null?void 0:e.config)==null?void 0:g.visible]),{handleShowModal:p,handleHideModal:s,setModalConfig:m,visible:n,config:r}};t();var dc=e=>{let{handleHideModal:o,visible:n,config:r}=sc({modalConfig:e.modalConfig,config:I(b({},e.modalInteractionConfig),{visible:e.visible})});return mc.default.createElement(er,{onHide:()=>{var s;o(),(s=e.onClose)==null||s.call(e)},visible:n,config:r,className:e.className},e.children)};t();var L=x(require("react")),Ec=require("@elrondnetwork/erdjs/out"),wn=require("@fortawesome/free-solid-svg-icons"),Cc=require("@fortawesome/react-fontawesome");t();var co=x(require("react")),_c=require("@fortawesome/react-fontawesome"),Eo=x(require("classnames"));t();var Vm=`.dapp-core-component__pageStateStyles__wrapper {
  color: inherit;
}

.dapp-core-component__pageStateStyles__state {
  color: inherit;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Vm));var lc={wrapper:"dapp-core-component__pageStateStyles__wrapper",state:"dapp-core-component__pageStateStyles__state"};var gc=({icon:e,title:o,action:n,iconClass:r,dataTestId:p,description:s,iconBgClass:m,iconSize:g="5x",className:_="dapp-page-state"})=>{let u={wrapper:(0,Eo.default)(lc.state,d.mAuto,d.p4,d.textCenter,_),iconContainer:(0,Eo.default)(d.iconState,d.mxAuto,{[m!=null?m:""]:Boolean(m)}),iconClass:(0,Eo.default)(r!=null&&r),title:(0,Eo.default)(d.h4,d.my4),description:d.mb3};return co.default.createElement("div",{className:u.wrapper,"data-testid":p},e&&co.default.createElement("span",{className:u.iconContainer},co.default.createElement(_c.FontAwesomeIcon,{icon:e,className:u.iconClass,size:g})),o&&co.default.createElement("p",{className:u.title},o),s&&co.default.createElement("div",{className:u.description},s),n)};t();var Co=x(require("react"));t();var uc=x(require("react")),fc=x(require("classnames"));var hc=({color:e,"data-testid":o,className:n="dapp-dot"})=>uc.default.createElement("span",{className:(0,fc.default)(d.dot,e,n),"data-testid":o});var bn=x(require("classnames")),Tc=({totalSteps:e,currentStep:o,className:n="dapp-progress-steps"})=>{let r=[];for(let p=1;p<=e;p+=1){let s=o===p,m=s||p<o?d.bgPrimary:d.bgSecondary;r.push(Co.default.createElement(hc,{"data-testid":`step${p}${s?"active":""}`,key:p,color:m}))}return Co.default.createElement("div",{className:(0,bn.default)(d.progressSteps,d.flexRow,d.justifyContentCenter,n)},Co.default.createElement("div",{className:(0,bn.default)(d.steps,d.flexRow,d.justifyContentCenter,d.alignItemsCenter,d.positionRelative)},Co.default.createElement("hr",{className:(0,bn.default)(d.positionAbsolute,d.w100,d.m0)}),r))};t();var $=x(require("react")),wc=require("@fortawesome/free-solid-svg-icons"),vc=require("@fortawesome/react-fontawesome");t();var k=x(require("react")),jm=e=>k.createElement("svg",b({viewBox:"0 0 100 100",xmlns:"http://www.w3.org/2000/svg",role:"img"},e),k.createElement("g",{fillRule:"evenodd"},k.createElement("g",{transform:"translate(0 -1)",fillRule:"nonzero"},k.createElement("g",{transform:"translate(0 1)",id:"egld-token"},k.createElement("path",{d:"m79.6 19.424c-0.4-0.4383-0.6-0.8766-0.6-1.4245 0-0.54787 0.2-0.98617 0.6-1.4245 0.8-0.76702 2-0.76702 2.8 0 0.4 0.4383 0.6 0.8766 0.6 1.4245 0 0.54787-0.2 0.98617-0.6 1.4245-0.4 0.32872-0.9 0.54787-1.4 0.54787-0.5 0.10957-1-0.10957-1.4-0.54787"}),k.createElement("path",{id:"Path",d:"m74.556 24.448c-0.33333-0.44138-0.55556-0.88276-0.55556-1.4345s0.22222-0.9931 0.55556-1.4345c0.77778-0.77241 2.1111-0.77241 2.8889 0 0.33333 0.44138 0.55556 0.88276 0.55556 1.4345s-0.22222 0.9931-0.55556 1.4345c-0.44444 0.33103-0.88889 0.55172-1.4444 0.55172s-1-0.22069-1.4444-0.55172"}),k.createElement("path",{id:"Path",d:"m69.556 30.448c-0.33333-0.33103-0.55556-0.88276-0.55556-1.4345s0.22222-0.9931 0.55556-1.4345c0.77778-0.77241 2.1111-0.77241 2.8889 0 0.33333 0.33103 0.55556 0.88276 0.55556 1.4345s-0.22222 0.9931-0.55556 1.4345c-0.44444 0.44138-0.88889 0.55172-1.4444 0.55172s-1.1111-0.22069-1.4444-0.55172"}),k.createElement("path",{id:"Path",d:"m64.6 35.448c-0.4-0.33103-0.6-0.88276-0.6-1.4345s0.2-0.9931 0.6-1.4345c0.8-0.77241 2-0.77241 2.8 0 0.4 0.33103 0.6 0.88276 0.6 1.4345s-0.2 1.1034-0.6 1.4345c-0.4 0.44138-0.9 0.55172-1.4 0.55172s-1-0.11034-1.4-0.55172"}),k.createElement("path",{id:"Path",d:"m59.556 40.424c-0.44444-0.32872-0.55556-0.8766-0.55556-1.4245 0-0.54787 0.22222-0.98617 0.55556-1.4245 0.77778-0.76702 2.1111-0.76702 2.8889 0 0.44444 0.32872 0.55556 0.8766 0.55556 1.4245 0 0.54787-0.22222 1.0957-0.55556 1.4245-0.33333 0.4383-0.88889 0.54787-1.4444 0.54787-0.55556 0.10957-1.1111-0.10957-1.4444-0.54787"}),k.createElement("path",{id:"Path",d:"m53 44.014c0-0.55172 0.22222-0.9931 0.55556-1.4345 0.77778-0.77241 2.1111-0.77241 2.8889 0 0.44444 0.44138 0.55556 0.88276 0.55556 1.4345s-0.22222 0.9931-0.55556 1.4345c-0.33333 0.33103-0.88889 0.55172-1.4444 0.55172s-1-0.22069-1.4444-0.55172c-0.33333-0.33103-0.55556-0.88276-0.55556-1.4345z"}),k.createElement("path",{id:"Path",d:"m43.556 56.424c-0.33333-0.32872-0.55556-0.8766-0.55556-1.4245 0-0.54787 0.22222-0.98617 0.55556-1.4245 0.77778-0.76702 2.1111-0.76702 2.8889 0 0.44444 0.4383 0.55556 0.8766 0.55556 1.4245 0 0.54787-0.22222 0.98617-0.55556 1.4245-0.33333 0.4383-0.88889 0.54787-1.4444 0.54787-0.55556 0.10957-1-0.10957-1.4444-0.54787"}),k.createElement("path",{id:"Path",d:"m38.6 61.424c-0.4-0.32872-0.6-0.8766-0.6-1.4245 0-0.54787 0.2-0.98617 0.6-1.4245 0.8-0.76702 2-0.76702 2.8 0 0.4 0.4383 0.6 0.8766 0.6 1.4245 0 0.54787-0.2 0.98617-0.6 1.4245-0.4 0.4383-0.9 0.54787-1.4 0.54787-0.5 0.10957-1-0.10957-1.4-0.54787"}),k.createElement("path",{id:"Path",d:"m33.556 66.448c-0.33333-0.33103-0.55556-0.88276-0.55556-1.4345s0.22222-1.1034 0.55556-1.4345c0.77778-0.77241 2.1111-0.77241 2.8889 0 0.33333 0.33103 0.55556 0.88276 0.55556 1.4345s-0.22222 0.9931-0.55556 1.4345c-0.44444 0.44138-0.88889 0.55172-1.4444 0.55172s-1-0.22069-1.4444-0.55172"}),k.createElement("path",{id:"Path",d:"m27.556 71.448c-0.44444-0.33103-0.55556-0.88276-0.55556-1.4345s0.22222-0.9931 0.55556-1.4345c0.77778-0.77241 2.1111-0.77241 2.8889 0 0.33333 0.44138 0.55556 0.88276 0.55556 1.4345s-0.22222 1.1034-0.55556 1.4345c-0.44444 0.44138-0.88889 0.55172-1.4444 0.55172s-1.1111-0.11034-1.4444-0.55172"}),k.createElement("path",{id:"Path",d:"m22.615 76.448c-0.41026-0.33103-0.61538-0.88276-0.61538-1.4345s0.20513-0.9931 0.61538-1.4345c0.82051-0.77241 2.0513-0.77241 2.7692 0 0.41026 0.44138 0.61538 0.88276 0.61538 1.4345s-0.20513 1.1034-0.61538 1.4345c-0.41026 0.33103-0.92308 0.55172-1.4359 0.55172-0.51282 0-1.0256-0.11034-1.3333-0.55172"}),k.createElement("path",{id:"Path",d:"m17.556 82.424c-0.44444-0.4383-0.55556-0.8766-0.55556-1.4245 0-0.54787 0.22222-0.98617 0.55556-1.4245 0.77778-0.76702 2.1111-0.76702 2.8889 0 0.44444 0.4383 0.55556 0.8766 0.55556 1.4245 0 0.54787-0.22222 0.98617-0.55556 1.4245-0.33333 0.32872-0.88889 0.54787-1.4444 0.54787-0.44444 0.10957-1-0.10957-1.4444-0.54787"}),k.createElement("path",{id:"Path",d:"m20.444 19.424c0.33333-0.4383 0.55556-0.8766 0.55556-1.4245 0-0.54787-0.22222-0.98617-0.55556-1.4245-0.77778-0.76702-2.1111-0.76702-2.8889 0-0.44444 0.4383-0.55556 0.8766-0.55556 1.4245 0 0.54787 0.22222 0.98617 0.55556 1.4245 0.33333 0.32872 0.88889 0.54787 1.4444 0.54787 0.55556 0.10957 1.1111-0.10957 1.4444-0.54787"}),k.createElement("path",{id:"Path",d:"m25.4 24.448c0.4-0.44138 0.6-0.88276 0.6-1.4345s-0.2-0.9931-0.6-1.4345c-0.8-0.77241-2-0.77241-2.8 0-0.4 0.44138-0.6 0.88276-0.6 1.4345s0.2 0.9931 0.6 1.4345c0.4 0.33103 0.9 0.55172 1.4 0.55172s1-0.22069 1.4-0.55172"}),k.createElement("path",{id:"Path",d:"m30.444 30.448c0.33333-0.33103 0.55556-0.88276 0.55556-1.4345s-0.22222-0.9931-0.55556-1.4345c-0.77778-0.77241-2.1111-0.77241-2.8889 0-0.33333 0.33103-0.55556 0.88276-0.55556 1.4345s0.22222 0.9931 0.55556 1.4345c0.44444 0.44138 0.88889 0.55172 1.4444 0.55172 0.44444 0 1-0.22069 1.4444-0.55172"}),k.createElement("path",{id:"Path",d:"m36.444 35.448c0.33333-0.33103 0.55556-0.88276 0.55556-1.4345s-0.22222-0.9931-0.55556-1.4345c-0.77778-0.77241-2.1111-0.77241-2.8889 0-0.33333 0.33103-0.55556 0.88276-0.55556 1.4345s0.22222 1.1034 0.55556 1.4345c0.44444 0.44138 0.88889 0.55172 1.4444 0.55172 0.44444 0 1-0.11034 1.4444-0.55172"}),k.createElement("path",{id:"Path",d:"m41.385 40.424c0.41026-0.32872 0.61538-0.8766 0.61538-1.4245 0-0.54787-0.20513-0.98617-0.61538-1.4245-0.71795-0.76702-2.0513-0.76702-2.7692 0-0.41026 0.32872-0.61538 0.8766-0.61538 1.4245 0 0.54787 0.20513 1.0957 0.61538 1.4245 0.41026 0.4383 0.92308 0.54787 1.4359 0.54787 0.51282 0.10957 1.0256-0.10957 1.3333-0.54787"}),k.createElement("path",{id:"Path",d:"m47 44.014c0-0.55172-0.22222-0.9931-0.55556-1.4345-0.77778-0.77241-2.1111-0.77241-2.8889 0-0.44444 0.44138-0.55556 0.88276-0.55556 1.4345s0.22222 0.9931 0.55556 1.4345c0.33333 0.33103 0.88889 0.55172 1.4444 0.55172s1-0.22069 1.4444-0.55172c0.44444-0.33103 0.55556-0.88276 0.55556-1.4345z"}),k.createElement("path",{id:"Path",d:"m51.444 51.448c0.44444-0.44138 0.55556-0.88276 0.55556-1.4345s-0.22222-0.9931-0.55556-1.4345c-0.77778-0.77241-2.1111-0.77241-2.8889 0-0.44444 0.44138-0.55556 0.88276-0.55556 1.4345s0.22222 0.9931 0.55556 1.4345c0.33333 0.33103 0.88889 0.55172 1.4444 0.55172s1.1111-0.22069 1.4444-0.55172"}),k.createElement("path",{id:"Path",d:"m56.385 56.424c0.41026-0.32872 0.61538-0.8766 0.61538-1.4245 0-0.54787-0.20513-0.98617-0.61538-1.4245-0.82051-0.76702-2.0513-0.76702-2.7692 0-0.41026 0.4383-0.61538 0.8766-0.61538 1.4245 0 0.54787 0.20513 0.98617 0.61538 1.4245 0.41026 0.4383 0.92308 0.54787 1.4359 0.54787 0.41026 0.10957 0.92308-0.10957 1.3333-0.54787"}),k.createElement("path",{id:"Path",d:"m62.444 61.424c0.33333-0.32872 0.55556-0.8766 0.55556-1.4245 0-0.54787-0.22222-0.98617-0.55556-1.4245-0.77778-0.76702-2.1111-0.76702-2.8889 0-0.33333 0.4383-0.55556 0.8766-0.55556 1.4245 0 0.54787 0.22222 0.98617 0.55556 1.4245 0.44444 0.4383 0.88889 0.54787 1.4444 0.54787 0.44444 0.10957 1-0.10957 1.4444-0.54787"}),k.createElement("path",{id:"Path",d:"m67.444 66.448c0.33333-0.33103 0.55556-0.88276 0.55556-1.4345s-0.22222-1.1034-0.55556-1.4345c-0.77778-0.77241-2.1111-0.77241-2.8889 0-0.33333 0.33103-0.55556 0.88276-0.55556 1.4345s0.22222 0.9931 0.55556 1.4345c0.44444 0.44138 0.88889 0.55172 1.4444 0.55172s1-0.22069 1.4444-0.55172"}),k.createElement("path",{id:"Path",d:"m72.444 71.448c0.44444-0.33103 0.55556-0.88276 0.55556-1.4345s-0.22222-0.9931-0.55556-1.4345c-0.77778-0.77241-2.1111-0.77241-2.8889 0-0.33333 0.44138-0.55556 0.88276-0.55556 1.4345s0.22222 1.1034 0.55556 1.4345c0.44444 0.44138 0.88889 0.55172 1.4444 0.55172s1.1111-0.11034 1.4444-0.55172"}),k.createElement("path",{id:"a",d:"m77.444 76.448c0.44444-0.33103 0.55556-0.88276 0.55556-1.4345s-0.22222-0.9931-0.55556-1.4345c-0.77778-0.77241-2.1111-0.77241-2.8889 0-0.44444 0.44138-0.55556 0.88276-0.55556 1.4345s0.22222 1.1034 0.55556 1.4345c0.33333 0.33103 0.88889 0.55172 1.4444 0.55172s1.1111-0.11034 1.4444-0.55172"}),k.createElement("path",{d:"m82.385 82.424c0.41026-0.4383 0.61538-0.8766 0.61538-1.4245 0-0.54787-0.20513-0.98617-0.61538-1.4245-0.82051-0.76702-2.0513-0.76702-2.7692 0-0.41026 0.4383-0.61538 0.8766-0.61538 1.4245 0 0.54787 0.20513 0.98617 0.61538 1.4245 0.41026 0.32872 0.92308 0.54787 1.4359 0.54787 0.41026 0.10957 0.92308-0.10957 1.3333-0.54787"}),k.createElement("path",{d:"m33.036 21c5.2647-3.4327 11.036-5.0481 17.414-5.149 6.2771 0 12.048 1.7163 17.414 5.149l11.137-11.106c-8.7069-6.5625-18.123-9.8942-28.551-9.8942-10.428 0-19.945 3.3317-28.449 9.8942l11.036 11.106z"}),k.createElement("path",{d:"m21 66.742c-3.5854-5.3754-5.3268-11.157-5.3268-17.343 0-6.2883 1.7415-12.069 5.3268-17.343l-11.166-11.055c-6.5561 8.5196-9.8341 17.952-9.8341 28.5 0 10.345 3.278 19.879 9.9366 28.5l11.063-11.258z"}),k.createElement("path",{d:"m67.964 78c-5.2647 3.5333-11.036 5.3504-17.313 5.4513-6.3783 0-12.25-1.7162-17.515-5.3504l-11.137 11.004c8.6057 6.6628 18.123 9.9941 28.652 9.8932 10.428-0.10095 19.844-3.4323 28.348-9.8932l-11.036-11.105z"}),k.createElement("path",{d:"m79 32.258c3.5333 5.3754 5.3504 11.258 5.2494 17.648-0.10095 6.1868-1.8171 11.867-5.2494 17.039l11.105 11.055c6.5618-8.5196 9.8932-17.952 9.8932-28.297 0.10095-10.548-3.2304-20.082-9.8932-28.703l-11.105 11.258z"}),k.createElement("path",{d:"m90 81c4.3636 0 8 3.5325 8 8 0 4.3636-3.5325 8-8 8-4.3636 0-8-3.5325-8-8s3.5325-8 8-8"}),k.createElement("path",{d:"m10 81c4.4211 0 8 3.5325 8 8 0 4.3636-3.5789 8-8 8-4.4211 0-8-3.5325-8-8s3.5789-8 8-8"}),k.createElement("path",{d:"m90 1c4.4211 0 8 3.5325 8 8 0 4.4675-3.5789 8-8 8s-8-3.5325-8-8c0-4.4675 3.5789-8 8-8"}),k.createElement("path",{d:"m10 1c4.3636 0 8 3.5789 8 8 0 4.4211-3.5325 8-8 8-4.3636 0-8-3.5789-8-8 0-4.4211 3.5325-8 8-8"}))))),xc=jm;t();var or=x(require("react")),nr=x(require("classnames"));t();var zm=`.dapp-core-component__tokenSymbol__token-symbol,
.dapp-core-component__tokenSymbol__token-symbol-combined {
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #fff;
  border-radius: 50%;
}
.dapp-core-component__tokenSymbol__token-symbol svg #dapp-core-component__tokenSymbol__egld-token,
.dapp-core-component__tokenSymbol__token-symbol-combined svg #dapp-core-component__tokenSymbol__egld-token {
  fill: #000;
}

.dapp-core-component__tokenSymbol__token-symbol-combined {
  background-color: rgba(0, 0, 0, 0);
}

.dapp-core-component__tokenSymbol__token-symbol {
  border: 1px solid #000;
  width: 2rem;
  height: 2rem;
}
.dapp-core-component__tokenSymbol__token-symbol img {
  width: 100%;
  height: 100%;
}
.dapp-core-component__tokenSymbol__token-symbol svg {
  width: 1.25rem;
  height: 1.25rem;
}
.dapp-core-component__tokenSymbol__token-symbol svg.dapp-core-component__tokenSymbol__expanded {
  width: 100% !important;
  height: 100% !important;
}

.dapp-core-component__tokenSymbol__token-symbol-combined {
  width: 4.5rem;
  height: 4.5rem;
}
.dapp-core-component__tokenSymbol__token-symbol-combined svg circle {
  fill: #fff;
}
.dapp-core-component__tokenSymbol__token-symbol-combined svg {
  width: 4.25rem;
  height: 4.25rem;
}
.dapp-core-component__tokenSymbol__token-symbol-combined.dapp-core-component__tokenSymbol__small {
  width: 2.5rem;
  min-width: 2.5rem;
  height: 2.5rem;
}
.dapp-core-component__tokenSymbol__token-symbol-combined.dapp-core-component__tokenSymbol__small svg {
  width: 2.5rem;
  height: 2.5rem;
}

.dapp-core-component__tokenSymbol__token-icon-lg .dapp-core-component__tokenSymbol__token-symbol {
  width: 3rem;
  height: 3rem;
}
.dapp-core-component__tokenSymbol__token-icon-lg .dapp-core-component__tokenSymbol__token-symbol svg {
  width: 2.25rem;
  height: 2.25rem;
}

.dapp-core-component__tokenSymbol__token-icon-xl .dapp-core-component__tokenSymbol__token-symbol {
  width: 4rem;
  height: 4rem;
}
.dapp-core-component__tokenSymbol__token-icon-xl .dapp-core-component__tokenSymbol__token-symbol svg {
  width: 2.3rem;
  height: 2.3rem;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(zm));var Sn={"token-symbol":"dapp-core-component__tokenSymbol__token-symbol",tokenSymbol:"dapp-core-component__tokenSymbol__token-symbol","token-symbol-combined":"dapp-core-component__tokenSymbol__token-symbol-combined",tokenSymbolCombined:"dapp-core-component__tokenSymbol__token-symbol-combined","egld-token":"dapp-core-component__tokenSymbol__egld-token",egldToken:"dapp-core-component__tokenSymbol__egld-token",expanded:"dapp-core-component__tokenSymbol__expanded",small:"dapp-core-component__tokenSymbol__small","token-icon-lg":"dapp-core-component__tokenSymbol__token-icon-lg",tokenIconLg:"dapp-core-component__tokenSymbol__token-icon-lg","token-icon-xl":"dapp-core-component__tokenSymbol__token-icon-xl",tokenIconXl:"dapp-core-component__tokenSymbol__token-icon-xl"};var yc=({children:e,className:o="dapp-simple-token-symbol"})=>or.default.createElement("div",{className:(0,nr.default)(Sn.tokenSymbol,o)},e),bc=({small:e,children:o,className:n="dapp-combined-token-symbol"})=>or.default.createElement("div",{className:(0,nr.default)(Sn.tokenSymbolCombined,{[Sn.small]:e},n)},o);t();var qm=`.dapp-core-component__tokenDetailsStyles__token-symbol-custom-token {
  color: inherit;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(qm));var Sc={"token-symbol-custom-token":"dapp-core-component__tokenDetailsStyles__token-symbol-custom-token",tokenSymbolCustomToken:"dapp-core-component__tokenDetailsStyles__token-symbol-custom-token"};var tr=e=>{let o=e.split("-");return e.includes("-")?`${o[0]}-${o[1]}`:e};function Km(e,o){return o?$.default.createElement("img",{className:Sc.tokenSymbolCustomToken,src:o}):e?$.default.createElement(xc,null):$.default.createElement(vc.FontAwesomeIcon,{icon:wc.faDiamond})}var rr=(e,o)=>{let n=Tn(),r=e===n;return{token:e,symbol:e?e.split("-")[0]:"",label:e,icon:()=>Km(r,o)}},_e=class extends $.Component{render(){return null}};_e.Token=o=>$.default.createElement($.default.Fragment,null,o.token),_e.Symbol=o=>$.default.createElement($.default.Fragment,null,rr(tr(o.token),o.tokenAvatar).symbol),_e.Label=o=>$.default.createElement($.default.Fragment,null,rr(tr(o.token),o.tokenAvatar).label),_e.Icon=o=>{var r;let n=rr(tr(o.token),o.tokenAvatar).icon;return $.default.createElement("div",{className:(r=o.className)!=null?r:"dapp-token-details-icon"},o.combined?$.default.createElement(bc,{small:o.small},$.default.createElement(n,null)):$.default.createElement(yc,null,$.default.createElement(n,null)))};t();var F=x(require("react")),Ue=x(require("classnames"));N();var Xm=(e,o)=>[...e.matchAll(new RegExp(o,"gi"))].map(n=>n.index),kc=({data:e,highlight:o,isScCall:n,className:r="dapp-transaction-data"})=>{let p=F.default.createElement(F.default.Fragment,null,e),[s,...m]=o&&n?o.split("@"):[];if(e&&o&&Xm(e,o).length===1)switch(!0){case e.startsWith(o):{let[,g]=e.split(o);p=F.default.createElement(F.default.Fragment,null,o,F.default.createElement("span",{className:d.textMuted},g));break}case e.endsWith(o):{let[g]=e.split(o);p=F.default.createElement(F.default.Fragment,null,F.default.createElement("span",{className:d.textMuted},g),o);break}default:{let[g,_]=e.split(o);p=F.default.createElement(F.default.Fragment,null,F.default.createElement("span",{className:d.textMuted},g),F.default.createElement("span",{className:d.highlighted},o),F.default.createElement("span",{className:d.textMuted},_));break}}return F.default.createElement(F.default.Fragment,null,s&&F.default.createElement("div",{className:(0,Ue.default)(d.formGroup,d.mb0,d.dataField,d.mw100,d.w100,r)},F.default.createElement("span",{className:(0,Ue.default)(d.formLabel,d.textSecondary,d.dBlock)},"SC Call"),F.default.createElement("div",{"data-testid":"confirmScCall",className:(0,Ue.default)(d.textarea,d.scCall,d.formControl,d.mt1,d.textBreak,d.w100)},[Fe(s),...m].join("@"))),F.default.createElement("div",{className:(0,Ue.default)(d.formGroup,d.mb0,d.dataField,d.mw100,d.w100,r)},F.default.createElement("span",{className:(0,Ue.default)(d.formLabel,d.textSecondary,d.dBlock)},"Data"),F.default.createElement("div",{"data-testid":"confirmData",className:(0,Ue.default)(d.textarea,d.formControl,d.mt1,d.textBreak,d.w100)},e?p:ea)))};t();var q=x(require("classnames"));var Ic=require("react"),Ac=(e=null)=>(0,Ic.useMemo)(()=>({formGroup:(0,q.default)(d.formGroup,d.textBreak,d.textLeft),formLabel:(0,q.default)(d.textBreak,d.textSecondary),icon:d.textWhite,contentWrapper:(0,q.default)(d.dFlex,d.flexColumn,d.justifyContentStart,d.flexRow,d.justifyContentBetween,d.mb3),tokenWrapper:(0,q.default)(d.mb3,d.mb0,d.dFlex,d.flexColumn,d.alignItemsStart),tokenLabel:(0,q.default)(d.textSecondary,d.textLeft),tokenValue:(0,q.default)(d.dFlex,d.alignItemsCenter,d.mt1),scamReport:d.textWarning,scamReportIcon:(0,q.default)(d.textWarning,d.mr1),tokenAmountLabel:(0,q.default)(d.textSecondary,d.textLeft),tokenAmountValue:(0,q.default)(d.dFlex,d.alignItemsCenter),dataFormGroup:(0,q.default)(d.formGroup,d.textLeft),errorMessage:(0,q.default)(d.textDanger,d.dFlex,d.justifyContentCenter,d.alignItemsCenter),buttonsWrapper:(0,q.default)(d.dFlex,d.alignItemsCenter,d.justifyContentEnd,d.mt1),cancelButton:(0,q.default)(d.btn,d.btnDark,d.textWhite,d.dFlex,d.mr2),signButton:(0,q.default)(d.btn,e?d.btnWarning:d.btnPrimary,d.dFlex,d.ml2)}),[e]);var Lc=({onSignTransaction:e,handleClose:o,onPrev:n,title:r,waitingForDevice:p,currentTransaction:s,error:m,allTransactions:g,isLastTransaction:_,currentStep:u,className:h})=>{let f=Tn();if(!s)return null;let y=s.transaction.getData().toString(),{network:T}=ko(),{tokenId:v,nonce:M,amount:te,type:we,multiTxData:D,receiver:Z}=s.transactionTokenInfo,W=Boolean(v&&cn({tokenId:v,erdLabel:f})),ae=u===0,ge=No=>{No.preventDefault(),ae?o():n()},po=we&&D&&!y.endsWith(D),re="Sign & Continue";re=p?"Check your Ledger":re,re=_&&!p?"Sign & Submit":re,re=po?"Continue":re;let kn=`${v}-${M}`,{tokenDecimals:Lo,tokenAvatar:In}=Wi({tokenId:M&&M.length>0?kn:v}),An=Be({input:W?te:s.transaction.getValue().toString(),decimals:W?Lo:Number(T.decimals),digits:Number(T.digits),showLastNonZeroDecimal:!1,addCommas:!0}),so=s.receiverScamInfo,Do=g.length>1,U=Ac(so);return L.default.createElement(gc,{icon:m?wn.faTimes:null,iconClass:U.icon,iconBgClass:m?d.bgDanger:d.bgWarning,iconSize:"3x",className:h,title:r||"Confirm on Ledger",description:L.default.createElement(L.default.Fragment,null,s.transaction&&L.default.createElement(L.default.Fragment,null,Do&&L.default.createElement(Tc,{totalSteps:g.length,currentStep:u+1,className:d.mb4}),L.default.createElement("div",{className:U.formGroup,"data-testid":"transactionTitle"},L.default.createElement("div",{className:U.formLabel},"To "),D?new Ec.Address(Z).bech32():s.transaction.getReceiver().toString(),so&&L.default.createElement("div",{className:U.scamReport},L.default.createElement("span",null,L.default.createElement(Cc.FontAwesomeIcon,{icon:wn.faExclamationTriangle,className:U.scamReportIcon}),L.default.createElement("small",null,so)))),L.default.createElement("div",{className:U.contentWrapper},L.default.createElement("div",{className:U.tokenWrapper},L.default.createElement("div",{className:U.tokenLabel},"Token"),L.default.createElement("div",{className:U.tokenValue},L.default.createElement(_e.Icon,{tokenAvatar:In,token:v||f}),L.default.createElement("div",{className:d.mr1}),L.default.createElement(_e.Label,{token:v||f}))),L.default.createElement("div",null,L.default.createElement("div",{className:U.tokenAmountLabel},"Amount"),L.default.createElement("div",{className:U.tokenAmountValue},L.default.createElement("div",{className:d.mr1},An),L.default.createElement(_e.Symbol,{token:v||f})))),L.default.createElement("div",{className:U.dataFormGroup},s.transaction.getData()&&L.default.createElement(kc,{isScCall:!v,data:s.transaction.getData().toString(),highlight:D})),m&&L.default.createElement("p",{className:U.errorMessage},m))),action:L.default.createElement("div",{className:U.buttonsWrapper},L.default.createElement("button",{id:"closeButton","data-testid":"closeButton",onClick:ge,className:U.cancelButton},ae?"Cancel":"Back"),L.default.createElement("button",{type:"button",className:U.signButton,id:"signBtn","data-testid":"signBtn",onClick:e,disabled:p},re))})};t();var Ym=`.dapp-core-component__signWithDeviceModalStyles__wallet-connect {
  color: inherit;
}

.dapp-core-component__signWithDeviceModalStyles__modal-container {
  color: inherit;
}

.dapp-core-component__signWithDeviceModalStyles__formGroup {
  color: inherit;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Ym));var ar={"wallet-connect":"dapp-core-component__signWithDeviceModalStyles__wallet-connect",walletConnect:"dapp-core-component__signWithDeviceModalStyles__wallet-connect","modal-container":"dapp-core-component__signWithDeviceModalStyles__modal-container",modalContainer:"dapp-core-component__signWithDeviceModalStyles__modal-container",formGroup:"dapp-core-component__signWithDeviceModalStyles__formGroup"};var Zm=({handleClose:e,error:o,className:n="dapp-device-modal",verifyReceiverScam:r=!0,title:p="Confirm transaction"})=>{let{onSignTransaction:s,onPrev:m,allTransactions:g,waitingForDevice:_,onAbort:u,isLastTransaction:h,currentStep:f,callbackRoute:y,currentTransaction:T}=Vt({onCancel:e,verifyReceiverScam:r}),v={wrapper:(0,ir.default)(ar.modalContainer,ar.walletConnect,n),container:(0,ir.default)(d.card,d.container),cardBody:d.cardBody};return vn.default.createElement(dc,{onClose:e,modalConfig:{modalDialogClassName:v.wrapper},visible:T!=null},vn.default.createElement("div",{className:v.cardBody},vn.default.createElement(Lc,{onSignTransaction:s,allTransactions:g,onPrev:m,waitingForDevice:_,currentStep:f,isLastTransaction:h,callbackRoute:y,currentTransaction:T,handleClose:u,error:o,title:p})))};0&&(module.exports={SignWithDeviceModal});
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
