// @elrondnetwork/dapp-core/__commonjs/UI/SignTransactionsModals/SignWithDeviceModal/SignStep.js
"use strict";var xi=Object.create;var Ke=Object.defineProperty,yi=Object.defineProperties,bi=Object.getOwnPropertyDescriptor,Si=Object.getOwnPropertyDescriptors,wi=Object.getOwnPropertyNames,so=Object.getOwnPropertySymbols,vi=Object.getPrototypeOf,qo=Object.prototype.hasOwnProperty,lt=Object.prototype.propertyIsEnumerable;var dt=(o,e,n)=>e in o?Ke(o,e,{enumerable:!0,configurable:!0,writable:!0,value:n}):o[e]=n,T=(o,e)=>{for(var n in e||(e={}))qo.call(e,n)&&dt(o,n,e[n]);if(so)for(var n of so(e))lt.call(e,n)&&dt(o,n,e[n]);return o},A=(o,e)=>yi(o,Si(e));var Ko=(o,e)=>{var n={};for(var r in o)qo.call(o,r)&&e.indexOf(r)<0&&(n[r]=o[r]);if(o!=null&&so)for(var r of so(o))e.indexOf(r)<0&&lt.call(o,r)&&(n[r]=o[r]);return n};var y=(o,e)=>()=>(o&&(e=o(o=0)),e);var mo=(o,e)=>()=>(e||o((e={exports:{}}).exports,e),e.exports),Te=(o,e)=>{for(var n in e)Ke(o,n,{get:e[n],enumerable:!0})},ut=(o,e,n,r)=>{if(e&&typeof e=="object"||typeof e=="function")for(let p of wi(e))!qo.call(o,p)&&p!==n&&Ke(o,p,{get:()=>e[p],enumerable:!(r=bi(e,p))||r.enumerable});return o};var x=(o,e,n)=>(n=o!=null?xi(vi(o)):{},ut(e||!o||!o.__esModule?Ke(n,"default",{value:o,enumerable:!0}):n,o)),Xe=o=>ut(Ke({},"__esModule",{value:!0}),o);var I=(o,e,n)=>new Promise((r,p)=>{var m=l=>{try{u(n.next(l))}catch(g){p(g)}},s=l=>{try{u(n.throw(l))}catch(g){p(g)}},u=l=>l.done?r(l.value):Promise.resolve(l.value).then(m,s);u((n=n.apply(o,e)).next())});var ht=mo(_o=>{"use strict";t();_o.byteLength=Ii;_o.toByteArray=Ei;_o.fromByteArray=Pi;var J=[],q=[],ki=typeof Uint8Array!="undefined"?Uint8Array:Array,Xo="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";for(xe=0,gt=Xo.length;xe<gt;++xe)J[xe]=Xo[xe],q[Xo.charCodeAt(xe)]=xe;var xe,gt;q["-".charCodeAt(0)]=62;q["_".charCodeAt(0)]=63;function ft(o){var e=o.length;if(e%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var n=o.indexOf("=");n===-1&&(n=e);var r=n===e?0:4-n%4;return[n,r]}function Ii(o){var e=ft(o),n=e[0],r=e[1];return(n+r)*3/4-r}function Ai(o,e,n){return(e+n)*3/4-n}function Ei(o){var e,n=ft(o),r=n[0],p=n[1],m=new ki(Ai(o,r,p)),s=0,u=p>0?r-4:r,l;for(l=0;l<u;l+=4)e=q[o.charCodeAt(l)]<<18|q[o.charCodeAt(l+1)]<<12|q[o.charCodeAt(l+2)]<<6|q[o.charCodeAt(l+3)],m[s++]=e>>16&255,m[s++]=e>>8&255,m[s++]=e&255;return p===2&&(e=q[o.charCodeAt(l)]<<2|q[o.charCodeAt(l+1)]>>4,m[s++]=e&255),p===1&&(e=q[o.charCodeAt(l)]<<10|q[o.charCodeAt(l+1)]<<4|q[o.charCodeAt(l+2)]>>2,m[s++]=e>>8&255,m[s++]=e&255),m}function Li(o){return J[o>>18&63]+J[o>>12&63]+J[o>>6&63]+J[o&63]}function Ci(o,e,n){for(var r,p=[],m=e;m<n;m+=3)r=(o[m]<<16&16711680)+(o[m+1]<<8&65280)+(o[m+2]&255),p.push(Li(r));return p.join("")}function Pi(o){for(var e,n=o.length,r=n%3,p=[],m=16383,s=0,u=n-r;s<u;s+=m)p.push(Ci(o,s,s+m>u?u:s+m));return r===1?(e=o[n-1],p.push(J[e>>2]+J[e<<4&63]+"==")):r===2&&(e=(o[n-2]<<8)+o[n-1],p.push(J[e>>10]+J[e>>4&63]+J[e<<2&63]+"=")),p.join("")}});var Tt=mo(Yo=>{t();Yo.read=function(o,e,n,r,p){var m,s,u=p*8-r-1,l=(1<<u)-1,g=l>>1,b=-7,h=n?p-1:0,w=n?-1:1,P=o[e+h];for(h+=w,m=P&(1<<-b)-1,P>>=-b,b+=u;b>0;m=m*256+o[e+h],h+=w,b-=8);for(s=m&(1<<-b)-1,m>>=-b,b+=r;b>0;s=s*256+o[e+h],h+=w,b-=8);if(m===0)m=1-g;else{if(m===l)return s?NaN:(P?-1:1)*(1/0);s=s+Math.pow(2,r),m=m-g}return(P?-1:1)*s*Math.pow(2,m-r)};Yo.write=function(o,e,n,r,p,m){var s,u,l,g=m*8-p-1,b=(1<<g)-1,h=b>>1,w=p===23?Math.pow(2,-24)-Math.pow(2,-77):0,P=r?0:m-1,D=r?1:-1,O=e<0||e===0&&1/e<0?1:0;for(e=Math.abs(e),isNaN(e)||e===1/0?(u=isNaN(e)?1:0,s=b):(s=Math.floor(Math.log(e)/Math.LN2),e*(l=Math.pow(2,-s))<1&&(s--,l*=2),s+h>=1?e+=w/l:e+=w*Math.pow(2,1-h),e*l>=2&&(s++,l/=2),s+h>=b?(u=0,s=b):s+h>=1?(u=(e*l-1)*Math.pow(2,p),s=s+h):(u=e*Math.pow(2,h-1)*Math.pow(2,p),s=0));p>=8;o[n+P]=u&255,P+=D,u/=256,p-=8);for(s=s<<p|u,g+=p;g>0;o[n+P]=s&255,P+=D,s/=256,g-=8);o[n+P-D]|=O*128}});var Pt=mo(Ne=>{"use strict";t();var Zo=ht(),De=Tt(),xt=typeof Symbol=="function"&&typeof Symbol.for=="function"?Symbol.for("nodejs.util.inspect.custom"):null;Ne.Buffer=_;Ne.SlowBuffer=Bi;Ne.INSPECT_MAX_BYTES=50;var lo=2147483647;Ne.kMaxLength=lo;_.TYPED_ARRAY_SUPPORT=Di();!_.TYPED_ARRAY_SUPPORT&&typeof console!="undefined"&&typeof console.error=="function"&&console.error("This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.");function Di(){try{var o=new Uint8Array(1),e={foo:function(){return 42}};return Object.setPrototypeOf(e,Uint8Array.prototype),Object.setPrototypeOf(o,e),o.foo()===42}catch(n){return!1}}Object.defineProperty(_.prototype,"parent",{enumerable:!0,get:function(){if(!!_.isBuffer(this))return this.buffer}});Object.defineProperty(_.prototype,"offset",{enumerable:!0,get:function(){if(!!_.isBuffer(this))return this.byteOffset}});function ie(o){if(o>lo)throw new RangeError('The value "'+o+'" is invalid for option "size"');var e=new Uint8Array(o);return Object.setPrototypeOf(e,_.prototype),e}function _(o,e,n){if(typeof o=="number"){if(typeof e=="string")throw new TypeError('The "string" argument must be of type string. Received type number');return on(o)}return St(o,e,n)}_.poolSize=8192;function St(o,e,n){if(typeof o=="string")return Ri(o,e);if(ArrayBuffer.isView(o))return Mi(o);if(o==null)throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof o);if(Q(o,ArrayBuffer)||o&&Q(o.buffer,ArrayBuffer)||typeof SharedArrayBuffer!="undefined"&&(Q(o,SharedArrayBuffer)||o&&Q(o.buffer,SharedArrayBuffer)))return Qo(o,e,n);if(typeof o=="number")throw new TypeError('The "value" argument must not be of type number. Received type number');var r=o.valueOf&&o.valueOf();if(r!=null&&r!==o)return _.from(r,e,n);var p=Fi(o);if(p)return p;if(typeof Symbol!="undefined"&&Symbol.toPrimitive!=null&&typeof o[Symbol.toPrimitive]=="function")return _.from(o[Symbol.toPrimitive]("string"),e,n);throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof o)}_.from=function(o,e,n){return St(o,e,n)};Object.setPrototypeOf(_.prototype,Uint8Array.prototype);Object.setPrototypeOf(_,Uint8Array);function wt(o){if(typeof o!="number")throw new TypeError('"size" argument must be of type number');if(o<0)throw new RangeError('The value "'+o+'" is invalid for option "size"')}function Ni(o,e,n){return wt(o),o<=0?ie(o):e!==void 0?typeof n=="string"?ie(o).fill(e,n):ie(o).fill(e):ie(o)}_.alloc=function(o,e,n){return Ni(o,e,n)};function on(o){return wt(o),ie(o<0?0:nn(o)|0)}_.allocUnsafe=function(o){return on(o)};_.allocUnsafeSlow=function(o){return on(o)};function Ri(o,e){if((typeof e!="string"||e==="")&&(e="utf8"),!_.isEncoding(e))throw new TypeError("Unknown encoding: "+e);var n=vt(o,e)|0,r=ie(n),p=r.write(o,e);return p!==n&&(r=r.slice(0,p)),r}function Jo(o){for(var e=o.length<0?0:nn(o.length)|0,n=ie(e),r=0;r<e;r+=1)n[r]=o[r]&255;return n}function Mi(o){if(Q(o,Uint8Array)){var e=new Uint8Array(o);return Qo(e.buffer,e.byteOffset,e.byteLength)}return Jo(o)}function Qo(o,e,n){if(e<0||o.byteLength<e)throw new RangeError('"offset" is outside of buffer bounds');if(o.byteLength<e+(n||0))throw new RangeError('"length" is outside of buffer bounds');var r;return e===void 0&&n===void 0?r=new Uint8Array(o):n===void 0?r=new Uint8Array(o,e):r=new Uint8Array(o,e,n),Object.setPrototypeOf(r,_.prototype),r}function Fi(o){if(_.isBuffer(o)){var e=nn(o.length)|0,n=ie(e);return n.length===0||o.copy(n,0,0,e),n}if(o.length!==void 0)return typeof o.length!="number"||tn(o.length)?ie(0):Jo(o);if(o.type==="Buffer"&&Array.isArray(o.data))return Jo(o.data)}function nn(o){if(o>=lo)throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x"+lo.toString(16)+" bytes");return o|0}function Bi(o){return+o!=o&&(o=0),_.alloc(+o)}_.isBuffer=function(e){return e!=null&&e._isBuffer===!0&&e!==_.prototype};_.compare=function(e,n){if(Q(e,Uint8Array)&&(e=_.from(e,e.offset,e.byteLength)),Q(n,Uint8Array)&&(n=_.from(n,n.offset,n.byteLength)),!_.isBuffer(e)||!_.isBuffer(n))throw new TypeError('The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array');if(e===n)return 0;for(var r=e.length,p=n.length,m=0,s=Math.min(r,p);m<s;++m)if(e[m]!==n[m]){r=e[m],p=n[m];break}return r<p?-1:p<r?1:0};_.isEncoding=function(e){switch(String(e).toLowerCase()){case"hex":case"utf8":case"utf-8":case"ascii":case"latin1":case"binary":case"base64":case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return!0;default:return!1}};_.concat=function(e,n){if(!Array.isArray(e))throw new TypeError('"list" argument must be an Array of Buffers');if(e.length===0)return _.alloc(0);var r;if(n===void 0)for(n=0,r=0;r<e.length;++r)n+=e[r].length;var p=_.allocUnsafe(n),m=0;for(r=0;r<e.length;++r){var s=e[r];if(Q(s,Uint8Array))m+s.length>p.length?_.from(s).copy(p,m):Uint8Array.prototype.set.call(p,s,m);else if(_.isBuffer(s))s.copy(p,m);else throw new TypeError('"list" argument must be an Array of Buffers');m+=s.length}return p};function vt(o,e){if(_.isBuffer(o))return o.length;if(ArrayBuffer.isView(o)||Q(o,ArrayBuffer))return o.byteLength;if(typeof o!="string")throw new TypeError('The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type '+typeof o);var n=o.length,r=arguments.length>2&&arguments[2]===!0;if(!r&&n===0)return 0;for(var p=!1;;)switch(e){case"ascii":case"latin1":case"binary":return n;case"utf8":case"utf-8":return en(o).length;case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return n*2;case"hex":return n>>>1;case"base64":return Ct(o).length;default:if(p)return r?-1:en(o).length;e=(""+e).toLowerCase(),p=!0}}_.byteLength=vt;function Oi(o,e,n){var r=!1;if((e===void 0||e<0)&&(e=0),e>this.length||((n===void 0||n>this.length)&&(n=this.length),n<=0)||(n>>>=0,e>>>=0,n<=e))return"";for(o||(o="utf8");;)switch(o){case"hex":return Ki(this,e,n);case"utf8":case"utf-8":return It(this,e,n);case"ascii":return zi(this,e,n);case"latin1":case"binary":return qi(this,e,n);case"base64":return Hi(this,e,n);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return Xi(this,e,n);default:if(r)throw new TypeError("Unknown encoding: "+o);o=(o+"").toLowerCase(),r=!0}}_.prototype._isBuffer=!0;function ye(o,e,n){var r=o[e];o[e]=o[n],o[n]=r}_.prototype.swap16=function(){var e=this.length;if(e%2!==0)throw new RangeError("Buffer size must be a multiple of 16-bits");for(var n=0;n<e;n+=2)ye(this,n,n+1);return this};_.prototype.swap32=function(){var e=this.length;if(e%4!==0)throw new RangeError("Buffer size must be a multiple of 32-bits");for(var n=0;n<e;n+=4)ye(this,n,n+3),ye(this,n+1,n+2);return this};_.prototype.swap64=function(){var e=this.length;if(e%8!==0)throw new RangeError("Buffer size must be a multiple of 64-bits");for(var n=0;n<e;n+=8)ye(this,n,n+7),ye(this,n+1,n+6),ye(this,n+2,n+5),ye(this,n+3,n+4);return this};_.prototype.toString=function(){var e=this.length;return e===0?"":arguments.length===0?It(this,0,e):Oi.apply(this,arguments)};_.prototype.toLocaleString=_.prototype.toString;_.prototype.equals=function(e){if(!_.isBuffer(e))throw new TypeError("Argument must be a Buffer");return this===e?!0:_.compare(this,e)===0};_.prototype.inspect=function(){var e="",n=Ne.INSPECT_MAX_BYTES;return e=this.toString("hex",0,n).replace(/(.{2})/g,"$1 ").trim(),this.length>n&&(e+=" ... "),"<Buffer "+e+">"};xt&&(_.prototype[xt]=_.prototype.inspect);_.prototype.compare=function(e,n,r,p,m){if(Q(e,Uint8Array)&&(e=_.from(e,e.offset,e.byteLength)),!_.isBuffer(e))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+typeof e);if(n===void 0&&(n=0),r===void 0&&(r=e?e.length:0),p===void 0&&(p=0),m===void 0&&(m=this.length),n<0||r>e.length||p<0||m>this.length)throw new RangeError("out of range index");if(p>=m&&n>=r)return 0;if(p>=m)return-1;if(n>=r)return 1;if(n>>>=0,r>>>=0,p>>>=0,m>>>=0,this===e)return 0;for(var s=m-p,u=r-n,l=Math.min(s,u),g=this.slice(p,m),b=e.slice(n,r),h=0;h<l;++h)if(g[h]!==b[h]){s=g[h],u=b[h];break}return s<u?-1:u<s?1:0};function kt(o,e,n,r,p){if(o.length===0)return-1;if(typeof n=="string"?(r=n,n=0):n>2147483647?n=2147483647:n<-2147483648&&(n=-2147483648),n=+n,tn(n)&&(n=p?0:o.length-1),n<0&&(n=o.length+n),n>=o.length){if(p)return-1;n=o.length-1}else if(n<0)if(p)n=0;else return-1;if(typeof e=="string"&&(e=_.from(e,r)),_.isBuffer(e))return e.length===0?-1:yt(o,e,n,r,p);if(typeof e=="number")return e=e&255,typeof Uint8Array.prototype.indexOf=="function"?p?Uint8Array.prototype.indexOf.call(o,e,n):Uint8Array.prototype.lastIndexOf.call(o,e,n):yt(o,[e],n,r,p);throw new TypeError("val must be string, number or Buffer")}function yt(o,e,n,r,p){var m=1,s=o.length,u=e.length;if(r!==void 0&&(r=String(r).toLowerCase(),r==="ucs2"||r==="ucs-2"||r==="utf16le"||r==="utf-16le")){if(o.length<2||e.length<2)return-1;m=2,s/=2,u/=2,n/=2}function l(P,D){return m===1?P[D]:P.readUInt16BE(D*m)}var g;if(p){var b=-1;for(g=n;g<s;g++)if(l(o,g)===l(e,b===-1?0:g-b)){if(b===-1&&(b=g),g-b+1===u)return b*m}else b!==-1&&(g-=g-b),b=-1}else for(n+u>s&&(n=s-u),g=n;g>=0;g--){for(var h=!0,w=0;w<u;w++)if(l(o,g+w)!==l(e,w)){h=!1;break}if(h)return g}return-1}_.prototype.includes=function(e,n,r){return this.indexOf(e,n,r)!==-1};_.prototype.indexOf=function(e,n,r){return kt(this,e,n,r,!0)};_.prototype.lastIndexOf=function(e,n,r){return kt(this,e,n,r,!1)};function Ui(o,e,n,r){n=Number(n)||0;var p=o.length-n;r?(r=Number(r),r>p&&(r=p)):r=p;var m=e.length;r>m/2&&(r=m/2);for(var s=0;s<r;++s){var u=parseInt(e.substr(s*2,2),16);if(tn(u))return s;o[n+s]=u}return s}function Wi(o,e,n,r){return uo(en(e,o.length-n),o,n,r)}function Gi(o,e,n,r){return uo(Ji(e),o,n,r)}function $i(o,e,n,r){return uo(Ct(e),o,n,r)}function Vi(o,e,n,r){return uo(Qi(e,o.length-n),o,n,r)}_.prototype.write=function(e,n,r,p){if(n===void 0)p="utf8",r=this.length,n=0;else if(r===void 0&&typeof n=="string")p=n,r=this.length,n=0;else if(isFinite(n))n=n>>>0,isFinite(r)?(r=r>>>0,p===void 0&&(p="utf8")):(p=r,r=void 0);else throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");var m=this.length-n;if((r===void 0||r>m)&&(r=m),e.length>0&&(r<0||n<0)||n>this.length)throw new RangeError("Attempt to write outside buffer bounds");p||(p="utf8");for(var s=!1;;)switch(p){case"hex":return Ui(this,e,n,r);case"utf8":case"utf-8":return Wi(this,e,n,r);case"ascii":case"latin1":case"binary":return Gi(this,e,n,r);case"base64":return $i(this,e,n,r);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return Vi(this,e,n,r);default:if(s)throw new TypeError("Unknown encoding: "+p);p=(""+p).toLowerCase(),s=!0}};_.prototype.toJSON=function(){return{type:"Buffer",data:Array.prototype.slice.call(this._arr||this,0)}};function Hi(o,e,n){return e===0&&n===o.length?Zo.fromByteArray(o):Zo.fromByteArray(o.slice(e,n))}function It(o,e,n){n=Math.min(o.length,n);for(var r=[],p=e;p<n;){var m=o[p],s=null,u=m>239?4:m>223?3:m>191?2:1;if(p+u<=n){var l,g,b,h;switch(u){case 1:m<128&&(s=m);break;case 2:l=o[p+1],(l&192)===128&&(h=(m&31)<<6|l&63,h>127&&(s=h));break;case 3:l=o[p+1],g=o[p+2],(l&192)===128&&(g&192)===128&&(h=(m&15)<<12|(l&63)<<6|g&63,h>2047&&(h<55296||h>57343)&&(s=h));break;case 4:l=o[p+1],g=o[p+2],b=o[p+3],(l&192)===128&&(g&192)===128&&(b&192)===128&&(h=(m&15)<<18|(l&63)<<12|(g&63)<<6|b&63,h>65535&&h<1114112&&(s=h))}}s===null?(s=65533,u=1):s>65535&&(s-=65536,r.push(s>>>10&1023|55296),s=56320|s&1023),r.push(s),p+=u}return ji(r)}var bt=4096;function ji(o){var e=o.length;if(e<=bt)return String.fromCharCode.apply(String,o);for(var n="",r=0;r<e;)n+=String.fromCharCode.apply(String,o.slice(r,r+=bt));return n}function zi(o,e,n){var r="";n=Math.min(o.length,n);for(var p=e;p<n;++p)r+=String.fromCharCode(o[p]&127);return r}function qi(o,e,n){var r="";n=Math.min(o.length,n);for(var p=e;p<n;++p)r+=String.fromCharCode(o[p]);return r}function Ki(o,e,n){var r=o.length;(!e||e<0)&&(e=0),(!n||n<0||n>r)&&(n=r);for(var p="",m=e;m<n;++m)p+=ec[o[m]];return p}function Xi(o,e,n){for(var r=o.slice(e,n),p="",m=0;m<r.length-1;m+=2)p+=String.fromCharCode(r[m]+r[m+1]*256);return p}_.prototype.slice=function(e,n){var r=this.length;e=~~e,n=n===void 0?r:~~n,e<0?(e+=r,e<0&&(e=0)):e>r&&(e=r),n<0?(n+=r,n<0&&(n=0)):n>r&&(n=r),n<e&&(n=e);var p=this.subarray(e,n);return Object.setPrototypeOf(p,_.prototype),p};function F(o,e,n){if(o%1!==0||o<0)throw new RangeError("offset is not uint");if(o+e>n)throw new RangeError("Trying to access beyond buffer length")}_.prototype.readUintLE=_.prototype.readUIntLE=function(e,n,r){e=e>>>0,n=n>>>0,r||F(e,n,this.length);for(var p=this[e],m=1,s=0;++s<n&&(m*=256);)p+=this[e+s]*m;return p};_.prototype.readUintBE=_.prototype.readUIntBE=function(e,n,r){e=e>>>0,n=n>>>0,r||F(e,n,this.length);for(var p=this[e+--n],m=1;n>0&&(m*=256);)p+=this[e+--n]*m;return p};_.prototype.readUint8=_.prototype.readUInt8=function(e,n){return e=e>>>0,n||F(e,1,this.length),this[e]};_.prototype.readUint16LE=_.prototype.readUInt16LE=function(e,n){return e=e>>>0,n||F(e,2,this.length),this[e]|this[e+1]<<8};_.prototype.readUint16BE=_.prototype.readUInt16BE=function(e,n){return e=e>>>0,n||F(e,2,this.length),this[e]<<8|this[e+1]};_.prototype.readUint32LE=_.prototype.readUInt32LE=function(e,n){return e=e>>>0,n||F(e,4,this.length),(this[e]|this[e+1]<<8|this[e+2]<<16)+this[e+3]*16777216};_.prototype.readUint32BE=_.prototype.readUInt32BE=function(e,n){return e=e>>>0,n||F(e,4,this.length),this[e]*16777216+(this[e+1]<<16|this[e+2]<<8|this[e+3])};_.prototype.readIntLE=function(e,n,r){e=e>>>0,n=n>>>0,r||F(e,n,this.length);for(var p=this[e],m=1,s=0;++s<n&&(m*=256);)p+=this[e+s]*m;return m*=128,p>=m&&(p-=Math.pow(2,8*n)),p};_.prototype.readIntBE=function(e,n,r){e=e>>>0,n=n>>>0,r||F(e,n,this.length);for(var p=n,m=1,s=this[e+--p];p>0&&(m*=256);)s+=this[e+--p]*m;return m*=128,s>=m&&(s-=Math.pow(2,8*n)),s};_.prototype.readInt8=function(e,n){return e=e>>>0,n||F(e,1,this.length),this[e]&128?(255-this[e]+1)*-1:this[e]};_.prototype.readInt16LE=function(e,n){e=e>>>0,n||F(e,2,this.length);var r=this[e]|this[e+1]<<8;return r&32768?r|4294901760:r};_.prototype.readInt16BE=function(e,n){e=e>>>0,n||F(e,2,this.length);var r=this[e+1]|this[e]<<8;return r&32768?r|4294901760:r};_.prototype.readInt32LE=function(e,n){return e=e>>>0,n||F(e,4,this.length),this[e]|this[e+1]<<8|this[e+2]<<16|this[e+3]<<24};_.prototype.readInt32BE=function(e,n){return e=e>>>0,n||F(e,4,this.length),this[e]<<24|this[e+1]<<16|this[e+2]<<8|this[e+3]};_.prototype.readFloatLE=function(e,n){return e=e>>>0,n||F(e,4,this.length),De.read(this,e,!0,23,4)};_.prototype.readFloatBE=function(e,n){return e=e>>>0,n||F(e,4,this.length),De.read(this,e,!1,23,4)};_.prototype.readDoubleLE=function(e,n){return e=e>>>0,n||F(e,8,this.length),De.read(this,e,!0,52,8)};_.prototype.readDoubleBE=function(e,n){return e=e>>>0,n||F(e,8,this.length),De.read(this,e,!1,52,8)};function H(o,e,n,r,p,m){if(!_.isBuffer(o))throw new TypeError('"buffer" argument must be a Buffer instance');if(e>p||e<m)throw new RangeError('"value" argument is out of bounds');if(n+r>o.length)throw new RangeError("Index out of range")}_.prototype.writeUintLE=_.prototype.writeUIntLE=function(e,n,r,p){if(e=+e,n=n>>>0,r=r>>>0,!p){var m=Math.pow(2,8*r)-1;H(this,e,n,r,m,0)}var s=1,u=0;for(this[n]=e&255;++u<r&&(s*=256);)this[n+u]=e/s&255;return n+r};_.prototype.writeUintBE=_.prototype.writeUIntBE=function(e,n,r,p){if(e=+e,n=n>>>0,r=r>>>0,!p){var m=Math.pow(2,8*r)-1;H(this,e,n,r,m,0)}var s=r-1,u=1;for(this[n+s]=e&255;--s>=0&&(u*=256);)this[n+s]=e/u&255;return n+r};_.prototype.writeUint8=_.prototype.writeUInt8=function(e,n,r){return e=+e,n=n>>>0,r||H(this,e,n,1,255,0),this[n]=e&255,n+1};_.prototype.writeUint16LE=_.prototype.writeUInt16LE=function(e,n,r){return e=+e,n=n>>>0,r||H(this,e,n,2,65535,0),this[n]=e&255,this[n+1]=e>>>8,n+2};_.prototype.writeUint16BE=_.prototype.writeUInt16BE=function(e,n,r){return e=+e,n=n>>>0,r||H(this,e,n,2,65535,0),this[n]=e>>>8,this[n+1]=e&255,n+2};_.prototype.writeUint32LE=_.prototype.writeUInt32LE=function(e,n,r){return e=+e,n=n>>>0,r||H(this,e,n,4,4294967295,0),this[n+3]=e>>>24,this[n+2]=e>>>16,this[n+1]=e>>>8,this[n]=e&255,n+4};_.prototype.writeUint32BE=_.prototype.writeUInt32BE=function(e,n,r){return e=+e,n=n>>>0,r||H(this,e,n,4,4294967295,0),this[n]=e>>>24,this[n+1]=e>>>16,this[n+2]=e>>>8,this[n+3]=e&255,n+4};_.prototype.writeIntLE=function(e,n,r,p){if(e=+e,n=n>>>0,!p){var m=Math.pow(2,8*r-1);H(this,e,n,r,m-1,-m)}var s=0,u=1,l=0;for(this[n]=e&255;++s<r&&(u*=256);)e<0&&l===0&&this[n+s-1]!==0&&(l=1),this[n+s]=(e/u>>0)-l&255;return n+r};_.prototype.writeIntBE=function(e,n,r,p){if(e=+e,n=n>>>0,!p){var m=Math.pow(2,8*r-1);H(this,e,n,r,m-1,-m)}var s=r-1,u=1,l=0;for(this[n+s]=e&255;--s>=0&&(u*=256);)e<0&&l===0&&this[n+s+1]!==0&&(l=1),this[n+s]=(e/u>>0)-l&255;return n+r};_.prototype.writeInt8=function(e,n,r){return e=+e,n=n>>>0,r||H(this,e,n,1,127,-128),e<0&&(e=255+e+1),this[n]=e&255,n+1};_.prototype.writeInt16LE=function(e,n,r){return e=+e,n=n>>>0,r||H(this,e,n,2,32767,-32768),this[n]=e&255,this[n+1]=e>>>8,n+2};_.prototype.writeInt16BE=function(e,n,r){return e=+e,n=n>>>0,r||H(this,e,n,2,32767,-32768),this[n]=e>>>8,this[n+1]=e&255,n+2};_.prototype.writeInt32LE=function(e,n,r){return e=+e,n=n>>>0,r||H(this,e,n,4,2147483647,-2147483648),this[n]=e&255,this[n+1]=e>>>8,this[n+2]=e>>>16,this[n+3]=e>>>24,n+4};_.prototype.writeInt32BE=function(e,n,r){return e=+e,n=n>>>0,r||H(this,e,n,4,2147483647,-2147483648),e<0&&(e=4294967295+e+1),this[n]=e>>>24,this[n+1]=e>>>16,this[n+2]=e>>>8,this[n+3]=e&255,n+4};function At(o,e,n,r,p,m){if(n+r>o.length)throw new RangeError("Index out of range");if(n<0)throw new RangeError("Index out of range")}function Et(o,e,n,r,p){return e=+e,n=n>>>0,p||At(o,e,n,4,34028234663852886e22,-34028234663852886e22),De.write(o,e,n,r,23,4),n+4}_.prototype.writeFloatLE=function(e,n,r){return Et(this,e,n,!0,r)};_.prototype.writeFloatBE=function(e,n,r){return Et(this,e,n,!1,r)};function Lt(o,e,n,r,p){return e=+e,n=n>>>0,p||At(o,e,n,8,17976931348623157e292,-17976931348623157e292),De.write(o,e,n,r,52,8),n+8}_.prototype.writeDoubleLE=function(e,n,r){return Lt(this,e,n,!0,r)};_.prototype.writeDoubleBE=function(e,n,r){return Lt(this,e,n,!1,r)};_.prototype.copy=function(e,n,r,p){if(!_.isBuffer(e))throw new TypeError("argument should be a Buffer");if(r||(r=0),!p&&p!==0&&(p=this.length),n>=e.length&&(n=e.length),n||(n=0),p>0&&p<r&&(p=r),p===r||e.length===0||this.length===0)return 0;if(n<0)throw new RangeError("targetStart out of bounds");if(r<0||r>=this.length)throw new RangeError("Index out of range");if(p<0)throw new RangeError("sourceEnd out of bounds");p>this.length&&(p=this.length),e.length-n<p-r&&(p=e.length-n+r);var m=p-r;return this===e&&typeof Uint8Array.prototype.copyWithin=="function"?this.copyWithin(n,r,p):Uint8Array.prototype.set.call(e,this.subarray(r,p),n),m};_.prototype.fill=function(e,n,r,p){if(typeof e=="string"){if(typeof n=="string"?(p=n,n=0,r=this.length):typeof r=="string"&&(p=r,r=this.length),p!==void 0&&typeof p!="string")throw new TypeError("encoding must be a string");if(typeof p=="string"&&!_.isEncoding(p))throw new TypeError("Unknown encoding: "+p);if(e.length===1){var m=e.charCodeAt(0);(p==="utf8"&&m<128||p==="latin1")&&(e=m)}}else typeof e=="number"?e=e&255:typeof e=="boolean"&&(e=Number(e));if(n<0||this.length<n||this.length<r)throw new RangeError("Out of range index");if(r<=n)return this;n=n>>>0,r=r===void 0?this.length:r>>>0,e||(e=0);var s;if(typeof e=="number")for(s=n;s<r;++s)this[s]=e;else{var u=_.isBuffer(e)?e:_.from(e,p),l=u.length;if(l===0)throw new TypeError('The value "'+e+'" is invalid for argument "value"');for(s=0;s<r-n;++s)this[s+n]=u[s%l]}return this};var Yi=/[^+/0-9A-Za-z-_]/g;function Zi(o){if(o=o.split("=")[0],o=o.trim().replace(Yi,""),o.length<2)return"";for(;o.length%4!==0;)o=o+"=";return o}function en(o,e){e=e||1/0;for(var n,r=o.length,p=null,m=[],s=0;s<r;++s){if(n=o.charCodeAt(s),n>55295&&n<57344){if(!p){if(n>56319){(e-=3)>-1&&m.push(239,191,189);continue}else if(s+1===r){(e-=3)>-1&&m.push(239,191,189);continue}p=n;continue}if(n<56320){(e-=3)>-1&&m.push(239,191,189),p=n;continue}n=(p-55296<<10|n-56320)+65536}else p&&(e-=3)>-1&&m.push(239,191,189);if(p=null,n<128){if((e-=1)<0)break;m.push(n)}else if(n<2048){if((e-=2)<0)break;m.push(n>>6|192,n&63|128)}else if(n<65536){if((e-=3)<0)break;m.push(n>>12|224,n>>6&63|128,n&63|128)}else if(n<1114112){if((e-=4)<0)break;m.push(n>>18|240,n>>12&63|128,n>>6&63|128,n&63|128)}else throw new Error("Invalid code point")}return m}function Ji(o){for(var e=[],n=0;n<o.length;++n)e.push(o.charCodeAt(n)&255);return e}function Qi(o,e){for(var n,r,p,m=[],s=0;s<o.length&&!((e-=2)<0);++s)n=o.charCodeAt(s),r=n>>8,p=n%256,m.push(p),m.push(r);return m}function Ct(o){return Zo.toByteArray(Zi(o))}function uo(o,e,n,r){for(var p=0;p<r&&!(p+n>=e.length||p>=o.length);++p)e[p+n]=o[p];return p}function Q(o,e){return o instanceof e||o!=null&&o.constructor!=null&&o.constructor.name!=null&&o.constructor.name===e.name}function tn(o){return o!==o}var ec=function(){for(var o="0123456789abcdef",e=new Array(256),n=0;n<16;++n)for(var r=n*16,p=0;p<16;++p)e[r+p]=o[n]+o[p];return e}()});var Ft=mo((Ks,Mt)=>{t();var M=Mt.exports={},ee,oe;function rn(){throw new Error("setTimeout has not been defined")}function an(){throw new Error("clearTimeout has not been defined")}(function(){try{typeof setTimeout=="function"?ee=setTimeout:ee=rn}catch(o){ee=rn}try{typeof clearTimeout=="function"?oe=clearTimeout:oe=an}catch(o){oe=an}})();function Dt(o){if(ee===setTimeout)return setTimeout(o,0);if((ee===rn||!ee)&&setTimeout)return ee=setTimeout,setTimeout(o,0);try{return ee(o,0)}catch(e){try{return ee.call(null,o,0)}catch(n){return ee.call(this,o,0)}}}function oc(o){if(oe===clearTimeout)return clearTimeout(o);if((oe===an||!oe)&&clearTimeout)return oe=clearTimeout,clearTimeout(o);try{return oe(o)}catch(e){try{return oe.call(null,o)}catch(n){return oe.call(this,o)}}}var ce=[],Re=!1,be,go=-1;function nc(){!Re||!be||(Re=!1,be.length?ce=be.concat(ce):go=-1,ce.length&&Nt())}function Nt(){if(!Re){var o=Dt(nc);Re=!0;for(var e=ce.length;e;){for(be=ce,ce=[];++go<e;)be&&be[go].run();go=-1,e=ce.length}be=null,Re=!1,oc(o)}}M.nextTick=function(o){var e=new Array(arguments.length-1);if(arguments.length>1)for(var n=1;n<arguments.length;n++)e[n-1]=arguments[n];ce.push(new Rt(o,e)),ce.length===1&&!Re&&Dt(Nt)};function Rt(o,e){this.fun=o,this.array=e}Rt.prototype.run=function(){this.fun.apply(null,this.array)};M.title="browser";M.browser=!0;M.env={};M.argv=[];M.version="";M.versions={};function pe(){}M.on=pe;M.addListener=pe;M.once=pe;M.off=pe;M.removeListener=pe;M.removeAllListeners=pe;M.emit=pe;M.prependListener=pe;M.prependOnceListener=pe;M.listeners=function(o){return[]};M.binding=function(o){throw new Error("process.binding is not supported")};M.cwd=function(){return"/"};M.chdir=function(o){throw new Error("process.chdir is not supported")};M.umask=function(){return 0}});var a,i,Xs,t=y(()=>{a=x(Pt()),i=x(Ft()),Xs=function(o){function e(){var r=this||self;return delete o.prototype.__magic__,r}if(typeof globalThis=="object")return globalThis;if(this)return e();o.defineProperty(o.prototype,"__magic__",{configurable:!0,get:e});var n=__magic__;return n}(Object)});var fo,cn=y(()=>{"use strict";t();fo=()=>Date.now()/1e3});var Bt=y(()=>{"use strict";t()});var Ot=y(()=>{"use strict";t()});var pn=y(()=>{"use strict";t();cn();Bt();Ot()});var mn={};Te(mn,{getItem:()=>rc,localStorageKeys:()=>ue,removeItem:()=>ac,setItem:()=>tc});var ue,sn,tc,rc,ac,Me=y(()=>{"use strict";t();pn();ue={loginExpiresAt:"dapp-core-login-expires-at",logoutEvent:"dapp-core-logout-event"},sn=typeof localStorage!="undefined",tc=({key:o,data:e,expires:n})=>{!sn||localStorage.setItem(String(o),JSON.stringify({expires:n,data:e}))},rc=o=>{if(!sn)return;let e=localStorage.getItem(String(o));if(!e)return null;let n=JSON.parse(e);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:fo()>=n.expires?(localStorage.removeItem(String(o)),null):n.data},ac=o=>{!sn||localStorage.removeItem(String(o))}});var _n={};Te(_n,{clear:()=>$t,getItem:()=>Wt,removeItem:()=>Gt,setItem:()=>Ut,storage:()=>ic});var Ut,Wt,Gt,$t,ic,Vt=y(()=>{"use strict";t();Ut=({key:o,data:e,expires:n})=>{sessionStorage.setItem(String(o),JSON.stringify({expires:n,data:e}))},Wt=o=>{let e=sessionStorage.getItem(String(o));if(!e)return null;let n=JSON.parse(e);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:Date.now()>=n.expires?(sessionStorage.removeItem(String(o)),null):n.data},Gt=o=>sessionStorage.removeItem(String(o)),$t=()=>sessionStorage.clear(),ic={setItem:Ut,getItem:Wt,removeItem:Gt,clear:$t}});var Se,Fe=y(()=>{"use strict";t();Me();Vt();Se={session:_n,local:mn}});function Ye(){return new Date().setHours(new Date().getHours()+24)}function Ze(o){Se.local.setItem({key:ue.loginExpiresAt,data:o,expires:o})}var dn=y(()=>{"use strict";t();Fe();Me()});var N=y(()=>{"use strict";t()});var Be=y(()=>{"use strict";t()});var Ht=y(()=>{"use strict";t()});var W=y(()=>{"use strict";t()});var jt=y(()=>{"use strict";t()});var zt=y(()=>{"use strict";t()});var qt=y(()=>{"use strict";t()});var Kt=y(()=>{"use strict";t()});var K=y(()=>{"use strict";t();Ht();W();N();jt();zt();qt();Kt()});var ln,Xt,wm,Yt,vm,Zt,km,Im,Am,Jt=y(()=>{"use strict";t();K();ln={devnet:{id:"devnet",chainId:"D",name:"Devnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://devnet-wallet.elrond.com",apiAddress:"https://devnet-api.elrond.com",explorerAddress:"http://devnet-explorer.elrond.com",apiTimeout:"4000"},testnet:{id:"testnet",chainId:"T",name:"Testnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://testnet-wallet.elrond.com",apiAddress:"https://testnet-api.elrond.com",explorerAddress:"http://testnet-explorer.elrond.com",apiTimeout:"4000"},mainnet:{id:"mainnet",chainId:"1",name:"Mainnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://wallet.elrond.com",apiAddress:"https://api.elrond.com",explorerAddress:"https://explorer.elrond.com",apiTimeout:"4000"}},{chainId:Xt,egldLabel:wm}=ln["devnet"],{chainId:Yt,egldLabel:vm}=ln["testnet"],{chainId:Zt,egldLabel:km}=ln["mainnet"],Im={["devnet"]:Xt,["testnet"]:Yt,["mainnet"]:Zt},Am={[Xt]:"devnet",[Yt]:"testnet",[Zt]:"mainnet"}});var un=y(()=>{"use strict";t()});var Qt=y(()=>{"use strict";t()});var gn,fn=y(()=>{"use strict";t();gn="transaction-status-toast-id"});var Je,ho,ne,To,er,xo,or,nr,we,E=y(()=>{"use strict";t();Be();Jt();un();Qt();fn();Je=5e4,ho=1e9,ne=18,To=4,er=1,xo="logout",or="login",nr="N/A",we="0"});var hn,G,se,j=y(()=>{"use strict";t();hn=require("@reduxjs/toolkit");E();G=(0,hn.createAction)(xo),se=(0,hn.createAction)(or,o=>({payload:o}))});var rr,tr,ar,jm,ir,cc,Qe,cr,pc,pr,sr,Tn=y(()=>{"use strict";t();rr=require("@reduxjs/toolkit");dn();N();j();tr={loginMethod:"",walletConnectLogin:null,ledgerLogin:null,tokenLogin:null,walletLogin:null,extensionLogin:null,isLoginSessionInvalid:!1},ar=(0,rr.createSlice)({name:"loginInfoSlice",initialState:tr,reducers:{setLoginMethod:(o,e)=>{o.loginMethod=e.payload},setTokenLogin:(o,e)=>{o.tokenLogin=e.payload},setTokenLoginSignature:(o,e)=>{(o==null?void 0:o.tokenLogin)!=null&&(o.tokenLogin.signature=e.payload)},setWalletLogin:(o,e)=>{o.walletLogin=e.payload},setWalletConnectLogin:(o,e)=>{o.walletConnectLogin=e.payload},setLedgerLogin:(o,e)=>{o.ledgerLogin=e.payload},invalidateLoginSession:o=>{o.isLoginSessionInvalid=!0}},extraReducers:o=>{o.addCase(G,()=>tr),o.addCase(se,(e,n)=>{e.loginMethod=n.payload.loginMethod,Ze(Ye())})}}),{setLoginMethod:jm,setWalletConnectLogin:ir,setLedgerLogin:cc,setTokenLogin:Qe,setTokenLoginSignature:cr,setWalletLogin:pc,invalidateLoginSession:pr}=ar.actions,sr=ar.reducer});var eo=y(()=>{"use strict";t()});var _r=y(()=>{"use strict";t();eo()});function xn(o){return o[Math.floor(Math.random()*o.length)]}var dr=y(()=>{"use strict";t()});var yn=y(()=>{"use strict";t();un()});var Oe=y(()=>{"use strict";t();_r();dr();yn()});var lr,ur,bn,sc,gr,i_,c_,fr,Sn=y(()=>{"use strict";t();lr=require("@reduxjs/toolkit"),ur=x(require("lodash.omit"));Oe();bn={id:"not-configured",chainId:"",name:"NOT CONFIGURED",egldLabel:"",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"",walletConnectBridgeAddress:"",walletConnectV2RelayAddress:"",walletConnectV2ProjectId:"",walletAddress:"",apiAddress:"",explorerAddress:"",apiTimeout:"4000"},sc={network:bn,chainID:"-1"},gr=(0,lr.createSlice)({name:"appConfig",initialState:sc,reducers:{initializeNetworkConfig:(o,e)=>{let n=xn(e.payload.walletConnectBridgeAddresses),r=xn(e.payload.walletConnectV2RelayAddresses),p=(0,ur.default)(e.payload,"walletConnectBridgeAddresses","walletConnectV2RelayAddresses");o.network=A(T(T({},o.network),p),{walletConnectBridgeAddress:n,walletConnectV2RelayAddress:r})},setChainID:(o,e)=>{o.chainID=e.payload}}}),{initializeNetworkConfig:i_,setChainID:c_}=gr.actions,fr=gr.reducer});var vn,hr,Tr,yo,wn,xr,kn,g_,In,f_,mc,_c,h_,T_,x_,dc,yr,An=y(()=>{"use strict";t();vn=require("@elrondnetwork/erdjs"),hr=require("@reduxjs/toolkit"),Tr=require("redux-persist");E();Fe();Me();j();yo={balance:"...",address:"",nonce:0,txCount:0,scrCount:0,claimableRewards:we},wn={address:"",websocketEvent:null,accounts:{"":yo},ledgerAccount:null,publicKey:"",walletConnectAccount:null,isAccountLoading:!0,accountLoadingError:null},xr=(0,hr.createSlice)({name:"accountInfoSlice",initialState:wn,reducers:{setAddress:(o,e)=>{let n=e.payload;o.address=n,o.publicKey=new vn.Address(n).hex()},setAccount:(o,e)=>{let n=o.address===e.payload.address;o.accounts={[o.address]:n?e.payload:yo},o.isAccountLoading=!1,o.accountLoadingError=null},setAccountNonce:(o,e)=>{let{address:n}=o;o.accounts[n].nonce=e.payload},setAccountShard:(o,e)=>{o.shard=e.payload},setLedgerAccount:(o,e)=>{o.ledgerAccount=e.payload},updateLedgerAccount:(o,e)=>{o.ledgerAccount!=null&&(o.ledgerAccount.index=e.payload.index,o.ledgerAccount.address=e.payload.address)},setWalletConnectAccount:(o,e)=>{o.walletConnectAccount=e.payload},setIsAccountLoading:(o,e)=>{o.isAccountLoading=e.payload,o.accountLoadingError=null},setAccountLoadingError:(o,e)=>{o.accountLoadingError=e.payload,o.isAccountLoading=!1},setWebsocketEvent:(o,e)=>{o.websocketEvent={timestamp:Date.now(),message:e.payload}}},extraReducers:o=>{o.addCase(G,()=>(Se.local.removeItem(ue.loginExpiresAt),wn)),o.addCase(se,(e,n)=>{let{address:r}=n.payload;e.address=r,e.publicKey=new vn.Address(r).hex()}),o.addCase(Tr.REHYDRATE,(e,n)=>{var g;if(!((g=n.payload)!=null&&g.account))return;let{account:r}=n.payload,{address:p,shard:m,accounts:s,publicKey:u}=r;e.address=p,e.shard=m;let l=s&&p in s;e.accounts=l?s:wn.accounts,e.publicKey=u})}}),{setAccount:kn,setAddress:g_,setAccountNonce:In,setAccountShard:f_,setLedgerAccount:mc,updateLedgerAccount:_c,setWalletConnectAccount:h_,setIsAccountLoading:T_,setAccountLoadingError:x_,setWebsocketEvent:dc}=xr.actions,yr=xr.reducer});function Ue(o){return o!=null&&(yc(o)||vc(o))}function We(o){return o!=null&&(bc(o)||kc(o))}function Ge(o){return o!=null&&(Sc(o)||Ic(o))}function En(o){return o!=null&&wc(o)}function yc(o){return o!=null&&lc.includes(o)}function bc(o){return o!=null&&uc.includes(o)}function Sc(o){return o!=null&&gc.includes(o)}function wc(o){return o!=null&&fc.includes(o)}function vc(o){return o!=null&&hc.includes(o)}function kc(o){return o!=null&&Tc.includes(o)}function Ic(o){return o!=null&&xc.includes(o)}var lc,uc,gc,fc,hc,Tc,xc,$e=y(()=>{"use strict";t();N();lc=["sent"],uc=["success"],gc=["fail","cancelled","timedOut"],fc=["timedOut"],hc=["pending"],Tc=["success"],xc=["fail","invalid"]});var br,Sr,no,Ac,wr,vr,kr,Ec,Ln,Lc,Cc,E_,Pc,Ir,Cn,bo,So=y(()=>{"use strict";t();br=require("@reduxjs/toolkit"),Sr=require("redux-persist");N();$e();j();no={signedTransactions:{},transactionsToSign:null,signTransactionsError:null,signTransactionsCancelMessage:null,customTransactionInformationForSessionId:{}},Ac={signWithoutSending:!1,sessionInformation:null,redirectAfterSign:!1},wr=(0,br.createSlice)({name:"transactionsSlice",initialState:no,reducers:{moveTransactionsToSignedState:(o,e)=>{var l,g;let{sessionId:n,transactions:r,errorMessage:p,status:m,redirectRoute:s}=e.payload,u=((l=o.customTransactionInformationForSessionId)==null?void 0:l[n])||Ac;o.signedTransactions[n]={transactions:r,status:m,errorMessage:p,redirectRoute:s,customTransactionInformation:u},((g=o==null?void 0:o.transactionsToSign)==null?void 0:g.sessionId)===n&&(o.transactionsToSign=no.transactionsToSign)},clearSignedTransaction:(o,e)=>{o.signedTransactions[e.payload]&&delete o.signedTransactions[e.payload]},clearTransactionToSign:o=>{o!=null&&o.transactionsToSign&&(o.transactionsToSign=null)},updateSignedTransaction:(o,e)=>{o.signedTransactions=T(T({},o.signedTransactions),e.payload)},updateSignedTransactions:(o,e)=>{let{sessionId:n,status:r,errorMessage:p,transactions:m}=e.payload;o.signedTransactions[n]!=null&&(o.signedTransactions[n].status=r,p!=null&&(o.signedTransactions[n].errorMessage=p),m!=null&&(o.signedTransactions[n].transactions=m))},updateSignedTransactionStatus:(o,e)=>{var u,l,g,b,h,w;let{sessionId:n,status:r,errorMessage:p,transactionHash:m}=e.payload,s=(l=(u=o.signedTransactions)==null?void 0:u[n])==null?void 0:l.transactions;if(s!=null){o.signedTransactions[n].transactions=s.map(O=>O.hash===m?A(T({},O),{status:r,errorMessage:p}):O);let P=(b=(g=o.signedTransactions[n])==null?void 0:g.transactions)==null?void 0:b.every(O=>We(O.status)),D=(w=(h=o.signedTransactions[n])==null?void 0:h.transactions)==null?void 0:w.every(O=>Ge(O.status));P&&(o.signedTransactions[n].status="success"),D&&(o.signedTransactions[n].status="fail")}},setTransactionsToSign:(o,e)=>{o.transactionsToSign=e.payload;let{sessionId:n,customTransactionInformation:r}=e.payload;o.customTransactionInformationForSessionId[n]=r,o.signTransactionsError=null},clearAllTransactionsToSign:o=>{o.transactionsToSign=no.transactionsToSign,o.signTransactionsError=null},clearAllSignedTransactions:o=>{o.signedTransactions=no.signedTransactions},setSignTransactionsError:(o,e)=>{o.signTransactionsError=e.payload},setSignTransactionsCancelMessage:(o,e)=>{o.signTransactionsCancelMessage=e.payload}},extraReducers:o=>{o.addCase(G,()=>no),o.addCase(Sr.REHYDRATE,(e,n)=>{var s;if(!((s=n.payload)!=null&&s.transactions))return;let{signedTransactions:r,customTransactionInformationForSessionId:p}=n.payload.transactions,m=Object.entries(r).reduce((u,[l,g])=>{let b=new Date(l),h=new Date;return h.setHours(h.getHours()+5),h-b>0||(u[l]=g),u},{});p!=null&&(e.customTransactionInformationForSessionId=p),r!=null&&(e.signedTransactions=m)})}}),{updateSignedTransactionStatus:vr,updateSignedTransactions:kr,setTransactionsToSign:Ec,clearAllTransactionsToSign:Ln,clearAllSignedTransactions:Lc,clearSignedTransaction:Cc,clearTransactionToSign:E_,setSignTransactionsError:Pc,setSignTransactionsCancelMessage:Ir,moveTransactionsToSignedState:Cn}=wr.actions,bo=wr.reducer});var Ar,Dn,Nn,Rn,Dc,Pn,Er,D_,Nc,Lr,wo,vo=y(()=>{"use strict";t();Ar=require("@reduxjs/toolkit");j();Dn="Transaction failed",Nn="Transaction successful",Rn="Processing transaction",Dc="Transaction submitted",Pn={},Er=(0,Ar.createSlice)({name:"transactionsInfo",initialState:Pn,reducers:{setTransactionsDisplayInfo(o,e){let{sessionId:n,transactionsDisplayInfo:r}=e.payload;n!=null&&(o[n]={errorMessage:(r==null?void 0:r.errorMessage)||Dn,successMessage:(r==null?void 0:r.successMessage)||Nn,processingMessage:(r==null?void 0:r.processingMessage)||Rn,submittedMessage:(r==null?void 0:r.submittedMessage)||Dc,transactionDuration:r==null?void 0:r.transactionDuration})},clearTransactionsInfoForSessionId(o,e){e.payload!=null&&delete o[e.payload]},clearTransactionsInfo:()=>Pn},extraReducers:o=>{o.addCase(G,()=>Pn)}}),{clearTransactionsInfo:D_,setTransactionsDisplayInfo:Nc,clearTransactionsInfoForSessionId:Lr}=Er.actions,wo=Er.reducer});var Pr,Cr,Dr,F_,Rc,B_,Mc,Nr,Mn=y(()=>{"use strict";t();Pr=require("@reduxjs/toolkit");j();Cr={},Dr=(0,Pr.createSlice)({name:"modalsSlice",initialState:Cr,reducers:{setTxSubmittedModal:(o,e)=>{o.txSubmittedModal=e.payload},setNotificationModal:(o,e)=>{o.notificationModal=e.payload},clearTxSubmittedModal:o=>{o.txSubmittedModal=void 0},clearNotificationModal:o=>{o.notificationModal=void 0}},extraReducers:o=>{o.addCase(G,()=>Cr)}}),{setTxSubmittedModal:F_,setNotificationModal:Rc,clearTxSubmittedModal:B_,clearNotificationModal:Mc}=Dr.actions,Nr=Dr.reducer});var Mr,Rr,Fr,Fc,Br,j_,z_,q_,K_,ko,Io=y(()=>{"use strict";t();Mr=require("@reduxjs/toolkit");fn();K();cn();j();Rr={customToasts:[],transactionToasts:[],failTransactionToast:null},Fr=(0,Mr.createSlice)({name:"toastsSlice",initialState:Rr,reducers:{addCustomToast:(o,e)=>{o.customToasts.push(A(T({},e.payload),{type:"custom",toastId:e.payload.toastId||`custom-toast-${o.customToasts.length+1}`}))},removeCustomToast:(o,e)=>{o.customToasts=o.customToasts.filter(n=>n.toastId!==e.payload)},addTransactionToast:(o,e)=>{o.transactionToasts.push({type:"transaction",startTimestamp:fo(),toastId:e.payload||`custom-toast-${o.transactionToasts.length+1}`})},removeTransactionToast:(o,e)=>{o.transactionToasts=o.transactionToasts.filter(n=>n.toastId!==e.payload)},addFailTransactionToast:(o,e)=>{o.failTransactionToast=A(T({},e.payload),{toastId:gn})},removeFailTransactionToast:o=>{o.failTransactionToast=null}},extraReducers:o=>{o.addCase(G,()=>Rr)}}),{addCustomToast:Fc,removeCustomToast:Br,addTransactionToast:j_,removeTransactionToast:z_,addFailTransactionToast:q_,removeFailTransactionToast:K_}=Fr.actions,ko=Fr.reducer});var L=y(()=>{"use strict";t();Tn();Sn();An();So();vo();Mn();Io()});var Fn=y(()=>{"use strict";t()});var Gr,Wr,$r,ud,gd,fd,hd,Td,xd,Vr,Hr=y(()=>{"use strict";t();Gr=require("@reduxjs/toolkit");j();Wr={visible:!1,modal:{showHeader:!0,showFooter:!1,headerText:"",footerText:""}},$r=(0,Gr.createSlice)({name:"dappModalsSlice",initialState:Wr,reducers:{setDappModalVisibility:(o,e)=>{o.visible=e.payload},setDappModalHeaderVisibility:(o,e)=>{o.modal.showHeader=e.payload},setDappModalFooterVisibility:(o,e)=>{o.modal.showFooter=e.payload},setDappModalHeaderText:(o,e)=>{o.modal.headerText=e.payload},setDappModalFooterText:(o,e)=>{o.modal.footerText=e.payload},setDappModalConfig:(o,e)=>{o.modal=T({},e.payload)}},extraReducers:o=>{o.addCase(G,()=>Wr)}}),{setDappModalVisibility:ud,setDappModalHeaderVisibility:gd,setDappModalFooterVisibility:fd,setDappModalHeaderText:hd,setDappModalFooterText:Td,setDappModalConfig:xd}=$r.actions,Vr=$r.reducer});var jr,Wc,Gc,Ve,Eo=y(()=>{"use strict";t();jr=require("@reduxjs/toolkit");Fn();An();Hr();Tn();Mn();Sn();Io();vo();So();Wc={["account"]:yr,["networkConfig"]:fr,["loginInfo"]:sr,["modals"]:Nr,["dappModal"]:Vr,["toasts"]:ko,["transactions"]:bo,["transactionsInfo"]:wo},Gc=(o={})=>(0,jr.combineReducers)(T(T({},Wc),o)),Ve=Gc});var Kr={};Te(Kr,{default:()=>Kc});function Bn(o,e=[]){return{key:o,version:1,storage:qr.default,blacklist:e}}var ke,zr,qr,$c,Vc,Hc,jc,zc,qc,Kc,Xr=y(()=>{"use strict";t();ke=require("redux-persist"),zr=x(require("redux-persist/lib/storage")),qr=x(require("redux-persist/lib/storage/session"));Eo();L();Io();vo();So();Fn();$c={2:o=>A(T({},o),{networkConfig:bn})};Vc=Bn("dapp-core-transactionsInfo"),Hc=Bn("dapp-core-transactions",["transactionsToSign"]),jc=Bn("dapp-core-toasts"),zc={key:"dapp-core-store",version:2,storage:zr.default,whitelist:["account","loginInfo","modals","networkConfig"],migrate:(0,ke.createMigrate)($c,{debug:!1})},qc={["toasts"]:(0,ke.persistReducer)(jc,ko),["transactions"]:(0,ke.persistReducer)(Hc,bo),["transactionsInfo"]:(0,ke.persistReducer)(Vc,wo)},Kc=(0,ke.persistReducer)(zc,Ve(qc))});var Yr={};Te(Yr,{default:()=>Xc});var Xc,Zr=y(()=>{"use strict";t();Eo();Xc=Ve()});var Jr={};Te(Jr,{default:()=>Zc});var Y,Yc,Zc,Qr=y(()=>{"use strict";t();Y=require("redux-persist"),Yc=[Y.FLUSH,Y.REHYDRATE,Y.PAUSE,Y.PERSIST,Y.PURGE,Y.REGISTER],Zc=Yc});var na={};Te(na,{default:()=>oa});function oa(o){return(0,ea.persistStore)(o)}var ea,ta=y(()=>{"use strict";t();ea=require("redux-persist")});var Vs={};Te(Vs,{SignStep:()=>$s});module.exports=Xe(Vs);t();var v=x(require("react")),_i=require("@elrondnetwork/erdjs/out"),Vo=require("@fortawesome/free-solid-svg-icons"),di=require("@fortawesome/react-fontawesome");t();t();t();var sa=require("react"),He=require("react-redux");t();var On=require("@reduxjs/toolkit"),ca=require("react-redux/es/utils/Subscription");L();t();var Or=x(require("lodash.throttle"));E();L();dn();Fe();Me();var Bc=[xo],Oc=(0,Or.default)(()=>{Ze(Ye())},5e3),Ur=o=>e=>n=>{var l;if(Bc.includes(n.type))return e(n);let r=o.getState(),p=Se.local.getItem(ue.loginExpiresAt);if(!((l=r==null?void 0:r.loginInfo)==null?void 0:l.isLoggedIn))return e(n);if(p==null)return Ze(Ye());let s=Date.now();return p-s<0?setTimeout(()=>I(void 0,null,function*(){console.log("session expired");try{o.dispatch(pr())}catch(g){console.error("error logging out",g)}}),1e3):(Oc(),e(n))};t();t();function Ao(){return typeof sessionStorage!="undefined"}var ra=Ao()?(Xr(),Xe(Kr)).default:(Zr(),Xe(Yr)).default,aa=Ao()?(Qr(),Xe(Jr)).default:[],ia=Ao()?(ta(),Xe(na)).default:o=>o;Eo();var k=(0,On.configureStore)({reducer:ra,middleware:o=>o({serializableCheck:{ignoredActions:[...aa,In.type,kn.type],ignoredPaths:["payload.nonce","account.account.nonce","providers.accountProvider"]}}).concat(Ur)}),pa=(0,ca.createSubscription)(k),Hd=ia(k),jd=(0,On.configureStore)({reducer:Ve});var Jc={store:k,subscription:pa},Un=(0,sa.createContext)(Jc),Xd=(0,He.createStoreHook)(Un),Z=(0,He.createDispatchHook)(Un),C=(0,He.createSelectorHook)(Un);t();t();L();t();var ma=x(require("lodash.isequal")),Lo=require("reselect"),f=(0,Lo.createSelectorCreator)(Lo.defaultMemoize,ma.default);var me=o=>o.account,je=f(me,o=>o.address),to=f(me,je,(o,e)=>e in o.accounts?o.accounts[e]:yo),Qc=f(me,to,(o,e)=>{let p=o,{accounts:n}=p,r=Ko(p,["accounts"]);return A(T({},r),{account:e})}),ep=f(to,o=>o.balance),op=f(to,o=>{var e;return((e=o==null?void 0:o.nonce)==null?void 0:e.valueOf())||0}),el=f(me,o=>o.shard),np=f(me,o=>o.ledgerAccount),ol=f(me,o=>o.walletConnectAccount),nl=f(me,o=>o.isAccountLoading),tl=f(me,o=>o.accountLoadingError),_a=f(me,o=>o.websocketEvent);t();var _e=o=>o.networkConfig,Co=f(_e,o=>o.chainID),tp=f(_e,o=>o.network.walletConnectBridgeAddress),rp=f(_e,o=>o.network.walletConnectV2RelayAddress),ap=f(_e,o=>o.network.walletConnectV2ProjectId),da=f(_e,o=>o.network.walletConnectDeepLink),Ie=f(_e,o=>o.network),la=f(Ie,o=>o.apiAddress),ua=f(Ie,o=>o.explorerAddress),Wn=f(Ie,o=>o.egldLabel);t();N();var Ae=o=>o.loginInfo,ml=f(Ae,o=>o.loginMethod),Po=f(Ae,je,(o,e)=>o.loginMethod!=""&&Boolean(e)),_l=f(Ae,o=>o.walletConnectLogin),dl=f(Ae,o=>o.ledgerLogin),ip=f(Ae,o=>o.walletLogin),ll=f(Ae,o=>o.isLoginSessionInvalid);t();t();var z=require("@elrondnetwork/erdjs");E();t();function Gn(o){try{return a.Buffer.from(o,"base64").toString()===atob(o)}catch(e){return!1}}function Do(o){var p,m,s;let{data:e}=o,n=Gn(e!=null?e:"")?z.TransactionPayload.fromEncoded(e):new z.TransactionPayload(e),r=new z.Transaction(T({value:o.value.valueOf(),data:n,nonce:o.nonce.valueOf(),receiver:new z.Address(o.receiver),sender:new z.Address(o.sender),gasLimit:(p=o.gasLimit.valueOf())!=null?p:Je,gasPrice:(m=o.gasPrice.valueOf())!=null?m:ho,chainID:o.chainID.valueOf(),version:new z.TransactionVersion((s=o.version)!=null?s:er)},o.options?{options:new z.TransactionOptions(o.options)}:{}));return r.applySignature({hex:()=>o.signature||""},new z.Address(o.sender)),r}t();$e();t();var cp=x(require("bignumber.js"));K();t();t();function $n(o){for(let e=0;e<o.length;e++)if(o.charCodeAt(e)>127)return!1;return!0}function ro(o){let e=o;try{let n=a.Buffer.from(o,"hex").toString();$n(n)&&n.length>1&&(e=n)}catch(n){}return e}t();t();var sp=require("@elrondnetwork/erdjs"),mp=x(require("bignumber.js"));N();t();var pp=require("@elrondnetwork/erdjs");t();function Hn({tokenId:o,erdLabel:e}){return Boolean(o&&o!==e)}t();t();var dp=require("@elrondnetwork/erdjs/out");t();E();t();var fa=require("@elrondnetwork/erdjs");K();t();t();t();t();N();t();t();t();t();t();t();t();var jn="tokens";var zn="collections";t();t();t();t();W();t();t();W();t();t();t();var fp=x(require("bignumber.js"));E();t();var ya=require("@elrondnetwork/erdjs"),ge=x(require("bignumber.js"));E();t();var xa=x(require("bignumber.js")),qn=(o,e=!0)=>{let n=String(o);if(!n.match(/^[-]?\d+$/))return!1;let r=new xa.default(n),p=e?0:-1;return r.toString(10)===n&&r.comparedTo(0)>=p};t();function de(o){return{if:function(e){return e?{then:n=>n instanceof Function?de(n(o)):de(n)}:{then:()=>de(o)}},then:e=>e instanceof Function?de(e(o)):de(e),valueOf:function(){return o}}}ge.default.config({ROUNDING_MODE:ge.default.ROUND_FLOOR});function Ee({input:o,decimals:e=ne,digits:n=To,showLastNonZeroDecimal:r=!0,showIsLessThanDecimalsLabel:p=!1,addCommas:m=!1}){if(!qn(o,!1))throw new Error("Invalid input");let s=new ge.default(o).isNegative(),u=o;return s&&(u=o.substring(1)),de(u).then(()=>ya.TokenPayment.fungibleFromBigInteger("",u,e).toRationalNumber()).then(l=>{let g=new ge.default(l);if(g.isZero())return we;let b=g.toString(10),[h,w]=b.split("."),P=new ge.default(w||0),D=de(0).if(Boolean(w&&r)).then(()=>Math.max(w.length,n)).if(P.isZero()&&!r).then(0).if(Boolean(w&&!r)).then(()=>Math.min(w.length,n)).valueOf(),O=w&&n>=1&&n<=w.length&&P.isGreaterThan(0)&&new ge.default(w.substring(0,n)).isZero(),Ho=g.toFormat(D);return de(b).if(m).then(Ho).if(Boolean(O)).then(re=>{let fe=new ge.default(h).isZero(),[V,he]=re.split("."),Pe=new Array(n-1).fill(0),jo=[...Pe,0].join(""),ae=[...Pe,1].join("");return fe?p?`<${V}.${ae}`:r?`${V}.${he}`:V:`${V}.${jo}`}).if(Boolean(!O&&w)).then(re=>{let[fe]=re.split("."),V=w.substring(0,D);if(r){let he=n-V.length;if(he>0){let Pe=Array(he).fill(0).join("");return V=`${V}${Pe}`,`${fe}.${V}`}return re}return V?`${fe}.${V}`:fe}).valueOf()}).if(s).then(l=>`-${l}`).valueOf()}t();t();t();E();t();t();t();t();E();t();E();t();var hp=require("@elrondnetwork/erdjs");E();t();var Ro=require("@elrondnetwork/erdjs"),yp=require("@elrondnetwork/erdjs"),bp=require("@elrondnetwork/erdjs-network-providers");E();t();t();var Tp=x(require("bignumber.js"));E();t();E();t();var Kn={isEsdt:!1,isNft:!1,isEgld:!1};function Xn(o){let e=o==null?void 0:o.split("-").length;return e===2?A(T({},Kn),{isEsdt:!0}):e===3?A(T({},Kn),{isNft:!0}):A(T({},Kn),{isEgld:!0})}t();t();t();t();E();t();W();var wp=["reDelegateRewards","claimRewards","unBond"],vp=["wrapEgld","unwrapEgld"],kp=["unStake"],Ip=["unDelegate"];t();t();t();E();t();var Lp=x(require("bignumber.js"));t();t();t();W();t();var Cp=x(require("bignumber.js"));t();t();t();W();t();W();t();var Np=x(require("bignumber.js"));E();W();t();W();t();W();t();var Rp=x(require("bignumber.js"));t();N();t();N();t();t();var Mp=require("react");W();var Z0=["create","localMint","ESDTLocalMint","addQuantity","burn","localBurn","ESDTLocalBurn","wipe","writeLog","signalError"];t();var Fp=x(require("anchorme"));t();var wa=require("react");K();t();t();W();t();var Bp=require("@elrondnetwork/erdjs/out"),Op=x(require("bignumber.js"));W();t();W();t();t();N();var Eh=[{label:"Raw",value:"raw"},{label:"Text",value:"text"},{label:"Decimal",value:"decimal"},{label:"Smart",value:"smart"}];t();var va=require("react");K();t();t();t();t();t();E();var Fo=o=>o.transactions,ze=f(Fo,o=>o.signedTransactions),Wp=f(Fo,o=>o.signTransactionsError),Gp=f(Fo,o=>o.signTransactionsCancelMessage),Bo=o=>e=>Object.entries(e).reduce((n,[r,p])=>(o(p.status)&&(n[r]=p),n),{}),ka=f(ze,Bo(Ue)),Ia=f(ze,Bo(We)),Aa=f(ze,Bo(Ge)),$p=f(ze,Bo(En)),Zn=f(Fo,o=>{var e;return(o==null?void 0:o.transactionsToSign)==null?null:A(T({},o.transactionsToSign),{transactions:((e=o==null?void 0:o.transactionsToSign)==null?void 0:e.transactions.map(n=>Do(n)))||[]})}),Vp=f(ze,(o,e)=>e,(o,e)=>e!=null?(o==null?void 0:o[e])||{}:{});t();L();var Ea={errorMessage:Dn,successMessage:Nn,processingMessage:Rn},Hp=o=>o.transactionsInfo,jp=f(Hp,(o,e)=>e,(o,e)=>e!=null&&(o==null?void 0:o[Number(e)])||Ea);t();var La=o=>o.modals,kT=f(La,o=>o.txSubmittedModal),zp=f(La,o=>o.notificationModal);t();var Jn=o=>o.toasts,ET=f(Jn,o=>o.customToasts),LT=f(Jn,o=>o.transactionToasts),CT=f(Jn,o=>o.failTransactionToast);var ao=()=>C(_e);t();var Ca=require("react");t();L();t();t();t();var Kp=require("react"),Xp=require("@elrondnetwork/erdjs-web-wallet-provider"),Yp=x(require("qs"));E();L();N();t();var nt=require("react"),is=require("@elrondnetwork/erdjs-extension-provider");E();t();t();t();t();var Zp=require("@elrondnetwork/erdjs-extension-provider"),Jp=require("@elrondnetwork/erdjs-hw-provider"),Da=require("@elrondnetwork/erdjs-wallet-connect-provider"),Qp=require("@elrondnetwork/erdjs-web-wallet-provider");E();N();var le=o=>`Unable to perform ${o}, Provider not initialized`,Qn=class{init(){return Promise.resolve(!1)}login(e){throw new Error(le(`login with options: ${e}`))}logout(e){throw new Error(le(`logout with options: ${e}`))}getAddress(){throw new Error(le("getAddress"))}isInitialized(){return!1}isConnected(){return Promise.resolve(!1)}sendTransaction(e,n){throw new Error(le(`sendTransaction with transactions: ${e} options: ${n}`))}signTransaction(e,n){throw new Error(le(`signTransaction with transactions: ${e} options: ${n}`))}signTransactions(e,n){throw new Error(le(`signTransactions with transactions: ${e} options: ${n}`))}signMessage(e,n){throw new Error(le(`signTransactions with ${e} and options ${n}`))}sendCustomMessage({method:e,params:n}){throw new Error(le(`sendCustomMessage with method: ${e} params: ${n}`))}sendCustomRequest(e){throw new Error(le(`sendSessionEvent with options: ${e}`))}ping(){return Promise.resolve(!1)}},es=new Qn;L();N();t();t();j();K();t();t();t();N();t();t();t();t();var os=x(require("axios"));t();t();t();L();t();t();t();L();t();var Ma=require("@elrondnetwork/erdjs");t();j();K();eo();Fe();Me();t();t();t();t();L();t();t();t();t();t();function Uo(){return Wn(k.getState())}Fe();eo();pn();t();t();t();t();t();var ns=x(require("axios"));t();t();var ts=x(require("platform"));t();t();t();var Ba=x(require("axios")),Oa=x(require("swr"));var cs=o=>Ba.default.get(o).then(e=>e.data);function Ua({tokenId:o}){var w;let{network:e}=ao(),{isEsdt:n,isNft:r}=Xn(o),p=n?jn:zn,m=o;if(r){let[P,D]=o.split("-");m=`${P}-${D}`}let{data:s,error:u}=(0,Oa.default)(Boolean(m)?`${e.apiAddress}/${p}/${m}`:null,cs);if(!m)return{tokenDecimals:Number(e.decimals),tokenLabel:"",tokenAvatar:""};let l=s?s==null?void 0:s.decimals:Number(e.decimals),g=s?s==null?void 0:s.name:"",b=s?`${(w=s==null?void 0:s.assets)==null?void 0:w.svgUrl}`:"",h=s==null?void 0:s.assets;return{tokenDecimals:l,tokenLabel:g,tokenAvatar:b,assets:h,error:u}}t();t();var ps=x(require("axios"));t();t();var Ga=require("react");t();var Wa=require("react");yn();L();K();eo();t();Be();L();t();t();t();t();t();$e();t();t();t();var ms=x(require("axios"));t();var _s=x(require("axios"));t();var ls=x(require("axios"));t();t();var us=x(require("axios"));t();var gs=x(require("axios"));t();L();N();t();$e();t();L();N();t();L();K();t();var ws=require("react");t();t();t();L();t();t();var ys=x(require("bignumber.js"));t();var hs=`.dapp-core-component__main__link-style {
  color: #1392ff;
}

.dapp-core-component__main__link-second-style {
  color: #9dabbd;
}
.dapp-core-component__main__link-second-style:hover {
  color: #1392ff;
}
.dapp-core-component__main__link-second-style:hover svg {
  color: #1392ff;
}

.dapp-core-component__main__wrapper {
  min-height: 100vh;
}

.dapp-core-component__main__opacity-6 {
  opacity: 0.6;
}

.dapp-core-component__main__qr-code-svg-container {
  width: 15rem;
  height: 15rem;
}

.dapp-core-component__main__pair-list {
  max-height: 20rem;
  overflow-y: auto;
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__pair-image {
  width: 4rem;
  overflow: hidden;
  height: auto;
  min-width: 2rem;
}

.dapp-core-component__main__pair-remove {
  padding: 0 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

.dapp-core-component__main__dapp-icon {
  padding: 5px;
  border-radius: 50%;
  background-color: #fafafa;
  width: 4.6428571429rem;
  height: 4.6428571429rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}
.dapp-core-component__main__dapp-icon.dapp-core-component__main__icon-medium {
  width: 80px;
  height: 80px;
}

.dapp-core-component__main__centering {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dapp-core-component__main__border-n {
  border: none !important;
}

.dapp-core-component__main__t-shadow {
  text-shadow: 0px 4px 7px rgba(0, 0, 0, 0.3);
}

.dapp-core-component__main__btn {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.04) !important;
}

.dapp-core-component__main__media {
  display: flex;
  align-items: flex-start;
}

.dapp-core-component__main__media-body {
  flex: 1;
}

.dapp-core-component__main__connect-btns .dapp-core-component__main__btn {
  display: flex;
  height: 48px;
  justify-content: center;
  align-items: center;
  color: #fff !important;
  background-color: #1392ff;
  padding: 15px 20px 15px 16px;
  border-radius: 6px;
  gap: 8px;
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__btn:hover, .dapp-core-component__main__connect-btns .dapp-core-component__main__btn:active {
  background-color: #1392ff;
  background-image: linear-gradient(rgba(0, 0, 0, 0.1) 0 0);
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__info {
  color: #1392ff;
  margin-left: 48px;
}

.dapp-core-component__main__btn {
  transition: none !important;
}

.dapp-core-component__main__trust-badge {
  width: 20px;
  height: 20px;
}

/* Navbar
  -------------------------------------------------- */
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name {
  display: flex;
  align-items: center;
  font-size: 1.7142857143rem;
  font-weight: 400;
  color: #9ba5b4;
  margin-left: 1.6428571429rem;
}
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name:before {
  content: "";
  display: block;
  width: 1px;
  height: 3rem;
  margin-right: 1.7142857143rem;
}

/* Dashboard
  -------------------------------------------------- */
.dapp-core-component__main__action-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  margin: 0 7px;
}
.dapp-core-component__main__action-btn button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #fff;
}
.dapp-core-component__main__action-btn button:hover {
  opacity: 0.9;
}

.dapp-core-component__main__transactions .dapp-core-component__main__transaction-icon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table td {
  white-space: nowrap;
  vertical-align: middle;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table tr:last-of-type td {
  border-bottom: 1px solid #dee2e6;
}

/* Loader, Transaction Success/Fail
  -------------------------------------------------- */
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div {
  position: absolute;
  top: 27px;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.1);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.dapp-core-component__main__page-state .dapp-core-component__main__bg-blue .dapp-core-component__main__lds-ellipsis div {
  background: rgba(255, 255, 255, 0.4);
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(1) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis1 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(2) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(3) {
  left: 26px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(4) {
  left: 45px;
  animation: dapp-core-component__main__lds-ellipsis3 0.6s infinite;
}
@keyframes dapp-core-component__main__lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(19px, 0);
  }
}

.dapp-core-component__main__btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
}
.dapp-core-component__main__btn .dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn .dapp-core-component__main__disabled {
  opacity: 0.65;
  cursor: default;
}
.dapp-core-component__main__btn:hover {
  color: #212529;
  text-decoration: none;
}
.dapp-core-component__main__btn:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn:disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn-primary {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary:hover {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-primary:focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary:hover {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-secondary:focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-warning {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning:hover {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning:focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-dark {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark:hover {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
}
.dapp-core-component__main__btn-dark:focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-link {
  font-weight: 400;
  color: #007bff;
  text-decoration: none;
}
.dapp-core-component__main__btn-link:hover {
  color: #0056b3;
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:focus, .dapp-core-component__main__btn-link.dapp-core-component__main__focus {
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:disabled, .dapp-core-component__main__btn-link.dapp-core-component__main__disabled {
  color: #6c757d;
  pointer-events: none;
}

.dapp-core-component__main__btn-light {
  color: #212529;
  background-color: #fafafa;
  border-color: #fafafa;
}
.dapp-core-component__main__btn-light .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}
.dapp-core-component__main__btn-light .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #f8f9fa;
  border-color: #f8f9fa;
}
.dapp-core-component__main__btn-light:hover {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
}
.dapp-core-component__main__btn-light:focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .dapp-core-component__main__badge {
    transition: none;
  }
}
a.dapp-core-component__main__badge:hover, a.dapp-core-component__main__badge:focus {
  text-decoration: none;
}

.dapp-core-component__main__badge:empty {
  display: none;
}

.dapp-core-component__main__btn .dapp-core-component__main__badge {
  position: relative;
  top: -1px;
}

.dapp-core-component__main__badge-pill {
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.dapp-core-component__main__badge-primary {
  color: #fff;
  background-color: #007bff;
}

a.dapp-core-component__main__badge-primary:hover, a.dapp-core-component__main__badge-primary:focus {
  color: #fff;
  background-color: #0062cc;
}

a.dapp-core-component__main__badge-primary:focus, a.dapp-core-component__main__badge-primary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.5);
}

.dapp-core-component__main__badge-secondary {
  color: #fff;
  background-color: #6c757d;
}

a.dapp-core-component__main__badge-secondary:hover, a.dapp-core-component__main__badge-secondary:focus {
  color: #fff;
  background-color: #545b62;
}

a.dapp-core-component__main__badge-secondary:focus, a.dapp-core-component__main__badge-secondary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108, 117, 125, 0.5);
}

.dapp-core-component__main__badge-success {
  color: #fff;
  background-color: #28a745;
}

a.dapp-core-component__main__badge-success:hover, a.dapp-core-component__main__badge-success:focus {
  color: #fff;
  background-color: #1e7e34;
}

a.dapp-core-component__main__badge-success:focus, a.dapp-core-component__main__badge-success.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
}

.dapp-core-component__main__badge-info {
  color: #fff;
  background-color: #17a2b8;
}

a.dapp-core-component__main__badge-info:hover, a.dapp-core-component__main__badge-info:focus {
  color: #fff;
  background-color: #117a8b;
}

a.dapp-core-component__main__badge-info:focus, a.dapp-core-component__main__badge-info.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(23, 162, 184, 0.5);
}

.dapp-core-component__main__badge-warning {
  color: #212529;
  background-color: #ffc107;
}

a.dapp-core-component__main__badge-warning:hover, a.dapp-core-component__main__badge-warning:focus {
  color: #212529;
  background-color: #d39e00;
}

a.dapp-core-component__main__badge-warning:focus, a.dapp-core-component__main__badge-warning.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 193, 7, 0.5);
}

.dapp-core-component__main__badge-danger {
  color: #fff;
  background-color: #dc3545;
}

a.dapp-core-component__main__badge-danger:hover, a.dapp-core-component__main__badge-danger:focus {
  color: #fff;
  background-color: #bd2130;
}

a.dapp-core-component__main__badge-danger:focus, a.dapp-core-component__main__badge-danger.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(220, 53, 69, 0.5);
}

.dapp-core-component__main__badge-light {
  color: #212529;
  background-color: #f8f9fa;
}

a.dapp-core-component__main__badge-light:hover, a.dapp-core-component__main__badge-light:focus {
  color: #212529;
  background-color: #dae0e5;
}

a.dapp-core-component__main__badge-light:focus, a.dapp-core-component__main__badge-light.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(248, 249, 250, 0.5);
}

.dapp-core-component__main__badge-dark {
  color: #fff;
  background-color: #343a40;
}

a.dapp-core-component__main__badge-dark:hover, a.dapp-core-component__main__badge-dark:focus {
  color: #fff;
  background-color: #1d2124;
}

a.dapp-core-component__main__badge-dark:focus, a.dapp-core-component__main__badge-dark.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}

.dapp-core-component__main__card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 5px;
}
.dapp-core-component__main__card > hr {
  margin-right: 0;
  margin-left: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group {
  border-top: inherit;
  border-bottom: inherit;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:first-child {
  border-top-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:last-child {
  border-bottom-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__card-header + .dapp-core-component__main__list-group,
.dapp-core-component__main__card > .dapp-core-component__main__list-group + .dapp-core-component__main__card-footer {
  border-top: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header {
  border: 0;
  padding: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header:first-child {
  border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item {
  border-bottom: 1px solid #dee2e6;
  padding: 1.1rem 1rem;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item h6,
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item .dapp-core-component__main__h6 {
  line-height: 2.5rem;
  letter-spacing: unset;
  margin: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item.dapp-core-component__main__compact {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fafafa;
}

.dapp-core-component__main__card-body {
  flex: 1 1 auto;
  min-height: 1px;
  padding: 1.25rem;
}

.dapp-core-component__main__card-title {
  margin-bottom: 0.75rem;
}

.dapp-core-component__main__card-subtitle {
  margin-top: -0.375rem;
  margin-bottom: 0;
}

.dapp-core-component__main__card-text:last-child {
  margin-bottom: 0;
}

.dapp-core-component__main__card-link:hover {
  text-decoration: none;
}
.dapp-core-component__main__card-link + .dapp-core-component__main__card-link {
  margin-left: 1.25rem;
}

.dapp-core-component__main__card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: #fff;
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__card-footer {
  padding: 0.75rem 1.25rem;
  background-color: #fff;
  border-top: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__text {
  color: #9dabbd;
}

.dapp-core-component__main__text-white {
  color: #fff;
}

.dapp-core-component__main__text-monospace {
  font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-size-xl .dapp-core-component__main__trim {
  max-width: 13rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim {
  max-width: 10rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-only-sm .dapp-core-component__main__trim {
  max-width: none;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-left {
  text-align: left;
}

.dapp-core-component__main__text-right {
  text-align: right;
}

.dapp-core-component__main__text-center {
  text-align: center;
}

.dapp-core-component__main__text-primary {
  color: #1b46c2;
}

.dapp-core-component__main__text-secondary {
  color: #6c757d;
}

.dapp-core-component__main__text-primary-highlight {
  color: #1b46c2;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-success {
  color: #28a745;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-body {
  color: #212529;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-black-50 {
  color: rgba(0, 0, 0, 0.5);
}

.dapp-core-component__main__text-white-50 {
  color: rgba(255, 255, 255, 0.5);
}

@media (min-width: 992px) {
  .dapp-core-component__main__text-lg-right {
    text-align: right;
  }
}

.dapp-core-component__main__text-hide {
  font: 0/0 a;
  color: rgba(0, 0, 0, 0);
  text-shadow: none;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
}

.dapp-core-component__main__text-decoration-none {
  text-decoration: none;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__text-reset {
  color: inherit;
}

.dapp-core-component__main__text-lowercase {
  text-transform: lowercase;
}

.dapp-core-component__main__text-uppercase {
  text-transform: uppercase;
}

.dapp-core-component__main__text-capitalize {
  text-transform: capitalize;
}

.dapp-core-component__main__text-danger {
  color: #d9534f;
}

.dapp-core-component__main__text-warning {
  color: #f0ad4e;
}

.dapp-core-component__main__trim-fs-sm {
  color: inherit;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__m-0 {
  margin: 0;
}

.dapp-core-component__main__mt-0 {
  margin-top: 0;
}

.dapp-core-component__main__my-0 {
  margin-top: 0;
  margin-bottom: 0;
}

.dapp-core-component__main__mr-0 {
  margin-right: 0;
}

.dapp-core-component__main__mx-0 {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__mb-0 {
  margin-bottom: 0;
}

.dapp-core-component__main__my-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.dapp-core-component__main__ml-0 {
  margin-left: 0;
}

.dapp-core-component__main__mx-0 {
  margin-left: 0;
  margin-right: 0;
}

.dapp-core-component__main__m-1 {
  margin: 0.25rem;
}

.dapp-core-component__main__mt-1 {
  margin-top: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__mr-1 {
  margin-right: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.dapp-core-component__main__mb-1 {
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-bottom: 0.25rem;
  margin-top: 0.25rem;
}

.dapp-core-component__main__ml-1 {
  margin-left: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-right: 0.25rem;
  margin-left: 0.25rem;
}

.dapp-core-component__main__m-2 {
  margin: 0.5rem;
}

.dapp-core-component__main__mt-2 {
  margin-top: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-bottom: 0.5rem;
  margin-top: 0.5rem;
}

.dapp-core-component__main__mr-2 {
  margin-right: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__mb-2 {
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__ml-2 {
  margin-left: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__m-3 {
  margin: 1rem;
}

.dapp-core-component__main__mt-3,
.dapp-core-component__main__my-3 {
  margin-top: 1rem;
}

.dapp-core-component__main__mr-3 {
  margin-right: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__mb-3 {
  margin-bottom: 1rem;
}

.dapp-core-component__main__my-3 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.dapp-core-component__main__ml-3 {
  margin-left: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__m-4 {
  margin: 1.5rem;
}

.dapp-core-component__main__mt-4 {
  margin-top: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__mr-4 {
  margin-right: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-right: 1.5rem;
  margin-left: 1.5rem;
}

.dapp-core-component__main__mb-4 {
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.dapp-core-component__main__ml-4 {
  margin-left: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}

.dapp-core-component__main__m-5 {
  margin: 3rem;
}

.dapp-core-component__main__mt-5 {
  margin-top: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__mr-5 {
  margin-right: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-right: 3rem;
  margin-left: 3rem;
}

.dapp-core-component__main__mb-5 {
  margin-bottom: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__ml-5 {
  margin-left: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-left: 3rem;
  margin-right: 3rem;
}

.dapp-core-component__main__p-0 {
  padding: 0;
}

.dapp-core-component__main__pt-0 {
  padding-top: 0;
}

.dapp-core-component__main__py-0 {
  padding-top: 0;
  padding-bottom: 0;
}

.dapp-core-component__main__pr-0 {
  padding-right: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__pb-0 {
  padding-bottom: 0;
}

.dapp-core-component__main__py-0 {
  padding-bottom: 0;
  padding-top: 0;
}

.dapp-core-component__main__pl-0 {
  padding-left: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__p-1 {
  padding: 0.25rem;
}

.dapp-core-component__main__pt-1 {
  padding-top: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pr-1 {
  padding-right: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-right: 0.25rem;
  padding-left: 0.25rem;
}

.dapp-core-component__main__pb-1 {
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pl-1 {
  padding-left: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

.dapp-core-component__main__p-2 {
  padding: 0.5rem;
}

.dapp-core-component__main__pt-2 {
  padding-top: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.dapp-core-component__main__pr-2 {
  padding-right: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__pb-2 {
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__pl-2 {
  padding-left: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__p-3 {
  padding: 1rem;
}

.dapp-core-component__main__pt-3 {
  padding-top: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.dapp-core-component__main__pr-3 {
  padding-right: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__pb-3 {
  padding-bottom: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.dapp-core-component__main__pl-3 {
  padding-left: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__p-4 {
  padding: 1.5rem;
}

.dapp-core-component__main__pt-4 {
  padding-top: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pr-4 {
  padding-right: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.dapp-core-component__main__pb-4 {
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pl-4 {
  padding-left: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
}

.dapp-core-component__main__p-5 {
  padding: 3rem;
}

.dapp-core-component__main__pt-5 {
  padding-top: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-bottom: 3rem;
  padding-top: 3rem;
}

.dapp-core-component__main__pr-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__pb-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.dapp-core-component__main__pl-5 {
  padding-left: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__m-n1 {
  margin: -0.25rem;
}

.dapp-core-component__main__mt-n1 {
  margin-top: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__mr-n1 {
  margin-right: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__mb-n1 {
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-bottom: -0.25rem;
  margin-top: -0.25rem;
}

.dapp-core-component__main__ml-n1 {
  margin-left: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__m-n2 {
  margin: -0.5rem;
}

.dapp-core-component__main__mt-n2 {
  margin-top: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-top: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mr-n2 {
  margin-right: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mb-n2 {
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-bottom: -0.5rem;
  margin-top: -0.5rem;
}

.dapp-core-component__main__ml-n2 {
  margin-left: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.dapp-core-component__main__m-n3 {
  margin: -1rem;
}

.dapp-core-component__main__mt-n3 {
  margin-top: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-top: -1rem;
  margin-bottom: -1rem;
}

.dapp-core-component__main__mr-n3 {
  margin-right: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-right: -1rem;
  margin-left: -1rem;
}

.dapp-core-component__main__mb-n3 {
  margin-bottom: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-bottom: -1rem;
  margin-top: -1rem;
}

.dapp-core-component__main__ml-n3 {
  margin-left: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-left: -1rem;
  margin-right: -1rem;
}

.dapp-core-component__main__m-n4 {
  margin: -1.5rem;
}

.dapp-core-component__main__mt-n4 {
  margin-top: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-top: -1.5rem;
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__mr-n4 {
  margin-right: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__mb-n4 {
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-bottom: -1.5rem;
  margin-top: -1.5rem;
}

.dapp-core-component__main__ml-n4 {
  margin-left: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__m-n5 {
  margin: -3rem;
}

.dapp-core-component__main__mt-n5 {
  margin-top: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-bottom: -3rem;
  margin-top: -3rem;
}

.dapp-core-component__main__mr-n5 {
  margin-right: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-left: -3rem;
  margin-right: -3rem;
}

.dapp-core-component__main__mb-n5 {
  margin-bottom: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-top: -3rem;
  margin-bottom: -3rem;
}

.dapp-core-component__main__ml-n5 {
  margin-left: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-right: -3rem;
  margin-left: -3rem;
}

.dapp-core-component__main__m-auto {
  margin: auto;
}

.dapp-core-component__main__mt-auto {
  margin-top: auto;
}

.dapp-core-component__main__my-auto {
  margin-bottom: auto;
  margin-top: auto;
}

.dapp-core-component__main__mr-auto {
  margin-right: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__mb-auto {
  margin-bottom: auto;
}

.dapp-core-component__main__my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.dapp-core-component__main__ml-auto {
  margin-left: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-1 {
    margin-right: 0.25rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-4 {
    margin-right: 1.5rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mx-lg-4 {
    margin-left: 1.5rem;
    margin-right: 1.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__mb-xl-2 {
    margin-bottom: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__pr-sm-2 {
    padding-right: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__px-sm-2 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__pr-xl-0,
.dapp-core-component__main__px-xl-0 {
    padding-right: 0;
  }
}

.dapp-core-component__main__col {
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

.dapp-core-component__main__col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__col-xl,
.dapp-core-component__main__col-xl-auto,
.dapp-core-component__main__col-xl-12,
.dapp-core-component__main__col-xl-11,
.dapp-core-component__main__col-xl-10,
.dapp-core-component__main__col-xl-9,
.dapp-core-component__main__col-xl-8,
.dapp-core-component__main__col-xl-7,
.dapp-core-component__main__col-xl-6,
.dapp-core-component__main__col-xl-5,
.dapp-core-component__main__col-xl-4,
.dapp-core-component__main__col-xl-3,
.dapp-core-component__main__col-xl-2,
.dapp-core-component__main__col-xl-1,
.dapp-core-component__main__col-lg,
.dapp-core-component__main__col-lg-auto,
.dapp-core-component__main__col-lg-12,
.dapp-core-component__main__col-lg-11,
.dapp-core-component__main__col-lg-10,
.dapp-core-component__main__col-lg-9,
.dapp-core-component__main__col-lg-8,
.dapp-core-component__main__col-lg-7,
.dapp-core-component__main__col-lg-6,
.dapp-core-component__main__col-lg-5,
.dapp-core-component__main__col-lg-4,
.dapp-core-component__main__col-lg-3,
.dapp-core-component__main__col-lg-2,
.dapp-core-component__main__col-lg-1,
.dapp-core-component__main__col-md,
.dapp-core-component__main__col-md-auto,
.dapp-core-component__main__col-md-12,
.dapp-core-component__main__col-md-11,
.dapp-core-component__main__col-md-10,
.dapp-core-component__main__col-md-9,
.dapp-core-component__main__col-md-8,
.dapp-core-component__main__col-md-7,
.dapp-core-component__main__col-md-6,
.dapp-core-component__main__col-md-5,
.dapp-core-component__main__col-md-4,
.dapp-core-component__main__col-md-3,
.dapp-core-component__main__col-md-2,
.dapp-core-component__main__col-md-1,
.dapp-core-component__main__col-sm,
.dapp-core-component__main__col-sm-auto,
.dapp-core-component__main__col-sm-12,
.dapp-core-component__main__col-sm-11,
.dapp-core-component__main__col-sm-10,
.dapp-core-component__main__col-sm-9,
.dapp-core-component__main__col-sm-8,
.dapp-core-component__main__col-sm-7,
.dapp-core-component__main__col-sm-6,
.dapp-core-component__main__col-sm-5,
.dapp-core-component__main__col-sm-4,
.dapp-core-component__main__col-sm-3,
.dapp-core-component__main__col-sm-2,
.dapp-core-component__main__col-sm-1,
.dapp-core-component__main__col,
.dapp-core-component__main__col-auto,
.dapp-core-component__main__col-12,
.dapp-core-component__main__col-11,
.dapp-core-component__main__col-10,
.dapp-core-component__main__col-9,
.dapp-core-component__main__col-8,
.dapp-core-component__main__col-7,
.dapp-core-component__main__col-6,
.dapp-core-component__main__col-5,
.dapp-core-component__main__col-4,
.dapp-core-component__main__col-3,
.dapp-core-component__main__col-2,
.dapp-core-component__main__col-1 {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
}

.dapp-core-component__main__col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

@media (min-width: 576px) {
  .dapp-core-component__main__col-sm-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-sm-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-sm-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-sm-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-sm-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-sm-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-sm-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-sm-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-sm-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-sm-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-sm-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-sm-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 768px) {
  .dapp-core-component__main__col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__col-xl-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-xl-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-xl-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-xl-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-xl-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-xl-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-xl-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-xl-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-xl-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-xl-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-xl-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-xl-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
.dapp-core-component__main__flex-row {
  flex-direction: row;
}

.dapp-core-component__main__flex-column {
  flex-direction: column;
}

.dapp-core-component__main__flex-row-reverse {
  flex-direction: row-reverse;
}

.dapp-core-component__main__flex-column-reverse {
  flex-direction: column-reverse;
}

.dapp-core-component__main__flex-wrap {
  flex-wrap: wrap;
}

.dapp-core-component__main__flex-nowrap {
  flex-wrap: nowrap;
}

.dapp-core-component__main__flex-wrap-reverse {
  flex-wrap: wrap-reverse;
}

.dapp-core-component__main__flex-fill {
  flex: 1 1 auto;
}

.dapp-core-component__main__flex-grow-0 {
  flex-grow: 0;
}

.dapp-core-component__main__flex-grow-1 {
  flex-grow: 1;
}

.dapp-core-component__main__flex-shrink-0 {
  flex-shrink: 0;
}

.dapp-core-component__main__flex-shrink-1 {
  flex-shrink: 1;
}

.dapp-core-component__main__justify-content-start {
  justify-content: flex-start;
}

.dapp-core-component__main__justify-content-end {
  justify-content: flex-end;
}

.dapp-core-component__main__justify-content-center {
  justify-content: center;
}

.dapp-core-component__main__justify-content-between {
  justify-content: space-between;
}

.dapp-core-component__main__justify-content-around {
  justify-content: space-around;
}

.dapp-core-component__main__align-items-start {
  align-items: flex-start;
}

.dapp-core-component__main__align-items-end {
  align-items: flex-end;
}

.dapp-core-component__main__align-items-center {
  align-items: center;
}

.dapp-core-component__main__align-items-baseline {
  align-items: baseline;
}

.dapp-core-component__main__align-items-stretch {
  align-items: stretch;
}

.dapp-core-component__main__align-content-start {
  align-content: flex-start;
}

.dapp-core-component__main__align-content-end {
  align-content: flex-end;
}

.dapp-core-component__main__align-content-center {
  align-content: center;
}

.dapp-core-component__main__align-content-between {
  align-content: space-between;
}

.dapp-core-component__main__align-content-around {
  align-content: space-around;
}

.dapp-core-component__main__align-content-stretch {
  align-content: stretch;
}

.dapp-core-component__main__align-self-auto {
  align-self: auto;
}

.dapp-core-component__main__align-self-start {
  align-self: flex-start;
}

.dapp-core-component__main__align-self-end {
  align-self: flex-end;
}

.dapp-core-component__main__align-self-center {
  align-self: center;
}

.dapp-core-component__main__align-self-baseline {
  align-self: baseline;
}

.dapp-core-component__main__align-self-stretch {
  align-self: stretch;
}

@media (min-width: 576px) {
  .dapp-core-component__main__flex-sm-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-wrap {
    flex-wrap: wrap;
  }
}

.dapp-core-component__main__form-check {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: auto;
  padding-left: 0;
}

.dapp-core-component__main__form-check-label {
  margin-bottom: 0;
}

.dapp-core-component__main__form-check-input {
  position: absolute;
  margin-top: 0.3rem;
  margin-left: -1.25rem;
}
.dapp-core-component__main__form-check-input[disabled] ~ .dapp-core-component__main__form-check-label, .dapp-core-component__main__form-check-input:disabled ~ .dapp-core-component__main__form-check-label {
  color: #6c757d;
}

.dapp-core-component__main__form-group {
  display: flex;
  flex: 0 0 auto;
  flex-flow: row wrap;
  align-items: center;
  margin-bottom: 0;
}

.dapp-core-component__main__form-control {
  display: block;
  width: 100%;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.dapp-core-component__main__form-label {
  padding-top: calc(0.375rem + 1px);
  padding-bottom: calc(0.375rem + 1px);
  margin-bottom: 0;
  font-size: inherit;
  line-height: 1.5;
}

input,
button,
select,
optgroup,
textarea {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.dapp-core-component__main__form-group {
  margin-bottom: 1rem;
}

.dapp-core-component__main__form-text {
  display: block;
  margin-top: 0.25rem;
  color: inherit;
}

.dapp-core-component__main__border {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__border-top {
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__border-right {
  border-right: 1px solid #dee2e6;
}

.dapp-core-component__main__border-bottom {
  border-bottom: 1px solid #dee2e6;
}

.dapp-core-component__main__border-left {
  border-left: 1px solid #dee2e6;
}

.dapp-core-component__main__border-0 {
  border: 0;
}

.dapp-core-component__main__border-top-0 {
  border-top: 0;
}

.dapp-core-component__main__border-right-0 {
  border-right: 0;
}

.dapp-core-component__main__border-bottom-0 {
  border-bottom: 0;
}

.dapp-core-component__main__border-left-0 {
  border-left: 0;
}

.dapp-core-component__main__rounded-sm {
  border-radius: 0.2rem;
}

.dapp-core-component__main__rounded {
  border-radius: 0.25rem;
}

.dapp-core-component__main__rounded-top {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-right {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-bottom {
  border-bottom-right-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-left {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-lg {
  border-radius: 0.3rem;
}

.dapp-core-component__main__rounded-circle {
  border-radius: 50%;
}

.dapp-core-component__main__rounded-pill {
  border-radius: 50rem;
}

.dapp-core-component__main__rounded-0 {
  border-radius: 0;
}

.dapp-core-component__main__h1,
.dapp-core-component__main__h2,
.dapp-core-component__main__h3,
.dapp-core-component__main__h4,
.dapp-core-component__main__h5,
.dapp-core-component__main__h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2;
}

.dapp-core-component__main__h1 {
  font-size: 2.5rem;
}

.dapp-core-component__main__h2 {
  font-size: 2rem;
}

.dapp-core-component__main__h3 {
  font-size: 1.75rem;
}

.dapp-core-component__main__h4 {
  font-size: 1.5rem;
}

.dapp-core-component__main__h5 {
  font-size: 1.25rem;
}

.dapp-core-component__main__h6 {
  font-size: 1rem;
}

.dapp-core-component__main__lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

.dapp-core-component__main__bg-warning,
.dapp-core-component__main__warning {
  background-color: #f0ad4e;
}

.dapp-core-component__main__bg-danger,
.dapp-core-component__main__danger {
  background-color: #d9534f;
}

.dapp-core-component__main__bg-success,
.dapp-core-component__main__success {
  background-color: #5cb85c;
}

.dapp-core-component__main__position-relative {
  position: relative;
}

.dapp-core-component__main__position-absolute {
  position: absolute;
}

.dapp-core-component__main__w-25 {
  width: 25%;
}

.dapp-core-component__main__w-50 {
  width: 50%;
}

.dapp-core-component__main__w-75 {
  width: 75%;
}

.dapp-core-component__main__w-100 {
  width: 100%;
}

.dapp-core-component__main__w-auto {
  width: auto;
}

.dapp-core-component__main__h-25 {
  height: 25%;
}

.dapp-core-component__main__h-50 {
  height: 50%;
}

.dapp-core-component__main__h-75 {
  height: 75%;
}

.dapp-core-component__main__h-100 {
  height: 100%;
}

.dapp-core-component__main__h-auto {
  height: auto;
}

.dapp-core-component__main__mw-100 {
  max-width: 100%;
}

.dapp-core-component__main__mh-100 {
  max-height: 100%;
}

.dapp-core-component__main__min-vw-100 {
  min-width: 100vw;
}

.dapp-core-component__main__min-vh-100 {
  min-height: 100vh;
}

.dapp-core-component__main__vw-100 {
  width: 100vw;
}

.dapp-core-component__main__vh-100 {
  height: 100vh;
}

.dapp-core-component__main__locked-icon {
  color: inherit;
}

.dapp-core-component__main__shadow-sm {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
}

.dapp-core-component__main__shadow {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.dapp-core-component__main__shadow-lg {
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.175);
}

.dapp-core-component__main__shadow-none {
  box-shadow: none;
}

.dapp-core-component__main__d-none {
  display: none;
}

.dapp-core-component__main__d-inline {
  display: inline;
}

.dapp-core-component__main__d-inline-block {
  display: inline-block;
}

.dapp-core-component__main__d-block {
  display: block;
}

.dapp-core-component__main__d-table {
  display: table;
}

.dapp-core-component__main__d-table-row {
  display: table-row;
}

.dapp-core-component__main__d-table-cell {
  display: table-cell;
}

.dapp-core-component__main__d-flex {
  display: -ms-flexbox;
  display: flex;
}

.dapp-core-component__main__d-inline-flex {
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media (min-width: 576px) {
  .dapp-core-component__main__d-sm-flex {
    display: flex;
  }
}

.dapp-core-component__main__table {
  width: 100%;
  margin-bottom: 1rem;
  color: #212529;
}
.dapp-core-component__main__table th,
.dapp-core-component__main__table td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #dee2e6;
}
.dapp-core-component__main__table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #dee2e6;
}
.dapp-core-component__main__table tbody + tbody {
  border-top: 2px solid #dee2e6;
}

.dapp-core-component__main__table-sm th,
.dapp-core-component__main__table-sm td {
  padding: 0.3rem;
}

.dapp-core-component__main__table-bordered {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered th,
.dapp-core-component__main__table-bordered td {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered thead th,
.dapp-core-component__main__table-bordered thead td {
  border-bottom-width: 2px;
}

.dapp-core-component__main__table-borderless th,
.dapp-core-component__main__table-borderless td,
.dapp-core-component__main__table-borderless thead th,
.dapp-core-component__main__table-borderless tbody + tbody {
  border: 0;
}

.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(0, 0, 0, 0.05);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-dark th {
  color: #fff;
  background-color: #343a40;
  border-color: #454d55;
}
.dapp-core-component__main__table .dapp-core-component__main__thead-light th {
  color: #495057;
  background-color: #e9ecef;
  border-color: #dee2e6;
}

.dapp-core-component__main__table-dark {
  color: #fff;
  background-color: #343a40;
}
.dapp-core-component__main__table-dark th,
.dapp-core-component__main__table-dark td,
.dapp-core-component__main__table-dark thead th {
  border-color: #454d55;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-bordered {
  border: 0;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(255, 255, 255, 0.05);
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-hover tbody tr {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.075);
}

.dapp-core-component__main__table-responsive {
  display: block;
  width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}
.dapp-core-component__main__table-responsive > .dapp-core-component__main__table-bordered {
  border: 0;
}

@media (max-width: 767.98px) {
  .dapp-core-component__main__transactions.dapp-core-component__main__table .dapp-core-component__main__transaction-hash .dapp-core-component__main__trim {
    max-width: 8rem;
  }
}
.dapp-core-component__main__font-weight-light {
  font-weight: 300 !important;
}

.dapp-core-component__main__font-weight-lighter {
  font-weight: lighter !important;
}

.dapp-core-component__main__font-weight-normal {
  font-weight: 400 !important;
}

.dapp-core-component__main__font-weight-medium {
  font-weight: 500 !important;
}

.dapp-core-component__main__font-weight-bold {
  font-weight: 700 !important;
}

.dapp-core-component__main__font-weight-bolder {
  font-weight: bolder !important;
}

.dapp-core-component__main__font-italic {
  font-style: italic !important;
}

.dapp-core-component__main__icon-state {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
}
.dapp-core-component__main__icon-state.dapp-core-component__main__half {
  width: 3rem;
  height: 3rem;
}

.dapp-core-component__main__side-icon {
  width: 1.5rem;
  height: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  color: transparent;
  border-radius: 50%;
  overflow: hidden;
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__side-icon svg {
  color: #cacbcc;
}

div.dapp-core-component__main__side-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
div.dapp-core-component__main__side-icon img {
  transform: scale(0.75);
}

.dapp-core-component__main__toast-messages {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast {
  display: block;
  flex-basis: 4rem;
  width: 100%;
  max-width: 28.875rem;
  transition: transform 0.3s ease-in-out;
  transform: translateX(120%);
  background-color: #fff;
  border-radius: 0.25rem;
  border-color: #dee2e6;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__clickable {
  cursor: pointer;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible {
  margin-bottom: 0.3rem;
  transform: translateX(0);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress {
  background-color: rgba(0, 0, 0, 0);
  height: inherit;
  font-size: inherit;
  line-height: inherit;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar {
  background-color: rgba(108, 117, 125, 0.2);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar .dapp-core-component__main__content-height {
  z-index: -1;
  visibility: hidden;
  width: 0 !important;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast .dapp-core-component__main__tx-description {
  margin-bottom: 0.2rem;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close {
  opacity: 1;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close svg {
  color: #6c757d;
}

.dapp-core-component__main__data-field .dapp-core-component__main__textarea {
  -moz-appearance: textfield-multiline;
  -webkit-appearance: textarea;
  border: 1px solid #000;
  background-color: #fff;
  border-radius: 5px;
  font: medium -moz-fixed;
  font-size: 0.8125rem;
  min-height: 5rem;
  overflow: auto;
  padding: 0.375rem 0.75rem;
  resize: both;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea:hover {
  border: 1px solid #000;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea .dapp-core-component__main__highlighted {
  font-size: 1rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea.dapp-core-component__main__sc-call {
  min-height: 3rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__form-control:disabled,
.dapp-core-component__main__data-field .dapp-core-component__main__form-control[readonly] {
  background-color: #fafafa;
}
.dapp-core-component__main__data-field .dapp-core-component__main__small-font {
  font-size: 0.8125rem;
}

.dapp-core-component__main__progress-steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps hr {
  background-color: #6c757d;
  border-color: #dee2e6;
  z-index: 2;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot {
  height: 0.625rem;
  width: 0.625rem;
  border-radius: 50%;
  margin: 0 1rem;
  z-index: 5;
  background-color: #6c757d;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:first-of-type {
  margin-left: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:last-of-type {
  margin-right: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

a.dapp-core-component__main__side-link::after {
  right: 0;
  left: unset;
  width: calc(100% - 1.85rem);
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(hs));var d={"link-style":"dapp-core-component__main__link-style",linkStyle:"dapp-core-component__main__link-style","link-second-style":"dapp-core-component__main__link-second-style",linkSecondStyle:"dapp-core-component__main__link-second-style",wrapper:"dapp-core-component__main__wrapper","opacity-6":"dapp-core-component__main__opacity-6",opacity6:"dapp-core-component__main__opacity-6","qr-code-svg-container":"dapp-core-component__main__qr-code-svg-container",qrCodeSvgContainer:"dapp-core-component__main__qr-code-svg-container","pair-list":"dapp-core-component__main__pair-list",pairList:"dapp-core-component__main__pair-list","pair-image":"dapp-core-component__main__pair-image",pairImage:"dapp-core-component__main__pair-image","pair-remove":"dapp-core-component__main__pair-remove",pairRemove:"dapp-core-component__main__pair-remove","dapp-icon":"dapp-core-component__main__dapp-icon",dappIcon:"dapp-core-component__main__dapp-icon","icon-medium":"dapp-core-component__main__icon-medium",iconMedium:"dapp-core-component__main__icon-medium",centering:"dapp-core-component__main__centering","border-n":"dapp-core-component__main__border-n",borderN:"dapp-core-component__main__border-n","t-shadow":"dapp-core-component__main__t-shadow",tShadow:"dapp-core-component__main__t-shadow",btn:"dapp-core-component__main__btn",media:"dapp-core-component__main__media","media-body":"dapp-core-component__main__media-body",mediaBody:"dapp-core-component__main__media-body","connect-btns":"dapp-core-component__main__connect-btns",connectBtns:"dapp-core-component__main__connect-btns",info:"dapp-core-component__main__info","trust-badge":"dapp-core-component__main__trust-badge",trustBadge:"dapp-core-component__main__trust-badge",navbar:"dapp-core-component__main__navbar","dapp-name":"dapp-core-component__main__dapp-name",dappName:"dapp-core-component__main__dapp-name","action-btn":"dapp-core-component__main__action-btn",actionBtn:"dapp-core-component__main__action-btn",transactions:"dapp-core-component__main__transactions","transaction-icon":"dapp-core-component__main__transaction-icon",transactionIcon:"dapp-core-component__main__transaction-icon",table:"dapp-core-component__main__table","page-state":"dapp-core-component__main__page-state",pageState:"dapp-core-component__main__page-state","lds-ellipsis":"dapp-core-component__main__lds-ellipsis",ldsEllipsis:"dapp-core-component__main__lds-ellipsis","bg-blue":"dapp-core-component__main__bg-blue",bgBlue:"dapp-core-component__main__bg-blue","lds-ellipsis1":"dapp-core-component__main__lds-ellipsis1",ldsEllipsis1:"dapp-core-component__main__lds-ellipsis1","lds-ellipsis2":"dapp-core-component__main__lds-ellipsis2",ldsEllipsis2:"dapp-core-component__main__lds-ellipsis2","lds-ellipsis3":"dapp-core-component__main__lds-ellipsis3",ldsEllipsis3:"dapp-core-component__main__lds-ellipsis3",focus:"dapp-core-component__main__focus",disabled:"dapp-core-component__main__disabled","btn-primary":"dapp-core-component__main__btn-primary",btnPrimary:"dapp-core-component__main__btn-primary","btn-secondary":"dapp-core-component__main__btn-secondary",btnSecondary:"dapp-core-component__main__btn-secondary","btn-warning":"dapp-core-component__main__btn-warning",btnWarning:"dapp-core-component__main__btn-warning","btn-dark":"dapp-core-component__main__btn-dark",btnDark:"dapp-core-component__main__btn-dark","btn-link":"dapp-core-component__main__btn-link",btnLink:"dapp-core-component__main__btn-link","btn-light":"dapp-core-component__main__btn-light",btnLight:"dapp-core-component__main__btn-light",badge:"dapp-core-component__main__badge","badge-pill":"dapp-core-component__main__badge-pill",badgePill:"dapp-core-component__main__badge-pill","badge-primary":"dapp-core-component__main__badge-primary",badgePrimary:"dapp-core-component__main__badge-primary","badge-secondary":"dapp-core-component__main__badge-secondary",badgeSecondary:"dapp-core-component__main__badge-secondary","badge-success":"dapp-core-component__main__badge-success",badgeSuccess:"dapp-core-component__main__badge-success","badge-info":"dapp-core-component__main__badge-info",badgeInfo:"dapp-core-component__main__badge-info","badge-warning":"dapp-core-component__main__badge-warning",badgeWarning:"dapp-core-component__main__badge-warning","badge-danger":"dapp-core-component__main__badge-danger",badgeDanger:"dapp-core-component__main__badge-danger","badge-light":"dapp-core-component__main__badge-light",badgeLight:"dapp-core-component__main__badge-light","badge-dark":"dapp-core-component__main__badge-dark",badgeDark:"dapp-core-component__main__badge-dark",card:"dapp-core-component__main__card","list-group":"dapp-core-component__main__list-group",listGroup:"dapp-core-component__main__list-group","card-header":"dapp-core-component__main__card-header",cardHeader:"dapp-core-component__main__card-header","card-footer":"dapp-core-component__main__card-footer",cardFooter:"dapp-core-component__main__card-footer","card-header-item":"dapp-core-component__main__card-header-item",cardHeaderItem:"dapp-core-component__main__card-header-item",h6:"dapp-core-component__main__h6",compact:"dapp-core-component__main__compact","card-body":"dapp-core-component__main__card-body",cardBody:"dapp-core-component__main__card-body","card-title":"dapp-core-component__main__card-title",cardTitle:"dapp-core-component__main__card-title","card-subtitle":"dapp-core-component__main__card-subtitle",cardSubtitle:"dapp-core-component__main__card-subtitle","card-text":"dapp-core-component__main__card-text",cardText:"dapp-core-component__main__card-text","card-link":"dapp-core-component__main__card-link",cardLink:"dapp-core-component__main__card-link",text:"dapp-core-component__main__text","text-white":"dapp-core-component__main__text-white",textWhite:"dapp-core-component__main__text-white","text-monospace":"dapp-core-component__main__text-monospace",textMonospace:"dapp-core-component__main__text-monospace","trim-size-xl":"dapp-core-component__main__trim-size-xl",trimSizeXl:"dapp-core-component__main__trim-size-xl",trim:"dapp-core-component__main__trim","trim-only-sm":"dapp-core-component__main__trim-only-sm",trimOnlySm:"dapp-core-component__main__trim-only-sm","text-justify":"dapp-core-component__main__text-justify",textJustify:"dapp-core-component__main__text-justify","text-wrap":"dapp-core-component__main__text-wrap",textWrap:"dapp-core-component__main__text-wrap","text-nowrap":"dapp-core-component__main__text-nowrap",textNowrap:"dapp-core-component__main__text-nowrap","text-truncate":"dapp-core-component__main__text-truncate",textTruncate:"dapp-core-component__main__text-truncate","text-left":"dapp-core-component__main__text-left",textLeft:"dapp-core-component__main__text-left","text-right":"dapp-core-component__main__text-right",textRight:"dapp-core-component__main__text-right","text-center":"dapp-core-component__main__text-center",textCenter:"dapp-core-component__main__text-center","text-primary":"dapp-core-component__main__text-primary",textPrimary:"dapp-core-component__main__text-primary","text-secondary":"dapp-core-component__main__text-secondary",textSecondary:"dapp-core-component__main__text-secondary","text-primary-highlight":"dapp-core-component__main__text-primary-highlight",textPrimaryHighlight:"dapp-core-component__main__text-primary-highlight","text-muted":"dapp-core-component__main__text-muted",textMuted:"dapp-core-component__main__text-muted","text-success":"dapp-core-component__main__text-success",textSuccess:"dapp-core-component__main__text-success","text-body":"dapp-core-component__main__text-body",textBody:"dapp-core-component__main__text-body","text-black-50":"dapp-core-component__main__text-black-50",textBlack50:"dapp-core-component__main__text-black-50","text-white-50":"dapp-core-component__main__text-white-50",textWhite50:"dapp-core-component__main__text-white-50","text-lg-right":"dapp-core-component__main__text-lg-right",textLgRight:"dapp-core-component__main__text-lg-right","text-hide":"dapp-core-component__main__text-hide",textHide:"dapp-core-component__main__text-hide","text-decoration-none":"dapp-core-component__main__text-decoration-none",textDecorationNone:"dapp-core-component__main__text-decoration-none","text-break":"dapp-core-component__main__text-break",textBreak:"dapp-core-component__main__text-break","text-reset":"dapp-core-component__main__text-reset",textReset:"dapp-core-component__main__text-reset","text-lowercase":"dapp-core-component__main__text-lowercase",textLowercase:"dapp-core-component__main__text-lowercase","text-uppercase":"dapp-core-component__main__text-uppercase",textUppercase:"dapp-core-component__main__text-uppercase","text-capitalize":"dapp-core-component__main__text-capitalize",textCapitalize:"dapp-core-component__main__text-capitalize","text-danger":"dapp-core-component__main__text-danger",textDanger:"dapp-core-component__main__text-danger","text-warning":"dapp-core-component__main__text-warning",textWarning:"dapp-core-component__main__text-warning","trim-fs-sm":"dapp-core-component__main__trim-fs-sm",trimFsSm:"dapp-core-component__main__trim-fs-sm","mx-auto":"dapp-core-component__main__mx-auto",mxAuto:"dapp-core-component__main__mx-auto","m-0":"dapp-core-component__main__m-0",m0:"dapp-core-component__main__m-0","mt-0":"dapp-core-component__main__mt-0",mt0:"dapp-core-component__main__mt-0","my-0":"dapp-core-component__main__my-0",my0:"dapp-core-component__main__my-0","mr-0":"dapp-core-component__main__mr-0",mr0:"dapp-core-component__main__mr-0","mx-0":"dapp-core-component__main__mx-0",mx0:"dapp-core-component__main__mx-0","mb-0":"dapp-core-component__main__mb-0",mb0:"dapp-core-component__main__mb-0","ml-0":"dapp-core-component__main__ml-0",ml0:"dapp-core-component__main__ml-0","m-1":"dapp-core-component__main__m-1",m1:"dapp-core-component__main__m-1","mt-1":"dapp-core-component__main__mt-1",mt1:"dapp-core-component__main__mt-1","my-1":"dapp-core-component__main__my-1",my1:"dapp-core-component__main__my-1","mr-1":"dapp-core-component__main__mr-1",mr1:"dapp-core-component__main__mr-1","mx-1":"dapp-core-component__main__mx-1",mx1:"dapp-core-component__main__mx-1","mb-1":"dapp-core-component__main__mb-1",mb1:"dapp-core-component__main__mb-1","ml-1":"dapp-core-component__main__ml-1",ml1:"dapp-core-component__main__ml-1","m-2":"dapp-core-component__main__m-2",m2:"dapp-core-component__main__m-2","mt-2":"dapp-core-component__main__mt-2",mt2:"dapp-core-component__main__mt-2","my-2":"dapp-core-component__main__my-2",my2:"dapp-core-component__main__my-2","mr-2":"dapp-core-component__main__mr-2",mr2:"dapp-core-component__main__mr-2","mx-2":"dapp-core-component__main__mx-2",mx2:"dapp-core-component__main__mx-2","mb-2":"dapp-core-component__main__mb-2",mb2:"dapp-core-component__main__mb-2","ml-2":"dapp-core-component__main__ml-2",ml2:"dapp-core-component__main__ml-2","m-3":"dapp-core-component__main__m-3",m3:"dapp-core-component__main__m-3","mt-3":"dapp-core-component__main__mt-3",mt3:"dapp-core-component__main__mt-3","my-3":"dapp-core-component__main__my-3",my3:"dapp-core-component__main__my-3","mr-3":"dapp-core-component__main__mr-3",mr3:"dapp-core-component__main__mr-3","mx-3":"dapp-core-component__main__mx-3",mx3:"dapp-core-component__main__mx-3","mb-3":"dapp-core-component__main__mb-3",mb3:"dapp-core-component__main__mb-3","ml-3":"dapp-core-component__main__ml-3",ml3:"dapp-core-component__main__ml-3","m-4":"dapp-core-component__main__m-4",m4:"dapp-core-component__main__m-4","mt-4":"dapp-core-component__main__mt-4",mt4:"dapp-core-component__main__mt-4","my-4":"dapp-core-component__main__my-4",my4:"dapp-core-component__main__my-4","mr-4":"dapp-core-component__main__mr-4",mr4:"dapp-core-component__main__mr-4","mx-4":"dapp-core-component__main__mx-4",mx4:"dapp-core-component__main__mx-4","mb-4":"dapp-core-component__main__mb-4",mb4:"dapp-core-component__main__mb-4","ml-4":"dapp-core-component__main__ml-4",ml4:"dapp-core-component__main__ml-4","m-5":"dapp-core-component__main__m-5",m5:"dapp-core-component__main__m-5","mt-5":"dapp-core-component__main__mt-5",mt5:"dapp-core-component__main__mt-5","my-5":"dapp-core-component__main__my-5",my5:"dapp-core-component__main__my-5","mr-5":"dapp-core-component__main__mr-5",mr5:"dapp-core-component__main__mr-5","mx-5":"dapp-core-component__main__mx-5",mx5:"dapp-core-component__main__mx-5","mb-5":"dapp-core-component__main__mb-5",mb5:"dapp-core-component__main__mb-5","ml-5":"dapp-core-component__main__ml-5",ml5:"dapp-core-component__main__ml-5","p-0":"dapp-core-component__main__p-0",p0:"dapp-core-component__main__p-0","pt-0":"dapp-core-component__main__pt-0",pt0:"dapp-core-component__main__pt-0","py-0":"dapp-core-component__main__py-0",py0:"dapp-core-component__main__py-0","pr-0":"dapp-core-component__main__pr-0",pr0:"dapp-core-component__main__pr-0","px-0":"dapp-core-component__main__px-0",px0:"dapp-core-component__main__px-0","pb-0":"dapp-core-component__main__pb-0",pb0:"dapp-core-component__main__pb-0","pl-0":"dapp-core-component__main__pl-0",pl0:"dapp-core-component__main__pl-0","p-1":"dapp-core-component__main__p-1",p1:"dapp-core-component__main__p-1","pt-1":"dapp-core-component__main__pt-1",pt1:"dapp-core-component__main__pt-1","py-1":"dapp-core-component__main__py-1",py1:"dapp-core-component__main__py-1","pr-1":"dapp-core-component__main__pr-1",pr1:"dapp-core-component__main__pr-1","px-1":"dapp-core-component__main__px-1",px1:"dapp-core-component__main__px-1","pb-1":"dapp-core-component__main__pb-1",pb1:"dapp-core-component__main__pb-1","pl-1":"dapp-core-component__main__pl-1",pl1:"dapp-core-component__main__pl-1","p-2":"dapp-core-component__main__p-2",p2:"dapp-core-component__main__p-2","pt-2":"dapp-core-component__main__pt-2",pt2:"dapp-core-component__main__pt-2","py-2":"dapp-core-component__main__py-2",py2:"dapp-core-component__main__py-2","pr-2":"dapp-core-component__main__pr-2",pr2:"dapp-core-component__main__pr-2","px-2":"dapp-core-component__main__px-2",px2:"dapp-core-component__main__px-2","pb-2":"dapp-core-component__main__pb-2",pb2:"dapp-core-component__main__pb-2","pl-2":"dapp-core-component__main__pl-2",pl2:"dapp-core-component__main__pl-2","p-3":"dapp-core-component__main__p-3",p3:"dapp-core-component__main__p-3","pt-3":"dapp-core-component__main__pt-3",pt3:"dapp-core-component__main__pt-3","py-3":"dapp-core-component__main__py-3",py3:"dapp-core-component__main__py-3","pr-3":"dapp-core-component__main__pr-3",pr3:"dapp-core-component__main__pr-3","px-3":"dapp-core-component__main__px-3",px3:"dapp-core-component__main__px-3","pb-3":"dapp-core-component__main__pb-3",pb3:"dapp-core-component__main__pb-3","pl-3":"dapp-core-component__main__pl-3",pl3:"dapp-core-component__main__pl-3","p-4":"dapp-core-component__main__p-4",p4:"dapp-core-component__main__p-4","pt-4":"dapp-core-component__main__pt-4",pt4:"dapp-core-component__main__pt-4","py-4":"dapp-core-component__main__py-4",py4:"dapp-core-component__main__py-4","pr-4":"dapp-core-component__main__pr-4",pr4:"dapp-core-component__main__pr-4","px-4":"dapp-core-component__main__px-4",px4:"dapp-core-component__main__px-4","pb-4":"dapp-core-component__main__pb-4",pb4:"dapp-core-component__main__pb-4","pl-4":"dapp-core-component__main__pl-4",pl4:"dapp-core-component__main__pl-4","p-5":"dapp-core-component__main__p-5",p5:"dapp-core-component__main__p-5","pt-5":"dapp-core-component__main__pt-5",pt5:"dapp-core-component__main__pt-5","py-5":"dapp-core-component__main__py-5",py5:"dapp-core-component__main__py-5","pr-5":"dapp-core-component__main__pr-5",pr5:"dapp-core-component__main__pr-5","px-5":"dapp-core-component__main__px-5",px5:"dapp-core-component__main__px-5","pb-5":"dapp-core-component__main__pb-5",pb5:"dapp-core-component__main__pb-5","pl-5":"dapp-core-component__main__pl-5",pl5:"dapp-core-component__main__pl-5","m-n1":"dapp-core-component__main__m-n1",mN1:"dapp-core-component__main__m-n1","mt-n1":"dapp-core-component__main__mt-n1",mtN1:"dapp-core-component__main__mt-n1","my-n1":"dapp-core-component__main__my-n1",myN1:"dapp-core-component__main__my-n1","mr-n1":"dapp-core-component__main__mr-n1",mrN1:"dapp-core-component__main__mr-n1","mx-n1":"dapp-core-component__main__mx-n1",mxN1:"dapp-core-component__main__mx-n1","mb-n1":"dapp-core-component__main__mb-n1",mbN1:"dapp-core-component__main__mb-n1","ml-n1":"dapp-core-component__main__ml-n1",mlN1:"dapp-core-component__main__ml-n1","m-n2":"dapp-core-component__main__m-n2",mN2:"dapp-core-component__main__m-n2","mt-n2":"dapp-core-component__main__mt-n2",mtN2:"dapp-core-component__main__mt-n2","my-n2":"dapp-core-component__main__my-n2",myN2:"dapp-core-component__main__my-n2","mr-n2":"dapp-core-component__main__mr-n2",mrN2:"dapp-core-component__main__mr-n2","mx-n2":"dapp-core-component__main__mx-n2",mxN2:"dapp-core-component__main__mx-n2","mb-n2":"dapp-core-component__main__mb-n2",mbN2:"dapp-core-component__main__mb-n2","ml-n2":"dapp-core-component__main__ml-n2",mlN2:"dapp-core-component__main__ml-n2","m-n3":"dapp-core-component__main__m-n3",mN3:"dapp-core-component__main__m-n3","mt-n3":"dapp-core-component__main__mt-n3",mtN3:"dapp-core-component__main__mt-n3","my-n3":"dapp-core-component__main__my-n3",myN3:"dapp-core-component__main__my-n3","mr-n3":"dapp-core-component__main__mr-n3",mrN3:"dapp-core-component__main__mr-n3","mx-n3":"dapp-core-component__main__mx-n3",mxN3:"dapp-core-component__main__mx-n3","mb-n3":"dapp-core-component__main__mb-n3",mbN3:"dapp-core-component__main__mb-n3","ml-n3":"dapp-core-component__main__ml-n3",mlN3:"dapp-core-component__main__ml-n3","m-n4":"dapp-core-component__main__m-n4",mN4:"dapp-core-component__main__m-n4","mt-n4":"dapp-core-component__main__mt-n4",mtN4:"dapp-core-component__main__mt-n4","my-n4":"dapp-core-component__main__my-n4",myN4:"dapp-core-component__main__my-n4","mr-n4":"dapp-core-component__main__mr-n4",mrN4:"dapp-core-component__main__mr-n4","mx-n4":"dapp-core-component__main__mx-n4",mxN4:"dapp-core-component__main__mx-n4","mb-n4":"dapp-core-component__main__mb-n4",mbN4:"dapp-core-component__main__mb-n4","ml-n4":"dapp-core-component__main__ml-n4",mlN4:"dapp-core-component__main__ml-n4","m-n5":"dapp-core-component__main__m-n5",mN5:"dapp-core-component__main__m-n5","mt-n5":"dapp-core-component__main__mt-n5",mtN5:"dapp-core-component__main__mt-n5","my-n5":"dapp-core-component__main__my-n5",myN5:"dapp-core-component__main__my-n5","mr-n5":"dapp-core-component__main__mr-n5",mrN5:"dapp-core-component__main__mr-n5","mx-n5":"dapp-core-component__main__mx-n5",mxN5:"dapp-core-component__main__mx-n5","mb-n5":"dapp-core-component__main__mb-n5",mbN5:"dapp-core-component__main__mb-n5","ml-n5":"dapp-core-component__main__ml-n5",mlN5:"dapp-core-component__main__ml-n5","m-auto":"dapp-core-component__main__m-auto",mAuto:"dapp-core-component__main__m-auto","mt-auto":"dapp-core-component__main__mt-auto",mtAuto:"dapp-core-component__main__mt-auto","my-auto":"dapp-core-component__main__my-auto",myAuto:"dapp-core-component__main__my-auto","mr-auto":"dapp-core-component__main__mr-auto",mrAuto:"dapp-core-component__main__mr-auto","mb-auto":"dapp-core-component__main__mb-auto",mbAuto:"dapp-core-component__main__mb-auto","ml-auto":"dapp-core-component__main__ml-auto",mlAuto:"dapp-core-component__main__ml-auto","mr-lg-1":"dapp-core-component__main__mr-lg-1",mrLg1:"dapp-core-component__main__mr-lg-1","mr-lg-4":"dapp-core-component__main__mr-lg-4",mrLg4:"dapp-core-component__main__mr-lg-4","mx-lg-4":"dapp-core-component__main__mx-lg-4",mxLg4:"dapp-core-component__main__mx-lg-4","mb-xl-2":"dapp-core-component__main__mb-xl-2",mbXl2:"dapp-core-component__main__mb-xl-2","pr-sm-2":"dapp-core-component__main__pr-sm-2",prSm2:"dapp-core-component__main__pr-sm-2","px-sm-2":"dapp-core-component__main__px-sm-2",pxSm2:"dapp-core-component__main__px-sm-2","pr-xl-0":"dapp-core-component__main__pr-xl-0",prXl0:"dapp-core-component__main__pr-xl-0","px-xl-0":"dapp-core-component__main__px-xl-0",pxXl0:"dapp-core-component__main__px-xl-0",col:"dapp-core-component__main__col",row:"dapp-core-component__main__row","col-xl":"dapp-core-component__main__col-xl",colXl:"dapp-core-component__main__col-xl","col-xl-auto":"dapp-core-component__main__col-xl-auto",colXlAuto:"dapp-core-component__main__col-xl-auto","col-xl-12":"dapp-core-component__main__col-xl-12",colXl12:"dapp-core-component__main__col-xl-12","col-xl-11":"dapp-core-component__main__col-xl-11",colXl11:"dapp-core-component__main__col-xl-11","col-xl-10":"dapp-core-component__main__col-xl-10",colXl10:"dapp-core-component__main__col-xl-10","col-xl-9":"dapp-core-component__main__col-xl-9",colXl9:"dapp-core-component__main__col-xl-9","col-xl-8":"dapp-core-component__main__col-xl-8",colXl8:"dapp-core-component__main__col-xl-8","col-xl-7":"dapp-core-component__main__col-xl-7",colXl7:"dapp-core-component__main__col-xl-7","col-xl-6":"dapp-core-component__main__col-xl-6",colXl6:"dapp-core-component__main__col-xl-6","col-xl-5":"dapp-core-component__main__col-xl-5",colXl5:"dapp-core-component__main__col-xl-5","col-xl-4":"dapp-core-component__main__col-xl-4",colXl4:"dapp-core-component__main__col-xl-4","col-xl-3":"dapp-core-component__main__col-xl-3",colXl3:"dapp-core-component__main__col-xl-3","col-xl-2":"dapp-core-component__main__col-xl-2",colXl2:"dapp-core-component__main__col-xl-2","col-xl-1":"dapp-core-component__main__col-xl-1",colXl1:"dapp-core-component__main__col-xl-1","col-lg":"dapp-core-component__main__col-lg",colLg:"dapp-core-component__main__col-lg","col-lg-auto":"dapp-core-component__main__col-lg-auto",colLgAuto:"dapp-core-component__main__col-lg-auto","col-lg-12":"dapp-core-component__main__col-lg-12",colLg12:"dapp-core-component__main__col-lg-12","col-lg-11":"dapp-core-component__main__col-lg-11",colLg11:"dapp-core-component__main__col-lg-11","col-lg-10":"dapp-core-component__main__col-lg-10",colLg10:"dapp-core-component__main__col-lg-10","col-lg-9":"dapp-core-component__main__col-lg-9",colLg9:"dapp-core-component__main__col-lg-9","col-lg-8":"dapp-core-component__main__col-lg-8",colLg8:"dapp-core-component__main__col-lg-8","col-lg-7":"dapp-core-component__main__col-lg-7",colLg7:"dapp-core-component__main__col-lg-7","col-lg-6":"dapp-core-component__main__col-lg-6",colLg6:"dapp-core-component__main__col-lg-6","col-lg-5":"dapp-core-component__main__col-lg-5",colLg5:"dapp-core-component__main__col-lg-5","col-lg-4":"dapp-core-component__main__col-lg-4",colLg4:"dapp-core-component__main__col-lg-4","col-lg-3":"dapp-core-component__main__col-lg-3",colLg3:"dapp-core-component__main__col-lg-3","col-lg-2":"dapp-core-component__main__col-lg-2",colLg2:"dapp-core-component__main__col-lg-2","col-lg-1":"dapp-core-component__main__col-lg-1",colLg1:"dapp-core-component__main__col-lg-1","col-md":"dapp-core-component__main__col-md",colMd:"dapp-core-component__main__col-md","col-md-auto":"dapp-core-component__main__col-md-auto",colMdAuto:"dapp-core-component__main__col-md-auto","col-md-12":"dapp-core-component__main__col-md-12",colMd12:"dapp-core-component__main__col-md-12","col-md-11":"dapp-core-component__main__col-md-11",colMd11:"dapp-core-component__main__col-md-11","col-md-10":"dapp-core-component__main__col-md-10",colMd10:"dapp-core-component__main__col-md-10","col-md-9":"dapp-core-component__main__col-md-9",colMd9:"dapp-core-component__main__col-md-9","col-md-8":"dapp-core-component__main__col-md-8",colMd8:"dapp-core-component__main__col-md-8","col-md-7":"dapp-core-component__main__col-md-7",colMd7:"dapp-core-component__main__col-md-7","col-md-6":"dapp-core-component__main__col-md-6",colMd6:"dapp-core-component__main__col-md-6","col-md-5":"dapp-core-component__main__col-md-5",colMd5:"dapp-core-component__main__col-md-5","col-md-4":"dapp-core-component__main__col-md-4",colMd4:"dapp-core-component__main__col-md-4","col-md-3":"dapp-core-component__main__col-md-3",colMd3:"dapp-core-component__main__col-md-3","col-md-2":"dapp-core-component__main__col-md-2",colMd2:"dapp-core-component__main__col-md-2","col-md-1":"dapp-core-component__main__col-md-1",colMd1:"dapp-core-component__main__col-md-1","col-sm":"dapp-core-component__main__col-sm",colSm:"dapp-core-component__main__col-sm","col-sm-auto":"dapp-core-component__main__col-sm-auto",colSmAuto:"dapp-core-component__main__col-sm-auto","col-sm-12":"dapp-core-component__main__col-sm-12",colSm12:"dapp-core-component__main__col-sm-12","col-sm-11":"dapp-core-component__main__col-sm-11",colSm11:"dapp-core-component__main__col-sm-11","col-sm-10":"dapp-core-component__main__col-sm-10",colSm10:"dapp-core-component__main__col-sm-10","col-sm-9":"dapp-core-component__main__col-sm-9",colSm9:"dapp-core-component__main__col-sm-9","col-sm-8":"dapp-core-component__main__col-sm-8",colSm8:"dapp-core-component__main__col-sm-8","col-sm-7":"dapp-core-component__main__col-sm-7",colSm7:"dapp-core-component__main__col-sm-7","col-sm-6":"dapp-core-component__main__col-sm-6",colSm6:"dapp-core-component__main__col-sm-6","col-sm-5":"dapp-core-component__main__col-sm-5",colSm5:"dapp-core-component__main__col-sm-5","col-sm-4":"dapp-core-component__main__col-sm-4",colSm4:"dapp-core-component__main__col-sm-4","col-sm-3":"dapp-core-component__main__col-sm-3",colSm3:"dapp-core-component__main__col-sm-3","col-sm-2":"dapp-core-component__main__col-sm-2",colSm2:"dapp-core-component__main__col-sm-2","col-sm-1":"dapp-core-component__main__col-sm-1",colSm1:"dapp-core-component__main__col-sm-1","col-auto":"dapp-core-component__main__col-auto",colAuto:"dapp-core-component__main__col-auto","col-12":"dapp-core-component__main__col-12",col12:"dapp-core-component__main__col-12","col-11":"dapp-core-component__main__col-11",col11:"dapp-core-component__main__col-11","col-10":"dapp-core-component__main__col-10",col10:"dapp-core-component__main__col-10","col-9":"dapp-core-component__main__col-9",col9:"dapp-core-component__main__col-9","col-8":"dapp-core-component__main__col-8",col8:"dapp-core-component__main__col-8","col-7":"dapp-core-component__main__col-7",col7:"dapp-core-component__main__col-7","col-6":"dapp-core-component__main__col-6",col6:"dapp-core-component__main__col-6","col-5":"dapp-core-component__main__col-5",col5:"dapp-core-component__main__col-5","col-4":"dapp-core-component__main__col-4",col4:"dapp-core-component__main__col-4","col-3":"dapp-core-component__main__col-3",col3:"dapp-core-component__main__col-3","col-2":"dapp-core-component__main__col-2",col2:"dapp-core-component__main__col-2","col-1":"dapp-core-component__main__col-1",col1:"dapp-core-component__main__col-1","flex-row":"dapp-core-component__main__flex-row",flexRow:"dapp-core-component__main__flex-row","flex-column":"dapp-core-component__main__flex-column",flexColumn:"dapp-core-component__main__flex-column","flex-row-reverse":"dapp-core-component__main__flex-row-reverse",flexRowReverse:"dapp-core-component__main__flex-row-reverse","flex-column-reverse":"dapp-core-component__main__flex-column-reverse",flexColumnReverse:"dapp-core-component__main__flex-column-reverse","flex-wrap":"dapp-core-component__main__flex-wrap",flexWrap:"dapp-core-component__main__flex-wrap","flex-nowrap":"dapp-core-component__main__flex-nowrap",flexNowrap:"dapp-core-component__main__flex-nowrap","flex-wrap-reverse":"dapp-core-component__main__flex-wrap-reverse",flexWrapReverse:"dapp-core-component__main__flex-wrap-reverse","flex-fill":"dapp-core-component__main__flex-fill",flexFill:"dapp-core-component__main__flex-fill","flex-grow-0":"dapp-core-component__main__flex-grow-0",flexGrow0:"dapp-core-component__main__flex-grow-0","flex-grow-1":"dapp-core-component__main__flex-grow-1",flexGrow1:"dapp-core-component__main__flex-grow-1","flex-shrink-0":"dapp-core-component__main__flex-shrink-0",flexShrink0:"dapp-core-component__main__flex-shrink-0","flex-shrink-1":"dapp-core-component__main__flex-shrink-1",flexShrink1:"dapp-core-component__main__flex-shrink-1","justify-content-start":"dapp-core-component__main__justify-content-start",justifyContentStart:"dapp-core-component__main__justify-content-start","justify-content-end":"dapp-core-component__main__justify-content-end",justifyContentEnd:"dapp-core-component__main__justify-content-end","justify-content-center":"dapp-core-component__main__justify-content-center",justifyContentCenter:"dapp-core-component__main__justify-content-center","justify-content-between":"dapp-core-component__main__justify-content-between",justifyContentBetween:"dapp-core-component__main__justify-content-between","justify-content-around":"dapp-core-component__main__justify-content-around",justifyContentAround:"dapp-core-component__main__justify-content-around","align-items-start":"dapp-core-component__main__align-items-start",alignItemsStart:"dapp-core-component__main__align-items-start","align-items-end":"dapp-core-component__main__align-items-end",alignItemsEnd:"dapp-core-component__main__align-items-end","align-items-center":"dapp-core-component__main__align-items-center",alignItemsCenter:"dapp-core-component__main__align-items-center","align-items-baseline":"dapp-core-component__main__align-items-baseline",alignItemsBaseline:"dapp-core-component__main__align-items-baseline","align-items-stretch":"dapp-core-component__main__align-items-stretch",alignItemsStretch:"dapp-core-component__main__align-items-stretch","align-content-start":"dapp-core-component__main__align-content-start",alignContentStart:"dapp-core-component__main__align-content-start","align-content-end":"dapp-core-component__main__align-content-end",alignContentEnd:"dapp-core-component__main__align-content-end","align-content-center":"dapp-core-component__main__align-content-center",alignContentCenter:"dapp-core-component__main__align-content-center","align-content-between":"dapp-core-component__main__align-content-between",alignContentBetween:"dapp-core-component__main__align-content-between","align-content-around":"dapp-core-component__main__align-content-around",alignContentAround:"dapp-core-component__main__align-content-around","align-content-stretch":"dapp-core-component__main__align-content-stretch",alignContentStretch:"dapp-core-component__main__align-content-stretch","align-self-auto":"dapp-core-component__main__align-self-auto",alignSelfAuto:"dapp-core-component__main__align-self-auto","align-self-start":"dapp-core-component__main__align-self-start",alignSelfStart:"dapp-core-component__main__align-self-start","align-self-end":"dapp-core-component__main__align-self-end",alignSelfEnd:"dapp-core-component__main__align-self-end","align-self-center":"dapp-core-component__main__align-self-center",alignSelfCenter:"dapp-core-component__main__align-self-center","align-self-baseline":"dapp-core-component__main__align-self-baseline",alignSelfBaseline:"dapp-core-component__main__align-self-baseline","align-self-stretch":"dapp-core-component__main__align-self-stretch",alignSelfStretch:"dapp-core-component__main__align-self-stretch","flex-sm-row":"dapp-core-component__main__flex-sm-row",flexSmRow:"dapp-core-component__main__flex-sm-row","flex-lg-row":"dapp-core-component__main__flex-lg-row",flexLgRow:"dapp-core-component__main__flex-lg-row","flex-lg-wrap":"dapp-core-component__main__flex-lg-wrap",flexLgWrap:"dapp-core-component__main__flex-lg-wrap","form-check":"dapp-core-component__main__form-check",formCheck:"dapp-core-component__main__form-check","form-check-label":"dapp-core-component__main__form-check-label",formCheckLabel:"dapp-core-component__main__form-check-label","form-check-input":"dapp-core-component__main__form-check-input",formCheckInput:"dapp-core-component__main__form-check-input","form-group":"dapp-core-component__main__form-group",formGroup:"dapp-core-component__main__form-group","form-control":"dapp-core-component__main__form-control",formControl:"dapp-core-component__main__form-control","form-label":"dapp-core-component__main__form-label",formLabel:"dapp-core-component__main__form-label","form-text":"dapp-core-component__main__form-text",formText:"dapp-core-component__main__form-text",border:"dapp-core-component__main__border","border-top":"dapp-core-component__main__border-top",borderTop:"dapp-core-component__main__border-top","border-right":"dapp-core-component__main__border-right",borderRight:"dapp-core-component__main__border-right","border-bottom":"dapp-core-component__main__border-bottom",borderBottom:"dapp-core-component__main__border-bottom","border-left":"dapp-core-component__main__border-left",borderLeft:"dapp-core-component__main__border-left","border-0":"dapp-core-component__main__border-0",border0:"dapp-core-component__main__border-0","border-top-0":"dapp-core-component__main__border-top-0",borderTop0:"dapp-core-component__main__border-top-0","border-right-0":"dapp-core-component__main__border-right-0",borderRight0:"dapp-core-component__main__border-right-0","border-bottom-0":"dapp-core-component__main__border-bottom-0",borderBottom0:"dapp-core-component__main__border-bottom-0","border-left-0":"dapp-core-component__main__border-left-0",borderLeft0:"dapp-core-component__main__border-left-0","rounded-sm":"dapp-core-component__main__rounded-sm",roundedSm:"dapp-core-component__main__rounded-sm",rounded:"dapp-core-component__main__rounded","rounded-top":"dapp-core-component__main__rounded-top",roundedTop:"dapp-core-component__main__rounded-top","rounded-right":"dapp-core-component__main__rounded-right",roundedRight:"dapp-core-component__main__rounded-right","rounded-bottom":"dapp-core-component__main__rounded-bottom",roundedBottom:"dapp-core-component__main__rounded-bottom","rounded-left":"dapp-core-component__main__rounded-left",roundedLeft:"dapp-core-component__main__rounded-left","rounded-lg":"dapp-core-component__main__rounded-lg",roundedLg:"dapp-core-component__main__rounded-lg","rounded-circle":"dapp-core-component__main__rounded-circle",roundedCircle:"dapp-core-component__main__rounded-circle","rounded-pill":"dapp-core-component__main__rounded-pill",roundedPill:"dapp-core-component__main__rounded-pill","rounded-0":"dapp-core-component__main__rounded-0",rounded0:"dapp-core-component__main__rounded-0",h1:"dapp-core-component__main__h1",h2:"dapp-core-component__main__h2",h3:"dapp-core-component__main__h3",h4:"dapp-core-component__main__h4",h5:"dapp-core-component__main__h5",lead:"dapp-core-component__main__lead","bg-primary":"dapp-core-component__main__bg-primary",bgPrimary:"dapp-core-component__main__bg-primary","bg-warning":"dapp-core-component__main__bg-warning",bgWarning:"dapp-core-component__main__bg-warning",warning:"dapp-core-component__main__warning","bg-danger":"dapp-core-component__main__bg-danger",bgDanger:"dapp-core-component__main__bg-danger",danger:"dapp-core-component__main__danger","bg-success":"dapp-core-component__main__bg-success",bgSuccess:"dapp-core-component__main__bg-success",success:"dapp-core-component__main__success","position-relative":"dapp-core-component__main__position-relative",positionRelative:"dapp-core-component__main__position-relative","position-absolute":"dapp-core-component__main__position-absolute",positionAbsolute:"dapp-core-component__main__position-absolute","w-25":"dapp-core-component__main__w-25",w25:"dapp-core-component__main__w-25","w-50":"dapp-core-component__main__w-50",w50:"dapp-core-component__main__w-50","w-75":"dapp-core-component__main__w-75",w75:"dapp-core-component__main__w-75","w-100":"dapp-core-component__main__w-100",w100:"dapp-core-component__main__w-100","w-auto":"dapp-core-component__main__w-auto",wAuto:"dapp-core-component__main__w-auto","h-25":"dapp-core-component__main__h-25",h25:"dapp-core-component__main__h-25","h-50":"dapp-core-component__main__h-50",h50:"dapp-core-component__main__h-50","h-75":"dapp-core-component__main__h-75",h75:"dapp-core-component__main__h-75","h-100":"dapp-core-component__main__h-100",h100:"dapp-core-component__main__h-100","h-auto":"dapp-core-component__main__h-auto",hAuto:"dapp-core-component__main__h-auto","mw-100":"dapp-core-component__main__mw-100",mw100:"dapp-core-component__main__mw-100","mh-100":"dapp-core-component__main__mh-100",mh100:"dapp-core-component__main__mh-100","min-vw-100":"dapp-core-component__main__min-vw-100",minVw100:"dapp-core-component__main__min-vw-100","min-vh-100":"dapp-core-component__main__min-vh-100",minVh100:"dapp-core-component__main__min-vh-100","vw-100":"dapp-core-component__main__vw-100",vw100:"dapp-core-component__main__vw-100","vh-100":"dapp-core-component__main__vh-100",vh100:"dapp-core-component__main__vh-100","locked-icon":"dapp-core-component__main__locked-icon",lockedIcon:"dapp-core-component__main__locked-icon","shadow-sm":"dapp-core-component__main__shadow-sm",shadowSm:"dapp-core-component__main__shadow-sm",shadow:"dapp-core-component__main__shadow","shadow-lg":"dapp-core-component__main__shadow-lg",shadowLg:"dapp-core-component__main__shadow-lg","shadow-none":"dapp-core-component__main__shadow-none",shadowNone:"dapp-core-component__main__shadow-none","d-none":"dapp-core-component__main__d-none",dNone:"dapp-core-component__main__d-none","d-inline":"dapp-core-component__main__d-inline",dInline:"dapp-core-component__main__d-inline","d-inline-block":"dapp-core-component__main__d-inline-block",dInlineBlock:"dapp-core-component__main__d-inline-block","d-block":"dapp-core-component__main__d-block",dBlock:"dapp-core-component__main__d-block","d-table":"dapp-core-component__main__d-table",dTable:"dapp-core-component__main__d-table","d-table-row":"dapp-core-component__main__d-table-row",dTableRow:"dapp-core-component__main__d-table-row","d-table-cell":"dapp-core-component__main__d-table-cell",dTableCell:"dapp-core-component__main__d-table-cell","d-flex":"dapp-core-component__main__d-flex",dFlex:"dapp-core-component__main__d-flex","d-inline-flex":"dapp-core-component__main__d-inline-flex",dInlineFlex:"dapp-core-component__main__d-inline-flex","d-sm-flex":"dapp-core-component__main__d-sm-flex",dSmFlex:"dapp-core-component__main__d-sm-flex","table-sm":"dapp-core-component__main__table-sm",tableSm:"dapp-core-component__main__table-sm","table-bordered":"dapp-core-component__main__table-bordered",tableBordered:"dapp-core-component__main__table-bordered","table-borderless":"dapp-core-component__main__table-borderless",tableBorderless:"dapp-core-component__main__table-borderless","table-striped":"dapp-core-component__main__table-striped",tableStriped:"dapp-core-component__main__table-striped","thead-dark":"dapp-core-component__main__thead-dark",theadDark:"dapp-core-component__main__thead-dark","thead-light":"dapp-core-component__main__thead-light",theadLight:"dapp-core-component__main__thead-light","table-dark":"dapp-core-component__main__table-dark",tableDark:"dapp-core-component__main__table-dark","table-hover":"dapp-core-component__main__table-hover",tableHover:"dapp-core-component__main__table-hover","table-responsive":"dapp-core-component__main__table-responsive",tableResponsive:"dapp-core-component__main__table-responsive","transaction-hash":"dapp-core-component__main__transaction-hash",transactionHash:"dapp-core-component__main__transaction-hash","font-weight-light":"dapp-core-component__main__font-weight-light",fontWeightLight:"dapp-core-component__main__font-weight-light","font-weight-lighter":"dapp-core-component__main__font-weight-lighter",fontWeightLighter:"dapp-core-component__main__font-weight-lighter","font-weight-normal":"dapp-core-component__main__font-weight-normal",fontWeightNormal:"dapp-core-component__main__font-weight-normal","font-weight-medium":"dapp-core-component__main__font-weight-medium",fontWeightMedium:"dapp-core-component__main__font-weight-medium","font-weight-bold":"dapp-core-component__main__font-weight-bold",fontWeightBold:"dapp-core-component__main__font-weight-bold","font-weight-bolder":"dapp-core-component__main__font-weight-bolder",fontWeightBolder:"dapp-core-component__main__font-weight-bolder","font-italic":"dapp-core-component__main__font-italic",fontItalic:"dapp-core-component__main__font-italic","icon-state":"dapp-core-component__main__icon-state",iconState:"dapp-core-component__main__icon-state",half:"dapp-core-component__main__half","side-icon":"dapp-core-component__main__side-icon",sideIcon:"dapp-core-component__main__side-icon","toast-messages":"dapp-core-component__main__toast-messages",toastMessages:"dapp-core-component__main__toast-messages",toast:"dapp-core-component__main__toast",clickable:"dapp-core-component__main__clickable","toast-visible":"dapp-core-component__main__toast-visible",toastVisible:"dapp-core-component__main__toast-visible",progress:"dapp-core-component__main__progress","progress-bar":"dapp-core-component__main__progress-bar",progressBar:"dapp-core-component__main__progress-bar","content-height":"dapp-core-component__main__content-height",contentHeight:"dapp-core-component__main__content-height","tx-description":"dapp-core-component__main__tx-description",txDescription:"dapp-core-component__main__tx-description",close:"dapp-core-component__main__close","data-field":"dapp-core-component__main__data-field",dataField:"dapp-core-component__main__data-field",textarea:"dapp-core-component__main__textarea",highlighted:"dapp-core-component__main__highlighted","sc-call":"dapp-core-component__main__sc-call",scCall:"dapp-core-component__main__sc-call","small-font":"dapp-core-component__main__small-font",smallFont:"dapp-core-component__main__small-font","progress-steps":"dapp-core-component__main__progress-steps",progressSteps:"dapp-core-component__main__progress-steps",steps:"dapp-core-component__main__steps",dot:"dapp-core-component__main__dot","side-link":"dapp-core-component__main__side-link",sideLink:"dapp-core-component__main__side-link"};E();L();K();t();var xs=x(require("bignumber.js"));E();t();var bs=require("@elrondnetwork/erdjs"),Ss=x(require("bignumber.js"));E();N();$e();t();t();t();t();t();t();t();t();t();var Va=require("react");t();t();var vs=require("react"),ks=require("@elrondnetwork/erdjs-extension-provider");Be();j();L();N();Oe();t();var Is=require("react");Be();L();t();var ja=require("react"),As=require("@elrondnetwork/erdjs-hw-provider");Be();j();L();N();Oe();t();var at=require("react"),Cs=require("@elrondnetwork/erdjs-wallet-connect-provider");j();L();N();Oe();t();var za=require("react"),Ps=require("@elrondnetwork/erdjs-wallet-connect-provider");j();L();N();Oe();t();t();var qa=require("react");t();var Rs=require("react");t();var it=require("react"),Ns=require("socket.io-client");L();t();t();t();t();var Ka=o=>o.dappModal,Ms=f(Ka,o=>o.visible),Wo=f(Ka,o=>o.modal),K3=f(Wo,o=>o.showHeader),X3=f(Wo,o=>o.showFooter),Y3=f(Wo,o=>o.headerText),Z3=f(Wo,o=>o.footerText);t();var qe=x(require("react")),Ya=require("@fortawesome/react-fontawesome"),co=x(require("classnames"));t();var Fs=`.dapp-core-component__pageStateStyles__wrapper {
  color: inherit;
}

.dapp-core-component__pageStateStyles__state {
  color: inherit;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Fs));var Xa={wrapper:"dapp-core-component__pageStateStyles__wrapper",state:"dapp-core-component__pageStateStyles__state"};var Za=({icon:o,title:e,action:n,iconClass:r,dataTestId:p,description:m,iconBgClass:s,iconSize:u="5x",className:l="dapp-page-state"})=>{let g={wrapper:(0,co.default)(Xa.state,d.mAuto,d.p4,d.textCenter,l),iconContainer:(0,co.default)(d.iconState,d.mxAuto,{[s!=null?s:""]:Boolean(s)}),iconClass:(0,co.default)(r!=null&&r),title:(0,co.default)(d.h4,d.my4),description:d.mb3};return qe.default.createElement("div",{className:g.wrapper,"data-testid":p},o&&qe.default.createElement("span",{className:g.iconContainer},qe.default.createElement(Ya.FontAwesomeIcon,{icon:o,className:g.iconClass,size:u})),e&&qe.default.createElement("p",{className:g.title},e),m&&qe.default.createElement("div",{className:g.description},m),n)};t();var po=x(require("react"));t();var Ja=x(require("react")),Qa=x(require("classnames"));var ei=({color:o,"data-testid":e,className:n="dapp-dot"})=>Ja.default.createElement("span",{className:(0,Qa.default)(d.dot,o,n),"data-testid":e});var Go=x(require("classnames")),oi=({totalSteps:o,currentStep:e,className:n="dapp-progress-steps"})=>{let r=[];for(let p=1;p<=o;p+=1){let m=e===p,s=m||p<e?d.bgPrimary:d.bgSecondary;r.push(po.default.createElement(ei,{"data-testid":`step${p}${m?"active":""}`,key:p,color:s}))}return po.default.createElement("div",{className:(0,Go.default)(d.progressSteps,d.flexRow,d.justifyContentCenter,n)},po.default.createElement("div",{className:(0,Go.default)(d.steps,d.flexRow,d.justifyContentCenter,d.alignItemsCenter,d.positionRelative)},po.default.createElement("hr",{className:(0,Go.default)(d.positionAbsolute,d.w100,d.m0)}),r))};t();var B=x(require("react")),ii=require("@fortawesome/free-solid-svg-icons"),ci=require("@fortawesome/react-fontawesome");t();var S=x(require("react")),Bs=o=>S.createElement("svg",T({viewBox:"0 0 100 100",xmlns:"http://www.w3.org/2000/svg",role:"img"},o),S.createElement("g",{fillRule:"evenodd"},S.createElement("g",{transform:"translate(0 -1)",fillRule:"nonzero"},S.createElement("g",{transform:"translate(0 1)",id:"egld-token"},S.createElement("path",{d:"m79.6 19.424c-0.4-0.4383-0.6-0.8766-0.6-1.4245 0-0.54787 0.2-0.98617 0.6-1.4245 0.8-0.76702 2-0.76702 2.8 0 0.4 0.4383 0.6 0.8766 0.6 1.4245 0 0.54787-0.2 0.98617-0.6 1.4245-0.4 0.32872-0.9 0.54787-1.4 0.54787-0.5 0.10957-1-0.10957-1.4-0.54787"}),S.createElement("path",{id:"Path",d:"m74.556 24.448c-0.33333-0.44138-0.55556-0.88276-0.55556-1.4345s0.22222-0.9931 0.55556-1.4345c0.77778-0.77241 2.1111-0.77241 2.8889 0 0.33333 0.44138 0.55556 0.88276 0.55556 1.4345s-0.22222 0.9931-0.55556 1.4345c-0.44444 0.33103-0.88889 0.55172-1.4444 0.55172s-1-0.22069-1.4444-0.55172"}),S.createElement("path",{id:"Path",d:"m69.556 30.448c-0.33333-0.33103-0.55556-0.88276-0.55556-1.4345s0.22222-0.9931 0.55556-1.4345c0.77778-0.77241 2.1111-0.77241 2.8889 0 0.33333 0.33103 0.55556 0.88276 0.55556 1.4345s-0.22222 0.9931-0.55556 1.4345c-0.44444 0.44138-0.88889 0.55172-1.4444 0.55172s-1.1111-0.22069-1.4444-0.55172"}),S.createElement("path",{id:"Path",d:"m64.6 35.448c-0.4-0.33103-0.6-0.88276-0.6-1.4345s0.2-0.9931 0.6-1.4345c0.8-0.77241 2-0.77241 2.8 0 0.4 0.33103 0.6 0.88276 0.6 1.4345s-0.2 1.1034-0.6 1.4345c-0.4 0.44138-0.9 0.55172-1.4 0.55172s-1-0.11034-1.4-0.55172"}),S.createElement("path",{id:"Path",d:"m59.556 40.424c-0.44444-0.32872-0.55556-0.8766-0.55556-1.4245 0-0.54787 0.22222-0.98617 0.55556-1.4245 0.77778-0.76702 2.1111-0.76702 2.8889 0 0.44444 0.32872 0.55556 0.8766 0.55556 1.4245 0 0.54787-0.22222 1.0957-0.55556 1.4245-0.33333 0.4383-0.88889 0.54787-1.4444 0.54787-0.55556 0.10957-1.1111-0.10957-1.4444-0.54787"}),S.createElement("path",{id:"Path",d:"m53 44.014c0-0.55172 0.22222-0.9931 0.55556-1.4345 0.77778-0.77241 2.1111-0.77241 2.8889 0 0.44444 0.44138 0.55556 0.88276 0.55556 1.4345s-0.22222 0.9931-0.55556 1.4345c-0.33333 0.33103-0.88889 0.55172-1.4444 0.55172s-1-0.22069-1.4444-0.55172c-0.33333-0.33103-0.55556-0.88276-0.55556-1.4345z"}),S.createElement("path",{id:"Path",d:"m43.556 56.424c-0.33333-0.32872-0.55556-0.8766-0.55556-1.4245 0-0.54787 0.22222-0.98617 0.55556-1.4245 0.77778-0.76702 2.1111-0.76702 2.8889 0 0.44444 0.4383 0.55556 0.8766 0.55556 1.4245 0 0.54787-0.22222 0.98617-0.55556 1.4245-0.33333 0.4383-0.88889 0.54787-1.4444 0.54787-0.55556 0.10957-1-0.10957-1.4444-0.54787"}),S.createElement("path",{id:"Path",d:"m38.6 61.424c-0.4-0.32872-0.6-0.8766-0.6-1.4245 0-0.54787 0.2-0.98617 0.6-1.4245 0.8-0.76702 2-0.76702 2.8 0 0.4 0.4383 0.6 0.8766 0.6 1.4245 0 0.54787-0.2 0.98617-0.6 1.4245-0.4 0.4383-0.9 0.54787-1.4 0.54787-0.5 0.10957-1-0.10957-1.4-0.54787"}),S.createElement("path",{id:"Path",d:"m33.556 66.448c-0.33333-0.33103-0.55556-0.88276-0.55556-1.4345s0.22222-1.1034 0.55556-1.4345c0.77778-0.77241 2.1111-0.77241 2.8889 0 0.33333 0.33103 0.55556 0.88276 0.55556 1.4345s-0.22222 0.9931-0.55556 1.4345c-0.44444 0.44138-0.88889 0.55172-1.4444 0.55172s-1-0.22069-1.4444-0.55172"}),S.createElement("path",{id:"Path",d:"m27.556 71.448c-0.44444-0.33103-0.55556-0.88276-0.55556-1.4345s0.22222-0.9931 0.55556-1.4345c0.77778-0.77241 2.1111-0.77241 2.8889 0 0.33333 0.44138 0.55556 0.88276 0.55556 1.4345s-0.22222 1.1034-0.55556 1.4345c-0.44444 0.44138-0.88889 0.55172-1.4444 0.55172s-1.1111-0.11034-1.4444-0.55172"}),S.createElement("path",{id:"Path",d:"m22.615 76.448c-0.41026-0.33103-0.61538-0.88276-0.61538-1.4345s0.20513-0.9931 0.61538-1.4345c0.82051-0.77241 2.0513-0.77241 2.7692 0 0.41026 0.44138 0.61538 0.88276 0.61538 1.4345s-0.20513 1.1034-0.61538 1.4345c-0.41026 0.33103-0.92308 0.55172-1.4359 0.55172-0.51282 0-1.0256-0.11034-1.3333-0.55172"}),S.createElement("path",{id:"Path",d:"m17.556 82.424c-0.44444-0.4383-0.55556-0.8766-0.55556-1.4245 0-0.54787 0.22222-0.98617 0.55556-1.4245 0.77778-0.76702 2.1111-0.76702 2.8889 0 0.44444 0.4383 0.55556 0.8766 0.55556 1.4245 0 0.54787-0.22222 0.98617-0.55556 1.4245-0.33333 0.32872-0.88889 0.54787-1.4444 0.54787-0.44444 0.10957-1-0.10957-1.4444-0.54787"}),S.createElement("path",{id:"Path",d:"m20.444 19.424c0.33333-0.4383 0.55556-0.8766 0.55556-1.4245 0-0.54787-0.22222-0.98617-0.55556-1.4245-0.77778-0.76702-2.1111-0.76702-2.8889 0-0.44444 0.4383-0.55556 0.8766-0.55556 1.4245 0 0.54787 0.22222 0.98617 0.55556 1.4245 0.33333 0.32872 0.88889 0.54787 1.4444 0.54787 0.55556 0.10957 1.1111-0.10957 1.4444-0.54787"}),S.createElement("path",{id:"Path",d:"m25.4 24.448c0.4-0.44138 0.6-0.88276 0.6-1.4345s-0.2-0.9931-0.6-1.4345c-0.8-0.77241-2-0.77241-2.8 0-0.4 0.44138-0.6 0.88276-0.6 1.4345s0.2 0.9931 0.6 1.4345c0.4 0.33103 0.9 0.55172 1.4 0.55172s1-0.22069 1.4-0.55172"}),S.createElement("path",{id:"Path",d:"m30.444 30.448c0.33333-0.33103 0.55556-0.88276 0.55556-1.4345s-0.22222-0.9931-0.55556-1.4345c-0.77778-0.77241-2.1111-0.77241-2.8889 0-0.33333 0.33103-0.55556 0.88276-0.55556 1.4345s0.22222 0.9931 0.55556 1.4345c0.44444 0.44138 0.88889 0.55172 1.4444 0.55172 0.44444 0 1-0.22069 1.4444-0.55172"}),S.createElement("path",{id:"Path",d:"m36.444 35.448c0.33333-0.33103 0.55556-0.88276 0.55556-1.4345s-0.22222-0.9931-0.55556-1.4345c-0.77778-0.77241-2.1111-0.77241-2.8889 0-0.33333 0.33103-0.55556 0.88276-0.55556 1.4345s0.22222 1.1034 0.55556 1.4345c0.44444 0.44138 0.88889 0.55172 1.4444 0.55172 0.44444 0 1-0.11034 1.4444-0.55172"}),S.createElement("path",{id:"Path",d:"m41.385 40.424c0.41026-0.32872 0.61538-0.8766 0.61538-1.4245 0-0.54787-0.20513-0.98617-0.61538-1.4245-0.71795-0.76702-2.0513-0.76702-2.7692 0-0.41026 0.32872-0.61538 0.8766-0.61538 1.4245 0 0.54787 0.20513 1.0957 0.61538 1.4245 0.41026 0.4383 0.92308 0.54787 1.4359 0.54787 0.51282 0.10957 1.0256-0.10957 1.3333-0.54787"}),S.createElement("path",{id:"Path",d:"m47 44.014c0-0.55172-0.22222-0.9931-0.55556-1.4345-0.77778-0.77241-2.1111-0.77241-2.8889 0-0.44444 0.44138-0.55556 0.88276-0.55556 1.4345s0.22222 0.9931 0.55556 1.4345c0.33333 0.33103 0.88889 0.55172 1.4444 0.55172s1-0.22069 1.4444-0.55172c0.44444-0.33103 0.55556-0.88276 0.55556-1.4345z"}),S.createElement("path",{id:"Path",d:"m51.444 51.448c0.44444-0.44138 0.55556-0.88276 0.55556-1.4345s-0.22222-0.9931-0.55556-1.4345c-0.77778-0.77241-2.1111-0.77241-2.8889 0-0.44444 0.44138-0.55556 0.88276-0.55556 1.4345s0.22222 0.9931 0.55556 1.4345c0.33333 0.33103 0.88889 0.55172 1.4444 0.55172s1.1111-0.22069 1.4444-0.55172"}),S.createElement("path",{id:"Path",d:"m56.385 56.424c0.41026-0.32872 0.61538-0.8766 0.61538-1.4245 0-0.54787-0.20513-0.98617-0.61538-1.4245-0.82051-0.76702-2.0513-0.76702-2.7692 0-0.41026 0.4383-0.61538 0.8766-0.61538 1.4245 0 0.54787 0.20513 0.98617 0.61538 1.4245 0.41026 0.4383 0.92308 0.54787 1.4359 0.54787 0.41026 0.10957 0.92308-0.10957 1.3333-0.54787"}),S.createElement("path",{id:"Path",d:"m62.444 61.424c0.33333-0.32872 0.55556-0.8766 0.55556-1.4245 0-0.54787-0.22222-0.98617-0.55556-1.4245-0.77778-0.76702-2.1111-0.76702-2.8889 0-0.33333 0.4383-0.55556 0.8766-0.55556 1.4245 0 0.54787 0.22222 0.98617 0.55556 1.4245 0.44444 0.4383 0.88889 0.54787 1.4444 0.54787 0.44444 0.10957 1-0.10957 1.4444-0.54787"}),S.createElement("path",{id:"Path",d:"m67.444 66.448c0.33333-0.33103 0.55556-0.88276 0.55556-1.4345s-0.22222-1.1034-0.55556-1.4345c-0.77778-0.77241-2.1111-0.77241-2.8889 0-0.33333 0.33103-0.55556 0.88276-0.55556 1.4345s0.22222 0.9931 0.55556 1.4345c0.44444 0.44138 0.88889 0.55172 1.4444 0.55172s1-0.22069 1.4444-0.55172"}),S.createElement("path",{id:"Path",d:"m72.444 71.448c0.44444-0.33103 0.55556-0.88276 0.55556-1.4345s-0.22222-0.9931-0.55556-1.4345c-0.77778-0.77241-2.1111-0.77241-2.8889 0-0.33333 0.44138-0.55556 0.88276-0.55556 1.4345s0.22222 1.1034 0.55556 1.4345c0.44444 0.44138 0.88889 0.55172 1.4444 0.55172s1.1111-0.11034 1.4444-0.55172"}),S.createElement("path",{id:"a",d:"m77.444 76.448c0.44444-0.33103 0.55556-0.88276 0.55556-1.4345s-0.22222-0.9931-0.55556-1.4345c-0.77778-0.77241-2.1111-0.77241-2.8889 0-0.44444 0.44138-0.55556 0.88276-0.55556 1.4345s0.22222 1.1034 0.55556 1.4345c0.33333 0.33103 0.88889 0.55172 1.4444 0.55172s1.1111-0.11034 1.4444-0.55172"}),S.createElement("path",{d:"m82.385 82.424c0.41026-0.4383 0.61538-0.8766 0.61538-1.4245 0-0.54787-0.20513-0.98617-0.61538-1.4245-0.82051-0.76702-2.0513-0.76702-2.7692 0-0.41026 0.4383-0.61538 0.8766-0.61538 1.4245 0 0.54787 0.20513 0.98617 0.61538 1.4245 0.41026 0.32872 0.92308 0.54787 1.4359 0.54787 0.41026 0.10957 0.92308-0.10957 1.3333-0.54787"}),S.createElement("path",{d:"m33.036 21c5.2647-3.4327 11.036-5.0481 17.414-5.149 6.2771 0 12.048 1.7163 17.414 5.149l11.137-11.106c-8.7069-6.5625-18.123-9.8942-28.551-9.8942-10.428 0-19.945 3.3317-28.449 9.8942l11.036 11.106z"}),S.createElement("path",{d:"m21 66.742c-3.5854-5.3754-5.3268-11.157-5.3268-17.343 0-6.2883 1.7415-12.069 5.3268-17.343l-11.166-11.055c-6.5561 8.5196-9.8341 17.952-9.8341 28.5 0 10.345 3.278 19.879 9.9366 28.5l11.063-11.258z"}),S.createElement("path",{d:"m67.964 78c-5.2647 3.5333-11.036 5.3504-17.313 5.4513-6.3783 0-12.25-1.7162-17.515-5.3504l-11.137 11.004c8.6057 6.6628 18.123 9.9941 28.652 9.8932 10.428-0.10095 19.844-3.4323 28.348-9.8932l-11.036-11.105z"}),S.createElement("path",{d:"m79 32.258c3.5333 5.3754 5.3504 11.258 5.2494 17.648-0.10095 6.1868-1.8171 11.867-5.2494 17.039l11.105 11.055c6.5618-8.5196 9.8932-17.952 9.8932-28.297 0.10095-10.548-3.2304-20.082-9.8932-28.703l-11.105 11.258z"}),S.createElement("path",{d:"m90 81c4.3636 0 8 3.5325 8 8 0 4.3636-3.5325 8-8 8-4.3636 0-8-3.5325-8-8s3.5325-8 8-8"}),S.createElement("path",{d:"m10 81c4.4211 0 8 3.5325 8 8 0 4.3636-3.5789 8-8 8-4.4211 0-8-3.5325-8-8s3.5789-8 8-8"}),S.createElement("path",{d:"m90 1c4.4211 0 8 3.5325 8 8 0 4.4675-3.5789 8-8 8s-8-3.5325-8-8c0-4.4675 3.5789-8 8-8"}),S.createElement("path",{d:"m10 1c4.3636 0 8 3.5789 8 8 0 4.4211-3.5325 8-8 8-4.3636 0-8-3.5789-8-8 0-4.4211 3.5325-8 8-8"}))))),ni=Bs;t();var ct=x(require("react")),pt=x(require("classnames"));t();var Os=`.dapp-core-component__tokenSymbol__token-symbol,
.dapp-core-component__tokenSymbol__token-symbol-combined {
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #fff;
  border-radius: 50%;
}
.dapp-core-component__tokenSymbol__token-symbol svg #dapp-core-component__tokenSymbol__egld-token,
.dapp-core-component__tokenSymbol__token-symbol-combined svg #dapp-core-component__tokenSymbol__egld-token {
  fill: #000;
}

.dapp-core-component__tokenSymbol__token-symbol-combined {
  background-color: rgba(0, 0, 0, 0);
}

.dapp-core-component__tokenSymbol__token-symbol {
  border: 1px solid #000;
  width: 2rem;
  height: 2rem;
}
.dapp-core-component__tokenSymbol__token-symbol img {
  width: 100%;
  height: 100%;
}
.dapp-core-component__tokenSymbol__token-symbol svg {
  width: 1.25rem;
  height: 1.25rem;
}
.dapp-core-component__tokenSymbol__token-symbol svg.dapp-core-component__tokenSymbol__expanded {
  width: 100% !important;
  height: 100% !important;
}

.dapp-core-component__tokenSymbol__token-symbol-combined {
  width: 4.5rem;
  height: 4.5rem;
}
.dapp-core-component__tokenSymbol__token-symbol-combined svg circle {
  fill: #fff;
}
.dapp-core-component__tokenSymbol__token-symbol-combined svg {
  width: 4.25rem;
  height: 4.25rem;
}
.dapp-core-component__tokenSymbol__token-symbol-combined.dapp-core-component__tokenSymbol__small {
  width: 2.5rem;
  min-width: 2.5rem;
  height: 2.5rem;
}
.dapp-core-component__tokenSymbol__token-symbol-combined.dapp-core-component__tokenSymbol__small svg {
  width: 2.5rem;
  height: 2.5rem;
}

.dapp-core-component__tokenSymbol__token-icon-lg .dapp-core-component__tokenSymbol__token-symbol {
  width: 3rem;
  height: 3rem;
}
.dapp-core-component__tokenSymbol__token-icon-lg .dapp-core-component__tokenSymbol__token-symbol svg {
  width: 2.25rem;
  height: 2.25rem;
}

.dapp-core-component__tokenSymbol__token-icon-xl .dapp-core-component__tokenSymbol__token-symbol {
  width: 4rem;
  height: 4rem;
}
.dapp-core-component__tokenSymbol__token-icon-xl .dapp-core-component__tokenSymbol__token-symbol svg {
  width: 2.3rem;
  height: 2.3rem;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Os));var $o={"token-symbol":"dapp-core-component__tokenSymbol__token-symbol",tokenSymbol:"dapp-core-component__tokenSymbol__token-symbol","token-symbol-combined":"dapp-core-component__tokenSymbol__token-symbol-combined",tokenSymbolCombined:"dapp-core-component__tokenSymbol__token-symbol-combined","egld-token":"dapp-core-component__tokenSymbol__egld-token",egldToken:"dapp-core-component__tokenSymbol__egld-token",expanded:"dapp-core-component__tokenSymbol__expanded",small:"dapp-core-component__tokenSymbol__small","token-icon-lg":"dapp-core-component__tokenSymbol__token-icon-lg",tokenIconLg:"dapp-core-component__tokenSymbol__token-icon-lg","token-icon-xl":"dapp-core-component__tokenSymbol__token-icon-xl",tokenIconXl:"dapp-core-component__tokenSymbol__token-icon-xl"};var ti=({children:o,className:e="dapp-simple-token-symbol"})=>ct.default.createElement("div",{className:(0,pt.default)($o.tokenSymbol,e)},o),ri=({small:o,children:e,className:n="dapp-combined-token-symbol"})=>ct.default.createElement("div",{className:(0,pt.default)($o.tokenSymbolCombined,{[$o.small]:o},n)},e);t();var Us=`.dapp-core-component__tokenDetailsStyles__token-symbol-custom-token {
  color: inherit;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Us));var ai={"token-symbol-custom-token":"dapp-core-component__tokenDetailsStyles__token-symbol-custom-token",tokenSymbolCustomToken:"dapp-core-component__tokenDetailsStyles__token-symbol-custom-token"};var st=o=>{let e=o.split("-");return o.includes("-")?`${e[0]}-${e[1]}`:o};function Ws(o,e){return e?B.default.createElement("img",{className:ai.tokenSymbolCustomToken,src:e}):o?B.default.createElement(ni,null):B.default.createElement(ci.FontAwesomeIcon,{icon:ii.faDiamond})}var mt=(o,e)=>{let n=Uo(),r=o===n;return{token:o,symbol:o?o.split("-")[0]:"",label:o,icon:()=>Ws(r,e)}},te=class extends B.Component{render(){return null}};te.Token=e=>B.default.createElement(B.default.Fragment,null,e.token),te.Symbol=e=>B.default.createElement(B.default.Fragment,null,mt(st(e.token),e.tokenAvatar).symbol),te.Label=e=>B.default.createElement(B.default.Fragment,null,mt(st(e.token),e.tokenAvatar).label),te.Icon=e=>{var r;let n=mt(st(e.token),e.tokenAvatar).icon;return B.default.createElement("div",{className:(r=e.className)!=null?r:"dapp-token-details-icon"},e.combined?B.default.createElement(ri,{small:e.small},B.default.createElement(n,null)):B.default.createElement(ti,null,B.default.createElement(n,null)))};t();var R=x(require("react")),Ce=x(require("classnames"));E();var Gs=(o,e)=>[...o.matchAll(new RegExp(e,"gi"))].map(n=>n.index),pi=({data:o,highlight:e,isScCall:n,className:r="dapp-transaction-data"})=>{let p=R.default.createElement(R.default.Fragment,null,o),[m,...s]=e&&n?e.split("@"):[];if(o&&e&&Gs(o,e).length===1)switch(!0){case o.startsWith(e):{let[,u]=o.split(e);p=R.default.createElement(R.default.Fragment,null,e,R.default.createElement("span",{className:d.textMuted},u));break}case o.endsWith(e):{let[u]=o.split(e);p=R.default.createElement(R.default.Fragment,null,R.default.createElement("span",{className:d.textMuted},u),e);break}default:{let[u,l]=o.split(e);p=R.default.createElement(R.default.Fragment,null,R.default.createElement("span",{className:d.textMuted},u),R.default.createElement("span",{className:d.highlighted},e),R.default.createElement("span",{className:d.textMuted},l));break}}return R.default.createElement(R.default.Fragment,null,m&&R.default.createElement("div",{className:(0,Ce.default)(d.formGroup,d.mb0,d.dataField,d.mw100,d.w100,r)},R.default.createElement("span",{className:(0,Ce.default)(d.formLabel,d.textSecondary,d.dBlock)},"SC Call"),R.default.createElement("div",{"data-testid":"confirmScCall",className:(0,Ce.default)(d.textarea,d.scCall,d.formControl,d.mt1,d.textBreak,d.w100)},[ro(m),...s].join("@"))),R.default.createElement("div",{className:(0,Ce.default)(d.formGroup,d.mb0,d.dataField,d.mw100,d.w100,r)},R.default.createElement("span",{className:(0,Ce.default)(d.formLabel,d.textSecondary,d.dBlock)},"Data"),R.default.createElement("div",{"data-testid":"confirmData",className:(0,Ce.default)(d.textarea,d.formControl,d.mt1,d.textBreak,d.w100)},o?p:nr)))};t();var $=x(require("classnames"));var si=require("react"),mi=(o=null)=>(0,si.useMemo)(()=>({formGroup:(0,$.default)(d.formGroup,d.textBreak,d.textLeft),formLabel:(0,$.default)(d.textBreak,d.textSecondary),icon:d.textWhite,contentWrapper:(0,$.default)(d.dFlex,d.flexColumn,d.justifyContentStart,d.flexRow,d.justifyContentBetween,d.mb3),tokenWrapper:(0,$.default)(d.mb3,d.mb0,d.dFlex,d.flexColumn,d.alignItemsStart),tokenLabel:(0,$.default)(d.textSecondary,d.textLeft),tokenValue:(0,$.default)(d.dFlex,d.alignItemsCenter,d.mt1),scamReport:d.textWarning,scamReportIcon:(0,$.default)(d.textWarning,d.mr1),tokenAmountLabel:(0,$.default)(d.textSecondary,d.textLeft),tokenAmountValue:(0,$.default)(d.dFlex,d.alignItemsCenter),dataFormGroup:(0,$.default)(d.formGroup,d.textLeft),errorMessage:(0,$.default)(d.textDanger,d.dFlex,d.justifyContentCenter,d.alignItemsCenter),buttonsWrapper:(0,$.default)(d.dFlex,d.alignItemsCenter,d.justifyContentEnd,d.mt1),cancelButton:(0,$.default)(d.btn,d.btnDark,d.textWhite,d.dFlex,d.mr2),signButton:(0,$.default)(d.btn,o?d.btnWarning:d.btnPrimary,d.dFlex,d.ml2)}),[o]);var $s=({onSignTransaction:o,handleClose:e,onPrev:n,title:r,waitingForDevice:p,currentTransaction:m,error:s,allTransactions:u,isLastTransaction:l,currentStep:g,className:b})=>{let h=Uo();if(!m)return null;let w=m.transaction.getData().toString(),{network:P}=ao(),{tokenId:D,nonce:O,amount:Ho,type:_t,multiTxData:re,receiver:fe}=m.transactionTokenInfo,V=Boolean(D&&Hn({tokenId:D,erdLabel:h})),he=g===0,Pe=Ti=>{Ti.preventDefault(),he?e():n()},jo=_t&&re&&!w.endsWith(re),ae="Sign & Continue";ae=p?"Check your Ledger":ae,ae=l&&!p?"Sign & Submit":ae,ae=jo?"Continue":ae;let li=`${D}-${O}`,{tokenDecimals:ui,tokenAvatar:gi}=Ua({tokenId:O&&O.length>0?li:D}),fi=Ee({input:V?Ho:m.transaction.getValue().toString(),decimals:V?ui:Number(P.decimals),digits:Number(P.digits),showLastNonZeroDecimal:!1,addCommas:!0}),zo=m.receiverScamInfo,hi=u.length>1,U=mi(zo);return v.default.createElement(Za,{icon:s?Vo.faTimes:null,iconClass:U.icon,iconBgClass:s?d.bgDanger:d.bgWarning,iconSize:"3x",className:b,title:r||"Confirm on Ledger",description:v.default.createElement(v.default.Fragment,null,m.transaction&&v.default.createElement(v.default.Fragment,null,hi&&v.default.createElement(oi,{totalSteps:u.length,currentStep:g+1,className:d.mb4}),v.default.createElement("div",{className:U.formGroup,"data-testid":"transactionTitle"},v.default.createElement("div",{className:U.formLabel},"To "),re?new _i.Address(fe).bech32():m.transaction.getReceiver().toString(),zo&&v.default.createElement("div",{className:U.scamReport},v.default.createElement("span",null,v.default.createElement(di.FontAwesomeIcon,{icon:Vo.faExclamationTriangle,className:U.scamReportIcon}),v.default.createElement("small",null,zo)))),v.default.createElement("div",{className:U.contentWrapper},v.default.createElement("div",{className:U.tokenWrapper},v.default.createElement("div",{className:U.tokenLabel},"Token"),v.default.createElement("div",{className:U.tokenValue},v.default.createElement(te.Icon,{tokenAvatar:gi,token:D||h}),v.default.createElement("div",{className:d.mr1}),v.default.createElement(te.Label,{token:D||h}))),v.default.createElement("div",null,v.default.createElement("div",{className:U.tokenAmountLabel},"Amount"),v.default.createElement("div",{className:U.tokenAmountValue},v.default.createElement("div",{className:d.mr1},fi),v.default.createElement(te.Symbol,{token:D||h})))),v.default.createElement("div",{className:U.dataFormGroup},m.transaction.getData()&&v.default.createElement(pi,{isScCall:!D,data:m.transaction.getData().toString(),highlight:re})),s&&v.default.createElement("p",{className:U.errorMessage},s))),action:v.default.createElement("div",{className:U.buttonsWrapper},v.default.createElement("button",{id:"closeButton","data-testid":"closeButton",onClick:Pe,className:U.cancelButton},he?"Cancel":"Back"),v.default.createElement("button",{type:"button",className:U.signButton,id:"signBtn","data-testid":"signBtn",onClick:o,disabled:p},ae))})};0&&(module.exports={SignStep});
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
