// @elrondnetwork/dapp-core/__commonjs/UI/TransactionsTable/components/tests/TransactionMethod.test.js
"use strict";var Mn=Object.create;var pe=Object.defineProperty,In=Object.defineProperties,Ln=Object.getOwnPropertyDescriptor,Rn=Object.getOwnPropertyDescriptors,jn=Object.getOwnPropertyNames,Me=Object.getOwnPropertySymbols,zn=Object.getPrototypeOf,Le=Object.prototype.hasOwnProperty,Fn=Object.prototype.propertyIsEnumerable;var Ie=(o,e,n)=>e in o?pe(o,e,{enumerable:!0,configurable:!0,writable:!0,value:n}):o[e]=n,ie=(o,e)=>{for(var n in e||(e={}))Le.call(e,n)&&Ie(o,n,e[n]);if(Me)for(var n of Me(e))Fn.call(e,n)&&Ie(o,n,e[n]);return o},_e=(o,e)=>In(o,Rn(e));var Un=(o,e)=>()=>(o&&(e=o(o=0)),e);var Q=(o,e)=>()=>(e||o((e={exports:{}}).exports,e),e.exports);var Cn=(o,e,n,a)=>{if(e&&typeof e=="object"||typeof e=="function")for(let r of jn(e))!Le.call(o,r)&&r!==n&&pe(o,r,{get:()=>e[r],enumerable:!(a=Ln(e,r))||a.enumerable});return o};var T=(o,e,n)=>(n=o!=null?Mn(zn(o)):{},Cn(e||!o||!o.__esModule?pe(n,"default",{value:o,enumerable:!0}):n,o));var ze=Q($=>{"use strict";c();$.byteLength=Pn;$.toByteArray=Gn;$.fromByteArray=Wn;var v=[],k=[],Bn=typeof Uint8Array!="undefined"?Uint8Array:Array,me="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";for(z=0,Re=me.length;z<Re;++z)v[z]=me[z],k[me.charCodeAt(z)]=z;var z,Re;k["-".charCodeAt(0)]=62;k["_".charCodeAt(0)]=63;function je(o){var e=o.length;if(e%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var n=o.indexOf("=");n===-1&&(n=e);var a=n===e?0:4-n%4;return[n,a]}function Pn(o){var e=je(o),n=e[0],a=e[1];return(n+a)*3/4-a}function On(o,e,n){return(e+n)*3/4-n}function Gn(o){var e,n=je(o),a=n[0],r=n[1],t=new Bn(On(o,a,r)),p=0,s=r>0?a-4:a,l;for(l=0;l<s;l+=4)e=k[o.charCodeAt(l)]<<18|k[o.charCodeAt(l+1)]<<12|k[o.charCodeAt(l+2)]<<6|k[o.charCodeAt(l+3)],t[p++]=e>>16&255,t[p++]=e>>8&255,t[p++]=e&255;return r===2&&(e=k[o.charCodeAt(l)]<<2|k[o.charCodeAt(l+1)]>>4,t[p++]=e&255),r===1&&(e=k[o.charCodeAt(l)]<<10|k[o.charCodeAt(l+1)]<<4|k[o.charCodeAt(l+2)]>>2,t[p++]=e>>8&255,t[p++]=e&255),t}function Vn(o){return v[o>>18&63]+v[o>>12&63]+v[o>>6&63]+v[o&63]}function Zn(o,e,n){for(var a,r=[],t=e;t<n;t+=3)a=(o[t]<<16&16711680)+(o[t+1]<<8&65280)+(o[t+2]&255),r.push(Vn(a));return r.join("")}function Wn(o){for(var e,n=o.length,a=n%3,r=[],t=16383,p=0,s=n-a;p<s;p+=t)r.push(Zn(o,p,p+t>s?s:p+t));return a===1?(e=o[n-1],r.push(v[e>>2]+v[e<<4&63]+"==")):a===2&&(e=(o[n-2]<<8)+o[n-1],r.push(v[e>>10]+v[e>>4&63]+v[e<<2&63]+"=")),r.join("")}});var Fe=Q(de=>{c();de.read=function(o,e,n,a,r){var t,p,s=r*8-a-1,l=(1<<s)-1,f=l>>1,u=-7,g=n?r-1:0,y=n?-1:1,x=o[e+g];for(g+=y,t=x&(1<<-u)-1,x>>=-u,u+=s;u>0;t=t*256+o[e+g],g+=y,u-=8);for(p=t&(1<<-u)-1,t>>=-u,u+=a;u>0;p=p*256+o[e+g],g+=y,u-=8);if(t===0)t=1-f;else{if(t===l)return p?NaN:(x?-1:1)*(1/0);p=p+Math.pow(2,a),t=t-f}return(x?-1:1)*p*Math.pow(2,t-a)};de.write=function(o,e,n,a,r,t){var p,s,l,f=t*8-r-1,u=(1<<f)-1,g=u>>1,y=r===23?Math.pow(2,-24)-Math.pow(2,-77):0,x=a?0:t-1,I=a?1:-1,Y=e<0||e===0&&1/e<0?1:0;for(e=Math.abs(e),isNaN(e)||e===1/0?(s=isNaN(e)?1:0,p=u):(p=Math.floor(Math.log(e)/Math.LN2),e*(l=Math.pow(2,-p))<1&&(p--,l*=2),p+g>=1?e+=y/l:e+=y*Math.pow(2,1-g),e*l>=2&&(p++,l/=2),p+g>=u?(s=0,p=u):p+g>=1?(s=(e*l-1)*Math.pow(2,r),p=p+g):(s=e*Math.pow(2,g-1)*Math.pow(2,r),p=0));r>=8;o[n+x]=s&255,x+=I,s/=256,r-=8);for(p=p<<r|s,f+=r;f>0;o[n+x]=p&255,x+=I,p/=256,f-=8);o[n+x-I]|=Y*128}});var $e=Q(P=>{"use strict";c();var se=ze(),B=Fe(),Ue=typeof Symbol=="function"&&typeof Symbol.for=="function"?Symbol.for("nodejs.util.inspect.custom"):null;P.Buffer=i;P.SlowBuffer=Kn;P.INSPECT_MAX_BYTES=50;var X=2147483647;P.kMaxLength=X;i.TYPED_ARRAY_SUPPORT=Hn();!i.TYPED_ARRAY_SUPPORT&&typeof console!="undefined"&&typeof console.error=="function"&&console.error("This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.");function Hn(){try{var o=new Uint8Array(1),e={foo:function(){return 42}};return Object.setPrototypeOf(e,Uint8Array.prototype),Object.setPrototypeOf(o,e),o.foo()===42}catch(n){return!1}}Object.defineProperty(i.prototype,"parent",{enumerable:!0,get:function(){if(!!i.isBuffer(this))return this.buffer}});Object.defineProperty(i.prototype,"offset",{enumerable:!0,get:function(){if(!!i.isBuffer(this))return this.byteOffset}});function E(o){if(o>X)throw new RangeError('The value "'+o+'" is invalid for option "size"');var e=new Uint8Array(o);return Object.setPrototypeOf(e,i.prototype),e}function i(o,e,n){if(typeof o=="number"){if(typeof e=="string")throw new TypeError('The "string" argument must be of type string. Received type number');return ue(o)}return Pe(o,e,n)}i.poolSize=8192;function Pe(o,e,n){if(typeof o=="string")return Qn(o,e);if(ArrayBuffer.isView(o))return $n(o);if(o==null)throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof o);if(N(o,ArrayBuffer)||o&&N(o.buffer,ArrayBuffer)||typeof SharedArrayBuffer!="undefined"&&(N(o,SharedArrayBuffer)||o&&N(o.buffer,SharedArrayBuffer)))return ge(o,e,n);if(typeof o=="number")throw new TypeError('The "value" argument must not be of type number. Received type number');var a=o.valueOf&&o.valueOf();if(a!=null&&a!==o)return i.from(a,e,n);var r=Xn(o);if(r)return r;if(typeof Symbol!="undefined"&&Symbol.toPrimitive!=null&&typeof o[Symbol.toPrimitive]=="function")return i.from(o[Symbol.toPrimitive]("string"),e,n);throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof o)}i.from=function(o,e,n){return Pe(o,e,n)};Object.setPrototypeOf(i.prototype,Uint8Array.prototype);Object.setPrototypeOf(i,Uint8Array);function Oe(o){if(typeof o!="number")throw new TypeError('"size" argument must be of type number');if(o<0)throw new RangeError('The value "'+o+'" is invalid for option "size"')}function Yn(o,e,n){return Oe(o),o<=0?E(o):e!==void 0?typeof n=="string"?E(o).fill(e,n):E(o).fill(e):E(o)}i.alloc=function(o,e,n){return Yn(o,e,n)};function ue(o){return Oe(o),E(o<0?0:be(o)|0)}i.allocUnsafe=function(o){return ue(o)};i.allocUnsafeSlow=function(o){return ue(o)};function Qn(o,e){if((typeof e!="string"||e==="")&&(e="utf8"),!i.isEncoding(e))throw new TypeError("Unknown encoding: "+e);var n=Ge(o,e)|0,a=E(n),r=a.write(o,e);return r!==n&&(a=a.slice(0,r)),a}function le(o){for(var e=o.length<0?0:be(o.length)|0,n=E(e),a=0;a<e;a+=1)n[a]=o[a]&255;return n}function $n(o){if(N(o,Uint8Array)){var e=new Uint8Array(o);return ge(e.buffer,e.byteOffset,e.byteLength)}return le(o)}function ge(o,e,n){if(e<0||o.byteLength<e)throw new RangeError('"offset" is outside of buffer bounds');if(o.byteLength<e+(n||0))throw new RangeError('"length" is outside of buffer bounds');var a;return e===void 0&&n===void 0?a=new Uint8Array(o):n===void 0?a=new Uint8Array(o,e):a=new Uint8Array(o,e,n),Object.setPrototypeOf(a,i.prototype),a}function Xn(o){if(i.isBuffer(o)){var e=be(o.length)|0,n=E(e);return n.length===0||o.copy(n,0,0,e),n}if(o.length!==void 0)return typeof o.length!="number"||he(o.length)?E(0):le(o);if(o.type==="Buffer"&&Array.isArray(o.data))return le(o.data)}function be(o){if(o>=X)throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x"+X.toString(16)+" bytes");return o|0}function Kn(o){return+o!=o&&(o=0),i.alloc(+o)}i.isBuffer=function(e){return e!=null&&e._isBuffer===!0&&e!==i.prototype};i.compare=function(e,n){if(N(e,Uint8Array)&&(e=i.from(e,e.offset,e.byteLength)),N(n,Uint8Array)&&(n=i.from(n,n.offset,n.byteLength)),!i.isBuffer(e)||!i.isBuffer(n))throw new TypeError('The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array');if(e===n)return 0;for(var a=e.length,r=n.length,t=0,p=Math.min(a,r);t<p;++t)if(e[t]!==n[t]){a=e[t],r=n[t];break}return a<r?-1:r<a?1:0};i.isEncoding=function(e){switch(String(e).toLowerCase()){case"hex":case"utf8":case"utf-8":case"ascii":case"latin1":case"binary":case"base64":case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return!0;default:return!1}};i.concat=function(e,n){if(!Array.isArray(e))throw new TypeError('"list" argument must be an Array of Buffers');if(e.length===0)return i.alloc(0);var a;if(n===void 0)for(n=0,a=0;a<e.length;++a)n+=e[a].length;var r=i.allocUnsafe(n),t=0;for(a=0;a<e.length;++a){var p=e[a];if(N(p,Uint8Array))t+p.length>r.length?i.from(p).copy(r,t):Uint8Array.prototype.set.call(r,p,t);else if(i.isBuffer(p))p.copy(r,t);else throw new TypeError('"list" argument must be an Array of Buffers');t+=p.length}return r};function Ge(o,e){if(i.isBuffer(o))return o.length;if(ArrayBuffer.isView(o)||N(o,ArrayBuffer))return o.byteLength;if(typeof o!="string")throw new TypeError('The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type '+typeof o);var n=o.length,a=arguments.length>2&&arguments[2]===!0;if(!a&&n===0)return 0;for(var r=!1;;)switch(e){case"ascii":case"latin1":case"binary":return n;case"utf8":case"utf-8":return fe(o).length;case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return n*2;case"hex":return n>>>1;case"base64":return Qe(o).length;default:if(r)return a?-1:fe(o).length;e=(""+e).toLowerCase(),r=!0}}i.byteLength=Ge;function Jn(o,e,n){var a=!1;if((e===void 0||e<0)&&(e=0),e>this.length||((n===void 0||n>this.length)&&(n=this.length),n<=0)||(n>>>=0,e>>>=0,n<=e))return"";for(o||(o="utf8");;)switch(o){case"hex":return _o(this,e,n);case"utf8":case"utf-8":return Ze(this,e,n);case"ascii":return po(this,e,n);case"latin1":case"binary":return io(this,e,n);case"base64":return to(this,e,n);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return mo(this,e,n);default:if(a)throw new TypeError("Unknown encoding: "+o);o=(o+"").toLowerCase(),a=!0}}i.prototype._isBuffer=!0;function F(o,e,n){var a=o[e];o[e]=o[n],o[n]=a}i.prototype.swap16=function(){var e=this.length;if(e%2!==0)throw new RangeError("Buffer size must be a multiple of 16-bits");for(var n=0;n<e;n+=2)F(this,n,n+1);return this};i.prototype.swap32=function(){var e=this.length;if(e%4!==0)throw new RangeError("Buffer size must be a multiple of 32-bits");for(var n=0;n<e;n+=4)F(this,n,n+3),F(this,n+1,n+2);return this};i.prototype.swap64=function(){var e=this.length;if(e%8!==0)throw new RangeError("Buffer size must be a multiple of 64-bits");for(var n=0;n<e;n+=8)F(this,n,n+7),F(this,n+1,n+6),F(this,n+2,n+5),F(this,n+3,n+4);return this};i.prototype.toString=function(){var e=this.length;return e===0?"":arguments.length===0?Ze(this,0,e):Jn.apply(this,arguments)};i.prototype.toLocaleString=i.prototype.toString;i.prototype.equals=function(e){if(!i.isBuffer(e))throw new TypeError("Argument must be a Buffer");return this===e?!0:i.compare(this,e)===0};i.prototype.inspect=function(){var e="",n=P.INSPECT_MAX_BYTES;return e=this.toString("hex",0,n).replace(/(.{2})/g,"$1 ").trim(),this.length>n&&(e+=" ... "),"<Buffer "+e+">"};Ue&&(i.prototype[Ue]=i.prototype.inspect);i.prototype.compare=function(e,n,a,r,t){if(N(e,Uint8Array)&&(e=i.from(e,e.offset,e.byteLength)),!i.isBuffer(e))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+typeof e);if(n===void 0&&(n=0),a===void 0&&(a=e?e.length:0),r===void 0&&(r=0),t===void 0&&(t=this.length),n<0||a>e.length||r<0||t>this.length)throw new RangeError("out of range index");if(r>=t&&n>=a)return 0;if(r>=t)return-1;if(n>=a)return 1;if(n>>>=0,a>>>=0,r>>>=0,t>>>=0,this===e)return 0;for(var p=t-r,s=a-n,l=Math.min(p,s),f=this.slice(r,t),u=e.slice(n,a),g=0;g<l;++g)if(f[g]!==u[g]){p=f[g],s=u[g];break}return p<s?-1:s<p?1:0};function Ve(o,e,n,a,r){if(o.length===0)return-1;if(typeof n=="string"?(a=n,n=0):n>2147483647?n=2147483647:n<-2147483648&&(n=-2147483648),n=+n,he(n)&&(n=r?0:o.length-1),n<0&&(n=o.length+n),n>=o.length){if(r)return-1;n=o.length-1}else if(n<0)if(r)n=0;else return-1;if(typeof e=="string"&&(e=i.from(e,a)),i.isBuffer(e))return e.length===0?-1:Ce(o,e,n,a,r);if(typeof e=="number")return e=e&255,typeof Uint8Array.prototype.indexOf=="function"?r?Uint8Array.prototype.indexOf.call(o,e,n):Uint8Array.prototype.lastIndexOf.call(o,e,n):Ce(o,[e],n,a,r);throw new TypeError("val must be string, number or Buffer")}function Ce(o,e,n,a,r){var t=1,p=o.length,s=e.length;if(a!==void 0&&(a=String(a).toLowerCase(),a==="ucs2"||a==="ucs-2"||a==="utf16le"||a==="utf-16le")){if(o.length<2||e.length<2)return-1;t=2,p/=2,s/=2,n/=2}function l(x,I){return t===1?x[I]:x.readUInt16BE(I*t)}var f;if(r){var u=-1;for(f=n;f<p;f++)if(l(o,f)===l(e,u===-1?0:f-u)){if(u===-1&&(u=f),f-u+1===s)return u*t}else u!==-1&&(f-=f-u),u=-1}else for(n+s>p&&(n=p-s),f=n;f>=0;f--){for(var g=!0,y=0;y<s;y++)if(l(o,f+y)!==l(e,y)){g=!1;break}if(g)return f}return-1}i.prototype.includes=function(e,n,a){return this.indexOf(e,n,a)!==-1};i.prototype.indexOf=function(e,n,a){return Ve(this,e,n,a,!0)};i.prototype.lastIndexOf=function(e,n,a){return Ve(this,e,n,a,!1)};function eo(o,e,n,a){n=Number(n)||0;var r=o.length-n;a?(a=Number(a),a>r&&(a=r)):a=r;var t=e.length;a>t/2&&(a=t/2);for(var p=0;p<a;++p){var s=parseInt(e.substr(p*2,2),16);if(he(s))return p;o[n+p]=s}return p}function no(o,e,n,a){return K(fe(e,o.length-n),o,n,a)}function oo(o,e,n,a){return K(go(e),o,n,a)}function ao(o,e,n,a){return K(Qe(e),o,n,a)}function ro(o,e,n,a){return K(fo(e,o.length-n),o,n,a)}i.prototype.write=function(e,n,a,r){if(n===void 0)r="utf8",a=this.length,n=0;else if(a===void 0&&typeof n=="string")r=n,a=this.length,n=0;else if(isFinite(n))n=n>>>0,isFinite(a)?(a=a>>>0,r===void 0&&(r="utf8")):(r=a,a=void 0);else throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");var t=this.length-n;if((a===void 0||a>t)&&(a=t),e.length>0&&(a<0||n<0)||n>this.length)throw new RangeError("Attempt to write outside buffer bounds");r||(r="utf8");for(var p=!1;;)switch(r){case"hex":return eo(this,e,n,a);case"utf8":case"utf-8":return no(this,e,n,a);case"ascii":case"latin1":case"binary":return oo(this,e,n,a);case"base64":return ao(this,e,n,a);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return ro(this,e,n,a);default:if(p)throw new TypeError("Unknown encoding: "+r);r=(""+r).toLowerCase(),p=!0}};i.prototype.toJSON=function(){return{type:"Buffer",data:Array.prototype.slice.call(this._arr||this,0)}};function to(o,e,n){return e===0&&n===o.length?se.fromByteArray(o):se.fromByteArray(o.slice(e,n))}function Ze(o,e,n){n=Math.min(o.length,n);for(var a=[],r=e;r<n;){var t=o[r],p=null,s=t>239?4:t>223?3:t>191?2:1;if(r+s<=n){var l,f,u,g;switch(s){case 1:t<128&&(p=t);break;case 2:l=o[r+1],(l&192)===128&&(g=(t&31)<<6|l&63,g>127&&(p=g));break;case 3:l=o[r+1],f=o[r+2],(l&192)===128&&(f&192)===128&&(g=(t&15)<<12|(l&63)<<6|f&63,g>2047&&(g<55296||g>57343)&&(p=g));break;case 4:l=o[r+1],f=o[r+2],u=o[r+3],(l&192)===128&&(f&192)===128&&(u&192)===128&&(g=(t&15)<<18|(l&63)<<12|(f&63)<<6|u&63,g>65535&&g<1114112&&(p=g))}}p===null?(p=65533,s=1):p>65535&&(p-=65536,a.push(p>>>10&1023|55296),p=56320|p&1023),a.push(p),r+=s}return co(a)}var Be=4096;function co(o){var e=o.length;if(e<=Be)return String.fromCharCode.apply(String,o);for(var n="",a=0;a<e;)n+=String.fromCharCode.apply(String,o.slice(a,a+=Be));return n}function po(o,e,n){var a="";n=Math.min(o.length,n);for(var r=e;r<n;++r)a+=String.fromCharCode(o[r]&127);return a}function io(o,e,n){var a="";n=Math.min(o.length,n);for(var r=e;r<n;++r)a+=String.fromCharCode(o[r]);return a}function _o(o,e,n){var a=o.length;(!e||e<0)&&(e=0),(!n||n<0||n>a)&&(n=a);for(var r="",t=e;t<n;++t)r+=uo[o[t]];return r}function mo(o,e,n){for(var a=o.slice(e,n),r="",t=0;t<a.length-1;t+=2)r+=String.fromCharCode(a[t]+a[t+1]*256);return r}i.prototype.slice=function(e,n){var a=this.length;e=~~e,n=n===void 0?a:~~n,e<0?(e+=a,e<0&&(e=0)):e>a&&(e=a),n<0?(n+=a,n<0&&(n=0)):n>a&&(n=a),n<e&&(n=e);var r=this.subarray(e,n);return Object.setPrototypeOf(r,i.prototype),r};function h(o,e,n){if(o%1!==0||o<0)throw new RangeError("offset is not uint");if(o+e>n)throw new RangeError("Trying to access beyond buffer length")}i.prototype.readUintLE=i.prototype.readUIntLE=function(e,n,a){e=e>>>0,n=n>>>0,a||h(e,n,this.length);for(var r=this[e],t=1,p=0;++p<n&&(t*=256);)r+=this[e+p]*t;return r};i.prototype.readUintBE=i.prototype.readUIntBE=function(e,n,a){e=e>>>0,n=n>>>0,a||h(e,n,this.length);for(var r=this[e+--n],t=1;n>0&&(t*=256);)r+=this[e+--n]*t;return r};i.prototype.readUint8=i.prototype.readUInt8=function(e,n){return e=e>>>0,n||h(e,1,this.length),this[e]};i.prototype.readUint16LE=i.prototype.readUInt16LE=function(e,n){return e=e>>>0,n||h(e,2,this.length),this[e]|this[e+1]<<8};i.prototype.readUint16BE=i.prototype.readUInt16BE=function(e,n){return e=e>>>0,n||h(e,2,this.length),this[e]<<8|this[e+1]};i.prototype.readUint32LE=i.prototype.readUInt32LE=function(e,n){return e=e>>>0,n||h(e,4,this.length),(this[e]|this[e+1]<<8|this[e+2]<<16)+this[e+3]*16777216};i.prototype.readUint32BE=i.prototype.readUInt32BE=function(e,n){return e=e>>>0,n||h(e,4,this.length),this[e]*16777216+(this[e+1]<<16|this[e+2]<<8|this[e+3])};i.prototype.readIntLE=function(e,n,a){e=e>>>0,n=n>>>0,a||h(e,n,this.length);for(var r=this[e],t=1,p=0;++p<n&&(t*=256);)r+=this[e+p]*t;return t*=128,r>=t&&(r-=Math.pow(2,8*n)),r};i.prototype.readIntBE=function(e,n,a){e=e>>>0,n=n>>>0,a||h(e,n,this.length);for(var r=n,t=1,p=this[e+--r];r>0&&(t*=256);)p+=this[e+--r]*t;return t*=128,p>=t&&(p-=Math.pow(2,8*n)),p};i.prototype.readInt8=function(e,n){return e=e>>>0,n||h(e,1,this.length),this[e]&128?(255-this[e]+1)*-1:this[e]};i.prototype.readInt16LE=function(e,n){e=e>>>0,n||h(e,2,this.length);var a=this[e]|this[e+1]<<8;return a&32768?a|4294901760:a};i.prototype.readInt16BE=function(e,n){e=e>>>0,n||h(e,2,this.length);var a=this[e+1]|this[e]<<8;return a&32768?a|4294901760:a};i.prototype.readInt32LE=function(e,n){return e=e>>>0,n||h(e,4,this.length),this[e]|this[e+1]<<8|this[e+2]<<16|this[e+3]<<24};i.prototype.readInt32BE=function(e,n){return e=e>>>0,n||h(e,4,this.length),this[e]<<24|this[e+1]<<16|this[e+2]<<8|this[e+3]};i.prototype.readFloatLE=function(e,n){return e=e>>>0,n||h(e,4,this.length),B.read(this,e,!0,23,4)};i.prototype.readFloatBE=function(e,n){return e=e>>>0,n||h(e,4,this.length),B.read(this,e,!1,23,4)};i.prototype.readDoubleLE=function(e,n){return e=e>>>0,n||h(e,8,this.length),B.read(this,e,!0,52,8)};i.prototype.readDoubleBE=function(e,n){return e=e>>>0,n||h(e,8,this.length),B.read(this,e,!1,52,8)};function q(o,e,n,a,r,t){if(!i.isBuffer(o))throw new TypeError('"buffer" argument must be a Buffer instance');if(e>r||e<t)throw new RangeError('"value" argument is out of bounds');if(n+a>o.length)throw new RangeError("Index out of range")}i.prototype.writeUintLE=i.prototype.writeUIntLE=function(e,n,a,r){if(e=+e,n=n>>>0,a=a>>>0,!r){var t=Math.pow(2,8*a)-1;q(this,e,n,a,t,0)}var p=1,s=0;for(this[n]=e&255;++s<a&&(p*=256);)this[n+s]=e/p&255;return n+a};i.prototype.writeUintBE=i.prototype.writeUIntBE=function(e,n,a,r){if(e=+e,n=n>>>0,a=a>>>0,!r){var t=Math.pow(2,8*a)-1;q(this,e,n,a,t,0)}var p=a-1,s=1;for(this[n+p]=e&255;--p>=0&&(s*=256);)this[n+p]=e/s&255;return n+a};i.prototype.writeUint8=i.prototype.writeUInt8=function(e,n,a){return e=+e,n=n>>>0,a||q(this,e,n,1,255,0),this[n]=e&255,n+1};i.prototype.writeUint16LE=i.prototype.writeUInt16LE=function(e,n,a){return e=+e,n=n>>>0,a||q(this,e,n,2,65535,0),this[n]=e&255,this[n+1]=e>>>8,n+2};i.prototype.writeUint16BE=i.prototype.writeUInt16BE=function(e,n,a){return e=+e,n=n>>>0,a||q(this,e,n,2,65535,0),this[n]=e>>>8,this[n+1]=e&255,n+2};i.prototype.writeUint32LE=i.prototype.writeUInt32LE=function(e,n,a){return e=+e,n=n>>>0,a||q(this,e,n,4,4294967295,0),this[n+3]=e>>>24,this[n+2]=e>>>16,this[n+1]=e>>>8,this[n]=e&255,n+4};i.prototype.writeUint32BE=i.prototype.writeUInt32BE=function(e,n,a){return e=+e,n=n>>>0,a||q(this,e,n,4,4294967295,0),this[n]=e>>>24,this[n+1]=e>>>16,this[n+2]=e>>>8,this[n+3]=e&255,n+4};i.prototype.writeIntLE=function(e,n,a,r){if(e=+e,n=n>>>0,!r){var t=Math.pow(2,8*a-1);q(this,e,n,a,t-1,-t)}var p=0,s=1,l=0;for(this[n]=e&255;++p<a&&(s*=256);)e<0&&l===0&&this[n+p-1]!==0&&(l=1),this[n+p]=(e/s>>0)-l&255;return n+a};i.prototype.writeIntBE=function(e,n,a,r){if(e=+e,n=n>>>0,!r){var t=Math.pow(2,8*a-1);q(this,e,n,a,t-1,-t)}var p=a-1,s=1,l=0;for(this[n+p]=e&255;--p>=0&&(s*=256);)e<0&&l===0&&this[n+p+1]!==0&&(l=1),this[n+p]=(e/s>>0)-l&255;return n+a};i.prototype.writeInt8=function(e,n,a){return e=+e,n=n>>>0,a||q(this,e,n,1,127,-128),e<0&&(e=255+e+1),this[n]=e&255,n+1};i.prototype.writeInt16LE=function(e,n,a){return e=+e,n=n>>>0,a||q(this,e,n,2,32767,-32768),this[n]=e&255,this[n+1]=e>>>8,n+2};i.prototype.writeInt16BE=function(e,n,a){return e=+e,n=n>>>0,a||q(this,e,n,2,32767,-32768),this[n]=e>>>8,this[n+1]=e&255,n+2};i.prototype.writeInt32LE=function(e,n,a){return e=+e,n=n>>>0,a||q(this,e,n,4,2147483647,-2147483648),this[n]=e&255,this[n+1]=e>>>8,this[n+2]=e>>>16,this[n+3]=e>>>24,n+4};i.prototype.writeInt32BE=function(e,n,a){return e=+e,n=n>>>0,a||q(this,e,n,4,2147483647,-2147483648),e<0&&(e=4294967295+e+1),this[n]=e>>>24,this[n+1]=e>>>16,this[n+2]=e>>>8,this[n+3]=e&255,n+4};function We(o,e,n,a,r,t){if(n+a>o.length)throw new RangeError("Index out of range");if(n<0)throw new RangeError("Index out of range")}function He(o,e,n,a,r){return e=+e,n=n>>>0,r||We(o,e,n,4,34028234663852886e22,-34028234663852886e22),B.write(o,e,n,a,23,4),n+4}i.prototype.writeFloatLE=function(e,n,a){return He(this,e,n,!0,a)};i.prototype.writeFloatBE=function(e,n,a){return He(this,e,n,!1,a)};function Ye(o,e,n,a,r){return e=+e,n=n>>>0,r||We(o,e,n,8,17976931348623157e292,-17976931348623157e292),B.write(o,e,n,a,52,8),n+8}i.prototype.writeDoubleLE=function(e,n,a){return Ye(this,e,n,!0,a)};i.prototype.writeDoubleBE=function(e,n,a){return Ye(this,e,n,!1,a)};i.prototype.copy=function(e,n,a,r){if(!i.isBuffer(e))throw new TypeError("argument should be a Buffer");if(a||(a=0),!r&&r!==0&&(r=this.length),n>=e.length&&(n=e.length),n||(n=0),r>0&&r<a&&(r=a),r===a||e.length===0||this.length===0)return 0;if(n<0)throw new RangeError("targetStart out of bounds");if(a<0||a>=this.length)throw new RangeError("Index out of range");if(r<0)throw new RangeError("sourceEnd out of bounds");r>this.length&&(r=this.length),e.length-n<r-a&&(r=e.length-n+a);var t=r-a;return this===e&&typeof Uint8Array.prototype.copyWithin=="function"?this.copyWithin(n,a,r):Uint8Array.prototype.set.call(e,this.subarray(a,r),n),t};i.prototype.fill=function(e,n,a,r){if(typeof e=="string"){if(typeof n=="string"?(r=n,n=0,a=this.length):typeof a=="string"&&(r=a,a=this.length),r!==void 0&&typeof r!="string")throw new TypeError("encoding must be a string");if(typeof r=="string"&&!i.isEncoding(r))throw new TypeError("Unknown encoding: "+r);if(e.length===1){var t=e.charCodeAt(0);(r==="utf8"&&t<128||r==="latin1")&&(e=t)}}else typeof e=="number"?e=e&255:typeof e=="boolean"&&(e=Number(e));if(n<0||this.length<n||this.length<a)throw new RangeError("Out of range index");if(a<=n)return this;n=n>>>0,a=a===void 0?this.length:a>>>0,e||(e=0);var p;if(typeof e=="number")for(p=n;p<a;++p)this[p]=e;else{var s=i.isBuffer(e)?e:i.from(e,r),l=s.length;if(l===0)throw new TypeError('The value "'+e+'" is invalid for argument "value"');for(p=0;p<a-n;++p)this[p+n]=s[p%l]}return this};var so=/[^+/0-9A-Za-z-_]/g;function lo(o){if(o=o.split("=")[0],o=o.trim().replace(so,""),o.length<2)return"";for(;o.length%4!==0;)o=o+"=";return o}function fe(o,e){e=e||1/0;for(var n,a=o.length,r=null,t=[],p=0;p<a;++p){if(n=o.charCodeAt(p),n>55295&&n<57344){if(!r){if(n>56319){(e-=3)>-1&&t.push(239,191,189);continue}else if(p+1===a){(e-=3)>-1&&t.push(239,191,189);continue}r=n;continue}if(n<56320){(e-=3)>-1&&t.push(239,191,189),r=n;continue}n=(r-55296<<10|n-56320)+65536}else r&&(e-=3)>-1&&t.push(239,191,189);if(r=null,n<128){if((e-=1)<0)break;t.push(n)}else if(n<2048){if((e-=2)<0)break;t.push(n>>6|192,n&63|128)}else if(n<65536){if((e-=3)<0)break;t.push(n>>12|224,n>>6&63|128,n&63|128)}else if(n<1114112){if((e-=4)<0)break;t.push(n>>18|240,n>>12&63|128,n>>6&63|128,n&63|128)}else throw new Error("Invalid code point")}return t}function go(o){for(var e=[],n=0;n<o.length;++n)e.push(o.charCodeAt(n)&255);return e}function fo(o,e){for(var n,a,r,t=[],p=0;p<o.length&&!((e-=2)<0);++p)n=o.charCodeAt(p),a=n>>8,r=n%256,t.push(r),t.push(a);return t}function Qe(o){return se.toByteArray(lo(o))}function K(o,e,n,a){for(var r=0;r<a&&!(r+n>=e.length||r>=o.length);++r)e[r+n]=o[r];return r}function N(o,e){return o instanceof e||o!=null&&o.constructor!=null&&o.constructor.name!=null&&o.constructor.name===e.name}function he(o){return o!==o}var uo=function(){for(var o="0123456789abcdef",e=new Array(256),n=0;n<16;++n)for(var a=n*16,r=0;r<16;++r)e[a+r]=o[n]+o[r];return e}()});var nn=Q((Ko,en)=>{c();var b=en.exports={},D,A;function xe(){throw new Error("setTimeout has not been defined")}function qe(){throw new Error("clearTimeout has not been defined")}(function(){try{typeof setTimeout=="function"?D=setTimeout:D=xe}catch(o){D=xe}try{typeof clearTimeout=="function"?A=clearTimeout:A=qe}catch(o){A=qe}})();function Xe(o){if(D===setTimeout)return setTimeout(o,0);if((D===xe||!D)&&setTimeout)return D=setTimeout,setTimeout(o,0);try{return D(o,0)}catch(e){try{return D.call(null,o,0)}catch(n){return D.call(this,o,0)}}}function bo(o){if(A===clearTimeout)return clearTimeout(o);if((A===qe||!A)&&clearTimeout)return A=clearTimeout,clearTimeout(o);try{return A(o)}catch(e){try{return A.call(null,o)}catch(n){return A.call(this,o)}}}var S=[],O=!1,U,J=-1;function ho(){!O||!U||(O=!1,U.length?S=U.concat(S):J=-1,S.length&&Ke())}function Ke(){if(!O){var o=Xe(ho);O=!0;for(var e=S.length;e;){for(U=S,S=[];++J<e;)U&&U[J].run();J=-1,e=S.length}U=null,O=!1,bo(o)}}b.nextTick=function(o){var e=new Array(arguments.length-1);if(arguments.length>1)for(var n=1;n<arguments.length;n++)e[n-1]=arguments[n];S.push(new Je(o,e)),S.length===1&&!O&&Xe(Ke)};function Je(o,e){this.fun=o,this.array=e}Je.prototype.run=function(){this.fun.apply(null,this.array)};b.title="browser";b.browser=!0;b.env={};b.argv=[];b.version="";b.versions={};function M(){}b.on=M;b.addListener=M;b.once=M;b.off=M;b.removeListener=M;b.removeAllListeners=M;b.emit=M;b.prependListener=M;b.prependOnceListener=M;b.listeners=function(o){return[]};b.binding=function(o){throw new Error("process.binding is not supported")};b.cwd=function(){return"/"};b.chdir=function(o){throw new Error("process.chdir is not supported")};b.umask=function(){return 0}});var _,m,Jo,c=Un(()=>{_=T($e()),m=T(nn()),Jo=function(o){function e(){var a=this||self;return delete o.prototype.__magic__,a}if(typeof globalThis=="object")return globalThis;if(this)return e();o.defineProperty(o.prototype,"__magic__",{configurable:!0,get:e});var n=__magic__;return n}(Object)});c();var En=T(require("react")),Sn=require("@testing-library/react");c();var Z=T(require("react")),oe=T(require("classnames"));c();c();var ee=o=>{var a,r,t,p,s,l;let e="Transaction";return((a=o.action)==null?void 0:a.name)&&((r=o.action)==null?void 0:r.category)&&(((t=o.action)==null?void 0:t.category)==="esdtNft"&&((p=o.action)==null?void 0:p.name)==="transfer"?e="Transaction":o.action&&(e=o.action.name),(l=(s=o.action)==null?void 0:s.arguments)!=null&&l.functionName&&(e=o.action.arguments.functionName)),e};c();var qo=`.dapp-core-component__main__link-style {
  color: #1392ff;
}

.dapp-core-component__main__link-second-style {
  color: #9dabbd;
}
.dapp-core-component__main__link-second-style:hover {
  color: #1392ff;
}
.dapp-core-component__main__link-second-style:hover svg {
  color: #1392ff;
}

.dapp-core-component__main__wrapper {
  min-height: 100vh;
}

.dapp-core-component__main__opacity-6 {
  opacity: 0.6;
}

.dapp-core-component__main__qr-code-svg-container {
  width: 15rem;
  height: 15rem;
}

.dapp-core-component__main__pair-list {
  max-height: 20rem;
  overflow-y: auto;
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__pair-image {
  width: 4rem;
  overflow: hidden;
  height: auto;
  min-width: 2rem;
}

.dapp-core-component__main__pair-remove {
  padding: 0 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

.dapp-core-component__main__dapp-icon {
  padding: 5px;
  border-radius: 50%;
  background-color: #fafafa;
  width: 4.6428571429rem;
  height: 4.6428571429rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}
.dapp-core-component__main__dapp-icon.dapp-core-component__main__icon-medium {
  width: 80px;
  height: 80px;
}

.dapp-core-component__main__centering {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dapp-core-component__main__border-n {
  border: none !important;
}

.dapp-core-component__main__t-shadow {
  text-shadow: 0px 4px 7px rgba(0, 0, 0, 0.3);
}

.dapp-core-component__main__btn {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.04) !important;
}

.dapp-core-component__main__media {
  display: flex;
  align-items: flex-start;
}

.dapp-core-component__main__media-body {
  flex: 1;
}

.dapp-core-component__main__connect-btns .dapp-core-component__main__btn {
  display: flex;
  height: 48px;
  justify-content: center;
  align-items: center;
  color: #fff !important;
  background-color: #1392ff;
  padding: 15px 20px 15px 16px;
  border-radius: 6px;
  gap: 8px;
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__btn:hover, .dapp-core-component__main__connect-btns .dapp-core-component__main__btn:active {
  background-color: #1392ff;
  background-image: linear-gradient(rgba(0, 0, 0, 0.1) 0 0);
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__info {
  color: #1392ff;
  margin-left: 48px;
}

.dapp-core-component__main__btn {
  transition: none !important;
}

.dapp-core-component__main__trust-badge {
  width: 20px;
  height: 20px;
}

/* Navbar
  -------------------------------------------------- */
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name {
  display: flex;
  align-items: center;
  font-size: 1.7142857143rem;
  font-weight: 400;
  color: #9ba5b4;
  margin-left: 1.6428571429rem;
}
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name:before {
  content: "";
  display: block;
  width: 1px;
  height: 3rem;
  margin-right: 1.7142857143rem;
}

/* Dashboard
  -------------------------------------------------- */
.dapp-core-component__main__action-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  margin: 0 7px;
}
.dapp-core-component__main__action-btn button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #fff;
}
.dapp-core-component__main__action-btn button:hover {
  opacity: 0.9;
}

.dapp-core-component__main__transactions .dapp-core-component__main__transaction-icon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table td {
  white-space: nowrap;
  vertical-align: middle;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table tr:last-of-type td {
  border-bottom: 1px solid #dee2e6;
}

/* Loader, Transaction Success/Fail
  -------------------------------------------------- */
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div {
  position: absolute;
  top: 27px;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.1);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.dapp-core-component__main__page-state .dapp-core-component__main__bg-blue .dapp-core-component__main__lds-ellipsis div {
  background: rgba(255, 255, 255, 0.4);
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(1) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis1 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(2) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(3) {
  left: 26px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(4) {
  left: 45px;
  animation: dapp-core-component__main__lds-ellipsis3 0.6s infinite;
}
@keyframes dapp-core-component__main__lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(19px, 0);
  }
}

.dapp-core-component__main__btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
}
.dapp-core-component__main__btn .dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn .dapp-core-component__main__disabled {
  opacity: 0.65;
  cursor: default;
}
.dapp-core-component__main__btn:hover {
  color: #212529;
  text-decoration: none;
}
.dapp-core-component__main__btn:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn:disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn-primary {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary:hover {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-primary:focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary:hover {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-secondary:focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-warning {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning:hover {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning:focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-dark {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark:hover {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
}
.dapp-core-component__main__btn-dark:focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-link {
  font-weight: 400;
  color: #007bff;
  text-decoration: none;
}
.dapp-core-component__main__btn-link:hover {
  color: #0056b3;
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:focus, .dapp-core-component__main__btn-link.dapp-core-component__main__focus {
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:disabled, .dapp-core-component__main__btn-link.dapp-core-component__main__disabled {
  color: #6c757d;
  pointer-events: none;
}

.dapp-core-component__main__btn-light {
  color: #212529;
  background-color: #fafafa;
  border-color: #fafafa;
}
.dapp-core-component__main__btn-light .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}
.dapp-core-component__main__btn-light .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #f8f9fa;
  border-color: #f8f9fa;
}
.dapp-core-component__main__btn-light:hover {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
}
.dapp-core-component__main__btn-light:focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .dapp-core-component__main__badge {
    transition: none;
  }
}
a.dapp-core-component__main__badge:hover, a.dapp-core-component__main__badge:focus {
  text-decoration: none;
}

.dapp-core-component__main__badge:empty {
  display: none;
}

.dapp-core-component__main__btn .dapp-core-component__main__badge {
  position: relative;
  top: -1px;
}

.dapp-core-component__main__badge-pill {
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.dapp-core-component__main__badge-primary {
  color: #fff;
  background-color: #007bff;
}

a.dapp-core-component__main__badge-primary:hover, a.dapp-core-component__main__badge-primary:focus {
  color: #fff;
  background-color: #0062cc;
}

a.dapp-core-component__main__badge-primary:focus, a.dapp-core-component__main__badge-primary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.5);
}

.dapp-core-component__main__badge-secondary {
  color: #fff;
  background-color: #6c757d;
}

a.dapp-core-component__main__badge-secondary:hover, a.dapp-core-component__main__badge-secondary:focus {
  color: #fff;
  background-color: #545b62;
}

a.dapp-core-component__main__badge-secondary:focus, a.dapp-core-component__main__badge-secondary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108, 117, 125, 0.5);
}

.dapp-core-component__main__badge-success {
  color: #fff;
  background-color: #28a745;
}

a.dapp-core-component__main__badge-success:hover, a.dapp-core-component__main__badge-success:focus {
  color: #fff;
  background-color: #1e7e34;
}

a.dapp-core-component__main__badge-success:focus, a.dapp-core-component__main__badge-success.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
}

.dapp-core-component__main__badge-info {
  color: #fff;
  background-color: #17a2b8;
}

a.dapp-core-component__main__badge-info:hover, a.dapp-core-component__main__badge-info:focus {
  color: #fff;
  background-color: #117a8b;
}

a.dapp-core-component__main__badge-info:focus, a.dapp-core-component__main__badge-info.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(23, 162, 184, 0.5);
}

.dapp-core-component__main__badge-warning {
  color: #212529;
  background-color: #ffc107;
}

a.dapp-core-component__main__badge-warning:hover, a.dapp-core-component__main__badge-warning:focus {
  color: #212529;
  background-color: #d39e00;
}

a.dapp-core-component__main__badge-warning:focus, a.dapp-core-component__main__badge-warning.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 193, 7, 0.5);
}

.dapp-core-component__main__badge-danger {
  color: #fff;
  background-color: #dc3545;
}

a.dapp-core-component__main__badge-danger:hover, a.dapp-core-component__main__badge-danger:focus {
  color: #fff;
  background-color: #bd2130;
}

a.dapp-core-component__main__badge-danger:focus, a.dapp-core-component__main__badge-danger.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(220, 53, 69, 0.5);
}

.dapp-core-component__main__badge-light {
  color: #212529;
  background-color: #f8f9fa;
}

a.dapp-core-component__main__badge-light:hover, a.dapp-core-component__main__badge-light:focus {
  color: #212529;
  background-color: #dae0e5;
}

a.dapp-core-component__main__badge-light:focus, a.dapp-core-component__main__badge-light.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(248, 249, 250, 0.5);
}

.dapp-core-component__main__badge-dark {
  color: #fff;
  background-color: #343a40;
}

a.dapp-core-component__main__badge-dark:hover, a.dapp-core-component__main__badge-dark:focus {
  color: #fff;
  background-color: #1d2124;
}

a.dapp-core-component__main__badge-dark:focus, a.dapp-core-component__main__badge-dark.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}

.dapp-core-component__main__card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 5px;
}
.dapp-core-component__main__card > hr {
  margin-right: 0;
  margin-left: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group {
  border-top: inherit;
  border-bottom: inherit;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:first-child {
  border-top-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:last-child {
  border-bottom-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__card-header + .dapp-core-component__main__list-group,
.dapp-core-component__main__card > .dapp-core-component__main__list-group + .dapp-core-component__main__card-footer {
  border-top: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header {
  border: 0;
  padding: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header:first-child {
  border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item {
  border-bottom: 1px solid #dee2e6;
  padding: 1.1rem 1rem;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item h6,
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item .dapp-core-component__main__h6 {
  line-height: 2.5rem;
  letter-spacing: unset;
  margin: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item.dapp-core-component__main__compact {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fafafa;
}

.dapp-core-component__main__card-body {
  flex: 1 1 auto;
  min-height: 1px;
  padding: 1.25rem;
}

.dapp-core-component__main__card-title {
  margin-bottom: 0.75rem;
}

.dapp-core-component__main__card-subtitle {
  margin-top: -0.375rem;
  margin-bottom: 0;
}

.dapp-core-component__main__card-text:last-child {
  margin-bottom: 0;
}

.dapp-core-component__main__card-link:hover {
  text-decoration: none;
}
.dapp-core-component__main__card-link + .dapp-core-component__main__card-link {
  margin-left: 1.25rem;
}

.dapp-core-component__main__card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: #fff;
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__card-footer {
  padding: 0.75rem 1.25rem;
  background-color: #fff;
  border-top: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__text {
  color: #9dabbd;
}

.dapp-core-component__main__text-white {
  color: #fff;
}

.dapp-core-component__main__text-monospace {
  font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-size-xl .dapp-core-component__main__trim {
  max-width: 13rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim {
  max-width: 10rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-only-sm .dapp-core-component__main__trim {
  max-width: none;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-left {
  text-align: left;
}

.dapp-core-component__main__text-right {
  text-align: right;
}

.dapp-core-component__main__text-center {
  text-align: center;
}

.dapp-core-component__main__text-primary {
  color: #1b46c2;
}

.dapp-core-component__main__text-secondary {
  color: #6c757d;
}

.dapp-core-component__main__text-primary-highlight {
  color: #1b46c2;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-success {
  color: #28a745;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-body {
  color: #212529;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-black-50 {
  color: rgba(0, 0, 0, 0.5);
}

.dapp-core-component__main__text-white-50 {
  color: rgba(255, 255, 255, 0.5);
}

@media (min-width: 992px) {
  .dapp-core-component__main__text-lg-right {
    text-align: right;
  }
}

.dapp-core-component__main__text-hide {
  font: 0/0 a;
  color: rgba(0, 0, 0, 0);
  text-shadow: none;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
}

.dapp-core-component__main__text-decoration-none {
  text-decoration: none;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__text-reset {
  color: inherit;
}

.dapp-core-component__main__text-lowercase {
  text-transform: lowercase;
}

.dapp-core-component__main__text-uppercase {
  text-transform: uppercase;
}

.dapp-core-component__main__text-capitalize {
  text-transform: capitalize;
}

.dapp-core-component__main__text-danger {
  color: #d9534f;
}

.dapp-core-component__main__text-warning {
  color: #f0ad4e;
}

.dapp-core-component__main__trim-fs-sm {
  color: inherit;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__m-0 {
  margin: 0;
}

.dapp-core-component__main__mt-0 {
  margin-top: 0;
}

.dapp-core-component__main__my-0 {
  margin-top: 0;
  margin-bottom: 0;
}

.dapp-core-component__main__mr-0 {
  margin-right: 0;
}

.dapp-core-component__main__mx-0 {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__mb-0 {
  margin-bottom: 0;
}

.dapp-core-component__main__my-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.dapp-core-component__main__ml-0 {
  margin-left: 0;
}

.dapp-core-component__main__mx-0 {
  margin-left: 0;
  margin-right: 0;
}

.dapp-core-component__main__m-1 {
  margin: 0.25rem;
}

.dapp-core-component__main__mt-1 {
  margin-top: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__mr-1 {
  margin-right: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.dapp-core-component__main__mb-1 {
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-bottom: 0.25rem;
  margin-top: 0.25rem;
}

.dapp-core-component__main__ml-1 {
  margin-left: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-right: 0.25rem;
  margin-left: 0.25rem;
}

.dapp-core-component__main__m-2 {
  margin: 0.5rem;
}

.dapp-core-component__main__mt-2 {
  margin-top: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-bottom: 0.5rem;
  margin-top: 0.5rem;
}

.dapp-core-component__main__mr-2 {
  margin-right: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__mb-2 {
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__ml-2 {
  margin-left: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__m-3 {
  margin: 1rem;
}

.dapp-core-component__main__mt-3,
.dapp-core-component__main__my-3 {
  margin-top: 1rem;
}

.dapp-core-component__main__mr-3 {
  margin-right: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__mb-3 {
  margin-bottom: 1rem;
}

.dapp-core-component__main__my-3 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.dapp-core-component__main__ml-3 {
  margin-left: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__m-4 {
  margin: 1.5rem;
}

.dapp-core-component__main__mt-4 {
  margin-top: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__mr-4 {
  margin-right: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-right: 1.5rem;
  margin-left: 1.5rem;
}

.dapp-core-component__main__mb-4 {
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.dapp-core-component__main__ml-4 {
  margin-left: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}

.dapp-core-component__main__m-5 {
  margin: 3rem;
}

.dapp-core-component__main__mt-5 {
  margin-top: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__mr-5 {
  margin-right: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-right: 3rem;
  margin-left: 3rem;
}

.dapp-core-component__main__mb-5 {
  margin-bottom: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__ml-5 {
  margin-left: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-left: 3rem;
  margin-right: 3rem;
}

.dapp-core-component__main__p-0 {
  padding: 0;
}

.dapp-core-component__main__pt-0 {
  padding-top: 0;
}

.dapp-core-component__main__py-0 {
  padding-top: 0;
  padding-bottom: 0;
}

.dapp-core-component__main__pr-0 {
  padding-right: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__pb-0 {
  padding-bottom: 0;
}

.dapp-core-component__main__py-0 {
  padding-bottom: 0;
  padding-top: 0;
}

.dapp-core-component__main__pl-0 {
  padding-left: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__p-1 {
  padding: 0.25rem;
}

.dapp-core-component__main__pt-1 {
  padding-top: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pr-1 {
  padding-right: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-right: 0.25rem;
  padding-left: 0.25rem;
}

.dapp-core-component__main__pb-1 {
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pl-1 {
  padding-left: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

.dapp-core-component__main__p-2 {
  padding: 0.5rem;
}

.dapp-core-component__main__pt-2 {
  padding-top: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.dapp-core-component__main__pr-2 {
  padding-right: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__pb-2 {
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__pl-2 {
  padding-left: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__p-3 {
  padding: 1rem;
}

.dapp-core-component__main__pt-3 {
  padding-top: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.dapp-core-component__main__pr-3 {
  padding-right: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__pb-3 {
  padding-bottom: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.dapp-core-component__main__pl-3 {
  padding-left: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__p-4 {
  padding: 1.5rem;
}

.dapp-core-component__main__pt-4 {
  padding-top: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pr-4 {
  padding-right: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.dapp-core-component__main__pb-4 {
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pl-4 {
  padding-left: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
}

.dapp-core-component__main__p-5 {
  padding: 3rem;
}

.dapp-core-component__main__pt-5 {
  padding-top: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-bottom: 3rem;
  padding-top: 3rem;
}

.dapp-core-component__main__pr-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__pb-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.dapp-core-component__main__pl-5 {
  padding-left: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__m-n1 {
  margin: -0.25rem;
}

.dapp-core-component__main__mt-n1 {
  margin-top: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__mr-n1 {
  margin-right: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__mb-n1 {
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-bottom: -0.25rem;
  margin-top: -0.25rem;
}

.dapp-core-component__main__ml-n1 {
  margin-left: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__m-n2 {
  margin: -0.5rem;
}

.dapp-core-component__main__mt-n2 {
  margin-top: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-top: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mr-n2 {
  margin-right: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mb-n2 {
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-bottom: -0.5rem;
  margin-top: -0.5rem;
}

.dapp-core-component__main__ml-n2 {
  margin-left: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.dapp-core-component__main__m-n3 {
  margin: -1rem;
}

.dapp-core-component__main__mt-n3 {
  margin-top: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-top: -1rem;
  margin-bottom: -1rem;
}

.dapp-core-component__main__mr-n3 {
  margin-right: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-right: -1rem;
  margin-left: -1rem;
}

.dapp-core-component__main__mb-n3 {
  margin-bottom: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-bottom: -1rem;
  margin-top: -1rem;
}

.dapp-core-component__main__ml-n3 {
  margin-left: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-left: -1rem;
  margin-right: -1rem;
}

.dapp-core-component__main__m-n4 {
  margin: -1.5rem;
}

.dapp-core-component__main__mt-n4 {
  margin-top: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-top: -1.5rem;
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__mr-n4 {
  margin-right: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__mb-n4 {
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-bottom: -1.5rem;
  margin-top: -1.5rem;
}

.dapp-core-component__main__ml-n4 {
  margin-left: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__m-n5 {
  margin: -3rem;
}

.dapp-core-component__main__mt-n5 {
  margin-top: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-bottom: -3rem;
  margin-top: -3rem;
}

.dapp-core-component__main__mr-n5 {
  margin-right: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-left: -3rem;
  margin-right: -3rem;
}

.dapp-core-component__main__mb-n5 {
  margin-bottom: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-top: -3rem;
  margin-bottom: -3rem;
}

.dapp-core-component__main__ml-n5 {
  margin-left: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-right: -3rem;
  margin-left: -3rem;
}

.dapp-core-component__main__m-auto {
  margin: auto;
}

.dapp-core-component__main__mt-auto {
  margin-top: auto;
}

.dapp-core-component__main__my-auto {
  margin-bottom: auto;
  margin-top: auto;
}

.dapp-core-component__main__mr-auto {
  margin-right: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__mb-auto {
  margin-bottom: auto;
}

.dapp-core-component__main__my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.dapp-core-component__main__ml-auto {
  margin-left: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-1 {
    margin-right: 0.25rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-4 {
    margin-right: 1.5rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mx-lg-4 {
    margin-left: 1.5rem;
    margin-right: 1.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__mb-xl-2 {
    margin-bottom: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__pr-sm-2 {
    padding-right: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__px-sm-2 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__pr-xl-0,
.dapp-core-component__main__px-xl-0 {
    padding-right: 0;
  }
}

.dapp-core-component__main__col {
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

.dapp-core-component__main__col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__col-xl,
.dapp-core-component__main__col-xl-auto,
.dapp-core-component__main__col-xl-12,
.dapp-core-component__main__col-xl-11,
.dapp-core-component__main__col-xl-10,
.dapp-core-component__main__col-xl-9,
.dapp-core-component__main__col-xl-8,
.dapp-core-component__main__col-xl-7,
.dapp-core-component__main__col-xl-6,
.dapp-core-component__main__col-xl-5,
.dapp-core-component__main__col-xl-4,
.dapp-core-component__main__col-xl-3,
.dapp-core-component__main__col-xl-2,
.dapp-core-component__main__col-xl-1,
.dapp-core-component__main__col-lg,
.dapp-core-component__main__col-lg-auto,
.dapp-core-component__main__col-lg-12,
.dapp-core-component__main__col-lg-11,
.dapp-core-component__main__col-lg-10,
.dapp-core-component__main__col-lg-9,
.dapp-core-component__main__col-lg-8,
.dapp-core-component__main__col-lg-7,
.dapp-core-component__main__col-lg-6,
.dapp-core-component__main__col-lg-5,
.dapp-core-component__main__col-lg-4,
.dapp-core-component__main__col-lg-3,
.dapp-core-component__main__col-lg-2,
.dapp-core-component__main__col-lg-1,
.dapp-core-component__main__col-md,
.dapp-core-component__main__col-md-auto,
.dapp-core-component__main__col-md-12,
.dapp-core-component__main__col-md-11,
.dapp-core-component__main__col-md-10,
.dapp-core-component__main__col-md-9,
.dapp-core-component__main__col-md-8,
.dapp-core-component__main__col-md-7,
.dapp-core-component__main__col-md-6,
.dapp-core-component__main__col-md-5,
.dapp-core-component__main__col-md-4,
.dapp-core-component__main__col-md-3,
.dapp-core-component__main__col-md-2,
.dapp-core-component__main__col-md-1,
.dapp-core-component__main__col-sm,
.dapp-core-component__main__col-sm-auto,
.dapp-core-component__main__col-sm-12,
.dapp-core-component__main__col-sm-11,
.dapp-core-component__main__col-sm-10,
.dapp-core-component__main__col-sm-9,
.dapp-core-component__main__col-sm-8,
.dapp-core-component__main__col-sm-7,
.dapp-core-component__main__col-sm-6,
.dapp-core-component__main__col-sm-5,
.dapp-core-component__main__col-sm-4,
.dapp-core-component__main__col-sm-3,
.dapp-core-component__main__col-sm-2,
.dapp-core-component__main__col-sm-1,
.dapp-core-component__main__col,
.dapp-core-component__main__col-auto,
.dapp-core-component__main__col-12,
.dapp-core-component__main__col-11,
.dapp-core-component__main__col-10,
.dapp-core-component__main__col-9,
.dapp-core-component__main__col-8,
.dapp-core-component__main__col-7,
.dapp-core-component__main__col-6,
.dapp-core-component__main__col-5,
.dapp-core-component__main__col-4,
.dapp-core-component__main__col-3,
.dapp-core-component__main__col-2,
.dapp-core-component__main__col-1 {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
}

.dapp-core-component__main__col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

@media (min-width: 576px) {
  .dapp-core-component__main__col-sm-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-sm-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-sm-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-sm-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-sm-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-sm-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-sm-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-sm-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-sm-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-sm-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-sm-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-sm-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 768px) {
  .dapp-core-component__main__col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__col-xl-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-xl-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-xl-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-xl-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-xl-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-xl-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-xl-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-xl-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-xl-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-xl-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-xl-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-xl-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
.dapp-core-component__main__flex-row {
  flex-direction: row;
}

.dapp-core-component__main__flex-column {
  flex-direction: column;
}

.dapp-core-component__main__flex-row-reverse {
  flex-direction: row-reverse;
}

.dapp-core-component__main__flex-column-reverse {
  flex-direction: column-reverse;
}

.dapp-core-component__main__flex-wrap {
  flex-wrap: wrap;
}

.dapp-core-component__main__flex-nowrap {
  flex-wrap: nowrap;
}

.dapp-core-component__main__flex-wrap-reverse {
  flex-wrap: wrap-reverse;
}

.dapp-core-component__main__flex-fill {
  flex: 1 1 auto;
}

.dapp-core-component__main__flex-grow-0 {
  flex-grow: 0;
}

.dapp-core-component__main__flex-grow-1 {
  flex-grow: 1;
}

.dapp-core-component__main__flex-shrink-0 {
  flex-shrink: 0;
}

.dapp-core-component__main__flex-shrink-1 {
  flex-shrink: 1;
}

.dapp-core-component__main__justify-content-start {
  justify-content: flex-start;
}

.dapp-core-component__main__justify-content-end {
  justify-content: flex-end;
}

.dapp-core-component__main__justify-content-center {
  justify-content: center;
}

.dapp-core-component__main__justify-content-between {
  justify-content: space-between;
}

.dapp-core-component__main__justify-content-around {
  justify-content: space-around;
}

.dapp-core-component__main__align-items-start {
  align-items: flex-start;
}

.dapp-core-component__main__align-items-end {
  align-items: flex-end;
}

.dapp-core-component__main__align-items-center {
  align-items: center;
}

.dapp-core-component__main__align-items-baseline {
  align-items: baseline;
}

.dapp-core-component__main__align-items-stretch {
  align-items: stretch;
}

.dapp-core-component__main__align-content-start {
  align-content: flex-start;
}

.dapp-core-component__main__align-content-end {
  align-content: flex-end;
}

.dapp-core-component__main__align-content-center {
  align-content: center;
}

.dapp-core-component__main__align-content-between {
  align-content: space-between;
}

.dapp-core-component__main__align-content-around {
  align-content: space-around;
}

.dapp-core-component__main__align-content-stretch {
  align-content: stretch;
}

.dapp-core-component__main__align-self-auto {
  align-self: auto;
}

.dapp-core-component__main__align-self-start {
  align-self: flex-start;
}

.dapp-core-component__main__align-self-end {
  align-self: flex-end;
}

.dapp-core-component__main__align-self-center {
  align-self: center;
}

.dapp-core-component__main__align-self-baseline {
  align-self: baseline;
}

.dapp-core-component__main__align-self-stretch {
  align-self: stretch;
}

@media (min-width: 576px) {
  .dapp-core-component__main__flex-sm-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-wrap {
    flex-wrap: wrap;
  }
}

.dapp-core-component__main__form-check {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: auto;
  padding-left: 0;
}

.dapp-core-component__main__form-check-label {
  margin-bottom: 0;
}

.dapp-core-component__main__form-check-input {
  position: absolute;
  margin-top: 0.3rem;
  margin-left: -1.25rem;
}
.dapp-core-component__main__form-check-input[disabled] ~ .dapp-core-component__main__form-check-label, .dapp-core-component__main__form-check-input:disabled ~ .dapp-core-component__main__form-check-label {
  color: #6c757d;
}

.dapp-core-component__main__form-group {
  display: flex;
  flex: 0 0 auto;
  flex-flow: row wrap;
  align-items: center;
  margin-bottom: 0;
}

.dapp-core-component__main__form-control {
  display: block;
  width: 100%;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.dapp-core-component__main__form-label {
  padding-top: calc(0.375rem + 1px);
  padding-bottom: calc(0.375rem + 1px);
  margin-bottom: 0;
  font-size: inherit;
  line-height: 1.5;
}

input,
button,
select,
optgroup,
textarea {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.dapp-core-component__main__form-group {
  margin-bottom: 1rem;
}

.dapp-core-component__main__form-text {
  display: block;
  margin-top: 0.25rem;
  color: inherit;
}

.dapp-core-component__main__border {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__border-top {
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__border-right {
  border-right: 1px solid #dee2e6;
}

.dapp-core-component__main__border-bottom {
  border-bottom: 1px solid #dee2e6;
}

.dapp-core-component__main__border-left {
  border-left: 1px solid #dee2e6;
}

.dapp-core-component__main__border-0 {
  border: 0;
}

.dapp-core-component__main__border-top-0 {
  border-top: 0;
}

.dapp-core-component__main__border-right-0 {
  border-right: 0;
}

.dapp-core-component__main__border-bottom-0 {
  border-bottom: 0;
}

.dapp-core-component__main__border-left-0 {
  border-left: 0;
}

.dapp-core-component__main__rounded-sm {
  border-radius: 0.2rem;
}

.dapp-core-component__main__rounded {
  border-radius: 0.25rem;
}

.dapp-core-component__main__rounded-top {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-right {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-bottom {
  border-bottom-right-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-left {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-lg {
  border-radius: 0.3rem;
}

.dapp-core-component__main__rounded-circle {
  border-radius: 50%;
}

.dapp-core-component__main__rounded-pill {
  border-radius: 50rem;
}

.dapp-core-component__main__rounded-0 {
  border-radius: 0;
}

.dapp-core-component__main__h1,
.dapp-core-component__main__h2,
.dapp-core-component__main__h3,
.dapp-core-component__main__h4,
.dapp-core-component__main__h5,
.dapp-core-component__main__h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2;
}

.dapp-core-component__main__h1 {
  font-size: 2.5rem;
}

.dapp-core-component__main__h2 {
  font-size: 2rem;
}

.dapp-core-component__main__h3 {
  font-size: 1.75rem;
}

.dapp-core-component__main__h4 {
  font-size: 1.5rem;
}

.dapp-core-component__main__h5 {
  font-size: 1.25rem;
}

.dapp-core-component__main__h6 {
  font-size: 1rem;
}

.dapp-core-component__main__lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

.dapp-core-component__main__bg-warning,
.dapp-core-component__main__warning {
  background-color: #f0ad4e;
}

.dapp-core-component__main__bg-danger,
.dapp-core-component__main__danger {
  background-color: #d9534f;
}

.dapp-core-component__main__bg-success,
.dapp-core-component__main__success {
  background-color: #5cb85c;
}

.dapp-core-component__main__position-relative {
  position: relative;
}

.dapp-core-component__main__position-absolute {
  position: absolute;
}

.dapp-core-component__main__w-25 {
  width: 25%;
}

.dapp-core-component__main__w-50 {
  width: 50%;
}

.dapp-core-component__main__w-75 {
  width: 75%;
}

.dapp-core-component__main__w-100 {
  width: 100%;
}

.dapp-core-component__main__w-auto {
  width: auto;
}

.dapp-core-component__main__h-25 {
  height: 25%;
}

.dapp-core-component__main__h-50 {
  height: 50%;
}

.dapp-core-component__main__h-75 {
  height: 75%;
}

.dapp-core-component__main__h-100 {
  height: 100%;
}

.dapp-core-component__main__h-auto {
  height: auto;
}

.dapp-core-component__main__mw-100 {
  max-width: 100%;
}

.dapp-core-component__main__mh-100 {
  max-height: 100%;
}

.dapp-core-component__main__min-vw-100 {
  min-width: 100vw;
}

.dapp-core-component__main__min-vh-100 {
  min-height: 100vh;
}

.dapp-core-component__main__vw-100 {
  width: 100vw;
}

.dapp-core-component__main__vh-100 {
  height: 100vh;
}

.dapp-core-component__main__locked-icon {
  color: inherit;
}

.dapp-core-component__main__shadow-sm {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
}

.dapp-core-component__main__shadow {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.dapp-core-component__main__shadow-lg {
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.175);
}

.dapp-core-component__main__shadow-none {
  box-shadow: none;
}

.dapp-core-component__main__d-none {
  display: none;
}

.dapp-core-component__main__d-inline {
  display: inline;
}

.dapp-core-component__main__d-inline-block {
  display: inline-block;
}

.dapp-core-component__main__d-block {
  display: block;
}

.dapp-core-component__main__d-table {
  display: table;
}

.dapp-core-component__main__d-table-row {
  display: table-row;
}

.dapp-core-component__main__d-table-cell {
  display: table-cell;
}

.dapp-core-component__main__d-flex {
  display: -ms-flexbox;
  display: flex;
}

.dapp-core-component__main__d-inline-flex {
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media (min-width: 576px) {
  .dapp-core-component__main__d-sm-flex {
    display: flex;
  }
}

.dapp-core-component__main__table {
  width: 100%;
  margin-bottom: 1rem;
  color: #212529;
}
.dapp-core-component__main__table th,
.dapp-core-component__main__table td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #dee2e6;
}
.dapp-core-component__main__table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #dee2e6;
}
.dapp-core-component__main__table tbody + tbody {
  border-top: 2px solid #dee2e6;
}

.dapp-core-component__main__table-sm th,
.dapp-core-component__main__table-sm td {
  padding: 0.3rem;
}

.dapp-core-component__main__table-bordered {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered th,
.dapp-core-component__main__table-bordered td {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered thead th,
.dapp-core-component__main__table-bordered thead td {
  border-bottom-width: 2px;
}

.dapp-core-component__main__table-borderless th,
.dapp-core-component__main__table-borderless td,
.dapp-core-component__main__table-borderless thead th,
.dapp-core-component__main__table-borderless tbody + tbody {
  border: 0;
}

.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(0, 0, 0, 0.05);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-dark th {
  color: #fff;
  background-color: #343a40;
  border-color: #454d55;
}
.dapp-core-component__main__table .dapp-core-component__main__thead-light th {
  color: #495057;
  background-color: #e9ecef;
  border-color: #dee2e6;
}

.dapp-core-component__main__table-dark {
  color: #fff;
  background-color: #343a40;
}
.dapp-core-component__main__table-dark th,
.dapp-core-component__main__table-dark td,
.dapp-core-component__main__table-dark thead th {
  border-color: #454d55;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-bordered {
  border: 0;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(255, 255, 255, 0.05);
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-hover tbody tr {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.075);
}

.dapp-core-component__main__table-responsive {
  display: block;
  width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}
.dapp-core-component__main__table-responsive > .dapp-core-component__main__table-bordered {
  border: 0;
}

@media (max-width: 767.98px) {
  .dapp-core-component__main__transactions.dapp-core-component__main__table .dapp-core-component__main__transaction-hash .dapp-core-component__main__trim {
    max-width: 8rem;
  }
}
.dapp-core-component__main__font-weight-light {
  font-weight: 300 !important;
}

.dapp-core-component__main__font-weight-lighter {
  font-weight: lighter !important;
}

.dapp-core-component__main__font-weight-normal {
  font-weight: 400 !important;
}

.dapp-core-component__main__font-weight-medium {
  font-weight: 500 !important;
}

.dapp-core-component__main__font-weight-bold {
  font-weight: 700 !important;
}

.dapp-core-component__main__font-weight-bolder {
  font-weight: bolder !important;
}

.dapp-core-component__main__font-italic {
  font-style: italic !important;
}

.dapp-core-component__main__icon-state {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
}
.dapp-core-component__main__icon-state.dapp-core-component__main__half {
  width: 3rem;
  height: 3rem;
}

.dapp-core-component__main__side-icon {
  width: 1.5rem;
  height: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  color: transparent;
  border-radius: 50%;
  overflow: hidden;
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__side-icon svg {
  color: #cacbcc;
}

div.dapp-core-component__main__side-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
div.dapp-core-component__main__side-icon img {
  transform: scale(0.75);
}

.dapp-core-component__main__toast-messages {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast {
  display: block;
  flex-basis: 4rem;
  width: 100%;
  max-width: 28.875rem;
  transition: transform 0.3s ease-in-out;
  transform: translateX(120%);
  background-color: #fff;
  border-radius: 0.25rem;
  border-color: #dee2e6;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__clickable {
  cursor: pointer;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible {
  margin-bottom: 0.3rem;
  transform: translateX(0);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress {
  background-color: rgba(0, 0, 0, 0);
  height: inherit;
  font-size: inherit;
  line-height: inherit;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar {
  background-color: rgba(108, 117, 125, 0.2);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar .dapp-core-component__main__content-height {
  z-index: -1;
  visibility: hidden;
  width: 0 !important;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast .dapp-core-component__main__tx-description {
  margin-bottom: 0.2rem;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close {
  opacity: 1;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close svg {
  color: #6c757d;
}

.dapp-core-component__main__data-field .dapp-core-component__main__textarea {
  -moz-appearance: textfield-multiline;
  -webkit-appearance: textarea;
  border: 1px solid #000;
  background-color: #fff;
  border-radius: 5px;
  font: medium -moz-fixed;
  font-size: 0.8125rem;
  min-height: 5rem;
  overflow: auto;
  padding: 0.375rem 0.75rem;
  resize: both;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea:hover {
  border: 1px solid #000;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea .dapp-core-component__main__highlighted {
  font-size: 1rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea.dapp-core-component__main__sc-call {
  min-height: 3rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__form-control:disabled,
.dapp-core-component__main__data-field .dapp-core-component__main__form-control[readonly] {
  background-color: #fafafa;
}
.dapp-core-component__main__data-field .dapp-core-component__main__small-font {
  font-size: 0.8125rem;
}

.dapp-core-component__main__progress-steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps hr {
  background-color: #6c757d;
  border-color: #dee2e6;
  z-index: 2;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot {
  height: 0.625rem;
  width: 0.625rem;
  border-radius: 50%;
  margin: 0 1rem;
  z-index: 5;
  background-color: #6c757d;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:first-of-type {
  margin-left: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:last-of-type {
  margin-right: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

a.dapp-core-component__main__side-link::after {
  right: 0;
  left: unset;
  width: calc(100% - 1.85rem);
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(qo));var w={"link-style":"dapp-core-component__main__link-style",linkStyle:"dapp-core-component__main__link-style","link-second-style":"dapp-core-component__main__link-second-style",linkSecondStyle:"dapp-core-component__main__link-second-style",wrapper:"dapp-core-component__main__wrapper","opacity-6":"dapp-core-component__main__opacity-6",opacity6:"dapp-core-component__main__opacity-6","qr-code-svg-container":"dapp-core-component__main__qr-code-svg-container",qrCodeSvgContainer:"dapp-core-component__main__qr-code-svg-container","pair-list":"dapp-core-component__main__pair-list",pairList:"dapp-core-component__main__pair-list","pair-image":"dapp-core-component__main__pair-image",pairImage:"dapp-core-component__main__pair-image","pair-remove":"dapp-core-component__main__pair-remove",pairRemove:"dapp-core-component__main__pair-remove","dapp-icon":"dapp-core-component__main__dapp-icon",dappIcon:"dapp-core-component__main__dapp-icon","icon-medium":"dapp-core-component__main__icon-medium",iconMedium:"dapp-core-component__main__icon-medium",centering:"dapp-core-component__main__centering","border-n":"dapp-core-component__main__border-n",borderN:"dapp-core-component__main__border-n","t-shadow":"dapp-core-component__main__t-shadow",tShadow:"dapp-core-component__main__t-shadow",btn:"dapp-core-component__main__btn",media:"dapp-core-component__main__media","media-body":"dapp-core-component__main__media-body",mediaBody:"dapp-core-component__main__media-body","connect-btns":"dapp-core-component__main__connect-btns",connectBtns:"dapp-core-component__main__connect-btns",info:"dapp-core-component__main__info","trust-badge":"dapp-core-component__main__trust-badge",trustBadge:"dapp-core-component__main__trust-badge",navbar:"dapp-core-component__main__navbar","dapp-name":"dapp-core-component__main__dapp-name",dappName:"dapp-core-component__main__dapp-name","action-btn":"dapp-core-component__main__action-btn",actionBtn:"dapp-core-component__main__action-btn",transactions:"dapp-core-component__main__transactions","transaction-icon":"dapp-core-component__main__transaction-icon",transactionIcon:"dapp-core-component__main__transaction-icon",table:"dapp-core-component__main__table","page-state":"dapp-core-component__main__page-state",pageState:"dapp-core-component__main__page-state","lds-ellipsis":"dapp-core-component__main__lds-ellipsis",ldsEllipsis:"dapp-core-component__main__lds-ellipsis","bg-blue":"dapp-core-component__main__bg-blue",bgBlue:"dapp-core-component__main__bg-blue","lds-ellipsis1":"dapp-core-component__main__lds-ellipsis1",ldsEllipsis1:"dapp-core-component__main__lds-ellipsis1","lds-ellipsis2":"dapp-core-component__main__lds-ellipsis2",ldsEllipsis2:"dapp-core-component__main__lds-ellipsis2","lds-ellipsis3":"dapp-core-component__main__lds-ellipsis3",ldsEllipsis3:"dapp-core-component__main__lds-ellipsis3",focus:"dapp-core-component__main__focus",disabled:"dapp-core-component__main__disabled","btn-primary":"dapp-core-component__main__btn-primary",btnPrimary:"dapp-core-component__main__btn-primary","btn-secondary":"dapp-core-component__main__btn-secondary",btnSecondary:"dapp-core-component__main__btn-secondary","btn-warning":"dapp-core-component__main__btn-warning",btnWarning:"dapp-core-component__main__btn-warning","btn-dark":"dapp-core-component__main__btn-dark",btnDark:"dapp-core-component__main__btn-dark","btn-link":"dapp-core-component__main__btn-link",btnLink:"dapp-core-component__main__btn-link","btn-light":"dapp-core-component__main__btn-light",btnLight:"dapp-core-component__main__btn-light",badge:"dapp-core-component__main__badge","badge-pill":"dapp-core-component__main__badge-pill",badgePill:"dapp-core-component__main__badge-pill","badge-primary":"dapp-core-component__main__badge-primary",badgePrimary:"dapp-core-component__main__badge-primary","badge-secondary":"dapp-core-component__main__badge-secondary",badgeSecondary:"dapp-core-component__main__badge-secondary","badge-success":"dapp-core-component__main__badge-success",badgeSuccess:"dapp-core-component__main__badge-success","badge-info":"dapp-core-component__main__badge-info",badgeInfo:"dapp-core-component__main__badge-info","badge-warning":"dapp-core-component__main__badge-warning",badgeWarning:"dapp-core-component__main__badge-warning","badge-danger":"dapp-core-component__main__badge-danger",badgeDanger:"dapp-core-component__main__badge-danger","badge-light":"dapp-core-component__main__badge-light",badgeLight:"dapp-core-component__main__badge-light","badge-dark":"dapp-core-component__main__badge-dark",badgeDark:"dapp-core-component__main__badge-dark",card:"dapp-core-component__main__card","list-group":"dapp-core-component__main__list-group",listGroup:"dapp-core-component__main__list-group","card-header":"dapp-core-component__main__card-header",cardHeader:"dapp-core-component__main__card-header","card-footer":"dapp-core-component__main__card-footer",cardFooter:"dapp-core-component__main__card-footer","card-header-item":"dapp-core-component__main__card-header-item",cardHeaderItem:"dapp-core-component__main__card-header-item",h6:"dapp-core-component__main__h6",compact:"dapp-core-component__main__compact","card-body":"dapp-core-component__main__card-body",cardBody:"dapp-core-component__main__card-body","card-title":"dapp-core-component__main__card-title",cardTitle:"dapp-core-component__main__card-title","card-subtitle":"dapp-core-component__main__card-subtitle",cardSubtitle:"dapp-core-component__main__card-subtitle","card-text":"dapp-core-component__main__card-text",cardText:"dapp-core-component__main__card-text","card-link":"dapp-core-component__main__card-link",cardLink:"dapp-core-component__main__card-link",text:"dapp-core-component__main__text","text-white":"dapp-core-component__main__text-white",textWhite:"dapp-core-component__main__text-white","text-monospace":"dapp-core-component__main__text-monospace",textMonospace:"dapp-core-component__main__text-monospace","trim-size-xl":"dapp-core-component__main__trim-size-xl",trimSizeXl:"dapp-core-component__main__trim-size-xl",trim:"dapp-core-component__main__trim","trim-only-sm":"dapp-core-component__main__trim-only-sm",trimOnlySm:"dapp-core-component__main__trim-only-sm","text-justify":"dapp-core-component__main__text-justify",textJustify:"dapp-core-component__main__text-justify","text-wrap":"dapp-core-component__main__text-wrap",textWrap:"dapp-core-component__main__text-wrap","text-nowrap":"dapp-core-component__main__text-nowrap",textNowrap:"dapp-core-component__main__text-nowrap","text-truncate":"dapp-core-component__main__text-truncate",textTruncate:"dapp-core-component__main__text-truncate","text-left":"dapp-core-component__main__text-left",textLeft:"dapp-core-component__main__text-left","text-right":"dapp-core-component__main__text-right",textRight:"dapp-core-component__main__text-right","text-center":"dapp-core-component__main__text-center",textCenter:"dapp-core-component__main__text-center","text-primary":"dapp-core-component__main__text-primary",textPrimary:"dapp-core-component__main__text-primary","text-secondary":"dapp-core-component__main__text-secondary",textSecondary:"dapp-core-component__main__text-secondary","text-primary-highlight":"dapp-core-component__main__text-primary-highlight",textPrimaryHighlight:"dapp-core-component__main__text-primary-highlight","text-muted":"dapp-core-component__main__text-muted",textMuted:"dapp-core-component__main__text-muted","text-success":"dapp-core-component__main__text-success",textSuccess:"dapp-core-component__main__text-success","text-body":"dapp-core-component__main__text-body",textBody:"dapp-core-component__main__text-body","text-black-50":"dapp-core-component__main__text-black-50",textBlack50:"dapp-core-component__main__text-black-50","text-white-50":"dapp-core-component__main__text-white-50",textWhite50:"dapp-core-component__main__text-white-50","text-lg-right":"dapp-core-component__main__text-lg-right",textLgRight:"dapp-core-component__main__text-lg-right","text-hide":"dapp-core-component__main__text-hide",textHide:"dapp-core-component__main__text-hide","text-decoration-none":"dapp-core-component__main__text-decoration-none",textDecorationNone:"dapp-core-component__main__text-decoration-none","text-break":"dapp-core-component__main__text-break",textBreak:"dapp-core-component__main__text-break","text-reset":"dapp-core-component__main__text-reset",textReset:"dapp-core-component__main__text-reset","text-lowercase":"dapp-core-component__main__text-lowercase",textLowercase:"dapp-core-component__main__text-lowercase","text-uppercase":"dapp-core-component__main__text-uppercase",textUppercase:"dapp-core-component__main__text-uppercase","text-capitalize":"dapp-core-component__main__text-capitalize",textCapitalize:"dapp-core-component__main__text-capitalize","text-danger":"dapp-core-component__main__text-danger",textDanger:"dapp-core-component__main__text-danger","text-warning":"dapp-core-component__main__text-warning",textWarning:"dapp-core-component__main__text-warning","trim-fs-sm":"dapp-core-component__main__trim-fs-sm",trimFsSm:"dapp-core-component__main__trim-fs-sm","mx-auto":"dapp-core-component__main__mx-auto",mxAuto:"dapp-core-component__main__mx-auto","m-0":"dapp-core-component__main__m-0",m0:"dapp-core-component__main__m-0","mt-0":"dapp-core-component__main__mt-0",mt0:"dapp-core-component__main__mt-0","my-0":"dapp-core-component__main__my-0",my0:"dapp-core-component__main__my-0","mr-0":"dapp-core-component__main__mr-0",mr0:"dapp-core-component__main__mr-0","mx-0":"dapp-core-component__main__mx-0",mx0:"dapp-core-component__main__mx-0","mb-0":"dapp-core-component__main__mb-0",mb0:"dapp-core-component__main__mb-0","ml-0":"dapp-core-component__main__ml-0",ml0:"dapp-core-component__main__ml-0","m-1":"dapp-core-component__main__m-1",m1:"dapp-core-component__main__m-1","mt-1":"dapp-core-component__main__mt-1",mt1:"dapp-core-component__main__mt-1","my-1":"dapp-core-component__main__my-1",my1:"dapp-core-component__main__my-1","mr-1":"dapp-core-component__main__mr-1",mr1:"dapp-core-component__main__mr-1","mx-1":"dapp-core-component__main__mx-1",mx1:"dapp-core-component__main__mx-1","mb-1":"dapp-core-component__main__mb-1",mb1:"dapp-core-component__main__mb-1","ml-1":"dapp-core-component__main__ml-1",ml1:"dapp-core-component__main__ml-1","m-2":"dapp-core-component__main__m-2",m2:"dapp-core-component__main__m-2","mt-2":"dapp-core-component__main__mt-2",mt2:"dapp-core-component__main__mt-2","my-2":"dapp-core-component__main__my-2",my2:"dapp-core-component__main__my-2","mr-2":"dapp-core-component__main__mr-2",mr2:"dapp-core-component__main__mr-2","mx-2":"dapp-core-component__main__mx-2",mx2:"dapp-core-component__main__mx-2","mb-2":"dapp-core-component__main__mb-2",mb2:"dapp-core-component__main__mb-2","ml-2":"dapp-core-component__main__ml-2",ml2:"dapp-core-component__main__ml-2","m-3":"dapp-core-component__main__m-3",m3:"dapp-core-component__main__m-3","mt-3":"dapp-core-component__main__mt-3",mt3:"dapp-core-component__main__mt-3","my-3":"dapp-core-component__main__my-3",my3:"dapp-core-component__main__my-3","mr-3":"dapp-core-component__main__mr-3",mr3:"dapp-core-component__main__mr-3","mx-3":"dapp-core-component__main__mx-3",mx3:"dapp-core-component__main__mx-3","mb-3":"dapp-core-component__main__mb-3",mb3:"dapp-core-component__main__mb-3","ml-3":"dapp-core-component__main__ml-3",ml3:"dapp-core-component__main__ml-3","m-4":"dapp-core-component__main__m-4",m4:"dapp-core-component__main__m-4","mt-4":"dapp-core-component__main__mt-4",mt4:"dapp-core-component__main__mt-4","my-4":"dapp-core-component__main__my-4",my4:"dapp-core-component__main__my-4","mr-4":"dapp-core-component__main__mr-4",mr4:"dapp-core-component__main__mr-4","mx-4":"dapp-core-component__main__mx-4",mx4:"dapp-core-component__main__mx-4","mb-4":"dapp-core-component__main__mb-4",mb4:"dapp-core-component__main__mb-4","ml-4":"dapp-core-component__main__ml-4",ml4:"dapp-core-component__main__ml-4","m-5":"dapp-core-component__main__m-5",m5:"dapp-core-component__main__m-5","mt-5":"dapp-core-component__main__mt-5",mt5:"dapp-core-component__main__mt-5","my-5":"dapp-core-component__main__my-5",my5:"dapp-core-component__main__my-5","mr-5":"dapp-core-component__main__mr-5",mr5:"dapp-core-component__main__mr-5","mx-5":"dapp-core-component__main__mx-5",mx5:"dapp-core-component__main__mx-5","mb-5":"dapp-core-component__main__mb-5",mb5:"dapp-core-component__main__mb-5","ml-5":"dapp-core-component__main__ml-5",ml5:"dapp-core-component__main__ml-5","p-0":"dapp-core-component__main__p-0",p0:"dapp-core-component__main__p-0","pt-0":"dapp-core-component__main__pt-0",pt0:"dapp-core-component__main__pt-0","py-0":"dapp-core-component__main__py-0",py0:"dapp-core-component__main__py-0","pr-0":"dapp-core-component__main__pr-0",pr0:"dapp-core-component__main__pr-0","px-0":"dapp-core-component__main__px-0",px0:"dapp-core-component__main__px-0","pb-0":"dapp-core-component__main__pb-0",pb0:"dapp-core-component__main__pb-0","pl-0":"dapp-core-component__main__pl-0",pl0:"dapp-core-component__main__pl-0","p-1":"dapp-core-component__main__p-1",p1:"dapp-core-component__main__p-1","pt-1":"dapp-core-component__main__pt-1",pt1:"dapp-core-component__main__pt-1","py-1":"dapp-core-component__main__py-1",py1:"dapp-core-component__main__py-1","pr-1":"dapp-core-component__main__pr-1",pr1:"dapp-core-component__main__pr-1","px-1":"dapp-core-component__main__px-1",px1:"dapp-core-component__main__px-1","pb-1":"dapp-core-component__main__pb-1",pb1:"dapp-core-component__main__pb-1","pl-1":"dapp-core-component__main__pl-1",pl1:"dapp-core-component__main__pl-1","p-2":"dapp-core-component__main__p-2",p2:"dapp-core-component__main__p-2","pt-2":"dapp-core-component__main__pt-2",pt2:"dapp-core-component__main__pt-2","py-2":"dapp-core-component__main__py-2",py2:"dapp-core-component__main__py-2","pr-2":"dapp-core-component__main__pr-2",pr2:"dapp-core-component__main__pr-2","px-2":"dapp-core-component__main__px-2",px2:"dapp-core-component__main__px-2","pb-2":"dapp-core-component__main__pb-2",pb2:"dapp-core-component__main__pb-2","pl-2":"dapp-core-component__main__pl-2",pl2:"dapp-core-component__main__pl-2","p-3":"dapp-core-component__main__p-3",p3:"dapp-core-component__main__p-3","pt-3":"dapp-core-component__main__pt-3",pt3:"dapp-core-component__main__pt-3","py-3":"dapp-core-component__main__py-3",py3:"dapp-core-component__main__py-3","pr-3":"dapp-core-component__main__pr-3",pr3:"dapp-core-component__main__pr-3","px-3":"dapp-core-component__main__px-3",px3:"dapp-core-component__main__px-3","pb-3":"dapp-core-component__main__pb-3",pb3:"dapp-core-component__main__pb-3","pl-3":"dapp-core-component__main__pl-3",pl3:"dapp-core-component__main__pl-3","p-4":"dapp-core-component__main__p-4",p4:"dapp-core-component__main__p-4","pt-4":"dapp-core-component__main__pt-4",pt4:"dapp-core-component__main__pt-4","py-4":"dapp-core-component__main__py-4",py4:"dapp-core-component__main__py-4","pr-4":"dapp-core-component__main__pr-4",pr4:"dapp-core-component__main__pr-4","px-4":"dapp-core-component__main__px-4",px4:"dapp-core-component__main__px-4","pb-4":"dapp-core-component__main__pb-4",pb4:"dapp-core-component__main__pb-4","pl-4":"dapp-core-component__main__pl-4",pl4:"dapp-core-component__main__pl-4","p-5":"dapp-core-component__main__p-5",p5:"dapp-core-component__main__p-5","pt-5":"dapp-core-component__main__pt-5",pt5:"dapp-core-component__main__pt-5","py-5":"dapp-core-component__main__py-5",py5:"dapp-core-component__main__py-5","pr-5":"dapp-core-component__main__pr-5",pr5:"dapp-core-component__main__pr-5","px-5":"dapp-core-component__main__px-5",px5:"dapp-core-component__main__px-5","pb-5":"dapp-core-component__main__pb-5",pb5:"dapp-core-component__main__pb-5","pl-5":"dapp-core-component__main__pl-5",pl5:"dapp-core-component__main__pl-5","m-n1":"dapp-core-component__main__m-n1",mN1:"dapp-core-component__main__m-n1","mt-n1":"dapp-core-component__main__mt-n1",mtN1:"dapp-core-component__main__mt-n1","my-n1":"dapp-core-component__main__my-n1",myN1:"dapp-core-component__main__my-n1","mr-n1":"dapp-core-component__main__mr-n1",mrN1:"dapp-core-component__main__mr-n1","mx-n1":"dapp-core-component__main__mx-n1",mxN1:"dapp-core-component__main__mx-n1","mb-n1":"dapp-core-component__main__mb-n1",mbN1:"dapp-core-component__main__mb-n1","ml-n1":"dapp-core-component__main__ml-n1",mlN1:"dapp-core-component__main__ml-n1","m-n2":"dapp-core-component__main__m-n2",mN2:"dapp-core-component__main__m-n2","mt-n2":"dapp-core-component__main__mt-n2",mtN2:"dapp-core-component__main__mt-n2","my-n2":"dapp-core-component__main__my-n2",myN2:"dapp-core-component__main__my-n2","mr-n2":"dapp-core-component__main__mr-n2",mrN2:"dapp-core-component__main__mr-n2","mx-n2":"dapp-core-component__main__mx-n2",mxN2:"dapp-core-component__main__mx-n2","mb-n2":"dapp-core-component__main__mb-n2",mbN2:"dapp-core-component__main__mb-n2","ml-n2":"dapp-core-component__main__ml-n2",mlN2:"dapp-core-component__main__ml-n2","m-n3":"dapp-core-component__main__m-n3",mN3:"dapp-core-component__main__m-n3","mt-n3":"dapp-core-component__main__mt-n3",mtN3:"dapp-core-component__main__mt-n3","my-n3":"dapp-core-component__main__my-n3",myN3:"dapp-core-component__main__my-n3","mr-n3":"dapp-core-component__main__mr-n3",mrN3:"dapp-core-component__main__mr-n3","mx-n3":"dapp-core-component__main__mx-n3",mxN3:"dapp-core-component__main__mx-n3","mb-n3":"dapp-core-component__main__mb-n3",mbN3:"dapp-core-component__main__mb-n3","ml-n3":"dapp-core-component__main__ml-n3",mlN3:"dapp-core-component__main__ml-n3","m-n4":"dapp-core-component__main__m-n4",mN4:"dapp-core-component__main__m-n4","mt-n4":"dapp-core-component__main__mt-n4",mtN4:"dapp-core-component__main__mt-n4","my-n4":"dapp-core-component__main__my-n4",myN4:"dapp-core-component__main__my-n4","mr-n4":"dapp-core-component__main__mr-n4",mrN4:"dapp-core-component__main__mr-n4","mx-n4":"dapp-core-component__main__mx-n4",mxN4:"dapp-core-component__main__mx-n4","mb-n4":"dapp-core-component__main__mb-n4",mbN4:"dapp-core-component__main__mb-n4","ml-n4":"dapp-core-component__main__ml-n4",mlN4:"dapp-core-component__main__ml-n4","m-n5":"dapp-core-component__main__m-n5",mN5:"dapp-core-component__main__m-n5","mt-n5":"dapp-core-component__main__mt-n5",mtN5:"dapp-core-component__main__mt-n5","my-n5":"dapp-core-component__main__my-n5",myN5:"dapp-core-component__main__my-n5","mr-n5":"dapp-core-component__main__mr-n5",mrN5:"dapp-core-component__main__mr-n5","mx-n5":"dapp-core-component__main__mx-n5",mxN5:"dapp-core-component__main__mx-n5","mb-n5":"dapp-core-component__main__mb-n5",mbN5:"dapp-core-component__main__mb-n5","ml-n5":"dapp-core-component__main__ml-n5",mlN5:"dapp-core-component__main__ml-n5","m-auto":"dapp-core-component__main__m-auto",mAuto:"dapp-core-component__main__m-auto","mt-auto":"dapp-core-component__main__mt-auto",mtAuto:"dapp-core-component__main__mt-auto","my-auto":"dapp-core-component__main__my-auto",myAuto:"dapp-core-component__main__my-auto","mr-auto":"dapp-core-component__main__mr-auto",mrAuto:"dapp-core-component__main__mr-auto","mb-auto":"dapp-core-component__main__mb-auto",mbAuto:"dapp-core-component__main__mb-auto","ml-auto":"dapp-core-component__main__ml-auto",mlAuto:"dapp-core-component__main__ml-auto","mr-lg-1":"dapp-core-component__main__mr-lg-1",mrLg1:"dapp-core-component__main__mr-lg-1","mr-lg-4":"dapp-core-component__main__mr-lg-4",mrLg4:"dapp-core-component__main__mr-lg-4","mx-lg-4":"dapp-core-component__main__mx-lg-4",mxLg4:"dapp-core-component__main__mx-lg-4","mb-xl-2":"dapp-core-component__main__mb-xl-2",mbXl2:"dapp-core-component__main__mb-xl-2","pr-sm-2":"dapp-core-component__main__pr-sm-2",prSm2:"dapp-core-component__main__pr-sm-2","px-sm-2":"dapp-core-component__main__px-sm-2",pxSm2:"dapp-core-component__main__px-sm-2","pr-xl-0":"dapp-core-component__main__pr-xl-0",prXl0:"dapp-core-component__main__pr-xl-0","px-xl-0":"dapp-core-component__main__px-xl-0",pxXl0:"dapp-core-component__main__px-xl-0",col:"dapp-core-component__main__col",row:"dapp-core-component__main__row","col-xl":"dapp-core-component__main__col-xl",colXl:"dapp-core-component__main__col-xl","col-xl-auto":"dapp-core-component__main__col-xl-auto",colXlAuto:"dapp-core-component__main__col-xl-auto","col-xl-12":"dapp-core-component__main__col-xl-12",colXl12:"dapp-core-component__main__col-xl-12","col-xl-11":"dapp-core-component__main__col-xl-11",colXl11:"dapp-core-component__main__col-xl-11","col-xl-10":"dapp-core-component__main__col-xl-10",colXl10:"dapp-core-component__main__col-xl-10","col-xl-9":"dapp-core-component__main__col-xl-9",colXl9:"dapp-core-component__main__col-xl-9","col-xl-8":"dapp-core-component__main__col-xl-8",colXl8:"dapp-core-component__main__col-xl-8","col-xl-7":"dapp-core-component__main__col-xl-7",colXl7:"dapp-core-component__main__col-xl-7","col-xl-6":"dapp-core-component__main__col-xl-6",colXl6:"dapp-core-component__main__col-xl-6","col-xl-5":"dapp-core-component__main__col-xl-5",colXl5:"dapp-core-component__main__col-xl-5","col-xl-4":"dapp-core-component__main__col-xl-4",colXl4:"dapp-core-component__main__col-xl-4","col-xl-3":"dapp-core-component__main__col-xl-3",colXl3:"dapp-core-component__main__col-xl-3","col-xl-2":"dapp-core-component__main__col-xl-2",colXl2:"dapp-core-component__main__col-xl-2","col-xl-1":"dapp-core-component__main__col-xl-1",colXl1:"dapp-core-component__main__col-xl-1","col-lg":"dapp-core-component__main__col-lg",colLg:"dapp-core-component__main__col-lg","col-lg-auto":"dapp-core-component__main__col-lg-auto",colLgAuto:"dapp-core-component__main__col-lg-auto","col-lg-12":"dapp-core-component__main__col-lg-12",colLg12:"dapp-core-component__main__col-lg-12","col-lg-11":"dapp-core-component__main__col-lg-11",colLg11:"dapp-core-component__main__col-lg-11","col-lg-10":"dapp-core-component__main__col-lg-10",colLg10:"dapp-core-component__main__col-lg-10","col-lg-9":"dapp-core-component__main__col-lg-9",colLg9:"dapp-core-component__main__col-lg-9","col-lg-8":"dapp-core-component__main__col-lg-8",colLg8:"dapp-core-component__main__col-lg-8","col-lg-7":"dapp-core-component__main__col-lg-7",colLg7:"dapp-core-component__main__col-lg-7","col-lg-6":"dapp-core-component__main__col-lg-6",colLg6:"dapp-core-component__main__col-lg-6","col-lg-5":"dapp-core-component__main__col-lg-5",colLg5:"dapp-core-component__main__col-lg-5","col-lg-4":"dapp-core-component__main__col-lg-4",colLg4:"dapp-core-component__main__col-lg-4","col-lg-3":"dapp-core-component__main__col-lg-3",colLg3:"dapp-core-component__main__col-lg-3","col-lg-2":"dapp-core-component__main__col-lg-2",colLg2:"dapp-core-component__main__col-lg-2","col-lg-1":"dapp-core-component__main__col-lg-1",colLg1:"dapp-core-component__main__col-lg-1","col-md":"dapp-core-component__main__col-md",colMd:"dapp-core-component__main__col-md","col-md-auto":"dapp-core-component__main__col-md-auto",colMdAuto:"dapp-core-component__main__col-md-auto","col-md-12":"dapp-core-component__main__col-md-12",colMd12:"dapp-core-component__main__col-md-12","col-md-11":"dapp-core-component__main__col-md-11",colMd11:"dapp-core-component__main__col-md-11","col-md-10":"dapp-core-component__main__col-md-10",colMd10:"dapp-core-component__main__col-md-10","col-md-9":"dapp-core-component__main__col-md-9",colMd9:"dapp-core-component__main__col-md-9","col-md-8":"dapp-core-component__main__col-md-8",colMd8:"dapp-core-component__main__col-md-8","col-md-7":"dapp-core-component__main__col-md-7",colMd7:"dapp-core-component__main__col-md-7","col-md-6":"dapp-core-component__main__col-md-6",colMd6:"dapp-core-component__main__col-md-6","col-md-5":"dapp-core-component__main__col-md-5",colMd5:"dapp-core-component__main__col-md-5","col-md-4":"dapp-core-component__main__col-md-4",colMd4:"dapp-core-component__main__col-md-4","col-md-3":"dapp-core-component__main__col-md-3",colMd3:"dapp-core-component__main__col-md-3","col-md-2":"dapp-core-component__main__col-md-2",colMd2:"dapp-core-component__main__col-md-2","col-md-1":"dapp-core-component__main__col-md-1",colMd1:"dapp-core-component__main__col-md-1","col-sm":"dapp-core-component__main__col-sm",colSm:"dapp-core-component__main__col-sm","col-sm-auto":"dapp-core-component__main__col-sm-auto",colSmAuto:"dapp-core-component__main__col-sm-auto","col-sm-12":"dapp-core-component__main__col-sm-12",colSm12:"dapp-core-component__main__col-sm-12","col-sm-11":"dapp-core-component__main__col-sm-11",colSm11:"dapp-core-component__main__col-sm-11","col-sm-10":"dapp-core-component__main__col-sm-10",colSm10:"dapp-core-component__main__col-sm-10","col-sm-9":"dapp-core-component__main__col-sm-9",colSm9:"dapp-core-component__main__col-sm-9","col-sm-8":"dapp-core-component__main__col-sm-8",colSm8:"dapp-core-component__main__col-sm-8","col-sm-7":"dapp-core-component__main__col-sm-7",colSm7:"dapp-core-component__main__col-sm-7","col-sm-6":"dapp-core-component__main__col-sm-6",colSm6:"dapp-core-component__main__col-sm-6","col-sm-5":"dapp-core-component__main__col-sm-5",colSm5:"dapp-core-component__main__col-sm-5","col-sm-4":"dapp-core-component__main__col-sm-4",colSm4:"dapp-core-component__main__col-sm-4","col-sm-3":"dapp-core-component__main__col-sm-3",colSm3:"dapp-core-component__main__col-sm-3","col-sm-2":"dapp-core-component__main__col-sm-2",colSm2:"dapp-core-component__main__col-sm-2","col-sm-1":"dapp-core-component__main__col-sm-1",colSm1:"dapp-core-component__main__col-sm-1","col-auto":"dapp-core-component__main__col-auto",colAuto:"dapp-core-component__main__col-auto","col-12":"dapp-core-component__main__col-12",col12:"dapp-core-component__main__col-12","col-11":"dapp-core-component__main__col-11",col11:"dapp-core-component__main__col-11","col-10":"dapp-core-component__main__col-10",col10:"dapp-core-component__main__col-10","col-9":"dapp-core-component__main__col-9",col9:"dapp-core-component__main__col-9","col-8":"dapp-core-component__main__col-8",col8:"dapp-core-component__main__col-8","col-7":"dapp-core-component__main__col-7",col7:"dapp-core-component__main__col-7","col-6":"dapp-core-component__main__col-6",col6:"dapp-core-component__main__col-6","col-5":"dapp-core-component__main__col-5",col5:"dapp-core-component__main__col-5","col-4":"dapp-core-component__main__col-4",col4:"dapp-core-component__main__col-4","col-3":"dapp-core-component__main__col-3",col3:"dapp-core-component__main__col-3","col-2":"dapp-core-component__main__col-2",col2:"dapp-core-component__main__col-2","col-1":"dapp-core-component__main__col-1",col1:"dapp-core-component__main__col-1","flex-row":"dapp-core-component__main__flex-row",flexRow:"dapp-core-component__main__flex-row","flex-column":"dapp-core-component__main__flex-column",flexColumn:"dapp-core-component__main__flex-column","flex-row-reverse":"dapp-core-component__main__flex-row-reverse",flexRowReverse:"dapp-core-component__main__flex-row-reverse","flex-column-reverse":"dapp-core-component__main__flex-column-reverse",flexColumnReverse:"dapp-core-component__main__flex-column-reverse","flex-wrap":"dapp-core-component__main__flex-wrap",flexWrap:"dapp-core-component__main__flex-wrap","flex-nowrap":"dapp-core-component__main__flex-nowrap",flexNowrap:"dapp-core-component__main__flex-nowrap","flex-wrap-reverse":"dapp-core-component__main__flex-wrap-reverse",flexWrapReverse:"dapp-core-component__main__flex-wrap-reverse","flex-fill":"dapp-core-component__main__flex-fill",flexFill:"dapp-core-component__main__flex-fill","flex-grow-0":"dapp-core-component__main__flex-grow-0",flexGrow0:"dapp-core-component__main__flex-grow-0","flex-grow-1":"dapp-core-component__main__flex-grow-1",flexGrow1:"dapp-core-component__main__flex-grow-1","flex-shrink-0":"dapp-core-component__main__flex-shrink-0",flexShrink0:"dapp-core-component__main__flex-shrink-0","flex-shrink-1":"dapp-core-component__main__flex-shrink-1",flexShrink1:"dapp-core-component__main__flex-shrink-1","justify-content-start":"dapp-core-component__main__justify-content-start",justifyContentStart:"dapp-core-component__main__justify-content-start","justify-content-end":"dapp-core-component__main__justify-content-end",justifyContentEnd:"dapp-core-component__main__justify-content-end","justify-content-center":"dapp-core-component__main__justify-content-center",justifyContentCenter:"dapp-core-component__main__justify-content-center","justify-content-between":"dapp-core-component__main__justify-content-between",justifyContentBetween:"dapp-core-component__main__justify-content-between","justify-content-around":"dapp-core-component__main__justify-content-around",justifyContentAround:"dapp-core-component__main__justify-content-around","align-items-start":"dapp-core-component__main__align-items-start",alignItemsStart:"dapp-core-component__main__align-items-start","align-items-end":"dapp-core-component__main__align-items-end",alignItemsEnd:"dapp-core-component__main__align-items-end","align-items-center":"dapp-core-component__main__align-items-center",alignItemsCenter:"dapp-core-component__main__align-items-center","align-items-baseline":"dapp-core-component__main__align-items-baseline",alignItemsBaseline:"dapp-core-component__main__align-items-baseline","align-items-stretch":"dapp-core-component__main__align-items-stretch",alignItemsStretch:"dapp-core-component__main__align-items-stretch","align-content-start":"dapp-core-component__main__align-content-start",alignContentStart:"dapp-core-component__main__align-content-start","align-content-end":"dapp-core-component__main__align-content-end",alignContentEnd:"dapp-core-component__main__align-content-end","align-content-center":"dapp-core-component__main__align-content-center",alignContentCenter:"dapp-core-component__main__align-content-center","align-content-between":"dapp-core-component__main__align-content-between",alignContentBetween:"dapp-core-component__main__align-content-between","align-content-around":"dapp-core-component__main__align-content-around",alignContentAround:"dapp-core-component__main__align-content-around","align-content-stretch":"dapp-core-component__main__align-content-stretch",alignContentStretch:"dapp-core-component__main__align-content-stretch","align-self-auto":"dapp-core-component__main__align-self-auto",alignSelfAuto:"dapp-core-component__main__align-self-auto","align-self-start":"dapp-core-component__main__align-self-start",alignSelfStart:"dapp-core-component__main__align-self-start","align-self-end":"dapp-core-component__main__align-self-end",alignSelfEnd:"dapp-core-component__main__align-self-end","align-self-center":"dapp-core-component__main__align-self-center",alignSelfCenter:"dapp-core-component__main__align-self-center","align-self-baseline":"dapp-core-component__main__align-self-baseline",alignSelfBaseline:"dapp-core-component__main__align-self-baseline","align-self-stretch":"dapp-core-component__main__align-self-stretch",alignSelfStretch:"dapp-core-component__main__align-self-stretch","flex-sm-row":"dapp-core-component__main__flex-sm-row",flexSmRow:"dapp-core-component__main__flex-sm-row","flex-lg-row":"dapp-core-component__main__flex-lg-row",flexLgRow:"dapp-core-component__main__flex-lg-row","flex-lg-wrap":"dapp-core-component__main__flex-lg-wrap",flexLgWrap:"dapp-core-component__main__flex-lg-wrap","form-check":"dapp-core-component__main__form-check",formCheck:"dapp-core-component__main__form-check","form-check-label":"dapp-core-component__main__form-check-label",formCheckLabel:"dapp-core-component__main__form-check-label","form-check-input":"dapp-core-component__main__form-check-input",formCheckInput:"dapp-core-component__main__form-check-input","form-group":"dapp-core-component__main__form-group",formGroup:"dapp-core-component__main__form-group","form-control":"dapp-core-component__main__form-control",formControl:"dapp-core-component__main__form-control","form-label":"dapp-core-component__main__form-label",formLabel:"dapp-core-component__main__form-label","form-text":"dapp-core-component__main__form-text",formText:"dapp-core-component__main__form-text",border:"dapp-core-component__main__border","border-top":"dapp-core-component__main__border-top",borderTop:"dapp-core-component__main__border-top","border-right":"dapp-core-component__main__border-right",borderRight:"dapp-core-component__main__border-right","border-bottom":"dapp-core-component__main__border-bottom",borderBottom:"dapp-core-component__main__border-bottom","border-left":"dapp-core-component__main__border-left",borderLeft:"dapp-core-component__main__border-left","border-0":"dapp-core-component__main__border-0",border0:"dapp-core-component__main__border-0","border-top-0":"dapp-core-component__main__border-top-0",borderTop0:"dapp-core-component__main__border-top-0","border-right-0":"dapp-core-component__main__border-right-0",borderRight0:"dapp-core-component__main__border-right-0","border-bottom-0":"dapp-core-component__main__border-bottom-0",borderBottom0:"dapp-core-component__main__border-bottom-0","border-left-0":"dapp-core-component__main__border-left-0",borderLeft0:"dapp-core-component__main__border-left-0","rounded-sm":"dapp-core-component__main__rounded-sm",roundedSm:"dapp-core-component__main__rounded-sm",rounded:"dapp-core-component__main__rounded","rounded-top":"dapp-core-component__main__rounded-top",roundedTop:"dapp-core-component__main__rounded-top","rounded-right":"dapp-core-component__main__rounded-right",roundedRight:"dapp-core-component__main__rounded-right","rounded-bottom":"dapp-core-component__main__rounded-bottom",roundedBottom:"dapp-core-component__main__rounded-bottom","rounded-left":"dapp-core-component__main__rounded-left",roundedLeft:"dapp-core-component__main__rounded-left","rounded-lg":"dapp-core-component__main__rounded-lg",roundedLg:"dapp-core-component__main__rounded-lg","rounded-circle":"dapp-core-component__main__rounded-circle",roundedCircle:"dapp-core-component__main__rounded-circle","rounded-pill":"dapp-core-component__main__rounded-pill",roundedPill:"dapp-core-component__main__rounded-pill","rounded-0":"dapp-core-component__main__rounded-0",rounded0:"dapp-core-component__main__rounded-0",h1:"dapp-core-component__main__h1",h2:"dapp-core-component__main__h2",h3:"dapp-core-component__main__h3",h4:"dapp-core-component__main__h4",h5:"dapp-core-component__main__h5",lead:"dapp-core-component__main__lead","bg-primary":"dapp-core-component__main__bg-primary",bgPrimary:"dapp-core-component__main__bg-primary","bg-warning":"dapp-core-component__main__bg-warning",bgWarning:"dapp-core-component__main__bg-warning",warning:"dapp-core-component__main__warning","bg-danger":"dapp-core-component__main__bg-danger",bgDanger:"dapp-core-component__main__bg-danger",danger:"dapp-core-component__main__danger","bg-success":"dapp-core-component__main__bg-success",bgSuccess:"dapp-core-component__main__bg-success",success:"dapp-core-component__main__success","position-relative":"dapp-core-component__main__position-relative",positionRelative:"dapp-core-component__main__position-relative","position-absolute":"dapp-core-component__main__position-absolute",positionAbsolute:"dapp-core-component__main__position-absolute","w-25":"dapp-core-component__main__w-25",w25:"dapp-core-component__main__w-25","w-50":"dapp-core-component__main__w-50",w50:"dapp-core-component__main__w-50","w-75":"dapp-core-component__main__w-75",w75:"dapp-core-component__main__w-75","w-100":"dapp-core-component__main__w-100",w100:"dapp-core-component__main__w-100","w-auto":"dapp-core-component__main__w-auto",wAuto:"dapp-core-component__main__w-auto","h-25":"dapp-core-component__main__h-25",h25:"dapp-core-component__main__h-25","h-50":"dapp-core-component__main__h-50",h50:"dapp-core-component__main__h-50","h-75":"dapp-core-component__main__h-75",h75:"dapp-core-component__main__h-75","h-100":"dapp-core-component__main__h-100",h100:"dapp-core-component__main__h-100","h-auto":"dapp-core-component__main__h-auto",hAuto:"dapp-core-component__main__h-auto","mw-100":"dapp-core-component__main__mw-100",mw100:"dapp-core-component__main__mw-100","mh-100":"dapp-core-component__main__mh-100",mh100:"dapp-core-component__main__mh-100","min-vw-100":"dapp-core-component__main__min-vw-100",minVw100:"dapp-core-component__main__min-vw-100","min-vh-100":"dapp-core-component__main__min-vh-100",minVh100:"dapp-core-component__main__min-vh-100","vw-100":"dapp-core-component__main__vw-100",vw100:"dapp-core-component__main__vw-100","vh-100":"dapp-core-component__main__vh-100",vh100:"dapp-core-component__main__vh-100","locked-icon":"dapp-core-component__main__locked-icon",lockedIcon:"dapp-core-component__main__locked-icon","shadow-sm":"dapp-core-component__main__shadow-sm",shadowSm:"dapp-core-component__main__shadow-sm",shadow:"dapp-core-component__main__shadow","shadow-lg":"dapp-core-component__main__shadow-lg",shadowLg:"dapp-core-component__main__shadow-lg","shadow-none":"dapp-core-component__main__shadow-none",shadowNone:"dapp-core-component__main__shadow-none","d-none":"dapp-core-component__main__d-none",dNone:"dapp-core-component__main__d-none","d-inline":"dapp-core-component__main__d-inline",dInline:"dapp-core-component__main__d-inline","d-inline-block":"dapp-core-component__main__d-inline-block",dInlineBlock:"dapp-core-component__main__d-inline-block","d-block":"dapp-core-component__main__d-block",dBlock:"dapp-core-component__main__d-block","d-table":"dapp-core-component__main__d-table",dTable:"dapp-core-component__main__d-table","d-table-row":"dapp-core-component__main__d-table-row",dTableRow:"dapp-core-component__main__d-table-row","d-table-cell":"dapp-core-component__main__d-table-cell",dTableCell:"dapp-core-component__main__d-table-cell","d-flex":"dapp-core-component__main__d-flex",dFlex:"dapp-core-component__main__d-flex","d-inline-flex":"dapp-core-component__main__d-inline-flex",dInlineFlex:"dapp-core-component__main__d-inline-flex","d-sm-flex":"dapp-core-component__main__d-sm-flex",dSmFlex:"dapp-core-component__main__d-sm-flex","table-sm":"dapp-core-component__main__table-sm",tableSm:"dapp-core-component__main__table-sm","table-bordered":"dapp-core-component__main__table-bordered",tableBordered:"dapp-core-component__main__table-bordered","table-borderless":"dapp-core-component__main__table-borderless",tableBorderless:"dapp-core-component__main__table-borderless","table-striped":"dapp-core-component__main__table-striped",tableStriped:"dapp-core-component__main__table-striped","thead-dark":"dapp-core-component__main__thead-dark",theadDark:"dapp-core-component__main__thead-dark","thead-light":"dapp-core-component__main__thead-light",theadLight:"dapp-core-component__main__thead-light","table-dark":"dapp-core-component__main__table-dark",tableDark:"dapp-core-component__main__table-dark","table-hover":"dapp-core-component__main__table-hover",tableHover:"dapp-core-component__main__table-hover","table-responsive":"dapp-core-component__main__table-responsive",tableResponsive:"dapp-core-component__main__table-responsive","transaction-hash":"dapp-core-component__main__transaction-hash",transactionHash:"dapp-core-component__main__transaction-hash","font-weight-light":"dapp-core-component__main__font-weight-light",fontWeightLight:"dapp-core-component__main__font-weight-light","font-weight-lighter":"dapp-core-component__main__font-weight-lighter",fontWeightLighter:"dapp-core-component__main__font-weight-lighter","font-weight-normal":"dapp-core-component__main__font-weight-normal",fontWeightNormal:"dapp-core-component__main__font-weight-normal","font-weight-medium":"dapp-core-component__main__font-weight-medium",fontWeightMedium:"dapp-core-component__main__font-weight-medium","font-weight-bold":"dapp-core-component__main__font-weight-bold",fontWeightBold:"dapp-core-component__main__font-weight-bold","font-weight-bolder":"dapp-core-component__main__font-weight-bolder",fontWeightBolder:"dapp-core-component__main__font-weight-bolder","font-italic":"dapp-core-component__main__font-italic",fontItalic:"dapp-core-component__main__font-italic","icon-state":"dapp-core-component__main__icon-state",iconState:"dapp-core-component__main__icon-state",half:"dapp-core-component__main__half","side-icon":"dapp-core-component__main__side-icon",sideIcon:"dapp-core-component__main__side-icon","toast-messages":"dapp-core-component__main__toast-messages",toastMessages:"dapp-core-component__main__toast-messages",toast:"dapp-core-component__main__toast",clickable:"dapp-core-component__main__clickable","toast-visible":"dapp-core-component__main__toast-visible",toastVisible:"dapp-core-component__main__toast-visible",progress:"dapp-core-component__main__progress","progress-bar":"dapp-core-component__main__progress-bar",progressBar:"dapp-core-component__main__progress-bar","content-height":"dapp-core-component__main__content-height",contentHeight:"dapp-core-component__main__content-height","tx-description":"dapp-core-component__main__tx-description",txDescription:"dapp-core-component__main__tx-description",close:"dapp-core-component__main__close","data-field":"dapp-core-component__main__data-field",dataField:"dapp-core-component__main__data-field",textarea:"dapp-core-component__main__textarea",highlighted:"dapp-core-component__main__highlighted","sc-call":"dapp-core-component__main__sc-call",scCall:"dapp-core-component__main__sc-call","small-font":"dapp-core-component__main__small-font",smallFont:"dapp-core-component__main__small-font","progress-steps":"dapp-core-component__main__progress-steps",progressSteps:"dapp-core-component__main__progress-steps",steps:"dapp-core-component__main__steps",dot:"dapp-core-component__main__dot","side-link":"dapp-core-component__main__side-link",sideLink:"dapp-core-component__main__side-link"};c();var yo=`.dapp-core-component__transactionsTable-styles__transactions-table {
  border: none;
  box-shadow: 0 0.75rem 1.875rem 0 rgba(90, 97, 105, 0.05);
  border-radius: 0.75rem;
  color: #000;
}
.dapp-core-component__transactionsTable-styles__transactions-table .dapp-core-component__transactionsTable-styles__transactions-table-header {
  border-bottom: none !important;
  border-radius: 0.75rem !important;
  padding: 1.75rem 2rem;
}
.dapp-core-component__transactionsTable-styles__transactions-table .dapp-core-component__transactionsTable-styles__transactions-table-header h6 {
  margin-bottom: 0;
}
.dapp-core-component__transactionsTable-styles__transactions-table .dapp-core-component__transactionsTable-styles__table-wrapper {
  overflow-x: auto;
}
.dapp-core-component__transactionsTable-styles__transactions-table .dapp-core-component__transactionsTable-styles__table-wrapper .dapp-core-component__transactionsTable-styles__table {
  white-space: nowrap;
  max-height: 500px;
  width: 100%;
}
.dapp-core-component__transactionsTable-styles__transactions-table .dapp-core-component__transactionsTable-styles__table-wrapper .dapp-core-component__transactionsTable-styles__table tr {
  border-top: 1px solid #e2e2e2;
}
.dapp-core-component__transactionsTable-styles__transactions-table .dapp-core-component__transactionsTable-styles__table-wrapper .dapp-core-component__transactionsTable-styles__table th {
  white-space: nowrap;
  padding: 0.75rem !important;
  background-color: #fbfbfb;
  border: 0;
  vertical-align: middle;
  font-weight: 400;
}
.dapp-core-component__transactionsTable-styles__transactions-table .dapp-core-component__transactionsTable-styles__table-wrapper .dapp-core-component__transactionsTable-styles__table th:first-child {
  padding-left: 2rem !important;
}
.dapp-core-component__transactionsTable-styles__transactions-table .dapp-core-component__transactionsTable-styles__table-wrapper .dapp-core-component__transactionsTable-styles__table td {
  white-space: nowrap;
  width: 0.1%;
  white-space: nowrap;
}
.dapp-core-component__transactionsTable-styles__transactions-table .dapp-core-component__transactionsTable-styles__table-wrapper .dapp-core-component__transactionsTable-styles__table td:first-child {
  padding-left: 1.25rem !important;
}
.dapp-core-component__transactionsTable-styles__transactions-table .dapp-core-component__transactionsTable-styles__table-wrapper .dapp-core-component__transactionsTable-styles__table .dapp-core-component__transactionsTable-styles__transaction-cell {
  padding: 0;
  margin: 0.75rem;
  font-variant-numeric: slashed-zero;
  color: #000;
}
.dapp-core-component__transactionsTable-styles__transactions-table .dapp-core-component__transactionsTable-styles__table-wrapper .dapp-core-component__transactionsTable-styles__table .dapp-core-component__transactionsTable-styles__transaction-cell a {
  color: #1f43f4 !important;
  font-weight: 500;
}
.dapp-core-component__transactionsTable-styles__transactions-table .dapp-core-component__transactionsTable-styles__table-wrapper .dapp-core-component__transactionsTable-styles__table .dapp-core-component__transactionsTable-styles__transaction-cell-margin {
  margin-left: 0;
}
.dapp-core-component__transactionsTable-styles__transactions-table .dapp-core-component__transactionsTable-styles__table-wrapper .dapp-core-component__transactionsTable-styles__table .dapp-core-component__transactionsTable-styles__transaction-cell-margin [class*=left] span,
.dapp-core-component__transactionsTable-styles__transactions-table .dapp-core-component__transactionsTable-styles__table-wrapper .dapp-core-component__transactionsTable-styles__table .dapp-core-component__transactionsTable-styles__transaction-cell-margin [class*=right] span {
  font-size: 1rem;
  color: #1f43f4 !important;
}
.dapp-core-component__transactionsTable-styles__transactions-table .dapp-core-component__transactionsTable-styles__table-wrapper .dapp-core-component__transactionsTable-styles__table .dapp-core-component__transactionsTable-styles__transaction-cell-margin [class*=ellipsis] {
  color: #1f43f4 !important;
}
.dapp-core-component__transactionsTable-styles__transactions-table .dapp-core-component__transactionsTable-styles__table-wrapper .dapp-core-component__transactionsTable-styles__table .dapp-core-component__transactionsTable-styles__transaction-cell-overflow {
  max-width: 9rem;
  overflow: hidden;
  text-overflow: ellipsis;
}
.dapp-core-component__transactionsTable-styles__transactions-table .dapp-core-component__transactionsTable-styles__table-wrapper .dapp-core-component__transactionsTable-styles__table .dapp-core-component__transactionsTable-styles__transaction-cell-link {
  position: relative;
  text-decoration: none;
}
.dapp-core-component__transactionsTable-styles__transactions-table .dapp-core-component__transactionsTable-styles__table-wrapper .dapp-core-component__transactionsTable-styles__table .dapp-core-component__transactionsTable-styles__transaction-cell-font-small {
  font-size: 0.75rem;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(yo));var ne={"transactions-table":"dapp-core-component__transactionsTable-styles__transactions-table",transactionsTable:"dapp-core-component__transactionsTable-styles__transactions-table","transactions-table-header":"dapp-core-component__transactionsTable-styles__transactions-table-header",transactionsTableHeader:"dapp-core-component__transactionsTable-styles__transactions-table-header","table-wrapper":"dapp-core-component__transactionsTable-styles__table-wrapper",tableWrapper:"dapp-core-component__transactionsTable-styles__table-wrapper",table:"dapp-core-component__transactionsTable-styles__table","transaction-cell":"dapp-core-component__transactionsTable-styles__transaction-cell",transactionCell:"dapp-core-component__transactionsTable-styles__transaction-cell","transaction-cell-margin":"dapp-core-component__transactionsTable-styles__transaction-cell-margin",transactionCellMargin:"dapp-core-component__transactionsTable-styles__transaction-cell-margin","transaction-cell-overflow":"dapp-core-component__transactionsTable-styles__transaction-cell-overflow",transactionCellOverflow:"dapp-core-component__transactionsTable-styles__transaction-cell-overflow","transaction-cell-link":"dapp-core-component__transactionsTable-styles__transaction-cell-link",transactionCellLink:"dapp-core-component__transactionsTable-styles__transaction-cell-link","transaction-cell-font-small":"dapp-core-component__transactionsTable-styles__transaction-cell-font-small",transactionCellFontSmall:"dapp-core-component__transactionsTable-styles__transaction-cell-font-small"};var on=({transaction:o})=>{var e;return Z.default.createElement("div",{className:ne.transactionCell},Z.default.createElement("span",{title:(e=o.action)==null?void 0:e.description,className:(0,oe.default)(w.badge,w.badgeSecondary,w.badgePill,w.fontWeightLight,w.p0),"data-testid":"transactionMethod"},Z.default.createElement("div",{className:(0,oe.default)(w.badge,w.badgeSecondary,w.badgePill)},Z.default.createElement("div",{className:(0,oe.default)(ne.transactionFunctionBadge,w.textTruncate,w.textCapitalize,w.textWhite,w.p1,ne.transactionCellFontSmall)},ee(o)))))};c();c();c();c();c();c();c();c();c();c();c();var W={devnet:{id:"devnet",chainId:"D",name:"Devnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://devnet-wallet.elrond.com",apiAddress:"https://devnet-api.elrond.com",explorerAddress:"http://devnet-explorer.elrond.com",apiTimeout:"4000"},testnet:{id:"testnet",chainId:"T",name:"Testnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://testnet-wallet.elrond.com",apiAddress:"https://testnet-api.elrond.com",explorerAddress:"http://testnet-explorer.elrond.com",apiTimeout:"4000"},mainnet:{id:"mainnet",chainId:"1",name:"Mainnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://maiar.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://wallet.elrond.com",apiAddress:"https://api.elrond.com",explorerAddress:"https://explorer.elrond.com",apiTimeout:"4000"}},{chainId:an,egldLabel:va}=W["devnet"],{chainId:rn,egldLabel:Na}=W["testnet"],{chainId:tn,egldLabel:Da}=W["mainnet"];var Aa={["devnet"]:an,["testnet"]:rn,["mainnet"]:tn},Ea={[an]:"devnet",[rn]:"testnet",[tn]:"mainnet"};c();c();c();c();c();var H=require("@elrondnetwork/erdjs");c();var cn=require("@elrondnetwork/erdjs");function wo(o){try{let e=new cn.Address(o);return Boolean(e.bech32())}catch(e){return!1}}function ae(o){return(o==null?void 0:o.startsWith("erd"))&&o.length===62&&/^\w+$/.test(o)&&wo(o)}c();function pn(o){try{return _.Buffer.from(o,"base64").toString()===atob(o)}catch(e){return!1}}var To=["ESDTNFTTransfer","ESDTNFTBurn","ESDTNFTAddQuantity","ESDTNFTCreate","MultiESDTNFTTransfer","ESDTTransfer","ESDTBurn","ESDTLocalMint","ESDTLocalBurn","ESDTWipe","ESDTFreeze"];function _n(o,e,n=""){if(!ae(o))return!1;if(new H.Address(o).isContractAddress())return!0;let t=Do({receiver:o,data:n});return t?new H.Address(t).isContractAddress()||No(o,e,n):!1}var ko=o=>o.toLowerCase().match(/[0-9a-f]/g),vo=o=>o.length%2===0;function No(o,e,n){let a=n==null?void 0:n.split("@");if(a==null)return!1;let[r,...t]=a,p=e!=null&&o!=null&&o===e,s=To.includes(r),l=t.every(f=>ko(f)&&vo(f));return p&&s&&l}function Do({receiver:o,data:e}){try{if(!e)return o;let n=pn(e)?H.TransactionPayload.fromEncoded(e).toString():e,a=Ao(n),r=n.split("@");return a>-1?r[a]:o}catch(n){console.log(n);return}}function Ao(o){return o.includes("MultiESDTNFTTransfer")?1:o.includes("ESDTNFTTransfer")?4:-1}c();var we=require("@elrondnetwork/erdjs"),Te=T(require("bignumber.js"));c();c();function mn(o){for(let e=0;e<o.length;e++)if(o.charCodeAt(e)>127)return!1;return!0}function dn(o){let e=o;try{let n=_.Buffer.from(o,"hex").toString();mn(n)&&n.length>1&&(e=n)}catch(n){}return e}var ye={tokenId:"",amount:""},Eo=o=>o.split("@").map((t,p)=>[2,3].includes(p)?t:dn(t));function sn(o){if(!o)return ye;let e=o.startsWith("ESDTTransfer"),n=o.startsWith("ESDTNFTTransfer")&&o.includes("@");if(e){let[,a,r]=o.split("@");try{let t=_.Buffer.from(a,"hex").toString("ascii");if(!t)return ye;let p=new Te.default("0x"+r.replace("0x","")).toString(10);return{tokenId:t,amount:p}}catch(t){console.error("Error getting token from transaction data",t)}}if(n)try{let[,a,r,t,p]=Eo(o);if([a,r,t,p].every(s=>Boolean(s))&&ae(new we.Address(p).bech32()))return{tokenId:`${a}-${r}`,amount:new Te.default(t,16).toString(10),collection:a,nonce:r,receiver:new we.Address(p).bech32()}}catch(a){}return ye}c();c();c();var L="accounts",C="transactions";var G="tokens",ln="blocks";var gn="nodes",fn="identities",un="collections",ke="nfts",ve="providers",Ne="sc-results",bn="contracts",hn="logs",xn="code",qn="locked-accounts",yn="roles";var R={shard:o=>`/${ln}?shard=${o}`,receiverShard:o=>`/${C}?receivershard=${o}`,senderShard:o=>`/${C}?sendershard=${o}`,transactionDetails:o=>`/${C}/${o}`,transactionDetailsScResults:o=>`/${C}/${o}/${Ne}`,transactionDetailsLogs:o=>`/${C}/${o}/${hn}`,nodeDetails:o=>`/${gn}/${o}`,accountDetails:o=>`/${L}/${o}`,accountDetailsContractCode:o=>`/${L}/${o}/${xn}`,accountDetailsTokens:o=>`/${L}/${o}/${G}`,accountDetailsNfts:o=>`/${L}/${o}/${ke}`,accountDetailsScResults:o=>`/${L}/${o}/${Ne}`,accountDetailsContracts:o=>`/${L}/${o}/${bn}`,identityDetails:o=>`/${fn}/${o}`,tokenDetails:o=>`/${G}/${o}`,tokenDetailsAccounts:o=>`/${G}/${o}/${L}`,tokenDetailsLockedAccounts:o=>`/${G}/${o}/${qn}`,tokenDetailsRoles:o=>`/${G}/${o}/${yn}`,collectionDetails:o=>`/${un}/${o}`,nftDetails:o=>`/${ke}/${o}`,providerDetails:o=>`/${ve}/${o}`,providerDetailsTransactions:o=>`/${ve}/${o}/${C}`};c();var De=o=>{var e,n,a,r;if(o.action){let t=[(e=o.action.arguments)==null?void 0:e.token,(n=o.action.arguments)==null?void 0:n.token1,(a=o.action.arguments)==null?void 0:a.token2,(r=o.action.arguments)==null?void 0:r.transfers].filter(s=>s!=null);return[].concat(...t)}return[]};c();c();function re(o){var n,a;let e=o.receiver;return(a=(n=o.action)==null?void 0:n.arguments)!=null&&a.receiver&&(e=o.action.arguments.receiver),e}function wn(o){let e=re(o);return o.receiver===e?o.receiverAssets:void 0}c();var Tn=!1;function So(o){Tn||(console.error(o),Tn=!0)}function V({explorerAddress:o,to:e}){try{return new URL(e).href}catch(n){return e.startsWith("/")||(So(`Link not prepended by / : ${e}`),e=`/${e}`),o?`${o}${e}`:e}}c();function kn(o,e,n){let a=o===e.sender,r=o===n,t=a&&r,p=(e==null?void 0:e.type)==="SmartContractResult";switch(!0){case p:return"Internal";case t:return"Self";case r:return"In";case a:default:return"Out"}}c();c();c();var Ro=T(require("bignumber.js"));c();var Lo=require("@elrondnetwork/erdjs"),Ae=T(require("bignumber.js"));c();var Mo=T(require("bignumber.js"));c();Ae.default.config({ROUNDING_MODE:Ae.default.ROUND_FLOOR});c();c();c();c();c();c();c();c();c();var jo=require("@elrondnetwork/erdjs");c();var ce=require("@elrondnetwork/erdjs"),Fo=require("@elrondnetwork/erdjs"),Uo=require("@elrondnetwork/erdjs-network-providers");c();c();var zo=T(require("bignumber.js"));c();c();c();c();c();c();c();var Bo=["reDelegateRewards","claimRewards","unBond"],Po=["wrapEgld","unwrapEgld"],Oo=["unStake"],Go=["unDelegate"];c();c();c();c();var Wo=T(require("bignumber.js"));c();c();c();c();var Ho=T(require("bignumber.js"));function Dn({transaction:o,address:e,explorerAddress:n}){var Y;let a=(Y=o.tokenIdentifier)!=null?Y:sn(o.data).tokenId,r=re(o),t=wn(o),p=kn(e,o,r),s=ee(o),l=De(o),f=V({explorerAddress:n,to:R.accountDetails(o.sender)}),u=V({explorerAddress:n,to:R.accountDetails(r)}),g=V({explorerAddress:n,to:R.senderShard(o.senderShard)}),y=V({explorerAddress:n,to:R.receiverShard(o.receiverShard)}),x=o.originalTxHash?`${o.originalTxHash}#${o.txHash}`:o.txHash,I=V({explorerAddress:n,to:R.transactionDetails(x)});return _e(ie({},o),{tokenIdentifier:a,receiver:r,receiverAssets:t,transactionDetails:{direction:p,method:s,transactionTokens:l,isContract:_n(o.sender)},links:{senderLink:f,receiverLink:u,senderShardLink:g,receiverShardLink:y,transactionLink:I}})}c();var Se=[{txHash:"5996572de1fc5df2c456602467e8cc1a446b8145ae79df8f6c670e0732ab3b04",gasLimit:12e6,gasPrice:1e9,gasUsed:6062e3,miniBlockHash:"54d07f7c1e4d3f66e9f2c30d44d1e35d768a2da98dfe50cc437d485b9c2afb92",nonce:241,receiver:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqz8llllsh6u4jp",receiverAssets:{name:"Staking: ARC Stake",description:"ARC Staking provides staking services on the Elrond Blockchain, the most advanced blockchain platform in the world.",tags:["staking","provider"],iconPng:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"},receiverShard:4294967295,round:7430784,sender:"erd1zwq3qaa3vk5suenlkj4cf0ullwefa6h3n2394k25pxv4sz0pwhhsj9u9vk",senderShard:1,signature:"b9d589aa9aabf521864ccd52647850d224bb0ce76e89aaa8ad62bf67b4b9434d586e1b5be0db10ffc251be2fa495a3284585a226f57ee147ae2158efe651900b",status:"success",value:"14167802221131358682",fee:"122000000000000",timestamp:1640702304,data:"ZGVsZWdhdGU=",function:"delegate",action:{category:"stake",name:"delegate",description:"Delegate 14.167802221131358682 eGLD to staking provider ARC Stake",arguments:{value:"14167802221131358682",providerName:"ARC Stake",providerAvatar:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"}},results:[{hash:"8970fe5f5548caa994e495798ee1874b07f4ac2156d6a3a9636f450211350784",timestamp:1640702304,nonce:0,gasLimit:0,gasPrice:1e9,value:"14167802221131358682",sender:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqz8llllsh6u4jp",receiver:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqplllst77y4l",senderAssets:{name:"Staking: ARC Stake",description:"ARC Staking provides staking services on the Elrond Blockchain, the most advanced blockchain platform in the world.",tags:["staking","provider"],iconPng:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"},receiverAssets:{name:"System: Staking module",description:"Smart contract containing all staked eGLD on the network",tags:["system","staking"],iconPng:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.png",iconSvg:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.svg"},prevTxHash:"5996572de1fc5df2c456602467e8cc1a446b8145ae79df8f6c670e0732ab3b04",originalTxHash:"5996572de1fc5df2c456602467e8cc1a446b8145ae79df8f6c670e0732ab3b04",callType:"0",miniBlockHash:"432d6b8022301b67f1d76b5a96d525758e934b026128251d7ad462edb003c4b9"},{hash:"d4d84be4aa09cf5d45f79ffed87c971e5d883e7581d31e00d585f65b2da0564d",timestamp:1640702316,nonce:242,gasLimit:0,gasPrice:1e9,value:"59380000000000",sender:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqz8llllsh6u4jp",receiver:"erd1zwq3qaa3vk5suenlkj4cf0ullwefa6h3n2394k25pxv4sz0pwhhsj9u9vk",senderAssets:{name:"Staking: ARC Stake",description:"ARC Staking provides staking services on the Elrond Blockchain, the most advanced blockchain platform in the world.",tags:["staking","provider"],iconPng:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"},data:"QDZmNmI=",prevTxHash:"5996572de1fc5df2c456602467e8cc1a446b8145ae79df8f6c670e0732ab3b04",originalTxHash:"5996572de1fc5df2c456602467e8cc1a446b8145ae79df8f6c670e0732ab3b04",callType:"0",miniBlockHash:"578fc3b632fa3899ae00a6c7c399bb80d9b8f3e7c9e9396f6730efa73900cf3e",logs:{id:"d4d84be4aa09cf5d45f79ffed87c971e5d883e7581d31e00d585f65b2da0564d",address:"erd1zwq3qaa3vk5suenlkj4cf0ullwefa6h3n2394k25pxv4sz0pwhhsj9u9vk",events:[{address:"erd1zwq3qaa3vk5suenlkj4cf0ullwefa6h3n2394k25pxv4sz0pwhhsj9u9vk",identifier:"completedTxEvent",topics:["WZZXLeH8XfLEVmAkZ+jMGkRrgUWued+PbGcOBzKrOwQ="],order:0}]}}],price:240.75,logs:{id:"5996572de1fc5df2c456602467e8cc1a446b8145ae79df8f6c670e0732ab3b04",address:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqz8llllsh6u4jp",addressAssets:{name:"Staking: ARC Stake",description:"ARC Staking provides staking services on the Elrond Blockchain, the most advanced blockchain platform in the world.",tags:["staking","provider"],iconPng:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"},events:[{address:"erd1zwq3qaa3vk5suenlkj4cf0ullwefa6h3n2394k25pxv4sz0pwhhsj9u9vk",identifier:"delegate",topics:["xJ4lDk6GYdo=","BKAbisZRO5yq","E8I=","Ha6dNo2YLEc53A=="],order:0}]},operations:[{id:"8970fe5f5548caa994e495798ee1874b07f4ac2156d6a3a9636f450211350784",action:"transfer",type:"egld",sender:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqz8llllsh6u4jp",receiver:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqplllst77y4l",senderAssets:{name:"Staking: ARC Stake",description:"ARC Staking provides staking services on the Elrond Blockchain, the most advanced blockchain platform in the world.",tags:["staking","provider"],iconPng:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"},receiverAssets:{name:"System: Staking module",description:"Smart contract containing all staked eGLD on the network",tags:["system","staking"],iconPng:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.png",iconSvg:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.svg"},value:"14167802221131358682"}]},{txHash:"5e32a3f1808e7909a25381784731d6fc247325cc0a2737595129e783d50a3065",gasLimit:12e6,gasPrice:1e9,gasUsed:12e6,miniBlockHash:"e233dd9bc6f3b886ef3c4c5dfd722b291ea189bd4cd7065720730bee84553774",nonce:348,receiver:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqz8llllsh6u4jp",receiverAssets:{name:"Staking: ARC Stake",description:"ARC Staking provides staking services on the Elrond Blockchain, the most advanced blockchain platform in the world.",tags:["staking","provider"],iconPng:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"},receiverShard:4294967295,round:7430390,sender:"erd122h3x0nk8jddu3tfqujs67jffwuwu300c23da5lyxl9z5hyqtq8shceft6",senderShard:1,signature:"c449a086173d86c6c136278cd129c402cc64b88c757c85273bafcaad40be2b5128bf638d549d123448b09e1b4467954b8d1db43ff5aaf0a2cec66f63e779ec0f",status:"success",value:"0",fee:"209595000000000",timestamp:1640699940,data:"dW5EZWxlZ2F0ZUA0NTYzOTE4MjQ0ZjQwMDAw",function:"unDelegate",action:{category:"stake",name:"unDelegate",description:"Undelegate 5 eGLD from staking provider ARC Stake",arguments:{value:"5000000000000000000",providerName:"ARC Stake",providerAvatar:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"}}},{txHash:"813b7314a27b5aa4fd7e7c9565a826eca12984b620f674c756b01595b74a7ced",gasLimit:6e6,gasPrice:1e9,gasUsed:1068e3,miniBlockHash:"4bd1cc674dffdb322aa9d88d023fa8ec8a4f1f93bee73b00ee261bef7e5c4ce0",nonce:55,receiver:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqz8llllsh6u4jp",receiverAssets:{name:"Staking: ARC Stake",description:"ARC Staking provides staking services on the Elrond Blockchain, the most advanced blockchain platform in the world.",tags:["staking","provider"],iconPng:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"},receiverShard:4294967295,round:7430236,sender:"erd194ym3hhkw5dl6qxnxvpnnafjj54dyr8lycnev6sg32mqx270pnpq98ryvf",senderShard:2,signature:"b03ebdd464c37c1906bf2c2f1d957736c551fa311f6f5ce2c34c5b6d7b6516e46cac110d2ce8cc846530f2f3ac136d1d3d496b15ec421a73610b73da32480207",status:"success",value:"0",fee:"78000000000000",timestamp:1640699016,data:"Y2xhaW1SZXdhcmRz",function:"claimRewards",action:{category:"stake",name:"claimRewards",description:"Claim rewards from staking provider ARC Stake",arguments:{providerName:"ARC Stake",providerAvatar:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"}}},{txHash:"96f70d60de9fd2b29ea42841a507fe8b8aa0de07e9ed877c9f0fa8ec1feccc51",gasLimit:12e6,gasPrice:1e9,gasUsed:12e6,miniBlockHash:"f7ce09752494703b0f7e70f65a6e8ce7cf91684db94056e95300bc94e0f2a4da",nonce:174,receiver:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqz8llllsh6u4jp",receiverAssets:{name:"Staking: ARC Stake",description:"ARC Staking provides staking services on the Elrond Blockchain, the most advanced blockchain platform in the world.",tags:["staking","provider"],iconPng:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"},receiverShard:4294967295,round:7429921,sender:"erd1vcsum7fapl6crfrpqg7petydufgxm9qtqnuvhdqygdc6xc6uutksuzfatm",senderShard:1,signature:"ae7c8ae1cfa88474088bf886f1aff12e0ff85c9e1b61facfc46c79aa5314746e7c0ccb4df79197879c4c2171278399f833f8173ec5e7c0f25d64cfb6558d8401",status:"success",value:"0",fee:"181380000000000",timestamp:1640697126,data:"d2l0aGRyYXc=",function:"withdraw",action:{category:"stake",name:"withdraw",description:"Withdraw from staking provider ARC Stake",arguments:{providerName:"ARC Stake",providerAvatar:"https://s3.amazonaws.com/keybase_processed_uploads/d7555347e2e2dc7c68974d39f8e7b405_360_360.jpg"}}},{txHash:"6f570b2e75883df0a525786bbe69da141e0e4c4fc39accac17c9819162e77fbc",gasLimit:1175500,gasPrice:1e9,gasUsed:463200,miniBlockHash:"599c1a26a61dcd9efa5de88deeb856a0278a7e4abeaee017dc72c22d6a118425",nonce:408,receiver:"erd1y8jm2342j4jcyy4kz7hv823lr36wd9ur2mmlq4ea5wn6jndhpnjqh5shcg",receiverShard:0,round:7228515,sender:"erd1y8jm2342j4jcyy4kz7hv823lr36wd9ur2mmlq4ea5wn6jndhpnjqh5shcg",senderShard:0,signature:"b1ee08ee093949d5974461ea8d673d1761139a9bf208cac75a271b93f5f6e3a1b1ebd1bac62bf04c240a7f39e097c0e5f2730df8fc33f113069917a4d1c9c509",status:"success",value:"0",fee:"227877000000000",timestamp:1639488690,data:"RVNEVE5GVFRyYW5zZmVyQDQxNTI1NDQzNTI0MTQ2NTQyZDMzMzIzMjYzMzY2NUAxMUAwMUA3MTc3YjMwYjU0OGM3N2VjMjVlNzQ2NmYwMzg2ZjZlNGQ0NmFlNGZhMzk3Mzg0ZGEzYmVkYWQxNmEwNTA0NzA3",action:{category:"esdtNft",name:"transfer",description:"Transfer",arguments:{transfers:[{type:"NonFungibleESDT",name:"Genesis",ticker:"ARTCRAFT-322c6e",collection:"ARTCRAFT-322c6e",identifier:"ARTCRAFT-322c6e-11",value:"1"}],receiver:"erd1w9mmxz6533m7cf08gehs8phkun2x4e8689ecfk3makk3dgzsgurszhsxk4"}}},{txHash:"d51a4d6c74f1ab359e01dd18148f01ef64ab49162570f0b325fa35103d9784fb",gasLimit:12e6,gasPrice:1e9,gasUsed:6075500,miniBlockHash:"f5d84f705198e940905b36bc6231a90bdb3c1cf6f7af1776bc33bae34e103180",nonce:256,receiver:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqq2lllllspsw9sj",receiverAssets:{name:"Staking: Unity Stake",description:"We are a dedicated Elrond Network staking provider aiming to make things as simple and safe as possible by providing personal and caring support.",tags:["staking","provider"],iconPng:"https://s3.amazonaws.com/keybase_processed_uploads/4dd9f0dc7321d396f61f271951bba405_360_360.jpg"},receiverShard:4294967295,round:7071577,sender:"erd1vt2c72styh9eyrlsss855tprxgvtjdhmjpw8kyuapnjudlggatks0q0gm4",senderShard:1,signature:"5d198321defff0c4f451bc16434913f827efb25bb9b86fbf60aeca5ccefc896e27faa7de821f3c998b47dba9f18bae60aa5cba882cef9ee9d37389b5ed262200",status:"success",value:"0",fee:"135500000000000",timestamp:1638547062,data:"cmVEZWxlZ2F0ZVJld2FyZHM=",function:"reDelegateRewards",action:{category:"stake",name:"reDelegateRewards",description:"Redelegate rewards from staking provider Unity Stake",arguments:{providerName:"Unity Stake",providerAvatar:"https://s3.amazonaws.com/keybase_processed_uploads/4dd9f0dc7321d396f61f271951bba405_360_360.jpg"}}},{txHash:"20ab299a06956e3ea122445b3d8a181a226c4dacb0730dd9c13ede7ab5ee9328",gasLimit:14e6,gasPrice:1e9,gasUsed:11520759,miniBlockHash:"4e53661c4ac0803d4e53aa3fd6fa1eae0b4da337bd4630d107229ad0d3af0a85",nonce:112,receiver:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",receiverShard:0,round:7057306,sender:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",senderShard:0,signature:"07ca4425308793064273435b2fdbfea61acdd7d34bbe6371c3a0e459256003c8f72d52c31cf4538e173965da8a91729686814f5d321426a56b85746f9ddcb002",status:"success",value:"0",fee:"501802590000000",timestamp:1638461436,data:"TXVsdGlFU0RUTkZUVHJhbnNmZXJAMDAwMDAwMDAwMDAwMDAwMDA1MDA5MDU2ZjAwMTEwMWI2ZTE1MTgwNDQwNjRiNDhjMmU0ZmI0OTJjZDg4NTQ4M0AwMkA0YzRiNGQ0NTU4MmQ2MTYxNjIzOTMxMzBAMDNAMDEzZmJlODVlZGM5MDAwMEA0YzRiNGQ0NTU4MmQ2MTYxNjIzOTMxMzBAMGQzZEAxMTJkNmFiZGQ5NWUxZDVlNDBANmQ2NTcyNjc2NTRjNmY2MzZiNjU2NDQxNzM3MzY1NzQ1NDZmNmI2NTZlNzM=",function:"mergeLockedAssetTokens",action:{category:"mex",name:"mergeLockedAssetTokens",description:"Merge 2 LKMEX positions into a single LKMEX position of value 316.957286073585",arguments:{transfers:[{type:"MetaESDT",name:"LockedMEX",ticker:"LKMEX",svgUrl:"https://media.elrond.com/tokens/asset/LKMEX-aab910/logo.svg",collection:"LKMEX-aab910",decimals:18,identifier:"LKMEX-aab910-03",value:"90000000000000000"},{type:"MetaESDT",name:"LockedMEX",ticker:"LKMEX",svgUrl:"https://media.elrond.com/tokens/asset/LKMEX-aab910/logo.svg",collection:"LKMEX-aab910",decimals:18,identifier:"LKMEX-aab910-0d3d",value:"316867286073585000000"}],receiver:"erd1qqqqqqqqqqqqqpgqjpt0qqgsrdhp2xqygpjtfrpwf76f9nvg2jpsg4q7th",functionName:"mergeLockedAssetTokens",receiverAssets:{name:"Maiar Exchange: Locked asset Contract",tags:["mex","lockedasset"]}}}},{txHash:"91d1be9345b8b93983f105bca08ec62c99e13cacb40098773c549a7450aacf3b",gasLimit:1e6,gasPrice:1e9,gasUsed:459600,miniBlockHash:"2c5add74b4224c785440fc957ea5389f288953fdd4c9ddab91d11a1e7f436939",nonce:170,receiver:"erd180yut8q4ve7mzuqtens26ct0tdg0hwmhjfr0s8t2z0ujukmet8rqtt2px0",receiverShard:2,round:7044527,sender:"erd180yut8q4ve7mzuqtens26ct0tdg0hwmhjfr0s8t2z0ujukmet8rqtt2px0",senderShard:2,signature:"cda937a0e6c62ff1f745c5500d1aa9cf0a2009051dc3fdfb2abeaa3c6701946bd3946c8811a8cea6addbdf598f57d27d4ab059d765c88af2085758c4ca2c5807",status:"success",value:"0",fee:"248631000000000",timestamp:1638384762,data:"RVNEVE5GVFRyYW5zZmVyQDRjNGI0NjQxNTI0ZDJkMzk2NDMxNjU2MTM4QDEyNmMxN0AxNTU1ODA4OWJkYmJmNmI5QGNjZmUxOTY4N2JkN2QxMjgwZTY1YTQ3ZTI5NDI0OGE3ODhiNjFkZmUxYTIyNDRmODBhNDEwMTEyMDFlN2MwMTU=",action:{category:"esdtNft",name:"transfer",description:"Transfer",arguments:{transfers:[{type:"MetaESDT",name:"LockedLPStaked",ticker:"LKFARM",svgUrl:"https://media.elrond.com/tokens/asset/LKFARM-9d1ea8/logo.svg",collection:"LKFARM-9d1ea8",decimals:18,identifier:"LKFARM-9d1ea8-126c17",value:"1537276176898979513"}],receiver:"erd1enlpj6rm6lgjsrn953lzjsjg57ytv807rg3yf7q2gyq3yq08cq2sfphazl"}}},{txHash:"5bbe1a53027e6850362fcb4633108627bbe241adbb48001e25e3cd8129861189",gasLimit:105e5,gasPrice:1e9,gasUsed:9755137,miniBlockHash:"29e8111d14131cfc1b80b354ef47899aecff43873b333636e80997bb58d39f6d",nonce:864,receiver:"erd1qqqqqqqqqqqqqpgqa0fsfshnff4n76jhcye6k7uvd7qacsq42jpsp6shh2",receiverAssets:{name:"Maiar Exchange: MEX/WEGLD Liquidity Pool",tags:["mex","liquiditypool"]},receiverShard:1,round:6885918,sender:"erd1dwpev7tgza3hv58u8syzlk8nxsnkm94slwvvsfs44qwu3dqega7sqq3uxx",senderShard:1,signature:"8f028a4635be9cd8e02c2ae180da1a65f95a2b65dea4f65c3b25863f9b2be0c1d6c1059d1c9aafe88dd05e80a709685a120042099a4f182116a071ed7e11500b",status:"success",value:"0",fee:"350496370000000",timestamp:1637433108,data:"RVNEVFRyYW5zZmVyQDU3NDU0NzRjNDQyZDYyNjQzNDY0MzczOUAwZGUwYjZiM2E3NjQwMDAwQDczNzc2MTcwNTQ2ZjZiNjU2ZTczNDY2OTc4NjU2NDQ5NmU3MDc1NzRANGQ0NTU4MmQzNDM1MzU2MzM1MzdAM2I4NGE4YzZmYmY2YjJhZDA5Y2Y=",function:"swapTokensFixedInput",action:{category:"mex",name:"swap",description:"Swap 1 WEGLD for a minimum of 281066.754391919467235791 MEX",arguments:{transfers:[{type:"FungibleESDT",name:"WrappedEGLD",ticker:"WEGLD",svgUrl:"https://media.elrond.com/tokens/asset/WEGLD-bd4d79/logo.svg",token:"WEGLD-bd4d79",decimals:18,value:"1000000000000000000"},{type:"FungibleESDT",name:"MEX",ticker:"MEX",svgUrl:"https://media.elrond.com/tokens/asset/MEX-455c57/logo.svg",token:"MEX-455c57",decimals:18,value:"281066754391919467235791"}],receiver:"erd1qqqqqqqqqqqqqpgqa0fsfshnff4n76jhcye6k7uvd7qacsq42jpsp6shh2",functionName:"swapTokensFixedInput",functionArgs:["4d45582d343535633537","3b84a8c6fbf6b2ad09cf"],receiverAssets:{name:"Maiar Exchange: MEX/WEGLD Liquidity Pool",tags:["mex","liquiditypool"]}}}},{txHash:"6b65404efa716ad6dc6a2f37b6b6a7677cea70815fb699451d6ff1875f2111e2",gasLimit:4e6,gasPrice:1e9,gasUsed:2111255,miniBlockHash:"29e8111d14131cfc1b80b354ef47899aecff43873b333636e80997bb58d39f6d",nonce:863,receiver:"erd1qqqqqqqqqqqqqpgqhe8t5jewej70zupmh44jurgn29psua5l2jps3ntjj3",receiverAssets:{name:"ESDT: WrappedEGLD Contract Shard 1",tags:["mex","wegld"]},receiverShard:1,round:6885918,sender:"erd1dwpev7tgza3hv58u8syzlk8nxsnkm94slwvvsfs44qwu3dqega7sqq3uxx",senderShard:1,signature:"521ce31af51e02f2bf31b9198807784cf7a37c97798acc7859e7c5c5c3e24651b15fe392af24febf90d5a2864614f12f1852eb6ad541e61c456448b6d1c86807",status:"success",value:"1000000000000000000",fee:"82492550000000",timestamp:1637433108,data:"d3JhcEVnbGQ=",function:"wrapEgld",action:{category:"mex",name:"wrapEgld",description:"Wrap 1 eGLD",arguments:{token:{type:"FungibleESDT",name:"WrappedEGLD",token:"WEGLD-bd4d79",ticker:"WEGLD",decimals:18,value:"1000000000000000000"},receiver:"erd1qqqqqqqqqqqqqpgqhe8t5jewej70zupmh44jurgn29psua5l2jps3ntjj3",receiverAssets:{name:"ESDT: WrappedEGLD Contract Shard 1",tags:["mex","wegld"]}}}},{txHash:"8db34cc6b7a8f8bcdfb4dbf0ca55cb885f625da1dfe316fc9e9d722ebc687524",gasLimit:5e5,gasPrice:1e9,gasUsed:316e3,miniBlockHash:"966bcf89dd731f1ed563e37dcab3ec8d923382a769eec6d4ab03eb490d573dee",nonce:813,receiver:"erd1k50edvacs6qvsfyuex38d8plzswus35knd4wxh39apjlpz0j308qpeywwr",receiverShard:2,round:6885918,sender:"erd1fdcfhqsa2qp0l6grzpehndgeum8kzavd5hxjn9h7f0llafwrvyfqvt6k2y",senderShard:2,signature:"8f7caff6c24b5a685008cb93b3bf4c8c53d50215d4e3c8de3f69aeeec4220f183840031a0b220dd38630f4456af11c80d550c486b174273a91fe47f27ae9c302",status:"success",value:"0",fee:"118000000000000",timestamp:1637433108,data:"RVNEVFRyYW5zZmVyQDU1NTM0NDQzMmQ2MzM3MzY2NjMxNjZAYWZkNTZkODA=",action:{category:"esdtNft",name:"transfer",description:"Transfer",arguments:{transfers:[{type:"FungibleESDT",name:"WrappedUSDC",ticker:"USDC",svgUrl:"https://media.elrond.com/tokens/asset/USDC-c76f1f/logo.svg",token:"USDC-c76f1f",decimals:6,value:"2950000000"}],receiver:"erd1k50edvacs6qvsfyuex38d8plzswus35knd4wxh39apjlpz0j308qpeywwr"}}},{txHash:"4abc9f3a18b2685bc8e34f26e0815512899787e10ef66b8ace105a8f48e3f694",gasLimit:4e6,gasPrice:1e9,gasUsed:2213320,miniBlockHash:"29e8111d14131cfc1b80b354ef47899aecff43873b333636e80997bb58d39f6d",nonce:761,receiver:"erd1qqqqqqqqqqqqqpgqhe8t5jewej70zupmh44jurgn29psua5l2jps3ntjj3",receiverAssets:{name:"ESDT: WrappedEGLD Contract Shard 1",tags:["mex","wegld"]},receiverShard:1,round:6885918,sender:"erd1e7ycjzurhan9tcryu5t65sv3pcj0rg5xh490cg9mmpy7k582zsgsyufld5",senderShard:1,signature:"27b2329b6fbd6b717dc664d5c081f9763f4731b8d6d8905001f07dce2781faa98c8fc99deccf2a7b7bf1e3e6666b732ab99c420ebdc1af9f8c5a8cc47808820e",status:"success",value:"0",fee:"183008200000000",timestamp:1637433108,data:"RVNEVFRyYW5zZmVyQDU3NDU0NzRjNDQyZDYyNjQzNDY0MzczOUA0NGVjYzgzNzEwYzU3OWQzQDc1NmU3NzcyNjE3MDQ1Njc2YzY0",function:"unwrapEgld",action:{category:"mex",name:"unwrapEgld",description:"Unwrap 4.966564627912948179 WEGLD",arguments:{transfers:[{type:"FungibleESDT",name:"WrappedEGLD",ticker:"WEGLD",svgUrl:"https://media.elrond.com/tokens/asset/WEGLD-bd4d79/logo.svg",token:"WEGLD-bd4d79",decimals:18,value:"4966564627912948179"}],receiver:"erd1qqqqqqqqqqqqqpgqhe8t5jewej70zupmh44jurgn29psua5l2jps3ntjj3",functionName:"unwrapEgld",receiverAssets:{name:"ESDT: WrappedEGLD Contract Shard 1",tags:["mex","wegld"]}}}},{txHash:"e060a7820af21bae38686ec036c2e28df6295d2492dde473c23c5ebbf83a40cf",gasLimit:6e7,gasPrice:1e9,gasUsed:51054664,miniBlockHash:"29e8111d14131cfc1b80b354ef47899aecff43873b333636e80997bb58d39f6d",nonce:377,receiver:"erd1096gglmcpgrfudgfr6ya8vjxev7gslzfnngyn5qgspt3df9nvlfsttsfwl",receiverShard:1,round:6885918,sender:"erd1096gglmcpgrfudgfr6ya8vjxev7gslzfnngyn5qgspt3df9nvlfsttsfwl",senderShard:1,signature:"bea080cfc7329bc4b181332935100f7772f0ba4133d5e8be807ca78dc9b09218ff4b9a02f730a2cfd28a3256f40324373d78df90c4b559ad6947207d40383500",status:"success",value:"0",fee:"903081640000000",timestamp:1637433108,data:"RVNEVE5GVFRyYW5zZmVyQDRjNGI0NjQxNTI0ZDJkMzk2NDMxNjU2MTM4QDAxNjcxMUA3MDlhNWYxNjJiOGM3YmU2QDAwMDAwMDAwMDAwMDAwMDAwNTAwMWUyYTE0MjhkZDFlM2E1MTQ2YjM5NjBkOWUwZjRhNTAzNjk5MDRlZTU0ODNANjM2YzYxNjk2ZDUyNjU3NzYxNzI2NDczNTA3MjZmNzg3OUAwMDAwMDAwMDAwMDAwMDAwMDUwMDI2NzUxODkzZDY3ODliZTllNWE5OTg2M2JhOWVlYWE4MDg4ZGQyNWY1NDgz",function:"claimRewardsProxy",action:{category:"mex",name:"claimRewards",description:"Claim rewards for 8.113902227485916134 LKFARM",arguments:{transfers:[{type:"MetaESDT",name:"LockedLPStaked",ticker:"LKFARM",svgUrl:"https://media.elrond.com/tokens/asset/LKFARM-9d1ea8/logo.svg",collection:"LKFARM-9d1ea8",decimals:18,identifier:"LKFARM-9d1ea8-016711",value:"8113902227485916134"}],receiver:"erd1qqqqqqqqqqqqqpgqrc4pg2xarca9z34njcxeur622qmfjp8w2jps89fxnl",functionName:"claimRewardsProxy",functionArgs:["0000000000000000050026751893d6789be9e5a99863ba9eeaa8088dd25f5483"],receiverAssets:{name:"Maiar Exchange: Proxy",tags:["maiar","exchange","proxy"],iconPng:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.png",iconSvg:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.svg"}}}},{txHash:"7b4ac0ef53f9b43c80cc2af6ae0a1e4e9065a3054a1358775edb6161eaa02811",gasLimit:12e6,gasPrice:1e9,gasUsed:10717267,miniBlockHash:"29e8111d14131cfc1b80b354ef47899aecff43873b333636e80997bb58d39f6d",nonce:274,receiver:"erd1x0syqh7tcauqx70qdakmaa48mkmnczh8kl5zw3y0tnp08kfuz9zsqtmmtk",receiverShard:1,round:6885918,sender:"erd1x0syqh7tcauqx70qdakmaa48mkmnczh8kl5zw3y0tnp08kfuz9zsqtmmtk",senderShard:1,signature:"c8ae158d499004313c8d7e0cbede71c9302eb080ea131bd6a53fc7d7fc9d71364f0ff80408948e2a09652003903352a0a13b8f0625497bda3dc1e03297150f07",status:"success",value:"0",fee:"508617670000000",timestamp:1637433108,data:"TXVsdGlFU0RUTkZUVHJhbnNmZXJAMDAwMDAwMDAwMDAwMDAwMDA1MDBlYmQzMDRjMmYzNGE2YjNmNmE1N2MxMzNhYjdiOGM2ZjgxZGM0MDE1NTQ4M0AwMkA1NzQ1NDc0YzQ0MmQ2MjY0MzQ2NDM3MzlAQDA0NGNmNjhjMjkwODUwMDlANGQ0NTU4MmQzNDM1MzU2MzM1MzdAQDEyYWZlMGQzODQ2MTMzMjBjNTQ0QDYxNjQ2NDRjNjk3MTc1Njk2NDY5NzQ3OUAwNDQxZjQxNGZmYTk4MjZmQDEyODAwYTE5MTRmOWQzZTU5MDE3",function:"addLiquidity",action:{category:"mex",name:"addLiquidity",description:"Added liquidity for 0.309893556225986569 WEGLD, 88246.97734239037166522 MEX",arguments:{transfers:[{type:"FungibleESDT",name:"WrappedEGLD",ticker:"WEGLD",svgUrl:"https://media.elrond.com/tokens/asset/WEGLD-bd4d79/logo.svg",token:"WEGLD-bd4d79",decimals:18,value:"309893556225986569"},{type:"FungibleESDT",name:"MEX",ticker:"MEX",svgUrl:"https://media.elrond.com/tokens/asset/MEX-455c57/logo.svg",token:"MEX-455c57",decimals:18,value:"88246977342390371665220"}],receiver:"erd1qqqqqqqqqqqqqpgqa0fsfshnff4n76jhcye6k7uvd7qacsq42jpsp6shh2",functionName:"addLiquidity",functionArgs:["0441f414ffa9826f","12800a1914f9d3e59017"],receiverAssets:{name:"Maiar Exchange: MEX/WEGLD Liquidity Pool",tags:["mex","liquiditypool"]}}}},{txHash:"a13bf18c6d91edb7dc403d915fd89298806f67d59fdd72f12370f2c6fe31966c",gasLimit:38e6,gasPrice:1e9,gasUsed:34290478,miniBlockHash:"29e8111d14131cfc1b80b354ef47899aecff43873b333636e80997bb58d39f6d",nonce:222,receiver:"erd1znusjpmfpukrtf0wscvf4su8yjqg393c092j8sxxvvrfk9m0rdss0646y0",receiverShard:1,round:6885918,sender:"erd1znusjpmfpukrtf0wscvf4su8yjqg393c092j8sxxvvrfk9m0rdss0646y0",senderShard:1,signature:"e06d88a54af0d9fc220050b69d5c2c4368765ce2a382871cc341531ad309cc695b983a11a24efe99697407ac028e7f4cbdd8ff1a79ed17765760fa33897e340b",status:"success",value:"0",fee:"753259780000000",timestamp:1637433108,data:"RVNEVE5GVFRyYW5zZmVyQDRjNGI0NjQxNTI0ZDJkMzk2NDMxNjU2MTM4QDAxNmEzMUAwYTAxMTBiYjNkMGZmNDJmN2ZlZDQ4QDAwMDAwMDAwMDAwMDAwMDAwNTAwMWUyYTE0MjhkZDFlM2E1MTQ2YjM5NjBkOWUwZjRhNTAzNjk5MDRlZTU0ODNANjM2ZjZkNzA2Zjc1NmU2NDUyNjU3NzYxNzI2NDczNTA3MjZmNzg3OUAwMDAwMDAwMDAwMDAwMDAwMDUwMDY1NmQwYWNjNTM1NjFjNWQ2ZjZmZDdkN2U4MmJmMTMyNDcwMTRmNjE1NDgz",function:"compoundRewardsProxy",action:{category:"mex",name:"compoundRewards",description:"Reinvest rewards for 12094289.202491940860325192 LKFARM",arguments:{transfers:[{type:"MetaESDT",name:"LockedLPStaked",ticker:"LKFARM",svgUrl:"https://media.elrond.com/tokens/asset/LKFARM-9d1ea8/logo.svg",collection:"LKFARM-9d1ea8",decimals:18,identifier:"LKFARM-9d1ea8-016a31",value:"12094289202491940860325192"}],receiver:"erd1qqqqqqqqqqqqqpgqrc4pg2xarca9z34njcxeur622qmfjp8w2jps89fxnl",functionName:"compoundRewardsProxy",functionArgs:["00000000000000000500656d0acc53561c5d6f6fd7d7e82bf13247014f615483"],receiverAssets:{name:"Maiar Exchange: Proxy",tags:["maiar","exchange","proxy"],iconPng:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.png",iconSvg:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.svg"}}}},{txHash:"8f3b73756ecf8500737d0f68d300585c19e9d9f38f3b112171ca71e6890c2ab2",gasLimit:165e5,gasPrice:1e9,gasUsed:11554861,miniBlockHash:"7b6cf2eb70d414973ac1294fb02ceec49d5b2f48a8fcf56e96eb13e71299912d",nonce:150,receiver:"erd1z3q7g93vj2vqz6sevse7cugauh6e5uszelwsfjledpzz9qc2xg2q97sqvr",receiverShard:0,round:6885918,sender:"erd1z3q7g93vj2vqz6sevse7cugauh6e5uszelwsfjledpzz9qc2xg2q97sqvr",senderShard:0,signature:"9ef7c45815f4147b75aaff150f55347f73efbbc7158731273c9c363c163aa34cabdd324d24449d8657a9faba896cc33fc64f86224123dab6ca71d7aa556b5004",status:"success",value:"0",fee:"438288610000000",timestamp:1637433108,data:"TXVsdGlFU0RUTkZUVHJhbnNmZXJAMDAwMDAwMDAwMDAwMDAwMDA1MDAyNjc1MTg5M2Q2Nzg5YmU5ZTVhOTk4NjNiYTllZWFhODA4OGRkMjVmNTQ4M0AwMUA0NTQ3NGM0NDRkNDU1ODJkMzA2MjY1Mzk2NTM1QEAwMzUxYmFiNTY3YWFkODE2NTlANjU2ZTc0NjU3MjQ2NjE3MjZkNDE2ZTY0NGM2ZjYzNmI1MjY1Nzc2MTcyNjQ3Mw==",function:"enterFarmAndLockRewards",action:{category:"mex",name:"enterFarm",description:"Enter farm and lock rewards with 61.229451140721546841 EGLDMEX",arguments:{transfers:[{type:"FungibleESDT",name:"EGLDMEXLP",ticker:"EGLDMEX",svgUrl:"https://media.elrond.com/tokens/asset/EGLDMEX-0be9e5/logo.svg",token:"EGLDMEX-0be9e5",decimals:18,value:"61229451140721546841"}],receiver:"erd1qqqqqqqqqqqqqpgqye633y7k0zd7nedfnp3m48h24qygm5jl2jpslxallh",functionName:"enterFarmAndLockRewards",receiverAssets:{name:"Maiar Exchange: EGLDMEXLPStaked Farm",tags:["mex","farm"]}}}},{txHash:"6e29bb7454bad9c145d76308f7c36efb5cc8a83619d2ca26a5ed165a302e50e7",gasLimit:36e6,gasPrice:1e9,gasUsed:34299559,miniBlockHash:"7b6cf2eb70d414973ac1294fb02ceec49d5b2f48a8fcf56e96eb13e71299912d",nonce:70,receiver:"erd1dasthqh3ydgtjnxj4xmzrm02rru7dk4ytq6vqsjdrq4p9lul7hjqms7q4m",receiverShard:0,round:6885918,sender:"erd1dasthqh3ydgtjnxj4xmzrm02rru7dk4ytq6vqsjdrq4p9lul7hjqms7q4m",senderShard:0,signature:"edafd4d9f8255f1bf8e416fd3944b74784faaf86baa242978e11a52f67e07b6273d4bc209f46e11d192b60b1181075b36af243d938da64bdf8940b4426a0900d",status:"success",value:"0",fee:"738500590000000",timestamp:1637433108,data:"TXVsdGlFU0RUTkZUVHJhbnNmZXJAMDAwMDAwMDAwMDAwMDAwMDA1MDAxZTJhMTQyOGRkMWUzYTUxNDZiMzk2MGQ5ZTBmNGE1MDM2OTkwNGVlNTQ4M0AwMUA0YzRiNGQ0NTU4MmQ2MTYxNjIzOTMxMzBAMDRAMDUxZDVkZDAzNjA3Y2ZhMmJlMDAwMEA2NTZlNzQ2NTcyNDY2MTcyNmQ1MDcyNmY3ODc5QDAwMDAwMDAwMDAwMDAwMDAwNTAwNjU2ZDBhY2M1MzU2MWM1ZDZmNmZkN2Q3ZTgyYmYxMzI0NzAxNGY2MTU0ODM=",function:"enterFarmProxy",action:{category:"mex",name:"enterFarm",description:"Enter farm with 6183308.2764630175744 LKMEX",arguments:{transfers:[{type:"MetaESDT",name:"LockedMEX",ticker:"LKMEX",svgUrl:"https://media.elrond.com/tokens/asset/LKMEX-aab910/logo.svg",collection:"LKMEX-aab910",decimals:18,identifier:"LKMEX-aab910-04",value:"6183308276463017574400000"}],receiver:"erd1qqqqqqqqqqqqqpgqrc4pg2xarca9z34njcxeur622qmfjp8w2jps89fxnl",functionName:"enterFarmProxy",functionArgs:["00000000000000000500656d0acc53561c5d6f6fd7d7e82bf13247014f615483"],receiverAssets:{name:"Maiar Exchange: Proxy",tags:["maiar","exchange","proxy"],iconPng:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.png",iconSvg:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.svg"}}}},{txHash:"e47917728c85e355baf0bec96f3f51e0fe7b732594905c4b7ce8e03f49866d0f",gasLimit:115e5,gasPrice:1e9,gasUsed:9757888,miniBlockHash:"452b91830aebdfe7c73028af68221b84977a26455eb1d074bb71ce8f35591eaf",nonce:806,receiver:"erd1qqqqqqqqqqqqqpgqa0fsfshnff4n76jhcye6k7uvd7qacsq42jpsp6shh2",receiverAssets:{name:"Maiar Exchange: MEX/WEGLD Liquidity Pool",tags:["mex","liquiditypool"]},receiverShard:1,round:6885907,sender:"erd1esdzudw5wfzq4egwjwaxc9yueqes3cxw88cumaqs9dak9tsdseps95k0xt",senderShard:1,signature:"d3a30b6660ef25612e7ce551e2a7daab257dfb362bf363a86795e4aa7c3153be78d3f6e1223cc19b867ff37fa2ba769a2157bbd2e1a75f589d1f705b1fd3d505",status:"success",value:"0",fee:"335673880000000",timestamp:1637433042,data:"RVNEVFRyYW5zZmVyQDQ1NDc0YzQ0NGQ0NTU4MmQzMDYyNjUzOTY1MzVAMDQxZjI3ZWQyYTRhODAwMEA3MjY1NmQ2Zjc2NjU0YzY5NzE3NTY5NjQ2OTc0NzlAMDlhN2I0OTUyMDA1MmVkYkAyOWY1MTU3ODBmNDI1ZGYzNjMzMQ==",function:"removeLiquidity",action:{category:"mex",name:"removeLiquidity",description:"Removed liquidity with 0.297 EGLDMEX",arguments:{transfers:[{type:"FungibleESDT",name:"EGLDMEXLP",ticker:"EGLDMEX",svgUrl:"https://media.elrond.com/tokens/asset/EGLDMEX-0be9e5/logo.svg",token:"EGLDMEX-0be9e5",decimals:18,value:"297000000000000000"}],receiver:"erd1qqqqqqqqqqqqqpgqa0fsfshnff4n76jhcye6k7uvd7qacsq42jpsp6shh2",functionName:"removeLiquidity",functionArgs:["09a7b49520052edb","29f515780f425df36331"],receiverAssets:{name:"Maiar Exchange: MEX/WEGLD Liquidity Pool",tags:["mex","liquiditypool"]}}}},{txHash:"4f6559320c4687c8bfa065e0ed7d3de8c45ef4c1606af29616306cd5dabf33f1",gasLimit:615e5,gasPrice:1e9,gasUsed:48392131,miniBlockHash:"452b91830aebdfe7c73028af68221b84977a26455eb1d074bb71ce8f35591eaf",nonce:33,receiver:"erd1ecr8cdkpafrh9nw688hn2fq80lll53lqd4w9wl72kvde8qurntrssqgez2",receiverShard:1,round:6885907,sender:"erd1ecr8cdkpafrh9nw688hn2fq80lll53lqd4w9wl72kvde8qurntrssqgez2",senderShard:1,signature:"95e286ce67ec20bf7097e9a8f38d68a5fb318df31b8a73fffb7c1b729af0c65bf36f1d205d1d7cefbfd76fbb52c306f0f1f7c703aa0a974a8f04aaf3d8fbfc05",status:"success",value:"0",fee:"870516310000000",timestamp:1637433042,data:"RVNEVE5GVFRyYW5zZmVyQDRjNGI0NjQxNTI0ZDJkMzk2NDMxNjU2MTM4QDdmZjFAMDExZmE2OWY4ZTdiMDgzNDBkMDAwMEAwMDAwMDAwMDAwMDAwMDAwMDUwMDFlMmExNDI4ZGQxZTNhNTE0NmIzOTYwZDllMGY0YTUwMzY5OTA0ZWU1NDgzQDY1Nzg2OTc0NDY2MTcyNmQ1MDcyNmY3ODc5QDAwMDAwMDAwMDAwMDAwMDAwNTAwNjU2ZDBhY2M1MzU2MWM1ZDZmNmZkN2Q3ZTgyYmYxMzI0NzAxNGY2MTU0ODM=",function:"exitFarmProxy",action:{category:"mex",name:"exitFarm",description:"Exit farm with 1358392.837361997848576 LKFARM",arguments:{transfers:[{type:"MetaESDT",name:"LockedLPStaked",ticker:"LKFARM",svgUrl:"https://media.elrond.com/tokens/asset/LKFARM-9d1ea8/logo.svg",collection:"LKFARM-9d1ea8",decimals:18,identifier:"LKFARM-9d1ea8-7ff1",value:"1358392837361997848576000"}],receiver:"erd1qqqqqqqqqqqqqpgqrc4pg2xarca9z34njcxeur622qmfjp8w2jps89fxnl",functionName:"exitFarmProxy",functionArgs:["00000000000000000500656d0acc53561c5d6f6fd7d7e82bf13247014f615483"],receiverAssets:{name:"Maiar Exchange: Proxy",tags:["maiar","exchange","proxy"],iconPng:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.png",iconSvg:"https://raw.githubusercontent.com/ElrondNetwork/assets/master/accounts/icons/elrond.svg"}}}},{txHash:"02c3c2660f6c2bcfa767f9539201cd9062ab0f398c3db8bcdca1ae2cd718c639",gasLimit:75e6,gasPrice:1e9,gasUsed:7953928,miniBlockHash:"8c56f215588dd57815a0ca0147d643b958eda75ddbafb95fa24a84a9c98101e2",nonce:448,receiver:"erd1qqqqqqqqqqqqqpgqp699jngundfqw07d8jzkepucvpzush6k3wvqyc44rx",receiverShard:0,round:1812254,sender:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",senderShard:0,signature:"425d8db21db62692983095d9a2148bd0c12a337c743b50a480961ae0f135c1e48a7bcd3bb4e9eb5ce11bd8f96537aefd5660152748d7a09e294e90cb543d7904",status:"success",value:"0",fee:"167649280000000",timestamp:1664195124,data:"dW5TdGFrZUAwMTE1OGU0NjA5MTNkMDAwMDA=",function:"unStake",action:{category:"scCall",name:"unStake"}},{txHash:"ce1c1d6c4b36a3f239f822307624491cae6f0c4872970a3d0d02c3cc48b2844f",gasLimit:6e6,gasPrice:1e9,gasUsed:1068e3,miniBlockHash:"c72fca9867293054d9182aad167674142cf57c2fc3958282792ecddc55a74cca",nonce:446,receiver:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqphllllsndz99p",receiverShard:4294967295,round:1812226,sender:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",senderShard:0,signature:"6ed3d00d103e1ade45bd9a6e5ca8194661e414e5a2c8a30cf3360ea9bf6ccc71f14cbc56a68eefbc020502f9a2835278c9cb027c6407face5c6dc225e85c430e",status:"success",value:"0",fee:"78000000000000",timestamp:1664194956,data:"Y2xhaW1SZXdhcmRz",function:"claimRewards",action:{category:"stake",name:"claimRewards",description:"Claim rewards from staking provider https://keybase.io/drfonix",arguments:{providerName:"https://keybase.io/drfonix"}},results:[{hash:"03ec460f4037114403828f357dc91495149ab35c3556c6ad0155acd8c65c4b04",timestamp:1664194968,nonce:0,gasLimit:0,gasPrice:1e9,value:"67861704531764158",sender:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqphllllsndz99p",receiver:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",prevTxHash:"ce1c1d6c4b36a3f239f822307624491cae6f0c4872970a3d0d02c3cc48b2844f",originalTxHash:"ce1c1d6c4b36a3f239f822307624491cae6f0c4872970a3d0d02c3cc48b2844f",callType:"0",miniBlockHash:"8da210d07a8e099ced579f7ff634b3159f9589cbad106c55c51b4e1c5bf98c7d"},{hash:"ff5d9b11e3fddf3ee4bdf3dec0e82915f1c0b36233e1d05366b9950d78f96fea",timestamp:1664194968,nonce:447,gasLimit:0,gasPrice:1e9,value:"49320000000000",sender:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqphllllsndz99p",receiver:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",data:"QDZmNmI=",prevTxHash:"ce1c1d6c4b36a3f239f822307624491cae6f0c4872970a3d0d02c3cc48b2844f",originalTxHash:"ce1c1d6c4b36a3f239f822307624491cae6f0c4872970a3d0d02c3cc48b2844f",callType:"0",miniBlockHash:"8da210d07a8e099ced579f7ff634b3159f9589cbad106c55c51b4e1c5bf98c7d"}],operations:[{id:"03ec460f4037114403828f357dc91495149ab35c3556c6ad0155acd8c65c4b04",action:"transfer",type:"egld",sender:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqphllllsndz99p",receiver:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",value:"67861704531764158"}]},{txHash:"5e73704d40fdef680bd8839cd277827209f5decd3679b235efddd921f1c9fb3b",gasLimit:12e6,gasPrice:1e9,gasUsed:6075500,miniBlockHash:"3a7470a369d5e5bf6b5e984df1f6fdbed2fb260bb81574b1392e8f42d3c17f96",nonce:445,receiver:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqq9lllllsf3mp40",receiverShard:4294967295,round:1809456,sender:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",senderShard:0,signature:"3a25067722f2c8a98f693ccb2861f6ea1f91565d99e6b6a1aee57788d612a78d22cf1c3bfb55e94d9f5e881efad8632a4ad5be83e0830ad8f6181c8ac1552205",status:"success",value:"0",fee:"135500000000000",timestamp:1664178336,data:"cmVEZWxlZ2F0ZVJld2FyZHM=",function:"reDelegateRewards",action:{category:"stake",name:"reDelegateRewards",description:"Redelegate rewards from staking provider figment.io_testnet",arguments:{providerName:"figment.io_testnet",providerAvatar:"https://s3.amazonaws.com/keybase_processed_uploads/8cba5dc15c79565402849fd869654705_360_360.jpg"}},results:[{hash:"1bf45b782580cd035efdf600aafbfdd6d3fc437e93b4fb9219fe76f7526fad65",timestamp:1664178336,nonce:0,gasLimit:0,gasPrice:1e9,value:"392502810808819111",sender:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqq9lllllsf3mp40",receiver:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqplllst77y4l",prevTxHash:"5e73704d40fdef680bd8839cd277827209f5decd3679b235efddd921f1c9fb3b",originalTxHash:"5e73704d40fdef680bd8839cd277827209f5decd3679b235efddd921f1c9fb3b",callType:"0",miniBlockHash:"e74217b57b02069cf1b3294db1acbce08c5174811464c283cd1c739dbbf27b74"},{hash:"5fef839b3abc2f07be0923acfcb086a19ad2cf9522e9cf2eee4f398c84b301ef",timestamp:1664178348,nonce:446,gasLimit:0,gasPrice:1e9,value:"59245000000000",sender:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqq9lllllsf3mp40",receiver:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",data:"QDZmNmI=",prevTxHash:"5e73704d40fdef680bd8839cd277827209f5decd3679b235efddd921f1c9fb3b",originalTxHash:"5e73704d40fdef680bd8839cd277827209f5decd3679b235efddd921f1c9fb3b",callType:"0",miniBlockHash:"8f15889818ae5f2b63bebfae42deb1265ef500832aa5d12510246535140e1465"}],operations:[{id:"1bf45b782580cd035efdf600aafbfdd6d3fc437e93b4fb9219fe76f7526fad65",action:"transfer",type:"egld",sender:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqq9lllllsf3mp40",receiver:"erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqplllst77y4l",value:"392502810808819111"}]},{txHash:"5bdfd566dcc2c19a8bd62caa138acd2662a9d5f3c76213698e11c0f74c28cda6",gasLimit:75e6,gasPrice:1e9,gasUsed:75e6,miniBlockHash:"bde6aad6833c6eac6a58f26fbed6ae3960615b900ba41cf634138eb3c4902a1e",nonce:449,receiver:"erd1qqqqqqqqqqqqqpgqp699jngundfqw07d8jzkepucvpzush6k3wvqyc44rx",receiverShard:0,round:1813482,sender:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",senderShard:0,signature:"414b703c1e998daacf0d7ddf809b88ccd9ded2e673dc4ff978ec1c01f59a2457ddef21196ae7c566e2268be4fd0cb09be876cca938352f7acd6793b377c99e0f",status:"success",value:"0",fee:"808410000000000",timestamp:1664202492,data:"dW5Cb25k",function:"unBond",action:{category:"scCall",name:"unBond"},results:[{hash:"62ad6e43a172cfc39ae1620e61a0e200cb5cd51e9cf1675a19cc4963971fdc65",timestamp:1664202492,nonce:0,gasLimit:0,gasPrice:1e9,value:"20000000000000000000",sender:"erd1qqqqqqqqqqqqqpgqp699jngundfqw07d8jzkepucvpzush6k3wvqyc44rx",receiver:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",data:"ZGVsZWdhdGlvbiBzdGFrZSB1bmJvbmQ=",prevTxHash:"5bdfd566dcc2c19a8bd62caa138acd2662a9d5f3c76213698e11c0f74c28cda6",originalTxHash:"5bdfd566dcc2c19a8bd62caa138acd2662a9d5f3c76213698e11c0f74c28cda6",callType:"0"}],operations:[{id:"5bdfd566dcc2c19a8bd62caa138acd2662a9d5f3c76213698e11c0f74c28cda6",action:"transfer",type:"egld",sender:"erd1qqqqqqqqqqqqqpgqp699jngundfqw07d8jzkepucvpzush6k3wvqyc44rx",receiver:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",value:"20000000000000000000"},{id:"5bdfd566dcc2c19a8bd62caa138acd2662a9d5f3c76213698e11c0f74c28cda6",action:"writeLog",type:"log",sender:"erd1qqqqqqqqqqqqqpgqp699jngundfqw07d8jzkepucvpzush6k3wvqyc44rx",receiver:"erd1wh9c0sjr2xn8hzf02lwwcr4jk2s84tat9ud2kaq6zr7xzpvl9l5q8awmex",data:"@6f6b@01158e460913d00000",message:"@too much gas provided for processing: gas provided = 74941000, gas used = 6578498"}]}];var An=(o=0)=>Dn({transaction:Se[o],address:Se[o].sender,explorerAddress:W.devnet.explorerAddress});describe("TransactionsMethod tests",()=>{it("should display hash",()=>{let e=(0,Sn.render)(En.default.createElement(on,{transaction:An(0)})).getByTestId("transactionMethod"),n="delegate";expect(e.textContent).toContain(n),expect(e.getAttribute("title")).toBe("Delegate 14.167802221131358682 eGLD to staking provider ARC Stake")})});
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
